

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../../../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Group by variant category &mdash; GPSEA 0.9.4 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../../../_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/gpsea.css?v=547657ed" />

  
      <script src="../../../../_static/jquery.js?v=5d32c60e"></script>
      <script src="../../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../../../../_static/documentation_options.js?v=c5ca06ca"></script>
      <script src="../../../../_static/doctools.js?v=9bcbadda"></script>
      <script src="../../../../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../../../../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../../../../_static/copybutton.js?v=6dbb43f8"></script>
      <script>window.MathJax = {"TeX": {"Macros": {"RR": "{\\mathbb{R}}"}}}</script>
      <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" />
    <link rel="next" title="Group by allele count" href="allele_count.html" />
    <link rel="prev" title="Variant Predicates" href="variant_predicates.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../index.html">User guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../../input-data.html">Input data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../exploratory.html">Cohort exploratory analysis</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../index.html">Statistical analyses</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="../index.html">Partitioning</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="index.html">Genotype classifiers</a></li>
<li class="toctree-l4"><a class="reference internal" href="../phenotype/index.html">Phenotype classifiers and scorers</a></li>
<li class="toctree-l4"><a class="reference internal" href="../gallery.html">Gallery</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../phenotype-classes.html">Compare genotype and phenotype classes</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../mtc.html">Multiple-testing correction</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../phenotype-scores.html">Compare phenotype scores</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../measurements.html">Compare measurements</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../survival.html">Survival analysis</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../custom-components.html">Custom components</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../glossary.html">Glossary</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../../index.html">User guide</a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Statistical analyses</a></li>
          <li class="breadcrumb-item"><a href="../index.html">Partitioning</a></li>
          <li class="breadcrumb-item"><a href="index.html">Genotype classifiers</a></li>
      <li class="breadcrumb-item active">Group by variant category</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../../../_sources/user-guide/analyses/partitioning/genotype/variant_category.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="group-by-variant-category">
<span id="variant-category"></span><h1>Group by variant category<a class="headerlink" href="#group-by-variant-category" title="Link to this heading"></a></h1>
<p>Sometimes we want to compare the individuals who have the same allele count (<cite>AC</cite>) of variant categories <span class="math notranslate nohighlight">\(A\)</span> and <span class="math notranslate nohighlight">\(B\)</span>.
For example, in the context of an autosomal dominant disease,
we may want to compare the individuals with <span class="math notranslate nohighlight">\(AC_{A}=1\)</span> (where <span class="math notranslate nohighlight">\(A\)</span> is e.g. a predicted loss-of-function mutation)
with those harboring <span class="math notranslate nohighlight">\(AC_{B}=1\)</span> (where <span class="math notranslate nohighlight">\(B\)</span> is e.g. a missense mutation).
Similarly, in an autosomal recessive disease, we may be interested in comparing the individuals
with <span class="math notranslate nohighlight">\(AC_{A} \ge 1\)</span> with those with <span class="math notranslate nohighlight">\(AC_{A} = 0\)</span>.
In both analyses, we compare two variant categories <span class="math notranslate nohighlight">\(A\)</span> and <span class="math notranslate nohighlight">\(B\)</span>
which are described by a <a class="reference internal" href="../../../../apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.VariantPredicate" title="gpsea.analysis.predicate.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a>
(see <a class="reference internal" href="variant_predicates.html#variant-predicates"><span class="std std-ref">Variant Predicates</span></a> section),
while ensuring the allele count sum of both variant categories is <span class="math notranslate nohighlight">\(k\)</span>.</p>
<p><span class="math notranslate nohighlight">\(k = \sum_{i \in \{A, B\}} AC_{i}\)</span></p>
<p>GPSEA provides two genotype classifiers:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p><span class="math notranslate nohighlight">\(k\)</span></p></th>
<th class="head"><p>Name</p></th>
<th class="head"><p>Function</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>1</p></td>
<td><p>Monoallelic classifier</p></td>
<td><p><a class="reference internal" href="../../../../apidocs/gpsea.analysis.clf.html#gpsea.analysis.clf.monoallelic_classifier" title="gpsea.analysis.clf.monoallelic_classifier"><code class="xref py py-func docutils literal notranslate"><span class="pre">monoallelic_classifier()</span></code></a></p></td>
</tr>
<tr class="row-odd"><td><p>2</p></td>
<td><p>Biallelic classifier</p></td>
<td><p><a class="reference internal" href="../../../../apidocs/gpsea.analysis.clf.html#gpsea.analysis.clf.biallelic_classifier" title="gpsea.analysis.clf.biallelic_classifier"><code class="xref py py-func docutils literal notranslate"><span class="pre">biallelic_classifier()</span></code></a></p></td>
</tr>
</tbody>
</table>
<section id="monoallelic-classifier">
<span id="id1"></span><h2>Monoallelic classifier<a class="headerlink" href="#monoallelic-classifier" title="Link to this heading"></a></h2>
<p>Monoallelic classifier compares the individuals who have <em>one</em> allele of a variants of interest.
The classifier uses a pair of variant predicates, <cite>A</cite> and <cite>B</cite>,
to compute the allele counts <span class="math notranslate nohighlight">\(AC_{A}\)</span> and <span class="math notranslate nohighlight">\(AC_{B}\)</span>,
in order to assign an individual into a genotype class.</p>
<table class="docutils align-default" id="id4">
<caption><span class="caption-text">Monoallelic genotype classes</span><a class="headerlink" href="#id4" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"><p>Genotype class</p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(AC_{A}\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(AC_{B}\)</span></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>A</p></td>
<td><p>1</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><p>B</p></td>
<td><p>0</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">None</span></code></p></td>
<td><p>other</p></td>
<td><p>other</p></td>
</tr>
</tbody>
</table>
<p>The individuals with <span class="math notranslate nohighlight">\(\sum_{i \in \{A, B\}} AC_{i} \neq 1\)</span>
are omitted from the analysis.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../../../../_images/monoallelic-classifier.png"><img alt="Monoallelic classifier" src="../../../../_images/monoallelic-classifier.png" style="width: 600px;" />
</a>
</figure>
<section id="example">
<h3>Example<a class="headerlink" href="#example" title="Link to this heading"></a></h3>
<p>Let’s create a classifier to categorize the individuals
to those having one missense allele or to those having
one frameshift allele with respect to fictional transcript <code class="docutils literal notranslate"><span class="pre">NM_1234.5</span></code>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s2">&quot;NM_1234.5&quot;</span>
</pre></div>
</div>
<p>We start by creating the variant predicates for missense (<cite>A</cite>)
and frameshift (<cite>B</cite>) variants:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.model</span><span class="w"> </span><span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.predicate</span><span class="w"> </span><span class="kn">import</span> <span class="n">variant_effect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_missense</span> <span class="o">=</span> <span class="n">variant_effect</span><span class="p">(</span><span class="n">effect</span><span class="o">=</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_frameshift</span> <span class="o">=</span> <span class="n">variant_effect</span><span class="p">(</span><span class="n">effect</span><span class="o">=</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">FRAMESHIFT_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">tx_id</span><span class="p">)</span>
</pre></div>
</div>
<p>Monoallelic classifier lets us customize the category names.
Let’s use <cite>Missense</cite> and <cite>Frameshift</cite> instead of the defaults:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">a_label</span> <span class="o">=</span> <span class="s2">&quot;Missense&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b_label</span> <span class="o">=</span> <span class="s2">&quot;Frameshift&quot;</span>
</pre></div>
</div>
<p>Now we have all we need to create the predicate:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.clf</span><span class="w"> </span><span class="kn">import</span> <span class="n">monoallelic_classifier</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span> <span class="o">=</span> <span class="n">monoallelic_classifier</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">a_predicate</span><span class="o">=</span><span class="n">is_missense</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">b_predicate</span><span class="o">=</span><span class="n">is_frameshift</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">a_label</span><span class="o">=</span><span class="n">a_label</span><span class="p">,</span> <span class="n">b_label</span><span class="o">=</span><span class="n">b_label</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span><span class="o">.</span><span class="n">class_labels</span>
<span class="go">(&#39;Missense&#39;, &#39;Frameshift&#39;)</span>
</pre></div>
</div>
</section>
</section>
<section id="biallelic-classifier">
<span id="id2"></span><h2>Biallelic classifier<a class="headerlink" href="#biallelic-classifier" title="Link to this heading"></a></h2>
<p>Biallelic classifier compares the individuals with <em>two</em> alleles of the variants of interest.
The functionality is very similar to that of monoallelic classifier, with two differences:
(1) genotype classes and (2) partitions.</p>
<section id="genotype-classes">
<h3>Genotype classes<a class="headerlink" href="#genotype-classes" title="Link to this heading"></a></h3>
<p>Biallelic locus can be present in one of three genotypes, allowing an individual
to be assigned into one of the three genotype classes:</p>
<span id="biallelic-gt-classes"></span><table class="docutils align-default" id="id5">
<caption><span class="caption-text">Biallelic genotype classes</span><a class="headerlink" href="#id5" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"><p>Index</p></th>
<th class="head"><p>Genotype class</p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(AC_{A}\)</span></p></th>
<th class="head"><p><span class="math notranslate nohighlight">\(AC_{B}\)</span></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>0</p></td>
<td><p>A/A</p></td>
<td><p>2</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><p>1</p></td>
<td><p>A/B</p></td>
<td><p>1</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><p>2</p></td>
<td><p>B/B</p></td>
<td><p>0</p></td>
<td><p>2</p></td>
</tr>
<tr class="row-odd"><td></td>
<td><p><code class="docutils literal notranslate"><span class="pre">None</span></code></p></td>
<td><p>other</p></td>
<td><p>other</p></td>
</tr>
</tbody>
</table>
<p>Note that <span class="math notranslate nohighlight">\(\sum_{i \in \{A, B\}} AC_{i} = 2\)</span> and the individuals
with a different allele count sum are omitted from the analysis.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../../../../_images/biallelic-classifier.png"><img alt="Biallelic classifier" src="../../../../_images/biallelic-classifier.png" style="width: 600px;" />
</a>
</figure>
<section id="id3">
<h4>Example<a class="headerlink" href="#id3" title="Link to this heading"></a></h4>
<p>Let <cite>A</cite> and <cite>B</cite> correspond to <em>MISSENSE</em> and <em>FRAMESHIFT</em> variants,
and let’s reuse the variant predicates <code class="docutils literal notranslate"><span class="pre">is_missense</span></code> and <code class="docutils literal notranslate"><span class="pre">is_frameshift</span></code> from the previous section,
to compare missense and frameshift variants in the context of an autosomal recessive disease.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.clf</span><span class="w"> </span><span class="kn">import</span> <span class="n">biallelic_classifier</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span> <span class="o">=</span> <span class="n">biallelic_classifier</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">a_predicate</span><span class="o">=</span><span class="n">is_missense</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">b_predicate</span><span class="o">=</span><span class="n">is_frameshift</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">a_label</span><span class="o">=</span><span class="s2">&quot;Missense&quot;</span><span class="p">,</span> <span class="n">b_label</span><span class="o">=</span><span class="s2">&quot;Frameshift&quot;</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span><span class="o">.</span><span class="n">class_labels</span>
<span class="go">(&#39;Missense/Missense&#39;, &#39;Missense/Frameshift&#39;, &#39;Frameshift/Frameshift&#39;)</span>
</pre></div>
</div>
<p>The classifier assigns an individual into one of three genotype classes:</p>
<ul class="simple">
<li><p><cite>Missense/Missense</cite> - two missense alleles</p></li>
<li><p><cite>Missense/Frameshift</cite> - one missense and one frameshift allele</p></li>
<li><p><cite>Frameshift/Frameshift</cite> - two frameshift alleles</p></li>
</ul>
</section>
</section>
<section id="partitions">
<h3>Partitions<a class="headerlink" href="#partitions" title="Link to this heading"></a></h3>
<p>Sometimes we are interested in lumping several genotype classes into a partition
and then comparing the partitions.
For instance, in the context of an autosomal recessive disease,
we may want to compare individuals with two “mild” mutations with the individuals
with at least one “severe” mutation.
This comparison can be implemented using the <cite>partitions</cite> option.</p>
<p>A partition is a set of one or more genotype class indices
(see <a class="reference internal" href="#biallelic-gt-classes"><span class="std std-ref">Biallelic genotype classes</span></a> table).
Then, two (or more) partitions are provided to biallelic classifier
via the <cite>partitions</cite> option.</p>
<p>For example, we can compare the individuals with two missense alleles with those harboring
one frameshift and one missense alleles, or two frameshift alleles.</p>
<p>Let <cite>A</cite> and <cite>B</cite> correspond to <em>MISSENSE</em> and <em>FRAMESHIFT</em> variant.
According to <a class="reference internal" href="#biallelic-gt-classes"><span class="std std-ref">Biallelic genotype classes</span></a> table,
the <cite>A/A</cite> genotype class corresponds to index <cite>0</cite>,
and the <cite>A/B</cite> and <cite>B/B</cite> genotype class correspond to indices <cite>1</cite> and <cite>2</cite>, respectively.
We form the partitions accordingly:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">partitions</span> <span class="o">=</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="p">{</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">})</span>
</pre></div>
</div>
<p>With the <code class="docutils literal notranslate"><span class="pre">partitions</span></code>, the biallelic classifier splits the individuals
into two classes:</p>
<ul class="simple">
<li><p>two missense alleles</p></li>
<li><p>one missense alelele and one frameshift allele or two frameshift alleles</p></li>
</ul>
<figure class="align-center">
<a class="reference internal image-reference" href="../../../../_images/biallelic-classifier-w-partitions.png"><img alt="Biallelic classifier with partitions" src="../../../../_images/biallelic-classifier-w-partitions.png" style="width: 600px;" />
</a>
</figure>
<p>Using the <code class="docutils literal notranslate"><span class="pre">partitions</span></code> in code is a no-brainer:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span> <span class="o">=</span> <span class="n">biallelic_classifier</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">a_predicate</span><span class="o">=</span><span class="n">is_missense</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">b_predicate</span><span class="o">=</span><span class="n">is_frameshift</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">a_label</span><span class="o">=</span><span class="s2">&quot;Missense&quot;</span><span class="p">,</span> <span class="n">b_label</span><span class="o">=</span><span class="s2">&quot;Frameshift&quot;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">partitions</span><span class="o">=</span><span class="n">partitions</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span><span class="o">.</span><span class="n">class_labels</span>
<span class="go">(&#39;Missense/Missense&#39;, &#39;Missense/Frameshift OR Frameshift/Frameshift&#39;)</span>
</pre></div>
</div>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="variant_predicates.html" class="btn btn-neutral float-left" title="Variant Predicates" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="allele_count.html" class="btn btn-neutral float-right" title="Group by allele count" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>