

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Cohort exploratory analysis &mdash; GPSEA 0.9.3.dev0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/gpsea.css?v=547657ed" />

  
      <script src="../_static/jquery.js?v=5d32c60e"></script>
      <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../_static/documentation_options.js?v=bc6e4f25"></script>
      <script src="../_static/doctools.js?v=9bcbadda"></script>
      <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Statistical analyses" href="analyses/index.html" />
    <link rel="prev" title="Input data" href="input-data.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="index.html">User guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="input-data.html">Input data</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Cohort exploratory analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#background">Background</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#types-of-statistical-tests">Types of statistical tests</a></li>
<li class="toctree-l4"><a class="reference internal" href="#power">Power</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#exploration">Exploration</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#interactive-exploration">Interactive exploration</a></li>
<li class="toctree-l4"><a class="reference internal" href="#cohort-summary">Cohort summary</a></li>
<li class="toctree-l4"><a class="reference internal" href="#distribution-of-variants-across-protein-domains">Distribution of variants across protein domains</a></li>
<li class="toctree-l4"><a class="reference internal" href="#plot-distribution-of-variants-with-respect-to-the-protein-sequence">Plot distribution of variants with respect to the protein sequence</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="analyses/index.html">Statistical analyses</a></li>
<li class="toctree-l2"><a class="reference internal" href="custom-components.html">Custom components</a></li>
<li class="toctree-l2"><a class="reference internal" href="glossary.html">Glossary</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="index.html">User guide</a></li>
      <li class="breadcrumb-item active">Cohort exploratory analysis</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/user-guide/exploratory.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="cohort-exploratory-analysis">
<span id="exploratory"></span><h1>Cohort exploratory analysis<a class="headerlink" href="#cohort-exploratory-analysis" title="Link to this heading"></a></h1>
<section id="background">
<h2>Background<a class="headerlink" href="#background" title="Link to this heading"></a></h2>
<p>As a general rule for statistical testing, it is preferable to formulate one or several hypotheses and to test
these in a targeted way. Performing numerous tests without any specific plan comes with an increased
danger of false-positive results (although it may be appropriate to generate hypotheses if a validation cohort is
available).</p>
<section id="types-of-statistical-tests">
<h3>Types of statistical tests<a class="headerlink" href="#types-of-statistical-tests" title="Link to this heading"></a></h3>
<p>GPSEA offers four statistical tests:</p>
<ul class="simple">
<li><p>Fisher Exact Test (FET): Association between two categorical variables.
In this case, the first categorical variable is presence/absense of a genotype and the second is presence/absense of annotation to an HPO term.</p></li>
<li><p>Mann Whitney U test: Association between presence/absence of a genotype with a phenotype score (e.g., <a class="reference internal" href="analyses/partitioning/phenotype/devries.html#devries-scorer"><span class="std std-ref">De Vries Score</span></a>).</p></li>
<li><p>t test: Association between presence/absence of a genotype with a numerical test result (e.g., potassium level).</p></li>
<li><p>log rank test: Association between presence/absence of a genotype with age-of-onset of a disease or of an HPO feature or with mortality.</p></li>
</ul>
<p>Please consult the documentation pages for each of these tests if you are unsure which to use.</p>
</section>
<section id="power">
<h3>Power<a class="headerlink" href="#power" title="Link to this heading"></a></h3>
<p>Two things must be true to identify a valid genotype-phenotype correlation. First, and obviously,
the correlation must actually exist in the dataset being analyzed. Second, there must be statistical power to
identify the effect.
Power is a measure of the ability of an experimental design and hypothesis testing setup to detect a
particular effect if it is truly present (<a class="reference external" href="https://en.wikipedia.org/wiki/Power_(statistics)">Wikipedia</a>).</p>
<p>It will generally not be possible to perform a formal power calculation because the effect size of
most genotype-phenotype correlations is not well known. But, for instance it would not
make much sense to test for correlation with a specific variant that occurs in only one of 50 individuals in a cohort.</p>
</section>
</section>
<section id="exploration">
<h2>Exploration<a class="headerlink" href="#exploration" title="Link to this heading"></a></h2>
<p>The purpose of the exploratory analysis is thus to decide which tests to perform.
GPSEA provides tables and graphics that visualize some of the salient aspects
of the cohort and the distribution of the identified variants.
We exemplify the exploratory analysis on a cohort of 156 individuals with mutations
in <em>TBX5</em> from Phenopacket Store <em>0.1.20</em>. The cohort was preprocessed as described
in the <a class="reference internal" href="input-data.html#input-data"><span class="std std-ref">Input data</span></a> section and stored
at <a class="reference external" href="https://github.com/monarch-initiative/gpsea/tree/main/docs/cohort-data/TBX5.0.1.20.json">docs/cohort-data/TBX5.0.1.20.json</a>
in JSON format.</p>
<p>We start by loading the cohort from the JSON file:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.io</span><span class="w"> </span><span class="kn">import</span> <span class="n">GpseaJSONDecoder</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fpath_cohort</span> <span class="o">=</span> <span class="s1">&#39;docs/cohort-data/TBX5.0.1.20.json&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">fpath_cohort</span><span class="p">)</span> <span class="k">as</span> <span class="n">fh</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">cohort</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">fh</span><span class="p">,</span> <span class="bp">cls</span><span class="o">=</span><span class="n">GpseaJSONDecoder</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
<span class="go">156</span>
</pre></div>
</div>
<p>Then we choose the transcript and protein identifiers, and we fetch the corresponding data:
(see <a class="reference internal" href="input-data.html#choose-tx-and-protein"><span class="std std-ref">Choose the transcript and protein of interest</span></a> for more info):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.preprocessing</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_default_tx_coordinate_service</span><span class="p">,</span> <span class="n">configure_default_protein_metadata_service</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s2">&quot;NM_181486.4&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">px_id</span> <span class="o">=</span> <span class="s2">&quot;NP_852259.1&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_service</span> <span class="o">=</span> <span class="n">configure_default_tx_coordinate_service</span><span class="p">(</span><span class="n">genome_build</span><span class="o">=</span><span class="s2">&quot;GRCh38.p13&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span> <span class="o">=</span> <span class="n">tx_service</span><span class="o">.</span><span class="n">fetch</span><span class="p">(</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pm_service</span> <span class="o">=</span> <span class="n">configure_default_protein_metadata_service</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">protein_meta</span> <span class="o">=</span> <span class="n">pm_service</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="n">px_id</span><span class="p">)</span>
</pre></div>
</div>
<p>Last, we load HPO <cite>v2024-07-01</cite> to use in the exploratory analysis:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<section id="interactive-exploration">
<h3>Interactive exploration<a class="headerlink" href="#interactive-exploration" title="Link to this heading"></a></h3>
<p>The code for generating reports with tables and figures is located in <a class="reference internal" href="../apidocs/gpsea.view.html#module-gpsea.view" title="gpsea.view"><code class="xref py py-mod docutils literal notranslate"><span class="pre">gpsea.view</span></code></a> module.
As a rule of thumb, the reports leverage IPython’s “magic”
to integrate with the environments such as Jupyter notebook.
Returning the report object as the last cell expression will render the report in the cell output.</p>
</section>
<section id="cohort-summary">
<h3>Cohort summary<a class="headerlink" href="#cohort-summary" title="Link to this heading"></a></h3>
<p>We recommend that users start be generating a cohort summary
with an overview about the HPO terms, variants, diseases, and variant effects that occur most frequently.
We create the <cite>viewer</cite> and we generate the <cite>report</cite> for a cohort and transcript.
The implicit return of the <cite>report</cite> at the end of the cell renders the report in the cell output:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">CohortViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">viewer</span> <span class="o">=</span> <span class="n">CohortViewer</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span> <span class="o">=</span> <span class="n">viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span> <span class="n">transcript_id</span><span class="o">=</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span>  
</pre></div>
</div>
<div>
    <style>
        /** Light theme is the default */
        :root {
            --gpsea-color-text: black;
            --gpsea-color-link-text: rgb(0, 118, 187);
            --gpsea-color-link-hover: rgb(0, 99, 156);
            --gpsea-color-primary-a0: #2f5170;
            --gpsea-color-primary-a10: #4778a5;
            --gpsea-color-primary-a20: #5995ce;
            --gpsea-color-primary-a60: LightSkyBlue;
            --gpsea-color-surface-a0: #fcfcfc;
            --gpsea-color-surface-a10: #ffffff;
            --gpsea-color-surface-a20: #b3b3b3;
            --gpsea-color-surface-a30: #8d8d8d;
            --gpsea-color-surface-a40: #646464;
            --gpsea-color-surface-a50: #000000;
        }

        /** Dark theme */
        /*
        @media (prefers-color-scheme: dark) {
            :root {
                --gpsea-color-text: rgb(240, 240, 240);
                --gpsea-color-link-text: rgb(240, 240, 240);
                --gpsea-color-link-hover: lightblue;
                --gpsea-color-primary-a0: #64a3df;
                --gpsea-color-primary-a10: #5286b8;
                --gpsea-color-primary-a20: #375c7e;
                --gpsea-color-primary-a60: #233b52;
                --gpsea-color-surface-a0: #121212;
                --gpsea-color-surface-a10: #282828;
                --gpsea-color-surface-a20: #3f3f3f;
                --gpsea-color-surface-a30: #575757;
                --gpsea-color-surface-a40: #717171;
                --gpsea-color-surface-a50: #8b8b8b;
            }
        }
        */

        #gpsea-body {
            max-width: 1600px;
            color: var(--gpsea-color-text);
            background-color: var(--gpsea-color-surface-a0);
            padding: 10px 0;
        }

        #gpsea-body {

            h1,
            h2,
            h3,
            h4,
            h5 {
                color: var(--gpsea-color-primary-a0);
            }

            a {
                color: var(--gpsea-color-link-text);
                text-decoration: none;
            }

            a:hover {
                color: var(--gpsea-color-link-hover);
            }

            #cohort-summary {
                max-width: 800px;
            }

            td {
                padding: 5px 0;
            }

            .cards {
                display: flex;
                flex-flow: row wrap;
                align-items: top;
                gap: 1rem;
            }

            .card {
                border-radius: 5px;
                padding: 0px 10px;
                background: var(--gpsea-color-surface-a10);
                border: 1px solid var(--gpsea-color-surface-a30);
                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);
            }

            table {
                width: 100%;
                border-collapse: collapse;
                margin: 10px 0;
                font-size: 0.9em;
                font-family: sans-serif;
                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
                border: 1px solid var(--gpsea-color-primary-a60);
            }

            th {
                text-align: left;
                padding: 2px;
                font-weight: bold;
                font-size: 120%;
                color: var(--gpsea-color-text);
                background-color: var(--gpsea-color-primary-a60);
            }

            table td,
            tr {
                text-align: left;
                padding: 0px 5px;
            }

            tr:nth-child(even) {
                background-color: var(--gpsea-color-surface-a10);
            }

            tr:nth-child(odd) {
                background-color: var(--gpsea-color-surface-a20);
            }

            caption {
                caption-side: top;
                text-align: left;
                padding-bottom: 10px;
                font-weight: bold;
            }
        }
    </style>
    <div id="gpsea-body">
        

<h1>GPSEA cohort analysis</h1>
<div id="cohort-summary">
  <p>
    Successfully loaded 156 individuals.

    
    46 were recorded as male,
    56 as female,
    and 54 as unknown sex.

    
    
    No information about individuals' vital status was reported.
    

    
    52 individuals had disease onset information
    and 0 had information about the age of last encounter.
  </p>

  
  <p>No errors encountered.</p>
  
</div>


<div class="cards">
  
  <div id="hpo" class="card">
    <h2>HPO terms</h2>
    
    
    <table>
      The cohort included 90 distinct HPO terms.
      <caption>
        
        Top 10 most common HPO terms.
        
        
      </caption>
      <tbody>
        <tr class="strng">
          <th><em>n</em></th>
          <th>HPO Term</th>
        </tr>

        
        <tr>
          <td>50</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0001631">Atrial septal defect</a>
          </td>
        </tr>
        
        <tr>
          <td>41</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0001629">Ventricular septal defect</a>
          </td>
        </tr>
        
        <tr>
          <td>40</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0002984">Hypoplasia of the radius</a>
          </td>
        </tr>
        
        <tr>
          <td>36</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0001199">Triphalangeal thumb</a>
          </td>
        </tr>
        
        <tr>
          <td>32</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0009777">Absent thumb</a>
          </td>
        </tr>
        
        <tr>
          <td>32</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0009778">Short thumb</a>
          </td>
        </tr>
        
        <tr>
          <td>30</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0001191">Abnormal carpal morphology</a>
          </td>
        </tr>
        
        <tr>
          <td>27</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0001684">Secundum atrial septal defect</a>
          </td>
        </tr>
        
        <tr>
          <td>15</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0003974">Absent radius</a>
          </td>
        </tr>
        
        <tr>
          <td>14</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0031546">Cardiac conduction abnormality</a>
          </td>
        </tr>
        
      </tbody>
    </table>
    
    
  </div>

  
  <div id="measurements" class="card">
    <h2>Measurements</h2>
    
    
    No data regarding measurement assays were provided.
    
  </div>

  
  <div id="diseases" class="card">
    <h2>Diseases</h2>
    
    
    The cohort members were diagnosed with 1 disease.
    <table>
      

      <tbody>
        <tr class="strng">
          <th><em>n</em></th>
          <th>Disease</th>
        </tr>
        
        <tr>
          <td>156</td>
          <td><a href="https://hpo.jax.org/browse/disease/OMIM:142900">Holt-Oram syndrome</a>
          </td>
        </tr>
        
      </tbody>
    </table>
    
  </div>

  
  
  <div class="card">
    <h2>Variants</h2>
    <p>
      A total of 53 unique variants
      were identified in the cohort. Variants were annotated with respect to NM_181486.4.
    </p>
    <table>
      <caption>
        <h3>Top 10 variants</h3>
      </caption>
      <tbody>
        <tr class="strng">
          <th><em>n</em></th>
          <th>Variant key</th>
          <th>HGVS</th>
          <th>Variant Class</th>
        </tr>
        
        <tr>
          <td>22</td>
          <td>12_114385521_114385521_C_T</td>
          <td>c.710G>A (p.Arg237Gln)</td>
          <td>MISSENSE_VARIANT</td>
        </tr>
        
        <tr>
          <td>20</td>
          <td>12_114401830_114401830_C_T</td>
          <td>c.238G>A (p.Gly80Arg)</td>
          <td>MISSENSE_VARIANT</td>
        </tr>
        
        <tr>
          <td>8</td>
          <td>12_114385563_114385563_G_A</td>
          <td>c.668C>T (p.Thr223Met)</td>
          <td>MISSENSE_VARIANT</td>
        </tr>
        
        <tr>
          <td>6</td>
          <td>12_114398675_114398675_G_T</td>
          <td>c.408C>A (p.Tyr136Ter)</td>
          <td>STOP_GAINED</td>
        </tr>
        
        <tr>
          <td>5</td>
          <td>12_114398682_114398682_C_CG</td>
          <td>c.400dup (p.Arg134ProfsTer49)</td>
          <td>FRAMESHIFT_VARIANT</td>
        </tr>
        
        <tr>
          <td>5</td>
          <td>12_114399514_114399514_A_C</td>
          <td>c.361T>G (p.Trp121Gly)</td>
          <td>MISSENSE_VARIANT, SPLICE_REGION_VARIANT</td>
        </tr>
        
        <tr>
          <td>5</td>
          <td>12_114403792_114403792_C_CG</td>
          <td>c.106_107insC (p.Ser36ThrfsTer25)</td>
          <td>FRAMESHIFT_VARIANT</td>
        </tr>
        
        <tr>
          <td>4</td>
          <td>12_114385522_114385522_G_A</td>
          <td>c.709C>T (p.Arg237Trp)</td>
          <td>MISSENSE_VARIANT</td>
        </tr>
        
        <tr>
          <td>4</td>
          <td>12_114398656_114398656_C_CG</td>
          <td>c.426dup (p.Ala143ArgfsTer40)</td>
          <td>FRAMESHIFT_VARIANT</td>
        </tr>
        
        <tr>
          <td>4</td>
          <td>12_114403798_114403798_G_GC</td>
          <td>c.100dup (p.Ala34GlyfsTer27)</td>
          <td>FRAMESHIFT_VARIANT</td>
        </tr>
        
      </tbody>
    </table>
  </div>

  
  <div class="card">
    
    <h2>Variant effects</h2>
    <p>
      The effects were predicted for NM_181486.4
    </p>
    <table>
      <tbody>
        <tr class="strng">
          <th>Variant effect</th>
          <th>Count</th>
        </tr>
        
        <tr>
          <td>MISSENSE_VARIANT</td>
          <td>85 (50%)</td>
        </tr>
        
        <tr>
          <td>FRAMESHIFT_VARIANT</td>
          <td>38 (22%)</td>
        </tr>
        
        <tr>
          <td>STOP_GAINED</td>
          <td>19 (11%)</td>
        </tr>
        
        <tr>
          <td>SPLICE_REGION_VARIANT</td>
          <td>10 (6%)</td>
        </tr>
        
        <tr>
          <td>SPLICE_DONOR_VARIANT</td>
          <td>7 (4%)</td>
        </tr>
        
        <tr>
          <td>INFRAME_INSERTION</td>
          <td>2 (1%)</td>
        </tr>
        
        <tr>
          <td>STOP_RETAINED_VARIANT</td>
          <td>2 (1%)</td>
        </tr>
        
        <tr>
          <td>SPLICE_ACCEPTOR_VARIANT</td>
          <td>2 (1%)</td>
        </tr>
        
        <tr>
          <td>SPLICE_DONOR_5TH_BASE_VARIANT</td>
          <td>2 (1%)</td>
        </tr>
        
        <tr>
          <td>INTRON_VARIANT</td>
          <td>2 (1%)</td>
        </tr>
        
        <tr>
          <td>INFRAME_DELETION</td>
          <td>1 (1%)</td>
        </tr>
        
      </tbody>
    </table>
    
  </div>
</div>

    </div>
</div></section>
<section id="distribution-of-variants-across-protein-domains">
<h3>Distribution of variants across protein domains<a class="headerlink" href="#distribution-of-variants-across-protein-domains" title="Link to this heading"></a></h3>
<p>GPSEA gathers information about protein domains from the UniProt API, and alternatively allows users to
enter domain information manually (See <a class="reference internal" href="input-data.html#fetch-protein-data"><span class="std std-ref">Fetch protein data</span></a>).
Protein domains are distinct functional or structural units in a protein. For instance, the following graphic shows domains of
the <em>PLD1</em> protein. The HKD domains contribute to the catalytic activity of the protein whereas the PX and PH domains
regulation PLD1 localization within the cell. Observations such as this may suggest testable hypotheses.</p>
<figure class="align-center" id="id1">
<a class="reference internal image-reference" href="../_images/PLD1.png"><img alt="PLD1" src="../_images/PLD1.png" style="width: 600px;" />
</a>
<figcaption>
<p><span class="caption-text">Human <em>PLD1</em> with PX, PH, and two HKD domains.</span><a class="headerlink" href="#id1" title="Link to this image"></a></p>
</figcaption>
</figure>
<p>Users can create a table to display the protein domains and the variants
located in them in order to decide whether it might be sensible to test for correlation between variants
located in one or more protein domains and a certain phenotype.</p>
<p>This code will produce the following table on the basis of a cohort of individuals
with variants in the <em>TBX5</em> gene:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">ProteinVariantViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cpd_viewer</span> <span class="o">=</span> <span class="n">ProteinVariantViewer</span><span class="p">(</span><span class="n">tx_id</span><span class="o">=</span><span class="n">tx_id</span><span class="p">,</span> <span class="n">protein_metadata</span><span class="o">=</span><span class="n">protein_meta</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span> <span class="o">=</span> <span class="n">cpd_viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span>  
</pre></div>
</div>
<div>
    <style>
        /** Light theme is the default */
        :root {
            --gpsea-color-text: black;
            --gpsea-color-link-text: rgb(0, 118, 187);
            --gpsea-color-link-hover: rgb(0, 99, 156);
            --gpsea-color-primary-a0: #2f5170;
            --gpsea-color-primary-a10: #4778a5;
            --gpsea-color-primary-a20: #5995ce;
            --gpsea-color-primary-a60: LightSkyBlue;
            --gpsea-color-surface-a0: #fcfcfc;
            --gpsea-color-surface-a10: #ffffff;
            --gpsea-color-surface-a20: #b3b3b3;
            --gpsea-color-surface-a30: #8d8d8d;
            --gpsea-color-surface-a40: #646464;
            --gpsea-color-surface-a50: #000000;
        }

        /** Dark theme */
        /*
        @media (prefers-color-scheme: dark) {
            :root {
                --gpsea-color-text: rgb(240, 240, 240);
                --gpsea-color-link-text: rgb(240, 240, 240);
                --gpsea-color-link-hover: lightblue;
                --gpsea-color-primary-a0: #64a3df;
                --gpsea-color-primary-a10: #5286b8;
                --gpsea-color-primary-a20: #375c7e;
                --gpsea-color-primary-a60: #233b52;
                --gpsea-color-surface-a0: #121212;
                --gpsea-color-surface-a10: #282828;
                --gpsea-color-surface-a20: #3f3f3f;
                --gpsea-color-surface-a30: #575757;
                --gpsea-color-surface-a40: #717171;
                --gpsea-color-surface-a50: #8b8b8b;
            }
        }
        */

        #gpsea-body {
            max-width: 1600px;
            color: var(--gpsea-color-text);
            background-color: var(--gpsea-color-surface-a0);
            padding: 10px 0;
        }

        #gpsea-body {

            h1,
            h2,
            h3,
            h4,
            h5 {
                color: var(--gpsea-color-primary-a0);
            }

            a {
                color: var(--gpsea-color-link-text);
                text-decoration: none;
            }

            a:hover {
                color: var(--gpsea-color-link-hover);
            }

            #cohort-summary {
                max-width: 800px;
            }

            td {
                padding: 5px 0;
            }

            .cards {
                display: flex;
                flex-flow: row wrap;
                align-items: top;
                gap: 1rem;
            }

            .card {
                border-radius: 5px;
                padding: 0px 10px;
                background: var(--gpsea-color-surface-a10);
                border: 1px solid var(--gpsea-color-surface-a30);
                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);
            }

            table {
                width: 100%;
                border-collapse: collapse;
                margin: 10px 0;
                font-size: 0.9em;
                font-family: sans-serif;
                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
                border: 1px solid var(--gpsea-color-primary-a60);
            }

            th {
                text-align: left;
                padding: 2px;
                font-weight: bold;
                font-size: 120%;
                color: var(--gpsea-color-text);
                background-color: var(--gpsea-color-primary-a60);
            }

            table td,
            tr {
                text-align: left;
                padding: 0px 5px;
            }

            tr:nth-child(even) {
                background-color: var(--gpsea-color-surface-a10);
            }

            tr:nth-child(odd) {
                background-color: var(--gpsea-color-surface-a20);
            }

            caption {
                caption-side: top;
                text-align: left;
                padding-bottom: 10px;
                font-weight: bold;
            }
        }
    </style>
    <div id="gpsea-body">
        
<h1>T-box transcription factor TBX5: NP_852259.1</h1>
<div class="card">
  <p>
    T-box transcription factor TBX5 has 5 annotated protein featureses.
    139 were found to be located within these features,
    and 6 were located elsewhere.
    The total length of the protein is 518 amino acid residues.
  </p>
  <table>
    <tbody>
        <tr>
          <th>Count</th>
          <th>Name</th>
          <th>Type</th>
          <th>Coordinates</th>
          <th>Variants</th>
        </tr>
        
        <tr>
          <td>13</td>
          <td>Disordered</td>
          <td>Region</td>
          <td>2 - 46</td>
          <td>p.Ala34GlyfsTer27; p.Ala34ProfsTer32; p.Ser36ThrfsTer25; p.Pro14Thr</td>
        </tr>
        
        <tr>
          <td>107</td>
          <td>T-box</td>
          <td>DNA binding</td>
          <td>59 - 238</td>
          <td>p.Pro139GlnfsTer11; p.Ile106Val; p.Phe168LeufsTer6; p.Ser196Ter; p.Arg237Trp; p.Arg237Pro; p.Gly195Ala; p.Ala143ArgfsTer40; p.Thr161Pro; p.Leu65GlnfsTer10; p.Met83PhefsTer6; p.Leu94Arg; p.His220del; p.Tyr136Ter; p.Gln151Ter; p.Ile227_Glu228insTer; p.Arg237Gln; p.Arg81Trp; p.Trp121Gly; p.Thr72Lys; p.Lys226Asn; p.Met74Ile; p.Gly125AlafsTer25; p.Gly80Arg; p.Arg134ProfsTer49; p.Pro85Thr; p.Val214GlyfsTer12; p.Lys88Ter; p.Val153SerfsTer21; p.Thr223Met</td>
        </tr>
        
        <tr>
          <td>19</td>
          <td>Disordered</td>
          <td>Region</td>
          <td>251 - 356</td>
          <td>p.Tyr291Ter; p.Gln302ArgfsTer92; p.Val267TrpfsTer127; p.Val263Met; p.Gln315ArgfsTer79; p.Ser261Cys; p.Glu294Ter; p.Arg279Ter; p.Tyr342ThrfsTer52</td>
        </tr>
        
        <tr>
          <td>8</td>
          <td>Polar residues</td>
          <td>Compositional bias</td>
          <td>264 - 299</td>
          <td>p.Arg279Ter; p.Tyr291Ter; p.Glu294Ter; p.Val267TrpfsTer127</td>
        </tr>
        
        <tr>
          <td>1</td>
          <td>Basic and acidic residues</td>
          <td>Compositional bias</td>
          <td>321 - 346</td>
          <td>p.Tyr342ThrfsTer52</td>
        </tr>
        
    </tbody>
</table>
</div>

    </div>
</div></section>
<section id="plot-distribution-of-variants-with-respect-to-the-protein-sequence">
<h3>Plot distribution of variants with respect to the protein sequence<a class="headerlink" href="#plot-distribution-of-variants-with-respect-to-the-protein-sequence" title="Link to this heading"></a></h3>
<section id="cohort-artist">
<h4>Cohort artist<a class="headerlink" href="#cohort-artist" title="Link to this heading"></a></h4>
<p>The simplest way to plot the variant distribution is to use the <a class="reference internal" href="../apidocs/gpsea.view.html#gpsea.view.CohortArtist" title="gpsea.view.CohortArtist"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortArtist</span></code></a> API:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_default_cohort_artist</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_artist</span> <span class="o">=</span> <span class="n">configure_default_cohort_artist</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span> <span class="mi">8</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_artist</span><span class="o">.</span><span class="n">draw_protein</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">protein_id</span><span class="o">=</span><span class="n">px_id</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/TBX5_protein_diagram.from_artist.png"><img alt="TBX5 protein diagram" class="align-center" src="../_images/TBX5_protein_diagram.from_artist.png" style="width: 600px;" />
</a>
<p>The <a class="reference internal" href="../apidocs/gpsea.view.html#gpsea.view.configure_default_cohort_artist" title="gpsea.view.configure_default_cohort_artist"><code class="xref py py-func docutils literal notranslate"><span class="pre">configure_default_cohort_artist()</span></code></a> function gets the default artist
which we use to plot the diagram with the variant distribution across the protein sequence
on Matplotlib axes.</p>
</section>
<section id="protein-visualizer">
<h4>Protein visualizer<a class="headerlink" href="#protein-visualizer" title="Link to this heading"></a></h4>
<p>Sometimes, however, things do not work out-of-the-box, e.g. because protein metadata
is not available from Uniprot (the default), and we may need to use the lower-level components.</p>
<p>The <a class="reference internal" href="../apidocs/gpsea.view.html#gpsea.view.ProteinVisualizer" title="gpsea.view.ProteinVisualizer"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinVisualizer</span></code></a> takes cohort and the protein metadata
to plot the distribution of variants on a protein diagram:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">ProteinVisualizer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span> <span class="mi">8</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">visualizer</span> <span class="o">=</span> <span class="n">ProteinVisualizer</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">visualizer</span><span class="o">.</span><span class="n">draw_protein</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">protein_metadata</span><span class="o">=</span><span class="n">protein_meta</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/TBX5_protein_diagram.from_protein_visualizer.png"><img alt="TBX5 protein diagram" class="align-center" src="../_images/TBX5_protein_diagram.from_protein_visualizer.png" style="width: 600px;" />
</a>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="input-data.html" class="btn btn-neutral float-left" title="Input data" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="analyses/index.html" class="btn btn-neutral float-right" title="Statistical analyses" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>