

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Input data &mdash; GPSEA 0.9.4.dev0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/gpsea.css?v=547657ed" />

  
      <script src="../_static/jquery.js?v=5d32c60e"></script>
      <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../_static/documentation_options.js?v=d641ac61"></script>
      <script src="../_static/doctools.js?v=9bcbadda"></script>
      <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Cohort exploratory analysis" href="exploratory.html" />
    <link rel="prev" title="User guide" href="index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="index.html">User guide</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">Input data</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#create-a-cohort">Create a cohort</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#load-hpo">Load HPO</a></li>
<li class="toctree-l4"><a class="reference internal" href="#cohort-creator">Cohort creator</a></li>
<li class="toctree-l4"><a class="reference internal" href="#load-phenopackets">Load phenopackets</a></li>
<li class="toctree-l4"><a class="reference internal" href="#alternative-phenopacket-sources">Alternative phenopacket sources</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#quality-control">Quality control</a></li>
<li class="toctree-l3"><a class="reference internal" href="#persist-the-cohort-for-later">Persist the cohort for later</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#example">Example</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#choose-the-transcript-and-protein-of-interest">Choose the transcript and protein of interest</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#choose-the-transcript">Choose the transcript</a></li>
<li class="toctree-l4"><a class="reference internal" href="#get-the-transcript-data">Get the transcript data</a></li>
<li class="toctree-l4"><a class="reference internal" href="#showcase-transcript-data">Showcase transcript data</a></li>
<li class="toctree-l4"><a class="reference internal" href="#fetch-protein-data">Fetch protein data</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="exploratory.html">Cohort exploratory analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="analyses/index.html">Statistical analyses</a></li>
<li class="toctree-l2"><a class="reference internal" href="custom-components.html">Custom components</a></li>
<li class="toctree-l2"><a class="reference internal" href="glossary.html">Glossary</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="index.html">User guide</a></li>
      <li class="breadcrumb-item active">Input data</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/user-guide/input-data.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="input-data">
<span id="id1"></span><h1>Input data<a class="headerlink" href="#input-data" title="Link to this heading"></a></h1>
<p>Each GPSEA analysis starts with the ingest of genotype and phenotype data.
There are many things to be done. We need to check the HPO terms
to point out any logical inconsistencies or to remove redundancies,
annotate the variants with respect to genes and transcripts,
or check uniqueness of the identifiers of the cohort members.
Additionally, most GPSEA analyses need an information about the “anatomy”
of the transcript and/or protein corresponding to the gene(s) of interest.</p>
<p>This section will explain how to ingest the phenopackets,
how to persist the cohort for later, to save the preprocessing time,
and how to get the transcript and protein information.</p>
<nav class="contents local" id="section-overview">
<p class="topic-title">Section overview</p>
<ul class="simple">
<li><p><a class="reference internal" href="#create-a-cohort" id="id6">Create a cohort</a></p></li>
<li><p><a class="reference internal" href="#quality-control" id="id7">Quality control</a></p></li>
<li><p><a class="reference internal" href="#persist-the-cohort-for-later" id="id8">Persist the cohort for later</a></p></li>
<li><p><a class="reference internal" href="#choose-the-transcript-and-protein-of-interest" id="id9">Choose the transcript and protein of interest</a></p></li>
</ul>
</nav>
<section id="create-a-cohort">
<span id="id2"></span><h2><a class="toc-backref" href="#id6" role="doc-backlink">Create a cohort</a><a class="headerlink" href="#create-a-cohort" title="Link to this heading"></a></h2>
<p>GPSEA workflow starts with a collection
of <a class="reference external" href="https://phenopacket-schema.readthedocs.io/en/latest">GA4GH Phenopacket Schema</a> phenopackets
which are transformed into a <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>.</p>
<section id="load-hpo">
<h3>Load HPO<a class="headerlink" href="#load-hpo" title="Link to this heading"></a></h3>
<p>The Q/C requires access to Human Phenotype Ontology, which we can load using
the <a class="reference external" href="https://github.com/ielis/hpo-toolkit">hpo-toolkit</a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>The code downloads the HPO data from GitHub and stores it in the user’s home directory,
to speed up the analysis by downloading the ontology data only once.
The latest HPO can be loaded by leaving out the <code class="docutils literal notranslate"><span class="pre">release</span></code> argument.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If storing HPO locally is not desired, HPO can also be loaded
directly from GitHub, at the expense of additional network traffic:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">release</span> <span class="o">=</span> <span class="s1">&#39;v2024-07-01&#39;</span>
<span class="n">hpo</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">load_minimal_ontology</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;https://github.com/obophenotype/human-phenotype-ontology/releases/download/</span><span class="si">{</span><span class="n">release</span><span class="si">}</span><span class="s1">/hp.json&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</section>
<section id="cohort-creator">
<h3>Cohort creator<a class="headerlink" href="#cohort-creator" title="Link to this heading"></a></h3>
<p>The <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.CohortCreator" title="gpsea.preprocessing.CohortCreator"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortCreator</span></code></a> is responsible for transforming
phenopackets into a <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>.
The transformation is very flexible, with an array of parameters to tweak.
However, the majority of cases should be covered by the default settings
and we recommend to configure the default cohort creator with
the <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_caching_cohort_creator" title="gpsea.preprocessing.configure_caching_cohort_creator"><code class="xref py py-func docutils literal notranslate"><span class="pre">configure_caching_cohort_creator()</span></code></a> method:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.preprocessing</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">cohort_creator</span></code> will check the HPO terms for any invalid or obsolete
terms, as well as point out any logical inconsistencies.
The functional annotation and validation of variants is delegated to
Variant Effect Predictor and Variant Validator REST APIs,
and the API responses are cached in the current working directory, to reduce the network bandwith.
The cohort is also checked for individuals with non-unique ID.</p>
</section>
<section id="load-phenopackets">
<h3>Load phenopackets<a class="headerlink" href="#load-phenopackets" title="Link to this heading"></a></h3>
<p>We can create a cohort starting from a collection of <cite>Phenopacket</cite> objects
provided by Python <a class="reference external" href="https://pypi.org/project/phenopackets">Phenopackets</a> library.
For the purpose of this example, we will load a cohort of patients with pathogenic mutations in <em>RERE</em> gene
which are included in the release <cite>0.1.18</cite> of <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a>.
We use <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store-toolkit">Phenopacket Store Toolkit</a>
(<code class="docutils literal notranslate"><span class="pre">ppktstore</span></code> in the code) to reduce the boilerplate code
associated with extracting phenopacket data from Phenopacket Store release:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">ppktstore.registry</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_phenopacket_registry</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">registry</span> <span class="o">=</span> <span class="n">configure_phenopacket_registry</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">registry</span><span class="o">.</span><span class="n">open_phenopacket_store</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s2">&quot;0.1.19&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">ps</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">phenopackets</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">ps</span><span class="o">.</span><span class="n">iter_cohort_phenopackets</span><span class="p">(</span><span class="s1">&#39;RERE&#39;</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">)</span>
<span class="go">19</span>
</pre></div>
</div>
<p>The code creates a <code class="docutils literal notranslate"><span class="pre">registry</span></code> which will download the ZIP file
for a given Phenopacket Store data release and store it in user’s home directory.
Then, phenopackets of the <em>RERE</em> cohort are extracted,
resulting in a tuple with 19 phenopackets.</p>
<p>The actual transformation of the phenopackets into a <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>
is orchestrated by the <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.load_phenopackets" title="gpsea.preprocessing.load_phenopackets"><code class="xref py py-func docutils literal notranslate"><span class="pre">load_phenopackets()</span></code></a>
loader function:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.preprocessing</span><span class="w"> </span><span class="kn">import</span> <span class="n">load_phenopackets</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">qc_results</span> <span class="o">=</span> <span class="n">load_phenopackets</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">phenopackets</span><span class="o">=</span><span class="n">phenopackets</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">cohort_creator</span><span class="o">=</span><span class="n">cohort_creator</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>  
<span class="go">Individuals Processed: ...</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
<span class="go">19</span>
</pre></div>
</div>
<p>The loader applies the <code class="docutils literal notranslate"><span class="pre">CohortCreator</span></code> to phenopackets, while keeping track of any issues,
and we get back a <code class="docutils literal notranslate"><span class="pre">cohort</span></code> as well as a <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PreprocessingValidationResult" title="gpsea.preprocessing.PreprocessingValidationResult"><code class="xref py py-class docutils literal notranslate"><span class="pre">PreprocessingValidationResult</span></code></a> (<code class="docutils literal notranslate"><span class="pre">qc_results</span></code>)
with any Q/C issues.</p>
<p>The <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PreprocessingValidationResult.summarize" title="gpsea.preprocessing.PreprocessingValidationResult.summarize"><code class="xref py py-meth docutils literal notranslate"><span class="pre">summarize()</span></code></a> method summarizes the found issues:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">qc_results</span><span class="o">.</span><span class="n">summarize</span><span class="p">()</span>  
<span class="go">Validated under permissive policy</span>
<span class="go">No errors or warnings were found</span>
</pre></div>
</div>
<p>No issues were found in the current cohort.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>See <a class="reference internal" href="#quality-control"><span class="std std-ref">Quality control</span></a> (few paragraphs below) for more info on <code class="docutils literal notranslate"><span class="pre">qc_results</span></code>.</p>
</div>
</section>
<section id="alternative-phenopacket-sources">
<h3>Alternative phenopacket sources<a class="headerlink" href="#alternative-phenopacket-sources" title="Link to this heading"></a></h3>
<p>More often than not, the phenopackets of interest will not be deposited in Phenopacket Store.
For these use cases, GPSEA simplifies loading phenopackets
from a list of JSON files, or from a folder with phenopacket JSON files.</p>
<p>The <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.load_phenopacket_files" title="gpsea.preprocessing.load_phenopacket_files"><code class="xref py py-func docutils literal notranslate"><span class="pre">load_phenopacket_files()</span></code></a> function loads
phenopackets from one or more paths that point to phenopacket JSON files:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.preprocessing</span><span class="w"> </span><span class="kn">import</span> <span class="n">load_phenopacket_files</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pp_file_paths</span> <span class="o">=</span> <span class="p">(</span><span class="s1">&#39;path/to/phenopacket1.json&#39;</span><span class="p">,</span> <span class="s1">&#39;path/to/phenopacket2.json&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">qc_results</span> <span class="o">=</span> <span class="n">load_phenopacket_files</span><span class="p">(</span><span class="n">pp_file_paths</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>  
</pre></div>
</div>
<p>Alternatively, you can load an entire directory of phenopackets
with the <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.load_phenopacket_folder" title="gpsea.preprocessing.load_phenopacket_folder"><code class="xref py py-func docutils literal notranslate"><span class="pre">load_phenopacket_folder()</span></code></a> loader function.
Starting with path to a directory with phenopacket JSON files,
the loader includes all files that end with <code class="docutils literal notranslate"><span class="pre">*.json</span></code> suffix
and ignores any other files or sub-directories:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.preprocessing</span><span class="w"> </span><span class="kn">import</span> <span class="n">load_phenopacket_folder</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pp_dir</span> <span class="o">=</span> <span class="s1">&#39;path/to/folder/with/many/phenopacket/json/files&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">qc_results</span> <span class="o">=</span> <span class="n">load_phenopacket_folder</span><span class="p">(</span><span class="n">pp_dir</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>  
</pre></div>
</div>
</section>
</section>
<section id="quality-control">
<span id="id3"></span><h2><a class="toc-backref" href="#id7" role="doc-backlink">Quality control</a><a class="headerlink" href="#quality-control" title="Link to this heading"></a></h2>
<p>Besides the <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>, the loader functions also provide Q/C results (<code class="docutils literal notranslate"><span class="pre">qc_results</span></code>)
as <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PreprocessingValidationResult" title="gpsea.preprocessing.PreprocessingValidationResult"><code class="xref py py-class docutils literal notranslate"><span class="pre">PreprocessingValidationResult</span></code></a>.
The Q/C checker points out as many issues as possible (not just the first one),
to address all issues at once, as opposed to time-consuming iterative fixing.
The issues can be explored programmatically
through the <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PreprocessingValidationResult" title="gpsea.preprocessing.PreprocessingValidationResult"><code class="xref py py-class docutils literal notranslate"><span class="pre">PreprocessingValidationResult</span></code></a> API,
or we can display a summary with the
<a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PreprocessingValidationResult.summarize" title="gpsea.preprocessing.PreprocessingValidationResult.summarize"><code class="xref py py-meth docutils literal notranslate"><span class="pre">summarize()</span></code></a> method:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">qc_results</span><span class="o">.</span><span class="n">summarize</span><span class="p">()</span>  
<span class="go">Validated under permissive policy</span>
<span class="go">No errors or warnings were found</span>
</pre></div>
</div>
<p>In this case, no Q/C issues were found.</p>
</section>
<section id="persist-the-cohort-for-later">
<span id="cohort-persistence"></span><h2><a class="toc-backref" href="#id8" role="doc-backlink">Persist the cohort for later</a><a class="headerlink" href="#persist-the-cohort-for-later" title="Link to this heading"></a></h2>
<p>The preprocessing of a cohort can take some time even if we cache the responses from remote resources,
such as Variant Effect Predictor, Variant Validator, or Uniprot.
GPSEA ships with a custom encoder and decoder
that integrates with Python’s built-in <a class="reference external" href="https://docs.python.org/3/library/json.html#module-json" title="(in Python v3.13)"><code class="xref py py-mod docutils literal notranslate"><span class="pre">json</span></code></a> module,
to persist a <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a> to a JSON file on Friday afternoon,
and load it back on Monday morning.</p>
<section id="example">
<h3>Example<a class="headerlink" href="#example" title="Link to this heading"></a></h3>
<p>We can dump the <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a> into JSON
by providing <a class="reference internal" href="../apidocs/gpsea.io.html#gpsea.io.GpseaJSONEncoder" title="gpsea.io.GpseaJSONEncoder"><code class="xref py py-class docutils literal notranslate"><span class="pre">GpseaJSONEncoder</span></code></a> via the <cite>cls</cite> option to the <cite>json</cite> module functions,
such as the <a class="reference external" href="https://docs.python.org/3/library/json.html#json.dumps" title="(in Python v3.13)"><code class="xref py py-func docutils literal notranslate"><span class="pre">json.dumps()</span></code></a> which encodes an object into a JSON <cite>str</cite>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.io</span><span class="w"> </span><span class="kn">import</span> <span class="n">GpseaJSONEncoder</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">encoded</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">dumps</span><span class="p">(</span><span class="n">cohort</span><span class="p">,</span> <span class="bp">cls</span><span class="o">=</span><span class="n">GpseaJSONEncoder</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">encoded</span><span class="p">[:</span><span class="mi">80</span><span class="p">]</span>
<span class="go">&#39;{&quot;members&quot;: [{&quot;labels&quot;: {&quot;label&quot;: &quot;Subject 8&quot;, &quot;meta_label&quot;: &quot;PMID_29330883_Subj&#39;</span>
</pre></div>
</div>
<p>Here we see the first 80 letters of the JSON object.</p>
<p>We can decode the JSON object with <a class="reference internal" href="../apidocs/gpsea.io.html#gpsea.io.GpseaJSONDecoder" title="gpsea.io.GpseaJSONDecoder"><code class="xref py py-class docutils literal notranslate"><span class="pre">GpseaJSONDecoder</span></code></a> to get the same cohort back:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.io</span><span class="w"> </span><span class="kn">import</span> <span class="n">GpseaJSONDecoder</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">decoded</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">loads</span><span class="p">(</span><span class="n">encoded</span><span class="p">,</span> <span class="bp">cls</span><span class="o">=</span><span class="n">GpseaJSONDecoder</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span> <span class="o">==</span> <span class="n">decoded</span>
<span class="go">True</span>
</pre></div>
</div>
<p>We will leave persisting the cohort into an actual file or another data store
as an exercise for the interested readers.</p>
</section>
</section>
<section id="choose-the-transcript-and-protein-of-interest">
<span id="choose-tx-and-protein"></span><h2><a class="toc-backref" href="#id9" role="doc-backlink">Choose the transcript and protein of interest</a><a class="headerlink" href="#choose-the-transcript-and-protein-of-interest" title="Link to this heading"></a></h2>
<p>G/P association analysis is performed with reference to a specific gene transcript and the corresponding protein sequence.
The transcript and protein information is also needed for summary statistics generation,
to visualize variant distribution with respect to transcript/protein sequence,
and in most genotype partitioning schemes.</p>
<p>Here we point out our recommendations for choosing the transcript of interest
and let GPSEA fetch the associated data.</p>
<section id="choose-the-transcript">
<h3>Choose the transcript<a class="headerlink" href="#choose-the-transcript" title="Link to this heading"></a></h3>
<p>For the analysis, the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/refseq/MANE/">MANE</a> transcript
(i.e., the “main” biomedically relevant transcript of a gene) should be chosen unless
there is a specific reason not to (which should occur rarely if at all).</p>
<p>A good way to find the MANE transcript is to search on the gene symbol (e.g., <em>TBX5</em>) in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a>
and to choose a variant that is specifically located in the gene.
The MANE transcript will be displayed here
(e.g., <a class="reference external" href="https://www.ncbi.nlm.nih.gov/clinvar/variation/495227/">NM_181486.4(TBX5):c.1221C&gt;G (p.Tyr407Ter)</a>).
The RefSeq identifier of the encoded protein (e.g. <cite>NP_852259.1</cite> for <em>TBX5</em>) should be also readily available on the ClinVar website:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s2">&quot;NM_181486.4&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pt_id</span> <span class="o">=</span> <span class="s2">&quot;NP_852259.1&quot;</span>
</pre></div>
</div>
</section>
<section id="get-the-transcript-data">
<h3>Get the transcript data<a class="headerlink" href="#get-the-transcript-data" title="Link to this heading"></a></h3>
<p>Besides the transcript accession ID, the downstream analysis may need more information
about the “anatomy” of the transcript of interest, such as the coordinates of the exons
or the untranslated regions.
The coordinates can be represented either in <code class="xref py py-const docutils literal notranslate"><span class="pre">GRCh37</span></code>
or <code class="xref py py-class docutils literal notranslate"><span class="pre">GRCh38</span></code> (recommended) reference genomes.</p>
<p>GPSEA models the transcript anatomy with
<a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates" title="gpsea.model.TranscriptCoordinates"><code class="xref py py-class docutils literal notranslate"><span class="pre">TranscriptCoordinates</span></code></a> class,
and there are several ways to prepare the transcript coordinates,
which we list here by their convenience in decreasing order.</p>
<section id="fetch-transcript-coordinates-from-variant-validator-rest-api">
<h4>Fetch transcript coordinates from Variant Validator REST API<a class="headerlink" href="#fetch-transcript-coordinates-from-variant-validator-rest-api" title="Link to this heading"></a></h4>
<p>The most convenient way for getting the transcript coordinates is to use
the REST API of the amazing <a class="reference external" href="https://variantvalidator.org/">Variant Validator</a>.
GPSEA simplifies querying the API and parsing the response
with a <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.TranscriptCoordinateService" title="gpsea.preprocessing.TranscriptCoordinateService"><code class="xref py py-class docutils literal notranslate"><span class="pre">TranscriptCoordinateService</span></code></a>.
We use <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_default_tx_coordinate_service" title="gpsea.preprocessing.configure_default_tx_coordinate_service"><code class="xref py py-func docutils literal notranslate"><span class="pre">configure_default_tx_coordinate_service()</span></code></a> to get a service
that caches the response locally to prevent repeated API queries for the same transcript accession:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.preprocessing</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_default_tx_coordinate_service</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">txc_service</span> <span class="o">=</span> <span class="n">configure_default_tx_coordinate_service</span><span class="p">(</span><span class="n">genome_build</span><span class="o">=</span><span class="s2">&quot;GRCh38.p13&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>Now we can fetch the coordinates of the MANE transcript of <em>TBX5</em> on GRCh38:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span> <span class="o">=</span> <span class="n">txc_service</span><span class="o">.</span><span class="n">fetch</span><span class="p">(</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">identifier</span>
<span class="go">&#39;NM_181486.4&#39;</span>
</pre></div>
</div>
</section>
<section id="provide-the-transcript-coordinates-manually">
<h4>Provide the transcript coordinates manually<a class="headerlink" href="#provide-the-transcript-coordinates-manually" title="Link to this heading"></a></h4>
<p>TODO: implement and document!</p>
</section>
</section>
<section id="showcase-transcript-data">
<h3>Showcase transcript data<a class="headerlink" href="#showcase-transcript-data" title="Link to this heading"></a></h3>
<p>Based on <code class="docutils literal notranslate"><span class="pre">tx_coordinates</span></code>, GPSEA knows about the location of the transcript region in the reference genome:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">region</span><span class="o">.</span><span class="n">contig</span><span class="o">.</span><span class="n">name</span>
<span class="go">&#39;12&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">region</span><span class="o">.</span><span class="n">start</span>
<span class="go">18869165</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">region</span><span class="o">.</span><span class="n">end</span>
<span class="go">18921399</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">region</span><span class="o">.</span><span class="n">strand</span><span class="o">.</span><span class="n">symbol</span>
<span class="go">&#39;-&#39;</span>
</pre></div>
</div>
<p>or the count and coordinates of exons:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">exons</span><span class="p">)</span>
<span class="go">9</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">exons</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>  <span class="c1"># coordinate of the 1st exon</span>
<span class="go">GenomicRegion(contig=12, start=18869165, end=18869682, strand=-)</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The regions spanned by transcripts, exons, UTRs, as well as by variants
are represented either as a <a class="reference internal" href="../apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomicRegion" title="gpsea.model.genome.GenomicRegion"><code class="xref py py-class docutils literal notranslate"><span class="pre">GenomicRegion</span></code></a>
or a <a class="reference internal" href="../apidocs/gpsea.model.genome.html#gpsea.model.genome.Region" title="gpsea.model.genome.Region"><code class="xref py py-class docutils literal notranslate"><span class="pre">Region</span></code></a>.</p>
</div>
<p>Furthermore, we know that the transcript is coding</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">is_coding</span><span class="p">()</span>
<span class="go">True</span>
</pre></div>
</div>
<p>and so we can see that 8 exons include protein coding sequences,</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">get_cds_regions</span><span class="p">())</span>
<span class="go">8</span>
</pre></div>
</div>
<p>2 exons include 5’ untranslated regions,</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">get_five_prime_utrs</span><span class="p">())</span>
<span class="go">2</span>
</pre></div>
</div>
<p>and the coding sequence includes 1554 coding bases and 518 codons.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">get_coding_base_count</span><span class="p">()</span>
<span class="go">1554</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span><span class="o">.</span><span class="n">get_codon_count</span><span class="p">()</span>
<span class="go">518</span>
</pre></div>
</div>
</section>
<section id="fetch-protein-data">
<span id="id4"></span><h3>Fetch protein data<a class="headerlink" href="#fetch-protein-data" title="Link to this heading"></a></h3>
<p>Specific domains of a protein may be associated with genotype-phenotype correlations.
For instance, variants in the pore domain of <em>PIEZO1</em> are associated with more severe clinical
manifestions in dehydrated hereditary stomatocytosis <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30187933">Andolfo et al., 2018</a>.</p>
<p>GPSEA uses the protein data in several places: to show distribution of variants with respect to the protein domains
or other features of biological interest, and to group the individuals based on presence of a variant predicted
to affect the protein features.
In all cases, the protein data must be formatted as an instance of <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.ProteinMetadata" title="gpsea.model.ProteinMetadata"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinMetadata</span></code></a>
and here we show how to get the data and use it in the analysis.</p>
<p>The protein data (<a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.ProteinMetadata" title="gpsea.model.ProteinMetadata"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinMetadata</span></code></a>) can be obtained in several ways,
ordered (again) by their convenience:</p>
<ul class="simple">
<li><p>fetched from UniProt REST API</p></li>
<li><p>parsed from a JSON file downloaded from UniProt</p></li>
<li><p>entered manually from a data frame</p></li>
</ul>
<section id="fetch-data-from-uniprot-rest-api">
<h4>Fetch data from UniProt REST API<a class="headerlink" href="#fetch-data-from-uniprot-rest-api" title="Link to this heading"></a></h4>
<p>The most convenient way to obtain the protein data is to use a <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.ProteinMetadataService" title="gpsea.preprocessing.ProteinMetadataService"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinMetadataService</span></code></a>.
We recommend using the <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_default_protein_metadata_service" title="gpsea.preprocessing.configure_default_protein_metadata_service"><code class="xref py py-func docutils literal notranslate"><span class="pre">configure_default_protein_metadata_service()</span></code></a>
to reduce the amount of the associated boiler-plate code:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.preprocessing</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_default_protein_metadata_service</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pms</span> <span class="o">=</span> <span class="n">configure_default_protein_metadata_service</span><span class="p">()</span>
</pre></div>
</div>
<p>Then, fetching the data for protein accession <em>NP_852259.1</em> encoded by the <em>NM_181486.4</em> transcript of <em>TBX5</em>
is as simple as running:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">protein_meta</span> <span class="o">=</span> <span class="n">pms</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="n">pt_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">protein_meta</span><span class="o">.</span><span class="n">protein_id</span>
<span class="go">&#39;NP_852259.1&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">protein_meta</span><span class="o">.</span><span class="n">protein_length</span>
<span class="go">518</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">protein_meta</span><span class="o">.</span><span class="n">protein_features</span><span class="p">)</span>
<span class="go">5</span>
</pre></div>
</div>
<p>The <cite>protein_meta</cite> represents the <em>TBX5</em> isoform that includes 518 aminoacids and two features of interest,
which we can see on the following screenshot of the UniProt entry for <em>TBX5</em>:</p>
<figure class="align-center" id="id5">
<a class="reference internal image-reference" href="../_images/TBX5_uniprot_features.png"><img alt="*TBX5* (P37173, UniProt entry)" src="../_images/TBX5_uniprot_features.png" style="width: 800px;" />
</a>
<figcaption>
<p><span class="caption-text">Protein features of <em>TBX5</em> (Q99593, UniProt entry)</span><a class="headerlink" href="#id5" title="Link to this image"></a></p>
</figcaption>
</figure>
<p>UniProt shows four protein features:</p>
<ul class="simple">
<li><p>the Disordered region (1-46)</p></li>
<li><p>the Disordered region (250-356)</p></li>
<li><p>presence of Polar residues (263-299)</p></li>
<li><p>presence of Basic and acidic residues (320-346).</p></li>
</ul>
</section>
<section id="parse-uniprot-json-dump">
<h4>Parse UniProt JSON dump<a class="headerlink" href="#parse-uniprot-json-dump" title="Link to this heading"></a></h4>
<p>In the cases, when the REST API cannot give us the data for a protein of interest,
we can download a JSON file representing the protein features manually,
and load the file into <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.ProteinMetadata" title="gpsea.model.ProteinMetadata"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinMetadata</span></code></a>.</p>
<p>To do this, click on the <em>Download</em> symbol (see the UniProt screenshot figure above). This will open a dialog
that allows the user to choose the contents of the JSON file.
Do not change the default option (Features - Domain, Region).
Provided that the file has been saved as <cite>docs/user-guide/data/Q99593.json</cite>,
the <code class="docutils literal notranslate"><span class="pre">ProteinMetadata</span></code> can be loaded using <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.from_uniprot_json" title="gpsea.model.ProteinMetadata.from_uniprot_json"><code class="xref py py-func docutils literal notranslate"><span class="pre">from_uniprot_json()</span></code></a> function.
Note that you will need to obtain information about the protein name (<code class="docutils literal notranslate"><span class="pre">label</span></code>)
and <code class="docutils literal notranslate"><span class="pre">protein_length</span></code>, but these are shown in the UniProt entry:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.model</span><span class="w"> </span><span class="kn">import</span> <span class="n">ProteinMetadata</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">downloaded_uniprot_json</span> <span class="o">=</span> <span class="s2">&quot;docs/user-guide/data/Q99593.json&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">protein_meta</span> <span class="o">=</span> <span class="n">ProteinMetadata</span><span class="o">.</span><span class="n">from_uniprot_json</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">protein_id</span><span class="o">=</span><span class="s2">&quot;NP_852259.1&quot;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">label</span><span class="o">=</span><span class="s2">&quot;transforming growth factor beta receptor 2&quot;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">uniprot_json</span><span class="o">=</span><span class="n">downloaded_uniprot_json</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">protein_length</span><span class="o">=</span><span class="mi">518</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="enter-features-manually">
<h4>Enter features manually<a class="headerlink" href="#enter-features-manually" title="Link to this heading"></a></h4>
<p>The information about protein features provided by UniProt entries may not always be complete.
Here we show how to enter the same information manually, in a custom protein dataframe.</p>
<p>The frame can be created e.g. by running:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">domains</span> <span class="o">=</span> <span class="p">[</span>
<span class="gp">... </span>   <span class="p">{</span><span class="s2">&quot;region&quot;</span><span class="p">:</span> <span class="s2">&quot;Disordered&quot;</span><span class="p">,</span><span class="s2">&quot;category&quot;</span><span class="p">:</span> <span class="s2">&quot;region&quot;</span><span class="p">,</span> <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="mi">46</span><span class="p">,</span> <span class="p">},</span>
<span class="gp">... </span>   <span class="p">{</span><span class="s2">&quot;region&quot;</span><span class="p">:</span> <span class="s2">&quot;Disordered&quot;</span><span class="p">,</span> <span class="s2">&quot;category&quot;</span><span class="p">:</span> <span class="s2">&quot;region&quot;</span><span class="p">,</span> <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="mi">250</span><span class="p">,</span> <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="mi">356</span><span class="p">,</span> <span class="p">},</span>
<span class="gp">... </span>   <span class="p">{</span><span class="s2">&quot;region&quot;</span><span class="p">:</span> <span class="s2">&quot;Polar residues&quot;</span><span class="p">,</span> <span class="s2">&quot;category&quot;</span><span class="p">:</span> <span class="s2">&quot;compositional bias&quot;</span><span class="p">,</span> <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="mi">263</span><span class="p">,</span> <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="mi">299</span><span class="p">,</span> <span class="p">},</span>
<span class="gp">... </span>   <span class="p">{</span><span class="s2">&quot;region&quot;</span><span class="p">:</span> <span class="s2">&quot;Basic and acidic residues&quot;</span><span class="p">,</span> <span class="s2">&quot;category&quot;</span><span class="p">:</span> <span class="s2">&quot;compositional bias&quot;</span><span class="p">,</span> <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="mi">320</span><span class="p">,</span> <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="mi">346</span><span class="p">,</span> <span class="p">},</span>
<span class="gp">... </span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">domains</span><span class="p">)</span>
</pre></div>
</div>
<p>The <cite>ProteinMetadata</cite> is then created using <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.from_feature_frame" title="gpsea.model.ProteinMetadata.from_feature_frame"><code class="xref py py-func docutils literal notranslate"><span class="pre">from_feature_frame()</span></code></a> function:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">protein_meta</span> <span class="o">=</span> <span class="n">ProteinMetadata</span><span class="o">.</span><span class="n">from_feature_frame</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">protein_id</span><span class="o">=</span><span class="s2">&quot;NP_852259.1&quot;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">label</span><span class="o">=</span><span class="s2">&quot;transforming growth factor beta receptor 2&quot;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">features</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">protein_length</span><span class="o">=</span><span class="mi">518</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="index.html" class="btn btn-neutral float-left" title="User guide" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="exploratory.html" class="btn btn-neutral float-right" title="Cohort exploratory analysis" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>