<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Predicates &mdash; genophenocorr 0.1.1dev documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/genophenocorr.css?v=a9fa5eb9" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=8427294d"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="API reference" href="../apidocs/modules.html" />
    <link rel="prev" title="Input data" href="input-data.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            genophenocorr
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="index.html">User guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="input-data.html">Input data</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Predicates</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#simple-predicates">Simple predicates</a></li>
<li class="toctree-l3"><a class="reference internal" href="#compound-predicates">Compound predicates</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">genophenocorr</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="index.html">User guide</a></li>
      <li class="breadcrumb-item active">Predicates</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/user-guide/predicates.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="predicates">
<span id="id1"></span><h1>Predicates<a class="headerlink" href="#predicates" title="Link to this heading"></a></h1>
<p>Genophenocorr uses predicates to test if variant, patient, or any tested item
meets a condition. Based on the test results, the items are assigned into groups.</p>
<p>As described in the <a class="reference internal" href="../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PolyPredicate" title="genophenocorr.analysis.predicate.PolyPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.analysis.predicate.PolyPredicate</span></code></a> API,
the groups must be <em>exclusive</em> - the item can be assigned with one and only one group,
and <em>exhaustive</em> - the groups must cover all possible scenarios.</p>
<p>However, if the item cannot be assigned into any meaningful category,
the predicate can return <cite>None</cite>, and the item will be omitted from the analysis.</p>
<p>The predicates can be chained to test for more complex conditions.
For instance, “test if a patient has a missense or synonymous variant located in exon 6 of transcript <cite>NM_013275.6</cite>”.</p>
<p>Let’s demonstrate this on an example with a <a class="reference internal" href="../apidocs/genophenocorr.analysis.predicate.genotype.html#genophenocorr.analysis.predicate.genotype.VariantPredicate" title="genophenocorr.analysis.predicate.genotype.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.analysis.predicate.genotype.VariantPredicate</span></code></a>.
We will load a cohort of 5 subjects with variants in <em>ANKRD11</em>, leading to KBG syndrome.
The the clinical signs and symptoms of the subjects were encoded into HPO terms
along with the pathogenic <em>ANKRD11</em> variant.</p>
<p>Let’s load the phenopackets, as previously described in greater detail the <a class="reference internal" href="input-data.html#input-data"><span class="std std-ref">Input data</span></a> section.</p>
<p>First, we load HPO using HPO toolkit:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-03-06&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>then, we will configure the cohort creator:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.preprocessing</span> <span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span><span class="p">,</span> <span class="n">load_phenopacket_folder</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
</pre></div>
</div>
<p>last, we will load the cohort from a directory with phenopackets:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">os</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_dir</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="s1">&#39;docs&#39;</span><span class="p">,</span> <span class="s1">&#39;data&#39;</span><span class="p">,</span> <span class="s1">&#39;simple_cohort&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span> <span class="o">=</span> <span class="n">load_phenopacket_folder</span><span class="p">(</span><span class="n">cohort_dir</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
<span class="go">5</span>
</pre></div>
</div>
<p>We loaded a cohort of 5 patients.</p>
<p>Let’s use the variant <code class="docutils literal notranslate"><span class="pre">16_89281397_89281397_G_C</span></code> that corresponds
to a pathogenic variant <a class="reference external" href="https://www.ncbi.nlm.nih.gov/clinvar/variation/1029215/">VCV001029215.1</a>
that replaces the tyrosine encoded by the 1715th codon of <cite>NM_013275.6</cite> with a premature stop codon: <code class="docutils literal notranslate"><span class="pre">NM_013275.6(ANKRD11):c.5145C&gt;G</span> <span class="pre">(p.Tyr1715Ter)</span></code>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">variant_key_of_interest</span> <span class="o">=</span> <span class="s1">&#39;16_89281397_89281397_G_C&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant</span> <span class="o">=</span> <span class="n">cohort</span><span class="o">.</span><span class="n">get_variant_by_key</span><span class="p">(</span><span class="n">variant_key_of_interest</span><span class="p">)</span>
</pre></div>
</div>
<p>We will use the variant to exemplify the predicate API.</p>
<section id="simple-predicates">
<h2>Simple predicates<a class="headerlink" href="#simple-predicates" title="Link to this heading"></a></h2>
<p>We can check that the variant overlaps with <em>ANKRD11</em>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gene</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">gene</span><span class="p">(</span><span class="s1">&#39;ANKRD11&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gene</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>it overlaps with the <em>MANE</em> transcript:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">ankrd11_mane_tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_013275.6&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">transcript</span><span class="p">(</span><span class="n">ankrd11_mane_tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>it in fact overlaps with the exon 9:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">exon9</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">exon</span><span class="p">(</span><span class="n">exon</span><span class="o">=</span><span class="mi">9</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">ankrd11_mane_tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">exon9</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>and it is predicted to introduce a premature termination codon to the MANE transcript:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">nonsense</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">STOP_GAINED</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">ankrd11_mane_tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">nonsense</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>See <a class="reference internal" href="../apidocs/genophenocorr.analysis.predicate.genotype.html#genophenocorr.analysis.predicate.genotype.VariantPredicates" title="genophenocorr.analysis.predicate.genotype.VariantPredicates"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.analysis.predicate.genotype.VariantPredicates</span></code></a>
for more info on the predicates available off the shelf.</p>
</section>
<section id="compound-predicates">
<h2>Compound predicates<a class="headerlink" href="#compound-predicates" title="Link to this heading"></a></h2>
<p>The simple predicates can be combined to test for more elaborate conditions.
For instance, we can test if the variant meets <em>any</em> or several conditions:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">missense</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">ankrd11_mane_tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">missense_or_nonsense</span> <span class="o">=</span> <span class="n">missense</span> <span class="o">|</span> <span class="n">nonsense</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">missense_or_nonsense</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>or <em>all</em> conditions:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">nonsense_and_exon9</span> <span class="o">=</span> <span class="n">nonsense</span> <span class="o">&amp;</span> <span class="n">exon9</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">nonsense_and_exon9</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>The <cite>VariantPredicate</cite> overloads Python <cite>&amp;</cite> (AND) and <cite>|</cite> (OR) operators to build a compound predicate from a building block.</p>
<p>That’s it for predicates.</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="input-data.html" class="btn btn-neutral float-left" title="Input data" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../apidocs/modules.html" class="btn btn-neutral float-right" title="API reference" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>