<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Statistical tests &mdash; GPSEA 0.1.1dev documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/gpsea.css?v=547657ed" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=8427294d"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../_static/copybutton.js?v=6dbb43f8"></script>
        <script>window.MathJax = {"TeX": {"Macros": {"RR": "{\\mathbb{R}}"}}}</script>
        <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Multiple-testing correction" href="mtc.html" />
    <link rel="prev" title="Predicates" href="predicates.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            GPSEA
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="index.html">User guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="input-data.html">Input data</a></li>
<li class="toctree-l2"><a class="reference internal" href="exploratory.html">Cohort exploratory analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="predicates.html">Predicates</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Statistical tests</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#compare-genotype-and-phenotype-groups">Compare genotype and phenotype groups</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#fisher-exact-test-fet">Fisher exact test (FET)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#phenotype-score">Phenotype score</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#mann-whitney-u-test">Mann-Whitney U Test</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="mtc.html">Multiple-testing correction</a></li>
<li class="toctree-l2"><a class="reference internal" href="glossary.html">Glossary</a></li>
<li class="toctree-l2"><a class="reference internal" href="autosomal_recessive.html">Genotype-Phenotype Correlations in Autosomal Recessive Diseases</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="index.html">User guide</a></li>
      <li class="breadcrumb-item active">Statistical tests</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/user-guide/stats.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="statistical-tests">
<span id="stats"></span><h1>Statistical tests<a class="headerlink" href="#statistical-tests" title="Link to this heading"></a></h1>
<p>There are many different ways of statistically testing for genotype-phenotype correlations,
and the appropriate statistical test depends on the question.
This document provides an overview of the tests offered by the GPSEA library
and explanations of how they are implemented by our software.</p>
<section id="compare-genotype-and-phenotype-groups">
<h2>Compare genotype and phenotype groups<a class="headerlink" href="#compare-genotype-and-phenotype-groups" title="Link to this heading"></a></h2>
<section id="fisher-exact-test-fet">
<span id="fisher-exact-test"></span><h3>Fisher exact test (FET)<a class="headerlink" href="#fisher-exact-test-fet" title="Link to this heading"></a></h3>
<p>The Fisher exact test (FET) calculates the exact probability value
for the relationship between two dichotomous variables.
In our implementation, the two dichotomous variables are the genotype and the phenotype.
For instance, the individuals of the cohort may be divided
according to whether or not they have a nonsense variant
and according to whether or not they have Strabismus (<a class="reference external" href="https://hpo.jax.org/browse/term/HP:0000486">HP:0000486</a>).</p>
<p>The results of FET are expressed in terms of an exact probability (P-value), varying within 0 and 1.
Two groups are considered statistically significant if the P-value is less
than the chosen significance level (usually <span class="math notranslate nohighlight">\(\alpha = 0.05\)</span>).</p>
<p>The following graphic shows an example contingency table that is used to conduct a Fisher exact test.
We are comparing the frequency of <em>strabismus</em> in individuals with missense and nonsense variants:</p>
<a class="reference internal image-reference" href="../_images/fisher.png"><img alt="Fisher exact text example" class="align-center" src="../_images/fisher.png" style="width: 600px;" /></a>
<p>To perform the corresponding test in Python, we would use the following code.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">scipy.stats</span> <span class="k">as</span> <span class="nn">stats</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">contingency_table</span> <span class="o">=</span> <span class="p">[</span>
<span class="gp">... </span><span class="c1">#   Missense,    Nonsense</span>
<span class="gp">... </span>   <span class="p">[</span><span class="mi">17</span><span class="p">,</span>          <span class="mi">6</span>       <span class="p">],</span>  <span class="c1"># Strabismus observed</span>
<span class="gp">... </span>   <span class="p">[</span><span class="mi">1</span><span class="p">,</span>          <span class="mi">19</span>       <span class="p">],</span>  <span class="c1"># Strabismus excluded</span>
<span class="gp">... </span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">oddsratio</span><span class="p">,</span> <span class="n">p_value</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">fisher_exact</span><span class="p">(</span><span class="n">contingency_table</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">float</span><span class="p">(</span><span class="n">oddsratio</span><span class="p">)</span>
<span class="go">53.833333333333336</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">float</span><span class="p">(</span><span class="n">p_value</span><span class="p">)</span>
<span class="go">5.432292015291845e-06</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">p_value</span></code> evaluates to <cite>5.432292015291845e-06</cite>, meaning there is a significant difference between the groups.</p>
<p>The Fisher exact test evaluates whether the observed frequencies in a contingency table significantly
deviate from the frequencies we would expect if there were no association between the variables.
We want to test whether the frequency of <cite>HP:0000486`</cite> is significantly higher or lower in
one genotype group compared to what would be expected if there were no association.
Note that by default, the <em>two-tailed</em> Fisher exact test is performed, meaning we have no
hypothesis as to whether there is a higher or lower frequency in one of the genotype groups.</p>
<p>However, we are typically interested in testing the associations between the genotype and multiple phenotypic features at once.
GPSEA takes advantage of the HPO structure and simplifies the testing for all HPO terms encoded in the cohort.</p>
<section id="example">
<h4>Example<a class="headerlink" href="#example" title="Link to this heading"></a></h4>
<p>Let’s illustrate this in a real-life example of the analysis of the association between frameshift variants in <em>TBX5</em> gene
and congenital heart defects in the dataset of 156 individuals with mutations in <em>TBX5</em> whose signs and symptoms were
encoded into HPO terms, stored as phenopackets of the <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/35705716">GA4GH Phenopacket Schema</a>,
and deposited in <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a> (version <cite>0.1.18</cite>).</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The shorter version of the same analysis has been presented in the <a class="reference internal" href="../tutorial.html#tutorial"><span class="std std-ref">Tutorial</span></a>.</p>
</div>
<section id="create-cohort">
<h5>Create cohort<a class="headerlink" href="#create-cohort" title="Link to this heading"></a></h5>
<p>We will load and transform the phenopackets into a <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>,
as described in <a class="reference internal" href="input-data.html#input-data"><span class="std std-ref">Input data</span></a> section. Briefly, we will load the phenopackets:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">ppktstore.registry</span> <span class="kn">import</span> <span class="n">configure_phenopacket_registry</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">registry</span> <span class="o">=</span> <span class="n">configure_phenopacket_registry</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">registry</span><span class="o">.</span><span class="n">open_phenopacket_store</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;0.1.18&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">ps</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">phenopackets</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">ps</span><span class="o">.</span><span class="n">iter_cohort_phenopackets</span><span class="p">(</span><span class="s1">&#39;TBX5&#39;</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">)</span>
<span class="go">156</span>
</pre></div>
</div>
<p>followed by loading HPO release <cite>v2024-07-01</cite>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>and we will perform Q/C and functional annotations for the mutations
with the default cohort creator:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span><span class="p">,</span> <span class="n">load_phenopackets</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">qc_results</span> <span class="o">=</span> <span class="n">load_phenopackets</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>  
<span class="go">Patients Created: ...</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">qc_results</span><span class="o">.</span><span class="n">summarize</span><span class="p">()</span>  
<span class="go">Validated under none policy</span>
<span class="go">No errors or warnings were found</span>
</pre></div>
</div>
</section>
<section id="configure-analysis">
<h5>Configure analysis<a class="headerlink" href="#configure-analysis" title="Link to this heading"></a></h5>
<p>We want to test the association between frameshift <em>TBX5</em> variants and phenotypic abnormalities.
GPSEA exposes a flexible predicate API that lets us create genotype and phenotype predicates
to assign the cohort members into genotype and phenotype categories based on the variants
and the HPO terms. We need to create one genotype predicate and one or more phenotype predicates.</p>
<p><strong>Genotype predicate</strong></p>
<p>We want to separate the patients into two groups: a group <em>with</em> a frameshift variant
and a group <em>without</em> a frameshift variant, based on the functional annotation.
We will use the <em>MANE</em> transcript for the analysis:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span><span class="p">,</span> <span class="n">boolean_predicate</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_181486.4&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span> <span class="o">=</span> <span class="n">boolean_predicate</span><span class="p">(</span><span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">FRAMESHIFT_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;FRAMESHIFT_VARIANT on NM_181486.4&#39;</span>
</pre></div>
</div>
<p><strong>Phenotype predicates</strong></p>
<p>We recommend testing the genotype phenotype association for all HPO terms that are present in 2 or more cohort members,
while taking advantage of the HPO graph structure and of the <a class="reference internal" href="glossary.html#true-path-rule"><span class="std std-ref">True path rule</span></a>.
We will use the <a class="reference internal" href="../apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.prepare_predicates_for_terms_of_interest" title="gpsea.analysis.predicate.phenotype.prepare_predicates_for_terms_of_interest"><code class="xref py py-func docutils literal notranslate"><span class="pre">prepare_predicates_for_terms_of_interest()</span></code></a>
utility function to generate phenotype predicates for all HPO terms:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.phenotype</span> <span class="kn">import</span> <span class="n">prepare_predicates_for_terms_of_interest</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pheno_predicates</span> <span class="o">=</span> <span class="n">prepare_predicates_for_terms_of_interest</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">hpo</span><span class="o">=</span><span class="n">hpo</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">min_n_of_patients_with_term</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">pheno_predicates</span><span class="p">)</span>
<span class="go">260</span>
</pre></div>
</div>
<p>The function finds all HPO terms that annotate at least <em>n</em> (<code class="docutils literal notranslate"><span class="pre">min_n_of_patients_with_term=2</span></code> above) individuals,
including the <em>indirect</em> annotations whose presence is implied by the true path rule.</p>
<p><strong>Statistical test</strong></p>
<p>We will use :ref:&lt;fisher-exact-test&gt; to test the association
between genotype and phenotype groups, as described previously.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pcats.stats</span> <span class="kn">import</span> <span class="n">ScipyFisherExact</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">count_statistic</span> <span class="o">=</span> <span class="n">ScipyFisherExact</span><span class="p">()</span>
</pre></div>
</div>
<p>FET will compute a p value for each genotype phenotype group.</p>
<p><strong>Multiple testing correction</strong></p>
<p>In the case of this cohort, we could test association between having a frameshift variant and one of 260 HPO terms.
However, testing multiple hypotheses on the same dataset increases the risk of finding a significant association
by chance.
GPSEA uses a two-pronged strategy to mitigate this risk - use Phenotype MTC filter and multiple testing correction.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>See the <a class="reference internal" href="mtc.html#mtc"><span class="std std-ref">Multiple-testing correction</span></a> section for more info on multiple testing procedures.</p>
</div>
<p>Here we will use a combination of the HPO MTC filter (<a class="reference internal" href="../apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter" title="gpsea.analysis.mtc_filter.HpoMtcFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">HpoMtcFilter</span></code></a>)
with Benjamini-Hochberg procedure (<code class="docutils literal notranslate"><span class="pre">mtc_correction='fdr_bh'</span></code>)
with a false discovery control level set to <cite>0.05</cite> (<code class="docutils literal notranslate"><span class="pre">mtc_alpha=0.05</span></code>):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.mtc_filter</span> <span class="kn">import</span> <span class="n">HpoMtcFilter</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_filter</span> <span class="o">=</span> <span class="n">HpoMtcFilter</span><span class="o">.</span><span class="n">default_filter</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">term_frequency_threshold</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_correction</span> <span class="o">=</span> <span class="s1">&#39;fdr_bh&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_alpha</span> <span class="o">=</span> <span class="mf">0.05</span>
</pre></div>
</div>
<p><strong>Final analysis</strong></p>
<p>We finalize the analysis setup by putting all components together
into <a class="reference internal" href="../apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysis" title="gpsea.analysis.pcats.HpoTermAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">HpoTermAnalysis</span></code></a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pcats</span> <span class="kn">import</span> <span class="n">HpoTermAnalysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">analysis</span> <span class="o">=</span> <span class="n">HpoTermAnalysis</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">count_statistic</span><span class="o">=</span><span class="n">count_statistic</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_filter</span><span class="o">=</span><span class="n">mtc_filter</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_correction</span><span class="o">=</span><span class="n">mtc_correction</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_alpha</span><span class="o">=</span><span class="n">mtc_alpha</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="analysis">
<h5>Analysis<a class="headerlink" href="#analysis" title="Link to this heading"></a></h5>
<p>We can now execute the analysis:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span> <span class="o">=</span> <span class="n">analysis</span><span class="o">.</span><span class="n">compare_genotype_vs_phenotypes</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">gt_predicate</span><span class="o">=</span><span class="n">gt_predicate</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">pheno_predicates</span><span class="o">=</span><span class="n">pheno_predicates</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">result</span><span class="o">.</span><span class="n">phenotypes</span><span class="p">)</span>
<span class="go">260</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">total_tests</span>
<span class="go">17</span>
</pre></div>
</div>
<p>Thanks to Phenotype MTC filter, we only tested 17 out of 260 terms.
We can learn more by showing the MTC filter report:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">MtcStatsViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_viewer</span> <span class="o">=</span> <span class="n">MtcStatsViewer</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_report</span> <span class="o">=</span> <span class="n">mtc_viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;docs/user-guide/report/tbx5_frameshift.mtc_report.html&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">fh</span><span class="p">:</span>  
<span class="gp">... </span>    <span class="n">_</span> <span class="o">=</span> <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">mtc_report</span><span class="p">)</span>
</pre></div>
</div>
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Cohort</title>
  <style>table {
    border-collapse: collapse;
    margin: 25px 0;
      font-size: 0.9em;
      font-family: sans-serif;
      min-width: 400px;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
  }


  th {
    background-color: #f2f2f2;
    border: 1px solid #dddddd;
    text-align: left;
    padding: 2px;
  }

  tr {
    border: 1px solid #dddddd;
  }

  td {
    padding: 2px;
    font-weight: bold;
  }

  tr:nth-child(even) {
    background-color: #f2f2f2;
  }

  .table td,tr {
    text-align: left;
}

.table td:first-child, tr:first-child {
    text-align: left;
}

  </style>
</head>

<body>
  <h1>Phenotype testing report</h1>
  <p>Phenotype MTC filter: <em>HPO MTC filter</em></p>
  <p>Multiple testing correction: <em>fdr_bh</em></p>
  <p>Performed statistical tests for 17 out of the total of 260 HPO terms.</p>
    <table>
      <caption>Using <em>HPO MTC filter</em>, 243 term(s) were omitted from statistical analysis.</caption>
        <tbody>
          <tr>
            <th>Code</th>
            <th>Reason</th>
            <th>Count</th>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 2 observations (not powered for 2x2)</td>
            <td>51</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term because all genotypes have same HPO observed proportions</td>
            <td>50</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping general term</td>
            <td>44</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 3 observations (not powered for 2x2)</td>
            <td>27</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 6 observations (not powered for 2x2)</td>
            <td>19</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 4 observations (not powered for 2x2)</td>
            <td>16</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 5 observations (not powered for 2x2)</td>
            <td>10</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with maximum frequency that was less than threshold 0.2</td>
            <td>10</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 1 observations (not powered for 2x2)</td>
            <td>7</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term because one genotype had zero observations</td>
            <td>5</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term because no genotype has more than one observed HPO count</td>
            <td>4</td>
          </tr>
          
        </tbody>
    </table>
</body>
</html>
</section>
<section id="genotype-phenotype-associations">
<h5>Genotype phenotype associations<a class="headerlink" href="#genotype-phenotype-associations" title="Link to this heading"></a></h5>
<p>Last, let’s explore the associations. This is a table of the tested HPO terms
ordered by the corrected p value (Benjamini-Hochberg FDR):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate</span> <span class="kn">import</span> <span class="n">PatientCategories</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span> <span class="o">=</span> <span class="n">result</span><span class="o">.</span><span class="n">summarize</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">PatientCategories</span><span class="o">.</span><span class="n">YES</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;docs/user-guide/report/tbx5_frameshift.csv&#39;</span><span class="p">)</span>  
</pre></div>
</div>
<table class="docutils align-default" id="id5">
<caption><span class="caption-text"><em>TBX5</em> frameshift vs rest</span><a class="headerlink" href="#id5" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"><p>FRAMESHIFT_VARIANT on NM_181486.4</p></th>
<th class="head"><p>Yes</p></th>
<th class="head"><p>Yes</p></th>
<th class="head"><p>No</p></th>
<th class="head"><p>No</p></th>
<th class="head"></th>
<th class="head"></th>
</tr>
<tr class="row-even"><th class="head"></th>
<th class="head"><p>Count</p></th>
<th class="head"><p>Percent</p></th>
<th class="head"><p>Count</p></th>
<th class="head"><p>Percent</p></th>
<th class="head"><p>Corrected p values</p></th>
<th class="head"><p>p values</p></th>
</tr>
</thead>
<tbody>
<tr class="row-odd"><td><p>Ventricular septal defect [HP:0001629]</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td><p>42/71</p></td>
<td><p>59%</p></td>
<td><p>0.004112753923262261</p></td>
<td><p>0.00024192670136836828</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal atrioventricular conduction [HP:0005150]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>1/23</p></td>
<td><p>4%</p></td>
<td><p>0.013076923076923076</p></td>
<td><p>0.0015384615384615385</p></td>
</tr>
<tr class="row-odd"><td><p>Absent thumb [HP:0009777]</p></td>
<td><p>14/31</p></td>
<td><p>45%</p></td>
<td><p>18/100</p></td>
<td><p>18%</p></td>
<td><p>0.021044138590779585</p></td>
<td><p>0.0037136715160199273</p></td>
</tr>
<tr class="row-even"><td><p>Atrioventricular block [HP:0001678]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>1/23</p></td>
<td><p>4%</p></td>
<td><p>0.034</p></td>
<td><p>0.010000000000000002</p></td>
</tr>
<tr class="row-odd"><td><p>Heart block [HP:0012722]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>1/23</p></td>
<td><p>4%</p></td>
<td><p>0.034</p></td>
<td><p>0.010000000000000002</p></td>
</tr>
<tr class="row-even"><td><p>Patent ductus arteriosus [HP:0001643]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>6/40</p></td>
<td><p>15%</p></td>
<td><p>0.09214092140921408</p></td>
<td><p>0.03252032520325203</p></td>
</tr>
<tr class="row-odd"><td><p>Secundum atrial septal defect [HP:0001684]</p></td>
<td><p>4/22</p></td>
<td><p>18%</p></td>
<td><p>23/55</p></td>
<td><p>42%</p></td>
<td><p>0.1440020479198931</p></td>
<td><p>0.06544319142266644</p></td>
</tr>
<tr class="row-even"><td><p>Triphalangeal thumb [HP:0001199]</p></td>
<td><p>13/32</p></td>
<td><p>41%</p></td>
<td><p>23/99</p></td>
<td><p>23%</p></td>
<td><p>0.1440020479198931</p></td>
<td><p>0.06932119159387057</p></td>
</tr>
<tr class="row-odd"><td><p>Cardiac conduction abnormality [HP:0031546]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>15/37</p></td>
<td><p>41%</p></td>
<td><p>0.1440020479198931</p></td>
<td><p>0.08259109311740892</p></td>
</tr>
<tr class="row-even"><td><p>Muscular ventricular septal defect [HP:0011623]</p></td>
<td><p>6/25</p></td>
<td><p>24%</p></td>
<td><p>8/84</p></td>
<td><p>10%</p></td>
<td><p>0.1440020479198931</p></td>
<td><p>0.08470708701170182</p></td>
</tr>
<tr class="row-odd"><td><p>Pulmonary arterial hypertension [HP:0002092]</p></td>
<td><p>0/2</p></td>
<td><p>0%</p></td>
<td><p>8/14</p></td>
<td><p>57%</p></td>
<td><p>0.6899307928951144</p></td>
<td><p>0.4666666666666667</p></td>
</tr>
<tr class="row-even"><td><p>Short thumb [HP:0009778]</p></td>
<td><p>8/30</p></td>
<td><p>27%</p></td>
<td><p>25/69</p></td>
<td><p>36%</p></td>
<td><p>0.6899307928951144</p></td>
<td><p>0.4870099714553749</p></td>
</tr>
<tr class="row-odd"><td><p>Absent radius [HP:0003974]</p></td>
<td><p>6/25</p></td>
<td><p>24%</p></td>
<td><p>9/43</p></td>
<td><p>21%</p></td>
<td><p>1.0</p></td>
<td><p>0.7703831604944444</p></td>
</tr>
<tr class="row-even"><td><p>Hypoplasia of the radius [HP:0002984]</p></td>
<td><p>6/14</p></td>
<td><p>43%</p></td>
<td><p>34/75</p></td>
<td><p>45%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Atrial septal defect [HP:0001631]</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>63/65</p></td>
<td><p>97%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Hypoplasia of the ulna [HP:0003022]</p></td>
<td><p>2/10</p></td>
<td><p>20%</p></td>
<td><p>3/17</p></td>
<td><p>18%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Short humerus [HP:0005792]</p></td>
<td><p>4/9</p></td>
<td><p>44%</p></td>
<td><p>8/21</p></td>
<td><p>38%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal ventricular septum morphology [HP:0010438]</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td><p>42/42</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiac ventricle morphology [HP:0001713]</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td><p>43/43</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal heart morphology [HP:0001627]</p></td>
<td><p>30/30</p></td>
<td><p>100%</p></td>
<td><p>89/89</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiovascular system morphology [HP:0030680]</p></td>
<td><p>30/30</p></td>
<td><p>100%</p></td>
<td><p>92/92</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the cardiovascular system [HP:0001626]</p></td>
<td><p>32/32</p></td>
<td><p>100%</p></td>
<td><p>94/94</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Phenotypic abnormality [HP:0000118]</p></td>
<td><p>38/38</p></td>
<td><p>100%</p></td>
<td><p>114/114</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>All [HP:0000001]</p></td>
<td><p>38/38</p></td>
<td><p>100%</p></td>
<td><p>114/114</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiac septum morphology [HP:0001671]</p></td>
<td><p>28/28</p></td>
<td><p>100%</p></td>
<td><p>89/89</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Forearm undergrowth [HP:0009821]</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td><p>35/35</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal upper limb bone morphology [HP:0040070]</p></td>
<td><p>14/14</p></td>
<td><p>100%</p></td>
<td><p>50/50</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the upper limb [HP:0002817]</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td><p>102/102</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of limbs [HP:0040064]</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td><p>102/102</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal limb bone morphology [HP:0002813]</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td><p>92/92</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of limb bone [HP:0040068]</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td><p>92/92</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the skeletal system [HP:0000924]</p></td>
<td><p>35/35</p></td>
<td><p>100%</p></td>
<td><p>103/103</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the musculoskeletal system [HP:0033127]</p></td>
<td><p>35/35</p></td>
<td><p>100%</p></td>
<td><p>104/104</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal appendicular skeleton morphology [HP:0011844]</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td><p>93/93</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal skeletal morphology [HP:0011842]</p></td>
<td><p>35/35</p></td>
<td><p>100%</p></td>
<td><p>103/103</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Upper limb undergrowth [HP:0009824]</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td><p>38/38</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Limb undergrowth [HP:0009826]</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td><p>38/38</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia of the extremities [HP:0009815]</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td><p>78/78</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia involving the skeleton [HP:0009115]</p></td>
<td><p>23/23</p></td>
<td><p>100%</p></td>
<td><p>80/80</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving bones of the upper limbs [HP:0006496]</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td><p>78/78</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia involving bones of the extremities [HP:0045060]</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td><p>78/78</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving forearm bones [HP:0006503]</p></td>
<td><p>12/12</p></td>
<td><p>100%</p></td>
<td><p>43/43</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal forearm bone morphology [HP:0040072]</p></td>
<td><p>14/14</p></td>
<td><p>100%</p></td>
<td><p>43/43</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal forearm morphology [HP:0002973]</p></td>
<td><p>14/14</p></td>
<td><p>100%</p></td>
<td><p>43/43</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Short long bone [HP:0003026]</p></td>
<td><p>9/9</p></td>
<td><p>100%</p></td>
<td><p>41/41</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal long bone morphology [HP:0011314]</p></td>
<td><p>13/13</p></td>
<td><p>100%</p></td>
<td><p>50/50</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the radius [HP:0006501]</p></td>
<td><p>11/11</p></td>
<td><p>100%</p></td>
<td><p>43/43</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal morphology of the radius [HP:0002818]</p></td>
<td><p>13/13</p></td>
<td><p>100%</p></td>
<td><p>43/43</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>1-2 finger syndactyly [HP:0010704]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Finger syndactyly [HP:0006101]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Syndactyly [HP:0001159]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal digit morphology [HP:0011297]</p></td>
<td><p>33/33</p></td>
<td><p>100%</p></td>
<td><p>67/67</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal atrial septum morphology [HP:0011994]</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>64/64</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal cardiac atrium morphology [HP:0005120]</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>64/64</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Perimembranous ventricular septal defect [HP:0011682]</p></td>
<td><p>3/25</p></td>
<td><p>12%</p></td>
<td><p>6/84</p></td>
<td><p>7%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the thumb [HP:0009601]</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td><p>40/40</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of fingers [HP:0006265]</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td><p>44/44</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving bones of the hand [HP:0005927]</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td><p>44/44</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal hand morphology [HP:0005922]</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>75/75</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the hand [HP:0001155]</p></td>
<td><p>31/31</p></td>
<td><p>100%</p></td>
<td><p>88/88</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal finger morphology [HP:0001167]</p></td>
<td><p>31/31</p></td>
<td><p>100%</p></td>
<td><p>64/64</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal thumb morphology [HP:0001172]</p></td>
<td><p>31/31</p></td>
<td><p>100%</p></td>
<td><p>58/58</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Finger aplasia [HP:0009380]</p></td>
<td><p>14/14</p></td>
<td><p>100%</p></td>
<td><p>23/23</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia involving forearm bones [HP:0009822]</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td><p>9/9</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Absent forearm bone [HP:0003953]</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td><p>9/9</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the ulna [HP:0006495]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal morphology of ulna [HP:0040071]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal carpal morphology [HP:0001191]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>30/32</p></td>
<td><p>94%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the wrist [HP:0003019]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>30/30</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of upper limb joint [HP:0009810]</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td><p>32/32</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal joint morphology [HP:0001367]</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td><p>33/33</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of cardiovascular system electrophysiology [HP:0030956]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>18/18</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiovascular system physiology [HP:0011025]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>30/30</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of thumb phalanx [HP:0009602]</p></td>
<td><p>13/13</p></td>
<td><p>100%</p></td>
<td><p>26/26</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Upper limb phocomelia [HP:0009813]</p></td>
<td><p>2/37</p></td>
<td><p>5%</p></td>
<td><p>8/116</p></td>
<td><p>7%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Phocomelia [HP:0009829]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Short finger [HP:0009381]</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td><p>27/27</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short digit [HP:0011927]</p></td>
<td><p>10/10</p></td>
<td><p>100%</p></td>
<td><p>28/28</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Pre-capillary pulmonary hypertension [HP:0033578]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Elevated pulmonary artery pressure [HP:0004890]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of pulmonary circulation [HP:0030875]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal vascular physiology [HP:0030163]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the vasculature [HP:0002597]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>17/17</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal respiratory system physiology [HP:0002795]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the respiratory system [HP:0002086]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Congenital malformation of the great arteries [HP:0011603]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal morphology of the great vessels [HP:0030962]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>10/10</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal blood vessel morphology [HP:0033353]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>11/11</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal vascular morphology [HP:0025015]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>11/11</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Patent foramen ovale [HP:0001655]</p></td>
<td><p>0/36</p></td>
<td><p>0%</p></td>
<td><p>4/69</p></td>
<td><p>6%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Synostosis of joints [HP:0100240]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of joint mobility [HP:0011729]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal joint physiology [HP:0034430]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal musculoskeletal physiology [HP:0011843]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the vertebral column [HP:0000925]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal axial skeleton morphology [HP:0009121]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>9/9</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Complete atrioventricular canal defect [HP:0001674]</p></td>
<td><p>3/36</p></td>
<td><p>8%</p></td>
<td><p>6/67</p></td>
<td><p>9%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Atrioventricular canal defect [HP:0006695]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Hypoplasia of deltoid muscle [HP:0030241]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Shoulder muscle hypoplasia [HP:0008952]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Hypoplasia of the musculature [HP:0009004]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia involving the skeletal musculature [HP:0001460]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of muscle size [HP:0030236]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal skeletal muscle morphology [HP:0011805]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the musculature [HP:0003011]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia involving the shoulder musculature [HP:0001464]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia involving the musculature of the upper limbs [HP:0001467]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia involving the musculature of the extremities [HP:0009128]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the musculature of the limbs [HP:0009127]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the musculature of the upper limbs [HP:0001446]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the shoulder girdle musculature [HP:0001435]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Bowed forearm bones [HP:0003956]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Bowing of the arm [HP:0006488]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Bowing of the long bones [HP:0006487]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal diaphysis morphology [HP:0000940]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Micrognathia [HP:0000347]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the mandible [HP:0009118]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia involving bones of the skull [HP:0009116]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia affecting bones of the axial skeleton [HP:0009122]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal skull morphology [HP:0000929]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the head [HP:0000234]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of head or neck [HP:0000152]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal mandible morphology [HP:0000277]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal jaw morphology [HP:0030791]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal facial skeleton morphology [HP:0011821]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal oral cavity morphology [HP:0000163]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal oral morphology [HP:0031816]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the mouth [HP:0000153]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the face [HP:0000271]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia involving bones of the thorax [HP:0006711]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal thorax morphology [HP:0000765]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Persistent left superior vena cava [HP:0005301]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/39</p></td>
<td><p>10%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal superior vena cava morphology [HP:0025575]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal vena cava morphology [HP:0005345]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal venous morphology [HP:0002624]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia of the humerus [HP:0006507]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the humerus [HP:0003063]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the upper arm [HP:0001454]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Right atrial enlargement [HP:0030718]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal right atrium morphology [HP:0025580]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Atrial septal dilatation [HP:0011995]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Left ventricular noncompaction cardiomyopathy [HP:0011664]</p></td>
<td><p>1/5</p></td>
<td><p>20%</p></td>
<td><p>1/7</p></td>
<td><p>14%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Noncompaction cardiomyopathy [HP:0012817]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Cardiomyopathy [HP:0001638]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal myocardium morphology [HP:0001637]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Hypoplastic scapulae [HP:0000882]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the scapulae [HP:0006713]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal scapula morphology [HP:0000782]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal toe morphology [HP:0001780]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal foot morphology [HP:0001760]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the lower limb [HP:0002814]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal toe phalanx morphology [HP:0010161]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the distal phalanges of the toes [HP:0010182]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal lower limb bone morphology [HP:0040069]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia involving bones of the extremities [HP:0009825]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia of the 1st metacarpal [HP:0010035]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia of the proximal phalanges of the hand [HP:0010242]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the proximal phalanges of the hand [HP:0009851]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal proximal phalanx morphology of the hand [HP:0009834]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal finger phalanx morphology [HP:0005918]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>9/9</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the phalanges of the hand [HP:0009767]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia of the phalanges of the hand [HP:0009802]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia involving bones of the upper limbs [HP:0009823]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia of metacarpal bones [HP:0010048]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia involving the metacarpal bones [HP:0005914]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal metacarpal morphology [HP:0005916]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the 1st metacarpal [HP:0010026]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal 1st metacarpal morphology [HP:0010009]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the phalanges of the thumb [HP:0009658]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Partial absence of thumb [HP:0009659]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Limited pronation/supination of forearm [HP:0006394]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Limited elbow movement [HP:0002996]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the elbow [HP:0009811]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Limitation of joint mobility [HP:0001376]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Mitral regurgitation [HP:0001653]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Atrioventricular valve regurgitation [HP:0034376]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal atrioventricular valve physiology [HP:0031650]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal heart valve physiology [HP:0031653]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal mitral valve physiology [HP:0031481]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Tricuspid regurgitation [HP:0005180]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal tricuspid valve physiology [HP:0031651]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Cleft soft palate [HP:0000185]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal soft palate morphology [HP:0100736]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal palate morphology [HP:0000174]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Cleft palate [HP:0000175]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Orofacial cleft [HP:0000202]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Craniofacial cleft [HP:5201015]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Sinus bradycardia [HP:0001688]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal electrophysiology of sinoatrial node origin [HP:0011702]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Arrhythmia [HP:0011675]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Bradycardia [HP:0001662]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Pectus excavatum [HP:0000767]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>3/4</p></td>
<td><p>75%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal sternum morphology [HP:0000766]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Deviation of the thumb [HP:0009603]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Deviation of finger [HP:0004097]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Deviation of the hand or of fingers of the hand [HP:0009484]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal skin morphology [HP:0011121]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the skin [HP:0000951]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the integument [HP:0001574]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Sinus venosus atrial septal defect [HP:0011567]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/2</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>First degree atrioventricular block [HP:0011705]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/23</p></td>
<td><p>4%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Postaxial hand polydactyly [HP:0001162]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/4</p></td>
<td><p>75%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Postaxial polydactyly [HP:0100259]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Polydactyly [HP:0010442]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal 5th finger morphology [HP:0004207]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Hand polydactyly [HP:0001161]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Duplication of phalanx of hand [HP:0009997]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Duplication of hand bones [HP:0004275]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Duplication of bones involving the upper extremities [HP:0009142]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>High palate [HP:0000218]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Short neck [HP:0000470]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal neck morphology [HP:0025668]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the neck [HP:0000464]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the cervical spine [HP:0003319]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Shield chest [HP:0000914]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Enlarged thorax [HP:0100625]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal rib cage morphology [HP:0001547]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Y-shaped metatarsals [HP:0010567]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal metatarsal morphology [HP:0001832]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>11 pairs of ribs [HP:0000878]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Missing ribs [HP:0000921]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the ribs [HP:0006712]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal rib morphology [HP:0000772]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the 2nd finger [HP:0006264]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal 2nd finger morphology [HP:0004100]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Small hypothenar eminence [HP:0010487]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the hypothenar eminence [HP:0010486]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the musculature of the hand [HP:0001421]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Upper extremity joint dislocation [HP:0030310]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Joint dislocation [HP:0001373]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Amelia involving the upper limbs [HP:0009812]</p></td>
<td><p>1/37</p></td>
<td><p>3%</p></td>
<td><p>0/114</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Third degree atrioventricular block [HP:0001709]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Atrioventricular dissociation [HP:0011709]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal shoulder morphology [HP:0003043]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Proximal placement of thumb [HP:0009623]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short middle phalanx of the 5th finger [HP:0004220]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Type A brachydactyly [HP:0009370]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Brachydactyly [HP:0001156]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Short 5th finger [HP:0009237]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the 5th finger [HP:0006262]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the middle phalanx of the 5th finger [HP:0009161]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the phalanges of the 5th finger [HP:0009376]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal 5th finger phalanx morphology [HP:0004213]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the middle phalanx of the 5th finger [HP:0004219]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Short middle phalanx of finger [HP:0005819]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the middle phalanges of the hand [HP:0009843]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal middle phalanx morphology of the hand [HP:0009833]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short phalanx of finger [HP:0009803]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Common atrium [HP:0011565]</p></td>
<td><p>0/38</p></td>
<td><p>0%</p></td>
<td><p>1/115</p></td>
<td><p>1%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Unroofed coronary sinus [HP:0031297]</p></td>
<td><p>0/38</p></td>
<td><p>0%</p></td>
<td><p>1/117</p></td>
<td><p>1%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Clinodactyly of the 5th finger [HP:0004209]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Finger clinodactyly [HP:0040019]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Clinodactyly [HP:0030084]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Deviation of the 5th finger [HP:0009179]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Mitral valve prolapse [HP:0001634]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal mitral valve morphology [HP:0001633]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal atrioventricular valve morphology [HP:0006705]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal heart valve morphology [HP:0001654]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Short 1st metacarpal [HP:0010034]</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td><p>1/45</p></td>
<td><p>2%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short phalanx of the thumb [HP:0009660]</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td><p>1/45</p></td>
<td><p>2%</p></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</section>
</section>
</section>
</section>
<section id="phenotype-score">
<span id="phenotype-score-stats"></span><h2>Phenotype score<a class="headerlink" href="#phenotype-score" title="Link to this heading"></a></h2>
<section id="mann-whitney-u-test">
<span id="id1"></span><h3>Mann-Whitney U Test<a class="headerlink" href="#mann-whitney-u-test" title="Link to this heading"></a></h3>
<p>We may want to compare the total number of occurences of a specific set of phenotypic features between two different genotype groups.
For instance, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/29330883/">Jordan et al (2018)</a> found that the total number of structural defects
of the brain, eye, heart, and kidney and sensorineural hearing loss seen in individuals with point mutations in the Atrophin-1 domain of the RERE gene
is significantly higher than expected based on the number of similar defects seen in individuals with putative loss-of-function variants.
Since there are five potential defects, each individual has a count ranging between 0 and 5.</p>
<p>We perform a Mann-Whitney U Test (or Wilcoxon Rank-Sum Test) to compare the distribution of such counts between genotype groups.
This is a non-parametric test that compares the medians of the two groups to determine if they come from the same distribution.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">scipy.stats</span> <span class="k">as</span> <span class="nn">stats</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">group1</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">0</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">group2</span> <span class="o">=</span> <span class="p">[</span><span class="mi">4</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">3</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">r</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">mannwhitneyu</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">group1</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">group2</span><span class="p">,</span> <span class="n">alternative</span> <span class="o">=</span> <span class="s1">&#39;two-sided&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">p_value</span> <span class="o">=</span> <span class="n">r</span><span class="o">.</span><span class="n">pvalue</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">float</span><span class="p">(</span><span class="n">p_value</span><span class="p">)</span>
<span class="go">6.348081479150902e-06</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">p_value</span></code> evaluates to <cite>6.348081479150901e-06</cite>, meaning there is a significant difference between the groups.</p>
<section id="id2">
<h4>Example<a class="headerlink" href="#id2" title="Link to this heading"></a></h4>
<p>Let’s now analyze the subjects reported in <em>Jordan et al</em>.
We will load 19 phenopackets that represent individuals with mutations in <em>RERE</em>
whose signs and symptoms were encoded into HPO terms and deposited into Phenopacket Store.
The phenopackets will be processed into <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>
as described in the <a class="reference internal" href="input-data.html#input-data"><span class="std std-ref">Input data</span></a> section.</p>
<p>Briefly, we will first load 19 phenopackets</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">ppktstore.registry</span> <span class="kn">import</span> <span class="n">configure_phenopacket_registry</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">registry</span> <span class="o">=</span> <span class="n">configure_phenopacket_registry</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">registry</span><span class="o">.</span><span class="n">open_phenopacket_store</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;0.1.18&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">ps</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">phenopackets</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">ps</span><span class="o">.</span><span class="n">iter_cohort_phenopackets</span><span class="p">(</span><span class="s1">&#39;RERE&#39;</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">)</span>
<span class="go">19</span>
</pre></div>
</div>
<p>and load HPO (version <cite>v2024-07-01</cite>)</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>to create a <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.CohortCreator" title="gpsea.preprocessing.CohortCreator"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortCreator</span></code></a></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
</pre></div>
</div>
<p>which we will use to preprocess the cohort</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">load_phenopackets</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">load_phenopackets</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>  
<span class="go">Patients Created: ...</span>
</pre></div>
</div>
<p>resulting in a cohort consisting of 19 individuals</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
<span class="go">19</span>
</pre></div>
</div>
<section id="id3">
<h5>Configure analysis<a class="headerlink" href="#id3" title="Link to this heading"></a></h5>
<p>Now we can set up the analysis of genotype and phenotype.
We will perform the analysis using the <em>RERE</em> transcript selected
as the “main” biologically relevant by the <cite>MANE</cite> consortium.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_001042681.2&#39;</span>
</pre></div>
</div>
<p><strong>Genotype predicate</strong></p>
<p><em>Jordan et al.</em> compare phenotype of individuals harboring point mutations
with the individuals carrying loss of function mutations.
Let’s create a predicate for testing if the variant
is a point mutation or a loss of function mutation.</p>
<p>In this example, the point mutation is a mutation that meets the following conditions:</p>
<ul class="simple">
<li><p>predicted to lead to a missense variant on the <cite>MANE</cite> transcript</p></li>
<li><p>the <a class="reference internal" href="glossary.html#length-of-the-reference-allele"><span class="std std-ref">Length of the reference allele</span></a> is equal to <cite>1</cite></p></li>
<li><p>the <a class="reference internal" href="glossary.html#change-length-of-an-allele"><span class="std std-ref">Change length of an allele</span></a> is equal to <cite>0</cite></p></li>
</ul>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">point_mutation_effects</span> <span class="o">=</span> <span class="p">(</span>
<span class="gp">... </span>    <span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">point_mutation</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">change_length</span><span class="p">(</span><span class="s1">&#39;==&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">)</span> \
<span class="gp">... </span>    <span class="o">&amp;</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">ref_length</span><span class="p">(</span><span class="s1">&#39;==&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span> \
<span class="gp">... </span>    <span class="o">&amp;</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">any</span><span class="p">(</span><span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">effect</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">)</span> <span class="k">for</span> <span class="n">effect</span> <span class="ow">in</span> <span class="n">point_mutation_effects</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">point_mutation</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;((change length == 0 AND ref allele length == 1) AND MISSENSE_VARIANT on NM_001042681.2)&#39;</span>
</pre></div>
</div>
<p>For the loss of function predicate, the following variant effects are considered loss of function:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">lof_effects</span> <span class="o">=</span> <span class="p">(</span>
<span class="gp">... </span>    <span class="n">VariantEffect</span><span class="o">.</span><span class="n">TRANSCRIPT_ABLATION</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">VariantEffect</span><span class="o">.</span><span class="n">FRAMESHIFT_VARIANT</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">VariantEffect</span><span class="o">.</span><span class="n">START_LOST</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">VariantEffect</span><span class="o">.</span><span class="n">STOP_GAINED</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lof_mutation</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">any</span><span class="p">(</span><span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">eff</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">)</span> <span class="k">for</span> <span class="n">eff</span> <span class="ow">in</span> <span class="n">lof_effects</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lof_mutation</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;(TRANSCRIPT_ABLATION on NM_001042681.2 OR FRAMESHIFT_VARIANT on NM_001042681.2 OR START_LOST on NM_001042681.2 OR STOP_GAINED on NM_001042681.2)&#39;</span>
</pre></div>
</div>
<p>The genotype predicate will bin the patient into two groups: a point mutation group or the loss of function group:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">groups_predicate</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span> <span class="o">=</span> <span class="n">groups_predicate</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">predicates</span><span class="o">=</span><span class="p">(</span><span class="n">point_mutation</span><span class="p">,</span> <span class="n">lof_mutation</span><span class="p">),</span>
<span class="gp">... </span>    <span class="n">group_names</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;Point&#39;</span><span class="p">,</span> <span class="s1">&#39;LoF&#39;</span><span class="p">),</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;Genotype group: Point, LoF&#39;</span>
</pre></div>
</div>
<p><strong>Phenotype score</strong></p>
<p>The authors score the individuals based on the number of structural defects
from the following 5 categories:</p>
<ul class="simple">
<li><p>Brain anomalies</p></li>
<li><p>Eye anomalies</p></li>
<li><p>Congenital heart defects</p></li>
<li><p>Renal anomalies</p></li>
<li><p>Sensorineural hearing loss</p></li>
</ul>
<p>and they assign each individual a score based on the number structural defects.
For example, an individual with a congenital heart defect would be assigned a score of <cite>1</cite>,
an individual with congenital heart defect and a renal anomaly would be assigned a score of <cite>2</cite>,
and so on.</p>
<p>We automatize this scoring method by encoding the categories into HPO terms</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">structural_defects</span> <span class="o">=</span> <span class="p">(</span>
<span class="gp">... </span>    <span class="s1">&#39;HP:0012443&#39;</span><span class="p">,</span>  <span class="c1"># Abnormal brain morphology (Brain anomalies)</span>
<span class="gp">... </span>    <span class="s1">&#39;HP:0012372&#39;</span><span class="p">,</span>  <span class="c1"># Abnormal eye morphology (Eye anomalies)</span>
<span class="gp">... </span>    <span class="s1">&#39;HP:0001627&#39;</span><span class="p">,</span>  <span class="c1"># Abnormal heart morphology (Congenital heart defects)</span>
<span class="gp">... </span>    <span class="s1">&#39;HP:0012210&#39;</span><span class="p">,</span>  <span class="c1"># Abnormal renal morphology (Renal anomalies)</span>
<span class="gp">... </span>    <span class="s1">&#39;HP:0000407&#39;</span><span class="p">,</span>  <span class="c1"># Sensorineural hearing impairment (Sensorineural hearing loss)</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>and then test the individuals for presence of at least one HPO term
that corresponds to the structural defect category
(e.g. <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0012443">Abnormal brain morphology</a>)
or that is its descendant
(e.g. <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001272">Cerebellar atrophy</a>).</p>
<p>GPSEA implements this scoring method in <a class="reference internal" href="../apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.CountingPhenotypeScorer" title="gpsea.analysis.pscore.CountingPhenotypeScorer"><code class="xref py py-class docutils literal notranslate"><span class="pre">CountingPhenotypeScorer</span></code></a>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pscore</span> <span class="kn">import</span> <span class="n">CountingPhenotypeScorer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pheno_scorer</span> <span class="o">=</span> <span class="n">CountingPhenotypeScorer</span><span class="o">.</span><span class="n">from_query_curies</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">hpo</span><span class="o">=</span><span class="n">hpo</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">query</span><span class="o">=</span><span class="n">structural_defects</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Statistical test</strong></p>
<p>We will use <a class="reference internal" href="#mann-whitney-u-test"><span class="std std-ref">Mann-Whitney U Test</span></a> as described above.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pscore.stats</span> <span class="kn">import</span> <span class="n">MannWhitneyStatistic</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">score_statistic</span> <span class="o">=</span> <span class="n">MannWhitneyStatistic</span><span class="p">()</span>
</pre></div>
</div>
<p><strong>Final analysis</strong></p>
<p>We will put the final analysis together into <a class="reference internal" href="../apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysis" title="gpsea.analysis.pscore.PhenotypeScoreAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">PhenotypeScoreAnalysis</span></code></a>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pscore</span> <span class="kn">import</span> <span class="n">PhenotypeScoreAnalysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">score_analysis</span> <span class="o">=</span> <span class="n">PhenotypeScoreAnalysis</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">score_statistic</span><span class="o">=</span><span class="n">score_statistic</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="id4">
<h5>Analysis<a class="headerlink" href="#id4" title="Link to this heading"></a></h5>
<p>We execute the analysis by running</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span> <span class="o">=</span> <span class="n">score_analysis</span><span class="o">.</span><span class="n">compare_genotype_vs_phenotype_score</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">gt_predicate</span><span class="o">=</span><span class="n">gt_predicate</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">pheno_scorer</span><span class="o">=</span><span class="n">pheno_scorer</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>The analysis shows a significant difference between the number of structural defects
in individuals with point vs. loss-of-function mutations.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">pval</span>
<span class="go">0.012074957610483744</span>
</pre></div>
</div>
<p>To explore further, we can access a data frame with genotype categories and phenotype counts:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">scores</span> <span class="o">=</span> <span class="n">result</span><span class="o">.</span><span class="n">genotype_phenotype_scores</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">scores</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>  
<span class="go">                                     genotype phenotype</span>
<span class="go">patient_id</span>
<span class="go">Subject 10[PMID_27087320_Subject_10]        1         0</span>
<span class="go">Subject 1[PMID_27087320_Subject_1]          0         4</span>
<span class="go">Subject 1[PMID_29330883_Subject_1]          1         0</span>
<span class="go">Subject 2[PMID_27087320_Subject_2]       None         4</span>
<span class="go">Subject 2[PMID_29330883_Subject_2]          1         1</span>
</pre></div>
</div>
<p>The data frame provides a <cite>genotype</cite> category and a <cite>phenotype</cite> score for each patient.
The genotype category should be interpreted in the context of the genotype predicate:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">gt_id_to_name</span> <span class="o">=</span> <span class="p">{</span><span class="n">c</span><span class="o">.</span><span class="n">category</span><span class="o">.</span><span class="n">cat_id</span><span class="p">:</span> <span class="n">c</span><span class="o">.</span><span class="n">category</span><span class="o">.</span><span class="n">name</span> <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">gt_predicate</span><span class="o">.</span><span class="n">get_categorizations</span><span class="p">()}</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_id_to_name</span>
<span class="go">{0: &#39;Point&#39;, 1: &#39;LoF&#39;}</span>
</pre></div>
</div>
<p>The genotype code <cite>0</cite> is assigned to patients with a point mutation, <cite>1</cite> corresponds to the loss-of-function mutations,
and <cite>None</cite> is assigned to patients who cannot be assigned into any of the groups.</p>
<p>Last, let’s use <a class="reference internal" href="../apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.plot_boxplots" title="gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.plot_boxplots"><code class="xref py py-meth docutils literal notranslate"><span class="pre">plot_boxplots()</span></code></a> method
to show a box plot of the phenotype score distributions:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">4</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">120</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">plot_boxplots</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">gt_predicate</span><span class="o">=</span><span class="n">gt_predicate</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">_</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s2">&quot;y&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">_</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;Phenotype score&quot;</span><span class="p">,</span> <span class="n">ylim</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">structural_defects</span><span class="p">)</span> <span class="o">+</span> <span class="mf">0.5</span><span class="p">)</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;docs/img/rere_phenotype_score_boxplot.png&#39;</span><span class="p">)</span>  
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/rere_phenotype_score_boxplot.png"><img alt="Phenotype score distribution" class="align-center" src="../_images/rere_phenotype_score_boxplot.png" style="width: 600px;" /></a>
<p>We see that the individuals with the point mutations feature structural defects
than the individuals with the loss-of-function mutations.</p>
<p>The box extends from the first quartile (Q1) to the third quartile (Q3) of the data,
with a red line at the median.
The whiskers extend from the box to the farthest data point
lying within 1.5x the inter-quartile range (IQR) from the box.</p>
</section>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="predicates.html" class="btn btn-neutral float-left" title="Predicates" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="mtc.html" class="btn btn-neutral float-right" title="Multiple-testing correction" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>