

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>gpsea.analysis package &mdash; GPSEA 0.9.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/gpsea.css?v=547657ed" />

  
      <script src="../_static/jquery.js?v=5d32c60e"></script>
      <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../_static/documentation_options.js?v=9dc39874"></script>
      <script src="../_static/doctools.js?v=9bcbadda"></script>
      <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../_static/copybutton.js?v=6dbb43f8"></script>
      <script>window.MathJax = {"TeX": {"Macros": {"RR": "{\\mathbb{R}}"}}}</script>
      <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="gpsea.analysis.clf package" href="gpsea.analysis.clf.html" />
    <link rel="prev" title="gpsea package" href="gpsea.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../user-guide/index.html">User guide</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">API reference</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="gpsea.html">gpsea package</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="gpsea.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l4 current"><a class="current reference internal" href="#">gpsea.analysis package</a></li>
<li class="toctree-l4"><a class="reference internal" href="gpsea.model.html">gpsea.model package</a></li>
<li class="toctree-l4"><a class="reference internal" href="gpsea.preprocessing.html">gpsea.preprocessing package</a></li>
<li class="toctree-l4"><a class="reference internal" href="gpsea.view.html">gpsea.view package</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="gpsea.html#submodules">Submodules</a></li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="modules.html">API reference</a></li>
          <li class="breadcrumb-item"><a href="gpsea.html">gpsea package</a></li>
      <li class="breadcrumb-item active">gpsea.analysis package</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/apidocs/gpsea.analysis.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-gpsea.analysis">
<span id="gpsea-analysis-package"></span><h1>gpsea.analysis package<a class="headerlink" href="#module-gpsea.analysis" title="Link to this heading"></a></h1>
<dl class="py exception">
<dt class="sig sig-object py" id="gpsea.analysis.AnalysisException">
<em class="property"><span class="pre">exception</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">AnalysisException</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.13)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.13)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.13)"><span class="pre">Any</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span><span class="n"><span class="pre">args</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_base.html#AnalysisException"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.AnalysisException" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/exceptions.html#Exception" title="(in Python v3.13)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Exception</span></code></a></p>
<p>Reports analysis issues that need user’s attention.</p>
<p>To aid troubleshooting, the exception includes <a class="reference internal" href="#gpsea.analysis.AnalysisException.data" title="gpsea.analysis.AnalysisException.data"><code class="xref py py-attr docutils literal notranslate"><span class="pre">data</span></code></a> -
a mapping with any data that has been computed prior encountering the issues.</p>
<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.AnalysisException.data">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">data</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.13)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.13)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.13)"><span class="pre">Any</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.AnalysisException.data" title="Link to this definition"></a></dt>
<dd><p>Get a mapping with (partial) data to aid troubleshooting.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.AnalysisResult">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">AnalysisResult</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gt_clf</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.GenotypeClassifier" title="gpsea.analysis.clf._api.GenotypeClassifier"><span class="pre">GenotypeClassifier</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">statistic</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#gpsea.analysis.Statistic" title="gpsea.analysis._base.Statistic"><span class="pre">Statistic</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_base.html#AnalysisResult"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.AnalysisResult" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.13)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>AnalysisResult</cite> includes the common parts of results of all analyses.</p>
<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.AnalysisResult.gt_clf">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">gt_clf</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.GenotypeClassifier" title="gpsea.analysis.clf._api.GenotypeClassifier"><span class="pre">GenotypeClassifier</span></a></em><a class="headerlink" href="#gpsea.analysis.AnalysisResult.gt_clf" title="Link to this definition"></a></dt>
<dd><p>Get the genotype classifier used in the survival analysis that produced this result.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.AnalysisResult.statistic">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">statistic</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#gpsea.analysis.Statistic" title="gpsea.analysis._base.Statistic"><span class="pre">Statistic</span></a></em><a class="headerlink" href="#gpsea.analysis.AnalysisResult.statistic" title="Link to this definition"></a></dt>
<dd><p>Get the statistic which computed the (nominal) p values for this result.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.MonoPhenotypeAnalysisResult">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">MonoPhenotypeAnalysisResult</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gt_clf</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.GenotypeClassifier" title="gpsea.analysis.clf._api.GenotypeClassifier"><span class="pre">GenotypeClassifier</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">phenotype</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#gpsea.analysis.Partitioning" title="gpsea.analysis._partition.Partitioning"><span class="pre">Partitioning</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">statistic</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#gpsea.analysis.Statistic" title="gpsea.analysis._base.Statistic"><span class="pre">Statistic</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">statistic_result</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#gpsea.analysis.StatisticResult" title="gpsea.analysis._base.StatisticResult"><span class="pre">StatisticResult</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_base.html#MonoPhenotypeAnalysisResult"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.MonoPhenotypeAnalysisResult" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#gpsea.analysis.AnalysisResult" title="gpsea.analysis._base.AnalysisResult"><code class="xref py py-class docutils literal notranslate"><span class="pre">AnalysisResult</span></code></a></p>
<p><cite>MonoPhenotypeAnalysisResult</cite> reports the outcome of an analysis
that tested a single genotype-phenotype association.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="gpsea.analysis.MonoPhenotypeAnalysisResult.GT_COL">
<span class="sig-name descname"><span class="pre">GT_COL</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'genotype'</span></em><a class="headerlink" href="#gpsea.analysis.MonoPhenotypeAnalysisResult.GT_COL" title="Link to this definition"></a></dt>
<dd><p>Name of column for storing genotype data.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="gpsea.analysis.MonoPhenotypeAnalysisResult.PH_COL">
<span class="sig-name descname"><span class="pre">PH_COL</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'phenotype'</span></em><a class="headerlink" href="#gpsea.analysis.MonoPhenotypeAnalysisResult.PH_COL" title="Link to this definition"></a></dt>
<dd><p>Name of column for storing phenotype data.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="gpsea.analysis.MonoPhenotypeAnalysisResult.DATA_COLUMNS">
<span class="sig-name descname"><span class="pre">DATA_COLUMNS</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">('genotype',</span> <span class="pre">'phenotype')</span></em><a class="headerlink" href="#gpsea.analysis.MonoPhenotypeAnalysisResult.DATA_COLUMNS" title="Link to this definition"></a></dt>
<dd><p>The required columns of the <cite>data</cite> data frame.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MonoPhenotypeAnalysisResult.phenotype">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">phenotype</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#gpsea.analysis.Partitioning" title="gpsea.analysis._partition.Partitioning"><span class="pre">Partitioning</span></a></em><a class="headerlink" href="#gpsea.analysis.MonoPhenotypeAnalysisResult.phenotype" title="Link to this definition"></a></dt>
<dd><p>Get the <a class="reference internal" href="#gpsea.analysis.Partitioning" title="gpsea.analysis.Partitioning"><code class="xref py py-class docutils literal notranslate"><span class="pre">Partitioning</span></code></a> that produced the phenotype.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MonoPhenotypeAnalysisResult.data">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">data</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></em><a class="headerlink" href="#gpsea.analysis.MonoPhenotypeAnalysisResult.data" title="Link to this definition"></a></dt>
<dd><p>Get the data frame with genotype and phenotype values for each tested individual.</p>
<p>The index of the data frame contains the identifiers of the tested individuals,
and the values are stored in <cite>genotype</cite> and <cite>phenotype</cite> columns.</p>
<p>The <cite>genotype</cite> column includes the genotype category ID
(<a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.PatientCategory.cat_id" title="gpsea.analysis.clf.PatientCategory.cat_id"><code class="xref py py-attr docutils literal notranslate"><span class="pre">cat_id</span></code></a>)
or <cite>None</cite> if the individual could not be assigned into a genotype group.
The <cite>phenotype</cite> contains the phenotype values, and the data type depends on the analysis.</p>
<p>Here are some common phenotype data types:</p>
<ul class="simple">
<li><p>a phenotype score computed in <a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysis" title="gpsea.analysis.pscore.PhenotypeScoreAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">PhenotypeScoreAnalysis</span></code></a> is a <cite>float</cite></p></li>
<li><p>survival computed in <a class="reference internal" href="gpsea.analysis.temporal.html#gpsea.analysis.temporal.SurvivalAnalysis" title="gpsea.analysis.temporal.SurvivalAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">SurvivalAnalysis</span></code></a>
is of type <a class="reference internal" href="gpsea.analysis.temporal.html#gpsea.analysis.temporal.Survival" title="gpsea.analysis.temporal.Survival"><code class="xref py py-class docutils literal notranslate"><span class="pre">Survival</span></code></a></p></li>
</ul>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.MonoPhenotypeAnalysisResult.complete_records">
<span class="sig-name descname"><span class="pre">complete_records</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_base.html#MonoPhenotypeAnalysisResult.complete_records"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.MonoPhenotypeAnalysisResult.complete_records" title="Link to this definition"></a></dt>
<dd><p>Get the <a class="reference internal" href="#gpsea.analysis.MonoPhenotypeAnalysisResult.data" title="gpsea.analysis.MonoPhenotypeAnalysisResult.data"><code class="xref py py-attr docutils literal notranslate"><span class="pre">data</span></code></a> rows
where both <cite>genotype</cite> and <cite>phenotype</cite> columns are available (i.e. not <cite>None</cite> or <cite>NaN</cite>).</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.MonoPhenotypeAnalysisResult.statistic_result">
<span class="sig-name descname"><span class="pre">statistic_result</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#gpsea.analysis.StatisticResult" title="gpsea.analysis._base.StatisticResult"><span class="pre">StatisticResult</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_base.html#MonoPhenotypeAnalysisResult.statistic_result"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.MonoPhenotypeAnalysisResult.statistic_result" title="Link to this definition"></a></dt>
<dd><p>Get statistic result with the nominal p value and the associated statistics.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MonoPhenotypeAnalysisResult.pval">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pval</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.13)"><span class="pre">float</span></a></em><a class="headerlink" href="#gpsea.analysis.MonoPhenotypeAnalysisResult.pval" title="Link to this definition"></a></dt>
<dd><p>Get the p value of the test.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">MultiPhenotypeAnalysisResult</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gt_clf</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.GenotypeClassifier" title="gpsea.analysis.clf._api.GenotypeClassifier"><span class="pre">GenotypeClassifier</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">pheno_clfs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.13)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.PhenotypeClassifier" title="gpsea.analysis.clf._api.PhenotypeClassifier"><span class="pre">PhenotypeClassifier</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">statistic</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#gpsea.analysis.Statistic" title="gpsea.analysis._base.Statistic"><span class="pre">Statistic</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_usable</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.13)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">all_counts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">statistic_results</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#gpsea.analysis.StatisticResult" title="gpsea.analysis._base.StatisticResult"><span class="pre">StatisticResult</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.13)"><span class="pre">None</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">corrected_pvals</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.13)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.13)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">mtc_correction</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.13)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.13)"><span class="pre">None</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_base.html#MultiPhenotypeAnalysisResult"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Generic" title="(in Python v3.13)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Generic</span></code></a>[<code class="xref py py-obj docutils literal notranslate"><span class="pre">P</span></code>], <a class="reference internal" href="#gpsea.analysis.AnalysisResult" title="gpsea.analysis._base.AnalysisResult"><code class="xref py py-class docutils literal notranslate"><span class="pre">AnalysisResult</span></code></a></p>
<p><cite>MultiPhenotypeAnalysisResult</cite> reports the outcome of an analysis
that tested the association of genotype with two or more phenotypes.</p>
<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.pheno_clfs">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pheno_clfs</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.PhenotypeClassifier" title="gpsea.analysis.clf._api.PhenotypeClassifier"><span class="pre">PhenotypeClassifier</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.pheno_clfs" title="Link to this definition"></a></dt>
<dd><p>Get the phenotype classifiers used in the analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.phenotypes">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">phenotypes</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.phenotypes" title="Link to this definition"></a></dt>
<dd><p>Get the phenotypes that were tested for association with genotype in the analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.n_usable">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">n_usable</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.13)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.n_usable" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of numbers of patients where the phenotype was assessable,
and are, thus, usable for genotype-phenotype correlation analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.all_counts">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">all_counts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.all_counts" title="Link to this definition"></a></dt>
<dd><p>Get a <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">DataFrame</span></code></a> sequence where each <cite>DataFrame</cite> includes the counts of patients
in genotype and phenotype groups.</p>
<p>An example for a genotype predicate that bins into two categories (<cite>Yes</cite> and <cite>No</cite>) based on presence
of a missense variant in transcript <cite>NM_123456.7</cite>, and phenotype predicate that checks
presence/absence of <cite>HP:0001166</cite> (a phenotype term):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>           <span class="n">Has</span> <span class="n">MISSENSE_VARIANT</span> <span class="ow">in</span> <span class="n">NM_123456</span><span class="mf">.7</span>
           <span class="n">No</span>       <span class="n">Yes</span>
<span class="n">Present</span>
<span class="n">Yes</span>        <span class="mi">1</span>        <span class="mi">13</span>
<span class="n">No</span>         <span class="mi">7</span>        <span class="mi">5</span>
</pre></div>
</div>
<p>The rows correspond to the phenotype categories, and the columns represent the genotype categories.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.statistic_results">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">statistic_results</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#gpsea.analysis.StatisticResult" title="gpsea.analysis._base.StatisticResult"><span class="pre">StatisticResult</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.13)"><span class="pre">None</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.statistic_results" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of <a class="reference internal" href="#gpsea.analysis.StatisticResult" title="gpsea.analysis.StatisticResult"><code class="xref py py-class docutils literal notranslate"><span class="pre">StatisticResult</span></code></a> items with nominal p values and the associated statistic values
for each tested phenotype or <cite>None</cite> for the untested phenotypes.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.pvals">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pvals</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.13)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.pvals" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of nominal p values for each tested phenotype.
The sequence includes a <cite>NaN</cite> value for each phenotype that was <em>not</em> tested.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.corrected_pvals">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">corrected_pvals</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.13)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.13)"><span class="pre">None</span></a></em><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.corrected_pvals" title="Link to this definition"></a></dt>
<dd><p>Get a sequence with p values for each tested phenotype after multiple testing correction
or <cite>None</cite> if the correction was not applied.
The sequence includes a <cite>NaN</cite> value for each phenotype that was <em>not</em> tested.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.n_significant_for_alpha">
<span class="sig-name descname"><span class="pre">n_significant_for_alpha</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.13)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.05</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.13)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.13)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_base.html#MultiPhenotypeAnalysisResult.n_significant_for_alpha"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.n_significant_for_alpha" title="Link to this definition"></a></dt>
<dd><p>Get the count of the corrected p values with the value being less than or equal to <cite>alpha</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>alpha</strong> – a <cite>float</cite> with significance level.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.significant_phenotype_indices">
<span class="sig-name descname"><span class="pre">significant_phenotype_indices</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.13)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pval_kind</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Literal" title="(in Python v3.13)"><span class="pre">Literal</span></a><span class="p"><span class="pre">[</span></span><span class="s"><span class="pre">'corrected'</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="s"><span class="pre">'nominal'</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'corrected'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.13)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.13)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.13)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_base.html#MultiPhenotypeAnalysisResult.significant_phenotype_indices"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.significant_phenotype_indices" title="Link to this definition"></a></dt>
<dd><p>Get the indices of phenotypes that attain significance for provided <cite>alpha</cite>.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.total_tests">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">total_tests</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.13)"><span class="pre">int</span></a></em><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.total_tests" title="Link to this definition"></a></dt>
<dd><p>Get total count of genotype-phenotype associations that were tested in this analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.MultiPhenotypeAnalysisResult.mtc_correction">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mtc_correction</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.13)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.13)"><span class="pre">None</span></a></em><a class="headerlink" href="#gpsea.analysis.MultiPhenotypeAnalysisResult.mtc_correction" title="Link to this definition"></a></dt>
<dd><p>Get name/code of the used multiple testing correction
(e.g. <cite>fdr_bh</cite> for Benjamini-Hochberg) or <cite>None</cite> if no correction was applied.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.Statistic">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">Statistic</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.13)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_base.html#Statistic"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.Statistic" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.13)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>Mixin for classes that are used to compute a nominal p value for a genotype-phenotype association.</p>
<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.Statistic.name">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">name</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.13)"><span class="pre">str</span></a></em><a class="headerlink" href="#gpsea.analysis.Statistic.name" title="Link to this definition"></a></dt>
<dd><p>Get the name of the statistic (e.g. <cite>Fisher Exact Test</cite>, <cite>Logrank test</cite>).</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.StatisticResult">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">StatisticResult</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">statistic</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.13)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.13)"><span class="pre">float</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.13)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">pval</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.13)"><span class="pre">float</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_base.html#StatisticResult"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.StatisticResult" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.13)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>StatisticResult</cite> reports result of a <a class="reference internal" href="#gpsea.analysis.Statistic" title="gpsea.analysis.Statistic"><code class="xref py py-class docutils literal notranslate"><span class="pre">Statistic</span></code></a>.</p>
<p>It includes a statistic (optional) and a corresponding p value.
The p value can be <cite>NaN</cite> if it is impossible to compute for a given dataset.</p>
<p>Raises an <a class="reference external" href="https://docs.python.org/3/library/exceptions.html#AssertionError" title="(in Python v3.13)"><code class="xref py py-class docutils literal notranslate"><span class="pre">AssertionError</span></code></a> for an invalid input.</p>
<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.StatisticResult.statistic">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">statistic</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.13)"><span class="pre">float</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.13)"><span class="pre">None</span></a></em><a class="headerlink" href="#gpsea.analysis.StatisticResult.statistic" title="Link to this definition"></a></dt>
<dd><p>Get a <cite>float</cite> with the test statistic or <cite>None</cite> if not available.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.StatisticResult.pval">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pval</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.13)"><span class="pre">float</span></a></em><a class="headerlink" href="#gpsea.analysis.StatisticResult.pval" title="Link to this definition"></a></dt>
<dd><p>Get a p value (a value or a <cite>NaN</cite>).</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.Partitioning">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">Partitioning</span></span><a class="reference internal" href="../_modules/gpsea/analysis/_partition.html#Partitioning"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.Partitioning" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#gpsea.analysis.Summarizable" title="gpsea.analysis._util.Summarizable"><code class="xref py py-class docutils literal notranslate"><span class="pre">Summarizable</span></code></a></p>
<p><cite>Partitioning</cite> is a superclass of all classes that assign a group,
compute a score or survival for an individual.</p>
<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.Partitioning.name">
<em class="property"><span class="pre">abstract</span><span class="w"> </span><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">name</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.13)"><span class="pre">str</span></a></em><a class="headerlink" href="#gpsea.analysis.Partitioning.name" title="Link to this definition"></a></dt>
<dd><p>Get the name of the partitioning.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.Partitioning.description">
<em class="property"><span class="pre">abstract</span><span class="w"> </span><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">description</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.13)"><span class="pre">str</span></a></em><a class="headerlink" href="#gpsea.analysis.Partitioning.description" title="Link to this definition"></a></dt>
<dd><p>Get a description of the partitioning.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.Partitioning.variable_name">
<em class="property"><span class="pre">abstract</span><span class="w"> </span><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variable_name</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.13)"><span class="pre">str</span></a></em><a class="headerlink" href="#gpsea.analysis.Partitioning.variable_name" title="Link to this definition"></a></dt>
<dd><p>Get a <cite>str</cite> with the name of the variable investigated by the partitioning.</p>
<p>For instance <cite>Sex</cite>, <cite>Allele groups</cite>, <cite>HP:0001250</cite>, <cite>OMIM:256000</cite></p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.Partitioning.summarize">
<span class="sig-name descname"><span class="pre">summarize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">out</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.TextIO" title="(in Python v3.13)"><span class="pre">TextIO</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_partition.html#Partitioning.summarize"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.Partitioning.summarize" title="Link to this definition"></a></dt>
<dd><p>Summarize the item while also considering <cite>other</cite> (default <cite>None</cite>).</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.ContinuousPartitioning">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">ContinuousPartitioning</span></span><a class="reference internal" href="../_modules/gpsea/analysis/_partition.html#ContinuousPartitioning"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.ContinuousPartitioning" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#gpsea.analysis.Partitioning" title="gpsea.analysis._partition.Partitioning"><code class="xref py py-class docutils literal notranslate"><span class="pre">Partitioning</span></code></a></p>
<p><cite>ContinuousPartitioning</cite> computes a score that is a real number.</p>
<p>The class is just a marker class at this time.</p>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.Summarizable">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">Summarizable</span></span><a class="reference internal" href="../_modules/gpsea/analysis/_util.html#Summarizable"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.Summarizable" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.13)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>A mixin for entities that can summarize themselves into a provided IO handle.</p>
<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.Summarizable.summarize">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">summarize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">out</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.TextIO" title="(in Python v3.13)"><span class="pre">TextIO</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_util.html#Summarizable.summarize"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.Summarizable.summarize" title="Link to this definition"></a></dt>
<dd><p>Summarize the item into the provided IO handle.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>out</strong> – an IO handle to write into.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.Summarizable.summary">
<span class="sig-name descname"><span class="pre">summary</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.13)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_util.html#Summarizable.summary"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.Summarizable.summary" title="Link to this definition"></a></dt>
<dd><p>Get the summary.</p>
</dd></dl>

</dd></dl>

<section id="subpackages">
<h2>Subpackages<a class="headerlink" href="#subpackages" title="Link to this heading"></a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="gpsea.analysis.clf.html">gpsea.analysis.clf package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.Classifier"><code class="docutils literal notranslate"><span class="pre">Classifier</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.PatientCategory"><code class="docutils literal notranslate"><span class="pre">PatientCategory</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.Categorization"><code class="docutils literal notranslate"><span class="pre">Categorization</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.GenotypeClassifier"><code class="docutils literal notranslate"><span class="pre">GenotypeClassifier</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.AlleleCounter"><code class="docutils literal notranslate"><span class="pre">AlleleCounter</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.sex_classifier"><code class="docutils literal notranslate"><span class="pre">sex_classifier()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.diagnosis_classifier"><code class="docutils literal notranslate"><span class="pre">diagnosis_classifier()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.monoallelic_classifier"><code class="docutils literal notranslate"><span class="pre">monoallelic_classifier()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.biallelic_classifier"><code class="docutils literal notranslate"><span class="pre">biallelic_classifier()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.allele_count"><code class="docutils literal notranslate"><span class="pre">allele_count()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.PhenotypeClassifier"><code class="docutils literal notranslate"><span class="pre">PhenotypeClassifier</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.PhenotypeCategorization"><code class="docutils literal notranslate"><span class="pre">PhenotypeCategorization</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.HpoClassifier"><code class="docutils literal notranslate"><span class="pre">HpoClassifier</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.DiseasePresenceClassifier"><code class="docutils literal notranslate"><span class="pre">DiseasePresenceClassifier</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.prepare_classifiers_for_terms_of_interest"><code class="docutils literal notranslate"><span class="pre">prepare_classifiers_for_terms_of_interest()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.clf.html#gpsea.analysis.clf.prepare_hpo_terms_of_interest"><code class="docutils literal notranslate"><span class="pre">prepare_hpo_terms_of_interest()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gpsea.analysis.mtc_filter.html">gpsea.analysis.mtc_filter package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcFilter"><code class="docutils literal notranslate"><span class="pre">PhenotypeMtcFilter</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcResult"><code class="docutils literal notranslate"><span class="pre">PhenotypeMtcResult</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcIssue"><code class="docutils literal notranslate"><span class="pre">PhenotypeMtcIssue</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.UseAllTermsMtcFilter"><code class="docutils literal notranslate"><span class="pre">UseAllTermsMtcFilter</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter"><code class="docutils literal notranslate"><span class="pre">SpecifiedTermsMtcFilter</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter"><code class="docutils literal notranslate"><span class="pre">HpoMtcFilter</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gpsea.analysis.pcats.html">gpsea.analysis.pcats package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysis"><code class="docutils literal notranslate"><span class="pre">MultiPhenotypeAnalysis</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysisResult"><code class="docutils literal notranslate"><span class="pre">MultiPhenotypeAnalysisResult</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.DiseaseAnalysis"><code class="docutils literal notranslate"><span class="pre">DiseaseAnalysis</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysis"><code class="docutils literal notranslate"><span class="pre">HpoTermAnalysis</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysisResult"><code class="docutils literal notranslate"><span class="pre">HpoTermAnalysisResult</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.apply_classifiers_on_individuals"><code class="docutils literal notranslate"><span class="pre">apply_classifiers_on_individuals()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.configure_hpo_term_analysis"><code class="docutils literal notranslate"><span class="pre">configure_hpo_term_analysis()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#subpackages">Subpackages</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gpsea.analysis.predicate.html">gpsea.analysis.predicate package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.VariantPredicate"><code class="docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.true"><code class="docutils literal notranslate"><span class="pre">true()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.allof"><code class="docutils literal notranslate"><span class="pre">allof()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.anyof"><code class="docutils literal notranslate"><span class="pre">anyof()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.variant_effect"><code class="docutils literal notranslate"><span class="pre">variant_effect()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.variant_key"><code class="docutils literal notranslate"><span class="pre">variant_key()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.gene"><code class="docutils literal notranslate"><span class="pre">gene()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.transcript"><code class="docutils literal notranslate"><span class="pre">transcript()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.exon"><code class="docutils literal notranslate"><span class="pre">exon()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.protein_region"><code class="docutils literal notranslate"><span class="pre">protein_region()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.is_large_imprecise_sv"><code class="docutils literal notranslate"><span class="pre">is_large_imprecise_sv()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.is_structural_variant"><code class="docutils literal notranslate"><span class="pre">is_structural_variant()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.structural_type"><code class="docutils literal notranslate"><span class="pre">structural_type()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.variant_class"><code class="docutils literal notranslate"><span class="pre">variant_class()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.ref_length"><code class="docutils literal notranslate"><span class="pre">ref_length()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.change_length"><code class="docutils literal notranslate"><span class="pre">change_length()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.is_structural_deletion"><code class="docutils literal notranslate"><span class="pre">is_structural_deletion()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.protein_feature_type"><code class="docutils literal notranslate"><span class="pre">protein_feature_type()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.protein_feature"><code class="docutils literal notranslate"><span class="pre">protein_feature()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gpsea.analysis.pscore.html">gpsea.analysis.pscore package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScorer"><code class="docutils literal notranslate"><span class="pre">PhenotypeScorer</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysis"><code class="docutils literal notranslate"><span class="pre">PhenotypeScoreAnalysis</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysisResult"><code class="docutils literal notranslate"><span class="pre">PhenotypeScoreAnalysisResult</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.CountingPhenotypeScorer"><code class="docutils literal notranslate"><span class="pre">CountingPhenotypeScorer</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.DeVriesPhenotypeScorer"><code class="docutils literal notranslate"><span class="pre">DeVriesPhenotypeScorer</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.MeasurementPhenotypeScorer"><code class="docutils literal notranslate"><span class="pre">MeasurementPhenotypeScorer</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#subpackages">Subpackages</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gpsea.analysis.temporal.html">gpsea.analysis.temporal package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.temporal.html#gpsea.analysis.temporal.Endpoint"><code class="docutils literal notranslate"><span class="pre">Endpoint</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.temporal.html#gpsea.analysis.temporal.SurvivalAnalysis"><code class="docutils literal notranslate"><span class="pre">SurvivalAnalysis</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.temporal.html#gpsea.analysis.temporal.SurvivalAnalysisResult"><code class="docutils literal notranslate"><span class="pre">SurvivalAnalysisResult</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.temporal.html#gpsea.analysis.temporal.Survival"><code class="docutils literal notranslate"><span class="pre">Survival</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.temporal.html#subpackages">Subpackages</a></li>
</ul>
</li>
</ul>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="gpsea.html" class="btn btn-neutral float-left" title="gpsea package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="gpsea.analysis.clf.html" class="btn btn-neutral float-right" title="gpsea.analysis.clf package" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>