<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>gpsea.analysis package &mdash; GPSEA 0.1.1dev documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/gpsea.css?v=547657ed" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=8427294d"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../_static/copybutton.js?v=6dbb43f8"></script>
        <script>window.MathJax = {"TeX": {"Macros": {"RR": "{\\mathbb{R}}"}}}</script>
        <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="gpsea.analysis.mtc_filter package" href="gpsea.analysis.mtc_filter.html" />
    <link rel="prev" title="gpsea package" href="gpsea.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            GPSEA
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../user-guide/index.html">User guide</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">API reference</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="gpsea.html">gpsea package</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="gpsea.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l4 current"><a class="current reference internal" href="#">gpsea.analysis package</a></li>
<li class="toctree-l4"><a class="reference internal" href="gpsea.data.html">gpsea.data package</a></li>
<li class="toctree-l4"><a class="reference internal" href="gpsea.model.html">gpsea.model package</a></li>
<li class="toctree-l4"><a class="reference internal" href="gpsea.preprocessing.html">gpsea.preprocessing package</a></li>
<li class="toctree-l4"><a class="reference internal" href="gpsea.view.html">gpsea.view package</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="gpsea.html#submodules">Submodules</a></li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="modules.html">API reference</a></li>
          <li class="breadcrumb-item"><a href="gpsea.html">gpsea package</a></li>
      <li class="breadcrumb-item active">gpsea.analysis package</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/apidocs/gpsea.analysis.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-gpsea.analysis">
<span id="gpsea-analysis-package"></span><h1>gpsea.analysis package<a class="headerlink" href="#module-gpsea.analysis" title="Link to this heading"></a></h1>
<dl class="py function">
<dt class="sig sig-object py" id="gpsea.analysis.configure_cohort_analysis">
<span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">configure_cohort_analysis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cohort</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.model.html#gpsea.model.Cohort" title="gpsea.model._cohort.Cohort"><span class="pre">Cohort</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/apidocs/hpotk.ontology.html#hpotk.ontology.MinimalOntology" title="(in hpo-toolkit)"><span class="pre">MinimalOntology</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'UNIPROT'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cache_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#gpsea.analysis.CohortAnalysisConfiguration" title="gpsea.analysis._config.CohortAnalysisConfiguration"><span class="pre">CohortAnalysisConfiguration</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#gpsea.analysis.CohortAnalysis" title="gpsea.analysis._api.CohortAnalysis"><span class="pre">CohortAnalysis</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_config.html#configure_cohort_analysis"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.configure_cohort_analysis" title="Link to this definition"></a></dt>
<dd><p>Configure <a class="reference internal" href="#gpsea.analysis.CohortAnalysis" title="gpsea.analysis.CohortAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortAnalysis</span></code></a> for given <cite>cohort</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>cohort</strong> – a <a class="reference internal" href="gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a> to analyze</p></li>
<li><p><strong>hpo</strong> – a <code class="xref py py-class docutils literal notranslate"><span class="pre">MinimalOntology</span></code> with HPO to use in the analysis</p></li>
<li><p><strong>protein_source</strong> – the resource to retrieve protein annotations from if we cannot find the annotations locally.
Choose from <code class="docutils literal notranslate"><span class="pre">{'UNIPROT'}</span></code> (just one fallback implementation is available at the moment).</p></li>
<li><p><strong>config</strong> – an optional <a class="reference internal" href="#gpsea.analysis.CohortAnalysisConfiguration" title="gpsea.analysis.CohortAnalysisConfiguration"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortAnalysisConfiguration</span></code></a> to parameterize the analysis.
The default parameters will be used if <cite>None</cite>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysis">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">CohortAnalysis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/apidocs/hpotk.ontology.html#hpotk.ontology.MinimalOntology" title="(in hpo-toolkit)"><span class="pre">MinimalOntology</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_service</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.preprocessing.html#gpsea.preprocessing.ProteinMetadataService" title="gpsea.preprocessing._api.ProteinMetadataService"><span class="pre">ProteinMetadataService</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#CohortAnalysis"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysis" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>CohortAnalysis</cite> is a driver class for running genotype-phenotype correlation analyses.</p>
<p>The class provides various methods to test genotype-phenotype correlations. All methods wrap results
into <a class="reference internal" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult" title="gpsea.analysis.GenotypePhenotypeAnalysisResult"><code class="xref py py-class docutils literal notranslate"><span class="pre">GenotypePhenotypeAnalysisResult</span></code></a>.</p>
<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysis.compare_hpo_vs_genotype">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_hpo_vs_genotype</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate" title="gpsea.analysis.predicate.genotype._api.VariantPredicate"><span class="pre">VariantPredicate</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult" title="gpsea.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#CohortAnalysis.compare_hpo_vs_genotype"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysis.compare_hpo_vs_genotype" title="Link to this definition"></a></dt>
<dd><p>Bin patients according to a presence of at least one allele that matches <cite>predicate</cite>
and test for genotype-phenotype correlations.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysis.compare_hpo_vs_recessive_genotype">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_hpo_vs_recessive_genotype</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate" title="gpsea.analysis.predicate.genotype._api.VariantPredicate"><span class="pre">VariantPredicate</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult" title="gpsea.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#CohortAnalysis.compare_hpo_vs_recessive_genotype"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysis.compare_hpo_vs_recessive_genotype" title="Link to this definition"></a></dt>
<dd><p>Bin patients according to a presence of zero, one, or two alleles that matche the <cite>predicate</cite>
and test for genotype-phenotype correlations.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysis.compare_hpo_vs_genotype_groups">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_hpo_vs_genotype_groups</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">predicates</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate" title="gpsea.analysis.predicate.genotype._api.VariantPredicate"><span class="pre">VariantPredicate</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">group_names</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult" title="gpsea.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#CohortAnalysis.compare_hpo_vs_genotype_groups"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysis.compare_hpo_vs_genotype_groups" title="Link to this definition"></a></dt>
<dd><p>Bin patients according to a presence of at least one allele that matches
any of the provided <cite>predicates</cite> and test for genotype-phenotype correlations
between the groups.</p>
<p>Note, the patients that pass testing by &gt;1 genotype predicate are <em>OMITTED</em> from the analysis!</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysis.compare_disease_vs_genotype">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_disease_vs_genotype</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate" title="gpsea.analysis.predicate.genotype._api.VariantPredicate"><span class="pre">VariantPredicate</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">disease_ids</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult" title="gpsea.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#CohortAnalysis.compare_disease_vs_genotype"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysis.compare_disease_vs_genotype" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysis.compare_genotype_vs_phenotype_group_count">
<span class="sig-name descname"><span class="pre">compare_genotype_vs_phenotype_group_count</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gt_predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.GenotypePolyPredicate" title="gpsea.analysis.predicate.genotype._api.GenotypePolyPredicate"><span class="pre">GenotypePolyPredicate</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">phenotype_group_terms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">PhenotypeScoreAnalysisResult</span></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#CohortAnalysis.compare_genotype_vs_phenotype_group_count"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysis.compare_genotype_vs_phenotype_group_count" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysis.compare_genotype_vs_phenotype_score">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_genotype_vs_phenotype_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gt_predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.GenotypePolyPredicate" title="gpsea.analysis.predicate.genotype._api.GenotypePolyPredicate"><span class="pre">GenotypePolyPredicate</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">phenotype_scorer</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScorer" title="gpsea.analysis.pscore._api.PhenotypeScorer"><span class="pre">PhenotypeScorer</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">PhenotypeScoreAnalysisResult</span></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#CohortAnalysis.compare_genotype_vs_phenotype_score"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysis.compare_genotype_vs_phenotype_score" title="Link to this definition"></a></dt>
<dd><p>Score the patients with a phenotype scoring method and test for correlation between the genotype group
and the phenotype score.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>gt_predicate</strong> – a genotype predicate for binning the patients along the genotype axis.</p></li>
<li><p><strong>phenotype_scorer</strong> – a callable that computes a phenotype score for a given <cite>Patient</cite>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysis.compare_genotype_vs_cohort_phenotypes">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_genotype_vs_cohort_phenotypes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gt_predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.GenotypePolyPredicate" title="gpsea.analysis.predicate.genotype._api.GenotypePolyPredicate"><span class="pre">GenotypePolyPredicate</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult" title="gpsea.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#CohortAnalysis.compare_genotype_vs_cohort_phenotypes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysis.compare_genotype_vs_cohort_phenotypes" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysis.compare_genotype_vs_phenotypes">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_genotype_vs_phenotypes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gt_predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.GenotypePolyPredicate" title="gpsea.analysis.predicate.genotype._api.GenotypePolyPredicate"><span class="pre">GenotypePolyPredicate</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">pheno_predicates</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate" title="gpsea.analysis.predicate.phenotype._pheno.PhenotypePolyPredicate"><span class="pre">PhenotypePolyPredicate</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#CohortAnalysis.compare_genotype_vs_phenotypes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysis.compare_genotype_vs_phenotypes" title="Link to this definition"></a></dt>
<dd><p>All analysis functions go through this function.</p>
<p>The genotype predicate will partition the individuals into non-overlapping groups
along the genotype axis.
The phenotype predicates represent the phenotypes we want to test.
Less phenotypes may actually be tested thanks to <code class="xref py py-class docutils literal notranslate"><span class="pre">PhenotypeMtcFilter</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>gt_predicate</strong> – a predicate for binning the individuals along the genotype axis</p></li>
<li><p><strong>pheno_predicates</strong> – phenotype predicates for test the individuals along the phenotype axis</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.GenotypePhenotypeAnalysisResult">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">GenotypePhenotypeAnalysisResult</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_usable</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">all_counts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvals</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">corrected_pvals</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">phenotype_categories</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategory" title="gpsea.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geno_predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate" title="gpsea.analysis.predicate._api.PolyPredicate"><span class="pre">PolyPredicate</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">mtc_filter_report</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#gpsea.analysis.HpoMtcReport" title="gpsea.analysis._api.HpoMtcReport"><span class="pre">HpoMtcReport</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#GenotypePhenotypeAnalysisResult"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>GenotypePhenotypeAnalysisResult</cite> summarizes results of genotype-phenotype correlation analysis of a cohort.</p>
<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.GenotypePhenotypeAnalysisResult.n_usable">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">n_usable</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult.n_usable" title="Link to this definition"></a></dt>
<dd><p>Get a mapping from a phenotype <cite>P</cite> (either an HPO term or a disease ID)
to an <cite>int</cite> with the number of patients where the phenotype was assessable,
and are, thus, usable for genotype-phenotype correlation analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.GenotypePhenotypeAnalysisResult.all_counts">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">all_counts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult.all_counts" title="Link to this definition"></a></dt>
<dd><p>Get a mapping from the phenotype item to <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">pandas.DataFrame</span></code></a> with counts of patients
in genotype and phenotype groups.</p>
<p>An example for a genotype predicate that bins into two categories (<cite>Yes</cite> and <cite>No</cite>) based on presence
of a missense variant in transcript <cite>NM_123456.7</cite>, and phenotype predicate that checks
presence/absence of <cite>HP:0001166</cite> (a phenotype term):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>           <span class="n">Has</span> <span class="n">MISSENSE_VARIANT</span> <span class="ow">in</span> <span class="n">NM_123456</span><span class="mf">.7</span>
           <span class="n">No</span>       <span class="n">Yes</span>
<span class="n">Present</span>
<span class="n">Yes</span>        <span class="mi">1</span>        <span class="mi">13</span>
<span class="n">No</span>         <span class="mi">7</span>        <span class="mi">5</span>
</pre></div>
</div>
<p>The rows correspond to the phenotype categories, and the columns represent the genotype categories.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.GenotypePhenotypeAnalysisResult.pvals">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pvals</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a></em><a class="headerlink" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult.pvals" title="Link to this definition"></a></dt>
<dd><p>Get a <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">pandas.Series</span></code></a> with p values for each tested HPO term.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.GenotypePhenotypeAnalysisResult.corrected_pvals">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">corrected_pvals</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult.corrected_pvals" title="Link to this definition"></a></dt>
<dd><p>Get an optional <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">pandas.Series</span></code></a> with p values for each tested HPO term after multiple testing correction.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.GenotypePhenotypeAnalysisResult.phenotype_categories">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">phenotype_categories</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategory" title="gpsea.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult.phenotype_categories" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of phenotype patient categories that can be investigated.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.GenotypePhenotypeAnalysisResult.total_tests">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">total_tests</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult.total_tests" title="Link to this definition"></a></dt>
<dd><p>Get total count of tests that were run for this analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.GenotypePhenotypeAnalysisResult.mtc_filter_report">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mtc_filter_report</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#gpsea.analysis.HpoMtcReport" title="gpsea.analysis._api.HpoMtcReport"><span class="pre">HpoMtcReport</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult.mtc_filter_report" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.GenotypePhenotypeAnalysisResult.summarize">
<span class="sig-name descname"><span class="pre">summarize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/apidocs/hpotk.ontology.html#hpotk.ontology.MinimalOntology" title="(in hpo-toolkit)"><span class="pre">MinimalOntology</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">category</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategory" title="gpsea.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#GenotypePhenotypeAnalysisResult.summarize"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.GenotypePhenotypeAnalysisResult.summarize" title="Link to this definition"></a></dt>
<dd><p>Create a data frame with summary of the genotype phenotype analysis.</p>
<p>The <em>rows</em> of the frame correspond to the analyzed HPO terms.</p>
<p>The columns of the data frame have <cite>Count</cite> and <cite>Percentage</cite> per used genotype predicate.</p>
<p><strong>Example</strong></p>
<p>If we use <code class="xref py py-class docutils literal notranslate"><span class="pre">VariantEffectPredicate</span></code>
which can compare phenotype with and without a missense variant, we will have a data frame
that looks like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>MISSENSE_VARIANT on `NM_1234.5`                       No                  Yes
                                                      Count    Percent    Count    Percent   p value   Corrected p value
Arachnodactyly [HP:0001166]                           1/10         10%    13/16        81%   0.000781  0.020299
Abnormality of the musculature [HP:0003011]           6/6         100%    11/11       100%   1.000000  1.000000
Abnormal nervous system physiology [HP:0012638]       9/9         100%    15/15       100%   1.000000  1.000000
...                                                   ...      ...        ...      ...       ...       ...
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">CohortAnalysisConfiguration</span></span><a class="reference internal" href="../_modules/gpsea/analysis/_config.html#CohortAnalysisConfiguration"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>CohortAnalysisConfiguration</cite> is a value class for storing <a class="reference internal" href="#gpsea.analysis.CohortAnalysis" title="gpsea.analysis.CohortAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortAnalysis</span></code></a>
configuration options.</p>
<p>The class contains the default values upon creation and the configuration option values can be set as properties.</p>
<p>If an invalid value option is passed to the property setter, a warning is logged and the previous value is retained.
Therefore, it is <em>impossible</em> to mis-configure the analysis.</p>
<section id="default-values">
<h2>Default values<a class="headerlink" href="#default-values" title="Link to this heading"></a></h2>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Option</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Default value</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">pval_correction</span></code></p></td>
<td><p><cite>str</cite></p></td>
<td><p><cite>bonferroni</cite></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">min_n_patients_with_term</span></code></p></td>
<td><p><cite>int</cite></p></td>
<td><p><cite>2</cite></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">mtc_alpha</span></code></p></td>
<td><p><cite>float</cite></p></td>
<td><p><cite>0.05</cite></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">include_sv</span></code></p></td>
<td><p><cite>bool</cite></p></td>
<td><p><cite>False</cite></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">mtc_strategy</span></code></p></td>
<td><p><a class="reference internal" href="#gpsea.analysis.MtcStrategy" title="gpsea.analysis.MtcStrategy"><code class="xref py py-class docutils literal notranslate"><span class="pre">MtcStrategy</span></code></a></p></td>
<td><p><a class="reference internal" href="#gpsea.analysis.MtcStrategy.ALL_PHENOTYPE_TERMS" title="gpsea.analysis.MtcStrategy.ALL_PHENOTYPE_TERMS"><code class="xref py py-class docutils literal notranslate"><span class="pre">MtcStrategy.ALL_PHENOTYPE_TERMS</span></code></a></p></td>
</tr>
</tbody>
</table>
<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration.pval_correction">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pval_correction</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration.pval_correction" title="Link to this definition"></a></dt>
<dd><p><cite>bonferroni</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Get method for multiple testing p value correction. Default</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration.min_n_patients_with_term">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">min_n_patients_with_term</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration.min_n_patients_with_term" title="Link to this definition"></a></dt>
<dd><p>Get the minimum number of patients that must be annotated with an HPO term
for including the term in the analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration.mtc_alpha">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mtc_alpha</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></em><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration.mtc_alpha" title="Link to this definition"></a></dt>
<dd><p>The alpha value for multiple testing correction.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration.include_sv">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">include_sv</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></em><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration.include_sv" title="Link to this definition"></a></dt>
<dd><p><cite>True</cite> if we want to include structural variants in the analysis
(i.e. the variants that use symbolic VCF notation).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration.mtc_strategy">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mtc_strategy</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#gpsea.analysis.MtcStrategy" title="gpsea.analysis._config.MtcStrategy"><span class="pre">MtcStrategy</span></a></em><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration.mtc_strategy" title="Link to this definition"></a></dt>
<dd><p>Get the MTC filtering strategy to be used.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration.all_terms_strategy">
<span class="sig-name descname"><span class="pre">all_terms_strategy</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_config.html#CohortAnalysisConfiguration.all_terms_strategy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration.all_terms_strategy" title="Link to this definition"></a></dt>
<dd><p>Test all phenotype terms.</p>
<p>See <a class="reference internal" href="../user-guide/mtc.html#use-all-terms-strategy"><span class="std std-ref">Test all terms</span></a> for more info.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration.hpo_mtc_strategy">
<span class="sig-name descname"><span class="pre">hpo_mtc_strategy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">min_patients_w_hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_config.html#CohortAnalysisConfiguration.hpo_mtc_strategy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration.hpo_mtc_strategy" title="Link to this definition"></a></dt>
<dd><p>Only test the HPO terms that pass all rules of the HPO filter strategy.</p>
<p>See <a class="reference internal" href="../user-guide/mtc.html#hpo-mtc-filter-strategy"><span class="std std-ref">HPO MTC filter strategy</span></a> section for more info on the rules.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>threshold_HPO_observed_frequency</strong> – a float in range <span class="math notranslate nohighlight">\((0, 1]\)</span> to represent
the minimum fraction of patients for an HPO term to be included.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration.specify_terms_strategy">
<span class="sig-name descname"><span class="pre">specify_terms_strategy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">terms_to_test</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_config.html#CohortAnalysisConfiguration.specify_terms_strategy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration.specify_terms_strategy" title="Link to this definition"></a></dt>
<dd><p>Mitigate the MTC burden by only testing the specified HPO terms.</p>
<p>The HPO terms are validated before running the analysis,
to point out invalid CURIE (e.g. <cite>WHATEVER</cite>) values,
or HPO term IDs that are not in the currently used HPO.</p>
<p>Calling this method will clear any previously specified terms.</p>
<p>See <a class="reference internal" href="../user-guide/mtc.html#specify-terms-strategy"><span class="std std-ref">Specify terms strategy</span></a> for more info.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>terms_to_test</strong> – an iterable with CURIEs (e.g. <cite>HP:0001250</cite>)
or <code class="xref py py-class docutils literal notranslate"><span class="pre">hpotk.TermId</span></code> instances representing the terms to test.</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration.terms_to_test">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">terms_to_test</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration.terms_to_test" title="Link to this definition"></a></dt>
<dd><p>Get the ids of the terms to be tested in <cite>specify_terms_strategy</cite>
or <cite>None</cite> if <a class="reference internal" href="#gpsea.analysis.MtcStrategy.SPECIFY_TERMS" title="gpsea.analysis.MtcStrategy.SPECIFY_TERMS"><code class="xref py py-class docutils literal notranslate"><span class="pre">MtcStrategy.SPECIFY_TERMS</span></code></a> will <em>not</em> be used.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.CohortAnalysisConfiguration.min_patients_w_hpo">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">min_patients_w_hpo</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#gpsea.analysis.CohortAnalysisConfiguration.min_patients_w_hpo" title="Link to this definition"></a></dt>
<dd><p>Get the minimum fraction of patients needed to be annotated with an HPO term
to justify being tested or <cite>None</cite> if <a class="reference internal" href="#gpsea.analysis.MtcStrategy.HPO_MTC" title="gpsea.analysis.MtcStrategy.HPO_MTC"><code class="xref py py-class docutils literal notranslate"><span class="pre">MtcStrategy.HPO_MTC</span></code></a> will <em>not</em> be used.</p>
</dd></dl>

</section>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.MtcStrategy">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">MtcStrategy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">value</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">module</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qualname</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boundary</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_config.html#MtcStrategy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.MtcStrategy" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/enum.html#enum.Enum" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Enum</span></code></a></p>
<p>A strategy for mitigating the multiple testing correction (MTC) burden.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="gpsea.analysis.MtcStrategy.ALL_PHENOTYPE_TERMS">
<span class="sig-name descname"><span class="pre">ALL_PHENOTYPE_TERMS</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">0</span></em><a class="headerlink" href="#gpsea.analysis.MtcStrategy.ALL_PHENOTYPE_TERMS" title="Link to this definition"></a></dt>
<dd><p>All phenotype terms (HPO or disease IDs) will be tested.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="gpsea.analysis.MtcStrategy.SPECIFY_TERMS">
<span class="sig-name descname"><span class="pre">SPECIFY_TERMS</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">1</span></em><a class="headerlink" href="#gpsea.analysis.MtcStrategy.SPECIFY_TERMS" title="Link to this definition"></a></dt>
<dd><p>Only the manually provided HPO terms will be tested.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="gpsea.analysis.MtcStrategy.HPO_MTC">
<span class="sig-name descname"><span class="pre">HPO_MTC</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">2</span></em><a class="headerlink" href="#gpsea.analysis.MtcStrategy.HPO_MTC" title="Link to this definition"></a></dt>
<dd><p>Only HPO terms present in at least a certain fraction of patients will be tested.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="gpsea.analysis.HpoMtcReport">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">HpoMtcReport</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filter_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">mtc_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_results_map</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_terms_before_filtering</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/gpsea/analysis/_api.html#HpoMtcReport"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.HpoMtcReport" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>Class to simplify reporting results of multiple testing filtering by HpoMtcFilter subclasses.</p>
<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.HpoMtcReport.filter_method">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">filter_method</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#gpsea.analysis.HpoMtcReport.filter_method" title="Link to this definition"></a></dt>
<dd><p>Returns:
the name of the HpoMtcFilter method used.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.HpoMtcReport.skipped_terms_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">skipped_terms_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#gpsea.analysis.HpoMtcReport.skipped_terms_dict" title="Link to this definition"></a></dt>
<dd><p>Returns:
a mapping with reasons why an HPO term was skipped as keys and counts of the skipped terms as values.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.HpoMtcReport.mtc_method">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mtc_method</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#gpsea.analysis.HpoMtcReport.mtc_method" title="Link to this definition"></a></dt>
<dd><p>Returns:
the name of the multiple testing correction method used (e.g. <cite>bonferroni</cite>).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="gpsea.analysis.HpoMtcReport.n_terms_before_filtering">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">n_terms_before_filtering</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#gpsea.analysis.HpoMtcReport.n_terms_before_filtering" title="Link to this definition"></a></dt>
<dd><p>Get the number of terms before filtering.</p>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="gpsea.analysis.apply_predicates_on_patients">
<span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">apply_predicates_on_patients</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patients</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="gpsea.model.html#gpsea.model.Patient" title="gpsea.model._cohort.Patient"><span class="pre">Patient</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pheno_predicates</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate" title="gpsea.analysis.predicate.phenotype._pheno.PhenotypePolyPredicate"><span class="pre">PhenotypePolyPredicate</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gt_predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.GenotypePolyPredicate" title="gpsea.analysis.predicate.genotype._api.GenotypePolyPredicate"><span class="pre">GenotypePolyPredicate</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.12)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Collection" title="(in Python v3.12)"><span class="pre">Collection</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategory" title="gpsea.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_gp_analysis.html#apply_predicates_on_patients"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.apply_predicates_on_patients" title="Link to this definition"></a></dt>
<dd><p>Apply the phenotype predicates <cite>pheno_predicates</cite> and the genotype predicate <cite>gt_predicate</cite>
to bin the <cite>patients</cite> into categories.</p>
<p>Note, it may not be possible to bin <em>all</em> patients with a genotype/phenotype pair,
since a predicate is allowed to return <cite>None</cite> (e.g. if it bins the patient into MISSENSE or NONSENSE groups
but the patient has no MISSENSE or NONSENSE variants). If this happens, the patient will not be “usable”
for the phenotype <cite>P</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>patients</strong> – an iterable with the patients to bin into categories</p></li>
<li><p><strong>pheno_predicates</strong> – an iterable with the phenotype predicates to apply</p></li>
<li><p><strong>gt_predicate</strong> – a genotype predicate to apply</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p>a collection of unique <code class="xref py py-class docutils literal notranslate"><span class="pre">PatientCategory</span></code> items that the patients were binned into</p></li>
<li><p>a mapping from a phenotype <code class="xref py py-class docutils literal notranslate"><span class="pre">P</span></code> (e.g. an HPO term or a disease)
to an <cite>int</cite> with count of patients that could be binned according to the phenotype <cite>P</cite></p></li>
<li><p>a mapping from phenotype <code class="xref py py-class docutils literal notranslate"><span class="pre">P</span></code> to a data frame with counts of patients
in i-th phenotype category and j-th genotype category where i and j are rows and columns of the data frame</p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>a tuple with 3 items</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="gpsea.analysis.configure_default_protein_metadata_service">
<span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">configure_default_protein_metadata_service</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'UNIPROT'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cache_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="gpsea.preprocessing.html#gpsea.preprocessing.ProteinMetadataService" title="gpsea.preprocessing._api.ProteinMetadataService"><span class="pre">ProteinMetadataService</span></a></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_config.html#configure_default_protein_metadata_service"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.configure_default_protein_metadata_service" title="Link to this definition"></a></dt>
<dd><p>Create default protein metadata service that will cache the protein metadata
in current working directory under <cite>.gpsea_cache/protein_cache</cite>
and reach out to UNIPROT REST API if a cache entry is missing.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="gpsea.analysis.prepare_hpo_terms_of_interest">
<span class="sig-prename descclassname"><span class="pre">gpsea.analysis.</span></span><span class="sig-name descname"><span class="pre">prepare_hpo_terms_of_interest</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/apidocs/hpotk.graph.html#hpotk.graph.GraphAware" title="(in hpo-toolkit)"><span class="pre">GraphAware</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">patients</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="gpsea.model.html#gpsea.model.Patient" title="gpsea.model._cohort.Patient"><span class="pre">Patient</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_n_of_patients_with_term</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Collection" title="(in Python v3.12)"><span class="pre">Collection</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/gpsea/analysis/_util.html#prepare_hpo_terms_of_interest"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#gpsea.analysis.prepare_hpo_terms_of_interest" title="Link to this definition"></a></dt>
<dd><p>Prepare a collection of HPO terms to test.</p>
<p>This includes the direct HPO patient annotations
as well as the ancestors of the present terms and the descendants of the excluded terms.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>hpo</strong> – an entity with an HPO graph (e.g. <code class="xref py py-class docutils literal notranslate"><span class="pre">hpotk.MinimalOntology</span></code>).</p></li>
<li><p><strong>patients</strong> – an iterable with patients.</p></li>
<li><p><strong>min_n_of_patients_with_term</strong> – the minimum number of patients that must feature an HPO term
(either directly or indirectly) for the term to be included in the analysis.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<section id="subpackages">
<h2>Subpackages<a class="headerlink" href="#subpackages" title="Link to this heading"></a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="gpsea.analysis.mtc_filter.html">gpsea.analysis.mtc_filter package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcFilter"><code class="docutils literal notranslate"><span class="pre">PhenotypeMtcFilter</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcResult"><code class="docutils literal notranslate"><span class="pre">PhenotypeMtcResult</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.UseAllTermsMtcFilter"><code class="docutils literal notranslate"><span class="pre">UseAllTermsMtcFilter</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter"><code class="docutils literal notranslate"><span class="pre">SpecifiedTermsMtcFilter</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter"><code class="docutils literal notranslate"><span class="pre">HpoMtcFilter</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gpsea.analysis.pcats.html">gpsea.analysis.pcats package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysis"><code class="docutils literal notranslate"><span class="pre">MultiPhenotypeAnalysis</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysisResult"><code class="docutils literal notranslate"><span class="pre">MultiPhenotypeAnalysisResult</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.DiseaseAnalysis"><code class="docutils literal notranslate"><span class="pre">DiseaseAnalysis</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysis"><code class="docutils literal notranslate"><span class="pre">HpoTermAnalysis</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysisResult"><code class="docutils literal notranslate"><span class="pre">HpoTermAnalysisResult</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pcats.html#subpackages">Subpackages</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gpsea.analysis.predicate.html">gpsea.analysis.predicate package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategory"><code class="docutils literal notranslate"><span class="pre">PatientCategory</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategories"><code class="docutils literal notranslate"><span class="pre">PatientCategories</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.Categorization"><code class="docutils literal notranslate"><span class="pre">Categorization</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate"><code class="docutils literal notranslate"><span class="pre">PolyPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.predicate.html#subpackages">Subpackages</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gpsea.analysis.pscore.html">gpsea.analysis.pscore package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScorer"><code class="docutils literal notranslate"><span class="pre">PhenotypeScorer</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysis"><code class="docutils literal notranslate"><span class="pre">PhenotypeScoreAnalysis</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysisResult"><code class="docutils literal notranslate"><span class="pre">PhenotypeScoreAnalysisResult</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#gpsea.analysis.pscore.CountingPhenotypeScorer"><code class="docutils literal notranslate"><span class="pre">CountingPhenotypeScorer</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="gpsea.analysis.pscore.html#subpackages">Subpackages</a></li>
</ul>
</li>
</ul>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="gpsea.html" class="btn btn-neutral float-left" title="gpsea package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="gpsea.analysis.mtc_filter.html" class="btn btn-neutral float-right" title="gpsea.analysis.mtc_filter package" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>