<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../../">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>gpsea.model._variant &mdash; GPSEA 0.4.2.dev0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../../../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../../_static/gpsea.css?v=547657ed" />

  
  <!--[if lt IE 9]>
    <script src="../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../../_static/documentation_options.js?v=272485b7"></script>
        <script src="../../../_static/doctools.js?v=9a2dae69"></script>
        <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../../../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../index.html" class="icon icon-home">
            GPSEA
          </a>
              <div class="version">
                0.4
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Module code</a></li>
      <li class="breadcrumb-item active">gpsea.model._variant</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <h1>Source code for gpsea.model._variant</h1><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">abc</span>
<span class="kn">import</span> <span class="nn">enum</span>
<span class="kn">import</span> <span class="nn">typing</span>

<span class="kn">import</span> <span class="nn">hpotk</span>

<span class="kn">from</span> <span class="nn">.genome</span> <span class="kn">import</span> <span class="n">Region</span><span class="p">,</span> <span class="n">GenomicRegion</span><span class="p">,</span> <span class="n">Contig</span><span class="p">,</span> <span class="n">Strand</span>
<span class="kn">from</span> <span class="nn">._gt</span> <span class="kn">import</span> <span class="n">Genotyped</span><span class="p">,</span> <span class="n">Genotypes</span>
<span class="kn">from</span> <span class="nn">._variant_effects</span> <span class="kn">import</span> <span class="n">VariantEffect</span>


<div class="viewcode-block" id="TranscriptInfoAware">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.TranscriptInfoAware">[docs]</a>
<span class="k">class</span> <span class="nc">TranscriptInfoAware</span><span class="p">(</span><span class="n">metaclass</span><span class="o">=</span><span class="n">abc</span><span class="o">.</span><span class="n">ABCMeta</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    The implementors know about basic gene/transcript identifiers.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="nd">@property</span>
    <span class="nd">@abc</span><span class="o">.</span><span class="n">abstractmethod</span>
    <span class="k">def</span> <span class="nf">gene_id</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Returns:</span>
<span class="sd">            string: The gene symbol (e.g. SURF1)</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">pass</span>

    <span class="nd">@property</span>
    <span class="nd">@abc</span><span class="o">.</span><span class="n">abstractmethod</span>
    <span class="k">def</span> <span class="nf">transcript_id</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Returns:</span>
<span class="sd">            string: The transcript RefSeq identifier (e.g. NM_123456.7)</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">pass</span></div>



<div class="viewcode-block" id="TranscriptAnnotation">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation">[docs]</a>
<span class="k">class</span> <span class="nc">TranscriptAnnotation</span><span class="p">(</span><span class="n">TranscriptInfoAware</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    `TranscriptAnnotation` represent a result of the functional annotation of a variant</span>
<span class="sd">    with respect to single transcript of a gene.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">gene_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
        <span class="n">tx_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
        <span class="n">hgvs_cdna</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
        <span class="n">is_preferred</span><span class="p">:</span> <span class="nb">bool</span><span class="p">,</span>
        <span class="n">variant_effects</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Iterable</span><span class="p">[</span><span class="n">VariantEffect</span><span class="p">],</span>
        <span class="n">affected_exons</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">typing</span><span class="o">.</span><span class="n">Iterable</span><span class="p">[</span><span class="nb">int</span><span class="p">]],</span>
        <span class="n">protein_id</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
        <span class="n">hgvsp</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
        <span class="n">protein_effect_coordinates</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">Region</span><span class="p">],</span>
    <span class="p">):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">gene_id</span><span class="p">,</span> <span class="nb">str</span><span class="p">,</span> <span class="s2">&quot;gene_id&quot;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_tx_id</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">tx_id</span><span class="p">,</span> <span class="nb">str</span><span class="p">,</span> <span class="s2">&quot;tx_id&quot;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_hgvs_cdna</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_optional_instance</span><span class="p">(</span>
            <span class="n">hgvs_cdna</span><span class="p">,</span> <span class="nb">str</span><span class="p">,</span> <span class="s2">&quot;hgvs_cdna&quot;</span>
        <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_is_preferred</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span>
            <span class="n">is_preferred</span><span class="p">,</span> <span class="nb">bool</span><span class="p">,</span> <span class="s2">&quot;is_preferred&quot;</span>
        <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_variant_effects</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">variant_effects</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">affected_exons</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_affected_exons</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">affected_exons</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_affected_exons</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="k">if</span> <span class="n">protein_id</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_protein_id</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span>
                <span class="n">protein_id</span><span class="p">,</span> <span class="nb">str</span><span class="p">,</span> <span class="s2">&quot;protein_id&quot;</span>
            <span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_protein_id</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="k">if</span> <span class="n">hgvsp</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_hgvsp</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">hgvsp</span><span class="p">,</span> <span class="nb">str</span><span class="p">,</span> <span class="s2">&quot;hgvsp&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_hgvsp</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_protein_effect_location</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_optional_instance</span><span class="p">(</span>
            <span class="n">protein_effect_coordinates</span><span class="p">,</span>
            <span class="n">Region</span><span class="p">,</span>
            <span class="s2">&quot;protein_effect_coordinates&quot;</span><span class="p">,</span>
        <span class="p">)</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">gene_id</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the HGVS symbol of the affected gene (e.g. *SURF1*).</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">transcript_id</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the RefSeq identifier of the transcript (e.g. `NM_123456.7`).</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_tx_id</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">is_preferred</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Return `True` if the transcript is the preferred transcript of a gene,</span>
<span class="sd">        such as MANE transcript, canonical Ensembl transcript.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_is_preferred</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">hgvs_cdna</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the HGVS description of the sequence variant (e.g. ``NM_123456.7:c.9876G&gt;T``)</span>
<span class="sd">        or `None` if not available.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hgvs_cdna</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">variant_effects</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">VariantEffect</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get a sequence of the predicted functional variant effects.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_variant_effects</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">overlapping_exons</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="nb">int</span><span class="p">]]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get a sequence of 1-based exon indices (the index of the 1st exon is `1`)</span>
<span class="sd">        that overlap with the variant.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_affected_exons</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">protein_id</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the ID of the protein encoded by the :attr:`transcript_id`</span>
<span class="sd">        or `None` if the transcript is not protein-coding.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_id</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">hgvsp</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the HGVS description of the protein sequence variant (e.g. ``NP_001027559.1:p.Gln421Ter``)</span>
<span class="sd">        or `None` if not available.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hgvsp</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">protein_effect_location</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">Region</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the :class:`~gpsea.model.genome.Region` with start and end amino-acid coordinates</span>
<span class="sd">        affected by the variant.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_effect_location</span>

    <span class="k">def</span> <span class="fm">__str__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="sa">f</span><span class="s2">&quot;TranscriptAnnotation(gene_id:</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span><span class="si">}</span><span class="s2">,&quot;</span>
            <span class="sa">f</span><span class="s2">&quot;transcript_id:</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_tx_id</span><span class="si">}</span><span class="s2">,&quot;</span>
            <span class="sa">f</span><span class="s2">&quot;hgvs_cdna:</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_hgvs_cdna</span><span class="si">}</span><span class="s2">,&quot;</span>
            <span class="sa">f</span><span class="s2">&quot;is_preferred:</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_is_preferred</span><span class="si">}</span><span class="s2">,&quot;</span>
            <span class="sa">f</span><span class="s2">&quot;variant_effects:</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_variant_effects</span><span class="si">}</span><span class="s2">,&quot;</span>
            <span class="sa">f</span><span class="s2">&quot;overlapping_exons:</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_affected_exons</span><span class="si">}</span><span class="s2">,&quot;</span>
            <span class="sa">f</span><span class="s2">&quot;protein_id:</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_protein_id</span><span class="si">}</span><span class="s2">,&quot;</span>
            <span class="sa">f</span><span class="s2">&quot;hgvsp:</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_hgvsp</span><span class="si">}</span><span class="s2">,&quot;</span>
            <span class="sa">f</span><span class="s2">&quot;protein_effect_location:</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_protein_effect_location</span><span class="si">}</span><span class="s2">)&quot;</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__eq__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">other</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="nb">isinstance</span><span class="p">(</span><span class="n">other</span><span class="p">,</span> <span class="n">TranscriptAnnotation</span><span class="p">)</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_gene_id</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_tx_id</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_tx_id</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hgvs_cdna</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_hgvs_cdna</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_is_preferred</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_is_preferred</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_variant_effects</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_variant_effects</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_affected_exons</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_affected_exons</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_id</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_protein_id</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hgvsp</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_hgvsp</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_effect_location</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_protein_effect_location</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__repr__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span>

    <span class="k">def</span> <span class="fm">__hash__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">hash</span><span class="p">(</span>
            <span class="p">(</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_tx_id</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_hgvs_cdna</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_is_preferred</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_variant_effects</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_affected_exons</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_protein_id</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_hgvsp</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_protein_effect_location</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="p">)</span></div>



<div class="viewcode-block" id="VariantClass">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.VariantClass">[docs]</a>
<span class="k">class</span> <span class="nc">VariantClass</span><span class="p">(</span><span class="n">enum</span><span class="o">.</span><span class="n">Enum</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    `VariantClass` represents a high-level variant category</span>
<span class="sd">    which mostly corresponds to the structural variant categories</span>
<span class="sd">    of the Variant Call Format specification,</span>
<span class="sd">    but includes type for single nucleotide variants (SNV) and multi-nucleotide variant (MNV).</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">DEL</span> <span class="o">=</span> <span class="mi">0</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    A deletion - a variant with a net loss of sequence from the alternative allele</span>
<span class="sd">    regardless of its size.</span>
<span class="sd">    </span>
<span class="sd">    Both a deletion of 1 base pair and a deletion of 1000 base pairs are acceptable.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">DUP</span> <span class="o">=</span> <span class="mi">1</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Duplication (tandem or interspersed).</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">INS</span> <span class="o">=</span> <span class="mi">2</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Insertion of a novel sequence.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">INV</span> <span class="o">=</span> <span class="mi">3</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Inversion of a chromosome segment.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">MNV</span> <span class="o">=</span> <span class="mi">4</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Multi-nucleotide variant (e.g. `AG&gt;CT`) that is not a duplication, deletion, or insertion.</span>
<span class="sd">    May be called INDEL.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">SNV</span> <span class="o">=</span> <span class="mi">5</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Single nucleotide variant.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="n">TRANSLOCATION</span> <span class="o">=</span> <span class="mi">6</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    A chromosomal translocation, which occurs when a chromosome breaks</span>
<span class="sd">    and the (typically two) fragmented pieces re-attach to different chromosomes.</span>
<span class="sd">    &quot;&quot;&quot;</span></div>



<div class="viewcode-block" id="VariantCoordinates">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.VariantCoordinates">[docs]</a>
<span class="k">class</span> <span class="nc">VariantCoordinates</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    A representation of coordinates of sequence and symbolic variants.</span>

<span class="sd">    Note, the breakend variants are not currently supported.</span>
<span class="sd">    &quot;&quot;&quot;</span>

<div class="viewcode-block" id="VariantCoordinates.from_vcf_literal">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.from_vcf_literal">[docs]</a>
    <span class="nd">@staticmethod</span>
    <span class="k">def</span> <span class="nf">from_vcf_literal</span><span class="p">(</span>
        <span class="n">contig</span><span class="p">:</span> <span class="n">Contig</span><span class="p">,</span>
        <span class="n">pos</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
        <span class="n">ref</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
        <span class="n">alt</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Create `VariantCoordinates` from a variant in VCF literal notation.</span>

<span class="sd">        Note, this function must *not* be used to create a VCF symbolic variant</span>
<span class="sd">        (e.g. `&lt;DEL&gt;` or translocation).</span>
<span class="sd">        Use :func:`from_vcf_symbolic` instead.</span>

<span class="sd">        **Example**</span>

<span class="sd">        Create a variant from a VCF line:</span>
<span class="sd">        ```</span>
<span class="sd">        #CHROM  POS     ID  REF ALT ...</span>
<span class="sd">        chr1    1001    .   C   T</span>
<span class="sd">        ```</span>

<span class="sd">        We first must decide on the genome build. Most of the time, we should use GRCh38:</span>

<span class="sd">        &gt;&gt;&gt; from gpsea.model.genome import GRCh38</span>
<span class="sd">        &gt;&gt;&gt; build = GRCh38</span>

<span class="sd">        Then, we access the contig for ``&#39;chr1&#39;``:</span>

<span class="sd">        &gt;&gt;&gt; chr1 = build.contig_by_name(&#39;chr1&#39;)</span>

<span class="sd">        Last, we create the variant coordinates:</span>

<span class="sd">        &gt;&gt;&gt; from gpsea.model import VariantCoordinates</span>
<span class="sd">        &gt;&gt;&gt; vc = VariantCoordinates.from_vcf_literal(</span>
<span class="sd">        ...     contig=chr1, pos=1001, ref=&#39;C&#39;, alt=&#39;T&#39;,</span>
<span class="sd">        ... )</span>

<span class="sd">        Now can test the properties:</span>

<span class="sd">        &gt;&gt;&gt; vc.start, vc.end, vc.ref, vc.alt, len(vc), vc.change_length</span>
<span class="sd">        (1000, 1001, &#39;C&#39;, &#39;T&#39;, 1, 0)</span>

<span class="sd">        Args:</span>
<span class="sd">            contig: a :class:`~gpsea.model.genome.Contig` for the chromosome</span>
<span class="sd">            pos: a 1-based coordinate of the first base of the reference allele, as described in VCF standard</span>
<span class="sd">            ref: a `str` with the REF allele. Should meet the requirements of the VCF standard.</span>
<span class="sd">            alt: a `str` with the ALT allele. Should meet the requirements of the VCF standard.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="c1"># TODO: test that we&#39;re getting proper alleles</span>
        <span class="n">region</span> <span class="o">=</span> <span class="n">GenomicRegion</span><span class="p">(</span>
            <span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">,</span>
            <span class="n">start</span><span class="o">=</span><span class="n">pos</span> <span class="o">-</span> <span class="mi">1</span><span class="p">,</span>
            <span class="n">end</span><span class="o">=</span><span class="n">pos</span> <span class="o">+</span> <span class="nb">len</span><span class="p">(</span><span class="n">ref</span><span class="p">)</span> <span class="o">-</span> <span class="mi">1</span><span class="p">,</span>
            <span class="n">strand</span><span class="o">=</span><span class="n">Strand</span><span class="o">.</span><span class="n">POSITIVE</span><span class="p">,</span>
        <span class="p">)</span>

        <span class="n">change_length</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">ref</span><span class="p">)</span> <span class="o">-</span> <span class="nb">len</span><span class="p">(</span><span class="n">alt</span><span class="p">)</span>

        <span class="k">return</span> <span class="n">VariantCoordinates</span><span class="p">(</span>
            <span class="n">region</span><span class="o">=</span><span class="n">region</span><span class="p">,</span>
            <span class="n">ref</span><span class="o">=</span><span class="n">ref</span><span class="p">,</span>
            <span class="n">alt</span><span class="o">=</span><span class="n">alt</span><span class="p">,</span>
            <span class="n">change_length</span><span class="o">=</span><span class="n">change_length</span><span class="p">,</span>
        <span class="p">)</span></div>


<div class="viewcode-block" id="VariantCoordinates.from_vcf_symbolic">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.from_vcf_symbolic">[docs]</a>
    <span class="nd">@staticmethod</span>
    <span class="k">def</span> <span class="nf">from_vcf_symbolic</span><span class="p">(</span>
        <span class="n">contig</span><span class="p">:</span> <span class="n">Contig</span><span class="p">,</span>
        <span class="n">pos</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
        <span class="n">end</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
        <span class="n">ref</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
        <span class="n">alt</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
        <span class="n">svlen</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
    <span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Create `VariantCoordinates` from a variant in VCF symbolic notation.</span>

<span class="sd">        Note, this function must *not* be used to create a VCF sequence/literal variant.</span>
<span class="sd">        Use :func:`from_vcf_literal` instead.</span>

<span class="sd">        **Example**</span>

<span class="sd">        Let&#39;s create a symbolic variant from the line:</span>

<span class="sd">        ```</span>
<span class="sd">        #CHROM   POS      ID   REF   ALT     QUAL   FILTER   INFO</span>
<span class="sd">        2        321682   .    T     &lt;DEL&gt;   6      PASS     SVTYPE=DEL;END=321887;SVLEN=-205</span>
<span class="sd">        ```</span>

<span class="sd">        We first must decide on the genome build. Most of the time, we should use GRCh38:</span>

<span class="sd">        &gt;&gt;&gt; from gpsea.model.genome import GRCh38</span>
<span class="sd">        &gt;&gt;&gt; contig = GRCh38.contig_by_name(&#39;2&#39;)</span>

<span class="sd">        Now, we create the coordinates as:</span>

<span class="sd">        &gt;&gt;&gt; vc = VariantCoordinates.from_vcf_symbolic(</span>
<span class="sd">        ...     contig=contig, pos=321682, end=321887,</span>
<span class="sd">        ...     ref=&#39;T&#39;, alt=&#39;&lt;DEL&gt;&#39;, svlen=-205,</span>
<span class="sd">        ... )</span>

<span class="sd">        Now can test the properties:</span>

<span class="sd">        &gt;&gt;&gt; vc.start, vc.end, vc.ref, vc.alt, len(vc), vc.change_length</span>
<span class="sd">        (321681, 321887, &#39;T&#39;, &#39;&lt;DEL&gt;&#39;, 206, -205)</span>

<span class="sd">        Args:</span>
<span class="sd">            contig: a :class:`~gpsea.model.genome.Contig` for the chromosome</span>
<span class="sd">            pos: a 1-based coordinate of the first base of the affected reference allele region</span>
<span class="sd">            end: a 1-based coordinate of the last base of the affected reference allele region</span>
<span class="sd">            ref: a `str` with the REF allele. Most of the time, it is one of `{&#39;N&#39;, &#39;A&#39;, &#39;C&#39;, &#39;G&#39;, &#39;T&#39;}`</span>
<span class="sd">            alt: a `str` with the ALT allele, e.g. one of `{&#39;&lt;DEL&gt;&#39;, &#39;&lt;DUP&gt;&#39;, &#39;&lt;INS&gt;&#39;, &#39;&lt;INV&gt;&#39;}`</span>
<span class="sd">            svlen: an `int` with change length (the difference between ref and alt allele lengths)</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="c1"># TODO: test that we&#39;re getting proper alleles</span>
        <span class="n">region</span> <span class="o">=</span> <span class="n">GenomicRegion</span><span class="p">(</span>
            <span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">,</span>
            <span class="n">start</span><span class="o">=</span><span class="n">pos</span> <span class="o">-</span> <span class="mi">1</span><span class="p">,</span>  <span class="c1"># convert to 0-based coordinates,</span>
            <span class="n">end</span><span class="o">=</span><span class="n">end</span><span class="p">,</span>
            <span class="n">strand</span><span class="o">=</span><span class="n">Strand</span><span class="o">.</span><span class="n">POSITIVE</span><span class="p">,</span>
        <span class="p">)</span>

        <span class="k">return</span> <span class="n">VariantCoordinates</span><span class="p">(</span>
            <span class="n">region</span><span class="o">=</span><span class="n">region</span><span class="p">,</span>
            <span class="n">ref</span><span class="o">=</span><span class="n">ref</span><span class="p">,</span>
            <span class="n">alt</span><span class="o">=</span><span class="n">alt</span><span class="p">,</span>
            <span class="n">change_length</span><span class="o">=</span><span class="n">svlen</span><span class="p">,</span>
        <span class="p">)</span></div>


    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">region</span><span class="p">:</span> <span class="n">GenomicRegion</span><span class="p">,</span> <span class="n">ref</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">alt</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">change_length</span><span class="p">:</span> <span class="nb">int</span><span class="p">):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_region</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">region</span><span class="p">,</span> <span class="n">GenomicRegion</span><span class="p">,</span> <span class="s2">&quot;region&quot;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_ref</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">ref</span><span class="p">,</span> <span class="nb">str</span><span class="p">,</span> <span class="s2">&quot;ref&quot;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_alt</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">alt</span><span class="p">,</span> <span class="nb">str</span><span class="p">,</span> <span class="s2">&quot;alt&quot;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_change_length</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span>
            <span class="n">change_length</span><span class="p">,</span> <span class="nb">int</span><span class="p">,</span> <span class="s2">&quot;change_length&quot;</span>
        <span class="p">)</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">chrom</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the label of the chromosome/contig where the variant is located.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_region</span><span class="o">.</span><span class="n">contig</span><span class="o">.</span><span class="n">name</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">start</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the 0-based start coordinate (excluded) of the first base of the :attr:`ref` allele.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_region</span><span class="o">.</span><span class="n">start</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">end</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the 0-based end coordinate (included) of the last base of the :attr:`ref` allele.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_region</span><span class="o">.</span><span class="n">end</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">region</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">GenomicRegion</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the genomic region spanned by the :attr:`ref` allele.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_region</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">ref</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the reference allele (e.g. &quot;A&quot;, &quot;CCT&quot;, &quot;N&quot;). The allele may be an empty string.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_ref</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">alt</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the alternate allele (e.g. &quot;A&quot;, &quot;GG&quot;, &quot;&lt;DEL&gt;&quot;).</span>

<span class="sd">        The allele may be an empty string for sequence variants.</span>
<span class="sd">        The symbolic alternate allele follow the VCF notation and use the `&lt;` and `&gt;` characters</span>
<span class="sd">        (e.g. &quot;&lt;DEL&gt;&quot;, &quot;&lt;INS:ME:SINE&gt;&quot;).</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_alt</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">change_length</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the change of length between the `ref` and `alt` alleles due to the variant presence.</span>

<span class="sd">        See :ref:`change-length-of-an-allele` for more info.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_change_length</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">variant_key</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get a readable representation of the variant&#39;s coordinates.</span>

<span class="sd">        For instance, ``X_12345_12345_C_G`` for a sequence variant or ``22_10001_20000_INV`` for a symbolic variant.</span>
<span class="sd">        If the key is larger than 50 characters, the &#39;ref&#39; and/or &#39;alt&#39; (if over 10 bps)</span>
<span class="sd">        are changed to just show number of bps.</span>
<span class="sd">        Example: ``X_1000001_1000027_TAAAAAAAAAAAAAAAAAAAAAAAAAA_T`` -&gt; ``X_1000001_1000027_--27bp--_T``</span>

<span class="sd">        .. note::</span>

<span class="sd">          Both *start* and *end* coordinates use 1-based (included) coordinate system.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">is_structural</span><span class="p">():</span>
            <span class="k">return</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">chrom</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="mi">1</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">alt</span><span class="p">[</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s2">&quot;</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">key</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">chrom</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="mi">1</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">ref</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">alt</span><span class="si">}</span><span class="s2">&quot;</span>
            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">key</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">50</span><span class="p">:</span>
                <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">10</span><span class="p">:</span>
                    <span class="n">ref</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;--</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref</span><span class="p">)</span><span class="si">}</span><span class="s2">bp--&quot;</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">ref</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref</span>
                <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">alt</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">10</span><span class="p">:</span>
                    <span class="n">alt</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;--</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">alt</span><span class="p">)</span><span class="si">}</span><span class="s2">bp--&quot;</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">alt</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt</span>
                <span class="k">return</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">chrom</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="mi">1</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="n">ref</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="n">alt</span><span class="si">}</span><span class="s2">&quot;</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">return</span> <span class="n">key</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">variant_class</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">VariantClass</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get a :class:`VariantClass` category.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">is_structural</span><span class="p">():</span>
            <span class="c1"># Expecting a `str` like &lt;DEL&gt;, &lt;INS&gt;, &lt;DUP&gt;, &lt;INV&gt;, ...</span>
            <span class="k">return</span> <span class="n">VariantClass</span><span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">alt</span><span class="p">[</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">]]</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref</span><span class="p">)</span> <span class="o">&gt;</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">alt</span><span class="p">):</span>
                <span class="k">return</span> <span class="n">VariantClass</span><span class="o">.</span><span class="n">DEL</span>
            <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref</span><span class="p">)</span> <span class="o">&lt;</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">alt</span><span class="p">):</span>
                <span class="c1"># may also be a duplication, but it&#39;s hard to say from this</span>
                <span class="k">return</span> <span class="n">VariantClass</span><span class="o">.</span><span class="n">INS</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">VariantClass</span><span class="o">.</span><span class="n">SNV</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">VariantClass</span><span class="o">.</span><span class="n">MNV</span>

<div class="viewcode-block" id="VariantCoordinates.is_structural">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.is_structural">[docs]</a>
    <span class="k">def</span> <span class="nf">is_structural</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Checks if the variant coordinates use structural variant notation as described by Variant Call Format</span>
<span class="sd">        (`VCF &lt;https://en.wikipedia.org/wiki/Variant_Call_Format&gt;`_).</span>

<span class="sd">        Ane example of *structural* variant notation::</span>

<span class="sd">          chr5  101 . N &lt;DEL&gt; .  .  SVTYPE=DEL;END=120;SVLEN=-10</span>


<span class="sd">        as opposed to the *sequence* (literal) notation::</span>

<span class="sd">          chr5  101 . NACGTACGTAC N</span>

<span class="sd">        :return: `True` if the variant coordinates use structural variant notation.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">_alt</span><span class="p">)</span> <span class="o">!=</span> <span class="mi">0</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_alt</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;&lt;&quot;</span><span class="p">)</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_alt</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s2">&quot;&gt;&quot;</span><span class="p">)</span>
        <span class="p">)</span></div>


    <span class="k">def</span> <span class="fm">__len__</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the number of bases on the ref allele that are affected by the variant.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">_region</span><span class="p">)</span>

    <span class="k">def</span> <span class="fm">__eq__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">other</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="nb">isinstance</span><span class="p">(</span><span class="n">other</span><span class="p">,</span> <span class="n">VariantCoordinates</span><span class="p">)</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">region</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">region</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">ref</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">alt</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">change_length</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">change_length</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__hash__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">hash</span><span class="p">((</span><span class="bp">self</span><span class="o">.</span><span class="n">_region</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">_ref</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">_alt</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">_change_length</span><span class="p">))</span>

    <span class="k">def</span> <span class="fm">__str__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="sa">f</span><span class="s2">&quot;VariantCoordinates(region=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">region</span><span class="si">}</span><span class="s2">, &quot;</span>
            <span class="sa">f</span><span class="s2">&quot;ref=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">ref</span><span class="si">}</span><span class="s2">, alt=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">alt</span><span class="si">}</span><span class="s2">, &quot;</span>
            <span class="sa">f</span><span class="s2">&quot;change_length=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">change_length</span><span class="si">}</span><span class="s2">)&quot;</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__repr__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span></div>



<div class="viewcode-block" id="ImpreciseSvInfo">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.ImpreciseSvInfo">[docs]</a>
<span class="k">class</span> <span class="nc">ImpreciseSvInfo</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Data regarding a structural variant (SV) with imprecise breakpoint coordinates.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">structural_type</span><span class="p">:</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">TermId</span><span class="p">,</span>
        <span class="n">variant_class</span><span class="p">:</span> <span class="n">VariantClass</span><span class="p">,</span>
        <span class="n">gene_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
        <span class="n">gene_symbol</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="p">):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_structural_type</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span>
            <span class="n">structural_type</span><span class="p">,</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">TermId</span><span class="p">,</span> <span class="s2">&quot;structural_type&quot;</span>
        <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_variant_class</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span>
            <span class="n">variant_class</span><span class="p">,</span> <span class="n">VariantClass</span><span class="p">,</span> <span class="s2">&quot;variant_class&quot;</span>
        <span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span> <span class="o">=</span> <span class="n">gene_id</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_gene_symbol</span> <span class="o">=</span> <span class="n">gene_symbol</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">structural_type</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">TermId</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get term ID of the structural type (e.g. ``SO:1000029`` for chromosomal deletion).</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_structural_type</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">variant_class</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">VariantClass</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get a :class:`VariantClass` category.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_variant_class</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">gene_id</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get a `str` with gene identifier CURIE (e.g. ``HGNC:3603``) or `None` if the identifier is not available.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">gene_symbol</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get a `str` with HGVS gene symbol (e.g. *FBN1*) or `None` if the symbol is not available.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gene_symbol</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">variant_key</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get a readable representation of the variant.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_structural_type</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span><span class="si">}</span><span class="s2">_</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_gene_symbol</span><span class="si">}</span><span class="s2">&quot;</span>

    <span class="k">def</span> <span class="fm">__eq__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">value</span><span class="p">:</span> <span class="nb">object</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="nb">isinstance</span><span class="p">(</span><span class="n">value</span><span class="p">,</span> <span class="n">ImpreciseSvInfo</span><span class="p">)</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_structural_type</span> <span class="o">==</span> <span class="n">value</span><span class="o">.</span><span class="n">_structural_type</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_variant_class</span> <span class="o">==</span> <span class="n">value</span><span class="o">.</span><span class="n">_variant_class</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span> <span class="o">==</span> <span class="n">value</span><span class="o">.</span><span class="n">_gene_id</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gene_symbol</span> <span class="o">==</span> <span class="n">value</span><span class="o">.</span><span class="n">_gene_symbol</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__hash__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">hash</span><span class="p">(</span>
            <span class="p">(</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_structural_type</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_variant_class</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span><span class="p">,</span>
                <span class="bp">self</span><span class="o">.</span><span class="n">_gene_symbol</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__str__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="sa">f</span><span class="s2">&quot;ImpreciseSv(&quot;</span>
            <span class="sa">f</span><span class="s2">&quot;structural_type=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_structural_type</span><span class="si">}</span><span class="s2">, &quot;</span>
            <span class="sa">f</span><span class="s2">&quot;variant_class=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_variant_class</span><span class="si">}</span><span class="s2">, &quot;</span>
            <span class="sa">f</span><span class="s2">&quot;gene_id=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_gene_id</span><span class="si">}</span><span class="s2">, &quot;</span>
            <span class="sa">f</span><span class="s2">&quot;gene_symbol=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_gene_symbol</span><span class="si">}</span><span class="s2">)&quot;</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__repr__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span></div>



<div class="viewcode-block" id="VariantInfo">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.VariantInfo">[docs]</a>
<span class="k">class</span> <span class="nc">VariantInfo</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    `VariantInfo` consists of either variant coordinates or imprecise SV data.</span>

<span class="sd">    The class is conceptually similar to Rust enum - only one of the fields can be set at any point in time.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">variant_coordinates</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">VariantCoordinates</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
        <span class="n">sv_info</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">ImpreciseSvInfo</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="p">):</span>
        <span class="k">if</span> <span class="p">(</span><span class="n">variant_coordinates</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">)</span> <span class="o">!=</span> <span class="p">(</span><span class="n">sv_info</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">):</span>
            <span class="c1"># At most one field can be set.</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_variant_coordinates</span> <span class="o">=</span> <span class="n">variant_coordinates</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_sv_info</span> <span class="o">=</span> <span class="n">sv_info</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="sa">f</span><span class="s2">&quot;Only one field can be set: variant_coordinates=</span><span class="si">{</span><span class="n">variant_coordinates</span><span class="si">}</span><span class="s2">, sv_info=</span><span class="si">{</span><span class="n">sv_info</span><span class="si">}</span><span class="s2">&quot;</span>
            <span class="p">)</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">variant_coordinates</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">VariantCoordinates</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get variant coordinates if available.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_variant_coordinates</span>

<div class="viewcode-block" id="VariantInfo.has_variant_coordinates">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.VariantInfo.has_variant_coordinates">[docs]</a>
    <span class="k">def</span> <span class="nf">has_variant_coordinates</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Returns `True` if the variant coordinates are available.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">variant_coordinates</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span></div>


    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">sv_info</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">ImpreciseSvInfo</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get information about large imprecise SV.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_sv_info</span>

<div class="viewcode-block" id="VariantInfo.has_sv_info">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.VariantInfo.has_sv_info">[docs]</a>
    <span class="k">def</span> <span class="nf">has_sv_info</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Returns `True` if the variant is a large imprecise SV and the exact coordinates are thus unavailable.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">sv_info</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span></div>


    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">variant_key</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get a readable representation of the variant&#39;s coordinates or the large SV info.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">has_variant_coordinates</span><span class="p">():</span>
            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">variant_coordinates</span><span class="o">.</span><span class="n">variant_key</span>
        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">has_sv_info</span><span class="p">():</span>
            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">sv_info</span><span class="o">.</span><span class="n">variant_key</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_handle_missing_state</span><span class="p">()</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">variant_class</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">VariantClass</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get a :class:`VariantClass` category.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">has_variant_coordinates</span><span class="p">():</span>
            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">variant_coordinates</span><span class="o">.</span><span class="n">variant_class</span>
        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">has_sv_info</span><span class="p">():</span>
            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">sv_info</span><span class="o">.</span><span class="n">variant_class</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_handle_missing_state</span><span class="p">()</span>

<div class="viewcode-block" id="VariantInfo.is_structural">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.VariantInfo.is_structural">[docs]</a>
    <span class="k">def</span> <span class="nf">is_structural</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Test if the variant is a structural variant.</span>

<span class="sd">        This can either be because the variant coordinates</span>
<span class="sd">        use the structural variant notation (see :meth:`VariantCoordinates.is_structural`)</span>
<span class="sd">        or if the variant is large imprecise SV.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">has_variant_coordinates</span><span class="p">():</span>
            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">variant_coordinates</span><span class="o">.</span><span class="n">is_structural</span><span class="p">()</span>
        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">has_sv_info</span><span class="p">():</span>
            <span class="k">return</span> <span class="kc">True</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_handle_missing_state</span><span class="p">()</span></div>


    <span class="k">def</span> <span class="nf">_handle_missing_state</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
            <span class="s2">&quot;VariantInfo should have either variant coordinates or SV info!&quot;</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__eq__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">value</span><span class="p">:</span> <span class="nb">object</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="nb">isinstance</span><span class="p">(</span><span class="n">value</span><span class="p">,</span> <span class="n">VariantInfo</span><span class="p">)</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_variant_coordinates</span> <span class="o">==</span> <span class="n">value</span><span class="o">.</span><span class="n">_variant_coordinates</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_sv_info</span> <span class="o">==</span> <span class="n">value</span><span class="o">.</span><span class="n">_sv_info</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__hash__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">hash</span><span class="p">((</span><span class="bp">self</span><span class="o">.</span><span class="n">_variant_coordinates</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">_sv_info</span><span class="p">))</span>

    <span class="k">def</span> <span class="fm">__str__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="sa">f</span><span class="s2">&quot;VariantInfo(&quot;</span>
            <span class="sa">f</span><span class="s2">&quot;variant_coordinates=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_variant_coordinates</span><span class="si">}</span><span class="s2">, &quot;</span>
            <span class="sa">f</span><span class="s2">&quot;sv_info=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_sv_info</span><span class="si">}</span><span class="s2">)&quot;</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__repr__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span></div>



<div class="viewcode-block" id="VariantInfoAware">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.VariantInfoAware">[docs]</a>
<span class="k">class</span> <span class="nc">VariantInfoAware</span><span class="p">(</span><span class="n">metaclass</span><span class="o">=</span><span class="n">abc</span><span class="o">.</span><span class="n">ABCMeta</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    An entity where :class:`VariantInfo` is available.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="nd">@property</span>
    <span class="nd">@abc</span><span class="o">.</span><span class="n">abstractmethod</span>
    <span class="k">def</span> <span class="nf">variant_info</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">VariantInfo</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the variant data with coordinates or other info available for large imprecise SVs.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">pass</span></div>



<div class="viewcode-block" id="FunctionalAnnotationAware">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.FunctionalAnnotationAware">[docs]</a>
<span class="k">class</span> <span class="nc">FunctionalAnnotationAware</span><span class="p">(</span><span class="n">metaclass</span><span class="o">=</span><span class="n">abc</span><span class="o">.</span><span class="n">ABCMeta</span><span class="p">):</span>

    <span class="nd">@property</span>
    <span class="nd">@abc</span><span class="o">.</span><span class="n">abstractmethod</span>
    <span class="k">def</span> <span class="nf">tx_annotations</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">TranscriptAnnotation</span><span class="p">]:</span>
        <span class="k">pass</span>

<div class="viewcode-block" id="FunctionalAnnotationAware.get_tx_anno_by_tx_id">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.FunctionalAnnotationAware.get_tx_anno_by_tx_id">[docs]</a>
    <span class="k">def</span> <span class="nf">get_tx_anno_by_tx_id</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span> <span class="n">transcript_id</span><span class="p">:</span> <span class="nb">str</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">TranscriptAnnotation</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Given a transcript ID, this will return the `TranscriptAnnotation` associated with that</span>
<span class="sd">        variant and transcript.</span>

<span class="sd">        Args:</span>
<span class="sd">            transcript_id (str): A transcript ID - i.e. &#39;NM_170707.4&#39;</span>

<span class="sd">        Returns:</span>
<span class="sd">            typing.Optional[TranscriptAnnotation]: The Transcript Annotation if available.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">for</span> <span class="n">tx_ann</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_annotations</span><span class="p">:</span>
            <span class="k">if</span> <span class="n">tx_ann</span><span class="o">.</span><span class="n">transcript_id</span> <span class="o">==</span> <span class="n">transcript_id</span><span class="p">:</span>
                <span class="k">return</span> <span class="n">tx_ann</span>
        <span class="k">return</span> <span class="kc">None</span></div>


<div class="viewcode-block" id="FunctionalAnnotationAware.get_hgvs_cdna_by_tx_id">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.FunctionalAnnotationAware.get_hgvs_cdna_by_tx_id">[docs]</a>
    <span class="k">def</span> <span class="nf">get_hgvs_cdna_by_tx_id</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Given a transcript ID, will return the hgvs cdna string associated with that variant and transcript.</span>

<span class="sd">        Args:</span>
<span class="sd">            transcript_id (str): A transcript ID - i.e. &#39;NM_170707.4&#39;</span>

<span class="sd">        Returns:</span>
<span class="sd">            str or None: The hgvs cdna if available - i.e. &#39;NM_170707.4:c.1824C&gt;T&#39;</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">for</span> <span class="n">tx_ann</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_annotations</span><span class="p">:</span>
            <span class="k">if</span> <span class="n">tx_ann</span><span class="o">.</span><span class="n">transcript_id</span> <span class="o">==</span> <span class="n">transcript_id</span><span class="p">:</span>
                <span class="k">return</span> <span class="n">tx_ann</span><span class="o">.</span><span class="n">hgvs_cdna</span>
        <span class="k">return</span> <span class="kc">None</span></div>


<div class="viewcode-block" id="FunctionalAnnotationAware.get_preferred_tx_annotation">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.FunctionalAnnotationAware.get_preferred_tx_annotation">[docs]</a>
    <span class="k">def</span> <span class="nf">get_preferred_tx_annotation</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">TranscriptAnnotation</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Get the `TranscriptAnnotation` that represents the result of the functional annotation</span>
<span class="sd">        with respect to the preferred transcript of a gene.</span>

<span class="sd">        Returns `None` if transcript annotations is no preferred transcript found.</span>

<span class="sd">        Returns:</span>
<span class="sd">            typing.Optional[TranscriptAnnotation]: The `TranscriptAnnotation` with respect</span>
<span class="sd">                                                   to the preferred transcript</span>
<span class="sd">                                                   or `None` if the preferred transcript info is not available.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">for</span> <span class="n">tx</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_annotations</span><span class="p">:</span>
            <span class="k">if</span> <span class="n">tx</span><span class="o">.</span><span class="n">is_preferred</span><span class="p">:</span>
                <span class="k">return</span> <span class="n">tx</span>
        <span class="k">return</span> <span class="kc">None</span></div>
</div>



<div class="viewcode-block" id="Variant">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.Variant">[docs]</a>
<span class="k">class</span> <span class="nc">Variant</span><span class="p">(</span><span class="n">VariantInfoAware</span><span class="p">,</span> <span class="n">FunctionalAnnotationAware</span><span class="p">,</span> <span class="n">Genotyped</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Variant includes three lines of information:</span>
<span class="sd">     * the variant data with coordinates or other info available for large imprecise SVs,</span>
<span class="sd">     * results of the functional annotation with respect to relevant transcripts, and</span>
<span class="sd">     * the genotypes for the known samples</span>
<span class="sd">    &quot;&quot;&quot;</span>

<div class="viewcode-block" id="Variant.create_variant_from_scratch">
<a class="viewcode-back" href="../../../apidocs/gpsea.model.html#gpsea.model.Variant.create_variant_from_scratch">[docs]</a>
    <span class="nd">@staticmethod</span>
    <span class="k">def</span> <span class="nf">create_variant_from_scratch</span><span class="p">(</span>
        <span class="n">variant_coordinates</span><span class="p">:</span> <span class="n">VariantCoordinates</span><span class="p">,</span>
        <span class="n">gene_name</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
        <span class="n">trans_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
        <span class="n">hgvs_cdna</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
        <span class="n">is_preferred</span><span class="p">:</span> <span class="nb">bool</span><span class="p">,</span>
        <span class="n">consequences</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Iterable</span><span class="p">[</span><span class="n">VariantEffect</span><span class="p">],</span>
        <span class="n">exons_effected</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="nb">int</span><span class="p">],</span>
        <span class="n">protein_id</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
        <span class="n">hgvsp</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
        <span class="n">protein_effect_start</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">int</span><span class="p">],</span>
        <span class="n">protein_effect_end</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">int</span><span class="p">],</span>
        <span class="n">genotypes</span><span class="p">:</span> <span class="n">Genotypes</span><span class="p">,</span>
    <span class="p">):</span>
        <span class="n">variant_info</span> <span class="o">=</span> <span class="n">VariantInfo</span><span class="p">(</span><span class="n">variant_coordinates</span><span class="o">=</span><span class="n">variant_coordinates</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">protein_effect_start</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="ow">and</span> <span class="n">protein_effect_end</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">protein_effect</span> <span class="o">=</span> <span class="n">Region</span><span class="p">(</span><span class="n">protein_effect_start</span><span class="p">,</span> <span class="n">protein_effect_end</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">protein_effect</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="n">transcript</span> <span class="o">=</span> <span class="n">TranscriptAnnotation</span><span class="p">(</span>
            <span class="n">gene_name</span><span class="p">,</span>
            <span class="n">trans_id</span><span class="p">,</span>
            <span class="n">hgvs_cdna</span><span class="p">,</span>
            <span class="n">is_preferred</span><span class="p">,</span>
            <span class="n">consequences</span><span class="p">,</span>
            <span class="n">exons_effected</span><span class="p">,</span>
            <span class="n">protein_id</span><span class="p">,</span>
            <span class="n">hgvsp</span><span class="p">,</span>
            <span class="n">protein_effect</span><span class="p">,</span>
        <span class="p">)</span>
        <span class="k">return</span> <span class="n">Variant</span><span class="p">(</span><span class="n">variant_info</span><span class="p">,</span> <span class="p">(</span><span class="n">transcript</span><span class="p">,),</span> <span class="n">genotypes</span><span class="p">)</span></div>


    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">variant_info</span><span class="p">:</span> <span class="n">VariantInfo</span><span class="p">,</span>
        <span class="n">tx_annotations</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Iterable</span><span class="p">[</span><span class="n">TranscriptAnnotation</span><span class="p">],</span>
        <span class="n">genotypes</span><span class="p">:</span> <span class="n">Genotypes</span><span class="p">,</span>
    <span class="p">):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_variant_info</span> <span class="o">=</span> <span class="n">variant_info</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_tx_annotations</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">tx_annotations</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_gts</span> <span class="o">=</span> <span class="n">genotypes</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">variant_info</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">VariantInfo</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get the representation of the variant data for sequence and symbolic variants, </span>
<span class="sd">        as well as for large imprecise SVs.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_variant_info</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">tx_annotations</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">TranscriptAnnotation</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;A collection of TranscriptAnnotations that each represent results of the functional annotation</span>
<span class="sd">        of a variant with respect to single transcript of a gene.</span>

<span class="sd">        Returns:</span>
<span class="sd">            Sequence[TranscriptAnnotation]: A sequence of TranscriptAnnotation objects</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_tx_annotations</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">genotypes</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Genotypes</span><span class="p">:</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gts</span>

    <span class="k">def</span> <span class="fm">__eq__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">other</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="nb">isinstance</span><span class="p">(</span><span class="n">other</span><span class="p">,</span> <span class="n">Variant</span><span class="p">)</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_variant_info</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_variant_info</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_tx_annotations</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_tx_annotations</span>
            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gts</span> <span class="o">==</span> <span class="n">other</span><span class="o">.</span><span class="n">_gts</span>
        <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__hash__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">hash</span><span class="p">((</span><span class="bp">self</span><span class="o">.</span><span class="n">_variant_info</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">_tx_annotations</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gts</span><span class="p">))</span>

    <span class="k">def</span> <span class="fm">__repr__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span>

    <span class="k">def</span> <span class="fm">__str__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="k">return</span> <span class="p">(</span>
            <span class="sa">f</span><span class="s2">&quot;Variant(variant_info=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_variant_info</span><span class="si">}</span><span class="s2">, &quot;</span>
            <span class="sa">f</span><span class="s2">&quot;tx_annotations=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_tx_annotations</span><span class="si">}</span><span class="s2">, &quot;</span>
            <span class="sa">f</span><span class="s2">&quot;genotypes=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_gts</span><span class="si">}</span><span class="s2">)&quot;</span>
        <span class="p">)</span></div>

</pre></div>

           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>