<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>genophenocorr.analysis package &mdash; genophenocorr 0.1.1dev documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/genophenocorr.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=8427294d"></script>
        <script src="../_static/doctools.js?v=888ff710"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="genophenocorr.analysis.predicate package" href="genophenocorr.analysis.predicate.html" />
    <link rel="prev" title="genophenocorr package" href="genophenocorr.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            genophenocorr
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../user-guide/index.html">User guide</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">API reference</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="genophenocorr.html">genophenocorr package</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="genophenocorr.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l4 current"><a class="current reference internal" href="#">genophenocorr.analysis package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.constants.html">genophenocorr.constants package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.data.html">genophenocorr.data package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.model.html">genophenocorr.model package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.preprocessing.html">genophenocorr.preprocessing package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.view.html">genophenocorr.view package</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">genophenocorr</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="modules.html">API reference</a></li>
          <li class="breadcrumb-item"><a href="genophenocorr.html">genophenocorr package</a></li>
      <li class="breadcrumb-item active">genophenocorr.analysis package</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/apidocs/genophenocorr.analysis.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-genophenocorr.analysis">
<span id="genophenocorr-analysis-package"></span><h1>genophenocorr.analysis package<a class="headerlink" href="#module-genophenocorr.analysis" title="Link to this heading"></a></h1>
<dl class="py function">
<dt class="sig sig-object py" id="genophenocorr.analysis.configure_cohort_analysis">
<span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">configure_cohort_analysis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cohort</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Cohort" title="genophenocorr.model._cohort.Cohort"><span class="pre">Cohort</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://thejacksonlaboratory.github.io/hpo-toolkit/latest/apidocs/hpotk.ontology.html#hpotk.ontology.MinimalOntology" title="(in hpo-toolkit)"><span class="pre">MinimalOntology</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.analysis.CohortAnalysisConfiguration" title="genophenocorr.analysis._config.CohortAnalysisConfiguration"><span class="pre">CohortAnalysisConfiguration</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.CohortAnalysis" title="genophenocorr.analysis._api.CohortAnalysis"><span class="pre">CohortAnalysis</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#configure_cohort_analysis"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.configure_cohort_analysis" title="Link to this definition"></a></dt>
<dd><p>Configure <a class="reference internal" href="#genophenocorr.analysis.CohortAnalysis" title="genophenocorr.analysis.CohortAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.analysis.CohortAnalysis</span></code></a> for given <cite>cohort</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>cohort</strong> – a <a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Cohort" title="genophenocorr.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.model.Cohort</span></code></a> to analyze</p></li>
<li><p><strong>hpo</strong> – a <code class="xref py py-class docutils literal notranslate"><span class="pre">hpotk.MinimalOntology</span></code> with HPO to use in the analysis</p></li>
<li><p><strong>config</strong> – an optional <a class="reference internal" href="#genophenocorr.analysis.CohortAnalysisConfiguration" title="genophenocorr.analysis.CohortAnalysisConfiguration"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortAnalysisConfiguration</span></code></a> to parameterize the analysis.
The default parameters will be used if <cite>None</cite>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">CohortAnalysis</span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>CohortAnalysis</cite> is a driver class for running genotype-phenotype correlation analyses.</p>
<p>The class provides various methods to test genotype-phenotype correlations. All methods wrap results
into <a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis.GenotypePhenotypeAnalysisResult"><code class="xref py py-class docutils literal notranslate"><span class="pre">GenotypePhenotypeAnalysisResult</span></code></a>.</p>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_variant_effect">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_by_variant_effect</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">effect</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_variant_effect"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_variant_effect" title="Link to this definition"></a></dt>
<dd><p>Compares variants with <cite>effect</cite> vs. variants with all other variant effects.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>effect</strong> – variant effect</p></li>
<li><p><strong>tx_id</strong> – the accession of the transcript of interest</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_variant_key">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_by_variant_key</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant_key</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_variant_key"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_variant_key" title="Link to this definition"></a></dt>
<dd><p>Compares variant with <cite>variant_key</cite> vs all the other variants.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantCoordinates.variant_key" title="genophenocorr.model.VariantCoordinates.variant_key"><code class="xref py py-attr docutils literal notranslate"><span class="pre">genophenocorr.model.VariantCoordinates.variant_key</span></code></a></p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>variant_key</strong> – a <cite>str</cite> with the variant key, e.g. <code class="docutils literal notranslate"><span class="pre">X_12345_12345_C_G</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_exon">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_by_exon</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">exon_number</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_exon"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_exon" title="Link to this definition"></a></dt>
<dd><p>Compares variants with <cite>effect</cite> vs. variants with all other variant effects.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>We use 1-based numbering to number the exons, not the usual 0-based numbering of the computer science.
Therefore, the first exon of the transcript has <code class="docutils literal notranslate"><span class="pre">exon_number==1</span></code>, the second exon is <code class="docutils literal notranslate"><span class="pre">2</span></code>, and so on …</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>We do not check if the <cite>exon_number</cite> spans beyond the number of exons of the given <cite>transcript_id</cite>!
Therefore, <code class="docutils literal notranslate"><span class="pre">exon_number==10,000</span></code> will effectively return <code class="xref py py-attr docutils literal notranslate"><span class="pre">BooleanPredicate.FALSE</span></code> for <em>all</em> patients!!! 😱
Well, at least the patients of the <em>Homo sapiens sapiens</em> taxon…</p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>exon_number</strong> – a positive <cite>int</cite> with exon number</p></li>
<li><p><strong>tx_id</strong> – the accession of the transcript of interest</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_variant_keys">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_by_variant_keys</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">a</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">b</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_variant_keys"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_variant_keys" title="Link to this definition"></a></dt>
<dd><p>Compare genotype-phenotype correlation between groups with variant <cite>a</cite> vs. variant <cite>b</cite>.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantCoordinates.variant_key" title="genophenocorr.model.VariantCoordinates.variant_key"><code class="xref py py-attr docutils literal notranslate"><span class="pre">genophenocorr.model.VariantCoordinates.variant_key</span></code></a></p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>a</strong> – a <cite>str</cite> with variant key of variant <cite>a</cite>, e.g. <code class="docutils literal notranslate"><span class="pre">X_12345_12345_C_G</span></code></p></li>
<li><p><strong>b</strong> – a <cite>str</cite> with variant key of variant <cite>b</cite></p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">GenotypePhenotypeAnalysisResult</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_usable</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">all_counts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvals</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">corrected_pvals</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">phenotype_categories</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">question</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#GenotypePhenotypeAnalysisResult"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>GenotypePhenotypeAnalysisResult</cite> summarizes results of genotype-phenotype correlation analysis of a cohort.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.n_usable">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">n_usable</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.n_usable" title="Link to this definition"></a></dt>
<dd><p>Get a <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">pandas.Series</span></code></a> with mapping from HPO term and number of patients where the term was either
present or excluded, and are, thus, usable for genotype-phenotype correlation analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.all_counts">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">all_counts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.all_counts" title="Link to this definition"></a></dt>
<dd><p>Get a <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">pandas.DataFrame</span></code></a> with counts of patients in genotype and phenotype groups.</p>
<p>An example for a genotype predicate that bins into two categories (<cite>Yes</cite> and <cite>No</cite>) based on presence
of a missense variant in transcript, and phenotype predicate that checks presence/absence of a phenotype term:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>                       <span class="n">Has</span> <span class="n">MISSENSE_VARIANT</span> <span class="ow">in</span> <span class="n">NM_123456</span><span class="mf">.7</span>
                       <span class="n">No</span>       <span class="n">Yes</span>
<span class="n">HPO</span>         <span class="n">Present</span>
<span class="n">HP</span><span class="p">:</span><span class="mi">0001166</span>  <span class="n">Yes</span>        <span class="mi">1</span>        <span class="mi">13</span>
<span class="n">HP</span><span class="p">:</span><span class="mi">0001166</span>  <span class="n">No</span>         <span class="mi">7</span>        <span class="mi">5</span>
<span class="n">HP</span><span class="p">:</span><span class="mi">0003011</span>  <span class="n">Yes</span>        <span class="mi">3</span>        <span class="mi">11</span>
<span class="n">HP</span><span class="p">:</span><span class="mi">0003011</span>  <span class="n">No</span>         <span class="mi">7</span>        <span class="mi">16</span>
<span class="n">HP</span><span class="p">:</span><span class="mi">0012638</span>  <span class="n">Yes</span>        <span class="mi">9</span>        <span class="mi">15</span>
<span class="n">HP</span><span class="p">:</span><span class="mi">0012638</span>  <span class="n">No</span>         <span class="mi">3</span>        <span class="mi">4</span>
</pre></div>
</div>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.pvals">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pvals</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.pvals" title="Link to this definition"></a></dt>
<dd><p>Get a <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">pandas.Series</span></code></a> with p values for each tested HPO term.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.corrected_pvals">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">corrected_pvals</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.corrected_pvals" title="Link to this definition"></a></dt>
<dd><p>Get an optional <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">pandas.Series</span></code></a> with p values for each tested HPO term after multiple testing correction.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.phenotype_categories">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">phenotype_categories</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.phenotype_categories" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of phenotype patient categories that can be investigated.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.summarize">
<span class="sig-name descname"><span class="pre">summarize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://thejacksonlaboratory.github.io/hpo-toolkit/latest/apidocs/hpotk.ontology.html#hpotk.ontology.MinimalOntology" title="(in hpo-toolkit)"><span class="pre">MinimalOntology</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">phenotype_category</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#GenotypePhenotypeAnalysisResult.summarize"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.summarize" title="Link to this definition"></a></dt>
<dd><p>Create a data frame with summary of the genotype phenotype analysis.</p>
<p>The <em>rows</em> of the frame correspond to the analyzed HPO terms.</p>
<p>The columns of the data frame have <cite>Count</cite> and <cite>Percentage</cite> per used genotype predicate.
For instance, if we used <a class="reference internal" href="genophenocorr.analysis.predicate.genotype.html#genophenocorr.analysis.predicate.genotype.VariantEffectPredicate" title="genophenocorr.analysis.predicate.genotype.VariantEffectPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.analysis.predicate.genotype.VariantEffectPredicate</span></code></a>
which compares phenotype with and without variants with certain <a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model.VariantEffect"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.model.VariantEffect</span></code></a>,
we will have the following columns:</p>
<table class="docutils align-default">
<tbody>
<tr class="row-odd"><td colspan="2"><p>Yes</p></td>
<td colspan="2"><p>No</p></td>
</tr>
<tr class="row-even"><td><p>Count</p></td>
<td><p>Percent</p></td>
<td><p>Count</p></td>
<td><p>Percent</p></td>
</tr>
</tbody>
</table>
<p>The final data frame may look something like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>MISSENSE_VARIANT on `NM_1234.5`                       No                  Yes
                                                      Count    Percent    Count    Percent   p value   Corrected p value
Arachnodactyly [HP:0001166]                           1         3.8461    13       50.0000   0.000781  0.020299
Abnormality of the musculature [HP:0003011]           6        35.2941    11       64.7058   1.000000  1.000000
Abnormal nervous system physiology [HP:0012638]       9        37.5000    15       62.5000   1.000000  1.000000
...                                                   ...      ...        ...      ...       ...       ...
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">CohortAnalysisConfiguration</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">missing_implies_excluded</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">pval_correction</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_perc_patients_w_hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">include_sv</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfiguration"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>CohortAnalysisConfiguration</cite> is a value class for storing <a class="reference internal" href="#genophenocorr.analysis.CohortAnalysis" title="genophenocorr.analysis.CohortAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.analysis.CohortAnalysis</span></code></a>
configuration options.</p>
<section id="default-values">
<h2>Default values<a class="headerlink" href="#default-values" title="Link to this heading"></a></h2>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Option</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Default value</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">missing_implies_excluded</span></code></p></td>
<td><p><cite>bool</cite></p></td>
<td><p><cite>False</cite></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">pval_correction</span></code></p></td>
<td><p><cite>str</cite></p></td>
<td><p><cite>bonferroni</cite></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">min_perc_patients_w_hpo</span></code></p></td>
<td><p><cite>float</cite></p></td>
<td><p><cite>0.1</cite></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">include_sv</span></code></p></td>
<td><p><cite>bool</cite></p></td>
<td><p><cite>False</cite></p></td>
</tr>
</tbody>
</table>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.builder">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">builder</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfiguration.builder"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.builder" title="Link to this definition"></a></dt>
<dd><p>Create a new <a class="reference internal" href="#genophenocorr.analysis.CohortAnalysisConfigurationBuilder" title="genophenocorr.analysis.CohortAnalysisConfigurationBuilder"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortAnalysisConfigurationBuilder</span></code></a> initialized with default configuration options.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.missing_implies_excluded">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">missing_implies_excluded</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.missing_implies_excluded" title="Link to this definition"></a></dt>
<dd><p><cite>True</cite> if we assume that a patient without a specific phenotype listed <em>does not</em> have the phenotype.
Otherwise, the only excluded phenotypes are those that are excluded explicitly.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.pval_correction">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pval_correction</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.pval_correction" title="Link to this definition"></a></dt>
<dd><p><cite>bonferroni</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Get method for multiple testing p value correction. Default</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.min_perc_patients_w_hpo">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">min_perc_patients_w_hpo</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.min_perc_patients_w_hpo" title="Link to this definition"></a></dt>
<dd><p>A threshold for removing rare HPO terms, only the terms that are observed in at least this fraction of patients
will be retained for the analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.include_sv">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">include_sv</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.include_sv" title="Link to this definition"></a></dt>
<dd><p><cite>True</cite> if we want to include structural variants in the analysis
(i.e. the variants that use symbolic VCF notation).</p>
</dd></dl>

</section>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfigurationBuilder">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">CohortAnalysisConfigurationBuilder</span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfigurationBuilder"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfigurationBuilder" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>CohortAnalysisConfigurationBuilder</cite> builds <a class="reference internal" href="#genophenocorr.analysis.CohortAnalysisConfiguration" title="genophenocorr.analysis.CohortAnalysisConfiguration"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortAnalysisConfiguration</span></code></a>. The configuration options
can be customized or set with default values.</p>
<p>If an invalid option is passed, a warning is logged and the previous value is retained.
Consequently, it is <em>impossible</em> to mis-configure the analysis.</p>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfigurationBuilder.missing_implies_excluded">
<span class="sig-name descname"><span class="pre">missing_implies_excluded</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">missing_implies_excluded</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfigurationBuilder.missing_implies_excluded"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfigurationBuilder.missing_implies_excluded" title="Link to this definition"></a></dt>
<dd><p>Set <cite>missing_implies_excluded</cite> option.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfigurationBuilder.pval_correction">
<span class="sig-name descname"><span class="pre">pval_correction</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pval_correction</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfigurationBuilder.pval_correction"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfigurationBuilder.pval_correction" title="Link to this definition"></a></dt>
<dd><p>Set <cite>pval_correction</cite> option.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfigurationBuilder.min_perc_patients_w_hpo">
<span class="sig-name descname"><span class="pre">min_perc_patients_w_hpo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">min_perc_patients_w_hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfigurationBuilder.min_perc_patients_w_hpo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfigurationBuilder.min_perc_patients_w_hpo" title="Link to this definition"></a></dt>
<dd><p>Set <cite>min_perc_patients_w_hpo</cite> option.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfigurationBuilder.include_sv">
<span class="sig-name descname"><span class="pre">include_sv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">include_sv</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfigurationBuilder.include_sv"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfigurationBuilder.include_sv" title="Link to this definition"></a></dt>
<dd><p>Set <cite>include_sv</cite> option.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfigurationBuilder.build">
<span class="sig-name descname"><span class="pre">build</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.CohortAnalysisConfiguration" title="genophenocorr.analysis._config.CohortAnalysisConfiguration"><span class="pre">CohortAnalysisConfiguration</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfigurationBuilder.build"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfigurationBuilder.build" title="Link to this definition"></a></dt>
<dd><p>Build the configuration.</p>
</dd></dl>

</dd></dl>

<section id="subpackages">
<h2>Subpackages<a class="headerlink" href="#subpackages" title="Link to this heading"></a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="genophenocorr.analysis.predicate.html">genophenocorr.analysis.predicate package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PolyPredicate"><code class="docutils literal notranslate"><span class="pre">PolyPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory"><code class="docutils literal notranslate"><span class="pre">PatientCategory</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.BooleanPredicate"><code class="docutils literal notranslate"><span class="pre">BooleanPredicate</span></code></a></li>
</ul>
</li>
</ul>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="genophenocorr.html" class="btn btn-neutral float-left" title="genophenocorr package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="genophenocorr.analysis.predicate.html" class="btn btn-neutral float-right" title="genophenocorr.analysis.predicate package" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>