<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>genophenocorr.model package &mdash; genophenocorr 0.1.1dev documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/genophenocorr.css?v=547657ed" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=8427294d"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="genophenocorr.preprocessing package" href="genophenocorr.preprocessing.html" />
    <link rel="prev" title="genophenocorr.data package" href="genophenocorr.data.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            genophenocorr
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../user-guide/index.html">User guide</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">API reference</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="genophenocorr.html">genophenocorr package</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="genophenocorr.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.analysis.html">genophenocorr.analysis package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.constants.html">genophenocorr.constants package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.data.html">genophenocorr.data package</a></li>
<li class="toctree-l4 current"><a class="current reference internal" href="#">genophenocorr.model package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.preprocessing.html">genophenocorr.preprocessing package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.view.html">genophenocorr.view package</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">genophenocorr</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="modules.html">API reference</a></li>
          <li class="breadcrumb-item"><a href="genophenocorr.html">genophenocorr package</a></li>
      <li class="breadcrumb-item active">genophenocorr.model package</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/apidocs/genophenocorr.model.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-genophenocorr.model">
<span id="genophenocorr-model-package"></span><h1>genophenocorr.model package<a class="headerlink" href="#module-genophenocorr.model" title="Link to this heading"></a></h1>
<p>The <cite>genophenocorr.model</cite> package defines data model classes used in genophenocorr. We start with the top-level elements,
such as <a class="reference internal" href="#genophenocorr.model.Cohort" title="genophenocorr.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a> and <a class="reference internal" href="#genophenocorr.model.Patient" title="genophenocorr.model.Patient"><code class="xref py py-class docutils literal notranslate"><span class="pre">Patient</span></code></a>, and we follow with data classes for phenotype, genotype, transcript,
and protein info.</p>
<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">Cohort</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">members</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">excluded_member_count</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sized" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Sized</span></code></a></p>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.from_patients">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">from_patients</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">members</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">include_patients_with_no_HPO</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">include_patients_with_no_variants</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.from_patients"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.from_patients" title="Link to this definition"></a></dt>
<dd><p>Create a cohort from a sequence of patients.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.all_patients">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">all_patients</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Collection" title="(in Python v3.12)"><span class="pre">Collection</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.model.Cohort.all_patients" title="Link to this definition"></a></dt>
<dd><p>Get a collection of all patients in the cohort.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.all_phenotypes">
<span class="sig-name descname"><span class="pre">all_phenotypes</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Set" title="(in Python v3.12)"><span class="pre">Set</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Phenotype" title="genophenocorr.model._phenotype.Phenotype"><span class="pre">Phenotype</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.all_phenotypes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.all_phenotypes" title="Link to this definition"></a></dt>
<dd><p>Get a set of all phenotypes (observed or excluded) in the cohort members.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.all_diseases">
<span class="sig-name descname"><span class="pre">all_diseases</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Set" title="(in Python v3.12)"><span class="pre">Set</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Disease" title="genophenocorr.model._phenotype.Disease"><span class="pre">Disease</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.all_diseases"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.all_diseases" title="Link to this definition"></a></dt>
<dd><p>Get a set of all diseases (observed or excluded) in the cohort members.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.all_variants">
<span class="sig-name descname"><span class="pre">all_variants</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Set" title="(in Python v3.12)"><span class="pre">Set</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Variant" title="genophenocorr.model._variant.Variant"><span class="pre">Variant</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.all_variants"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.all_variants" title="Link to this definition"></a></dt>
<dd><p>Get a set of all variants observed in the cohort members.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.all_transcript_ids">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">all_transcript_ids</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Set" title="(in Python v3.12)"><span class="pre">Set</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.model.Cohort.all_transcript_ids" title="Link to this definition"></a></dt>
<dd><p>Get a set of all transcript IDs affected by the cohort variants.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.total_patient_count">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">total_patient_count</span></span><a class="headerlink" href="#genophenocorr.model.Cohort.total_patient_count" title="Link to this definition"></a></dt>
<dd><p>Get the total number of cohort members.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.get_patient_ids">
<span class="sig-name descname"><span class="pre">get_patient_ids</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Set" title="(in Python v3.12)"><span class="pre">Set</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.get_patient_ids"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.get_patient_ids" title="Link to this definition"></a></dt>
<dd><p>Get a set of the patient IDs.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.list_present_phenotypes">
<span class="sig-name descname"><span class="pre">list_present_phenotypes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">top</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.12)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.list_present_phenotypes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.list_present_phenotypes" title="Link to this definition"></a></dt>
<dd><p>Get a sequence with counts of HPO terms used as direct annotations of the cohort members.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>typing.Optional</strong><strong>[</strong><strong>int</strong><strong>]</strong> (<em>top</em>) – If not given, lists all present phenotypes.
Otherwise, lists only the <cite>top</cite> highest counts</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><dl class="simple">
<dt>A sequence of tuples, formatted (phenotype CURIE,</dt><dd><p>number of patients with that phenotype)</p>
</dd>
</dl>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><em>Sequence</em></a>[<a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.12)"><em>Tuple</em></a>[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)">str</a>, <a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)">int</a>]]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.list_all_diseases">
<span class="sig-name descname"><span class="pre">list_all_diseases</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">top</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.12)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.list_all_diseases"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.list_all_diseases" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.list_all_variants">
<span class="sig-name descname"><span class="pre">list_all_variants</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">top</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.12)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.list_all_variants"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.list_all_variants" title="Link to this definition"></a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>typing.Optional</strong><strong>[</strong><strong>int</strong><strong>]</strong> (<em>top</em>) – If not given, lists all variants. Otherwise, lists only the <cite>top</cite> highest counts</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A sequence of tuples, formatted (variant key, number of patients with that variant)</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#list" title="(in Python v3.12)">list</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.list_all_proteins">
<span class="sig-name descname"><span class="pre">list_all_proteins</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">top</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.12)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.list_all_proteins"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.list_all_proteins" title="Link to this definition"></a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>typing.Optional</strong><strong>[</strong><strong>int</strong><strong>]</strong> (<em>top</em>) – If not given, lists all proteins. Otherwise, lists only the <cite>top</cite> highest counts.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A list of tuples, formatted (protein ID string, the count of variants that affect the protein)</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#list" title="(in Python v3.12)">list</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.variant_effect_count_by_tx">
<span class="sig-name descname"><span class="pre">variant_effect_count_by_tx</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.variant_effect_count_by_tx"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.variant_effect_count_by_tx" title="Link to this definition"></a></dt>
<dd><p>Count variant effects for all transcripts or for a transcript <cite>tx_id</cite> of choice.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>tx_id</strong> – a <cite>str</cite> with transcript accession (e.g. <cite>NM_123456.5</cite>)
or <cite>None</cite> if all transcripts should be listed.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><dl class="simple">
<dt>Each transcript ID references a Counter(), with the variant effect as the key</dt><dd><p>and the count of variants with that effect on the transcript id.</p>
</dd>
</dl>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>mapping</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.get_excluded_count">
<span class="sig-name descname"><span class="pre">get_excluded_count</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.get_excluded_count"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.get_excluded_count" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Cohort.get_variant_by_key">
<span class="sig-name descname"><span class="pre">get_variant_by_key</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant_key</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.model.Variant" title="genophenocorr.model._variant.Variant"><span class="pre">Variant</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Cohort.get_variant_by_key"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Cohort.get_variant_by_key" title="Link to this definition"></a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.Patient">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">Patient</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">labels</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.SampleLabels" title="genophenocorr.model._base.SampleLabels"><span class="pre">SampleLabels</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">phenotypes</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Phenotype" title="genophenocorr.model._phenotype.Phenotype"><span class="pre">Phenotype</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">diseases</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Disease" title="genophenocorr.model._phenotype.Disease"><span class="pre">Disease</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variants</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Variant" title="genophenocorr.model._variant.Variant"><span class="pre">Variant</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Patient"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Patient" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>Patient</cite> represents a single investigated individual.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Patient.patient_id">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">patient_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.Patient.patient_id" title="Link to this definition"></a></dt>
<dd><p>Get a unique patient ID.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Patient.labels">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">labels</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.SampleLabels" title="genophenocorr.model._base.SampleLabels"><span class="pre">SampleLabels</span></a></em><a class="headerlink" href="#genophenocorr.model.Patient.labels" title="Link to this definition"></a></dt>
<dd><p>Get the sample identifiers.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Patient.phenotypes">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">phenotypes</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Phenotype" title="genophenocorr.model._phenotype.Phenotype"><span class="pre">Phenotype</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.model.Patient.phenotypes" title="Link to this definition"></a></dt>
<dd><p>Get the phenotypes observed and excluded in the patient.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Patient.diseases">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">diseases</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Disease" title="genophenocorr.model._phenotype.Disease"><span class="pre">Disease</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.model.Patient.diseases" title="Link to this definition"></a></dt>
<dd><p>Get the diseases the patient has (not) been diagnosed with.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Patient.variants">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variants</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Variant" title="genophenocorr.model._variant.Variant"><span class="pre">Variant</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.model.Patient.variants" title="Link to this definition"></a></dt>
<dd><p>Get a list of variants observed in the patient.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Patient.present_phenotypes">
<span class="sig-name descname"><span class="pre">present_phenotypes</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterator" title="(in Python v3.12)"><span class="pre">Iterator</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Phenotype" title="genophenocorr.model._phenotype.Phenotype"><span class="pre">Phenotype</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Patient.present_phenotypes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Patient.present_phenotypes" title="Link to this definition"></a></dt>
<dd><p>Get an iterator over the <em>present</em> phenotypes of the patient.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Patient.excluded_phenotypes">
<span class="sig-name descname"><span class="pre">excluded_phenotypes</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterator" title="(in Python v3.12)"><span class="pre">Iterator</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Phenotype" title="genophenocorr.model._phenotype.Phenotype"><span class="pre">Phenotype</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Patient.excluded_phenotypes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Patient.excluded_phenotypes" title="Link to this definition"></a></dt>
<dd><p>Get an iterator over the <em>excluded</em> phenotypes of the patient.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Patient.present_diseases">
<span class="sig-name descname"><span class="pre">present_diseases</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterator" title="(in Python v3.12)"><span class="pre">Iterator</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Disease" title="genophenocorr.model._phenotype.Disease"><span class="pre">Disease</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Patient.present_diseases"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Patient.present_diseases" title="Link to this definition"></a></dt>
<dd><p>Get an iterator with diseases the patient was diagnosed with.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Patient.excluded_diseases">
<span class="sig-name descname"><span class="pre">excluded_diseases</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterator" title="(in Python v3.12)"><span class="pre">Iterator</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Disease" title="genophenocorr.model._phenotype.Disease"><span class="pre">Disease</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_cohort.html#Patient.excluded_diseases"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Patient.excluded_diseases" title="Link to this definition"></a></dt>
<dd><p>Get an iterator with diseases whose presence was excluded in the patient.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.SampleLabels">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">SampleLabels</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">label</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">meta_label</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_base.html#SampleLabels"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.SampleLabels" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>A data model for subject identifiers.</p>
<p>The subject has a mandatory <a class="reference internal" href="#genophenocorr.model.SampleLabels.label" title="genophenocorr.model.SampleLabels.label"><code class="xref py py-attr docutils literal notranslate"><span class="pre">label</span></code></a> and an optional <a class="reference internal" href="#genophenocorr.model.SampleLabels.meta_label" title="genophenocorr.model.SampleLabels.meta_label"><code class="xref py py-attr docutils literal notranslate"><span class="pre">meta_label</span></code></a>.</p>
<p>The identifiers support natural ordering, equality tests, and are hashable.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.SampleLabels.label">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">label</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.SampleLabels.label" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.SampleLabels.meta_label">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">meta_label</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.SampleLabels.meta_label" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.SampleLabels.label_summary">
<span class="sig-name descname"><span class="pre">label_summary</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_base.html#SampleLabels.label_summary"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.SampleLabels.label_summary" title="Link to this definition"></a></dt>
<dd><p>Summarize <cite>label</cite> and <cite>meta_label</cite> into a <cite>str</cite> where the sub-parts are inserted as <code class="docutils literal notranslate"><span class="pre">&lt;label&gt;[&lt;meta_label&gt;]</span></code>.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.Phenotype">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">Phenotype</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">term_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_observed</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_phenotype.html#Phenotype"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Phenotype" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.Identified" title="(in hpo-toolkit)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Identified</span></code></a>, <a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.ObservableFeature" title="(in hpo-toolkit)"><code class="xref py py-class docutils literal notranslate"><span class="pre">ObservableFeature</span></code></a></p>
<p><cite>Phenotype</cite> represents a clinical sign or symptom represented as an HPO term.</p>
<p>The phenotype can be either present in the patient or excluded.</p>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Phenotype.from_term">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">from_term</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">term</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.MinimalTerm" title="(in hpo-toolkit)"><span class="pre">MinimalTerm</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_observed</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_phenotype.html#Phenotype.from_term"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Phenotype.from_term" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Phenotype.identifier">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">identifier</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a></em><a class="headerlink" href="#genophenocorr.model.Phenotype.identifier" title="Link to this definition"></a></dt>
<dd><p>Get the phenotype ID; an HPO term ID most of the time.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Phenotype.is_present">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">is_present</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></em><a class="headerlink" href="#genophenocorr.model.Phenotype.is_present" title="Link to this definition"></a></dt>
<dd><p>Return <cite>True</cite> if the phenotype feature was observed in the subject or <cite>False</cite> if the feature’s presence was explicitly excluded.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Phenotype.observed">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">observed</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.Phenotype.observed" title="Link to this definition"></a></dt>
<dd><p>Returns a boolean for whether the phenotype is observed.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>True if this phenotype was observed in the respective patient.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>boolean</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Phenotype.is_observed">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">is_observed</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></em><a class="headerlink" href="#genophenocorr.model.Phenotype.is_observed" title="Link to this definition"></a></dt>
<dd><p>Returns <cite>True</cite> if the phenotype was <em>present</em> in the respective patient.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.Disease">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">Disease</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">term_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_observed</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_phenotype.html#Disease"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Disease" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.Identified" title="(in hpo-toolkit)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Identified</span></code></a>, <a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.ObservableFeature" title="(in hpo-toolkit)"><code class="xref py py-class docutils literal notranslate"><span class="pre">ObservableFeature</span></code></a>, <a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.Named" title="(in hpo-toolkit)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Named</span></code></a></p>
<p>Representation of a disease diagnosed (or excluded) in an investigated individual.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Disease.identifier">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">identifier</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a></em><a class="headerlink" href="#genophenocorr.model.Disease.identifier" title="Link to this definition"></a></dt>
<dd><p>Get the disease ID.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Disease.name">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">name</span></span><a class="headerlink" href="#genophenocorr.model.Disease.name" title="Link to this definition"></a></dt>
<dd><p>Get the disease label (e.g. <cite>LEIGH SYNDROME, NUCLEAR; NULS</cite>).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Disease.is_present">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">is_present</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></em><a class="headerlink" href="#genophenocorr.model.Disease.is_present" title="Link to this definition"></a></dt>
<dd><p>Return <cite>True</cite> if the disease was diagnosed in the individual or <cite>False</cite> if it was excluded.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.Variant">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">Variant</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant_info</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.VariantInfo" title="genophenocorr.model._variant.VariantInfo"><span class="pre">VariantInfo</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_annotations</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.TranscriptAnnotation" title="genophenocorr.model._variant.TranscriptAnnotation"><span class="pre">TranscriptAnnotation</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genotypes</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.Genotypes" title="genophenocorr.model._gt.Genotypes"><span class="pre">Genotypes</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#Variant"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Variant" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#genophenocorr.model.VariantInfoAware" title="genophenocorr.model._variant.VariantInfoAware"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantInfoAware</span></code></a>, <code class="xref py py-class docutils literal notranslate"><span class="pre">FunctionalAnnotationAware</span></code>, <a class="reference internal" href="#genophenocorr.model.Genotyped" title="genophenocorr.model._gt.Genotyped"><code class="xref py py-class docutils literal notranslate"><span class="pre">Genotyped</span></code></a></p>
<dl class="simple">
<dt>Variant includes three lines of information:</dt><dd><ul class="simple">
<li><p>the variant data with coordinates or other info available for large imprecise SVs,</p></li>
<li><p>results of the functional annotation with respect to relevant transcripts, and</p></li>
<li><p>the genotypes for the known samples</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Variant.create_variant_from_scratch">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">create_variant_from_scratch</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant_coordinates</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.VariantCoordinates" title="genophenocorr.model._variant.VariantCoordinates"><span class="pre">VariantCoordinates</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">trans_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">hgvs_cdna</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_preferred</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">consequences</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">exons_effected</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">hgvsp</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_effect_start</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_effect_end</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">genotypes</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.Genotypes" title="genophenocorr.model._gt.Genotypes"><span class="pre">Genotypes</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#Variant.create_variant_from_scratch"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Variant.create_variant_from_scratch" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Variant.variant_info">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variant_info</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.VariantInfo" title="genophenocorr.model._variant.VariantInfo"><span class="pre">VariantInfo</span></a></em><a class="headerlink" href="#genophenocorr.model.Variant.variant_info" title="Link to this definition"></a></dt>
<dd><p>Returns:
VariantInfo: A representation of the variant data for sequence and symbolic variants, as well as for large imprecise SVs.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Variant.tx_annotations">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">tx_annotations</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.TranscriptAnnotation" title="genophenocorr.model._variant.TranscriptAnnotation"><span class="pre">TranscriptAnnotation</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.model.Variant.tx_annotations" title="Link to this definition"></a></dt>
<dd><p>A collection of TranscriptAnnotations that each represent results of the functional annotation
of a variant with respect to single transcript of a gene.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A sequence of TranscriptAnnotation objects</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Sequence[<a class="reference internal" href="#genophenocorr.model.TranscriptAnnotation" title="genophenocorr.model.TranscriptAnnotation">TranscriptAnnotation</a>]</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Variant.genotypes">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">genotypes</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.Genotypes" title="genophenocorr.model._gt.Genotypes"><span class="pre">Genotypes</span></a></em><a class="headerlink" href="#genophenocorr.model.Variant.genotypes" title="Link to this definition"></a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.VariantClass">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">VariantClass</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">value</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">module</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qualname</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boundary</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#VariantClass"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantClass" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/enum.html#enum.Enum" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Enum</span></code></a></p>
<p><cite>VariantClass</cite> represents a high-level variant category
which mostly corresponds to the structural variant categories
of the Variant Call Format specification,
but includes type for single nucleotide variants (SNV) and multi-nucleotide variant (MNV).</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantClass.DEL">
<span class="sig-name descname"><span class="pre">DEL</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">0</span></em><a class="headerlink" href="#genophenocorr.model.VariantClass.DEL" title="Link to this definition"></a></dt>
<dd><p>A deletion - a variant with a net loss of sequence from the alternative allele
regardless of its size.</p>
<p>Both a deletion of 1 base pair and a deletion of 1000 base pairs are acceptable.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantClass.DUP">
<span class="sig-name descname"><span class="pre">DUP</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">1</span></em><a class="headerlink" href="#genophenocorr.model.VariantClass.DUP" title="Link to this definition"></a></dt>
<dd><p>Duplication (tandem or interspersed).</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantClass.INS">
<span class="sig-name descname"><span class="pre">INS</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">2</span></em><a class="headerlink" href="#genophenocorr.model.VariantClass.INS" title="Link to this definition"></a></dt>
<dd><p>Insertion of a novel sequence.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantClass.INV">
<span class="sig-name descname"><span class="pre">INV</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">3</span></em><a class="headerlink" href="#genophenocorr.model.VariantClass.INV" title="Link to this definition"></a></dt>
<dd><p>Inversion of a chromosome segment.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantClass.MNV">
<span class="sig-name descname"><span class="pre">MNV</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">4</span></em><a class="headerlink" href="#genophenocorr.model.VariantClass.MNV" title="Link to this definition"></a></dt>
<dd><p>Multi-nucleotide variant (e.g. <cite>AG&gt;CT</cite>) that is not a duplication, deletion, or insertion.
May be called INDEL.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantClass.SNV">
<span class="sig-name descname"><span class="pre">SNV</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">5</span></em><a class="headerlink" href="#genophenocorr.model.VariantClass.SNV" title="Link to this definition"></a></dt>
<dd><p>Single nucleotide variant.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantClass.BND">
<span class="sig-name descname"><span class="pre">BND</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">6</span></em><a class="headerlink" href="#genophenocorr.model.VariantClass.BND" title="Link to this definition"></a></dt>
<dd><p>A breakend.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">VariantCoordinates</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">GenomicRegion</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">alt</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">change_length</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#VariantCoordinates"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantCoordinates" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>A representation of coordinates of sequence and symbolic variants.</p>
<p>Note, the breakend variants are not currently supported.</p>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.from_vcf_literal">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">from_vcf_literal</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contig</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Contig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">alt</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#VariantCoordinates.from_vcf_literal"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.from_vcf_literal" title="Link to this definition"></a></dt>
<dd><p>Create <cite>VariantCoordinates</cite> from a variant in VCF literal notation.</p>
<p>Note, this function must <em>not</em> be used to create a VCF symbolic variant
(e.g. <cite>&lt;DEL&gt;</cite> or translocation).
Use <a class="reference internal" href="#genophenocorr.model.VariantCoordinates.from_vcf_symbolic" title="genophenocorr.model.VariantCoordinates.from_vcf_symbolic"><code class="xref py py-func docutils literal notranslate"><span class="pre">from_vcf_symbolic()</span></code></a> instead.</p>
<p><strong>Example</strong></p>
<p>Create a variant from a VCF line:
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">#CHROM</span>&#160; <span class="pre">POS</span>&#160;&#160;&#160;&#160; <span class="pre">ID</span>&#160; <span class="pre">REF</span> <span class="pre">ALT</span> <span class="pre">...</span>
<span class="pre">chr1</span>&#160;&#160;&#160; <span class="pre">1001</span>&#160;&#160;&#160; <span class="pre">.</span>&#160;&#160; <span class="pre">C</span>&#160;&#160; <span class="pre">T</span>
<span class="pre">`</span></code></p>
<p>We first must decide on the genome build. Most of the time, we should use GRCh38:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.model.genome</span> <span class="kn">import</span> <span class="n">GRCh38</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">build</span> <span class="o">=</span> <span class="n">GRCh38</span>
</pre></div>
</div>
<p>Then, we access the contig for <code class="docutils literal notranslate"><span class="pre">'chr1'</span></code>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">chr1</span> <span class="o">=</span> <span class="n">build</span><span class="o">.</span><span class="n">contig_by_name</span><span class="p">(</span><span class="s1">&#39;chr1&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Last, we create the variant coordinates:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.model</span> <span class="kn">import</span> <span class="n">VariantCoordinates</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">vc</span> <span class="o">=</span> <span class="n">VariantCoordinates</span><span class="o">.</span><span class="n">from_vcf_literal</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">contig</span><span class="o">=</span><span class="n">chr1</span><span class="p">,</span> <span class="n">pos</span><span class="o">=</span><span class="mi">1001</span><span class="p">,</span> <span class="n">ref</span><span class="o">=</span><span class="s1">&#39;C&#39;</span><span class="p">,</span> <span class="n">alt</span><span class="o">=</span><span class="s1">&#39;T&#39;</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>Now can test the properties:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">vc</span><span class="o">.</span><span class="n">start</span><span class="p">,</span> <span class="n">vc</span><span class="o">.</span><span class="n">end</span><span class="p">,</span> <span class="n">vc</span><span class="o">.</span><span class="n">ref</span><span class="p">,</span> <span class="n">vc</span><span class="o">.</span><span class="n">alt</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">vc</span><span class="p">),</span> <span class="n">vc</span><span class="o">.</span><span class="n">change_length</span>
<span class="go">(1000, 1001, &#39;C&#39;, &#39;T&#39;, 1, 0)</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>contig</strong> – a <code class="xref py py-class docutils literal notranslate"><span class="pre">Contig</span></code> for the chromosome</p></li>
<li><p><strong>pos</strong> – a 1-based coordinate of the first base of the reference allele, as described in VCF standard</p></li>
<li><p><strong>ref</strong> – a <cite>str</cite> with the REF allele. Should meet the requirements of the VCF standard.</p></li>
<li><p><strong>alt</strong> – a <cite>str</cite> with the ALT allele. Should meet the requirements of the VCF standard.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.from_vcf_symbolic">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">from_vcf_symbolic</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">contig</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Contig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">end</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">alt</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">svlen</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#VariantCoordinates.from_vcf_symbolic"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.from_vcf_symbolic" title="Link to this definition"></a></dt>
<dd><p>Create <cite>VariantCoordinates</cite> from a variant in VCF symbolic notation.</p>
<p>Note, this function must <em>not</em> be used to create a VCF sequence/literal variant.
Use <a class="reference internal" href="#genophenocorr.model.VariantCoordinates.from_vcf_literal" title="genophenocorr.model.VariantCoordinates.from_vcf_literal"><code class="xref py py-func docutils literal notranslate"><span class="pre">from_vcf_literal()</span></code></a> instead.</p>
<p><strong>Example</strong></p>
<p>Let’s create a symbolic variant from the line:</p>
<p><code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">#CHROM</span>&#160;&#160; <span class="pre">POS</span>&#160;&#160;&#160;&#160;&#160; <span class="pre">ID</span>&#160;&#160; <span class="pre">REF</span>&#160;&#160; <span class="pre">ALT</span>&#160;&#160;&#160;&#160; <span class="pre">QUAL</span>&#160;&#160; <span class="pre">FILTER</span>&#160;&#160; <span class="pre">INFO</span>
<span class="pre">2</span>&#160;&#160;&#160;&#160;&#160;&#160;&#160; <span class="pre">321682</span>&#160;&#160; <span class="pre">.</span>&#160;&#160;&#160; <span class="pre">T</span>&#160;&#160;&#160;&#160; <span class="pre">&lt;DEL&gt;</span>&#160;&#160; <span class="pre">6</span>&#160;&#160;&#160;&#160;&#160; <span class="pre">PASS</span>&#160;&#160;&#160;&#160; <span class="pre">SVTYPE=DEL;END=321887;SVLEN=-205</span>
<span class="pre">`</span></code></p>
<p>We first must decide on the genome build. Most of the time, we should use GRCh38:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.model.genome</span> <span class="kn">import</span> <span class="n">GRCh38</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">contig</span> <span class="o">=</span> <span class="n">GRCh38</span><span class="o">.</span><span class="n">contig_by_name</span><span class="p">(</span><span class="s1">&#39;2&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Now, we create the coordinates as:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">vc</span> <span class="o">=</span> <span class="n">VariantCoordinates</span><span class="o">.</span><span class="n">from_vcf_symbolic</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">contig</span><span class="o">=</span><span class="n">contig</span><span class="p">,</span> <span class="n">pos</span><span class="o">=</span><span class="mi">321682</span><span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="mi">321887</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">ref</span><span class="o">=</span><span class="s1">&#39;T&#39;</span><span class="p">,</span> <span class="n">alt</span><span class="o">=</span><span class="s1">&#39;&lt;DEL&gt;&#39;</span><span class="p">,</span> <span class="n">svlen</span><span class="o">=-</span><span class="mi">205</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>Now can test the properties:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">vc</span><span class="o">.</span><span class="n">start</span><span class="p">,</span> <span class="n">vc</span><span class="o">.</span><span class="n">end</span><span class="p">,</span> <span class="n">vc</span><span class="o">.</span><span class="n">ref</span><span class="p">,</span> <span class="n">vc</span><span class="o">.</span><span class="n">alt</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">vc</span><span class="p">),</span> <span class="n">vc</span><span class="o">.</span><span class="n">change_length</span>
<span class="go">(321681, 321887, &#39;T&#39;, &#39;&lt;DEL&gt;&#39;, 206, -205)</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>contig</strong> – a <code class="xref py py-class docutils literal notranslate"><span class="pre">Contig</span></code> for the chromosome</p></li>
<li><p><strong>pos</strong> – a 1-based coordinate of the first base of the affected reference allele region</p></li>
<li><p><strong>end</strong> – a 1-based coordinate of the last base of the affected reference allele region</p></li>
<li><p><strong>ref</strong> – a <cite>str</cite> with the REF allele. Most of the time, it is one of <cite>{‘N’, ‘A’, ‘C’, ‘G’, ‘T’}</cite></p></li>
<li><p><strong>alt</strong> – a <cite>str</cite> with the ALT allele, e.g. one of <cite>{‘&lt;DEL&gt;’, ‘&lt;DUP&gt;’, ‘&lt;INS&gt;’, ‘&lt;INV&gt;’}</cite></p></li>
<li><p><strong>svlen</strong> – an <cite>int</cite> with change length (the difference between ref and alt allele lengths)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.chrom">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">chrom</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.chrom" title="Link to this definition"></a></dt>
<dd><p>Get the label of the chromosome/contig where the variant is located.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.start">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">start</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.start" title="Link to this definition"></a></dt>
<dd><p>Get the 0-based start coordinate (excluded) of the first base of the <a class="reference internal" href="#genophenocorr.model.VariantCoordinates.ref" title="genophenocorr.model.VariantCoordinates.ref"><code class="xref py py-attr docutils literal notranslate"><span class="pre">ref</span></code></a> allele.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.end">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">end</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.end" title="Link to this definition"></a></dt>
<dd><p>Get the 0-based end coordinate (included) of the last base of the <a class="reference internal" href="#genophenocorr.model.VariantCoordinates.ref" title="genophenocorr.model.VariantCoordinates.ref"><code class="xref py py-attr docutils literal notranslate"><span class="pre">ref</span></code></a> allele.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.region">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">region</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">GenomicRegion</span></em><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.region" title="Link to this definition"></a></dt>
<dd><p>Get the genomic region spanned by the <a class="reference internal" href="#genophenocorr.model.VariantCoordinates.ref" title="genophenocorr.model.VariantCoordinates.ref"><code class="xref py py-attr docutils literal notranslate"><span class="pre">ref</span></code></a> allele.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.ref">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">ref</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.ref" title="Link to this definition"></a></dt>
<dd><p>Get the reference allele (e.g. “A”, “CCT”, “N”). The allele may be an empty string.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.alt">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">alt</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.alt" title="Link to this definition"></a></dt>
<dd><p>Get the alternate allele (e.g. “A”, “GG”, “&lt;DEL&gt;”).</p>
<p>The allele may be an empty string for sequence variants.
The symbolic alternate allele follow the VCF notation and use the <cite>&lt;</cite> and <cite>&gt;</cite> characters
(e.g. “&lt;DEL&gt;”, “&lt;INS:ME:SINE&gt;”).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.change_length">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">change_length</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.change_length" title="Link to this definition"></a></dt>
<dd><p>Get the change of length between the <cite>ref</cite> and <cite>alt</cite> alleles due to the variant presence.</p>
<p>SNVs lead to change length of zero, deletions and insertions/duplications lead to negative
and positive change lengths, respectively.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.variant_key">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variant_key</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.variant_key" title="Link to this definition"></a></dt>
<dd><p>Get a readable representation of the variant’s coordinates.</p>
<p>For instance, <code class="docutils literal notranslate"><span class="pre">X_12345_12345_C_G</span></code> for a sequence variant or <code class="docutils literal notranslate"><span class="pre">22_10001_20000_INV</span></code> for a symbolic variant.
If the key is larger than 50 characters, the ‘ref’ and/or ‘alt’ (if over 10 bps) are changed to just show number of bps.
Example: <code class="docutils literal notranslate"><span class="pre">X_1000001_1000027_TAAAAAAAAAAAAAAAAAAAAAAAAAA_T</span></code> -&gt; <code class="docutils literal notranslate"><span class="pre">X_1000001_1000027_--27bp--_T</span></code></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Both <em>start</em> and <em>end</em> coordinates use 1-based (included) coordinate system.</p>
</div>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.variant_class">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variant_class</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.VariantClass" title="genophenocorr.model._variant.VariantClass"><span class="pre">VariantClass</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.variant_class" title="Link to this definition"></a></dt>
<dd><p>Get a <a class="reference internal" href="#genophenocorr.model.VariantClass" title="genophenocorr.model.VariantClass"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantClass</span></code></a> category.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.VariantCoordinates.is_structural">
<span class="sig-name descname"><span class="pre">is_structural</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#VariantCoordinates.is_structural"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantCoordinates.is_structural" title="Link to this definition"></a></dt>
<dd><p>Checks if the variant coordinates use structural variant notation as described by Variant Call Format
(<a class="reference external" href="https://en.wikipedia.org/wiki/Variant_Call_Format">VCF</a>).</p>
<p>Ane example of <em>structural</em> variant notation:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">chr5</span>  <span class="mi">101</span> <span class="o">.</span> <span class="n">N</span> <span class="o">&lt;</span><span class="n">DEL</span><span class="o">&gt;</span> <span class="o">.</span>  <span class="o">.</span>  <span class="n">SVTYPE</span><span class="o">=</span><span class="n">DEL</span><span class="p">;</span><span class="n">END</span><span class="o">=</span><span class="mi">120</span><span class="p">;</span><span class="n">SVLEN</span><span class="o">=-</span><span class="mi">10</span>
</pre></div>
</div>
<p>as opposed to the <em>sequence</em> (literal) notation:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">chr5</span>  <span class="mi">101</span> <span class="o">.</span> <span class="n">NACGTACGTAC</span> <span class="n">N</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><cite>True</cite> if the variant coordinates use structural variant notation.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.ImpreciseSvInfo">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">ImpreciseSvInfo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">structural_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_class</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.VariantClass" title="genophenocorr.model._variant.VariantClass"><span class="pre">VariantClass</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_symbol</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#ImpreciseSvInfo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.ImpreciseSvInfo" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>Data regarding a structural variant (SV) with imprecise breakpoint coordinates.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ImpreciseSvInfo.structural_type">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">structural_type</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a></em><a class="headerlink" href="#genophenocorr.model.ImpreciseSvInfo.structural_type" title="Link to this definition"></a></dt>
<dd><p>Get term ID of the structural type (e.g. <code class="docutils literal notranslate"><span class="pre">SO:1000029</span></code> for chromosomal deletion).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ImpreciseSvInfo.variant_class">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variant_class</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.VariantClass" title="genophenocorr.model._variant.VariantClass"><span class="pre">VariantClass</span></a></em><a class="headerlink" href="#genophenocorr.model.ImpreciseSvInfo.variant_class" title="Link to this definition"></a></dt>
<dd><p>Get a <a class="reference internal" href="#genophenocorr.model.VariantClass" title="genophenocorr.model.VariantClass"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantClass</span></code></a> category.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ImpreciseSvInfo.gene_id">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">gene_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.ImpreciseSvInfo.gene_id" title="Link to this definition"></a></dt>
<dd><p>Get a <cite>str</cite> with gene identifier CURIE (e.g. <code class="docutils literal notranslate"><span class="pre">HGNC:3603</span></code>) or <cite>None</cite> if the identifier is not available.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ImpreciseSvInfo.gene_symbol">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">gene_symbol</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.ImpreciseSvInfo.gene_symbol" title="Link to this definition"></a></dt>
<dd><p>Get a <cite>str</cite> with HGVS gene symbol (e.g. <em>FBN1</em>) or <cite>None</cite> if the symbol is not available.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ImpreciseSvInfo.variant_key">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variant_key</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.ImpreciseSvInfo.variant_key" title="Link to this definition"></a></dt>
<dd><p>Get a readable representation of the variant.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.VariantInfo">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">VariantInfo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant_coordinates</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.VariantCoordinates" title="genophenocorr.model._variant.VariantCoordinates"><span class="pre">VariantCoordinates</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sv_info</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.ImpreciseSvInfo" title="genophenocorr.model._variant.ImpreciseSvInfo"><span class="pre">ImpreciseSvInfo</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#VariantInfo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantInfo" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>VariantInfo</cite> consists of either variant coordinates or imprecise SV data.</p>
<p>The class is conceptually similar to Rust enum - only one of the fields can be set at any point in time.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantInfo.variant_coordinates">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variant_coordinates</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.VariantCoordinates" title="genophenocorr.model._variant.VariantCoordinates"><span class="pre">VariantCoordinates</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantInfo.variant_coordinates" title="Link to this definition"></a></dt>
<dd><p>Get variant coordinates if available.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.VariantInfo.has_variant_coordinates">
<span class="sig-name descname"><span class="pre">has_variant_coordinates</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#VariantInfo.has_variant_coordinates"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantInfo.has_variant_coordinates" title="Link to this definition"></a></dt>
<dd><p>Returns <cite>True</cite> if the variant coordinates are available.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantInfo.sv_info">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">sv_info</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.ImpreciseSvInfo" title="genophenocorr.model._variant.ImpreciseSvInfo"><span class="pre">ImpreciseSvInfo</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantInfo.sv_info" title="Link to this definition"></a></dt>
<dd><p>Get information about large imprecise SV.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.VariantInfo.has_sv_info">
<span class="sig-name descname"><span class="pre">has_sv_info</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#VariantInfo.has_sv_info"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantInfo.has_sv_info" title="Link to this definition"></a></dt>
<dd><p>Returns <cite>True</cite> if the variant is a large imprecise SV and the exact coordinates are thus unavailable.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantInfo.variant_key">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variant_key</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantInfo.variant_key" title="Link to this definition"></a></dt>
<dd><p>Get a readable representation of the variant’s coordinates or the large SV info.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantInfo.variant_class">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variant_class</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.VariantClass" title="genophenocorr.model._variant.VariantClass"><span class="pre">VariantClass</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantInfo.variant_class" title="Link to this definition"></a></dt>
<dd><p>Get a <a class="reference internal" href="#genophenocorr.model.VariantClass" title="genophenocorr.model.VariantClass"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantClass</span></code></a> category.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.VariantInfo.is_structural">
<span class="sig-name descname"><span class="pre">is_structural</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#VariantInfo.is_structural"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantInfo.is_structural" title="Link to this definition"></a></dt>
<dd><p>Test if the variant is a structural variant.</p>
<p>This can either be because the variant coordinates
use the structural variant notation (see <a class="reference internal" href="#genophenocorr.model.VariantCoordinates.is_structural" title="genophenocorr.model.VariantCoordinates.is_structural"><code class="xref py py-meth docutils literal notranslate"><span class="pre">VariantCoordinates.is_structural()</span></code></a>)
or if the variant is large imprecise SV.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.VariantInfoAware">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">VariantInfoAware</span></span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#VariantInfoAware"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantInfoAware" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>An entity where <a class="reference internal" href="#genophenocorr.model.VariantInfo" title="genophenocorr.model.VariantInfo"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantInfo</span></code></a> is available.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantInfoAware.variant_info">
<em class="property"><span class="pre">abstract</span><span class="w"> </span><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variant_info</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.VariantInfo" title="genophenocorr.model._variant.VariantInfo"><span class="pre">VariantInfo</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantInfoAware.variant_info" title="Link to this definition"></a></dt>
<dd><p>Get the variant data with coordinates or other info available for large imprecise SVs.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.Genotype">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">Genotype</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">value</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">module</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qualname</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boundary</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_gt.html#Genotype"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Genotype" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/enum.html#enum.Enum" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Enum</span></code></a></p>
<p><cite>Genotype</cite> represents state of a variable locus in a diploid genome.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.Genotype.NO_CALL">
<span class="sig-name descname"><span class="pre">NO_CALL</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">('.',)</span></em><a class="headerlink" href="#genophenocorr.model.Genotype.NO_CALL" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.Genotype.HOMOZYGOUS_REFERENCE">
<span class="sig-name descname"><span class="pre">HOMOZYGOUS_REFERENCE</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">('0/0',)</span></em><a class="headerlink" href="#genophenocorr.model.Genotype.HOMOZYGOUS_REFERENCE" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.Genotype.HETEROZYGOUS">
<span class="sig-name descname"><span class="pre">HETEROZYGOUS</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">('0/1',)</span></em><a class="headerlink" href="#genophenocorr.model.Genotype.HETEROZYGOUS" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.Genotype.HOMOZYGOUS_ALTERNATE">
<span class="sig-name descname"><span class="pre">HOMOZYGOUS_ALTERNATE</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">('1/1',)</span></em><a class="headerlink" href="#genophenocorr.model.Genotype.HOMOZYGOUS_ALTERNATE" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.Genotype.HEMIZYGOUS">
<span class="sig-name descname"><span class="pre">HEMIZYGOUS</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">('1',)</span></em><a class="headerlink" href="#genophenocorr.model.Genotype.HEMIZYGOUS" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Genotype.code">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">code</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.Genotype.code" title="Link to this definition"></a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.Genotypes">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">Genotypes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">samples</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.SampleLabels" title="genophenocorr.model._base.SampleLabels"><span class="pre">SampleLabels</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genotypes</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.Genotype" title="genophenocorr.model._gt.Genotype"><span class="pre">Genotype</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_gt.html#Genotypes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Genotypes" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sized" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Sized</span></code></a>, <a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Iterable</span></code></a></p>
<p><cite>Genotypes</cite> is a container for mapping between sample ID and its genotype.</p>
<p>Let’s consider a pair of samples:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">a</span> <span class="o">=</span> <span class="n">SampleLabels</span><span class="p">(</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span> <span class="o">=</span> <span class="n">SampleLabels</span><span class="p">(</span><span class="s1">&#39;B&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>We can use one of the static methods to create an instance. Either a single genotype:
&gt;&gt;&gt; gt = Genotypes.single(a, Genotype.HETEROZYGOUS)</p>
<p>Or genotypes of several samples:
&gt;&gt;&gt; gts = Genotypes.from_mapping({a: Genotype.HETEROZYGOUS, b: Genotype.HOMOZYGOUS_ALTERNATE})</p>
<p>There are 2 genotypes in the container:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">gts</span><span class="p">)</span>
<span class="go">2</span>
</pre></div>
</div>
<p>You can get a genotype for a sample ID:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">g</span> <span class="o">=</span> <span class="n">gts</span><span class="o">.</span><span class="n">for_sample</span><span class="p">(</span><span class="n">a</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">g</span><span class="o">.</span><span class="n">code</span>
<span class="go">&#39;0/1&#39;</span>
</pre></div>
</div>
<p>You will get <cite>None</cite> if the sample is not present:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">gts</span><span class="o">.</span><span class="n">for_sample</span><span class="p">(</span><span class="n">SampleLabels</span><span class="p">(</span><span class="s1">&#39;UNKNOWN&#39;</span><span class="p">))</span>
</pre></div>
</div>
<p>You can iterate over sample-genotype pairs:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="k">for</span> <span class="n">sample_id</span><span class="p">,</span> <span class="n">genotype</span> <span class="ow">in</span> <span class="n">gts</span><span class="p">:</span>
<span class="gp">... </span>  <span class="nb">print</span><span class="p">(</span><span class="n">sample_id</span><span class="p">,</span> <span class="n">genotype</span><span class="p">)</span>
<span class="go">A 0/1</span>
<span class="go">B 1/1</span>
</pre></div>
</div>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Genotypes.empty">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">empty</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_gt.html#Genotypes.empty"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Genotypes.empty" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Genotypes.single">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">single</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.SampleLabels" title="genophenocorr.model._base.SampleLabels"><span class="pre">SampleLabels</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">genotype</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.Genotype" title="genophenocorr.model._gt.Genotype"><span class="pre">Genotype</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_gt.html#Genotypes.single"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Genotypes.single" title="Link to this definition"></a></dt>
<dd><p>A shortcut for creating <cite>Genotypes</cite> for a single sample:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">a</span> <span class="o">=</span> <span class="n">SampleLabels</span><span class="p">(</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gts</span> <span class="o">=</span> <span class="n">Genotypes</span><span class="o">.</span><span class="n">single</span><span class="p">(</span><span class="n">a</span><span class="p">,</span> <span class="n">Genotype</span><span class="o">.</span><span class="n">HOMOZYGOUS_ALTERNATE</span><span class="p">)</span>
</pre></div>
</div>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="k">assert</span> <span class="nb">len</span><span class="p">(</span><span class="n">gts</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">assert</span> <span class="n">gts</span><span class="o">.</span><span class="n">for_sample</span><span class="p">(</span><span class="n">a</span><span class="p">)</span> <span class="o">==</span> <span class="n">Genotype</span><span class="o">.</span><span class="n">HOMOZYGOUS_ALTERNATE</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Genotypes.from_mapping">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">from_mapping</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mapping</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.SampleLabels" title="genophenocorr.model._base.SampleLabels"><span class="pre">SampleLabels</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.Genotype" title="genophenocorr.model._gt.Genotype"><span class="pre">Genotype</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_gt.html#Genotypes.from_mapping"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Genotypes.from_mapping" title="Link to this definition"></a></dt>
<dd><p>Create <cite>Genotypes</cite> from mapping between sample IDs and genotypes.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">a</span> <span class="o">=</span> <span class="n">SampleLabels</span><span class="p">(</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span> <span class="o">=</span> <span class="n">SampleLabels</span><span class="p">(</span><span class="s1">&#39;B&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gts</span> <span class="o">=</span> <span class="n">Genotypes</span><span class="o">.</span><span class="n">from_mapping</span><span class="p">({</span><span class="n">a</span><span class="p">:</span> <span class="n">Genotype</span><span class="o">.</span><span class="n">HETEROZYGOUS</span><span class="p">,</span> <span class="n">b</span><span class="p">:</span> <span class="n">Genotype</span><span class="o">.</span><span class="n">HOMOZYGOUS_ALTERNATE</span><span class="p">})</span>
</pre></div>
</div>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="k">assert</span> <span class="nb">len</span><span class="p">(</span><span class="n">gts</span><span class="p">)</span> <span class="o">==</span> <span class="mi">2</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Genotypes.for_sample">
<span class="sig-name descname"><span class="pre">for_sample</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.SampleLabels" title="genophenocorr.model._base.SampleLabels"><span class="pre">SampleLabels</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.model.Genotype" title="genophenocorr.model._gt.Genotype"><span class="pre">Genotype</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_gt.html#Genotypes.for_sample"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Genotypes.for_sample" title="Link to this definition"></a></dt>
<dd><p>Get a genotype for a sample or <cite>None</cite> if the genotype is not present.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>sample_id</strong> – a <a class="reference internal" href="#genophenocorr.model.SampleLabels" title="genophenocorr.model.SampleLabels"><code class="xref py py-class docutils literal notranslate"><span class="pre">SampleLabels</span></code></a> with sample’s identifier.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.Genotyped">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">Genotyped</span></span><a class="reference internal" href="../_modules/genophenocorr/model/_gt.html#Genotyped"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Genotyped" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>Genotyped</cite> entities</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.Genotyped.genotypes">
<em class="property"><span class="pre">abstract</span><span class="w"> </span><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">genotypes</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.Genotypes" title="genophenocorr.model._gt.Genotypes"><span class="pre">Genotypes</span></a></em><a class="headerlink" href="#genophenocorr.model.Genotyped.genotypes" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.Genotyped.genotype_for_sample">
<span class="sig-name descname"><span class="pre">genotype_for_sample</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.SampleLabels" title="genophenocorr.model._base.SampleLabels"><span class="pre">SampleLabels</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.model.Genotype" title="genophenocorr.model._gt.Genotype"><span class="pre">Genotype</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_gt.html#Genotyped.genotype_for_sample"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.Genotyped.genotype_for_sample" title="Link to this definition"></a></dt>
<dd><p>Get a genotype for a sample or <cite>None</cite> if the genotype is not present.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>sample_id</strong> – a <a class="reference internal" href="#genophenocorr.model.SampleLabels" title="genophenocorr.model.SampleLabels"><code class="xref py py-class docutils literal notranslate"><span class="pre">SampleLabels</span></code></a> with sample’s identifier.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptAnnotation">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">TranscriptAnnotation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gene_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">hgvs_cdna</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_preferred</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_effects</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">affected_exons</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">hgvsp</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_effect_coordinates</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Region</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#TranscriptAnnotation"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.TranscriptAnnotation" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#genophenocorr.model.TranscriptInfoAware" title="genophenocorr.model._variant.TranscriptInfoAware"><code class="xref py py-class docutils literal notranslate"><span class="pre">TranscriptInfoAware</span></code></a></p>
<p><cite>TranscriptAnnotation</cite> represent a result of the functional annotation of a variant
with respect to single transcript of a gene.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptAnnotation.gene_id">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">gene_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptAnnotation.gene_id" title="Link to this definition"></a></dt>
<dd><p>Get the HGVS symbol of the affected gene (e.g. <em>SURF1</em>).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptAnnotation.transcript_id">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">transcript_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptAnnotation.transcript_id" title="Link to this definition"></a></dt>
<dd><p>Get the RefSeq identifier of the transcript (e.g. <cite>NM_123456.7</cite>).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptAnnotation.is_preferred">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">is_preferred</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptAnnotation.is_preferred" title="Link to this definition"></a></dt>
<dd><p>Return <cite>True</cite> if the transcript is the preferred transcript of a gene,
such as MANE transcript, canonical Ensembl transcript.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptAnnotation.hgvs_cdna">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">hgvs_cdna</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptAnnotation.hgvs_cdna" title="Link to this definition"></a></dt>
<dd><p>Get the HGVS description of the sequence variant (e.g. <code class="docutils literal notranslate"><span class="pre">NM_123456.7:c.9876G&gt;T</span></code>)
or <cite>None</cite> if not available.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptAnnotation.variant_effects">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">variant_effects</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.model.TranscriptAnnotation.variant_effects" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of the predicted functional variant effects.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptAnnotation.overlapping_exons">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">overlapping_exons</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptAnnotation.overlapping_exons" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of 1-based exon indices (the index of the 1st exon is <cite>1</cite>)
that overlap with the variant.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptAnnotation.protein_id">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">protein_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptAnnotation.protein_id" title="Link to this definition"></a></dt>
<dd><p>Get the ID of the protein encoded by the <a class="reference internal" href="#genophenocorr.model.TranscriptAnnotation.transcript_id" title="genophenocorr.model.TranscriptAnnotation.transcript_id"><code class="xref py py-attr docutils literal notranslate"><span class="pre">transcript_id</span></code></a>
or <cite>None</cite> if the transcript is not protein-coding.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptAnnotation.hgvsp">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">hgvsp</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptAnnotation.hgvsp" title="Link to this definition"></a></dt>
<dd><p>Get the HGVS description of the protein sequence variant (e.g. <code class="docutils literal notranslate"><span class="pre">NP_001027559.1:p.Gln421Ter</span></code>)
or <cite>None</cite> if not available.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptAnnotation.protein_effect_location">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">protein_effect_location</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Region</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptAnnotation.protein_effect_location" title="Link to this definition"></a></dt>
<dd><p>Get the <code class="xref py py-class docutils literal notranslate"><span class="pre">Region</span></code> with start and end amino-acid coordinates
affected by the variant.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">VariantEffect</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">value</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">module</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qualname</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boundary</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_variant_effects.html#VariantEffect"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.VariantEffect" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/enum.html#enum.Enum" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Enum</span></code></a></p>
<p><cite>VariantEffect</cite> represents consequences of a variant on transcript that are supported by Genophenocorr.</p>
<div class="highlight-pycon notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">missense</span> <span class="o">=</span> <span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">missense</span><span class="p">)</span>
<span class="go">missense_variant</span>
</pre></div>
</div>
<p>The <cite>VariantEffect</cite> has a <a class="reference internal" href="#genophenocorr.model.VariantEffect.curie" title="genophenocorr.model.VariantEffect.curie"><code class="xref py py-attr docutils literal notranslate"><span class="pre">curie</span></code></a> attribute that represents the ontology class from
<a class="reference external" href="http://www.sequenceontology.org/">Sequence Ontology</a>.</p>
<div class="highlight-pycon notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">missense</span><span class="o">.</span><span class="n">curie</span>
<span class="go">&#39;SO:0001583&#39;</span>
</pre></div>
</div>
<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.TRANSCRIPT_ABLATION">
<span class="sig-name descname"><span class="pre">TRANSCRIPT_ABLATION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001893'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.TRANSCRIPT_ABLATION" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.SPLICE_ACCEPTOR_VARIANT">
<span class="sig-name descname"><span class="pre">SPLICE_ACCEPTOR_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001574'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.SPLICE_ACCEPTOR_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.SPLICE_DONOR_VARIANT">
<span class="sig-name descname"><span class="pre">SPLICE_DONOR_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001575'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.SPLICE_DONOR_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.STOP_GAINED">
<span class="sig-name descname"><span class="pre">STOP_GAINED</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001587'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.STOP_GAINED" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.FRAMESHIFT_VARIANT">
<span class="sig-name descname"><span class="pre">FRAMESHIFT_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001589'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.FRAMESHIFT_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.STOP_LOST">
<span class="sig-name descname"><span class="pre">STOP_LOST</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001578'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.STOP_LOST" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.START_LOST">
<span class="sig-name descname"><span class="pre">START_LOST</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0002012'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.START_LOST" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.TRANSCRIPT_AMPLIFICATION">
<span class="sig-name descname"><span class="pre">TRANSCRIPT_AMPLIFICATION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001889'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.TRANSCRIPT_AMPLIFICATION" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.INFRAME_INSERTION">
<span class="sig-name descname"><span class="pre">INFRAME_INSERTION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001821'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.INFRAME_INSERTION" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.INFRAME_DELETION">
<span class="sig-name descname"><span class="pre">INFRAME_DELETION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001822'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.INFRAME_DELETION" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.MISSENSE_VARIANT">
<span class="sig-name descname"><span class="pre">MISSENSE_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001583'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.MISSENSE_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.PROTEIN_ALTERING_VARIANT">
<span class="sig-name descname"><span class="pre">PROTEIN_ALTERING_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001818'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.PROTEIN_ALTERING_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.SPLICE_REGION_VARIANT">
<span class="sig-name descname"><span class="pre">SPLICE_REGION_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001630'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.SPLICE_REGION_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.SPLICE_DONOR_5TH_BASE_VARIANT">
<span class="sig-name descname"><span class="pre">SPLICE_DONOR_5TH_BASE_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001787'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.SPLICE_DONOR_5TH_BASE_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.SPLICE_DONOR_REGION_VARIANT">
<span class="sig-name descname"><span class="pre">SPLICE_DONOR_REGION_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0002170'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.SPLICE_DONOR_REGION_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.SPLICE_POLYPYRIMIDINE_TRACT_VARIANT">
<span class="sig-name descname"><span class="pre">SPLICE_POLYPYRIMIDINE_TRACT_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0002169'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.SPLICE_POLYPYRIMIDINE_TRACT_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.INCOMPLETE_TERMINAL_CODON_VARIANT">
<span class="sig-name descname"><span class="pre">INCOMPLETE_TERMINAL_CODON_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001626'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.INCOMPLETE_TERMINAL_CODON_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.START_RETAINED_VARIANT">
<span class="sig-name descname"><span class="pre">START_RETAINED_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0002019'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.START_RETAINED_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.STOP_RETAINED_VARIANT">
<span class="sig-name descname"><span class="pre">STOP_RETAINED_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001567'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.STOP_RETAINED_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.SYNONYMOUS_VARIANT">
<span class="sig-name descname"><span class="pre">SYNONYMOUS_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001819'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.SYNONYMOUS_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.CODING_SEQUENCE_VARIANT">
<span class="sig-name descname"><span class="pre">CODING_SEQUENCE_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001580'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.CODING_SEQUENCE_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.MATURE_MIRNA_VARIANT">
<span class="sig-name descname"><span class="pre">MATURE_MIRNA_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001620'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.MATURE_MIRNA_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.FIVE_PRIME_UTR_VARIANT">
<span class="sig-name descname"><span class="pre">FIVE_PRIME_UTR_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001623'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.FIVE_PRIME_UTR_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.THREE_PRIME_UTR_VARIANT">
<span class="sig-name descname"><span class="pre">THREE_PRIME_UTR_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001624'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.THREE_PRIME_UTR_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.NON_CODING_TRANSCRIPT_EXON_VARIANT">
<span class="sig-name descname"><span class="pre">NON_CODING_TRANSCRIPT_EXON_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001792'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.NON_CODING_TRANSCRIPT_EXON_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.INTRON_VARIANT">
<span class="sig-name descname"><span class="pre">INTRON_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001627'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.INTRON_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.NMD_TRANSCRIPT_VARIANT">
<span class="sig-name descname"><span class="pre">NMD_TRANSCRIPT_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">('SO:0001621',)</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.NMD_TRANSCRIPT_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.NON_CODING_TRANSCRIPT_VARIANT">
<span class="sig-name descname"><span class="pre">NON_CODING_TRANSCRIPT_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001619'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.NON_CODING_TRANSCRIPT_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.UPSTREAM_GENE_VARIANT">
<span class="sig-name descname"><span class="pre">UPSTREAM_GENE_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001631'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.UPSTREAM_GENE_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.DOWNSTREAM_GENE_VARIANT">
<span class="sig-name descname"><span class="pre">DOWNSTREAM_GENE_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001632'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.DOWNSTREAM_GENE_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.TFBS_ABLATION">
<span class="sig-name descname"><span class="pre">TFBS_ABLATION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001895'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.TFBS_ABLATION" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.TFBS_AMPLIFICATION">
<span class="sig-name descname"><span class="pre">TFBS_AMPLIFICATION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001892'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.TFBS_AMPLIFICATION" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.TF_BINDING_SITE_VARIANT">
<span class="sig-name descname"><span class="pre">TF_BINDING_SITE_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001782'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.TF_BINDING_SITE_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.REGULATORY_REGION_ABLATION">
<span class="sig-name descname"><span class="pre">REGULATORY_REGION_ABLATION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001894'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.REGULATORY_REGION_ABLATION" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.REGULATORY_REGION_AMPLIFICATION">
<span class="sig-name descname"><span class="pre">REGULATORY_REGION_AMPLIFICATION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001891'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.REGULATORY_REGION_AMPLIFICATION" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.FEATURE_ELONGATION">
<span class="sig-name descname"><span class="pre">FEATURE_ELONGATION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001907'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.FEATURE_ELONGATION" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.REGULATORY_REGION_VARIANT">
<span class="sig-name descname"><span class="pre">REGULATORY_REGION_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001566'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.REGULATORY_REGION_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.FEATURE_TRUNCATION">
<span class="sig-name descname"><span class="pre">FEATURE_TRUNCATION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001906'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.FEATURE_TRUNCATION" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.INTERGENIC_VARIANT">
<span class="sig-name descname"><span class="pre">INTERGENIC_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001628'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.INTERGENIC_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.SEQUENCE_VARIANT">
<span class="sig-name descname"><span class="pre">SEQUENCE_VARIANT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'SO:0001060'</span></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.SEQUENCE_VARIANT" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.VariantEffect.curie">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">curie</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.VariantEffect.curie" title="Link to this definition"></a></dt>
<dd><p>Get a compact URI (CURIE) of the variant effect
(e.g. <cite>SO:0001583</cite> for a missense variant).</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptInfoAware">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">TranscriptInfoAware</span></span><a class="reference internal" href="../_modules/genophenocorr/model/_variant.html#TranscriptInfoAware"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.TranscriptInfoAware" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>The implementors know about basic gene/transcript identifiers.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptInfoAware.gene_id">
<em class="property"><span class="pre">abstract</span><span class="w"> </span><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">gene_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptInfoAware.gene_id" title="Link to this definition"></a></dt>
<dd><p>Returns:
string: The gene symbol (e.g. SURF1)</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptInfoAware.transcript_id">
<em class="property"><span class="pre">abstract</span><span class="w"> </span><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">transcript_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptInfoAware.transcript_id" title="Link to this definition"></a></dt>
<dd><p>Returns:
string: The transcript RefSeq identifier (e.g. NM_123456.7)</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">TranscriptCoordinates</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">identifier</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">region</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">GenomicRegion</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">exons</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><span class="pre">GenomicRegion</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cds_start</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">cds_end</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_preferred</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_tx.html#TranscriptCoordinates"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>TranscriptCoordinates</cite> knows about genomic region of the transcript, exonic/intronic regions, as well as
the coding and non-coding regions.</p>
<p>If both CDS coordinates are <cite>None</cite>, then the transcript coordinates are assumed to represent a non-coding transcript.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.identifier">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">identifier</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.identifier" title="Link to this definition"></a></dt>
<dd><p>Get transcript’s identifier (e.g. <cite>ENST00000123456.7</cite>, <cite>NM_123456.7</cite>).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.region">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">region</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">GenomicRegion</span></em><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.region" title="Link to this definition"></a></dt>
<dd><p>Get the genomic region spanned by the transcript, corresponding to 5’UTR, exonic, intronic, and 3’UTR regions.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.exons">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">exons</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><span class="pre">GenomicRegion</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.exons" title="Link to this definition"></a></dt>
<dd><p>Get the exon regions.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.cds_start">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">cds_start</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.cds_start" title="Link to this definition"></a></dt>
<dd><p>Get the 0-based (excluded) start coordinate of the first base of the start codon of the transcript or <cite>None</cite>
if the transcript is not coding.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.cds_end">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">cds_end</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.cds_end" title="Link to this definition"></a></dt>
<dd><p>Get the 0-based (included) end coordinate of the last base of the termination codon of the transcript or <cite>None</cite>
if the transcript is not coding.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.is_coding">
<span class="sig-name descname"><span class="pre">is_coding</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_tx.html#TranscriptCoordinates.is_coding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.is_coding" title="Link to this definition"></a></dt>
<dd><p>Return <cite>True</cite> if the transcript is coding/translated.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.get_coding_base_count">
<span class="sig-name descname"><span class="pre">get_coding_base_count</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_tx.html#TranscriptCoordinates.get_coding_base_count"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.get_coding_base_count" title="Link to this definition"></a></dt>
<dd><p>Calculate the number of coding bases present in the transcript. Note, the count does <em>NOT</em> include
the termination codon since it does not code for an aminoacid.
Returns: an <cite>int</cite> with the coding base count or <cite>None</cite> if the transcript is non-coding.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.get_codon_count">
<span class="sig-name descname"><span class="pre">get_codon_count</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_tx.html#TranscriptCoordinates.get_codon_count"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.get_codon_count" title="Link to this definition"></a></dt>
<dd><p>Calculate the count of codons present in the transcript. Note, the count does <em>NOT</em> include the termination codon!</p>
<p>Returns: the number of codons of the transcript or <cite>None</cite> if the transcript is non-coding.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.get_five_prime_utrs">
<span class="sig-name descname"><span class="pre">get_five_prime_utrs</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><span class="pre">GenomicRegion</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_tx.html#TranscriptCoordinates.get_five_prime_utrs"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.get_five_prime_utrs" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of genomic regions that correspond to 5’ untranslated regions of the transcript.</p>
<p>Returns: a sequence of genomic regions, an empty sequence if the transcript is non-coding.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.get_three_prime_utrs">
<span class="sig-name descname"><span class="pre">get_three_prime_utrs</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><span class="pre">GenomicRegion</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_tx.html#TranscriptCoordinates.get_three_prime_utrs"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.get_three_prime_utrs" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of genomic regions that correspond to 3’ untranslated regions of the transcript.</p>
<p>Note, the termination codon is <em>NOT</em> included in the regions!</p>
<p>Returns: a sequence of genomic regions, an empty sequence if the transcript is non-coding.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.get_cds_regions">
<span class="sig-name descname"><span class="pre">get_cds_regions</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><span class="pre">GenomicRegion</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_tx.html#TranscriptCoordinates.get_cds_regions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.get_cds_regions" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of genomic regions that correspond to coding regions of the transcript, including
BOTH the initiation and termination codons.</p>
<p>Returns: a sequence of genomic regions, an empty sequence if the transcript is non-coding.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.TranscriptCoordinates.is_preferred">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">is_preferred</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.model.TranscriptCoordinates.is_preferred" title="Link to this definition"></a></dt>
<dd><p>Check if the transcript belongs among the preferred transcripts of the gene.
This usually means that the transcript was chosen by the MANE project or similar.</p>
<p>Returns <cite>None</cite> if the info is not available.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinMetadata">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">ProteinMetadata</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">label</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_features</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model._protein.ProteinFeature"><span class="pre">ProteinFeature</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_length</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#ProteinMetadata"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.ProteinMetadata" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>An info regarding a protein sequence, including an ID, a label,
and location of protein features, such as motifs, domains, or other regions.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinMetadata.protein_id">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">protein_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.ProteinMetadata.protein_id" title="Link to this definition"></a></dt>
<dd><p>Returns:
string: A string unique to this protein</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinMetadata.label">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">label</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.ProteinMetadata.label" title="Link to this definition"></a></dt>
<dd><p>Returns:
string: The full name of the protein</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinMetadata.protein_features">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">protein_features</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model._protein.ProteinFeature"><span class="pre">ProteinFeature</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.model.ProteinMetadata.protein_features" title="Link to this definition"></a></dt>
<dd><p>Returns:
Sequence[ProteinFeature]: A sequence of ProteinFeatures objects</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinMetadata.protein_length">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">protein_length</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#genophenocorr.model.ProteinMetadata.protein_length" title="Link to this definition"></a></dt>
<dd><p>Returns:
int: length of protein</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinMetadata.domains">
<span class="sig-name descname"><span class="pre">domains</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model._protein.ProteinFeature"><span class="pre">ProteinFeature</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#ProteinMetadata.domains"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.ProteinMetadata.domains" title="Link to this definition"></a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A subgroup of protein_features, where the ProteinFeature object has a FeatureType equal to “DOMAIN”</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Iterable[<a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model.ProteinFeature">ProteinFeature</a>]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinMetadata.repeats">
<span class="sig-name descname"><span class="pre">repeats</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model._protein.ProteinFeature"><span class="pre">ProteinFeature</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#ProteinMetadata.repeats"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.ProteinMetadata.repeats" title="Link to this definition"></a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A subgroup of protein_features, where the ProteinFeature object has a FeatureType equal to “REPEAT”</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Iterable[<a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model.ProteinFeature">ProteinFeature</a>]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinMetadata.regions">
<span class="sig-name descname"><span class="pre">regions</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model._protein.ProteinFeature"><span class="pre">ProteinFeature</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#ProteinMetadata.regions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.ProteinMetadata.regions" title="Link to this definition"></a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A subgroup of protein_features, where the ProteinFeature object has a FeatureType equal to “REGIONS”</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Iterable[<a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model.ProteinFeature">ProteinFeature</a>]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinMetadata.motifs">
<span class="sig-name descname"><span class="pre">motifs</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model._protein.ProteinFeature"><span class="pre">ProteinFeature</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#ProteinMetadata.motifs"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.ProteinMetadata.motifs" title="Link to this definition"></a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A subgroup of protein_features, where the ProteinFeature object has a FeatureType equal to “MOTIF”</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Iterable[<a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model.ProteinFeature">ProteinFeature</a>]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinMetadata.get_features_variant_overlaps">
<span class="sig-name descname"><span class="pre">get_features_variant_overlaps</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Region</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Collection" title="(in Python v3.12)"><span class="pre">Collection</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model._protein.ProteinFeature"><span class="pre">ProteinFeature</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#ProteinMetadata.get_features_variant_overlaps"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.ProteinMetadata.get_features_variant_overlaps" title="Link to this definition"></a></dt>
<dd><p>Get a collection of protein features that overlap with the <cite>region</cite>.
:param region: the query region.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>a collection of overlapping protein features.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Collection[<a class="reference internal" href="#genophenocorr.model.ProteinFeature" title="genophenocorr.model.ProteinFeature">ProteinFeature</a>]</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinFeature">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">ProteinFeature</span></span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#ProteinFeature"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.ProteinFeature" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinFeature.create">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">create</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">info</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.FeatureInfo" title="genophenocorr.model._protein.FeatureInfo"><span class="pre">FeatureInfo</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.model.FeatureType" title="genophenocorr.model._protein.FeatureType"><span class="pre">FeatureType</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#ProteinFeature.create"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.ProteinFeature.create" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinFeature.info">
<em class="property"><span class="pre">abstract</span><span class="w"> </span><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">info</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.FeatureInfo" title="genophenocorr.model._protein.FeatureInfo"><span class="pre">FeatureInfo</span></a></em><a class="headerlink" href="#genophenocorr.model.ProteinFeature.info" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinFeature.feature_type">
<em class="property"><span class="pre">abstract</span><span class="w"> </span><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">feature_type</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.model.FeatureType" title="genophenocorr.model._protein.FeatureType"><span class="pre">FeatureType</span></a></em><a class="headerlink" href="#genophenocorr.model.ProteinFeature.feature_type" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.model.ProteinFeature.to_string">
<span class="sig-name descname"><span class="pre">to_string</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#ProteinFeature.to_string"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.ProteinFeature.to_string" title="Link to this definition"></a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.FeatureInfo">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">FeatureInfo</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">region</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Region</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#FeatureInfo"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.FeatureInfo" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>FeatureInfo</cite> represents a protein feature
(e.g. a repeated sequence given the name “ANK 1” in protein “Ankyrin repeat domain-containing protein 11”)</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.FeatureInfo.name">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">name</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.model.FeatureInfo.name" title="Link to this definition"></a></dt>
<dd><p>Returns:
string: the name of the protein feature</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.FeatureInfo.start">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">start</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#genophenocorr.model.FeatureInfo.start" title="Link to this definition"></a></dt>
<dd><p>Returns:
integer: A 0-based (excluded) start coordinate of the protein feature.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.FeatureInfo.end">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">end</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#genophenocorr.model.FeatureInfo.end" title="Link to this definition"></a></dt>
<dd><p>Returns:
integer: A 0-based (included) end coordinate of the protein feature.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.model.FeatureInfo.region">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">region</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Region</span></em><a class="headerlink" href="#genophenocorr.model.FeatureInfo.region" title="Link to this definition"></a></dt>
<dd><p>Returns:
Region: a protein region spanned by the feature.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.model.FeatureType">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.model.</span></span><span class="sig-name descname"><span class="pre">FeatureType</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">value</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">module</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qualname</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boundary</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/model/_protein.html#FeatureType"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.model.FeatureType" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/enum.html#enum.Enum" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Enum</span></code></a></p>
<p>An enum representing the protein feature types supported in GPSEA.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.FeatureType.REPEAT">
<span class="sig-name descname"><span class="pre">REPEAT</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">1</span></em><a class="headerlink" href="#genophenocorr.model.FeatureType.REPEAT" title="Link to this definition"></a></dt>
<dd><p>A repeated sequence motif or repeated domain within the protein.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.FeatureType.MOTIF">
<span class="sig-name descname"><span class="pre">MOTIF</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">2</span></em><a class="headerlink" href="#genophenocorr.model.FeatureType.MOTIF" title="Link to this definition"></a></dt>
<dd><p>A short (usually not more than 20 amino acids) conserved sequence motif of biological significance.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.FeatureType.DOMAIN">
<span class="sig-name descname"><span class="pre">DOMAIN</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">3</span></em><a class="headerlink" href="#genophenocorr.model.FeatureType.DOMAIN" title="Link to this definition"></a></dt>
<dd><p>A specific combination of secondary structures organized into a characteristic three-dimensional structure or fold.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.model.FeatureType.REGION">
<span class="sig-name descname"><span class="pre">REGION</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">4</span></em><a class="headerlink" href="#genophenocorr.model.FeatureType.REGION" title="Link to this definition"></a></dt>
<dd><p>A region of interest that cannot be described in other subsections.</p>
</dd></dl>

</dd></dl>

</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="genophenocorr.data.html" class="btn btn-neutral float-left" title="genophenocorr.data package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="genophenocorr.preprocessing.html" class="btn btn-neutral float-right" title="genophenocorr.preprocessing package" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>