<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>genophenocorr.analysis package &mdash; genophenocorr 0.1.1dev documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/genophenocorr.css?v=a9fa5eb9" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=8427294d"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../_static/copybutton.js?v=6dbb43f8"></script>
        <script>window.MathJax = {"TeX": {"Macros": {"RR": "{\\mathbb{R}}"}}}</script>
        <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="genophenocorr.analysis.predicate package" href="genophenocorr.analysis.predicate.html" />
    <link rel="prev" title="genophenocorr package" href="genophenocorr.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            genophenocorr
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../user-guide/index.html">User guide</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">API reference</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="genophenocorr.html">genophenocorr package</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="genophenocorr.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l4 current"><a class="current reference internal" href="#">genophenocorr.analysis package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.constants.html">genophenocorr.constants package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.data.html">genophenocorr.data package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.model.html">genophenocorr.model package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.preprocessing.html">genophenocorr.preprocessing package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.view.html">genophenocorr.view package</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">genophenocorr</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="modules.html">API reference</a></li>
          <li class="breadcrumb-item"><a href="genophenocorr.html">genophenocorr package</a></li>
      <li class="breadcrumb-item active">genophenocorr.analysis package</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/apidocs/genophenocorr.analysis.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-genophenocorr.analysis">
<span id="genophenocorr-analysis-package"></span><h1>genophenocorr.analysis package<a class="headerlink" href="#module-genophenocorr.analysis" title="Link to this heading"></a></h1>
<dl class="py function">
<dt class="sig sig-object py" id="genophenocorr.analysis.configure_cohort_analysis">
<span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">configure_cohort_analysis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cohort</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Cohort" title="genophenocorr.model._cohort.Cohort"><span class="pre">Cohort</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.ontology.html#hpotk.ontology.MinimalOntology" title="(in hpo-toolkit)"><span class="pre">MinimalOntology</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'UNIPROT'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cache_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.analysis.CohortAnalysisConfiguration" title="genophenocorr.analysis._config.CohortAnalysisConfiguration"><span class="pre">CohortAnalysisConfiguration</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.CohortAnalysis" title="genophenocorr.analysis._api.CohortAnalysis"><span class="pre">CohortAnalysis</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#configure_cohort_analysis"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.configure_cohort_analysis" title="Link to this definition"></a></dt>
<dd><p>Configure <a class="reference internal" href="#genophenocorr.analysis.CohortAnalysis" title="genophenocorr.analysis.CohortAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.analysis.CohortAnalysis</span></code></a> for given <cite>cohort</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>cohort</strong> – a <a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Cohort" title="genophenocorr.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.model.Cohort</span></code></a> to analyze</p></li>
<li><p><strong>hpo</strong> – a <code class="xref py py-class docutils literal notranslate"><span class="pre">hpotk.MinimalOntology</span></code> with HPO to use in the analysis</p></li>
<li><p><strong>protein_source</strong> – the resource to retrieve protein annotations from if we cannot find the annotations locally.
Choose from <code class="docutils literal notranslate"><span class="pre">{'UNIPROT'}</span></code> (just one fallback implementation is available at the moment).</p></li>
<li><p><strong>config</strong> – an optional <a class="reference internal" href="#genophenocorr.analysis.CohortAnalysisConfiguration" title="genophenocorr.analysis.CohortAnalysisConfiguration"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortAnalysisConfiguration</span></code></a> to parameterize the analysis.
The default parameters will be used if <cite>None</cite>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">CohortAnalysis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_service</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.preprocessing.html#genophenocorr.preprocessing.ProteinMetadataService" title="genophenocorr.preprocessing._api.ProteinMetadataService"><span class="pre">ProteinMetadataService</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>CohortAnalysis</cite> is a driver class for running genotype-phenotype correlation analyses.</p>
<p>The class provides various methods to test genotype-phenotype correlations. All methods wrap results
into <a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis.GenotypePhenotypeAnalysisResult"><code class="xref py py-class docutils literal notranslate"><span class="pre">GenotypePhenotypeAnalysisResult</span></code></a>.</p>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_variant_effect">
<span class="sig-name descname"><span class="pre">compare_by_variant_effect</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">effect</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_variant_effect"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_variant_effect" title="Link to this definition"></a></dt>
<dd><p>Compares variants with <cite>effect</cite> vs. variants with all other variant effects.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>effect</strong> – variant effect</p></li>
<li><p><strong>tx_id</strong> – the accession of the transcript of interest</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_variant_key">
<span class="sig-name descname"><span class="pre">compare_by_variant_key</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant_key</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_variant_key"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_variant_key" title="Link to this definition"></a></dt>
<dd><p>Compares variant with <cite>variant_key</cite> vs all the other variants.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantCoordinates.variant_key" title="genophenocorr.model.VariantCoordinates.variant_key"><code class="xref py py-attr docutils literal notranslate"><span class="pre">genophenocorr.model.VariantCoordinates.variant_key</span></code></a></p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>variant_key</strong> – a <cite>str</cite> with the variant key, e.g. <code class="docutils literal notranslate"><span class="pre">X_12345_12345_C_G</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_exon">
<span class="sig-name descname"><span class="pre">compare_by_exon</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">exon_number</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_exon"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_exon" title="Link to this definition"></a></dt>
<dd><p>Compares variants with <cite>effect</cite> vs. variants with all other variant effects.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>We use 1-based numbering to number the exons, not the usual 0-based numbering of the computer science.
Therefore, the first exon of the transcript has <code class="docutils literal notranslate"><span class="pre">exon_number==1</span></code>, the second exon is <code class="docutils literal notranslate"><span class="pre">2</span></code>, and so on …</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>We do not check if the <cite>exon_number</cite> spans beyond the number of exons of the given <cite>transcript_id</cite>!
Therefore, <code class="docutils literal notranslate"><span class="pre">exon_number==10,000</span></code> will effectively return <code class="xref py py-attr docutils literal notranslate"><span class="pre">BooleanPredicate.FALSE</span></code> for <em>all</em> patients!!! 😱
Well, at least the patients of the <em>Homo sapiens sapiens</em> taxon…</p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>exon_number</strong> – a positive <cite>int</cite> with exon number</p></li>
<li><p><strong>tx_id</strong> – the accession of the transcript of interest</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature_type">
<span class="sig-name descname"><span class="pre">compare_by_protein_feature_type</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">feature_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model._protein.FeatureType"><span class="pre">FeatureType</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_protein_feature_type"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature_type" title="Link to this definition"></a></dt>
<dd><p>Compare genotype-phenotype correlation between variants that affect a given <cite>feature_type</cite> and the variants
affecting the rest of the protein.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>feature_type</strong> (<a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model.FeatureType"><em>FeatureType</em></a>) – the protein feature type of interest</p></li>
<li><p><strong>tx_id</strong> – the accession of the transcript of interest</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature">
<span class="sig-name descname"><span class="pre">compare_by_protein_feature</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">feature</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_protein_feature"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature" title="Link to this definition"></a></dt>
<dd><p>Compare genotype-phenotype correlation between variants that affect a given <cite>feature</cite> and the variants
affecting the rest of the protein.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>The protein feature names can be accessed at <code class="xref py py-attr docutils literal notranslate"><span class="pre">genophenocorr.model.ProteinFeature.name</span></code>.</p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>feature</strong> (<em>string</em>) – feature identifier, e.g. <code class="docutils literal notranslate"><span class="pre">DNA-binding</span></code>.</p></li>
<li><p><strong>tx_id</strong> (<em>string</em>) – the accession of the transcript of interest</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_protein_region">
<span class="sig-name descname"><span class="pre">compare_by_protein_region</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_region</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_protein_region"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_protein_region" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_variant_effects">
<span class="sig-name descname"><span class="pre">compare_by_variant_effects</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">effect1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">effect2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_variant_effects"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_variant_effects" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_variant_keys">
<span class="sig-name descname"><span class="pre">compare_by_variant_keys</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant_key1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_key2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_variant_keys"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_variant_keys" title="Link to this definition"></a></dt>
<dd><p>Compare genotype-phenotype correlation between groups with variant <cite>a</cite> vs. variant <cite>b</cite>.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantCoordinates.variant_key" title="genophenocorr.model.VariantCoordinates.variant_key"><code class="xref py py-attr docutils literal notranslate"><span class="pre">genophenocorr.model.VariantCoordinates.variant_key</span></code></a></p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>a</strong> – a <cite>str</cite> with variant key of variant <cite>a</cite>, e.g. <code class="docutils literal notranslate"><span class="pre">X_12345_12345_C_G</span></code></p></li>
<li><p><strong>b</strong> – a <cite>str</cite> with variant key of variant <cite>b</cite></p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_exons">
<span class="sig-name descname"><span class="pre">compare_by_exons</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">exon1_number</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">exon2_number</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_exons"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_exons" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature_types">
<span class="sig-name descname"><span class="pre">compare_by_protein_feature_types</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">feature_type1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model._protein.FeatureType"><span class="pre">FeatureType</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_type2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model._protein.FeatureType"><span class="pre">FeatureType</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_protein_feature_types"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature_types" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_protein_features">
<span class="sig-name descname"><span class="pre">compare_by_protein_features</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">feature1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_protein_features"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_protein_features" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_protein_regions">
<span class="sig-name descname"><span class="pre">compare_by_protein_regions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">region1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">region2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Region</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tx_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_by_protein_regions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_protein_regions" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_hpo_vs_genotype">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_hpo_vs_genotype</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.analysis.predicate.genotype.html#genophenocorr.analysis.predicate.genotype.VariantPredicate" title="genophenocorr.analysis.predicate.genotype._api.VariantPredicate"><span class="pre">VariantPredicate</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_hpo_vs_genotype"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_hpo_vs_genotype" title="Link to this definition"></a></dt>
<dd><p>Bin patients according to a presence of at least one allele that matches <cite>predicate</cite>
and test for genotype-phenotype correlations.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_hpo_vs_genotype_groups">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_hpo_vs_genotype_groups</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">first</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.analysis.predicate.genotype.html#genophenocorr.analysis.predicate.genotype.VariantPredicate" title="genophenocorr.analysis.predicate.genotype._api.VariantPredicate"><span class="pre">VariantPredicate</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">second</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.analysis.predicate.genotype.html#genophenocorr.analysis.predicate.genotype.VariantPredicate" title="genophenocorr.analysis.predicate.genotype._api.VariantPredicate"><span class="pre">VariantPredicate</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="genophenocorr.analysis._api.GenotypePhenotypeAnalysisResult"><span class="pre">GenotypePhenotypeAnalysisResult</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_hpo_vs_genotype_groups"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_hpo_vs_genotype_groups" title="Link to this definition"></a></dt>
<dd><p>Bin patients according to a presence of at least one allele that matches <cite>first</cite> or <cite>second</cite> predicate
and test for genotype-phenotype correlations between the groups.</p>
<p>Note, the patients that pass testing by both predicates are <em>OMITTED</em> from the analysis!</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_symptom_count_vs_genotype">
<em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">compare_symptom_count_vs_genotype</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">query</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.analysis.predicate.genotype.html#genophenocorr.analysis.predicate.genotype.VariantPredicate" title="genophenocorr.analysis.predicate.genotype._api.VariantPredicate"><span class="pre">VariantPredicate</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#CohortAnalysis.compare_symptom_count_vs_genotype"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_symptom_count_vs_genotype" title="Link to this definition"></a></dt>
<dd><p>Bins the patients according to the count of present phenotypes that are either
an exact match to the <cite>query</cite> terms or their descendants
and test for genotype-phenotype correlation between the counts
and the genotype bins.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>query</strong> – an iterable with HPO term CURIEs (e.g. <cite>‘HP:0001250’</cite> for seizure)
or HPO term IDs</p></li>
<li><p><strong>variant_predicate</strong> – the variant predicate for binning the patients
along the genotype axis</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">GenotypePhenotypeAnalysisResult</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_usable</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">all_counts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pvals</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">corrected_pvals</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">phenotype_categories</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geno_predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PolyPredicate" title="genophenocorr.analysis.predicate._api.PolyPredicate"><span class="pre">PolyPredicate</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">mtc_filter_report</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#genophenocorr.analysis.HpoMtcReport" title="genophenocorr.analysis._api.HpoMtcReport"><span class="pre">HpoMtcReport</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#GenotypePhenotypeAnalysisResult"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>GenotypePhenotypeAnalysisResult</cite> summarizes results of genotype-phenotype correlation analysis of a cohort.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.n_usable">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">n_usable</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.n_usable" title="Link to this definition"></a></dt>
<dd><p>Get a mapping from a phenotype <cite>P</cite> (either an HPO term or a disease ID)
to an <cite>int</cite> with the number of patients where the phenotype was assessable,
and are, thus, usable for genotype-phenotype correlation analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.all_counts">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">all_counts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.all_counts" title="Link to this definition"></a></dt>
<dd><p>Get a mapping from the phenotype item to <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">pandas.DataFrame</span></code></a> with counts of patients
in genotype and phenotype groups.</p>
<p>An example for a genotype predicate that bins into two categories (<cite>Yes</cite> and <cite>No</cite>) based on presence
of a missense variant in transcript <cite>NM_123456.7</cite>, and phenotype predicate that checks
presence/absence of <cite>HP:0001166</cite> (a phenotype term):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>           <span class="n">Has</span> <span class="n">MISSENSE_VARIANT</span> <span class="ow">in</span> <span class="n">NM_123456</span><span class="mf">.7</span>
           <span class="n">No</span>       <span class="n">Yes</span>
<span class="n">Present</span>
<span class="n">Yes</span>        <span class="mi">1</span>        <span class="mi">13</span>
<span class="n">No</span>         <span class="mi">7</span>        <span class="mi">5</span>
</pre></div>
</div>
<p>The rows correspond to the phenotype categories, and the columns represent the genotype categories.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.pvals">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pvals</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.pvals" title="Link to this definition"></a></dt>
<dd><p>Get a <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">pandas.Series</span></code></a> with p values for each tested HPO term.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.corrected_pvals">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">corrected_pvals</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.corrected_pvals" title="Link to this definition"></a></dt>
<dd><p>Get an optional <a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><code class="xref py py-class docutils literal notranslate"><span class="pre">pandas.Series</span></code></a> with p values for each tested HPO term after multiple testing correction.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.phenotype_categories">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">phenotype_categories</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Sequence" title="(in Python v3.12)"><span class="pre">Sequence</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.phenotype_categories" title="Link to this definition"></a></dt>
<dd><p>Get a sequence of phenotype patient categories that can be investigated.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.total_tests">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">total_tests</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.total_tests" title="Link to this definition"></a></dt>
<dd><p>Get total count of tests that were run for this analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.mtc_filter_report">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mtc_filter_report</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.analysis.HpoMtcReport" title="genophenocorr.analysis._api.HpoMtcReport"><span class="pre">HpoMtcReport</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.mtc_filter_report" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.GenotypePhenotypeAnalysisResult.summarize">
<span class="sig-name descname"><span class="pre">summarize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.ontology.html#hpotk.ontology.MinimalOntology" title="(in hpo-toolkit)"><span class="pre">MinimalOntology</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">category</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#GenotypePhenotypeAnalysisResult.summarize"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.GenotypePhenotypeAnalysisResult.summarize" title="Link to this definition"></a></dt>
<dd><p>Create a data frame with summary of the genotype phenotype analysis.</p>
<p>The <em>rows</em> of the frame correspond to the analyzed HPO terms.</p>
<p>The columns of the data frame have <cite>Count</cite> and <cite>Percentage</cite> per used genotype predicate.</p>
<p><strong>Example</strong></p>
<p>If we use <code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.analysis.predicate.genotype.VariantEffectPredicate</span></code>
which can compare phenotype with and without a missense variant, we will have a data frame
that looks like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>MISSENSE_VARIANT on `NM_1234.5`                       No                  Yes
                                                      Count    Percent    Count    Percent   p value   Corrected p value
Arachnodactyly [HP:0001166]                           1/10         10%    13/16        81%   0.000781  0.020299
Abnormality of the musculature [HP:0003011]           6/6         100%    11/11       100%   1.000000  1.000000
Abnormal nervous system physiology [HP:0012638]       9/9         100%    15/15       100%   1.000000  1.000000
...                                                   ...      ...        ...      ...       ...       ...
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">CohortAnalysisConfiguration</span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfiguration"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p><cite>CohortAnalysisConfiguration</cite> is a value class for storing <a class="reference internal" href="#genophenocorr.analysis.CohortAnalysis" title="genophenocorr.analysis.CohortAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.analysis.CohortAnalysis</span></code></a>
configuration options.</p>
<p>The class contains the default values upon creation and the configuration option values can be set as properties.</p>
<p>If an invalid value option is passed to the property setter, a warning is logged and the previous value is retained.
Therefore, it is <em>impossible</em> to mis-configure the analysis.</p>
<section id="default-values">
<h2>Default values<a class="headerlink" href="#default-values" title="Link to this heading"></a></h2>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Option</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Default value</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">missing_implies_excluded</span></code></p></td>
<td><p><cite>bool</cite></p></td>
<td><p><cite>False</cite></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">pval_correction</span></code></p></td>
<td><p><cite>str</cite></p></td>
<td><p><cite>bonferroni</cite></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">min_perc_patients_w_hpo</span></code></p></td>
<td><p><cite>float</cite></p></td>
<td><p><cite>0.1</cite></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">mtc_alpha</span></code></p></td>
<td><p><cite>float</cite></p></td>
<td><p><cite>0.05</cite></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">include_sv</span></code></p></td>
<td><p><cite>bool</cite></p></td>
<td><p><cite>False</cite></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">mtc_strategy</span></code></p></td>
<td><p><a class="reference internal" href="#genophenocorr.analysis.MTC_Strategy" title="genophenocorr.analysis.MTC_Strategy"><code class="xref py py-class docutils literal notranslate"><span class="pre">MTC_Strategy</span></code></a></p></td>
<td><p><a class="reference internal" href="#genophenocorr.analysis.MTC_Strategy.ALL_HPO_TERMS" title="genophenocorr.analysis.MTC_Strategy.ALL_HPO_TERMS"><code class="xref py py-class docutils literal notranslate"><span class="pre">MTC_Strategy.ALL_HPO_TERMS</span></code></a></p></td>
</tr>
</tbody>
</table>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.missing_implies_excluded">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">missing_implies_excluded</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.missing_implies_excluded" title="Link to this definition"></a></dt>
<dd><p><cite>True</cite> if we assume that a patient without a specific phenotype listed <em>does not</em> have the phenotype.
Otherwise, the only excluded phenotypes are those that are excluded explicitly.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.pval_correction">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pval_correction</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.pval_correction" title="Link to this definition"></a></dt>
<dd><p><cite>bonferroni</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Get method for multiple testing p value correction. Default</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.min_perc_patients_w_hpo">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">min_perc_patients_w_hpo</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.min_perc_patients_w_hpo" title="Link to this definition"></a></dt>
<dd><p>A threshold for removing rare HPO terms, only the terms that are observed in at least this fraction of patients
will be retained for the analysis (default 0.2).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.mtc_alpha">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mtc_alpha</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.mtc_alpha" title="Link to this definition"></a></dt>
<dd><p>The alpha value for multiple testing correction.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.include_sv">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">include_sv</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.include_sv" title="Link to this definition"></a></dt>
<dd><p><cite>True</cite> if we want to include structural variants in the analysis
(i.e. the variants that use symbolic VCF notation).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.mtc_strategy">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mtc_strategy</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#genophenocorr.analysis.MTC_Strategy" title="genophenocorr.analysis._config.MTC_Strategy"><span class="pre">MTC_Strategy</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.mtc_strategy" title="Link to this definition"></a></dt>
<dd><p>Get the MTC strategy to be used.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.all_terms_strategy">
<span class="sig-name descname"><span class="pre">all_terms_strategy</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfiguration.all_terms_strategy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.all_terms_strategy" title="Link to this definition"></a></dt>
<dd><p>Test all HPO terms (the default).</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.heuristic_strategy">
<span class="sig-name descname"><span class="pre">heuristic_strategy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">threshold_HPO_observed_frequency</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.2</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfiguration.heuristic_strategy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.heuristic_strategy" title="Link to this definition"></a></dt>
<dd><p>Only test the HPO terms that pass all rules of the heuristic strategy.</p>
<p>See <a class="reference internal" href="../user-guide/mtc.html#heuristic-sampler-strategy-section"><span class="std std-ref">heuristic sampler strategy</span></a> section
for more info on the rules.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>threshold_HPO_observed_frequency</strong> – the float in range <em>(0, 1]</em> to represent
the minimum fraction of patients for an HPO term to be included.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.specify_terms_strategy">
<span class="sig-name descname"><span class="pre">specify_terms_strategy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">specified_term_set</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#CohortAnalysisConfiguration.specify_terms_strategy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.specify_terms_strategy" title="Link to this definition"></a></dt>
<dd><p>Mitigate the MTC burden by only testing the specified HPO terms.</p>
<p>Note, the HPO terms are validated before running the analysis,
to point out invalid CURIE (e.g. <cite>WHATEVER</cite>) values,
or HPO term IDs that are not in the currently used HPO.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>specified_term_set</strong> – an iterable with CURIEs (e.g. <cite>HP:0001250</cite>)
or <code class="xref py py-class docutils literal notranslate"><span class="pre">hpotk.TermId</span></code> instances representing the terms to test.</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysisConfiguration.terms_to_test">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">terms_to_test</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></em><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysisConfiguration.terms_to_test" title="Link to this definition"></a></dt>
<dd></dd></dl>

</section>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.MTC_Strategy">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">MTC_Strategy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">value</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">module</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">qualname</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">start</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boundary</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#MTC_Strategy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.MTC_Strategy" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/enum.html#enum.Enum" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Enum</span></code></a></p>
<p>A strategy for mitigating the multiple testing correction (MTC) burden.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.analysis.MTC_Strategy.ALL_HPO_TERMS">
<span class="sig-name descname"><span class="pre">ALL_HPO_TERMS</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">0</span></em><a class="headerlink" href="#genophenocorr.analysis.MTC_Strategy.ALL_HPO_TERMS" title="Link to this definition"></a></dt>
<dd><p>All HPO terms will be tested.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.analysis.MTC_Strategy.HEURISTIC_SAMPLER">
<span class="sig-name descname"><span class="pre">HEURISTIC_SAMPLER</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">1</span></em><a class="headerlink" href="#genophenocorr.analysis.MTC_Strategy.HEURISTIC_SAMPLER" title="Link to this definition"></a></dt>
<dd><p>Only HPO terms present in at least a certain fraction of patients will be tested.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.analysis.MTC_Strategy.SPECIFY_TERMS">
<span class="sig-name descname"><span class="pre">SPECIFY_TERMS</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">2</span></em><a class="headerlink" href="#genophenocorr.analysis.MTC_Strategy.SPECIFY_TERMS" title="Link to this definition"></a></dt>
<dd><p>Only the manually provided HPO terms will be tested.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.HpoMtcReport">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">HpoMtcReport</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">filter_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">mtc_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_results_map</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">term_count</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_api.html#HpoMtcReport"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.HpoMtcReport" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>Class to simplify reporting results of multiple testing filtering by HpoMtcFilter subclasses.</p>
<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.HpoMtcReport.filter_method">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">filter_method</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.analysis.HpoMtcReport.filter_method" title="Link to this definition"></a></dt>
<dd><p>Returns:
the name of the HpoMtcFilter method used.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.HpoMtcReport.skipped_terms_dict">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">skipped_terms_dict</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#genophenocorr.analysis.HpoMtcReport.skipped_terms_dict" title="Link to this definition"></a></dt>
<dd><p>Returns:
a mapping with reasons why an HPO term was skipped as keys and counts of the skipped terms as values.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.HpoMtcReport.mtc_method">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mtc_method</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></em><a class="headerlink" href="#genophenocorr.analysis.HpoMtcReport.mtc_method" title="Link to this definition"></a></dt>
<dd><p>Returns:
the name of the multiple testing correction method used (e.g. <cite>bonferroni</cite>).</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="genophenocorr.analysis.HpoMtcReport.n_terms_tested">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">n_terms_tested</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></em><a class="headerlink" href="#genophenocorr.analysis.HpoMtcReport.n_terms_tested" title="Link to this definition"></a></dt>
<dd><p>Returns:
the number of HPO terms before filtering.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.HeuristicMtcFilter">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">HeuristicMtcFilter</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.ontology.html#hpotk.ontology.MinimalOntology" title="(in hpo-toolkit)"><span class="pre">MinimalOntology</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">term_frequency_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">general_hpo_terms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#HeuristicMtcFilter"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.HeuristicMtcFilter" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">HpoMtcFilter</span></code></p>
<p><cite>HeuristicMtcFilter</cite> decides which phenotypes should be tested and which phenotypes are not worth testing.</p>
<p>The class leverages a number of heuristics and domain decisions. See <span class="xref std std-ref">heuristic-sampler-strategy-section</span> for more info.</p>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.HeuristicMtcFilter.default_filter">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">default_filter</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.ontology.html#hpotk.ontology.MinimalOntology" title="(in hpo-toolkit)"><span class="pre">MinimalOntology</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">term_frequency_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#HeuristicMtcFilter.default_filter"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.HeuristicMtcFilter.default_filter" title="Link to this definition"></a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>hpo</strong> – HPO</p></li>
<li><p><strong>term_frequency_threshold</strong> – a <cite>float</cite> in range <span class="math notranslate nohighlight">\((0, 1]\)</span> with the minimum frequency
for an HPO term to have in at least one of the genotype groups
(e.g., 22% in missense and 3% in nonsense genotypes would be OK,
but not 13% missense and 10% nonsense genotypes if the threshold is 0.2)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.HeuristicMtcFilter.filter_terms_to_test">
<span class="sig-name descname"><span class="pre">filter_terms_to_test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_usable</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">all_counts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.12)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#HeuristicMtcFilter.filter_terms_to_test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.HeuristicMtcFilter.filter_terms_to_test" title="Link to this definition"></a></dt>
<dd><p>Decide which terms to pass through for statistical testing.
The intention of this class is to reduce multiple testing burden by removing terms that are unlikely
to lead to interesting statistical/analytical results.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>n_usable</strong> – a mapping from the <code class="xref py py-class docutils literal notranslate"><span class="pre">hpotk.TermId</span></code> to an <cite>int</cite> with the count of patients
that could be binned according to the used genotype/phenotype predicate.</p></li>
<li><p><strong>all_counts</strong> – a mapping from the <code class="xref py py-class docutils literal notranslate"><span class="pre">hpotk.TermId</span></code> to</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p>a mapping from <code class="xref py py-class docutils literal notranslate"><span class="pre">hpotk.TermId</span></code> -&gt;</p></li>
<li><p>a mapping from <code class="xref py py-class docutils literal notranslate"><span class="pre">hpotk.TermId</span></code> -&gt;</p></li>
<li><p>a mapping from a <cite>str</cite> with reason why a term was filtered out (e.g. <em>Skipping top level term</em>)</p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>a tuple with three items</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.HeuristicMtcFilter.filter_method_name">
<span class="sig-name descname"><span class="pre">filter_method_name</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#HeuristicMtcFilter.filter_method_name"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.HeuristicMtcFilter.filter_method_name" title="Link to this definition"></a></dt>
<dd><p>returns the name of the heuristic used to limit multiple testing</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.HeuristicMtcFilter.get_number_of_observed_hpo_observations">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">get_number_of_observed_hpo_observations</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">counts_frame</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#HeuristicMtcFilter.get_number_of_observed_hpo_observations"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.HeuristicMtcFilter.get_number_of_observed_hpo_observations" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.HeuristicMtcFilter.get_maximum_group_observed_HPO_frequency">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">get_maximum_group_observed_HPO_frequency</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">counts_frame</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#HeuristicMtcFilter.get_maximum_group_observed_HPO_frequency"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.HeuristicMtcFilter.get_maximum_group_observed_HPO_frequency" title="Link to this definition"></a></dt>
<dd><dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>The maximum frequency of observed HPO annotations across all genotypes.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.HeuristicMtcFilter.one_genotype_has_zero_hpo_observations">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">one_genotype_has_zero_hpo_observations</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">counts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#HeuristicMtcFilter.one_genotype_has_zero_hpo_observations"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.HeuristicMtcFilter.one_genotype_has_zero_hpo_observations" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.HeuristicMtcFilter.some_cell_has_greater_than_one_count">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">some_cell_has_greater_than_one_count</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">counts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#HeuristicMtcFilter.some_cell_has_greater_than_one_count"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.HeuristicMtcFilter.some_cell_has_greater_than_one_count" title="Link to this definition"></a></dt>
<dd><p>If no genotype has more than one HPO count, we do not want to do a test. For instance, if MISSENSE has one
observed HPO and N excluded, and NOT MISSENSE has zero or one observed HPO, then we will skip the test
:param counts: pandas DataFrame with counts</p>
<p>Returns: true if at least one of the genotypes has more than one observed HPO count</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.HeuristicMtcFilter.genotypes_have_same_hpo_proportions">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">genotypes_have_same_hpo_proportions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">counts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">delta</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.12)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.01</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.12)"><span class="pre">bool</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#HeuristicMtcFilter.genotypes_have_same_hpo_proportions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.HeuristicMtcFilter.genotypes_have_same_hpo_proportions" title="Link to this definition"></a></dt>
<dd><p>If each genotype has the same proportion of observed HPOs, then we do not want to do a test.
For instance, if MISSENSE has 5/5 observed HPOs and NOT MISSENSE has 7/7 it makes not sense
to do a statistical test.
:param counts: pandas DataFrame with counts
:param delta: a <cite>float</cite> for tolerance comparing the proportion tolerance</p>
<p>Returns: true if the genotypes differ by more than <cite>delta</cite></p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.SpecifiedTermsMtcFilter">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">SpecifiedTermsMtcFilter</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">hpo</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.ontology.html#hpotk.ontology.MinimalOntology" title="(in hpo-toolkit)"><span class="pre">MinimalOntology</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">terms_to_test</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#SpecifiedTermsMtcFilter"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.SpecifiedTermsMtcFilter" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">HpoMtcFilter</span></code></p>
<p>SpecifiedTermsMtcFilter` limits the HPO terms to be tested to a selection of provided terms.</p>
<p>In cases where we have a hypothesis about which phenotypes are relevant for
testing genotype-pehnotype correlations, we can pass the corresponding
terms to the constructor of this class, thereby preventing other terms from
being tested and reducing the multiple testing burden.</p>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.SpecifiedTermsMtcFilter.filter_terms_to_test">
<span class="sig-name descname"><span class="pre">filter_terms_to_test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n_usable</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">all_counts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.12)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://ielis.github.io/hpo-toolkit/latest/apidocs/hpotk.model.html#hpotk.model.TermId" title="(in hpo-toolkit)"><span class="pre">TermId</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#SpecifiedTermsMtcFilter.filter_terms_to_test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.SpecifiedTermsMtcFilter.filter_terms_to_test" title="Link to this definition"></a></dt>
<dd><p>Remove terms that are not members of the specific set of HPO terms to be tested.
:param n_usable: dictionary with HPO term ids seen in our cohort and their counts
:param all_counts: Dictionary with HPO term ids from cohort and DataFrames with detailed GPC counts</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>filtered versions of the two dictionaries above and dataframe with reasons for skipping</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.SpecifiedTermsMtcFilter.filter_method_name">
<span class="sig-name descname"><span class="pre">filter_method_name</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#SpecifiedTermsMtcFilter.filter_method_name"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.SpecifiedTermsMtcFilter.filter_method_name" title="Link to this definition"></a></dt>
<dd><p>returns the name of the heuristic used to limit multiple testing</p>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="genophenocorr.analysis.apply_predicates_on_patients">
<span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">apply_predicates_on_patients</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patients</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pheno_predicates</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Iterable" title="(in Python v3.12)"><span class="pre">Iterable</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="genophenocorr.analysis.predicate.phenotype.html#genophenocorr.analysis.predicate.phenotype.PhenotypePolyPredicate" title="genophenocorr.analysis.predicate.phenotype._pheno.PhenotypePolyPredicate"><span class="pre">PhenotypePolyPredicate</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gt_predicate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.GenotypePolyPredicate" title="genophenocorr.analysis.predicate._api.GenotypePolyPredicate"><span class="pre">GenotypePolyPredicate</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.12)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Collection" title="(in Python v3.12)"><span class="pre">Collection</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v2.0.0)"><span class="pre">Series</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Mapping" title="(in Python v3.12)"><span class="pre">Mapping</span></a><span class="p"><span class="pre">[</span></span><span class="pre">P</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/pandas-docs/version/2.0.0/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v2.0.0)"><span class="pre">DataFrame</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_gp_analysis.html#apply_predicates_on_patients"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.apply_predicates_on_patients" title="Link to this definition"></a></dt>
<dd><p>Apply the phenotype predicates <cite>pheno_predicates</cite> and the genotype predicate <cite>gt_predicate</cite>
to bin the <cite>patients</cite> into categories.</p>
<p>Note, it may not be possible to bin <em>all</em> patients with a genotype/phenotype pair,
since a predicate is allowed to return <cite>None</cite> (e.g. if it bins the patient into MISSENSE or NONSENSE groups
but the patient has no MISSENSE or NONSENSE variants). If this happens, the patient will not be “usable”
for the phenotype <cite>P</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>patients</strong> – an iterable with the patients to bin into categories</p></li>
<li><p><strong>pheno_predicates</strong> – an iterable with the phenotype predicates to apply</p></li>
<li><p><strong>gt_predicate</strong> – a genotype predicate to apply</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p>a collection of unique <code class="xref py py-class docutils literal notranslate"><span class="pre">PatientCategory</span></code> items that the patients were binned into</p></li>
<li><p>a <code class="xref py py-class docutils literal notranslate"><span class="pre">pd.Series</span></code> with mapping from a phenotype <code class="xref py py-class docutils literal notranslate"><span class="pre">P</span></code> (e.g. an HPO term or a disease)
to an <cite>int</cite> with count of patients that could be binned according to the phenotype <cite>P</cite></p></li>
<li><p>a mapping from phenotype <code class="xref py py-class docutils literal notranslate"><span class="pre">P</span></code> to a data frame with counts of patients
in i-th phenotype category and j-th genotype category where i and j are rows and columns of the data frame</p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>a tuple with 3 items</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="genophenocorr.analysis.configure_default_protein_metadata_service">
<span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">configure_default_protein_metadata_service</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'UNIPROT'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cache_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.preprocessing.html#genophenocorr.preprocessing.ProteinMetadataService" title="genophenocorr.preprocessing._api.ProteinMetadataService"><span class="pre">ProteinMetadataService</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/_config.html#configure_default_protein_metadata_service"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.configure_default_protein_metadata_service" title="Link to this definition"></a></dt>
<dd><p>Create default protein metadata service that will cache the protein metadata
in current working directory under <cite>.genophenocorr_cache/protein_cache</cite>
and reach out to UNIPROT REST API if a cache entry is missing.</p>
</dd></dl>

<section id="subpackages">
<h2>Subpackages<a class="headerlink" href="#subpackages" title="Link to this heading"></a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="genophenocorr.analysis.predicate.html">genophenocorr.analysis.predicate package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory"><code class="docutils literal notranslate"><span class="pre">PatientCategory</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategories"><code class="docutils literal notranslate"><span class="pre">PatientCategories</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.Categorization"><code class="docutils literal notranslate"><span class="pre">Categorization</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PolyPredicate"><code class="docutils literal notranslate"><span class="pre">PolyPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.GroupingPredicate"><code class="docutils literal notranslate"><span class="pre">GroupingPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.GenotypePolyPredicate"><code class="docutils literal notranslate"><span class="pre">GenotypePolyPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.GenotypeBooleanPredicate"><code class="docutils literal notranslate"><span class="pre">GenotypeBooleanPredicate</span></code></a></li>
</ul>
</li>
</ul>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="genophenocorr.html" class="btn btn-neutral float-left" title="genophenocorr package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="genophenocorr.analysis.predicate.html" class="btn btn-neutral float-right" title="genophenocorr.analysis.predicate package" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>