<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>genophenocorr.analysis package &mdash; genophenocorr 0.1.1dev documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/copybutton.css" type="text/css" />
      <link rel="stylesheet" href="../_static/genophenocorr.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=8427294d"></script>
        <script src="../_static/doctools.js?v=888ff710"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="genophenocorr.analysis.predicate package" href="genophenocorr.analysis.predicate.html" />
    <link rel="prev" title="genophenocorr package" href="genophenocorr.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            genophenocorr
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../user-guide/index.html">User guide</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">API reference</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="genophenocorr.html">genophenocorr package</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="genophenocorr.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l4 current"><a class="current reference internal" href="#">genophenocorr.analysis package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.constants.html">genophenocorr.constants package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.data.html">genophenocorr.data package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.model.html">genophenocorr.model package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.preprocessing.html">genophenocorr.preprocessing package</a></li>
<li class="toctree-l4"><a class="reference internal" href="genophenocorr.view.html">genophenocorr.view package</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">genophenocorr</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="modules.html">API reference</a></li>
          <li class="breadcrumb-item"><a href="genophenocorr.html">genophenocorr package</a></li>
      <li class="breadcrumb-item active">genophenocorr.analysis package</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/apidocs/genophenocorr.analysis.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-genophenocorr.analysis">
<span id="genophenocorr-analysis-package"></span><h1>genophenocorr.analysis package<a class="headerlink" href="#module-genophenocorr.analysis" title="Link to this heading"></a></h1>
<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.</span></span><span class="sig-name descname"><span class="pre">CohortAnalysis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cohort</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transcript</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hpo</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recessive</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">include_unmeasured</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_val_correction</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'bonferroni'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">include_large_SV</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_perc_patients_w_hpo</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_analyzers.html#CohortAnalysis"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference external" href="https://docs.python.org/3/library/functions.html#object" title="(in Python v3.12)"><code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></a></p>
<p>This class can be used to run different analyses of a given Cohort.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.analysis_cohort">
<span class="sig-name descname"><span class="pre">analysis_cohort</span></span><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.analysis_cohort" title="Link to this definition"></a></dt>
<dd><p>The Cohort object given for this analysis</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Cohort" title="genophenocorr.model.Cohort">Cohort</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.analysis_patients">
<span class="sig-name descname"><span class="pre">analysis_patients</span></span><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.analysis_patients" title="Link to this definition"></a></dt>
<dd><p>A subset of patients with no structural variants if they are being removed from the analysis. Otherwise, all patients in the cohort.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#list" title="(in Python v3.12)">list</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.analysis_transcript">
<span class="sig-name descname"><span class="pre">analysis_transcript</span></span><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.analysis_transcript" title="Link to this definition"></a></dt>
<dd><p>The given transcript ID that will be used in this analysis</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>string</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.is_recessive">
<span class="sig-name descname"><span class="pre">is_recessive</span></span><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.is_recessive" title="Link to this definition"></a></dt>
<dd><p>True if the variant is recessive. Default is False.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>boolean</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.include_unmeasured">
<span class="sig-name descname"><span class="pre">include_unmeasured</span></span><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.include_unmeasured" title="Link to this definition"></a></dt>
<dd><p>True if we want to assume a patient without a specific phenotype listed does not have that phenotype. Otherwise, will only include those that explicitely say the phenotype was not observed. Defaults to True.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>boolean</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.include_large_SV">
<span class="sig-name descname"><span class="pre">include_large_SV</span></span><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.include_large_SV" title="Link to this definition"></a></dt>
<dd><p>True if we want to include large structural variants in the analysis. Defaults to True.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>boolean</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.min_perc_patients_w_hpo">
<span class="sig-name descname"><span class="pre">min_perc_patients_w_hpo</span></span><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.min_perc_patients_w_hpo" title="Link to this definition"></a></dt>
<dd><p>Will only test phenotypes that are listed in at least this percent of patients. Defaults to 10 (10%).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>integer</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">compare_by_variant_type(var_type1</span></span></dt>
<dd><p>VariantEffect, var_type2:Optional VariantEffect): Runs Fisher Exact analysis, finds any correlation between given variant effects and phenotypes</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">compare_by_variant_type(variant1</span></span></dt>
<dd><p>string, variant2:Optional string): Runs Fisher Exact analysis, finds any correlation between given variants and phenotypes</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">compare_by_variant_type(exon1</span></span></dt>
<dd><p>integer, exon2:Optional integer): Runs Fisher Exact analysis, finds any correlation between given affected exons and phenotypes</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">compare_by_variant_type(feature1</span></span></dt>
<dd><p>FeatureType, feature2:Optional FeatureType): Runs Fisher Exact analysis, finds any correlation between given protein feature types and phenotypes</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">compare_by_variant_type(feature1</span></span></dt>
<dd><p>string, feature2:Optional string): Runs Fisher Exact analysis, finds any correlation between given protein features and phenotypes</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="id0">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">analysis_cohort</span></span><a class="headerlink" href="#id0" title="Link to this definition"></a></dt>
<dd><p>Returns:
Cohort: The Cohort object being analyzed</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="id1">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">analysis_patients</span></span><a class="headerlink" href="#id1" title="Link to this definition"></a></dt>
<dd><p>Returns:
list: A subset of patients with no structural variants if they are being removed from the analysis. Otherwise, all patients in the cohort.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="id2">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">analysis_transcript</span></span><a class="headerlink" href="#id2" title="Link to this definition"></a></dt>
<dd><p>Returns:
string: The given transcript ID that will be used in this analysis</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="id3">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">is_recessive</span></span><a class="headerlink" href="#id3" title="Link to this definition"></a></dt>
<dd><p>Returns:
boolean: True if the variant is recessive.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="id4">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">include_unmeasured</span></span><a class="headerlink" href="#id4" title="Link to this definition"></a></dt>
<dd><p>Returns:
boolean: True if we want to assume a patient without a specific phenotype listed does not have that phenotype.
Otherwise, will only include those that explicitely say the phenotype was not observed.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="id5">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">include_large_SV</span></span><a class="headerlink" href="#id5" title="Link to this definition"></a></dt>
<dd><p>Returns:
boolean: True if we want to include large structural variants in the analysis.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="id6">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">min_perc_patients_w_hpo</span></span><a class="headerlink" href="#id6" title="Link to this definition"></a></dt>
<dd><p>Returns:
integer: Will only test phenotypes that are listed in at least this percent of patients.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_variant_type">
<span class="sig-name descname"><span class="pre">compare_by_variant_type</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">var_type1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">var_type2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_analyzers.html#CohortAnalysis.compare_by_variant_type"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_variant_type" title="Link to this definition"></a></dt>
<dd><p>Runs Fisher Exact analysis, finds any correlation between given variant effects across phenotypes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>var_type1</strong> (<a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model.VariantEffect"><em>VariantEffect</em></a>) – </p></li>
<li><p><strong>var_type2</strong> (<a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model.VariantEffect"><em>VariantEffect</em></a><em>, </em><em>Optional</em>) – If None, we compare between those with var_type1 and those without var_type1</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A pandas DataFrame showing the results of the analysis</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_variant">
<span class="sig-name descname"><span class="pre">compare_by_variant</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_analyzers.html#CohortAnalysis.compare_by_variant"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_variant" title="Link to this definition"></a></dt>
<dd><p>Runs Fisher Exact analysis, finds any correlation between given variants across phenotypes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>variant1</strong> (<em>string</em>) – </p></li>
<li><p><strong>variant2</strong> (<em>string</em><em>, </em><em>Optional</em>) – If None, we compare between those with variant1 and those without variant1</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A pandas DataFrame showing the results of the analysis</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_exon">
<span class="sig-name descname"><span class="pre">compare_by_exon</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">exon1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">exon2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_analyzers.html#CohortAnalysis.compare_by_exon"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_exon" title="Link to this definition"></a></dt>
<dd><p>Runs Fisher Exact analysis, finds any correlation between given exons across phenotypes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>exon1</strong> (<em>integer</em>) – </p></li>
<li><p><strong>exon2</strong> (<em>integer</em><em>, </em><em>Optional</em>) – If None, we compare between those within exon1 and those outside exon1</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A pandas DataFrame showing the results of the analysis</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature_type">
<span class="sig-name descname"><span class="pre">compare_by_protein_feature_type</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">feature1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model._protein.FeatureType"><span class="pre">FeatureType</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model._protein.FeatureType"><span class="pre">FeatureType</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_analyzers.html#CohortAnalysis.compare_by_protein_feature_type"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature_type" title="Link to this definition"></a></dt>
<dd><p>Runs Fisher Exact analysis, finds any correlation between given feature type across phenotypes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>feature1</strong> (<a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model.FeatureType"><em>FeatureType</em></a>) – </p></li>
<li><p><strong>feature2</strong> (<a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model.FeatureType"><em>FeatureType</em></a><em>, </em><em>Optional</em>) – If None, we compare between those with feature1 and those without feature1</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A pandas DataFrame showing the results of the analysis</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature">
<span class="sig-name descname"><span class="pre">compare_by_protein_feature</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">feature1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/_analyzers.html#CohortAnalysis.compare_by_protein_feature"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature" title="Link to this definition"></a></dt>
<dd><p>Runs Fisher Exact analysis, finds any correlation between given feature and phenotypes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>feature1</strong> (<em>string</em>) – </p></li>
<li><p><strong>feature2</strong> (<em>string</em><em>, </em><em>Optional</em>) – If None, we compare between those within feature1 and those outside feature1</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A pandas DataFrame showing the results of the analysis</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>DataFrame</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<section id="subpackages">
<h2>Subpackages<a class="headerlink" href="#subpackages" title="Link to this heading"></a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="genophenocorr.analysis.predicate.html">genophenocorr.analysis.predicate package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.SimplePredicate"><code class="docutils literal notranslate"><span class="pre">SimplePredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory"><code class="docutils literal notranslate"><span class="pre">PatientCategory</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PolyPredicate"><code class="docutils literal notranslate"><span class="pre">PolyPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.HPOPresentPredicate"><code class="docutils literal notranslate"><span class="pre">HPOPresentPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.VariantEffectPredicate"><code class="docutils literal notranslate"><span class="pre">VariantEffectPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.VariantPredicate"><code class="docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ExonPredicate"><code class="docutils literal notranslate"><span class="pre">ExonPredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ProtFeaturePredicate"><code class="docutils literal notranslate"><span class="pre">ProtFeaturePredicate</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ProtFeatureTypePredicate"><code class="docutils literal notranslate"><span class="pre">ProtFeatureTypePredicate</span></code></a></li>
</ul>
</li>
</ul>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="genophenocorr.html" class="btn btn-neutral float-left" title="genophenocorr package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="genophenocorr.analysis.predicate.html" class="btn btn-neutral float-right" title="genophenocorr.analysis.predicate package" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>