<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>genophenocorr.analysis.predicate.genotype package &mdash; genophenocorr 0.1.1dev documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/genophenocorr.css?v=a9fa5eb9" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=8427294d"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            genophenocorr
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">genophenocorr</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">genophenocorr.analysis.predicate.genotype package</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/apidocs/genophenocorr.analysis.predicate.genotype.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-genophenocorr.analysis.predicate.genotype">
<span id="genophenocorr-analysis-predicate-genotype-package"></span><h1>genophenocorr.analysis.predicate.genotype package<a class="headerlink" href="#module-genophenocorr.analysis.predicate.genotype" title="Link to this heading"></a></h1>
<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantEffectPredicate">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.predicate.genotype.</span></span><span class="sig-name descname"><span class="pre">VariantEffectPredicate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">effect</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#VariantEffectPredicate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantEffectPredicate" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.BooleanPredicate" title="genophenocorr.analysis.predicate._api.BooleanPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">BooleanPredicate</span></code></a></p>
<p><cite>VariantEffectPredicate</cite> tests if the <cite>patient</cite> has at least one variant that is predicted to have
the functional <cite>effect</cite> on the transcript of interest.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>transcript_id</strong> – the accession of the transcript of interest.</p></li>
<li><p><strong>effect</strong> – the tested variant effect.</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantEffectPredicate.get_question">
<span class="sig-name descname"><span class="pre">get_question</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#VariantEffectPredicate.get_question"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantEffectPredicate.get_question" title="Link to this definition"></a></dt>
<dd><p>Prepare a <cite>str</cite> with the question the predicate can answer.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantEffectPredicate.test">
<span class="sig-name descname"><span class="pre">test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patient</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#VariantEffectPredicate.test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantEffectPredicate.test" title="Link to this definition"></a></dt>
<dd><p>Assign a <cite>patient</cite> into a category.</p>
<p>Return <cite>None</cite> if the patient cannot be assigned into any meaningful category.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.predicate.genotype.</span></span><span class="sig-name descname"><span class="pre">VariantEffectsPredicate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">effect1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">effect2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantEffect" title="genophenocorr.model._variant_effects.VariantEffect"><span class="pre">VariantEffect</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#VariantEffectsPredicate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.GroupingPredicate" title="genophenocorr.analysis.predicate._api.GroupingPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">GroupingPredicate</span></code></a></p>
<p><cite>VariantEffectsPredicate</cite> tests if the <cite>patient</cite> has at least one variant that is predicted to have
one of the two functional <cite>effects</cite> on the transcript of interest.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>transcript_id</strong> – the accession of the transcript of interest.</p></li>
<li><p><strong>effect1</strong> – the first tested variant effect.</p></li>
<li><p><strong>effect2</strong> – the second tested variant effect.</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate.get_question">
<span class="sig-name descname"><span class="pre">get_question</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#VariantEffectsPredicate.get_question"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate.get_question" title="Link to this definition"></a></dt>
<dd><p>Prepare a <cite>str</cite> with the question the predicate can answer.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate.test">
<span class="sig-name descname"><span class="pre">test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patient</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#VariantEffectsPredicate.test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate.test" title="Link to this definition"></a></dt>
<dd><p>Assign a <cite>patient</cite> into a category.</p>
<p>Return <cite>None</cite> if the patient cannot be assigned into any meaningful category.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantPredicate">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.predicate.genotype.</span></span><span class="sig-name descname"><span class="pre">VariantPredicate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant_key</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#VariantPredicate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantPredicate" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.BooleanPredicate" title="genophenocorr.analysis.predicate._api.BooleanPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">BooleanPredicate</span></code></a></p>
<p><cite>VariantPredicate</cite> tests if the <cite>patient</cite> has ar least one allele of the variant described by the <cite>variant_key</cite>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can get the variant key by calling <a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantCoordinates.variant_key" title="genophenocorr.model.VariantCoordinates.variant_key"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.model.VariantCoordinates.variant_key</span></code></a>.</p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>variant_key</strong> – a <cite>str</cite> with the variant key.</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantPredicate.get_question">
<span class="sig-name descname"><span class="pre">get_question</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#VariantPredicate.get_question"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantPredicate.get_question" title="Link to this definition"></a></dt>
<dd><p>Prepare a <cite>str</cite> with the question the predicate can answer.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantPredicate.test">
<span class="sig-name descname"><span class="pre">test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patient</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#VariantPredicate.test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantPredicate.test" title="Link to this definition"></a></dt>
<dd><p>Assign a <cite>patient</cite> into a category.</p>
<p>Return <cite>None</cite> if the patient cannot be assigned into any meaningful category.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantsPredicate">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.predicate.genotype.</span></span><span class="sig-name descname"><span class="pre">VariantsPredicate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant_key1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">variant_key2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#VariantsPredicate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantsPredicate" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.GroupingPredicate" title="genophenocorr.analysis.predicate._api.GroupingPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">GroupingPredicate</span></code></a></p>
<p><cite>VariantsPredicate</cite> tests if the <cite>patient</cite> has at least one allele of one of the variants described by
the <cite>variant_key1</cite> and <cite>variant_key2</cite>.
<strong>If patient has both variant_keys, it is not included</strong></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can get the variant key by calling <a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.VariantCoordinates.variant_key" title="genophenocorr.model.VariantCoordinates.variant_key"><code class="xref py py-class docutils literal notranslate"><span class="pre">genophenocorr.model.VariantCoordinates.variant_key</span></code></a>.</p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>variant_key1</strong> – a <cite>str</cite> with a variant key.</p></li>
<li><p><strong>variant_key2</strong> – a <cite>str</cite> with a variant key.</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantsPredicate.get_question">
<span class="sig-name descname"><span class="pre">get_question</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#VariantsPredicate.get_question"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantsPredicate.get_question" title="Link to this definition"></a></dt>
<dd><p>Prepare a <cite>str</cite> with the question the predicate can answer.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.VariantsPredicate.test">
<span class="sig-name descname"><span class="pre">test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patient</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#VariantsPredicate.test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.VariantsPredicate.test" title="Link to this definition"></a></dt>
<dd><p>Assign a <cite>patient</cite> into a category.</p>
<p>Return <cite>None</cite> if the patient cannot be assigned into any meaningful category.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ExonPredicate">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.predicate.genotype.</span></span><span class="sig-name descname"><span class="pre">ExonPredicate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">exon_number</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#ExonPredicate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ExonPredicate" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.BooleanPredicate" title="genophenocorr.analysis.predicate._api.BooleanPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">BooleanPredicate</span></code></a></p>
<p><cite>ExonPredicate</cite> tests if the <cite>patient</cite> has a variant that affects <em>n</em>-th exon of the transcript of interest.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>We use 1-based numbering to number the exons, not the usual 0-based numbering of the computer science.
Therefore, the first exon of the transcript has <code class="docutils literal notranslate"><span class="pre">exon_number==1</span></code>, the second exon is <code class="docutils literal notranslate"><span class="pre">2</span></code>, and so on …</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>We do not check if the <cite>exon_number</cite> spans beyond the number of exons of the given <cite>transcript_id</cite>!
Therefore, <code class="docutils literal notranslate"><span class="pre">exon_number==10,000</span></code> will effectively return <code class="xref py py-attr docutils literal notranslate"><span class="pre">BooleanPredicate.FALSE</span></code> for <em>all</em> patients!!! 😱
Well, at least the patients of the <em>Homo sapiens sapiens</em> taxon…</p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>transcript_id</strong> – the accession of the transcript of interest.</p></li>
<li><p><strong>exon_number</strong> – a positive <cite>int</cite> of the target exon.</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ExonPredicate.get_question">
<span class="sig-name descname"><span class="pre">get_question</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#ExonPredicate.get_question"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ExonPredicate.get_question" title="Link to this definition"></a></dt>
<dd><p>Prepare a <cite>str</cite> with the question the predicate can answer.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ExonPredicate.test">
<span class="sig-name descname"><span class="pre">test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patient</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#ExonPredicate.test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ExonPredicate.test" title="Link to this definition"></a></dt>
<dd><p>Assign a <cite>patient</cite> into a category.</p>
<p>Return <cite>None</cite> if the patient cannot be assigned into any meaningful category.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ExonsPredicate">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.predicate.genotype.</span></span><span class="sig-name descname"><span class="pre">ExonsPredicate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">exon1_number</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">exon2_number</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.12)"><span class="pre">int</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#ExonsPredicate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ExonsPredicate" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.GroupingPredicate" title="genophenocorr.analysis.predicate._api.GroupingPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">GroupingPredicate</span></code></a></p>
<p><cite>ExonsPredicate</cite> tests if the <cite>patient</cite> has a variant that affects one of the
two given <em>n</em>-th exons of the transcript of interest.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>We use 1-based numbering to number the exons, not the usual 0-based numbering of the computer science.
Therefore, the first exon of the transcript has <code class="docutils literal notranslate"><span class="pre">exon_number==1</span></code>, the second exon is <code class="docutils literal notranslate"><span class="pre">2</span></code>, and so on …</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>We do not check if the <cite>exon_number</cite> spans beyond the number of exons of the given <cite>transcript_id</cite>!
Therefore, <code class="docutils literal notranslate"><span class="pre">exon_number==10,000</span></code> will effectively return <cite>None</cite> for <em>all</em> patients!!! 😱
Well, at least the patients of the <em>Homo sapiens sapiens</em> taxon…</p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>transcript_id</strong> – the accession of the transcript of interest.</p></li>
<li><p><strong>exon1_number</strong> – a positive <cite>int</cite> of the first target exon.</p></li>
<li><p><strong>exon2_number</strong> – a positive <cite>int</cite> of the second target exon.</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ExonsPredicate.get_question">
<span class="sig-name descname"><span class="pre">get_question</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#ExonsPredicate.get_question"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ExonsPredicate.get_question" title="Link to this definition"></a></dt>
<dd><p>Prepare a <cite>str</cite> with the question the predicate can answer.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ExonsPredicate.test">
<span class="sig-name descname"><span class="pre">test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patient</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#ExonsPredicate.test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ExonsPredicate.test" title="Link to this definition"></a></dt>
<dd><p>Assign a <cite>patient</cite> into a category.</p>
<p>Return <cite>None</cite> if the patient cannot be assigned into any meaningful category.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.predicate.genotype.</span></span><span class="sig-name descname"><span class="pre">ProtFeaturePredicate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_feature_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_service</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.preprocessing.html#genophenocorr.preprocessing.ProteinMetadataService" title="genophenocorr.preprocessing._api.ProteinMetadataService"><span class="pre">ProteinMetadataService</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#ProtFeaturePredicate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.BooleanPredicate" title="genophenocorr.analysis.predicate._api.BooleanPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">BooleanPredicate</span></code></a></p>
<p><cite>ProtFeaturePredicate</cite> tests if the <cite>patient</cite> has a variant that overlaps with a protein feature.</p>
<p>The predicate needs the name of the protein feature.
For instance, <cite>EGF-like 2</cite> for <a class="reference external" href="https://www.uniprot.org/uniprotkb/P35555/entry#family_and_domains">FBN1</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>transcript_id</strong> – the accession of the transcript of interest.</p></li>
<li><p><strong>protein_feature_name</strong> – a <cite>str</cite> with the name of the protein feature.</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate.get_question">
<span class="sig-name descname"><span class="pre">get_question</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#ProtFeaturePredicate.get_question"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate.get_question" title="Link to this definition"></a></dt>
<dd><p>Prepare a <cite>str</cite> with the question the predicate can answer.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate.test">
<span class="sig-name descname"><span class="pre">test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patient</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#ProtFeaturePredicate.test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate.test" title="Link to this definition"></a></dt>
<dd><p>Assign a <cite>patient</cite> into a category.</p>
<p>Return <cite>None</cite> if the patient cannot be assigned into any meaningful category.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.predicate.genotype.</span></span><span class="sig-name descname"><span class="pre">ProtFeaturesPredicate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_feature1_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_feature2_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_service</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.preprocessing.html#genophenocorr.preprocessing.ProteinMetadataService" title="genophenocorr.preprocessing._api.ProteinMetadataService"><span class="pre">ProteinMetadataService</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#ProtFeaturesPredicate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.GroupingPredicate" title="genophenocorr.analysis.predicate._api.GroupingPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">GroupingPredicate</span></code></a></p>
<dl class="simple">
<dt><cite>ProtFeaturesPredicate</cite> tests if the <cite>patient</cite> has a variant that overlaps with one of the two given</dt><dd><p>protein features.</p>
</dd>
</dl>
<p>The predicate needs the name of the protein feature.
For instance, <cite>EGF-like 2</cite> for <a class="reference external" href="https://www.uniprot.org/uniprotkb/P35555/entry#family_and_domains">FBN1</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>transcript_id</strong> – the accession of the transcript of interest.</p></li>
<li><p><strong>protein_feature1_name</strong> – a <cite>str</cite> with the name of the first protein feature.</p></li>
<li><p><strong>protein_feature2_name</strong> – a <cite>str</cite> with the name of the second protein feature.</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate.get_question">
<span class="sig-name descname"><span class="pre">get_question</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#ProtFeaturesPredicate.get_question"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate.get_question" title="Link to this definition"></a></dt>
<dd><p>Prepare a <cite>str</cite> with the question the predicate can answer.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate.test">
<span class="sig-name descname"><span class="pre">test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patient</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#ProtFeaturesPredicate.test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate.test" title="Link to this definition"></a></dt>
<dd><p>Assign a <cite>patient</cite> into a category.</p>
<p>Return <cite>None</cite> if the patient cannot be assigned into any meaningful category.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.predicate.genotype.</span></span><span class="sig-name descname"><span class="pre">ProtFeatureTypePredicate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model._protein.FeatureType"><span class="pre">FeatureType</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_service</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.preprocessing.html#genophenocorr.preprocessing.ProteinMetadataService" title="genophenocorr.preprocessing._api.ProteinMetadataService"><span class="pre">ProteinMetadataService</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#ProtFeatureTypePredicate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.BooleanPredicate" title="genophenocorr.analysis.predicate._api.BooleanPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">BooleanPredicate</span></code></a></p>
<p><cite>ProtFeatureTypePredicate</cite> tests if the <cite>patient</cite> has a variant that affects a <code class="xref py py-class docutils literal notranslate"><span class="pre">FeatureType</span></code>
in the protein encoded by the transcript of interest.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>transcript_id</strong> – the accession of the transcript of interest.</p></li>
<li><p><strong>feature_type</strong> – an instance of the target <code class="xref py py-class docutils literal notranslate"><span class="pre">FeatureType</span></code>.</p></li>
<li><p><strong>protein_service</strong> – an instance of a <code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinMetadataService</span></code>.</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate.get_question">
<span class="sig-name descname"><span class="pre">get_question</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#ProtFeatureTypePredicate.get_question"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate.get_question" title="Link to this definition"></a></dt>
<dd><p>Prepare a <cite>str</cite> with the question the predicate can answer.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate.test">
<span class="sig-name descname"><span class="pre">test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patient</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_bool.html#ProtFeatureTypePredicate.test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate.test" title="Link to this definition"></a></dt>
<dd><p>Assign a <cite>patient</cite> into a category.</p>
<p>Return <cite>None</cite> if the patient cannot be assigned into any meaningful category.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">genophenocorr.analysis.predicate.genotype.</span></span><span class="sig-name descname"><span class="pre">ProtFeatureTypesPredicate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">transcript_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_type1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model._protein.FeatureType"><span class="pre">FeatureType</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_type2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.FeatureType" title="genophenocorr.model._protein.FeatureType"><span class="pre">FeatureType</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_service</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.preprocessing.html#genophenocorr.preprocessing.ProteinMetadataService" title="genophenocorr.preprocessing._api.ProteinMetadataService"><span class="pre">ProteinMetadataService</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#ProtFeatureTypesPredicate"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.GroupingPredicate" title="genophenocorr.analysis.predicate._api.GroupingPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">GroupingPredicate</span></code></a></p>
<p><cite>ProtFeatureTypesPredicate</cite> tests if the <cite>patient</cite> has a variant that affects one of the two given
:class:<a href="#id2"><span class="problematic" id="id3">`</span></a>FeatureType`s in the transcript of interest.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>transcript_id</strong> – the accession of the transcript of interest.</p></li>
<li><p><strong>feature_type1</strong> – an instance of the first target <code class="xref py py-class docutils literal notranslate"><span class="pre">FeatureType</span></code>.</p></li>
<li><p><strong>feature_type2</strong> – an instance of the second target <code class="xref py py-class docutils literal notranslate"><span class="pre">FeatureType</span></code>.</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate.get_question">
<span class="sig-name descname"><span class="pre">get_question</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.12)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#ProtFeatureTypesPredicate.get_question"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate.get_question" title="Link to this definition"></a></dt>
<dd><p>Prepare a <cite>str</cite> with the question the predicate can answer.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate.test">
<span class="sig-name descname"><span class="pre">test</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">patient</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="genophenocorr.model.html#genophenocorr.model.Patient" title="genophenocorr.model._cohort.Patient"><span class="pre">Patient</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.PatientCategory" title="genophenocorr.analysis.predicate._api.PatientCategory"><span class="pre">PatientCategory</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.12)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/genophenocorr/analysis/predicate/genotype/_geno_group.html#ProtFeatureTypesPredicate.test"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate.test" title="Link to this definition"></a></dt>
<dd><p>Assign a <cite>patient</cite> into a category.</p>
<p>Return <cite>None</cite> if the patient cannot be assigned into any meaningful category.</p>
</dd></dl>

</dd></dl>

</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>