

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Index &mdash; GPSEA 0.5.1.dev0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="_static/gpsea.css?v=547657ed" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=215e94d7"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="_static/copybutton.js?v=6dbb43f8"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="#" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Index</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             

<h1 id="index">Index</h1>

<div class="genindex-jumpbox">
 <a href="#A"><strong>A</strong></a>
 | <a href="#B"><strong>B</strong></a>
 | <a href="#C"><strong>C</strong></a>
 | <a href="#D"><strong>D</strong></a>
 | <a href="#E"><strong>E</strong></a>
 | <a href="#F"><strong>F</strong></a>
 | <a href="#G"><strong>G</strong></a>
 | <a href="#H"><strong>H</strong></a>
 | <a href="#I"><strong>I</strong></a>
 | <a href="#L"><strong>L</strong></a>
 | <a href="#M"><strong>M</strong></a>
 | <a href="#N"><strong>N</strong></a>
 | <a href="#O"><strong>O</strong></a>
 | <a href="#P"><strong>P</strong></a>
 | <a href="#R"><strong>R</strong></a>
 | <a href="#S"><strong>S</strong></a>
 | <a href="#T"><strong>T</strong></a>
 | <a href="#U"><strong>U</strong></a>
 | <a href="#V"><strong>V</strong></a>
 | <a href="#W"><strong>W</strong></a>
 | <a href="#Y"><strong>Y</strong></a>
 
</div>
<h2 id="A">A</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad.add_error">add_error() (gpsea.preprocessing.Notepad method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad.add_issue">add_issue() (gpsea.preprocessing.Notepad method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad.add_subsection">add_subsection() (gpsea.preprocessing.Notepad method)</a>

      <ul>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.NotepadTree.add_subsection">(gpsea.preprocessing.NotepadTree method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad.add_warning">add_warning() (gpsea.preprocessing.Notepad method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.all">all() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.all_counts">all_counts (gpsea.analysis.pcats.MultiPhenotypeAnalysisResult property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.all_diseases">all_diseases() (gpsea.model.Cohort method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.all_patients">all_patients (gpsea.model.Cohort property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.all_phenotypes">all_phenotypes() (gpsea.model.Cohort method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.all_transcript_ids">all_transcript_ids (gpsea.model.Cohort property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.all_variants">all_variants() (gpsea.model.Cohort method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.AlleleCounter">AlleleCounter (class in gpsea.analysis.predicate.genotype)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.alt">alt (gpsea.model.VariantCoordinates property)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator.annotate">annotate() (gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator method)</a>

      <ul>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.FunctionalAnnotator.annotate">(gpsea.preprocessing.FunctionalAnnotator method)</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.ImpreciseSvFunctionalAnnotator.annotate">(gpsea.preprocessing.ImpreciseSvFunctionalAnnotator method)</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.ProtCachingMetadataService.annotate">(gpsea.preprocessing.ProtCachingMetadataService method)</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.ProteinMetadataService.annotate">(gpsea.preprocessing.ProteinMetadataService method)</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.UniprotProteinMetadataService.annotate">(gpsea.preprocessing.UniprotProteinMetadataService method)</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VarCachingFunctionalAnnotator.annotate">(gpsea.preprocessing.VarCachingFunctionalAnnotator method)</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VepFunctionalAnnotator.annotate">(gpsea.preprocessing.VepFunctionalAnnotator method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.any">any() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.apply_predicates_on_patients">apply_predicates_on_patients() (in module gpsea.analysis.pcats)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Auditor">Auditor (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.autosomal_dominant">autosomal_dominant() (in module gpsea.analysis.predicate.genotype)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.autosomal_recessive">autosomal_recessive() (in module gpsea.analysis.predicate.genotype)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="B">B</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.biallelic_predicate">biallelic_predicate() (in module gpsea.analysis.predicate.genotype)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="C">C</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.config.html#gpsea.config.CACHE_ENV">CACHE_ENV (in module gpsea.config)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategory.cat_id">cat_id (gpsea.analysis.predicate.PatientCategory property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.Categorization">Categorization (class in gpsea.analysis.predicate)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.Categorization.category">category (gpsea.analysis.predicate.Categorization property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.cds_end">cds_end (gpsea.model.TranscriptCoordinates property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.cds_start">cds_start (gpsea.model.TranscriptCoordinates property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.change_length">change_length (gpsea.model.VariantCoordinates property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.change_length">change_length() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.NotepadTree.children">children (gpsea.preprocessing.NotepadTree property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.chrom">chrom (gpsea.model.VariantCoordinates property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcIssue.code">code (gpsea.analysis.mtc_filter.PhenotypeMtcIssue attribute)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotype.code">(gpsea.model.Genotype property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.CODING_SEQUENCE_VARIANT">CODING_SEQUENCE_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort">Cohort (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.CohortCreator">CohortCreator (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.CohortVariantViewer">CohortVariantViewer (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.CohortViewable">CohortViewable (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureType.COILED_COIL">COILED_COIL (gpsea.model.FeatureType attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysis.compare_genotype_vs_phenotype_score">compare_genotype_vs_phenotype_score() (gpsea.analysis.pscore.PhenotypeScoreAnalysis method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysis.compare_genotype_vs_phenotypes">compare_genotype_vs_phenotypes() (gpsea.analysis.pcats.MultiPhenotypeAnalysis method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureType.COMPOSITIONAL_BIAS">COMPOSITIONAL_BIAS (gpsea.model.FeatureType attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.stats.html#gpsea.analysis.pcats.stats.CountStatistic.compute_pval">compute_pval() (gpsea.analysis.pcats.stats.CountStatistic method)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.pcats.stats.html#gpsea.analysis.pcats.stats.FisherExactTest.compute_pval">(gpsea.analysis.pcats.stats.FisherExactTest method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.pscore.stats.html#gpsea.analysis.pscore.stats.MannWhitneyStatistic.compute_pval">(gpsea.analysis.pscore.stats.MannWhitneyStatistic method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.pscore.stats.html#gpsea.analysis.pscore.stats.PhenotypeScoreStatistic.compute_pval">(gpsea.analysis.pscore.stats.PhenotypeScoreStatistic method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.pscore.stats.html#gpsea.analysis.pscore.stats.TTestStatistic.compute_pval">(gpsea.analysis.pscore.stats.TTestStatistic method)</a>
</li>
      </ul></li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_caching_cohort_creator">configure_caching_cohort_creator() (in module gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_caching_patient_creator">configure_caching_patient_creator() (in module gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_cohort_creator">configure_cohort_creator() (in module gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_default_protein_metadata_service">configure_default_protein_metadata_service() (in module gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_patient_creator">configure_patient_creator() (in module gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_protein_metadata_service">configure_protein_metadata_service() (in module gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomicRegion.contains">contains() (gpsea.model.genome.GenomicRegion method)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Region.contains">(gpsea.model.genome.Region method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Region.contains_pos">contains_pos() (gpsea.model.genome.Region method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Contig">Contig (class in gpsea.model.genome)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomicRegion.contig">contig (gpsea.model.genome.GenomicRegion property)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomeBuild.contig_by_name">contig_by_name() (gpsea.model.genome.GenomeBuild method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomeBuild.contigs">contigs (gpsea.model.genome.GenomeBuild property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.corrected_pvals">corrected_pvals (gpsea.analysis.pcats.MultiPhenotypeAnalysisResult property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.AlleleCounter.count">count() (gpsea.analysis.predicate.genotype.AlleleCounter method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.CountingPhenotypeScorer">CountingPhenotypeScorer (class in gpsea.analysis.pscore)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.stats.html#gpsea.analysis.pcats.stats.CountStatistic">CountStatistic (class in gpsea.analysis.pcats.stats)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinFeature.create">create() (gpsea.model.ProteinFeature static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Variant.create_variant_from_scratch">create_variant_from_scratch() (gpsea.model.Variant static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.curie">curie (gpsea.model.VariantEffect property)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="D">D</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.DataSanityIssue">DataSanityIssue (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.io.html#gpsea.io.GpseaJSONEncoder.default">default() (gpsea.io.GpseaJSONEncoder method)</a>
</li>
      <li><a href="apidocs/gpsea.config.html#gpsea.config.DEFAULT_CACHE_PATH">DEFAULT_CACHE_PATH (in module gpsea.config)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.default_filter">default_filter() (gpsea.analysis.mtc_filter.HpoMtcFilter static method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator">DefaultImpreciseSvFunctionalAnnotator (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantClass.DEL">DEL (gpsea.model.VariantClass attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategory.description">description (gpsea.analysis.predicate.PatientCategory property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.DeVriesPhenotypeScorer">DeVriesPhenotypeScorer (class in gpsea.analysis.pscore)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.diagnosis_predicate">diagnosis_predicate() (in module gpsea.analysis.predicate.genotype)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Disease">Disease (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.DiseaseAnalysis">DiseaseAnalysis (class in gpsea.analysis.pcats)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.DiseasePresencePredicate">DiseasePresencePredicate (class in gpsea.analysis.predicate.phenotype)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.diseases">diseases (gpsea.model.Patient property)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.DiseaseViewable">DiseaseViewable (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate.display_question">display_question() (gpsea.analysis.predicate.PolyPredicate method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomicRegion.distance_to">distance_to() (gpsea.model.genome.GenomicRegion method)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Region.distance_to">(gpsea.model.genome.Region method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureType.DOMAIN">DOMAIN (gpsea.model.FeatureType attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.domains">domains() (gpsea.model.ProteinMetadata method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.DOWNSTREAM_GENE_VARIANT">DOWNSTREAM_GENE_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizer.draw_fig">draw_fig() (gpsea.view.ProteinVisualizer method)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizer.draw_protein_diagram">draw_protein_diagram() (gpsea.view.ProteinVisualizer method)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.VariantTranscriptVisualizer.draw_variants">draw_variants() (gpsea.view.VariantTranscriptVisualizer method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantClass.DUP">DUP (gpsea.model.VariantClass attribute)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="E">E</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotypes.empty">empty() (gpsea.model.Genotypes static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureInfo.end">end (gpsea.model.FeatureInfo property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Region.end">(gpsea.model.genome.Region property)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.end">(gpsea.model.VariantCoordinates property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomicRegion.end_on_strand">end_on_strand() (gpsea.model.genome.GenomicRegion method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Level.ERROR">ERROR (gpsea.preprocessing.Level attribute)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad.error_count">error_count() (gpsea.preprocessing.Notepad method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad.errors">errors() (gpsea.preprocessing.Notepad method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.excluded_diseases">excluded_diseases() (gpsea.model.Patient method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.excluded_phenotypes">excluded_phenotypes() (gpsea.model.Patient method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.exon">exon() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.exons">exons (gpsea.model.TranscriptCoordinates property)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="F">F</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcResult.fail">fail() (gpsea.analysis.mtc_filter.PhenotypeMtcResult static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.FEATURE_ELONGATION">FEATURE_ELONGATION (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.FEATURE_TRUNCATION">FEATURE_TRUNCATION (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinFeature.feature_type">feature_type (gpsea.model.ProteinFeature property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureInfo">FeatureInfo (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureType">FeatureType (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Sex.FEMALE">FEMALE (gpsea.model.Sex attribute)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.TranscriptCoordinateService.fetch">fetch() (gpsea.preprocessing.TranscriptCoordinateService method)</a>

      <ul>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VVMultiCoordinateService.fetch">(gpsea.preprocessing.VVMultiCoordinateService method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.GeneCoordinateService.fetch_for_gene">fetch_for_gene() (gpsea.preprocessing.GeneCoordinateService method)</a>

      <ul>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VVMultiCoordinateService.fetch_for_gene">(gpsea.preprocessing.VVMultiCoordinateService method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VepFunctionalAnnotator.fetch_response">fetch_response() (gpsea.preprocessing.VepFunctionalAnnotator method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.filter">filter() (gpsea.analysis.mtc_filter.HpoMtcFilter method)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcFilter.filter">(gpsea.analysis.mtc_filter.PhenotypeMtcFilter method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.filter">(gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.filter">(gpsea.analysis.mtc_filter.UseAllTermsMtcFilter method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.filter_method_name">filter_method_name() (gpsea.analysis.mtc_filter.HpoMtcFilter method)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcFilter.filter_method_name">(gpsea.analysis.mtc_filter.PhenotypeMtcFilter method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.filter_method_name">(gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.filter_method_name">(gpsea.analysis.mtc_filter.UseAllTermsMtcFilter method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PhenopacketVariantCoordinateFinder.find_coordinates">find_coordinates() (gpsea.preprocessing.PhenopacketVariantCoordinateFinder method)</a>

      <ul>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VariantCoordinateFinder.find_coordinates">(gpsea.preprocessing.VariantCoordinateFinder method)</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VVHgvsVariantCoordinateFinder.find_coordinates">(gpsea.preprocessing.VVHgvsVariantCoordinateFinder method)</a>
</li>
      </ul></li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.pcats.stats.html#gpsea.analysis.pcats.stats.FisherExactTest">FisherExactTest (class in gpsea.analysis.pcats.stats)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.FIVE_PRIME_UTR_VARIANT">FIVE_PRIME_UTR_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotypes.for_sample">for_sample() (gpsea.model.Genotypes method)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.Formatter.format_as_string">format_as_string() (gpsea.view.Formatter method)</a>

      <ul>
        <li><a href="apidocs/gpsea.view.html#gpsea.view.VariantFormatter.format_as_string">(gpsea.view.VariantFormatter method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VepFunctionalAnnotator.format_coordinates_for_vep_query">format_coordinates_for_vep_query() (gpsea.preprocessing.VepFunctionalAnnotator static method)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.Formatter">Formatter (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.FRAMESHIFT_VARIANT">FRAMESHIFT_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.from_feature_frame">from_feature_frame() (gpsea.model.ProteinMetadata static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotypes.from_mapping">from_mapping() (gpsea.model.Genotypes static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.MeasurementPhenotypeScorer.from_measurement_id">from_measurement_id() (gpsea.analysis.pscore.MeasurementPhenotypeScorer static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.from_patients">from_patients() (gpsea.model.Cohort static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.CountingPhenotypeScorer.from_query_curies">from_query_curies() (gpsea.analysis.pscore.CountingPhenotypeScorer static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.Categorization.from_raw_parts">from_raw_parts() (gpsea.analysis.predicate.Categorization static method)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.from_raw_parts">(gpsea.model.Patient static method)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.Phenotype.from_raw_parts">(gpsea.model.Phenotype static method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureType.from_string">from_string() (gpsea.model.FeatureType static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Phenotype.from_term">from_term() (gpsea.model.Phenotype static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.from_uniprot_json">from_uniprot_json() (gpsea.model.ProteinMetadata static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.from_vcf_literal">from_vcf_literal() (gpsea.model.VariantCoordinates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.from_vcf_symbolic">from_vcf_symbolic() (gpsea.model.VariantCoordinates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FunctionalAnnotationAware">FunctionalAnnotationAware (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.FunctionalAnnotator">FunctionalAnnotator (class in gpsea.preprocessing)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="G">G</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Contig.genbank_acc">genbank_acc (gpsea.model.genome.Contig property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.gene">gene() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ImpreciseSvInfo.gene_id">gene_id (gpsea.model.ImpreciseSvInfo property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation.gene_id">(gpsea.model.TranscriptAnnotation property)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptInfoAware.gene_id">(gpsea.model.TranscriptInfoAware property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ImpreciseSvInfo.gene_symbol">gene_symbol (gpsea.model.ImpreciseSvInfo property)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.GeneCoordinateService">GeneCoordinateService (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomeBuild.genome_build_id">genome_build_id (gpsea.model.genome.GenomeBuild property)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomeBuild">GenomeBuild (class in gpsea.model.genome)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomeBuildIdentifier">GenomeBuildIdentifier (class in gpsea.model.genome)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomicRegion">GenomicRegion (class in gpsea.model.genome)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotype">Genotype (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotyped.genotype_for_sample">genotype_for_sample() (gpsea.model.Genotyped method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.genotype_phenotype_scores">genotype_phenotype_scores (gpsea.analysis.pscore.PhenotypeScoreAnalysisResult property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotyped">Genotyped (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.GenotypePolyPredicate">GenotypePolyPredicate (class in gpsea.analysis.predicate.genotype)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotypes">Genotypes (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotyped.genotypes">genotypes (gpsea.model.Genotyped property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.Variant.genotypes">(gpsea.model.Variant property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.ProteinAnnotationCache.get_annotations">get_annotations() (gpsea.preprocessing.ProteinAnnotationCache method)</a>

      <ul>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VariantAnnotationCache.get_annotations">(gpsea.preprocessing.VariantAnnotationCache method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.config.html#gpsea.config.get_cache_dir_path">get_cache_dir_path() (in module gpsea.config)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate.get_categories">get_categories() (gpsea.analysis.predicate.PolyPredicate method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.get_categorizations">get_categorizations() (gpsea.analysis.predicate.phenotype.DiseasePresencePredicate method)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.HpoPredicate.get_categorizations">(gpsea.analysis.predicate.phenotype.HpoPredicate method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate.get_categorizations">(gpsea.analysis.predicate.PolyPredicate method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate.get_category">get_category() (gpsea.analysis.predicate.PolyPredicate method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate.get_category_name">get_category_name() (gpsea.analysis.predicate.PolyPredicate method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate.get_category_names">get_category_names() (gpsea.analysis.predicate.PolyPredicate method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.get_cds_regions">get_cds_regions() (gpsea.model.TranscriptCoordinates method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.get_coding_base_count">get_coding_base_count() (gpsea.model.TranscriptCoordinates method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.get_codon_count">get_codon_count() (gpsea.model.TranscriptCoordinates method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.get_excluded_count">get_excluded_count() (gpsea.model.Cohort method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.get_features_variant_overlaps">get_features_variant_overlaps() (gpsea.model.ProteinMetadata method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.get_five_prime_utrs">get_five_prime_utrs() (gpsea.model.TranscriptCoordinates method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FunctionalAnnotationAware.get_hgvs_cdna_by_tx_id">get_hgvs_cdna_by_tx_id() (gpsea.model.FunctionalAnnotationAware method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.get_maximum_group_observed_HPO_frequency">get_maximum_group_observed_HPO_frequency() (gpsea.analysis.mtc_filter.HpoMtcFilter static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.get_number_of_observed_hpo_observations">get_number_of_observed_hpo_observations() (gpsea.analysis.mtc_filter.HpoMtcFilter static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.get_patient_ids">get_patient_ids() (gpsea.model.Cohort method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FunctionalAnnotationAware.get_preferred_tx_annotation">get_preferred_tx_annotation() (gpsea.model.FunctionalAnnotationAware method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.AlleleCounter.get_question">get_question() (gpsea.analysis.predicate.genotype.AlleleCounter method)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate.get_question">(gpsea.analysis.predicate.genotype.VariantPredicate method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.CountingPhenotypeScorer.get_question">(gpsea.analysis.pscore.CountingPhenotypeScorer method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.get_question_base">get_question_base() (gpsea.analysis.predicate.phenotype.DiseasePresencePredicate method)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.HpoPredicate.get_question_base">(gpsea.analysis.predicate.phenotype.HpoPredicate method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate.get_question_base">(gpsea.analysis.predicate.PolyPredicate method)</a>
</li>
      </ul></li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VVMultiCoordinateService.get_response">get_response() (gpsea.preprocessing.VVMultiCoordinateService method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.get_three_prime_utrs">get_three_prime_utrs() (gpsea.model.TranscriptCoordinates method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FunctionalAnnotationAware.get_tx_anno_by_tx_id">get_tx_anno_by_tx_id() (gpsea.model.FunctionalAnnotationAware method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.get_variant_by_key">get_variant_by_key() (gpsea.model.Cohort method)</a>
</li>
      <li>
    gpsea

      <ul>
        <li><a href="apidocs/gpsea.html#module-gpsea">module</a>
</li>
      </ul></li>
      <li>
    gpsea.analysis

      <ul>
        <li><a href="apidocs/gpsea.analysis.html#module-gpsea.analysis">module</a>
</li>
      </ul></li>
      <li>
    gpsea.analysis.mtc_filter

      <ul>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#module-gpsea.analysis.mtc_filter">module</a>
</li>
      </ul></li>
      <li>
    gpsea.analysis.pcats

      <ul>
        <li><a href="apidocs/gpsea.analysis.pcats.html#module-gpsea.analysis.pcats">module</a>
</li>
      </ul></li>
      <li>
    gpsea.analysis.pcats.stats

      <ul>
        <li><a href="apidocs/gpsea.analysis.pcats.stats.html#module-gpsea.analysis.pcats.stats">module</a>
</li>
      </ul></li>
      <li>
    gpsea.analysis.predicate

      <ul>
        <li><a href="apidocs/gpsea.analysis.predicate.html#module-gpsea.analysis.predicate">module</a>
</li>
      </ul></li>
      <li>
    gpsea.analysis.predicate.genotype

      <ul>
        <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#module-gpsea.analysis.predicate.genotype">module</a>
</li>
      </ul></li>
      <li>
    gpsea.analysis.predicate.phenotype

      <ul>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#module-gpsea.analysis.predicate.phenotype">module</a>
</li>
      </ul></li>
      <li>
    gpsea.analysis.pscore

      <ul>
        <li><a href="apidocs/gpsea.analysis.pscore.html#module-gpsea.analysis.pscore">module</a>
</li>
      </ul></li>
      <li>
    gpsea.analysis.pscore.stats

      <ul>
        <li><a href="apidocs/gpsea.analysis.pscore.stats.html#module-gpsea.analysis.pscore.stats">module</a>
</li>
      </ul></li>
      <li>
    gpsea.config

      <ul>
        <li><a href="apidocs/gpsea.config.html#module-gpsea.config">module</a>
</li>
      </ul></li>
      <li>
    gpsea.io

      <ul>
        <li><a href="apidocs/gpsea.io.html#module-gpsea.io">module</a>
</li>
      </ul></li>
      <li>
    gpsea.model

      <ul>
        <li><a href="apidocs/gpsea.model.html#module-gpsea.model">module</a>
</li>
      </ul></li>
      <li>
    gpsea.model.genome

      <ul>
        <li><a href="apidocs/gpsea.model.genome.html#module-gpsea.model.genome">module</a>
</li>
      </ul></li>
      <li>
    gpsea.preprocessing

      <ul>
        <li><a href="apidocs/gpsea.preprocessing.html#module-gpsea.preprocessing">module</a>
</li>
      </ul></li>
      <li>
    gpsea.util

      <ul>
        <li><a href="apidocs/gpsea.util.html#module-gpsea.util">module</a>
</li>
      </ul></li>
      <li>
    gpsea.view

      <ul>
        <li><a href="apidocs/gpsea.view.html#module-gpsea.view">module</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.io.html#gpsea.io.GpseaJSONDecoder">GpseaJSONDecoder (class in gpsea.io)</a>
</li>
      <li><a href="apidocs/gpsea.io.html#gpsea.io.GpseaJSONEncoder">GpseaJSONEncoder (class in gpsea.io)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.GpseaReport">GpseaReport (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.groups_predicate">groups_predicate() (in module gpsea.analysis.predicate.genotype)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="H">H</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.NotepadTree.has_errors">has_errors() (gpsea.preprocessing.NotepadTree method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.NotepadTree.has_errors_or_warnings">has_errors_or_warnings() (gpsea.preprocessing.NotepadTree method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantInfo.has_sv_info">has_sv_info() (gpsea.model.VariantInfo method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantInfo.has_variant_coordinates">has_variant_coordinates() (gpsea.model.VariantInfo method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.NotepadTree.has_warnings">has_warnings() (gpsea.preprocessing.NotepadTree method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotype.HEMIZYGOUS">HEMIZYGOUS (gpsea.model.Genotype attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotype.HETEROZYGOUS">HETEROZYGOUS (gpsea.model.Genotype attribute)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation.hgvs_cdna">hgvs_cdna (gpsea.model.TranscriptAnnotation property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation.hgvsp">hgvsp (gpsea.model.TranscriptAnnotation property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotype.HOMOZYGOUS_ALTERNATE">HOMOZYGOUS_ALTERNATE (gpsea.model.Genotype attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotype.HOMOZYGOUS_REFERENCE">HOMOZYGOUS_REFERENCE (gpsea.model.Genotype attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter">HpoMtcFilter (class in gpsea.analysis.mtc_filter)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.HpoPredicate">HpoPredicate (class in gpsea.analysis.predicate.phenotype)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysis">HpoTermAnalysis (class in gpsea.analysis.pcats)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysisResult">HpoTermAnalysisResult (class in gpsea.analysis.pcats)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="I">I</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Disease.identifier">identifier (gpsea.model.Disease property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomeBuild.identifier">(gpsea.model.genome.GenomeBuild property)</a>
</li>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomeBuildIdentifier.identifier">(gpsea.model.genome.GenomeBuildIdentifier property)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.Measurement.identifier">(gpsea.model.Measurement property)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.Phenotype.identifier">(gpsea.model.Phenotype property)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.identifier">(gpsea.model.TranscriptCoordinates property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.ImpreciseSvFunctionalAnnotator">ImpreciseSvFunctionalAnnotator (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ImpreciseSvInfo">ImpreciseSvInfo (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.INCOMPLETE_TERMINAL_CODON_VARIANT">INCOMPLETE_TERMINAL_CODON_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinFeature.info">info (gpsea.model.ProteinFeature property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.INFRAME_DELETION">INFRAME_DELETION (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.INFRAME_INSERTION">INFRAME_INSERTION (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantClass.INS">INS (gpsea.model.VariantClass attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.INTERGENIC_VARIANT">INTERGENIC_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.INTRON_VARIANT">INTRON_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantClass.INV">INV (gpsea.model.VariantClass attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.is_coding">is_coding() (gpsea.model.TranscriptCoordinates method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Region.is_empty">is_empty() (gpsea.model.genome.Region method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Sex.is_female">is_female() (gpsea.model.Sex method)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcResult.is_filtered_out">is_filtered_out() (gpsea.analysis.mtc_filter.PhenotypeMtcResult method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.is_large_imprecise_sv">is_large_imprecise_sv() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Sex.is_male">is_male() (gpsea.model.Sex method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Strand.is_negative">is_negative() (gpsea.model.genome.Strand method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Phenotype.is_observed">is_observed (gpsea.model.Phenotype property)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PreprocessingValidationResult.is_ok">is_ok() (gpsea.preprocessing.PreprocessingValidationResult method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcResult.is_passed">is_passed() (gpsea.analysis.mtc_filter.PhenotypeMtcResult method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Strand.is_positive">is_positive() (gpsea.model.genome.Strand method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation.is_preferred">is_preferred (gpsea.model.TranscriptAnnotation property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.is_preferred">(gpsea.model.TranscriptCoordinates property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Disease.is_present">is_present (gpsea.model.Disease property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.Phenotype.is_present">(gpsea.model.Phenotype property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Sex.is_provided">is_provided() (gpsea.model.Sex method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.is_structural">is_structural() (gpsea.model.VariantCoordinates method)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantInfo.is_structural">(gpsea.model.VariantInfo method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.is_structural_deletion">is_structural_deletion() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.is_structural_variant">is_structural_variant() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Sex.is_unknown">is_unknown() (gpsea.model.Sex method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad.issues">issues (gpsea.preprocessing.Notepad property)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.NotepadTree.iterate_nodes">iterate_nodes() (gpsea.preprocessing.NotepadTree method)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="L">L</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.label">label (gpsea.model.ProteinMetadata property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.SampleLabels.label">(gpsea.model.SampleLabels property)</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad.label">(gpsea.preprocessing.Notepad property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.SampleLabels.label_summary">label_summary() (gpsea.model.SampleLabels method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.labels">labels (gpsea.model.Patient property)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Level">Level (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.DataSanityIssue.level">level (gpsea.preprocessing.DataSanityIssue property)</a>

      <ul>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.NotepadTree.level">(gpsea.preprocessing.NotepadTree property)</a>
</li>
      </ul></li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.list_all_diseases">list_all_diseases() (gpsea.model.Cohort method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.list_all_proteins">list_all_proteins() (gpsea.model.Cohort method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.list_all_variants">list_all_variants() (gpsea.model.Cohort method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.list_present_phenotypes">list_present_phenotypes() (gpsea.model.Cohort method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.load_phenopacket_files">load_phenopacket_files() (in module gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.load_phenopacket_folder">load_phenopacket_folder() (in module gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.load_phenopackets">load_phenopackets() (in module gpsea.preprocessing)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="M">M</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomeBuildIdentifier.major_assembly">major_assembly (gpsea.model.genome.GenomeBuildIdentifier property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Sex.MALE">MALE (gpsea.model.Sex attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.stats.html#gpsea.analysis.pscore.stats.MannWhitneyStatistic">MannWhitneyStatistic (class in gpsea.analysis.pscore.stats)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.marker_counts">marker_counts (gpsea.view.ProteinVisualizable property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.MATURE_MIRNA_VARIANT">MATURE_MIRNA_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Measurement">Measurement (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.measurement_by_id">measurement_by_id() (gpsea.model.Patient method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.MeasurementPhenotypeScorer">MeasurementPhenotypeScorer (class in gpsea.analysis.pscore)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.measurements">measurements (gpsea.model.Patient property)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.DataSanityIssue.message">message (gpsea.preprocessing.DataSanityIssue property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.SampleLabels.meta_label">meta_label (gpsea.model.SampleLabels property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.MISSENSE_VARIANT">MISSENSE_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantClass.MNV">MNV (gpsea.model.VariantClass attribute)</a>
</li>
      <li>
    module

      <ul>
        <li><a href="apidocs/gpsea.html#module-gpsea">gpsea</a>
</li>
        <li><a href="apidocs/gpsea.analysis.html#module-gpsea.analysis">gpsea.analysis</a>
</li>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#module-gpsea.analysis.mtc_filter">gpsea.analysis.mtc_filter</a>
</li>
        <li><a href="apidocs/gpsea.analysis.pcats.html#module-gpsea.analysis.pcats">gpsea.analysis.pcats</a>
</li>
        <li><a href="apidocs/gpsea.analysis.pcats.stats.html#module-gpsea.analysis.pcats.stats">gpsea.analysis.pcats.stats</a>
</li>
        <li><a href="apidocs/gpsea.analysis.predicate.html#module-gpsea.analysis.predicate">gpsea.analysis.predicate</a>
</li>
        <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#module-gpsea.analysis.predicate.genotype">gpsea.analysis.predicate.genotype</a>
</li>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#module-gpsea.analysis.predicate.phenotype">gpsea.analysis.predicate.phenotype</a>
</li>
        <li><a href="apidocs/gpsea.analysis.pscore.html#module-gpsea.analysis.pscore">gpsea.analysis.pscore</a>
</li>
        <li><a href="apidocs/gpsea.analysis.pscore.stats.html#module-gpsea.analysis.pscore.stats">gpsea.analysis.pscore.stats</a>
</li>
        <li><a href="apidocs/gpsea.config.html#module-gpsea.config">gpsea.config</a>
</li>
        <li><a href="apidocs/gpsea.io.html#module-gpsea.io">gpsea.io</a>
</li>
        <li><a href="apidocs/gpsea.model.html#module-gpsea.model">gpsea.model</a>
</li>
        <li><a href="apidocs/gpsea.model.genome.html#module-gpsea.model.genome">gpsea.model.genome</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#module-gpsea.preprocessing">gpsea.preprocessing</a>
</li>
        <li><a href="apidocs/gpsea.util.html#module-gpsea.util">gpsea.util</a>
</li>
        <li><a href="apidocs/gpsea.view.html#module-gpsea.view">gpsea.view</a>
</li>
      </ul></li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.monoallelic_predicate">monoallelic_predicate() (in module gpsea.analysis.predicate.genotype)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureType.MOTIF">MOTIF (gpsea.model.FeatureType attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.motifs">motifs() (gpsea.model.ProteinMetadata method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_filter_name">mtc_filter_name (gpsea.analysis.pcats.HpoTermAnalysisResult property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_filter_results">mtc_filter_results (gpsea.analysis.pcats.HpoTermAnalysisResult property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcResult.mtc_issue">mtc_issue (gpsea.analysis.mtc_filter.PhenotypeMtcResult property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_name">mtc_name (gpsea.analysis.pcats.HpoTermAnalysisResult property)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.MtcStatsViewer">MtcStatsViewer (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysis">MultiPhenotypeAnalysis (class in gpsea.analysis.pcats)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysisResult">MultiPhenotypeAnalysisResult (class in gpsea.analysis.pcats)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="N">N</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate.n_categorizations">n_categorizations() (gpsea.analysis.predicate.PolyPredicate method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.n_usable">n_usable (gpsea.analysis.pcats.MultiPhenotypeAnalysisResult property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategory.name">name (gpsea.analysis.predicate.PatientCategory property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.Disease.name">(gpsea.model.Disease property)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureInfo.name">(gpsea.model.FeatureInfo property)</a>
</li>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Contig.name">(gpsea.model.genome.Contig property)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.Measurement.name">(gpsea.model.Measurement property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Strand.NEGATIVE">NEGATIVE (gpsea.model.genome.Strand attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.NMD_TRANSCRIPT_VARIANT">NMD_TRANSCRIPT_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategories.NO">NO (gpsea.analysis.predicate.PatientCategories attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotype.NO_CALL">NO_CALL (gpsea.model.Genotype attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.NO_GENOTYPE_HAS_MORE_THAN_ONE_HPO">NO_GENOTYPE_HAS_MORE_THAN_ONE_HPO (gpsea.analysis.mtc_filter.HpoMtcFilter attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.NON_CODING_TRANSCRIPT_EXON_VARIANT">NON_CODING_TRANSCRIPT_EXON_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.NON_CODING_TRANSCRIPT_VARIANT">NON_CODING_TRANSCRIPT_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.NON_SPECIFIED_TERM">NON_SPECIFIED_TERM (gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter attribute)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VepFunctionalAnnotator.NONCODING_EFFECTS">NONCODING_EFFECTS (gpsea.preprocessing.VepFunctionalAnnotator attribute)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad">Notepad (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.NotepadTree">NotepadTree (class in gpsea.preprocessing)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="O">O</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.io.html#gpsea.io.GpseaJSONDecoder.object_hook">object_hook() (gpsea.io.GpseaJSONDecoder static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Phenotype.observed">observed (gpsea.model.Phenotype property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcFilter.OK">OK (gpsea.analysis.mtc_filter.PhenotypeMtcFilter attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcResult.ok">ok() (gpsea.analysis.mtc_filter.PhenotypeMtcResult static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.one_genotype_has_zero_hpo_observations">one_genotype_has_zero_hpo_observations() (gpsea.analysis.mtc_filter.HpoMtcFilter static method)</a>
</li>
      <li><a href="apidocs/gpsea.util.html#gpsea.util.open_text_io_handle_for_reading">open_text_io_handle_for_reading() (in module gpsea.util)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.util.html#gpsea.util.open_text_io_handle_for_writing">open_text_io_handle_for_writing() (in module gpsea.util)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Strand.opposite">opposite() (gpsea.model.genome.Strand method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation.overlapping_exons">overlapping_exons (gpsea.model.TranscriptAnnotation property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureInfo.overlaps_with">overlaps_with() (gpsea.model.FeatureInfo method)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomicRegion.overlaps_with">(gpsea.model.genome.GenomicRegion method)</a>
</li>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Region.overlaps_with">(gpsea.model.genome.Region method)</a>
</li>
      </ul></li>
  </ul></td>
</tr></table>

<h2 id="P">P</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VVMultiCoordinateService.parse_multiple">parse_multiple() (gpsea.preprocessing.VVMultiCoordinateService method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VVMultiCoordinateService.parse_response">parse_response() (gpsea.preprocessing.VVMultiCoordinateService method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.UniprotProteinMetadataService.parse_uniprot_json">parse_uniprot_json() (gpsea.preprocessing.UniprotProteinMetadataService static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomeBuildIdentifier.patch">patch (gpsea.model.genome.GenomeBuildIdentifier property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient">Patient (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.patient_id">patient_id (gpsea.model.Patient property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategories">PatientCategories (class in gpsea.analysis.predicate)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategory">PatientCategory (class in gpsea.analysis.predicate)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PatientCreator">PatientCreator (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PhenopacketPatientCreator">PhenopacketPatientCreator (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PhenopacketVariantCoordinateFinder">PhenopacketVariantCoordinateFinder (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Phenotype">Phenotype (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.phenotype">phenotype (gpsea.analysis.predicate.phenotype.DiseasePresencePredicate property)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.HpoPredicate.phenotype">(gpsea.analysis.predicate.phenotype.HpoPredicate property)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.PhenotypeCategorization.phenotype">(gpsea.analysis.predicate.phenotype.PhenotypeCategorization property)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.phenotype">(gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.PhenotypeCategorization">PhenotypeCategorization (class in gpsea.analysis.predicate.phenotype)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PhenotypeCreator">PhenotypeCreator (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcFilter">PhenotypeMtcFilter (class in gpsea.analysis.mtc_filter)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcIssue">PhenotypeMtcIssue (class in gpsea.analysis.mtc_filter)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcResult">PhenotypeMtcResult (class in gpsea.analysis.mtc_filter)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate">PhenotypePolyPredicate (class in gpsea.analysis.predicate.phenotype)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.phenotypes">phenotypes (gpsea.model.Patient property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysis">PhenotypeScoreAnalysis (class in gpsea.analysis.pscore)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysisResult">PhenotypeScoreAnalysisResult (class in gpsea.analysis.pscore)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScorer">PhenotypeScorer (class in gpsea.analysis.pscore)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.stats.html#gpsea.analysis.pscore.stats.PhenotypeScoreStatistic">PhenotypeScoreStatistic (class in gpsea.analysis.pscore.stats)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.plot_boxplots">plot_boxplots() (gpsea.analysis.pscore.PhenotypeScoreAnalysisResult method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PreprocessingValidationResult.policy">policy (gpsea.preprocessing.PreprocessingValidationResult property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate">PolyPredicate (class in gpsea.analysis.predicate)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Strand.POSITIVE">POSITIVE (gpsea.model.genome.Strand attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.possible_results">possible_results() (gpsea.analysis.mtc_filter.HpoMtcFilter method)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcFilter.possible_results">(gpsea.analysis.mtc_filter.PhenotypeMtcFilter method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.possible_results">(gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.possible_results">(gpsea.analysis.mtc_filter.UseAllTermsMtcFilter method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.prepare_hpo_terms_of_interest">prepare_hpo_terms_of_interest() (in module gpsea.analysis.predicate.phenotype)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Auditor.prepare_notepad">prepare_notepad() (gpsea.preprocessing.Auditor static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.prepare_predicates_for_terms_of_interest">prepare_predicates_for_terms_of_interest() (in module gpsea.analysis.predicate.phenotype)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PreprocessingValidationResult">PreprocessingValidationResult (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.present_diseases">present_diseases() (gpsea.model.Patient method)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.present_phenotype_categorization">present_phenotype_categorization (gpsea.analysis.predicate.phenotype.DiseasePresencePredicate property)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.HpoPredicate.present_phenotype_categorization">(gpsea.analysis.predicate.phenotype.HpoPredicate property)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.present_phenotype_categorization">(gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.present_phenotype_category">present_phenotype_category (gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.present_phenotypes">present_phenotypes() (gpsea.model.Patient method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Auditor.process">process() (gpsea.preprocessing.Auditor method)</a>

      <ul>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.CohortCreator.process">(gpsea.preprocessing.CohortCreator method)</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PhenopacketPatientCreator.process">(gpsea.preprocessing.PhenopacketPatientCreator method)</a>
</li>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PhenotypeCreator.process">(gpsea.preprocessing.PhenotypeCreator method)</a>
</li>
        <li><a href="apidocs/gpsea.view.html#gpsea.view.CohortVariantViewer.process">(gpsea.view.CohortVariantViewer method)</a>
</li>
        <li><a href="apidocs/gpsea.view.html#gpsea.view.CohortViewable.process">(gpsea.view.CohortViewable method)</a>
</li>
        <li><a href="apidocs/gpsea.view.html#gpsea.view.DiseaseViewable.process">(gpsea.view.DiseaseViewable method)</a>
</li>
        <li><a href="apidocs/gpsea.view.html#gpsea.view.MtcStatsViewer.process">(gpsea.view.MtcStatsViewer method)</a>
</li>
        <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVariantViewer.process">(gpsea.view.ProteinVariantViewer method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VepFunctionalAnnotator.process_response">process_response() (gpsea.preprocessing.VepFunctionalAnnotator method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.ProtCachingMetadataService">ProtCachingMetadataService (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.PROTEIN_ALTERING_VARIANT">PROTEIN_ALTERING_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation.protein_effect_location">protein_effect_location (gpsea.model.TranscriptAnnotation property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.ProteinPredicates.protein_feature">protein_feature() (gpsea.analysis.predicate.genotype.ProteinPredicates method)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.protein_feature_ends">protein_feature_ends (gpsea.view.ProteinVisualizable property)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.protein_feature_names">protein_feature_names (gpsea.view.ProteinVisualizable property)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.protein_feature_starts">protein_feature_starts (gpsea.view.ProteinVisualizable property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.ProteinPredicates.protein_feature_type">protein_feature_type() (gpsea.analysis.predicate.genotype.ProteinPredicates method)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.protein_feature_types">protein_feature_types (gpsea.view.ProteinVisualizable property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.protein_features">protein_features (gpsea.model.ProteinMetadata property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.protein_id">protein_id (gpsea.model.ProteinMetadata property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation.protein_id">(gpsea.model.TranscriptAnnotation property)</a>
</li>
        <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.protein_id">(gpsea.view.ProteinVisualizable property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.protein_length">protein_length (gpsea.model.ProteinMetadata property)</a>

      <ul>
        <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.protein_length">(gpsea.view.ProteinVisualizable property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.protein_metadata">protein_metadata (gpsea.view.ProteinVisualizable property)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.ProteinAnnotationCache">ProteinAnnotationCache (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinFeature">ProteinFeature (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata">ProteinMetadata (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.ProteinMetadataService">ProteinMetadataService (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.ProteinPredicates">ProteinPredicates (class in gpsea.analysis.predicate.genotype)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVariantViewer">ProteinVariantViewer (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable">ProteinVisualizable (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizer">ProteinVisualizer (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.pval">pval (gpsea.analysis.pscore.PhenotypeScoreAnalysisResult property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.pvals">pvals (gpsea.analysis.pcats.MultiPhenotypeAnalysisResult property)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="R">R</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcIssue.reason">reason (gpsea.analysis.mtc_filter.PhenotypeMtcIssue attribute)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcResult.reason">(gpsea.analysis.mtc_filter.PhenotypeMtcResult property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.ref">ref (gpsea.model.VariantCoordinates property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.ref_length">ref_length() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Contig.refseq_name">refseq_name (gpsea.model.genome.Contig property)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Region">Region (class in gpsea.model.genome)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureInfo.region">region (gpsea.model.FeatureInfo property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureType.REGION">REGION (gpsea.model.FeatureType attribute)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates.region">region (gpsea.model.TranscriptCoordinates property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.region">(gpsea.model.VariantCoordinates property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.region">region() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.regions">regions() (gpsea.model.ProteinMetadata method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.REGULATORY_REGION_ABLATION">REGULATORY_REGION_ABLATION (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.REGULATORY_REGION_AMPLIFICATION">REGULATORY_REGION_AMPLIFICATION (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.REGULATORY_REGION_VARIANT">REGULATORY_REGION_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureType.REPEAT">REPEAT (gpsea.model.FeatureType attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata.repeats">repeats() (gpsea.model.ProteinMetadata method)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="S">S</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.SAME_COUNT_AS_THE_ONLY_CHILD">SAME_COUNT_AS_THE_ONLY_CHILD (gpsea.analysis.mtc_filter.HpoMtcFilter attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.SampleLabels">SampleLabels (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.CountingPhenotypeScorer.score">score() (gpsea.analysis.pscore.CountingPhenotypeScorer method)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.DeVriesPhenotypeScorer.score">(gpsea.analysis.pscore.DeVriesPhenotypeScorer method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.MeasurementPhenotypeScorer.score">(gpsea.analysis.pscore.MeasurementPhenotypeScorer method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScorer.score">(gpsea.analysis.pscore.PhenotypeScorer method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.SEQUENCE_VARIANT">SEQUENCE_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Sex">Sex (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.sex">sex (gpsea.model.Patient property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.sex_predicate">sex_predicate() (in module gpsea.analysis.predicate.genotype)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Genotypes.single">single() (gpsea.model.Genotypes static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.SKIPPING_GENERAL_TERM">SKIPPING_GENERAL_TERM (gpsea.analysis.mtc_filter.HpoMtcFilter attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.SKIPPING_NON_PHENOTYPE_TERM">SKIPPING_NON_PHENOTYPE_TERM (gpsea.analysis.mtc_filter.HpoMtcFilter attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.SKIPPING_SINCE_ONE_GENOTYPE_HAD_ZERO_OBSERVATIONS">SKIPPING_SINCE_ONE_GENOTYPE_HAD_ZERO_OBSERVATIONS (gpsea.analysis.mtc_filter.HpoMtcFilter attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantClass.SNV">SNV (gpsea.model.VariantClass attribute)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.DataSanityIssue.solution">solution (gpsea.preprocessing.DataSanityIssue property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter.some_cell_has_greater_than_one_count">some_cell_has_greater_than_one_count() (gpsea.analysis.mtc_filter.HpoMtcFilter static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter">SpecifiedTermsMtcFilter (class in gpsea.analysis.mtc_filter)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.SPLICE_ACCEPTOR_VARIANT">SPLICE_ACCEPTOR_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.SPLICE_DONOR_5TH_BASE_VARIANT">SPLICE_DONOR_5TH_BASE_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.SPLICE_DONOR_REGION_VARIANT">SPLICE_DONOR_REGION_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.SPLICE_DONOR_VARIANT">SPLICE_DONOR_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.SPLICE_POLYPYRIMIDINE_TRACT_VARIANT">SPLICE_POLYPYRIMIDINE_TRACT_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.SPLICE_REGION_VARIANT">SPLICE_REGION_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FeatureInfo.start">start (gpsea.model.FeatureInfo property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Region.start">(gpsea.model.genome.Region property)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.start">(gpsea.model.VariantCoordinates property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.START_LOST">START_LOST (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomicRegion.start_on_strand">start_on_strand() (gpsea.model.genome.GenomicRegion method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.START_RETAINED_VARIANT">START_RETAINED_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.STOP_GAINED">STOP_GAINED (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.STOP_LOST">STOP_LOST (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.STOP_RETAINED_VARIANT">STOP_RETAINED_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.ProteinAnnotationCache.store_annotations">store_annotations() (gpsea.preprocessing.ProteinAnnotationCache method)</a>

      <ul>
        <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VariantAnnotationCache.store_annotations">(gpsea.preprocessing.VariantAnnotationCache method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Strand">Strand (class in gpsea.model.genome)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomicRegion.strand">strand (gpsea.model.genome.GenomicRegion property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Stranded.strand">(gpsea.model.genome.Stranded property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Stranded">Stranded (class in gpsea.model.genome)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.structural_so_id_to_display">structural_so_id_to_display() (gpsea.model.VariantEffect static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ImpreciseSvInfo.structural_type">structural_type (gpsea.model.ImpreciseSvInfo property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.structural_type">structural_type() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.PreprocessingValidationResult.summarize">summarize() (gpsea.preprocessing.PreprocessingValidationResult method)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.summarize_hpo_analysis">summarize_hpo_analysis() (in module gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.stats.html#gpsea.analysis.pcats.stats.CountStatistic.supports_shape">supports_shape (gpsea.analysis.pcats.stats.CountStatistic property)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.pcats.stats.html#gpsea.analysis.pcats.stats.FisherExactTest.supports_shape">(gpsea.analysis.pcats.stats.FisherExactTest property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantInfo.sv_info">sv_info (gpsea.model.VariantInfo property)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Strand.symbol">symbol (gpsea.model.genome.Strand property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.SYNONYMOUS_VARIANT">SYNONYMOUS_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="T">T</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.terms_to_test">terms_to_test (gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate.test">test() (gpsea.analysis.predicate.genotype.VariantPredicate method)</a>

      <ul>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.test">(gpsea.analysis.predicate.phenotype.DiseasePresencePredicate method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.HpoPredicate.test">(gpsea.analysis.predicate.phenotype.HpoPredicate method)</a>
</li>
        <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PolyPredicate.test">(gpsea.analysis.predicate.PolyPredicate method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Measurement.test_result">test_result (gpsea.model.Measurement property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.TF_BINDING_SITE_VARIANT">TF_BINDING_SITE_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.TFBS_ABLATION">TFBS_ABLATION (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.TFBS_AMPLIFICATION">TFBS_AMPLIFICATION (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.THREE_PRIME_UTR_VARIANT">THREE_PRIME_UTR_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.to_display">to_display() (gpsea.model.VariantEffect method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Transposable.to_negative_strand">to_negative_strand() (gpsea.model.genome.Transposable method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Transposable.to_opposite_strand">to_opposite_strand() (gpsea.model.genome.Transposable method)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Transposable.to_positive_strand">to_positive_strand() (gpsea.model.genome.Transposable method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ProteinFeature.to_string">to_string() (gpsea.model.ProteinFeature method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.total_patient_count">total_patient_count (gpsea.model.Cohort property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.total_tests">total_tests (gpsea.analysis.pcats.MultiPhenotypeAnalysisResult property)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.transcript">transcript() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.TRANSCRIPT_ABLATION">TRANSCRIPT_ABLATION (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.TRANSCRIPT_AMPLIFICATION">TRANSCRIPT_AMPLIFICATION (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.transcript_coordinates">transcript_coordinates (gpsea.view.ProteinVisualizable property)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation.transcript_id">transcript_id (gpsea.model.TranscriptAnnotation property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptInfoAware.transcript_id">(gpsea.model.TranscriptInfoAware property)</a>
</li>
        <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.transcript_id">(gpsea.view.ProteinVisualizable property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation">TranscriptAnnotation (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates">TranscriptCoordinates (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.TranscriptCoordinateService">TranscriptCoordinateService (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptInfoAware">TranscriptInfoAware (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantClass.TRANSLOCATION">TRANSLOCATION (gpsea.model.VariantClass attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Transposable">Transposable (class in gpsea.model.genome)</a>
</li>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.transpose_coordinate">transpose_coordinate() (in module gpsea.model.genome)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.true">true() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.pscore.stats.html#gpsea.analysis.pscore.stats.TTestStatistic">TTestStatistic (class in gpsea.analysis.pscore.stats)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.FunctionalAnnotationAware.tx_annotations">tx_annotations (gpsea.model.FunctionalAnnotationAware property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.Variant.tx_annotations">(gpsea.model.Variant property)</a>
</li>
      </ul></li>
  </ul></td>
</tr></table>

<h2 id="U">U</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Contig.ucsc_name">ucsc_name (gpsea.model.genome.Contig property)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.UniprotProteinMetadataService">UniprotProteinMetadataService (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Measurement.unit">unit (gpsea.model.Measurement property)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Sex.UNKNOWN_SEX">UNKNOWN_SEX (gpsea.model.Sex attribute)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect.UPSTREAM_GENE_VARIANT">UPSTREAM_GENE_VARIANT (gpsea.model.VariantEffect attribute)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.UseAllTermsMtcFilter">UseAllTermsMtcFilter (class in gpsea.analysis.mtc_filter)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="V">V</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VarCachingFunctionalAnnotator">VarCachingFunctionalAnnotator (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Variant">Variant (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ImpreciseSvInfo.variant_class">variant_class (gpsea.model.ImpreciseSvInfo property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.variant_class">(gpsea.model.VariantCoordinates property)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantInfo.variant_class">(gpsea.model.VariantInfo property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.variant_class">variant_class() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantInfo.variant_coordinates">variant_coordinates (gpsea.model.VariantInfo property)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.variant_effect">variant_effect() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Cohort.variant_effect_count_by_tx">variant_effect_count_by_tx() (gpsea.model.Cohort method)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.TranscriptAnnotation.variant_effects">variant_effects (gpsea.model.TranscriptAnnotation property)</a>

      <ul>
        <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.variant_effects">(gpsea.view.ProteinVisualizable property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Variant.variant_info">variant_info (gpsea.model.Variant property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantInfoAware.variant_info">(gpsea.model.VariantInfoAware property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.ImpreciseSvInfo.variant_key">variant_key (gpsea.model.ImpreciseSvInfo property)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates.variant_key">(gpsea.model.VariantCoordinates property)</a>
</li>
        <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantInfo.variant_key">(gpsea.model.VariantInfo property)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates.variant_key">variant_key() (gpsea.analysis.predicate.genotype.VariantPredicates static method)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.variant_locations">variant_locations (gpsea.view.ProteinVisualizable property)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.ProteinVisualizable.variant_locations_counted_absolute">variant_locations_counted_absolute (gpsea.view.ProteinVisualizable property)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VariantAnnotationCache">VariantAnnotationCache (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantClass">VariantClass (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VariantCoordinateFinder">VariantCoordinateFinder (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantCoordinates">VariantCoordinates (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantEffect">VariantEffect (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.VariantFormatter">VariantFormatter (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantInfo">VariantInfo (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.VariantInfoAware">VariantInfoAware (class in gpsea.model)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate">VariantPredicate (class in gpsea.analysis.predicate.genotype)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates">VariantPredicates (class in gpsea.analysis.predicate.genotype)</a>
</li>
      <li><a href="apidocs/gpsea.model.html#gpsea.model.Patient.variants">variants (gpsea.model.Patient property)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.VariantTranscriptVisualizer">VariantTranscriptVisualizer (class in gpsea.view)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VepFunctionalAnnotator">VepFunctionalAnnotator (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.verify_term_id">verify_term_id() (gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter static method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.NotepadTree.visit">visit() (gpsea.preprocessing.NotepadTree method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VVHgvsVariantCoordinateFinder">VVHgvsVariantCoordinateFinder (class in gpsea.preprocessing)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VVMultiCoordinateService">VVMultiCoordinateService (class in gpsea.preprocessing)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="W">W</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Level.WARN">WARN (gpsea.preprocessing.Level attribute)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad.warning_count">warning_count() (gpsea.preprocessing.Notepad method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.Notepad.warnings">warnings() (gpsea.preprocessing.Notepad method)</a>
</li>
      <li><a href="apidocs/gpsea.preprocessing.html#gpsea.preprocessing.VarCachingFunctionalAnnotator.with_cache_folder">with_cache_folder() (gpsea.preprocessing.VarCachingFunctionalAnnotator static method)</a>
</li>
  </ul></td>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.GenomicRegion.with_strand">with_strand() (gpsea.model.genome.GenomicRegion method)</a>

      <ul>
        <li><a href="apidocs/gpsea.model.genome.html#gpsea.model.genome.Transposable.with_strand">(gpsea.model.genome.Transposable method)</a>
</li>
      </ul></li>
      <li><a href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScorer.wrap_scoring_function">wrap_scoring_function() (gpsea.analysis.pscore.PhenotypeScorer static method)</a>
</li>
      <li><a href="apidocs/gpsea.view.html#gpsea.view.GpseaReport.write">write() (gpsea.view.GpseaReport method)</a>
</li>
  </ul></td>
</tr></table>

<h2 id="Y">Y</h2>
<table style="width: 100%" class="indextable genindextable"><tr>
  <td style="width: 33%; vertical-align: top;"><ul>
      <li><a href="apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.PatientCategories.YES">YES (gpsea.analysis.predicate.PatientCategories attribute)</a>
</li>
  </ul></td>
</tr></table>



           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>