Search.setIndex({"alltitles": {"API reference": [[17, null]], "Alternative phenopacket sources": [[26, "alternative-phenopacket-sources"]], "Analysis": [[38, "analysis"], [38, "id4"]], "Assign individuals into genotype groups": [[35, "assign-individuals-into-genotype-groups"]], "Background": [[27, "background"]], "Building blocks": [[37, "building-blocks"]], "Change length of an allele": [[24, "change-length-of-an-allele"]], "Choose the transcript": [[26, "choose-the-transcript"]], "Cohort exploratory analysis": [[22, null]], "Compare genotype and phenotype groups": [[38, "compare-genotype-and-phenotype-groups"]], "Complex conditions": [[37, "complex-conditions"]], "Configure analysis": [[38, "configure-analysis"], [38, "id3"]], "Contents:": [[25, null], [28, null], [31, null], [36, null]], "Create a cohort from GA4GH phenopackets": [[26, "create-a-cohort-from-ga4gh-phenopackets"]], "Create cohort": [[38, "create-cohort"]], "Create cohort creator": [[26, "create-cohort-creator"]], "De Vries Score": [[29, null]], "Developmental delay": [[29, "developmental-delay"]], "Enter features manually": [[26, "enter-features-manually"]], "Example": [[32, "example"], [33, "example"], [38, "example"], [38, "id2"]], "Explore cohort": [[20, "explore-cohort"]], "Facial dysmorphic features": [[29, "facial-dysmorphic-features"]], "Feedback": [[18, "feedback"]], "Fetch data from UniProt REST API": [[26, "fetch-data-from-uniprot-rest-api"]], "Fetch transcript coordinates from Variant Validator REST API": [[26, "fetch-transcript-coordinates-from-variant-validator-rest-api"]], "Final score": [[29, "final-score"]], "Fisher exact test (FET)": [[38, "fisher-exact-test-fet"]], "GPSEA": [[18, null]], "Gallery": [[28, "gallery"]], "General HPO terms": [[23, null]], "Genotype Predicates": [[31, null]], "Genotype phenotype associations": [[38, "genotype-phenotype-associations"]], "Genotype-Phenotype Correlations in Autosomal Recessive Diseases": [[21, null]], "Get data for the transcript and protein of interest": [[26, "get-data-for-the-transcript-and-protein-of-interest"]], "Get the protein data": [[26, "get-the-protein-data"]], "Get the transcript data": [[26, "get-the-transcript-data"]], "Glossary": [[24, null]], "Groups Predicate": [[32, null]], "HMF01 - Skip terms that occur very rarely": [[27, "hmf01-skip-terms-that-occur-very-rarely"]], "HMF02 - Skip terms if no genotype group has more than one count": [[27, "hmf02-skip-terms-if-no-genotype-group-has-more-than-one-count"]], "HMF03 - Skip terms if all counts are identical to counts for a child term": [[27, "hmf03-skip-terms-if-all-counts-are-identical-to-counts-for-a-child-term"]], "HMF04 - Skip terms if genotypes have same HPO proportions": [[27, "hmf04-skip-terms-if-genotypes-have-same-hpo-proportions"]], "HMF05 - Skip term if one of the genotype groups has neither observed nor excluded observations": [[27, "hmf05-skip-term-if-one-of-the-genotype-groups-has-neither-observed-nor-excluded-observations"]], "HMF06 - Skip term if underpowered for 2x2 or 2x3 analysis": [[27, "hmf06-skip-term-if-underpowered-for-2x2-or-2x3-analysis"]], "HMF07 - Skipping terms that are not descendents of Phenotypic abnormality": [[27, "hmf07-skipping-terms-that-are-not-descendents-of-phenotypic-abnormality"]], "HMF08 - Skipping \u201cgeneral\u201d level terms": [[27, "hmf08-skipping-general-level-terms"]], "HPO MTC filter strategy": [[27, "hpo-mtc-filter-strategy"]], "HPO predicate": [[33, null]], "Implementation in GPSEA": [[27, "implementation-in-gpsea"]], "Input data": [[26, null]], "Installation": [[19, null]], "Inverting conditions": [[37, "inverting-conditions"]], "Latest release": [[19, "latest-release"]], "Length of the reference allele": [[24, "length-of-the-reference-allele"]], "Literature": [[18, "literature"]], "Load HPO": [[20, "load-hpo"]], "Load phenopackets": [[26, "load-phenopackets"]], "MTC filters: Choosing which terms to test": [[27, "mtc-filters-choosing-which-terms-to-test"]], "Mann-Whitney U Test": [[38, "mann-whitney-u-test"]], "Mode of Inheritance Predicates": [[35, null]], "Multiple-testing correction": [[27, null]], "Multiple-testing correction procedures": [[27, "multiple-testing-correction-procedures"]], "Need more?": [[28, "need-more"]], "Non-facial dysmorphism and congenital abnormalities": [[29, "non-facial-dysmorphism-and-congenital-abnormalities"]], "Parse UniProt JSON dump": [[26, "parse-uniprot-json-dump"]], "Partition by a diagnosis": [[30, null]], "Partition by the sex of the individual": [[34, null]], "Phenotype Predicates": [[36, null]], "Phenotype score": [[38, "phenotype-score"]], "Plot distribution of cohort variants on the protein": [[26, "plot-distribution-of-cohort-variants-on-the-protein"]], "Plot distribution of variants with respect to the protein sequence": [[20, "plot-distribution-of-variants-with-respect-to-the-protein-sequence"]], "Postnatal growth abnormalities": [[29, "postnatal-growth-abnormalities"]], "Predicates": [[28, null]], "Predicates for all cohort phenotypes": [[33, "predicates-for-all-cohort-phenotypes"]], "Prenatal-onset growth retardation": [[29, "prenatal-onset-growth-retardation"]], "Prepare cohort": [[20, "prepare-cohort"]], "Prepare genotype and phenotype predicates": [[20, "prepare-genotype-and-phenotype-predicates"]], "Provide the transcript coordinates manually": [[26, "provide-the-transcript-coordinates-manually"]], "Run tests": [[19, "run-tests"]], "Show cohort summary": [[20, "show-cohort-summary"]], "Showcase transcript data": [[26, "showcase-transcript-data"]], "Specify terms strategy": [[27, "specify-terms-strategy"]], "Stable release": [[19, "stable-release"]], "Standardize genotype and phenotype data": [[26, "standardize-genotype-and-phenotype-data"]], "Statistical tests": [[38, null]], "Submodules": [[0, "submodules"]], "Subpackages": [[0, "subpackages"], [1, "subpackages"], [3, "subpackages"], [5, "subpackages"], [8, "subpackages"], [12, "subpackages"]], "Summarize all variant alleles": [[20, "summarize-all-variant-alleles"]], "Supports shape": [[4, "supports-shape"]], "TBX5 frameshift vs missense": [[20, "id5"]], "TBX5 frameshift vs rest": [[38, "id5"]], "Test all terms": [[27, "test-all-terms"]], "The analysis": [[20, "the-analysis"]], "True path rule": [[24, "true-path-rule"]], "Tutorial": [[20, null], [21, "tutorial"]], "Use a subset of variants for choosing the genotype group": [[35, "use-a-subset-of-variants-for-choosing-the-genotype-group"]], "User guide": [[25, null]], "Using the De Vries Scorer in code": [[29, "using-the-de-vries-scorer-in-code"]], "Variant Predicates": [[37, null]], "gpsea package": [[0, null]], "gpsea.analysis package": [[1, null]], "gpsea.analysis.mtc_filter package": [[2, null]], "gpsea.analysis.pcats package": [[3, null]], "gpsea.analysis.pcats.stats package": [[4, null]], "gpsea.analysis.predicate package": [[5, null]], "gpsea.analysis.predicate.genotype package": [[6, null]], "gpsea.analysis.predicate.phenotype package": [[7, null]], "gpsea.analysis.pscore package": [[8, null]], "gpsea.analysis.pscore.stats package": [[9, null]], "gpsea.config module": [[10, null]], "gpsea.io module": [[11, null]], "gpsea.model package": [[12, null]], "gpsea.model.genome package": [[13, null]], "gpsea.preprocessing package": [[14, null]], "gpsea.util module": [[15, null]], "gpsea.view package": [[16, null]], "missing_implies_phenotype_excluded": [[33, "missing-implies-phenotype-excluded"]]}, "docnames": ["apidocs/gpsea", "apidocs/gpsea.analysis", "apidocs/gpsea.analysis.mtc_filter", "apidocs/gpsea.analysis.pcats", "apidocs/gpsea.analysis.pcats.stats", "apidocs/gpsea.analysis.predicate", "apidocs/gpsea.analysis.predicate.genotype", "apidocs/gpsea.analysis.predicate.phenotype", "apidocs/gpsea.analysis.pscore", "apidocs/gpsea.analysis.pscore.stats", "apidocs/gpsea.config", "apidocs/gpsea.io", "apidocs/gpsea.model", "apidocs/gpsea.model.genome", "apidocs/gpsea.preprocessing", "apidocs/gpsea.util", "apidocs/gpsea.view", "apidocs/modules", "index", "installation", "tutorial", "user-guide/autosomal_recessive", "user-guide/exploratory", "user-guide/general_hpo_terms", "user-guide/glossary", "user-guide/index", "user-guide/input-data", "user-guide/mtc", "user-guide/predicates", "user-guide/predicates/devries", "user-guide/predicates/diagnosis_predicate", "user-guide/predicates/genotype_predicates", "user-guide/predicates/groups_predicate", "user-guide/predicates/hpo_predicate", "user-guide/predicates/male_female_predicate", "user-guide/predicates/mode_of_inheritance_predicate", "user-guide/predicates/phenotype_predicates", "user-guide/predicates/variant_predicates", "user-guide/stats"], "envversion": {"sphinx": 62, "sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1}, "filenames": ["apidocs/gpsea.rst", "apidocs/gpsea.analysis.rst", "apidocs/gpsea.analysis.mtc_filter.rst", "apidocs/gpsea.analysis.pcats.rst", "apidocs/gpsea.analysis.pcats.stats.rst", "apidocs/gpsea.analysis.predicate.rst", "apidocs/gpsea.analysis.predicate.genotype.rst", "apidocs/gpsea.analysis.predicate.phenotype.rst", "apidocs/gpsea.analysis.pscore.rst", "apidocs/gpsea.analysis.pscore.stats.rst", "apidocs/gpsea.config.rst", "apidocs/gpsea.io.rst", "apidocs/gpsea.model.rst", "apidocs/gpsea.model.genome.rst", "apidocs/gpsea.preprocessing.rst", "apidocs/gpsea.util.rst", "apidocs/gpsea.view.rst", "apidocs/modules.rst", "index.rst", "installation.rst", "tutorial.rst", "user-guide/autosomal_recessive.rst", "user-guide/exploratory.rst", "user-guide/general_hpo_terms.rst", "user-guide/glossary.rst", "user-guide/index.rst", "user-guide/input-data.rst", "user-guide/mtc.rst", "user-guide/predicates.rst", "user-guide/predicates/devries.rst", "user-guide/predicates/diagnosis_predicate.rst", "user-guide/predicates/genotype_predicates.rst", "user-guide/predicates/groups_predicate.rst", "user-guide/predicates/hpo_predicate.rst", "user-guide/predicates/male_female_predicate.rst", "user-guide/predicates/mode_of_inheritance_predicate.rst", "user-guide/predicates/phenotype_predicates.rst", "user-guide/predicates/variant_predicates.rst", "user-guide/stats.rst"], "indexentries": {"add_error() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.add_error", false]], "add_issue() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.add_issue", false]], "add_subsection() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.add_subsection", false]], "add_subsection() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.add_subsection", false]], "add_warning() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.add_warning", false]], "all() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.all", false]], "all_counts (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.all_counts", false]], "all_diseases() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.all_diseases", false]], "all_patients (gpsea.model.cohort property)": [[12, "gpsea.model.Cohort.all_patients", false]], "all_phenotypes() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.all_phenotypes", false]], "all_transcript_ids (gpsea.model.cohort property)": [[12, "gpsea.model.Cohort.all_transcript_ids", false]], "all_variants() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.all_variants", false]], "allelecounter (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.AlleleCounter", false]], "alt (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.alt", false]], "annotate() (gpsea.preprocessing.defaultimprecisesvfunctionalannotator method)": [[14, "gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.functionalannotator method)": [[14, "gpsea.preprocessing.FunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.imprecisesvfunctionalannotator method)": [[14, "gpsea.preprocessing.ImpreciseSvFunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.protcachingmetadataservice method)": [[14, "gpsea.preprocessing.ProtCachingMetadataService.annotate", false]], "annotate() (gpsea.preprocessing.proteinmetadataservice method)": [[14, "gpsea.preprocessing.ProteinMetadataService.annotate", false]], "annotate() (gpsea.preprocessing.uniprotproteinmetadataservice method)": [[14, "gpsea.preprocessing.UniprotProteinMetadataService.annotate", false]], "annotate() (gpsea.preprocessing.varcachingfunctionalannotator method)": [[14, "gpsea.preprocessing.VarCachingFunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.vepfunctionalannotator method)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator.annotate", false]], "any() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.any", false]], "apply_predicates_on_patients() (in module gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.apply_predicates_on_patients", false]], "auditor (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.Auditor", false]], "autosomal_dominant() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate static method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.autosomal_dominant", false]], "autosomal_dominant() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.autosomal_dominant", false]], "autosomal_recessive() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate static method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.autosomal_recessive", false]], "autosomal_recessive() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.autosomal_recessive", false]], "biallelic_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.biallelic_predicate", false]], "cache_env (in module gpsea.config)": [[10, "gpsea.config.CACHE_ENV", false]], "cat_id (gpsea.analysis.predicate.patientcategory property)": [[5, "gpsea.analysis.predicate.PatientCategory.cat_id", false]], "categorization (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.Categorization", false]], "category (gpsea.analysis.predicate.categorization property)": [[5, "gpsea.analysis.predicate.Categorization.category", false]], "cds_end (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.cds_end", false]], "cds_start (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.cds_start", false]], "change_length (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.change_length", false]], "change_length() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.change_length", false]], "children (gpsea.preprocessing.notepadtree property)": [[14, "gpsea.preprocessing.NotepadTree.children", false]], "chrom (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.chrom", false]], "code (gpsea.analysis.mtc_filter.phenotypemtcissue attribute)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcIssue.code", false]], "code (gpsea.model.genotype property)": [[12, "gpsea.model.Genotype.code", false]], "coding_sequence_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.CODING_SEQUENCE_VARIANT", false]], "cohort (class in gpsea.model)": [[12, "gpsea.model.Cohort", false]], "cohortcreator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.CohortCreator", false]], "cohortvariantviewer (class in gpsea.view)": [[16, "gpsea.view.CohortVariantViewer", false]], "cohortviewable (class in gpsea.view)": [[16, "gpsea.view.CohortViewable", false]], "coiled_coil (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.COILED_COIL", false]], "compare_genotype_vs_phenotype_score() (gpsea.analysis.pscore.phenotypescoreanalysis method)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysis.compare_genotype_vs_phenotype_score", false]], "compare_genotype_vs_phenotypes() (gpsea.analysis.pcats.multiphenotypeanalysis method)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysis.compare_genotype_vs_phenotypes", false]], "compositional_bias (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.COMPOSITIONAL_BIAS", false]], "compute_pval() (gpsea.analysis.pcats.stats.countstatistic method)": [[4, "gpsea.analysis.pcats.stats.CountStatistic.compute_pval", false]], "compute_pval() (gpsea.analysis.pcats.stats.fisherexacttest method)": [[4, "gpsea.analysis.pcats.stats.FisherExactTest.compute_pval", false]], "compute_pval() (gpsea.analysis.pscore.stats.mannwhitneystatistic method)": [[9, "gpsea.analysis.pscore.stats.MannWhitneyStatistic.compute_pval", false]], "compute_pval() (gpsea.analysis.pscore.stats.phenotypescorestatistic method)": [[9, "gpsea.analysis.pscore.stats.PhenotypeScoreStatistic.compute_pval", false]], "compute_pval() (gpsea.analysis.pscore.stats.tteststatistic method)": [[9, "gpsea.analysis.pscore.stats.TTestStatistic.compute_pval", false]], "configure_caching_cohort_creator() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_caching_cohort_creator", false]], "configure_caching_patient_creator() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_caching_patient_creator", false]], "configure_cohort_creator() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_cohort_creator", false]], "configure_default_protein_metadata_service() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_default_protein_metadata_service", false]], "configure_patient_creator() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_patient_creator", false]], "configure_protein_metadata_service() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_protein_metadata_service", false]], "contains() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.contains", false]], "contains() (gpsea.model.genome.region method)": [[13, "gpsea.model.genome.Region.contains", false]], "contains_pos() (gpsea.model.genome.region method)": [[13, "gpsea.model.genome.Region.contains_pos", false]], "contig (class in gpsea.model.genome)": [[13, "gpsea.model.genome.Contig", false]], "contig (gpsea.model.genome.genomicregion property)": [[13, "gpsea.model.genome.GenomicRegion.contig", false]], "contig_by_name() (gpsea.model.genome.genomebuild method)": [[13, "gpsea.model.genome.GenomeBuild.contig_by_name", false]], "contigs (gpsea.model.genome.genomebuild property)": [[13, "gpsea.model.genome.GenomeBuild.contigs", false]], "corrected_pvals (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.corrected_pvals", false]], "count() (gpsea.analysis.predicate.genotype.allelecounter method)": [[6, "gpsea.analysis.predicate.genotype.AlleleCounter.count", false]], "countingphenotypescorer (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer", false]], "countstatistic (class in gpsea.analysis.pcats.stats)": [[4, "gpsea.analysis.pcats.stats.CountStatistic", false]], "create() (gpsea.model.proteinfeature static method)": [[12, "gpsea.model.ProteinFeature.create", false]], "create_variant_from_scratch() (gpsea.model.variant static method)": [[12, "gpsea.model.Variant.create_variant_from_scratch", false]], "curie (gpsea.model.varianteffect property)": [[12, "gpsea.model.VariantEffect.curie", false]], "datasanityissue (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.DataSanityIssue", false]], "default() (gpsea.io.gpseajsonencoder method)": [[11, "gpsea.io.GpseaJSONEncoder.default", false]], "default_cache_path (in module gpsea.config)": [[10, "gpsea.config.DEFAULT_CACHE_PATH", false]], "default_filter() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.default_filter", false]], "defaultimprecisesvfunctionalannotator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator", false]], "del (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.DEL", false]], "description (gpsea.analysis.predicate.patientcategory property)": [[5, "gpsea.analysis.predicate.PatientCategory.description", false]], "devriesphenotypescorer (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.DeVriesPhenotypeScorer", false]], "diagnosis_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.diagnosis_predicate", false]], "disease (class in gpsea.model)": [[12, "gpsea.model.Disease", false]], "diseaseanalysis (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.DiseaseAnalysis", false]], "diseasepresencepredicate (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate", false]], "diseases (gpsea.model.patient property)": [[12, "gpsea.model.Patient.diseases", false]], "diseaseviewable (class in gpsea.view)": [[16, "gpsea.view.DiseaseViewable", false]], "display_question() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.display_question", false]], "distance_to() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.distance_to", false]], "distance_to() (gpsea.model.genome.region method)": [[13, "gpsea.model.genome.Region.distance_to", false]], "domain (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.DOMAIN", false]], "domains() (gpsea.model.proteinmetadata method)": [[12, "gpsea.model.ProteinMetadata.domains", false]], "downstream_gene_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.DOWNSTREAM_GENE_VARIANT", false]], "draw_fig() (gpsea.view.proteinvisualizer method)": [[16, "gpsea.view.ProteinVisualizer.draw_fig", false]], "draw_protein_diagram() (gpsea.view.proteinvisualizer method)": [[16, "gpsea.view.ProteinVisualizer.draw_protein_diagram", false]], "draw_variants() (gpsea.view.varianttranscriptvisualizer method)": [[16, "gpsea.view.VariantTranscriptVisualizer.draw_variants", false]], "dup (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.DUP", false]], "empty() (gpsea.model.genotypes static method)": [[12, "gpsea.model.Genotypes.empty", false]], "end (gpsea.model.featureinfo property)": [[12, "gpsea.model.FeatureInfo.end", false]], "end (gpsea.model.genome.region property)": [[13, "gpsea.model.genome.Region.end", false]], "end (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.end", false]], "end_on_strand() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.end_on_strand", false]], "error (gpsea.preprocessing.level attribute)": [[14, "gpsea.preprocessing.Level.ERROR", false]], "error_count() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.error_count", false]], "errors() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.errors", false]], "excluded_diseases() (gpsea.model.patient method)": [[12, "gpsea.model.Patient.excluded_diseases", false]], "excluded_phenotypes() (gpsea.model.patient method)": [[12, "gpsea.model.Patient.excluded_phenotypes", false]], "exon() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.exon", false]], "exons (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.exons", false]], "fail() (gpsea.analysis.mtc_filter.phenotypemtcresult static method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.fail", false]], "feature_elongation (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.FEATURE_ELONGATION", false]], "feature_truncation (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.FEATURE_TRUNCATION", false]], "feature_type (gpsea.model.proteinfeature property)": [[12, "gpsea.model.ProteinFeature.feature_type", false]], "featureinfo (class in gpsea.model)": [[12, "gpsea.model.FeatureInfo", false]], "featuretype (class in gpsea.model)": [[12, "gpsea.model.FeatureType", false]], "female (gpsea.model.sex attribute)": [[12, "gpsea.model.Sex.FEMALE", false]], "fetch() (gpsea.preprocessing.transcriptcoordinateservice method)": [[14, "gpsea.preprocessing.TranscriptCoordinateService.fetch", false]], "fetch() (gpsea.preprocessing.vvmulticoordinateservice method)": [[14, "gpsea.preprocessing.VVMultiCoordinateService.fetch", false]], "fetch_for_gene() (gpsea.preprocessing.genecoordinateservice method)": [[14, "gpsea.preprocessing.GeneCoordinateService.fetch_for_gene", false]], "fetch_for_gene() (gpsea.preprocessing.vvmulticoordinateservice method)": [[14, "gpsea.preprocessing.VVMultiCoordinateService.fetch_for_gene", false]], "fetch_response() (gpsea.preprocessing.vepfunctionalannotator method)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator.fetch_response", false]], "filter() (gpsea.analysis.mtc_filter.hpomtcfilter method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.filter", false]], "filter() (gpsea.analysis.mtc_filter.phenotypemtcfilter method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter.filter", false]], "filter() (gpsea.analysis.mtc_filter.specifiedtermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.filter", false]], "filter() (gpsea.analysis.mtc_filter.usealltermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.filter", false]], "filter_method_name() (gpsea.analysis.mtc_filter.hpomtcfilter method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.filter_method_name", false]], "filter_method_name() (gpsea.analysis.mtc_filter.phenotypemtcfilter method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter.filter_method_name", false]], "filter_method_name() (gpsea.analysis.mtc_filter.specifiedtermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.filter_method_name", false]], "filter_method_name() (gpsea.analysis.mtc_filter.usealltermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.filter_method_name", false]], "find_coordinates() (gpsea.preprocessing.phenopacketvariantcoordinatefinder method)": [[14, "gpsea.preprocessing.PhenopacketVariantCoordinateFinder.find_coordinates", false]], "find_coordinates() (gpsea.preprocessing.variantcoordinatefinder method)": [[14, "gpsea.preprocessing.VariantCoordinateFinder.find_coordinates", false]], "find_coordinates() (gpsea.preprocessing.vvhgvsvariantcoordinatefinder method)": [[14, "gpsea.preprocessing.VVHgvsVariantCoordinateFinder.find_coordinates", false]], "fisherexacttest (class in gpsea.analysis.pcats.stats)": [[4, "gpsea.analysis.pcats.stats.FisherExactTest", false]], "five_prime_utr_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.FIVE_PRIME_UTR_VARIANT", false]], "for_sample() (gpsea.model.genotypes method)": [[12, "gpsea.model.Genotypes.for_sample", false]], "format_as_string() (gpsea.view.variantformatter method)": [[16, "gpsea.view.VariantFormatter.format_as_string", false]], "format_coordinates_for_vep_query() (gpsea.preprocessing.vepfunctionalannotator static method)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator.format_coordinates_for_vep_query", false]], "frameshift_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.FRAMESHIFT_VARIANT", false]], "from_feature_frame() (gpsea.model.proteinmetadata static method)": [[12, "gpsea.model.ProteinMetadata.from_feature_frame", false]], "from_mapping() (gpsea.model.genotypes static method)": [[12, "gpsea.model.Genotypes.from_mapping", false]], "from_patients() (gpsea.model.cohort static method)": [[12, "gpsea.model.Cohort.from_patients", false]], "from_query_curies() (gpsea.analysis.pscore.countingphenotypescorer static method)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer.from_query_curies", false]], "from_raw_parts() (gpsea.analysis.predicate.categorization static method)": [[5, "gpsea.analysis.predicate.Categorization.from_raw_parts", false]], "from_raw_parts() (gpsea.model.patient static method)": [[12, "gpsea.model.Patient.from_raw_parts", false]], "from_raw_parts() (gpsea.model.phenotype static method)": [[12, "gpsea.model.Phenotype.from_raw_parts", false]], "from_string() (gpsea.model.featuretype static method)": [[12, "gpsea.model.FeatureType.from_string", false]], "from_term() (gpsea.model.phenotype static method)": [[12, "gpsea.model.Phenotype.from_term", false]], "from_uniprot_json() (gpsea.model.proteinmetadata static method)": [[12, "gpsea.model.ProteinMetadata.from_uniprot_json", false]], "from_vcf_literal() (gpsea.model.variantcoordinates static method)": [[12, "gpsea.model.VariantCoordinates.from_vcf_literal", false]], "from_vcf_symbolic() (gpsea.model.variantcoordinates static method)": [[12, "gpsea.model.VariantCoordinates.from_vcf_symbolic", false]], "functionalannotator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.FunctionalAnnotator", false]], "genbank_acc (gpsea.model.genome.contig property)": [[13, "gpsea.model.genome.Contig.genbank_acc", false]], "gene() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.gene", false]], "gene_id (gpsea.model.imprecisesvinfo property)": [[12, "gpsea.model.ImpreciseSvInfo.gene_id", false]], "gene_id (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.gene_id", false]], "gene_id (gpsea.model.transcriptinfoaware property)": [[12, "gpsea.model.TranscriptInfoAware.gene_id", false]], "gene_symbol (gpsea.model.imprecisesvinfo property)": [[12, "gpsea.model.ImpreciseSvInfo.gene_symbol", false]], "genecoordinateservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.GeneCoordinateService", false]], "genome_build_id (gpsea.model.genome.genomebuild property)": [[13, "gpsea.model.genome.GenomeBuild.genome_build_id", false]], "genomebuild (class in gpsea.model.genome)": [[13, "gpsea.model.genome.GenomeBuild", false]], "genomebuildidentifier (class in gpsea.model.genome)": [[13, "gpsea.model.genome.GenomeBuildIdentifier", false]], "genomicregion (class in gpsea.model.genome)": [[13, "gpsea.model.genome.GenomicRegion", false]], "genotype (class in gpsea.model)": [[12, "gpsea.model.Genotype", false]], "genotype_for_sample() (gpsea.model.genotyped method)": [[12, "gpsea.model.Genotyped.genotype_for_sample", false]], "genotype_phenotype_scores (gpsea.analysis.pscore.phenotypescoreanalysisresult property)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.genotype_phenotype_scores", false]], "genotyped (class in gpsea.model)": [[12, "gpsea.model.Genotyped", false]], "genotypepolypredicate (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.GenotypePolyPredicate", false]], "genotypes (class in gpsea.model)": [[12, "gpsea.model.Genotypes", false]], "genotypes (gpsea.model.genotyped property)": [[12, "gpsea.model.Genotyped.genotypes", false]], "genotypes (gpsea.model.variant property)": [[12, "gpsea.model.Variant.genotypes", false]], "genotypes_have_same_hpo_proportions() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.genotypes_have_same_hpo_proportions", false]], "get_annotations() (gpsea.preprocessing.proteinannotationcache method)": [[14, "gpsea.preprocessing.ProteinAnnotationCache.get_annotations", false]], "get_annotations() (gpsea.preprocessing.variantannotationcache method)": [[14, "gpsea.preprocessing.VariantAnnotationCache.get_annotations", false]], "get_cache_dir_path() (in module gpsea.config)": [[10, "gpsea.config.get_cache_dir_path", false]], "get_categories() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_categories", false]], "get_categorizations() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.get_categorizations", false]], "get_categorizations() (gpsea.analysis.predicate.phenotype.diseasepresencepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.get_categorizations", false]], "get_categorizations() (gpsea.analysis.predicate.phenotype.hpopredicate method)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate.get_categorizations", false]], "get_categorizations() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_categorizations", false]], "get_category() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_category", false]], "get_category_name() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_category_name", false]], "get_category_names() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_category_names", false]], "get_cds_regions() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.get_cds_regions", false]], "get_coding_base_count() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.get_coding_base_count", false]], "get_codon_count() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.get_codon_count", false]], "get_excluded_count() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.get_excluded_count", false]], "get_features_variant_overlaps() (gpsea.model.proteinmetadata method)": [[12, "gpsea.model.ProteinMetadata.get_features_variant_overlaps", false]], "get_five_prime_utrs() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.get_five_prime_utrs", false]], "get_maximum_group_observed_hpo_frequency() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.get_maximum_group_observed_HPO_frequency", false]], "get_number_of_observed_hpo_observations() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.get_number_of_observed_hpo_observations", false]], "get_patient_ids() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.get_patient_ids", false]], "get_question() (gpsea.analysis.predicate.genotype.allelecounter method)": [[6, "gpsea.analysis.predicate.genotype.AlleleCounter.get_question", false]], "get_question() (gpsea.analysis.predicate.genotype.variantpredicate method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicate.get_question", false]], "get_question() (gpsea.analysis.pscore.countingphenotypescorer method)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer.get_question", false]], "get_question_base() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.get_question_base", false]], "get_question_base() (gpsea.analysis.predicate.phenotype.diseasepresencepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.get_question_base", false]], "get_question_base() (gpsea.analysis.predicate.phenotype.hpopredicate method)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate.get_question_base", false]], "get_question_base() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_question_base", false]], "get_response() (gpsea.preprocessing.vvmulticoordinateservice method)": [[14, "gpsea.preprocessing.VVMultiCoordinateService.get_response", false]], "get_three_prime_utrs() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.get_three_prime_utrs", false]], "get_variant_by_key() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.get_variant_by_key", false]], "gpsea": [[0, "module-gpsea", false]], "gpsea.analysis": [[1, "module-gpsea.analysis", false]], "gpsea.analysis.mtc_filter": [[2, "module-gpsea.analysis.mtc_filter", false]], "gpsea.analysis.pcats": [[3, "module-gpsea.analysis.pcats", false]], "gpsea.analysis.pcats.stats": [[4, "module-gpsea.analysis.pcats.stats", false]], "gpsea.analysis.predicate": [[5, "module-gpsea.analysis.predicate", false]], "gpsea.analysis.predicate.genotype": [[6, "module-gpsea.analysis.predicate.genotype", false]], "gpsea.analysis.predicate.phenotype": [[7, "module-gpsea.analysis.predicate.phenotype", false]], "gpsea.analysis.pscore": [[8, "module-gpsea.analysis.pscore", false]], "gpsea.analysis.pscore.stats": [[9, "module-gpsea.analysis.pscore.stats", false]], "gpsea.config": [[10, "module-gpsea.config", false]], "gpsea.io": [[11, "module-gpsea.io", false]], "gpsea.model": [[12, "module-gpsea.model", false]], "gpsea.model.genome": [[13, "module-gpsea.model.genome", false]], "gpsea.preprocessing": [[14, "module-gpsea.preprocessing", false]], "gpsea.util": [[15, "module-gpsea.util", false]], "gpsea.view": [[16, "module-gpsea.view", false]], "gpseajsondecoder (class in gpsea.io)": [[11, "gpsea.io.GpseaJSONDecoder", false]], "gpseajsonencoder (class in gpsea.io)": [[11, "gpsea.io.GpseaJSONEncoder", false]], "groups_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.groups_predicate", false]], "has_errors() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.has_errors", false]], "has_errors_or_warnings() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.has_errors_or_warnings", false]], "has_sv_info() (gpsea.model.variantinfo method)": [[12, "gpsea.model.VariantInfo.has_sv_info", false]], "has_variant_coordinates() (gpsea.model.variantinfo method)": [[12, "gpsea.model.VariantInfo.has_variant_coordinates", false]], "has_warnings() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.has_warnings", false]], "hemizygous (gpsea.model.genotype attribute)": [[12, "gpsea.model.Genotype.HEMIZYGOUS", false]], "heterozygous (gpsea.model.genotype attribute)": [[12, "gpsea.model.Genotype.HETEROZYGOUS", false]], "hgvs_cdna (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.hgvs_cdna", false]], "hgvsp (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.hgvsp", false]], "homozygous_alternate (gpsea.model.genotype attribute)": [[12, "gpsea.model.Genotype.HOMOZYGOUS_ALTERNATE", false]], "homozygous_reference (gpsea.model.genotype attribute)": [[12, "gpsea.model.Genotype.HOMOZYGOUS_REFERENCE", false]], "hpomtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter", false]], "hpopredicate (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate", false]], "hpotermanalysis (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.HpoTermAnalysis", false]], "hpotermanalysisresult (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult", false]], "identifier (gpsea.model.disease property)": [[12, "gpsea.model.Disease.identifier", false]], "identifier (gpsea.model.genome.genomebuild property)": [[13, "gpsea.model.genome.GenomeBuild.identifier", false]], "identifier (gpsea.model.genome.genomebuildidentifier property)": [[13, "gpsea.model.genome.GenomeBuildIdentifier.identifier", false]], "identifier (gpsea.model.phenotype property)": [[12, "gpsea.model.Phenotype.identifier", false]], "identifier (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.identifier", false]], "imprecisesvfunctionalannotator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.ImpreciseSvFunctionalAnnotator", false]], "imprecisesvinfo (class in gpsea.model)": [[12, "gpsea.model.ImpreciseSvInfo", false]], "incomplete_terminal_codon_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.INCOMPLETE_TERMINAL_CODON_VARIANT", false]], "info (gpsea.model.proteinfeature property)": [[12, "gpsea.model.ProteinFeature.info", false]], "inframe_deletion (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.INFRAME_DELETION", false]], "inframe_insertion (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.INFRAME_INSERTION", false]], "ins (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.INS", false]], "intergenic_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.INTERGENIC_VARIANT", false]], "intron_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.INTRON_VARIANT", false]], "inv (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.INV", false]], "is_coding() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.is_coding", false]], "is_empty() (gpsea.model.genome.region method)": [[13, "gpsea.model.genome.Region.is_empty", false]], "is_female() (gpsea.model.sex method)": [[12, "gpsea.model.Sex.is_female", false]], "is_filtered_out() (gpsea.analysis.mtc_filter.phenotypemtcresult method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.is_filtered_out", false]], "is_large_imprecise_sv() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.is_large_imprecise_sv", false]], "is_male() (gpsea.model.sex method)": [[12, "gpsea.model.Sex.is_male", false]], "is_negative() (gpsea.model.genome.strand method)": [[13, "gpsea.model.genome.Strand.is_negative", false]], "is_observed (gpsea.model.phenotype property)": [[12, "gpsea.model.Phenotype.is_observed", false]], "is_ok() (gpsea.preprocessing.preprocessingvalidationresult method)": [[14, "gpsea.preprocessing.PreprocessingValidationResult.is_ok", false]], "is_passed() (gpsea.analysis.mtc_filter.phenotypemtcresult method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.is_passed", false]], "is_positive() (gpsea.model.genome.strand method)": [[13, "gpsea.model.genome.Strand.is_positive", false]], "is_preferred (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.is_preferred", false]], "is_preferred (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.is_preferred", false]], "is_present (gpsea.model.disease property)": [[12, "gpsea.model.Disease.is_present", false]], "is_present (gpsea.model.phenotype property)": [[12, "gpsea.model.Phenotype.is_present", false]], "is_provided() (gpsea.model.sex method)": [[12, "gpsea.model.Sex.is_provided", false]], "is_structural() (gpsea.model.variantcoordinates method)": [[12, "gpsea.model.VariantCoordinates.is_structural", false]], "is_structural() (gpsea.model.variantinfo method)": [[12, "gpsea.model.VariantInfo.is_structural", false]], "is_structural_deletion() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.is_structural_deletion", false]], "is_structural_variant() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.is_structural_variant", false]], "is_unknown() (gpsea.model.sex method)": [[12, "gpsea.model.Sex.is_unknown", false]], "issues (gpsea.preprocessing.notepad property)": [[14, "gpsea.preprocessing.Notepad.issues", false]], "iterate_nodes() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.iterate_nodes", false]], "label (gpsea.model.proteinmetadata property)": [[12, "gpsea.model.ProteinMetadata.label", false]], "label (gpsea.model.samplelabels property)": [[12, "gpsea.model.SampleLabels.label", false]], "label (gpsea.preprocessing.notepad property)": [[14, "gpsea.preprocessing.Notepad.label", false]], "label_summary() (gpsea.model.samplelabels method)": [[12, "gpsea.model.SampleLabels.label_summary", false]], "labels (gpsea.model.patient property)": [[12, "gpsea.model.Patient.labels", false]], "level (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.Level", false]], "level (gpsea.preprocessing.datasanityissue property)": [[14, "gpsea.preprocessing.DataSanityIssue.level", false]], "level (gpsea.preprocessing.notepadtree property)": [[14, "gpsea.preprocessing.NotepadTree.level", false]], "list_all_diseases() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.list_all_diseases", false]], "list_all_proteins() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.list_all_proteins", false]], "list_all_variants() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.list_all_variants", false]], "list_present_phenotypes() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.list_present_phenotypes", false]], "load_phenopacket_files() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.load_phenopacket_files", false]], "load_phenopacket_folder() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.load_phenopacket_folder", false]], "load_phenopackets() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.load_phenopackets", false]], "major_assembly (gpsea.model.genome.genomebuildidentifier property)": [[13, "gpsea.model.genome.GenomeBuildIdentifier.major_assembly", false]], "male (gpsea.model.sex attribute)": [[12, "gpsea.model.Sex.MALE", false]], "mannwhitneystatistic (class in gpsea.analysis.pscore.stats)": [[9, "gpsea.analysis.pscore.stats.MannWhitneyStatistic", false]], "marker_counts (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.marker_counts", false]], "mature_mirna_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.MATURE_MIRNA_VARIANT", false]], "message (gpsea.preprocessing.datasanityissue property)": [[14, "gpsea.preprocessing.DataSanityIssue.message", false]], "meta_label (gpsea.model.samplelabels property)": [[12, "gpsea.model.SampleLabels.meta_label", false]], "missense_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.MISSENSE_VARIANT", false]], "mnv (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.MNV", false]], "modeofinheritancepredicate (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate", false]], "module": [[0, "module-gpsea", false], [1, "module-gpsea.analysis", false], [2, "module-gpsea.analysis.mtc_filter", false], [3, "module-gpsea.analysis.pcats", false], [4, "module-gpsea.analysis.pcats.stats", false], [5, "module-gpsea.analysis.predicate", false], [6, "module-gpsea.analysis.predicate.genotype", false], [7, "module-gpsea.analysis.predicate.phenotype", false], [8, "module-gpsea.analysis.pscore", false], [9, "module-gpsea.analysis.pscore.stats", false], [10, "module-gpsea.config", false], [11, "module-gpsea.io", false], [12, "module-gpsea.model", false], [13, "module-gpsea.model.genome", false], [14, "module-gpsea.preprocessing", false], [15, "module-gpsea.util", false], [16, "module-gpsea.view", false]], "monoallelic_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.monoallelic_predicate", false]], "motif (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.MOTIF", false]], "motifs() (gpsea.model.proteinmetadata method)": [[12, "gpsea.model.ProteinMetadata.motifs", false]], "mtc_filter_name (gpsea.analysis.pcats.hpotermanalysisresult property)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_filter_name", false]], "mtc_filter_results (gpsea.analysis.pcats.hpotermanalysisresult property)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_filter_results", false]], "mtc_issue (gpsea.analysis.mtc_filter.phenotypemtcresult property)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.mtc_issue", false]], "mtc_name (gpsea.analysis.pcats.hpotermanalysisresult property)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_name", false]], "mtcstatsviewer (class in gpsea.view)": [[16, "gpsea.view.MtcStatsViewer", false]], "multiphenotypeanalysis (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysis", false]], "multiphenotypeanalysisresult (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult", false]], "n_categorizations() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.n_categorizations", false]], "n_usable (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.n_usable", false]], "name (gpsea.analysis.predicate.patientcategory property)": [[5, "gpsea.analysis.predicate.PatientCategory.name", false]], "name (gpsea.model.disease property)": [[12, "gpsea.model.Disease.name", false]], "name (gpsea.model.featureinfo property)": [[12, "gpsea.model.FeatureInfo.name", false]], "name (gpsea.model.genome.contig property)": [[13, "gpsea.model.genome.Contig.name", false]], "negative (gpsea.model.genome.strand attribute)": [[13, "gpsea.model.genome.Strand.NEGATIVE", false]], "nmd_transcript_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.NMD_TRANSCRIPT_VARIANT", false]], "no (gpsea.analysis.predicate.patientcategories attribute)": [[5, "gpsea.analysis.predicate.PatientCategories.NO", false]], "no_call (gpsea.model.genotype attribute)": [[12, "gpsea.model.Genotype.NO_CALL", false]], "no_genotype_has_more_than_one_hpo (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.NO_GENOTYPE_HAS_MORE_THAN_ONE_HPO", false]], "non_coding_transcript_exon_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.NON_CODING_TRANSCRIPT_EXON_VARIANT", false]], "non_coding_transcript_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.NON_CODING_TRANSCRIPT_VARIANT", false]], "non_specified_term (gpsea.analysis.mtc_filter.specifiedtermsmtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.NON_SPECIFIED_TERM", false]], "noncoding_effects (gpsea.preprocessing.vepfunctionalannotator attribute)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator.NONCODING_EFFECTS", false]], "notepad (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.Notepad", false]], "notepadtree (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.NotepadTree", false]], "object_hook() (gpsea.io.gpseajsondecoder static method)": [[11, "gpsea.io.GpseaJSONDecoder.object_hook", false]], "observed (gpsea.model.phenotype property)": [[12, "gpsea.model.Phenotype.observed", false]], "ok (gpsea.analysis.mtc_filter.phenotypemtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter.OK", false]], "ok() (gpsea.analysis.mtc_filter.phenotypemtcresult static method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.ok", false]], "one_genotype_has_zero_hpo_observations() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.one_genotype_has_zero_hpo_observations", false]], "open_text_io_handle_for_reading() (in module gpsea.util)": [[15, "gpsea.util.open_text_io_handle_for_reading", false]], "opposite() (gpsea.model.genome.strand method)": [[13, "gpsea.model.genome.Strand.opposite", false]], "overlapping_exons (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.overlapping_exons", false]], "overlaps_with() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.overlaps_with", false]], "overlaps_with() (gpsea.model.genome.region method)": [[13, "gpsea.model.genome.Region.overlaps_with", false]], "parse_multiple() (gpsea.preprocessing.vvmulticoordinateservice method)": [[14, "gpsea.preprocessing.VVMultiCoordinateService.parse_multiple", false]], "parse_response() (gpsea.preprocessing.vvmulticoordinateservice method)": [[14, "gpsea.preprocessing.VVMultiCoordinateService.parse_response", false]], "parse_uniprot_json() (gpsea.preprocessing.uniprotproteinmetadataservice static method)": [[14, "gpsea.preprocessing.UniprotProteinMetadataService.parse_uniprot_json", false]], "patch (gpsea.model.genome.genomebuildidentifier property)": [[13, "gpsea.model.genome.GenomeBuildIdentifier.patch", false]], "patient (class in gpsea.model)": [[12, "gpsea.model.Patient", false]], "patient_id (gpsea.model.patient property)": [[12, "gpsea.model.Patient.patient_id", false]], "patientcategories (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.PatientCategories", false]], "patientcategory (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.PatientCategory", false]], "patientcreator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.PatientCreator", false]], "phenopacketpatientcreator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.PhenopacketPatientCreator", false]], "phenopacketvariantcoordinatefinder (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.PhenopacketVariantCoordinateFinder", false]], "phenotype (class in gpsea.model)": [[12, "gpsea.model.Phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.diseasepresencepredicate property)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.hpopredicate property)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate.phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.phenotypecategorization property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypeCategorization.phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.phenotypepolypredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.phenotype", false]], "phenotypecategorization (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypeCategorization", false]], "phenotypecreator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.PhenotypeCreator", false]], "phenotypemtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter", false]], "phenotypemtcissue (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcIssue", false]], "phenotypemtcresult (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult", false]], "phenotypepolypredicate (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate", false]], "phenotypes (gpsea.model.patient property)": [[12, "gpsea.model.Patient.phenotypes", false]], "phenotypescoreanalysis (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysis", false]], "phenotypescoreanalysisresult (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult", false]], "phenotypescorer (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.PhenotypeScorer", false]], "phenotypescorestatistic (class in gpsea.analysis.pscore.stats)": [[9, "gpsea.analysis.pscore.stats.PhenotypeScoreStatistic", false]], "plot_boxplots() (gpsea.analysis.pscore.phenotypescoreanalysisresult method)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.plot_boxplots", false]], "policy (gpsea.preprocessing.preprocessingvalidationresult property)": [[14, "gpsea.preprocessing.PreprocessingValidationResult.policy", false]], "polypredicate (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.PolyPredicate", false]], "positive (gpsea.model.genome.strand attribute)": [[13, "gpsea.model.genome.Strand.POSITIVE", false]], "possible_results() (gpsea.analysis.mtc_filter.hpomtcfilter method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.possible_results", false]], "possible_results() (gpsea.analysis.mtc_filter.phenotypemtcfilter method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter.possible_results", false]], "possible_results() (gpsea.analysis.mtc_filter.specifiedtermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.possible_results", false]], "possible_results() (gpsea.analysis.mtc_filter.usealltermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.possible_results", false]], "prepare_hpo_terms_of_interest() (in module gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.prepare_hpo_terms_of_interest", false]], "prepare_notepad() (gpsea.preprocessing.auditor static method)": [[14, "gpsea.preprocessing.Auditor.prepare_notepad", false]], "prepare_predicates_for_terms_of_interest() (in module gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.prepare_predicates_for_terms_of_interest", false]], "preprocessingvalidationresult (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.PreprocessingValidationResult", false]], "present_diseases() (gpsea.model.patient method)": [[12, "gpsea.model.Patient.present_diseases", false]], "present_phenotype_categorization (gpsea.analysis.predicate.phenotype.diseasepresencepredicate property)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.present_phenotype_categorization", false]], "present_phenotype_categorization (gpsea.analysis.predicate.phenotype.hpopredicate property)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate.present_phenotype_categorization", false]], "present_phenotype_categorization (gpsea.analysis.predicate.phenotype.phenotypepolypredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.present_phenotype_categorization", false]], "present_phenotype_category (gpsea.analysis.predicate.phenotype.phenotypepolypredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.present_phenotype_category", false]], "present_phenotypes() (gpsea.model.patient method)": [[12, "gpsea.model.Patient.present_phenotypes", false]], "process() (gpsea.preprocessing.auditor method)": [[14, "gpsea.preprocessing.Auditor.process", false]], "process() (gpsea.preprocessing.cohortcreator method)": [[14, "gpsea.preprocessing.CohortCreator.process", false]], "process() (gpsea.preprocessing.phenopacketpatientcreator method)": [[14, "gpsea.preprocessing.PhenopacketPatientCreator.process", false]], "process() (gpsea.preprocessing.phenotypecreator method)": [[14, "gpsea.preprocessing.PhenotypeCreator.process", false]], "process() (gpsea.view.cohortvariantviewer method)": [[16, "gpsea.view.CohortVariantViewer.process", false]], "process() (gpsea.view.cohortviewable method)": [[16, "gpsea.view.CohortViewable.process", false]], "process() (gpsea.view.diseaseviewable method)": [[16, "gpsea.view.DiseaseViewable.process", false]], "process() (gpsea.view.mtcstatsviewer method)": [[16, "gpsea.view.MtcStatsViewer.process", false]], "process() (gpsea.view.proteinviewable method)": [[16, "gpsea.view.ProteinViewable.process", false]], "process_response() (gpsea.preprocessing.vepfunctionalannotator method)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator.process_response", false]], "protcachingmetadataservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.ProtCachingMetadataService", false]], "protein_altering_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.PROTEIN_ALTERING_VARIANT", false]], "protein_effect_location (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.protein_effect_location", false]], "protein_feature() (gpsea.analysis.predicate.genotype.proteinpredicates method)": [[6, "gpsea.analysis.predicate.genotype.ProteinPredicates.protein_feature", false]], "protein_feature_ends (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_ends", false]], "protein_feature_names (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_names", false]], "protein_feature_starts (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_starts", false]], "protein_feature_type() (gpsea.analysis.predicate.genotype.proteinpredicates method)": [[6, "gpsea.analysis.predicate.genotype.ProteinPredicates.protein_feature_type", false]], "protein_feature_types (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_types", false]], "protein_features (gpsea.model.proteinmetadata property)": [[12, "gpsea.model.ProteinMetadata.protein_features", false]], "protein_id (gpsea.model.proteinmetadata property)": [[12, "gpsea.model.ProteinMetadata.protein_id", false]], "protein_id (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.protein_id", false]], "protein_id (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_id", false]], "protein_length (gpsea.model.proteinmetadata property)": [[12, "gpsea.model.ProteinMetadata.protein_length", false]], "protein_length (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_length", false]], "protein_metadata (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_metadata", false]], "proteinannotationcache (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.ProteinAnnotationCache", false]], "proteinfeature (class in gpsea.model)": [[12, "gpsea.model.ProteinFeature", false]], "proteinmetadata (class in gpsea.model)": [[12, "gpsea.model.ProteinMetadata", false]], "proteinmetadataservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.ProteinMetadataService", false]], "proteinpredicates (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.ProteinPredicates", false]], "proteinviewable (class in gpsea.view)": [[16, "gpsea.view.ProteinViewable", false]], "proteinvisualizable (class in gpsea.view)": [[16, "gpsea.view.ProteinVisualizable", false]], "proteinvisualizer (class in gpsea.view)": [[16, "gpsea.view.ProteinVisualizer", false]], "pval (gpsea.analysis.pscore.phenotypescoreanalysisresult property)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.pval", false]], "pvals (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.pvals", false]], "reason (gpsea.analysis.mtc_filter.phenotypemtcissue attribute)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcIssue.reason", false]], "reason (gpsea.analysis.mtc_filter.phenotypemtcresult property)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.reason", false]], "ref (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.ref", false]], "ref_length() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.ref_length", false]], "refseq_name (gpsea.model.genome.contig property)": [[13, "gpsea.model.genome.Contig.refseq_name", false]], "region (class in gpsea.model.genome)": [[13, "gpsea.model.genome.Region", false]], "region (gpsea.model.featureinfo property)": [[12, "gpsea.model.FeatureInfo.region", false]], "region (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.REGION", false]], "region (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.region", false]], "region (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.region", false]], "region() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.region", false]], "regions() (gpsea.model.proteinmetadata method)": [[12, "gpsea.model.ProteinMetadata.regions", false]], "regulatory_region_ablation (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.REGULATORY_REGION_ABLATION", false]], "regulatory_region_amplification (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.REGULATORY_REGION_AMPLIFICATION", false]], "regulatory_region_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.REGULATORY_REGION_VARIANT", false]], "repeat (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.REPEAT", false]], "repeats() (gpsea.model.proteinmetadata method)": [[12, "gpsea.model.ProteinMetadata.repeats", false]], "same_count_as_the_only_child (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.SAME_COUNT_AS_THE_ONLY_CHILD", false]], "samplelabels (class in gpsea.model)": [[12, "gpsea.model.SampleLabels", false]], "score() (gpsea.analysis.pscore.countingphenotypescorer method)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer.score", false]], "score() (gpsea.analysis.pscore.devriesphenotypescorer method)": [[8, "gpsea.analysis.pscore.DeVriesPhenotypeScorer.score", false]], "score() (gpsea.analysis.pscore.phenotypescorer method)": [[8, "gpsea.analysis.pscore.PhenotypeScorer.score", false]], "sequence_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SEQUENCE_VARIANT", false]], "sex (class in gpsea.model)": [[12, "gpsea.model.Sex", false]], "sex (gpsea.model.patient property)": [[12, "gpsea.model.Patient.sex", false]], "sex_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.sex_predicate", false]], "single() (gpsea.model.genotypes static method)": [[12, "gpsea.model.Genotypes.single", false]], "skipping_general_term (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.SKIPPING_GENERAL_TERM", false]], "skipping_non_phenotype_term (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.SKIPPING_NON_PHENOTYPE_TERM", false]], "skipping_same_observed_proportions (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.SKIPPING_SAME_OBSERVED_PROPORTIONS", false]], "skipping_since_one_genotype_had_zero_observations (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.SKIPPING_SINCE_ONE_GENOTYPE_HAD_ZERO_OBSERVATIONS", false]], "snv (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.SNV", false]], "solution (gpsea.preprocessing.datasanityissue property)": [[14, "gpsea.preprocessing.DataSanityIssue.solution", false]], "some_cell_has_greater_than_one_count() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.some_cell_has_greater_than_one_count", false]], "specifiedtermsmtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter", false]], "splice_acceptor_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_ACCEPTOR_VARIANT", false]], "splice_donor_5th_base_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_DONOR_5TH_BASE_VARIANT", false]], "splice_donor_region_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_DONOR_REGION_VARIANT", false]], "splice_donor_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_DONOR_VARIANT", false]], "splice_polypyrimidine_tract_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_POLYPYRIMIDINE_TRACT_VARIANT", false]], "splice_region_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_REGION_VARIANT", false]], "start (gpsea.model.featureinfo property)": [[12, "gpsea.model.FeatureInfo.start", false]], "start (gpsea.model.genome.region property)": [[13, "gpsea.model.genome.Region.start", false]], "start (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.start", false]], "start_lost (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.START_LOST", false]], "start_on_strand() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.start_on_strand", false]], "start_retained_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.START_RETAINED_VARIANT", false]], "stop_gained (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.STOP_GAINED", false]], "stop_lost (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.STOP_LOST", false]], "stop_retained_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.STOP_RETAINED_VARIANT", false]], "store_annotations() (gpsea.preprocessing.proteinannotationcache method)": [[14, "gpsea.preprocessing.ProteinAnnotationCache.store_annotations", false]], "store_annotations() (gpsea.preprocessing.variantannotationcache method)": [[14, "gpsea.preprocessing.VariantAnnotationCache.store_annotations", false]], "strand (class in gpsea.model.genome)": [[13, "gpsea.model.genome.Strand", false]], "strand (gpsea.model.genome.genomicregion property)": [[13, "gpsea.model.genome.GenomicRegion.strand", false]], "strand (gpsea.model.genome.stranded property)": [[13, "gpsea.model.genome.Stranded.strand", false]], "stranded (class in gpsea.model.genome)": [[13, "gpsea.model.genome.Stranded", false]], "structural_so_id_to_display() (gpsea.model.varianteffect static method)": [[12, "gpsea.model.VariantEffect.structural_so_id_to_display", false]], "structural_type (gpsea.model.imprecisesvinfo property)": [[12, "gpsea.model.ImpreciseSvInfo.structural_type", false]], "structural_type() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.structural_type", false]], "summarize() (gpsea.preprocessing.preprocessingvalidationresult method)": [[14, "gpsea.preprocessing.PreprocessingValidationResult.summarize", false]], "summarize_hpo_analysis() (in module gpsea.view)": [[16, "gpsea.view.summarize_hpo_analysis", false]], "supports_shape (gpsea.analysis.pcats.stats.countstatistic property)": [[4, "gpsea.analysis.pcats.stats.CountStatistic.supports_shape", false]], "supports_shape (gpsea.analysis.pcats.stats.fisherexacttest property)": [[4, "gpsea.analysis.pcats.stats.FisherExactTest.supports_shape", false]], "sv_info (gpsea.model.variantinfo property)": [[12, "gpsea.model.VariantInfo.sv_info", false]], "symbol (gpsea.model.genome.strand property)": [[13, "gpsea.model.genome.Strand.symbol", false]], "synonymous_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SYNONYMOUS_VARIANT", false]], "test() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.test", false]], "test() (gpsea.analysis.predicate.genotype.variantpredicate method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicate.test", false]], "test() (gpsea.analysis.predicate.phenotype.diseasepresencepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.test", false]], "test() (gpsea.analysis.predicate.phenotype.hpopredicate method)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate.test", false]], "test() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.test", false]], "tf_binding_site_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.TF_BINDING_SITE_VARIANT", false]], "tfbs_ablation (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.TFBS_ABLATION", false]], "tfbs_amplification (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.TFBS_AMPLIFICATION", false]], "three_prime_utr_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.THREE_PRIME_UTR_VARIANT", false]], "to_display() (gpsea.model.varianteffect method)": [[12, "gpsea.model.VariantEffect.to_display", false]], "to_negative_strand() (gpsea.model.genome.transposable method)": [[13, "gpsea.model.genome.Transposable.to_negative_strand", false]], "to_opposite_strand() (gpsea.model.genome.transposable method)": [[13, "gpsea.model.genome.Transposable.to_opposite_strand", false]], "to_positive_strand() (gpsea.model.genome.transposable method)": [[13, "gpsea.model.genome.Transposable.to_positive_strand", false]], "to_string() (gpsea.model.proteinfeature method)": [[12, "gpsea.model.ProteinFeature.to_string", false]], "total_patient_count (gpsea.model.cohort property)": [[12, "gpsea.model.Cohort.total_patient_count", false]], "total_tests (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.total_tests", false]], "transcript() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.transcript", false]], "transcript_ablation (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.TRANSCRIPT_ABLATION", false]], "transcript_amplification (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.TRANSCRIPT_AMPLIFICATION", false]], "transcript_coordinates (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.transcript_coordinates", false]], "transcript_id (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.transcript_id", false]], "transcript_id (gpsea.model.transcriptinfoaware property)": [[12, "gpsea.model.TranscriptInfoAware.transcript_id", false]], "transcript_id (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.transcript_id", false]], "transcriptannotation (class in gpsea.model)": [[12, "gpsea.model.TranscriptAnnotation", false]], "transcriptcoordinates (class in gpsea.model)": [[12, "gpsea.model.TranscriptCoordinates", false]], "transcriptcoordinateservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.TranscriptCoordinateService", false]], "transcriptinfoaware (class in gpsea.model)": [[12, "gpsea.model.TranscriptInfoAware", false]], "translocation (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.TRANSLOCATION", false]], "transposable (class in gpsea.model.genome)": [[13, "gpsea.model.genome.Transposable", false]], "transpose_coordinate() (in module gpsea.model.genome)": [[13, "gpsea.model.genome.transpose_coordinate", false]], "true() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.true", false]], "tteststatistic (class in gpsea.analysis.pscore.stats)": [[9, "gpsea.analysis.pscore.stats.TTestStatistic", false]], "tx_annotations (gpsea.model.variant property)": [[12, "gpsea.model.Variant.tx_annotations", false]], "ucsc_name (gpsea.model.genome.contig property)": [[13, "gpsea.model.genome.Contig.ucsc_name", false]], "uniprotproteinmetadataservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.UniprotProteinMetadataService", false]], "unknown_sex (gpsea.model.sex attribute)": [[12, "gpsea.model.Sex.UNKNOWN_SEX", false]], "upstream_gene_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.UPSTREAM_GENE_VARIANT", false]], "usealltermsmtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter", false]], "varcachingfunctionalannotator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VarCachingFunctionalAnnotator", false]], "variant (class in gpsea.model)": [[12, "gpsea.model.Variant", false]], "variant_class (gpsea.model.imprecisesvinfo property)": [[12, "gpsea.model.ImpreciseSvInfo.variant_class", false]], "variant_class (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.variant_class", false]], "variant_class (gpsea.model.variantinfo property)": [[12, "gpsea.model.VariantInfo.variant_class", false]], "variant_class() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.variant_class", false]], "variant_coordinates (gpsea.model.variantinfo property)": [[12, "gpsea.model.VariantInfo.variant_coordinates", false]], "variant_effect() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.variant_effect", false]], "variant_effect_count_by_tx() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.variant_effect_count_by_tx", false]], "variant_effects (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.variant_effects", false]], "variant_effects (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.variant_effects", false]], "variant_info (gpsea.model.variant property)": [[12, "gpsea.model.Variant.variant_info", false]], "variant_info (gpsea.model.variantinfoaware property)": [[12, "gpsea.model.VariantInfoAware.variant_info", false]], "variant_key (gpsea.model.imprecisesvinfo property)": [[12, "gpsea.model.ImpreciseSvInfo.variant_key", false]], "variant_key (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.variant_key", false]], "variant_key (gpsea.model.variantinfo property)": [[12, "gpsea.model.VariantInfo.variant_key", false]], "variant_key() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.variant_key", false]], "variant_locations (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.variant_locations", false]], "variant_locations_counted_absolute (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.variant_locations_counted_absolute", false]], "variantannotationcache (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VariantAnnotationCache", false]], "variantclass (class in gpsea.model)": [[12, "gpsea.model.VariantClass", false]], "variantcoordinatefinder (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VariantCoordinateFinder", false]], "variantcoordinates (class in gpsea.model)": [[12, "gpsea.model.VariantCoordinates", false]], "varianteffect (class in gpsea.model)": [[12, "gpsea.model.VariantEffect", false]], "variantformatter (class in gpsea.view)": [[16, "gpsea.view.VariantFormatter", false]], "variantinfo (class in gpsea.model)": [[12, "gpsea.model.VariantInfo", false]], "variantinfoaware (class in gpsea.model)": [[12, "gpsea.model.VariantInfoAware", false]], "variantpredicate (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicate", false]], "variantpredicates (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates", false]], "variants (gpsea.model.patient property)": [[12, "gpsea.model.Patient.variants", false]], "varianttranscriptvisualizer (class in gpsea.view)": [[16, "gpsea.view.VariantTranscriptVisualizer", false]], "vepfunctionalannotator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator", false]], "visit() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.visit", false]], "vvhgvsvariantcoordinatefinder (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VVHgvsVariantCoordinateFinder", false]], "vvmulticoordinateservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VVMultiCoordinateService", false]], "warn (gpsea.preprocessing.level attribute)": [[14, "gpsea.preprocessing.Level.WARN", false]], "warning_count() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.warning_count", false]], "warnings() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.warnings", false]], "with_cache_folder() (gpsea.preprocessing.varcachingfunctionalannotator static method)": [[14, "gpsea.preprocessing.VarCachingFunctionalAnnotator.with_cache_folder", false]], "with_strand() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.with_strand", false]], "with_strand() (gpsea.model.genome.transposable method)": [[13, "gpsea.model.genome.Transposable.with_strand", false]], "wrap_scoring_function() (gpsea.analysis.pscore.phenotypescorer static method)": [[8, "gpsea.analysis.pscore.PhenotypeScorer.wrap_scoring_function", false]], "yes (gpsea.analysis.predicate.patientcategories attribute)": [[5, "gpsea.analysis.predicate.PatientCategories.YES", false]]}, "objects": {"": [[0, 0, 0, "-", "gpsea"]], "gpsea": [[1, 0, 0, "-", "analysis"], [10, 0, 0, "-", "config"], [11, 0, 0, "-", "io"], [12, 0, 0, "-", "model"], [14, 0, 0, "-", "preprocessing"], [15, 0, 0, "-", "util"], [16, 0, 0, "-", "view"]], "gpsea.analysis": [[2, 0, 0, "-", "mtc_filter"], [3, 0, 0, "-", "pcats"], [5, 0, 0, "-", "predicate"], [8, 0, 0, "-", "pscore"]], "gpsea.analysis.mtc_filter": [[2, 1, 1, "", "HpoMtcFilter"], [2, 1, 1, "", "PhenotypeMtcFilter"], [2, 1, 1, "", "PhenotypeMtcIssue"], [2, 1, 1, "", "PhenotypeMtcResult"], [2, 1, 1, "", "SpecifiedTermsMtcFilter"], [2, 1, 1, "", "UseAllTermsMtcFilter"]], "gpsea.analysis.mtc_filter.HpoMtcFilter": [[2, 2, 1, "", "NO_GENOTYPE_HAS_MORE_THAN_ONE_HPO"], [2, 2, 1, "", "SAME_COUNT_AS_THE_ONLY_CHILD"], [2, 2, 1, "", "SKIPPING_GENERAL_TERM"], [2, 2, 1, "", "SKIPPING_NON_PHENOTYPE_TERM"], [2, 2, 1, "", "SKIPPING_SAME_OBSERVED_PROPORTIONS"], [2, 2, 1, "", "SKIPPING_SINCE_ONE_GENOTYPE_HAD_ZERO_OBSERVATIONS"], [2, 3, 1, "", "default_filter"], [2, 3, 1, "", "filter"], [2, 3, 1, "", "filter_method_name"], [2, 3, 1, "", "genotypes_have_same_hpo_proportions"], [2, 3, 1, "", "get_maximum_group_observed_HPO_frequency"], [2, 3, 1, "", "get_number_of_observed_hpo_observations"], [2, 3, 1, "", "one_genotype_has_zero_hpo_observations"], [2, 3, 1, "", "possible_results"], [2, 3, 1, "", "some_cell_has_greater_than_one_count"]], "gpsea.analysis.mtc_filter.PhenotypeMtcFilter": [[2, 2, 1, "", "OK"], [2, 3, 1, "", "filter"], [2, 3, 1, "", "filter_method_name"], [2, 3, 1, "", "possible_results"]], "gpsea.analysis.mtc_filter.PhenotypeMtcIssue": [[2, 2, 1, "", "code"], [2, 2, 1, "", "reason"]], "gpsea.analysis.mtc_filter.PhenotypeMtcResult": [[2, 3, 1, "", "fail"], [2, 3, 1, "", "is_filtered_out"], [2, 3, 1, "", "is_passed"], [2, 4, 1, "", "mtc_issue"], [2, 3, 1, "", "ok"], [2, 4, 1, "", "reason"]], "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter": [[2, 2, 1, "", "NON_SPECIFIED_TERM"], [2, 3, 1, "", "filter"], [2, 3, 1, "", "filter_method_name"], [2, 3, 1, "", "possible_results"]], "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter": [[2, 3, 1, "", "filter"], [2, 3, 1, "", "filter_method_name"], [2, 3, 1, "", "possible_results"]], "gpsea.analysis.pcats": [[3, 1, 1, "", "DiseaseAnalysis"], [3, 1, 1, "", "HpoTermAnalysis"], [3, 1, 1, "", "HpoTermAnalysisResult"], [3, 1, 1, "", "MultiPhenotypeAnalysis"], [3, 1, 1, "", "MultiPhenotypeAnalysisResult"], [3, 5, 1, "", "apply_predicates_on_patients"], [4, 0, 0, "-", "stats"]], "gpsea.analysis.pcats.HpoTermAnalysisResult": [[3, 4, 1, "", "mtc_filter_name"], [3, 4, 1, "", "mtc_filter_results"], [3, 4, 1, "", "mtc_name"]], "gpsea.analysis.pcats.MultiPhenotypeAnalysis": [[3, 3, 1, "", "compare_genotype_vs_phenotypes"]], "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult": [[3, 4, 1, "", "all_counts"], [3, 4, 1, "", "corrected_pvals"], [3, 4, 1, "", "n_usable"], [3, 4, 1, "", "pvals"], [3, 4, 1, "", "total_tests"]], "gpsea.analysis.pcats.stats": [[4, 1, 1, "", "CountStatistic"], [4, 1, 1, "", "FisherExactTest"]], "gpsea.analysis.pcats.stats.CountStatistic": [[4, 3, 1, "", "compute_pval"], [4, 4, 1, "", "supports_shape"]], "gpsea.analysis.pcats.stats.FisherExactTest": [[4, 3, 1, "", "compute_pval"], [4, 4, 1, "", "supports_shape"]], "gpsea.analysis.predicate": [[5, 1, 1, "", "Categorization"], [5, 1, 1, "", "PatientCategories"], [5, 1, 1, "", "PatientCategory"], [5, 1, 1, "", "PolyPredicate"], [6, 0, 0, "-", "genotype"], [7, 0, 0, "-", "phenotype"]], "gpsea.analysis.predicate.Categorization": [[5, 4, 1, "", "category"], [5, 3, 1, "", "from_raw_parts"]], "gpsea.analysis.predicate.PatientCategories": [[5, 2, 1, "", "NO"], [5, 2, 1, "", "YES"]], "gpsea.analysis.predicate.PatientCategory": [[5, 4, 1, "", "cat_id"], [5, 4, 1, "", "description"], [5, 4, 1, "", "name"]], "gpsea.analysis.predicate.PolyPredicate": [[5, 3, 1, "", "display_question"], [5, 3, 1, "", "get_categories"], [5, 3, 1, "", "get_categorizations"], [5, 3, 1, "", "get_category"], [5, 3, 1, "", "get_category_name"], [5, 3, 1, "", "get_category_names"], [5, 3, 1, "", "get_question_base"], [5, 3, 1, "", "n_categorizations"], [5, 3, 1, "", "test"]], "gpsea.analysis.predicate.genotype": [[6, 1, 1, "", "AlleleCounter"], [6, 1, 1, "", "GenotypePolyPredicate"], [6, 1, 1, "", "ModeOfInheritancePredicate"], [6, 1, 1, "", "ProteinPredicates"], [6, 1, 1, "", "VariantPredicate"], [6, 1, 1, "", "VariantPredicates"], [6, 5, 1, "", "autosomal_dominant"], [6, 5, 1, "", "autosomal_recessive"], [6, 5, 1, "", "biallelic_predicate"], [6, 5, 1, "", "diagnosis_predicate"], [6, 5, 1, "", "groups_predicate"], [6, 5, 1, "", "monoallelic_predicate"], [6, 5, 1, "", "sex_predicate"]], "gpsea.analysis.predicate.genotype.AlleleCounter": [[6, 3, 1, "", "count"], [6, 3, 1, "", "get_question"]], "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate": [[6, 3, 1, "", "autosomal_dominant"], [6, 3, 1, "", "autosomal_recessive"], [6, 3, 1, "", "get_categorizations"], [6, 3, 1, "", "get_question_base"], [6, 3, 1, "", "test"]], "gpsea.analysis.predicate.genotype.ProteinPredicates": [[6, 3, 1, "", "protein_feature"], [6, 3, 1, "", "protein_feature_type"]], "gpsea.analysis.predicate.genotype.VariantPredicate": [[6, 3, 1, "", "get_question"], [6, 3, 1, "", "test"]], "gpsea.analysis.predicate.genotype.VariantPredicates": [[6, 3, 1, "", "all"], [6, 3, 1, "", "any"], [6, 3, 1, "", "change_length"], [6, 3, 1, "", "exon"], [6, 3, 1, "", "gene"], [6, 3, 1, "", "is_large_imprecise_sv"], [6, 3, 1, "", "is_structural_deletion"], [6, 3, 1, "", "is_structural_variant"], [6, 3, 1, "", "ref_length"], [6, 3, 1, "", "region"], [6, 3, 1, "", "structural_type"], [6, 3, 1, "", "transcript"], [6, 3, 1, "", "true"], [6, 3, 1, "", "variant_class"], [6, 3, 1, "", "variant_effect"], [6, 3, 1, "", "variant_key"]], "gpsea.analysis.predicate.phenotype": [[7, 1, 1, "", "DiseasePresencePredicate"], [7, 1, 1, "", "HpoPredicate"], [7, 1, 1, "", "PhenotypeCategorization"], [7, 1, 1, "", "PhenotypePolyPredicate"], [7, 5, 1, "", "prepare_hpo_terms_of_interest"], [7, 5, 1, "", "prepare_predicates_for_terms_of_interest"]], "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate": [[7, 3, 1, "", "get_categorizations"], [7, 3, 1, "", "get_question_base"], [7, 4, 1, "", "phenotype"], [7, 4, 1, "", "present_phenotype_categorization"], [7, 3, 1, "", "test"]], "gpsea.analysis.predicate.phenotype.HpoPredicate": [[7, 3, 1, "", "get_categorizations"], [7, 3, 1, "", "get_question_base"], [7, 4, 1, "", "phenotype"], [7, 4, 1, "", "present_phenotype_categorization"], [7, 3, 1, "", "test"]], "gpsea.analysis.predicate.phenotype.PhenotypeCategorization": [[7, 4, 1, "", "phenotype"]], "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate": [[7, 4, 1, "", "phenotype"], [7, 4, 1, "", "present_phenotype_categorization"], [7, 4, 1, "", "present_phenotype_category"]], "gpsea.analysis.pscore": [[8, 1, 1, "", "CountingPhenotypeScorer"], [8, 1, 1, "", "DeVriesPhenotypeScorer"], [8, 1, 1, "", "PhenotypeScoreAnalysis"], [8, 1, 1, "", "PhenotypeScoreAnalysisResult"], [8, 1, 1, "", "PhenotypeScorer"], [9, 0, 0, "-", "stats"]], "gpsea.analysis.pscore.CountingPhenotypeScorer": [[8, 3, 1, "", "from_query_curies"], [8, 3, 1, "", "get_question"], [8, 3, 1, "", "score"]], "gpsea.analysis.pscore.DeVriesPhenotypeScorer": [[8, 3, 1, "", "score"]], "gpsea.analysis.pscore.PhenotypeScoreAnalysis": [[8, 3, 1, "", "compare_genotype_vs_phenotype_score"]], "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult": [[8, 4, 1, "", "genotype_phenotype_scores"], [8, 3, 1, "", "plot_boxplots"], [8, 4, 1, "", "pval"]], "gpsea.analysis.pscore.PhenotypeScorer": [[8, 3, 1, "", "score"], [8, 3, 1, "", "wrap_scoring_function"]], "gpsea.analysis.pscore.stats": [[9, 1, 1, "", "MannWhitneyStatistic"], [9, 1, 1, "", "PhenotypeScoreStatistic"], [9, 1, 1, "", "TTestStatistic"]], "gpsea.analysis.pscore.stats.MannWhitneyStatistic": [[9, 3, 1, "", "compute_pval"]], "gpsea.analysis.pscore.stats.PhenotypeScoreStatistic": [[9, 3, 1, "", "compute_pval"]], "gpsea.analysis.pscore.stats.TTestStatistic": [[9, 3, 1, "", "compute_pval"]], "gpsea.config": [[10, 6, 1, "", "CACHE_ENV"], [10, 6, 1, "", "DEFAULT_CACHE_PATH"], [10, 5, 1, "", "get_cache_dir_path"]], "gpsea.io": [[11, 1, 1, "", "GpseaJSONDecoder"], [11, 1, 1, "", "GpseaJSONEncoder"]], "gpsea.io.GpseaJSONDecoder": [[11, 3, 1, "", "object_hook"]], "gpsea.io.GpseaJSONEncoder": [[11, 3, 1, "", "default"]], "gpsea.model": [[12, 1, 1, "", "Cohort"], [12, 1, 1, "", "Disease"], [12, 1, 1, "", "FeatureInfo"], [12, 1, 1, "", "FeatureType"], [12, 1, 1, "", "Genotype"], [12, 1, 1, "", "Genotyped"], [12, 1, 1, "", "Genotypes"], [12, 1, 1, "", "ImpreciseSvInfo"], [12, 1, 1, "", "Patient"], [12, 1, 1, "", "Phenotype"], [12, 1, 1, "", "ProteinFeature"], [12, 1, 1, "", "ProteinMetadata"], [12, 1, 1, "", "SampleLabels"], [12, 1, 1, "", "Sex"], [12, 1, 1, "", "TranscriptAnnotation"], [12, 1, 1, "", "TranscriptCoordinates"], [12, 1, 1, "", "TranscriptInfoAware"], [12, 1, 1, "", "Variant"], [12, 1, 1, "", "VariantClass"], [12, 1, 1, "", "VariantCoordinates"], [12, 1, 1, "", "VariantEffect"], [12, 1, 1, "", "VariantInfo"], [12, 1, 1, "", "VariantInfoAware"], [13, 0, 0, "-", "genome"]], "gpsea.model.Cohort": [[12, 3, 1, "", "all_diseases"], [12, 4, 1, "", "all_patients"], [12, 3, 1, "", "all_phenotypes"], [12, 4, 1, "", "all_transcript_ids"], [12, 3, 1, "", "all_variants"], [12, 3, 1, "", "from_patients"], [12, 3, 1, "", "get_excluded_count"], [12, 3, 1, "", "get_patient_ids"], [12, 3, 1, "", "get_variant_by_key"], [12, 3, 1, "", "list_all_diseases"], [12, 3, 1, "", "list_all_proteins"], [12, 3, 1, "", "list_all_variants"], [12, 3, 1, "", "list_present_phenotypes"], [12, 4, 1, "", "total_patient_count"], [12, 3, 1, "", "variant_effect_count_by_tx"]], "gpsea.model.Disease": [[12, 4, 1, "", "identifier"], [12, 4, 1, "", "is_present"], [12, 4, 1, "", "name"]], "gpsea.model.FeatureInfo": [[12, 4, 1, "", "end"], [12, 4, 1, "", "name"], [12, 4, 1, "", "region"], [12, 4, 1, "", "start"]], "gpsea.model.FeatureType": [[12, 2, 1, "", "COILED_COIL"], [12, 2, 1, "", "COMPOSITIONAL_BIAS"], [12, 2, 1, "", "DOMAIN"], [12, 2, 1, "", "MOTIF"], [12, 2, 1, "", "REGION"], [12, 2, 1, "", "REPEAT"], [12, 3, 1, "", "from_string"]], "gpsea.model.Genotype": [[12, 2, 1, "", "HEMIZYGOUS"], [12, 2, 1, "", "HETEROZYGOUS"], [12, 2, 1, "", "HOMOZYGOUS_ALTERNATE"], [12, 2, 1, "", "HOMOZYGOUS_REFERENCE"], [12, 2, 1, "", "NO_CALL"], [12, 4, 1, "", "code"]], "gpsea.model.Genotyped": [[12, 3, 1, "", "genotype_for_sample"], [12, 4, 1, "", "genotypes"]], "gpsea.model.Genotypes": [[12, 3, 1, "", "empty"], [12, 3, 1, "", "for_sample"], [12, 3, 1, "", "from_mapping"], [12, 3, 1, "", "single"]], "gpsea.model.ImpreciseSvInfo": [[12, 4, 1, "", "gene_id"], [12, 4, 1, "", "gene_symbol"], [12, 4, 1, "", "structural_type"], [12, 4, 1, "", "variant_class"], [12, 4, 1, "", "variant_key"]], "gpsea.model.Patient": [[12, 4, 1, "", "diseases"], [12, 3, 1, "", "excluded_diseases"], [12, 3, 1, "", "excluded_phenotypes"], [12, 3, 1, "", "from_raw_parts"], [12, 4, 1, "", "labels"], [12, 4, 1, "", "patient_id"], [12, 4, 1, "", "phenotypes"], [12, 3, 1, "", "present_diseases"], [12, 3, 1, "", "present_phenotypes"], [12, 4, 1, "", "sex"], [12, 4, 1, "", "variants"]], "gpsea.model.Phenotype": [[12, 3, 1, "", "from_raw_parts"], [12, 3, 1, "", "from_term"], [12, 4, 1, "", "identifier"], [12, 4, 1, "", "is_observed"], [12, 4, 1, "", "is_present"], [12, 4, 1, "", "observed"]], "gpsea.model.ProteinFeature": [[12, 3, 1, "", "create"], [12, 4, 1, "", "feature_type"], [12, 4, 1, "", "info"], [12, 3, 1, "", "to_string"]], "gpsea.model.ProteinMetadata": [[12, 3, 1, "", "domains"], [12, 3, 1, "", "from_feature_frame"], [12, 3, 1, "", "from_uniprot_json"], [12, 3, 1, "", "get_features_variant_overlaps"], [12, 4, 1, "", "label"], [12, 3, 1, "", "motifs"], [12, 4, 1, "", "protein_features"], [12, 4, 1, "", "protein_id"], [12, 4, 1, "", "protein_length"], [12, 3, 1, "", "regions"], [12, 3, 1, "", "repeats"]], "gpsea.model.SampleLabels": [[12, 4, 1, "", "label"], [12, 3, 1, "", "label_summary"], [12, 4, 1, "", "meta_label"]], "gpsea.model.Sex": [[12, 2, 1, "", "FEMALE"], [12, 2, 1, "", "MALE"], [12, 2, 1, "", "UNKNOWN_SEX"], [12, 3, 1, "", "is_female"], [12, 3, 1, "", "is_male"], [12, 3, 1, "", "is_provided"], [12, 3, 1, "", "is_unknown"]], "gpsea.model.TranscriptAnnotation": [[12, 4, 1, "", "gene_id"], [12, 4, 1, "", "hgvs_cdna"], [12, 4, 1, "", "hgvsp"], [12, 4, 1, "", "is_preferred"], [12, 4, 1, "", "overlapping_exons"], [12, 4, 1, "", "protein_effect_location"], [12, 4, 1, "", "protein_id"], [12, 4, 1, "", "transcript_id"], [12, 4, 1, "", "variant_effects"]], "gpsea.model.TranscriptCoordinates": [[12, 4, 1, "", "cds_end"], [12, 4, 1, "", "cds_start"], [12, 4, 1, "", "exons"], [12, 3, 1, "", "get_cds_regions"], [12, 3, 1, "", "get_coding_base_count"], [12, 3, 1, "", "get_codon_count"], [12, 3, 1, "", "get_five_prime_utrs"], [12, 3, 1, "", "get_three_prime_utrs"], [12, 4, 1, "", "identifier"], [12, 3, 1, "", "is_coding"], [12, 4, 1, "", "is_preferred"], [12, 4, 1, "", "region"]], "gpsea.model.TranscriptInfoAware": [[12, 4, 1, "", "gene_id"], [12, 4, 1, "", "transcript_id"]], "gpsea.model.Variant": [[12, 3, 1, "", "create_variant_from_scratch"], [12, 4, 1, "", "genotypes"], [12, 4, 1, "", "tx_annotations"], [12, 4, 1, "", "variant_info"]], "gpsea.model.VariantClass": [[12, 2, 1, "", "DEL"], [12, 2, 1, "", "DUP"], [12, 2, 1, "", "INS"], [12, 2, 1, "", "INV"], [12, 2, 1, "", "MNV"], [12, 2, 1, "", "SNV"], [12, 2, 1, "", "TRANSLOCATION"]], "gpsea.model.VariantCoordinates": [[12, 4, 1, "", "alt"], [12, 4, 1, "", "change_length"], [12, 4, 1, "", "chrom"], [12, 4, 1, "", "end"], [12, 3, 1, "", "from_vcf_literal"], [12, 3, 1, "", "from_vcf_symbolic"], [12, 3, 1, "", "is_structural"], [12, 4, 1, "", "ref"], [12, 4, 1, "", "region"], [12, 4, 1, "", "start"], [12, 4, 1, "", "variant_class"], [12, 4, 1, "", "variant_key"]], "gpsea.model.VariantEffect": [[12, 2, 1, "", "CODING_SEQUENCE_VARIANT"], [12, 2, 1, "", "DOWNSTREAM_GENE_VARIANT"], [12, 2, 1, "", "FEATURE_ELONGATION"], [12, 2, 1, "", "FEATURE_TRUNCATION"], [12, 2, 1, "", "FIVE_PRIME_UTR_VARIANT"], [12, 2, 1, "", "FRAMESHIFT_VARIANT"], [12, 2, 1, "", "INCOMPLETE_TERMINAL_CODON_VARIANT"], [12, 2, 1, "", "INFRAME_DELETION"], [12, 2, 1, "", "INFRAME_INSERTION"], [12, 2, 1, "", "INTERGENIC_VARIANT"], [12, 2, 1, "", "INTRON_VARIANT"], [12, 2, 1, "", "MATURE_MIRNA_VARIANT"], [12, 2, 1, "", "MISSENSE_VARIANT"], [12, 2, 1, "", "NMD_TRANSCRIPT_VARIANT"], [12, 2, 1, "", "NON_CODING_TRANSCRIPT_EXON_VARIANT"], [12, 2, 1, "", "NON_CODING_TRANSCRIPT_VARIANT"], [12, 2, 1, "", "PROTEIN_ALTERING_VARIANT"], [12, 2, 1, "", "REGULATORY_REGION_ABLATION"], [12, 2, 1, "", "REGULATORY_REGION_AMPLIFICATION"], [12, 2, 1, "", "REGULATORY_REGION_VARIANT"], [12, 2, 1, "", "SEQUENCE_VARIANT"], [12, 2, 1, "", "SPLICE_ACCEPTOR_VARIANT"], [12, 2, 1, "", "SPLICE_DONOR_5TH_BASE_VARIANT"], [12, 2, 1, "", "SPLICE_DONOR_REGION_VARIANT"], [12, 2, 1, "", "SPLICE_DONOR_VARIANT"], [12, 2, 1, "", "SPLICE_POLYPYRIMIDINE_TRACT_VARIANT"], [12, 2, 1, "", "SPLICE_REGION_VARIANT"], [12, 2, 1, "", "START_LOST"], [12, 2, 1, "", "START_RETAINED_VARIANT"], [12, 2, 1, "", "STOP_GAINED"], [12, 2, 1, "", "STOP_LOST"], [12, 2, 1, "", "STOP_RETAINED_VARIANT"], [12, 2, 1, "", "SYNONYMOUS_VARIANT"], [12, 2, 1, "", "TFBS_ABLATION"], [12, 2, 1, "", "TFBS_AMPLIFICATION"], [12, 2, 1, "", "TF_BINDING_SITE_VARIANT"], [12, 2, 1, "", "THREE_PRIME_UTR_VARIANT"], [12, 2, 1, "", "TRANSCRIPT_ABLATION"], [12, 2, 1, "", "TRANSCRIPT_AMPLIFICATION"], [12, 2, 1, "", "UPSTREAM_GENE_VARIANT"], [12, 4, 1, "", "curie"], [12, 3, 1, "", "structural_so_id_to_display"], [12, 3, 1, "", "to_display"]], "gpsea.model.VariantInfo": [[12, 3, 1, "", "has_sv_info"], [12, 3, 1, "", "has_variant_coordinates"], [12, 3, 1, "", "is_structural"], [12, 4, 1, "", "sv_info"], [12, 4, 1, "", "variant_class"], [12, 4, 1, "", "variant_coordinates"], [12, 4, 1, "", "variant_key"]], "gpsea.model.VariantInfoAware": [[12, 4, 1, "", "variant_info"]], "gpsea.model.genome": [[13, 1, 1, "", "Contig"], [13, 1, 1, "", "GenomeBuild"], [13, 1, 1, "", "GenomeBuildIdentifier"], [13, 1, 1, "", "GenomicRegion"], [13, 1, 1, "", "Region"], [13, 1, 1, "", "Strand"], [13, 1, 1, "", "Stranded"], [13, 1, 1, "", "Transposable"], [13, 5, 1, "", "transpose_coordinate"]], "gpsea.model.genome.Contig": [[13, 4, 1, "", "genbank_acc"], [13, 4, 1, "", "name"], [13, 4, 1, "", "refseq_name"], [13, 4, 1, "", "ucsc_name"]], "gpsea.model.genome.GenomeBuild": [[13, 3, 1, "", "contig_by_name"], [13, 4, 1, "", "contigs"], [13, 4, 1, "", "genome_build_id"], [13, 4, 1, "", "identifier"]], "gpsea.model.genome.GenomeBuildIdentifier": [[13, 4, 1, "", "identifier"], [13, 4, 1, "", "major_assembly"], [13, 4, 1, "", "patch"]], "gpsea.model.genome.GenomicRegion": [[13, 3, 1, "", "contains"], [13, 4, 1, "", "contig"], [13, 3, 1, "", "distance_to"], [13, 3, 1, "", "end_on_strand"], [13, 3, 1, "", "overlaps_with"], [13, 3, 1, "", "start_on_strand"], [13, 4, 1, "", "strand"], [13, 3, 1, "", "with_strand"]], "gpsea.model.genome.Region": [[13, 3, 1, "", "contains"], [13, 3, 1, "", "contains_pos"], [13, 3, 1, "", "distance_to"], [13, 4, 1, "", "end"], [13, 3, 1, "", "is_empty"], [13, 3, 1, "", "overlaps_with"], [13, 4, 1, "", "start"]], "gpsea.model.genome.Strand": [[13, 2, 1, "", "NEGATIVE"], [13, 2, 1, "", "POSITIVE"], [13, 3, 1, "", "is_negative"], [13, 3, 1, "", "is_positive"], [13, 3, 1, "", "opposite"], [13, 4, 1, "", "symbol"]], "gpsea.model.genome.Stranded": [[13, 4, 1, "", "strand"]], "gpsea.model.genome.Transposable": [[13, 3, 1, "", "to_negative_strand"], [13, 3, 1, "", "to_opposite_strand"], [13, 3, 1, "", "to_positive_strand"], [13, 3, 1, "", "with_strand"]], "gpsea.preprocessing": [[14, 1, 1, "", "Auditor"], [14, 1, 1, "", "CohortCreator"], [14, 1, 1, "", "DataSanityIssue"], [14, 1, 1, "", "DefaultImpreciseSvFunctionalAnnotator"], [14, 1, 1, "", "FunctionalAnnotator"], [14, 1, 1, "", "GeneCoordinateService"], [14, 1, 1, "", "ImpreciseSvFunctionalAnnotator"], [14, 1, 1, "", "Level"], [14, 1, 1, "", "Notepad"], [14, 1, 1, "", "NotepadTree"], [14, 1, 1, "", "PatientCreator"], [14, 1, 1, "", "PhenopacketPatientCreator"], [14, 1, 1, "", "PhenopacketVariantCoordinateFinder"], [14, 1, 1, "", "PhenotypeCreator"], [14, 1, 1, "", "PreprocessingValidationResult"], [14, 1, 1, "", "ProtCachingMetadataService"], [14, 1, 1, "", "ProteinAnnotationCache"], [14, 1, 1, "", "ProteinMetadataService"], [14, 1, 1, "", "TranscriptCoordinateService"], [14, 1, 1, "", "UniprotProteinMetadataService"], [14, 1, 1, "", "VVHgvsVariantCoordinateFinder"], [14, 1, 1, "", "VVMultiCoordinateService"], [14, 1, 1, "", "VarCachingFunctionalAnnotator"], [14, 1, 1, "", "VariantAnnotationCache"], [14, 1, 1, "", "VariantCoordinateFinder"], [14, 1, 1, "", "VepFunctionalAnnotator"], [14, 5, 1, "", "configure_caching_cohort_creator"], [14, 5, 1, "", "configure_caching_patient_creator"], [14, 5, 1, "", "configure_cohort_creator"], [14, 5, 1, "", "configure_default_protein_metadata_service"], [14, 5, 1, "", "configure_patient_creator"], [14, 5, 1, "", "configure_protein_metadata_service"], [14, 5, 1, "", "load_phenopacket_files"], [14, 5, 1, "", "load_phenopacket_folder"], [14, 5, 1, "", "load_phenopackets"]], "gpsea.preprocessing.Auditor": [[14, 3, 1, "", "prepare_notepad"], [14, 3, 1, "", "process"]], "gpsea.preprocessing.CohortCreator": [[14, 3, 1, "", "process"]], "gpsea.preprocessing.DataSanityIssue": [[14, 4, 1, "", "level"], [14, 4, 1, "", "message"], [14, 4, 1, "", "solution"]], "gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator": [[14, 3, 1, "", "annotate"]], "gpsea.preprocessing.FunctionalAnnotator": [[14, 3, 1, "", "annotate"]], "gpsea.preprocessing.GeneCoordinateService": [[14, 3, 1, "", "fetch_for_gene"]], "gpsea.preprocessing.ImpreciseSvFunctionalAnnotator": [[14, 3, 1, "", "annotate"]], "gpsea.preprocessing.Level": [[14, 2, 1, "", "ERROR"], [14, 2, 1, "", "WARN"]], "gpsea.preprocessing.Notepad": [[14, 3, 1, "", "add_error"], [14, 3, 1, "", "add_issue"], [14, 3, 1, "", "add_subsection"], [14, 3, 1, "", "add_warning"], [14, 3, 1, "", "error_count"], [14, 3, 1, "", "errors"], [14, 4, 1, "", "issues"], [14, 4, 1, "", "label"], [14, 3, 1, "", "warning_count"], [14, 3, 1, "", "warnings"]], "gpsea.preprocessing.NotepadTree": [[14, 3, 1, "", "add_subsection"], [14, 4, 1, "", "children"], [14, 3, 1, "", "has_errors"], [14, 3, 1, "", "has_errors_or_warnings"], [14, 3, 1, "", "has_warnings"], [14, 3, 1, "", "iterate_nodes"], [14, 4, 1, "", "level"], [14, 3, 1, "", "visit"]], "gpsea.preprocessing.PhenopacketPatientCreator": [[14, 3, 1, "", "process"]], "gpsea.preprocessing.PhenopacketVariantCoordinateFinder": [[14, 3, 1, "", "find_coordinates"]], "gpsea.preprocessing.PhenotypeCreator": [[14, 3, 1, "", "process"]], "gpsea.preprocessing.PreprocessingValidationResult": [[14, 3, 1, "", "is_ok"], [14, 4, 1, "", "policy"], [14, 3, 1, "", "summarize"]], "gpsea.preprocessing.ProtCachingMetadataService": [[14, 3, 1, "", "annotate"]], "gpsea.preprocessing.ProteinAnnotationCache": [[14, 3, 1, "", "get_annotations"], [14, 3, 1, "", "store_annotations"]], "gpsea.preprocessing.ProteinMetadataService": [[14, 3, 1, "", "annotate"]], "gpsea.preprocessing.TranscriptCoordinateService": [[14, 3, 1, "", "fetch"]], "gpsea.preprocessing.UniprotProteinMetadataService": [[14, 3, 1, "", "annotate"], [14, 3, 1, "", "parse_uniprot_json"]], "gpsea.preprocessing.VVHgvsVariantCoordinateFinder": [[14, 3, 1, "", "find_coordinates"]], "gpsea.preprocessing.VVMultiCoordinateService": [[14, 3, 1, "", "fetch"], [14, 3, 1, "", "fetch_for_gene"], [14, 3, 1, "", "get_response"], [14, 3, 1, "", "parse_multiple"], [14, 3, 1, "", "parse_response"]], "gpsea.preprocessing.VarCachingFunctionalAnnotator": [[14, 3, 1, "", "annotate"], [14, 3, 1, "", "with_cache_folder"]], "gpsea.preprocessing.VariantAnnotationCache": [[14, 3, 1, "", "get_annotations"], [14, 3, 1, "", "store_annotations"]], "gpsea.preprocessing.VariantCoordinateFinder": [[14, 3, 1, "", "find_coordinates"]], "gpsea.preprocessing.VepFunctionalAnnotator": [[14, 2, 1, "", "NONCODING_EFFECTS"], [14, 3, 1, "", "annotate"], [14, 3, 1, "", "fetch_response"], [14, 3, 1, "", "format_coordinates_for_vep_query"], [14, 3, 1, "", "process_response"]], "gpsea.util": [[15, 5, 1, "", "open_text_io_handle_for_reading"]], "gpsea.view": [[16, 1, 1, "", "CohortVariantViewer"], [16, 1, 1, "", "CohortViewable"], [16, 1, 1, "", "DiseaseViewable"], [16, 1, 1, "", "MtcStatsViewer"], [16, 1, 1, "", "ProteinViewable"], [16, 1, 1, "", "ProteinVisualizable"], [16, 1, 1, "", "ProteinVisualizer"], [16, 1, 1, "", "VariantFormatter"], [16, 1, 1, "", "VariantTranscriptVisualizer"], [16, 5, 1, "", "summarize_hpo_analysis"]], "gpsea.view.CohortVariantViewer": [[16, 3, 1, "", "process"]], "gpsea.view.CohortViewable": [[16, 3, 1, "", "process"]], "gpsea.view.DiseaseViewable": [[16, 3, 1, "", "process"]], "gpsea.view.MtcStatsViewer": [[16, 3, 1, "", "process"]], "gpsea.view.ProteinViewable": [[16, 3, 1, "", "process"]], "gpsea.view.ProteinVisualizable": [[16, 4, 1, "", "marker_counts"], [16, 4, 1, "", "protein_feature_ends"], [16, 4, 1, "", "protein_feature_names"], [16, 4, 1, "", "protein_feature_starts"], [16, 4, 1, "", "protein_feature_types"], [16, 4, 1, "", "protein_id"], [16, 4, 1, "", "protein_length"], [16, 4, 1, "", "protein_metadata"], [16, 4, 1, "", "transcript_coordinates"], [16, 4, 1, "", "transcript_id"], [16, 4, 1, "", "variant_effects"], [16, 4, 1, "", "variant_locations"], [16, 4, 1, "", "variant_locations_counted_absolute"]], "gpsea.view.ProteinVisualizer": [[16, 3, 1, "", "draw_fig"], [16, 3, 1, "", "draw_protein_diagram"]], "gpsea.view.VariantFormatter": [[16, 3, 1, "", "format_as_string"]], "gpsea.view.VariantTranscriptVisualizer": [[16, 3, 1, "", "draw_variants"]]}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "attribute", "Python attribute"], "3": ["py", "method", "Python method"], "4": ["py", "property", "Python property"], "5": ["py", "function", "Python function"], "6": ["py", "data", "Python data"]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:attribute", "3": "py:method", "4": "py:property", "5": "py:function", "6": "py:data"}, "terms": {"": [4, 6, 7, 9, 11, 12, 13, 14, 19, 20, 24, 26, 27, 37, 38], "0": [2, 3, 5, 6, 8, 12, 13, 14, 16, 20, 24, 26, 27, 29, 33, 35, 37, 38], "000": 6, "0000002": 23, "0000047": 29, "0000078": 23, "0000079": 23, "0000080": 23, "0000098": 29, "0000118": [2, 23, 27], "0000119": 23, "0000152": 23, "0000223": 23, "0000234": 23, "0000252": 29, "0000256": 29, "0000271": 23, "0000316": 29, "0000356": 23, "0000359": 23, "0000370": 23, "0000377": 29, "0000407": 38, "0000464": 23, "0000478": 23, "0000486": 38, "0000496": 23, "0000504": 23, "0000517": 33, "0000539": 23, "0000553": 23, "0000562": 20, "0000591": 23, "0000598": 23, "0000707": [23, 27], "0000759": 23, "0000769": 23, "0000777": 23, "0000818": 23, "0000820": 23, "0000828": 23, "0000834": 23, "0000847": 23, "0000864": 23, "0000924": 23, "0000927": 23, "0000929": 23, "0000951": 23, "0000971": 23, "0001060": 12, "0001115": 27, "0001155": 23, "0001166": [3, 7], "0001191": 20, "0001194": 23, "0001197": 23, "0001199": [20, 38], "0001249": 29, "0001250": 12, "0001256": 29, "0001263": 29, "0001311": 23, "0001392": 23, "0001437": 23, "0001438": 23, "0001446": 23, "0001454": 23, "0001507": 23, "0001511": 29, "0001518": 29, "0001537": 6, "0001551": 23, "0001560": 23, "0001566": 12, "0001567": 12, "0001574": [12, 23], "0001575": 12, "0001578": 12, "0001580": 12, "0001583": 12, "0001587": 12, "0001589": 12, "0001595": 23, "0001597": 23, "0001608": 23, "0001619": 12, "0001620": 12, "0001621": 12, "0001623": 12, "0001624": 12, "0001626": [12, 23], "0001627": [12, 23, 29, 38], "0001628": 12, "0001629": [20, 38], "0001630": 12, "0001631": [12, 20, 38], "0001632": 12, "0001643": [20, 38], "0001678": [20, 38], "0001684": [20, 38], "0001732": 23, "0001743": 23, "0001760": 23, "0001782": 12, "0001787": [12, 23], "0001792": 12, "0001818": 12, "0001819": 12, "0001821": 12, "0001822": 12, "0001871": 23, "0001872": 23, "0001877": 23, "0001881": 23, "0001889": 12, "0001891": 12, "0001892": [12, 23], "0001893": 12, "0001894": 12, "0001895": 12, "0001906": 12, "0001907": 12, "0001928": 23, "0001939": 23, "0001965": 23, "0001977": 23, "0002012": [12, 23], "0002019": 12, "0002086": 23, "0002092": [20, 38], "0002169": 12, "0002170": 12, "0002187": 29, "0002270": 23, "0002339": 27, "0002342": 29, "00024192670136836825": 38, "000242": 38, "0002585": 23, "0002597": 23, "0002664": 23, "0002715": 23, "0002733": 23, "0002793": 23, "0002795": 23, "0002813": 23, "0002814": 23, "0002817": 23, "0002916": 23, "0002926": 23, "0002973": 23, "0002981": 23, "0002984": [20, 38], "0003011": 23, "0003022": [20, 38], "0003110": 23, "0003111": 23, "0003117": 23, "0003220": 23, "0003254": 23, "0003256": 23, "0003498": 29, "0003549": 23, "0003839": 23, "0003974": [20, 38], "0004298": 23, "0004312": 23, "0004322": 29, "0004323": 23, "0004328": 23, "0004329": 23, "0004332": 23, "0004337": 23, "00043478260869565214": 20, "0004352": 23, "0004353": 23, "0004356": 23, "0004360": 23, "0004361": 23, "0004370": 23, "0004371": 23, "0004378": 23, "0004404": 23, "0004408": 23, "0004930": 23, "0005": 2, "0005150": [20, 38], "0005368": 23, "0005561": 23, "0005792": [20, 38], "0005922": 29, "0006476": 23, "0006490": 23, "0006500": 23, "0006529": 23, "0006707": 23, "0006889": 29, "0007686": 23, "0008047": 23, "000899": 20, "0008990549794102921": 20, "0009124": 23, "0009127": 23, "0009777": [20, 38], "0009778": [20, 38], "0009809": 23, "0009810": 23, "0010696": 27, "0010864": 29, "0010876": 23, "0010881": 23, "0010932": 23, "0010935": 23, "0010936": 23, "0010938": 29, "0010948": 23, "0010968": 23, "0010969": 23, "0010974": 23, "0010977": 23, "0010978": 23, "0010987": 23, "0010991": 23, "0011014": 23, "0011017": 23, "0011018": 23, "0011019": 23, "0011024": 23, "0011025": 23, "0011028": 23, "0011032": 23, "0011043": 23, "0011111": 23, "0011121": 23, "0011122": 23, "0011138": 23, "0011277": 23, "0011342": 29, "0011343": 29, "0011344": 29, "0011409": 23, "0011442": 23, "0011446": 23, "0011620": 23, "0011623": [20, 38], "0011730": 23, "0011732": 23, "0011733": 23, "0011766": 23, "0011767": 23, "0011772": 23, "0011804": 23, "0011805": 23, "0011842": 23, "0011843": 23, "0011869": 23, "0011875": 23, "0011884": 23, "0011893": 23, "0011902": 23, "0012029": 23, "0012093": 23, "0012099": 23, "0012103": 23, "0012111": 23, "0012129": 23, "0012130": 23, "0012131": 23, "0012135": 23, "0012143": 23, "0012145": 23, "0012210": 38, "0012243": 23, "0012252": 23, "0012261": 23, "0012285": 23, "0012337": 23, "0012338": 23, "0012345": 23, "0012372": [23, 38], "0012373": 23, "0012379": 23, "0012415": 23, "0012443": [8, 38], "0012447": 23, "0012503": 23, "0012535": 23, "0012591": 23, "0012614": 23, "0012632": 23, "0012638": 23, "0012639": 23, "0012640": 23, "0012647": 23, "0012654": 23, "0012680": 23, "0012681": 23, "0012688": 23, "0012700": 23, "0012718": 23, "0012757": 23, "0012769": 23, "0012772": 23, "0015384615384615387": 38, "0020047": 23, "0020054": 23, "0020058": 23, "0020061": 23, "0020155": 23, "0020169": 23, "0025015": 23, "0025021": 23, "0025031": 23, "0025032": 23, "0025033": 23, "0025065": 23, "0025142": 23, "0025155": 23, "0025276": 23, "0025354": 23, "0025427": 23, "0025429": 23, "0025443": 23, "0025454": 23, "0025456": 23, "0025461": 23, "0025463": 23, "0025546": 23, "0025590": 23, "0025640": 23, "0025668": 23, "0025669": 23, "0025688": 23, "0030085": 23, "0030163": 23, "0030272": 23, "0030338": 23, "0030352": 23, "0030453": 23, "0030680": 23, "0030684": 23, "0030687": 23, "0030800": 23, "0030829": 23, "0030860": 23, "0030872": 23, "0030875": 23, "0030878": 23, "0030956": 23, "0030972": 23, "0031071": 23, "0031072": 23, "0031073": 23, "0031093": 23, "0031094": 23, "0031097": 23, "0031099": 23, "0031101": 23, "0031285": 23, "0031331": 23, "0031340": 23, "0031377": 23, "0031383": 23, "0031389": 23, "0031409": 23, "0031411": 23, "0031416": 23, "0031427": 23, "0031476": 23, "0031508": 23, "0031546": [20, 38], "0031550": 23, "0031602": 23, "0031653": 23, "0031657": 23, "0031685": 23, "0031703": 23, "0031704": 23, "0031818": 23, "0031850": 23, "0031871": 23, "0031884": 23, "0031910": 23, "0031982": 27, "0032120": 23, "0032180": 23, "0032226": 23, "0032243": 23, "0032245": 23, "0032251": 23, "0032314": 23, "0032367": 23, "0032481": 23, "0032485": 23, "0032488": 23, "0032943": 23, "0033012": 23, "0033013": 23, "0033072": 23, "0033127": 23, "0033170": 23, "0033334": 23, "0033335": 23, "0033354": 23, "0033358": 23, "0033796": 23, "0033799": 23, "0034190": 23, "0034200": 23, "0034251": 23, "0034263": 23, "0034370": 23, "0034430": 23, "0034442": 23, "0034482": 23, "0034552": 23, "0034644": 23, "0034684": 23, "0034698": 23, "0034737": 23, "003478260869565217": 20, "0034858": 23, "0034899": 23, "0034915": 23, "0034916": 23, "0034977": 23, "0036231884057971015": 20, "003713671516019927": 38, "00387": 38, "003870827221893892": 38, "0040064": 23, "0040068": 23, "0040069": 23, "0040070": 23, "0040084": 23, "0040085": 23, "0040127": 23, "0040172": 23, "0040203": 23, "0040207": 23, "0040214": 23, "0040224": 23, "0040231": 23, "0045026": 23, "0045027": 23, "0045081": 23, "005628510156750059": 20, "01": [20, 26, 27, 33, 37, 38], "0100016": 23, "0100022": 23, "0100491": 23, "0100508": 23, "0100530": 23, "0100536": 23, "0100685": 23, "0100705": 23, "0100763": 23, "0100765": 23, "0100766": 23, "0100767": 23, "0100886": 23, "0100887": 23, "012074957610483744": 38, "01230769230769231": 38, "01349527665317139": 20, "01932367149758454": 20, "01980624808543961": 38, "022514040627000236": 20, "02560162393963452": 20, "03252032520325203": 38, "04": 38, "0410008": 23, "0410009": 23, "0410014": 23, "0430071": 23, "04318488529014845": 20, "05": [3, 20, 27, 38], "0500012": 23, "0500015": 23, "0500016": 23, "0500017": 23, "0500018": 23, "0500019": 23, "0500020": 23, "0500114": 23, "0500117": 23, "0500166": 23, "0500183": 23, "0500238": 23, "06": 38, "06544319142266644": 38, "06827099717235872": 20, "06932119159387057": 38, "07": [20, 26, 27, 33, 37, 38], "07440639019586388": 20, "08259109311740892": 38, "08470708701170182": 38, "1": [2, 3, 5, 6, 8, 12, 13, 14, 20, 23, 24, 26, 27, 29, 33, 35, 37, 38], "10": [2, 6, 8, 12, 14, 16, 20, 27, 29, 38], "100": [20, 27, 38], "1000": 12, "1000029": [6, 12, 37], "1001": [12, 24], "1003": 24, "100del": 20, "100dup": 20, "101": 12, "102": 20, "1024del": 20, "10406504065040649": 38, "106_107insc": 20, "11": [12, 13, 20, 27], "116": 20, "12": [20, 26], "120": [12, 26, 38], "1221c": [20, 26], "123": 8, "129600": 30, "12_114355723_114355723_g_a": 20, "12_114355755_114355756_tg_t": 20, "12_114355784_114355785_ca_c": 20, "12_114356064_114356065_ta_t": 20, "12_114366207_114366208_gc_g": 20, "12_114366241_114366242_ct_c": 20, "12_114366267_114366267_c_a": 20, "12_114366274_114366274_g_t": 20, "12_114366312_114366312_g_a": 20, "12_114366348_114366349_ct_c": 20, "12_114366360_114366360_c_t": 20, "12_114366366_114366366_t_a": 20, "12_114385474_114385474_a_g": 20, "12_114385475_114385475_c_t": 20, "12_114385521_114385521_c_g": 20, "12_114385521_114385521_c_t": 20, "12_114385522_114385522_g_a": 20, "12_114385550_114385550_a_aattattctcag": 20, "12_114385553_114385553_c_a": 20, "12_114385563_114385563_g_a": 20, "12_114394743_114394746_tgtg_t": 20, "12_114394762_114394763_ca_c": 20, "12_114394817_114394817_g_c": 20, "12_114394820_114394820_c_g": 20, "12_114398568_114398568_c_a": 20, "12_114398578_114398579_ca_c": 20, "12_114398602_114398602_t_g": 20, "12_114398626_114398627_cg_c": 20, "12_114398632_114398632_g_a": 20, "12_114398656_114398656_c_cg": 20, "12_114398666_114398667_tg_t": 20, "12_114398675_114398675_g_t": 20, "12_114398682_114398682_c_cg": 20, "12_114398708_114398709_gc_g": 20, "12_114399514_114399514_a_c": 20, "12_114399559_114399559_t_c": 20, "12_114399594_114399594_a_c": 20, "12_114399613_114399613_t_a": 20, "12_114399622_114399622_g_t": 20, "12_114399625_114399629_acatc_a": 20, "12_114399633_114399633_c_g": 20, "12_114401827_114401827_t_a": 20, "12_114401830_114401830_c_t": 20, "12_114401846_114401846_c_g": 20, "12_114401853_114401853_g_t": 20, "12_114401873_114401874_ta_t": 20, "12_114401907_114401907_a_g": 20, "12_114401921_114401921_c_g": 20, "12_114403754_114403754_g_t": 20, "12_114403792_114403792_c_cg": 20, "12_114403798_114403798_g_gc": 20, "12_114403798_114403799_gc_g": 20, "12_114403859_114403859_g_t": 20, "13": [2, 3, 20, 38], "130": 38, "1304del": 20, "1333del": 20, "1366c": 20, "14": [20, 38], "1424522583078588": 20, "142900": [20, 37], "1435th": 37, "145c": 20, "148": 20, "15": [13, 20, 38], "15059037690969213": 38, "154700": 30, "1554": 26, "156": [20, 26, 33, 38], "16": [20, 38], "161t": 20, "1687456462342971": 20, "17": [20, 38], "17007174901911745": 20, "18": [20, 26, 33, 37, 38], "18869165": 26, "18869682": 26, "18921399": 26, "19": [20, 26, 37, 38], "194del": 20, "1_8358231_8358231_t_c": 37, "1d": 16, "1g": 20, "1st": [6, 12, 26], "2": [2, 3, 4, 6, 7, 8, 9, 12, 13, 14, 20, 23, 24, 26, 27, 29, 35, 37, 38], "20": [6, 12, 20, 27, 37, 38], "2011": 29, "2015": 18, "2018": [26, 38], "2022": [21, 29], "205": 12, "206": 12, "21": 38, "215c": 20, "22": [2, 20, 38], "222g": 20, "223": 12, "224": 12, "22_10001_20000_inv": [6, 12], "23": [23, 38], "238g": 20, "24": [20, 38], "241a": 20, "243": 20, "244": [20, 38], "246_249del": 20, "25": [20, 38], "250": 26, "253a": 14, "253c": 20, "260": [20, 33, 38], "262a": 20, "263": 26, "27": [20, 38], "27bp": 12, "281t": 20, "2849045166157176": 20, "299": 26, "29999225997208373": 20, "2d": 2, "2nd": 6, "2t": 20, "2x2": [4, 20, 38], "2x3": 4, "3": [2, 4, 8, 12, 14, 20, 23, 24, 27, 35, 38], "30": [14, 20, 38], "3000013": 23, "3000036": 23, "3000050": 23, "30552424": 20, "31": [20, 38], "316a": 20, "32": [20, 38], "320": 26, "321681": 12, "321682": 12, "321887": 12, "34": 38, "346": 26, "348081479150901e": 38, "348081479150902e": 38, "35": 20, "356": 26, "36": [20, 27, 38], "3603": [12, 14], "361t": 20, "36303223": 21, "37": [20, 38], "374del": 20, "38": [20, 38], "38336121": 20, "39": 20, "4": [6, 8, 12, 14, 20, 26, 38], "40": [20, 38], "4000056": 23, "4000183": 23, "400dup": 20, "408c": 20, "40c": 20, "41": [20, 38], "416del": 20, "42": [16, 20, 38], "426dup": 20, "42857142857142855": 20, "43": [20, 38], "4304a": 37, "432292015291845e": 38, "44": [20, 38], "45": [20, 38], "450": 27, "451c": 20, "456del": 20, "46": 26, "4666666666666667": 38, "48": 20, "481a": 20, "48700997145537483": 38, "5": [2, 3, 6, 12, 14, 20, 26, 29, 32, 35, 37, 38], "50": [6, 12, 20, 27, 37, 38], "504del": 20, "50bp": 6, "51": 20, "510": 20, "518": 26, "52": [20, 38], "53": [20, 38], "55": 38, "57": 38, "5714285714285713": 20, "578": 12, "584g": 20, "587c": 20, "59": [20, 38], "5g": 20, "5th": [12, 20], "5x": 38, "6": [12, 20, 21, 24, 27, 37, 38], "60": [20, 38], "6000062": 23, "6000231": 23, "6000489": 23, "6000673": 23, "6190936213143254e": 20, "62": 20, "63": 38, "64": 27, "641del": 20, "65": 38, "658_660del": 20, "668c": 20, "67": 20, "678g": 20, "680_681insctgagaataat": 20, "6857142857142857": 20, "69": 38, "7": [2, 3, 6, 12, 20, 27, 38], "708378140298727": 38, "709c": 20, "71": [20, 38], "710g": 20, "72": 20, "75": 38, "755": 20, "7703831604944444": 38, "7735491022101784": 20, "781a": 20, "787g": 20, "798del": 20, "8": [14, 15, 20, 26, 27, 38], "831168831168831": 20, "833333333333336": 38, "835c": 20, "84": 38, "85": 20, "873c": 20, "880g": 20, "9": [20, 26, 38], "90": 27, "905del": 20, "939del": 20, "95": [20, 27], "97": 38, "9876g": 12, "99": [27, 38], "A": [2, 3, 5, 6, 7, 8, 12, 14, 16, 18, 20, 24, 26, 27, 28, 31], "AND": [6, 37, 38], "AT": 24, "As": [5, 7, 28, 37], "By": [16, 20, 27, 33, 35], "For": [2, 4, 8, 11, 12, 13, 14, 20, 21, 24, 26, 27, 33, 35, 37, 38], "IN": 14, "INS": 12, "If": [2, 3, 6, 10, 12, 14, 15, 20, 27, 29], "In": [2, 4, 6, 8, 14, 18, 20, 21, 25, 26, 27, 28, 29, 33, 38], "It": [3, 4, 20, 27, 30, 34], "NO": [5, 7], "NOT": [2, 12, 13, 37], "No": [3, 5, 13, 20, 26, 33, 38], "Not": 12, "OR": [6, 37, 38], "On": [3, 7, 26], "One": [29, 37], "That": 19, "The": [2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 16, 18, 19, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 37, 38], "Then": [12, 19, 20, 26, 35], "There": [12, 20, 35, 38], "These": 12, "To": [14, 19, 20, 26, 27, 35, 37, 38], "_": [20, 26, 33, 37, 38], "__eq__": 6, "__hash__": 6, "__init__": [12, 27], "__repr__": 6, "__str__": 6, "_api": 2, "_io": 14, "_missing_implies_phenotype_exclud": 7, "_t": 12, "_term_id": 2, "a_pred": [6, 20], "aa": 6, "aat": 24, "ab": 6, "abbrevi": [16, 27], "abdomen": 23, "abdomin": 23, "abnorm": [8, 18, 20, 23, 24, 33, 38], "about": [2, 12, 14, 16, 19, 20, 22, 25, 26, 27, 29, 31, 33], "abov": [6, 19, 20, 23, 26, 27, 29, 38], "absenc": [3, 7, 20, 33], "absent": [20, 38], "absolut": 6, "abstract": [2, 3, 4, 5, 6, 7, 9, 12, 13, 14], "accept": [12, 14], "acceptor": 20, "access": [6, 12, 13, 14, 19, 20, 26, 27, 33, 38], "accommod": 23, "accord": [2, 3, 20, 23, 38], "account": 27, "acetylaspart": 23, "acid": [12, 23, 26], "across": [2, 16, 34], "actg": 24, "activ": 23, "ad": [14, 29], "adapt": [8, 23], "add": [14, 19, 31], "add_error": 14, "add_issu": 14, "add_subsect": 14, "add_warn": 14, "addit": [4, 19, 20], "addition": 20, "address": 3, "adenosin": 23, "adiponectin": 23, "adipos": 23, "adjac": 13, "adjust": 29, "adnexa": 23, "adren": 23, "adrenocorticotropin": 23, "advanc": 2, "advantag": [20, 27, 33, 38], "affect": [6, 12, 20, 21, 24, 26, 37], "affected_exon": 12, "after": [3, 14, 19], "ag": [12, 29], "against": 21, "al": [8, 18, 21, 26, 29, 38], "ala143argfster40": 20, "ala34glyfster27": 20, "ala34profster32": 20, "albeit": 14, "albumin": 23, "aldosteron": 23, "align": 12, "all": [2, 3, 5, 6, 7, 8, 12, 14, 19, 23, 24, 26, 28, 31, 32, 35, 36, 37, 38], "all_count": 3, "all_diseas": 12, "all_pati": 12, "all_phenotyp": 12, "all_transcript_id": 12, "all_vari": 12, "allel": [6, 12, 18, 21, 31, 32, 35, 37, 38], "allele_count": 6, "allelecount": [5, 6], "allianc": 18, "allow": [3, 25, 26, 27, 28, 32], "allow_nan": 11, "alon": 27, "along": [11, 26, 31, 37], "alpha": [23, 27, 38], "alreadi": 26, "also": [12, 20, 23, 26, 30, 31, 33], "alt": [12, 24], "altern": [6, 12, 24, 27, 35, 38], "alwai": [6, 13, 26], "amaz": 26, "amino": [12, 23], "aminoacid": [12, 13, 16, 26], "amniot": 23, "among": [2, 12], "amount": [26, 37], "amyloid": 23, "an": [2, 3, 4, 5, 6, 7, 8, 9, 12, 13, 14, 15, 16, 18, 19, 20, 21, 25, 26, 27, 28, 30, 31, 33, 35, 37, 38], "analys": [26, 28], "analysi": [0, 14, 16, 18, 21, 25, 26, 28, 30, 32, 33, 34, 35, 37], "analyz": [0, 8, 16, 20, 28, 38], "anatomi": [20, 26], "ancestor": [7, 20, 24, 33], "andolfo": 26, "aneurysm": 24, "angiotensin": 23, "ani": [2, 5, 6, 8, 11, 12, 14, 27, 29, 30, 37, 38], "ank": [6, 12], "ankyrin": 12, "annot": [2, 6, 7, 12, 14, 16, 20, 24, 26, 27, 33, 37, 38], "anomali": [29, 38], "anoth": [7, 13, 20], "answer": [5, 6, 7], "anterior": [23, 33], "antimullerian": 23, "anu": 23, "api": [12, 14, 16, 18, 28, 37, 38], "apoptosi": 23, "appear": 23, "appendix": 35, "appli": [3, 9, 24, 27], "applic": 20, "apply_predicates_on_pati": [1, 3], "approach": [25, 27], "appropri": [20, 38], "ar": [2, 3, 6, 8, 12, 13, 14, 16, 18, 19, 20, 21, 23, 26, 28, 29, 32, 33, 35, 37, 38], "arachnodactyli": 7, "arbitrari": [11, 13], "areolar": 23, "arg": 11, "arg134profster49": 20, "arg237gln": 20, "arg237pro": 20, "arg237trp": 20, "arg279ter": 20, "arg81trp": 20, "arginin": 37, "argument": [8, 10, 14, 16, 27, 33], "arithmet": 13, "arm": 23, "aromatas": 23, "around": [4, 9, 12], "arrai": [3, 4, 16], "arrang": 23, "arrhythmia": 20, "arriv": 27, "arteri": [20, 38], "arteriosu": [20, 38], "ask": [2, 18], "assembl": 14, "assembli": 13, "assert": [12, 13], "assess": [3, 12, 20, 29], "assign": [3, 5, 6, 7, 8, 12, 28, 29, 30, 31, 32, 37, 38], "assist": 25, "associ": [3, 8, 14, 18, 20, 21, 26, 28], "assum": [12, 13, 14, 27], "assume_karyotypic_sex": 14, "assumpt": [27, 33], "atrial": [20, 23, 38], "atrioventricular": [20, 38], "atrophi": 38, "atrophin": 38, "attach": 12, "attribut": [3, 6, 7, 12, 18], "audit": 14, "auditor": [0, 14], "author": 38, "automat": [30, 38], "autonom": 23, "autosom": [6, 20, 25, 35, 38], "autosomal_domin": [5, 6, 35, 38], "autosomal_recess": [5, 6, 35], "avail": [2, 12, 14, 18, 20, 26, 28, 37, 38], "avoid": [16, 23], "ax": [8, 16, 20, 26, 38], "axi": [6, 23, 31, 38], "b": [6, 12, 13, 27, 35], "b_predic": [6, 20], "back": [10, 14], "background": 18, "backup": 12, "bandwidth": 26, "base": [2, 3, 4, 5, 6, 7, 8, 9, 11, 12, 13, 14, 15, 16, 20, 23, 24, 26, 28, 29, 30, 34, 35, 38], "basemultiphenotypeanalysisresult": 3, "basic": [12, 13, 26, 35, 37], "batteri": 7, "bb": 6, "becaus": [2, 12, 20, 21, 23, 24, 27, 29, 34, 38], "been": [12, 20, 26, 27, 38], "befor": [27, 37], "behavior": 18, "behind": [20, 33], "being": [2, 13, 14, 16, 20, 27, 28, 29, 37], "belong": [2, 5, 12, 35], "below": [6, 24], "benjamini": [20, 27, 38], "besid": 26, "best": 18, "beta": [23, 26], "better": 37, "between": [3, 8, 12, 13, 16, 18, 20, 24, 27, 29, 32, 34, 35, 38], "beyond": 6, "bia": [12, 26], "biallel": [6, 35], "biallelic_alt": 35, "biallelic_pred": [5, 6], "bin": [3, 5, 6, 20, 27, 30, 38], "bind": 12, "biolog": [12, 13, 26, 38], "biologi": 18, "biomark": 23, "biomed": [20, 26], "biopterin": 23, "birth": [12, 23], "bit": 33, "bleed": 23, "block": [20, 38], "blood": 23, "bodi": 23, "boiler": 26, "boilerpl": 26, "bone": 23, "bonferroni": [3, 20, 27], "bool": [2, 6, 7, 12, 13, 14, 16], "boolean": 12, "borderlin": 29, "both": [4, 12, 14, 20, 27, 28, 29, 30, 35], "bound": 13, "boundari": [12, 13, 14], "box": [8, 26, 38], "bp": [6, 12, 24, 37], "brain": [8, 38], "branch": [19, 27], "break": 12, "breakend": 12, "breakpoint": [6, 12, 14], "breast": 23, "breath": 23, "brief": 18, "briefli": [37, 38], "bring": [20, 23], "bronchu": 23, "buccal": 23, "bug": 18, "build": [6, 12, 13, 14, 20, 32, 38], "bunch": [26, 37], "burden": [2, 3, 20, 27], "c": [5, 12, 14, 18, 20, 24, 26, 37, 38], "c17998": 12, "c46112": 12, "c46113": 12, "cach": [10, 14, 26], "cache_dir": 14, "cache_env": [0, 10], "calcium": 23, "calcul": [4, 8, 9, 12, 13, 38], "calf": 23, "call": [11, 12, 14, 23, 26, 27, 35], "callabl": [8, 14], "caller": 14, "can": [2, 3, 4, 5, 6, 7, 8, 12, 13, 14, 16, 19, 20, 22, 23, 26, 27, 29, 31, 32, 33, 35, 37, 38], "cannot": [5, 6, 8, 12, 14, 26, 28, 29, 38], "canon": 12, "cap": 27, "capabl": 28, "carboxyl": 23, "cardiac": [20, 23, 38], "cardiomyocyt": 23, "cardiovascular": 23, "carpal": 20, "carri": [21, 26, 38], "cart": 23, "cascad": 23, "case": [2, 6, 8, 12, 14, 20, 26, 33, 38], "cat_id": [5, 8, 38], "cataract": 27, "catecholamin": 23, "categor": [1, 2, 5, 6, 7, 31], "categori": [3, 5, 6, 7, 8, 12, 20, 26, 29, 35, 38], "catheter": 23, "caudat": 27, "caus": [18, 33], "caviti": 23, "ccc": 24, "cct": 12, "cd": [12, 19], "cdna": 20, "cds_end": 12, "cds_start": 12, "cell": 23, "cellular": 23, "central": 23, "cerebellar": 38, "certain": [6, 12, 14, 31, 37], "certainli": 26, "cg": 24, "chanc": [20, 27, 38], "chang": [6, 12, 26, 35, 37, 38], "change_length": [6, 12, 37, 38], "charact": 12, "character": [12, 20], "characterist": 12, "check": [3, 4, 6, 12, 13, 14, 20, 27, 37], "check_circular": 11, "checkout": 19, "child": [2, 20, 38], "children": [14, 23, 27], "choic": [12, 25], "choos": [6, 14, 20, 38], "chosen": [12, 20, 26, 27, 38], "chr1": [12, 13, 24], "chr5": 12, "chrom": [12, 24], "chromosom": [6, 12, 21, 23, 37], "chromosomal_delet": 37, "chrx": 13, "cilium": 23, "circul": 23, "cl": 11, "class": [2, 3, 4, 5, 6, 7, 8, 9, 11, 12, 13, 14, 16, 20, 26, 27, 37], "clearanc": 23, "clearli": 37, "click": [12, 26], "clinic": [12, 20, 26, 27, 29, 33, 35, 37], "clinvar": [20, 26], "clone": 19, "cluster": 31, "cm000663": 13, "coagul": 23, "code": [2, 12, 14, 19, 20, 26, 38], "coding_sequence_vari": 12, "codon": [12, 18, 26, 37], "cohort": [0, 3, 7, 8, 12, 14, 16, 18, 25, 27, 28, 35, 36, 37], "cohort_cr": [14, 20, 26, 33, 37, 38], "cohort_nam": 20, "cohortcr": [0, 14, 26, 38], "cohortvariantview": [0, 16, 20], "cohortview": [0, 16, 20], "coiled_coil": 12, "collect": [2, 7, 9, 12, 14, 16, 18, 20, 26, 33], "color": 8, "column": [2, 3, 8, 12], "com": 19, "combin": [12, 20, 37, 38], "come": [13, 38], "command": 19, "common": [5, 20, 21], "commonli": [13, 27], "compact": [6, 12], "compar": [2, 4, 12, 21, 32, 34], "compare_genotype_vs_phenotyp": [3, 20, 38], "compare_genotype_vs_phenotype_scor": [8, 38], "comparison": 21, "compat": 6, "complain": 14, "complet": [14, 26, 27], "complex": 27, "compon": [20, 37, 38], "composit": [12, 23, 26], "compositional_bia": 12, "compound": [6, 21, 23, 35, 37], "comprehens": [18, 33], "comput": [3, 6, 8, 9, 14, 20, 38], "compute_pv": [4, 9], "concentr": 23, "concept": [18, 24, 27], "conceptu": 12, "concern": 18, "concis": 12, "conclus": 27, "condens": 23, "condit": [27, 38], "conduct": [20, 38], "config": 0, "configur": [6, 14, 37], "configure_caching_cohort_cr": [0, 14, 20, 26, 33, 37, 38], "configure_caching_patient_cr": [0, 14], "configure_cohort_cr": [0, 14], "configure_default_protein_metadata_servic": [0, 14, 26], "configure_ontology_stor": [20, 26, 27, 33, 37, 38], "configure_patient_cr": [0, 14], "configure_phenopacket_registri": [20, 26, 33, 37, 38], "configure_protein_metadata_servic": [0, 14, 20], "congenit": [20, 38], "connect": 23, "consequ": 12, "conserv": 12, "consid": [4, 6, 12, 13, 14, 23, 29, 35, 38], "consider": [23, 37], "consist": [3, 12, 13, 26, 38], "consortium": 38, "constitut": 23, "construct": [14, 33], "constructor": [2, 12, 27], "consult": [6, 10], "contain": [2, 8, 12, 13, 14, 16, 20, 23], "contains_po": 13, "content": 26, "context": [2, 7, 18, 27, 38], "contig": [12, 13, 26], "contig_by_nam": [12, 13, 24], "contigu": 13, "conting": [2, 3, 4, 38], "contingency_t": 38, "control": [20, 27, 38], "conveni": [7, 14, 26], "convers": 24, "convert": [14, 37], "coordin": [6, 12, 13, 14, 16, 24], "copi": 21, "cord": 23, "corneal": 23, "corpuscular": 23, "correct": [3, 20, 23, 25, 38], "corrected_pv": 3, "correctli": 14, "correl": [0, 2, 3, 18, 20, 23, 25, 26, 27, 37, 38], "correspond": [2, 3, 7, 8, 12, 13, 14, 20, 24, 25, 27, 29, 37, 38], "corrrespond": 2, "cortisol": 23, "could": [3, 11, 38], "count": [2, 3, 4, 6, 7, 8, 12, 14, 16, 20, 22, 23, 26, 29, 31, 32, 35, 38], "count_": 6, "count_statist": [3, 20, 27, 38], "counter": 12, "countingphenotypescor": [1, 8, 38], "counts_fram": 2, "countstatist": [3, 4], "cover": [5, 28], "cranial": 23, "creat": [2, 5, 6, 7, 8, 10, 12, 13, 14, 16, 18, 20, 27, 33, 35, 37], "create_variant_from_scratch": 12, "creator": [14, 37, 38], "criteria": [6, 31, 37], "criterion": 6, "cry": 23, "csf": 23, "csv": [20, 38], "ct": 12, "culprit": 2, "curat": 20, "curi": [6, 8, 12, 14], "current": [10, 12, 13, 14, 16, 19, 26, 34], "custom": 26, "cut": 33, "cycl": 23, "cytologi": 23, "cytometri": 23, "darksalmon": 8, "data": [2, 3, 4, 12, 13, 14, 16, 18, 20, 25, 27, 31, 33, 37, 38], "databas": 14, "datadir": 14, "datafram": [2, 3, 4, 8, 12, 16, 26], "datasanityissu": [0, 14], "dataset": [20, 38], "de": [8, 36], "decid": [2, 12, 23], "decis": 2, "decod": [11, 15], "deem": 23, "deep": 18, "def": [8, 11], "default": [6, 10, 11, 14, 15, 16, 20, 26, 27, 33, 35, 38], "default_cache_path": [0, 10], "default_filt": [2, 20, 27, 38], "defaultimprecisesvfunctionalannot": [0, 14], "defaulttermid": 2, "defect": [20, 24, 38], "defin": [3, 5, 6, 12, 24], "definit": 12, "degre": 29, "dehydr": 26, "del": [6, 12, 24, 37], "delet": [6, 12, 20, 24, 37], "deliveri": 23, "delta": 2, "demand": 37, "demonstr": [20, 37], "denot": [12, 18], "depend": [20, 38], "deposit": 38, "depth": 14, "deriv": 27, "descend": [6, 7, 8, 20, 24, 29, 33, 38], "describ": [8, 12, 14, 19, 22, 27, 33, 37, 38], "descript": [5, 12, 14, 29], "design": [18, 27], "desir": [6, 14, 21, 23, 28, 35], "despit": 20, "detail": [5, 12, 20, 27, 37], "determin": [3, 6, 12, 14, 27, 29, 35, 38], "develop": [19, 23, 27, 29], "deviat": [18, 38], "devriesphenotypescor": [1, 8], "df": [12, 26], "diagnos": [6, 7, 12, 30, 31], "diagnosi": [6, 7, 31], "diagnosis_pred": [5, 6, 30], "diagram": [16, 20, 26], "dialog": 26, "dichotom": 38, "dict": [11, 14], "differ": [2, 6, 8, 12, 13, 20, 24, 25, 26, 27, 28, 29, 32, 34, 35, 38], "differenti": 18, "digest": 23, "dimension": 12, "dingeman": 29, "diploid": 12, "direct": [7, 12, 20, 27], "directli": [7, 20], "directori": [10, 14, 26], "disabl": [27, 29], "discharg": 23, "discoveri": [20, 27, 38], "discret": [3, 4, 5, 14, 28], "diseas": [0, 6, 7, 12, 18, 20, 25, 30, 31, 33, 35], "disease_id_queri": 7, "diseaseanalysi": [1, 3], "diseasepresencepred": [5, 7], "diseaseview": [0, 16], "disord": [20, 26], "displai": [2, 16, 20, 26], "display_quest": [5, 20, 30, 32, 33, 34, 35, 38], "disproportion": 29, "dissect": 38, "distanc": 13, "distance_to": 13, "distribut": [8, 12, 16, 22, 34, 38], "divid": [31, 38], "dna": [13, 23], "do": [2, 6, 8, 13, 14, 16, 19, 20, 26, 27, 32, 33, 37], "doc": [20, 26, 38], "document": [16, 18, 19, 26, 27, 38], "doe": [2, 5, 10, 12, 14, 24], "domain": [2, 6, 12, 23, 26, 27, 37, 38], "domin": [6, 20, 21, 35, 38], "done": [13, 14, 19, 31, 35, 37], "donor": [12, 20], "dopamin": 23, "doubl": [8, 13], "down": [14, 33], "download": [12, 26], "downloaded_uniprot_json": 26, "downstream": [2, 13, 14, 26, 28], "downstream_gene_vari": [12, 14], "dpi": [26, 38], "draw": [8, 16, 26], "draw_fig": [16, 26], "draw_protein_diagram": [16, 20], "draw_vari": 16, "drawn": 16, "driven": 23, "drug": 23, "ductu": [20, 38], "due": [12, 14], "dump": 11, "dup": 12, "duplic": [12, 14, 24], "dure": [12, 14], "e": [2, 3, 6, 7, 8, 12, 13, 14, 18, 20, 23, 24, 26, 27, 30, 33, 38], "each": [2, 3, 7, 8, 12, 14, 16, 18, 20, 21, 25, 27, 28, 29, 38], "ear": [23, 29], "easi": [19, 34], "easiest": 26, "ectopia": [30, 33], "eff": [37, 38], "effect": [6, 12, 14, 20, 22, 26, 37, 38], "either": [2, 6, 7, 8, 12, 13, 14, 16, 26, 29], "elabor": 37, "electrolyt": 23, "electrophysiologi": 23, "element": [12, 13, 14, 16], "els": 11, "embryon": 23, "empti": [12, 13, 14], "enabl": 19, "encod": [7, 11, 12, 14, 15, 20, 26, 37, 38], "encount": [14, 20], "end": [6, 12, 13, 20, 24, 26], "end_on_strand": 13, "endocrin": 23, "energi": 23, "enrich": 27, "ensembl": 12, "enst00000123456": 12, "ensur": 15, "ensure_ascii": 11, "entir": [14, 26], "entiti": [7, 12, 13, 14], "entri": [14, 26, 33], "enum": [12, 13, 14], "enumer": 16, "environ": [10, 19], "enzym": 23, "epiphysi": 23, "equal": [6, 12, 27, 38], "equival": 8, "error": [14, 20, 26, 27, 38], "error_count": 14, "erythrocyt": 23, "erythroid": 23, "erythropoietin": 23, "et": [8, 18, 21, 26, 29, 38], "etc": [12, 14], "evalu": [6, 18, 38], "even": [3, 23, 27], "ever": 2, "exact": [4, 6, 8, 12, 14, 18, 20, 21, 27], "exactli": [6, 8, 27], "exampl": [3, 4, 6, 7, 8, 9, 11, 12, 14, 16, 20, 21, 24, 26, 27, 28, 37], "except": [11, 14, 37], "exclud": [2, 5, 7, 12, 13, 20, 32, 33, 38], "excluded_diseas": 12, "excluded_member_count": 12, "excluded_phenotyp": 12, "exclus": [5, 7, 20, 24, 28], "execut": 38, "exercis": 23, "exhaust": 5, "exibit": 7, "exist": [10, 14, 27], "exon": [6, 12, 26, 37], "exon20": 37, "exon_numb": 6, "exons_effect": 12, "expect": [2, 6, 12, 14, 23, 27, 31, 38], "expenditur": 23, "explan": [2, 36, 38], "explicit": 7, "explicitli": [12, 33], "explor": [25, 33, 38], "exploratori": 25, "expos": 38, "express": [14, 23, 38], "extend": 38, "extern": 29, "extract": [12, 14], "extraocular": 23, "extrem": 27, "ey": [23, 33, 38], "f": [8, 18], "face": 23, "fact": [27, 37], "factor": [23, 26, 27, 31], "factori": 2, "fail": [2, 12], "fall": [8, 10], "fallback": [10, 14], "fals": [2, 3, 6, 7, 11, 12, 13, 14, 20, 27, 38], "famili": [12, 20, 27, 30], "farthest": 38, "fascia": 23, "fashion": 14, "fbn1": [6, 12], "fdr": [27, 38], "fdr_bh": [3, 20, 27, 38], "fdr_by": 27, "fdr_gb": 27, "fdr_tsbh": 27, "fdr_tsbky": 27, "featur": [6, 7, 12, 14, 18, 19, 22, 33, 37, 38], "feature_elong": 12, "feature_id": 6, "feature_trunc": 12, "feature_typ": [6, 12], "featureinfo": [0, 12], "featuretyp": [0, 6, 12], "fecal": 23, "feel": 28, "feenstra": [8, 29], "femal": [12, 34], "fetal": 23, "fetch": [14, 20], "fetch_for_gen": 14, "fetch_respons": 14, "few": [24, 26, 37], "fh": [15, 20, 38], "fiber": 23, "fibrillin": 12, "fibrinolysi": 23, "fibroblast": 23, "fiction": [6, 35], "field": [12, 21], "fig": [20, 26, 38], "figsiz": [20, 26, 38], "figur": [14, 16, 26], "file": [12, 14, 15, 16, 26], "filter": [2, 3, 12, 20, 23, 24, 38], "filter_method_nam": 2, "final": [20, 33, 38], "find": [3, 13, 14, 20, 26, 27, 31, 38], "find_coordin": 14, "finder": 14, "fingertip": 26, "first": [6, 10, 12, 14, 20, 26, 27, 31, 37, 38], "fisher": [4, 18, 20, 21, 27, 36], "fisher_exact": [4, 38], "fisherexacttest": [3, 4, 20, 27, 38], "fit": 12, "five": [35, 38], "five_prime_utr_vari": [12, 14], "flexibl": [37, 38], "flip": 13, "float": [2, 3, 4, 8, 9, 14, 38], "flow": 23, "fluid": 23, "focal": 24, "focu": [18, 20], "fold": 12, "folder": [10, 14, 26], "follow": [3, 6, 8, 12, 19, 20, 23, 24, 26, 27, 28, 29, 35, 38], "foot": 23, "for_sampl": 12, "forearm": 23, "form": [4, 23, 26], "format": [7, 12, 13, 14, 16, 26], "format_as_str": 16, "format_coordinates_for_vep_queri": 14, "formatt": 16, "formul": 26, "forward": 13, "found": [14, 20, 26, 27, 29, 32, 38], "four": 26, "fpath_cache_dir": 14, "fragment": 12, "frame": [2, 3, 4, 12, 16, 26, 37, 38], "frameshift": [32, 37], "frameshift_vari": [12, 20, 32, 37, 38], "free": 28, "frequenc": [2, 16, 20, 38], "frequent": [20, 24], "friendli": [2, 14], "from": [2, 5, 6, 8, 11, 12, 13, 14, 15, 16, 18, 19, 20, 23, 24, 27, 28, 29, 30, 32, 33, 34, 35, 37, 38], "from_curi": 33, "from_feature_fram": [12, 26], "from_map": 12, "from_pati": 12, "from_query_curi": [8, 38], "from_raw_part": [5, 12], "from_str": 12, "from_term": 12, "from_uniprot_json": [12, 26], "from_vcf_liter": [12, 24], "from_vcf_symbol": [12, 24], "fuction": 14, "func": 8, "function": [4, 6, 7, 8, 9, 12, 14, 15, 16, 20, 23, 26, 33, 37, 38], "functional_annot": 14, "functionalannot": [0, 14], "functionalannotationawar": 12, "further": [20, 33, 38], "furthermor": 26, "fwer": 27, "g": [2, 3, 6, 7, 8, 12, 13, 14, 18, 20, 23, 24, 26, 27, 30, 33, 37, 38], "ga": 23, "ga4gh": [18, 20, 38], "gain": 20, "ganglion": 23, "gastrin": 23, "gastrointestin": 23, "gb_acc": 13, "ge": 35, "genbank": 13, "genbank_acc": 13, "gene": [6, 12, 14, 20, 21, 26, 33, 35, 37, 38], "gene_coordinate_servic": 14, "gene_id": 12, "gene_nam": 12, "gene_symbol": 12, "genecoordinateservic": [0, 14], "gener": [2, 3, 5, 7, 12, 14, 16, 18, 20, 25, 28, 38], "general_hpo_term": 2, "genet": [18, 35], "genit": 23, "genitourinari": 23, "genom": [6, 12, 14, 18, 20, 24, 26, 37], "genome_build": [14, 20, 26], "genome_build_id": 13, "genomebuild": [12, 13, 14], "genomebuildidentifi": [12, 13], "genomicinterpret": 14, "genomicregion": [12, 13, 26], "genotyp": [0, 2, 3, 4, 5, 8, 12, 14, 18, 23, 25, 28, 30, 32, 34, 37], "genotype_for_sampl": 12, "genotype_phenotype_scor": [8, 38], "genotypepolypred": [2, 3, 5, 6, 8, 28, 32], "genotypes_have_same_hpo_proport": 2, "gestat": 29, "get": [2, 3, 4, 5, 6, 7, 8, 10, 12, 13, 14, 15, 16, 27], "get_annot": 14, "get_cache_dir_path": [0, 10, 14], "get_categor": [5, 6, 7, 38], "get_categori": 5, "get_category_nam": 5, "get_cds_region": [12, 26], "get_coding_base_count": [12, 26], "get_codon_count": [12, 26], "get_excluded_count": 12, "get_features_variant_overlap": 12, "get_five_prime_utr": [12, 26], "get_maximum_group_observed_hpo_frequ": 2, "get_number_of_observed_hpo_observ": 2, "get_patient_id": 12, "get_quest": [6, 8, 35, 37, 38], "get_question_bas": [5, 6, 7], "get_respons": 14, "get_three_prime_utr": 12, "get_variant_by_kei": [12, 37], "gg": [12, 14], "git": 19, "github": [19, 28], "give": [6, 26], "given": [2, 6, 7, 12, 14, 15, 16, 29, 33], "gland": 23, "glial": 23, "gln151ter": 20, "gln302argfster92": 20, "gln315argfster79": 20, "gln421ter": 12, "gln456ter": 20, "gln49ly": 20, "global": [18, 29], "globe": 23, "glossari": 25, "glu294ter": 20, "glucagon": 23, "glucocorticoid": 23, "glucos": 23, "gly125alafster25": 20, "gly195ala": 20, "gly80arg": 20, "glycolipid": 23, "glycoprotein": 23, "glycosaminoglycan": 23, "glycosyl": 23, "go": 12, "gonadotropin": 23, "good": [20, 26], "gov": 21, "gp": 20, "gpc": 25, "gpsea": [17, 19, 20, 21, 24, 25, 26, 28, 30, 32, 33, 34, 35, 36, 37, 38], "gpsea_cach": [10, 14], "gpsea_cachedir": 10, "gpseajsondecod": [0, 11], "gpseajsonencod": [0, 11], "granulocytopoiet": 23, "graph": [7, 38], "graphawar": 7, "graphic": [16, 38], "grch37": [13, 14, 26], "grch38": [12, 13, 14, 20, 24, 26], "great": 27, "greater": [27, 37], "grid": 38, "group": [2, 3, 4, 5, 6, 7, 8, 9, 20, 21, 26, 28, 29, 30, 31, 37], "group1": 38, "group2": 38, "group_nam": [6, 32, 38], "groups_pred": [5, 6, 32, 38], "growth": [23, 26], "gt": 12, "gt_id_to_nam": 38, "gt_predic": [2, 3, 8, 20, 30, 32, 34, 35, 38], "guid": [18, 20, 26, 38], "h": 27, "h_1": 4, "ha": [2, 3, 5, 6, 7, 8, 12, 13, 14, 20, 23, 26, 29, 32, 38], "had": [2, 20, 38], "hair": 23, "hallmark": 7, "hand": [23, 26, 29], "handl": 15, "happen": 3, "harbor": 38, "has_error": 14, "has_errors_or_warn": 14, "has_sv_info": 12, "has_variant_coordin": 12, "has_warn": 14, "hashabl": 12, "have": [2, 8, 12, 14, 20, 26, 29, 34, 38], "head": [23, 38], "health": 18, "hear": 38, "heart": [20, 23, 29, 38], "height": 23, "helic": 12, "help": 20, "hematocrit": 23, "hemizyg": 12, "hemoglobin": 23, "henc": [14, 19], "hepat": 23, "hepatobiliari": 23, "here": [6, 13, 20, 22, 24, 26, 28, 29, 32, 33, 35, 38], "hereditari": 26, "het": [35, 38], "heterozyg": [6, 12, 31, 35, 38], "heterozygot": 21, "heurist": [2, 20, 23, 27], "hgnc": [12, 14], "hgv": [12, 14, 16], "hgvs_cdna": 12, "hgvs_coordinate_find": 14, "hgvsp": 12, "hierarch": [14, 23], "hierarchi": [2, 33], "high": 12, "higher": [29, 38], "highest": 12, "hip": 23, "his1435arg": 37, "his220del": 20, "his445metfster137": 20, "histidin": 37, "histori": 27, "hmf01": [20, 38], "hmf02": [2, 20, 38], "hmf03": [2, 20, 38], "hmf04": [2, 20, 38], "hmf05": [2, 20, 38], "hmf06": [20, 38], "hmf07": 2, "hmf08": [2, 20, 38], "ho": 27, "hochberg": [20, 27, 38], "holm": 27, "holt": [20, 37], "hom_ref": [35, 38], "homeostasi": 23, "hommel": 27, "homo": 6, "homovanil": 23, "homozyg": [6, 35], "homozygot": 21, "homozygous_altern": 12, "homozygous_refer": 12, "honeydew": 8, "hormon": 23, "how": [13, 19, 20, 22, 26, 32, 33, 37, 38], "howev": [5, 13, 14, 19, 20, 24, 28, 31, 32, 35, 37, 38], "hp": [2, 3, 7, 8, 12, 20, 23, 27, 29, 33, 38], "hpo": [2, 3, 7, 8, 12, 14, 16, 18, 26, 28, 29, 36, 37, 38], "hpo_mtc": 27, "hpomtcfilt": [1, 2, 20, 27, 38], "hpopred": [5, 7, 33], "hpotermanalysi": [1, 3, 20, 27, 33, 38], "hpotermanalysisresult": [1, 3, 16], "hpotk": [2, 20, 26, 27, 33, 37, 38], "html": [16, 20, 38], "http": [19, 21], "human": [2, 5, 12, 13, 14, 16, 18, 26, 29], "humeru": [20, 38], "humor": 23, "hundr": 27, "hypertelor": 29, "hypertens": [20, 38], "hypoplasia": [20, 38], "hypospadia": 29, "hypothalamu": 23, "hypothes": [25, 26, 27, 38], "hypothesi": [2, 20, 27, 38], "i": [0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 18, 19, 20, 21, 23, 24, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 37, 38], "id": [5, 6, 7, 8, 12, 14, 16, 20, 23, 24, 26, 31], "ident": 21, "identifi": [5, 6, 8, 12, 13, 14, 20, 23, 26, 29], "idx": 2, "ignor": 14, "ii": 23, "ile106v": 20, "ile227_glu228inst": 20, "ile54thr": 20, "illustr": [20, 38], "img": [20, 26, 38], "immun": 23, "impact": 6, "impair": 38, "implement": [4, 6, 11, 14, 21, 26, 29, 34, 37, 38], "implementor": 12, "impli": [7, 24, 27, 38], "implicitli": [24, 33], "import": [4, 6, 8, 12, 13, 20, 24, 26, 27, 30, 32, 33, 34, 35, 37, 38], "impos": 4, "imprecis": 12, "imprecise_sv_functional_annot": 14, "imprecisesvfunctionalannot": [0, 14], "imprecisesvinfo": [0, 12, 14], "includ": [2, 3, 4, 5, 6, 7, 8, 12, 13, 14, 16, 18, 20, 25, 26, 31, 32, 33, 35, 38], "include_computational_tx": 14, "include_patients_with_no_hpo": 12, "include_patients_with_no_vari": 12, "include_subsect": 14, "inclus": [6, 31, 37], "incomplete_terminal_codon_vari": 12, "increas": [20, 38], "indel": 12, "indent": [11, 14], "independ": 27, "index": [6, 8, 12, 19, 23], "indic": 12, "indirect": [20, 27, 38], "indirectli": 7, "individu": [2, 3, 6, 7, 8, 12, 18, 20, 21, 24, 26, 27, 28, 29, 30, 31, 32, 33, 37, 38], "infer": 20, "inflammatori": 23, "info": [2, 3, 6, 12, 14, 20, 24, 37, 38], "inform": [12, 16, 18, 20, 25, 26, 27, 28, 29], "infram": 20, "inframe_delet": [12, 20], "inframe_insert": [12, 20], "ingest": 18, "inherit": [6, 27, 31, 38], "inhibin": 23, "initi": [7, 12, 19], "inner": 23, "input": [2, 3, 14, 15, 18, 20, 25, 37, 38], "insert": [12, 14, 20, 24], "insight": [20, 23, 26], "instal": [18, 26], "instanc": [2, 4, 5, 6, 7, 8, 12, 14, 21, 24, 26, 27, 29, 33, 35, 37, 38], "instanti": 26, "instead": [6, 12, 27], "instruct": 18, "insulin": 23, "int": [2, 3, 4, 5, 6, 7, 12, 13, 14, 16], "integ": 12, "integr": 19, "integu": 23, "intellectu": 29, "intend": [8, 12], "inter": [28, 38], "interact": 12, "interest": [6, 7, 12, 14, 20, 23, 27, 38], "intergenic_vari": [12, 14], "internet": 19, "interpret": [14, 15, 31, 38], "intersect": 13, "interspers": 12, "interv": 13, "intestin": 23, "intracrani": 23, "intraocular": 23, "intrauterin": 29, "introduct": 25, "intron": [12, 20], "intron_vari": [12, 14, 20, 37], "inv": 12, "invers": [12, 37], "investig": [7, 12, 20, 34], "invoc": 19, "involv": [12, 26, 27], "io": [0, 12, 14, 15], "iobas": [12, 15], "ion": 23, "ip3": 12, "ipython": [16, 20], "iqr": 38, "is_cod": [12, 26], "is_empti": 13, "is_femal": 12, "is_filtered_out": 2, "is_frameshift": 38, "is_large_imprecise_sv": 6, "is_mal": 12, "is_missens": 35, "is_neg": 13, "is_observ": 12, "is_ok": 14, "is_pass": 2, "is_posit": 13, "is_pref": 12, "is_pres": 12, "is_provid": 12, "is_structur": 12, "is_structural_delet": 6, "is_structural_vari": 6, "is_unknown": 12, "islet": 23, "isoform": [12, 26], "issu": [2, 12, 14, 18, 20, 28], "item": [2, 3, 5, 14, 16, 24], "iter": [2, 3, 5, 6, 7, 8, 11, 12, 13, 14, 16, 27], "iter_cohort_phenopacket": [20, 26, 33, 37, 38], "iterate_nod": 14, "its": [7, 12, 14, 16, 24, 28, 33, 37, 38], "j": 3, "jinja2": 16, "joint": 23, "jordan": 38, "json": [11, 12, 14], "jsondecod": 11, "jsonencod": 11, "jt": 21, "judgment": [23, 27, 35], "jupyt": [16, 18, 20], "just": [12, 14, 16, 20], "keep": [7, 14], "kei": [6, 12, 18, 20], "kidnei": [8, 38], "kind": [14, 21, 25], "know": [12, 13, 14, 26], "known": [12, 33], "kwarg": 11, "label": [6, 12, 14, 16, 23, 26, 30, 32], "label_summari": 12, "labeling_method": 16, "lack": [7, 14, 27], "lactat": 23, "langerhan": 23, "larg": [6, 12, 13, 14, 23], "larger": 12, "last": [10, 12, 20, 27, 38], "latest": 20, "lead": [6, 12, 20, 23, 24, 27, 35, 37, 38], "learn": [20, 22, 38], "least": [2, 6, 14, 27, 28, 32, 37, 38], "leav": 18, "left": 13, "legend": 16, "leigh": 12, "len": [12, 20, 24, 26, 33, 37, 38], "length": [6, 12, 13, 37, 38], "lenient": 14, "lenti": [30, 33], "leptin": 23, "less": [6, 20, 27, 38], "let": [11, 12, 24, 26, 27, 37, 38], "leu435argfster147": 20, "leu65glnfster10": 20, "leu94arg": 20, "leukocyt": 23, "level": [0, 8, 12, 14, 20, 23, 26, 37, 38], "leverag": [2, 26, 29, 33, 37], "li": [13, 21], "librari": [0, 13, 19, 26, 38], "life": 38, "like": [11, 18, 23, 27], "limb": [20, 23], "limit": [2, 27, 29], "line": [6, 12, 19, 38], "lineag": 23, "link": [31, 35], "liposaccharid": 23, "list": [6, 7, 8, 11, 12, 20, 23, 28, 33], "list_all_diseas": 12, "list_all_protein": 12, "list_all_vari": 12, "list_present_phenotyp": 12, "liter": [6, 12, 14, 16, 24], "literatur": [6, 20], "liver": [8, 23], "ll": 26, "load": [11, 14, 33, 37, 38], "load_minimal_hpo": [20, 26, 27, 33, 37, 38], "load_phenopacket": [0, 14, 20, 26, 33, 37, 38], "load_phenopacket_fil": [0, 14, 26], "load_phenopacket_fold": [0, 14, 26], "loader": 26, "local": [14, 15], "locat": [12, 13, 14, 18, 20, 23, 26, 37], "locu": 12, "lof": 38, "lof_effect": 38, "lof_mut": 38, "log": 36, "logic": [27, 29, 37], "loss": [12, 38], "lower": [23, 37, 38], "ly": 38, "lymph": 23, "lymphat": 23, "lymphocyt": 23, "lys226asn": 20, "lys88ter": 20, "lysosom": 23, "m": [4, 19], "macrocephali": 29, "macroscop": 23, "made": [20, 27], "mai": [3, 8, 12, 14, 26, 27, 37, 38], "main": [20, 26, 38], "major": [13, 28], "major_assembli": 13, "make": [2, 6, 27], "malarkei": 14, "male": [12, 34], "malform": 14, "mandatori": 12, "mane": [12, 20, 26, 37, 38], "mane_pt_id": 26, "mane_tx_id": 26, "mani": [20, 22, 23, 25, 26, 27, 28, 33, 37, 38], "manifest": [18, 26], "mann": [8, 9, 29, 36], "mannwhitneystatist": [8, 9, 38], "mannwhitneyu": [9, 38], "manual": 12, "map": [2, 12, 14, 16], "marfan": 30, "marker": [16, 23], "marker_count": 16, "marrow": 23, "mass": 23, "match": [6, 8, 30, 31, 32], "matern": 21, "math": 8, "matplotlib": [16, 20, 26, 38], "matrix": [2, 3], "matter": 14, "matur": 23, "mature_mirna_vari": 12, "maxim": 28, "maximum": [2, 20, 29, 38], "me": 12, "mean": [12, 23, 27, 38], "meaning": [5, 6], "measur": 23, "median": 38, "mediastinum": 23, "medic": [18, 23, 27], "medicin": 18, "meet": [6, 12, 31, 37, 38], "megakaryocyt": 23, "member": [12, 14, 22, 26, 38], "membran": 23, "mendelian": [18, 35], "mental": 23, "mesenteri": 23, "messag": [11, 14], "met": 37, "met74il": 20, "met83phefster6": 20, "meta_label": 12, "metabol": 23, "metabolit": 23, "metadata": [14, 16], "metaphysi": 23, "method": [2, 3, 6, 7, 11, 12, 13, 14, 20, 26, 27, 35, 38], "mhc": 23, "micro": 29, "microcephali": 29, "middl": 23, "midwif": 12, "might": 21, "mild": 29, "mim": [20, 37], "min_n_of_patients_with_term": [7, 20, 38], "miner": 23, "minimalontologi": [2, 7, 8, 14, 16, 33], "minimalterm": 12, "minimum": [2, 7], "mininum": 27, "miss": [8, 14, 20, 27, 28], "missens": [2, 3, 6, 12, 18, 21, 24, 27, 31, 32, 35, 37, 38], "missense_and_exon20": 37, "missense_or_nonsens": 37, "missense_vari": [3, 6, 12, 20, 32, 35, 37, 38], "missing_implies_exclud": 7, "missing_implies_phenotype_exclud": 7, "mitig": [27, 38], "mitochondri": 35, "mitochondrion": 23, "mixin": 13, "mnv": 12, "mode": [6, 14, 27, 31, 38], "mode_of_inheritance_info": 6, "model": [0, 2, 6, 20, 24, 26, 32, 35, 37, 38], "modeofinheritanceinfo": 6, "modeofinheritancepred": [5, 6], "moder": 29, "modifi": 27, "modul": [0, 12, 13, 14, 38], "moi": 35, "moment": 14, "monarch": 19, "mondo": 7, "monoallelic_pred": [5, 6, 20], "more": [2, 3, 6, 7, 8, 9, 12, 14, 19, 20, 23, 25, 26, 29, 30, 32, 37, 38], "morpholog": 23, "morphologi": [8, 18, 20, 23, 24, 27, 29, 33, 38], "most": [12, 13, 20, 23, 26, 27, 28], "mostli": [12, 28], "motif": [12, 18], "motil": 23, "motor": [20, 23], "movement": 23, "mri": 23, "mtc": [2, 3, 16, 20, 23, 25, 38], "mtc_alpha": [3, 20, 38], "mtc_correct": [3, 20, 27, 38], "mtc_filter": [1, 3, 20, 27, 38], "mtc_filter_nam": 3, "mtc_filter_result": 3, "mtc_issu": 2, "mtc_name": 3, "mtc_report": [20, 38], "mtc_viewer": [20, 38], "mtcstatsview": [0, 16, 20, 38], "much": [23, 27, 37], "mucociliari": 23, "mucosa": 23, "mucu": 23, "multi": [12, 37], "multiphenotypeanalysi": [1, 3, 27], "multiphenotypeanalysisresult": [1, 3], "multipl": [2, 3, 20, 23, 25, 37, 38], "multipli": 27, "muscl": 23, "muscular": [20, 38], "musculatur": 23, "musculoskelet": 23, "must": [2, 3, 5, 6, 7, 8, 12, 14, 19, 26, 27, 28, 30, 37], "mutat": [20, 26, 33, 37, 38], "mutlipl": 27, "myelin": 23, "myeloid": 23, "n": [2, 4, 6, 12, 14, 20, 23, 32, 33, 38], "n_categor": 5, "n_usabl": [2, 3], "nacgtacgtac": 12, "nail": 23, "name": [2, 3, 5, 6, 10, 12, 13, 14, 20, 26, 38], "nan": [3, 8], "nasal": [23, 29], "natur": 12, "nc_000001": 13, "ncbi": [20, 21], "ncit": 12, "ndarrai": 16, "necessari": 35, "neck": 23, "need": [6, 14, 20, 26, 27, 33, 38], "neg": [6, 13, 14, 24], "neoplasm": 23, "neopterin": 23, "nerv": 23, "nervou": [23, 24, 27], "net": 12, "network": 26, "neuron": 23, "never": [13, 27], "new": 16, "next": [20, 26, 37], "nih": 21, "nippl": 23, "nlm": 21, "nm_001042681": [37, 38], "nm_002834": 14, "nm_005912": 14, "nm_013275": 37, "nm_123": 6, "nm_1234": [32, 35], "nm_123456": [3, 6, 12], "nm_181486": [20, 26, 38], "nmd_transcript_vari": 12, "no_cal": 12, "no_genotype_has_more_than_one_hpo": 2, "node": [14, 23], "nomin": [3, 27], "non": [2, 3, 6, 12, 14, 20, 27, 32, 37, 38], "non_coding_transcript_exon_vari": [12, 14], "non_coding_transcript_vari": [12, 14], "non_frameshift_del": 37, "non_frameshift_effect": 37, "non_frameshift_pred": 37, "non_specified_term": 2, "noncoding_effect": 14, "none": [2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 16, 20, 26, 27, 28, 29, 32, 35, 38], "nonfaci": 29, "nons": 21, "nonsens": [2, 3, 6, 21, 27, 37, 38], "normal": 18, "nose": 29, "notat": [12, 24], "note": [2, 3, 7, 10, 12, 13, 20, 26, 27, 29, 30, 34, 37, 38], "notebook": [16, 18, 20], "notepad": [0, 14], "notepadtre": [0, 14], "noth": 37, "novel": 12, "now": [12, 20, 26, 27, 33, 38], "np_": 20, "np_000129": 12, "np_001027558": 14, "np_001027559": 12, "np_037407": 14, "np_852259": [20, 26], "nuclear": 12, "nucleobas": 23, "nucleotid": [12, 20, 24, 37], "nucleu": 27, "nul": 12, "null": 27, "number": [2, 3, 4, 5, 6, 7, 8, 12, 13, 20, 23, 27, 29, 35, 38], "numer": 5, "o": 11, "obj": 11, "object": [2, 4, 5, 6, 8, 9, 11, 12, 13, 14, 16, 26], "object_hook": 11, "observ": [2, 8, 12, 14, 18, 20, 33, 35, 38], "observablefeatur": 12, "obtain": [12, 14, 20, 26, 27, 29], "occipitofrontali": 23, "occur": [12, 20, 26, 38], "occurr": 20, "oddsratio": 38, "odontoid": 23, "off": [28, 37], "offer": [27, 29, 36, 37, 38], "offlin": 14, "often": 12, "ok": [2, 14], "oldest": 27, "omim": [7, 20, 30], "omit": [6, 20, 28, 30, 34, 38], "onc": 38, "one": [2, 5, 6, 7, 8, 12, 14, 19, 20, 21, 23, 28, 29, 30, 32, 35, 38], "one_genotype_has_zero_hpo_observ": 2, "ones": 14, "onli": [2, 5, 7, 12, 14, 20, 21, 27, 30, 32, 33, 35, 38], "onlin": 19, "only_hgv": 16, "onset": [23, 24], "onto": 16, "ontologi": [2, 7, 12, 18, 20, 24, 26, 29], "oocyt": 23, "open": [15, 20, 26, 28, 38], "open_phenopacket_stor": [20, 26, 33, 37, 38], "open_text_io_handle_for_read": [0, 15], "oper": [6, 37], "oppos": 12, "opposit": 13, "optim": 14, "option": [2, 5, 7, 12, 14, 19, 20, 26, 27, 33], "oram": [20, 37], "order": [2, 3, 12, 20, 26, 38], "organ": [12, 14, 23], "orgogozo": 18, "origin": 29, "orthogon": 31, "osmol": 23, "osteocalcin": 23, "other": [2, 12, 13, 14, 18, 20, 21, 26, 27, 29, 31, 34], "otherwis": [6, 7, 12, 29, 32], "our": [20, 23, 27, 28, 29, 38], "out": [2, 13, 14, 20, 24, 26, 27, 38], "outcom": 3, "outer": 23, "output": 14, "over": [12, 14], "overal": 27, "overlap": [3, 6, 12, 13, 16, 37], "overlapping_exon": 12, "overlaps_with": 13, "overload": 37, "overrepres": 12, "overview": [20, 38], "own": 13, "p": [2, 3, 4, 7, 8, 9, 12, 16, 18, 20, 26, 27, 33, 37, 38], "p13": [13, 14], "p_valu": 38, "packag": [17, 18, 19, 27], "page": [12, 20, 25, 28], "pair": [3, 4, 6, 12, 24], "pancrea": 23, "pancreat": 23, "panda": [2, 12, 26], "param": [2, 6, 12, 13, 14], "paramet": [2, 3, 5, 6, 7, 8, 11, 12, 13, 14, 15, 16], "parameter": 16, "parametr": 38, "parathyroid": 23, "parent": [3, 14, 27, 29], "pars": [12, 14], "parse_multipl": 14, "parse_respons": 14, "parse_uniprot_json": 14, "part": [12, 16, 26], "partit": [6, 28, 31], "pass": [2, 6, 11, 12, 24, 27], "past": 27, "patch": 13, "patent": [20, 38], "patern": 21, "path": [10, 12, 14, 15, 20, 26, 33, 38], "pathogen": [20, 26, 37], "patient": [0, 2, 3, 5, 6, 7, 8, 12, 14, 16, 18, 20, 26, 27, 28, 32, 33, 38], "patient_1": 8, "patient_2": 8, "patient_3": 8, "patient_4": 8, "patient_cr": 14, "patient_id": [8, 12, 38], "patientcategori": [1, 5, 7], "patientcr": [0, 14], "pattern": [14, 23], "payload": 14, "pcat": [1, 20, 27, 38], "pd": [12, 26], "pehnotyp": 2, "peptid": 23, "percent": [20, 38], "perform": [3, 4, 14, 18, 20, 21, 25, 26, 27, 38], "perifollicular": 23, "peripher": 23, "peritoneum": 23, "peroxisom": 23, "ph": 23, "ph_predic": 2, "phagocytosi": 23, "phe168leufster6": 20, "pheno_pred": [3, 20, 33, 38], "pheno_scor": [8, 38], "phenopacket": [12, 14, 18, 20, 33, 37, 38], "phenopacket1": 26, "phenopacket2": 26, "phenopacket_registri": 20, "phenopacketpatientcr": [0, 14], "phenopacketvariantcoordinatefind": [0, 14], "phenotyp": [0, 2, 3, 4, 5, 8, 9, 12, 14, 18, 22, 23, 25, 28, 29, 31, 34, 37], "phenotype_cr": 14, "phenotype_scor": 8, "phenotypecategor": [5, 7], "phenotypecr": [0, 14], "phenotypemtcfilt": [1, 2, 3, 20, 27], "phenotypemtcissu": [1, 2], "phenotypemtcresult": [1, 2, 3], "phenotypepolypred": [2, 3, 5, 7, 28], "phenotypescor": [1, 8, 9], "phenotypescoreanalysi": [1, 8, 38], "phenotypescoreanalysisresult": [1, 8], "phenotypescorestatist": [8, 9], "phenotypic_abnorm": 2, "phosphat": 23, "physician": 12, "physiolog": 23, "physiologi": [18, 23, 24], "pickl": 14, "piec": [12, 20], "piezo1": 26, "pineal": 23, "pinna": 29, "pip": 19, "pituitari": 23, "place": [18, 26, 34], "placent": 23, "placenta": 23, "plate": 26, "platelet": 23, "platysma": 23, "pleas": 28, "plot": [8, 16, 22, 38], "plot_boxplot": [8, 38], "plt": [16, 20, 26, 38], "plu": 6, "pm": [20, 26], "pmid": 20, "pmid_27087320_subject_1": 38, "pmid_27087320_subject_10": 38, "pmid_27087320_subject_2": 38, "pmid_29330883_subject_1": 38, "pmid_29330883_subject_2": 38, "png": [20, 26, 38], "po": [12, 13, 24], "point": [12, 14, 29, 38], "point_mut": 38, "point_mutation_effect": 38, "polar": [26, 27], "polici": [14, 20, 26, 38], "polypred": [1, 5, 6, 7], "pore": 26, "port": 13, "posit": [3, 6, 12, 13, 14, 20, 24, 27], "posixpath": 10, "possibl": [2, 3, 5, 8, 21, 27, 30, 34], "possible_result": 2, "possibli": 31, "posterior": [23, 27], "potenti": [25, 38], "power": [20, 27, 28, 38], "pp": 14, "pp_dir": 26, "pp_directori": 14, "pp_file": [14, 26], "ppktstore": [20, 26, 33, 37, 38], "practic": 21, "pre": 2, "precis": 18, "precursor": 23, "predic": [1, 2, 3, 4, 8, 25, 30, 34, 38], "predict": [6, 12, 20, 26, 37, 38], "predictor": [14, 26], "prefer": 12, "preimplant": 23, "prematur": 18, "prenat": 23, "prepar": [3, 5, 6, 7, 12, 14, 26, 30], "prepare_hpo_terms_of_interest": [5, 7], "prepare_notepad": 14, "prepare_predicates_for_terms_of_interest": [5, 7, 20, 33, 38], "preprocess": [0, 20, 26, 33, 37, 38], "preprocessingvalidationresult": [0, 14], "preproprotein": 12, "presenc": [3, 6, 7, 12, 20, 26, 27, 31, 33, 37, 38], "present": [3, 7, 8, 12, 13, 16, 20, 29, 33, 38], "present_diseas": 12, "present_phenotyp": 12, "present_phenotype_categor": 7, "present_phenotype_categori": 7, "pressur": 23, "pretti": 16, "prevent": [2, 3, 37], "previou": 27, "previous": [37, 38], "primari": [12, 23, 37], "principl": 21, "print": [12, 14, 26], "pro139glnfster11": 20, "pro14thr": 20, "pro85thr": 20, "probabl": [27, 38], "problem": 27, "proce": 14, "procedur": [3, 20, 23, 38], "process": [14, 16, 20, 26, 33, 37, 38], "process_respons": 14, "produc": [2, 4, 5, 6, 7], "product": 16, "profound": 29, "programmat": [6, 14], "project": 12, "prolifer": 23, "prong": [28, 38], "pronounc": 18, "propagatingphenotypepred": 7, "properti": [2, 3, 4, 5, 7, 8, 12, 13, 14, 16], "proport": [2, 20, 38], "protcachingmetadataservic": [0, 14], "protein": [6, 12, 13, 14, 16, 18, 23, 37], "protein_altering_vari": 12, "protein_cach": 14, "protein_effect_coordin": 12, "protein_effect_end": 12, "protein_effect_loc": 12, "protein_effect_start": 12, "protein_featur": [6, 12, 26], "protein_feature_end": 16, "protein_feature_nam": 16, "protein_feature_start": 16, "protein_feature_typ": [6, 16], "protein_id": [12, 14, 16, 26], "protein_length": [12, 16, 26], "protein_meta": [12, 16, 20, 26], "protein_metadata": 16, "protein_metadata_servic": 6, "protein_sourc": 14, "proteinannotationcach": [0, 14], "proteinfeatur": [0, 12], "proteinmetadata": [0, 12, 14, 16, 20, 26], "proteinmetadataservic": [0, 6, 14, 26], "proteinpred": [5, 6, 28], "proteinview": [0, 16], "proteinvisu": [0, 16, 20, 26], "proteinvisualiz": [0, 16, 26], "provid": [2, 3, 6, 7, 8, 12, 13, 14, 16, 18, 20, 25, 27, 28, 29, 33, 34, 35, 38], "proxi": 29, "prune": 14, "pscore": [1, 38], "publish": [19, 33], "pubm": 21, "pulmonari": [20, 23, 38], "pupillari": 23, "purin": 23, "purpos": 26, "put": [12, 20, 38], "putamen": 27, "pval": [3, 8, 38], "pvalu": 38, "pvi": [16, 26], "px_id": 20, "py": 12, "pydoc": 26, "pypi": 19, "pyplot": [20, 26, 38], "pyrimidin": 23, "pytest": 19, "python": [4, 11, 18, 19, 26, 27, 37, 38], "python3": 19, "q": [20, 26, 38], "q1": 38, "q3": 38, "q99593": 26, "qc_result": [26, 38], "qual": [12, 24], "qualnam": [12, 13, 14], "quartil": 38, "queri": [7, 8, 12, 14, 26, 33, 38], "question": [5, 6, 7, 18, 37, 38], "r": 38, "radiu": [20, 38], "rais": [5, 11, 12, 13, 14], "random": [4, 12], "random_se": 16, "rang": [2, 29, 38], "rank": [9, 36, 38], "rare": [0, 2, 20, 26, 33], "rate": [20, 23, 27], "rather": 23, "ratio": 23, "re": 12, "reach": 14, "read": [15, 18, 37], "readabl": [5, 12, 16], "readi": 20, "readili": 26, "real": 38, "realiz": 27, "realli": 26, "reason": [2, 20, 23, 26, 38], "receiv": 20, "recent": 27, "receptor": 26, "recess": [6, 25, 35], "recommend": [2, 6, 12, 18, 20, 26, 27, 33, 38], "record": [14, 33, 35], "red": [23, 38], "redox": 23, "reduc": [2, 20, 26, 27], "redund": 27, "ref": [6, 12, 16, 24, 38], "ref_length": [6, 38], "refer": [6, 12, 18, 23, 26, 29, 35, 38], "reflect": 28, "reflex": 23, "refract": 23, "refseq": [12, 13, 14, 26], "refseq_nam": 13, "regard": [2, 12, 16], "regardless": 12, "region": [6, 12, 13, 16, 20, 24, 26, 37], "regist": 35, "registri": [20, 26, 33, 37, 38], "regul": 23, "regulatory_region_abl": 12, "regulatory_region_amplif": 12, "regulatory_region_vari": 12, "rel": [6, 29], "relat": [14, 27], "relationship": [18, 38], "releas": [20, 23, 26, 27, 33, 37, 38], "relev": [2, 12, 20, 26, 38], "remot": 14, "remov": [6, 14, 23, 24, 37], "renal": 38, "render": 16, "renin": 23, "repair": 23, "repeat": 12, "replac": 37, "report": [3, 8, 14, 16, 18, 20, 33, 38], "repositori": 19, "repres": [2, 3, 5, 7, 8, 12, 13, 14, 16, 18, 26, 27, 38], "represent": [12, 14], "reproduct": 23, "request": 14, "requir": [4, 12, 14, 19, 20, 27, 28], "requisit": 26, "rere": [26, 37, 38], "rere_mane_tx_id": 37, "rere_phenotype_score_boxplot": 38, "residu": 26, "resourc": [6, 12, 14], "respect": [3, 12, 24, 26, 32, 37], "respir": 23, "respiratori": 23, "respons": [14, 23, 26], "rest": [12, 14, 23], "restrict": 21, "result": [2, 3, 8, 12, 14, 16, 18, 20, 23, 26, 27, 28, 33, 37, 38], "retain": 27, "retainin": 20, "reticulocyt": 23, "retriev": [12, 13, 14], "return": [2, 3, 4, 5, 6, 7, 8, 11, 12, 13, 14, 15, 16, 27, 28, 32], "revers": 13, "right": [13, 14, 16], "risk": 38, "rna": 13, "root": [2, 14, 23, 27], "row": [2, 3, 20], "rule": [5, 20, 33, 37, 38], "run": [3, 14, 20, 26, 27, 38], "runner": 14, "runonlin": 19, "rust": 12, "sai": 27, "salivari": 23, "same": [2, 3, 12, 13, 20, 23, 24, 26, 29, 38], "same_count_as_the_only_child": 2, "sampl": [12, 20], "sample_id": 12, "samplelabel": [0, 12], "sanit": 14, "sanitationresult": 14, "saniti": 14, "sapien": 6, "save": 26, "savefig": [20, 26, 38], "scalp": 23, "scenario": 5, "schema": [12, 14, 16, 26, 38], "scienc": 6, "scipi": [4, 9, 38], "scipyfisherexact": 3, "sclera": 23, "score": [8, 9, 36], "score_analysi": 38, "score_statist": [8, 38], "scorer": 8, "screenshot": 26, "search": [14, 20, 23, 26, 28], "sebac": 23, "second": [6, 14], "secondari": 12, "secret": 23, "section": [2, 3, 6, 7, 12, 14, 20, 27, 29, 37, 38], "secundum": [20, 38], "sediment": 23, "see": [2, 3, 6, 7, 8, 9, 12, 16, 20, 21, 23, 24, 25, 26, 27, 28, 35, 37, 38], "seek": 31, "seem": 28, "seen": [20, 27, 38], "segment": [12, 23, 33], "segreg": 23, "seizur": [20, 24], "select": [2, 6, 35, 38], "self": [11, 13], "sens": [2, 23, 27], "sensat": 23, "sensori": 23, "sensorineur": 38, "separ": [11, 36, 38], "septal": [20, 24, 38], "septum": [20, 24], "sequenc": [2, 3, 4, 5, 6, 7, 9, 12, 13, 14, 16, 26], "sequence_vari": 12, "ser196ter": 20, "ser261ci": 20, "ser36thrfster25": 20, "serializ": 11, "seriou": 14, "serum": 23, "serv": [5, 28], "servic": 14, "set": [8, 10, 12, 20, 27, 33, 35, 38], "sete": 35, "setup": [20, 27, 38], "sever": [3, 5, 12, 14, 20, 24, 26, 27, 29, 36, 37, 38], "sex": [0, 6, 12, 23, 31], "sex_pred": [5, 6, 34], "sh": 27, "sharpei": 23, "shelf": [28, 37], "shift": 37, "short": [12, 20, 29, 38], "shortcut": 12, "shorter": [6, 38], "should": [2, 3, 6, 7, 10, 12, 13, 14, 16, 20, 26, 35, 38], "show": [12, 16, 23, 24, 26, 27, 28, 32, 33, 35, 37, 38], "shown": [16, 20, 25, 26], "sidak": 27, "side": [4, 38], "sign": [12, 13, 20, 37, 38], "signal": [23, 27], "signific": [12, 23, 27, 29, 38], "significantli": [20, 38], "sime": 27, "similar": [12, 38], "similarli": 33, "simpl": [6, 14, 19, 21, 24, 26, 29, 37], "simplest": 27, "simplifi": [20, 38], "sinc": [3, 6, 12, 38], "sine": 12, "singl": [2, 12, 14, 20, 35, 37], "situ": 23, "size": [12, 13, 23], "skelet": 23, "skin": [8, 23], "skinfold": 23, "skip": [2, 20, 23, 38], "skipkei": 11, "skipping_general_term": 2, "skipping_non_phenotype_term": 2, "skipping_same_observed_proport": 2, "skipping_since_one_genotype_had_zero_observ": 2, "skull": 23, "slice": 13, "small": [24, 27, 29], "smell": 23, "snv": [12, 24, 37], "so": [6, 12, 26, 27, 37, 38], "so_term": 12, "softwar": 38, "solut": [14, 26], "somat": 23, "some": [4, 14, 18, 19, 26, 27, 28], "some_cell_has_greater_than_one_count": 2, "someth": 14, "sometim": [32, 37], "sort": 14, "sort_index": 38, "sort_kei": 11, "sound": [23, 33], "sourc": [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 19, 37], "span": [6, 12, 13, 23, 24, 26], "specif": [3, 6, 12, 14, 18, 20, 23, 26, 27, 33, 35, 38], "specifi": 2, "specified_term": 27, "specifiedtermsmtcfilt": [1, 2, 27], "spell": 24, "spinal": 23, "spleen": 23, "splice": [12, 20], "splice_acceptor_vari": [12, 14, 20], "splice_donor_5th_base_vari": [12, 14, 20], "splice_donor_region_vari": 12, "splice_donor_vari": [12, 14, 20], "splice_polypyrimidine_tract_vari": [12, 14], "splice_region_vari": [12, 20], "spuriou": 27, "st1": 2, "stabil": 23, "stalk": 23, "standalon": 16, "standard": 12, "start": [6, 12, 13, 14, 16, 23, 26], "start_lost": [12, 38], "start_on_strand": 13, "start_retained_vari": 12, "stat": [3, 8, 20, 27, 38], "state": [12, 14, 24], "static": [2, 5, 6, 8, 11, 12, 14, 27], "statist": [2, 4, 18, 20, 23, 25, 27, 28, 29], "statsmodel": 27, "statu": [2, 14], "statur": 29, "stdout": 14, "step": [19, 20, 25, 26, 31, 38], "stimul": 23, "stomatocytosi": 26, "stool": 23, "stop": [20, 23], "stop_gain": [6, 12, 20, 37, 38], "stop_lost": 12, "stop_retained_vari": [12, 20], "storag": 14, "store": [14, 20, 26, 27, 33, 37, 38], "store_annot": 14, "str": [2, 3, 5, 6, 7, 8, 10, 12, 13, 14, 15, 16, 24], "strabismu": 38, "strand": [6, 12, 13, 26], "strategi": [2, 16, 23, 38], "stress": 23, "strict": 14, "string": [8, 12, 14, 16, 23], "stringio": 14, "stromal": 23, "strongli": [6, 12], "structur": [6, 8, 12, 14, 23, 24, 27, 29, 37, 38], "structural_defect": 38, "structural_so_id_to_displai": 12, "structural_typ": [6, 12, 37], "structural_vari": 6, "studi": 18, "stump": 23, "sub": [12, 14], "subclass": [6, 11], "subgroup": 12, "subject": [12, 33, 37, 38], "submodul": 17, "subpackag": 17, "subplot": [20, 26, 38], "subsect": [12, 14], "subsequ": 26, "subunit": 23, "successfulli": 20, "suffix": 14, "suitabl": 12, "sum": [29, 38], "summar": [12, 14, 16, 24, 26, 38], "summari": [14, 26], "summarize_hpo_analysi": [0, 16, 20, 38], "summary_df": [20, 38], "suox": 21, "super": 11, "supercoil": 12, "suppli": 12, "support": [6, 11, 12, 13, 18, 21, 35, 37], "supports_shap": 4, "suppos": 11, "suppresor": 12, "surf1": [6, 12], "surf2": 6, "surfac": 23, "surviv": [27, 36], "sv": [6, 12, 14], "sv_delet": 24, "sv_info": 12, "svart": 13, "svlen": [12, 24], "svtype": [12, 24], "sweat": 23, "switch": 19, "symbol": [6, 12, 13, 20, 24, 26], "symptom": [12, 20, 23, 37, 38], "synapt": 23, "syndrom": [12, 20, 30, 37], "synonym": [32, 37], "synonymous_vari": [12, 32, 37], "system": [12, 13, 23, 24, 27, 29], "t": [9, 12, 14, 20, 24], "tabl": [4, 6, 16, 20, 23, 24, 25, 27, 38], "tail": 38, "take": [4, 8, 14, 20, 27, 32, 33, 38], "tall": 29, "tandem": 12, "target": [6, 7, 26, 37], "tast": 23, "taxon": 6, "tbx5": [26, 33], "tbx5_all_vari": 20, "tbx5_cohort_info": 20, "tbx5_frameshift": 38, "tbx5_frameshift_vs_missens": 20, "tbx5_gpsea_with_uniprot_domain": 26, "tbx5_protein_diagram": 20, "technic": 18, "tediou": [33, 37], "tedium": 33, "temperatur": 23, "templat": 16, "ten": [12, 27], "tend": 12, "term": [2, 3, 7, 8, 12, 14, 16, 18, 20, 24, 28, 29, 33, 37, 38], "term_frequency_threshold": [2, 20, 27, 38], "term_id": 12, "termid": [2, 3, 6, 7, 8, 12, 33], "termin": [12, 18], "terms_to_test": [2, 27], "test": [2, 3, 4, 5, 6, 7, 8, 9, 12, 13, 14, 16, 18, 20, 21, 23, 25, 28, 29, 31, 32, 33, 36, 37], "test_uniprot_json": 12, "text": 15, "textio": 15, "textiobas": 14, "textiowrapp": [14, 15], "tf_binding_site_vari": 12, "tfbs_ablat": 12, "tfbs_amplif": 12, "th": 3, "than": [2, 6, 12, 20, 23, 38], "thank": 38, "thei": [13, 18, 23, 27, 37, 38], "them": 14, "themselv": 13, "therebi": 2, "therefor": [2, 3, 6, 23, 27, 37], "thi": [2, 3, 6, 7, 8, 11, 12, 13, 14, 18, 19, 20, 21, 23, 24, 26, 27, 29, 31, 32, 33, 34, 37, 38], "thick": 23, "thin": 4, "think": 27, "third": 38, "thorac": 23, "thr161pro": 20, "thr223met": 20, "thr72ly": 20, "three": [6, 12, 21, 27], "three_prime_utr_vari": [12, 14], "threshold": [2, 6, 20, 27, 38], "thrombocyt": 23, "thrombosi": 23, "through": 14, "thu": [3, 6, 12, 27, 29, 38], "thumb": [5, 20, 37, 38], "thymu": 23, "thyroid": 23, "tight_layout": [20, 26], "tild": 37, "time": [3, 12, 14, 19, 27], "timeout": 14, "tissu": 23, "to_csv": [20, 38], "to_displai": 12, "to_negative_strand": 13, "to_opposite_strand": 13, "to_positive_strand": 13, "to_str": 12, "todo": [16, 21, 22, 26, 28, 29, 31, 36], "togeth": [12, 20, 38], "toler": 2, "tonsil": 23, "too": 23, "toolkit": [8, 14, 20, 26], "top": [3, 7, 8, 12, 14, 20, 23, 26], "top_phenotype_count": 16, "top_variant_count": 16, "total": [3, 12, 20, 27, 29, 38], "total_patient_count": 12, "total_test": [3, 20, 38], "traceback": 27, "track": [7, 14], "tracker": [18, 28], "tract": 23, "trans_id": 12, "transcript": [3, 6, 12, 14, 16, 20, 35, 37, 38], "transcript_abl": [12, 38], "transcript_amplif": 12, "transcript_coordin": 16, "transcript_id": [6, 12, 16, 20], "transcriptannot": [0, 12, 14], "transcriptcoordin": [0, 12, 14, 16, 20, 26], "transcriptcoordinateservic": [0, 14], "transcriptinfoawar": [0, 12, 14], "transform": [14, 26, 38], "transit": 13, "translat": 12, "transloc": [6, 12], "transmiss": 23, "transpos": [12, 13], "transpose_coordin": [12, 13], "trascript": 20, "travers": 14, "tree": 14, "tri": 14, "triphalang": [20, 38], "triphosph": 23, "trp121gly": 20, "true": [2, 6, 7, 11, 12, 13, 14, 16, 20, 26, 27, 33, 37, 38], "truncat": 20, "try": [10, 11, 13, 14], "ttest_ind": 9, "tteststatist": [8, 9], "tupl": [3, 6, 12, 14, 20, 26, 27, 33, 37, 38], "turn": [14, 26], "tutori": [18, 25, 38], "twice": 27, "two": [3, 4, 6, 8, 12, 13, 20, 21, 26, 27, 28, 29, 30, 31, 38], "tx": [14, 16, 37], "tx_annot": 12, "tx_coordin": [16, 20, 26], "tx_id": [6, 12, 14, 16, 20, 32, 35, 37, 38], "txc_servic": [20, 26], "type": [2, 3, 6, 7, 8, 11, 12, 13, 14, 16, 18, 26, 27, 37], "typeerror": [11, 27], "typic": [3, 12, 28, 31, 38], "tyr136ter": 20, "tyr291ter": 20, "tyr342thrfster52": 20, "tyr407ter": [20, 26], "u": [8, 9, 20, 26, 29, 37], "ucsc": 13, "ucsc_nam": 13, "ulna": [20, 38], "umbil": 23, "umbilicu": 23, "unavail": 12, "under": [14, 20, 26, 27, 38], "underli": 37, "understand": 35, "undoubtedli": 26, "uniform": 6, "uninterest": 23, "uniprot": [12, 14, 16], "uniprot_json": [12, 26], "uniprotmetadataservic": 12, "uniprotproteinmetadataservic": [0, 14], "uniqu": [2, 3, 5, 12, 20], "unit": [13, 19], "unknown": [6, 12], "unknown_sex": [12, 34], "unless": [7, 20, 26, 27, 33], "unlik": [14, 23, 27], "unnecessari": 23, "untransl": [12, 26], "up": [20, 27, 29, 33, 35, 38], "upper": [20, 23], "upstream": 13, "upstream_gene_vari": [12, 14], "urat": 23, "uri": 12, "urin": 23, "urinari": 23, "urobilinogen": 23, "us": [2, 3, 6, 7, 8, 10, 11, 12, 13, 14, 15, 16, 18, 20, 24, 26, 27, 28, 31, 33, 34, 37, 38], "usabl": 3, "usag": [7, 9, 16], "use_al": 27, "usealltermsmtcfilt": [1, 2, 27], "user": [12, 18, 20, 23, 26, 38], "usual": [6, 12, 23, 28, 38], "utf": [14, 15], "util": [0, 5, 6, 38], "utr": [12, 26], "uvea": 23, "v": [18, 27, 32], "v2024": [20, 26, 27, 33, 37, 38], "vagin": 23, "val153serfster21": 20, "val214glyfster12": 20, "val263met": 20, "val267trpfster127": 20, "valid": [14, 16, 20, 27, 38], "validation_polici": 14, "validation_runn": 14, "validationrunn": 14, "valu": [2, 3, 4, 8, 9, 12, 13, 14, 16, 20, 27, 38], "valueerror": [5, 12, 13, 14], "valv": 23, "varcachingfunctionalannot": [0, 14], "vari": [6, 38], "variabl": [4, 10, 12, 31, 38], "variant": [0, 3, 4, 6, 12, 14, 16, 18, 22, 24, 27, 28, 31, 32, 38], "variant_class": [6, 12, 37], "variant_coordin": [12, 14], "variant_effect": [6, 12, 16, 20, 32, 35, 37, 38], "variant_effect_count_by_tx": 12, "variant_fallback": 14, "variant_info": 12, "variant_kei": [6, 12, 14], "variant_key_of_interest": 37, "variant_loc": 16, "variant_locations_counted_absolut": 16, "variant_pred": 6, "variantannotationcach": [0, 14], "variantclass": [0, 6, 12, 37], "variantcoordin": [0, 12, 14, 24], "variantcoordinatefind": [0, 14], "varianteffect": [0, 6, 12, 14, 16, 20, 32, 35, 37, 38], "variantformatt": [0, 16], "variantinfo": [0, 12], "variantinfoawar": [0, 12], "variantpred": [5, 6, 20, 28, 31, 32, 35, 37, 38], "varianttranscriptproteinartist": 16, "varianttranscriptvisu": [0, 16], "variou": 13, "vascular": 23, "vasculatur": 23, "vc": [12, 14], "vcf": 12, "vcv000522858": 37, "ventricular": [20, 23, 24, 38], "vep": 14, "vepfunctionalannot": [0, 14], "verbos": 5, "veri": 2, "version": [4, 20, 29, 33, 38], "vessel": 23, "via": [7, 11, 14, 16], "view": [0, 18, 20, 26, 38], "viewabl": 16, "viewer": 20, "virtual": 19, "vision": 23, "visit": 14, "visitor": 14, "visual": [16, 20, 23, 26], "vitamin": 23, "voic": 23, "volum": 23, "vri": [8, 36], "vvhgvsvariantcoordinatefind": [0, 14], "vvmulticoordinateservic": [0, 14, 20, 26], "w": [14, 20, 38], "wa": [2, 3, 5, 7, 12, 14, 16, 20, 27, 29, 30, 33, 38], "wai": [20, 26, 38], "waist": 23, "wall": 23, "want": [2, 8, 26, 27, 32, 37, 38], "warn": [14, 20, 26, 38], "warning_count": 14, "we": [2, 3, 6, 8, 12, 14, 15, 16, 18, 19, 20, 22, 23, 26, 27, 28, 29, 31, 32, 33, 34, 35, 37, 38], "websit": 26, "weight": 23, "well": [6, 7, 12, 26], "were": [3, 14, 16, 20, 26, 33, 37, 38], "what": [22, 30, 31, 35, 38], "whatsoev": 6, "when": [2, 7, 12, 14, 26, 27, 33, 35], "where": [2, 3, 12, 14, 20, 32], "whether": [7, 8, 12, 18, 38], "which": [2, 4, 5, 6, 7, 12, 20, 26, 29, 37, 38], "while": [4, 21, 28, 38], "whisker": 38, "whitnei": [8, 9, 29, 36], "who": [5, 7, 20, 21, 32, 38], "whose": [12, 20, 33, 38], "why": 2, "wilcoxon": 38, "wise": [20, 27], "with_cache_fold": 14, "with_strand": 13, "within": [12, 14, 20, 38], "without": [6, 38], "word": [18, 23], "work": [2, 10, 14, 26], "worth": 2, "would": [2, 8, 12, 18, 21, 27, 29, 37, 38], "wrap": [8, 12, 26, 37], "wrap_scoring_funct": 8, "wrapper": [4, 9, 15], "write": [14, 16, 20, 22, 38], "written": 16, "wwox": 33, "x": [32, 35, 38], "x_1000001_1000027_": 12, "x_1000001_1000027_taaaaaaaaaaaaaaaaaaaaaaaaaa_t": 12, "x_12345_12345_c_g": [6, 12], "xm_": 14, "y": [32, 38], "ye": [3, 5, 7, 33], "yet": 12, "yield": 27, "ylabel": 38, "ylim": 38, "you": [11, 12, 13, 26, 27], "your": [13, 19, 26], "z": 32, "zero": [2, 6, 13, 20, 24, 29, 32, 38], "\u03b1": 12, "\u03c72": 4}, "titles": ["gpsea package", "gpsea.analysis package", "gpsea.analysis.mtc_filter package", "gpsea.analysis.pcats package", "gpsea.analysis.pcats.stats package", "gpsea.analysis.predicate package", "gpsea.analysis.predicate.genotype package", "gpsea.analysis.predicate.phenotype package", "gpsea.analysis.pscore package", "gpsea.analysis.pscore.stats package", "gpsea.config module", "gpsea.io module", "gpsea.model package", "gpsea.model.genome package", "gpsea.preprocessing package", "gpsea.util module", "gpsea.view package", "API reference", "GPSEA", "Installation", "Tutorial", "Genotype-Phenotype Correlations in Autosomal Recessive Diseases", "Cohort exploratory analysis", "General HPO terms", "Glossary", "User guide", "Input data", "Multiple-testing correction", "Predicates", "De Vries Score", "Partition by a diagnosis", "Genotype Predicates", "Groups Predicate", "HPO predicate", "Partition by the sex of the individual", "Mode of Inheritance Predicates", "Phenotype Predicates", "Variant Predicates", "Statistical tests"], "titleterms": {"2x2": 27, "2x3": 27, "The": 20, "abnorm": [27, 29], "all": [20, 27, 33], "allel": [20, 24], "altern": 26, "an": 24, "analysi": [1, 2, 3, 4, 5, 6, 7, 8, 9, 20, 22, 27, 38], "api": [17, 26], "ar": 27, "assign": 35, "associ": 38, "autosom": 21, "background": 27, "block": 37, "build": 37, "chang": 24, "child": 27, "choos": [26, 27, 35], "code": 29, "cohort": [20, 22, 26, 33, 38], "compar": 38, "complex": 37, "condit": 37, "config": 10, "configur": 38, "congenit": 29, "content": [25, 28, 31, 36], "coordin": 26, "correct": 27, "correl": 21, "count": 27, "creat": [26, 38], "creator": 26, "data": 26, "de": 29, "delai": 29, "descend": 27, "development": 29, "diagnosi": 30, "diseas": 21, "distribut": [20, 26], "dump": 26, "dysmorph": 29, "enter": 26, "exact": 38, "exampl": [32, 33, 38], "exclud": 27, "explor": 20, "exploratori": 22, "facial": 29, "featur": [26, 29], "feedback": 18, "fet": 38, "fetch": 26, "filter": 27, "final": 29, "fisher": 38, "frameshift": [20, 38], "from": 26, "ga4gh": 26, "galleri": 28, "gener": [23, 27], "genom": 13, "genotyp": [6, 20, 21, 26, 27, 31, 35, 38], "get": 26, "glossari": 24, "gpsea": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 18, 27], "group": [27, 32, 35, 38], "growth": 29, "guid": 25, "ha": 27, "have": 27, "hmf01": 27, "hmf02": 27, "hmf03": 27, "hmf04": 27, "hmf05": 27, "hmf06": 27, "hmf07": 27, "hmf08": 27, "hpo": [20, 23, 27, 33], "ident": 27, "implement": 27, "individu": [34, 35], "inherit": 35, "input": 26, "instal": 19, "interest": 26, "invert": 37, "io": 11, "json": 26, "latest": 19, "length": 24, "level": 27, "literatur": 18, "load": [20, 26], "mann": 38, "manual": 26, "missens": 20, "missing_implies_phenotype_exclud": 33, "mode": 35, "model": [12, 13], "modul": [10, 11, 15], "more": [27, 28], "mtc": 27, "mtc_filter": 2, "multipl": 27, "need": 28, "neither": 27, "non": 29, "nor": 27, "observ": 27, "occur": 27, "one": 27, "onset": 29, "packag": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 12, 13, 14, 16], "pars": 26, "partit": [30, 34], "path": 24, "pcat": [3, 4], "phenopacket": 26, "phenotyp": [7, 20, 21, 26, 27, 33, 36, 38], "plot": [20, 26], "postnat": 29, "predic": [5, 6, 7, 20, 28, 31, 32, 33, 35, 36, 37], "prenat": 29, "prepar": 20, "preprocess": 14, "procedur": 27, "proport": 27, "protein": [20, 26], "provid": 26, "pscore": [8, 9], "rare": 27, "recess": 21, "refer": [17, 24], "releas": 19, "respect": 20, "rest": [26, 38], "retard": 29, "rule": 24, "run": 19, "same": 27, "score": [29, 38], "scorer": 29, "sequenc": 20, "sex": 34, "shape": 4, "show": 20, "showcas": 26, "skip": 27, "sourc": 26, "specifi": 27, "stabl": 19, "standard": 26, "stat": [4, 9], "statist": 38, "strategi": 27, "submodul": 0, "subpackag": [0, 1, 3, 5, 8, 12], "subset": 35, "summar": 20, "summari": 20, "support": 4, "tbx5": [20, 38], "term": [23, 27], "test": [19, 27, 38], "than": 27, "transcript": 26, "true": 24, "tutori": [20, 21], "u": 38, "underpow": 27, "uniprot": 26, "us": [29, 35], "user": 25, "util": 15, "v": [20, 38], "valid": 26, "variant": [20, 26, 35, 37], "veri": 27, "view": 16, "vri": 29, "which": 27, "whitnei": 38}})