Search.setIndex({"docnames": ["apidocs/genophenocorr", "apidocs/genophenocorr.analysis", "apidocs/genophenocorr.analysis.predicate", "apidocs/genophenocorr.analysis.predicate.genotype", "apidocs/genophenocorr.analysis.predicate.phenotype", "apidocs/genophenocorr.constants", "apidocs/genophenocorr.constants.variant_effects", "apidocs/genophenocorr.data", "apidocs/genophenocorr.model", "apidocs/genophenocorr.model.genome", "apidocs/genophenocorr.preprocessing", "apidocs/genophenocorr.view", "apidocs/modules", "autosomal_recessive", "index", "installation", "tutorial", "user-guide/exploratory", "user-guide/index", "user-guide/input-data"], "filenames": ["apidocs/genophenocorr.rst", "apidocs/genophenocorr.analysis.rst", "apidocs/genophenocorr.analysis.predicate.rst", "apidocs/genophenocorr.analysis.predicate.genotype.rst", "apidocs/genophenocorr.analysis.predicate.phenotype.rst", "apidocs/genophenocorr.constants.rst", "apidocs/genophenocorr.constants.variant_effects.rst", "apidocs/genophenocorr.data.rst", "apidocs/genophenocorr.model.rst", "apidocs/genophenocorr.model.genome.rst", "apidocs/genophenocorr.preprocessing.rst", "apidocs/genophenocorr.view.rst", "apidocs/modules.rst", "autosomal_recessive.rst", "index.rst", "installation.rst", "tutorial.rst", "user-guide/exploratory.rst", "user-guide/index.rst", "user-guide/input-data.rst"], "titles": ["genophenocorr package", "genophenocorr.analysis package", "genophenocorr.analysis.predicate package", "genophenocorr.analysis.predicate.genotype package", "genophenocorr.analysis.predicate.phenotype package", "genophenocorr.constants package", "genophenocorr.constants.variant_effects module", "genophenocorr.data package", "genophenocorr.model package", "genophenocorr.model.genome package", "genophenocorr.preprocessing package", "genophenocorr.view package", "API reference", "Genotype-Phenotype Correlations in Autosomal Recessive Diseases", "genophenocorr", "Installation", "Tutorial", "Cohort exploratory analysis", "User guide", "Input data"], "terms": {"i": [0, 1, 3, 4, 8, 10, 13, 14, 15, 16, 19], "librari": [0, 9, 15, 19], "analyz": [0, 1, 11], "genotyp": [0, 1, 8, 10, 11, 14, 16, 19], "phenotyp": [0, 1, 8, 10, 11, 14, 16, 17, 19], "correl": [0, 1, 11, 14], "cohort": [0, 1, 7, 8, 10, 11, 14, 16], "rare": [0, 1], "diseas": [0, 14], "patient": [0, 1, 2, 3, 4, 8, 10, 11, 14, 16, 19], "analysi": [0, 13, 14, 19], "configure_cohort_analysi": [0, 1, 16], "cohortanalysi": [0, 1], "genotypephenotypeanalysisresult": [0, 1], "cohortanalysisconfigur": [0, 1], "cohortanalysisconfigurationbuild": [0, 1], "constant": 0, "submodul": 0, "data": [0, 1, 8, 9, 10, 14, 16, 18], "get_toy_cohort": [0, 7, 16], "model": [0, 1, 2, 3, 10, 16, 19], "variant": [0, 1, 2, 3, 8, 10, 11, 14, 16, 17, 19], "variantcoordin": [0, 1, 3, 8, 10], "transcriptannot": [0, 8, 10], "varianteffect": [0, 1, 3, 8, 16], "transcriptinfoawar": [0, 8, 10], "transcriptcoordin": [0, 8, 10, 11], "proteinmetadata": [0, 8, 10, 11], "proteinfeatur": [0, 1, 8], "featureinfo": [0, 8], "featuretyp": [0, 1, 3, 8], "preprocess": [0, 19], "configure_caching_patient_cr": [0, 10, 19], "configure_patient_cr": [0, 10], "variantcoordinatefind": [0, 10], "functionalannot": [0, 10], "proteinmetadataservic": [0, 10], "patientcr": [0, 10], "phenopacketvariantcoordinatefind": [0, 10], "phenopacketpatientcr": [0, 10, 19], "load_phenopacket_fold": [0, 10], "load_phenopacket": [0, 10], "phenotypecr": [0, 10], "phenotypevalidationexcept": [0, 10], "proteinannotationcach": [0, 10], "protcachingfunctionalannot": [0, 10], "uniprotproteinmetadataservic": [0, 10], "vepfunctionalannot": [0, 10], "variantannotationcach": [0, 10], "varcachingfunctionalannot": [0, 10], "vvhgvsvariantcoordinatefind": [0, 10], "vvtranscriptcoordinateservic": [0, 10], "view": [0, 14, 16], "cohortview": [0, 11], "varianttranscriptvisu": [0, 11], "hpo": [1, 4, 8, 10, 11, 14, 16, 19], "minimalontologi": [1, 4, 10], "config": 1, "none": [1, 2, 3, 4, 8, 10], "sourc": [1, 2, 3, 4, 7, 8, 10, 11, 15], "configur": [1, 10, 19], "given": [1, 2, 3, 8, 10], "paramet": [1, 3, 4, 8, 10, 11], "hpotk": [1, 8, 16, 19], "us": [1, 2, 3, 8, 9, 10, 11, 14, 16, 19], "an": [1, 2, 3, 4, 8, 10, 11, 13, 14, 15, 16, 19], "option": [1, 2, 8, 10, 15, 19], "parameter": 1, "The": [1, 2, 3, 4, 8, 9, 10, 14, 15, 16, 19], "class": [1, 2, 3, 4, 8, 9, 10, 11, 19], "base": [1, 2, 3, 4, 8, 10, 11, 16], "object": [1, 2, 4, 8, 10, 11], "driver": 1, "run": [1, 10, 11, 16], "analys": [1, 16, 19], "provid": [1, 4, 9, 11, 14, 19], "variou": 1, "method": [1, 8, 10, 19], "test": [1, 2, 3, 4, 11, 13, 14], "all": [1, 2, 3, 4, 8, 15, 16, 19], "wrap": 1, "result": [1, 4, 8, 10, 14, 16], "abstract": [1, 2, 4, 8, 10], "compare_by_variant_effect": [1, 16], "effect": [1, 3, 8, 10, 17, 19], "tx_id": [1, 8, 10, 16], "str": [1, 2, 3, 4, 8, 10, 11], "compar": [1, 13], "v": [1, 14, 16], "other": [1, 8, 10, 13, 14], "access": [1, 3, 15], "transcript": [1, 3, 8, 10, 11], "interest": [1, 3, 8], "compare_by_variant_kei": 1, "variant_kei": [1, 3, 8], "kei": [1, 3, 8, 14], "e": [1, 8, 10, 14, 16], "g": [1, 8, 10, 14, 16], "x_12345_12345_c_g": [1, 8], "compare_by_exon": 1, "exon_numb": [1, 3], "int": [1, 2, 3, 8, 10, 11], "we": [1, 3, 8, 10, 14, 15, 16, 17, 19], "1": [1, 2, 3, 8, 11, 16], "number": [1, 2, 3, 8, 11], "exon": [1, 3, 8], "usual": [1, 3, 8], "0": [1, 2, 3, 8, 10, 16], "comput": [1, 3, 10], "scienc": [1, 3], "therefor": [1, 3], "first": [1, 2, 3, 8, 19], "ha": [1, 2, 3, 8, 10], "second": [1, 2, 3, 10], "2": [1, 3, 8, 11, 16], "so": [1, 3, 8], "do": [1, 3, 9, 10, 15, 16], "check": [1, 2, 3, 8, 10], "span": [1, 3, 8], "beyond": [1, 3], "transcript_id": [1, 3, 8], "10": [1, 3, 8], "000": [1, 3, 16], "return": [1, 2, 3, 4, 8, 10, 11], "booleanpred": [1, 2, 3, 16], "fals": [1, 3, 4, 8, 10], "well": [1, 3, 8, 16], "least": [1, 3, 10], "homo": [1, 3], "sapien": [1, 3], "taxon": [1, 3], "posit": [1, 3, 8, 10], "b": [1, 8, 16], "between": [1, 8, 9, 14], "group": [1, 2, 13, 16], "compare_by_protein_feature_typ": 1, "feature_typ": [1, 3, 8], "affect": [1, 3, 8, 13], "rest": [1, 10], "protein": [1, 3, 8, 10, 11, 14, 19], "featur": [1, 3, 8, 14, 15, 17], "type": [1, 8, 10, 14], "compare_by_protein_featur": 1, "name": [1, 2, 3, 8, 10], "can": [1, 2, 3, 4, 8, 9, 10, 15, 16, 17, 19], "string": [1, 8, 10], "identifi": [1, 2, 8], "dna": [1, 8, 9], "bind": 1, "n_usabl": 1, "seri": 1, "all_count": 1, "datafram": 1, "pval": 1, "corrected_pv": 1, "phenotype_categori": 1, "iter": [1, 8, 10, 11], "patientcategori": [1, 2, 3, 4], "question": [1, 2, 3, 4, 14], "summar": [1, 16], "properti": [1, 2, 8, 10], "get": [1, 2, 3, 4, 8, 10, 16], "panda": 1, "map": [1, 8], "from": [1, 8, 10, 14, 15, 16], "term": [1, 4, 8, 10, 11, 14, 16, 19], "where": [1, 8, 10], "wa": [1, 4, 8, 10], "either": 1, "present": [1, 4, 8, 16], "exclud": [1, 4, 8], "ar": [1, 2, 3, 8, 9, 10, 13, 14, 15, 16], "thu": 1, "usabl": 1, "count": [1, 8, 16, 17], "exampl": [1, 8, 13, 16, 19], "predic": [1, 16], "bin": [1, 2], "two": [1, 2, 3, 9, 13, 16], "categori": [1, 2, 3, 4], "ye": [1, 2, 16], "No": [1, 2, 16], "presenc": [1, 4, 8, 16], "missens": [1, 8, 13, 14, 16], "absenc": [1, 16], "missense_vari": [1, 8, 16], "nm_123456": [1, 8], "7": [1, 8], "hp": [1, 16, 19], "0001166": [1, 16], "13": [1, 16], "5": [1, 8, 10, 16, 19], "0003011": 1, "3": [1, 8, 10, 16], "11": [1, 8], "16": [1, 16], "0012638": 1, "9": 1, "15": 1, "4": [1, 8, 10], "p": [1, 16], "each": [1, 8, 10, 13, 14, 16], "after": [1, 15], "multipl": 1, "correct": [1, 16], "sequenc": [1, 2, 4, 8, 10], "investig": 1, "creat": [1, 4, 8, 10, 11, 14], "frame": 1, "summari": 1, "row": [1, 16], "correspond": [1, 8], "column": 1, "have": [1, 3, 16], "percentag": 1, "per": [1, 11], "For": [1, 3, 8, 13, 16, 19], "instanc": [1, 2, 3, 4, 8, 10, 13, 16, 19], "varianteffectpred": [1, 3], "which": [1, 2, 4, 16, 19], "without": 1, "certain": [1, 2], "follow": [1, 4, 8, 15], "percent": [1, 16], "final": 1, "mai": [1, 8], "look": 1, "someth": 1, "like": [1, 3, 14], "thi": [1, 4, 8, 10, 13, 14, 15, 16, 19], "nm_1234": [1, 16], "arachnodactyli": [1, 16], "8461": 1, "50": [1, 16], "0000": 1, "000781": [1, 16], "020299": [1, 16], "abnorm": [1, 14], "musculatur": 1, "6": [1, 13], "35": 1, "2941": 1, "64": 1, "7058": 1, "000000": 1, "nervou": 1, "system": [1, 8, 19], "physiologi": [1, 14], "37": 1, "5000": 1, "62": 1, "missing_implies_exclud": [1, 4], "bool": [1, 4, 8, 10], "pval_correct": 1, "min_perc_patients_w_hpo": 1, "float": [1, 10], "include_sv": 1, "store": [1, 10, 19], "bonferroni": 1, "static": [1, 2, 4, 8, 11], "builder": 1, "new": 1, "initi": [1, 8, 15], "true": [1, 8, 10], "assum": [1, 8], "specif": [1, 4, 8, 10, 14], "list": [1, 8, 10, 16], "doe": [1, 2, 8, 10], "otherwis": [1, 8], "onli": [1, 2, 8, 10, 13], "those": 1, "explicitli": [1, 8], "A": [1, 8, 10, 11, 14, 16], "threshold": 1, "remov": 1, "observ": [1, 8, 10, 14], "fraction": 1, "retain": 1, "want": [1, 16], "includ": [1, 3, 8, 10, 14], "structur": [1, 8, 10], "symbol": [1, 8], "vcf": [1, 8], "notat": [1, 8], "build": [1, 9, 10], "custom": 1, "set": [1, 8], "If": [1, 3, 8], "invalid": 1, "pass": 1, "warn": 1, "log": 1, "previou": 1, "consequ": [1, 8], "imposs": 1, "mi": 1, "polypred": [1, 2, 4], "groupingpred": [1, 2, 3], "one": [2, 3, 8, 10, 13, 15], "sever": 2, "discret": 2, "repres": [2, 8, 14], "must": [2, 15], "exclus": 2, "exhaust": 2, "cover": 2, "possibl": [2, 13], "scenario": 2, "howev": [2, 15], "cannot": [2, 3, 4, 8, 10], "assign": [2, 3, 4], "ani": [2, 3, 4, 10], "meaning": [2, 3, 4], "note": [2, 8], "happi": 2, "path": [2, 10, 19], "inher": 2, "make": 2, "sens": 2, "compat": 2, "mode": 2, "inherit": 2, "get_categori": [2, 4], "produc": [2, 4], "get_quest": [2, 3, 4], "prepar": [2, 3, 4, 19], "answer": [2, 3, 4], "classmethod": 2, "n_categori": 2, "cat_id": 2, "descript": [2, 8], "uniqu": [2, 8], "numer": 2, "human": [2, 9, 14, 19], "readabl": [2, 8], "verbos": 2, "detail": 2, "belong": 2, "boolean": [2, 8, 10], "NO": 2, "who": [2, 13], "predict": [3, 8], "function": [3, 8, 10, 19], "varianteffectspred": 3, "effect1": 3, "effect2": 3, "variantpred": 3, "allel": [3, 8, 13, 14], "describ": [3, 8, 15, 17, 19], "you": [3, 8, 16], "call": [3, 8, 19], "variantspred": 3, "variant_key1": 3, "variant_key2": 3, "both": [3, 8], "exonpred": 3, "n": [3, 8, 16], "th": 3, "target": 3, "exonspred": 3, "exon1_numb": 3, "exon2_numb": 3, "protfeaturepred": 3, "protein_feature_nam": 3, "overlap": [3, 8, 9], "need": [3, 10, 16, 19], "egf": 3, "fbn1": 3, "protfeaturespred": 3, "protein_feature1_nam": 3, "protein_feature2_nam": 3, "protfeaturetypepred": 3, "protfeaturetypespred": 3, "feature_type1": 3, "feature_type2": 3, "": [3, 8, 10, 15, 16, 19], "propagatingphenotypepred": 4, "phenotypic_featur": 4, "termid": [4, 8], "phenotypepolypred": 4, "annot": [4, 8, 10, 11, 16, 19], "queri": [4, 8], "its": [4, 8, 11], "descend": 4, "ancestor": 4, "not_measur": 4, "propagatingphenotypebooleanpred": 4, "basephenotypebooleanpred": 4, "phenotypepredicatefactori": 4, "assess": 4, "factori": 4, "get_pred": 4, "id": [4, 8, 10], "propagatingphenotypebooleanpredicatefactori": 4, "variant_effect": [5, 8], "modul": [5, 8, 9], "defin": 8, "start": [8, 19], "top": 8, "level": [8, 18], "element": [8, 10], "info": [8, 10, 19], "patient_set": 8, "phenotype_set": 8, "variant_set": 8, "protein_set": 8, "counts_dict": 8, "excluded_memb": 8, "recess": 8, "size": 8, "from_pati": [8, 19], "member": [8, 16, 17, 19], "include_patients_with_no_hpo": [8, 10], "include_patients_with_no_vari": 8, "all_pati": 8, "frozenset": 8, "frozen": 8, "all_phenotyp": 8, "all_vari": 8, "collect": [8, 14], "all_protein": 8, "all_transcript": 8, "all_excluded_pati": 8, "total_patient_count": 8, "integ": 8, "list_all_pati": [8, 16], "list_all_phenotyp": [8, 16], "highest": 8, "tupl": [8, 10], "format": [8, 10], "list_all_vari": [8, 16], "list_all_protein": [8, 16], "list_data_by_tx": [8, 16], "refer": [8, 14], "counter": 8, "total": 8, "valu": [8, 16], "dictionari": 8, "get_excluded_id": 8, "get_excluded_count": 8, "get_protein_features_affect": 8, "patient_id": 8, "individu": [8, 13, 14, 18], "associ": [8, 10, 13, 14], "present_phenotyp": 8, "over": 8, "excluded_phenotyp": 8, "term_id": [8, 10], "is_observ": 8, "verifi": 8, "offici": 8, "respect": 8, "from_term": 8, "minimalterm": 8, "find": [8, 9, 10], "more": [8, 10, 15, 16, 19], "inform": [8, 14], "about": [8, 10, 15, 16, 17], "search": [8, 10], "http": [8, 10, 13, 15, 16, 19], "jax": 8, "org": [8, 10, 16, 19], "app": 8, "whether": [8, 14], "is_exclud": 8, "var_coordin": 8, "tx_annot": 8, "variantcoordinateawar": 8, "functionalannotationawar": 8, "coordin": [8, 9, 10, 19], "variant_coordin": [8, 10], "variant_str": 8, "represent": [8, 10], "variant_class": 8, "duplic": 8, "snv": 8, "etc": 8, "create_variant_from_scratch": 8, "gene_nam": 8, "trans_id": 8, "hgvsc_id": 8, "is_pref": 8, "exons_effect": 8, "protein_effect_start": 8, "protein_effect_end": 8, "singl": 8, "gene": [8, 13], "region": [8, 9], "genomicregion": [8, 9], "ref": 8, "alt": 8, "change_length": 8, "breakend": 8, "support": [8, 13, 14], "altern": 8, "chang": 8, "due": [8, 16], "chrom": 8, "label": [8, 11], "chromosom": [8, 13], "contig": [8, 9], "locat": [8, 14, 19], "end": 8, "genom": [8, 10, 14], "empti": [8, 10], "gg": 8, "del": 8, "charact": 8, "INS": 8, "me": 8, "sine": 8, "length": 8, "lead": 8, "zero": 8, "delet": 8, "insert": 8, "neg": 8, "22_10001_20000_inv": 8, "dup": 8, "mnv": 8, "inv": 8, "is_structur": 8, "chr5": 8, "101": 8, "svtype": 8, "120": 8, "svlen": 8, "oppos": 8, "liter": 8, "nacgtacgtac": 8, "qualnam": 8, "boundari": 8, "enum": 8, "state": 8, "variabl": 8, "locu": 8, "diploid": 8, "no_cal": 8, "homozygous_refer": 8, "heterozyg": 8, "homozygous_altern": 8, "hemizyg": 8, "code": [8, 11, 15, 16], "sampl": 8, "contain": 8, "gt": 8, "from_map": 8, "There": 8, "len": [8, 19], "for_sampl": 8, "unknown": 8, "pair": 8, "sample_id": 8, "print": 8, "shortcut": 8, "assert": 8, "entiti": [8, 10], "genotype_for_sampl": 8, "gene_id": 8, "hgvsc": 8, "affected_exon": 8, "affected_protein": 8, "protein_effect_coordin": 8, "hgv": [8, 10], "avail": [8, 10, 14], "els": 8, "mane": 8, "canon": 8, "ensembl": 8, "vep": [8, 10], "surf1": 8, "refseq": [8, 10], "prefer": 8, "c": [8, 10, 14, 16, 19], "9876g": 8, "t": [8, 10, 16], "definit": 8, "found": [8, 10], "www": [8, 10], "sequenceontologi": 8, "overlapping_exon": 8, "indic": 8, "index": [8, 15], "1st": 8, "protein_affect": 8, "alter": 8, "protein_effect_loc": 8, "import": [8, 16, 19], "prop": 8, "curi": 8, "attribut": [8, 14], "ontologi": [8, 14, 16, 19], "0001583": 8, "transcript_abl": 8, "0001893": 8, "splice_acceptor_vari": 8, "0001574": 8, "splice_donor_vari": 8, "0001575": 8, "stop_gain": 8, "0001587": 8, "frameshift_vari": [8, 16], "0001589": 8, "stop_lost": 8, "0001578": 8, "start_lost": 8, "0002012": 8, "transcript_amplif": 8, "0001889": 8, "inframe_insert": 8, "0001821": 8, "inframe_delet": 8, "0001822": 8, "protein_altering_vari": 8, "0001818": 8, "splice_region_vari": 8, "0001630": 8, "splice_donor_5th_base_vari": 8, "0001787": 8, "splice_donor_region_vari": 8, "0002170": 8, "splice_polypyrimidine_tract_vari": 8, "0002169": 8, "incomplete_terminal_codon_vari": 8, "0001626": 8, "start_retained_vari": 8, "0002019": 8, "stop_retained_vari": 8, "0001567": 8, "synonymous_vari": 8, "0001819": 8, "coding_sequence_vari": 8, "0001580": 8, "mature_mirna_vari": 8, "0001620": 8, "five_prime_utr_vari": 8, "0001623": 8, "three_prime_utr_vari": 8, "0001624": 8, "non_coding_transcript_exon_vari": 8, "0001792": 8, "intron_vari": 8, "0001627": 8, "nmd_transcript_vari": 8, "0001621": 8, "non_coding_transcript_vari": 8, "0001619": 8, "upstream_gene_vari": 8, "0001631": 8, "downstream_gene_vari": 8, "0001632": 8, "tfbs_ablat": 8, "0001895": 8, "tfbs_amplif": 8, "0001892": 8, "tf_binding_site_vari": 8, "0001782": 8, "regulatory_region_abl": 8, "0001894": 8, "regulatory_region_amplif": 8, "0001891": 8, "feature_elong": 8, "0001907": 8, "regulatory_region_vari": 8, "0001566": 8, "feature_trunc": 8, "0001906": 8, "intergenic_vari": 8, "0001628": 8, "sequence_vari": 8, "0001060": 8, "compact": 8, "uri": 8, "implementor": 8, "know": [8, 10], "basic": [8, 9], "cds_start": 8, "cds_end": 8, "intron": 8, "non": [8, 10], "cd": [8, 15], "enst00000123456": 8, "utr": 8, "codon": [8, 14], "last": 8, "termin": [8, 14], "is_cod": 8, "translat": 8, "get_coding_base_count": 8, "calcul": 8, "NOT": 8, "sinc": 8, "aminoacid": 8, "get_codon_count": 8, "get_five_prime_utr": 8, "untransl": 8, "get_three_prime_utr": 8, "get_cds_region": 8, "protein_id": [8, 10], "protein_featur": 8, "full": 8, "domain": 8, "subgroup": 8, "equal": 8, "repeat": 8, "motif": [8, 14], "get_features_variant_overlap": 8, "param": [8, 10], "to_str": 8, "ank": 8, "ankyrin": 8, "within": 8, "short": 8, "than": 8, "20": [8, 16], "amino": 8, "acid": 8, "conserv": 8, "biolog": 8, "signific": 8, "combin": 8, "secondari": 8, "organ": 8, "characterist": 8, "three": [8, 13], "dimension": 8, "fold": 8, "subsect": 8, "genomebuild": [9, 10], "arithmet": 9, "intersect": 9, "distanc": 9, "transpos": 9, "thei": [9, 14], "flip": 9, "strand": 9, "grch37": [9, 10], "p13": [9, 10], "grch38": [9, 10], "most": 9, "commonli": 9, "larg": 9, "port": 9, "svart": 9, "genome_build": 10, "validation_runn": 10, "validationrunn": 10, "cache_dir": 10, "variant_fallback": 10, "protein_fallback": 10, "uniprot": [10, 19], "conveni": [10, 19], "cach": [10, 19], "To": [10, 15, 16], "creator": [10, 19], "toolkit": [10, 19], "choos": [10, 16], "valid": 10, "runner": 10, "folder": [10, 19], "fetch": 10, "remot": 10, "api": [10, 14, 16, 19], "should": 10, "current": [10, 15, 19], "work": [10, 19], "directori": [10, 19], "In": [10, 13, 14], "case": 10, "exist": 10, "parent": 10, "fallback": 10, "local": 10, "just": [10, 16], "implement": [10, 13], "moment": 10, "metadata": 10, "gener": [10, 11, 14], "find_coordin": 10, "item": [10, 11, 16], "determin": 10, "some": [10, 14, 15, 16], "sort": [10, 16], "metaclass": 10, "establish": 10, "create_pati": [10, 19], "subject": 10, "hgvs_coordinate_find": 10, "genomicinterpret": 10, "figur": 10, "out": [10, 19], "phenopacket": [10, 14, 16], "schema": [10, 19], "finder": 10, "pars": [10, 19], "express": 10, "interpret": 10, "variantcal": 10, "phenotype_cr": 10, "var_func_ann": 10, "transform": [10, 19], "pp_directori": 10, "patient_cr": [10, 19], "json": [10, 16, 19], "file": [10, 19], "error": 10, "rais": 10, "point": 10, "turn": 10, "made": 10, "phenopacket_path": 10, "load": [10, 16, 19], "create_phenotyp": 10, "satisfi": 10, "requir": [10, 15, 19], "issu": [10, 14], "except": 10, "baseexcept": 10, "datadir": 10, "retriev": 10, "pickl": 10, "get_annot": 10, "store_annot": 10, "desir": [10, 13], "storag": 10, "under": 10, "fuction": 10, "here": [10, 16, 17, 19], "help": 10, "programmatic_access": 10, "databas": 10, "np_037407": 10, "protein_annot": 10, "include_computational_tx": 10, "predictor": [10, 19], "servic": 10, "xm_": 10, "perform": [10, 13, 14, 16, 19], "timeout": 10, "30": 10, "take": 10, "nm_005912": 10, "253a": 10, "extract": 10, "respons": [10, 19], "construct": [10, 19], "request": 10, "transcriptcoordinateservic": 10, "tx": [10, 11], "nm_002834": 10, "parse_respons": 10, "html_row": 11, "anchor": 11, "url": 11, "payload": 11, "hpo_anchor": 11, "hpo_id": 11, "hpo_term_counts_t": 11, "min_count": 11, "html": 11, "design": [11, 14], "displai": 11, "jupyt": [11, 14], "notebook": [11, 14, 16], "ipython": 11, "show": [11, 16], "explan": 11, "minimum": 11, "tabl": [11, 16], "variants_t": 11, "preferred_transcript": 11, "could": 11, "cohort_summary_t": 11, "protein_features_t": 11, "varianttranscriptproteinartist": 11, "graphic": 11, "distribut": [11, 17], "across": 11, "product": 11, "draw_vari": 11, "genophenocorr": [12, 13, 15, 16, 19], "packag": [12, 14, 15, 16], "subpackag": 12, "domin": 13, "simpl": [13, 15, 16], "becaus": 13, "carri": 13, "while": 13, "would": [13, 14], "principl": 13, "restrict": 13, "homozygot": 13, "ident": 13, "copi": 13, "matern": 13, "patern": 13, "practic": 13, "compound": 13, "heterozygot": 13, "common": 13, "comparison": 13, "might": 13, "against": 13, "nons": 13, "nonsens": 13, "see": [13, 16, 19], "li": 13, "jt": 13, "et": [13, 14], "al": [13, 14], "2022": 13, "pubm": 13, "ncbi": 13, "nlm": 13, "nih": 13, "gov": 13, "36303223": 13, "suox": 13, "field": 13, "fischer": 13, "exact": [13, 14], "kind": 13, "concept": 14, "denot": 14, "medic": 14, "context": 14, "word": 14, "deviat": 14, "normal": 14, "morphologi": 14, "behavior": 14, "f": [14, 16, 19], "deep": 14, "precis": 14, "medicin": 14, "biologi": 14, "genet": 14, "concern": 14, "relationship": 14, "mendelian": 14, "focu": 14, "place": 14, "studi": 14, "caus": 14, "manifest": 14, "python": [14, 15], "input": [14, 16, 18], "global": 14, "allianc": 14, "health": 14, "ga4gh": 14, "ingest": 14, "prematur": 14, "statist": 14, "fisher": 14, "report": [14, 16], "recommend": 14, "user": 14, "document": [14, 15], "instal": [14, 19], "instruct": 14, "brief": 14, "tutori": 14, "comprehens": 14, "guid": 14, "technic": 14, "read": [14, 19], "background": 14, "orgogozo": 14, "2015": 14, "differenti": 14, "best": 14, "leav": 14, "ask": 14, "bug": 14, "tracker": 14, "how": [15, 16, 17, 19], "your": [15, 16], "environ": 15, "easi": 15, "publish": 15, "pypi": 15, "python3": 15, "m": [15, 16], "pip": 15, "clone": 15, "github": 15, "repositori": [15, 19], "git": 15, "com": 15, "monarch": 15, "switch": 15, "develop": 15, "branch": 15, "checkout": 15, "abov": [15, 16], "virtual": 15, "done": 15, "step": [15, 19], "addit": 15, "henc": 15, "time": 15, "enabl": 15, "Then": 15, "pytest": 15, "unit": 15, "integr": 15, "internet": 15, "onlin": 15, "add": 15, "runonlin": 15, "command": [15, 16], "line": 15, "invoc": 15, "That": 15, "demonstr": 16, "let": [16, 19], "load_minimal_ontologi": [16, 19], "toi": 16, "latest": 16, "purl": [16, 19], "obolibrari": [16, 19], "obo": [16, 19], "todo": [16, 17, 18, 19], "move": 16, "workflow": 16, "now": [16, 19], "keep": 16, "thing": 16, "ship": 16, "section": [16, 18, 19], "learn": [16, 17], "d": 16, "h": 16, "j": 16, "k": 16, "l": 16, "o": [16, 19], "q": [16, 19], "r": 16, "u": 16, "w": 16, "x": 16, "y": 16, "z": 16, "14": 16, "0001250": 16, "0001257": 16, "17": 16, "1_281_281_a_g": 16, "1_361_363_ttc_t": 16, "np_09876": 16, "26": 16, "tx_dict": 16, "what": [16, 17], "partit": 16, "explain": 16, "updat": 16, "cohort_analysi": 16, "summary_df": 16, "head": 16, "846154": 16, "re": 16, "unstabl": 16, "order": 16, "better": 16, "soon": 16, "ascend": 16, "write": [17, 18], "mani": 17, "plot": 17, "high": 18, "overview": 18, "bridg": 18, "standard": 19, "form": 19, "consist": 19, "hold": 19, "dataset": 19, "explor": 19, "exploratori": 19, "involv": 19, "downstream": 19, "easiest": 19, "wai": 19, "box": 19, "requisit": 19, "ll": 19, "amaz": 19, "along": 19, "next": 19, "By": 19, "default": 19, "save": 19, "bandwidth": 19, "purpos": 19, "simple_cohort": 19, "simple_cohort_path": 19, "join": 19, "getcwd": 19, "walk": 19, "googl": 19, "protobuf": 19, "json_format": 19, "dirpath": 19, "_": 19, "filenam": 19, "endswith": 19, "pp_path": 19, "open": 19, "fh": 19, "pp": 19, "append": 19, "raw": 19}, "objects": {"": [[0, 0, 0, "-", "genophenocorr"]], "genophenocorr": [[1, 0, 0, "-", "analysis"], [5, 0, 0, "-", "constants"], [7, 0, 0, "-", "data"], [8, 0, 0, "-", "model"], [10, 0, 0, "-", "preprocessing"], [11, 0, 0, "-", "view"]], "genophenocorr.analysis": [[1, 1, 1, "", "CohortAnalysis"], [1, 1, 1, "", "CohortAnalysisConfiguration"], [1, 1, 1, "", "CohortAnalysisConfigurationBuilder"], [1, 1, 1, "", "GenotypePhenotypeAnalysisResult"], [1, 4, 1, "", "configure_cohort_analysis"], [2, 0, 0, "-", "predicate"]], "genophenocorr.analysis.CohortAnalysis": [[1, 2, 1, "", "compare_by_exon"], [1, 2, 1, "", "compare_by_protein_feature"], [1, 2, 1, "", "compare_by_protein_feature_type"], [1, 2, 1, "", "compare_by_variant_effect"], [1, 2, 1, "", "compare_by_variant_key"], [1, 2, 1, "", "compare_by_variant_keys"]], "genophenocorr.analysis.CohortAnalysisConfiguration": [[1, 2, 1, "", "builder"], [1, 3, 1, "", "include_sv"], [1, 3, 1, "", "min_perc_patients_w_hpo"], [1, 3, 1, "", "missing_implies_excluded"], [1, 3, 1, "", "pval_correction"]], "genophenocorr.analysis.CohortAnalysisConfigurationBuilder": [[1, 2, 1, "", "build"], [1, 2, 1, "", "include_sv"], [1, 2, 1, "", "min_perc_patients_w_hpo"], [1, 2, 1, "", "missing_implies_excluded"], [1, 2, 1, "", "pval_correction"]], "genophenocorr.analysis.GenotypePhenotypeAnalysisResult": [[1, 3, 1, "", "all_counts"], [1, 3, 1, "", "corrected_pvals"], [1, 3, 1, "", "n_usable"], [1, 3, 1, "", "phenotype_categories"], [1, 3, 1, "", "pvals"], [1, 2, 1, "", "summarize"]], "genophenocorr.analysis.predicate": [[2, 1, 1, "", "BooleanPredicate"], [2, 1, 1, "", "GroupingPredicate"], [2, 1, 1, "", "PatientCategory"], [2, 1, 1, "", "PolyPredicate"], [3, 0, 0, "-", "genotype"], [4, 0, 0, "-", "phenotype"]], "genophenocorr.analysis.predicate.BooleanPredicate": [[2, 5, 1, "", "NO"], [2, 5, 1, "", "YES"], [2, 2, 1, "", "get_categories"]], "genophenocorr.analysis.predicate.GroupingPredicate": [[2, 5, 1, "", "FIRST"], [2, 5, 1, "", "SECOND"], [2, 2, 1, "", "get_categories"]], "genophenocorr.analysis.predicate.PatientCategory": [[2, 3, 1, "", "cat_id"], [2, 3, 1, "", "description"], [2, 3, 1, "", "name"]], "genophenocorr.analysis.predicate.PolyPredicate": [[2, 2, 1, "", "get_categories"], [2, 2, 1, "", "get_question"], [2, 2, 1, "", "n_categories"], [2, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype": [[3, 1, 1, "", "ExonPredicate"], [3, 1, 1, "", "ExonsPredicate"], [3, 1, 1, "", "ProtFeaturePredicate"], [3, 1, 1, "", "ProtFeatureTypePredicate"], [3, 1, 1, "", "ProtFeatureTypesPredicate"], [3, 1, 1, "", "ProtFeaturesPredicate"], [3, 1, 1, "", "VariantEffectPredicate"], [3, 1, 1, "", "VariantEffectsPredicate"], [3, 1, 1, "", "VariantPredicate"], [3, 1, 1, "", "VariantsPredicate"]], "genophenocorr.analysis.predicate.genotype.ExonPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.ExonsPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.VariantEffectPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.VariantPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.VariantsPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.phenotype": [[4, 1, 1, "", "PhenotypePredicateFactory"], [4, 1, 1, "", "PropagatingPhenotypeBooleanPredicate"], [4, 1, 1, "", "PropagatingPhenotypeBooleanPredicateFactory"], [4, 1, 1, "", "PropagatingPhenotypePredicate"]], "genophenocorr.analysis.predicate.phenotype.PhenotypePredicateFactory": [[4, 2, 1, "", "get_categories"], [4, 2, 1, "", "get_predicate"]], "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypeBooleanPredicate": [[4, 2, 1, "", "get_question"], [4, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypeBooleanPredicateFactory": [[4, 2, 1, "", "get_categories"], [4, 2, 1, "", "get_predicate"]], "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate": [[4, 5, 1, "", "EXCLUDED"], [4, 5, 1, "", "NOT_MEASURED"], [4, 5, 1, "", "PRESENT"], [4, 2, 1, "", "get_categories"], [4, 2, 1, "", "get_question"], [4, 2, 1, "", "test"]], "genophenocorr.data": [[7, 4, 1, "", "get_toy_cohort"]], "genophenocorr.model": [[8, 1, 1, "", "Cohort"], [8, 1, 1, "", "FeatureInfo"], [8, 1, 1, "", "FeatureType"], [8, 1, 1, "", "Genotype"], [8, 1, 1, "", "Genotyped"], [8, 1, 1, "", "Genotypes"], [8, 1, 1, "", "Patient"], [8, 1, 1, "", "Phenotype"], [8, 1, 1, "", "ProteinFeature"], [8, 1, 1, "", "ProteinMetadata"], [8, 1, 1, "", "TranscriptAnnotation"], [8, 1, 1, "", "TranscriptCoordinates"], [8, 1, 1, "", "TranscriptInfoAware"], [8, 1, 1, "", "Variant"], [8, 1, 1, "", "VariantCoordinates"], [8, 1, 1, "", "VariantEffect"], [9, 0, 0, "-", "genome"]], "genophenocorr.model.Cohort": [[8, 3, 1, "", "all_excluded_patients"], [8, 3, 1, "", "all_patients"], [8, 3, 1, "", "all_phenotypes"], [8, 3, 1, "", "all_proteins"], [8, 3, 1, "", "all_transcripts"], [8, 3, 1, "", "all_variants"], [8, 2, 1, "", "from_patients"], [8, 2, 1, "", "get_excluded_count"], [8, 2, 1, "", "get_excluded_ids"], [8, 2, 1, "", "get_protein_features_affected"], [8, 2, 1, "", "list_all_patients"], [8, 2, 1, "", "list_all_phenotypes"], [8, 2, 1, "", "list_all_proteins"], [8, 2, 1, "", "list_all_variants"], [8, 2, 1, "", "list_data_by_tx"], [8, 3, 1, "", "total_patient_count"]], "genophenocorr.model.FeatureInfo": [[8, 3, 1, "", "end"], [8, 3, 1, "id23", "name"], [8, 3, 1, "id24", "region"], [8, 3, 1, "", "start"]], "genophenocorr.model.FeatureType": [[8, 5, 1, "id27", "DOMAIN"], [8, 5, 1, "id26", "MOTIF"], [8, 5, 1, "id28", "REGION"], [8, 5, 1, "id25", "REPEAT"]], "genophenocorr.model.Genotype": [[8, 5, 1, "", "HEMIZYGOUS"], [8, 5, 1, "", "HETEROZYGOUS"], [8, 5, 1, "", "HOMOZYGOUS_ALTERNATE"], [8, 5, 1, "", "HOMOZYGOUS_REFERENCE"], [8, 5, 1, "", "NO_CALL"], [8, 3, 1, "", "code"]], "genophenocorr.model.Genotyped": [[8, 2, 1, "", "genotype_for_sample"], [8, 3, 1, "", "genotypes"]], "genophenocorr.model.Genotypes": [[8, 2, 1, "", "empty"], [8, 2, 1, "", "for_sample"], [8, 2, 1, "", "from_mapping"], [8, 2, 1, "", "single"]], "genophenocorr.model.Patient": [[8, 2, 1, "", "excluded_phenotypes"], [8, 3, 1, "id0", "patient_id"], [8, 3, 1, "id1", "phenotypes"], [8, 2, 1, "", "present_phenotypes"], [8, 3, 1, "id3", "proteins"], [8, 3, 1, "id2", "variants"]], "genophenocorr.model.Phenotype": [[8, 2, 1, "", "from_term"], [8, 3, 1, "", "identifier"], [8, 3, 1, "", "is_excluded"], [8, 3, 1, "id5", "is_observed"], [8, 3, 1, "id4", "name"], [8, 3, 1, "", "observed"], [8, 5, 1, "", "term_id"]], "genophenocorr.model.ProteinFeature": [[8, 2, 1, "", "create"], [8, 3, 1, "", "feature_type"], [8, 3, 1, "", "info"], [8, 2, 1, "", "to_string"]], "genophenocorr.model.ProteinMetadata": [[8, 2, 1, "id19", "domains"], [8, 2, 1, "", "get_features_variant_overlaps"], [8, 3, 1, "id17", "label"], [8, 2, 1, "id22", "motifs"], [8, 3, 1, "id18", "protein_features"], [8, 3, 1, "id16", "protein_id"], [8, 2, 1, "id21", "regions"], [8, 2, 1, "id20", "repeats"]], "genophenocorr.model.TranscriptAnnotation": [[8, 3, 1, "", "gene_id"], [8, 3, 1, "", "hgvsc_id"], [8, 3, 1, "", "is_preferred"], [8, 3, 1, "", "overlapping_exons"], [8, 3, 1, "", "protein_affected"], [8, 3, 1, "", "protein_effect_location"], [8, 3, 1, "", "transcript_id"], [8, 3, 1, "", "variant_effects"]], "genophenocorr.model.TranscriptCoordinates": [[8, 3, 1, "", "cds_end"], [8, 3, 1, "", "cds_start"], [8, 3, 1, "", "exons"], [8, 2, 1, "", "get_cds_regions"], [8, 2, 1, "", "get_coding_base_count"], [8, 2, 1, "", "get_codon_count"], [8, 2, 1, "", "get_five_prime_utrs"], [8, 2, 1, "", "get_three_prime_utrs"], [8, 3, 1, "", "identifier"], [8, 2, 1, "", "is_coding"], [8, 3, 1, "", "region"]], "genophenocorr.model.TranscriptInfoAware": [[8, 3, 1, "", "gene_id"], [8, 3, 1, "", "transcript_id"]], "genophenocorr.model.Variant": [[8, 2, 1, "", "create_variant_from_scratch"], [8, 3, 1, "", "genotypes"], [8, 3, 1, "id7", "tx_annotations"], [8, 5, 1, "", "variant_class"], [8, 3, 1, "id6", "variant_coordinates"], [8, 5, 1, "", "variant_string"]], "genophenocorr.model.VariantCoordinates": [[8, 3, 1, "id10", "alt"], [8, 3, 1, "id11", "change_length"], [8, 3, 1, "", "chrom"], [8, 3, 1, "", "end"], [8, 2, 1, "", "is_structural"], [8, 3, 1, "id9", "ref"], [8, 3, 1, "id8", "region"], [8, 3, 1, "", "start"], [8, 3, 1, "", "variant_class"], [8, 3, 1, "", "variant_key"]], "genophenocorr.model.VariantEffect": [[8, 5, 1, "", "CODING_SEQUENCE_VARIANT"], [8, 5, 1, "", "DOWNSTREAM_GENE_VARIANT"], [8, 5, 1, "", "FEATURE_ELONGATION"], [8, 5, 1, "", "FEATURE_TRUNCATION"], [8, 5, 1, "", "FIVE_PRIME_UTR_VARIANT"], [8, 5, 1, "", "FRAMESHIFT_VARIANT"], [8, 5, 1, "", "INCOMPLETE_TERMINAL_CODON_VARIANT"], [8, 5, 1, "", "INFRAME_DELETION"], [8, 5, 1, "", "INFRAME_INSERTION"], [8, 5, 1, "", "INTERGENIC_VARIANT"], [8, 5, 1, "", "INTRON_VARIANT"], [8, 5, 1, "", "MATURE_MIRNA_VARIANT"], [8, 5, 1, "", "MISSENSE_VARIANT"], [8, 5, 1, "", "NMD_TRANSCRIPT_VARIANT"], [8, 5, 1, "", "NON_CODING_TRANSCRIPT_EXON_VARIANT"], [8, 5, 1, "", "NON_CODING_TRANSCRIPT_VARIANT"], [8, 5, 1, "", "PROTEIN_ALTERING_VARIANT"], [8, 5, 1, "", "REGULATORY_REGION_ABLATION"], [8, 5, 1, "", "REGULATORY_REGION_AMPLIFICATION"], [8, 5, 1, "", "REGULATORY_REGION_VARIANT"], [8, 5, 1, "", "SEQUENCE_VARIANT"], [8, 5, 1, "", "SPLICE_ACCEPTOR_VARIANT"], [8, 5, 1, "", "SPLICE_DONOR_5TH_BASE_VARIANT"], [8, 5, 1, "", "SPLICE_DONOR_REGION_VARIANT"], [8, 5, 1, "", "SPLICE_DONOR_VARIANT"], [8, 5, 1, "", "SPLICE_POLYPYRIMIDINE_TRACT_VARIANT"], [8, 5, 1, "", "SPLICE_REGION_VARIANT"], [8, 5, 1, "", "START_LOST"], [8, 5, 1, "", "START_RETAINED_VARIANT"], [8, 5, 1, "", "STOP_GAINED"], [8, 5, 1, "", "STOP_LOST"], [8, 5, 1, "", "STOP_RETAINED_VARIANT"], [8, 5, 1, "", "SYNONYMOUS_VARIANT"], [8, 5, 1, "", "TFBS_ABLATION"], [8, 5, 1, "", "TFBS_AMPLIFICATION"], [8, 5, 1, "", "TF_BINDING_SITE_VARIANT"], [8, 5, 1, "", "THREE_PRIME_UTR_VARIANT"], [8, 5, 1, "", "TRANSCRIPT_ABLATION"], [8, 5, 1, "", "TRANSCRIPT_AMPLIFICATION"], [8, 5, 1, "", "UPSTREAM_GENE_VARIANT"], [8, 3, 1, "", "curie"]], "genophenocorr.preprocessing": [[10, 1, 1, "", "FunctionalAnnotator"], [10, 1, 1, "", "PatientCreator"], [10, 1, 1, "", "PhenopacketPatientCreator"], [10, 1, 1, "", "PhenopacketVariantCoordinateFinder"], [10, 1, 1, "", "PhenotypeCreator"], [10, 6, 1, "", "PhenotypeValidationException"], [10, 1, 1, "", "ProtCachingFunctionalAnnotator"], [10, 1, 1, "", "ProteinAnnotationCache"], [10, 1, 1, "", "ProteinMetadataService"], [10, 1, 1, "", "UniprotProteinMetadataService"], [10, 1, 1, "", "VVHgvsVariantCoordinateFinder"], [10, 1, 1, "", "VVTranscriptCoordinateService"], [10, 1, 1, "", "VarCachingFunctionalAnnotator"], [10, 1, 1, "", "VariantAnnotationCache"], [10, 1, 1, "", "VariantCoordinateFinder"], [10, 1, 1, "", "VepFunctionalAnnotator"], [10, 4, 1, "", "configure_caching_patient_creator"], [10, 4, 1, "", "configure_patient_creator"], [10, 4, 1, "", "load_phenopacket"], [10, 4, 1, "", "load_phenopacket_folder"]], "genophenocorr.preprocessing.FunctionalAnnotator": [[10, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.PatientCreator": [[10, 2, 1, "", "create_patient"]], "genophenocorr.preprocessing.PhenopacketPatientCreator": [[10, 2, 1, "", "create_patient"]], "genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder": [[10, 2, 1, "", "find_coordinates"]], "genophenocorr.preprocessing.PhenotypeCreator": [[10, 2, 1, "", "create_phenotype"]], "genophenocorr.preprocessing.PhenotypeValidationException": [[10, 3, 1, "", "issues"]], "genophenocorr.preprocessing.ProtCachingFunctionalAnnotator": [[10, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.ProteinAnnotationCache": [[10, 2, 1, "", "get_annotations"], [10, 2, 1, "", "store_annotations"]], "genophenocorr.preprocessing.ProteinMetadataService": [[10, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.UniprotProteinMetadataService": [[10, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.VVHgvsVariantCoordinateFinder": [[10, 2, 1, "", "find_coordinates"]], "genophenocorr.preprocessing.VVTranscriptCoordinateService": [[10, 2, 1, "", "fetch"], [10, 2, 1, "", "parse_response"]], "genophenocorr.preprocessing.VarCachingFunctionalAnnotator": [[10, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.VariantAnnotationCache": [[10, 2, 1, "", "get_annotations"], [10, 2, 1, "", "store_annotations"]], "genophenocorr.preprocessing.VariantCoordinateFinder": [[10, 2, 1, "", "find_coordinates"]], "genophenocorr.preprocessing.VepFunctionalAnnotator": [[10, 2, 1, "", "annotate"]], "genophenocorr.view": [[11, 1, 1, "", "CohortViewer"], [11, 1, 1, "", "VariantTranscriptVisualizer"]], "genophenocorr.view.CohortViewer": [[11, 2, 1, "", "anchor"], [11, 2, 1, "", "cohort_summary_table"], [11, 2, 1, "", "hpo_anchor"], [11, 2, 1, "", "hpo_term_counts_table"], [11, 2, 1, "", "html_row"], [11, 2, 1, "", "protein_features_table"], [11, 2, 1, "", "variants_table"]], "genophenocorr.view.VariantTranscriptVisualizer": [[11, 2, 1, "", "draw_variants"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:property", "4": "py:function", "5": "py:attribute", "6": "py:exception"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "property", "Python property"], "4": ["py", "function", "Python function"], "5": ["py", "attribute", "Python attribute"], "6": ["py", "exception", "Python exception"]}, "titleterms": {"genophenocorr": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 14], "packag": [0, 1, 2, 3, 4, 5, 7, 8, 9, 10, 11], "subpackag": [0, 1], "analysi": [1, 2, 3, 4, 16, 17], "default": 1, "valu": 1, "predic": [2, 3, 4], "genotyp": [3, 13], "phenotyp": [4, 13], "constant": [5, 6], "submodul": 5, "variant_effect": 6, "modul": 6, "data": [7, 19], "model": [8, 9], "genom": 9, "preprocess": 10, "view": 11, "api": 12, "refer": 12, "correl": 13, "autosom": 13, "recess": 13, "diseas": 13, "tutori": [13, 16], "literatur": 14, "feedback": 14, "instal": 15, "stabl": 15, "releas": 15, "latest": 15, "run": 15, "test": 15, "set": 16, "up": 16, "prepar": 16, "sampl": 16, "cohort": [17, 19], "exploratori": 17, "user": 18, "guid": 18, "content": 18, "input": 19, "creat": 19, "from": 19, "ga4gh": 19, "phenopacket": 19, "other": 19}, "envversion": {"sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.viewcode": 1, "sphinx.ext.intersphinx": 1, "sphinx": 60}, "alltitles": {"genophenocorr package": [[0, "module-genophenocorr"]], "Subpackages": [[0, "subpackages"], [1, "subpackages"]], "genophenocorr.analysis package": [[1, "module-genophenocorr.analysis"]], "Default values": [[1, "default-values"]], "genophenocorr.analysis.predicate package": [[2, "module-genophenocorr.analysis.predicate"]], "genophenocorr.analysis.predicate.genotype package": [[3, "module-genophenocorr.analysis.predicate.genotype"]], "genophenocorr.analysis.predicate.phenotype package": [[4, "module-genophenocorr.analysis.predicate.phenotype"]], "genophenocorr.constants package": [[5, "module-genophenocorr.constants"]], "Submodules": [[5, "submodules"]], "genophenocorr.constants.variant_effects module": [[6, "genophenocorr-constants-variant-effects-module"]], "genophenocorr.data package": [[7, "module-genophenocorr.data"]], "genophenocorr.model package": [[8, "module-genophenocorr.model"]], "genophenocorr.model.genome package": [[9, "module-genophenocorr.model.genome"]], "genophenocorr.preprocessing package": [[10, "module-genophenocorr.preprocessing"]], "genophenocorr.view package": [[11, "module-genophenocorr.view"]], "API reference": [[12, "api-reference"]], "Genotype-Phenotype Correlations in Autosomal Recessive Diseases": [[13, "genotype-phenotype-correlations-in-autosomal-recessive-diseases"]], "Tutorial": [[13, "tutorial"], [16, "tutorial"]], "genophenocorr": [[14, "genophenocorr"]], "Literature": [[14, "literature"]], "Feedback": [[14, "feedback"]], "Installation": [[15, "installation"]], "Stable release": [[15, "stable-release"]], "Latest release": [[15, "latest-release"]], "Run tests": [[15, "run-tests"]], "Set up analysis": [[16, "set-up-analysis"]], "Prepare samples": [[16, "prepare-samples"]], "Cohort exploratory analysis": [[17, "cohort-exploratory-analysis"]], "User guide": [[18, "user-guide"]], "Contents:": [[18, null]], "Input data": [[19, "input-data"]], "Create a cohort from GA4GH phenopackets": [[19, "create-a-cohort-from-ga4gh-phenopackets"]], "Create a cohort from other data": [[19, "create-a-cohort-from-other-data"]]}, "indexentries": {"genophenocorr": [[0, "module-genophenocorr"]], "module": [[0, "module-genophenocorr"], [1, "module-genophenocorr.analysis"], [2, "module-genophenocorr.analysis.predicate"], [3, "module-genophenocorr.analysis.predicate.genotype"], [4, "module-genophenocorr.analysis.predicate.phenotype"], [5, "module-genophenocorr.constants"], [7, "module-genophenocorr.data"], [8, "module-genophenocorr.model"], [9, "module-genophenocorr.model.genome"], [10, "module-genophenocorr.preprocessing"], [11, "module-genophenocorr.view"]], "cohortanalysis (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.CohortAnalysis"]], "cohortanalysisconfiguration (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration"]], "cohortanalysisconfigurationbuilder (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.CohortAnalysisConfigurationBuilder"]], "genotypephenotypeanalysisresult (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult"]], "all_counts (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.all_counts"]], "build() (genophenocorr.analysis.cohortanalysisconfigurationbuilder method)": [[1, "genophenocorr.analysis.CohortAnalysisConfigurationBuilder.build"]], "builder() (genophenocorr.analysis.cohortanalysisconfiguration static method)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.builder"]], "compare_by_exon() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_exon"]], "compare_by_protein_feature() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature"]], "compare_by_protein_feature_type() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature_type"]], "compare_by_variant_effect() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_variant_effect"]], "compare_by_variant_key() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_variant_key"]], "compare_by_variant_keys() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_variant_keys"]], "configure_cohort_analysis() (in module genophenocorr.analysis)": [[1, "genophenocorr.analysis.configure_cohort_analysis"]], "corrected_pvals (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.corrected_pvals"]], "genophenocorr.analysis": [[1, "module-genophenocorr.analysis"]], "include_sv (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.include_sv"]], "include_sv() (genophenocorr.analysis.cohortanalysisconfigurationbuilder method)": [[1, "genophenocorr.analysis.CohortAnalysisConfigurationBuilder.include_sv"]], "min_perc_patients_w_hpo (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.min_perc_patients_w_hpo"]], "min_perc_patients_w_hpo() (genophenocorr.analysis.cohortanalysisconfigurationbuilder method)": [[1, "genophenocorr.analysis.CohortAnalysisConfigurationBuilder.min_perc_patients_w_hpo"]], "missing_implies_excluded (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.missing_implies_excluded"]], "missing_implies_excluded() (genophenocorr.analysis.cohortanalysisconfigurationbuilder method)": [[1, "genophenocorr.analysis.CohortAnalysisConfigurationBuilder.missing_implies_excluded"]], "n_usable (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.n_usable"]], "phenotype_categories (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.phenotype_categories"]], "pval_correction (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.pval_correction"]], "pval_correction() (genophenocorr.analysis.cohortanalysisconfigurationbuilder method)": [[1, "genophenocorr.analysis.CohortAnalysisConfigurationBuilder.pval_correction"]], "pvals (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.pvals"]], "summarize() (genophenocorr.analysis.genotypephenotypeanalysisresult method)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.summarize"]], "booleanpredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.BooleanPredicate"]], "first (genophenocorr.analysis.predicate.groupingpredicate attribute)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate.FIRST"]], "groupingpredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate"]], "no (genophenocorr.analysis.predicate.booleanpredicate attribute)": [[2, "genophenocorr.analysis.predicate.BooleanPredicate.NO"]], "patientcategory (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.PatientCategory"]], "polypredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.PolyPredicate"]], "second (genophenocorr.analysis.predicate.groupingpredicate attribute)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate.SECOND"]], "yes (genophenocorr.analysis.predicate.booleanpredicate attribute)": [[2, "genophenocorr.analysis.predicate.BooleanPredicate.YES"]], "cat_id (genophenocorr.analysis.predicate.patientcategory property)": [[2, "genophenocorr.analysis.predicate.PatientCategory.cat_id"]], "description (genophenocorr.analysis.predicate.patientcategory property)": [[2, "genophenocorr.analysis.predicate.PatientCategory.description"]], "genophenocorr.analysis.predicate": [[2, "module-genophenocorr.analysis.predicate"]], "get_categories() (genophenocorr.analysis.predicate.booleanpredicate static method)": [[2, "genophenocorr.analysis.predicate.BooleanPredicate.get_categories"]], "get_categories() (genophenocorr.analysis.predicate.groupingpredicate static method)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate.get_categories"]], "get_categories() (genophenocorr.analysis.predicate.polypredicate static method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.get_categories"]], "get_question() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.get_question"]], "n_categories() (genophenocorr.analysis.predicate.polypredicate class method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.n_categories"]], "name (genophenocorr.analysis.predicate.patientcategory property)": [[2, "genophenocorr.analysis.predicate.PatientCategory.name"]], "test() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.test"]], "exonpredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ExonPredicate"]], "exonspredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ExonsPredicate"]], "protfeaturepredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate"]], "protfeaturetypepredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate"]], "protfeaturetypespredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate"]], "protfeaturespredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate"]], "varianteffectpredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectPredicate"]], "varianteffectspredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate"]], "variantpredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicate"]], "variantspredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.VariantsPredicate"]], "genophenocorr.analysis.predicate.genotype": [[3, "module-genophenocorr.analysis.predicate.genotype"]], "get_question() (genophenocorr.analysis.predicate.genotype.exonpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ExonPredicate.get_question"]], "get_question() (genophenocorr.analysis.predicate.genotype.exonspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ExonsPredicate.get_question"]], "get_question() (genophenocorr.analysis.predicate.genotype.protfeaturepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate.get_question"]], "get_question() (genophenocorr.analysis.predicate.genotype.protfeaturetypepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate.get_question"]], "get_question() (genophenocorr.analysis.predicate.genotype.protfeaturetypespredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate.get_question"]], "get_question() (genophenocorr.analysis.predicate.genotype.protfeaturespredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate.get_question"]], "get_question() (genophenocorr.analysis.predicate.genotype.varianteffectpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectPredicate.get_question"]], "get_question() (genophenocorr.analysis.predicate.genotype.varianteffectspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate.get_question"]], "get_question() (genophenocorr.analysis.predicate.genotype.variantpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicate.get_question"]], "get_question() (genophenocorr.analysis.predicate.genotype.variantspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantsPredicate.get_question"]], "test() (genophenocorr.analysis.predicate.genotype.exonpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ExonPredicate.test"]], "test() (genophenocorr.analysis.predicate.genotype.exonspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ExonsPredicate.test"]], "test() (genophenocorr.analysis.predicate.genotype.protfeaturepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate.test"]], "test() (genophenocorr.analysis.predicate.genotype.protfeaturetypepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate.test"]], "test() (genophenocorr.analysis.predicate.genotype.protfeaturetypespredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate.test"]], "test() (genophenocorr.analysis.predicate.genotype.protfeaturespredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate.test"]], "test() (genophenocorr.analysis.predicate.genotype.varianteffectpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectPredicate.test"]], "test() (genophenocorr.analysis.predicate.genotype.varianteffectspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate.test"]], "test() (genophenocorr.analysis.predicate.genotype.variantpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicate.test"]], "test() (genophenocorr.analysis.predicate.genotype.variantspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantsPredicate.test"]], "excluded (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate attribute)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.EXCLUDED"]], "not_measured (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate attribute)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.NOT_MEASURED"]], "present (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate attribute)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.PRESENT"]], "phenotypepredicatefactory (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypePredicateFactory"]], "propagatingphenotypebooleanpredicate (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypeBooleanPredicate"]], "propagatingphenotypebooleanpredicatefactory (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypeBooleanPredicateFactory"]], "propagatingphenotypepredicate (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate"]], "genophenocorr.analysis.predicate.phenotype": [[4, "module-genophenocorr.analysis.predicate.phenotype"]], "get_categories() (genophenocorr.analysis.predicate.phenotype.phenotypepredicatefactory static method)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypePredicateFactory.get_categories"]], "get_categories() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypebooleanpredicatefactory static method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypeBooleanPredicateFactory.get_categories"]], "get_categories() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate static method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.get_categories"]], "get_predicate() (genophenocorr.analysis.predicate.phenotype.phenotypepredicatefactory method)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypePredicateFactory.get_predicate"]], "get_predicate() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypebooleanpredicatefactory method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypeBooleanPredicateFactory.get_predicate"]], "get_question() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypebooleanpredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypeBooleanPredicate.get_question"]], "get_question() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.get_question"]], "test() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypebooleanpredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypeBooleanPredicate.test"]], "test() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.test"]], "genophenocorr.constants": [[5, "module-genophenocorr.constants"]], "genophenocorr.data": [[7, "module-genophenocorr.data"]], "get_toy_cohort() (in module genophenocorr.data)": [[7, "genophenocorr.data.get_toy_cohort"]], "coding_sequence_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.CODING_SEQUENCE_VARIANT"]], "cohort (class in genophenocorr.model)": [[8, "genophenocorr.model.Cohort"]], "domain (genophenocorr.model.featuretype attribute)": [[8, "genophenocorr.model.FeatureType.DOMAIN"], [8, "id27"]], "downstream_gene_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.DOWNSTREAM_GENE_VARIANT"]], "feature_elongation (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.FEATURE_ELONGATION"]], "feature_truncation (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.FEATURE_TRUNCATION"]], "five_prime_utr_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.FIVE_PRIME_UTR_VARIANT"]], "frameshift_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.FRAMESHIFT_VARIANT"]], "featureinfo (class in genophenocorr.model)": [[8, "genophenocorr.model.FeatureInfo"]], "featuretype (class in genophenocorr.model)": [[8, "genophenocorr.model.FeatureType"]], "genotype (class in genophenocorr.model)": [[8, "genophenocorr.model.Genotype"]], "genotyped (class in genophenocorr.model)": [[8, "genophenocorr.model.Genotyped"]], "genotypes (class in genophenocorr.model)": [[8, "genophenocorr.model.Genotypes"]], "hemizygous (genophenocorr.model.genotype attribute)": [[8, "genophenocorr.model.Genotype.HEMIZYGOUS"]], "heterozygous (genophenocorr.model.genotype attribute)": [[8, "genophenocorr.model.Genotype.HETEROZYGOUS"]], "homozygous_alternate (genophenocorr.model.genotype attribute)": [[8, "genophenocorr.model.Genotype.HOMOZYGOUS_ALTERNATE"]], "homozygous_reference (genophenocorr.model.genotype attribute)": [[8, "genophenocorr.model.Genotype.HOMOZYGOUS_REFERENCE"]], "incomplete_terminal_codon_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.INCOMPLETE_TERMINAL_CODON_VARIANT"]], "inframe_deletion (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.INFRAME_DELETION"]], "inframe_insertion (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.INFRAME_INSERTION"]], "intergenic_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.INTERGENIC_VARIANT"]], "intron_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.INTRON_VARIANT"]], "mature_mirna_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.MATURE_MIRNA_VARIANT"]], "missense_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.MISSENSE_VARIANT"]], "motif (genophenocorr.model.featuretype attribute)": [[8, "genophenocorr.model.FeatureType.MOTIF"], [8, "id26"]], "nmd_transcript_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.NMD_TRANSCRIPT_VARIANT"]], "non_coding_transcript_exon_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.NON_CODING_TRANSCRIPT_EXON_VARIANT"]], "non_coding_transcript_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.NON_CODING_TRANSCRIPT_VARIANT"]], "no_call (genophenocorr.model.genotype attribute)": [[8, "genophenocorr.model.Genotype.NO_CALL"]], "protein_altering_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.PROTEIN_ALTERING_VARIANT"]], "patient (class in genophenocorr.model)": [[8, "genophenocorr.model.Patient"]], "phenotype (class in genophenocorr.model)": [[8, "genophenocorr.model.Phenotype"]], "proteinfeature (class in genophenocorr.model)": [[8, "genophenocorr.model.ProteinFeature"]], "proteinmetadata (class in genophenocorr.model)": [[8, "genophenocorr.model.ProteinMetadata"]], "region (genophenocorr.model.featuretype attribute)": [[8, "genophenocorr.model.FeatureType.REGION"], [8, "id28"]], "regulatory_region_ablation (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.REGULATORY_REGION_ABLATION"]], "regulatory_region_amplification (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.REGULATORY_REGION_AMPLIFICATION"]], "regulatory_region_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.REGULATORY_REGION_VARIANT"]], "repeat (genophenocorr.model.featuretype attribute)": [[8, "genophenocorr.model.FeatureType.REPEAT"], [8, "id25"]], "sequence_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SEQUENCE_VARIANT"]], "splice_acceptor_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_ACCEPTOR_VARIANT"]], "splice_donor_5th_base_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_DONOR_5TH_BASE_VARIANT"]], "splice_donor_region_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_DONOR_REGION_VARIANT"]], "splice_donor_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_DONOR_VARIANT"]], "splice_polypyrimidine_tract_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_POLYPYRIMIDINE_TRACT_VARIANT"]], "splice_region_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_REGION_VARIANT"]], "start_lost (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.START_LOST"]], "start_retained_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.START_RETAINED_VARIANT"]], "stop_gained (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.STOP_GAINED"]], "stop_lost (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.STOP_LOST"]], "stop_retained_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.STOP_RETAINED_VARIANT"]], "synonymous_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SYNONYMOUS_VARIANT"]], "tfbs_ablation (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.TFBS_ABLATION"]], "tfbs_amplification (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.TFBS_AMPLIFICATION"]], "tf_binding_site_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.TF_BINDING_SITE_VARIANT"]], "three_prime_utr_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.THREE_PRIME_UTR_VARIANT"]], "transcript_ablation (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.TRANSCRIPT_ABLATION"]], "transcript_amplification (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.TRANSCRIPT_AMPLIFICATION"]], "transcriptannotation (class in genophenocorr.model)": [[8, "genophenocorr.model.TranscriptAnnotation"]], "transcriptcoordinates (class in genophenocorr.model)": [[8, "genophenocorr.model.TranscriptCoordinates"]], "transcriptinfoaware (class in genophenocorr.model)": [[8, "genophenocorr.model.TranscriptInfoAware"]], "upstream_gene_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.UPSTREAM_GENE_VARIANT"]], "variant (class in genophenocorr.model)": [[8, "genophenocorr.model.Variant"]], "variantcoordinates (class in genophenocorr.model)": [[8, "genophenocorr.model.VariantCoordinates"]], "varianteffect (class in genophenocorr.model)": [[8, "genophenocorr.model.VariantEffect"]], "all_excluded_patients (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_excluded_patients"]], "all_patients (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_patients"]], "all_phenotypes (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_phenotypes"]], "all_proteins (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_proteins"]], "all_transcripts (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_transcripts"]], "all_variants (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_variants"]], "alt (genophenocorr.model.variantcoordinates attribute)": [[8, "genophenocorr.model.VariantCoordinates.alt"]], "alt (genophenocorr.model.variantcoordinates property)": [[8, "id10"]], "cds_end (genophenocorr.model.transcriptcoordinates property)": [[8, "genophenocorr.model.TranscriptCoordinates.cds_end"]], "cds_start (genophenocorr.model.transcriptcoordinates property)": [[8, "genophenocorr.model.TranscriptCoordinates.cds_start"]], "change_length (genophenocorr.model.variantcoordinates attribute)": [[8, "genophenocorr.model.VariantCoordinates.change_length"]], "change_length (genophenocorr.model.variantcoordinates property)": [[8, "id11"]], "chrom (genophenocorr.model.variantcoordinates property)": [[8, "genophenocorr.model.VariantCoordinates.chrom"]], "code (genophenocorr.model.genotype property)": [[8, "genophenocorr.model.Genotype.code"]], "create() (genophenocorr.model.proteinfeature static method)": [[8, "genophenocorr.model.ProteinFeature.create"]], "create_variant_from_scratch() (genophenocorr.model.variant static method)": [[8, "genophenocorr.model.Variant.create_variant_from_scratch"]], "curie (genophenocorr.model.varianteffect property)": [[8, "genophenocorr.model.VariantEffect.curie"]], "domains() (genophenocorr.model.proteinmetadata method)": [[8, "genophenocorr.model.ProteinMetadata.domains"], [8, "id19"]], "empty() (genophenocorr.model.genotypes static method)": [[8, "genophenocorr.model.Genotypes.empty"]], "end (genophenocorr.model.featureinfo property)": [[8, "genophenocorr.model.FeatureInfo.end"]], "end (genophenocorr.model.variantcoordinates property)": [[8, "genophenocorr.model.VariantCoordinates.end"]], "excluded_phenotypes() (genophenocorr.model.patient method)": [[8, "genophenocorr.model.Patient.excluded_phenotypes"]], "exons (genophenocorr.model.transcriptcoordinates property)": [[8, "genophenocorr.model.TranscriptCoordinates.exons"]], "feature_type (genophenocorr.model.proteinfeature property)": [[8, "genophenocorr.model.ProteinFeature.feature_type"]], "for_sample() (genophenocorr.model.genotypes method)": [[8, "genophenocorr.model.Genotypes.for_sample"]], "from_mapping() (genophenocorr.model.genotypes static method)": [[8, "genophenocorr.model.Genotypes.from_mapping"]], "from_patients() (genophenocorr.model.cohort static method)": [[8, "genophenocorr.model.Cohort.from_patients"]], "from_term() (genophenocorr.model.phenotype static method)": [[8, "genophenocorr.model.Phenotype.from_term"]], "gene_id (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.gene_id"]], "gene_id (genophenocorr.model.transcriptinfoaware property)": [[8, "genophenocorr.model.TranscriptInfoAware.gene_id"]], "genophenocorr.model": [[8, "module-genophenocorr.model"]], "genotype_for_sample() (genophenocorr.model.genotyped method)": [[8, "genophenocorr.model.Genotyped.genotype_for_sample"]], "genotypes (genophenocorr.model.genotyped property)": [[8, "genophenocorr.model.Genotyped.genotypes"]], "genotypes (genophenocorr.model.variant property)": [[8, "genophenocorr.model.Variant.genotypes"]], "get_cds_regions() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.get_cds_regions"]], "get_coding_base_count() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.get_coding_base_count"]], "get_codon_count() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.get_codon_count"]], "get_excluded_count() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.get_excluded_count"]], "get_excluded_ids() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.get_excluded_ids"]], "get_features_variant_overlaps() (genophenocorr.model.proteinmetadata method)": [[8, "genophenocorr.model.ProteinMetadata.get_features_variant_overlaps"]], "get_five_prime_utrs() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.get_five_prime_utrs"]], "get_protein_features_affected() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.get_protein_features_affected"]], "get_three_prime_utrs() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.get_three_prime_utrs"]], "hgvsc_id (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.hgvsc_id"]], "identifier (genophenocorr.model.phenotype property)": [[8, "genophenocorr.model.Phenotype.identifier"]], "identifier (genophenocorr.model.transcriptcoordinates property)": [[8, "genophenocorr.model.TranscriptCoordinates.identifier"]], "info (genophenocorr.model.proteinfeature property)": [[8, "genophenocorr.model.ProteinFeature.info"]], "is_coding() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.is_coding"]], "is_excluded (genophenocorr.model.phenotype property)": [[8, "genophenocorr.model.Phenotype.is_excluded"]], "is_observed (genophenocorr.model.phenotype attribute)": [[8, "genophenocorr.model.Phenotype.is_observed"]], "is_observed (genophenocorr.model.phenotype property)": [[8, "id5"]], "is_preferred (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.is_preferred"]], "is_structural() (genophenocorr.model.variantcoordinates method)": [[8, "genophenocorr.model.VariantCoordinates.is_structural"]], "label (genophenocorr.model.proteinmetadata attribute)": [[8, "genophenocorr.model.ProteinMetadata.label"]], "label (genophenocorr.model.proteinmetadata property)": [[8, "id17"]], "list_all_patients() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_all_patients"]], "list_all_phenotypes() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_all_phenotypes"]], "list_all_proteins() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_all_proteins"]], "list_all_variants() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_all_variants"]], "list_data_by_tx() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_data_by_tx"]], "motifs() (genophenocorr.model.proteinmetadata method)": [[8, "genophenocorr.model.ProteinMetadata.motifs"], [8, "id22"]], "name (genophenocorr.model.featureinfo attribute)": [[8, "genophenocorr.model.FeatureInfo.name"]], "name (genophenocorr.model.featureinfo property)": [[8, "id23"]], "name (genophenocorr.model.phenotype attribute)": [[8, "genophenocorr.model.Phenotype.name"]], "name (genophenocorr.model.phenotype property)": [[8, "id4"]], "observed (genophenocorr.model.phenotype property)": [[8, "genophenocorr.model.Phenotype.observed"]], "overlapping_exons (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.overlapping_exons"]], "patient_id (genophenocorr.model.patient attribute)": [[8, "genophenocorr.model.Patient.patient_id"]], "patient_id (genophenocorr.model.patient property)": [[8, "id0"]], "phenotypes (genophenocorr.model.patient attribute)": [[8, "genophenocorr.model.Patient.phenotypes"]], "phenotypes (genophenocorr.model.patient property)": [[8, "id1"]], "present_phenotypes() (genophenocorr.model.patient method)": [[8, "genophenocorr.model.Patient.present_phenotypes"]], "protein_affected (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.protein_affected"]], "protein_effect_location (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.protein_effect_location"]], "protein_features (genophenocorr.model.proteinmetadata attribute)": [[8, "genophenocorr.model.ProteinMetadata.protein_features"]], "protein_features (genophenocorr.model.proteinmetadata property)": [[8, "id18"]], "protein_id (genophenocorr.model.proteinmetadata attribute)": [[8, "genophenocorr.model.ProteinMetadata.protein_id"]], "protein_id (genophenocorr.model.proteinmetadata property)": [[8, "id16"]], "proteins (genophenocorr.model.patient attribute)": [[8, "genophenocorr.model.Patient.proteins"]], "proteins (genophenocorr.model.patient property)": [[8, "id3"]], "ref (genophenocorr.model.variantcoordinates attribute)": [[8, "genophenocorr.model.VariantCoordinates.ref"]], "ref (genophenocorr.model.variantcoordinates property)": [[8, "id9"]], "region (genophenocorr.model.featureinfo attribute)": [[8, "genophenocorr.model.FeatureInfo.region"]], "region (genophenocorr.model.featureinfo property)": [[8, "id24"]], "region (genophenocorr.model.transcriptcoordinates property)": [[8, "genophenocorr.model.TranscriptCoordinates.region"]], "region (genophenocorr.model.variantcoordinates attribute)": [[8, "genophenocorr.model.VariantCoordinates.region"]], "region (genophenocorr.model.variantcoordinates property)": [[8, "id8"]], "regions() (genophenocorr.model.proteinmetadata method)": [[8, "genophenocorr.model.ProteinMetadata.regions"], [8, "id21"]], "repeats() (genophenocorr.model.proteinmetadata method)": [[8, "genophenocorr.model.ProteinMetadata.repeats"], [8, "id20"]], "single() (genophenocorr.model.genotypes static method)": [[8, "genophenocorr.model.Genotypes.single"]], "start (genophenocorr.model.featureinfo property)": [[8, "genophenocorr.model.FeatureInfo.start"]], "start (genophenocorr.model.variantcoordinates property)": [[8, "genophenocorr.model.VariantCoordinates.start"]], "term_id (genophenocorr.model.phenotype attribute)": [[8, "genophenocorr.model.Phenotype.term_id"]], "to_string() (genophenocorr.model.proteinfeature method)": [[8, "genophenocorr.model.ProteinFeature.to_string"]], "total_patient_count (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.total_patient_count"]], "transcript_id (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.transcript_id"]], "transcript_id (genophenocorr.model.transcriptinfoaware property)": [[8, "genophenocorr.model.TranscriptInfoAware.transcript_id"]], "tx_annotations (genophenocorr.model.variant attribute)": [[8, "genophenocorr.model.Variant.tx_annotations"]], "tx_annotations (genophenocorr.model.variant property)": [[8, "id7"]], "variant_class (genophenocorr.model.variant attribute)": [[8, "genophenocorr.model.Variant.variant_class"]], "variant_class (genophenocorr.model.variantcoordinates property)": [[8, "genophenocorr.model.VariantCoordinates.variant_class"]], "variant_coordinates (genophenocorr.model.variant attribute)": [[8, "genophenocorr.model.Variant.variant_coordinates"]], "variant_coordinates (genophenocorr.model.variant property)": [[8, "id6"]], "variant_effects (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.variant_effects"]], "variant_key (genophenocorr.model.variantcoordinates property)": [[8, "genophenocorr.model.VariantCoordinates.variant_key"]], "variant_string (genophenocorr.model.variant attribute)": [[8, "genophenocorr.model.Variant.variant_string"]], "variants (genophenocorr.model.patient attribute)": [[8, "genophenocorr.model.Patient.variants"]], "variants (genophenocorr.model.patient property)": [[8, "id2"]], "genophenocorr.model.genome": [[9, "module-genophenocorr.model.genome"]], "functionalannotator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.FunctionalAnnotator"]], "patientcreator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.PatientCreator"]], "phenopacketpatientcreator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.PhenopacketPatientCreator"]], "phenopacketvariantcoordinatefinder (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder"]], "phenotypecreator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.PhenotypeCreator"]], "phenotypevalidationexception": [[10, "genophenocorr.preprocessing.PhenotypeValidationException"]], "protcachingfunctionalannotator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.ProtCachingFunctionalAnnotator"]], "proteinannotationcache (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.ProteinAnnotationCache"]], "proteinmetadataservice (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.ProteinMetadataService"]], "uniprotproteinmetadataservice (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.UniprotProteinMetadataService"]], "vvhgvsvariantcoordinatefinder (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VVHgvsVariantCoordinateFinder"]], "vvtranscriptcoordinateservice (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VVTranscriptCoordinateService"]], "varcachingfunctionalannotator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VarCachingFunctionalAnnotator"]], "variantannotationcache (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VariantAnnotationCache"]], "variantcoordinatefinder (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VariantCoordinateFinder"]], "vepfunctionalannotator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VepFunctionalAnnotator"]], "annotate() (genophenocorr.preprocessing.functionalannotator method)": [[10, "genophenocorr.preprocessing.FunctionalAnnotator.annotate"]], "annotate() (genophenocorr.preprocessing.protcachingfunctionalannotator method)": [[10, "genophenocorr.preprocessing.ProtCachingFunctionalAnnotator.annotate"]], "annotate() (genophenocorr.preprocessing.proteinmetadataservice method)": [[10, "genophenocorr.preprocessing.ProteinMetadataService.annotate"]], "annotate() (genophenocorr.preprocessing.uniprotproteinmetadataservice method)": [[10, "genophenocorr.preprocessing.UniprotProteinMetadataService.annotate"]], "annotate() (genophenocorr.preprocessing.varcachingfunctionalannotator method)": [[10, "genophenocorr.preprocessing.VarCachingFunctionalAnnotator.annotate"]], "annotate() (genophenocorr.preprocessing.vepfunctionalannotator method)": [[10, "genophenocorr.preprocessing.VepFunctionalAnnotator.annotate"]], "configure_caching_patient_creator() (in module genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.configure_caching_patient_creator"]], "configure_patient_creator() (in module genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.configure_patient_creator"]], "create_patient() (genophenocorr.preprocessing.patientcreator method)": [[10, "genophenocorr.preprocessing.PatientCreator.create_patient"]], "create_patient() (genophenocorr.preprocessing.phenopacketpatientcreator method)": [[10, "genophenocorr.preprocessing.PhenopacketPatientCreator.create_patient"]], "create_phenotype() (genophenocorr.preprocessing.phenotypecreator method)": [[10, "genophenocorr.preprocessing.PhenotypeCreator.create_phenotype"]], "fetch() (genophenocorr.preprocessing.vvtranscriptcoordinateservice method)": [[10, "genophenocorr.preprocessing.VVTranscriptCoordinateService.fetch"]], "find_coordinates() (genophenocorr.preprocessing.phenopacketvariantcoordinatefinder method)": [[10, "genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder.find_coordinates"]], "find_coordinates() (genophenocorr.preprocessing.vvhgvsvariantcoordinatefinder method)": [[10, "genophenocorr.preprocessing.VVHgvsVariantCoordinateFinder.find_coordinates"]], "find_coordinates() (genophenocorr.preprocessing.variantcoordinatefinder method)": [[10, "genophenocorr.preprocessing.VariantCoordinateFinder.find_coordinates"]], "genophenocorr.preprocessing": [[10, "module-genophenocorr.preprocessing"]], "get_annotations() (genophenocorr.preprocessing.proteinannotationcache method)": [[10, "genophenocorr.preprocessing.ProteinAnnotationCache.get_annotations"]], "get_annotations() (genophenocorr.preprocessing.variantannotationcache method)": [[10, "genophenocorr.preprocessing.VariantAnnotationCache.get_annotations"]], "issues (genophenocorr.preprocessing.phenotypevalidationexception property)": [[10, "genophenocorr.preprocessing.PhenotypeValidationException.issues"]], "load_phenopacket() (in module genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.load_phenopacket"]], "load_phenopacket_folder() (in module genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.load_phenopacket_folder"]], "parse_response() (genophenocorr.preprocessing.vvtranscriptcoordinateservice method)": [[10, "genophenocorr.preprocessing.VVTranscriptCoordinateService.parse_response"]], "store_annotations() (genophenocorr.preprocessing.proteinannotationcache method)": [[10, "genophenocorr.preprocessing.ProteinAnnotationCache.store_annotations"]], "store_annotations() (genophenocorr.preprocessing.variantannotationcache method)": [[10, "genophenocorr.preprocessing.VariantAnnotationCache.store_annotations"]], "cohortviewer (class in genophenocorr.view)": [[11, "genophenocorr.view.CohortViewer"]], "varianttranscriptvisualizer (class in genophenocorr.view)": [[11, "genophenocorr.view.VariantTranscriptVisualizer"]], "anchor() (genophenocorr.view.cohortviewer static method)": [[11, "genophenocorr.view.CohortViewer.anchor"]], "cohort_summary_table() (genophenocorr.view.cohortviewer method)": [[11, "genophenocorr.view.CohortViewer.cohort_summary_table"]], "draw_variants() (genophenocorr.view.varianttranscriptvisualizer method)": [[11, "genophenocorr.view.VariantTranscriptVisualizer.draw_variants"]], "genophenocorr.view": [[11, "module-genophenocorr.view"]], "hpo_anchor() (genophenocorr.view.cohortviewer static method)": [[11, "genophenocorr.view.CohortViewer.hpo_anchor"]], "hpo_term_counts_table() (genophenocorr.view.cohortviewer method)": [[11, "genophenocorr.view.CohortViewer.hpo_term_counts_table"]], "html_row() (genophenocorr.view.cohortviewer static method)": [[11, "genophenocorr.view.CohortViewer.html_row"]], "protein_features_table() (genophenocorr.view.cohortviewer method)": [[11, "genophenocorr.view.CohortViewer.protein_features_table"]], "variants_table() (genophenocorr.view.cohortviewer method)": [[11, "genophenocorr.view.CohortViewer.variants_table"]]}})