Search.setIndex({"alltitles": {"API reference": [[12, "api-reference"]], "Cohort exploratory analysis": [[17, "cohort-exploratory-analysis"]], "Contents:": [[18, null]], "Create a cohort from GA4GH phenopackets": [[19, "create-a-cohort-from-ga4gh-phenopackets"]], "Create a cohort from other data": [[19, "create-a-cohort-from-other-data"]], "Default values": [[1, "default-values"]], "Feedback": [[14, "feedback"]], "Genotype-Phenotype Correlations in Autosomal Recessive Diseases": [[13, "genotype-phenotype-correlations-in-autosomal-recessive-diseases"]], "Input data": [[19, "input-data"]], "Installation": [[15, "installation"]], "Latest release": [[15, "latest-release"]], "Literature": [[14, "literature"]], "Prepare samples": [[16, "prepare-samples"]], "Run tests": [[15, "run-tests"]], "Set up analysis": [[16, "set-up-analysis"]], "Stable release": [[15, "stable-release"]], "Submodules": [[5, "submodules"]], "Subpackages": [[0, "subpackages"], [1, "subpackages"]], "Tutorial": [[13, "tutorial"], [16, "tutorial"]], "User guide": [[18, "user-guide"]], "genophenocorr": [[14, "genophenocorr"]], "genophenocorr package": [[0, "module-genophenocorr"]], "genophenocorr.analysis package": [[1, "module-genophenocorr.analysis"]], "genophenocorr.analysis.predicate package": [[2, "module-genophenocorr.analysis.predicate"]], "genophenocorr.analysis.predicate.genotype package": [[3, "module-genophenocorr.analysis.predicate.genotype"]], "genophenocorr.analysis.predicate.phenotype package": [[4, "module-genophenocorr.analysis.predicate.phenotype"]], "genophenocorr.constants package": [[5, "module-genophenocorr.constants"]], "genophenocorr.constants.variant_effects module": [[6, "genophenocorr-constants-variant-effects-module"]], "genophenocorr.data package": [[7, "module-genophenocorr.data"]], "genophenocorr.model package": [[8, "module-genophenocorr.model"]], "genophenocorr.model.genome package": [[9, "module-genophenocorr.model.genome"]], "genophenocorr.preprocessing package": [[10, "module-genophenocorr.preprocessing"]], "genophenocorr.view package": [[11, "module-genophenocorr.view"]]}, "docnames": ["apidocs/genophenocorr", "apidocs/genophenocorr.analysis", "apidocs/genophenocorr.analysis.predicate", "apidocs/genophenocorr.analysis.predicate.genotype", "apidocs/genophenocorr.analysis.predicate.phenotype", "apidocs/genophenocorr.constants", "apidocs/genophenocorr.constants.variant_effects", "apidocs/genophenocorr.data", "apidocs/genophenocorr.model", "apidocs/genophenocorr.model.genome", "apidocs/genophenocorr.preprocessing", "apidocs/genophenocorr.view", "apidocs/modules", "autosomal_recessive", "index", "installation", "tutorial", "user-guide/exploratory", "user-guide/index", "user-guide/input-data"], "envversion": {"sphinx": 61, "sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1}, "filenames": ["apidocs/genophenocorr.rst", "apidocs/genophenocorr.analysis.rst", "apidocs/genophenocorr.analysis.predicate.rst", "apidocs/genophenocorr.analysis.predicate.genotype.rst", "apidocs/genophenocorr.analysis.predicate.phenotype.rst", "apidocs/genophenocorr.constants.rst", "apidocs/genophenocorr.constants.variant_effects.rst", "apidocs/genophenocorr.data.rst", "apidocs/genophenocorr.model.rst", "apidocs/genophenocorr.model.genome.rst", "apidocs/genophenocorr.preprocessing.rst", "apidocs/genophenocorr.view.rst", "apidocs/modules.rst", "autosomal_recessive.rst", "index.rst", "installation.rst", "tutorial.rst", "user-guide/exploratory.rst", "user-guide/index.rst", "user-guide/input-data.rst"], "indexentries": {"add_error() (genophenocorr.preprocessing.notepad method)": [[10, "genophenocorr.preprocessing.Notepad.add_error", false]], "add_issue() (genophenocorr.preprocessing.notepad method)": [[10, "genophenocorr.preprocessing.Notepad.add_issue", false]], "add_subsection() (genophenocorr.preprocessing.notepad method)": [[10, "genophenocorr.preprocessing.Notepad.add_subsection", false]], "add_subsection() (genophenocorr.preprocessing.notepadtree method)": [[10, "genophenocorr.preprocessing.NotepadTree.add_subsection", false]], "add_warning() (genophenocorr.preprocessing.notepad method)": [[10, "genophenocorr.preprocessing.Notepad.add_warning", false]], "all_counts (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.all_counts", false]], "all_diseases (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_diseases", false]], "all_excluded_patients (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_excluded_patients", false]], "all_patients (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_patients", false]], "all_phenotypes (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_phenotypes", false]], "all_transcripts (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_transcripts", false]], "all_variants (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.all_variants", false]], "alt (genophenocorr.model.variantcoordinates attribute)": [[8, "genophenocorr.model.VariantCoordinates.alt", false]], "alt (genophenocorr.model.variantcoordinates property)": [[8, "id10", false]], "anchor() (genophenocorr.view.cohortviewer static method)": [[11, "genophenocorr.view.CohortViewer.anchor", false]], "annotate() (genophenocorr.preprocessing.functionalannotator method)": [[10, "genophenocorr.preprocessing.FunctionalAnnotator.annotate", false]], "annotate() (genophenocorr.preprocessing.protcachingmetadataservice method)": [[10, "genophenocorr.preprocessing.ProtCachingMetadataService.annotate", false]], "annotate() (genophenocorr.preprocessing.proteinmetadataservice method)": [[10, "genophenocorr.preprocessing.ProteinMetadataService.annotate", false]], "annotate() (genophenocorr.preprocessing.uniprotproteinmetadataservice method)": [[10, "genophenocorr.preprocessing.UniprotProteinMetadataService.annotate", false]], "annotate() (genophenocorr.preprocessing.varcachingfunctionalannotator method)": [[10, "genophenocorr.preprocessing.VarCachingFunctionalAnnotator.annotate", false]], "annotate() (genophenocorr.preprocessing.vepfunctionalannotator method)": [[10, "genophenocorr.preprocessing.VepFunctionalAnnotator.annotate", false]], "auditor (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.Auditor", false]], "cat_id (genophenocorr.analysis.predicate.patientcategory property)": [[2, "genophenocorr.analysis.predicate.PatientCategory.cat_id", false]], "categorization (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.Categorization", false]], "category (genophenocorr.analysis.predicate.categorization property)": [[2, "genophenocorr.analysis.predicate.Categorization.category", false]], "cds_end (genophenocorr.model.transcriptcoordinates property)": [[8, "genophenocorr.model.TranscriptCoordinates.cds_end", false]], "cds_start (genophenocorr.model.transcriptcoordinates property)": [[8, "genophenocorr.model.TranscriptCoordinates.cds_start", false]], "change_length (genophenocorr.model.variantcoordinates attribute)": [[8, "genophenocorr.model.VariantCoordinates.change_length", false]], "change_length (genophenocorr.model.variantcoordinates property)": [[8, "id11", false]], "children (genophenocorr.preprocessing.notepadtree property)": [[10, "genophenocorr.preprocessing.NotepadTree.children", false]], "chrom (genophenocorr.model.variantcoordinates property)": [[8, "genophenocorr.model.VariantCoordinates.chrom", false]], "code (genophenocorr.model.genotype property)": [[8, "genophenocorr.model.Genotype.code", false]], "coding_sequence_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.CODING_SEQUENCE_VARIANT", false]], "cohort (class in genophenocorr.model)": [[8, "genophenocorr.model.Cohort", false]], "cohort_summary_table() (genophenocorr.view.cohortviewer method)": [[11, "genophenocorr.view.CohortViewer.cohort_summary_table", false]], "cohortanalysis (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.CohortAnalysis", false]], "cohortanalysisconfiguration (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration", false]], "cohortcreator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.CohortCreator", false]], "cohortviewer (class in genophenocorr.view)": [[11, "genophenocorr.view.CohortViewer", false]], "comparable (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.Comparable", false]], "compare_by_exon() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_exon", false]], "compare_by_protein_feature() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature", false]], "compare_by_protein_feature_type() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature_type", false]], "compare_by_variant_effect() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_variant_effect", false]], "compare_by_variant_key() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_variant_key", false]], "compare_by_variant_keys() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_variant_keys", false]], "configure_caching_cohort_creator() (in module genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.configure_caching_cohort_creator", false]], "configure_caching_patient_creator() (in module genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.configure_caching_patient_creator", false]], "configure_cohort_analysis() (in module genophenocorr.analysis)": [[1, "genophenocorr.analysis.configure_cohort_analysis", false]], "configure_cohort_creator() (in module genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.configure_cohort_creator", false]], "configure_patient_creator() (in module genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.configure_patient_creator", false]], "corrected_pvals (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.corrected_pvals", false]], "create() (genophenocorr.model.proteinfeature static method)": [[8, "genophenocorr.model.ProteinFeature.create", false]], "create_variant_from_scratch() (genophenocorr.model.variant static method)": [[8, "genophenocorr.model.Variant.create_variant_from_scratch", false]], "curie (genophenocorr.model.varianteffect property)": [[8, "genophenocorr.model.VariantEffect.curie", false]], "datasanityissue (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.DataSanityIssue", false]], "description (genophenocorr.analysis.predicate.patientcategory property)": [[2, "genophenocorr.analysis.predicate.PatientCategory.description", false]], "disease (class in genophenocorr.model)": [[8, "genophenocorr.model.Disease", false]], "diseasepresencepredicate (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate", false]], "diseases (genophenocorr.analysis.predicate.phenotype.diseasepresencepredicate property)": [[4, "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate.diseases", false]], "diseases (genophenocorr.model.patient property)": [[8, "genophenocorr.model.Patient.diseases", false]], "domain (genophenocorr.model.featuretype attribute)": [[8, "genophenocorr.model.FeatureType.DOMAIN", false], [8, "id27", false]], "domains() (genophenocorr.model.proteinmetadata method)": [[8, "genophenocorr.model.ProteinMetadata.domains", false], [8, "id19", false]], "downstream_gene_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.DOWNSTREAM_GENE_VARIANT", false]], "draw_variants() (genophenocorr.view.varianttranscriptvisualizer method)": [[11, "genophenocorr.view.VariantTranscriptVisualizer.draw_variants", false]], "empty() (genophenocorr.model.genotypes static method)": [[8, "genophenocorr.model.Genotypes.empty", false]], "end (genophenocorr.model.featureinfo property)": [[8, "genophenocorr.model.FeatureInfo.end", false]], "end (genophenocorr.model.variantcoordinates property)": [[8, "genophenocorr.model.VariantCoordinates.end", false]], "error (genophenocorr.preprocessing.level attribute)": [[10, "genophenocorr.preprocessing.Level.ERROR", false]], "error_count() (genophenocorr.preprocessing.notepad method)": [[10, "genophenocorr.preprocessing.Notepad.error_count", false]], "errors() (genophenocorr.preprocessing.notepad method)": [[10, "genophenocorr.preprocessing.Notepad.errors", false]], "excluded_diseases() (genophenocorr.model.patient method)": [[8, "genophenocorr.model.Patient.excluded_diseases", false]], "excluded_phenotypes() (genophenocorr.model.patient method)": [[8, "genophenocorr.model.Patient.excluded_phenotypes", false]], "exonpredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ExonPredicate", false]], "exons (genophenocorr.model.transcriptcoordinates property)": [[8, "genophenocorr.model.TranscriptCoordinates.exons", false]], "exonspredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ExonsPredicate", false]], "feature_elongation (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.FEATURE_ELONGATION", false]], "feature_truncation (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.FEATURE_TRUNCATION", false]], "feature_type (genophenocorr.model.proteinfeature property)": [[8, "genophenocorr.model.ProteinFeature.feature_type", false]], "featureinfo (class in genophenocorr.model)": [[8, "genophenocorr.model.FeatureInfo", false]], "featuretype (class in genophenocorr.model)": [[8, "genophenocorr.model.FeatureType", false]], "fetch() (genophenocorr.preprocessing.vvtranscriptcoordinateservice method)": [[10, "genophenocorr.preprocessing.VVTranscriptCoordinateService.fetch", false]], "find_coordinates() (genophenocorr.preprocessing.phenopacketvariantcoordinatefinder method)": [[10, "genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder.find_coordinates", false]], "find_coordinates() (genophenocorr.preprocessing.variantcoordinatefinder method)": [[10, "genophenocorr.preprocessing.VariantCoordinateFinder.find_coordinates", false]], "find_coordinates() (genophenocorr.preprocessing.vvhgvsvariantcoordinatefinder method)": [[10, "genophenocorr.preprocessing.VVHgvsVariantCoordinateFinder.find_coordinates", false]], "first (genophenocorr.analysis.predicate.groupingpredicate attribute)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate.FIRST", false]], "five_prime_utr_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.FIVE_PRIME_UTR_VARIANT", false]], "for_sample() (genophenocorr.model.genotypes method)": [[8, "genophenocorr.model.Genotypes.for_sample", false]], "frameshift_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.FRAMESHIFT_VARIANT", false]], "from_mapping() (genophenocorr.model.genotypes static method)": [[8, "genophenocorr.model.Genotypes.from_mapping", false]], "from_patients() (genophenocorr.model.cohort static method)": [[8, "genophenocorr.model.Cohort.from_patients", false]], "from_term() (genophenocorr.model.phenotype static method)": [[8, "genophenocorr.model.Phenotype.from_term", false]], "functionalannotator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.FunctionalAnnotator", false]], "gene_id (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.gene_id", false]], "gene_id (genophenocorr.model.transcriptinfoaware property)": [[8, "genophenocorr.model.TranscriptInfoAware.gene_id", false]], "genophenocorr": [[0, "module-genophenocorr", false]], "genophenocorr.analysis": [[1, "module-genophenocorr.analysis", false]], "genophenocorr.analysis.predicate": [[2, "module-genophenocorr.analysis.predicate", false]], "genophenocorr.analysis.predicate.genotype": [[3, "module-genophenocorr.analysis.predicate.genotype", false]], "genophenocorr.analysis.predicate.phenotype": [[4, "module-genophenocorr.analysis.predicate.phenotype", false]], "genophenocorr.constants": [[5, "module-genophenocorr.constants", false]], "genophenocorr.data": [[7, "module-genophenocorr.data", false]], "genophenocorr.model": [[8, "module-genophenocorr.model", false]], "genophenocorr.model.genome": [[9, "module-genophenocorr.model.genome", false]], "genophenocorr.preprocessing": [[10, "module-genophenocorr.preprocessing", false]], "genophenocorr.view": [[11, "module-genophenocorr.view", false]], "genotype (class in genophenocorr.model)": [[8, "genophenocorr.model.Genotype", false]], "genotype_for_sample() (genophenocorr.model.genotyped method)": [[8, "genophenocorr.model.Genotyped.genotype_for_sample", false]], "genotypebooleanpredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.GenotypeBooleanPredicate", false]], "genotyped (class in genophenocorr.model)": [[8, "genophenocorr.model.Genotyped", false]], "genotypephenotypeanalysisresult (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult", false]], "genotypepolypredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.GenotypePolyPredicate", false]], "genotypes (class in genophenocorr.model)": [[8, "genophenocorr.model.Genotypes", false]], "genotypes (genophenocorr.model.genotyped property)": [[8, "genophenocorr.model.Genotyped.genotypes", false]], "genotypes (genophenocorr.model.variant property)": [[8, "genophenocorr.model.Variant.genotypes", false]], "get_annotations() (genophenocorr.preprocessing.proteinannotationcache method)": [[10, "genophenocorr.preprocessing.ProteinAnnotationCache.get_annotations", false]], "get_annotations() (genophenocorr.preprocessing.variantannotationcache method)": [[10, "genophenocorr.preprocessing.VariantAnnotationCache.get_annotations", false]], "get_categories() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.get_categories", false]], "get_categorizations() (genophenocorr.analysis.predicate.genotypebooleanpredicate method)": [[2, "genophenocorr.analysis.predicate.GenotypeBooleanPredicate.get_categorizations", false]], "get_categorizations() (genophenocorr.analysis.predicate.groupingpredicate method)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate.get_categorizations", false]], "get_categorizations() (genophenocorr.analysis.predicate.phenotype.diseasepresencepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate.get_categorizations", false]], "get_categorizations() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.get_categorizations", false]], "get_categorizations() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.get_categorizations", false]], "get_cds_regions() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.get_cds_regions", false]], "get_coding_base_count() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.get_coding_base_count", false]], "get_codon_count() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.get_codon_count", false]], "get_excluded_count() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.get_excluded_count", false]], "get_excluded_ids() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.get_excluded_ids", false]], "get_features_variant_overlaps() (genophenocorr.model.proteinmetadata method)": [[8, "genophenocorr.model.ProteinMetadata.get_features_variant_overlaps", false]], "get_five_prime_utrs() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.get_five_prime_utrs", false]], "get_question() (genophenocorr.analysis.predicate.genotype.exonpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ExonPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.exonspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ExonsPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.protfeaturepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.protfeaturespredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.protfeaturetypepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.protfeaturetypespredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.recessiveexonpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveExonPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.recessiveprotfeaturepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveProtFeaturePredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.recessiveprotfeaturetypepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveProtFeatureTypePredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.recessivevarianteffectpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveVariantEffectPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.recessivevariantpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveVariantPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.varianteffectpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.varianteffectspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.variantpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.variantspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantsPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.phenotype.diseasepresencepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.get_question", false]], "get_three_prime_utrs() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.get_three_prime_utrs", false]], "get_toy_cohort() (in module genophenocorr.data)": [[7, "genophenocorr.data.get_toy_cohort", false]], "groupingpredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate", false]], "has_errors() (genophenocorr.preprocessing.notepadtree method)": [[10, "genophenocorr.preprocessing.NotepadTree.has_errors", false]], "has_errors_or_warnings() (genophenocorr.preprocessing.notepadtree method)": [[10, "genophenocorr.preprocessing.NotepadTree.has_errors_or_warnings", false]], "has_warnings() (genophenocorr.preprocessing.notepadtree method)": [[10, "genophenocorr.preprocessing.NotepadTree.has_warnings", false]], "hemizygous (genophenocorr.model.genotype attribute)": [[8, "genophenocorr.model.Genotype.HEMIZYGOUS", false]], "heterozygous (genophenocorr.model.genotype attribute)": [[8, "genophenocorr.model.Genotype.HETEROZYGOUS", false]], "hgvs_cdna (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.hgvs_cdna", false]], "homozygous_alternate (genophenocorr.model.genotype attribute)": [[8, "genophenocorr.model.Genotype.HOMOZYGOUS_ALTERNATE", false]], "homozygous_reference (genophenocorr.model.genotype attribute)": [[8, "genophenocorr.model.Genotype.HOMOZYGOUS_REFERENCE", false]], "hpo_anchor() (genophenocorr.view.cohortviewer static method)": [[11, "genophenocorr.view.CohortViewer.hpo_anchor", false]], "hpo_term_counts_table() (genophenocorr.view.cohortviewer method)": [[11, "genophenocorr.view.CohortViewer.hpo_term_counts_table", false]], "html_row() (genophenocorr.view.cohortviewer static method)": [[11, "genophenocorr.view.CohortViewer.html_row", false]], "identifier (genophenocorr.model.disease property)": [[8, "genophenocorr.model.Disease.identifier", false]], "identifier (genophenocorr.model.phenotype property)": [[8, "genophenocorr.model.Phenotype.identifier", false]], "identifier (genophenocorr.model.transcriptcoordinates property)": [[8, "genophenocorr.model.TranscriptCoordinates.identifier", false]], "include_sv (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.include_sv", false]], "incomplete_terminal_codon_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.INCOMPLETE_TERMINAL_CODON_VARIANT", false]], "info (genophenocorr.model.proteinfeature property)": [[8, "genophenocorr.model.ProteinFeature.info", false]], "inframe_deletion (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.INFRAME_DELETION", false]], "inframe_insertion (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.INFRAME_INSERTION", false]], "intergenic_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.INTERGENIC_VARIANT", false]], "intron_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.INTRON_VARIANT", false]], "is_coding() (genophenocorr.model.transcriptcoordinates method)": [[8, "genophenocorr.model.TranscriptCoordinates.is_coding", false]], "is_observed (genophenocorr.model.disease attribute)": [[8, "genophenocorr.model.Disease.is_observed", false]], "is_observed (genophenocorr.model.phenotype attribute)": [[8, "genophenocorr.model.Phenotype.is_observed", false]], "is_observed (genophenocorr.model.phenotype property)": [[8, "id4", false]], "is_preferred (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.is_preferred", false]], "is_present (genophenocorr.model.disease property)": [[8, "genophenocorr.model.Disease.is_present", false]], "is_present (genophenocorr.model.phenotype property)": [[8, "genophenocorr.model.Phenotype.is_present", false]], "is_structural() (genophenocorr.model.variantcoordinates method)": [[8, "genophenocorr.model.VariantCoordinates.is_structural", false]], "issues (genophenocorr.preprocessing.notepad property)": [[10, "genophenocorr.preprocessing.Notepad.issues", false]], "iterate_nodes() (genophenocorr.preprocessing.notepadtree method)": [[10, "genophenocorr.preprocessing.NotepadTree.iterate_nodes", false]], "label (genophenocorr.model.proteinmetadata attribute)": [[8, "genophenocorr.model.ProteinMetadata.label", false]], "label (genophenocorr.model.proteinmetadata property)": [[8, "id17", false]], "label (genophenocorr.model.samplelabels property)": [[8, "genophenocorr.model.SampleLabels.label", false]], "label (genophenocorr.preprocessing.notepad property)": [[10, "genophenocorr.preprocessing.Notepad.label", false]], "label_summary() (genophenocorr.model.samplelabels method)": [[8, "genophenocorr.model.SampleLabels.label_summary", false]], "labels (genophenocorr.model.patient property)": [[8, "genophenocorr.model.Patient.labels", false]], "level (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.Level", false]], "level (genophenocorr.preprocessing.datasanityissue property)": [[10, "genophenocorr.preprocessing.DataSanityIssue.level", false]], "level (genophenocorr.preprocessing.notepadtree property)": [[10, "genophenocorr.preprocessing.NotepadTree.level", false]], "list_all_diseases() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_all_diseases", false]], "list_all_patients() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_all_patients", false]], "list_all_phenotypes() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_all_phenotypes", false]], "list_all_proteins() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_all_proteins", false]], "list_all_variants() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_all_variants", false]], "list_data_by_tx() (genophenocorr.model.cohort method)": [[8, "genophenocorr.model.Cohort.list_data_by_tx", false]], "load_phenopacket() (in module genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.load_phenopacket", false]], "load_phenopacket_folder() (in module genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.load_phenopacket_folder", false]], "mature_mirna_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.MATURE_MIRNA_VARIANT", false]], "message (genophenocorr.preprocessing.datasanityissue property)": [[10, "genophenocorr.preprocessing.DataSanityIssue.message", false]], "meta_label (genophenocorr.model.samplelabels property)": [[8, "genophenocorr.model.SampleLabels.meta_label", false]], "min_perc_patients_w_hpo (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.min_perc_patients_w_hpo", false]], "missense_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.MISSENSE_VARIANT", false]], "missing_implies_excluded (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.missing_implies_excluded", false]], "module": [[0, "module-genophenocorr", false], [1, "module-genophenocorr.analysis", false], [2, "module-genophenocorr.analysis.predicate", false], [3, "module-genophenocorr.analysis.predicate.genotype", false], [4, "module-genophenocorr.analysis.predicate.phenotype", false], [5, "module-genophenocorr.constants", false], [7, "module-genophenocorr.data", false], [8, "module-genophenocorr.model", false], [9, "module-genophenocorr.model.genome", false], [10, "module-genophenocorr.preprocessing", false], [11, "module-genophenocorr.view", false]], "motif (genophenocorr.model.featuretype attribute)": [[8, "genophenocorr.model.FeatureType.MOTIF", false], [8, "id26", false]], "motifs() (genophenocorr.model.proteinmetadata method)": [[8, "genophenocorr.model.ProteinMetadata.motifs", false], [8, "id22", false]], "mtc_alpha (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.mtc_alpha", false]], "n_categorizations() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.n_categorizations", false]], "n_usable (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.n_usable", false]], "name (genophenocorr.analysis.predicate.patientcategory property)": [[2, "genophenocorr.analysis.predicate.PatientCategory.name", false]], "name (genophenocorr.model.disease attribute)": [[8, "genophenocorr.model.Disease.name", false]], "name (genophenocorr.model.disease property)": [[8, "id5", false]], "name (genophenocorr.model.featureinfo attribute)": [[8, "genophenocorr.model.FeatureInfo.name", false]], "name (genophenocorr.model.featureinfo property)": [[8, "id23", false]], "name (genophenocorr.model.phenotype attribute)": [[8, "genophenocorr.model.Phenotype.name", false]], "name (genophenocorr.model.phenotype property)": [[8, "id3", false]], "nmd_transcript_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.NMD_TRANSCRIPT_VARIANT", false]], "no (genophenocorr.analysis.predicate.genotypebooleanpredicate attribute)": [[2, "genophenocorr.analysis.predicate.GenotypeBooleanPredicate.NO", false]], "no (genophenocorr.analysis.predicate.patientcategories attribute)": [[2, "genophenocorr.analysis.predicate.PatientCategories.NO", false]], "no_call (genophenocorr.model.genotype attribute)": [[8, "genophenocorr.model.Genotype.NO_CALL", false]], "non_coding_transcript_exon_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.NON_CODING_TRANSCRIPT_EXON_VARIANT", false]], "non_coding_transcript_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.NON_CODING_TRANSCRIPT_VARIANT", false]], "noncoding_effects (genophenocorr.preprocessing.vepfunctionalannotator attribute)": [[10, "genophenocorr.preprocessing.VepFunctionalAnnotator.NONCODING_EFFECTS", false]], "notepad (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.Notepad", false]], "notepadtree (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.NotepadTree", false]], "observed (genophenocorr.model.phenotype property)": [[8, "genophenocorr.model.Phenotype.observed", false]], "overlapping_exons (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.overlapping_exons", false]], "parse_response() (genophenocorr.preprocessing.vvtranscriptcoordinateservice method)": [[10, "genophenocorr.preprocessing.VVTranscriptCoordinateService.parse_response", false]], "patient (class in genophenocorr.model)": [[8, "genophenocorr.model.Patient", false]], "patient_id (genophenocorr.model.patient attribute)": [[8, "genophenocorr.model.Patient.patient_id", false]], "patient_id (genophenocorr.model.patient property)": [[8, "id0", false]], "patientcategories (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.PatientCategories", false]], "patientcategory (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.PatientCategory", false]], "patientcreator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.PatientCreator", false]], "phenopacketpatientcreator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.PhenopacketPatientCreator", false]], "phenopacketvariantcoordinatefinder (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder", false]], "phenotype (class in genophenocorr.model)": [[8, "genophenocorr.model.Phenotype", false]], "phenotype (genophenocorr.analysis.predicate.phenotype.phenotypecategorization property)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypeCategorization.phenotype", false]], "phenotype_categories (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.phenotype_categories", false]], "phenotypecategorization (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypeCategorization", false]], "phenotypecreator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.PhenotypeCreator", false]], "phenotypepolypredicate (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypePolyPredicate", false]], "phenotypes (genophenocorr.analysis.predicate.phenotype.phenotypepolypredicate property)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypePolyPredicate.phenotypes", false]], "phenotypes (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate property)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.phenotypes", false]], "phenotypes (genophenocorr.model.patient attribute)": [[8, "genophenocorr.model.Patient.phenotypes", false]], "phenotypes (genophenocorr.model.patient property)": [[8, "id1", false]], "polypredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.PolyPredicate", false]], "prepare_notepad() (genophenocorr.preprocessing.auditor static method)": [[10, "genophenocorr.preprocessing.Auditor.prepare_notepad", false]], "present_diseases() (genophenocorr.model.patient method)": [[8, "genophenocorr.model.Patient.present_diseases", false]], "present_phenotypes() (genophenocorr.model.patient method)": [[8, "genophenocorr.model.Patient.present_phenotypes", false]], "process() (genophenocorr.preprocessing.auditor method)": [[10, "genophenocorr.preprocessing.Auditor.process", false]], "process() (genophenocorr.preprocessing.cohortcreator method)": [[10, "genophenocorr.preprocessing.CohortCreator.process", false]], "process() (genophenocorr.preprocessing.phenopacketpatientcreator method)": [[10, "genophenocorr.preprocessing.PhenopacketPatientCreator.process", false]], "process() (genophenocorr.preprocessing.phenotypecreator method)": [[10, "genophenocorr.preprocessing.PhenotypeCreator.process", false]], "propagatingphenotypepredicate (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate", false]], "protcachingmetadataservice (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.ProtCachingMetadataService", false]], "protein_altering_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.PROTEIN_ALTERING_VARIANT", false]], "protein_effect_location (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.protein_effect_location", false]], "protein_features (genophenocorr.model.proteinmetadata attribute)": [[8, "genophenocorr.model.ProteinMetadata.protein_features", false]], "protein_features (genophenocorr.model.proteinmetadata property)": [[8, "id18", false]], "protein_features_table() (genophenocorr.view.cohortviewer method)": [[11, "genophenocorr.view.CohortViewer.protein_features_table", false]], "protein_id (genophenocorr.model.proteinmetadata attribute)": [[8, "genophenocorr.model.ProteinMetadata.protein_id", false]], "protein_id (genophenocorr.model.proteinmetadata property)": [[8, "id16", false]], "protein_id (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.protein_id", false]], "proteinannotationcache (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.ProteinAnnotationCache", false]], "proteinfeature (class in genophenocorr.model)": [[8, "genophenocorr.model.ProteinFeature", false]], "proteinmetadata (class in genophenocorr.model)": [[8, "genophenocorr.model.ProteinMetadata", false]], "proteinmetadataservice (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.ProteinMetadataService", false]], "protfeaturepredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate", false]], "protfeaturespredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate", false]], "protfeaturetypepredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate", false]], "protfeaturetypespredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate", false]], "pval_correction (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.pval_correction", false]], "pvals (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.pvals", false]], "recessiveexonpredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveExonPredicate", false]], "recessiveprotfeaturepredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveProtFeaturePredicate", false]], "recessiveprotfeaturetypepredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveProtFeatureTypePredicate", false]], "recessivevarianteffectpredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveVariantEffectPredicate", false]], "recessivevariantpredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveVariantPredicate", false]], "ref (genophenocorr.model.variantcoordinates attribute)": [[8, "genophenocorr.model.VariantCoordinates.ref", false]], "ref (genophenocorr.model.variantcoordinates property)": [[8, "id9", false]], "region (genophenocorr.model.featureinfo attribute)": [[8, "genophenocorr.model.FeatureInfo.region", false]], "region (genophenocorr.model.featureinfo property)": [[8, "id24", false]], "region (genophenocorr.model.featuretype attribute)": [[8, "genophenocorr.model.FeatureType.REGION", false], [8, "id28", false]], "region (genophenocorr.model.transcriptcoordinates property)": [[8, "genophenocorr.model.TranscriptCoordinates.region", false]], "region (genophenocorr.model.variantcoordinates attribute)": [[8, "genophenocorr.model.VariantCoordinates.region", false]], "region (genophenocorr.model.variantcoordinates property)": [[8, "id8", false]], "regions() (genophenocorr.model.proteinmetadata method)": [[8, "genophenocorr.model.ProteinMetadata.regions", false], [8, "id21", false]], "regulatory_region_ablation (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.REGULATORY_REGION_ABLATION", false]], "regulatory_region_amplification (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.REGULATORY_REGION_AMPLIFICATION", false]], "regulatory_region_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.REGULATORY_REGION_VARIANT", false]], "repeat (genophenocorr.model.featuretype attribute)": [[8, "genophenocorr.model.FeatureType.REPEAT", false], [8, "id25", false]], "repeats() (genophenocorr.model.proteinmetadata method)": [[8, "genophenocorr.model.ProteinMetadata.repeats", false], [8, "id20", false]], "samplelabels (class in genophenocorr.model)": [[8, "genophenocorr.model.SampleLabels", false]], "second (genophenocorr.analysis.predicate.groupingpredicate attribute)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate.SECOND", false]], "sequence_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SEQUENCE_VARIANT", false]], "single() (genophenocorr.model.genotypes static method)": [[8, "genophenocorr.model.Genotypes.single", false]], "solution (genophenocorr.preprocessing.datasanityissue property)": [[10, "genophenocorr.preprocessing.DataSanityIssue.solution", false]], "splice_acceptor_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_ACCEPTOR_VARIANT", false]], "splice_donor_5th_base_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_DONOR_5TH_BASE_VARIANT", false]], "splice_donor_region_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_DONOR_REGION_VARIANT", false]], "splice_donor_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_DONOR_VARIANT", false]], "splice_polypyrimidine_tract_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_POLYPYRIMIDINE_TRACT_VARIANT", false]], "splice_region_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SPLICE_REGION_VARIANT", false]], "start (genophenocorr.model.featureinfo property)": [[8, "genophenocorr.model.FeatureInfo.start", false]], "start (genophenocorr.model.variantcoordinates property)": [[8, "genophenocorr.model.VariantCoordinates.start", false]], "start_lost (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.START_LOST", false]], "start_retained_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.START_RETAINED_VARIANT", false]], "stop_gained (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.STOP_GAINED", false]], "stop_lost (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.STOP_LOST", false]], "stop_retained_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.STOP_RETAINED_VARIANT", false]], "store_annotations() (genophenocorr.preprocessing.proteinannotationcache method)": [[10, "genophenocorr.preprocessing.ProteinAnnotationCache.store_annotations", false]], "store_annotations() (genophenocorr.preprocessing.variantannotationcache method)": [[10, "genophenocorr.preprocessing.VariantAnnotationCache.store_annotations", false]], "summarize() (genophenocorr.analysis.genotypephenotypeanalysisresult method)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.summarize", false]], "synonymous_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.SYNONYMOUS_VARIANT", false]], "term_id (genophenocorr.model.disease attribute)": [[8, "genophenocorr.model.Disease.term_id", false]], "term_id (genophenocorr.model.phenotype attribute)": [[8, "genophenocorr.model.Phenotype.term_id", false]], "test() (genophenocorr.analysis.predicate.genotype.exonpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ExonPredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.exonspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ExonsPredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.protfeaturepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.protfeaturespredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.protfeaturetypepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.protfeaturetypespredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.recessiveexonpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveExonPredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.recessiveprotfeaturepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveProtFeaturePredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.recessiveprotfeaturetypepredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveProtFeatureTypePredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.recessivevarianteffectpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveVariantEffectPredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.recessivevariantpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.RecessiveVariantPredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.varianteffectpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectPredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.varianteffectspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.variantpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicate.test", false]], "test() (genophenocorr.analysis.predicate.genotype.variantspredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantsPredicate.test", false]], "test() (genophenocorr.analysis.predicate.phenotype.diseasepresencepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate.test", false]], "test() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.test", false]], "test() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.test", false]], "tf_binding_site_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.TF_BINDING_SITE_VARIANT", false]], "tfbs_ablation (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.TFBS_ABLATION", false]], "tfbs_amplification (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.TFBS_AMPLIFICATION", false]], "three_prime_utr_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.THREE_PRIME_UTR_VARIANT", false]], "to_string() (genophenocorr.model.proteinfeature method)": [[8, "genophenocorr.model.ProteinFeature.to_string", false]], "total_patient_count (genophenocorr.model.cohort property)": [[8, "genophenocorr.model.Cohort.total_patient_count", false]], "transcript_ablation (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.TRANSCRIPT_ABLATION", false]], "transcript_amplification (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.TRANSCRIPT_AMPLIFICATION", false]], "transcript_id (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.transcript_id", false]], "transcript_id (genophenocorr.model.transcriptinfoaware property)": [[8, "genophenocorr.model.TranscriptInfoAware.transcript_id", false]], "transcriptannotation (class in genophenocorr.model)": [[8, "genophenocorr.model.TranscriptAnnotation", false]], "transcriptcoordinates (class in genophenocorr.model)": [[8, "genophenocorr.model.TranscriptCoordinates", false]], "transcriptinfoaware (class in genophenocorr.model)": [[8, "genophenocorr.model.TranscriptInfoAware", false]], "tx_annotations (genophenocorr.model.variant attribute)": [[8, "genophenocorr.model.Variant.tx_annotations", false]], "tx_annotations (genophenocorr.model.variant property)": [[8, "id7", false]], "uniprotproteinmetadataservice (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.UniprotProteinMetadataService", false]], "upstream_gene_variant (genophenocorr.model.varianteffect attribute)": [[8, "genophenocorr.model.VariantEffect.UPSTREAM_GENE_VARIANT", false]], "varcachingfunctionalannotator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VarCachingFunctionalAnnotator", false]], "variant (class in genophenocorr.model)": [[8, "genophenocorr.model.Variant", false]], "variant_class (genophenocorr.model.variant attribute)": [[8, "genophenocorr.model.Variant.variant_class", false]], "variant_class (genophenocorr.model.variantcoordinates property)": [[8, "genophenocorr.model.VariantCoordinates.variant_class", false]], "variant_coordinates (genophenocorr.model.variant attribute)": [[8, "genophenocorr.model.Variant.variant_coordinates", false]], "variant_coordinates (genophenocorr.model.variant property)": [[8, "id6", false]], "variant_effects (genophenocorr.model.transcriptannotation property)": [[8, "genophenocorr.model.TranscriptAnnotation.variant_effects", false]], "variant_key (genophenocorr.model.variantcoordinates property)": [[8, "genophenocorr.model.VariantCoordinates.variant_key", false]], "variant_string (genophenocorr.model.variant attribute)": [[8, "genophenocorr.model.Variant.variant_string", false]], "variantannotationcache (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VariantAnnotationCache", false]], "variantcoordinatefinder (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VariantCoordinateFinder", false]], "variantcoordinates (class in genophenocorr.model)": [[8, "genophenocorr.model.VariantCoordinates", false]], "varianteffect (class in genophenocorr.model)": [[8, "genophenocorr.model.VariantEffect", false]], "varianteffectpredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectPredicate", false]], "varianteffectspredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate", false]], "variantpredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicate", false]], "variants (genophenocorr.model.patient attribute)": [[8, "genophenocorr.model.Patient.variants", false]], "variants (genophenocorr.model.patient property)": [[8, "id2", false]], "variants_table() (genophenocorr.view.cohortviewer method)": [[11, "genophenocorr.view.CohortViewer.variants_table", false]], "variantspredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.VariantsPredicate", false]], "varianttranscriptvisualizer (class in genophenocorr.view)": [[11, "genophenocorr.view.VariantTranscriptVisualizer", false]], "vepfunctionalannotator (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VepFunctionalAnnotator", false]], "visit() (genophenocorr.preprocessing.notepadtree method)": [[10, "genophenocorr.preprocessing.NotepadTree.visit", false]], "vvhgvsvariantcoordinatefinder (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VVHgvsVariantCoordinateFinder", false]], "vvtranscriptcoordinateservice (class in genophenocorr.preprocessing)": [[10, "genophenocorr.preprocessing.VVTranscriptCoordinateService", false]], "warn (genophenocorr.preprocessing.level attribute)": [[10, "genophenocorr.preprocessing.Level.WARN", false]], "warning_count() (genophenocorr.preprocessing.notepad method)": [[10, "genophenocorr.preprocessing.Notepad.warning_count", false]], "warnings() (genophenocorr.preprocessing.notepad method)": [[10, "genophenocorr.preprocessing.Notepad.warnings", false]], "yes (genophenocorr.analysis.predicate.genotypebooleanpredicate attribute)": [[2, "genophenocorr.analysis.predicate.GenotypeBooleanPredicate.YES", false]], "yes (genophenocorr.analysis.predicate.patientcategories attribute)": [[2, "genophenocorr.analysis.predicate.PatientCategories.YES", false]]}, "objects": {"": [[0, 0, 0, "-", "genophenocorr"]], "genophenocorr": [[1, 0, 0, "-", "analysis"], [5, 0, 0, "-", "constants"], [7, 0, 0, "-", "data"], [8, 0, 0, "-", "model"], [10, 0, 0, "-", "preprocessing"], [11, 0, 0, "-", "view"]], "genophenocorr.analysis": [[1, 1, 1, "", "CohortAnalysis"], [1, 1, 1, "", "CohortAnalysisConfiguration"], [1, 1, 1, "", "GenotypePhenotypeAnalysisResult"], [1, 4, 1, "", "configure_cohort_analysis"], [2, 0, 0, "-", "predicate"]], "genophenocorr.analysis.CohortAnalysis": [[1, 2, 1, "", "compare_by_exon"], [1, 2, 1, "", "compare_by_protein_feature"], [1, 2, 1, "", "compare_by_protein_feature_type"], [1, 2, 1, "", "compare_by_variant_effect"], [1, 2, 1, "", "compare_by_variant_key"], [1, 2, 1, "", "compare_by_variant_keys"]], "genophenocorr.analysis.CohortAnalysisConfiguration": [[1, 3, 1, "", "include_sv"], [1, 3, 1, "", "min_perc_patients_w_hpo"], [1, 3, 1, "", "missing_implies_excluded"], [1, 3, 1, "", "mtc_alpha"], [1, 3, 1, "", "pval_correction"]], "genophenocorr.analysis.GenotypePhenotypeAnalysisResult": [[1, 3, 1, "", "all_counts"], [1, 3, 1, "", "corrected_pvals"], [1, 3, 1, "", "n_usable"], [1, 3, 1, "", "phenotype_categories"], [1, 3, 1, "", "pvals"], [1, 2, 1, "", "summarize"]], "genophenocorr.analysis.predicate": [[2, 1, 1, "", "Categorization"], [2, 1, 1, "", "GenotypeBooleanPredicate"], [2, 1, 1, "", "GenotypePolyPredicate"], [2, 1, 1, "", "GroupingPredicate"], [2, 1, 1, "", "PatientCategories"], [2, 1, 1, "", "PatientCategory"], [2, 1, 1, "", "PolyPredicate"], [3, 0, 0, "-", "genotype"], [4, 0, 0, "-", "phenotype"]], "genophenocorr.analysis.predicate.Categorization": [[2, 3, 1, "", "category"]], "genophenocorr.analysis.predicate.GenotypeBooleanPredicate": [[2, 5, 1, "", "NO"], [2, 5, 1, "", "YES"], [2, 2, 1, "", "get_categorizations"]], "genophenocorr.analysis.predicate.GroupingPredicate": [[2, 5, 1, "", "FIRST"], [2, 5, 1, "", "SECOND"], [2, 2, 1, "", "get_categorizations"]], "genophenocorr.analysis.predicate.PatientCategories": [[2, 5, 1, "", "NO"], [2, 5, 1, "", "YES"]], "genophenocorr.analysis.predicate.PatientCategory": [[2, 3, 1, "", "cat_id"], [2, 3, 1, "", "description"], [2, 3, 1, "", "name"]], "genophenocorr.analysis.predicate.PolyPredicate": [[2, 2, 1, "", "get_categories"], [2, 2, 1, "", "get_categorizations"], [2, 2, 1, "", "get_question"], [2, 2, 1, "", "n_categorizations"], [2, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype": [[3, 1, 1, "", "ExonPredicate"], [3, 1, 1, "", "ExonsPredicate"], [3, 1, 1, "", "ProtFeaturePredicate"], [3, 1, 1, "", "ProtFeatureTypePredicate"], [3, 1, 1, "", "ProtFeatureTypesPredicate"], [3, 1, 1, "", "ProtFeaturesPredicate"], [3, 1, 1, "", "RecessiveExonPredicate"], [3, 1, 1, "", "RecessiveProtFeaturePredicate"], [3, 1, 1, "", "RecessiveProtFeatureTypePredicate"], [3, 1, 1, "", "RecessiveVariantEffectPredicate"], [3, 1, 1, "", "RecessiveVariantPredicate"], [3, 1, 1, "", "VariantEffectPredicate"], [3, 1, 1, "", "VariantEffectsPredicate"], [3, 1, 1, "", "VariantPredicate"], [3, 1, 1, "", "VariantsPredicate"]], "genophenocorr.analysis.predicate.genotype.ExonPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.ExonsPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.ProtFeaturePredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.ProtFeatureTypePredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.ProtFeatureTypesPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.ProtFeaturesPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.RecessiveExonPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.RecessiveProtFeaturePredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.RecessiveProtFeatureTypePredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.RecessiveVariantEffectPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.RecessiveVariantPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.VariantEffectPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.VariantEffectsPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.VariantPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.VariantsPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.phenotype": [[4, 1, 1, "", "Comparable"], [4, 1, 1, "", "DiseasePresencePredicate"], [4, 1, 1, "", "PhenotypeCategorization"], [4, 1, 1, "", "PhenotypePolyPredicate"], [4, 1, 1, "", "PropagatingPhenotypePredicate"]], "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate": [[4, 3, 1, "", "diseases"], [4, 2, 1, "", "get_categorizations"], [4, 2, 1, "", "get_question"], [4, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.phenotype.PhenotypeCategorization": [[4, 3, 1, "", "phenotype"]], "genophenocorr.analysis.predicate.phenotype.PhenotypePolyPredicate": [[4, 3, 1, "", "phenotypes"]], "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate": [[4, 2, 1, "", "get_categorizations"], [4, 2, 1, "", "get_question"], [4, 3, 1, "", "phenotypes"], [4, 2, 1, "", "test"]], "genophenocorr.data": [[7, 4, 1, "", "get_toy_cohort"]], "genophenocorr.model": [[8, 1, 1, "", "Cohort"], [8, 1, 1, "", "Disease"], [8, 1, 1, "", "FeatureInfo"], [8, 1, 1, "", "FeatureType"], [8, 1, 1, "", "Genotype"], [8, 1, 1, "", "Genotyped"], [8, 1, 1, "", "Genotypes"], [8, 1, 1, "", "Patient"], [8, 1, 1, "", "Phenotype"], [8, 1, 1, "", "ProteinFeature"], [8, 1, 1, "", "ProteinMetadata"], [8, 1, 1, "", "SampleLabels"], [8, 1, 1, "", "TranscriptAnnotation"], [8, 1, 1, "", "TranscriptCoordinates"], [8, 1, 1, "", "TranscriptInfoAware"], [8, 1, 1, "", "Variant"], [8, 1, 1, "", "VariantCoordinates"], [8, 1, 1, "", "VariantEffect"], [9, 0, 0, "-", "genome"]], "genophenocorr.model.Cohort": [[8, 3, 1, "", "all_diseases"], [8, 3, 1, "", "all_excluded_patients"], [8, 3, 1, "", "all_patients"], [8, 3, 1, "", "all_phenotypes"], [8, 3, 1, "", "all_transcripts"], [8, 3, 1, "", "all_variants"], [8, 2, 1, "", "from_patients"], [8, 2, 1, "", "get_excluded_count"], [8, 2, 1, "", "get_excluded_ids"], [8, 2, 1, "", "list_all_diseases"], [8, 2, 1, "", "list_all_patients"], [8, 2, 1, "", "list_all_phenotypes"], [8, 2, 1, "", "list_all_proteins"], [8, 2, 1, "", "list_all_variants"], [8, 2, 1, "", "list_data_by_tx"], [8, 3, 1, "", "total_patient_count"]], "genophenocorr.model.Disease": [[8, 3, 1, "", "identifier"], [8, 5, 1, "", "is_observed"], [8, 3, 1, "", "is_present"], [8, 3, 1, "id5", "name"], [8, 5, 1, "", "term_id"]], "genophenocorr.model.FeatureInfo": [[8, 3, 1, "", "end"], [8, 3, 1, "id23", "name"], [8, 3, 1, "id24", "region"], [8, 3, 1, "", "start"]], "genophenocorr.model.FeatureType": [[8, 5, 1, "id27", "DOMAIN"], [8, 5, 1, "id26", "MOTIF"], [8, 5, 1, "id28", "REGION"], [8, 5, 1, "id25", "REPEAT"]], "genophenocorr.model.Genotype": [[8, 5, 1, "", "HEMIZYGOUS"], [8, 5, 1, "", "HETEROZYGOUS"], [8, 5, 1, "", "HOMOZYGOUS_ALTERNATE"], [8, 5, 1, "", "HOMOZYGOUS_REFERENCE"], [8, 5, 1, "", "NO_CALL"], [8, 3, 1, "", "code"]], "genophenocorr.model.Genotyped": [[8, 2, 1, "", "genotype_for_sample"], [8, 3, 1, "", "genotypes"]], "genophenocorr.model.Genotypes": [[8, 2, 1, "", "empty"], [8, 2, 1, "", "for_sample"], [8, 2, 1, "", "from_mapping"], [8, 2, 1, "", "single"]], "genophenocorr.model.Patient": [[8, 3, 1, "", "diseases"], [8, 2, 1, "", "excluded_diseases"], [8, 2, 1, "", "excluded_phenotypes"], [8, 3, 1, "", "labels"], [8, 3, 1, "id0", "patient_id"], [8, 3, 1, "id1", "phenotypes"], [8, 2, 1, "", "present_diseases"], [8, 2, 1, "", "present_phenotypes"], [8, 3, 1, "id2", "variants"]], "genophenocorr.model.Phenotype": [[8, 2, 1, "", "from_term"], [8, 3, 1, "", "identifier"], [8, 3, 1, "id4", "is_observed"], [8, 3, 1, "", "is_present"], [8, 3, 1, "id3", "name"], [8, 3, 1, "", "observed"], [8, 5, 1, "", "term_id"]], "genophenocorr.model.ProteinFeature": [[8, 2, 1, "", "create"], [8, 3, 1, "", "feature_type"], [8, 3, 1, "", "info"], [8, 2, 1, "", "to_string"]], "genophenocorr.model.ProteinMetadata": [[8, 2, 1, "id19", "domains"], [8, 2, 1, "", "get_features_variant_overlaps"], [8, 3, 1, "id17", "label"], [8, 2, 1, "id22", "motifs"], [8, 3, 1, "id18", "protein_features"], [8, 3, 1, "id16", "protein_id"], [8, 2, 1, "id21", "regions"], [8, 2, 1, "id20", "repeats"]], "genophenocorr.model.SampleLabels": [[8, 3, 1, "", "label"], [8, 2, 1, "", "label_summary"], [8, 3, 1, "", "meta_label"]], "genophenocorr.model.TranscriptAnnotation": [[8, 3, 1, "", "gene_id"], [8, 3, 1, "", "hgvs_cdna"], [8, 3, 1, "", "is_preferred"], [8, 3, 1, "", "overlapping_exons"], [8, 3, 1, "", "protein_effect_location"], [8, 3, 1, "", "protein_id"], [8, 3, 1, "", "transcript_id"], [8, 3, 1, "", "variant_effects"]], "genophenocorr.model.TranscriptCoordinates": [[8, 3, 1, "", "cds_end"], [8, 3, 1, "", "cds_start"], [8, 3, 1, "", "exons"], [8, 2, 1, "", "get_cds_regions"], [8, 2, 1, "", "get_coding_base_count"], [8, 2, 1, "", "get_codon_count"], [8, 2, 1, "", "get_five_prime_utrs"], [8, 2, 1, "", "get_three_prime_utrs"], [8, 3, 1, "", "identifier"], [8, 2, 1, "", "is_coding"], [8, 3, 1, "", "region"]], "genophenocorr.model.TranscriptInfoAware": [[8, 3, 1, "", "gene_id"], [8, 3, 1, "", "transcript_id"]], "genophenocorr.model.Variant": [[8, 2, 1, "", "create_variant_from_scratch"], [8, 3, 1, "", "genotypes"], [8, 3, 1, "id7", "tx_annotations"], [8, 5, 1, "", "variant_class"], [8, 3, 1, "id6", "variant_coordinates"], [8, 5, 1, "", "variant_string"]], "genophenocorr.model.VariantCoordinates": [[8, 3, 1, "id10", "alt"], [8, 3, 1, "id11", "change_length"], [8, 3, 1, "", "chrom"], [8, 3, 1, "", "end"], [8, 2, 1, "", "is_structural"], [8, 3, 1, "id9", "ref"], [8, 3, 1, "id8", "region"], [8, 3, 1, "", "start"], [8, 3, 1, "", "variant_class"], [8, 3, 1, "", "variant_key"]], "genophenocorr.model.VariantEffect": [[8, 5, 1, "", "CODING_SEQUENCE_VARIANT"], [8, 5, 1, "", "DOWNSTREAM_GENE_VARIANT"], [8, 5, 1, "", "FEATURE_ELONGATION"], [8, 5, 1, "", "FEATURE_TRUNCATION"], [8, 5, 1, "", "FIVE_PRIME_UTR_VARIANT"], [8, 5, 1, "", "FRAMESHIFT_VARIANT"], [8, 5, 1, "", "INCOMPLETE_TERMINAL_CODON_VARIANT"], [8, 5, 1, "", "INFRAME_DELETION"], [8, 5, 1, "", "INFRAME_INSERTION"], [8, 5, 1, "", "INTERGENIC_VARIANT"], [8, 5, 1, "", "INTRON_VARIANT"], [8, 5, 1, "", "MATURE_MIRNA_VARIANT"], [8, 5, 1, "", "MISSENSE_VARIANT"], [8, 5, 1, "", "NMD_TRANSCRIPT_VARIANT"], [8, 5, 1, "", "NON_CODING_TRANSCRIPT_EXON_VARIANT"], [8, 5, 1, "", "NON_CODING_TRANSCRIPT_VARIANT"], [8, 5, 1, "", "PROTEIN_ALTERING_VARIANT"], [8, 5, 1, "", "REGULATORY_REGION_ABLATION"], [8, 5, 1, "", "REGULATORY_REGION_AMPLIFICATION"], [8, 5, 1, "", "REGULATORY_REGION_VARIANT"], [8, 5, 1, "", "SEQUENCE_VARIANT"], [8, 5, 1, "", "SPLICE_ACCEPTOR_VARIANT"], [8, 5, 1, "", "SPLICE_DONOR_5TH_BASE_VARIANT"], [8, 5, 1, "", "SPLICE_DONOR_REGION_VARIANT"], [8, 5, 1, "", "SPLICE_DONOR_VARIANT"], [8, 5, 1, "", "SPLICE_POLYPYRIMIDINE_TRACT_VARIANT"], [8, 5, 1, "", "SPLICE_REGION_VARIANT"], [8, 5, 1, "", "START_LOST"], [8, 5, 1, "", "START_RETAINED_VARIANT"], [8, 5, 1, "", "STOP_GAINED"], [8, 5, 1, "", "STOP_LOST"], [8, 5, 1, "", "STOP_RETAINED_VARIANT"], [8, 5, 1, "", "SYNONYMOUS_VARIANT"], [8, 5, 1, "", "TFBS_ABLATION"], [8, 5, 1, "", "TFBS_AMPLIFICATION"], [8, 5, 1, "", "TF_BINDING_SITE_VARIANT"], [8, 5, 1, "", "THREE_PRIME_UTR_VARIANT"], [8, 5, 1, "", "TRANSCRIPT_ABLATION"], [8, 5, 1, "", "TRANSCRIPT_AMPLIFICATION"], [8, 5, 1, "", "UPSTREAM_GENE_VARIANT"], [8, 3, 1, "", "curie"]], "genophenocorr.preprocessing": [[10, 1, 1, "", "Auditor"], [10, 1, 1, "", "CohortCreator"], [10, 1, 1, "", "DataSanityIssue"], [10, 1, 1, "", "FunctionalAnnotator"], [10, 1, 1, "", "Level"], [10, 1, 1, "", "Notepad"], [10, 1, 1, "", "NotepadTree"], [10, 1, 1, "", "PatientCreator"], [10, 1, 1, "", "PhenopacketPatientCreator"], [10, 1, 1, "", "PhenopacketVariantCoordinateFinder"], [10, 1, 1, "", "PhenotypeCreator"], [10, 1, 1, "", "ProtCachingMetadataService"], [10, 1, 1, "", "ProteinAnnotationCache"], [10, 1, 1, "", "ProteinMetadataService"], [10, 1, 1, "", "UniprotProteinMetadataService"], [10, 1, 1, "", "VVHgvsVariantCoordinateFinder"], [10, 1, 1, "", "VVTranscriptCoordinateService"], [10, 1, 1, "", "VarCachingFunctionalAnnotator"], [10, 1, 1, "", "VariantAnnotationCache"], [10, 1, 1, "", "VariantCoordinateFinder"], [10, 1, 1, "", "VepFunctionalAnnotator"], [10, 4, 1, "", "configure_caching_cohort_creator"], [10, 4, 1, "", "configure_caching_patient_creator"], [10, 4, 1, "", "configure_cohort_creator"], [10, 4, 1, "", "configure_patient_creator"], [10, 4, 1, "", "load_phenopacket"], [10, 4, 1, "", "load_phenopacket_folder"]], "genophenocorr.preprocessing.Auditor": [[10, 2, 1, "", "prepare_notepad"], [10, 2, 1, "", "process"]], "genophenocorr.preprocessing.CohortCreator": [[10, 2, 1, "", "process"]], "genophenocorr.preprocessing.DataSanityIssue": [[10, 3, 1, "", "level"], [10, 3, 1, "", "message"], [10, 3, 1, "", "solution"]], "genophenocorr.preprocessing.FunctionalAnnotator": [[10, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.Level": [[10, 5, 1, "", "ERROR"], [10, 5, 1, "", "WARN"]], "genophenocorr.preprocessing.Notepad": [[10, 2, 1, "", "add_error"], [10, 2, 1, "", "add_issue"], [10, 2, 1, "", "add_subsection"], [10, 2, 1, "", "add_warning"], [10, 2, 1, "", "error_count"], [10, 2, 1, "", "errors"], [10, 3, 1, "", "issues"], [10, 3, 1, "", "label"], [10, 2, 1, "", "warning_count"], [10, 2, 1, "", "warnings"]], "genophenocorr.preprocessing.NotepadTree": [[10, 2, 1, "", "add_subsection"], [10, 3, 1, "", "children"], [10, 2, 1, "", "has_errors"], [10, 2, 1, "", "has_errors_or_warnings"], [10, 2, 1, "", "has_warnings"], [10, 2, 1, "", "iterate_nodes"], [10, 3, 1, "", "level"], [10, 2, 1, "", "visit"]], "genophenocorr.preprocessing.PhenopacketPatientCreator": [[10, 2, 1, "", "process"]], "genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder": [[10, 2, 1, "", "find_coordinates"]], "genophenocorr.preprocessing.PhenotypeCreator": [[10, 2, 1, "", "process"]], "genophenocorr.preprocessing.ProtCachingMetadataService": [[10, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.ProteinAnnotationCache": [[10, 2, 1, "", "get_annotations"], [10, 2, 1, "", "store_annotations"]], "genophenocorr.preprocessing.ProteinMetadataService": [[10, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.UniprotProteinMetadataService": [[10, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.VVHgvsVariantCoordinateFinder": [[10, 2, 1, "", "find_coordinates"]], "genophenocorr.preprocessing.VVTranscriptCoordinateService": [[10, 2, 1, "", "fetch"], [10, 2, 1, "", "parse_response"]], "genophenocorr.preprocessing.VarCachingFunctionalAnnotator": [[10, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.VariantAnnotationCache": [[10, 2, 1, "", "get_annotations"], [10, 2, 1, "", "store_annotations"]], "genophenocorr.preprocessing.VariantCoordinateFinder": [[10, 2, 1, "", "find_coordinates"]], "genophenocorr.preprocessing.VepFunctionalAnnotator": [[10, 5, 1, "", "NONCODING_EFFECTS"], [10, 2, 1, "", "annotate"]], "genophenocorr.view": [[11, 1, 1, "", "CohortViewer"], [11, 1, 1, "", "VariantTranscriptVisualizer"]], "genophenocorr.view.CohortViewer": [[11, 2, 1, "", "anchor"], [11, 2, 1, "", "cohort_summary_table"], [11, 2, 1, "", "hpo_anchor"], [11, 2, 1, "", "hpo_term_counts_table"], [11, 2, 1, "", "html_row"], [11, 2, 1, "", "protein_features_table"], [11, 2, 1, "", "variants_table"]], "genophenocorr.view.VariantTranscriptVisualizer": [[11, 2, 1, "", "draw_variants"]]}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "property", "Python property"], "4": ["py", "function", "Python function"], "5": ["py", "attribute", "Python attribute"]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:property", "4": "py:function", "5": "py:attribute"}, "terms": {"": [3, 4, 8, 10, 15, 16, 19], "0": [1, 2, 3, 8, 10, 16], "000": [1, 3, 16], "000000": 1, "0001060": 8, "0001166": [1, 4, 16], "0001250": 16, "0001257": 16, "0001566": 8, "0001567": 8, "0001574": 8, "0001575": 8, "0001578": 8, "0001580": 8, "0001583": 8, "0001587": 8, "0001589": 8, "0001619": 8, "0001620": 8, "0001621": 8, "0001623": 8, "0001624": 8, "0001626": 8, "0001627": 8, "0001628": 8, "0001630": 8, "0001631": 8, "0001632": 8, "0001782": 8, "0001787": 8, "0001792": 8, "0001818": 8, "0001819": 8, "0001821": 8, "0001822": 8, "0001889": 8, "0001891": 8, "0001892": 8, "0001893": 8, "0001894": 8, "0001895": 8, "0001906": 8, "0001907": 8, "0002012": 8, "0002019": 8, "0002169": 8, "0002170": 8, "0003011": 1, "000781": [1, 16], "0012638": 1, "020299": [1, 16], "03": 19, "05": 1, "06": 19, "1": [1, 2, 3, 8, 10, 11, 16], "10": [1, 3, 8, 10, 16], "100": 1, "101": 8, "11": [1, 8], "120": 8, "13": [1, 16], "14": 16, "15": 1, "16": [1, 16], "17": 16, "1_281_281_a_g": 16, "1_361_363_ttc_t": 16, "1st": 8, "2": [1, 3, 8, 10, 11, 16], "20": [8, 16], "2015": 14, "2022": 13, "22_10001_20000_inv": 8, "253a": 10, "29": 16, "3": [8, 10], "30": 10, "36303223": 13, "4": [8, 10], "5": [1, 8, 10, 16, 19], "6": [1, 13], "7": [1, 8], "81": [1, 16], "9": 1, "9876g": 8, "A": [1, 2, 3, 4, 8, 10, 11, 14, 16], "For": [3, 8, 10, 13, 16, 19], "IN": 10, "INS": 8, "If": [1, 3, 8], "In": [10, 13, 14], "NO": [2, 3, 4], "NOT": 8, "No": [1, 2, 16], "On": 4, "Or": 8, "That": 15, "The": [1, 2, 3, 4, 8, 9, 10, 14, 15, 16, 19], "Then": [4, 15], "There": 8, "To": [10, 15, 16], "_missing_implies_phenotype_exclud": 4, "abnorm": [1, 14], "abort": 10, "about": [8, 10, 15, 16, 17], "abov": [15, 16, 19], "absenc": [1, 4, 16], "abstract": [1, 2, 4, 8, 10], "access": [1, 3, 8, 10, 15], "acid": 8, "across": 11, "ad": 10, "add": [10, 15], "add_error": 10, "add_issu": 10, "add_subsect": 10, "add_warn": 10, "addit": 15, "affect": [1, 3, 8, 13], "affected_exon": 8, "after": [1, 15], "against": 13, "al": [13, 14], "albeit": 10, "all": [1, 2, 3, 4, 8, 10, 15, 16], "all_count": 1, "all_diseas": 8, "all_excluded_pati": 8, "all_pati": 8, "all_phenotyp": 8, "all_transcript": 8, "all_vari": 8, "allel": [3, 8, 13, 14], "allianc": 14, "along": 19, "alpha": 1, "alt": 8, "altern": 8, "amaz": 19, "amino": 8, "aminoacid": 8, "an": [1, 2, 3, 4, 8, 10, 11, 13, 14, 15, 16, 19], "analys": [1, 16, 19], "analysi": [0, 10, 13, 14, 19], "analyz": [0, 1, 11], "anchor": 11, "ani": [2, 10], "ank": 8, "ankyrin": 8, "annot": [1, 4, 8, 10, 11, 16, 19], "anoth": 19, "answer": [2, 3, 4], "api": [10, 14, 16, 19], "app": 8, "ar": [1, 2, 3, 8, 9, 10, 13, 14, 15, 16], "arachnodactyli": [1, 4, 16], "argument": 10, "arithmet": 9, "ascend": 16, "ask": 14, "aspect": 2, "assert": 8, "assign": 2, "associ": [8, 10, 13, 14], "assum": [1, 8], "attribut": [4, 8, 14], "audit": 10, "auditor": [0, 10], "avail": [1, 8, 10, 14], "b": [1, 8, 16], "background": 14, "bandwidth": 19, "base": [1, 2, 3, 4, 8, 10, 11, 16], "basic": [8, 9], "becaus": 13, "behavior": 14, "being": 10, "belong": 2, "best": 14, "better": 16, "between": [1, 8, 9, 14], "beyond": [1, 3], "bin": [1, 2, 4], "bind": 1, "biolog": 8, "biologi": 14, "bonferroni": 1, "bool": [1, 4, 8, 10], "boolean": [2, 8, 10], "booleanpred": [1, 2], "both": [3, 8, 10], "boundari": [8, 10], "box": 19, "branch": 15, "breakend": 8, "bridg": 18, "brief": 14, "bug": 14, "build": [9, 10], "c": [2, 4, 8, 10, 14, 16, 19], "cach": [10, 19], "cache_dir": [1, 10], "calcul": 8, "call": [3, 8, 10, 19], "callabl": 10, "caller": 10, "can": [1, 2, 3, 4, 8, 9, 10, 15, 16, 17, 19], "cannot": [1, 2, 8, 10], "canon": 8, "carri": 13, "case": 10, "cat_id": 2, "categor": [1, 2, 3, 4], "categori": [1, 2, 3, 4], "caus": 14, "cd": [8, 15], "cds_end": 8, "cds_start": 8, "certain": [2, 10], "chang": 8, "change_length": 8, "charact": 8, "characterist": 8, "check": [1, 2, 3, 8, 10], "checkout": 15, "children": 10, "choos": [1, 10, 16], "chr5": 8, "chrom": 8, "chromosom": [8, 13], "class": [1, 2, 3, 4, 8, 9, 10, 11], "clone": 15, "code": [8, 10, 11, 15, 16], "coding_sequence_vari": 8, "codon": [8, 14], "cohort": [0, 1, 7, 8, 10, 11, 14, 16], "cohort_analysi": 16, "cohort_cr": [10, 19], "cohort_summary_t": 11, "cohortanalysi": [0, 1], "cohortanalysisconfigur": [0, 1], "cohortcr": [0, 10, 19], "cohortview": [0, 11], "collect": [8, 10, 14], "column": 1, "com": 15, "combin": 8, "command": [15, 16], "common": [2, 13], "commonli": 9, "compact": 8, "compar": [1, 4, 13], "compare_by_exon": 1, "compare_by_protein_featur": 1, "compare_by_protein_feature_typ": 1, "compare_by_variant_effect": [1, 16], "compare_by_variant_kei": 1, "comparison": 13, "compat": 2, "complain": 10, "complet": 10, "compound": 13, "comprehens": 14, "comput": [1, 3, 10], "concept": 14, "concern": 14, "config": 1, "configur": [1, 10, 19], "configure_caching_cohort_cr": [0, 10, 19], "configure_caching_patient_cr": [0, 10], "configure_cohort_analysi": [0, 1, 16], "configure_cohort_cr": [0, 10], "configure_ontology_stor": 19, "configure_patient_cr": [0, 10], "consequ": 8, "conserv": 8, "consid": 8, "consist": 19, "constant": 0, "constrain": 2, "construct": [10, 19], "contain": [1, 8, 10], "context": [4, 14], "contig": [8, 9], "conveni": [10, 19], "coordin": [8, 9, 10, 19], "copi": 13, "correct": [1, 16], "corrected_pv": 1, "correl": [0, 1, 11, 14], "correspond": [1, 8, 10], "could": 11, "count": [1, 8, 10, 16, 17], "counter": 8, "counts_dict": 8, "cover": 2, "creat": [1, 8, 10, 11, 14], "create_variant_from_scratch": 8, "creation": 1, "creator": 10, "curi": [8, 10], "current": [10, 15, 19], "d": 16, "data": [0, 1, 8, 9, 10, 14, 16, 18], "databas": 10, "datadir": 10, "datafram": 1, "datasanityissu": [0, 10], "dataset": 19, "deep": 14, "default": 19, "defin": [8, 19], "definit": 8, "del": 8, "delet": 8, "demonstr": 16, "denot": 14, "depth": 10, "describ": [3, 8, 15, 17, 19], "descript": [2, 8, 10], "design": [11, 14], "desir": [10, 13], "detail": 2, "determin": 10, "develop": 15, "deviat": 14, "diagnos": 4, "dictionari": 8, "differenti": 14, "dimension": 8, "diploid": 8, "directori": [10, 19], "discret": [2, 4, 10], "diseas": [0, 4, 8, 14], "disease_id_queri": 4, "disease_set": 8, "diseasepresencepred": 4, "displai": 11, "distanc": 9, "distribut": [11, 17], "dna": [1, 8, 9], "do": [1, 3, 9, 10, 15, 16], "doc": [16, 19], "document": [14, 15], "doe": [1, 2, 3, 8, 10], "domain": 8, "domin": 13, "done": [10, 15], "down": 10, "downstream": [10, 19], "downstream_gene_vari": [8, 10], "draw_vari": 11, "driver": 1, "due": [8, 10, 16], "dup": 8, "duplic": 8, "dure": 10, "dysplasia": 4, "e": [1, 4, 8, 10, 14, 16], "each": [1, 4, 8, 10, 13, 14, 16], "easi": 15, "easiest": 19, "effect": [1, 3, 8, 10, 17, 19], "effect1": 3, "effect2": 3, "egf": 3, "either": [1, 3, 4, 8, 10], "element": [4, 8, 10], "els": 8, "empti": [8, 10], "enabl": 15, "encod": [3, 8], "encount": 10, "end": 8, "ensembl": 8, "enst00000123456": 8, "entir": 10, "entiti": [4, 8, 10], "enum": [8, 10], "environ": 15, "equal": 8, "error": 10, "error_count": 10, "establish": 10, "et": [13, 14], "etc": [8, 10], "exact": [13, 14], "exampl": [1, 8, 13, 16, 19], "except": 10, "exclud": [1, 4, 8], "excluded_diseas": 8, "excluded_memb": 8, "excluded_phenotyp": 8, "exclus": 2, "exhaust": 2, "exist": 10, "exon": [1, 3, 8], "exon1_numb": 3, "exon2_numb": 3, "exon_numb": [1, 3], "exonpred": 3, "exons_effect": 8, "exonspred": 3, "expand_frame_repr": 16, "expect": 10, "explain": 16, "explan": 11, "explicit": 4, "explicitli": [1, 8], "explor": 19, "exploratori": 19, "express": 10, "extract": 10, "f": [14, 16], "fallback": [1, 10], "fals": [1, 3, 4, 8, 10, 16], "fashion": 10, "fbn1": 3, "featur": [1, 3, 8, 10, 14, 15, 17], "feature_elong": 8, "feature_trunc": 8, "feature_typ": [1, 3, 8], "feature_type1": 3, "feature_type2": 3, "featureinfo": [0, 8], "featuretyp": [0, 1, 3, 8], "fetch": 10, "field": 13, "figur": 10, "file": [10, 19], "find": [1, 8, 9, 10], "find_coordin": 10, "finder": 10, "first": [1, 2, 3, 8, 10, 19], "fischer": 13, "fisher": 14, "five_prime_utr_vari": [8, 10], "fix": 10, "flip": 9, "float": [1, 10], "focu": 14, "fold": 8, "folder": [10, 19], "follow": [8, 15], "for_sampl": 8, "form": 19, "format": [4, 8, 10], "found": [8, 10], "fraction": 1, "frame": 1, "frameshift_vari": [8, 16], "friendli": 10, "from": [1, 8, 10, 14, 15, 16], "from_map": 8, "from_pati": 8, "from_term": 8, "frozen": 8, "frozenset": 8, "fuction": 10, "full": 8, "function": [3, 4, 8, 10, 19], "functionalannot": [0, 10], "functionalannotationawar": 8, "g": [1, 4, 8, 10, 14, 16], "ga4gh": 14, "geleophys": 4, "gene": [8, 13], "gene_id": 8, "gene_nam": 8, "gener": [2, 4, 10, 11, 14], "genet": 14, "geno_pred": 1, "genom": [8, 10, 14], "genome_build": 10, "genomebuild": [9, 10], "genomicinterpret": 10, "genomicregion": [8, 9], "genophenocorr": [12, 13, 15, 16, 19], "genotyp": [0, 1, 2, 8, 10, 11, 14, 16, 19], "genotype_for_sampl": 8, "genotypebooleanpred": [1, 2, 3], "genotypephenotypeanalysisresult": [0, 1], "genotypepolypred": [1, 2], "get": [1, 2, 3, 4, 8, 10, 16, 19], "get_annot": 10, "get_categor": [2, 4], "get_categori": 2, "get_cds_region": 8, "get_coding_base_count": 8, "get_codon_count": 8, "get_excluded_count": 8, "get_excluded_id": 8, "get_features_variant_overlap": 8, "get_five_prime_utr": 8, "get_quest": [2, 3, 4], "get_three_prime_utr": 8, "get_toy_cohort": [0, 7, 16], "gg": 8, "git": 15, "github": 15, "given": [1, 2, 3, 4, 8, 10], "global": 14, "gov": 13, "graphic": 11, "grch37": [9, 10], "grch38": [9, 10], "group": [1, 2, 13, 16], "groupingpred": [1, 2, 3], "gt": 8, "guid": 14, "h": 16, "ha": [1, 2, 3, 4, 8, 10], "happi": 2, "has_error": 10, "has_errors_or_warn": 10, "has_warn": 10, "hashabl": 8, "have": [1, 3, 10, 16], "head": 16, "health": 14, "help": 10, "hemizyg": 8, "henc": [10, 15], "here": [10, 16, 17, 19], "heterozyg": 8, "heterozygot": 13, "hgv": [8, 10], "hgvs_cdna": 8, "hgvs_coordinate_find": 10, "hgvsc": 8, "hierarch": 10, "high": 18, "highest": 8, "hold": 19, "homo": [1, 3], "homozygot": 13, "homozygous_altern": 8, "homozygous_refer": 8, "how": [15, 16, 17, 19], "howev": [2, 4, 10, 15], "hp": [1, 4, 16], "hpo": [1, 4, 8, 10, 11, 14, 16, 19], "hpo_anchor": 11, "hpo_id": 11, "hpo_term_counts_t": 11, "hpotk": [1, 4, 8, 16, 19], "html": 11, "html_row": 11, "http": [8, 10, 13, 15, 16], "human": [2, 9, 10, 14, 19], "i": [0, 1, 3, 4, 8, 10, 13, 14, 15, 16, 19], "id": [4, 8, 10], "ident": 13, "identifi": [1, 2, 4, 8, 10], "implement": [1, 10, 13], "implementor": 8, "impli": 4, "import": [8, 16, 19], "imposs": 1, "includ": [1, 3, 4, 8, 10, 14], "include_computational_tx": 10, "include_patients_with_no_hpo": 8, "include_patients_with_no_vari": 8, "include_subsect": 10, "include_sv": 1, "incomplete_terminal_codon_vari": 8, "index": [8, 15], "indic": 8, "individu": [8, 13, 14, 18], "info": [8, 10, 19], "inform": [8, 14], "inframe_delet": 8, "inframe_insert": 8, "ingest": 14, "inher": 2, "inherit": 2, "initi": [3, 4, 8, 15], "input": [10, 14, 16, 18], "insert": 8, "instal": [14, 19], "instanc": [2, 3, 8, 10, 13, 16, 19], "instruct": 14, "int": [1, 2, 3, 8, 10, 11], "integ": 8, "integr": 15, "interest": [1, 3, 4, 8], "intergenic_vari": [8, 10], "internet": 15, "interpret": 10, "intersect": 9, "intron": 8, "intron_vari": [8, 10], "inv": 8, "invalid": 1, "investig": [1, 2, 4], "invoc": 15, "involv": 19, "ipython": 11, "is_cod": 8, "is_observ": 8, "is_pref": 8, "is_pres": 8, "is_structur": 8, "issu": [10, 14], "item": [1, 10, 11, 16], "iter": [1, 8, 10, 11], "iterate_nod": 10, "its": [8, 10, 11], "j": 16, "jax": 8, "join": [16, 19], "json": [10, 16, 19], "jt": 13, "jupyt": [11, 14], "just": [1, 10, 16], "k": 16, "keep": [4, 10, 16], "kei": [1, 3, 8, 14], "kind": [10, 13], "know": [8, 10], "l": 16, "label": [8, 10, 11], "label_summari": 8, "lack": [4, 10], "larg": 9, "last": 8, "latest": 16, "lead": 8, "learn": [16, 17], "least": [1, 3, 10], "leav": 14, "len": [8, 19], "length": 8, "lenient": 10, "let": [8, 16, 19], "level": [0, 8, 10, 18], "li": 13, "librari": [0, 9, 15, 19], "like": [1, 3, 14], "line": 15, "list": [1, 4, 8, 10, 16], "list_all_diseas": 8, "list_all_pati": [8, 16], "list_all_phenotyp": [8, 16], "list_all_protein": [8, 16], "list_all_vari": [8, 16], "list_data_by_tx": [8, 16], "liter": 8, "ll": 19, "load": [10, 16, 19], "load_minimal_hpo": 19, "load_minimal_ontologi": 16, "load_phenopacket": [0, 10], "load_phenopacket_fold": [0, 10, 19], "local": [1, 10], "locat": [8, 14, 19], "locu": 8, "log": 1, "look": 1, "m": [15, 16], "made": 10, "mai": [8, 10], "make": 2, "malarkei": 10, "malform": 10, "mandatori": 8, "mane": 8, "mani": 17, "manifest": 14, "map": [1, 8], "marfan": 4, "matern": 13, "matter": 10, "mature_mirna_vari": 8, "me": 8, "meaning": 2, "medic": 14, "medicin": 14, "member": [8, 10, 16, 17, 19], "mendelian": 14, "messag": 10, "meta_label": 8, "metaclass": 10, "metadata": 10, "method": [1, 8, 10, 19], "mi": 1, "might": 13, "min_count": 11, "min_perc_patients_w_hpo": 1, "minimalontologi": [1, 4, 10], "minimalterm": 8, "minimum": 11, "missens": [1, 8, 13, 14, 16], "missense_vari": [1, 8, 16], "missing_implies_exclud": 1, "missing_implies_phenotype_exclud": 4, "mnv": 8, "mode": 2, "model": [0, 1, 2, 3, 4, 10, 16, 19], "modul": [5, 8, 9, 10], "moment": [1, 10], "monarch": 15, "mondo": 4, "more": [4, 8, 10, 15, 16, 19], "morphologi": 14, "most": [4, 9], "motif": [8, 14], "move": 16, "mtc_alpha": 1, "multipl": [1, 4], "musculatur": 1, "must": [2, 4, 10, 15], "n": [3, 8, 10, 16], "n_categor": 2, "n_usabl": 1, "nacgtacgtac": 8, "name": [1, 2, 3, 8, 10], "natur": 8, "ncbi": 13, "need": [3, 10, 16, 19], "neg": 8, "neither": 3, "nervou": 1, "next": 19, "nih": 13, "nlm": 13, "nm_002834": 10, "nm_005912": 10, "nm_1234": [1, 16], "nm_123456": [1, 8], "nmd_transcript_vari": 8, "no_cal": 8, "node": 10, "non": [8, 10], "non_coding_transcript_exon_vari": [8, 10], "non_coding_transcript_vari": [8, 10], "noncoding_effect": 10, "none": [1, 2, 3, 4, 8, 10], "nons": 13, "nonsens": 13, "normal": 14, "notat": [1, 8], "note": [2, 4, 8], "notebook": [11, 14, 16], "notepad": [0, 10], "notepadtre": [0, 10], "now": 16, "np_037407": 10, "np_09876": 16, "number": [1, 2, 3, 8, 11], "numer": 2, "o": [16, 19], "object": [1, 2, 4, 8, 10, 11], "obo": 16, "obolibrari": 16, "observ": [1, 8, 10, 14], "observablefeatur": 8, "off": 3, "offici": 8, "offlin": 10, "omim": 4, "one": [1, 2, 3, 4, 8, 10, 13, 15], "onli": [1, 2, 4, 8, 10, 13], "onlin": 15, "ontologi": [8, 14, 16, 19], "oppos": 8, "optim": 10, "option": [1, 2, 3, 4, 8, 10, 15, 19], "order": [8, 16], "org": [8, 10, 16], "organ": [8, 10], "orgogozo": 14, "other": [1, 8, 10, 13, 14], "otherwis": [1, 4, 8], "out": [10, 19], "over": [8, 10], "overlap": [3, 8, 9], "overlapping_exon": 8, "overview": 18, "p": [1, 4, 16], "p13": [9, 10], "packag": [12, 14, 15, 16], "pair": 8, "panda": [1, 16], "param": [8, 10], "paramet": [1, 3, 4, 8, 10, 11], "parameter": 1, "parent": 10, "pars": 10, "parse_respons": 10, "part": 8, "partit": 16, "pass": 1, "patern": 13, "path": [2, 10, 16, 19], "patient": [0, 1, 2, 3, 4, 8, 10, 11, 14, 16, 19], "patient_cr": 10, "patient_id": 8, "patient_set": 8, "patientcategori": [1, 2, 3, 4, 16], "patientcr": [0, 10], "pattern": 10, "payload": 11, "pd": 16, "per": [1, 11], "percent": [1, 16], "percentag": 1, "perform": [10, 13, 14, 16, 19], "phenopacket": [10, 14, 16], "phenopacket_path": 10, "phenopacketpatientcr": [0, 10], "phenopacketvariantcoordinatefind": [0, 10], "phenotyp": [0, 1, 8, 10, 11, 14, 16, 17, 19], "phenotype_categori": 1, "phenotype_cr": 10, "phenotype_set": 8, "phenotypecategor": 4, "phenotypecr": [0, 10], "phenotypepolypred": 4, "physiologi": [1, 14], "pickl": 10, "pip": 15, "place": 14, "plot": 17, "point": 10, "polici": 10, "polypred": [1, 2, 4], "port": 9, "posit": [1, 3, 8, 10], "possibl": [2, 13], "pp_directori": 10, "practic": 13, "precis": 14, "predic": [1, 16], "predict": [3, 8], "predictor": [10, 19], "prefer": 8, "preferred_transcript": 11, "prematur": 14, "prepar": [2, 3, 4, 10, 19], "prepare_notepad": 10, "preprocess": [0, 19], "presenc": [1, 8, 16], "present": [1, 3, 8, 16], "present_diseas": 8, "present_phenotyp": 8, "previou": 1, "principl": 13, "print": 8, "proce": 10, "process": 10, "produc": [2, 4], "product": 11, "programmatic_access": 10, "prop": 8, "propagatingphenotypepred": 4, "properti": [1, 2, 4, 8, 10], "protcachingmetadataservic": [0, 10], "protein": [1, 3, 8, 10, 11, 14, 19], "protein_altering_vari": 8, "protein_annot": 10, "protein_effect_coordin": 8, "protein_effect_end": 8, "protein_effect_loc": 8, "protein_effect_start": 8, "protein_featur": 8, "protein_feature1_nam": 3, "protein_feature2_nam": 3, "protein_feature_nam": 3, "protein_features_t": 11, "protein_id": [8, 10], "protein_servic": 3, "protein_sourc": 1, "proteinannotationcach": [0, 10], "proteinfeatur": [0, 1, 8], "proteinmetadata": [0, 8, 10, 11], "proteinmetadataservic": [0, 3, 10], "protfeaturepred": 3, "protfeaturespred": 3, "protfeaturetypepred": 3, "protfeaturetypespred": 3, "protocol": 4, "provid": [1, 4, 9, 10, 11, 14, 19], "prune": 10, "publish": 15, "pubm": 13, "purl": 16, "purpos": 19, "pval": 1, "pval_correct": 1, "pypi": 15, "pytest": 15, "python": [14, 15], "python3": 15, "q": [16, 19], "qualnam": [8, 10], "queri": [4, 8], "question": [2, 3, 4, 14], "r": 16, "rais": 10, "rare": [0, 1], "raw": 19, "re": 16, "read": 14, "readabl": [2, 8], "recess": 8, "recessiveexonpred": 3, "recessivegroupingpred": 3, "recessiveprotfeaturepred": 3, "recessiveprotfeaturetypepred": 3, "recessivevarianteffectpred": 3, "recessivevariantpred": 3, "recommend": 14, "record": 10, "ref": 8, "refer": [8, 14], "refseq": [8, 10], "region": [8, 9], "regulatory_region_abl": 8, "regulatory_region_amplif": 8, "regulatory_region_vari": 8, "relat": 10, "relationship": 14, "releas": 19, "relev": 8, "remot": 10, "remov": [1, 10], "repeat": 8, "report": [10, 14, 16], "repositori": [15, 19], "repres": [1, 2, 3, 4, 8, 10, 14], "represent": [8, 10], "request": 10, "requir": [10, 15, 19], "requisit": 19, "resourc": [1, 10], "respect": [3, 8], "respons": [10, 19], "rest": [1, 10], "restrict": 13, "result": [1, 8, 10, 14, 16], "retain": 1, "retriev": [1, 10], "return": [1, 2, 3, 4, 8, 10, 11], "right": 10, "root": 10, "row": [1, 16], "run": [1, 10, 11, 16], "runner": 10, "runonlin": 15, "sampl": 8, "sample_id": 8, "samplelabel": [0, 8], "sanit": 10, "sanitationresult": 10, "saniti": 10, "sapien": [1, 3], "satisfi": 10, "save": 19, "scenario": 2, "schema": [10, 19], "scienc": [1, 3], "search": [8, 10], "second": [1, 2, 3, 10], "secondari": 8, "section": [10, 16, 18, 19], "see": [13, 16, 19], "sens": 2, "sequenc": [1, 2, 4, 8, 10], "sequence_vari": 8, "sequenceontologi": 8, "seri": 1, "seriou": 10, "serv": 2, "servic": 10, "set": [1, 8], "set_opt": 16, "setter": 1, "sever": [2, 8, 10], "ship": 16, "short": 8, "shortcut": 8, "should": [4, 10], "show": [11, 16], "signific": 8, "simpl": [10, 13, 15, 16], "simple_cohort": 19, "simple_cohort_path": 19, "sinc": 8, "sine": 8, "singl": [8, 10], "size": 8, "snv": 8, "so": [1, 3, 8], "solut": [10, 19], "some": [10, 14, 15, 16], "someth": 10, "soon": 16, "sort": [10, 16], "sought": 4, "sourc": [1, 2, 3, 4, 7, 8, 10, 11, 15], "span": [1, 3, 8], "specif": [1, 8, 10, 14], "splice_acceptor_vari": [8, 10], "splice_donor_5th_base_vari": [8, 10], "splice_donor_region_vari": 8, "splice_donor_vari": [8, 10], "splice_polypyrimidine_tract_vari": [8, 10], "splice_region_vari": 8, "standard": 19, "start": [8, 10, 19], "start_lost": 8, "start_retained_vari": 8, "state": 8, "static": [2, 8, 10, 11], "statist": 14, "statu": 10, "step": [15, 19], "stop_gain": 8, "stop_lost": 8, "stop_retained_vari": 8, "storag": 10, "store": [1, 10, 19], "store_annot": 10, "str": [1, 2, 3, 4, 8, 10, 11], "strand": 9, "strict": 10, "string": [1, 3, 8, 10], "structur": [1, 8, 10], "studi": 14, "sub": [8, 10], "subgroup": 8, "subject": 8, "submodul": 0, "subpackag": 12, "subsect": [8, 10], "summar": [1, 8, 10, 16], "summari": 1, "summary_df": 16, "suox": 13, "support": [8, 13, 14], "surf1": 8, "svart": 9, "svlen": 8, "svtype": 8, "switch": 15, "symbol": [1, 8], "syndrom": 4, "synonymous_vari": 8, "system": [1, 8], "t": [8, 10, 16], "tabl": [11, 16], "take": 10, "target": [3, 4], "taxon": [1, 3], "technic": 14, "term": [1, 4, 8, 10, 11, 14, 16, 19], "term_id": 8, "termid": [4, 8], "termin": [8, 14], "test": [1, 2, 3, 4, 8, 11, 13, 14], "tf_binding_site_vari": 8, "tfbs_ablat": 8, "tfbs_amplif": 8, "th": 3, "than": 8, "thei": [9, 14], "them": 10, "therefor": [1, 3], "thi": [1, 3, 4, 8, 10, 13, 14, 15, 16, 19], "thing": 16, "those": 1, "three": [8, 13], "three_prime_utr_vari": [8, 10], "threshold": 1, "through": 10, "thu": 1, "time": [4, 10, 15], "timeout": 10, "to_str": 8, "todo": [16, 17, 18, 19], "togeth": 19, "toi": 16, "toolkit": [10, 19], "top": [4, 8, 10], "total": 8, "total_patient_count": 8, "track": [4, 10], "tracker": 14, "trans_id": 8, "transcript": [1, 3, 8, 10, 11], "transcript_abl": 8, "transcript_amplif": 8, "transcript_id": [1, 3, 8], "transcriptannot": [0, 8, 10], "transcriptcoordin": [0, 8, 10, 11], "transcriptcoordinateservic": 10, "transcriptinfoawar": [0, 8, 10], "transform": 10, "translat": 8, "transpos": 9, "travers": 10, "tree": 10, "true": [1, 4, 8, 10], "tupl": [8, 10], "turn": 10, "tutori": 14, "two": [1, 2, 3, 9, 13, 16], "tx": [10, 11], "tx_annot": 8, "tx_dict": 16, "tx_id": [1, 8, 10, 16], "type": [1, 3, 4, 8, 10, 14], "u": 16, "under": 10, "unfix": 10, "uniprot": [1, 10, 19], "uniprotproteinmetadataservic": [0, 10], "uniqu": [2, 8], "unit": 15, "unknown": 8, "unless": [4, 10], "unlik": 10, "unrunn": 10, "unstabl": 16, "untransl": 8, "updat": 16, "upon": 1, "upstream_gene_vari": [8, 10], "uri": 8, "url": 11, "us": [1, 2, 3, 8, 9, 10, 11, 14, 16, 19], "usabl": 1, "user": [8, 14], "usual": [1, 3, 8], "util": 2, "utr": 8, "v": [1, 14, 16], "v2024": 19, "valid": 10, "validation_polici": 10, "validation_runn": 10, "validationrunn": 10, "valu": [8, 10, 16], "valueerror": 10, "var_coordin": 8, "var_func_ann": 10, "varcachingfunctionalannot": [0, 10], "variabl": 8, "variant": [0, 1, 2, 3, 8, 10, 11, 14, 16, 17, 19], "variant_class": 8, "variant_coordin": [8, 10], "variant_effect": [5, 8], "variant_fallback": 10, "variant_kei": [1, 3, 8], "variant_key1": 3, "variant_key2": 3, "variant_set": 8, "variant_str": 8, "variantannotationcach": [0, 10], "variantcal": 10, "variantcoordin": [0, 1, 3, 8, 10], "variantcoordinateawar": 8, "variantcoordinatefind": [0, 10], "varianteffect": [0, 1, 3, 8, 10, 16], "varianteffectpred": [1, 3], "varianteffectspred": 3, "variantpred": 3, "variants_t": 11, "variantspred": 3, "varianttranscriptproteinartist": 11, "varianttranscriptvisu": [0, 11], "variou": 1, "vcf": [1, 8], "vep": [8, 10], "vepfunctionalannot": [0, 10], "verbos": 2, "verifi": 8, "via": 10, "view": [0, 14, 16], "virtual": 15, "visit": 10, "visitor": 10, "vvhgvsvariantcoordinatefind": [0, 10], "vvtranscriptcoordinateservic": [0, 10], "w": 16, "wa": [1, 4, 8, 10], "wai": 19, "want": [1, 16], "warn": [1, 10], "warning_count": 10, "we": [1, 3, 8, 10, 14, 15, 16, 17, 19], "well": [1, 3, 4, 8, 16], "were": 10, "what": [10, 16, 17], "when": [3, 4, 10], "where": [1, 8, 10], "whether": [3, 4, 8, 14], "which": [1, 2, 3, 4, 16, 19], "while": 13, "who": [2, 13], "within": [8, 10], "without": 1, "word": 14, "work": [10, 19], "workflow": 16, "would": [13, 14], "wrap": 1, "write": [10, 17, 18], "www": [8, 10], "x": 16, "x_12345_12345_c_g": [1, 8], "xm_": 10, "y": 16, "ye": [1, 2, 3, 4, 16], "you": [3, 8, 16], "your": [15, 16], "z": 16, "zero": 8}, "titles": ["genophenocorr package", "genophenocorr.analysis package", "genophenocorr.analysis.predicate package", "genophenocorr.analysis.predicate.genotype package", "genophenocorr.analysis.predicate.phenotype package", "genophenocorr.constants package", "genophenocorr.constants.variant_effects module", "genophenocorr.data package", "genophenocorr.model package", "genophenocorr.model.genome package", "genophenocorr.preprocessing package", "genophenocorr.view package", "API reference", "Genotype-Phenotype Correlations in Autosomal Recessive Diseases", "genophenocorr", "Installation", "Tutorial", "Cohort exploratory analysis", "User guide", "Input data"], "titleterms": {"analysi": [1, 2, 3, 4, 16, 17], "api": 12, "autosom": 13, "cohort": [17, 19], "constant": [5, 6], "content": 18, "correl": 13, "creat": 19, "data": [7, 19], "default": 1, "diseas": 13, "exploratori": 17, "feedback": 14, "from": 19, "ga4gh": 19, "genom": 9, "genophenocorr": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 14], "genotyp": [3, 13], "guid": 18, "input": 19, "instal": 15, "latest": 15, "literatur": 14, "model": [8, 9], "modul": 6, "other": 19, "packag": [0, 1, 2, 3, 4, 5, 7, 8, 9, 10, 11], "phenopacket": 19, "phenotyp": [4, 13], "predic": [2, 3, 4], "prepar": 16, "preprocess": 10, "recess": 13, "refer": 12, "releas": 15, "run": 15, "sampl": 16, "set": 16, "stabl": 15, "submodul": 5, "subpackag": [0, 1], "test": 15, "tutori": [13, 16], "up": 16, "user": 18, "valu": 1, "variant_effect": 6, "view": 11}})