Search.setIndex({"alltitles": {"API reference": [[13, null]], "Background": [[22, "background"]], "Cohort exploratory analysis": [[18, null]], "Compound predicates": [[23, "compound-predicates"]], "Contents:": [[20, null]], "Create a cohort from GA4GH phenopackets": [[21, "create-a-cohort-from-ga4gh-phenopackets"]], "Create a cohort from other data": [[21, "create-a-cohort-from-other-data"]], "Default values": [[1, "default-values"]], "Feedback": [[14, "feedback"]], "Fisher exact test (FET)": [[24, "fisher-exact-test-fet"]], "General HPO terms": [[19, null]], "Genotype-Phenotype Correlations in Autosomal Recessive Diseases": [[17, null]], "Implementation in genophenocorr": [[22, "implementation-in-genophenocorr"]], "Input data": [[21, null]], "Installation": [[15, null]], "Latest release": [[15, "latest-release"]], "Literature": [[14, "literature"]], "MTC Heuristics: Choosing which terms to test": [[22, "mtc-heuristics-choosing-which-terms-to-test"]], "Mann-Whitney U Test": [[24, "mann-whitney-u-test"]], "Multiple-testing correction": [[22, null]], "Multiple-testing correction (MTC) procedures": [[22, "multiple-testing-correction-mtc-procedures"]], "Predicates": [[23, null]], "Prepare samples": [[16, "prepare-samples"]], "Run tests": [[15, "run-tests"]], "Set up analysis": [[16, "set-up-analysis"]], "Simple predicates": [[23, "simple-predicates"]], "Stable release": [[15, "stable-release"]], "Statistical tests": [[24, null]], "Submodules": [[5, "submodules"]], "Subpackages": [[0, "subpackages"], [1, "subpackages"]], "Tutorial": [[16, null], [17, "tutorial"]], "User guide": [[20, null]], "genophenocorr": [[14, null]], "genophenocorr package": [[0, null]], "genophenocorr.analysis package": [[1, null]], "genophenocorr.analysis.predicate package": [[2, null]], "genophenocorr.analysis.predicate.genotype package": [[3, null]], "genophenocorr.analysis.predicate.phenotype package": [[4, null]], "genophenocorr.constants package": [[5, null]], "genophenocorr.constants.variant_effects module": [[6, null]], "genophenocorr.data package": [[7, null]], "genophenocorr.io module": [[8, null]], "genophenocorr.model package": [[9, null]], "genophenocorr.model.genome package": [[10, null]], "genophenocorr.preprocessing package": [[11, null]], "genophenocorr.view package": [[12, null]], "heuristic sampler strategy": [[22, "heuristic-sampler-strategy"]], "specify terms strategy": [[22, "specify-terms-strategy"]]}, "docnames": ["apidocs/genophenocorr", "apidocs/genophenocorr.analysis", "apidocs/genophenocorr.analysis.predicate", "apidocs/genophenocorr.analysis.predicate.genotype", "apidocs/genophenocorr.analysis.predicate.phenotype", "apidocs/genophenocorr.constants", "apidocs/genophenocorr.constants.variant_effects", "apidocs/genophenocorr.data", "apidocs/genophenocorr.io", "apidocs/genophenocorr.model", "apidocs/genophenocorr.model.genome", "apidocs/genophenocorr.preprocessing", "apidocs/genophenocorr.view", "apidocs/modules", "index", "installation", "tutorial", "user-guide/autosomal_recessive", "user-guide/exploratory", "user-guide/general_hpo_terms", "user-guide/index", "user-guide/input-data", "user-guide/mtc", "user-guide/predicates", "user-guide/stats"], "envversion": {"sphinx": 62, "sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1}, "filenames": ["apidocs/genophenocorr.rst", "apidocs/genophenocorr.analysis.rst", "apidocs/genophenocorr.analysis.predicate.rst", "apidocs/genophenocorr.analysis.predicate.genotype.rst", "apidocs/genophenocorr.analysis.predicate.phenotype.rst", "apidocs/genophenocorr.constants.rst", "apidocs/genophenocorr.constants.variant_effects.rst", "apidocs/genophenocorr.data.rst", "apidocs/genophenocorr.io.rst", "apidocs/genophenocorr.model.rst", "apidocs/genophenocorr.model.genome.rst", "apidocs/genophenocorr.preprocessing.rst", "apidocs/genophenocorr.view.rst", "apidocs/modules.rst", "index.rst", "installation.rst", "tutorial.rst", "user-guide/autosomal_recessive.rst", "user-guide/exploratory.rst", "user-guide/general_hpo_terms.rst", "user-guide/index.rst", "user-guide/input-data.rst", "user-guide/mtc.rst", "user-guide/predicates.rst", "user-guide/stats.rst"], "indexentries": {"add_error() (genophenocorr.preprocessing.notepad method)": [[11, "genophenocorr.preprocessing.Notepad.add_error", false]], "add_issue() (genophenocorr.preprocessing.notepad method)": [[11, "genophenocorr.preprocessing.Notepad.add_issue", false]], "add_subsection() (genophenocorr.preprocessing.notepad method)": [[11, "genophenocorr.preprocessing.Notepad.add_subsection", false]], "add_subsection() (genophenocorr.preprocessing.notepadtree method)": [[11, "genophenocorr.preprocessing.NotepadTree.add_subsection", false]], "add_warning() (genophenocorr.preprocessing.notepad method)": [[11, "genophenocorr.preprocessing.Notepad.add_warning", false]], "all_counts (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.all_counts", false]], "all_diseases() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.all_diseases", false]], "all_hpo_terms (genophenocorr.analysis.mtc_strategy attribute)": [[1, "genophenocorr.analysis.MTC_Strategy.ALL_HPO_TERMS", false]], "all_patients (genophenocorr.model.cohort property)": [[9, "genophenocorr.model.Cohort.all_patients", false]], "all_phenotypes() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.all_phenotypes", false]], "all_terms_strategy() (genophenocorr.analysis.cohortanalysisconfiguration method)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.all_terms_strategy", false]], "all_transcript_ids (genophenocorr.model.cohort property)": [[9, "genophenocorr.model.Cohort.all_transcript_ids", false]], "all_variants() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.all_variants", false]], "allelecounter (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.AlleleCounter", false]], "alt (genophenocorr.model.variantcoordinates attribute)": [[9, "genophenocorr.model.VariantCoordinates.alt", false]], "alt (genophenocorr.model.variantcoordinates property)": [[9, "id9", false]], "annotate() (genophenocorr.preprocessing.defaultimprecisesvfunctionalannotator method)": [[11, "genophenocorr.preprocessing.DefaultImpreciseSvFunctionalAnnotator.annotate", false]], "annotate() (genophenocorr.preprocessing.functionalannotator method)": [[11, "genophenocorr.preprocessing.FunctionalAnnotator.annotate", false]], "annotate() (genophenocorr.preprocessing.imprecisesvfunctionalannotator method)": [[11, "genophenocorr.preprocessing.ImpreciseSvFunctionalAnnotator.annotate", false]], "annotate() (genophenocorr.preprocessing.protcachingmetadataservice method)": [[11, "genophenocorr.preprocessing.ProtCachingMetadataService.annotate", false]], "annotate() (genophenocorr.preprocessing.proteinmetadataservice method)": [[11, "genophenocorr.preprocessing.ProteinMetadataService.annotate", false]], "annotate() (genophenocorr.preprocessing.uniprotproteinmetadataservice method)": [[11, "genophenocorr.preprocessing.UniprotProteinMetadataService.annotate", false]], "annotate() (genophenocorr.preprocessing.varcachingfunctionalannotator method)": [[11, "genophenocorr.preprocessing.VarCachingFunctionalAnnotator.annotate", false]], "annotate() (genophenocorr.preprocessing.vepfunctionalannotator method)": [[11, "genophenocorr.preprocessing.VepFunctionalAnnotator.annotate", false]], "apply_predicates_on_patients() (in module genophenocorr.analysis)": [[1, "genophenocorr.analysis.apply_predicates_on_patients", false]], "auditor (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.Auditor", false]], "bnd (genophenocorr.model.variantclass attribute)": [[9, "genophenocorr.model.VariantClass.BND", false]], "boolean_predicate() (in module genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.boolean_predicate", false]], "cat_id (genophenocorr.analysis.predicate.patientcategory property)": [[2, "genophenocorr.analysis.predicate.PatientCategory.cat_id", false]], "categorization (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.Categorization", false]], "category (genophenocorr.analysis.predicate.categorization property)": [[2, "genophenocorr.analysis.predicate.Categorization.category", false]], "cds_end (genophenocorr.model.transcriptcoordinates property)": [[9, "genophenocorr.model.TranscriptCoordinates.cds_end", false]], "cds_start (genophenocorr.model.transcriptcoordinates property)": [[9, "genophenocorr.model.TranscriptCoordinates.cds_start", false]], "change_length (genophenocorr.model.variantcoordinates attribute)": [[9, "genophenocorr.model.VariantCoordinates.change_length", false]], "change_length (genophenocorr.model.variantcoordinates property)": [[9, "id10", false]], "change_length() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.change_length", false]], "children (genophenocorr.preprocessing.notepadtree property)": [[11, "genophenocorr.preprocessing.NotepadTree.children", false]], "chrom (genophenocorr.model.variantcoordinates property)": [[9, "genophenocorr.model.VariantCoordinates.chrom", false]], "code (genophenocorr.model.genotype property)": [[9, "genophenocorr.model.Genotype.code", false]], "coding_sequence_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.CODING_SEQUENCE_VARIANT", false]], "cohort (class in genophenocorr.model)": [[9, "genophenocorr.model.Cohort", false]], "cohortanalysis (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.CohortAnalysis", false]], "cohortanalysisconfiguration (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration", false]], "cohortcreator (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.CohortCreator", false]], "cohortviewable (class in genophenocorr.view)": [[12, "genophenocorr.view.CohortViewable", false]], "compare_by_exon() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_exon", false]], "compare_by_exons() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_exons", false]], "compare_by_protein_feature() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature", false]], "compare_by_protein_feature_type() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature_type", false]], "compare_by_protein_feature_types() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_protein_feature_types", false]], "compare_by_protein_features() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_protein_features", false]], "compare_by_protein_region() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_protein_region", false]], "compare_by_protein_regions() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_protein_regions", false]], "compare_by_variant_effect() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_variant_effect", false]], "compare_by_variant_effects() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_variant_effects", false]], "compare_by_variant_key() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_variant_key", false]], "compare_by_variant_keys() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_by_variant_keys", false]], "compare_hpo_vs_genotype() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_hpo_vs_genotype", false]], "compare_hpo_vs_genotype_groups() (genophenocorr.analysis.cohortanalysis method)": [[1, "genophenocorr.analysis.CohortAnalysis.compare_hpo_vs_genotype_groups", false]], "configure_caching_cohort_creator() (in module genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.configure_caching_cohort_creator", false]], "configure_caching_patient_creator() (in module genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.configure_caching_patient_creator", false]], "configure_cohort_analysis() (in module genophenocorr.analysis)": [[1, "genophenocorr.analysis.configure_cohort_analysis", false]], "configure_cohort_creator() (in module genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.configure_cohort_creator", false]], "configure_default_protein_metadata_service() (in module genophenocorr.analysis)": [[1, "genophenocorr.analysis.configure_default_protein_metadata_service", false]], "configure_patient_creator() (in module genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.configure_patient_creator", false]], "corrected_pvals (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.corrected_pvals", false]], "count() (genophenocorr.analysis.predicate.genotype.allelecounter method)": [[3, "genophenocorr.analysis.predicate.genotype.AlleleCounter.count", false]], "countingphenotypepredicate (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.CountingPhenotypePredicate", false]], "create() (genophenocorr.model.proteinfeature static method)": [[9, "genophenocorr.model.ProteinFeature.create", false]], "create_variant_from_scratch() (genophenocorr.model.variant static method)": [[9, "genophenocorr.model.Variant.create_variant_from_scratch", false]], "curie (genophenocorr.model.varianteffect property)": [[9, "genophenocorr.model.VariantEffect.curie", false]], "datasanityissue (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.DataSanityIssue", false]], "default() (genophenocorr.io.genophenocorrjsonencoder method)": [[8, "genophenocorr.io.GenophenocorrJSONEncoder.default", false]], "defaultimprecisesvfunctionalannotator (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.DefaultImpreciseSvFunctionalAnnotator", false]], "del (genophenocorr.model.variantclass attribute)": [[9, "genophenocorr.model.VariantClass.DEL", false]], "description (genophenocorr.analysis.predicate.patientcategory property)": [[2, "genophenocorr.analysis.predicate.PatientCategory.description", false]], "disease (class in genophenocorr.model)": [[9, "genophenocorr.model.Disease", false]], "diseasepresencepredicate (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate", false]], "diseases (genophenocorr.model.patient attribute)": [[9, "genophenocorr.model.Patient.diseases", false]], "diseases (genophenocorr.model.patient property)": [[9, "id2", false]], "diseaseviewable (class in genophenocorr.view)": [[12, "genophenocorr.view.DiseaseViewable", false]], "domain (genophenocorr.model.featuretype attribute)": [[9, "genophenocorr.model.FeatureType.DOMAIN", false], [9, "id30", false]], "domains() (genophenocorr.model.proteinmetadata method)": [[9, "genophenocorr.model.ProteinMetadata.domains", false], [9, "id22", false]], "downstream_gene_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.DOWNSTREAM_GENE_VARIANT", false]], "draw_fig() (genophenocorr.view.proteinvisualizer method)": [[12, "genophenocorr.view.ProteinVisualizer.draw_fig", false]], "draw_variants() (genophenocorr.view.varianttranscriptvisualizer method)": [[12, "genophenocorr.view.VariantTranscriptVisualizer.draw_variants", false]], "dup (genophenocorr.model.variantclass attribute)": [[9, "genophenocorr.model.VariantClass.DUP", false]], "empty() (genophenocorr.model.genotypes static method)": [[9, "genophenocorr.model.Genotypes.empty", false]], "end (genophenocorr.model.featureinfo property)": [[9, "genophenocorr.model.FeatureInfo.end", false]], "end (genophenocorr.model.variantcoordinates property)": [[9, "genophenocorr.model.VariantCoordinates.end", false]], "error (genophenocorr.preprocessing.level attribute)": [[11, "genophenocorr.preprocessing.Level.ERROR", false]], "error_count() (genophenocorr.preprocessing.notepad method)": [[11, "genophenocorr.preprocessing.Notepad.error_count", false]], "errors() (genophenocorr.preprocessing.notepad method)": [[11, "genophenocorr.preprocessing.Notepad.errors", false]], "excluded_diseases() (genophenocorr.model.patient method)": [[9, "genophenocorr.model.Patient.excluded_diseases", false]], "excluded_phenotypes() (genophenocorr.model.patient method)": [[9, "genophenocorr.model.Patient.excluded_phenotypes", false]], "exon() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.exon", false]], "exons (genophenocorr.model.transcriptcoordinates property)": [[9, "genophenocorr.model.TranscriptCoordinates.exons", false]], "feature_elongation (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.FEATURE_ELONGATION", false]], "feature_truncation (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.FEATURE_TRUNCATION", false]], "feature_type (genophenocorr.model.proteinfeature property)": [[9, "genophenocorr.model.ProteinFeature.feature_type", false]], "featureinfo (class in genophenocorr.model)": [[9, "genophenocorr.model.FeatureInfo", false]], "featuretype (class in genophenocorr.model)": [[9, "genophenocorr.model.FeatureType", false]], "fetch() (genophenocorr.preprocessing.transcriptcoordinateservice method)": [[11, "genophenocorr.preprocessing.TranscriptCoordinateService.fetch", false]], "fetch() (genophenocorr.preprocessing.vvmulticoordinateservice method)": [[11, "genophenocorr.preprocessing.VVMultiCoordinateService.fetch", false]], "fetch_for_gene() (genophenocorr.preprocessing.genecoordinateservice method)": [[11, "genophenocorr.preprocessing.GeneCoordinateService.fetch_for_gene", false]], "fetch_for_gene() (genophenocorr.preprocessing.vvmulticoordinateservice method)": [[11, "genophenocorr.preprocessing.VVMultiCoordinateService.fetch_for_gene", false]], "fetch_response() (genophenocorr.preprocessing.vepfunctionalannotator method)": [[11, "genophenocorr.preprocessing.VepFunctionalAnnotator.fetch_response", false]], "filter_method (genophenocorr.analysis.hpomtcreport property)": [[1, "genophenocorr.analysis.HpoMtcReport.filter_method", false]], "filter_method_name() (genophenocorr.analysis.heuristicmtcfilter method)": [[1, "genophenocorr.analysis.HeuristicMtcFilter.filter_method_name", false]], "filter_method_name() (genophenocorr.analysis.specifiedtermsmtcfilter method)": [[1, "genophenocorr.analysis.SpecifiedTermsMtcFilter.filter_method_name", false]], "filter_terms_to_test() (genophenocorr.analysis.heuristicmtcfilter method)": [[1, "genophenocorr.analysis.HeuristicMtcFilter.filter_terms_to_test", false]], "filter_terms_to_test() (genophenocorr.analysis.specifiedtermsmtcfilter method)": [[1, "genophenocorr.analysis.SpecifiedTermsMtcFilter.filter_terms_to_test", false]], "find_coordinates() (genophenocorr.preprocessing.phenopacketvariantcoordinatefinder method)": [[11, "genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder.find_coordinates", false]], "find_coordinates() (genophenocorr.preprocessing.variantcoordinatefinder method)": [[11, "genophenocorr.preprocessing.VariantCoordinateFinder.find_coordinates", false]], "find_coordinates() (genophenocorr.preprocessing.vvhgvsvariantcoordinatefinder method)": [[11, "genophenocorr.preprocessing.VVHgvsVariantCoordinateFinder.find_coordinates", false]], "first (genophenocorr.analysis.predicate.groupingpredicate attribute)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate.FIRST", false]], "five_prime_utr_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.FIVE_PRIME_UTR_VARIANT", false]], "for_sample() (genophenocorr.model.genotypes method)": [[9, "genophenocorr.model.Genotypes.for_sample", false]], "format_as_string() (genophenocorr.view.variantformatter method)": [[12, "genophenocorr.view.VariantFormatter.format_as_string", false]], "format_coordinates_for_vep_query() (genophenocorr.preprocessing.vepfunctionalannotator static method)": [[11, "genophenocorr.preprocessing.VepFunctionalAnnotator.format_coordinates_for_vep_query", false]], "frameshift_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.FRAMESHIFT_VARIANT", false]], "from_mapping() (genophenocorr.model.genotypes static method)": [[9, "genophenocorr.model.Genotypes.from_mapping", false]], "from_patients() (genophenocorr.model.cohort static method)": [[9, "genophenocorr.model.Cohort.from_patients", false]], "from_query_curies() (genophenocorr.analysis.predicate.phenotype.countingphenotypepredicate static method)": [[4, "genophenocorr.analysis.predicate.phenotype.CountingPhenotypePredicate.from_query_curies", false]], "from_term() (genophenocorr.model.phenotype static method)": [[9, "genophenocorr.model.Phenotype.from_term", false]], "functionalannotator (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.FunctionalAnnotator", false]], "gene() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.gene", false]], "gene_id (genophenocorr.model.imprecisesvinfo property)": [[9, "genophenocorr.model.ImpreciseSvInfo.gene_id", false]], "gene_id (genophenocorr.model.transcriptannotation property)": [[9, "genophenocorr.model.TranscriptAnnotation.gene_id", false]], "gene_id (genophenocorr.model.transcriptinfoaware property)": [[9, "genophenocorr.model.TranscriptInfoAware.gene_id", false]], "gene_symbol (genophenocorr.model.imprecisesvinfo property)": [[9, "genophenocorr.model.ImpreciseSvInfo.gene_symbol", false]], "genecoordinateservice (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.GeneCoordinateService", false]], "genophenocorr": [[0, "module-genophenocorr", false]], "genophenocorr.analysis": [[1, "module-genophenocorr.analysis", false]], "genophenocorr.analysis.predicate": [[2, "module-genophenocorr.analysis.predicate", false]], "genophenocorr.analysis.predicate.genotype": [[3, "module-genophenocorr.analysis.predicate.genotype", false]], "genophenocorr.analysis.predicate.phenotype": [[4, "module-genophenocorr.analysis.predicate.phenotype", false]], "genophenocorr.constants": [[5, "module-genophenocorr.constants", false]], "genophenocorr.data": [[7, "module-genophenocorr.data", false]], "genophenocorr.io": [[8, "module-genophenocorr.io", false]], "genophenocorr.model": [[9, "module-genophenocorr.model", false]], "genophenocorr.model.genome": [[10, "module-genophenocorr.model.genome", false]], "genophenocorr.preprocessing": [[11, "module-genophenocorr.preprocessing", false]], "genophenocorr.view": [[12, "module-genophenocorr.view", false]], "genophenocorrjsondecoder (class in genophenocorr.io)": [[8, "genophenocorr.io.GenophenocorrJSONDecoder", false]], "genophenocorrjsonencoder (class in genophenocorr.io)": [[8, "genophenocorr.io.GenophenocorrJSONEncoder", false]], "genotype (class in genophenocorr.model)": [[9, "genophenocorr.model.Genotype", false]], "genotype_for_sample() (genophenocorr.model.genotyped method)": [[9, "genophenocorr.model.Genotyped.genotype_for_sample", false]], "genotypebooleanpredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.GenotypeBooleanPredicate", false]], "genotyped (class in genophenocorr.model)": [[9, "genophenocorr.model.Genotyped", false]], "genotypephenotypeanalysisresult (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult", false]], "genotypepolypredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.GenotypePolyPredicate", false]], "genotypes (class in genophenocorr.model)": [[9, "genophenocorr.model.Genotypes", false]], "genotypes (genophenocorr.model.genotyped property)": [[9, "genophenocorr.model.Genotyped.genotypes", false]], "genotypes (genophenocorr.model.variant property)": [[9, "genophenocorr.model.Variant.genotypes", false]], "genotypes_have_same_hpo_proportions() (genophenocorr.analysis.heuristicmtcfilter static method)": [[1, "genophenocorr.analysis.HeuristicMtcFilter.genotypes_have_same_hpo_proportions", false]], "get_annotations() (genophenocorr.preprocessing.proteinannotationcache method)": [[11, "genophenocorr.preprocessing.ProteinAnnotationCache.get_annotations", false]], "get_annotations() (genophenocorr.preprocessing.variantannotationcache method)": [[11, "genophenocorr.preprocessing.VariantAnnotationCache.get_annotations", false]], "get_categories() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.get_categories", false]], "get_categorizations() (genophenocorr.analysis.predicate.genotypebooleanpredicate method)": [[2, "genophenocorr.analysis.predicate.GenotypeBooleanPredicate.get_categorizations", false]], "get_categorizations() (genophenocorr.analysis.predicate.groupingpredicate method)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate.get_categorizations", false]], "get_categorizations() (genophenocorr.analysis.predicate.phenotype.countingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.CountingPhenotypePredicate.get_categorizations", false]], "get_categorizations() (genophenocorr.analysis.predicate.phenotype.diseasepresencepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate.get_categorizations", false]], "get_categorizations() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.get_categorizations", false]], "get_categorizations() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.get_categorizations", false]], "get_cds_regions() (genophenocorr.model.transcriptcoordinates method)": [[9, "genophenocorr.model.TranscriptCoordinates.get_cds_regions", false]], "get_coding_base_count() (genophenocorr.model.transcriptcoordinates method)": [[9, "genophenocorr.model.TranscriptCoordinates.get_coding_base_count", false]], "get_codon_count() (genophenocorr.model.transcriptcoordinates method)": [[9, "genophenocorr.model.TranscriptCoordinates.get_codon_count", false]], "get_excluded_count() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.get_excluded_count", false]], "get_features_variant_overlaps() (genophenocorr.model.proteinmetadata method)": [[9, "genophenocorr.model.ProteinMetadata.get_features_variant_overlaps", false]], "get_five_prime_utrs() (genophenocorr.model.transcriptcoordinates method)": [[9, "genophenocorr.model.TranscriptCoordinates.get_five_prime_utrs", false]], "get_maximum_group_observed_hpo_frequency() (genophenocorr.analysis.heuristicmtcfilter static method)": [[1, "genophenocorr.analysis.HeuristicMtcFilter.get_maximum_group_observed_HPO_frequency", false]], "get_number_of_observed_hpo_observations() (genophenocorr.analysis.heuristicmtcfilter static method)": [[1, "genophenocorr.analysis.HeuristicMtcFilter.get_number_of_observed_hpo_observations", false]], "get_patient_ids() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.get_patient_ids", false]], "get_question() (genophenocorr.analysis.predicate.genotype.allelecounter method)": [[3, "genophenocorr.analysis.predicate.genotype.AlleleCounter.get_question", false]], "get_question() (genophenocorr.analysis.predicate.genotype.variantpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.phenotype.countingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.CountingPhenotypePredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.phenotype.diseasepresencepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.get_question", false]], "get_question() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.get_question", false]], "get_response() (genophenocorr.preprocessing.vvmulticoordinateservice method)": [[11, "genophenocorr.preprocessing.VVMultiCoordinateService.get_response", false]], "get_three_prime_utrs() (genophenocorr.model.transcriptcoordinates method)": [[9, "genophenocorr.model.TranscriptCoordinates.get_three_prime_utrs", false]], "get_toy_cohort() (in module genophenocorr.data)": [[7, "genophenocorr.data.get_toy_cohort", false]], "get_variant_by_key() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.get_variant_by_key", false]], "get_variant_description() (genophenocorr.view.cohortviewable static method)": [[12, "genophenocorr.view.CohortViewable.get_variant_description", false]], "grouping_predicate() (in module genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.grouping_predicate", false]], "groupingpredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate", false]], "has_errors() (genophenocorr.preprocessing.notepadtree method)": [[11, "genophenocorr.preprocessing.NotepadTree.has_errors", false]], "has_errors_or_warnings() (genophenocorr.preprocessing.notepadtree method)": [[11, "genophenocorr.preprocessing.NotepadTree.has_errors_or_warnings", false]], "has_sv_info() (genophenocorr.model.variantinfo method)": [[9, "genophenocorr.model.VariantInfo.has_sv_info", false]], "has_variant_coordinates() (genophenocorr.model.variantinfo method)": [[9, "genophenocorr.model.VariantInfo.has_variant_coordinates", false]], "has_warnings() (genophenocorr.preprocessing.notepadtree method)": [[11, "genophenocorr.preprocessing.NotepadTree.has_warnings", false]], "hemizygous (genophenocorr.model.genotype attribute)": [[9, "genophenocorr.model.Genotype.HEMIZYGOUS", false]], "heterozygous (genophenocorr.model.genotype attribute)": [[9, "genophenocorr.model.Genotype.HETEROZYGOUS", false]], "heuristic_sampler (genophenocorr.analysis.mtc_strategy attribute)": [[1, "genophenocorr.analysis.MTC_Strategy.HEURISTIC_SAMPLER", false]], "heuristic_strategy() (genophenocorr.analysis.cohortanalysisconfiguration method)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.heuristic_strategy", false]], "heuristicmtcfilter (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.HeuristicMtcFilter", false]], "hgvs_cdna (genophenocorr.model.transcriptannotation property)": [[9, "genophenocorr.model.TranscriptAnnotation.hgvs_cdna", false]], "hgvsp (genophenocorr.model.transcriptannotation property)": [[9, "genophenocorr.model.TranscriptAnnotation.hgvsp", false]], "homozygous_alternate (genophenocorr.model.genotype attribute)": [[9, "genophenocorr.model.Genotype.HOMOZYGOUS_ALTERNATE", false]], "homozygous_reference (genophenocorr.model.genotype attribute)": [[9, "genophenocorr.model.Genotype.HOMOZYGOUS_REFERENCE", false]], "hpomtcreport (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.HpoMtcReport", false]], "identifier (genophenocorr.model.disease property)": [[9, "genophenocorr.model.Disease.identifier", false]], "identifier (genophenocorr.model.phenotype property)": [[9, "genophenocorr.model.Phenotype.identifier", false]], "identifier (genophenocorr.model.transcriptcoordinates property)": [[9, "genophenocorr.model.TranscriptCoordinates.identifier", false]], "imprecisesvfunctionalannotator (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.ImpreciseSvFunctionalAnnotator", false]], "imprecisesvinfo (class in genophenocorr.model)": [[9, "genophenocorr.model.ImpreciseSvInfo", false]], "include_sv (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.include_sv", false]], "incomplete_terminal_codon_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.INCOMPLETE_TERMINAL_CODON_VARIANT", false]], "info (genophenocorr.model.proteinfeature property)": [[9, "genophenocorr.model.ProteinFeature.info", false]], "inframe_deletion (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.INFRAME_DELETION", false]], "inframe_insertion (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.INFRAME_INSERTION", false]], "ins (genophenocorr.model.variantclass attribute)": [[9, "genophenocorr.model.VariantClass.INS", false]], "intergenic_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.INTERGENIC_VARIANT", false]], "intron_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.INTRON_VARIANT", false]], "inv (genophenocorr.model.variantclass attribute)": [[9, "genophenocorr.model.VariantClass.INV", false]], "is_coding() (genophenocorr.model.transcriptcoordinates method)": [[9, "genophenocorr.model.TranscriptCoordinates.is_coding", false]], "is_large_imprecise_sv() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.is_large_imprecise_sv", false]], "is_observed (genophenocorr.model.disease attribute)": [[9, "genophenocorr.model.Disease.is_observed", false]], "is_observed (genophenocorr.model.phenotype attribute)": [[9, "genophenocorr.model.Phenotype.is_observed", false]], "is_observed (genophenocorr.model.phenotype property)": [[9, "id5", false]], "is_preferred (genophenocorr.model.transcriptannotation property)": [[9, "genophenocorr.model.TranscriptAnnotation.is_preferred", false]], "is_preferred (genophenocorr.model.transcriptcoordinates property)": [[9, "genophenocorr.model.TranscriptCoordinates.is_preferred", false]], "is_present (genophenocorr.model.disease property)": [[9, "genophenocorr.model.Disease.is_present", false]], "is_present (genophenocorr.model.phenotype property)": [[9, "genophenocorr.model.Phenotype.is_present", false]], "is_structural() (genophenocorr.model.variantcoordinates method)": [[9, "genophenocorr.model.VariantCoordinates.is_structural", false]], "is_structural() (genophenocorr.model.variantinfo method)": [[9, "genophenocorr.model.VariantInfo.is_structural", false]], "is_structural_deletion() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.is_structural_deletion", false]], "is_structural_variant() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.is_structural_variant", false]], "issues (genophenocorr.preprocessing.notepad property)": [[11, "genophenocorr.preprocessing.Notepad.issues", false]], "iterate_nodes() (genophenocorr.preprocessing.notepadtree method)": [[11, "genophenocorr.preprocessing.NotepadTree.iterate_nodes", false]], "label (genophenocorr.model.proteinmetadata attribute)": [[9, "genophenocorr.model.ProteinMetadata.label", false]], "label (genophenocorr.model.proteinmetadata property)": [[9, "id20", false]], "label (genophenocorr.model.samplelabels property)": [[9, "genophenocorr.model.SampleLabels.label", false]], "label (genophenocorr.preprocessing.notepad property)": [[11, "genophenocorr.preprocessing.Notepad.label", false]], "label_summary() (genophenocorr.model.samplelabels method)": [[9, "genophenocorr.model.SampleLabels.label_summary", false]], "labels (genophenocorr.model.patient attribute)": [[9, "genophenocorr.model.Patient.labels", false]], "labels (genophenocorr.model.patient property)": [[9, "id0", false]], "length (genophenocorr.model.proteinmetadata attribute)": [[9, "genophenocorr.model.ProteinMetadata.length", false]], "level (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.Level", false]], "level (genophenocorr.preprocessing.datasanityissue property)": [[11, "genophenocorr.preprocessing.DataSanityIssue.level", false]], "level (genophenocorr.preprocessing.notepadtree property)": [[11, "genophenocorr.preprocessing.NotepadTree.level", false]], "list_all_diseases() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.list_all_diseases", false]], "list_all_proteins() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.list_all_proteins", false]], "list_all_variants() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.list_all_variants", false]], "list_present_phenotypes() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.list_present_phenotypes", false]], "load_phenopacket() (in module genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.load_phenopacket", false]], "load_phenopacket_folder() (in module genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.load_phenopacket_folder", false]], "marker_counts (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.marker_counts", false]], "mature_mirna_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.MATURE_MIRNA_VARIANT", false]], "message (genophenocorr.preprocessing.datasanityissue property)": [[11, "genophenocorr.preprocessing.DataSanityIssue.message", false]], "meta_label (genophenocorr.model.samplelabels property)": [[9, "genophenocorr.model.SampleLabels.meta_label", false]], "min_perc_patients_w_hpo (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.min_perc_patients_w_hpo", false]], "missense_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.MISSENSE_VARIANT", false]], "missing_implies_excluded (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.missing_implies_excluded", false]], "mnv (genophenocorr.model.variantclass attribute)": [[9, "genophenocorr.model.VariantClass.MNV", false]], "module": [[0, "module-genophenocorr", false], [1, "module-genophenocorr.analysis", false], [2, "module-genophenocorr.analysis.predicate", false], [3, "module-genophenocorr.analysis.predicate.genotype", false], [4, "module-genophenocorr.analysis.predicate.phenotype", false], [5, "module-genophenocorr.constants", false], [7, "module-genophenocorr.data", false], [8, "module-genophenocorr.io", false], [9, "module-genophenocorr.model", false], [10, "module-genophenocorr.model.genome", false], [11, "module-genophenocorr.preprocessing", false], [12, "module-genophenocorr.view", false]], "motif (genophenocorr.model.featuretype attribute)": [[9, "genophenocorr.model.FeatureType.MOTIF", false], [9, "id29", false]], "motifs() (genophenocorr.model.proteinmetadata method)": [[9, "genophenocorr.model.ProteinMetadata.motifs", false], [9, "id25", false]], "mtc_alpha (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.mtc_alpha", false]], "mtc_filter_report (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.mtc_filter_report", false]], "mtc_method (genophenocorr.analysis.hpomtcreport property)": [[1, "genophenocorr.analysis.HpoMtcReport.mtc_method", false]], "mtc_strategy (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.MTC_Strategy", false]], "mtc_strategy (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.mtc_strategy", false]], "n_categorizations() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.n_categorizations", false]], "n_terms_tested (genophenocorr.analysis.hpomtcreport property)": [[1, "genophenocorr.analysis.HpoMtcReport.n_terms_tested", false]], "n_usable (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.n_usable", false]], "name (genophenocorr.analysis.predicate.patientcategory property)": [[2, "genophenocorr.analysis.predicate.PatientCategory.name", false]], "name (genophenocorr.model.disease attribute)": [[9, "genophenocorr.model.Disease.name", false]], "name (genophenocorr.model.disease property)": [[9, "id6", false]], "name (genophenocorr.model.featureinfo attribute)": [[9, "genophenocorr.model.FeatureInfo.name", false]], "name (genophenocorr.model.featureinfo property)": [[9, "id26", false]], "name (genophenocorr.model.phenotype attribute)": [[9, "genophenocorr.model.Phenotype.name", false]], "name (genophenocorr.model.phenotype property)": [[9, "id4", false]], "nmd_transcript_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.NMD_TRANSCRIPT_VARIANT", false]], "no (genophenocorr.analysis.predicate.genotypebooleanpredicate attribute)": [[2, "genophenocorr.analysis.predicate.GenotypeBooleanPredicate.NO", false]], "no (genophenocorr.analysis.predicate.patientcategories attribute)": [[2, "genophenocorr.analysis.predicate.PatientCategories.NO", false]], "no_call (genophenocorr.model.genotype attribute)": [[9, "genophenocorr.model.Genotype.NO_CALL", false]], "non_coding_transcript_exon_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.NON_CODING_TRANSCRIPT_EXON_VARIANT", false]], "non_coding_transcript_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.NON_CODING_TRANSCRIPT_VARIANT", false]], "noncoding_effects (genophenocorr.preprocessing.vepfunctionalannotator attribute)": [[11, "genophenocorr.preprocessing.VepFunctionalAnnotator.NONCODING_EFFECTS", false]], "notepad (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.Notepad", false]], "notepadtree (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.NotepadTree", false]], "object_hook() (genophenocorr.io.genophenocorrjsondecoder static method)": [[8, "genophenocorr.io.GenophenocorrJSONDecoder.object_hook", false]], "observed (genophenocorr.model.phenotype property)": [[9, "genophenocorr.model.Phenotype.observed", false]], "one_genotype_has_zero_hpo_observations() (genophenocorr.analysis.heuristicmtcfilter static method)": [[1, "genophenocorr.analysis.HeuristicMtcFilter.one_genotype_has_zero_hpo_observations", false]], "overlapping_exons (genophenocorr.model.transcriptannotation property)": [[9, "genophenocorr.model.TranscriptAnnotation.overlapping_exons", false]], "parse_multiple() (genophenocorr.preprocessing.vvmulticoordinateservice method)": [[11, "genophenocorr.preprocessing.VVMultiCoordinateService.parse_multiple", false]], "parse_response() (genophenocorr.preprocessing.vvmulticoordinateservice method)": [[11, "genophenocorr.preprocessing.VVMultiCoordinateService.parse_response", false]], "parse_uniprot_json() (genophenocorr.preprocessing.uniprotproteinmetadataservice static method)": [[11, "genophenocorr.preprocessing.UniprotProteinMetadataService.parse_uniprot_json", false]], "patient (class in genophenocorr.model)": [[9, "genophenocorr.model.Patient", false]], "patient_id (genophenocorr.model.patient property)": [[9, "genophenocorr.model.Patient.patient_id", false]], "patientcategories (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.PatientCategories", false]], "patientcategory (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.PatientCategory", false]], "patientcreator (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.PatientCreator", false]], "phenopacketpatientcreator (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.PhenopacketPatientCreator", false]], "phenopacketvariantcoordinatefinder (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder", false]], "phenotype (class in genophenocorr.model)": [[9, "genophenocorr.model.Phenotype", false]], "phenotype (genophenocorr.analysis.predicate.phenotype.phenotypecategorization property)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypeCategorization.phenotype", false]], "phenotype_categories (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.phenotype_categories", false]], "phenotypecategorization (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypeCategorization", false]], "phenotypecreator (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.PhenotypeCreator", false]], "phenotypepolypredicate (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypePolyPredicate", false]], "phenotypes (genophenocorr.analysis.predicate.phenotype.countingphenotypepredicate property)": [[4, "genophenocorr.analysis.predicate.phenotype.CountingPhenotypePredicate.phenotypes", false]], "phenotypes (genophenocorr.analysis.predicate.phenotype.diseasepresencepredicate property)": [[4, "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate.phenotypes", false]], "phenotypes (genophenocorr.analysis.predicate.phenotype.phenotypepolypredicate property)": [[4, "genophenocorr.analysis.predicate.phenotype.PhenotypePolyPredicate.phenotypes", false]], "phenotypes (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate property)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.phenotypes", false]], "phenotypes (genophenocorr.model.patient attribute)": [[9, "genophenocorr.model.Patient.phenotypes", false]], "phenotypes (genophenocorr.model.patient property)": [[9, "id1", false]], "polypredicate (class in genophenocorr.analysis.predicate)": [[2, "genophenocorr.analysis.predicate.PolyPredicate", false]], "prepare_notepad() (genophenocorr.preprocessing.auditor static method)": [[11, "genophenocorr.preprocessing.Auditor.prepare_notepad", false]], "present_diseases() (genophenocorr.model.patient method)": [[9, "genophenocorr.model.Patient.present_diseases", false]], "present_phenotypes() (genophenocorr.model.patient method)": [[9, "genophenocorr.model.Patient.present_phenotypes", false]], "process() (genophenocorr.preprocessing.auditor method)": [[11, "genophenocorr.preprocessing.Auditor.process", false]], "process() (genophenocorr.preprocessing.cohortcreator method)": [[11, "genophenocorr.preprocessing.CohortCreator.process", false]], "process() (genophenocorr.preprocessing.phenopacketpatientcreator method)": [[11, "genophenocorr.preprocessing.PhenopacketPatientCreator.process", false]], "process() (genophenocorr.preprocessing.phenotypecreator method)": [[11, "genophenocorr.preprocessing.PhenotypeCreator.process", false]], "process() (genophenocorr.view.cohortviewable method)": [[12, "genophenocorr.view.CohortViewable.process", false]], "process() (genophenocorr.view.diseaseviewable method)": [[12, "genophenocorr.view.DiseaseViewable.process", false]], "process() (genophenocorr.view.proteinviewable method)": [[12, "genophenocorr.view.ProteinViewable.process", false]], "process() (genophenocorr.view.statsviewer method)": [[12, "genophenocorr.view.StatsViewer.process", false]], "process_response() (genophenocorr.preprocessing.vepfunctionalannotator method)": [[11, "genophenocorr.preprocessing.VepFunctionalAnnotator.process_response", false]], "propagatingphenotypepredicate (class in genophenocorr.analysis.predicate.phenotype)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate", false]], "protcachingmetadataservice (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.ProtCachingMetadataService", false]], "protein_altering_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.PROTEIN_ALTERING_VARIANT", false]], "protein_effect_location (genophenocorr.model.transcriptannotation property)": [[9, "genophenocorr.model.TranscriptAnnotation.protein_effect_location", false]], "protein_feature() (genophenocorr.analysis.predicate.genotype.proteinpredicates method)": [[3, "genophenocorr.analysis.predicate.genotype.ProteinPredicates.protein_feature", false]], "protein_feature_ends (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.protein_feature_ends", false]], "protein_feature_names (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.protein_feature_names", false]], "protein_feature_starts (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.protein_feature_starts", false]], "protein_feature_type() (genophenocorr.analysis.predicate.genotype.proteinpredicates method)": [[3, "genophenocorr.analysis.predicate.genotype.ProteinPredicates.protein_feature_type", false]], "protein_feature_types (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.protein_feature_types", false]], "protein_features (genophenocorr.model.proteinmetadata attribute)": [[9, "genophenocorr.model.ProteinMetadata.protein_features", false]], "protein_features (genophenocorr.model.proteinmetadata property)": [[9, "id21", false]], "protein_id (genophenocorr.model.proteinmetadata attribute)": [[9, "genophenocorr.model.ProteinMetadata.protein_id", false]], "protein_id (genophenocorr.model.proteinmetadata property)": [[9, "id19", false]], "protein_id (genophenocorr.model.transcriptannotation property)": [[9, "genophenocorr.model.TranscriptAnnotation.protein_id", false]], "protein_id (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.protein_id", false]], "protein_length (genophenocorr.model.proteinmetadata property)": [[9, "genophenocorr.model.ProteinMetadata.protein_length", false]], "protein_length (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.protein_length", false]], "protein_metadata (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.protein_metadata", false]], "proteinannotationcache (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.ProteinAnnotationCache", false]], "proteinfeature (class in genophenocorr.model)": [[9, "genophenocorr.model.ProteinFeature", false]], "proteinmetadata (class in genophenocorr.model)": [[9, "genophenocorr.model.ProteinMetadata", false]], "proteinmetadataservice (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.ProteinMetadataService", false]], "proteinpredicates (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.ProteinPredicates", false]], "proteinviewable (class in genophenocorr.view)": [[12, "genophenocorr.view.ProteinViewable", false]], "proteinvisualizable (class in genophenocorr.view)": [[12, "genophenocorr.view.ProteinVisualizable", false]], "proteinvisualizer (class in genophenocorr.view)": [[12, "genophenocorr.view.ProteinVisualizer", false]], "pval_correction (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.pval_correction", false]], "pvals (genophenocorr.analysis.genotypephenotypeanalysisresult property)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.pvals", false]], "recessive_predicate() (in module genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.recessive_predicate", false]], "ref (genophenocorr.model.variantcoordinates attribute)": [[9, "genophenocorr.model.VariantCoordinates.ref", false]], "ref (genophenocorr.model.variantcoordinates property)": [[9, "id8", false]], "region (genophenocorr.model.featureinfo attribute)": [[9, "genophenocorr.model.FeatureInfo.region", false]], "region (genophenocorr.model.featureinfo property)": [[9, "id27", false]], "region (genophenocorr.model.featuretype attribute)": [[9, "genophenocorr.model.FeatureType.REGION", false], [9, "id31", false]], "region (genophenocorr.model.transcriptcoordinates property)": [[9, "genophenocorr.model.TranscriptCoordinates.region", false]], "region (genophenocorr.model.variantcoordinates attribute)": [[9, "genophenocorr.model.VariantCoordinates.region", false]], "region (genophenocorr.model.variantcoordinates property)": [[9, "id7", false]], "region() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.region", false]], "regions() (genophenocorr.model.proteinmetadata method)": [[9, "genophenocorr.model.ProteinMetadata.regions", false], [9, "id24", false]], "regulatory_region_ablation (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.REGULATORY_REGION_ABLATION", false]], "regulatory_region_amplification (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.REGULATORY_REGION_AMPLIFICATION", false]], "regulatory_region_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.REGULATORY_REGION_VARIANT", false]], "repeat (genophenocorr.model.featuretype attribute)": [[9, "genophenocorr.model.FeatureType.REPEAT", false], [9, "id28", false]], "repeats() (genophenocorr.model.proteinmetadata method)": [[9, "genophenocorr.model.ProteinMetadata.repeats", false], [9, "id23", false]], "samplelabels (class in genophenocorr.model)": [[9, "genophenocorr.model.SampleLabels", false]], "second (genophenocorr.analysis.predicate.groupingpredicate attribute)": [[2, "genophenocorr.analysis.predicate.GroupingPredicate.SECOND", false]], "sequence_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.SEQUENCE_VARIANT", false]], "single() (genophenocorr.model.genotypes static method)": [[9, "genophenocorr.model.Genotypes.single", false]], "skipped_terms_dict (genophenocorr.analysis.hpomtcreport property)": [[1, "genophenocorr.analysis.HpoMtcReport.skipped_terms_dict", false]], "snv (genophenocorr.model.variantclass attribute)": [[9, "genophenocorr.model.VariantClass.SNV", false]], "solution (genophenocorr.preprocessing.datasanityissue property)": [[11, "genophenocorr.preprocessing.DataSanityIssue.solution", false]], "some_cell_has_greater_than_one_count() (genophenocorr.analysis.heuristicmtcfilter static method)": [[1, "genophenocorr.analysis.HeuristicMtcFilter.some_cell_has_greater_than_one_count", false]], "specifiedtermsmtcfilter (class in genophenocorr.analysis)": [[1, "genophenocorr.analysis.SpecifiedTermsMtcFilter", false]], "specify_terms (genophenocorr.analysis.mtc_strategy attribute)": [[1, "genophenocorr.analysis.MTC_Strategy.SPECIFY_TERMS", false]], "specify_terms_strategy() (genophenocorr.analysis.cohortanalysisconfiguration method)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.specify_terms_strategy", false]], "splice_acceptor_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.SPLICE_ACCEPTOR_VARIANT", false]], "splice_donor_5th_base_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.SPLICE_DONOR_5TH_BASE_VARIANT", false]], "splice_donor_region_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.SPLICE_DONOR_REGION_VARIANT", false]], "splice_donor_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.SPLICE_DONOR_VARIANT", false]], "splice_polypyrimidine_tract_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.SPLICE_POLYPYRIMIDINE_TRACT_VARIANT", false]], "splice_region_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.SPLICE_REGION_VARIANT", false]], "start (genophenocorr.model.featureinfo property)": [[9, "genophenocorr.model.FeatureInfo.start", false]], "start (genophenocorr.model.variantcoordinates property)": [[9, "genophenocorr.model.VariantCoordinates.start", false]], "start_lost (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.START_LOST", false]], "start_retained_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.START_RETAINED_VARIANT", false]], "statsviewer (class in genophenocorr.view)": [[12, "genophenocorr.view.StatsViewer", false]], "stop_gained (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.STOP_GAINED", false]], "stop_lost (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.STOP_LOST", false]], "stop_retained_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.STOP_RETAINED_VARIANT", false]], "store_annotations() (genophenocorr.preprocessing.proteinannotationcache method)": [[11, "genophenocorr.preprocessing.ProteinAnnotationCache.store_annotations", false]], "store_annotations() (genophenocorr.preprocessing.variantannotationcache method)": [[11, "genophenocorr.preprocessing.VariantAnnotationCache.store_annotations", false]], "structural_type (genophenocorr.model.imprecisesvinfo property)": [[9, "genophenocorr.model.ImpreciseSvInfo.structural_type", false]], "structural_type() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.structural_type", false]], "summarize() (genophenocorr.analysis.genotypephenotypeanalysisresult method)": [[1, "genophenocorr.analysis.GenotypePhenotypeAnalysisResult.summarize", false]], "sv_info (genophenocorr.model.variantinfo property)": [[9, "genophenocorr.model.VariantInfo.sv_info", false]], "synonymous_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.SYNONYMOUS_VARIANT", false]], "term_id (genophenocorr.model.disease attribute)": [[9, "genophenocorr.model.Disease.term_id", false]], "term_id (genophenocorr.model.phenotype attribute)": [[9, "genophenocorr.model.Phenotype.term_id", false]], "terms_to_test (genophenocorr.analysis.cohortanalysisconfiguration property)": [[1, "genophenocorr.analysis.CohortAnalysisConfiguration.terms_to_test", false]], "test() (genophenocorr.analysis.predicate.genotype.variantpredicate method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicate.test", false]], "test() (genophenocorr.analysis.predicate.phenotype.countingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.CountingPhenotypePredicate.test", false]], "test() (genophenocorr.analysis.predicate.phenotype.diseasepresencepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate.test", false]], "test() (genophenocorr.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[4, "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate.test", false]], "test() (genophenocorr.analysis.predicate.polypredicate method)": [[2, "genophenocorr.analysis.predicate.PolyPredicate.test", false]], "tf_binding_site_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.TF_BINDING_SITE_VARIANT", false]], "tfbs_ablation (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.TFBS_ABLATION", false]], "tfbs_amplification (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.TFBS_AMPLIFICATION", false]], "three_prime_utr_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.THREE_PRIME_UTR_VARIANT", false]], "to_string() (genophenocorr.model.proteinfeature method)": [[9, "genophenocorr.model.ProteinFeature.to_string", false]], "total_patient_count (genophenocorr.model.cohort property)": [[9, "genophenocorr.model.Cohort.total_patient_count", false]], "transcript() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.transcript", false]], "transcript_ablation (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.TRANSCRIPT_ABLATION", false]], "transcript_amplification (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.TRANSCRIPT_AMPLIFICATION", false]], "transcript_coordinates (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.transcript_coordinates", false]], "transcript_id (genophenocorr.model.transcriptannotation property)": [[9, "genophenocorr.model.TranscriptAnnotation.transcript_id", false]], "transcript_id (genophenocorr.model.transcriptinfoaware property)": [[9, "genophenocorr.model.TranscriptInfoAware.transcript_id", false]], "transcript_id (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.transcript_id", false]], "transcriptannotation (class in genophenocorr.model)": [[9, "genophenocorr.model.TranscriptAnnotation", false]], "transcriptcoordinates (class in genophenocorr.model)": [[9, "genophenocorr.model.TranscriptCoordinates", false]], "transcriptcoordinateservice (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.TranscriptCoordinateService", false]], "transcriptinfoaware (class in genophenocorr.model)": [[9, "genophenocorr.model.TranscriptInfoAware", false]], "tx_annotations (genophenocorr.model.variant property)": [[9, "genophenocorr.model.Variant.tx_annotations", false]], "uniprotproteinmetadataservice (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.UniprotProteinMetadataService", false]], "upstream_gene_variant (genophenocorr.model.varianteffect attribute)": [[9, "genophenocorr.model.VariantEffect.UPSTREAM_GENE_VARIANT", false]], "varcachingfunctionalannotator (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.VarCachingFunctionalAnnotator", false]], "variant (class in genophenocorr.model)": [[9, "genophenocorr.model.Variant", false]], "variant_class (genophenocorr.model.imprecisesvinfo property)": [[9, "genophenocorr.model.ImpreciseSvInfo.variant_class", false]], "variant_class (genophenocorr.model.variantcoordinates property)": [[9, "genophenocorr.model.VariantCoordinates.variant_class", false]], "variant_class (genophenocorr.model.variantinfo property)": [[9, "genophenocorr.model.VariantInfo.variant_class", false]], "variant_class() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.variant_class", false]], "variant_coordinates (genophenocorr.model.variantinfo property)": [[9, "genophenocorr.model.VariantInfo.variant_coordinates", false]], "variant_effect() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.variant_effect", false]], "variant_effect_count_by_tx() (genophenocorr.model.cohort method)": [[9, "genophenocorr.model.Cohort.variant_effect_count_by_tx", false]], "variant_effects (genophenocorr.model.transcriptannotation property)": [[9, "genophenocorr.model.TranscriptAnnotation.variant_effects", false]], "variant_effects (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.variant_effects", false]], "variant_info (genophenocorr.model.variant property)": [[9, "genophenocorr.model.Variant.variant_info", false]], "variant_info (genophenocorr.model.variantinfoaware property)": [[9, "genophenocorr.model.VariantInfoAware.variant_info", false]], "variant_key (genophenocorr.model.imprecisesvinfo property)": [[9, "genophenocorr.model.ImpreciseSvInfo.variant_key", false]], "variant_key (genophenocorr.model.variantcoordinates property)": [[9, "genophenocorr.model.VariantCoordinates.variant_key", false]], "variant_key (genophenocorr.model.variantinfo property)": [[9, "genophenocorr.model.VariantInfo.variant_key", false]], "variant_key() (genophenocorr.analysis.predicate.genotype.variantpredicates static method)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates.variant_key", false]], "variant_locations (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.variant_locations", false]], "variant_locations_counted_absolute (genophenocorr.view.proteinvisualizable property)": [[12, "genophenocorr.view.ProteinVisualizable.variant_locations_counted_absolute", false]], "variantannotationcache (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.VariantAnnotationCache", false]], "variantclass (class in genophenocorr.model)": [[9, "genophenocorr.model.VariantClass", false]], "variantcoordinatefinder (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.VariantCoordinateFinder", false]], "variantcoordinates (class in genophenocorr.model)": [[9, "genophenocorr.model.VariantCoordinates", false]], "varianteffect (class in genophenocorr.model)": [[9, "genophenocorr.model.VariantEffect", false]], "variantformatter (class in genophenocorr.view)": [[12, "genophenocorr.view.VariantFormatter", false]], "variantinfo (class in genophenocorr.model)": [[9, "genophenocorr.model.VariantInfo", false]], "variantinfoaware (class in genophenocorr.model)": [[9, "genophenocorr.model.VariantInfoAware", false]], "variantpredicate (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicate", false]], "variantpredicates (class in genophenocorr.analysis.predicate.genotype)": [[3, "genophenocorr.analysis.predicate.genotype.VariantPredicates", false]], "variants (genophenocorr.model.patient attribute)": [[9, "genophenocorr.model.Patient.variants", false]], "variants (genophenocorr.model.patient property)": [[9, "id3", false]], "varianttranscriptvisualizer (class in genophenocorr.view)": [[12, "genophenocorr.view.VariantTranscriptVisualizer", false]], "vepfunctionalannotator (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.VepFunctionalAnnotator", false]], "visit() (genophenocorr.preprocessing.notepadtree method)": [[11, "genophenocorr.preprocessing.NotepadTree.visit", false]], "vvhgvsvariantcoordinatefinder (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.VVHgvsVariantCoordinateFinder", false]], "vvmulticoordinateservice (class in genophenocorr.preprocessing)": [[11, "genophenocorr.preprocessing.VVMultiCoordinateService", false]], "warn (genophenocorr.preprocessing.level attribute)": [[11, "genophenocorr.preprocessing.Level.WARN", false]], "warning_count() (genophenocorr.preprocessing.notepad method)": [[11, "genophenocorr.preprocessing.Notepad.warning_count", false]], "warnings() (genophenocorr.preprocessing.notepad method)": [[11, "genophenocorr.preprocessing.Notepad.warnings", false]], "with_cache_folder() (genophenocorr.preprocessing.varcachingfunctionalannotator static method)": [[11, "genophenocorr.preprocessing.VarCachingFunctionalAnnotator.with_cache_folder", false]], "yes (genophenocorr.analysis.predicate.genotypebooleanpredicate attribute)": [[2, "genophenocorr.analysis.predicate.GenotypeBooleanPredicate.YES", false]], "yes (genophenocorr.analysis.predicate.patientcategories attribute)": [[2, "genophenocorr.analysis.predicate.PatientCategories.YES", false]]}, "objects": {"": [[0, 0, 0, "-", "genophenocorr"]], "genophenocorr": [[1, 0, 0, "-", "analysis"], [5, 0, 0, "-", "constants"], [7, 0, 0, "-", "data"], [8, 0, 0, "-", "io"], [9, 0, 0, "-", "model"], [11, 0, 0, "-", "preprocessing"], [12, 0, 0, "-", "view"]], "genophenocorr.analysis": [[1, 1, 1, "", "CohortAnalysis"], [1, 1, 1, "", "CohortAnalysisConfiguration"], [1, 1, 1, "", "GenotypePhenotypeAnalysisResult"], [1, 1, 1, "", "HeuristicMtcFilter"], [1, 1, 1, "", "HpoMtcReport"], [1, 1, 1, "", "MTC_Strategy"], [1, 1, 1, "", "SpecifiedTermsMtcFilter"], [1, 5, 1, "", "apply_predicates_on_patients"], [1, 5, 1, "", "configure_cohort_analysis"], [1, 5, 1, "", "configure_default_protein_metadata_service"], [2, 0, 0, "-", "predicate"]], "genophenocorr.analysis.CohortAnalysis": [[1, 2, 1, "", "compare_by_exon"], [1, 2, 1, "", "compare_by_exons"], [1, 2, 1, "", "compare_by_protein_feature"], [1, 2, 1, "", "compare_by_protein_feature_type"], [1, 2, 1, "", "compare_by_protein_feature_types"], [1, 2, 1, "", "compare_by_protein_features"], [1, 2, 1, "", "compare_by_protein_region"], [1, 2, 1, "", "compare_by_protein_regions"], [1, 2, 1, "", "compare_by_variant_effect"], [1, 2, 1, "", "compare_by_variant_effects"], [1, 2, 1, "", "compare_by_variant_key"], [1, 2, 1, "", "compare_by_variant_keys"], [1, 2, 1, "", "compare_hpo_vs_genotype"], [1, 2, 1, "", "compare_hpo_vs_genotype_groups"]], "genophenocorr.analysis.CohortAnalysisConfiguration": [[1, 2, 1, "", "all_terms_strategy"], [1, 2, 1, "", "heuristic_strategy"], [1, 3, 1, "", "include_sv"], [1, 3, 1, "", "min_perc_patients_w_hpo"], [1, 3, 1, "", "missing_implies_excluded"], [1, 3, 1, "", "mtc_alpha"], [1, 3, 1, "", "mtc_strategy"], [1, 3, 1, "", "pval_correction"], [1, 2, 1, "", "specify_terms_strategy"], [1, 3, 1, "", "terms_to_test"]], "genophenocorr.analysis.GenotypePhenotypeAnalysisResult": [[1, 3, 1, "", "all_counts"], [1, 3, 1, "", "corrected_pvals"], [1, 3, 1, "", "mtc_filter_report"], [1, 3, 1, "", "n_usable"], [1, 3, 1, "", "phenotype_categories"], [1, 3, 1, "", "pvals"], [1, 2, 1, "", "summarize"]], "genophenocorr.analysis.HeuristicMtcFilter": [[1, 2, 1, "", "filter_method_name"], [1, 2, 1, "", "filter_terms_to_test"], [1, 2, 1, "", "genotypes_have_same_hpo_proportions"], [1, 2, 1, "", "get_maximum_group_observed_HPO_frequency"], [1, 2, 1, "", "get_number_of_observed_hpo_observations"], [1, 2, 1, "", "one_genotype_has_zero_hpo_observations"], [1, 2, 1, "", "some_cell_has_greater_than_one_count"]], "genophenocorr.analysis.HpoMtcReport": [[1, 3, 1, "", "filter_method"], [1, 3, 1, "", "mtc_method"], [1, 3, 1, "", "n_terms_tested"], [1, 3, 1, "", "skipped_terms_dict"]], "genophenocorr.analysis.MTC_Strategy": [[1, 4, 1, "", "ALL_HPO_TERMS"], [1, 4, 1, "", "HEURISTIC_SAMPLER"], [1, 4, 1, "", "SPECIFY_TERMS"]], "genophenocorr.analysis.SpecifiedTermsMtcFilter": [[1, 2, 1, "", "filter_method_name"], [1, 2, 1, "", "filter_terms_to_test"]], "genophenocorr.analysis.predicate": [[2, 1, 1, "", "Categorization"], [2, 1, 1, "", "GenotypeBooleanPredicate"], [2, 1, 1, "", "GenotypePolyPredicate"], [2, 1, 1, "", "GroupingPredicate"], [2, 1, 1, "", "PatientCategories"], [2, 1, 1, "", "PatientCategory"], [2, 1, 1, "", "PolyPredicate"], [3, 0, 0, "-", "genotype"], [4, 0, 0, "-", "phenotype"]], "genophenocorr.analysis.predicate.Categorization": [[2, 3, 1, "", "category"]], "genophenocorr.analysis.predicate.GenotypeBooleanPredicate": [[2, 4, 1, "", "NO"], [2, 4, 1, "", "YES"], [2, 2, 1, "", "get_categorizations"]], "genophenocorr.analysis.predicate.GroupingPredicate": [[2, 4, 1, "", "FIRST"], [2, 4, 1, "", "SECOND"], [2, 2, 1, "", "get_categorizations"]], "genophenocorr.analysis.predicate.PatientCategories": [[2, 4, 1, "", "NO"], [2, 4, 1, "", "YES"]], "genophenocorr.analysis.predicate.PatientCategory": [[2, 3, 1, "", "cat_id"], [2, 3, 1, "", "description"], [2, 3, 1, "", "name"]], "genophenocorr.analysis.predicate.PolyPredicate": [[2, 2, 1, "", "get_categories"], [2, 2, 1, "", "get_categorizations"], [2, 2, 1, "", "get_question"], [2, 2, 1, "", "n_categorizations"], [2, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype": [[3, 1, 1, "", "AlleleCounter"], [3, 1, 1, "", "ProteinPredicates"], [3, 1, 1, "", "VariantPredicate"], [3, 1, 1, "", "VariantPredicates"], [3, 5, 1, "", "boolean_predicate"], [3, 5, 1, "", "grouping_predicate"], [3, 5, 1, "", "recessive_predicate"]], "genophenocorr.analysis.predicate.genotype.AlleleCounter": [[3, 2, 1, "", "count"], [3, 2, 1, "", "get_question"]], "genophenocorr.analysis.predicate.genotype.ProteinPredicates": [[3, 2, 1, "", "protein_feature"], [3, 2, 1, "", "protein_feature_type"]], "genophenocorr.analysis.predicate.genotype.VariantPredicate": [[3, 2, 1, "", "get_question"], [3, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.genotype.VariantPredicates": [[3, 2, 1, "", "change_length"], [3, 2, 1, "", "exon"], [3, 2, 1, "", "gene"], [3, 2, 1, "", "is_large_imprecise_sv"], [3, 2, 1, "", "is_structural_deletion"], [3, 2, 1, "", "is_structural_variant"], [3, 2, 1, "", "region"], [3, 2, 1, "", "structural_type"], [3, 2, 1, "", "transcript"], [3, 2, 1, "", "variant_class"], [3, 2, 1, "", "variant_effect"], [3, 2, 1, "", "variant_key"]], "genophenocorr.analysis.predicate.phenotype": [[4, 1, 1, "", "CountingPhenotypePredicate"], [4, 1, 1, "", "DiseasePresencePredicate"], [4, 1, 1, "", "PhenotypeCategorization"], [4, 1, 1, "", "PhenotypePolyPredicate"], [4, 1, 1, "", "PropagatingPhenotypePredicate"]], "genophenocorr.analysis.predicate.phenotype.CountingPhenotypePredicate": [[4, 2, 1, "", "from_query_curies"], [4, 2, 1, "", "get_categorizations"], [4, 2, 1, "", "get_question"], [4, 3, 1, "", "phenotypes"], [4, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.phenotype.DiseasePresencePredicate": [[4, 2, 1, "", "get_categorizations"], [4, 2, 1, "", "get_question"], [4, 3, 1, "", "phenotypes"], [4, 2, 1, "", "test"]], "genophenocorr.analysis.predicate.phenotype.PhenotypeCategorization": [[4, 3, 1, "", "phenotype"]], "genophenocorr.analysis.predicate.phenotype.PhenotypePolyPredicate": [[4, 3, 1, "", "phenotypes"]], "genophenocorr.analysis.predicate.phenotype.PropagatingPhenotypePredicate": [[4, 2, 1, "", "get_categorizations"], [4, 2, 1, "", "get_question"], [4, 3, 1, "", "phenotypes"], [4, 2, 1, "", "test"]], "genophenocorr.data": [[7, 5, 1, "", "get_toy_cohort"]], "genophenocorr.io": [[8, 1, 1, "", "GenophenocorrJSONDecoder"], [8, 1, 1, "", "GenophenocorrJSONEncoder"]], "genophenocorr.io.GenophenocorrJSONDecoder": [[8, 2, 1, "", "object_hook"]], "genophenocorr.io.GenophenocorrJSONEncoder": [[8, 2, 1, "", "default"]], "genophenocorr.model": [[9, 1, 1, "", "Cohort"], [9, 1, 1, "", "Disease"], [9, 1, 1, "", "FeatureInfo"], [9, 1, 1, "", "FeatureType"], [9, 1, 1, "", "Genotype"], [9, 1, 1, "", "Genotyped"], [9, 1, 1, "", "Genotypes"], [9, 1, 1, "", "ImpreciseSvInfo"], [9, 1, 1, "", "Patient"], [9, 1, 1, "", "Phenotype"], [9, 1, 1, "", "ProteinFeature"], [9, 1, 1, "", "ProteinMetadata"], [9, 1, 1, "", "SampleLabels"], [9, 1, 1, "", "TranscriptAnnotation"], [9, 1, 1, "", "TranscriptCoordinates"], [9, 1, 1, "", "TranscriptInfoAware"], [9, 1, 1, "", "Variant"], [9, 1, 1, "", "VariantClass"], [9, 1, 1, "", "VariantCoordinates"], [9, 1, 1, "", "VariantEffect"], [9, 1, 1, "", "VariantInfo"], [9, 1, 1, "", "VariantInfoAware"], [10, 0, 0, "-", "genome"]], "genophenocorr.model.Cohort": [[9, 2, 1, "", "all_diseases"], [9, 3, 1, "", "all_patients"], [9, 2, 1, "", "all_phenotypes"], [9, 3, 1, "", "all_transcript_ids"], [9, 2, 1, "", "all_variants"], [9, 2, 1, "", "from_patients"], [9, 2, 1, "", "get_excluded_count"], [9, 2, 1, "", "get_patient_ids"], [9, 2, 1, "", "get_variant_by_key"], [9, 2, 1, "", "list_all_diseases"], [9, 2, 1, "", "list_all_proteins"], [9, 2, 1, "", "list_all_variants"], [9, 2, 1, "", "list_present_phenotypes"], [9, 3, 1, "", "total_patient_count"], [9, 2, 1, "", "variant_effect_count_by_tx"]], "genophenocorr.model.Disease": [[9, 3, 1, "", "identifier"], [9, 4, 1, "", "is_observed"], [9, 3, 1, "", "is_present"], [9, 3, 1, "id6", "name"], [9, 4, 1, "", "term_id"]], "genophenocorr.model.FeatureInfo": [[9, 3, 1, "", "end"], [9, 3, 1, "id26", "name"], [9, 3, 1, "id27", "region"], [9, 3, 1, "", "start"]], "genophenocorr.model.FeatureType": [[9, 4, 1, "id30", "DOMAIN"], [9, 4, 1, "id29", "MOTIF"], [9, 4, 1, "id31", "REGION"], [9, 4, 1, "id28", "REPEAT"]], "genophenocorr.model.Genotype": [[9, 4, 1, "", "HEMIZYGOUS"], [9, 4, 1, "", "HETEROZYGOUS"], [9, 4, 1, "", "HOMOZYGOUS_ALTERNATE"], [9, 4, 1, "", "HOMOZYGOUS_REFERENCE"], [9, 4, 1, "", "NO_CALL"], [9, 3, 1, "", "code"]], "genophenocorr.model.Genotyped": [[9, 2, 1, "", "genotype_for_sample"], [9, 3, 1, "", "genotypes"]], "genophenocorr.model.Genotypes": [[9, 2, 1, "", "empty"], [9, 2, 1, "", "for_sample"], [9, 2, 1, "", "from_mapping"], [9, 2, 1, "", "single"]], "genophenocorr.model.ImpreciseSvInfo": [[9, 3, 1, "", "gene_id"], [9, 3, 1, "", "gene_symbol"], [9, 3, 1, "", "structural_type"], [9, 3, 1, "", "variant_class"], [9, 3, 1, "", "variant_key"]], "genophenocorr.model.Patient": [[9, 3, 1, "id2", "diseases"], [9, 2, 1, "", "excluded_diseases"], [9, 2, 1, "", "excluded_phenotypes"], [9, 3, 1, "id0", "labels"], [9, 3, 1, "", "patient_id"], [9, 3, 1, "id1", "phenotypes"], [9, 2, 1, "", "present_diseases"], [9, 2, 1, "", "present_phenotypes"], [9, 3, 1, "id3", "variants"]], "genophenocorr.model.Phenotype": [[9, 2, 1, "", "from_term"], [9, 3, 1, "", "identifier"], [9, 3, 1, "id5", "is_observed"], [9, 3, 1, "", "is_present"], [9, 3, 1, "id4", "name"], [9, 3, 1, "", "observed"], [9, 4, 1, "", "term_id"]], "genophenocorr.model.ProteinFeature": [[9, 2, 1, "", "create"], [9, 3, 1, "", "feature_type"], [9, 3, 1, "", "info"], [9, 2, 1, "", "to_string"]], "genophenocorr.model.ProteinMetadata": [[9, 2, 1, "id22", "domains"], [9, 2, 1, "", "get_features_variant_overlaps"], [9, 3, 1, "id20", "label"], [9, 4, 1, "", "length"], [9, 2, 1, "id25", "motifs"], [9, 3, 1, "id21", "protein_features"], [9, 3, 1, "id19", "protein_id"], [9, 3, 1, "", "protein_length"], [9, 2, 1, "id24", "regions"], [9, 2, 1, "id23", "repeats"]], "genophenocorr.model.SampleLabels": [[9, 3, 1, "", "label"], [9, 2, 1, "", "label_summary"], [9, 3, 1, "", "meta_label"]], "genophenocorr.model.TranscriptAnnotation": [[9, 3, 1, "", "gene_id"], [9, 3, 1, "", "hgvs_cdna"], [9, 3, 1, "", "hgvsp"], [9, 3, 1, "", "is_preferred"], [9, 3, 1, "", "overlapping_exons"], [9, 3, 1, "", "protein_effect_location"], [9, 3, 1, "", "protein_id"], [9, 3, 1, "", "transcript_id"], [9, 3, 1, "", "variant_effects"]], "genophenocorr.model.TranscriptCoordinates": [[9, 3, 1, "", "cds_end"], [9, 3, 1, "", "cds_start"], [9, 3, 1, "", "exons"], [9, 2, 1, "", "get_cds_regions"], [9, 2, 1, "", "get_coding_base_count"], [9, 2, 1, "", "get_codon_count"], [9, 2, 1, "", "get_five_prime_utrs"], [9, 2, 1, "", "get_three_prime_utrs"], [9, 3, 1, "", "identifier"], [9, 2, 1, "", "is_coding"], [9, 3, 1, "", "is_preferred"], [9, 3, 1, "", "region"]], "genophenocorr.model.TranscriptInfoAware": [[9, 3, 1, "", "gene_id"], [9, 3, 1, "", "transcript_id"]], "genophenocorr.model.Variant": [[9, 2, 1, "", "create_variant_from_scratch"], [9, 3, 1, "", "genotypes"], [9, 3, 1, "", "tx_annotations"], [9, 3, 1, "", "variant_info"]], "genophenocorr.model.VariantClass": [[9, 4, 1, "", "BND"], [9, 4, 1, "", "DEL"], [9, 4, 1, "", "DUP"], [9, 4, 1, "", "INS"], [9, 4, 1, "", "INV"], [9, 4, 1, "", "MNV"], [9, 4, 1, "", "SNV"]], "genophenocorr.model.VariantCoordinates": [[9, 3, 1, "id9", "alt"], [9, 3, 1, "id10", "change_length"], [9, 3, 1, "", "chrom"], [9, 3, 1, "", "end"], [9, 2, 1, "", "is_structural"], [9, 3, 1, "id8", "ref"], [9, 3, 1, "id7", "region"], [9, 3, 1, "", "start"], [9, 3, 1, "", "variant_class"], [9, 3, 1, "", "variant_key"]], "genophenocorr.model.VariantEffect": [[9, 4, 1, "", "CODING_SEQUENCE_VARIANT"], [9, 4, 1, "", "DOWNSTREAM_GENE_VARIANT"], [9, 4, 1, "", "FEATURE_ELONGATION"], [9, 4, 1, "", "FEATURE_TRUNCATION"], [9, 4, 1, "", "FIVE_PRIME_UTR_VARIANT"], [9, 4, 1, "", "FRAMESHIFT_VARIANT"], [9, 4, 1, "", "INCOMPLETE_TERMINAL_CODON_VARIANT"], [9, 4, 1, "", "INFRAME_DELETION"], [9, 4, 1, "", "INFRAME_INSERTION"], [9, 4, 1, "", "INTERGENIC_VARIANT"], [9, 4, 1, "", "INTRON_VARIANT"], [9, 4, 1, "", "MATURE_MIRNA_VARIANT"], [9, 4, 1, "", "MISSENSE_VARIANT"], [9, 4, 1, "", "NMD_TRANSCRIPT_VARIANT"], [9, 4, 1, "", "NON_CODING_TRANSCRIPT_EXON_VARIANT"], [9, 4, 1, "", "NON_CODING_TRANSCRIPT_VARIANT"], [9, 4, 1, "", "PROTEIN_ALTERING_VARIANT"], [9, 4, 1, "", "REGULATORY_REGION_ABLATION"], [9, 4, 1, "", "REGULATORY_REGION_AMPLIFICATION"], [9, 4, 1, "", "REGULATORY_REGION_VARIANT"], [9, 4, 1, "", "SEQUENCE_VARIANT"], [9, 4, 1, "", "SPLICE_ACCEPTOR_VARIANT"], [9, 4, 1, "", "SPLICE_DONOR_5TH_BASE_VARIANT"], [9, 4, 1, "", "SPLICE_DONOR_REGION_VARIANT"], [9, 4, 1, "", "SPLICE_DONOR_VARIANT"], [9, 4, 1, "", "SPLICE_POLYPYRIMIDINE_TRACT_VARIANT"], [9, 4, 1, "", "SPLICE_REGION_VARIANT"], [9, 4, 1, "", "START_LOST"], [9, 4, 1, "", "START_RETAINED_VARIANT"], [9, 4, 1, "", "STOP_GAINED"], [9, 4, 1, "", "STOP_LOST"], [9, 4, 1, "", "STOP_RETAINED_VARIANT"], [9, 4, 1, "", "SYNONYMOUS_VARIANT"], [9, 4, 1, "", "TFBS_ABLATION"], [9, 4, 1, "", "TFBS_AMPLIFICATION"], [9, 4, 1, "", "TF_BINDING_SITE_VARIANT"], [9, 4, 1, "", "THREE_PRIME_UTR_VARIANT"], [9, 4, 1, "", "TRANSCRIPT_ABLATION"], [9, 4, 1, "", "TRANSCRIPT_AMPLIFICATION"], [9, 4, 1, "", "UPSTREAM_GENE_VARIANT"], [9, 3, 1, "", "curie"]], "genophenocorr.model.VariantInfo": [[9, 2, 1, "", "has_sv_info"], [9, 2, 1, "", "has_variant_coordinates"], [9, 2, 1, "", "is_structural"], [9, 3, 1, "", "sv_info"], [9, 3, 1, "", "variant_class"], [9, 3, 1, "", "variant_coordinates"], [9, 3, 1, "", "variant_key"]], "genophenocorr.model.VariantInfoAware": [[9, 3, 1, "", "variant_info"]], "genophenocorr.preprocessing": [[11, 1, 1, "", "Auditor"], [11, 1, 1, "", "CohortCreator"], [11, 1, 1, "", "DataSanityIssue"], [11, 1, 1, "", "DefaultImpreciseSvFunctionalAnnotator"], [11, 1, 1, "", "FunctionalAnnotator"], [11, 1, 1, "", "GeneCoordinateService"], [11, 1, 1, "", "ImpreciseSvFunctionalAnnotator"], [11, 1, 1, "", "Level"], [11, 1, 1, "", "Notepad"], [11, 1, 1, "", "NotepadTree"], [11, 1, 1, "", "PatientCreator"], [11, 1, 1, "", "PhenopacketPatientCreator"], [11, 1, 1, "", "PhenopacketVariantCoordinateFinder"], [11, 1, 1, "", "PhenotypeCreator"], [11, 1, 1, "", "ProtCachingMetadataService"], [11, 1, 1, "", "ProteinAnnotationCache"], [11, 1, 1, "", "ProteinMetadataService"], [11, 1, 1, "", "TranscriptCoordinateService"], [11, 1, 1, "", "UniprotProteinMetadataService"], [11, 1, 1, "", "VVHgvsVariantCoordinateFinder"], [11, 1, 1, "", "VVMultiCoordinateService"], [11, 1, 1, "", "VarCachingFunctionalAnnotator"], [11, 1, 1, "", "VariantAnnotationCache"], [11, 1, 1, "", "VariantCoordinateFinder"], [11, 1, 1, "", "VepFunctionalAnnotator"], [11, 5, 1, "", "configure_caching_cohort_creator"], [11, 5, 1, "", "configure_caching_patient_creator"], [11, 5, 1, "", "configure_cohort_creator"], [11, 5, 1, "", "configure_patient_creator"], [11, 5, 1, "", "load_phenopacket"], [11, 5, 1, "", "load_phenopacket_folder"]], "genophenocorr.preprocessing.Auditor": [[11, 2, 1, "", "prepare_notepad"], [11, 2, 1, "", "process"]], "genophenocorr.preprocessing.CohortCreator": [[11, 2, 1, "", "process"]], "genophenocorr.preprocessing.DataSanityIssue": [[11, 3, 1, "", "level"], [11, 3, 1, "", "message"], [11, 3, 1, "", "solution"]], "genophenocorr.preprocessing.DefaultImpreciseSvFunctionalAnnotator": [[11, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.FunctionalAnnotator": [[11, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.GeneCoordinateService": [[11, 2, 1, "", "fetch_for_gene"]], "genophenocorr.preprocessing.ImpreciseSvFunctionalAnnotator": [[11, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.Level": [[11, 4, 1, "", "ERROR"], [11, 4, 1, "", "WARN"]], "genophenocorr.preprocessing.Notepad": [[11, 2, 1, "", "add_error"], [11, 2, 1, "", "add_issue"], [11, 2, 1, "", "add_subsection"], [11, 2, 1, "", "add_warning"], [11, 2, 1, "", "error_count"], [11, 2, 1, "", "errors"], [11, 3, 1, "", "issues"], [11, 3, 1, "", "label"], [11, 2, 1, "", "warning_count"], [11, 2, 1, "", "warnings"]], "genophenocorr.preprocessing.NotepadTree": [[11, 2, 1, "", "add_subsection"], [11, 3, 1, "", "children"], [11, 2, 1, "", "has_errors"], [11, 2, 1, "", "has_errors_or_warnings"], [11, 2, 1, "", "has_warnings"], [11, 2, 1, "", "iterate_nodes"], [11, 3, 1, "", "level"], [11, 2, 1, "", "visit"]], "genophenocorr.preprocessing.PhenopacketPatientCreator": [[11, 2, 1, "", "process"]], "genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder": [[11, 2, 1, "", "find_coordinates"]], "genophenocorr.preprocessing.PhenotypeCreator": [[11, 2, 1, "", "process"]], "genophenocorr.preprocessing.ProtCachingMetadataService": [[11, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.ProteinAnnotationCache": [[11, 2, 1, "", "get_annotations"], [11, 2, 1, "", "store_annotations"]], "genophenocorr.preprocessing.ProteinMetadataService": [[11, 2, 1, "", "annotate"]], "genophenocorr.preprocessing.TranscriptCoordinateService": [[11, 2, 1, "", "fetch"]], "genophenocorr.preprocessing.UniprotProteinMetadataService": [[11, 2, 1, "", "annotate"], [11, 2, 1, "", "parse_uniprot_json"]], "genophenocorr.preprocessing.VVHgvsVariantCoordinateFinder": [[11, 2, 1, "", "find_coordinates"]], "genophenocorr.preprocessing.VVMultiCoordinateService": [[11, 2, 1, "", "fetch"], [11, 2, 1, "", "fetch_for_gene"], [11, 2, 1, "", "get_response"], [11, 2, 1, "", "parse_multiple"], [11, 2, 1, "", "parse_response"]], "genophenocorr.preprocessing.VarCachingFunctionalAnnotator": [[11, 2, 1, "", "annotate"], [11, 2, 1, "", "with_cache_folder"]], "genophenocorr.preprocessing.VariantAnnotationCache": [[11, 2, 1, "", "get_annotations"], [11, 2, 1, "", "store_annotations"]], "genophenocorr.preprocessing.VariantCoordinateFinder": [[11, 2, 1, "", "find_coordinates"]], "genophenocorr.preprocessing.VepFunctionalAnnotator": [[11, 4, 1, "", "NONCODING_EFFECTS"], [11, 2, 1, "", "annotate"], [11, 2, 1, "", "fetch_response"], [11, 2, 1, "", "format_coordinates_for_vep_query"], [11, 2, 1, "", "process_response"]], "genophenocorr.view": [[12, 1, 1, "", "CohortViewable"], [12, 1, 1, "", "DiseaseViewable"], [12, 1, 1, "", "ProteinViewable"], [12, 1, 1, "", "ProteinVisualizable"], [12, 1, 1, "", "ProteinVisualizer"], [12, 1, 1, "", "StatsViewer"], [12, 1, 1, "", "VariantFormatter"], [12, 1, 1, "", "VariantTranscriptVisualizer"]], "genophenocorr.view.CohortViewable": [[12, 2, 1, "", "get_variant_description"], [12, 2, 1, "", "process"]], "genophenocorr.view.DiseaseViewable": [[12, 2, 1, "", "process"]], "genophenocorr.view.ProteinViewable": [[12, 2, 1, "", "process"]], "genophenocorr.view.ProteinVisualizable": [[12, 3, 1, "", "marker_counts"], [12, 3, 1, "", "protein_feature_ends"], [12, 3, 1, "", "protein_feature_names"], [12, 3, 1, "", "protein_feature_starts"], [12, 3, 1, "", "protein_feature_types"], [12, 3, 1, "", "protein_id"], [12, 3, 1, "", "protein_length"], [12, 3, 1, "", "protein_metadata"], [12, 3, 1, "", "transcript_coordinates"], [12, 3, 1, "", "transcript_id"], [12, 3, 1, "", "variant_effects"], [12, 3, 1, "", "variant_locations"], [12, 3, 1, "", "variant_locations_counted_absolute"]], "genophenocorr.view.ProteinVisualizer": [[12, 2, 1, "", "draw_fig"]], "genophenocorr.view.StatsViewer": [[12, 2, 1, "", "process"]], "genophenocorr.view.VariantFormatter": [[12, 2, 1, "", "format_as_string"]], "genophenocorr.view.VariantTranscriptVisualizer": [[12, 2, 1, "", "draw_variants"]]}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "property", "Python property"], "4": ["py", "attribute", "Python attribute"], "5": ["py", "function", "Python function"]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:property", "4": "py:attribute", "5": "py:function"}, "terms": {"": [3, 4, 8, 9, 11, 15, 16, 21, 22, 23], "0": [1, 2, 3, 4, 9, 11, 12, 16, 22, 24], "000": [1, 16], "000000": [1, 16], "0000002": 19, "0000078": 19, "0000079": 19, "0000080": 19, "0000118": [19, 22], "0000119": 19, "0000152": 19, "0000223": 19, "0000234": 19, "0000271": 19, "0000356": 19, "0000359": 19, "0000370": 19, "0000464": 19, "0000478": 19, "0000486": 24, "0000496": 19, "0000504": 19, "0000539": 19, "0000553": 19, "0000591": 19, "0000598": 19, "0000707": [1, 19, 22], "0000759": 19, "0000769": 19, "0000777": 19, "0000818": 19, "0000820": 19, "0000828": 19, "0000834": 19, "0000847": 19, "0000864": 19, "0000924": 19, "0000927": 19, "0000929": 19, "0000951": 19, "0000971": 19, "0001060": 9, "0001115": 22, "0001155": 19, "0001166": [1, 4, 16], "0001194": 19, "0001197": 19, "0001250": [1, 16], "0001257": 16, "0001311": 19, "0001392": 19, "0001437": 19, "0001438": 19, "0001446": 19, "0001454": 19, "0001507": 19, "0001537": 3, "0001551": 19, "0001560": 19, "0001566": 9, "0001567": 9, "0001574": [9, 19], "0001575": 9, "0001578": 9, "0001580": 9, "0001583": 9, "0001587": 9, "0001589": 9, "0001595": 19, "0001597": 19, "0001608": 19, "0001619": 9, "0001620": 9, "0001621": 9, "0001623": 9, "0001624": 9, "0001626": [9, 19], "0001627": [9, 19], "0001628": 9, "0001630": 9, "0001631": 9, "0001632": 9, "0001732": 19, "0001743": 19, "0001760": 19, "0001782": 9, "0001787": [9, 19], "0001792": 9, "0001818": 9, "0001819": 9, "0001821": 9, "0001822": 9, "0001871": 19, "0001872": 19, "0001877": 19, "0001881": 19, "0001889": 9, "0001891": 9, "0001892": [9, 19], "0001893": 9, "0001894": 9, "0001895": 9, "0001906": 9, "0001907": 9, "0001928": 19, "0001939": 19, "0001965": 19, "0001977": 19, "0002012": [9, 19], "0002019": 9, "0002086": 19, "0002169": 9, "0002170": 9, "0002270": 19, "0002339": 22, "0002585": 19, "0002597": 19, "0002664": 19, "0002715": 19, "0002733": 19, "0002793": 19, "0002795": 19, "0002813": 19, "0002814": 19, "0002817": 19, "0002916": 19, "0002926": 19, "0002973": 19, "0002981": 19, "0003011": [1, 19], "0003110": 19, "0003111": 19, "0003117": 19, "0003220": 19, "0003254": 19, "0003256": 19, "0003549": 19, "0003839": 19, "0004298": 19, "0004312": 19, "0004323": 19, "0004328": 19, "0004329": 19, "0004332": 19, "0004337": 19, "0004352": 19, "0004353": 19, "0004356": 19, "0004360": 19, "0004361": 19, "0004370": 19, "0004371": 19, "0004378": 19, "0004404": 19, "0004408": 19, "0004930": 19, "0005368": 19, "0005561": 19, "0006476": 19, "0006490": 19, "0006500": 19, "0006529": 19, "0006707": 19, "0007686": 19, "000781": [1, 16], "0008047": 19, "0009124": 19, "0009127": 19, "0009809": 19, "0009810": 19, "0010696": 22, "0010876": 19, "0010881": 19, "0010932": 19, "0010935": 19, "0010936": 19, "0010948": 19, "0010968": 19, "0010969": 19, "0010974": 19, "0010977": 19, "0010978": 19, "0010987": 19, "0010991": 19, "0011014": 19, "0011017": 19, "0011018": 19, "0011019": 19, "0011024": 19, "0011025": 19, "0011028": 19, "0011032": 19, "0011043": 19, "0011111": 19, "0011121": 19, "0011122": 19, "0011138": 19, "0011277": 19, "0011409": 19, "0011442": 19, "0011446": 19, "0011620": 19, "0011730": 19, "0011732": 19, "0011733": 19, "0011766": 19, "0011767": 19, "0011772": 19, "0011804": 19, "0011805": 19, "0011842": 19, "0011843": 19, "0011869": 19, "0011875": 19, "0011884": 19, "0011893": 19, "0011902": 19, "0012029": 19, "0012093": 19, "0012099": 19, "0012103": 19, "0012111": 19, "0012129": 19, "0012130": 19, "0012131": 19, "0012135": 19, "0012143": 19, "0012145": 19, "0012243": 19, "0012252": 19, "0012261": 19, "0012285": 19, "0012337": 19, "0012338": 19, "0012345": 19, "0012372": 19, "0012373": 19, "0012379": 19, "0012415": 19, "0012443": 4, "0012447": 19, "0012503": 19, "0012535": 19, "0012591": 19, "0012614": 19, "0012632": 19, "0012638": [1, 19], "0012639": 19, "0012640": 19, "0012647": 19, "0012654": 19, "0012680": 19, "0012681": 19, "0012688": 19, "0012700": 19, "0012718": 19, "0012757": 19, "0012769": 19, "0012772": 19, "0020047": 19, "0020054": 19, "0020058": 19, "0020061": 19, "0020155": 19, "0020169": 19, "0025015": 19, "0025021": 19, "0025031": 19, "0025032": 19, "0025033": 19, "0025065": 19, "0025142": 19, "0025155": 19, "0025276": 19, "0025354": 19, "0025427": 19, "0025429": 19, "0025443": 19, "0025454": 19, "0025456": 19, "0025461": 19, "0025463": 19, "0025546": 19, "0025590": 19, "0025640": 19, "0025668": 19, "0025669": 19, "0025688": 19, "0030085": 19, "0030163": 19, "0030272": 19, "0030338": 19, "0030352": 19, "0030453": 19, "0030680": 19, "0030684": 19, "0030687": 19, "0030800": 19, "0030829": 19, "0030860": 19, "0030872": 19, "0030875": 19, "0030878": 19, "0030956": 19, "0030972": 19, "0031071": 19, "0031072": 19, "0031073": 19, "0031093": 19, "0031094": 19, "0031097": 19, "0031099": 19, "0031101": 19, "0031285": 19, "0031331": 19, "0031340": 19, "0031377": 19, "0031383": 19, "0031389": 19, "0031409": 19, "0031411": 19, "0031416": 19, "0031427": 19, "0031476": 19, "0031508": 19, "0031550": 19, "0031602": 19, "0031653": 19, "0031657": 19, "0031685": 19, "0031703": 19, "0031704": 19, "0031818": 19, "0031850": 19, "0031871": 19, "0031884": 19, "0031910": 19, "0031982": 22, "0032120": 19, "0032180": 19, "0032226": 19, "0032243": 19, "0032245": 19, "0032251": 19, "0032314": 19, "0032367": 19, "0032481": 19, "0032485": 19, "0032488": 19, "0032943": 19, "0033012": 19, "0033013": 19, "0033072": 19, "0033127": 19, "0033170": 19, "0033334": 19, "0033335": 19, "0033354": 19, "0033358": 19, "0033796": 19, "0033799": 19, "0034190": 19, "0034200": 19, "0034251": 19, "0034263": 19, "0034370": 19, "0034430": 19, "0034442": 19, "0034482": 19, "0034552": 19, "0034644": 19, "0034684": 19, "0034698": 19, "0034737": 19, "0034858": 19, "0034899": 19, "0034915": 19, "0034916": 19, "0034977": 19, "0040064": 19, "0040068": 19, "0040069": 19, "0040070": 19, "0040084": 19, "0040085": 19, "0040127": 19, "0040172": 19, "0040203": 19, "0040207": 19, "0040214": 19, "0040224": 19, "0040231": 19, "0045026": 19, "0045027": 19, "0045081": 19, "01": [1, 22], "0100016": 19, "0100022": 19, "0100491": 19, "0100508": 19, "0100530": 19, "0100536": 19, "0100685": 19, "0100705": 19, "0100763": 19, "0100765": 19, "0100766": 19, "0100767": 19, "0100886": 19, "0100887": 19, "020299": [1, 16], "03": [21, 23], "0410008": 19, "0410009": 19, "0410014": 19, "0430071": 19, "05": [1, 22, 24], "0500012": 19, "0500015": 19, "0500016": 19, "0500017": 19, "0500018": 19, "0500019": 19, "0500020": 19, "0500114": 19, "0500117": 19, "0500166": 19, "0500183": 19, "0500238": 19, "06": [21, 23, 24], "1": [1, 2, 3, 9, 11, 16, 19, 22, 23, 24], "10": [1, 3, 9, 11, 12, 16, 22], "100": [1, 22], "1000": 9, "1000029": [3, 9, 23], "101": 9, "11": [1, 9, 16, 22], "120": 9, "13": [1, 16], "14": 16, "15": 1, "16": [1, 16], "16_89281397_89281397_g_c": 23, "17": [16, 24], "1715th": 23, "19": 24, "1_281_281_a_g": 16, "1_361_363_ttc_t": 16, "1d": 12, "1st": [3, 9], "2": [1, 3, 9, 11, 16, 19, 22, 24], "20": [9, 16, 22], "2015": 14, "2018": 24, "2022": 17, "22_10001_20000_inv": [3, 9], "23": 19, "253a": 11, "29": 16, "2nd": 3, "3": [1, 9, 11, 19, 22, 24], "30": 11, "3000013": 19, "3000036": 19, "3000050": 19, "348081479150901e": 24, "348081479150902e": 24, "36": 22, "3603": [9, 11], "36303223": 17, "3x2": 1, "4": [1, 3, 4, 9, 11, 24], "4000056": 19, "4000183": 19, "42": 12, "432292015291845e": 24, "450": 22, "5": [1, 4, 9, 11, 16, 21, 22, 23, 24], "50": [3, 22, 23], "50bp": 3, "5145c": 23, "53": 24, "6": [1, 9, 16, 17, 23, 24], "60": 16, "6000062": 19, "6000231": 19, "6000489": 19, "6000673": 19, "64": 22, "69": 16, "692449": 16, "7": [1, 9, 22], "8": 22, "81": [1, 16], "833333333333336": 24, "9": [1, 23], "90": 22, "95": 22, "9876g": 9, "99": 22, "A": [1, 2, 4, 9, 11, 12, 14, 16, 22], "AND": 23, "As": 23, "By": [12, 22], "For": [1, 4, 8, 9, 11, 16, 17, 21, 22, 23, 24], "IN": 11, "INS": 9, "If": [1, 3, 9, 22], "In": [1, 4, 11, 14, 17, 22, 24], "It": 22, "NO": [2, 4], "NOT": [1, 3, 9], "No": [1, 2, 16], "ONE": 3, "OR": 23, "On": 4, "Or": 9, "That": [1, 15, 23], "The": [1, 2, 3, 4, 8, 9, 10, 11, 12, 14, 15, 16, 19, 21, 22, 23, 24], "Then": [4, 15], "There": [9, 16, 24], "To": [11, 15, 16, 24], "_": 19, "__eq__": 3, "__hash__": 3, "__repr__": 3, "__str__": 3, "_max_term": 4, "_missing_implies_phenotype_exclud": 4, "_queri": 4, "abbrevi": [12, 22], "abdomen": 19, "abdomin": 19, "abn_caudate_nucleu": 22, "abn_putamen": 22, "abnorm": [1, 4, 14, 19, 22], "aborm": 4, "abort": 11, "about": [1, 9, 11, 12, 15, 16, 18, 22], "abov": [1, 3, 15, 16, 19, 21], "absenc": [1, 4, 16], "abstract": [1, 2, 3, 4, 9, 11], "accept": 9, "access": [1, 3, 9, 11, 15], "accommod": 19, "accord": [1, 4, 19, 24], "account": 22, "acetylaspart": 19, "acid": [9, 19], "across": [1, 12], "activ": 19, "ad": 11, "adapt": 19, "add": [11, 15], "add_error": 11, "add_issu": 11, "add_subsect": 11, "add_warn": 11, "addit": 15, "adenosin": 19, "adiponectin": 19, "adipos": 19, "adnexa": 19, "adren": 19, "adrenocorticotropin": 19, "affect": [1, 3, 9, 17], "affected_exon": 9, "after": [1, 11, 15], "ag": 9, "against": 17, "al": [14, 17, 24], "albeit": 11, "albumin": 19, "aldosteron": 19, "all": [1, 2, 3, 4, 9, 11, 15, 16, 19, 22, 23], "all_count": 1, "all_diseas": 9, "all_hpo_term": 1, "all_pati": 9, "all_phenotyp": 9, "all_terms_strategi": 1, "all_transcript_id": 9, "all_vari": 9, "allel": [1, 3, 9, 14, 17], "allelecount": 3, "allianc": 14, "allow": [1, 22], "allow_nan": 8, "alon": 22, "along": [8, 21, 23], "alpha": [1, 19, 22, 24], "also": [1, 19], "alt": 9, "altern": [1, 3, 9, 22, 24], "alwai": [1, 3], "amaz": 21, "amino": [9, 19], "aminoacid": [9, 12, 16], "amniot": 19, "among": 9, "amyloid": 19, "an": [1, 2, 3, 4, 9, 11, 12, 14, 15, 16, 17, 21, 22, 23, 24], "analys": [1, 16, 21], "analysi": [0, 11, 14, 17, 20, 21, 22, 23], "analyt": 1, "analyz": [0, 1, 12], "angiotensin": 19, "ani": [2, 3, 8, 9, 11, 22, 23], "ank": [3, 9], "ankrd11": 23, "ankrd11_mane_tx_id": 23, "ankyrin": 9, "annot": [1, 3, 4, 9, 11, 12, 16, 21, 22], "anoth": 21, "answer": [2, 3, 4], "anterior": 19, "antimullerian": 19, "anu": 19, "api": [1, 11, 12, 14, 16, 21, 23], "apoptosi": 19, "app": 9, "appear": 19, "appli": 1, "apply_predicates_on_pati": [0, 1], "approach": 22, "appropri": 24, "ar": [1, 2, 3, 4, 9, 10, 11, 12, 14, 15, 16, 17, 19, 22, 23, 24], "arachnodactyli": [1, 4, 16], "arbitrari": 8, "areolar": 19, "arg": 8, "argument": [9, 11, 12, 22], "arithmet": 10, "arm": 19, "aromatas": 19, "arrai": 12, "arrang": 19, "arriv": 22, "ascend": 16, "ask": 14, "aspect": 2, "assert": 9, "assess": 1, "assign": [2, 3, 16, 23], "associ": [9, 11, 14, 17, 24], "assum": [1, 9, 22], "assumpt": 22, "atrial": 19, "atrophin": 24, "attribut": [4, 9, 14], "audit": 11, "auditor": [0, 11], "autonom": 19, "autosom": 20, "avail": [1, 9, 11, 14, 23], "avoid": [12, 19], "ax": 12, "axi": 19, "b": [1, 9, 16, 22], "background": 14, "bandwidth": 21, "base": [1, 2, 3, 4, 8, 9, 11, 12, 16, 19, 23, 24], "basic": [9, 10], "becaus": [9, 17, 19, 22], "been": 22, "befor": 1, "behavior": 14, "being": [1, 11, 12, 22], "belong": [2, 9], "below": [3, 22], "benjamini": 22, "best": 14, "beta": 19, "better": 16, "between": [1, 4, 9, 10, 12, 14, 16, 22, 24], "beyond": 1, "bin": [1, 2, 3, 4], "bind": 1, "biolog": 9, "biologi": 14, "biomark": 19, "biopterin": 19, "birth": 19, "bleed": 19, "block": 23, "blood": 19, "bnd": [3, 9], "bodi": 19, "bone": 19, "bonferroni": [1, 22], "bool": [1, 3, 4, 9, 11, 12], "boolean": [2, 3, 9, 11], "boolean_pred": 3, "booleanpred": [1, 2], "both": [1, 3, 9, 11, 22], "boundari": [1, 9, 11], "box": 21, "bp": [3, 23], "brain": [4, 24], "branch": [15, 22], "break": 9, "breakend": 9, "breakpoint": [3, 9, 11], "breast": 19, "breath": 19, "bridg": 20, "brief": 14, "bring": 19, "bronchu": 19, "brows": 19, "buccal": 19, "bug": 14, "build": [3, 10, 11, 16, 23], "burden": [1, 22], "c": [2, 9, 11, 14, 16, 21, 23], "cach": [1, 11, 21], "cache_dir": [1, 11], "calcium": 19, "calcul": [1, 9, 24], "calf": 19, "call": [8, 9, 11, 19, 21, 22], "callabl": 11, "caller": 11, "can": [1, 2, 3, 4, 9, 10, 11, 12, 15, 16, 18, 19, 21, 22, 23], "cannot": [1, 2, 9, 11, 23], "canon": 9, "cap": 22, "carboxyl": 19, "cardiac": 19, "cardiomyocyt": 19, "cardiovascular": 19, "carri": 17, "cart": 19, "cascad": 19, "case": [1, 4, 11], "cat_id": 2, "cataract": 22, "catecholamin": 19, "categor": [1, 2, 3, 4], "categori": [1, 2, 4, 9, 23], "catheter": 19, "caudat": 22, "caus": [14, 22], "caviti": 19, "cd": [9, 15], "cds_end": 9, "cds_start": 9, "cell": [19, 22], "cellular": 19, "central": 19, "certain": [1, 2, 3, 11], "chain": [9, 23], "chanc": 22, "chang": [3, 9, 16, 23], "change_length": [3, 9, 23], "changelengthpred": 3, "charact": 9, "characterist": 9, "check": [1, 2, 9, 11, 22, 23], "check_circular": 8, "checkout": 15, "child": [1, 22], "children": [11, 19, 22], "choic": 9, "choos": [1, 11, 16], "chosen": [9, 22, 24], "chr5": 9, "chrom": 9, "chromosom": [3, 9, 12, 17, 19, 23], "chromosomal_delet": 23, "cilium": 19, "circul": 19, "cl": 8, "class": [1, 2, 3, 4, 8, 9, 10, 11, 12, 22, 23], "clearanc": 19, "clinic": [16, 22, 23], "clone": 15, "coagul": 19, "code": [1, 9, 11, 15, 16, 24], "coding_sequence_vari": 9, "codon": [9, 14, 23], "cohort": [0, 1, 7, 9, 11, 12, 14, 16, 20, 22, 23, 24], "cohort_analysi": 16, "cohort_cr": [11, 21, 23], "cohort_dir": 23, "cohortanalysi": [0, 1], "cohortanalysisconfigur": [0, 1, 22], "cohortcr": [0, 11, 21], "cohortview": [0, 12], "collect": [1, 9, 11, 12, 14], "column": 1, "com": 15, "combin": [9, 22, 23], "come": 24, "command": [15, 16], "common": [2, 17], "commonli": [10, 22], "compact": [3, 9], "compar": [1, 17, 24], "compare_by_exon": 1, "compare_by_protein_featur": 1, "compare_by_protein_feature_typ": 1, "compare_by_protein_region": 1, "compare_by_variant_effect": 1, "compare_by_variant_kei": 1, "compare_hpo_vs_genotyp": [1, 16], "compare_hpo_vs_genotype_group": 1, "comparison": 17, "compat": 2, "complain": 11, "complet": [11, 22], "complex": [22, 23], "composit": 19, "compound": [3, 17, 19], "comprehens": [14, 23], "comput": [1, 11], "concentr": 19, "concept": 14, "conceptu": 9, "concern": 14, "conclus": 22, "condens": 19, "condit": [22, 23], "conduct": 24, "config": [1, 22], "configur": [1, 3, 11, 21, 22, 23], "configure_caching_cohort_cr": [0, 11, 21, 23], "configure_caching_patient_cr": [0, 11], "configure_cohort_analysi": [0, 1, 16], "configure_cohort_cr": [0, 11], "configure_default_protein_metadata_servic": [0, 1], "configure_ontology_stor": [21, 23], "configure_patient_cr": [0, 11], "connect": 19, "consequ": 9, "conserv": 9, "consid": [3, 9, 19, 24], "consider": 19, "consist": [9, 21], "constant": 0, "constitut": 19, "constrain": 2, "construct": [11, 21], "constructor": 1, "consult": 3, "contain": [1, 3, 9, 11, 12, 19], "context": [4, 14], "contig": [9, 10], "conting": 24, "contingency_t": 24, "control": 22, "conveni": [11, 21], "convert": 11, "coordin": [3, 9, 10, 11, 12, 21], "copi": 17, "cord": 19, "corneal": 19, "corpuscular": 19, "correct": [1, 16, 19, 20], "corrected_pv": 1, "correctli": 11, "correl": [0, 1, 14, 16, 19, 20, 22, 24], "correspond": [1, 4, 9, 11, 22, 23, 24], "cortisol": 19, "could": [1, 8], "count": [1, 3, 4, 9, 11, 16, 18, 19, 22, 24], "counter": 9, "countingphenotypepred": 4, "counts_fram": 1, "cover": [2, 23], "cranial": 19, "creat": [1, 3, 4, 9, 11, 12, 14, 16, 23], "create_variant_from_scratch": 9, "creation": 1, "creator": [11, 23], "criterion": 3, "cry": 19, "csf": 19, "ct": 9, "curi": [1, 3, 9, 11], "current": [1, 11, 12, 15, 21], "cycl": 19, "cytologi": 19, "cytometri": 19, "d": 16, "data": [0, 1, 9, 10, 11, 12, 14, 16, 20, 22, 23], "databas": 11, "datadir": 11, "datafram": 1, "datasanityissu": [0, 11], "dataset": 21, "decid": [1, 19], "decod": 8, "deem": 19, "deep": 14, "def": 8, "default": [3, 8, 9, 12, 21, 22, 24], "defaultimprecisesvfunctionalannot": [0, 11], "defect": 24, "defin": [3, 9, 21], "definit": 9, "del": [9, 23], "delet": [3, 9, 23], "deliveri": 19, "delta": 1, "demonstr": [16, 23], "denot": 14, "depend": [3, 24], "depth": 11, "deriv": 22, "descend": [3, 4, 22], "describ": [9, 15, 18, 21, 22, 23], "descript": [2, 9, 11], "design": [14, 22], "desir": [3, 11, 17, 19], "destructur": 3, "detail": [1, 2, 22, 23], "determin": [3, 22, 24], "develop": [15, 19, 22], "deviat": [14, 24], "diagnos": 4, "diagram": 12, "dichotom": 24, "dict": [8, 11], "dictionari": 1, "differ": [1, 4, 22, 24], "differenti": 14, "digest": 19, "dimension": 9, "diploid": 9, "direct": [9, 22], "directori": [1, 11, 21, 23], "discharg": 19, "discoveri": 22, "discret": [2, 4, 11], "diseas": [0, 1, 4, 9, 14, 20], "disease_id_queri": 4, "diseasepresencepred": 4, "diseaseview": [0, 12], "displai": 12, "distanc": 10, "distribut": [12, 18, 24], "divid": 24, "dna": [1, 9, 10, 19], "do": [1, 4, 10, 11, 12, 15, 16, 22], "doc": [16, 21, 23], "document": [12, 14, 15, 22, 24], "doe": [1, 2, 9, 11], "domain": [9, 19, 24], "domin": 17, "done": [11, 15], "dopamin": 19, "doubl": 4, "down": 11, "downstream": [11, 16, 21], "downstream_gene_vari": [9, 11], "draw": 12, "draw_fig": 12, "draw_vari": 12, "drawn": 12, "driven": 19, "driver": 1, "drug": 19, "due": [9, 11, 16], "dump": 8, "dup": 9, "duplic": [9, 11], "dure": 11, "dysplasia": 4, "e": [1, 3, 4, 9, 11, 12, 14, 16, 19, 22], "each": [1, 4, 9, 11, 12, 14, 16, 17, 22, 24], "ear": 19, "easi": 15, "easiest": 21, "effect": [1, 3, 9, 11, 18, 21], "effect1": 1, "effect2": 1, "either": [1, 4, 9, 11], "elabor": 23, "electrolyt": 19, "electrophysiologi": 19, "element": [4, 9, 11, 12], "els": [8, 9], "embryon": 19, "empti": [9, 11], "enabl": 15, "encod": [8, 9, 16, 23], "encount": 11, "end": [3, 9], "endocrin": 19, "energi": 19, "enrich": 22, "ensembl": 9, "enst00000123456": 9, "ensure_ascii": 8, "entir": 11, "entiti": [4, 9, 11], "entri": 1, "enum": [1, 9, 11], "enumer": 12, "environ": 15, "enzym": 19, "epiphysi": 19, "equal": [3, 9], "error": [11, 22], "error_count": 11, "erythrocyt": 19, "erythroid": 19, "erythropoietin": 19, "et": [14, 17, 24], "etc": [9, 11], "evalu": 24, "even": [1, 19, 22], "exact": [3, 4, 9, 11, 14, 17, 22], "exactli": 22, "exampl": [1, 3, 8, 9, 11, 16, 17, 21, 22, 23, 24], "except": [8, 11], "exclud": [1, 4, 9, 22, 24], "excluded_diseas": 9, "excluded_member_count": 9, "excluded_phenotyp": 9, "exclus": [2, 23], "exemplifi": 23, "exercis": 19, "exhaust": [2, 23], "exist": [11, 22], "exon": [1, 3, 9, 23], "exon1_numb": 1, "exon2_numb": 1, "exon9": 23, "exon_numb": 1, "exons_effect": 9, "expand_frame_repr": 16, "expect": [3, 11, 19, 22, 24], "expenditur": 19, "explain": 16, "explan": 24, "explicit": 4, "explicitli": [1, 9], "explor": 21, "exploratori": [20, 21], "express": [11, 19, 24], "extract": 11, "extraocular": 19, "extrem": [1, 22], "ey": [19, 24], "f": [14, 16], "face": 19, "fact": [22, 23], "factor": [19, 22], "fallback": [1, 11], "fals": [1, 3, 4, 8, 9, 11, 16, 22], "famili": 22, "fascia": 19, "fashion": 11, "fbn1": [3, 9], "fdr": 22, "fdr_bh": 22, "fdr_by": 22, "fdr_gb": 22, "fdr_tsbh": 22, "fdr_tsbky": 22, "featur": [1, 3, 9, 11, 14, 15, 18, 24], "feature1": 1, "feature2": 1, "feature_elong": 9, "feature_id": 3, "feature_trunc": 9, "feature_typ": [1, 3, 9], "feature_type1": 1, "feature_type2": 1, "featureinfo": [0, 9], "featuretyp": [0, 1, 3, 9], "fecal": 19, "fetal": 19, "fetch": 11, "fetch_for_gen": 11, "fetch_respons": 11, "fiber": 19, "fibrinolysi": 19, "fibroblast": 19, "field": [9, 17], "figur": [11, 12], "file": [11, 12, 21], "filter": [1, 12, 22], "filter_method": 1, "filter_method_nam": 1, "filter_nam": 1, "filter_results_map": 1, "filter_terms_to_test": 1, "find": [1, 9, 10, 11, 22], "find_coordin": 11, "finder": 11, "first": [1, 2, 3, 9, 11, 16, 21, 23], "fischer": 17, "fisher": [14, 22], "fisher_exact": [1, 24], "five": 24, "five_prime_utr_vari": [9, 11], "fix": 11, "flip": 10, "float": [1, 11, 24], "flow": 19, "fluid": 19, "focu": 14, "fold": 9, "folder": [11, 21], "follow": [3, 9, 15, 19, 22, 24], "foot": 19, "for_sampl": 9, "forearm": 19, "form": [19, 21], "format": [4, 9, 11, 12], "format_as_str": 12, "format_coordinates_for_vep_queri": 11, "formatt": 12, "found": [9, 11, 22, 24], "fpath_cache_dir": 11, "fraction": 1, "frame": [1, 16], "frameshift_vari": [9, 16], "frequenc": [1, 24], "friendli": [11, 12], "from": [1, 3, 4, 8, 9, 11, 12, 14, 15, 16, 19, 22, 23, 24], "from_map": 9, "from_pati": 9, "from_query_curi": 4, "from_term": 9, "fuction": 11, "full": 9, "function": [3, 4, 9, 11, 19, 21, 22, 24], "functional_annot": 11, "functionalannot": [0, 11], "functionalannotationawar": 9, "fwer": 22, "g": [1, 3, 4, 9, 11, 12, 14, 16, 19, 22, 23], "ga": 19, "ga4gh": 14, "ganglion": 19, "gastrin": 19, "gastrointestin": 19, "geleophys": 4, "gene": [3, 9, 11, 17, 23, 24], "gene_coordinate_servic": 11, "gene_id": 9, "gene_nam": 9, "gene_symbol": 9, "genecoordinateservic": [0, 11], "genephenocorr": 22, "gener": [1, 2, 4, 11, 12, 14, 20, 22], "genet": 14, "genit": 19, "genitourinari": 19, "geno_pred": 1, "genom": [9, 11, 14], "genome_build": 11, "genomebuild": [10, 11], "genomicinterpret": 11, "genomicregion": [9, 10], "genophenocorr": [13, 15, 16, 17, 21, 23, 24], "genophenocorr_cach": 1, "genophenocorrjsondecod": 8, "genophenocorrjsonencod": 8, "genotyp": [0, 1, 2, 9, 11, 14, 16, 19, 20, 21, 22, 23, 24], "genotype_for_sampl": 9, "genotypebooleanpred": [1, 2, 3], "genotypephenotypeanalysisresult": [0, 1], "genotypepolypred": [1, 2, 3], "genotypes_have_same_hpo_proport": 1, "get": [1, 2, 3, 4, 9, 11, 12, 16, 21, 22], "get_annot": 11, "get_categor": [2, 4], "get_categori": 2, "get_cds_region": 9, "get_coding_base_count": 9, "get_codon_count": 9, "get_excluded_count": 9, "get_features_variant_overlap": 9, "get_five_prime_utr": 9, "get_maximum_group_observed_hpo_frequ": 1, "get_number_of_observed_hpo_observ": 1, "get_patient_id": [9, 16], "get_quest": [2, 3, 4, 23], "get_respons": 11, "get_three_prime_utr": 9, "get_toy_cohort": [0, 7, 16], "get_variant_by_kei": [9, 23], "get_variant_descript": 12, "gg": [9, 11], "git": 15, "github": [15, 23], "give": 3, "given": [1, 2, 3, 4, 9, 11, 12], "gland": 19, "glial": 19, "global": 14, "globe": 19, "glucagon": 19, "glucocorticoid": 19, "glucos": 19, "glycolipid": 19, "glycoprotein": 19, "glycosaminoglycan": 19, "glycosyl": 19, "gonadotropin": 19, "gov": 17, "gpc": 1, "granulocytopoiet": 19, "graphic": [12, 24], "grch37": [10, 11], "grch38": [10, 11], "great": 22, "greater": 23, "group": [1, 2, 3, 16, 17, 22, 23, 24], "group1": 24, "group2": 24, "grouping_pred": 3, "groupingpred": [1, 2, 3], "growth": 19, "gt": 9, "gt_predic": 1, "guid": 14, "h": [16, 22], "ha": [1, 2, 3, 4, 9, 11, 19, 22, 23, 24], "hair": 19, "hand": 19, "happen": 1, "happi": 2, "has_error": 11, "has_errors_or_warn": 11, "has_sv_info": 9, "has_variant_coordin": 9, "has_warn": 11, "hashabl": 9, "have": [1, 4, 11, 16, 22, 24], "head": [16, 19], "health": 14, "hear": 24, "heart": [19, 24], "height": 19, "help": 11, "hematocrit": 19, "hemizyg": 9, "hemoglobin": 19, "henc": [11, 15], "hepat": 19, "hepatobiliari": 19, "here": [3, 11, 16, 18, 21, 22], "heterozyg": 9, "heterozygot": 17, "heurist": [1, 12, 19], "heuristic_sampl": 1, "heuristic_strategi": [1, 22], "heuristicmtcfilt": [0, 1], "heursit": 19, "hgnc": [9, 11], "hgv": [9, 11, 12], "hgvs_cdna": 9, "hgvs_coordinate_find": 11, "hgvsc": 9, "hgvsp": [9, 12], "hierarch": [11, 19], "high": [9, 20], "higher": 24, "highest": 9, "hip": 19, "histori": 22, "ho": 22, "hochberg": 22, "hold": 21, "holm": 22, "homeostasi": 19, "hommel": 22, "homo": 1, "homovanil": 19, "homozygot": 17, "homozygous_altern": 9, "homozygous_refer": 9, "hormon": 19, "how": [15, 16, 18, 21, 22, 24], "howev": [2, 4, 11, 15, 23], "hp": [1, 4, 16, 19, 22, 24], "hpo": [1, 4, 9, 11, 12, 14, 16, 20, 21, 22, 23], "hpo_mtc_report": 12, "hpo_term_frequency_filt": 1, "hpomtcfilt": 1, "hpomtcreport": [0, 1, 12], "hpotk": [1, 4, 9, 16, 21, 23], "html": 12, "http": [9, 11, 15, 16, 17, 19], "human": [2, 10, 11, 12, 14, 21], "humor": 19, "hundr": 22, "hypothalamu": 19, "hypothes": 22, "hypothesi": [1, 22, 24], "i": [0, 1, 3, 4, 8, 9, 11, 12, 14, 15, 16, 17, 19, 21, 22, 23, 24], "id": [1, 3, 4, 9, 11, 12, 19], "ident": [17, 22], "identifi": [1, 2, 3, 4, 9, 11, 19], "ii": 19, "immun": 19, "implement": [1, 3, 8, 11, 17, 24], "implementor": 9, "impli": [4, 22], "import": [3, 9, 16, 21, 22, 23, 24], "imposs": 1, "imprecis": 9, "imprecise_sv_functional_annot": 11, "imprecisesvfunctionalannot": [0, 11], "imprecisesvinfo": [0, 9, 11], "includ": [1, 4, 9, 11, 14], "include_computational_tx": 11, "include_patients_with_no_hpo": 9, "include_patients_with_no_vari": 9, "include_subsect": 11, "include_sv": 1, "incomplete_terminal_codon_vari": 9, "indel": 9, "indent": 8, "independ": 22, "index": [3, 9, 15, 19], "indic": 9, "indirect": 22, "individu": [4, 9, 14, 17, 20, 22, 24], "inflammatori": 19, "info": [1, 3, 9, 11, 21, 22, 23], "inform": [9, 12, 14, 22], "inframe_delet": 9, "inframe_insert": 9, "ingest": 14, "inher": 2, "inherit": [2, 22], "inhibin": 19, "initi": [4, 9, 15], "inner": 19, "input": [11, 14, 16, 20, 23], "insert": [9, 11], "insight": 19, "instal": [14, 21], "instanc": [1, 2, 3, 4, 9, 11, 16, 17, 21, 22, 23, 24], "instead": 22, "instruct": 14, "insulin": 19, "int": [1, 2, 3, 4, 9, 11, 12], "integ": [4, 9], "integr": 15, "integu": 19, "intend": 4, "intent": 1, "interest": [1, 3, 4, 9, 11, 16, 19, 22], "intergenic_vari": [9, 11], "internet": 15, "interpret": 11, "intersect": 10, "interspers": 9, "intestin": 19, "intracrani": 19, "intraocular": 19, "introduc": 23, "intron": 9, "intron_vari": [9, 11], "inv": 9, "invalid": 1, "invers": 9, "investig": [1, 2, 4], "invoc": 15, "invok": 22, "involv": 21, "ion": 19, "ipython": 12, "is_cod": 9, "is_large_imprecise_sv": 3, "is_observ": 9, "is_pref": 9, "is_pres": 9, "is_structur": 9, "is_structural_delet": 3, "is_structural_vari": 3, "islet": 19, "issu": [11, 14, 23], "item": [1, 11, 12, 16, 23], "iter": [1, 4, 8, 9, 11, 12, 22], "iterate_nod": 11, "its": [9, 11, 12], "j": [1, 16], "jax": [9, 19], "jinja2": 12, "join": [16, 21, 23], "joint": 19, "jordan": 24, "json": [8, 11, 16, 21], "jsondecod": 8, "jsonencod": 8, "jt": 17, "judgment": 19, "jupyt": [12, 14], "just": [1, 11, 12, 16], "k": 16, "kbg": 23, "keep": [4, 11, 16], "kei": [1, 3, 9, 12, 14], "kidnei": [4, 24], "kind": [11, 17], "know": [9, 11], "known": 9, "kwarg": 8, "l": 16, "label": [9, 11, 12, 19], "label_summari": 9, "labeling_method": 12, "lack": [4, 11], "lactat": 19, "langerhan": 19, "larg": [3, 9, 10, 11, 19], "last": [9, 23], "later": 22, "latest": 16, "lead": [1, 3, 9, 16, 19, 22, 23], "learn": [16, 18], "least": [1, 3, 11, 22, 23], "leav": 14, "legend": 12, "len": [9, 21, 23], "length": [3, 9, 23], "lenient": 11, "leptin": 19, "less": [1, 3, 24], "let": [8, 9, 16, 21, 22, 23], "leukocyt": 19, "level": [0, 1, 9, 11, 19, 20, 22, 23, 24], "li": 17, "librari": [0, 10, 15, 21, 24], "like": [1, 8, 14, 19, 22], "limb": 19, "limit": [1, 22], "line": [9, 15], "lineag": 19, "liposaccharid": 19, "list": [1, 4, 8, 9, 11, 16, 19, 23], "list_all_diseas": 9, "list_all_protein": [9, 16], "list_all_vari": [9, 16], "list_present_phenotyp": [9, 16], "liter": [3, 9, 11, 12], "literatur": 3, "liver": [4, 19], "ll": 21, "load": [8, 11, 16, 21, 23], "load_minimal_hpo": [21, 23], "load_minimal_ontologi": 16, "load_phenopacket": [0, 11], "load_phenopacket_fold": [0, 11, 21, 23], "local": [1, 11], "locat": [9, 14, 19, 21, 23], "locu": 9, "log": 1, "logic": 22, "look": 1, "loss": [9, 24], "lower": [19, 23, 24], "lymph": 19, "lymphat": 19, "lymphocyt": 19, "lysosom": 19, "m": [15, 16], "macroscop": 19, "made": [11, 22], "mai": [1, 4, 9, 11, 22, 24], "make": [1, 2, 3, 22], "malarkei": 11, "malform": 11, "mandatori": 9, "mane": [9, 23], "mani": [16, 18, 19, 22, 24], "manifest": 14, "mann": 4, "manner": 3, "mannwhitneyu": 24, "manual": 1, "map": [1, 9, 11, 12], "marfan": 4, "marker": [12, 19], "marker_count": 12, "marrow": 19, "mass": 19, "match": [1, 3, 4, 12], "matern": 17, "matplotlib": 12, "matter": 11, "matur": 19, "mature_mirna_vari": 9, "maximum": 1, "me": 9, "mean": [9, 19, 22, 24], "meaning": [2, 23], "measur": 19, "median": 24, "mediastinum": 19, "medic": [14, 19, 22], "medicin": 14, "meet": [3, 23], "megakaryocyt": 19, "member": [1, 9, 11, 16, 18, 21], "membran": 19, "mendelian": 14, "mental": 19, "mesenteri": 19, "messag": [8, 11], "meta_label": 9, "metabol": 19, "metabolit": 19, "metadata": [1, 11, 12], "metaphysi": 19, "method": [1, 3, 8, 9, 11, 21, 22], "mhc": 19, "mi": 1, "middl": 19, "might": 17, "min_annot": 1, "min_perc_patients_w_hpo": 1, "miner": 19, "minimalontologi": [1, 4, 11, 12], "minimalterm": 9, "minimum": 1, "mininum": 22, "miss": [1, 11, 23], "missens": [1, 9, 14, 16, 17, 22, 23, 24], "missense_in_tx": 16, "missense_or_nonsens": 23, "missense_vari": [1, 9, 16, 23], "missing_implies_exclud": 1, "missing_implies_phenotype_exclud": 4, "mitig": [1, 22], "mitochondrion": 19, "mnv": 9, "mode": [2, 22], "model": [0, 1, 2, 3, 4, 11, 16, 21, 23], "modifi": 22, "modul": [1, 5, 9, 10, 11], "moment": [1, 11], "monarch": 15, "mondo": 4, "more": [1, 3, 4, 9, 11, 15, 16, 19, 21, 22, 23], "morpholog": 19, "morphologi": [4, 14, 19, 22], "most": [1, 4, 10, 19, 22], "mostli": 9, "motif": [9, 14], "motil": 19, "motor": 19, "move": 16, "movement": 19, "mri": 19, "mtc": 1, "mtc_alpha": 1, "mtc_filter_report": 1, "mtc_method": 1, "mtc_name": 1, "mtc_strategi": [0, 1, 22], "much": [19, 22], "mucociliari": 19, "mucosa": 19, "mucu": 19, "multi": [9, 23], "multipl": [1, 4, 19, 20], "multipli": 22, "muscl": 19, "musculatur": [1, 19], "musculoskelet": 19, "must": [2, 3, 4, 11, 15, 23], "mutat": 24, "mutlipl": 22, "myelin": 19, "myeloid": 19, "n": [1, 3, 9, 11, 16, 19], "n_categor": 2, "n_terms_test": 1, "n_usabl": 1, "nacgtacgtac": 9, "nail": 19, "name": [1, 2, 9, 11], "namedtupl": 12, "nasal": 19, "natur": 9, "ncbi": 17, "ndarrai": 12, "neck": 19, "need": [3, 11, 16, 21], "neg": [3, 9], "neither": [3, 22], "neoplasm": 19, "neopterin": 19, "nerv": 19, "nervou": [1, 19, 22], "net": 9, "neuron": 19, "new": 12, "next": [16, 21], "nih": 17, "nippl": 19, "nlm": 17, "nm_002834": 11, "nm_005912": 11, "nm_013275": 23, "nm_123": 3, "nm_1234": [1, 16], "nm_123456": [1, 9], "nmd_transcript_vari": 9, "no_cal": 9, "node": [11, 19], "nomin": [1, 22], "non": [3, 9, 11, 22, 24], "non_coding_transcript_exon_vari": [9, 11], "non_coding_transcript_vari": [9, 11], "noncoding_effect": 11, "none": [1, 2, 3, 4, 8, 9, 11, 12, 22, 23], "nons": 17, "nonsens": [1, 17, 22, 23, 24], "nonsense_and_exon9": 23, "nor": 22, "normal": 14, "notat": [1, 9], "note": [1, 2, 4, 9, 24], "notebook": [12, 14, 16], "notepad": [0, 11], "notepadtre": [0, 11], "noth": 23, "novel": 9, "now": [16, 22], "np_001027558": 11, "np_037407": 11, "np_09876": 16, "nucleobas": 19, "nucleotid": 9, "nucleu": 22, "null": 22, "number": [1, 2, 3, 4, 9, 19, 22, 24], "numer": 2, "o": [8, 16, 21, 23], "obj": 8, "object": [1, 2, 3, 4, 8, 9, 11, 12, 22], "object_hook": 8, "obo": 16, "obolibrari": 16, "observ": [1, 9, 11, 14, 22, 24], "observablefeatur": 9, "obtain": [11, 22], "occipitofrontali": 19, "occur": 24, "occurr": 22, "oddsr": 1, "oddsratio": 24, "odontoid": 19, "off": 23, "offer": [22, 24], "offici": 9, "offlin": 11, "oldest": 22, "omim": 4, "omit": [1, 23], "onc": 1, "one": [1, 2, 3, 4, 9, 11, 15, 17, 19, 22, 23, 24], "one_genotype_has_zero_hpo_observ": 1, "ones": 11, "onli": [1, 2, 4, 9, 11, 17, 22, 23], "onlin": 15, "only_hgv": 12, "onset": 19, "onto": 12, "ontologi": [9, 14, 16, 21], "oocyt": 19, "open": 23, "oper": [3, 23], "oppos": 9, "optim": 11, "option": [1, 2, 4, 9, 11, 15, 21, 22], "order": [9, 16], "org": [9, 11, 16, 19], "organ": [9, 11, 19], "orgogozo": 14, "osmol": 19, "osteocalcin": 19, "other": [1, 3, 9, 11, 14, 16, 17, 22], "otherwis": [1, 3, 4, 9], "our": [1, 12, 19, 22, 23, 24], "out": [1, 11, 21, 22], "outer": 19, "over": [9, 11], "overal": 22, "overlap": [3, 9, 10, 12, 23], "overlapping_exon": 9, "overload": 23, "overview": [20, 24], "p": [1, 4, 16, 22, 23, 24], "p13": [10, 11], "p_valu": 24, "packag": [13, 14, 15, 16, 22], "pair": [1, 3, 9], "pancrea": 19, "pancreat": 19, "panda": [1, 16], "param": [1, 3, 9, 11, 12], "paramet": [1, 3, 4, 8, 9, 11, 12], "parameter": [1, 12], "parametr": 24, "parathyroid": 19, "parent": [11, 22], "pars": 11, "parse_multipl": 11, "parse_respons": 11, "parse_uniprot_json": 11, "part": [9, 12], "partit": 16, "pass": [1, 3, 8, 22], "past": 22, "patern": 17, "path": [2, 11, 16, 21, 23], "pathogen": 23, "patient": [0, 1, 2, 3, 4, 9, 11, 12, 14, 16, 21, 22, 23], "patient_cr": 11, "patient_id": 9, "patientcategori": [1, 2, 4, 16], "patientcr": [0, 11], "pattern": [11, 19], "payload": 11, "pd": [1, 16], "pehnotyp": 1, "peptid": 19, "per": 1, "percent": [1, 16], "percentag": 1, "perform": [1, 11, 14, 16, 17, 21, 22, 24], "perifollicular": 19, "peripher": 19, "peritoneum": 19, "peroxisom": 19, "ph": 19, "phagocytosi": 19, "pheno_pred": 1, "phenopacket": [11, 14, 16, 23], "phenopacket_path": 11, "phenopacketpatientcr": [0, 11], "phenopacketvariantcoordinatefind": [0, 11], "phenotyp": [0, 1, 9, 11, 14, 16, 18, 19, 20, 21, 22, 24], "phenotype_categori": 1, "phenotype_cr": 11, "phenotypecategor": 4, "phenotypecr": [0, 11], "phenotypepolypred": [1, 4], "phosphat": 19, "physiolog": 19, "physiologi": [1, 14, 19], "pickl": 11, "pineal": 19, "pip": 15, "pituitari": 19, "place": 14, "placent": 19, "placenta": 19, "platelet": 19, "platysma": 19, "pleas": 23, "plot": [12, 18], "plt": 12, "plu": 3, "point": [1, 9, 11, 24], "polar": 22, "polici": 11, "polypred": [1, 2, 4, 23], "port": 10, "posit": [1, 9, 11, 22], "possibl": [1, 2, 17, 22, 23], "posterior": [19, 22], "potenti": 24, "power": 1, "pp_directori": 11, "practic": 17, "precis": 14, "precursor": 19, "predic": [1, 16, 20], "predict": [3, 9, 23], "predictor": [11, 21], "prefer": 9, "preimplant": 19, "prematur": [14, 23], "prenat": 19, "prepar": [2, 3, 4, 11, 21], "prepare_notepad": 11, "preprocess": [0, 21, 23], "presenc": [1, 3, 9, 16], "present": [1, 3, 4, 9, 16], "present_diseas": 9, "present_phenotyp": 9, "pressur": 19, "pretti": 12, "prevent": [1, 23], "previou": [1, 22], "previous": 23, "primari": 19, "principl": 17, "print": 9, "probabl": [22, 24], "problem": 22, "proce": 11, "procedur": [12, 19], "process": [11, 12], "process_respons": 11, "produc": [2, 4], "product": 12, "programmatic_access": 11, "project": 9, "prolifer": 19, "prop": 9, "propagatingphenotypepred": 4, "properti": [1, 2, 4, 9, 11, 12], "proport": [1, 22], "protcachingmetadataservic": [0, 11], "protein": [1, 3, 9, 11, 12, 14, 19, 21], "protein_altering_vari": 9, "protein_cach": 1, "protein_effect_coordin": 9, "protein_effect_end": 9, "protein_effect_loc": 9, "protein_effect_start": 9, "protein_featur": [3, 9], "protein_feature_end": 12, "protein_feature_nam": 12, "protein_feature_start": 12, "protein_feature_typ": [3, 12], "protein_id": [9, 11, 12], "protein_length": [9, 12], "protein_meta": 12, "protein_metadata": 12, "protein_metadata_servic": 3, "protein_region": 1, "protein_servic": 1, "protein_sourc": 1, "proteinannotationcach": [0, 11], "proteinfeatur": [0, 1, 9], "proteinmetadata": [0, 9, 11, 12], "proteinmetadataservic": [0, 1, 3, 11], "proteinpred": [3, 23], "proteinview": [0, 12], "proteinvisu": [0, 12], "proteinvisualiz": [0, 12], "provid": [1, 3, 4, 9, 10, 11, 12, 14, 21, 22, 24], "prune": 11, "publish": 15, "pubm": 17, "pulmonari": 19, "pupillari": 19, "purin": 19, "purl": 16, "purpos": 21, "put": 24, "putamen": 22, "pval": 1, "pval_correct": [1, 22], "pvalu": 24, "pvi": 12, "pypi": 15, "pyrimidin": 19, "pytest": 15, "python": [8, 14, 15, 22, 23, 24], "python3": 15, "q": [16, 21], "qualnam": [1, 9, 11], "queri": [3, 4, 9, 11], "question": [2, 3, 4, 14, 24], "r": [16, 24], "rais": [8, 11], "ran": 16, "random_se": 12, "rang": [1, 4, 24], "rank": 24, "rare": [0, 1, 22], "rate": [19, 22], "rather": 19, "ratio": 19, "raw": 21, "re": 16, "reach": 1, "read": 14, "readabl": [2, 9, 12], "realiz": 22, "reason": [1, 19], "recess": [3, 20], "recessive_pred": 3, "recessivegroupingpred": 3, "recommend": [3, 14], "record": 11, "red": 19, "redox": 19, "reduc": [1, 22], "redund": 22, "ref": 9, "refactor": 9, "refer": [3, 9, 14, 19], "reflex": 19, "refract": 19, "refseq": [9, 11], "regard": [9, 12], "regardless": 9, "region": [1, 3, 9, 10, 12], "region1": 1, "region2": 1, "regul": 19, "regulatory_region_abl": 9, "regulatory_region_amplif": 9, "regulatory_region_vari": 9, "rel": [1, 3], "relat": [11, 22], "relationship": [14, 24], "releas": [19, 21, 23], "relev": [1, 9], "remot": 11, "remov": [1, 3, 11, 19, 23], "render": 12, "renin": 19, "repair": 19, "repeat": 9, "replac": 23, "report": [1, 11, 14, 16], "repositori": [15, 21], "repres": [1, 2, 4, 9, 11, 12, 14], "represent": [9, 11], "reproduct": 19, "request": 11, "requir": [11, 15, 21], "requisit": 21, "rere": 24, "residu": 9, "resourc": [1, 3, 11], "respect": 9, "respir": 19, "respiratori": 19, "respons": [11, 19, 21], "rest": [1, 11, 19], "restrict": 17, "result": [1, 9, 11, 14, 16, 19, 22, 23, 24], "retain": [1, 22], "reticulocyt": 19, "retriev": [1, 11], "return": [1, 2, 3, 4, 8, 9, 11, 12, 22, 23], "reveal": 1, "right": [11, 12], "root": [11, 19, 22], "row": [1, 16], "rule": 1, "run": [1, 11, 16], "runner": 11, "runonlin": 15, "rust": 9, "sai": 22, "salivari": 19, "same": [1, 19, 22, 24], "sampl": 9, "sample_id": 9, "samplelabel": [0, 9], "sampler": [1, 19], "sanit": 11, "sanitationresult": 11, "saniti": 11, "sapien": 1, "satisfi": 11, "save": 21, "scalp": 19, "scenario": [2, 23], "schema": [11, 12, 21], "scienc": 1, "scipi": [1, 24], "sclera": 19, "search": [9, 11, 19], "sebac": 19, "second": [1, 2, 3, 11], "secondari": 9, "secret": 19, "section": [1, 11, 16, 20, 21, 23], "sediment": 19, "see": [1, 3, 9, 16, 17, 19, 21, 22, 23], "seem": 23, "seen": [1, 22, 24], "segment": [9, 19], "segreg": 19, "select": [1, 3], "self": [4, 8], "sens": [1, 2, 19, 22], "sensat": 19, "sensori": 19, "sensorineur": 24, "separ": 8, "sequenc": [1, 2, 4, 9, 11, 12], "sequence_vari": 9, "sequenceontologi": 9, "seri": 1, "serializ": 8, "seriou": 11, "serum": 19, "serv": 2, "servic": [1, 11], "set": [1, 4, 9, 22, 24], "set_opt": 16, "setter": 1, "sever": [2, 9, 11, 23], "sex": 19, "sh": 22, "sharpei": 19, "shelf": 23, "ship": 16, "short": 9, "shortcut": 9, "shorter": 3, "should": [1, 4, 9, 11, 12], "show": [12, 16, 19, 22, 24], "shown": 12, "sidak": 22, "side": [1, 24], "sign": [16, 23], "signal": [19, 22], "signific": [1, 9, 19, 22, 24], "significantli": 24, "sime": 22, "similar": [1, 9, 24], "simpl": [3, 11, 15, 16, 17], "simple_cohort": [21, 23], "simple_cohort_path": 21, "simplest": 22, "simplifi": 1, "sinc": [1, 3, 9, 24], "sine": 9, "singl": [9, 11], "situ": 19, "size": [9, 19], "skelet": 19, "skin": [4, 19], "skinfold": 19, "skip": [1, 19, 22], "skipkei": 8, "skipped_terms_dict": 1, "skull": 19, "small": 22, "smell": 19, "snv": 9, "so": [1, 3, 9, 22, 23], "softwar": 24, "solut": [11, 21], "somat": 19, "some": [11, 14, 15, 16, 22], "some_cell_has_greater_than_one_count": 1, "someth": 11, "soon": 16, "sort": [11, 16], "sort_kei": 8, "sought": 4, "sound": 19, "sourc": [1, 2, 3, 4, 7, 8, 9, 11, 12, 15], "span": [1, 9, 19], "spastic": 16, "specif": [1, 3, 9, 11, 14, 19, 22, 24], "specifi": 1, "specified_term_set": [1, 22], "specifiedtermsmtcfilt": [0, 1], "specify_term": [1, 22], "specify_terms_strategi": [1, 22], "spinal": 19, "spleen": 19, "splice_acceptor_vari": [9, 11], "splice_donor_5th_base_vari": [9, 11], "splice_donor_region_vari": 9, "splice_donor_vari": [9, 11], "splice_polypyrimidine_tract_vari": [9, 11], "splice_region_vari": 9, "spuriou": 22, "stabil": 19, "stalk": 19, "standalon": 12, "standard": 21, "start": [1, 3, 9, 11, 12, 19, 21], "start_lost": 9, "start_retained_vari": 9, "stat": [1, 24], "state": 9, "static": [1, 2, 3, 4, 8, 9, 11, 12], "statist": [1, 14, 19, 20, 22], "statsmodel": 22, "statsview": [0, 12], "statu": 11, "step": [15, 21], "stimul": 19, "stool": 19, "stop": [19, 23], "stop_gain": [9, 23], "stop_lost": 9, "stop_retained_vari": 9, "storag": 11, "store": [1, 11, 16, 21, 23], "store_annot": 11, "str": [1, 2, 3, 4, 9, 11, 12], "strabismu": 24, "strand": [3, 10], "strategi": [1, 12, 19], "stress": 19, "strict": 11, "string": [1, 9, 11, 12, 19], "stromal": 19, "strongli": 3, "structur": [1, 3, 9, 11, 19, 23, 24], "structural_typ": [3, 9, 23], "structural_vari": 3, "structuraltypepred": 3, "studi": 14, "stump": 19, "sub": [9, 11], "subclass": [1, 3, 8], "subgroup": 9, "subject": [9, 23], "submodul": 0, "subpackag": 13, "subsect": [9, 11], "subunit": 19, "sum": 24, "summar": [1, 9, 11, 12, 16], "summari": [1, 12], "summary_df": 16, "suox": 17, "super": 8, "support": [3, 8, 9, 14, 17], "suppos": 8, "surf1": 9, "surfac": 19, "surviv": 22, "sv": [3, 9, 11], "sv_info": 9, "svart": 10, "svlen": 9, "svtype": 9, "sweat": 19, "switch": 15, "symbol": [1, 3, 9], "symptom": [16, 19, 23], "synapt": 19, "syndrom": [4, 23], "synonym": 23, "synonymous_vari": 9, "system": [1, 9, 19, 22], "t": [9, 11, 16], "tabl": [1, 12, 16, 19, 22, 24], "tail": 24, "take": [11, 22], "tandem": 9, "target": [3, 4, 12], "tast": 19, "taxon": 1, "technic": 14, "temperatur": 19, "templat": 12, "ten": 22, "term": [1, 4, 9, 11, 12, 14, 16, 20, 21, 23, 24], "term_count": 1, "term_id": 9, "termid": [1, 3, 4, 9], "termin": [9, 14, 23], "terms_to_test": [1, 22], "test": [1, 2, 3, 4, 9, 14, 16, 17, 19, 20, 23], "tf_binding_site_vari": 9, "tfbs_ablat": 9, "tfbs_amplif": 9, "th": 1, "than": [1, 3, 9, 19, 22, 24], "thei": [10, 14, 19, 22, 24], "them": [1, 11], "therebi": 1, "therefor": [1, 19, 22, 23], "thi": [1, 3, 4, 8, 9, 11, 14, 15, 16, 17, 19, 21, 22, 23, 24], "thick": 19, "thing": 16, "think": 22, "thorac": 19, "those": 1, "three": [1, 9, 17], "three_prime_utr_vari": [9, 11], "threshold": [1, 3, 22], "threshold_hpo_observed_frequ": [1, 22], "thrombocyt": 19, "thrombosi": 19, "through": [1, 11], "thu": [1, 9, 22], "thumb": 1, "thymu": 19, "thyroid": 19, "time": [4, 9, 11, 15, 22], "timeout": 11, "tissu": 19, "to_str": 9, "todo": [1, 9, 12, 16, 18, 20, 21], "togeth": 21, "toi": 16, "toler": 1, "tonsil": 19, "too": 19, "toolkit": [11, 21, 23], "top": [1, 4, 9, 11, 19], "top_phenotype_count": 12, "top_variant_count": 12, "total": [22, 24], "total_patient_count": 9, "track": [4, 11], "tracker": [14, 23], "tract": 19, "trans_id": 9, "transcript": [1, 3, 9, 11, 12, 16, 23], "transcript_abl": 9, "transcript_amplif": 9, "transcript_coordin": 12, "transcript_id": [1, 9, 12], "transcriptannot": [0, 9, 11], "transcriptcoordin": [0, 9, 11, 12], "transcriptcoordinateservic": [0, 11], "transcriptinfoawar": [0, 9, 11], "transform": 11, "translat": 9, "transloc": 3, "transmiss": 19, "transpos": 10, "travers": 11, "tree": 11, "tri": 11, "triphosph": 19, "true": [1, 3, 4, 8, 9, 11, 12, 22, 23], "try": [8, 11], "tupl": [1, 9, 11], "turn": 11, "tutori": 14, "twice": 22, "two": [1, 2, 4, 10, 16, 17, 22, 24], "two_by_two_t": 1, "tx": [11, 12, 23], "tx_annot": 9, "tx_coordin": 12, "tx_dict": 16, "tx_id": [1, 3, 9, 11, 12, 16, 23], "type": [1, 3, 4, 8, 9, 11, 12, 14, 22, 23], "typeerror": 8, "tyr1715ter": 23, "tyrosin": 23, "u": [4, 16, 23], "umbil": 19, "umbilicu": 19, "unavail": 9, "under": [1, 11, 22], "unfix": 11, "uniform": 3, "uninterest": 19, "uniprot": [1, 11, 12, 21], "uniprotproteinmetadataservic": [0, 11], "uniqu": [1, 2, 9], "unit": 15, "unknown": [3, 9], "unless": [4, 11, 22], "unlik": [1, 11, 19, 22], "unnecessari": 19, "unrunn": 11, "unstabl": 16, "untransl": 9, "up": 22, "updat": 16, "upon": 1, "upper": 19, "upstream_gene_vari": [9, 11], "urat": 19, "uri": 9, "urin": 19, "urinari": 19, "urobilinogen": 19, "us": [1, 2, 3, 4, 8, 9, 10, 11, 12, 14, 16, 21, 22, 23, 24], "usabl": 1, "user": [9, 12, 14, 19], "usual": [1, 3, 9, 19, 24], "util": [2, 3], "utr": 9, "uvea": 19, "v": [1, 14, 16, 22], "v2024": [21, 23], "vagin": 19, "valid": [1, 11, 12, 22], "validation_polici": 11, "validation_runn": 11, "validationrunn": 11, "valu": [9, 11, 12, 16, 22, 24], "valueerror": 11, "valv": 19, "varcachingfunctionalannot": [0, 11], "vari": [3, 24], "variabl": [9, 16, 24], "variant": [0, 1, 2, 3, 9, 11, 12, 14, 16, 18, 21, 22, 23, 24], "variant_class": [3, 9, 23], "variant_coordin": [9, 11], "variant_effect": [3, 5, 9, 12, 16, 23], "variant_effect_count_by_tx": [9, 16], "variant_fallback": 11, "variant_info": 9, "variant_kei": [1, 3, 9, 11], "variant_key1": 1, "variant_key2": 1, "variant_key_of_interest": 23, "variant_loc": 12, "variant_locations_counted_absolut": 12, "variant_pred": 3, "variantannotationcach": [0, 11], "variantclass": [0, 3, 9, 23], "variantcoordin": [0, 1, 3, 9, 11], "variantcoordinatefind": [0, 11], "varianteffect": [0, 1, 3, 9, 11, 12, 16, 23], "varianteffectpred": 1, "variantformatt": [0, 12], "variantinfo": [0, 9], "variantinfoawar": [0, 9], "variantpred": [1, 3, 16, 23], "varianttranscriptproteinartist": 12, "varianttranscriptvisu": [0, 12], "variou": 1, "vascular": 19, "vasculatur": 19, "vc": 11, "vcf": [1, 9], "vcv001029215": 23, "ventricular": 19, "vep": [9, 11], "vepfunctionalannot": [0, 11], "verbos": 2, "veri": 22, "verifi": 9, "version": 1, "vessel": 19, "via": [8, 11, 12], "view": [0, 14, 16], "viewabl": 12, "virtual": 15, "vision": 19, "visit": 11, "visitor": 11, "visual": [12, 19], "vitamin": 19, "voic": 19, "volum": 19, "vvhgvsvariantcoordinatefind": [0, 11], "vvmulticoordinateservic": [0, 11], "w": 16, "wa": [1, 4, 9, 11, 12, 22], "wai": [16, 21, 24], "waist": 19, "wall": 19, "want": [1, 4, 16, 22, 24], "warn": [1, 11], "warning_count": 11, "we": [1, 3, 4, 9, 11, 12, 14, 15, 16, 18, 19, 21, 22, 23, 24], "weight": 19, "well": [1, 3, 4, 9, 16], "were": [1, 11, 23, 24], "what": [11, 16, 18, 24], "whatev": 1, "when": [4, 11, 22], "where": [1, 9, 11], "whether": [4, 9, 14, 24], "which": [1, 2, 4, 9, 16, 21, 23], "while": 17, "whitnei": 4, "who": [2, 17], "why": 1, "wilcoxon": 24, "wise": 22, "with_cache_fold": 11, "within": [9, 11, 24], "without": [1, 3], "word": [14, 19], "work": [1, 11, 21], "workflow": 16, "worth": 1, "would": [4, 14, 17, 22, 24], "wrap": 1, "write": [11, 12, 18, 20], "www": [9, 11], "x": [16, 24], "x_12345_12345_c_g": [1, 3, 9], "xm_": 11, "y": [16, 24], "ye": [1, 2, 4, 16], "yield": 22, "you": [8, 9, 16, 22], "your": [15, 16], "z": 16, "zero": [1, 9]}, "titles": ["genophenocorr package", "genophenocorr.analysis package", "genophenocorr.analysis.predicate package", "genophenocorr.analysis.predicate.genotype package", "genophenocorr.analysis.predicate.phenotype package", "genophenocorr.constants package", "genophenocorr.constants.variant_effects module", "genophenocorr.data package", "genophenocorr.io module", "genophenocorr.model package", "genophenocorr.model.genome package", "genophenocorr.preprocessing package", "genophenocorr.view package", "API reference", "genophenocorr", "Installation", "Tutorial", "Genotype-Phenotype Correlations in Autosomal Recessive Diseases", "Cohort exploratory analysis", "General HPO terms", "User guide", "Input data", "Multiple-testing correction", "Predicates", "Statistical tests"], "titleterms": {"analysi": [1, 2, 3, 4, 16, 18], "api": 13, "autosom": 17, "background": 22, "choos": 22, "cohort": [18, 21], "compound": 23, "constant": [5, 6], "content": 20, "correct": 22, "correl": 17, "creat": 21, "data": [7, 21], "default": 1, "diseas": 17, "exact": 24, "exploratori": 18, "feedback": 14, "fet": 24, "fisher": 24, "from": 21, "ga4gh": 21, "gener": 19, "genom": 10, "genophenocorr": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 14, 22], "genotyp": [3, 17], "guid": 20, "heurist": 22, "hpo": 19, "implement": 22, "input": 21, "instal": 15, "io": 8, "latest": 15, "literatur": 14, "mann": 24, "model": [9, 10], "modul": [6, 8], "mtc": 22, "multipl": 22, "other": 21, "packag": [0, 1, 2, 3, 4, 5, 7, 9, 10, 11, 12], "phenopacket": 21, "phenotyp": [4, 17], "predic": [2, 3, 4, 23], "prepar": 16, "preprocess": 11, "procedur": 22, "recess": 17, "refer": 13, "releas": 15, "run": 15, "sampl": 16, "sampler": 22, "set": 16, "simpl": 23, "specifi": 22, "stabl": 15, "statist": 24, "strategi": 22, "submodul": 5, "subpackag": [0, 1], "term": [19, 22], "test": [15, 22, 24], "tutori": [16, 17], "u": 24, "up": 16, "user": 20, "valu": 1, "variant_effect": 6, "view": 12, "which": 22, "whitnei": 24}})