Search.setIndex({"alltitles": {"API reference": [[17, null]], "Alternative phenopacket sources": [[27, "alternative-phenopacket-sources"]], "Analysis": [[31, "analysis"], [31, "id4"]], "Background": [[28, "background"]], "Building blocks": [[30, "building-blocks"]], "Change length of an allele": [[25, "change-length-of-an-allele"]], "Cohort exploratory analysis": [[23, null]], "Compare genotype and phenotype groups": [[31, "compare-genotype-and-phenotype-groups"]], "Complex conditions": [[30, "complex-conditions"]], "Configure analysis": [[31, "configure-analysis"], [31, "id3"]], "Contents:": [[26, null], [29, null]], "Create a cohort from GA4GH phenopackets": [[27, "create-a-cohort-from-ga4gh-phenopackets"]], "Create cohort": [[31, "create-cohort"]], "Create cohort creator": [[27, "create-cohort-creator"]], "De Vries Score": [[22, null]], "Developmental delay": [[22, "developmental-delay"]], "Example": [[30, "example"], [30, "id4"], [30, "id6"], [30, "id8"], [31, "example"], [31, "id2"]], "Explore cohort": [[20, "explore-cohort"]], "Facial dysmorphic features": [[22, "facial-dysmorphic-features"]], "Feedback": [[18, "feedback"]], "Filtering predicate": [[30, "filtering-predicate"]], "Final score": [[22, "final-score"]], "Fisher exact test (FET)": [[31, "fisher-exact-test-fet"]], "GPSEA": [[18, null]], "Gallery": [[30, "gallery"]], "General HPO terms": [[24, null]], "Genotype phenotype associations": [[31, "genotype-phenotype-associations"]], "Genotype predicates": [[30, "genotype-predicates"]], "Genotype-Phenotype Correlations in Autosomal Recessive Diseases": [[21, null]], "Glossary": [[25, null]], "Groups predicate": [[30, "groups-predicate"]], "HPO MTC filter strategy": [[28, "hpo-mtc-filter-strategy"]], "Implementation in GPSEA": [[28, "implementation-in-gpsea"]], "Input data": [[27, null]], "Installation": [[19, null]], "Inverting conditions": [[30, "inverting-conditions"]], "Latest release": [[19, "latest-release"]], "Length of the reference allele": [[25, "length-of-the-reference-allele"]], "Literature": [[18, "literature"]], "Load HPO": [[20, "load-hpo"]], "Load phenopackets": [[27, "load-phenopackets"]], "MTC filters: Choosing which terms to test": [[28, "mtc-filters-choosing-which-terms-to-test"]], "Mann-Whitney U Test": [[31, "mann-whitney-u-test"]], "Mode of inheritance predicate": [[30, "mode-of-inheritance-predicate"]], "Multiple-testing correction": [[28, null]], "Multiple-testing correction procedures": [[28, "multiple-testing-correction-procedures"]], "Need more?": [[30, "need-more"]], "Non-facial dysmorphism and congenital abnormalities": [[22, "non-facial-dysmorphism-and-congenital-abnormalities"]], "Phenotype Predicates": [[29, null]], "Phenotype predicates": [[30, "phenotype-predicates"]], "Phenotype score": [[31, "phenotype-score"]], "Postnatal growth abnormalities": [[22, "postnatal-growth-abnormalities"]], "Predicates": [[30, null]], "Predicates for all cohort phenotypes": [[30, "predicates-for-all-cohort-phenotypes"]], "Prenatal-onset growth retardation": [[22, "prenatal-onset-growth-retardation"]], "Prepare cohort": [[20, "prepare-cohort"]], "Prepare genotype and phenotype predicates": [[20, "prepare-genotype-and-phenotype-predicates"]], "Propagating phenotype predicate": [[30, "propagating-phenotype-predicate"]], "Run tests": [[19, "run-tests"]], "Specify terms strategy": [[28, "specify-terms-strategy"]], "Stable release": [[19, "stable-release"]], "Statistical tests": [[31, null]], "Submodules": [[0, "submodules"]], "Subpackages": [[0, "subpackages"], [1, "subpackages"], [3, "subpackages"], [5, "subpackages"], [8, "subpackages"], [13, "subpackages"]], "Supports shape": [[4, "supports-shape"]], "TBX5 frameshift vs missense": [[20, "id5"]], "TBX5 frameshift vs rest": [[31, "id5"]], "Test all terms": [[28, "test-all-terms"]], "The analysis": [[20, "the-analysis"]], "True path rule": [[25, "true-path-rule"]], "Tutorial": [[20, null], [21, "tutorial"]], "User guide": [[26, null]], "Using the De Vries Scorer in code": [[22, "using-the-de-vries-scorer-in-code"]], "Variant predicates": [[30, "variant-predicates"]], "gpsea package": [[0, null]], "gpsea.analysis package": [[1, null]], "gpsea.analysis.mtc_filter package": [[2, null]], "gpsea.analysis.pcats package": [[3, null]], "gpsea.analysis.pcats.stats package": [[4, null]], "gpsea.analysis.predicate package": [[5, null]], "gpsea.analysis.predicate.genotype package": [[6, null]], "gpsea.analysis.predicate.phenotype package": [[7, null]], "gpsea.analysis.pscore package": [[8, null]], "gpsea.analysis.pscore.stats package": [[9, null]], "gpsea.config module": [[10, null]], "gpsea.data package": [[11, null]], "gpsea.io module": [[12, null]], "gpsea.model package": [[13, null]], "gpsea.model.genome package": [[14, null]], "gpsea.preprocessing package": [[15, null]], "gpsea.view package": [[16, null]]}, "docnames": ["apidocs/gpsea", "apidocs/gpsea.analysis", "apidocs/gpsea.analysis.mtc_filter", "apidocs/gpsea.analysis.pcats", "apidocs/gpsea.analysis.pcats.stats", "apidocs/gpsea.analysis.predicate", "apidocs/gpsea.analysis.predicate.genotype", "apidocs/gpsea.analysis.predicate.phenotype", "apidocs/gpsea.analysis.pscore", "apidocs/gpsea.analysis.pscore.stats", "apidocs/gpsea.config", "apidocs/gpsea.data", "apidocs/gpsea.io", "apidocs/gpsea.model", "apidocs/gpsea.model.genome", "apidocs/gpsea.preprocessing", "apidocs/gpsea.view", "apidocs/modules", "index", "installation", "tutorial", "user-guide/autosomal_recessive", "user-guide/devries", "user-guide/exploratory", "user-guide/general_hpo_terms", "user-guide/glossary", "user-guide/index", "user-guide/input-data", "user-guide/mtc", "user-guide/phenotype_predicates", "user-guide/predicates", "user-guide/stats"], "envversion": {"sphinx": 62, "sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1}, "filenames": ["apidocs/gpsea.rst", "apidocs/gpsea.analysis.rst", "apidocs/gpsea.analysis.mtc_filter.rst", "apidocs/gpsea.analysis.pcats.rst", "apidocs/gpsea.analysis.pcats.stats.rst", "apidocs/gpsea.analysis.predicate.rst", "apidocs/gpsea.analysis.predicate.genotype.rst", "apidocs/gpsea.analysis.predicate.phenotype.rst", "apidocs/gpsea.analysis.pscore.rst", "apidocs/gpsea.analysis.pscore.stats.rst", "apidocs/gpsea.config.rst", "apidocs/gpsea.data.rst", "apidocs/gpsea.io.rst", "apidocs/gpsea.model.rst", "apidocs/gpsea.model.genome.rst", "apidocs/gpsea.preprocessing.rst", "apidocs/gpsea.view.rst", "apidocs/modules.rst", "index.rst", "installation.rst", "tutorial.rst", "user-guide/autosomal_recessive.rst", "user-guide/devries.rst", "user-guide/exploratory.rst", "user-guide/general_hpo_terms.rst", "user-guide/glossary.rst", "user-guide/index.rst", "user-guide/input-data.rst", "user-guide/mtc.rst", "user-guide/phenotype_predicates.rst", "user-guide/predicates.rst", "user-guide/stats.rst"], "indexentries": {"add_error() (gpsea.preprocessing.notepad method)": [[15, "gpsea.preprocessing.Notepad.add_error", false]], "add_issue() (gpsea.preprocessing.notepad method)": [[15, "gpsea.preprocessing.Notepad.add_issue", false]], "add_subsection() (gpsea.preprocessing.notepad method)": [[15, "gpsea.preprocessing.Notepad.add_subsection", false]], "add_subsection() (gpsea.preprocessing.notepadtree method)": [[15, "gpsea.preprocessing.NotepadTree.add_subsection", false]], "add_warning() (gpsea.preprocessing.notepad method)": [[15, "gpsea.preprocessing.Notepad.add_warning", false]], "all() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.all", false]], "all_counts (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.all_counts", false]], "all_diseases() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.all_diseases", false]], "all_patients (gpsea.model.cohort property)": [[13, "gpsea.model.Cohort.all_patients", false]], "all_phenotypes() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.all_phenotypes", false]], "all_transcript_ids (gpsea.model.cohort property)": [[13, "gpsea.model.Cohort.all_transcript_ids", false]], "all_variants() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.all_variants", false]], "allelecounter (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.AlleleCounter", false]], "alt (gpsea.model.variantcoordinates property)": [[13, "gpsea.model.VariantCoordinates.alt", false]], "annotate() (gpsea.preprocessing.defaultimprecisesvfunctionalannotator method)": [[15, "gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.functionalannotator method)": [[15, "gpsea.preprocessing.FunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.imprecisesvfunctionalannotator method)": [[15, "gpsea.preprocessing.ImpreciseSvFunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.protcachingmetadataservice method)": [[15, "gpsea.preprocessing.ProtCachingMetadataService.annotate", false]], "annotate() (gpsea.preprocessing.proteinmetadataservice method)": [[15, "gpsea.preprocessing.ProteinMetadataService.annotate", false]], "annotate() (gpsea.preprocessing.uniprotproteinmetadataservice method)": [[15, "gpsea.preprocessing.UniprotProteinMetadataService.annotate", false]], "annotate() (gpsea.preprocessing.varcachingfunctionalannotator method)": [[15, "gpsea.preprocessing.VarCachingFunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.vepfunctionalannotator method)": [[15, "gpsea.preprocessing.VepFunctionalAnnotator.annotate", false]], "any() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.any", false]], "apply_predicates_on_patients() (in module gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.apply_predicates_on_patients", false]], "auditor (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.Auditor", false]], "autosomal_dominant() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate static method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.autosomal_dominant", false]], "autosomal_recessive() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate static method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.autosomal_recessive", false]], "bnd (gpsea.model.variantclass attribute)": [[13, "gpsea.model.VariantClass.BND", false]], "boolean_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.boolean_predicate", false]], "cache_env (in module gpsea.config)": [[10, "gpsea.config.CACHE_ENV", false]], "cat_id (gpsea.analysis.predicate.patientcategory property)": [[5, "gpsea.analysis.predicate.PatientCategory.cat_id", false]], "categorization (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.Categorization", false]], "category (gpsea.analysis.predicate.categorization property)": [[5, "gpsea.analysis.predicate.Categorization.category", false]], "cds_end (gpsea.model.transcriptcoordinates property)": [[13, "gpsea.model.TranscriptCoordinates.cds_end", false]], "cds_start (gpsea.model.transcriptcoordinates property)": [[13, "gpsea.model.TranscriptCoordinates.cds_start", false]], "change_length (gpsea.model.variantcoordinates property)": [[13, "gpsea.model.VariantCoordinates.change_length", false]], "change_length() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.change_length", false]], "children (gpsea.preprocessing.notepadtree property)": [[15, "gpsea.preprocessing.NotepadTree.children", false]], "chrom (gpsea.model.variantcoordinates property)": [[13, "gpsea.model.VariantCoordinates.chrom", false]], "code (gpsea.model.genotype property)": [[13, "gpsea.model.Genotype.code", false]], "coding_sequence_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.CODING_SEQUENCE_VARIANT", false]], "cohort (class in gpsea.model)": [[13, "gpsea.model.Cohort", false]], "cohortcreator (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.CohortCreator", false]], "cohortviewable (class in gpsea.view)": [[16, "gpsea.view.CohortViewable", false]], "compare_genotype_vs_phenotype_score() (gpsea.analysis.pscore.phenotypescoreanalysis method)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysis.compare_genotype_vs_phenotype_score", false]], "compare_genotype_vs_phenotypes() (gpsea.analysis.pcats.multiphenotypeanalysis method)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysis.compare_genotype_vs_phenotypes", false]], "compute_pval() (gpsea.analysis.pcats.stats.countstatistic method)": [[4, "gpsea.analysis.pcats.stats.CountStatistic.compute_pval", false]], "compute_pval() (gpsea.analysis.pcats.stats.fisherexacttest method)": [[4, "gpsea.analysis.pcats.stats.FisherExactTest.compute_pval", false]], "compute_pval() (gpsea.analysis.pscore.stats.mannwhitneystatistic method)": [[9, "gpsea.analysis.pscore.stats.MannWhitneyStatistic.compute_pval", false]], "compute_pval() (gpsea.analysis.pscore.stats.phenotypescorestatistic method)": [[9, "gpsea.analysis.pscore.stats.PhenotypeScoreStatistic.compute_pval", false]], "compute_pval() (gpsea.analysis.pscore.stats.tteststatistic method)": [[9, "gpsea.analysis.pscore.stats.TTestStatistic.compute_pval", false]], "configure_caching_cohort_creator() (in module gpsea.preprocessing)": [[15, "gpsea.preprocessing.configure_caching_cohort_creator", false]], "configure_caching_patient_creator() (in module gpsea.preprocessing)": [[15, "gpsea.preprocessing.configure_caching_patient_creator", false]], "configure_cohort_creator() (in module gpsea.preprocessing)": [[15, "gpsea.preprocessing.configure_cohort_creator", false]], "configure_default_protein_metadata_service() (in module gpsea.preprocessing)": [[15, "gpsea.preprocessing.configure_default_protein_metadata_service", false]], "configure_patient_creator() (in module gpsea.preprocessing)": [[15, "gpsea.preprocessing.configure_patient_creator", false]], "configure_protein_metadata_service() (in module gpsea.preprocessing)": [[15, "gpsea.preprocessing.configure_protein_metadata_service", false]], "contains() (gpsea.model.genome.genomicregion method)": [[14, "gpsea.model.genome.GenomicRegion.contains", false]], "contains() (gpsea.model.genome.region method)": [[14, "gpsea.model.genome.Region.contains", false]], "contains_pos() (gpsea.model.genome.region method)": [[14, "gpsea.model.genome.Region.contains_pos", false]], "contig (class in gpsea.model.genome)": [[14, "gpsea.model.genome.Contig", false]], "contig (gpsea.model.genome.genomicregion property)": [[14, "gpsea.model.genome.GenomicRegion.contig", false]], "contig_by_name() (gpsea.model.genome.genomebuild method)": [[14, "gpsea.model.genome.GenomeBuild.contig_by_name", false]], "contigs (gpsea.model.genome.genomebuild property)": [[14, "gpsea.model.genome.GenomeBuild.contigs", false]], "corrected_pvals (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.corrected_pvals", false]], "count() (gpsea.analysis.predicate.genotype.allelecounter method)": [[6, "gpsea.analysis.predicate.genotype.AlleleCounter.count", false]], "countingphenotypescorer (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer", false]], "countstatistic (class in gpsea.analysis.pcats.stats)": [[4, "gpsea.analysis.pcats.stats.CountStatistic", false]], "create() (gpsea.model.proteinfeature static method)": [[13, "gpsea.model.ProteinFeature.create", false]], "create_variant_from_scratch() (gpsea.model.variant static method)": [[13, "gpsea.model.Variant.create_variant_from_scratch", false]], "curie (gpsea.model.varianteffect property)": [[13, "gpsea.model.VariantEffect.curie", false]], "datasanityissue (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.DataSanityIssue", false]], "default() (gpsea.io.gpseajsonencoder method)": [[12, "gpsea.io.GpseaJSONEncoder.default", false]], "default_cache_path (in module gpsea.config)": [[10, "gpsea.config.DEFAULT_CACHE_PATH", false]], "default_filter() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.default_filter", false]], "defaultimprecisesvfunctionalannotator (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator", false]], "del (gpsea.model.variantclass attribute)": [[13, "gpsea.model.VariantClass.DEL", false]], "description (gpsea.analysis.predicate.patientcategory property)": [[5, "gpsea.analysis.predicate.PatientCategory.description", false]], "devriesphenotypescorer (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.DeVriesPhenotypeScorer", false]], "disease (class in gpsea.model)": [[13, "gpsea.model.Disease", false]], "diseaseanalysis (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.DiseaseAnalysis", false]], "diseasepresencepredicate (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate", false]], "diseases (gpsea.model.patient property)": [[13, "gpsea.model.Patient.diseases", false]], "diseaseviewable (class in gpsea.view)": [[16, "gpsea.view.DiseaseViewable", false]], "display_question() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.display_question", false]], "distance_to() (gpsea.model.genome.genomicregion method)": [[14, "gpsea.model.genome.GenomicRegion.distance_to", false]], "distance_to() (gpsea.model.genome.region method)": [[14, "gpsea.model.genome.Region.distance_to", false]], "domain (gpsea.model.featuretype attribute)": [[13, "gpsea.model.FeatureType.DOMAIN", false]], "domains() (gpsea.model.proteinmetadata method)": [[13, "gpsea.model.ProteinMetadata.domains", false]], "downstream_gene_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.DOWNSTREAM_GENE_VARIANT", false]], "draw_fig() (gpsea.view.proteinvisualizer method)": [[16, "gpsea.view.ProteinVisualizer.draw_fig", false]], "draw_protein_diagram() (gpsea.view.proteinvisualizer method)": [[16, "gpsea.view.ProteinVisualizer.draw_protein_diagram", false]], "draw_variants() (gpsea.view.varianttranscriptvisualizer method)": [[16, "gpsea.view.VariantTranscriptVisualizer.draw_variants", false]], "dup (gpsea.model.variantclass attribute)": [[13, "gpsea.model.VariantClass.DUP", false]], "empty() (gpsea.model.genotypes static method)": [[13, "gpsea.model.Genotypes.empty", false]], "end (gpsea.model.featureinfo property)": [[13, "gpsea.model.FeatureInfo.end", false]], "end (gpsea.model.genome.region property)": [[14, "gpsea.model.genome.Region.end", false]], "end (gpsea.model.variantcoordinates property)": [[13, "gpsea.model.VariantCoordinates.end", false]], "end_on_strand() (gpsea.model.genome.genomicregion method)": [[14, "gpsea.model.genome.GenomicRegion.end_on_strand", false]], "error (gpsea.preprocessing.level attribute)": [[15, "gpsea.preprocessing.Level.ERROR", false]], "error_count() (gpsea.preprocessing.notepad method)": [[15, "gpsea.preprocessing.Notepad.error_count", false]], "errors() (gpsea.preprocessing.notepad method)": [[15, "gpsea.preprocessing.Notepad.errors", false]], "excluded_diseases() (gpsea.model.patient method)": [[13, "gpsea.model.Patient.excluded_diseases", false]], "excluded_phenotypes() (gpsea.model.patient method)": [[13, "gpsea.model.Patient.excluded_phenotypes", false]], "exon() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.exon", false]], "exons (gpsea.model.transcriptcoordinates property)": [[13, "gpsea.model.TranscriptCoordinates.exons", false]], "fail() (gpsea.analysis.mtc_filter.phenotypemtcresult static method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.fail", false]], "feature_elongation (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.FEATURE_ELONGATION", false]], "feature_truncation (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.FEATURE_TRUNCATION", false]], "feature_type (gpsea.model.proteinfeature property)": [[13, "gpsea.model.ProteinFeature.feature_type", false]], "featureinfo (class in gpsea.model)": [[13, "gpsea.model.FeatureInfo", false]], "featuretype (class in gpsea.model)": [[13, "gpsea.model.FeatureType", false]], "female (gpsea.model.sex attribute)": [[13, "gpsea.model.Sex.FEMALE", false]], "fetch() (gpsea.preprocessing.transcriptcoordinateservice method)": [[15, "gpsea.preprocessing.TranscriptCoordinateService.fetch", false]], "fetch() (gpsea.preprocessing.vvmulticoordinateservice method)": [[15, "gpsea.preprocessing.VVMultiCoordinateService.fetch", false]], "fetch_for_gene() (gpsea.preprocessing.genecoordinateservice method)": [[15, "gpsea.preprocessing.GeneCoordinateService.fetch_for_gene", false]], "fetch_for_gene() (gpsea.preprocessing.vvmulticoordinateservice method)": [[15, "gpsea.preprocessing.VVMultiCoordinateService.fetch_for_gene", false]], "fetch_response() (gpsea.preprocessing.vepfunctionalannotator method)": [[15, "gpsea.preprocessing.VepFunctionalAnnotator.fetch_response", false]], "filter() (gpsea.analysis.mtc_filter.hpomtcfilter method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.filter", false]], "filter() (gpsea.analysis.mtc_filter.phenotypemtcfilter method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter.filter", false]], "filter() (gpsea.analysis.mtc_filter.specifiedtermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.filter", false]], "filter() (gpsea.analysis.mtc_filter.usealltermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.filter", false]], "filter_method_name() (gpsea.analysis.mtc_filter.hpomtcfilter method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.filter_method_name", false]], "filter_method_name() (gpsea.analysis.mtc_filter.phenotypemtcfilter method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter.filter_method_name", false]], "filter_method_name() (gpsea.analysis.mtc_filter.specifiedtermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.filter_method_name", false]], "filter_method_name() (gpsea.analysis.mtc_filter.usealltermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.filter_method_name", false]], "filtering_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.filtering_predicate", false]], "find_coordinates() (gpsea.preprocessing.phenopacketvariantcoordinatefinder method)": [[15, "gpsea.preprocessing.PhenopacketVariantCoordinateFinder.find_coordinates", false]], "find_coordinates() (gpsea.preprocessing.variantcoordinatefinder method)": [[15, "gpsea.preprocessing.VariantCoordinateFinder.find_coordinates", false]], "find_coordinates() (gpsea.preprocessing.vvhgvsvariantcoordinatefinder method)": [[15, "gpsea.preprocessing.VVHgvsVariantCoordinateFinder.find_coordinates", false]], "fisherexacttest (class in gpsea.analysis.pcats.stats)": [[4, "gpsea.analysis.pcats.stats.FisherExactTest", false]], "five_prime_utr_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.FIVE_PRIME_UTR_VARIANT", false]], "for_sample() (gpsea.model.genotypes method)": [[13, "gpsea.model.Genotypes.for_sample", false]], "format_as_string() (gpsea.view.variantformatter method)": [[16, "gpsea.view.VariantFormatter.format_as_string", false]], "format_coordinates_for_vep_query() (gpsea.preprocessing.vepfunctionalannotator static method)": [[15, "gpsea.preprocessing.VepFunctionalAnnotator.format_coordinates_for_vep_query", false]], "frameshift_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.FRAMESHIFT_VARIANT", false]], "from_mapping() (gpsea.model.genotypes static method)": [[13, "gpsea.model.Genotypes.from_mapping", false]], "from_moi_info() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate static method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.from_moi_info", false]], "from_patients() (gpsea.model.cohort static method)": [[13, "gpsea.model.Cohort.from_patients", false]], "from_query_curies() (gpsea.analysis.pscore.countingphenotypescorer static method)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer.from_query_curies", false]], "from_raw_parts() (gpsea.analysis.predicate.categorization static method)": [[5, "gpsea.analysis.predicate.Categorization.from_raw_parts", false]], "from_raw_parts() (gpsea.model.patient static method)": [[13, "gpsea.model.Patient.from_raw_parts", false]], "from_raw_parts() (gpsea.model.phenotype static method)": [[13, "gpsea.model.Phenotype.from_raw_parts", false]], "from_term() (gpsea.model.phenotype static method)": [[13, "gpsea.model.Phenotype.from_term", false]], "from_vcf_literal() (gpsea.model.variantcoordinates static method)": [[13, "gpsea.model.VariantCoordinates.from_vcf_literal", false]], "from_vcf_symbolic() (gpsea.model.variantcoordinates static method)": [[13, "gpsea.model.VariantCoordinates.from_vcf_symbolic", false]], "functionalannotator (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.FunctionalAnnotator", false]], "genbank_acc (gpsea.model.genome.contig property)": [[14, "gpsea.model.genome.Contig.genbank_acc", false]], "gene() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.gene", false]], "gene_id (gpsea.model.imprecisesvinfo property)": [[13, "gpsea.model.ImpreciseSvInfo.gene_id", false]], "gene_id (gpsea.model.transcriptannotation property)": [[13, "gpsea.model.TranscriptAnnotation.gene_id", false]], "gene_id (gpsea.model.transcriptinfoaware property)": [[13, "gpsea.model.TranscriptInfoAware.gene_id", false]], "gene_symbol (gpsea.model.imprecisesvinfo property)": [[13, "gpsea.model.ImpreciseSvInfo.gene_symbol", false]], "genecoordinateservice (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.GeneCoordinateService", false]], "genome_build_id (gpsea.model.genome.genomebuild property)": [[14, "gpsea.model.genome.GenomeBuild.genome_build_id", false]], "genomebuild (class in gpsea.model.genome)": [[14, "gpsea.model.genome.GenomeBuild", false]], "genomebuildidentifier (class in gpsea.model.genome)": [[14, "gpsea.model.genome.GenomeBuildIdentifier", false]], "genomicregion (class in gpsea.model.genome)": [[14, "gpsea.model.genome.GenomicRegion", false]], "genotype (class in gpsea.model)": [[13, "gpsea.model.Genotype", false]], "genotype_for_sample() (gpsea.model.genotyped method)": [[13, "gpsea.model.Genotyped.genotype_for_sample", false]], "genotype_phenotype_scores (gpsea.analysis.pscore.phenotypescoreanalysisresult property)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.genotype_phenotype_scores", false]], "genotyped (class in gpsea.model)": [[13, "gpsea.model.Genotyped", false]], "genotypepolypredicate (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.GenotypePolyPredicate", false]], "genotypes (class in gpsea.model)": [[13, "gpsea.model.Genotypes", false]], "genotypes (gpsea.model.genotyped property)": [[13, "gpsea.model.Genotyped.genotypes", false]], "genotypes (gpsea.model.variant property)": [[13, "gpsea.model.Variant.genotypes", false]], "genotypes_have_same_hpo_proportions() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.genotypes_have_same_hpo_proportions", false]], "get_annotations() (gpsea.preprocessing.proteinannotationcache method)": [[15, "gpsea.preprocessing.ProteinAnnotationCache.get_annotations", false]], "get_annotations() (gpsea.preprocessing.variantannotationcache method)": [[15, "gpsea.preprocessing.VariantAnnotationCache.get_annotations", false]], "get_cache_dir_path() (in module gpsea.config)": [[10, "gpsea.config.get_cache_dir_path", false]], "get_categories() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_categories", false]], "get_categorizations() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.get_categorizations", false]], "get_categorizations() (gpsea.analysis.predicate.phenotype.diseasepresencepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.get_categorizations", false]], "get_categorizations() (gpsea.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.PropagatingPhenotypePredicate.get_categorizations", false]], "get_categorizations() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_categorizations", false]], "get_category() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_category", false]], "get_category_name() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_category_name", false]], "get_category_names() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_category_names", false]], "get_cds_regions() (gpsea.model.transcriptcoordinates method)": [[13, "gpsea.model.TranscriptCoordinates.get_cds_regions", false]], "get_coding_base_count() (gpsea.model.transcriptcoordinates method)": [[13, "gpsea.model.TranscriptCoordinates.get_coding_base_count", false]], "get_codon_count() (gpsea.model.transcriptcoordinates method)": [[13, "gpsea.model.TranscriptCoordinates.get_codon_count", false]], "get_excluded_count() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.get_excluded_count", false]], "get_features_variant_overlaps() (gpsea.model.proteinmetadata method)": [[13, "gpsea.model.ProteinMetadata.get_features_variant_overlaps", false]], "get_five_prime_utrs() (gpsea.model.transcriptcoordinates method)": [[13, "gpsea.model.TranscriptCoordinates.get_five_prime_utrs", false]], "get_maximum_group_observed_hpo_frequency() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.get_maximum_group_observed_HPO_frequency", false]], "get_number_of_observed_hpo_observations() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.get_number_of_observed_hpo_observations", false]], "get_patient_ids() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.get_patient_ids", false]], "get_question() (gpsea.analysis.predicate.genotype.allelecounter method)": [[6, "gpsea.analysis.predicate.genotype.AlleleCounter.get_question", false]], "get_question() (gpsea.analysis.predicate.genotype.variantpredicate method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicate.get_question", false]], "get_question() (gpsea.analysis.pscore.countingphenotypescorer method)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer.get_question", false]], "get_question_base() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.get_question_base", false]], "get_question_base() (gpsea.analysis.predicate.phenotype.diseasepresencepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.get_question_base", false]], "get_question_base() (gpsea.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.PropagatingPhenotypePredicate.get_question_base", false]], "get_question_base() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_question_base", false]], "get_response() (gpsea.preprocessing.vvmulticoordinateservice method)": [[15, "gpsea.preprocessing.VVMultiCoordinateService.get_response", false]], "get_three_prime_utrs() (gpsea.model.transcriptcoordinates method)": [[13, "gpsea.model.TranscriptCoordinates.get_three_prime_utrs", false]], "get_toy_cohort() (in module gpsea.data)": [[11, "gpsea.data.get_toy_cohort", false]], "get_variant_by_key() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.get_variant_by_key", false]], "gpsea": [[0, "module-gpsea", false]], "gpsea.analysis": [[1, "module-gpsea.analysis", false]], "gpsea.analysis.mtc_filter": [[2, "module-gpsea.analysis.mtc_filter", false]], "gpsea.analysis.pcats": [[3, "module-gpsea.analysis.pcats", false]], "gpsea.analysis.pcats.stats": [[4, "module-gpsea.analysis.pcats.stats", false]], "gpsea.analysis.predicate": [[5, "module-gpsea.analysis.predicate", false]], "gpsea.analysis.predicate.genotype": [[6, "module-gpsea.analysis.predicate.genotype", false]], "gpsea.analysis.predicate.phenotype": [[7, "module-gpsea.analysis.predicate.phenotype", false]], "gpsea.analysis.pscore": [[8, "module-gpsea.analysis.pscore", false]], "gpsea.analysis.pscore.stats": [[9, "module-gpsea.analysis.pscore.stats", false]], "gpsea.config": [[10, "module-gpsea.config", false]], "gpsea.data": [[11, "module-gpsea.data", false]], "gpsea.io": [[12, "module-gpsea.io", false]], "gpsea.model": [[13, "module-gpsea.model", false]], "gpsea.model.genome": [[14, "module-gpsea.model.genome", false]], "gpsea.preprocessing": [[15, "module-gpsea.preprocessing", false]], "gpsea.view": [[16, "module-gpsea.view", false]], "gpseajsondecoder (class in gpsea.io)": [[12, "gpsea.io.GpseaJSONDecoder", false]], "gpseajsonencoder (class in gpsea.io)": [[12, "gpsea.io.GpseaJSONEncoder", false]], "groups_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.groups_predicate", false]], "has_errors() (gpsea.preprocessing.notepadtree method)": [[15, "gpsea.preprocessing.NotepadTree.has_errors", false]], "has_errors_or_warnings() (gpsea.preprocessing.notepadtree method)": [[15, "gpsea.preprocessing.NotepadTree.has_errors_or_warnings", false]], "has_sv_info() (gpsea.model.variantinfo method)": [[13, "gpsea.model.VariantInfo.has_sv_info", false]], "has_variant_coordinates() (gpsea.model.variantinfo method)": [[13, "gpsea.model.VariantInfo.has_variant_coordinates", false]], "has_warnings() (gpsea.preprocessing.notepadtree method)": [[15, "gpsea.preprocessing.NotepadTree.has_warnings", false]], "hemizygous (gpsea.model.genotype attribute)": [[13, "gpsea.model.Genotype.HEMIZYGOUS", false]], "heterozygous (gpsea.model.genotype attribute)": [[13, "gpsea.model.Genotype.HETEROZYGOUS", false]], "hgvs_cdna (gpsea.model.transcriptannotation property)": [[13, "gpsea.model.TranscriptAnnotation.hgvs_cdna", false]], "hgvsp (gpsea.model.transcriptannotation property)": [[13, "gpsea.model.TranscriptAnnotation.hgvsp", false]], "homozygous_alternate (gpsea.model.genotype attribute)": [[13, "gpsea.model.Genotype.HOMOZYGOUS_ALTERNATE", false]], "homozygous_reference (gpsea.model.genotype attribute)": [[13, "gpsea.model.Genotype.HOMOZYGOUS_REFERENCE", false]], "hpomtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter", false]], "hpotermanalysis (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.HpoTermAnalysis", false]], "hpotermanalysisresult (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult", false]], "identifier (gpsea.model.disease property)": [[13, "gpsea.model.Disease.identifier", false]], "identifier (gpsea.model.genome.genomebuild property)": [[14, "gpsea.model.genome.GenomeBuild.identifier", false]], "identifier (gpsea.model.genome.genomebuildidentifier property)": [[14, "gpsea.model.genome.GenomeBuildIdentifier.identifier", false]], "identifier (gpsea.model.phenotype property)": [[13, "gpsea.model.Phenotype.identifier", false]], "identifier (gpsea.model.transcriptcoordinates property)": [[13, "gpsea.model.TranscriptCoordinates.identifier", false]], "imprecisesvfunctionalannotator (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.ImpreciseSvFunctionalAnnotator", false]], "imprecisesvinfo (class in gpsea.model)": [[13, "gpsea.model.ImpreciseSvInfo", false]], "incomplete_terminal_codon_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.INCOMPLETE_TERMINAL_CODON_VARIANT", false]], "info (gpsea.model.proteinfeature property)": [[13, "gpsea.model.ProteinFeature.info", false]], "inframe_deletion (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.INFRAME_DELETION", false]], "inframe_insertion (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.INFRAME_INSERTION", false]], "ins (gpsea.model.variantclass attribute)": [[13, "gpsea.model.VariantClass.INS", false]], "intergenic_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.INTERGENIC_VARIANT", false]], "intron_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.INTRON_VARIANT", false]], "inv (gpsea.model.variantclass attribute)": [[13, "gpsea.model.VariantClass.INV", false]], "is_coding() (gpsea.model.transcriptcoordinates method)": [[13, "gpsea.model.TranscriptCoordinates.is_coding", false]], "is_empty() (gpsea.model.genome.region method)": [[14, "gpsea.model.genome.Region.is_empty", false]], "is_female() (gpsea.model.sex method)": [[13, "gpsea.model.Sex.is_female", false]], "is_filtered_out() (gpsea.analysis.mtc_filter.phenotypemtcresult method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.is_filtered_out", false]], "is_large_imprecise_sv() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.is_large_imprecise_sv", false]], "is_male() (gpsea.model.sex method)": [[13, "gpsea.model.Sex.is_male", false]], "is_negative() (gpsea.model.genome.strand method)": [[14, "gpsea.model.genome.Strand.is_negative", false]], "is_observed (gpsea.model.phenotype property)": [[13, "gpsea.model.Phenotype.is_observed", false]], "is_passed() (gpsea.analysis.mtc_filter.phenotypemtcresult method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.is_passed", false]], "is_positive() (gpsea.model.genome.strand method)": [[14, "gpsea.model.genome.Strand.is_positive", false]], "is_preferred (gpsea.model.transcriptannotation property)": [[13, "gpsea.model.TranscriptAnnotation.is_preferred", false]], "is_preferred (gpsea.model.transcriptcoordinates property)": [[13, "gpsea.model.TranscriptCoordinates.is_preferred", false]], "is_present (gpsea.model.disease property)": [[13, "gpsea.model.Disease.is_present", false]], "is_present (gpsea.model.phenotype property)": [[13, "gpsea.model.Phenotype.is_present", false]], "is_provided() (gpsea.model.sex method)": [[13, "gpsea.model.Sex.is_provided", false]], "is_structural() (gpsea.model.variantcoordinates method)": [[13, "gpsea.model.VariantCoordinates.is_structural", false]], "is_structural() (gpsea.model.variantinfo method)": [[13, "gpsea.model.VariantInfo.is_structural", false]], "is_structural_deletion() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.is_structural_deletion", false]], "is_structural_variant() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.is_structural_variant", false]], "is_unknown() (gpsea.model.sex method)": [[13, "gpsea.model.Sex.is_unknown", false]], "issues (gpsea.preprocessing.notepad property)": [[15, "gpsea.preprocessing.Notepad.issues", false]], "iterate_nodes() (gpsea.preprocessing.notepadtree method)": [[15, "gpsea.preprocessing.NotepadTree.iterate_nodes", false]], "label (gpsea.model.proteinmetadata property)": [[13, "gpsea.model.ProteinMetadata.label", false]], "label (gpsea.model.samplelabels property)": [[13, "gpsea.model.SampleLabels.label", false]], "label (gpsea.preprocessing.notepad property)": [[15, "gpsea.preprocessing.Notepad.label", false]], "label_summary() (gpsea.model.samplelabels method)": [[13, "gpsea.model.SampleLabels.label_summary", false]], "labels (gpsea.model.patient property)": [[13, "gpsea.model.Patient.labels", false]], "level (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.Level", false]], "level (gpsea.preprocessing.datasanityissue property)": [[15, "gpsea.preprocessing.DataSanityIssue.level", false]], "level (gpsea.preprocessing.notepadtree property)": [[15, "gpsea.preprocessing.NotepadTree.level", false]], "list_all_diseases() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.list_all_diseases", false]], "list_all_proteins() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.list_all_proteins", false]], "list_all_variants() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.list_all_variants", false]], "list_present_phenotypes() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.list_present_phenotypes", false]], "load_phenopacket_files() (in module gpsea.preprocessing)": [[15, "gpsea.preprocessing.load_phenopacket_files", false]], "load_phenopacket_folder() (in module gpsea.preprocessing)": [[15, "gpsea.preprocessing.load_phenopacket_folder", false]], "load_phenopackets() (in module gpsea.preprocessing)": [[15, "gpsea.preprocessing.load_phenopackets", false]], "major_assembly (gpsea.model.genome.genomebuildidentifier property)": [[14, "gpsea.model.genome.GenomeBuildIdentifier.major_assembly", false]], "male (gpsea.model.sex attribute)": [[13, "gpsea.model.Sex.MALE", false]], "mannwhitneystatistic (class in gpsea.analysis.pscore.stats)": [[9, "gpsea.analysis.pscore.stats.MannWhitneyStatistic", false]], "marker_counts (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.marker_counts", false]], "mature_mirna_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.MATURE_MIRNA_VARIANT", false]], "message (gpsea.preprocessing.datasanityissue property)": [[15, "gpsea.preprocessing.DataSanityIssue.message", false]], "meta_label (gpsea.model.samplelabels property)": [[13, "gpsea.model.SampleLabels.meta_label", false]], "missense_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.MISSENSE_VARIANT", false]], "mnv (gpsea.model.variantclass attribute)": [[13, "gpsea.model.VariantClass.MNV", false]], "modeofinheritancepredicate (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate", false]], "module": [[0, "module-gpsea", false], [1, "module-gpsea.analysis", false], [2, "module-gpsea.analysis.mtc_filter", false], [3, "module-gpsea.analysis.pcats", false], [4, "module-gpsea.analysis.pcats.stats", false], [5, "module-gpsea.analysis.predicate", false], [6, "module-gpsea.analysis.predicate.genotype", false], [7, "module-gpsea.analysis.predicate.phenotype", false], [8, "module-gpsea.analysis.pscore", false], [9, "module-gpsea.analysis.pscore.stats", false], [10, "module-gpsea.config", false], [11, "module-gpsea.data", false], [12, "module-gpsea.io", false], [13, "module-gpsea.model", false], [14, "module-gpsea.model.genome", false], [15, "module-gpsea.preprocessing", false], [16, "module-gpsea.view", false]], "motif (gpsea.model.featuretype attribute)": [[13, "gpsea.model.FeatureType.MOTIF", false]], "motifs() (gpsea.model.proteinmetadata method)": [[13, "gpsea.model.ProteinMetadata.motifs", false]], "mtc_filter_name (gpsea.analysis.pcats.hpotermanalysisresult property)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_filter_name", false]], "mtc_filter_results (gpsea.analysis.pcats.hpotermanalysisresult property)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_filter_results", false]], "mtc_name (gpsea.analysis.pcats.hpotermanalysisresult property)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_name", false]], "mtcstatsviewer (class in gpsea.view)": [[16, "gpsea.view.MtcStatsViewer", false]], "multiphenotypeanalysis (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysis", false]], "multiphenotypeanalysisresult (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult", false]], "n_categorizations() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.n_categorizations", false]], "n_usable (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.n_usable", false]], "name (gpsea.analysis.predicate.patientcategory property)": [[5, "gpsea.analysis.predicate.PatientCategory.name", false]], "name (gpsea.model.disease property)": [[13, "gpsea.model.Disease.name", false]], "name (gpsea.model.featureinfo property)": [[13, "gpsea.model.FeatureInfo.name", false]], "name (gpsea.model.genome.contig property)": [[14, "gpsea.model.genome.Contig.name", false]], "negative (gpsea.model.genome.strand attribute)": [[14, "gpsea.model.genome.Strand.NEGATIVE", false]], "nmd_transcript_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.NMD_TRANSCRIPT_VARIANT", false]], "no (gpsea.analysis.predicate.patientcategories attribute)": [[5, "gpsea.analysis.predicate.PatientCategories.NO", false]], "no_call (gpsea.model.genotype attribute)": [[13, "gpsea.model.Genotype.NO_CALL", false]], "non_coding_transcript_exon_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.NON_CODING_TRANSCRIPT_EXON_VARIANT", false]], "non_coding_transcript_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.NON_CODING_TRANSCRIPT_VARIANT", false]], "noncoding_effects (gpsea.preprocessing.vepfunctionalannotator attribute)": [[15, "gpsea.preprocessing.VepFunctionalAnnotator.NONCODING_EFFECTS", false]], "notepad (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.Notepad", false]], "notepadtree (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.NotepadTree", false]], "object_hook() (gpsea.io.gpseajsondecoder static method)": [[12, "gpsea.io.GpseaJSONDecoder.object_hook", false]], "observed (gpsea.model.phenotype property)": [[13, "gpsea.model.Phenotype.observed", false]], "ok() (gpsea.analysis.mtc_filter.phenotypemtcresult static method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.ok", false]], "one_genotype_has_zero_hpo_observations() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.one_genotype_has_zero_hpo_observations", false]], "opposite() (gpsea.model.genome.strand method)": [[14, "gpsea.model.genome.Strand.opposite", false]], "overlapping_exons (gpsea.model.transcriptannotation property)": [[13, "gpsea.model.TranscriptAnnotation.overlapping_exons", false]], "overlaps_with() (gpsea.model.genome.genomicregion method)": [[14, "gpsea.model.genome.GenomicRegion.overlaps_with", false]], "overlaps_with() (gpsea.model.genome.region method)": [[14, "gpsea.model.genome.Region.overlaps_with", false]], "parse_multiple() (gpsea.preprocessing.vvmulticoordinateservice method)": [[15, "gpsea.preprocessing.VVMultiCoordinateService.parse_multiple", false]], "parse_response() (gpsea.preprocessing.vvmulticoordinateservice method)": [[15, "gpsea.preprocessing.VVMultiCoordinateService.parse_response", false]], "parse_uniprot_json() (gpsea.preprocessing.uniprotproteinmetadataservice static method)": [[15, "gpsea.preprocessing.UniprotProteinMetadataService.parse_uniprot_json", false]], "patch (gpsea.model.genome.genomebuildidentifier property)": [[14, "gpsea.model.genome.GenomeBuildIdentifier.patch", false]], "patient (class in gpsea.model)": [[13, "gpsea.model.Patient", false]], "patient_id (gpsea.model.patient property)": [[13, "gpsea.model.Patient.patient_id", false]], "patientcategories (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.PatientCategories", false]], "patientcategory (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.PatientCategory", false]], "patientcreator (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.PatientCreator", false]], "phenopacketpatientcreator (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.PhenopacketPatientCreator", false]], "phenopacketvariantcoordinatefinder (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.PhenopacketVariantCoordinateFinder", false]], "phenotype (class in gpsea.model)": [[13, "gpsea.model.Phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.diseasepresencepredicate property)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.phenotypecategorization property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypeCategorization.phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.phenotypepolypredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.propagatingphenotypepredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PropagatingPhenotypePredicate.phenotype", false]], "phenotypecategorization (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypeCategorization", false]], "phenotypecreator (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.PhenotypeCreator", false]], "phenotypemtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter", false]], "phenotypemtcresult (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult", false]], "phenotypepolypredicate (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate", false]], "phenotypes (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.phenotypes", false]], "phenotypes (gpsea.model.patient property)": [[13, "gpsea.model.Patient.phenotypes", false]], "phenotypescoreanalysis (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysis", false]], "phenotypescoreanalysisresult (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult", false]], "phenotypescorer (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.PhenotypeScorer", false]], "phenotypescorestatistic (class in gpsea.analysis.pscore.stats)": [[9, "gpsea.analysis.pscore.stats.PhenotypeScoreStatistic", false]], "plot_boxplots() (gpsea.analysis.pscore.phenotypescoreanalysisresult method)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.plot_boxplots", false]], "polypredicate (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.PolyPredicate", false]], "positive (gpsea.model.genome.strand attribute)": [[14, "gpsea.model.genome.Strand.POSITIVE", false]], "prepare_hpo_terms_of_interest() (in module gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.prepare_hpo_terms_of_interest", false]], "prepare_notepad() (gpsea.preprocessing.auditor static method)": [[15, "gpsea.preprocessing.Auditor.prepare_notepad", false]], "prepare_predicates_for_terms_of_interest() (in module gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.prepare_predicates_for_terms_of_interest", false]], "present_diseases() (gpsea.model.patient method)": [[13, "gpsea.model.Patient.present_diseases", false]], "present_phenotype_categorization (gpsea.analysis.predicate.phenotype.diseasepresencepredicate property)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.present_phenotype_categorization", false]], "present_phenotype_categorization (gpsea.analysis.predicate.phenotype.phenotypepolypredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.present_phenotype_categorization", false]], "present_phenotype_categorization (gpsea.analysis.predicate.phenotype.propagatingphenotypepredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PropagatingPhenotypePredicate.present_phenotype_categorization", false]], "present_phenotype_category (gpsea.analysis.predicate.phenotype.phenotypepolypredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.present_phenotype_category", false]], "present_phenotypes() (gpsea.model.patient method)": [[13, "gpsea.model.Patient.present_phenotypes", false]], "process() (gpsea.preprocessing.auditor method)": [[15, "gpsea.preprocessing.Auditor.process", false]], "process() (gpsea.preprocessing.cohortcreator method)": [[15, "gpsea.preprocessing.CohortCreator.process", false]], "process() (gpsea.preprocessing.phenopacketpatientcreator method)": [[15, "gpsea.preprocessing.PhenopacketPatientCreator.process", false]], "process() (gpsea.preprocessing.phenotypecreator method)": [[15, "gpsea.preprocessing.PhenotypeCreator.process", false]], "process() (gpsea.view.cohortviewable method)": [[16, "gpsea.view.CohortViewable.process", false]], "process() (gpsea.view.diseaseviewable method)": [[16, "gpsea.view.DiseaseViewable.process", false]], "process() (gpsea.view.mtcstatsviewer method)": [[16, "gpsea.view.MtcStatsViewer.process", false]], "process() (gpsea.view.proteinviewable method)": [[16, "gpsea.view.ProteinViewable.process", false]], "process_response() (gpsea.preprocessing.vepfunctionalannotator method)": [[15, "gpsea.preprocessing.VepFunctionalAnnotator.process_response", false]], "propagatingphenotypepredicate (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.PropagatingPhenotypePredicate", false]], "protcachingmetadataservice (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.ProtCachingMetadataService", false]], "protein_altering_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.PROTEIN_ALTERING_VARIANT", false]], "protein_effect_location (gpsea.model.transcriptannotation property)": [[13, "gpsea.model.TranscriptAnnotation.protein_effect_location", false]], "protein_feature() (gpsea.analysis.predicate.genotype.proteinpredicates method)": [[6, "gpsea.analysis.predicate.genotype.ProteinPredicates.protein_feature", false]], "protein_feature_ends (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_ends", false]], "protein_feature_names (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_names", false]], "protein_feature_starts (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_starts", false]], "protein_feature_type() (gpsea.analysis.predicate.genotype.proteinpredicates method)": [[6, "gpsea.analysis.predicate.genotype.ProteinPredicates.protein_feature_type", false]], "protein_feature_types (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_types", false]], "protein_features (gpsea.model.proteinmetadata property)": [[13, "gpsea.model.ProteinMetadata.protein_features", false]], "protein_id (gpsea.model.proteinmetadata property)": [[13, "gpsea.model.ProteinMetadata.protein_id", false]], "protein_id (gpsea.model.transcriptannotation property)": [[13, "gpsea.model.TranscriptAnnotation.protein_id", false]], "protein_id (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_id", false]], "protein_length (gpsea.model.proteinmetadata property)": [[13, "gpsea.model.ProteinMetadata.protein_length", false]], "protein_length (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_length", false]], "protein_metadata (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_metadata", false]], "proteinannotationcache (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.ProteinAnnotationCache", false]], "proteinfeature (class in gpsea.model)": [[13, "gpsea.model.ProteinFeature", false]], "proteinmetadata (class in gpsea.model)": [[13, "gpsea.model.ProteinMetadata", false]], "proteinmetadataservice (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.ProteinMetadataService", false]], "proteinpredicates (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.ProteinPredicates", false]], "proteinviewable (class in gpsea.view)": [[16, "gpsea.view.ProteinViewable", false]], "proteinvisualizable (class in gpsea.view)": [[16, "gpsea.view.ProteinVisualizable", false]], "proteinvisualizer (class in gpsea.view)": [[16, "gpsea.view.ProteinVisualizer", false]], "pval (gpsea.analysis.pscore.phenotypescoreanalysisresult property)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.pval", false]], "pvals (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.pvals", false]], "reason (gpsea.analysis.mtc_filter.phenotypemtcresult property)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.reason", false]], "ref (gpsea.model.variantcoordinates property)": [[13, "gpsea.model.VariantCoordinates.ref", false]], "ref_length() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.ref_length", false]], "refseq_name (gpsea.model.genome.contig property)": [[14, "gpsea.model.genome.Contig.refseq_name", false]], "region (class in gpsea.model.genome)": [[14, "gpsea.model.genome.Region", false]], "region (gpsea.model.featureinfo property)": [[13, "gpsea.model.FeatureInfo.region", false]], "region (gpsea.model.featuretype attribute)": [[13, "gpsea.model.FeatureType.REGION", false]], "region (gpsea.model.transcriptcoordinates property)": [[13, "gpsea.model.TranscriptCoordinates.region", false]], "region (gpsea.model.variantcoordinates property)": [[13, "gpsea.model.VariantCoordinates.region", false]], "region() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.region", false]], "regions() (gpsea.model.proteinmetadata method)": [[13, "gpsea.model.ProteinMetadata.regions", false]], "regulatory_region_ablation (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.REGULATORY_REGION_ABLATION", false]], "regulatory_region_amplification (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.REGULATORY_REGION_AMPLIFICATION", false]], "regulatory_region_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.REGULATORY_REGION_VARIANT", false]], "repeat (gpsea.model.featuretype attribute)": [[13, "gpsea.model.FeatureType.REPEAT", false]], "repeats() (gpsea.model.proteinmetadata method)": [[13, "gpsea.model.ProteinMetadata.repeats", false]], "samplelabels (class in gpsea.model)": [[13, "gpsea.model.SampleLabels", false]], "score() (gpsea.analysis.pscore.countingphenotypescorer method)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer.score", false]], "score() (gpsea.analysis.pscore.devriesphenotypescorer method)": [[8, "gpsea.analysis.pscore.DeVriesPhenotypeScorer.score", false]], "score() (gpsea.analysis.pscore.phenotypescorer method)": [[8, "gpsea.analysis.pscore.PhenotypeScorer.score", false]], "sequence_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.SEQUENCE_VARIANT", false]], "sex (class in gpsea.model)": [[13, "gpsea.model.Sex", false]], "sex (gpsea.model.patient property)": [[13, "gpsea.model.Patient.sex", false]], "single() (gpsea.model.genotypes static method)": [[13, "gpsea.model.Genotypes.single", false]], "snv (gpsea.model.variantclass attribute)": [[13, "gpsea.model.VariantClass.SNV", false]], "solution (gpsea.preprocessing.datasanityissue property)": [[15, "gpsea.preprocessing.DataSanityIssue.solution", false]], "some_cell_has_greater_than_one_count() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.some_cell_has_greater_than_one_count", false]], "specifiedtermsmtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter", false]], "splice_acceptor_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.SPLICE_ACCEPTOR_VARIANT", false]], "splice_donor_5th_base_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.SPLICE_DONOR_5TH_BASE_VARIANT", false]], "splice_donor_region_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.SPLICE_DONOR_REGION_VARIANT", false]], "splice_donor_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.SPLICE_DONOR_VARIANT", false]], "splice_polypyrimidine_tract_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.SPLICE_POLYPYRIMIDINE_TRACT_VARIANT", false]], "splice_region_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.SPLICE_REGION_VARIANT", false]], "start (gpsea.model.featureinfo property)": [[13, "gpsea.model.FeatureInfo.start", false]], "start (gpsea.model.genome.region property)": [[14, "gpsea.model.genome.Region.start", false]], "start (gpsea.model.variantcoordinates property)": [[13, "gpsea.model.VariantCoordinates.start", false]], "start_lost (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.START_LOST", false]], "start_on_strand() (gpsea.model.genome.genomicregion method)": [[14, "gpsea.model.genome.GenomicRegion.start_on_strand", false]], "start_retained_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.START_RETAINED_VARIANT", false]], "stop_gained (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.STOP_GAINED", false]], "stop_lost (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.STOP_LOST", false]], "stop_retained_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.STOP_RETAINED_VARIANT", false]], "store_annotations() (gpsea.preprocessing.proteinannotationcache method)": [[15, "gpsea.preprocessing.ProteinAnnotationCache.store_annotations", false]], "store_annotations() (gpsea.preprocessing.variantannotationcache method)": [[15, "gpsea.preprocessing.VariantAnnotationCache.store_annotations", false]], "strand (class in gpsea.model.genome)": [[14, "gpsea.model.genome.Strand", false]], "strand (gpsea.model.genome.genomicregion property)": [[14, "gpsea.model.genome.GenomicRegion.strand", false]], "strand (gpsea.model.genome.stranded property)": [[14, "gpsea.model.genome.Stranded.strand", false]], "stranded (class in gpsea.model.genome)": [[14, "gpsea.model.genome.Stranded", false]], "structural_type (gpsea.model.imprecisesvinfo property)": [[13, "gpsea.model.ImpreciseSvInfo.structural_type", false]], "structural_type() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.structural_type", false]], "summarize() (gpsea.analysis.pcats.multiphenotypeanalysisresult method)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.summarize", false]], "supports_shape (gpsea.analysis.pcats.stats.countstatistic property)": [[4, "gpsea.analysis.pcats.stats.CountStatistic.supports_shape", false]], "supports_shape (gpsea.analysis.pcats.stats.fisherexacttest property)": [[4, "gpsea.analysis.pcats.stats.FisherExactTest.supports_shape", false]], "sv_info (gpsea.model.variantinfo property)": [[13, "gpsea.model.VariantInfo.sv_info", false]], "symbol (gpsea.model.genome.strand property)": [[14, "gpsea.model.genome.Strand.symbol", false]], "synonymous_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.SYNONYMOUS_VARIANT", false]], "test() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.test", false]], "test() (gpsea.analysis.predicate.genotype.variantpredicate method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicate.test", false]], "test() (gpsea.analysis.predicate.phenotype.diseasepresencepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.test", false]], "test() (gpsea.analysis.predicate.phenotype.propagatingphenotypepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.PropagatingPhenotypePredicate.test", false]], "test() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.test", false]], "tf_binding_site_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.TF_BINDING_SITE_VARIANT", false]], "tfbs_ablation (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.TFBS_ABLATION", false]], "tfbs_amplification (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.TFBS_AMPLIFICATION", false]], "three_prime_utr_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.THREE_PRIME_UTR_VARIANT", false]], "to_negative_strand() (gpsea.model.genome.transposable method)": [[14, "gpsea.model.genome.Transposable.to_negative_strand", false]], "to_opposite_strand() (gpsea.model.genome.transposable method)": [[14, "gpsea.model.genome.Transposable.to_opposite_strand", false]], "to_positive_strand() (gpsea.model.genome.transposable method)": [[14, "gpsea.model.genome.Transposable.to_positive_strand", false]], "to_string() (gpsea.model.proteinfeature method)": [[13, "gpsea.model.ProteinFeature.to_string", false]], "total_patient_count (gpsea.model.cohort property)": [[13, "gpsea.model.Cohort.total_patient_count", false]], "total_tests (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.total_tests", false]], "transcript() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.transcript", false]], "transcript_ablation (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.TRANSCRIPT_ABLATION", false]], "transcript_amplification (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.TRANSCRIPT_AMPLIFICATION", false]], "transcript_coordinates (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.transcript_coordinates", false]], "transcript_id (gpsea.model.transcriptannotation property)": [[13, "gpsea.model.TranscriptAnnotation.transcript_id", false]], "transcript_id (gpsea.model.transcriptinfoaware property)": [[13, "gpsea.model.TranscriptInfoAware.transcript_id", false]], "transcript_id (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.transcript_id", false]], "transcriptannotation (class in gpsea.model)": [[13, "gpsea.model.TranscriptAnnotation", false]], "transcriptcoordinates (class in gpsea.model)": [[13, "gpsea.model.TranscriptCoordinates", false]], "transcriptcoordinateservice (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.TranscriptCoordinateService", false]], "transcriptinfoaware (class in gpsea.model)": [[13, "gpsea.model.TranscriptInfoAware", false]], "transposable (class in gpsea.model.genome)": [[14, "gpsea.model.genome.Transposable", false]], "transpose_coordinate() (in module gpsea.model.genome)": [[14, "gpsea.model.genome.transpose_coordinate", false]], "tteststatistic (class in gpsea.analysis.pscore.stats)": [[9, "gpsea.analysis.pscore.stats.TTestStatistic", false]], "tx_annotations (gpsea.model.variant property)": [[13, "gpsea.model.Variant.tx_annotations", false]], "ucsc_name (gpsea.model.genome.contig property)": [[14, "gpsea.model.genome.Contig.ucsc_name", false]], "uniprotproteinmetadataservice (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.UniprotProteinMetadataService", false]], "unknown_sex (gpsea.model.sex attribute)": [[13, "gpsea.model.Sex.UNKNOWN_SEX", false]], "upstream_gene_variant (gpsea.model.varianteffect attribute)": [[13, "gpsea.model.VariantEffect.UPSTREAM_GENE_VARIANT", false]], "usealltermsmtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter", false]], "varcachingfunctionalannotator (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.VarCachingFunctionalAnnotator", false]], "variant (class in gpsea.model)": [[13, "gpsea.model.Variant", false]], "variant_class (gpsea.model.imprecisesvinfo property)": [[13, "gpsea.model.ImpreciseSvInfo.variant_class", false]], "variant_class (gpsea.model.variantcoordinates property)": [[13, "gpsea.model.VariantCoordinates.variant_class", false]], "variant_class (gpsea.model.variantinfo property)": [[13, "gpsea.model.VariantInfo.variant_class", false]], "variant_class() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.variant_class", false]], "variant_coordinates (gpsea.model.variantinfo property)": [[13, "gpsea.model.VariantInfo.variant_coordinates", false]], "variant_effect() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.variant_effect", false]], "variant_effect_count_by_tx() (gpsea.model.cohort method)": [[13, "gpsea.model.Cohort.variant_effect_count_by_tx", false]], "variant_effects (gpsea.model.transcriptannotation property)": [[13, "gpsea.model.TranscriptAnnotation.variant_effects", false]], "variant_effects (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.variant_effects", false]], "variant_info (gpsea.model.variant property)": [[13, "gpsea.model.Variant.variant_info", false]], "variant_info (gpsea.model.variantinfoaware property)": [[13, "gpsea.model.VariantInfoAware.variant_info", false]], "variant_key (gpsea.model.imprecisesvinfo property)": [[13, "gpsea.model.ImpreciseSvInfo.variant_key", false]], "variant_key (gpsea.model.variantcoordinates property)": [[13, "gpsea.model.VariantCoordinates.variant_key", false]], "variant_key (gpsea.model.variantinfo property)": [[13, "gpsea.model.VariantInfo.variant_key", false]], "variant_key() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.variant_key", false]], "variant_locations (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.variant_locations", false]], "variant_locations_counted_absolute (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.variant_locations_counted_absolute", false]], "variantannotationcache (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.VariantAnnotationCache", false]], "variantclass (class in gpsea.model)": [[13, "gpsea.model.VariantClass", false]], "variantcoordinatefinder (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.VariantCoordinateFinder", false]], "variantcoordinates (class in gpsea.model)": [[13, "gpsea.model.VariantCoordinates", false]], "varianteffect (class in gpsea.model)": [[13, "gpsea.model.VariantEffect", false]], "variantformatter (class in gpsea.view)": [[16, "gpsea.view.VariantFormatter", false]], "variantinfo (class in gpsea.model)": [[13, "gpsea.model.VariantInfo", false]], "variantinfoaware (class in gpsea.model)": [[13, "gpsea.model.VariantInfoAware", false]], "variantpredicate (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicate", false]], "variantpredicates (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates", false]], "variants (gpsea.model.patient property)": [[13, "gpsea.model.Patient.variants", false]], "varianttranscriptvisualizer (class in gpsea.view)": [[16, "gpsea.view.VariantTranscriptVisualizer", false]], "vepfunctionalannotator (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.VepFunctionalAnnotator", false]], "visit() (gpsea.preprocessing.notepadtree method)": [[15, "gpsea.preprocessing.NotepadTree.visit", false]], "vvhgvsvariantcoordinatefinder (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.VVHgvsVariantCoordinateFinder", false]], "vvmulticoordinateservice (class in gpsea.preprocessing)": [[15, "gpsea.preprocessing.VVMultiCoordinateService", false]], "warn (gpsea.preprocessing.level attribute)": [[15, "gpsea.preprocessing.Level.WARN", false]], "warning_count() (gpsea.preprocessing.notepad method)": [[15, "gpsea.preprocessing.Notepad.warning_count", false]], "warnings() (gpsea.preprocessing.notepad method)": [[15, "gpsea.preprocessing.Notepad.warnings", false]], "with_cache_folder() (gpsea.preprocessing.varcachingfunctionalannotator static method)": [[15, "gpsea.preprocessing.VarCachingFunctionalAnnotator.with_cache_folder", false]], "with_strand() (gpsea.model.genome.genomicregion method)": [[14, "gpsea.model.genome.GenomicRegion.with_strand", false]], "with_strand() (gpsea.model.genome.transposable method)": [[14, "gpsea.model.genome.Transposable.with_strand", false]], "wrap_scoring_function() (gpsea.analysis.pscore.phenotypescorer static method)": [[8, "gpsea.analysis.pscore.PhenotypeScorer.wrap_scoring_function", false]], "x_dominant() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate static method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.x_dominant", false]], "x_recessive() (gpsea.analysis.predicate.genotype.modeofinheritancepredicate static method)": [[6, "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate.x_recessive", false]], "yes (gpsea.analysis.predicate.patientcategories attribute)": [[5, "gpsea.analysis.predicate.PatientCategories.YES", false]]}, "objects": {"": [[0, 0, 0, "-", "gpsea"]], "gpsea": [[1, 0, 0, "-", "analysis"], [10, 0, 0, "-", "config"], [11, 0, 0, "-", "data"], [12, 0, 0, "-", "io"], [13, 0, 0, "-", "model"], [15, 0, 0, "-", "preprocessing"], [16, 0, 0, "-", "view"]], "gpsea.analysis": [[2, 0, 0, "-", "mtc_filter"], [3, 0, 0, "-", "pcats"], [5, 0, 0, "-", "predicate"], [8, 0, 0, "-", "pscore"]], "gpsea.analysis.mtc_filter": [[2, 1, 1, "", "HpoMtcFilter"], [2, 1, 1, "", "PhenotypeMtcFilter"], [2, 1, 1, "", "PhenotypeMtcResult"], [2, 1, 1, "", "SpecifiedTermsMtcFilter"], [2, 1, 1, "", "UseAllTermsMtcFilter"]], "gpsea.analysis.mtc_filter.HpoMtcFilter": [[2, 2, 1, "", "default_filter"], [2, 2, 1, "", "filter"], [2, 2, 1, "", "filter_method_name"], [2, 2, 1, "", "genotypes_have_same_hpo_proportions"], [2, 2, 1, "", "get_maximum_group_observed_HPO_frequency"], [2, 2, 1, "", "get_number_of_observed_hpo_observations"], [2, 2, 1, "", "one_genotype_has_zero_hpo_observations"], [2, 2, 1, "", "some_cell_has_greater_than_one_count"]], "gpsea.analysis.mtc_filter.PhenotypeMtcFilter": [[2, 2, 1, "", "filter"], [2, 2, 1, "", "filter_method_name"]], "gpsea.analysis.mtc_filter.PhenotypeMtcResult": [[2, 2, 1, "", "fail"], [2, 2, 1, "", "is_filtered_out"], [2, 2, 1, "", "is_passed"], [2, 2, 1, "", "ok"], [2, 3, 1, "", "reason"]], "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter": [[2, 2, 1, "", "filter"], [2, 2, 1, "", "filter_method_name"]], "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter": [[2, 2, 1, "", "filter"], [2, 2, 1, "", "filter_method_name"]], "gpsea.analysis.pcats": [[3, 1, 1, "", "DiseaseAnalysis"], [3, 1, 1, "", "HpoTermAnalysis"], [3, 1, 1, "", "HpoTermAnalysisResult"], [3, 1, 1, "", "MultiPhenotypeAnalysis"], [3, 1, 1, "", "MultiPhenotypeAnalysisResult"], [3, 4, 1, "", "apply_predicates_on_patients"], [4, 0, 0, "-", "stats"]], "gpsea.analysis.pcats.HpoTermAnalysisResult": [[3, 3, 1, "", "mtc_filter_name"], [3, 3, 1, "", "mtc_filter_results"], [3, 3, 1, "", "mtc_name"]], "gpsea.analysis.pcats.MultiPhenotypeAnalysis": [[3, 2, 1, "", "compare_genotype_vs_phenotypes"]], "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult": [[3, 3, 1, "", "all_counts"], [3, 3, 1, "", "corrected_pvals"], [3, 3, 1, "", "n_usable"], [3, 3, 1, "", "phenotypes"], [3, 3, 1, "", "pvals"], [3, 2, 1, "", "summarize"], [3, 3, 1, "", "total_tests"]], "gpsea.analysis.pcats.stats": [[4, 1, 1, "", "CountStatistic"], [4, 1, 1, "", "FisherExactTest"]], "gpsea.analysis.pcats.stats.CountStatistic": [[4, 2, 1, "", "compute_pval"], [4, 3, 1, "", "supports_shape"]], "gpsea.analysis.pcats.stats.FisherExactTest": [[4, 2, 1, "", "compute_pval"], [4, 3, 1, "", "supports_shape"]], "gpsea.analysis.predicate": [[5, 1, 1, "", "Categorization"], [5, 1, 1, "", "PatientCategories"], [5, 1, 1, "", "PatientCategory"], [5, 1, 1, "", "PolyPredicate"], [6, 0, 0, "-", "genotype"], [7, 0, 0, "-", "phenotype"]], "gpsea.analysis.predicate.Categorization": [[5, 3, 1, "", "category"], [5, 2, 1, "", "from_raw_parts"]], "gpsea.analysis.predicate.PatientCategories": [[5, 5, 1, "", "NO"], [5, 5, 1, "", "YES"]], "gpsea.analysis.predicate.PatientCategory": [[5, 3, 1, "", "cat_id"], [5, 3, 1, "", "description"], [5, 3, 1, "", "name"]], "gpsea.analysis.predicate.PolyPredicate": [[5, 2, 1, "", "display_question"], [5, 2, 1, "", "get_categories"], [5, 2, 1, "", "get_categorizations"], [5, 2, 1, "", "get_category"], [5, 2, 1, "", "get_category_name"], [5, 2, 1, "", "get_category_names"], [5, 2, 1, "", "get_question_base"], [5, 2, 1, "", "n_categorizations"], [5, 2, 1, "", "test"]], "gpsea.analysis.predicate.genotype": [[6, 1, 1, "", "AlleleCounter"], [6, 1, 1, "", "GenotypePolyPredicate"], [6, 1, 1, "", "ModeOfInheritancePredicate"], [6, 1, 1, "", "ProteinPredicates"], [6, 1, 1, "", "VariantPredicate"], [6, 1, 1, "", "VariantPredicates"], [6, 4, 1, "", "boolean_predicate"], [6, 4, 1, "", "filtering_predicate"], [6, 4, 1, "", "groups_predicate"]], "gpsea.analysis.predicate.genotype.AlleleCounter": [[6, 2, 1, "", "count"], [6, 2, 1, "", "get_question"]], "gpsea.analysis.predicate.genotype.ModeOfInheritancePredicate": [[6, 2, 1, "", "autosomal_dominant"], [6, 2, 1, "", "autosomal_recessive"], [6, 2, 1, "", "from_moi_info"], [6, 2, 1, "", "get_categorizations"], [6, 2, 1, "", "get_question_base"], [6, 2, 1, "", "test"], [6, 2, 1, "", "x_dominant"], [6, 2, 1, "", "x_recessive"]], "gpsea.analysis.predicate.genotype.ProteinPredicates": [[6, 2, 1, "", "protein_feature"], [6, 2, 1, "", "protein_feature_type"]], "gpsea.analysis.predicate.genotype.VariantPredicate": [[6, 2, 1, "", "get_question"], [6, 2, 1, "", "test"]], "gpsea.analysis.predicate.genotype.VariantPredicates": [[6, 2, 1, "", "all"], [6, 2, 1, "", "any"], [6, 2, 1, "", "change_length"], [6, 2, 1, "", "exon"], [6, 2, 1, "", "gene"], [6, 2, 1, "", "is_large_imprecise_sv"], [6, 2, 1, "", "is_structural_deletion"], [6, 2, 1, "", "is_structural_variant"], [6, 2, 1, "", "ref_length"], [6, 2, 1, "", "region"], [6, 2, 1, "", "structural_type"], [6, 2, 1, "", "transcript"], [6, 2, 1, "", "variant_class"], [6, 2, 1, "", "variant_effect"], [6, 2, 1, "", "variant_key"]], "gpsea.analysis.predicate.phenotype": [[7, 1, 1, "", "DiseasePresencePredicate"], [7, 1, 1, "", "PhenotypeCategorization"], [7, 1, 1, "", "PhenotypePolyPredicate"], [7, 1, 1, "", "PropagatingPhenotypePredicate"], [7, 4, 1, "", "prepare_hpo_terms_of_interest"], [7, 4, 1, "", "prepare_predicates_for_terms_of_interest"]], "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate": [[7, 2, 1, "", "get_categorizations"], [7, 2, 1, "", "get_question_base"], [7, 3, 1, "", "phenotype"], [7, 3, 1, "", "present_phenotype_categorization"], [7, 2, 1, "", "test"]], "gpsea.analysis.predicate.phenotype.PhenotypeCategorization": [[7, 3, 1, "", "phenotype"]], "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate": [[7, 3, 1, "", "phenotype"], [7, 3, 1, "", "present_phenotype_categorization"], [7, 3, 1, "", "present_phenotype_category"]], "gpsea.analysis.predicate.phenotype.PropagatingPhenotypePredicate": [[7, 2, 1, "", "get_categorizations"], [7, 2, 1, "", "get_question_base"], [7, 3, 1, "", "phenotype"], [7, 3, 1, "", "present_phenotype_categorization"], [7, 2, 1, "", "test"]], "gpsea.analysis.pscore": [[8, 1, 1, "", "CountingPhenotypeScorer"], [8, 1, 1, "", "DeVriesPhenotypeScorer"], [8, 1, 1, "", "PhenotypeScoreAnalysis"], [8, 1, 1, "", "PhenotypeScoreAnalysisResult"], [8, 1, 1, "", "PhenotypeScorer"], [9, 0, 0, "-", "stats"]], "gpsea.analysis.pscore.CountingPhenotypeScorer": [[8, 2, 1, "", "from_query_curies"], [8, 2, 1, "", "get_question"], [8, 2, 1, "", "score"]], "gpsea.analysis.pscore.DeVriesPhenotypeScorer": [[8, 2, 1, "", "score"]], "gpsea.analysis.pscore.PhenotypeScoreAnalysis": [[8, 2, 1, "", "compare_genotype_vs_phenotype_score"]], "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult": [[8, 3, 1, "", "genotype_phenotype_scores"], [8, 2, 1, "", "plot_boxplots"], [8, 3, 1, "", "pval"]], "gpsea.analysis.pscore.PhenotypeScorer": [[8, 2, 1, "", "score"], [8, 2, 1, "", "wrap_scoring_function"]], "gpsea.analysis.pscore.stats": [[9, 1, 1, "", "MannWhitneyStatistic"], [9, 1, 1, "", "PhenotypeScoreStatistic"], [9, 1, 1, "", "TTestStatistic"]], "gpsea.analysis.pscore.stats.MannWhitneyStatistic": [[9, 2, 1, "", "compute_pval"]], "gpsea.analysis.pscore.stats.PhenotypeScoreStatistic": [[9, 2, 1, "", "compute_pval"]], "gpsea.analysis.pscore.stats.TTestStatistic": [[9, 2, 1, "", "compute_pval"]], "gpsea.config": [[10, 6, 1, "", "CACHE_ENV"], [10, 6, 1, "", "DEFAULT_CACHE_PATH"], [10, 4, 1, "", "get_cache_dir_path"]], "gpsea.data": [[11, 4, 1, "", "get_toy_cohort"]], "gpsea.io": [[12, 1, 1, "", "GpseaJSONDecoder"], [12, 1, 1, "", "GpseaJSONEncoder"]], "gpsea.io.GpseaJSONDecoder": [[12, 2, 1, "", "object_hook"]], "gpsea.io.GpseaJSONEncoder": [[12, 2, 1, "", "default"]], "gpsea.model": [[13, 1, 1, "", "Cohort"], [13, 1, 1, "", "Disease"], [13, 1, 1, "", "FeatureInfo"], [13, 1, 1, "", "FeatureType"], [13, 1, 1, "", "Genotype"], [13, 1, 1, "", "Genotyped"], [13, 1, 1, "", "Genotypes"], [13, 1, 1, "", "ImpreciseSvInfo"], [13, 1, 1, "", "Patient"], [13, 1, 1, "", "Phenotype"], [13, 1, 1, "", "ProteinFeature"], [13, 1, 1, "", "ProteinMetadata"], [13, 1, 1, "", "SampleLabels"], [13, 1, 1, "", "Sex"], [13, 1, 1, "", "TranscriptAnnotation"], [13, 1, 1, "", "TranscriptCoordinates"], [13, 1, 1, "", "TranscriptInfoAware"], [13, 1, 1, "", "Variant"], [13, 1, 1, "", "VariantClass"], [13, 1, 1, "", "VariantCoordinates"], [13, 1, 1, "", "VariantEffect"], [13, 1, 1, "", "VariantInfo"], [13, 1, 1, "", "VariantInfoAware"], [14, 0, 0, "-", "genome"]], "gpsea.model.Cohort": [[13, 2, 1, "", "all_diseases"], [13, 3, 1, "", "all_patients"], [13, 2, 1, "", "all_phenotypes"], [13, 3, 1, "", "all_transcript_ids"], [13, 2, 1, "", "all_variants"], [13, 2, 1, "", "from_patients"], [13, 2, 1, "", "get_excluded_count"], [13, 2, 1, "", "get_patient_ids"], [13, 2, 1, "", "get_variant_by_key"], [13, 2, 1, "", "list_all_diseases"], [13, 2, 1, "", "list_all_proteins"], [13, 2, 1, "", "list_all_variants"], [13, 2, 1, "", "list_present_phenotypes"], [13, 3, 1, "", "total_patient_count"], [13, 2, 1, "", "variant_effect_count_by_tx"]], "gpsea.model.Disease": [[13, 3, 1, "", "identifier"], [13, 3, 1, "", "is_present"], [13, 3, 1, "", "name"]], "gpsea.model.FeatureInfo": [[13, 3, 1, "", "end"], [13, 3, 1, "", "name"], [13, 3, 1, "", "region"], [13, 3, 1, "", "start"]], "gpsea.model.FeatureType": [[13, 5, 1, "", "DOMAIN"], [13, 5, 1, "", "MOTIF"], [13, 5, 1, "", "REGION"], [13, 5, 1, "", "REPEAT"]], "gpsea.model.Genotype": [[13, 5, 1, "", "HEMIZYGOUS"], [13, 5, 1, "", "HETEROZYGOUS"], [13, 5, 1, "", "HOMOZYGOUS_ALTERNATE"], [13, 5, 1, "", "HOMOZYGOUS_REFERENCE"], [13, 5, 1, "", "NO_CALL"], [13, 3, 1, "", "code"]], "gpsea.model.Genotyped": [[13, 2, 1, "", "genotype_for_sample"], [13, 3, 1, "", "genotypes"]], "gpsea.model.Genotypes": [[13, 2, 1, "", "empty"], [13, 2, 1, "", "for_sample"], [13, 2, 1, "", "from_mapping"], [13, 2, 1, "", "single"]], "gpsea.model.ImpreciseSvInfo": [[13, 3, 1, "", "gene_id"], [13, 3, 1, "", "gene_symbol"], [13, 3, 1, "", "structural_type"], [13, 3, 1, "", "variant_class"], [13, 3, 1, "", "variant_key"]], "gpsea.model.Patient": [[13, 3, 1, "", "diseases"], [13, 2, 1, "", "excluded_diseases"], [13, 2, 1, "", "excluded_phenotypes"], [13, 2, 1, "", "from_raw_parts"], [13, 3, 1, "", "labels"], [13, 3, 1, "", "patient_id"], [13, 3, 1, "", "phenotypes"], [13, 2, 1, "", "present_diseases"], [13, 2, 1, "", "present_phenotypes"], [13, 3, 1, "", "sex"], [13, 3, 1, "", "variants"]], "gpsea.model.Phenotype": [[13, 2, 1, "", "from_raw_parts"], [13, 2, 1, "", "from_term"], [13, 3, 1, "", "identifier"], [13, 3, 1, "", "is_observed"], [13, 3, 1, "", "is_present"], [13, 3, 1, "", "observed"]], "gpsea.model.ProteinFeature": [[13, 2, 1, "", "create"], [13, 3, 1, "", "feature_type"], [13, 3, 1, "", "info"], [13, 2, 1, "", "to_string"]], "gpsea.model.ProteinMetadata": [[13, 2, 1, "", "domains"], [13, 2, 1, "", "get_features_variant_overlaps"], [13, 3, 1, "", "label"], [13, 2, 1, "", "motifs"], [13, 3, 1, "", "protein_features"], [13, 3, 1, "", "protein_id"], [13, 3, 1, "", "protein_length"], [13, 2, 1, "", "regions"], [13, 2, 1, "", "repeats"]], "gpsea.model.SampleLabels": [[13, 3, 1, "", "label"], [13, 2, 1, "", "label_summary"], [13, 3, 1, "", "meta_label"]], "gpsea.model.Sex": [[13, 5, 1, "", "FEMALE"], [13, 5, 1, "", "MALE"], [13, 5, 1, "", "UNKNOWN_SEX"], [13, 2, 1, "", "is_female"], [13, 2, 1, "", "is_male"], [13, 2, 1, "", "is_provided"], [13, 2, 1, "", "is_unknown"]], "gpsea.model.TranscriptAnnotation": [[13, 3, 1, "", "gene_id"], [13, 3, 1, "", "hgvs_cdna"], [13, 3, 1, "", "hgvsp"], [13, 3, 1, "", "is_preferred"], [13, 3, 1, "", "overlapping_exons"], [13, 3, 1, "", "protein_effect_location"], [13, 3, 1, "", "protein_id"], [13, 3, 1, "", "transcript_id"], [13, 3, 1, "", "variant_effects"]], "gpsea.model.TranscriptCoordinates": [[13, 3, 1, "", "cds_end"], [13, 3, 1, "", "cds_start"], [13, 3, 1, "", "exons"], [13, 2, 1, "", "get_cds_regions"], [13, 2, 1, "", "get_coding_base_count"], [13, 2, 1, "", "get_codon_count"], [13, 2, 1, "", "get_five_prime_utrs"], [13, 2, 1, "", "get_three_prime_utrs"], [13, 3, 1, "", "identifier"], [13, 2, 1, "", "is_coding"], [13, 3, 1, "", "is_preferred"], [13, 3, 1, "", "region"]], "gpsea.model.TranscriptInfoAware": [[13, 3, 1, "", "gene_id"], [13, 3, 1, "", "transcript_id"]], "gpsea.model.Variant": [[13, 2, 1, "", "create_variant_from_scratch"], [13, 3, 1, "", "genotypes"], [13, 3, 1, "", "tx_annotations"], [13, 3, 1, "", "variant_info"]], "gpsea.model.VariantClass": [[13, 5, 1, "", "BND"], [13, 5, 1, "", "DEL"], [13, 5, 1, "", "DUP"], [13, 5, 1, "", "INS"], [13, 5, 1, "", "INV"], [13, 5, 1, "", "MNV"], [13, 5, 1, "", "SNV"]], "gpsea.model.VariantCoordinates": [[13, 3, 1, "", "alt"], [13, 3, 1, "", "change_length"], [13, 3, 1, "", "chrom"], [13, 3, 1, "", "end"], [13, 2, 1, "", "from_vcf_literal"], [13, 2, 1, "", "from_vcf_symbolic"], [13, 2, 1, "", "is_structural"], [13, 3, 1, "", "ref"], [13, 3, 1, "", "region"], [13, 3, 1, "", "start"], [13, 3, 1, "", "variant_class"], [13, 3, 1, "", "variant_key"]], "gpsea.model.VariantEffect": [[13, 5, 1, "", "CODING_SEQUENCE_VARIANT"], [13, 5, 1, "", "DOWNSTREAM_GENE_VARIANT"], [13, 5, 1, "", "FEATURE_ELONGATION"], [13, 5, 1, "", "FEATURE_TRUNCATION"], [13, 5, 1, "", "FIVE_PRIME_UTR_VARIANT"], [13, 5, 1, "", "FRAMESHIFT_VARIANT"], [13, 5, 1, "", "INCOMPLETE_TERMINAL_CODON_VARIANT"], [13, 5, 1, "", "INFRAME_DELETION"], [13, 5, 1, "", "INFRAME_INSERTION"], [13, 5, 1, "", "INTERGENIC_VARIANT"], [13, 5, 1, "", "INTRON_VARIANT"], [13, 5, 1, "", "MATURE_MIRNA_VARIANT"], [13, 5, 1, "", "MISSENSE_VARIANT"], [13, 5, 1, "", "NMD_TRANSCRIPT_VARIANT"], [13, 5, 1, "", "NON_CODING_TRANSCRIPT_EXON_VARIANT"], [13, 5, 1, "", "NON_CODING_TRANSCRIPT_VARIANT"], [13, 5, 1, "", "PROTEIN_ALTERING_VARIANT"], [13, 5, 1, "", "REGULATORY_REGION_ABLATION"], [13, 5, 1, "", "REGULATORY_REGION_AMPLIFICATION"], [13, 5, 1, "", "REGULATORY_REGION_VARIANT"], [13, 5, 1, "", "SEQUENCE_VARIANT"], [13, 5, 1, "", "SPLICE_ACCEPTOR_VARIANT"], [13, 5, 1, "", "SPLICE_DONOR_5TH_BASE_VARIANT"], [13, 5, 1, "", "SPLICE_DONOR_REGION_VARIANT"], [13, 5, 1, "", "SPLICE_DONOR_VARIANT"], [13, 5, 1, "", "SPLICE_POLYPYRIMIDINE_TRACT_VARIANT"], [13, 5, 1, "", "SPLICE_REGION_VARIANT"], [13, 5, 1, "", "START_LOST"], [13, 5, 1, "", "START_RETAINED_VARIANT"], [13, 5, 1, "", "STOP_GAINED"], [13, 5, 1, "", "STOP_LOST"], [13, 5, 1, "", "STOP_RETAINED_VARIANT"], [13, 5, 1, "", "SYNONYMOUS_VARIANT"], [13, 5, 1, "", "TFBS_ABLATION"], [13, 5, 1, "", "TFBS_AMPLIFICATION"], [13, 5, 1, "", "TF_BINDING_SITE_VARIANT"], [13, 5, 1, "", "THREE_PRIME_UTR_VARIANT"], [13, 5, 1, "", "TRANSCRIPT_ABLATION"], [13, 5, 1, "", "TRANSCRIPT_AMPLIFICATION"], [13, 5, 1, "", "UPSTREAM_GENE_VARIANT"], [13, 3, 1, "", "curie"]], "gpsea.model.VariantInfo": [[13, 2, 1, "", "has_sv_info"], [13, 2, 1, "", "has_variant_coordinates"], [13, 2, 1, "", "is_structural"], [13, 3, 1, "", "sv_info"], [13, 3, 1, "", "variant_class"], [13, 3, 1, "", "variant_coordinates"], [13, 3, 1, "", "variant_key"]], "gpsea.model.VariantInfoAware": [[13, 3, 1, "", "variant_info"]], "gpsea.model.genome": [[14, 1, 1, "", "Contig"], [14, 1, 1, "", "GenomeBuild"], [14, 1, 1, "", "GenomeBuildIdentifier"], [14, 1, 1, "", "GenomicRegion"], [14, 1, 1, "", "Region"], [14, 1, 1, "", "Strand"], [14, 1, 1, "", "Stranded"], [14, 1, 1, "", "Transposable"], [14, 4, 1, "", "transpose_coordinate"]], "gpsea.model.genome.Contig": [[14, 3, 1, "", "genbank_acc"], [14, 3, 1, "", "name"], [14, 3, 1, "", "refseq_name"], [14, 3, 1, "", "ucsc_name"]], "gpsea.model.genome.GenomeBuild": [[14, 2, 1, "", "contig_by_name"], [14, 3, 1, "", "contigs"], [14, 3, 1, "", "genome_build_id"], [14, 3, 1, "", "identifier"]], "gpsea.model.genome.GenomeBuildIdentifier": [[14, 3, 1, "", "identifier"], [14, 3, 1, "", "major_assembly"], [14, 3, 1, "", "patch"]], "gpsea.model.genome.GenomicRegion": [[14, 2, 1, "", "contains"], [14, 3, 1, "", "contig"], [14, 2, 1, "", "distance_to"], [14, 2, 1, "", "end_on_strand"], [14, 2, 1, "", "overlaps_with"], [14, 2, 1, "", "start_on_strand"], [14, 3, 1, "", "strand"], [14, 2, 1, "", "with_strand"]], "gpsea.model.genome.Region": [[14, 2, 1, "", "contains"], [14, 2, 1, "", "contains_pos"], [14, 2, 1, "", "distance_to"], [14, 3, 1, "", "end"], [14, 2, 1, "", "is_empty"], [14, 2, 1, "", "overlaps_with"], [14, 3, 1, "", "start"]], "gpsea.model.genome.Strand": [[14, 5, 1, "", "NEGATIVE"], [14, 5, 1, "", "POSITIVE"], [14, 2, 1, "", "is_negative"], [14, 2, 1, "", "is_positive"], [14, 2, 1, "", "opposite"], [14, 3, 1, "", "symbol"]], "gpsea.model.genome.Stranded": [[14, 3, 1, "", "strand"]], "gpsea.model.genome.Transposable": [[14, 2, 1, "", "to_negative_strand"], [14, 2, 1, "", "to_opposite_strand"], [14, 2, 1, "", "to_positive_strand"], [14, 2, 1, "", "with_strand"]], "gpsea.preprocessing": [[15, 1, 1, "", "Auditor"], [15, 1, 1, "", "CohortCreator"], [15, 1, 1, "", "DataSanityIssue"], [15, 1, 1, "", "DefaultImpreciseSvFunctionalAnnotator"], [15, 1, 1, "", "FunctionalAnnotator"], [15, 1, 1, "", "GeneCoordinateService"], [15, 1, 1, "", "ImpreciseSvFunctionalAnnotator"], [15, 1, 1, "", "Level"], [15, 1, 1, "", "Notepad"], [15, 1, 1, "", "NotepadTree"], [15, 1, 1, "", "PatientCreator"], [15, 1, 1, "", "PhenopacketPatientCreator"], [15, 1, 1, "", "PhenopacketVariantCoordinateFinder"], [15, 1, 1, "", "PhenotypeCreator"], [15, 1, 1, "", "ProtCachingMetadataService"], [15, 1, 1, "", "ProteinAnnotationCache"], [15, 1, 1, "", "ProteinMetadataService"], [15, 1, 1, "", "TranscriptCoordinateService"], [15, 1, 1, "", "UniprotProteinMetadataService"], [15, 1, 1, "", "VVHgvsVariantCoordinateFinder"], [15, 1, 1, "", "VVMultiCoordinateService"], [15, 1, 1, "", "VarCachingFunctionalAnnotator"], [15, 1, 1, "", "VariantAnnotationCache"], [15, 1, 1, "", "VariantCoordinateFinder"], [15, 1, 1, "", "VepFunctionalAnnotator"], [15, 4, 1, "", "configure_caching_cohort_creator"], [15, 4, 1, "", "configure_caching_patient_creator"], [15, 4, 1, "", "configure_cohort_creator"], [15, 4, 1, "", "configure_default_protein_metadata_service"], [15, 4, 1, "", "configure_patient_creator"], [15, 4, 1, "", "configure_protein_metadata_service"], [15, 4, 1, "", "load_phenopacket_files"], [15, 4, 1, "", "load_phenopacket_folder"], [15, 4, 1, "", "load_phenopackets"]], "gpsea.preprocessing.Auditor": [[15, 2, 1, "", "prepare_notepad"], [15, 2, 1, "", "process"]], "gpsea.preprocessing.CohortCreator": [[15, 2, 1, "", "process"]], "gpsea.preprocessing.DataSanityIssue": [[15, 3, 1, "", "level"], [15, 3, 1, "", "message"], [15, 3, 1, "", "solution"]], "gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator": [[15, 2, 1, "", "annotate"]], "gpsea.preprocessing.FunctionalAnnotator": [[15, 2, 1, "", "annotate"]], "gpsea.preprocessing.GeneCoordinateService": [[15, 2, 1, "", "fetch_for_gene"]], "gpsea.preprocessing.ImpreciseSvFunctionalAnnotator": [[15, 2, 1, "", "annotate"]], "gpsea.preprocessing.Level": [[15, 5, 1, "", "ERROR"], [15, 5, 1, "", "WARN"]], "gpsea.preprocessing.Notepad": [[15, 2, 1, "", "add_error"], [15, 2, 1, "", "add_issue"], [15, 2, 1, "", "add_subsection"], [15, 2, 1, "", "add_warning"], [15, 2, 1, "", "error_count"], [15, 2, 1, "", "errors"], [15, 3, 1, "", "issues"], [15, 3, 1, "", "label"], [15, 2, 1, "", "warning_count"], [15, 2, 1, "", "warnings"]], "gpsea.preprocessing.NotepadTree": [[15, 2, 1, "", "add_subsection"], [15, 3, 1, "", "children"], [15, 2, 1, "", "has_errors"], [15, 2, 1, "", "has_errors_or_warnings"], [15, 2, 1, "", "has_warnings"], [15, 2, 1, "", "iterate_nodes"], [15, 3, 1, "", "level"], [15, 2, 1, "", "visit"]], "gpsea.preprocessing.PhenopacketPatientCreator": [[15, 2, 1, "", "process"]], "gpsea.preprocessing.PhenopacketVariantCoordinateFinder": [[15, 2, 1, "", "find_coordinates"]], "gpsea.preprocessing.PhenotypeCreator": [[15, 2, 1, "", "process"]], "gpsea.preprocessing.ProtCachingMetadataService": [[15, 2, 1, "", "annotate"]], "gpsea.preprocessing.ProteinAnnotationCache": [[15, 2, 1, "", "get_annotations"], [15, 2, 1, "", "store_annotations"]], "gpsea.preprocessing.ProteinMetadataService": [[15, 2, 1, "", "annotate"]], "gpsea.preprocessing.TranscriptCoordinateService": [[15, 2, 1, "", "fetch"]], "gpsea.preprocessing.UniprotProteinMetadataService": [[15, 2, 1, "", "annotate"], [15, 2, 1, "", "parse_uniprot_json"]], "gpsea.preprocessing.VVHgvsVariantCoordinateFinder": [[15, 2, 1, "", "find_coordinates"]], "gpsea.preprocessing.VVMultiCoordinateService": [[15, 2, 1, "", "fetch"], [15, 2, 1, "", "fetch_for_gene"], [15, 2, 1, "", "get_response"], [15, 2, 1, "", "parse_multiple"], [15, 2, 1, "", "parse_response"]], "gpsea.preprocessing.VarCachingFunctionalAnnotator": [[15, 2, 1, "", "annotate"], [15, 2, 1, "", "with_cache_folder"]], "gpsea.preprocessing.VariantAnnotationCache": [[15, 2, 1, "", "get_annotations"], [15, 2, 1, "", "store_annotations"]], "gpsea.preprocessing.VariantCoordinateFinder": [[15, 2, 1, "", "find_coordinates"]], "gpsea.preprocessing.VepFunctionalAnnotator": [[15, 5, 1, "", "NONCODING_EFFECTS"], [15, 2, 1, "", "annotate"], [15, 2, 1, "", "fetch_response"], [15, 2, 1, "", "format_coordinates_for_vep_query"], [15, 2, 1, "", "process_response"]], "gpsea.view": [[16, 1, 1, "", "CohortViewable"], [16, 1, 1, "", "DiseaseViewable"], [16, 1, 1, "", "MtcStatsViewer"], [16, 1, 1, "", "ProteinViewable"], [16, 1, 1, "", "ProteinVisualizable"], [16, 1, 1, "", "ProteinVisualizer"], [16, 1, 1, "", "VariantFormatter"], [16, 1, 1, "", "VariantTranscriptVisualizer"]], "gpsea.view.CohortViewable": [[16, 2, 1, "", "process"]], "gpsea.view.DiseaseViewable": [[16, 2, 1, "", "process"]], "gpsea.view.MtcStatsViewer": [[16, 2, 1, "", "process"]], "gpsea.view.ProteinViewable": [[16, 2, 1, "", "process"]], "gpsea.view.ProteinVisualizable": [[16, 3, 1, "", "marker_counts"], [16, 3, 1, "", "protein_feature_ends"], [16, 3, 1, "", "protein_feature_names"], [16, 3, 1, "", "protein_feature_starts"], [16, 3, 1, "", "protein_feature_types"], [16, 3, 1, "", "protein_id"], [16, 3, 1, "", "protein_length"], [16, 3, 1, "", "protein_metadata"], [16, 3, 1, "", "transcript_coordinates"], [16, 3, 1, "", "transcript_id"], [16, 3, 1, "", "variant_effects"], [16, 3, 1, "", "variant_locations"], [16, 3, 1, "", "variant_locations_counted_absolute"]], "gpsea.view.ProteinVisualizer": [[16, 2, 1, "", "draw_fig"], [16, 2, 1, "", "draw_protein_diagram"]], "gpsea.view.VariantFormatter": [[16, 2, 1, "", "format_as_string"]], "gpsea.view.VariantTranscriptVisualizer": [[16, 2, 1, "", "draw_variants"]]}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "property", "Python property"], "4": ["py", "function", "Python function"], "5": ["py", "attribute", "Python attribute"], "6": ["py", "data", "Python data"]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:property", "4": "py:function", "5": "py:attribute", "6": "py:data"}, "terms": {"": [4, 6, 7, 9, 12, 13, 14, 15, 19, 20, 25, 27, 28, 30, 31], "0": [2, 3, 5, 6, 8, 13, 14, 15, 16, 20, 22, 25, 27, 28, 30, 31], "000000": 3, "0000002": 24, "0000047": 22, "0000078": 24, "0000079": 24, "0000080": 24, "0000098": 22, "0000118": [2, 24, 28], "0000119": 24, "0000152": 24, "0000223": 24, "0000234": 24, "0000252": 22, "0000256": 22, "0000271": 24, "0000316": 22, "0000356": 24, "0000359": 24, "0000370": 24, "0000377": 22, "0000407": 31, "0000464": 24, "0000478": 24, "0000486": 31, "0000496": 24, "0000504": 24, "0000517": 30, "0000539": 24, "0000553": 24, "0000562": 20, "0000591": 24, "0000598": 24, "0000707": [24, 28], "0000759": 24, "0000769": 24, "0000777": 24, "0000818": 24, "0000820": 24, "0000828": 24, "0000834": 24, "0000847": 24, "0000864": 24, "0000924": 24, "0000927": 24, "0000929": 24, "0000951": 24, "0000971": 24, "0001060": 13, "0001115": 28, "0001155": 24, "0001166": [3, 7], "0001191": 20, "0001194": 24, "0001197": 24, "0001199": [20, 31], "0001249": 22, "0001250": 13, "0001256": 22, "0001263": 22, "0001311": 24, "0001392": 24, "0001437": 24, "0001438": 24, "0001446": 24, "0001454": 24, "0001507": 24, "0001511": 22, "0001518": 22, "0001537": 6, "0001551": 24, "0001560": 24, "0001566": 13, "0001567": 13, "0001574": [13, 24], "0001575": 13, "0001578": 13, "0001580": 13, "0001583": 13, "0001587": 13, "0001589": 13, "0001595": 24, "0001597": 24, "0001608": 24, "0001619": 13, "0001620": 13, "0001621": 13, "0001623": 13, "0001624": 13, "0001626": [13, 24], "0001627": [13, 20, 22, 24, 31], "0001628": 13, "0001629": [20, 31], "0001630": 13, "0001631": [13, 20, 31], "0001632": 13, "0001643": [20, 31], "0001671": 31, "0001678": [20, 31], "0001684": [20, 31], "0001713": 20, "0001732": 24, "0001743": 24, "0001760": 24, "0001782": 13, "0001787": [13, 24], "0001792": 13, "0001818": 13, "0001819": 13, "0001821": 13, "0001822": 13, "0001871": 24, "0001872": 24, "0001877": 24, "0001881": 24, "0001889": 13, "0001891": 13, "0001892": [13, 24], "0001893": 13, "0001894": 13, "0001895": 13, "0001906": 13, "0001907": 13, "0001928": 24, "0001939": 24, "0001965": 24, "0001977": 24, "0002012": [13, 24], "0002019": 13, "0002086": 24, "0002092": [20, 31], "0002169": 13, "0002170": 13, "0002187": 22, "0002270": 24, "0002339": 28, "0002342": 22, "00024192670136836825": 31, "000242": 31, "0002585": 24, "0002597": 24, "0002664": 24, "0002715": 24, "0002733": 24, "0002793": 24, "0002795": 24, "0002813": 24, "0002814": 24, "0002817": 24, "0002916": 24, "0002926": 24, "0002973": 24, "0002981": 24, "0002984": [20, 31], "0003011": [3, 24], "0003022": [20, 31], "0003110": 24, "0003111": 24, "0003117": 24, "0003220": 24, "0003254": 24, "0003256": 24, "0003498": 22, "0003549": 24, "0003839": 24, "0003974": [20, 31], "0004298": 24, "0004312": 24, "0004322": 22, "0004323": 24, "0004328": 24, "0004329": 24, "0004332": 24, "0004337": 24, "00043478260869565214": 20, "0004352": 24, "0004353": 24, "0004356": 24, "0004360": 24, "0004361": 24, "0004370": 24, "0004371": 24, "0004378": 24, "0004404": 24, "0004408": 24, "0004930": 24, "0005": 2, "0005150": [20, 31], "0005368": 24, "0005561": 24, "0005792": [20, 31], "0005922": 22, "0006476": 24, "0006490": 24, "0006500": 24, "0006529": 24, "0006707": 24, "0006889": 22, "0007686": 24, "000781": 3, "0008047": 24, "0009124": 24, "0009127": 24, "000955": 20, "0009552459156234353": 20, "0009777": [20, 31], "0009778": [20, 31], "0009809": 24, "0009810": 24, "0010438": 20, "0010696": 28, "0010864": 22, "0010876": 24, "0010881": 24, "0010932": 24, "0010935": 24, "0010936": 24, "0010938": 22, "0010948": 24, "0010968": 24, "0010969": 24, "0010974": 24, "0010977": 24, "0010978": 24, "0010987": 24, "0010991": 24, "0011014": 24, "0011017": 24, "0011018": 24, "0011019": 24, "0011024": 24, "0011025": 24, "0011028": 24, "0011032": 24, "0011043": 24, "0011111": 24, "0011121": 24, "0011122": 24, "0011138": 24, "0011277": 24, "0011342": 22, "0011343": 22, "0011344": 22, "0011409": 24, "0011442": 24, "0011446": 24, "0011620": 24, "0011623": [20, 31], "0011730": 24, "0011732": 24, "0011733": 24, "0011766": 24, "0011767": 24, "0011772": 24, "0011804": 24, "0011805": 24, "0011842": 24, "0011843": 24, "0011869": 24, "0011875": 24, "0011884": 24, "0011893": 24, "0011902": 24, "0011994": 31, "0012029": 24, "0012093": 24, "0012099": 24, "0012103": 24, "0012111": 24, "0012129": 24, "0012130": 24, "0012131": 24, "0012135": 24, "0012143": 24, "0012145": 24, "0012210": 31, "0012243": 24, "0012252": 24, "0012261": 24, "0012285": 24, "0012337": 24, "0012338": 24, "0012345": 24, "0012372": [24, 31], "0012373": 24, "0012379": 24, "0012415": 24, "0012443": [8, 31], "0012447": 24, "0012503": 24, "0012535": 24, "0012591": 24, "0012614": 24, "0012632": 24, "0012638": [3, 24], "0012639": 24, "0012640": 24, "0012647": 24, "0012654": 24, "0012680": 24, "0012681": 24, "0012688": 24, "0012700": 24, "0012718": 24, "0012722": [20, 31], "0012757": 24, "0012769": 24, "0012772": 24, "0015384615384615387": 31, "0020047": 24, "0020054": 24, "0020058": 24, "0020061": 24, "0020155": 24, "0020169": 24, "0025015": 24, "0025021": 24, "0025031": 24, "0025032": 24, "0025033": 24, "0025065": 24, "0025142": 24, "0025155": 24, "0025276": 24, "0025354": 24, "0025427": 24, "0025429": 24, "0025443": 24, "0025454": 24, "0025456": 24, "0025461": 24, "0025463": 24, "0025546": 24, "0025590": 24, "0025640": 24, "0025668": 24, "0025669": 24, "0025688": 24, "0030085": 24, "0030163": 24, "0030272": 24, "0030338": 24, "0030352": 24, "0030453": 24, "0030680": 24, "0030684": 24, "0030687": 24, "0030800": 24, "0030829": 24, "0030860": 24, "0030872": 24, "0030875": 24, "0030878": 24, "0030956": 24, "0030972": 24, "0031071": 24, "0031072": 24, "0031073": 24, "0031093": 24, "0031094": 24, "0031097": 24, "0031099": 24, "0031101": 24, "0031285": 24, "0031331": 24, "0031340": 24, "0031377": 24, "0031383": 24, "0031389": 24, "0031409": 24, "0031411": 24, "0031416": 24, "0031427": 24, "0031476": 24, "0031508": 24, "0031546": [20, 31], "0031550": 24, "0031602": 24, "0031653": 24, "0031657": 24, "0031685": 24, "0031703": 24, "0031704": 24, "0031818": 24, "0031850": 24, "0031871": 24, "0031884": 24, "0031910": 24, "0031982": 28, "0032120": 24, "0032180": 24, "0032226": 24, "0032243": 24, "0032245": 24, "0032251": 24, "0032314": 24, "0032367": 24, "0032481": 24, "0032485": 24, "0032488": 24, "0032943": 24, "0033012": 24, "0033013": 24, "0033072": 24, "0033127": 24, "0033170": 24, "0033334": 24, "0033335": 24, "0033354": 24, "0033358": 24, "0033796": 24, "0033799": 24, "0034190": 24, "0034200": 24, "0034251": 24, "0034263": 24, "0034370": 24, "0034430": 24, "0034442": 24, "0034482": 24, "0034552": 24, "0034644": 24, "0034684": 24, "0034698": 24, "0034737": 24, "0034858": 24, "0034899": 24, "0034915": 24, "0034916": 24, "0034977": 24, "0036231884057971015": 20, "003695652173913043": 20, "003713671516019927": 31, "0040064": 24, "0040068": 24, "0040069": 24, "0040070": 24, "0040084": 24, "0040085": 24, "0040127": 24, "0040172": 24, "0040203": 24, "0040207": 24, "0040214": 24, "0040224": 24, "0040231": 24, "00411": 31, "00411275392326226": 31, "0045026": 24, "0045027": 24, "0045081": 24, "005628510156750059": 20, "01": [20, 27, 28, 30, 31], "0100016": 24, "0100022": 24, "0100491": 24, "0100508": 24, "0100530": 24, "0100536": 24, "0100685": 24, "0100705": 24, "0100763": 24, "0100765": 24, "0100766": 24, "0100767": 24, "0100886": 24, "0100887": 24, "012074957610483744": 31, "01307692307692308": 31, "01349527665317139": 20, "015398550724637682": 20, "0191369345329502": 20, "020299": 3, "021044138590779585": 31, "02560162393963452": 20, "03252032520325203": 31, "034": 31, "038236617183985605": 20, "0410008": 24, "0410009": 24, "0410014": 24, "0430071": 24, "05": [3, 20, 28, 31], "0500012": 24, "0500015": 24, "0500016": 24, "0500017": 24, "0500018": 24, "0500019": 24, "0500020": 24, "0500114": 24, "0500117": 24, "0500166": 24, "0500183": 24, "0500238": 24, "06": 31, "062175372424826694": 20, "06544319142266644": 31, "06932119159387057": 31, "07": [20, 27, 28, 30, 31], "07440639019586388": 20, "08259109311740892": 31, "08470708701170182": 31, "09214092140921408": 31, "1": [2, 3, 5, 6, 8, 13, 14, 15, 20, 22, 24, 25, 27, 28, 30, 31], "10": [2, 3, 6, 8, 13, 15, 16, 20, 22, 28, 31], "100": [3, 20, 28, 31], "1000": 13, "1000029": [6, 13, 30], "1001": [13, 25], "1003": 25, "101": 13, "106_107insc": 20, "11": [3, 13, 14, 20, 28], "116": 20, "12": 20, "120": [13, 31], "1221c": 20, "123": 8, "12_114366360_114366360_c_t": 20, "12_114385474_114385474_a_g": 20, "12_114385521_114385521_c_t": 20, "12_114385563_114385563_g_a": 20, "12_114398656_114398656_c_cg": 20, "12_114398675_114398675_g_t": 20, "12_114398682_114398682_c_cg": 20, "12_114399514_114399514_a_c": 20, "12_114401830_114401830_c_t": 20, "12_114403792_114403792_c_cg": 20, "13": [2, 3, 20, 31], "14": [20, 31], "1424522583078588": 20, "142900": [20, 30], "1435th": 30, "1440020479198931": 31, "15": [3, 14, 20, 31], "156": [20, 31], "15811357916621074": 20, "16": [3, 31], "1687456462342971": 20, "17": [20, 31], "18": [20, 27, 30, 31], "19": [20, 27, 30, 31], "1_8358231_8358231_t_c": 30, "1d": 16, "1st": [6, 13], "1y": 20, "2": [2, 3, 4, 6, 7, 8, 9, 13, 14, 15, 20, 22, 24, 25, 28, 30, 31], "20": [6, 13, 20, 28, 30, 31], "2011": 22, "2015": 18, "2018": 31, "2022": [21, 22], "205": 13, "206": 13, "21": 31, "22": [2, 20, 31], "22_10001_20000_inv": [6, 13], "23": [24, 31], "238g": 20, "24": [20, 31], "243": [20, 31], "25": [20, 31], "253a": 15, "26": 20, "260": [20, 31], "2690764879148444": 20, "27": [20, 31], "27bp": 13, "28": 31, "2868675985983051": 20, "2d": 2, "2nd": 6, "2t": 20, "2x2": [4, 20, 31], "2x3": 4, "3": [2, 4, 8, 13, 15, 20, 24, 25, 28, 30, 31], "30": [15, 20, 31], "3000013": 24, "3000036": 24, "3000050": 24, "301": 30, "30552424": 20, "31": [20, 31], "32": [20, 31], "321681": 13, "321682": 13, "321887": 13, "34": 31, "348081479150901e": 31, "348081479150902e": 31, "35": 20, "36": [20, 28, 31], "3603": [13, 15], "361t": 20, "36303223": 21, "37": [20, 31], "38": [20, 31], "38336121": 20, "39": 20, "4": [6, 8, 13, 15, 20, 31], "40": [20, 31], "4000056": 24, "4000183": 24, "400dup": 20, "408c": 20, "41": [20, 31], "42": [16, 20, 31], "426dup": 20, "42857142857142855": 20, "43": [20, 31], "4304a": 30, "432292015291845e": 31, "44": [20, 31], "45": [20, 31], "450": 28, "4666666666666667": 31, "48": 20, "48700997145537483": 31, "5": [2, 3, 6, 13, 15, 20, 22, 30, 31], "50": [6, 13, 20, 28, 30, 31], "50bp": 6, "51": [20, 31], "52": [20, 31], "53": 31, "55": 31, "57": 31, "5714285714285713": 20, "59": [20, 31], "5x": 31, "6": [3, 13, 20, 21, 25, 30, 31], "60": [20, 31], "6000062": 24, "6000231": 24, "6000489": 24, "6000673": 24, "6190936213143254e": 20, "62": 20, "63": 31, "64": [28, 31], "65": 31, "6623376623376622": 20, "668c": 20, "67": 20, "6899307928951143": 31, "69": 31, "7": [2, 3, 6, 13, 20, 28, 31], "71": [20, 31], "710g": 20, "72": 20, "75": 31, "755": 20, "7703831604944444": 31, "7735491022101784": 20, "787g": 20, "8": [20, 28, 31], "8095238095238093": 20, "81": 3, "833333333333336": 31, "84": 31, "85": 20, "89": 31, "9": [3, 20, 31], "90": 28, "95": [20, 28], "97": 31, "9876g": 13, "99": [28, 31], "A": [3, 5, 7, 8, 13, 15, 16, 18, 20, 25, 28, 30], "AND": [6, 30, 31], "AT": 25, "As": [5, 7, 30], "By": [16, 20, 28], "For": [2, 4, 6, 8, 12, 13, 14, 15, 20, 21, 25, 27, 28, 30, 31], "IN": 15, "INS": 13, "If": [2, 3, 10, 13, 20, 22, 28], "In": [2, 4, 6, 8, 15, 18, 20, 21, 22, 27, 28, 30, 31], "It": [3, 4, 20, 28, 30], "NO": [5, 7], "NOT": [2, 13, 14, 30], "No": [3, 5, 14, 20, 27, 30, 31], "Not": 13, "OR": [6, 30, 31], "On": [3, 7, 27], "One": [22, 30], "That": 19, "The": [2, 3, 4, 5, 6, 7, 8, 10, 12, 13, 14, 15, 16, 18, 19, 22, 24, 25, 27, 28, 30, 31], "Then": [13, 19, 20, 30], "There": [13, 20, 31], "To": [15, 19, 20, 28, 30, 31], "_": [20, 30, 31], "__eq__": 6, "__hash__": 6, "__init__": [13, 28], "__repr__": 6, "__str__": 6, "_api": 2, "_missing_implies_phenotype_exclud": 7, "_t": 13, "_term_id": 2, "aat": 25, "abbrevi": [16, 28], "abdomen": 24, "abdomin": 24, "abl": 6, "abnorm": [3, 8, 18, 20, 24, 25, 28, 30, 31], "about": [2, 13, 15, 16, 19, 20, 22, 23, 28, 30], "abov": [6, 19, 20, 22, 24, 28, 31], "absenc": [3, 7, 20, 30], "absent": [20, 31], "abstract": [2, 3, 4, 5, 6, 7, 9, 13, 14, 15], "accept": [13, 15], "access": [6, 13, 14, 15, 19, 20, 28, 31], "accommod": 24, "accord": [2, 3, 20, 24, 30, 31], "account": 28, "acetylaspart": 24, "acid": [13, 24], "across": [2, 16], "actg": 25, "activ": 24, "ad": [15, 22], "adapt": [8, 24], "add": [15, 19, 30], "add_error": 15, "add_issu": 15, "add_subsect": 15, "add_warn": 15, "addit": [4, 19, 20], "addition": 20, "address": 3, "adenosin": 24, "adiponectin": 24, "adipos": 24, "adjac": 14, "adjust": 22, "adnexa": 24, "adren": 24, "adrenocorticotropin": 24, "advanc": 2, "advantag": [20, 28, 30, 31], "affect": [6, 13, 20, 21, 25, 30], "affected_exon": 13, "after": [3, 15, 19], "ag": [13, 22], "against": 21, "al": [8, 18, 21, 22, 31], "ala143argfster40": 20, "albeit": 15, "albumin": 24, "aldosteron": 24, "align": 13, "all": [2, 3, 5, 6, 7, 8, 13, 15, 19, 20, 24, 25, 27, 31], "all_count": 3, "all_diseas": 13, "all_pati": 13, "all_phenotyp": 13, "all_transcript_id": 13, "all_vari": 13, "allel": [6, 13, 18, 21, 30, 31], "allele_count": 6, "allelecount": [5, 6], "allianc": 18, "allow": [3, 28, 30], "allow_nan": 12, "alon": 28, "along": [12, 27, 30], "alpha": [24, 28, 31], "alreadi": 27, "also": [6, 20, 24, 27, 30], "alt": [6, 13, 25], "altern": [6, 13, 25, 28, 30, 31], "alwai": [6, 14], "amaz": 27, "amino": [13, 24], "aminoacid": [13, 14, 16], "amniot": 24, "among": 13, "amount": 30, "amyloid": 24, "an": [2, 3, 4, 5, 6, 7, 8, 9, 13, 14, 15, 16, 18, 19, 20, 21, 27, 28, 30, 31], "analys": [27, 30], "analysi": [0, 15, 18, 21, 26, 27, 28, 30], "analyz": [0, 3, 8, 16, 20, 31], "anatomi": 20, "ancestor": [7, 20, 25, 30], "aneurysm": 25, "angiotensin": 24, "ani": [2, 5, 6, 8, 12, 13, 15, 22, 28, 30, 31], "ank": [6, 13], "ankyrin": 13, "annot": [2, 6, 7, 13, 15, 20, 25, 27, 28, 30, 31], "anomali": [22, 31], "anoth": [7, 14, 20], "answer": [5, 6, 7], "anterior": [24, 30], "antimullerian": 24, "anu": 24, "api": [15, 16, 18, 27, 30, 31], "apoptosi": 24, "appear": 24, "appli": [3, 6, 9, 25, 28], "applic": 20, "apply_predicates_on_pati": [1, 3], "approach": 28, "appropri": [20, 31], "ar": [2, 3, 6, 8, 13, 14, 15, 16, 18, 19, 20, 21, 22, 24, 28, 30, 31], "arachnodactyli": [3, 7], "arbitrari": [12, 14], "areolar": 24, "arg": 12, "arg134profster49": 20, "arg237gln": 20, "arginin": 30, "argument": [8, 10, 15, 16, 28, 30], "arithmet": 14, "arm": 24, "aromatas": 24, "around": [4, 9], "arrai": [3, 4, 16], "arrang": 24, "arrhythmia": 20, "arriv": 28, "arteri": [20, 31], "arteriosu": [20, 31], "ask": [2, 18], "assembl": 15, "assembli": 14, "assert": [13, 14], "assess": [3, 13, 20, 22], "assign": [3, 5, 6, 7, 8, 22, 30, 31], "associ": [3, 8, 15, 18, 20, 21, 30], "assum": [13, 14, 15, 28], "assume_karyotypic_sex": 15, "assumpt": 28, "atrial": [20, 24, 31], "atrioventricular": [20, 31], "atrophi": 31, "atrophin": 31, "attribut": [3, 7, 13, 18], "audit": 15, "auditor": [0, 15], "author": 31, "automat": 31, "autonom": 24, "autosom": [6, 20, 26, 30, 31], "autosomal_domin": [6, 30, 31], "autosomal_recess": [6, 30], "avail": [2, 13, 15, 18, 20, 30, 31], "avoid": [16, 24], "ax": [8, 16, 20, 31], "axi": [6, 24, 30, 31], "b": [13, 14, 28], "back": [10, 15], "background": 18, "bandwidth": 27, "base": [2, 3, 4, 5, 6, 7, 8, 9, 12, 13, 14, 15, 16, 20, 22, 24, 25, 30, 31], "basemultiphenotypeanalysisresult": 3, "basic": [13, 14, 30], "batteri": 7, "becaus": [13, 20, 21, 22, 24, 25, 28, 30, 31], "been": [13, 20, 28, 31], "befor": [28, 30], "behavior": 18, "behind": 20, "being": [2, 6, 14, 15, 16, 20, 22, 28, 30], "belong": [5, 13], "below": [6, 25], "benjamini": [20, 28, 31], "best": 18, "beta": 24, "better": 30, "between": [3, 6, 8, 13, 14, 16, 18, 20, 22, 25, 28, 30, 31], "biallel": [6, 30], "biallelic_alt": 30, "bin": [3, 5, 6, 20, 30, 31], "biolog": [13, 14, 31], "biologi": 18, "biomark": 24, "biomed": 20, "biopterin": 24, "birth": [13, 24], "bit": 30, "bleed": 24, "block": [20, 31], "blood": 24, "bnd": [6, 13], "bodi": 24, "boilerpl": 27, "bone": 24, "bonferroni": [3, 20, 28], "bool": [2, 6, 7, 13, 14, 15], "boolean": [6, 13], "boolean_pred": [5, 6, 31], "borderlin": 22, "both": [4, 13, 15, 20, 22, 28, 30], "bound": 14, "boundari": [13, 14, 15], "box": [8, 27, 31], "bp": [6, 13, 25, 30], "brain": [8, 31], "branch": [19, 28], "breakend": 13, "breakpoint": [6, 13, 15], "breast": 24, "breath": 24, "bridg": 26, "brief": 18, "briefli": [30, 31], "bring": [20, 24], "bronchu": 24, "buccal": 24, "bug": 18, "build": [6, 13, 14, 15, 20, 31], "bunch": [27, 30], "burden": [2, 3, 20, 28], "c": [5, 13, 15, 18, 20, 25, 27, 30, 31], "c17998": 13, "c46112": 13, "c46113": 13, "cach": [10, 15, 27], "cache_dir": 15, "cache_env": [0, 10], "calcium": 24, "calcul": [4, 8, 9, 13, 14, 31], "calf": 24, "call": [12, 13, 15, 24, 27, 28], "callabl": [8, 15], "caller": 15, "can": [2, 3, 4, 5, 6, 7, 8, 13, 14, 15, 16, 19, 20, 22, 23, 24, 27, 28, 30, 31], "cannot": [5, 6, 8, 13, 15, 22, 30, 31], "canon": 13, "cap": 28, "capabl": 30, "carboxyl": 24, "cardiac": [20, 24, 31], "cardiomyocyt": 24, "cardiovascular": 24, "carpal": 20, "carri": [21, 31], "cart": 24, "cascad": 24, "case": [2, 6, 8, 15, 20, 27, 31], "cat": 30, "cat_id": [5, 8, 31], "cataract": 28, "catecholamin": 24, "categor": [1, 2, 5, 6, 7, 30], "categori": [3, 5, 6, 7, 8, 13, 20, 22, 30, 31], "catheter": 24, "caudat": 28, "caus": 18, "caviti": 24, "ccc": 25, "cct": 13, "cd": [13, 19], "cds_end": 13, "cds_start": 13, "cell": [24, 28], "cellular": 24, "central": 24, "cerebellar": 31, "certain": [6, 15, 30], "cg": 25, "chanc": [20, 28, 31], "chang": [6, 13, 30, 31], "change_length": [6, 13, 30, 31], "charact": 13, "character": 20, "characterist": 13, "check": [3, 4, 6, 13, 14, 15, 20, 28, 30], "check_circular": 12, "checkout": 19, "child": 28, "children": [15, 24, 28], "choic": 13, "choos": [6, 15, 20, 30, 31], "chosen": [13, 20, 28, 31], "chr1": [13, 14, 25], "chr5": 13, "chrom": [13, 25], "chromosom": [6, 13, 21, 24, 30], "chromosomal_delet": 30, "chrx": 14, "cilium": 24, "circul": 24, "cl": 12, "class": [2, 3, 4, 5, 6, 7, 8, 9, 12, 13, 14, 15, 16, 20, 27, 28, 30], "clearanc": 24, "clearli": 30, "clinic": [13, 20, 22, 28, 30], "clinvar": 20, "clone": 19, "cluster": 30, "cm000663": 14, "coagul": 24, "code": [13, 15, 19, 20, 27, 31], "coding_sequence_vari": 13, "codon": [13, 18, 30], "cohort": [0, 3, 7, 8, 11, 13, 15, 16, 18, 26, 28], "cohort_cr": [15, 20, 27, 30, 31], "cohort_nam": 20, "cohortcr": [0, 15, 27, 31], "cohortview": [0, 16, 20], "collect": [6, 7, 9, 13, 15, 16, 18, 20, 27], "color": 8, "column": [2, 3, 8], "com": 19, "combin": [13, 20, 30, 31], "come": [14, 31], "command": 19, "common": [5, 20, 21], "commonli": [14, 28], "compact": [6, 13], "compar": [2, 3, 4, 6, 21, 30], "compare_genotype_vs_phenotyp": [3, 20, 31], "compare_genotype_vs_phenotype_scor": [8, 31], "comparison": 21, "compat": 6, "complain": 15, "complet": [15, 28], "complex": 28, "compon": [20, 30, 31], "composit": 24, "compound": [6, 21, 24, 30], "comprehens": 18, "comput": [3, 8, 9, 15, 20, 31], "compute_pv": [4, 9], "concentr": 24, "concept": [18, 25, 28], "conceptu": 13, "concern": 18, "conclus": 28, "condens": 24, "condit": [28, 31], "conduct": [20, 31], "config": 0, "configur": [6, 15, 30], "configure_caching_cohort_cr": [0, 15, 20, 27, 30, 31], "configure_caching_patient_cr": [0, 15], "configure_cohort_cr": [0, 15], "configure_default_protein_metadata_servic": [0, 15], "configure_ontology_stor": [20, 27, 28, 30, 31], "configure_patient_cr": [0, 15], "configure_phenopacket_registri": [20, 27, 30, 31], "configure_protein_metadata_servic": [0, 15, 20], "congenit": [20, 31], "connect": 24, "consequ": 13, "conserv": 13, "consid": [4, 6, 13, 14, 15, 22, 24, 30, 31], "consider": [24, 30], "consist": [3, 13, 14, 27, 31], "consortium": 31, "constitut": 24, "construct": [15, 30], "constructor": [2, 13, 28, 30], "consult": [6, 10], "contain": [8, 13, 14, 15, 16, 20, 24], "contains_po": 14, "context": [7, 18, 28, 31], "contig": [13, 14], "contig_by_nam": [13, 14, 25], "contigu": 14, "conting": [2, 3, 4, 31], "contingency_t": 31, "control": [20, 28, 31], "conveni": [7, 15, 27], "convers": 25, "convert": [15, 30], "coordin": [6, 13, 14, 15, 16, 25], "copi": 21, "cord": 24, "corneal": 24, "corpuscular": 24, "correct": [3, 20, 24, 26, 31], "corrected_pv": 3, "correctli": 15, "correl": [0, 2, 3, 18, 20, 24, 26, 28, 30, 31], "correspond": [2, 3, 7, 8, 13, 14, 15, 20, 22, 25, 28, 30, 31], "corrrespond": 2, "cortisol": 24, "could": [3, 12, 31], "count": [2, 3, 4, 6, 7, 8, 13, 15, 20, 22, 23, 24, 28, 30, 31], "count_statist": [3, 20, 28, 31], "counter": 13, "countingphenotypescor": [1, 8, 31], "counts_fram": 2, "countstatist": [3, 4], "cover": [5, 30], "cranial": 24, "creat": [2, 3, 5, 6, 7, 8, 10, 13, 14, 15, 16, 18, 20, 28, 30], "create_variant_from_scratch": 13, "creator": [15, 30, 31], "criteria": 30, "criterion": 6, "cry": 24, "csf": 24, "csv": [20, 31], "ct": 13, "curat": 20, "curi": [6, 8, 13, 15], "current": [10, 13, 14, 15, 16, 19, 27], "cut": 30, "cycl": 24, "cytologi": 24, "cytometri": 24, "darksalmon": 8, "data": [0, 2, 3, 4, 6, 13, 14, 15, 16, 18, 20, 26, 28, 30, 31], "databas": 15, "datadir": 15, "datafram": [2, 3, 4, 8], "datasanityissu": [0, 15], "dataset": [20, 31], "de": [8, 29], "decid": [2, 13, 24], "decis": 2, "decod": 12, "deem": 24, "deep": 18, "def": [8, 12], "default": [6, 10, 12, 15, 16, 20, 27, 28, 31], "default_cache_path": [0, 10], "default_filt": [2, 20, 28, 31], "defaultimprecisesvfunctionalannot": [0, 15], "defaulttermid": 2, "defect": [20, 25, 31], "defin": [3, 5, 6, 13, 25], "definit": 13, "degre": 22, "del": [6, 13, 25, 30], "delet": [6, 13, 25, 30], "deliveri": 24, "delta": 2, "demand": 30, "demonstr": [20, 30], "denot": 18, "depend": [20, 31], "deposit": 31, "depth": 15, "deriv": 28, "descend": [6, 7, 8, 20, 22, 25, 28, 30, 31], "describ": [8, 13, 15, 19, 23, 27, 28, 30, 31], "descript": [5, 13, 15, 22], "design": [18, 28], "desir": [6, 15, 21, 24, 30], "despit": 20, "detail": [5, 20, 28, 30], "determin": [3, 6, 13, 15, 22, 28, 31], "develop": [19, 22, 24, 28], "deviat": [18, 31], "devriesphenotypescor": [1, 8], "diagnos": [7, 13, 30], "diagnosi": 7, "diagram": [16, 20], "dichotom": 31, "dict": [12, 15], "differ": [2, 6, 8, 13, 14, 20, 22, 25, 28, 30, 31], "differenti": 18, "digest": 24, "dimension": 13, "dingeman": 22, "diploid": 13, "direct": [7, 13, 20, 28], "directli": [7, 20], "directori": [10, 15, 27], "disabl": [22, 28], "discharg": 24, "discoveri": [20, 28, 31], "discret": [3, 4, 5, 15, 30], "discuss": 30, "diseas": [0, 7, 13, 18, 20, 26, 30], "disease_id_queri": 7, "diseaseanalysi": [1, 3], "diseasepresencepred": [5, 7], "diseaseview": [0, 16], "disord": 20, "displai": [2, 16, 20, 27], "display_quest": [5, 20, 30, 31], "disproportion": 22, "dissect": 31, "distanc": 14, "distance_to": 14, "distribut": [8, 16, 20, 23, 31], "divid": [30, 31], "dna": [14, 24], "do": [2, 8, 14, 15, 19, 20, 27, 28, 30], "doc": [20, 31], "document": [16, 18, 19, 28, 31], "doe": [5, 10, 13, 15, 25, 30, 31], "domain": [2, 6, 13, 24, 28, 30, 31], "domin": [6, 20, 21, 30, 31], "done": [14, 15, 19, 30], "dopamin": 24, "doubl": [8, 14], "down": [15, 30], "downstream": [2, 14, 15, 27, 30], "downstream_gene_vari": [13, 15], "dpi": 31, "draw": [8, 16], "draw_fig": 16, "draw_protein_diagram": [16, 20], "draw_vari": 16, "drawn": 16, "driven": 24, "drug": 24, "ductu": [20, 31], "due": [13, 15], "dump": 12, "dup": 13, "duplic": [13, 15, 25], "dure": 15, "e": [2, 3, 6, 7, 8, 13, 14, 15, 18, 20, 24, 25, 28, 31], "each": [2, 3, 7, 8, 13, 15, 16, 18, 20, 21, 22, 28, 30, 31], "ear": [22, 24], "easi": 19, "easiest": 27, "ectopia": 30, "eff": [30, 31], "effect": [6, 13, 15, 20, 23, 27, 30, 31], "either": [2, 6, 7, 8, 13, 14, 15, 22], "elabor": 30, "electrolyt": 24, "electrophysiologi": 24, "element": [13, 14, 15, 16], "els": 12, "embryon": 24, "empti": [13, 14, 15], "enabl": 19, "encod": [7, 12, 13, 20, 30, 31], "encount": [15, 20], "end": [6, 13, 14, 20, 25, 30], "end_on_strand": 14, "endocrin": 24, "energi": 24, "enrich": 28, "ensembl": 13, "enst00000123456": 13, "ensure_ascii": 12, "entir": [15, 27], "entiti": [7, 13, 14, 15], "entri": 15, "enum": [13, 14, 15], "enumer": 16, "environ": [10, 19], "enzym": 24, "epiphysi": 24, "equal": [6, 13, 28, 31], "equival": 8, "error": [15, 20, 27, 28, 31], "error_count": 15, "erythrocyt": 24, "erythroid": 24, "erythropoietin": 24, "et": [8, 18, 21, 22, 31], "etc": 15, "evalu": [6, 18, 31], "even": [3, 24, 28], "ever": 2, "exact": [4, 6, 8, 13, 15, 18, 20, 21, 28], "exactli": [8, 28], "exampl": [3, 4, 6, 8, 9, 12, 13, 15, 20, 21, 25, 27, 28], "except": [12, 15, 30], "exclud": [2, 5, 7, 13, 14, 20, 28, 30, 31], "excluded_diseas": 13, "excluded_member_count": 13, "excluded_phenotyp": 13, "exclus": [5, 7, 20, 25, 30], "execut": 31, "exercis": 24, "exhaust": [5, 30], "exibit": 7, "exist": [10, 15, 28], "exon": [6, 13, 30], "exon20": 30, "exons_effect": 13, "expect": [2, 6, 15, 24, 28, 30, 31], "expenditur": 24, "explain": 30, "explan": [29, 31], "explicit": 7, "explicitli": 13, "explor": 31, "exploratori": [26, 27], "expos": 31, "express": [15, 24, 31], "extend": 31, "extern": 22, "extract": 15, "extraocular": 24, "extrem": 28, "ey": [24, 30, 31], "f": [8, 18], "face": 24, "fact": [28, 30], "factor": [24, 28, 30], "factori": 2, "fail": 2, "fall": [8, 10], "fallback": [10, 15], "fals": [3, 6, 7, 12, 13, 14, 15, 20, 28, 31], "famili": [20, 28], "farthest": 31, "fascia": 24, "fashion": 15, "fbn1": [6, 13], "fdr": [28, 31], "fdr_bh": [3, 20, 28, 31], "fdr_by": 28, "fdr_gb": 28, "fdr_tsbh": 28, "fdr_tsbky": 28, "featur": [6, 7, 13, 15, 18, 19, 23, 30, 31], "feature_elong": 13, "feature_id": 6, "feature_trunc": 13, "feature_typ": [6, 13], "featureinfo": [0, 13], "featuretyp": [0, 6, 13], "fecal": 24, "feel": 30, "feenstra": [8, 22], "femal": [13, 30], "fetal": 24, "fetch": [15, 20], "fetch_for_gen": 15, "fetch_respons": 15, "few": [25, 27, 30], "fgt_predic": 30, "fh": [20, 31], "fiber": 24, "fibrinolysi": 24, "fibroblast": 24, "fiction": [6, 30], "field": [13, 21], "fig": [20, 31], "figsiz": [20, 31], "figur": [15, 16], "file": [15, 16, 27], "filter": [2, 3, 6, 13, 20, 24, 25, 31], "filter_method_nam": 2, "filtering_pred": [5, 6, 30], "final": [20, 31], "find": [3, 14, 15, 20, 28, 30, 31], "find_coordin": 15, "finder": 15, "fingertip": 27, "first": [10, 13, 15, 20, 27, 28, 30, 31], "fisher": [4, 18, 20, 21, 28, 29], "fisher_exact": [4, 31], "fisherexacttest": [3, 4, 20, 28, 31], "fit": [30, 31], "five": 31, "five_prime_utr_vari": [13, 15], "flexibl": [30, 31], "flip": 14, "float": [2, 3, 4, 8, 9, 15, 31], "flow": 24, "fluid": 24, "focal": 25, "focu": [18, 20], "fold": 13, "folder": [10, 15, 27], "follow": [3, 8, 13, 19, 20, 22, 24, 25, 28, 30, 31], "foot": 24, "for_sampl": 13, "forearm": 24, "form": [4, 24, 27], "format": [7, 13, 14, 15, 16], "format_as_str": 16, "format_coordinates_for_vep_queri": 15, "formatt": 16, "forward": 14, "found": [15, 20, 22, 27, 28, 30, 31], "fpath_cache_dir": 15, "frame": [2, 3, 4, 30, 31], "frameshift": 30, "frameshift_vari": [13, 20, 30, 31], "free": 30, "frequenc": [2, 20, 31], "frequent": [20, 25], "friendli": 15, "from": [2, 5, 6, 8, 12, 13, 14, 15, 16, 18, 19, 20, 22, 24, 25, 28, 30, 31], "from_curi": 30, "from_map": 13, "from_moi_info": 6, "from_pati": 13, "from_query_curi": [8, 31], "from_raw_part": [5, 13], "from_term": 13, "from_vcf_liter": [13, 25], "from_vcf_symbol": [13, 25], "fuction": 15, "full": 13, "func": 8, "function": [4, 6, 7, 8, 9, 13, 15, 16, 20, 24, 27, 30, 31], "functional_annot": 15, "functionalannot": [0, 15], "functionalannotationawar": 13, "further": [20, 31], "fwer": 28, "g": [2, 3, 6, 7, 8, 13, 14, 15, 18, 20, 24, 25, 28, 30, 31], "ga": 24, "ga4gh": [18, 20, 31], "gain": 30, "ganglion": 24, "gastrin": 24, "gastrointestin": 24, "gb_acc": 14, "ge": 30, "genbank": 14, "genbank_acc": 14, "gene": [6, 13, 15, 20, 21, 27, 30, 31], "gene_coordinate_servic": 15, "gene_id": 13, "gene_nam": 13, "gene_symbol": 13, "genecoordinateservic": [0, 15], "gener": [2, 3, 5, 7, 15, 16, 18, 20, 28, 31], "general_hpo_term": 2, "genet": 18, "genit": 24, "genitourinari": 24, "genom": [13, 15, 18, 20, 25, 30], "genome_build": [15, 20], "genome_build_id": 14, "genomebuild": [13, 14, 15], "genomebuildidentifi": [13, 14], "genomicinterpret": 15, "genomicregion": [13, 14], "genotyp": [0, 2, 3, 4, 5, 8, 13, 15, 18, 24, 26, 27, 28], "genotype_for_sampl": 13, "genotype_phenotype_scor": [8, 31], "genotypepolypred": [2, 3, 5, 6, 8, 30], "genotypes_have_same_hpo_proport": 2, "gestat": 22, "get": [2, 3, 4, 5, 6, 7, 8, 10, 13, 14, 15, 16, 27, 28, 30], "get_annot": 15, "get_cache_dir_path": [0, 10, 15], "get_categor": [5, 6, 7, 30, 31], "get_categori": 5, "get_category_nam": 5, "get_cds_region": 13, "get_coding_base_count": 13, "get_codon_count": 13, "get_excluded_count": 13, "get_features_variant_overlap": 13, "get_five_prime_utr": 13, "get_maximum_group_observed_hpo_frequ": 2, "get_number_of_observed_hpo_observ": 2, "get_patient_id": 13, "get_quest": [6, 8, 30, 31], "get_question_bas": [5, 6, 7], "get_respons": 15, "get_three_prime_utr": 13, "get_toy_cohort": [0, 11], "get_variant_by_kei": [13, 30], "gg": [13, 15], "git": 19, "github": [19, 30], "give": 6, "given": [2, 6, 7, 13, 15, 16, 22], "gland": 24, "glial": 24, "gln421ter": 13, "global": [18, 22], "globe": 24, "glossari": 26, "glucagon": 24, "glucocorticoid": 24, "glucos": 24, "gly80arg": 20, "glycolipid": 24, "glycoprotein": 24, "glycosaminoglycan": 24, "glycosyl": 24, "gonadotropin": 24, "good": 20, "gov": 21, "gp": 20, "gpsea": [17, 19, 20, 21, 25, 27, 29, 30, 31], "gpsea_cach": [10, 15], "gpsea_cachedir": 10, "gpseajsondecod": [0, 12], "gpseajsonencod": [0, 12], "granulocytopoiet": 24, "graph": [7, 31], "graphawar": 7, "graphic": [16, 31], "grch37": [14, 15], "grch38": [13, 14, 15, 20, 25], "great": 28, "greater": [28, 30], "grid": 31, "group": [2, 3, 4, 5, 6, 7, 8, 9, 20, 21, 22, 28], "group1": 31, "group2": 31, "group_nam": [6, 20, 30, 31], "groups_pred": [5, 6, 20, 30, 31], "growth": 24, "gt": 13, "gt_id_to_nam": 31, "gt_predic": [2, 3, 8, 20, 30, 31], "guid": [18, 20, 30, 31], "h": 28, "h_1": 4, "ha": [2, 3, 5, 6, 7, 8, 13, 14, 15, 20, 22, 24, 28, 30, 31], "had": [20, 31], "hair": 24, "hallmark": 7, "hand": [22, 24], "happen": 3, "harbor": 31, "has_error": 15, "has_errors_or_warn": 15, "has_sv_info": 13, "has_variant_coordin": 13, "has_warn": 15, "hashabl": 13, "have": [2, 3, 8, 15, 20, 22, 27, 28, 30, 31], "head": [20, 24, 31], "health": 18, "hear": 31, "heart": [20, 22, 24, 31], "height": 24, "help": 15, "hematocrit": 24, "hemi": 30, "hemizyg": [6, 13], "hemoglobin": 24, "henc": [15, 19], "hepat": 24, "hepatobiliari": 24, "here": [6, 14, 15, 20, 22, 23, 25, 27, 30, 31], "het": [30, 31], "heterozyg": [6, 13, 30, 31], "heterozygot": 21, "heurist": [2, 20, 24, 28], "hgnc": [13, 15], "hgv": [13, 15], "hgvs_cdna": 13, "hgvs_coordinate_find": 15, "hgvsp": 13, "hierarch": [15, 24], "hierarchi": [2, 30], "high": [13, 26], "higher": [22, 31], "highest": 13, "hip": 24, "his1435arg": 30, "histidin": 30, "histori": 28, "ho": 28, "hochberg": [20, 28, 31], "holm": 28, "holt": [20, 30], "hom_ref": [30, 31], "homeostasi": 24, "hommel": 28, "homovanil": 24, "homozyg": [6, 30], "homozygot": 21, "homozygous_altern": 13, "homozygous_refer": 13, "honeydew": 8, "hormon": 24, "how": [14, 19, 20, 23, 27, 30, 31], "howev": [5, 14, 15, 19, 20, 25, 30, 31], "hp": [2, 3, 7, 8, 13, 20, 22, 24, 28, 30, 31], "hpo": [2, 3, 7, 8, 13, 15, 16, 18, 22, 27, 29, 30, 31], "hpo_mtc": 28, "hpomtcfilt": [1, 2, 20, 28, 31], "hpotermanalysi": [1, 3, 20, 28, 31], "hpotermanalysisresult": [1, 3, 16], "hpotk": [2, 20, 27, 28, 30, 31], "html": [16, 20, 31], "http": [15, 19, 21], "human": [2, 5, 14, 15, 16, 18, 22, 27], "humeru": [20, 31], "humor": 24, "hundr": 28, "hypertelor": 22, "hypertens": [20, 31], "hypoplasia": [20, 31], "hypospadia": 22, "hypothalamu": 24, "hypothes": [28, 31], "hypothesi": [2, 20, 28, 31], "i": [0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 13, 14, 15, 16, 18, 19, 20, 21, 22, 24, 25, 27, 28, 30, 31], "id": [5, 6, 7, 8, 13, 15, 16, 20, 24, 25, 30], "ident": [21, 28], "identifi": [5, 6, 8, 13, 14, 15, 20, 22, 24], "idx": 2, "ignor": 15, "ii": 24, "illustr": [20, 31], "img": [20, 31], "immun": 24, "impact": 6, "impair": 31, "implement": [4, 6, 12, 15, 21, 22, 30, 31], "implementor": 13, "impli": [7, 25, 28, 31], "implicitli": [25, 30], "import": [4, 6, 8, 13, 14, 20, 25, 27, 28, 30, 31], "impos": 4, "imposs": 30, "imprecis": 13, "imprecise_sv_functional_annot": 15, "imprecisesvfunctionalannot": [0, 15], "imprecisesvinfo": [0, 13, 15], "includ": [2, 3, 4, 5, 6, 7, 8, 13, 14, 15, 18, 20, 27, 30, 31], "include_computational_tx": 15, "include_patients_with_no_hpo": 13, "include_patients_with_no_vari": 13, "include_subsect": 15, "inclus": 30, "incomplete_terminal_codon_vari": 13, "increas": [20, 31], "indel": 13, "indent": 12, "independ": 28, "index": [6, 8, 13, 19, 24], "indic": 13, "indirect": [20, 28, 31], "indirectli": 7, "individu": [3, 6, 7, 8, 13, 18, 20, 21, 22, 25, 26, 27, 28, 30, 31], "infer": 20, "inflammatori": 24, "info": [2, 3, 6, 13, 15, 20, 25, 27, 30, 31], "inform": [13, 16, 18, 20, 22, 28, 30], "inframe_delet": [13, 20], "inframe_insert": [13, 20], "ingest": 18, "inherit": [6, 28, 31], "inhibin": 24, "initi": [7, 13, 19], "inner": 24, "input": [2, 3, 15, 18, 20, 26, 30, 31], "insert": [13, 15, 25], "insight": 24, "instal": [18, 27], "instanc": [2, 4, 5, 6, 7, 8, 13, 15, 21, 22, 25, 27, 28, 30, 31], "instead": [13, 28], "instruct": 18, "insulin": 24, "int": [2, 3, 4, 5, 6, 7, 13, 14, 15, 16], "integ": 13, "integr": 19, "integu": 24, "intellectu": 22, "intend": 8, "inter": [30, 31], "interest": [6, 7, 13, 15, 20, 24, 28, 30, 31], "intergenic_vari": [13, 15], "internet": 19, "interpret": [15, 30, 31], "intersect": 14, "interspers": 13, "interv": 14, "intestin": 24, "intracrani": 24, "intraocular": 24, "intrauterin": 22, "introduc": 30, "intron": 13, "intron_vari": [13, 15, 20, 30], "inv": 13, "invers": [13, 30], "investig": [7, 13, 20], "invoc": 19, "involv": [27, 28], "io": 0, "ion": 24, "ipython": [16, 20], "iqr": 31, "is_cod": 13, "is_empti": 14, "is_femal": 13, "is_filtered_out": 2, "is_frameshift": 31, "is_frameshift_or_stop_gain": 30, "is_large_imprecise_sv": 6, "is_mal": 13, "is_neg": 14, "is_observ": 13, "is_pass": 2, "is_posit": 14, "is_pref": 13, "is_pres": 13, "is_provid": 13, "is_structur": 13, "is_structural_delet": 6, "is_structural_vari": 6, "is_unknown": 13, "islet": 24, "issu": [15, 18, 20, 30], "item": [2, 3, 5, 6, 15, 16, 25], "iter": [2, 3, 5, 6, 7, 8, 12, 13, 14, 15, 16, 28], "iter_cohort_phenopacket": [20, 27, 30, 31], "iterate_nod": 15, "its": [7, 13, 15, 16, 25, 30, 31], "j": 3, "jinja2": 16, "joint": 24, "jordan": 31, "json": [12, 15, 27], "jsondecod": 12, "jsonencod": 12, "jt": 21, "judgment": [24, 28, 30], "jupyt": [16, 18, 20], "just": [13, 15, 20], "keep": [7, 15], "kei": [6, 13, 18, 20], "kidnei": [8, 31], "kind": [15, 21], "know": [13, 14, 15, 30], "known": 13, "kwarg": 12, "label": [13, 15, 16, 24, 30], "label_summari": 13, "labeling_method": 16, "lack": [7, 15], "lactat": 24, "langerhan": 24, "larg": [6, 13, 14, 15, 24], "larger": 13, "last": [10, 13, 20, 28, 31], "latest": 20, "lead": [6, 20, 24, 25, 28, 30, 31], "learn": [20, 23, 31], "least": [2, 6, 15, 28, 30, 31], "leav": 18, "left": 14, "legend": 16, "leigh": 13, "len": [13, 20, 25, 27, 30, 31], "length": [6, 13, 14, 30, 31], "lenient": 15, "lenti": 30, "leptin": 24, "less": [6, 20, 31], "let": [12, 13, 25, 27, 28, 30, 31], "leukocyt": 24, "level": [0, 8, 13, 15, 20, 24, 26, 28, 30, 31], "leverag": [2, 22, 30], "li": [14, 21], "librari": [0, 14, 19, 27, 31], "life": 31, "like": [3, 12, 18, 24, 28], "limb": [20, 24], "limit": [2, 22, 28], "line": [6, 13, 19, 31], "lineag": 24, "link": [6, 30], "liposaccharid": 24, "list": [7, 8, 12, 13, 24, 30], "list_all_diseas": 13, "list_all_protein": 13, "list_all_vari": 13, "list_present_phenotyp": 13, "liter": [6, 13, 15, 16, 25, 30], "literatur": [6, 20], "liver": [8, 24], "ll": 27, "load": [12, 15, 30, 31], "load_minimal_hpo": [20, 27, 28, 30, 31], "load_phenopacket": [0, 15, 20, 27, 30, 31], "load_phenopacket_fil": [0, 15, 27], "load_phenopacket_fold": [0, 15, 27], "loader": 27, "local": 15, "locat": [13, 14, 15, 18, 20, 24, 30], "locu": 13, "lof": 31, "lof_effect": 31, "lof_mut": 31, "log": 29, "logic": [22, 28, 30], "look": 3, "loss": [13, 31], "lower": [24, 30, 31], "luckili": 30, "ly": 31, "lymph": 24, "lymphat": 24, "lymphocyt": 24, "lysosom": 24, "m": [4, 19], "macrocephali": 22, "macroscop": 24, "made": [20, 28], "mai": [3, 8, 13, 15, 28, 30, 31], "main": [20, 31], "major": 14, "major_assembli": 14, "make": [2, 6, 28], "malarkei": 15, "male": [13, 30], "malform": 15, "mandatori": 13, "mane": [13, 20, 30, 31], "mani": [20, 23, 24, 27, 28, 30, 31], "manifest": 18, "mann": [8, 9, 22, 29], "mannwhitneystatist": [8, 9, 31], "mannwhitneyu": [9, 31], "map": [2, 13, 15, 16], "marker": [16, 24], "marker_count": 16, "marrow": 24, "mass": 24, "match": [6, 8, 30], "matern": 21, "math": 8, "matplotlib": [16, 20, 31], "matrix": [2, 3], "matter": 15, "matur": 24, "mature_mirna_vari": 13, "maximum": [2, 20, 22, 30, 31], "me": 13, "mean": [13, 24, 28, 31], "meaning": [5, 6], "measur": 24, "median": 31, "mediastinum": 24, "medic": [18, 24, 28], "medicin": 18, "meet": [6, 13, 30, 31], "megakaryocyt": 24, "member": [13, 15, 23, 31], "membran": 24, "mendelian": [18, 30], "mental": 24, "mesenteri": 24, "messag": [12, 15], "met": 30, "meta_label": 13, "metabol": 24, "metabolit": 24, "metadata": [15, 16], "metaphysi": 24, "method": [2, 3, 6, 7, 12, 13, 14, 15, 20, 27, 28, 31], "mhc": 24, "micro": 22, "microcephali": 22, "middl": 24, "midwif": 13, "might": 21, "mild": 22, "mim": [20, 30], "min_n_of_patients_with_term": [7, 20, 31], "miner": 24, "minimalontologi": [2, 3, 7, 8, 15, 16], "minimalterm": 13, "minimum": [2, 7], "mininum": 28, "miss": [8, 15, 20, 28, 30], "missens": [2, 3, 6, 13, 18, 21, 25, 28, 30, 31], "missense_and_exon20": 30, "missense_or_nonsens": 30, "missense_vari": [3, 6, 13, 20, 30, 31], "missing_implies_exclud": 7, "missing_implies_phenotype_exclud": [7, 30], "mitig": [28, 31], "mitochondrion": 24, "mixin": 14, "mnv": 13, "mode": [6, 28, 31], "mode_of_inheritance_data": 6, "mode_of_inheritance_info": 6, "model": [0, 2, 6, 20, 25, 30, 31], "modeofinheritanceinfo": 6, "modeofinheritancepred": [5, 6, 30, 31], "moder": 22, "modifi": 28, "modul": [0, 13, 14, 15, 31], "moi": 30, "moment": 15, "monarch": 19, "mondo": 7, "more": [2, 3, 6, 7, 8, 9, 13, 15, 19, 20, 22, 24, 27, 28, 31], "moreov": 30, "morpholog": 24, "morphologi": [8, 18, 20, 22, 24, 25, 28, 30, 31], "most": [13, 14, 20, 24, 28, 30], "mostli": [13, 30], "motif": [13, 18], "motil": 24, "motor": [20, 24], "movement": 24, "mri": 24, "mtc": [2, 3, 16, 20, 24, 31], "mtc_alpha": [3, 20, 31], "mtc_correct": [3, 20, 28, 31], "mtc_filter": [1, 3, 20, 28, 31], "mtc_filter_nam": 3, "mtc_filter_result": 3, "mtc_name": 3, "mtc_report": [20, 31], "mtc_viewer": [20, 31], "mtcstatsview": [0, 16, 20, 31], "much": [24, 28, 30], "mucociliari": 24, "mucosa": 24, "mucu": 24, "multi": [13, 30], "multiphenotypeanalysi": [1, 3, 28], "multiphenotypeanalysisresult": [1, 3], "multipl": [2, 3, 20, 24, 26, 30, 31], "multipli": 28, "muscl": 24, "muscular": [20, 31], "musculatur": [3, 24], "musculoskelet": 24, "must": [2, 3, 5, 6, 7, 8, 13, 15, 19, 28, 30], "mutat": [20, 27, 30, 31], "mutlipl": 28, "myelin": 24, "myeloid": 24, "n": [2, 4, 6, 13, 15, 20, 24, 30, 31], "n_categor": 5, "n_usabl": [2, 3], "nacgtacgtac": 13, "nail": 24, "name": [2, 3, 5, 6, 10, 13, 14, 15, 20, 31], "nan": [3, 8], "nasal": [22, 24], "natur": 13, "nc_000001": 14, "ncbi": [20, 21], "ncit": 13, "ndarrai": 16, "necessari": 30, "neck": 24, "need": [6, 15, 20, 27, 28, 31], "neg": [6, 14, 25], "neither": 28, "neoplasm": 24, "neopterin": 24, "nerv": 24, "nervou": [3, 24, 25, 28], "net": 13, "network": 27, "neuron": 24, "never": 14, "new": 16, "next": [20, 27, 30], "nih": 21, "nippl": 24, "nlm": 21, "nm_001042681": [30, 31], "nm_002834": 15, "nm_005912": 15, "nm_013275": 30, "nm_123": 6, "nm_1234": [3, 30], "nm_123456": [3, 6, 13], "nm_181486": [20, 31], "nmd_transcript_vari": 13, "no_cal": 13, "node": [15, 24], "nomin": [3, 28], "non": [3, 6, 13, 15, 20, 28, 30, 31], "non_coding_transcript_exon_vari": [13, 15], "non_coding_transcript_vari": [13, 15], "non_frameshift_del": 30, "non_frameshift_effect": 30, "non_frameshift_pred": 30, "noncoding_effect": 15, "none": [2, 3, 4, 5, 6, 7, 8, 10, 12, 13, 14, 15, 16, 20, 22, 27, 28, 30, 31], "nonfaci": 22, "nons": 21, "nonsens": [2, 3, 6, 21, 28, 30, 31], "nor": 28, "normal": 18, "nose": 22, "notat": [13, 25], "note": [2, 3, 7, 10, 13, 14, 20, 22, 28, 30, 31], "notebook": [16, 18, 20], "notepad": [0, 15], "notepadtre": [0, 15], "noth": 30, "novel": 13, "now": [13, 20, 27, 28, 31], "np_": 20, "np_001027558": 15, "np_001027559": 13, "np_037407": 15, "np_852259": 20, "nuclear": 13, "nucleobas": 24, "nucleotid": [13, 20, 25, 30], "nucleu": 28, "nul": 13, "null": 28, "number": [2, 3, 4, 5, 6, 7, 8, 13, 14, 22, 24, 28, 30, 31], "numer": 5, "o": 12, "obj": 12, "object": [2, 4, 5, 6, 7, 8, 9, 12, 13, 14, 15, 16, 27], "object_hook": 12, "observ": [2, 8, 13, 15, 18, 20, 28, 31], "observablefeatur": 13, "obtain": [15, 20, 22, 28], "occipitofrontali": 24, "occur": [20, 28, 31], "oddsratio": 31, "odontoid": 24, "off": 30, "offer": [22, 28, 29, 30, 31], "offlin": 15, "ok": 2, "oldest": 28, "omim": [7, 20], "omit": [20, 30, 31], "onc": 31, "one": [2, 5, 6, 7, 8, 13, 15, 19, 20, 21, 22, 24, 28, 30, 31], "one_genotype_has_zero_hpo_observ": 2, "ones": 15, "onli": [2, 5, 6, 7, 13, 15, 20, 21, 28, 30, 31], "onlin": 19, "onset": [24, 25], "onto": 16, "ontologi": [2, 13, 18, 20, 22, 25, 27], "oocyt": 24, "open": [20, 30, 31], "open_phenopacket_stor": [20, 27, 30, 31], "oper": [6, 30], "oppos": 13, "opposit": 14, "optim": 15, "option": [2, 5, 7, 13, 15, 19, 20, 27, 28], "oram": [20, 30], "order": [2, 3, 13, 20, 31], "org": 15, "organ": [13, 15, 24], "orgogozo": 18, "origin": 22, "orthogon": 30, "osmol": 24, "osteocalcin": 24, "other": [2, 13, 14, 15, 18, 20, 21, 22, 27, 28, 30], "otherwis": [6, 7, 13, 22, 30], "our": [20, 22, 24, 28, 30, 31], "out": [2, 14, 15, 20, 25, 27, 28, 31], "outcom": 3, "outer": 24, "over": [13, 15], "overal": 28, "overlap": [3, 6, 13, 14, 16, 30], "overlapping_exon": 13, "overlaps_with": 14, "overload": 30, "overview": [26, 31], "own": 14, "p": [2, 3, 4, 7, 8, 9, 13, 18, 20, 27, 28, 30, 31], "p13": [14, 15], "p_valu": 31, "packag": [17, 18, 19, 28], "page": 20, "pair": [3, 4, 6, 13, 25], "pancrea": 24, "pancreat": 24, "panda": 2, "param": [2, 6, 13, 14, 15], "paramet": [2, 3, 5, 6, 7, 8, 12, 13, 14, 15, 16], "parameter": 16, "parametr": 31, "parathyroid": 24, "parent": [3, 15, 22, 28], "pars": 15, "parse_multipl": 15, "parse_respons": 15, "parse_uniprot_json": 15, "part": [13, 16], "partit": 30, "pass": [2, 6, 12, 13, 25, 28], "past": 28, "patch": 14, "patent": [20, 31], "patern": 21, "path": [10, 15, 20, 27, 30, 31], "pathogen": [20, 27, 30], "patient": [0, 2, 3, 5, 6, 7, 8, 13, 15, 16, 18, 20, 27, 28, 30, 31], "patient_1": 8, "patient_2": 8, "patient_3": 8, "patient_4": 8, "patient_cr": 15, "patient_id": [8, 13, 31], "patientcategori": [1, 3, 5, 7, 20, 31], "patientcr": [0, 15], "pattern": [15, 24], "payload": 15, "pcat": [1, 20, 28, 31], "pehnotyp": 2, "peptid": 24, "per": 3, "percent": [3, 20, 31], "percentag": 3, "perform": [3, 4, 15, 18, 20, 21, 27, 28, 31], "perhap": 30, "perifollicular": 24, "peripher": 24, "peritoneum": 24, "peroxisom": 24, "ph": 24, "ph_predic": 2, "phagocytosi": 24, "pheno_pred": [3, 20, 30, 31], "pheno_scor": [8, 31], "phenopacket": [13, 15, 18, 20, 30, 31], "phenopacket1": 27, "phenopacket2": 27, "phenopacket_registri": 20, "phenopacketpatientcr": [0, 15], "phenopacketvariantcoordinatefind": [0, 15], "phenotyp": [0, 2, 3, 4, 5, 8, 9, 13, 15, 18, 22, 23, 24, 26, 27, 28], "phenotype_cr": 15, "phenotype_scor": 8, "phenotypecategor": [5, 7], "phenotypecr": [0, 15], "phenotypemtcfilt": [1, 2, 3, 20, 28], "phenotypemtcresult": [1, 2, 3], "phenotypepolypred": [2, 3, 5, 7, 30], "phenotypescor": [1, 8, 9], "phenotypescoreanalysi": [1, 8, 31], "phenotypescoreanalysisresult": [1, 8], "phenotypescorestatist": [8, 9], "phenotypic_abnorm": 2, "phosphat": 24, "physician": 13, "physiolog": 24, "physiologi": [3, 18, 24, 25], "pickl": 15, "piec": 20, "pineal": 24, "pinna": 22, "pip": 19, "pituitari": 24, "place": 18, "placent": 24, "placenta": 24, "platelet": 24, "platysma": 24, "pleas": 30, "plot": [8, 16, 20, 23, 31], "plot_boxplot": [8, 31], "plt": [16, 20, 31], "plu": 6, "pm": 20, "pmid": 20, "pmid_27087320_subject_1": 31, "pmid_27087320_subject_10": 31, "pmid_27087320_subject_2": 31, "pmid_29330883_subject_1": 31, "pmid_29330883_subject_2": 31, "png": [20, 31], "po": [13, 14, 25], "point": [13, 15, 22, 31], "point_mut": 31, "point_mutation_effect": 31, "polar": 28, "polici": [15, 20, 27, 31], "polypred": [1, 5, 6, 7], "port": 14, "posit": [3, 6, 14, 15, 20, 25, 28], "posixpath": 10, "possibl": [3, 5, 8, 21, 28, 30], "possibli": 30, "posterior": [24, 28], "potenti": 31, "power": [20, 31], "pp": 15, "pp_dir": 27, "pp_directori": 15, "pp_file": [15, 27], "ppktstore": [20, 27, 30, 31], "practic": 21, "pre": 2, "precis": 18, "precursor": 24, "predic": [1, 2, 3, 4, 8, 26, 31], "predict": [6, 13, 30, 31], "predictor": [15, 27], "prefer": 13, "preimplant": 24, "prematur": [18, 30], "prenat": 24, "prepar": [3, 5, 6, 7, 15, 27, 30], "prepare_hpo_terms_of_interest": [5, 7], "prepare_notepad": 15, "prepare_predicates_for_terms_of_interest": [5, 7, 20, 30, 31], "preprocess": [0, 20, 27, 30, 31], "preprocessingvalidationresult": 15, "presenc": [3, 6, 7, 13, 20, 28, 30, 31], "present": [3, 7, 8, 13, 14, 16, 20, 22, 30, 31], "present_diseas": 13, "present_phenotyp": 13, "present_phenotype_categor": 7, "present_phenotype_categori": 7, "pressur": 24, "pretti": 16, "prevent": [2, 3, 30], "previou": 28, "previous": [30, 31], "primari": [13, 24, 30], "principl": 21, "print": 13, "probabl": [28, 31], "problem": 28, "proce": 15, "procedur": [3, 20, 24, 31], "process": [15, 16, 20, 31], "process_respons": 15, "produc": [2, 4, 5, 6, 7], "product": 16, "profound": 22, "programmat": 6, "programmatic_access": 15, "project": 13, "prolifer": 24, "prong": [30, 31], "pronounc": 18, "propagatingphenotypepred": [5, 7, 30], "properti": [2, 3, 4, 5, 7, 8, 13, 14, 15, 16], "proport": [2, 20, 28, 31], "protcachingmetadataservic": [0, 15], "protein": [6, 13, 14, 15, 16, 18, 20, 24, 27, 30], "protein_altering_vari": 13, "protein_cach": 15, "protein_effect_coordin": 13, "protein_effect_end": 13, "protein_effect_loc": 13, "protein_effect_start": 13, "protein_featur": [6, 13], "protein_feature_end": 16, "protein_feature_nam": 16, "protein_feature_start": 16, "protein_feature_typ": [6, 16], "protein_id": [13, 15, 16], "protein_length": [13, 16], "protein_meta": [16, 20], "protein_metadata": 16, "protein_metadata_servic": 6, "protein_sourc": 15, "proteinannotationcach": [0, 15], "proteinfeatur": [0, 13], "proteinmetadata": [0, 13, 15, 16, 20], "proteinmetadataservic": [0, 6, 15], "proteinpred": [5, 6, 30], "proteinview": [0, 16], "proteinvisu": [0, 16, 20], "proteinvisualiz": [0, 16], "provid": [2, 3, 6, 7, 8, 14, 15, 16, 18, 22, 27, 28, 31], "proxi": 22, "prune": 15, "pscore": [1, 31], "publish": 19, "pubm": 21, "pulmonari": [20, 24, 31], "pupillari": 24, "purin": 24, "purpos": [27, 30], "put": [20, 31], "putamen": 28, "pval": [3, 8, 31], "pvalu": 31, "pvi": 16, "px_id": 20, "pydoc": 27, "pypi": 19, "pyplot": [20, 31], "pyrimidin": 24, "pytest": 19, "python": [4, 12, 18, 19, 27, 28, 30, 31], "python3": 19, "q": [20, 27, 31], "q1": 31, "q3": 31, "qc_result": [27, 31], "qual": [13, 25], "qualnam": [13, 14, 15], "quartil": 31, "queri": [7, 8, 13, 15, 30, 31], "question": [5, 6, 7, 18, 30, 31], "r": 31, "radiu": [20, 31], "rais": [5, 12, 14, 15], "random": 4, "random_se": 16, "rang": [2, 22, 31], "rank": [9, 29, 31], "rare": [0, 2, 20, 28], "rate": [20, 24, 28], "rather": 24, "ratio": 24, "reach": 15, "read": [18, 30], "readabl": [5, 13, 16], "readi": 20, "real": 31, "realiz": 28, "reason": [2, 20, 24, 31], "receiv": 20, "recent": 28, "recess": [6, 26, 30], "recommend": [2, 6, 13, 18, 20, 28, 31], "record": 15, "red": [24, 31], "redox": 24, "reduc": [2, 20, 27, 28], "redund": 28, "ref": [6, 13, 25, 31], "ref_length": [6, 31], "refer": [6, 13, 18, 22, 24, 31], "reflect": 30, "reflex": 24, "refract": 24, "refseq": [13, 14, 15], "refseq_nam": 14, "regard": [13, 16], "regardless": 13, "region": [6, 13, 14, 16, 25, 30], "registri": [20, 27, 30, 31], "regul": 24, "regulatory_region_abl": 13, "regulatory_region_amplif": 13, "regulatory_region_vari": 13, "rel": [6, 22], "relat": [15, 28], "relationship": [18, 31], "releas": [20, 24, 27, 28, 30, 31], "relev": [2, 13, 20, 31], "remot": 15, "remov": [6, 15, 24, 25, 30], "renal": 31, "render": 16, "renin": 24, "repair": 24, "repeat": 13, "replac": 30, "report": [3, 8, 15, 18, 20, 31], "repositori": 19, "repres": [2, 3, 5, 7, 8, 13, 14, 15, 16, 18, 28, 31], "represent": [13, 15], "reproduct": 24, "request": 15, "requir": [4, 13, 15, 19, 20, 27, 28, 30], "requisit": 27, "rere": [27, 30, 31], "rere_mane_tx_id": 30, "rere_phenotype_score_boxplot": 31, "resourc": [6, 15], "respect": [3, 13, 20, 25, 30], "respir": 24, "respiratori": 24, "respons": [15, 24, 27], "rest": [15, 24], "restrict": 21, "result": [2, 3, 8, 13, 15, 16, 18, 20, 24, 27, 28, 30, 31], "retain": [6, 28, 30], "reticulocyt": 24, "retriev": [14, 15], "return": [2, 3, 4, 5, 6, 7, 8, 12, 13, 14, 15, 16, 28, 30], "revers": 14, "right": [14, 15, 16], "risk": 31, "rna": 14, "root": [2, 15, 24, 28], "row": [2, 3, 31], "rule": [5, 20, 30, 31], "run": [3, 15, 20, 28, 31], "runner": 15, "runonlin": 19, "rust": 13, "sai": 28, "salivari": 24, "same": [2, 3, 14, 20, 22, 24, 25, 28, 31], "sampl": [13, 20], "sample_id": 13, "samplelabel": [0, 13], "sanit": 15, "sanitationresult": 15, "saniti": 15, "savefig": [20, 31], "scalp": 24, "scenario": 5, "schema": [13, 15, 16, 27, 31], "scipi": [4, 9, 31], "scipyfisherexact": 3, "sclera": 24, "score": [8, 9, 29], "score_analysi": 31, "score_statist": [8, 31], "scorer": 8, "search": [15, 20, 24, 30], "sebac": 24, "second": 15, "secondari": 13, "secret": 24, "section": [2, 3, 6, 15, 20, 22, 26, 27, 28, 30, 31], "secundum": [20, 31], "sediment": 24, "see": [2, 3, 6, 8, 9, 13, 20, 21, 24, 25, 27, 28, 30, 31], "seek": [6, 30], "seem": 30, "seen": [20, 28, 31], "segment": [13, 24, 30], "segreg": 24, "seizur": [20, 25], "select": [2, 6, 31], "self": [12, 14], "sens": [2, 24, 28], "sensat": 24, "sensori": 24, "sensorineur": 31, "separ": [12, 29, 31], "septal": [20, 25, 31], "septum": [20, 25, 31], "sequenc": [2, 3, 4, 5, 6, 7, 9, 13, 14, 15, 16], "sequence_vari": 13, "ser36thrfster25": 20, "serializ": 12, "seriou": 15, "serum": 24, "serv": [5, 30], "servic": 15, "set": [8, 10, 13, 20, 28, 30, 31], "sete": 30, "setup": [20, 28, 31], "sever": [3, 5, 13, 15, 20, 22, 25, 28, 29, 30, 31], "sex": [0, 13, 24, 30], "sh": 28, "sharpei": 24, "shelf": 30, "shift": 30, "short": [13, 20, 22, 31], "shortcut": 13, "shorter": [6, 31], "should": [2, 3, 6, 7, 10, 13, 14, 15, 16, 20, 30, 31], "show": [13, 16, 20, 24, 25, 28, 30, 31], "shown": [16, 20], "sidak": 28, "side": [4, 31], "sign": [13, 14, 20, 30, 31], "signal": [24, 28], "signific": [13, 22, 24, 28, 31], "significantli": [20, 31], "sime": 28, "similar": [13, 31], "similarli": 30, "simpl": [6, 15, 19, 21, 22, 25, 30], "simplest": 28, "simplifi": [20, 31], "sinc": [3, 6, 13, 31], "sine": 13, "singl": [2, 13, 15, 30], "situ": 24, "size": [13, 14, 24], "skelet": 24, "skin": [8, 24], "skinfold": 24, "skip": [2, 20, 24, 28, 31], "skipkei": 12, "skull": 24, "slice": 14, "small": [22, 25, 28], "smell": 24, "snv": [13, 25, 30], "so": [6, 13, 28, 30, 31], "softwar": 31, "solut": [15, 27], "somat": 24, "some": [4, 6, 15, 18, 19, 28, 30], "some_cell_has_greater_than_one_count": 2, "someth": 15, "sometim": 30, "sort": 15, "sort_index": 31, "sort_kei": 12, "sound": [24, 30], "sourc": [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 19, 30], "span": [13, 14, 24, 25], "specif": [3, 6, 13, 15, 18, 20, 24, 28, 30, 31], "specifi": [2, 6, 30], "specified_term": 28, "specifiedtermsmtcfilt": [1, 2, 28], "spell": 25, "spinal": 24, "spleen": 24, "splice_acceptor_vari": [13, 15, 20], "splice_donor_5th_base_vari": [13, 15, 20], "splice_donor_region_vari": 13, "splice_donor_vari": [13, 15, 20], "splice_polypyrimidine_tract_vari": [13, 15], "splice_region_vari": [13, 20], "spuriou": 28, "stabil": 24, "stalk": 24, "standalon": 16, "standard": [13, 27], "start": [6, 13, 14, 15, 16, 24, 27], "start_lost": [13, 31], "start_on_strand": 14, "start_retained_vari": 13, "stat": [3, 8, 20, 28, 31], "state": [13, 15, 25, 30], "static": [2, 5, 6, 8, 12, 13, 15, 28, 30], "statist": [2, 4, 18, 20, 22, 24, 26, 28], "statsmodel": 28, "statu": [2, 15], "statur": 22, "step": [19, 20, 27, 30, 31], "stimul": 24, "stool": 24, "stop": [24, 30], "stop_gain": [6, 13, 20, 30, 31], "stop_lost": 13, "stop_retained_vari": [13, 20], "storag": 15, "store": [15, 20, 27, 28, 30, 31], "store_annot": 15, "str": [2, 3, 5, 6, 7, 8, 10, 13, 14, 15, 16, 25], "strabismu": 31, "strand": [6, 13, 14], "strategi": [2, 16, 24, 31], "stress": 24, "strict": 15, "string": [8, 13, 15, 16, 24], "stromal": 24, "strongli": [6, 13], "structur": [6, 8, 13, 15, 22, 24, 25, 28, 30, 31], "structural_defect": 31, "structural_typ": [6, 13, 30], "structural_vari": 6, "studi": 18, "stump": 24, "sub": [13, 15], "subclass": [6, 12], "subgroup": 13, "subject": [6, 13, 30, 31], "submodul": 17, "subpackag": 17, "subplot": [20, 31], "subsect": [13, 15], "subset": 30, "subunit": 24, "successfulli": 20, "suffix": 15, "sum": [22, 31], "summar": [3, 13, 15, 16, 20, 25, 27, 31], "summari": [3, 27], "summary_df": [20, 31], "suox": 21, "super": 12, "support": [6, 12, 13, 14, 18, 21, 30], "supports_shap": 4, "suppos": [12, 30], "surf1": [6, 13], "surf2": 6, "surfac": 24, "surviv": [28, 29], "sv": [6, 13, 15], "sv_delet": 25, "sv_info": 13, "svart": 14, "svlen": [13, 25], "svtype": [13, 25], "sweat": 24, "switch": 19, "symbol": [6, 13, 14, 20, 25], "symptom": [13, 20, 24, 30, 31], "synapt": 24, "syndrom": [13, 20, 30], "synonym": 30, "synonymous_vari": [13, 30], "system": [3, 13, 14, 22, 24, 25, 28], "t": [9, 13, 15, 20, 25], "tabl": [4, 16, 24, 25, 28, 31], "tail": 31, "take": [4, 8, 15, 20, 28, 30, 31], "tall": 22, "tandem": 13, "target": [6, 7, 30], "target_phenotype_categori": 3, "tast": 24, "tbx5_cohort_info": 20, "tbx5_frameshift": 31, "tbx5_frameshift_vs_missens": 20, "tbx5_protein_diagram": 20, "technic": 18, "tediou": 30, "tedium": 30, "temperatur": 24, "templat": 16, "ten": 28, "term": [2, 3, 7, 8, 13, 15, 18, 20, 22, 25, 30, 31], "term_frequency_threshold": [2, 20, 28, 31], "term_id": 13, "termid": [2, 3, 6, 7, 8, 13, 30], "termin": [13, 18], "terms_to_test": [2, 28], "test": [2, 3, 4, 5, 6, 7, 8, 9, 13, 14, 18, 20, 21, 22, 24, 26, 29, 30], "tf_binding_site_vari": 13, "tfbs_ablat": 13, "tfbs_amplif": 13, "th": 3, "than": [2, 6, 13, 20, 24, 28, 30, 31], "thank": 31, "thei": [14, 18, 24, 28, 30, 31], "them": 15, "themselv": 14, "therebi": 2, "therefor": [2, 3, 24, 28, 30], "thi": [2, 3, 6, 7, 8, 12, 13, 14, 15, 18, 19, 20, 21, 22, 24, 25, 27, 28, 30, 31], "thick": 24, "thin": 4, "think": 28, "third": 31, "thorac": 24, "thr223met": 20, "three": [13, 21, 28, 30], "three_prime_utr_vari": [13, 15], "threshold": [2, 6, 20, 28, 31], "thrombocyt": 24, "thrombosi": 24, "through": 15, "thu": [3, 13, 22, 28, 31], "thumb": [5, 20, 30, 31], "thymu": 24, "thyroid": 24, "tight_layout": 20, "tild": 30, "time": [3, 13, 15, 19, 28], "timeout": 15, "tissu": 24, "to_csv": [20, 31], "to_negative_strand": 14, "to_opposite_strand": 14, "to_positive_strand": 14, "to_str": 13, "todo": [16, 20, 21, 22, 23, 26, 29, 30, 31], "togeth": [20, 31], "toler": 2, "tonsil": 24, "too": 24, "toolkit": [8, 15, 20, 27], "top": [3, 7, 8, 13, 15, 20, 24, 27, 31], "top_phenotype_count": 16, "top_variant_count": 16, "total": [3, 13, 20, 22, 28, 31], "total_patient_count": 13, "total_test": [3, 20, 31], "traceback": 28, "track": [7, 15], "tracker": [18, 30], "tract": 24, "trans_id": 13, "transcript": [3, 6, 13, 15, 16, 20, 30, 31], "transcript_abl": [13, 31], "transcript_amplif": 13, "transcript_coordin": 16, "transcript_id": [13, 16, 20], "transcriptannot": [0, 13, 15], "transcriptcoordin": [0, 13, 15, 16, 20], "transcriptcoordinateservic": [0, 15], "transcriptinfoawar": [0, 13, 15], "transform": [15, 31], "transit": 14, "translat": 13, "transloc": [6, 13], "transmiss": 24, "transpos": [13, 14], "transpose_coordin": [13, 14], "trascript": 20, "travers": 15, "tree": 15, "tri": 15, "triphalang": [20, 31], "triphosph": 24, "trp121gly": 20, "true": [2, 6, 7, 12, 13, 14, 15, 20, 28, 30, 31], "truncat": 20, "try": [10, 12, 14, 15], "ttest_ind": 9, "tteststatist": [8, 9], "tupl": [3, 13, 15, 20, 27, 28, 30, 31], "turn": [15, 27], "tutori": [18, 31], "twice": 28, "two": [3, 4, 8, 14, 20, 21, 22, 28, 30, 31], "tx": [15, 16, 30], "tx_annot": 13, "tx_coordin": [16, 20], "tx_id": [6, 13, 15, 16, 20, 30, 31], "txc_servic": 20, "type": [3, 6, 7, 8, 12, 13, 14, 15, 16, 18, 28, 30], "typeerror": [12, 28], "typic": [3, 13, 30, 31], "tyr136ter": 20, "tyr407ter": 20, "u": [8, 9, 20, 22, 30], "ucsc": 14, "ucsc_nam": 14, "ulna": [20, 31], "umbil": 24, "umbilicu": 24, "unavail": 13, "under": [15, 20, 27, 28, 31], "underli": 30, "uniform": 6, "uninterest": 24, "uniprot": [15, 16, 27], "uniprotproteinmetadataservic": [0, 15], "uniqu": [3, 5, 13, 20], "unit": [14, 19], "unknown": [6, 13], "unknown_sex": 13, "unless": [7, 20, 28], "unlik": [15, 24, 28], "unnecessari": 24, "unsurprisingli": 30, "untransl": 13, "up": [20, 22, 28, 30, 31], "upper": [20, 24], "upstream": 14, "upstream_gene_vari": [13, 15], "urat": 24, "uri": 13, "urin": 24, "urinari": 24, "urobilinogen": 24, "us": [2, 3, 6, 7, 8, 10, 12, 13, 14, 15, 16, 18, 20, 25, 27, 28, 30, 31], "usabl": 3, "usag": 9, "use_al": 28, "usealltermsmtcfilt": [1, 2, 28], "user": [18, 20, 24, 31], "usual": [6, 13, 24, 30, 31], "util": [5, 6, 31], "utr": 13, "uvea": 24, "v": [18, 28, 30], "v2024": [20, 27, 28, 30, 31], "vagin": 24, "val263met": 20, "valid": [15, 16, 20, 27, 28, 31], "validation_polici": 15, "validation_runn": 15, "validationrunn": 15, "valu": [2, 3, 4, 8, 9, 13, 14, 15, 16, 20, 28, 31], "valueerror": [5, 14, 15], "valv": 24, "varcachingfunctionalannot": [0, 15], "vari": [6, 31], "variabl": [4, 10, 13, 30, 31], "variant": [0, 3, 4, 6, 13, 15, 16, 18, 20, 23, 25, 27, 28, 31], "variant_class": [6, 13, 30], "variant_coordin": [13, 15], "variant_effect": [6, 13, 16, 20, 30, 31], "variant_effect_count_by_tx": 13, "variant_fallback": 15, "variant_info": 13, "variant_kei": [6, 13, 15], "variant_key_of_interest": 30, "variant_loc": 16, "variant_locations_counted_absolut": 16, "variant_pred": 6, "variantannotationcach": [0, 15], "variantclass": [0, 6, 13, 30], "variantcoordin": [0, 13, 15, 25], "variantcoordinatefind": [0, 15], "varianteffect": [0, 6, 13, 15, 16, 20, 30, 31], "varianteffectpred": 3, "variantformatt": [0, 16], "variantinfo": [0, 13], "variantinfoawar": [0, 13], "variantpred": [5, 6, 20, 30, 31], "varianttranscriptproteinartist": 16, "varianttranscriptvisu": [0, 16], "variou": 14, "vascular": 24, "vasculatur": 24, "vc": [13, 15], "vcf": 13, "vcv000522858": 30, "ventricl": 20, "ventricular": [20, 24, 25, 31], "vep": 15, "vepfunctionalannot": [0, 15], "verbos": 5, "veri": [2, 28], "version": [4, 20, 22, 31], "vessel": 24, "via": [7, 12, 15, 16], "view": [0, 18, 20, 31], "viewabl": 16, "viewer": 20, "virtual": 19, "vision": 24, "visit": 15, "visitor": 15, "visual": [16, 20, 24], "vitamin": 24, "voic": 24, "volum": 24, "vri": [8, 29], "vvhgvsvariantcoordinatefind": [0, 15], "vvmulticoordinateservic": [0, 15, 20], "w": [20, 31], "wa": [3, 5, 7, 13, 15, 16, 20, 22, 28, 30, 31], "wai": [20, 27, 31], "waist": 24, "wall": 24, "want": [2, 8, 28, 30, 31], "warn": [15, 20, 27, 31], "warning_count": 15, "we": [2, 3, 6, 8, 13, 15, 16, 18, 19, 20, 22, 23, 24, 27, 28, 30, 31], "weight": 24, "well": [6, 7, 13], "were": [3, 15, 20, 27, 30, 31], "what": [23, 30, 31], "when": [2, 7, 15, 28, 30], "where": [2, 3, 13, 15, 20, 30], "whether": [7, 8, 13, 18, 31], "which": [2, 3, 4, 5, 6, 7, 13, 20, 22, 27, 30, 31], "while": [4, 21, 30, 31], "whisker": 31, "whitnei": [8, 9, 22, 29], "who": [5, 7, 21, 30, 31], "whose": [6, 13, 20, 30, 31], "wilcoxon": 31, "wise": [20, 28], "with_cache_fold": 15, "with_strand": 14, "within": [13, 15, 20, 31], "without": [3, 6, 30, 31], "word": [18, 24], "wordsmith": 30, "work": [2, 10, 15, 27], "worth": 2, "would": [2, 8, 13, 18, 21, 22, 28, 30, 31], "wrap": [8, 30], "wrap_scoring_funct": 8, "wrapper": [4, 9], "write": [15, 16, 20, 23, 26, 31], "www": 15, "x": [6, 30, 31], "x_1000001_1000027_": 13, "x_1000001_1000027_taaaaaaaaaaaaaaaaaaaaaaaaaa_t": 13, "x_12345_12345_c_g": [6, 13], "x_domin": [6, 30], "x_recess": [6, 30], "xm_": 15, "y": [30, 31], "ye": [3, 5, 7, 20, 30, 31], "yield": 28, "ylabel": 31, "ylim": 31, "you": [12, 13, 14, 27, 28], "your": [14, 19, 27], "z": 30, "zero": [2, 6, 14, 20, 22, 25, 30, 31], "\u03c72": 4}, "titles": ["gpsea package", "gpsea.analysis package", "gpsea.analysis.mtc_filter package", "gpsea.analysis.pcats package", "gpsea.analysis.pcats.stats package", "gpsea.analysis.predicate package", "gpsea.analysis.predicate.genotype package", "gpsea.analysis.predicate.phenotype package", "gpsea.analysis.pscore package", "gpsea.analysis.pscore.stats package", "gpsea.config module", "gpsea.data package", "gpsea.io module", "gpsea.model package", "gpsea.model.genome package", "gpsea.preprocessing package", "gpsea.view package", "API reference", "GPSEA", "Installation", "Tutorial", "Genotype-Phenotype Correlations in Autosomal Recessive Diseases", "De Vries Score", "Cohort exploratory analysis", "General HPO terms", "Glossary", "User guide", "Input data", "Multiple-testing correction", "Phenotype Predicates", "Predicates", "Statistical tests"], "titleterms": {"The": 20, "abnorm": 22, "all": [28, 30], "allel": 25, "altern": 27, "an": 25, "analysi": [1, 2, 3, 4, 5, 6, 7, 8, 9, 20, 23, 31], "api": 17, "associ": 31, "autosom": 21, "background": 28, "block": 30, "build": 30, "chang": 25, "choos": 28, "code": 22, "cohort": [20, 23, 27, 30, 31], "compar": 31, "complex": 30, "condit": 30, "config": 10, "configur": 31, "congenit": 22, "content": [26, 29], "correct": 28, "correl": 21, "creat": [27, 31], "creator": 27, "data": [11, 27], "de": 22, "delai": 22, "development": 22, "diseas": 21, "dysmorph": 22, "exact": 31, "exampl": [30, 31], "explor": 20, "exploratori": 23, "facial": 22, "featur": 22, "feedback": 18, "fet": 31, "filter": [28, 30], "final": 22, "fisher": 31, "frameshift": [20, 31], "from": 27, "ga4gh": 27, "galleri": 30, "gener": 24, "genom": 14, "genotyp": [6, 20, 21, 30, 31], "glossari": 25, "gpsea": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 18, 28], "group": [30, 31], "growth": 22, "guid": 26, "hpo": [20, 24, 28], "implement": 28, "inherit": 30, "input": 27, "instal": 19, "invert": 30, "io": 12, "latest": 19, "length": 25, "literatur": 18, "load": [20, 27], "mann": 31, "missens": 20, "mode": 30, "model": [13, 14], "modul": [10, 12], "more": 30, "mtc": 28, "mtc_filter": 2, "multipl": 28, "need": 30, "non": 22, "onset": 22, "packag": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 11, 13, 14, 15, 16], "path": 25, "pcat": [3, 4], "phenopacket": 27, "phenotyp": [7, 20, 21, 29, 30, 31], "postnat": 22, "predic": [5, 6, 7, 20, 29, 30], "prenat": 22, "prepar": 20, "preprocess": 15, "procedur": 28, "propag": 30, "pscore": [8, 9], "recess": 21, "refer": [17, 25], "releas": 19, "rest": 31, "retard": 22, "rule": 25, "run": 19, "score": [22, 31], "scorer": 22, "shape": 4, "sourc": 27, "specifi": 28, "stabl": 19, "stat": [4, 9], "statist": 31, "strategi": 28, "submodul": 0, "subpackag": [0, 1, 3, 5, 8, 13], "support": 4, "tbx5": [20, 31], "term": [24, 28], "test": [19, 28, 31], "true": 25, "tutori": [20, 21], "u": 31, "us": 22, "user": 26, "v": [20, 31], "variant": 30, "view": 16, "vri": 22, "which": 28, "whitnei": 31}})