

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Tutorial &mdash; GPSEA 0.9.6.dev0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="_static/gpsea.css?v=547657ed" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=33b58314"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="_static/copybutton.js?v=6dbb43f8"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="User guide" href="user-guide/index.html" />
    <link rel="prev" title="Setup" href="setup.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="setup.html">Setup</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Tutorial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#background">Background</a></li>
<li class="toctree-l2"><a class="reference internal" href="#analysis">Analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#inputs-and-q-c">Inputs and Q/C</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#gene-and-transcript">Gene and transcript</a></li>
<li class="toctree-l4"><a class="reference internal" href="#human-phenotype-ontology">Human Phenotype Ontology</a></li>
<li class="toctree-l4"><a class="reference internal" href="#load-phenopackets">Load phenopackets</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#explore-cohort">Explore cohort</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#show-cohort-summary">Show cohort summary</a></li>
<li class="toctree-l4"><a class="reference internal" href="#plot-distribution-of-variants-with-respect-to-the-protein-sequence">Plot distribution of variants with respect to the protein sequence</a></li>
<li class="toctree-l4"><a class="reference internal" href="#summarize-all-variant-alleles">Summarize all variant alleles</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#partition-the-cohort-by-genotype-and-phenotype">Partition the cohort by genotype and phenotype</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#partition-by-genotype">Partition by genotype</a></li>
<li class="toctree-l4"><a class="reference internal" href="#partition-by-phenotype">Partition by phenotype</a></li>
<li class="toctree-l4"><a class="reference internal" href="#multiple-testing-correction">Multiple testing correction</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#statistical-testing">Statistical testing</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#conclusion">Conclusion</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Tutorial</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/tutorial.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="tutorial">
<span id="id1"></span><h1>Tutorial<a class="headerlink" href="#tutorial" title="Link to this heading"></a></h1>
<p>Here we present an example genotype-phenotype (G/P) analysis with GPSEA.
We assume GPSEA was installed as described in the <a class="reference internal" href="setup.html#setup"><span class="std std-ref">Setup</span></a> section
and we encourage the users to execute the tutorial code in a Jupyter notebook
or a similar interactive environment.</p>
<section id="background">
<h2>Background<a class="headerlink" href="#background" title="Link to this heading"></a></h2>
<p>In this tutorial, we analyze a cohort of individuals with pathogenic variants in <em>TBX5</em> leading to
<a class="reference external" href="https://omim.org/entry/142900">Holt-Oram syndrome MIM:142900</a>.</p>
<p>Holt-Oram syndrome is an autosomal dominant disorder characterized by
upper limb defects, congenital heart defects, and arrhythmias (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/38336121/">PMID:38336121</a>).
It has been observed in the literature that congenital defects of the ventricular and atrial septum
are more common in the truncating than in the missense variants (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30552424/">PMID:30552424</a>).
Additionally, upper limb defects are more frequent in patients with protein-truncating variants (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/38336121/">PMID:38336121</a>).</p>
<p>We curated the literature and created a <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/35705716/">GA4GH phenopacket</a>
for each affected individual.
The phenopackets are published in <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a>.</p>
</section>
<section id="analysis">
<h2>Analysis<a class="headerlink" href="#analysis" title="Link to this heading"></a></h2>
<p>A typical GPSEA analysis will consist of several steps. Starting with a collection of phenopackets,
we perform input Q/C and functional variant annotation to prepare a cohort.
With the cohort on hand, we generate reports with summary statistics, variant distributions,
and most common Human Phenotype Ontology (HPO) terms or measurements.
We then configure the methods for partitioning the cohort into genotype and phenotype classes,
to test for possible associations between the classes.
We finalize the analysis by statistical testing and evaluation of the results.</p>
<nav class="contents local" id="gpsea-analysis-workflow">
<p class="topic-title">GPSEA analysis workflow</p>
<ul class="simple">
<li><p><a class="reference internal" href="#inputs-and-q-c" id="id6">Inputs and Q/C</a></p>
<ul>
<li><p><a class="reference internal" href="#gene-and-transcript" id="id7">Gene and transcript</a></p></li>
<li><p><a class="reference internal" href="#human-phenotype-ontology" id="id8">Human Phenotype Ontology</a></p></li>
<li><p><a class="reference internal" href="#load-phenopackets" id="id9">Load phenopackets</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#explore-cohort" id="id10">Explore cohort</a></p>
<ul>
<li><p><a class="reference internal" href="#show-cohort-summary" id="id11">Show cohort summary</a></p></li>
<li><p><a class="reference internal" href="#plot-distribution-of-variants-with-respect-to-the-protein-sequence" id="id12">Plot distribution of variants with respect to the protein sequence</a></p></li>
<li><p><a class="reference internal" href="#summarize-all-variant-alleles" id="id13">Summarize all variant alleles</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#partition-the-cohort-by-genotype-and-phenotype" id="id14">Partition the cohort by genotype and phenotype</a></p>
<ul>
<li><p><a class="reference internal" href="#partition-by-genotype" id="id15">Partition by genotype</a></p></li>
<li><p><a class="reference internal" href="#partition-by-phenotype" id="id16">Partition by phenotype</a></p></li>
<li><p><a class="reference internal" href="#multiple-testing-correction" id="id17">Multiple testing correction</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#statistical-testing" id="id18">Statistical testing</a></p></li>
</ul>
</nav>
<section id="inputs-and-q-c">
<h3><a class="toc-backref" href="#id6" role="doc-backlink">Inputs and Q/C</a><a class="headerlink" href="#inputs-and-q-c" title="Link to this heading"></a></h3>
<section id="gene-and-transcript">
<h4><a class="toc-backref" href="#id7" role="doc-backlink">Gene and transcript</a><a class="headerlink" href="#gene-and-transcript" title="Link to this heading"></a></h4>
<p>For the analysis, the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/refseq/MANE/">MANE</a> transcript
(i.e., the “main” biomedically relevant transcript of a gene) should be chosen unless
there is a specific reason not to (which should occur rarely if at all).</p>
<p>In the case of <em>TBX5</em> the MANE transcript is <cite>NM_181486.4</cite>. Note that the trascript identifier (<cite>NM_181486</cite>) and the version (<cite>4</cite>) are both required.
A good way to find the MANE transcript is to search on the gene symbol (e.g., <em>TBX5</em>) in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a> and to
choose a variant that is specifically located in the gene. The MANE transcript will be displayed here (e.g., <a class="reference external" href="https://www.ncbi.nlm.nih.gov/clinvar/variation/495227/">NM_181486.4(TBX5):c.1221C&gt;G (p.Tyr407Ter)</a>).</p>
<p>We additionally need the corresponding protein identifier.
A good way to find this is to search on the transcript id in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/">NCBI Nucleotide</a>.
In our case, search on <cite>NM_181486.4</cite> will bring us to <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/NM_181486.4">this page</a>.
If we search within this page for <cite>“NP_”</cite>, this will bring us to the
corresponding protein accession <cite>NP_852259.1</cite>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_name</span> <span class="o">=</span> <span class="s1">&#39;TBX5&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_181486.4&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">px_id</span> <span class="o">=</span> <span class="s1">&#39;NP_852259.1&#39;</span>
</pre></div>
</div>
</section>
<section id="human-phenotype-ontology">
<h4><a class="toc-backref" href="#id8" role="doc-backlink">Human Phenotype Ontology</a><a class="headerlink" href="#human-phenotype-ontology" title="Link to this heading"></a></h4>
<p>The analysis in this tutorial needs to access Human Phenotype Ontology (HPO).
We use <a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/">HPO toolkit</a>
to load the version <cite>v2024-07-01</cite> of HPO:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Use the latest HPO release by omitting the <cite>release</cite> option from the loader method.</p>
</div>
</section>
<section id="load-phenopackets">
<h4><a class="toc-backref" href="#id9" role="doc-backlink">Load phenopackets</a><a class="headerlink" href="#load-phenopackets" title="Link to this heading"></a></h4>
<p>Now we will load the samples to analyze. We will use the cohort of 156 individuals with mutations in <em>TBX5</em>
whose clinical signs and symptoms were encoded into HPO terms
and stored in <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">ppktstore.registry</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_phenopacket_registry</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">phenopacket_registry</span> <span class="o">=</span> <span class="n">configure_phenopacket_registry</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">phenopacket_registry</span><span class="o">.</span><span class="n">open_phenopacket_store</span><span class="p">(</span><span class="s2">&quot;0.1.20&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">ps</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">phenopackets</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">ps</span><span class="o">.</span><span class="n">iter_cohort_phenopackets</span><span class="p">(</span><span class="n">cohort_name</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">)</span>
<span class="go">156</span>
</pre></div>
</div>
<p>We loaded 156 phenopackets which need further preprocessing to prepare for the analysis.
We will compute functional annotations for the mutations and then include the individuals into
a <a class="reference internal" href="apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.preprocessing</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span><span class="p">,</span> <span class="n">load_phenopackets</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">validation</span> <span class="o">=</span> <span class="n">load_phenopackets</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">phenopackets</span><span class="o">=</span><span class="n">phenopackets</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">cohort_creator</span><span class="o">=</span><span class="n">cohort_creator</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="go">Individuals Processed: ...</span>
</pre></div>
</div>
<p>and we will check that there are no Q/C issues:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">validation</span><span class="o">.</span><span class="n">summarize</span><span class="p">()</span>
<span class="go">Validated under none policy</span>
<span class="go">No errors or warnings were found</span>
</pre></div>
</div>
<p>We loaded the patient data into a <cite>cohort</cite> which is ready for the next steps.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>Here we show how to create a <a class="reference internal" href="apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a> from phenopackets.
See <a class="reference internal" href="user-guide/input-data.html#input-data"><span class="std std-ref">Input data</span></a> section to learn how to create a cohort from another inputs.</p>
</div>
</section>
</section>
<section id="explore-cohort">
<h3><a class="toc-backref" href="#id10" role="doc-backlink">Explore cohort</a><a class="headerlink" href="#explore-cohort" title="Link to this heading"></a></h3>
<p>Once the genotype and phenotype has been standardized, we can generate reports
to gain insight for the cohort data.</p>
<section id="show-cohort-summary">
<h4><a class="toc-backref" href="#id11" role="doc-backlink">Show cohort summary</a><a class="headerlink" href="#show-cohort-summary" title="Link to this heading"></a></h4>
<p>The cohort summary report provides an overview about
the most common HPO terms, variants, diseases, and variant effects:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">CohortViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">viewer</span> <span class="o">=</span> <span class="n">CohortViewer</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span> <span class="o">=</span> <span class="n">viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span> <span class="n">transcript_id</span><span class="o">=</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span>
</pre></div>
</div>
<div>
    <style>
        /** Light theme is the default */
        :root {
            --gpsea-color-text: black;
            --gpsea-color-link-text: rgb(0, 118, 187);
            --gpsea-color-link-hover: rgb(0, 99, 156);
            --gpsea-color-primary-a0: #2f5170;
            --gpsea-color-primary-a10: #4778a5;
            --gpsea-color-primary-a20: #5995ce;
            --gpsea-color-primary-a60: LightSkyBlue;
            --gpsea-color-surface-a0: #fcfcfc;
            --gpsea-color-surface-a10: #ffffff;
            --gpsea-color-surface-a20: #b3b3b3;
            --gpsea-color-surface-a30: #8d8d8d;
            --gpsea-color-surface-a40: #646464;
            --gpsea-color-surface-a50: #000000;
        }

        /** Dark theme */
        /*
        @media (prefers-color-scheme: dark) {
            :root {
                --gpsea-color-text: rgb(240, 240, 240);
                --gpsea-color-link-text: rgb(240, 240, 240);
                --gpsea-color-link-hover: lightblue;
                --gpsea-color-primary-a0: #64a3df;
                --gpsea-color-primary-a10: #5286b8;
                --gpsea-color-primary-a20: #375c7e;
                --gpsea-color-primary-a60: #233b52;
                --gpsea-color-surface-a0: #121212;
                --gpsea-color-surface-a10: #282828;
                --gpsea-color-surface-a20: #3f3f3f;
                --gpsea-color-surface-a30: #575757;
                --gpsea-color-surface-a40: #717171;
                --gpsea-color-surface-a50: #8b8b8b;
            }
        }
        */

        #gpsea-body {
            max-width: 1600px;
            color: var(--gpsea-color-text);
            background-color: var(--gpsea-color-surface-a0);
            padding: 10px 0;
        }

        #gpsea-body {

            h1,
            h2,
            h3,
            h4,
            h5 {
                color: var(--gpsea-color-primary-a0);
            }

            a {
                color: var(--gpsea-color-link-text);
                text-decoration: none;
            }

            a:hover {
                color: var(--gpsea-color-link-hover);
            }

            #cohort-summary {
                max-width: 800px;
            }

            td {
                padding: 5px 0;
            }

            .cards {
                display: flex;
                flex-flow: row wrap;
                align-items: top;
                gap: 1rem;
            }

            .card {
                border-radius: 5px;
                padding: 0px 10px;
                background: var(--gpsea-color-surface-a10);
                border: 1px solid var(--gpsea-color-surface-a30);
                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);
            }

            table {
                width: 100%;
                border-collapse: collapse;
                margin: 10px 0;
                font-size: 0.9em;
                font-family: sans-serif;
                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
                border: 1px solid var(--gpsea-color-primary-a60);
            }

            th {
                text-align: left;
                padding: 2px;
                font-weight: bold;
                font-size: 120%;
                color: var(--gpsea-color-text);
                background-color: var(--gpsea-color-primary-a60);
            }

            table td,
            tr {
                text-align: left;
                padding: 0px 5px;
            }

            tr:nth-child(even) {
                background-color: var(--gpsea-color-surface-a10);
            }

            tr:nth-child(odd) {
                background-color: var(--gpsea-color-surface-a20);
            }

            caption {
                caption-side: top;
                text-align: left;
                padding-bottom: 10px;
                font-weight: bold;
            }
        }
    </style>
    <div id="gpsea-body">
        

<h1>GPSEA cohort analysis</h1>
<div id="cohort-summary">
  <p>
    Successfully loaded 156 individuals.

    
    46 were recorded as male,
    56 as female,
    and 54 as unknown sex.

    
    
    No information about individuals' vital status was reported.
    

    
    52 individuals had disease onset information
    and 0 had information about the age of last encounter.
  </p>

  
  <p>No errors encountered.</p>
  
</div>


<div class="cards">
  
  <div id="hpo" class="card">
    <h2>HPO terms</h2>
    
    
    <table>
      The cohort included 90 distinct HPO terms.
      <caption>
        
        Top 10 most common HPO terms.
        
        
      </caption>
      <tbody>
        <tr class="strng">
          <th><em>n</em></th>
          <th>HPO Term</th>
        </tr>

        
        <tr>
          <td>50</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0001631">Atrial septal defect</a>
          </td>
        </tr>
        
        <tr>
          <td>41</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0001629">Ventricular septal defect</a>
          </td>
        </tr>
        
        <tr>
          <td>40</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0002984">Hypoplasia of the radius</a>
          </td>
        </tr>
        
        <tr>
          <td>36</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0001199">Triphalangeal thumb</a>
          </td>
        </tr>
        
        <tr>
          <td>32</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0009778">Short thumb</a>
          </td>
        </tr>
        
        <tr>
          <td>32</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0009777">Absent thumb</a>
          </td>
        </tr>
        
        <tr>
          <td>30</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0001191">Abnormal carpal morphology</a>
          </td>
        </tr>
        
        <tr>
          <td>27</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0001684">Secundum atrial septal defect</a>
          </td>
        </tr>
        
        <tr>
          <td>15</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0003974">Absent radius</a>
          </td>
        </tr>
        
        <tr>
          <td>14</td>
          <td><a href="https://hpo.jax.org/browse/term/HP:0031546">Cardiac conduction abnormality</a>
          </td>
        </tr>
        
      </tbody>
    </table>
    
    
  </div>

  
  <div id="measurements" class="card">
    <h2>Measurements</h2>
    
    
    No data regarding measurement assays were provided.
    
  </div>

  
  <div id="diseases" class="card">
    <h2>Diseases</h2>
    
    
    The cohort members were diagnosed with 1 disease.
    <table>
      

      <tbody>
        <tr class="strng">
          <th><em>n</em></th>
          <th>Disease</th>
        </tr>
        
        <tr>
          <td>156</td>
          <td><a href="https://hpo.jax.org/browse/disease/OMIM:142900">Holt-Oram syndrome</a>
          </td>
        </tr>
        
      </tbody>
    </table>
    
  </div>

  
  
  <div class="card">
    <h2>Variants</h2>
    <p>
      A total of 53 unique variants
      were identified in the cohort. Variants were annotated with respect to NM_181486.4.
    </p>
    <table>
      <caption>
        <h3>Top 10 variants</h3>
      </caption>
      <tbody>
        <tr class="strng">
          <th><em>n</em></th>
          <th>Variant key</th>
          <th>HGVS</th>
          <th>Variant Class</th>
        </tr>
        
        <tr>
          <td>22</td>
          <td>12_114385521_114385521_C_T</td>
          <td>c.710G>A (p.Arg237Gln)</td>
          <td>MISSENSE_VARIANT</td>
        </tr>
        
        <tr>
          <td>20</td>
          <td>12_114401830_114401830_C_T</td>
          <td>c.238G>A (p.Gly80Arg)</td>
          <td>MISSENSE_VARIANT</td>
        </tr>
        
        <tr>
          <td>8</td>
          <td>12_114385563_114385563_G_A</td>
          <td>c.668C>T (p.Thr223Met)</td>
          <td>MISSENSE_VARIANT</td>
        </tr>
        
        <tr>
          <td>6</td>
          <td>12_114398675_114398675_G_T</td>
          <td>c.408C>A (p.Tyr136Ter)</td>
          <td>STOP_GAINED</td>
        </tr>
        
        <tr>
          <td>5</td>
          <td>12_114399514_114399514_A_C</td>
          <td>c.361T>G (p.Trp121Gly)</td>
          <td>MISSENSE_VARIANT, SPLICE_REGION_VARIANT</td>
        </tr>
        
        <tr>
          <td>5</td>
          <td>12_114398682_114398682_C_CG</td>
          <td>c.400dup (p.Arg134ProfsTer49)</td>
          <td>FRAMESHIFT_VARIANT</td>
        </tr>
        
        <tr>
          <td>5</td>
          <td>12_114403792_114403792_C_CG</td>
          <td>c.106_107insC (p.Ser36ThrfsTer25)</td>
          <td>FRAMESHIFT_VARIANT</td>
        </tr>
        
        <tr>
          <td>4</td>
          <td>12_114385474_114385474_A_G</td>
          <td>c.755+2T>C (None)</td>
          <td>SPLICE_DONOR_VARIANT</td>
        </tr>
        
        <tr>
          <td>4</td>
          <td>12_114398656_114398656_C_CG</td>
          <td>c.426dup (p.Ala143ArgfsTer40)</td>
          <td>FRAMESHIFT_VARIANT</td>
        </tr>
        
        <tr>
          <td>4</td>
          <td>12_114403798_114403798_G_GC</td>
          <td>c.100dup (p.Ala34GlyfsTer27)</td>
          <td>FRAMESHIFT_VARIANT</td>
        </tr>
        
      </tbody>
    </table>
  </div>

  
  <div class="card">
    
    <h2>Variant effects</h2>
    <p>
      The effects were predicted for NM_181486.4
    </p>
    <table>
      <tbody>
        <tr class="strng">
          <th>Variant effect</th>
          <th>Count</th>
        </tr>
        
        <tr>
          <td>MISSENSE_VARIANT</td>
          <td>85 (50%)</td>
        </tr>
        
        <tr>
          <td>FRAMESHIFT_VARIANT</td>
          <td>38 (22%)</td>
        </tr>
        
        <tr>
          <td>STOP_GAINED</td>
          <td>19 (11%)</td>
        </tr>
        
        <tr>
          <td>SPLICE_REGION_VARIANT</td>
          <td>10 (6%)</td>
        </tr>
        
        <tr>
          <td>SPLICE_DONOR_VARIANT</td>
          <td>7 (4%)</td>
        </tr>
        
        <tr>
          <td>INFRAME_INSERTION</td>
          <td>2 (1%)</td>
        </tr>
        
        <tr>
          <td>STOP_RETAINED_VARIANT</td>
          <td>2 (1%)</td>
        </tr>
        
        <tr>
          <td>SPLICE_ACCEPTOR_VARIANT</td>
          <td>2 (1%)</td>
        </tr>
        
        <tr>
          <td>SPLICE_DONOR_5TH_BASE_VARIANT</td>
          <td>2 (1%)</td>
        </tr>
        
        <tr>
          <td>INTRON_VARIANT</td>
          <td>2 (1%)</td>
        </tr>
        
        <tr>
          <td>INFRAME_DELETION</td>
          <td>1 (1%)</td>
        </tr>
        
      </tbody>
    </table>
    
  </div>
</div>

    </div>
</div></section>
<section id="plot-distribution-of-variants-with-respect-to-the-protein-sequence">
<h4><a class="toc-backref" href="#id12" role="doc-backlink">Plot distribution of variants with respect to the protein sequence</a><a class="headerlink" href="#plot-distribution-of-variants-with-respect-to-the-protein-sequence" title="Link to this heading"></a></h4>
<p>We can use <a class="reference internal" href="apidocs/gpsea.view.html#gpsea.view.CohortArtist" title="gpsea.view.CohortArtist"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortArtist</span></code></a> to plot the distribution of variants
with respect to the encoded protein on
a Matplotlib <a class="reference external" href="https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.html">Axes</a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_default_cohort_artist</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_artist</span> <span class="o">=</span> <span class="n">configure_default_cohort_artist</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span> <span class="mi">8</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_artist</span><span class="o">.</span><span class="n">draw_protein</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">protein_id</span><span class="o">=</span><span class="n">px_id</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/tbx5_protein_diagram.png"><img alt="TBX5 protein diagram" class="align-center" src="_images/tbx5_protein_diagram.png" style="width: 600px;" />
</a>
<p>The diagram plots the location of the variants with respect to the protein sequence.
The variant location is represented by a “lollipop”.
The lollipop color represents the predicted variant effect and the lollipop size corresponds to the allele count within the cohort.
The diagram also highlights the protein features (domains, repeats, etc.).</p>
</section>
<section id="summarize-all-variant-alleles">
<span id="show-cohort-variants"></span><h4><a class="toc-backref" href="#id13" role="doc-backlink">Summarize all variant alleles</a><a class="headerlink" href="#summarize-all-variant-alleles" title="Link to this heading"></a></h4>
<p>We can prepare a table of all variant alleles that occur in the cohort.</p>
<p>Each table row corresponds to a single allele and lists the variant key,
the predicted effect on the transcript (<em>cDNA</em>) and protein of interest,
the variant effects, and the number of patients who present
with one or more variant alleles (<em>Count</em>):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">CohortVariantViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">viewer</span> <span class="o">=</span> <span class="n">CohortVariantViewer</span><span class="p">(</span><span class="n">tx_id</span><span class="o">=</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span> <span class="o">=</span> <span class="n">viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span>
</pre></div>
</div>
<div>
    <style>
        /** Light theme is the default */
        :root {
            --gpsea-color-text: black;
            --gpsea-color-link-text: rgb(0, 118, 187);
            --gpsea-color-link-hover: rgb(0, 99, 156);
            --gpsea-color-primary-a0: #2f5170;
            --gpsea-color-primary-a10: #4778a5;
            --gpsea-color-primary-a20: #5995ce;
            --gpsea-color-primary-a60: LightSkyBlue;
            --gpsea-color-surface-a0: #fcfcfc;
            --gpsea-color-surface-a10: #ffffff;
            --gpsea-color-surface-a20: #b3b3b3;
            --gpsea-color-surface-a30: #8d8d8d;
            --gpsea-color-surface-a40: #646464;
            --gpsea-color-surface-a50: #000000;
        }

        /** Dark theme */
        /*
        @media (prefers-color-scheme: dark) {
            :root {
                --gpsea-color-text: rgb(240, 240, 240);
                --gpsea-color-link-text: rgb(240, 240, 240);
                --gpsea-color-link-hover: lightblue;
                --gpsea-color-primary-a0: #64a3df;
                --gpsea-color-primary-a10: #5286b8;
                --gpsea-color-primary-a20: #375c7e;
                --gpsea-color-primary-a60: #233b52;
                --gpsea-color-surface-a0: #121212;
                --gpsea-color-surface-a10: #282828;
                --gpsea-color-surface-a20: #3f3f3f;
                --gpsea-color-surface-a30: #575757;
                --gpsea-color-surface-a40: #717171;
                --gpsea-color-surface-a50: #8b8b8b;
            }
        }
        */

        #gpsea-body {
            max-width: 1600px;
            color: var(--gpsea-color-text);
            background-color: var(--gpsea-color-surface-a0);
            padding: 10px 0;
        }

        #gpsea-body {

            h1,
            h2,
            h3,
            h4,
            h5 {
                color: var(--gpsea-color-primary-a0);
            }

            a {
                color: var(--gpsea-color-link-text);
                text-decoration: none;
            }

            a:hover {
                color: var(--gpsea-color-link-hover);
            }

            #cohort-summary {
                max-width: 800px;
            }

            td {
                padding: 5px 0;
            }

            .cards {
                display: flex;
                flex-flow: row wrap;
                align-items: top;
                gap: 1rem;
            }

            .card {
                border-radius: 5px;
                padding: 0px 10px;
                background: var(--gpsea-color-surface-a10);
                border: 1px solid var(--gpsea-color-surface-a30);
                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);
            }

            table {
                width: 100%;
                border-collapse: collapse;
                margin: 10px 0;
                font-size: 0.9em;
                font-family: sans-serif;
                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
                border: 1px solid var(--gpsea-color-primary-a60);
            }

            th {
                text-align: left;
                padding: 2px;
                font-weight: bold;
                font-size: 120%;
                color: var(--gpsea-color-text);
                background-color: var(--gpsea-color-primary-a60);
            }

            table td,
            tr {
                text-align: left;
                padding: 0px 5px;
            }

            tr:nth-child(even) {
                background-color: var(--gpsea-color-surface-a10);
            }

            tr:nth-child(odd) {
                background-color: var(--gpsea-color-surface-a20);
            }

            caption {
                caption-side: top;
                text-align: left;
                padding-bottom: 10px;
                font-weight: bold;
            }
        }
    </style>
    <div id="gpsea-body">
        
<h1>Variant alleles</h1>
<div class="card">
  <table>
    <caption>
      A total of 53 unique alleles were identified in the cohort.
    </caption>

    <tbody>
      <tr>
        <th>Count</th>
        <th>Variant key</th>
        <th>HGVS</th>
        <th>Overlapping Exons</th>
        <th>Effects</th>
      </tr>
      
      <tr>
        <td>22</td>
        <td>12_114385521_114385521_C_T</td>
        <td>c.710G>A (p.Arg237Gln)</td>
        <td>7</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>20</td>
        <td>12_114401830_114401830_C_T</td>
        <td>c.238G>A (p.Gly80Arg)</td>
        <td>3</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>8</td>
        <td>12_114385563_114385563_G_A</td>
        <td>c.668C>T (p.Thr223Met)</td>
        <td>7</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>6</td>
        <td>12_114398675_114398675_G_T</td>
        <td>c.408C>A (p.Tyr136Ter)</td>
        <td>5</td>
        <td>stop gained</td>
      </tr>
      
      <tr>
        <td>5</td>
        <td>12_114399514_114399514_A_C</td>
        <td>c.361T>G (p.Trp121Gly)</td>
        <td>4</td>
        <td>missense, splice region</td>
      </tr>
      
      <tr>
        <td>5</td>
        <td>12_114398682_114398682_C_CG</td>
        <td>c.400dup (p.Arg134ProfsTer49)</td>
        <td>5</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>5</td>
        <td>12_114403792_114403792_C_CG</td>
        <td>c.106_107insC (p.Ser36ThrfsTer25)</td>
        <td>2</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>4</td>
        <td>12_114385474_114385474_A_G</td>
        <td>c.755+2T>C (-)</td>
        <td>-</td>
        <td>splice donor</td>
      </tr>
      
      <tr>
        <td>4</td>
        <td>12_114403798_114403798_G_GC</td>
        <td>c.100dup (p.Ala34GlyfsTer27)</td>
        <td>2</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>4</td>
        <td>12_114398656_114398656_C_CG</td>
        <td>c.426dup (p.Ala143ArgfsTer40)</td>
        <td>5</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>4</td>
        <td>12_114385522_114385522_G_A</td>
        <td>c.709C>T (p.Arg237Trp)</td>
        <td>7</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>4</td>
        <td>12_114366360_114366360_C_T</td>
        <td>c.787G>A (p.Val263Met)</td>
        <td>8</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>3</td>
        <td>12_114366366_114366366_T_A</td>
        <td>c.781A>T (p.Ser261Cys)</td>
        <td>8</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>3</td>
        <td>12_114399613_114399613_T_A</td>
        <td>c.262A>T (p.Lys88Ter)</td>
        <td>4</td>
        <td>stop gained</td>
      </tr>
      
      <tr>
        <td>3</td>
        <td>12_114401853_114401853_G_T</td>
        <td>c.215C>A (p.Thr72Lys)</td>
        <td>3</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>3</td>
        <td>12_114366312_114366312_G_A</td>
        <td>c.835C>T (p.Arg279Ter)</td>
        <td>8</td>
        <td>stop gained</td>
      </tr>
      
      <tr>
        <td>3</td>
        <td>12_114403798_114403799_GC_G</td>
        <td>c.100del (p.Ala34ProfsTer32)</td>
        <td>2</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>3</td>
        <td>12_114385475_114385475_C_T</td>
        <td>c.755+1G>A (-)</td>
        <td>-</td>
        <td>splice donor</td>
      </tr>
      
      <tr>
        <td>3</td>
        <td>12_114401827_114401827_T_A</td>
        <td>c.241A>T (p.Arg81Trp)</td>
        <td>3</td>
        <td>missense, splice region</td>
      </tr>
      
      <tr>
        <td>2</td>
        <td>12_114398666_114398667_TG_T</td>
        <td>c.416del (p.Pro139GlnfsTer11)</td>
        <td>5</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>2</td>
        <td>12_114366274_114366274_G_T</td>
        <td>c.873C>A (p.Tyr291Ter)</td>
        <td>8</td>
        <td>stop gained</td>
      </tr>
      
      <tr>
        <td>2</td>
        <td>12_114385521_114385521_C_G</td>
        <td>c.710G>C (p.Arg237Pro)</td>
        <td>7</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>2</td>
        <td>12_114366207_114366208_GC_G</td>
        <td>c.939del (p.Gln315ArgfsTer79)</td>
        <td>8</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>2</td>
        <td>12_114394762_114394763_CA_C</td>
        <td>c.641del (p.Val214GlyfsTer12)</td>
        <td>6</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>2</td>
        <td>12_114366267_114366267_C_A</td>
        <td>c.880G>T (p.Glu294Ter)</td>
        <td>8</td>
        <td>stop gained</td>
      </tr>
      
      <tr>
        <td>2</td>
        <td>12_114385550_114385550_A_AATTATTCTCAG</td>
        <td>c.680_681insCTGAGAATAAT (p.Ile227_Glu228insTer)</td>
        <td>7</td>
        <td>inframe insertion, stop retainined</td>
      </tr>
      
      <tr>
        <td>2</td>
        <td>12_114403754_114403754_G_T</td>
        <td>c.145C>A (p.Gln49Lys)</td>
        <td>2</td>
        <td>missense, splice region</td>
      </tr>
      
      <tr>
        <td>2</td>
        <td>12_114398578_114398579_CA_C</td>
        <td>c.504del (p.Phe168LeufsTer6)</td>
        <td>5</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>2</td>
        <td>12_114398568_114398568_C_A</td>
        <td>c.510+5G>T (-)</td>
        <td>-</td>
        <td>splice donor 5th base, intronic</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114401907_114401907_A_G</td>
        <td>c.161T>C (p.Ile54Thr)</td>
        <td>3</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114401846_114401846_C_G</td>
        <td>c.222G>C (p.Met74Ile)</td>
        <td>3</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114356064_114356065_TA_T</td>
        <td>c.1024del (p.Tyr342ThrfsTer52)</td>
        <td>9</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114355784_114355785_CA_C</td>
        <td>c.1304del (p.Leu435ArgfsTer147)</td>
        <td>9</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114399559_114399559_T_C</td>
        <td>c.316A>G (p.Ile106Val)</td>
        <td>4</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114398708_114398709_GC_G</td>
        <td>c.374del (p.Gly125AlafsTer25)</td>
        <td>5</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114355755_114355756_TG_T</td>
        <td>c.1333del (p.His445MetfsTer137)</td>
        <td>9</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114394820_114394820_C_G</td>
        <td>c.584G>C (p.Gly195Ala)</td>
        <td>6</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114399625_114399629_ACATC_A</td>
        <td>c.246_249del (p.Met83PhefsTer6)</td>
        <td>4</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114399594_114399594_A_C</td>
        <td>c.281T>G (p.Leu94Arg)</td>
        <td>4</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114401921_114401921_C_G</td>
        <td>c.148-1G>C (-)</td>
        <td>-</td>
        <td>splice acceptor</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114399622_114399622_G_T</td>
        <td>c.253C>A (p.Pro85Thr)</td>
        <td>4</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114385553_114385553_C_A</td>
        <td>c.678G>T (p.Lys226Asn)</td>
        <td>7</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114398632_114398632_G_A</td>
        <td>c.451C>T (p.Gln151Ter)</td>
        <td>5</td>
        <td>stop gained</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114355723_114355723_G_A</td>
        <td>c.1366C>T (p.Gln456Ter)</td>
        <td>9</td>
        <td>stop gained</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114399633_114399633_C_G</td>
        <td>c.243-1G>C (-)</td>
        <td>-</td>
        <td>splice acceptor</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114394817_114394817_G_C</td>
        <td>c.587C>G (p.Ser196Ter)</td>
        <td>6</td>
        <td>stop gained</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114394743_114394746_TGTG_T</td>
        <td>c.658_660del (p.His220del)</td>
        <td>6</td>
        <td>inframe deletion</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114403859_114403859_G_T</td>
        <td>c.40C>A (p.Pro14Thr)</td>
        <td>2</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114366348_114366349_CT_C</td>
        <td>c.798del (p.Val267TrpfsTer127)</td>
        <td>8</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114366241_114366242_CT_C</td>
        <td>c.905del (p.Gln302ArgfsTer92)</td>
        <td>8</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114398602_114398602_T_G</td>
        <td>c.481A>C (p.Thr161Pro)</td>
        <td>5</td>
        <td>missense</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114398626_114398627_CG_C</td>
        <td>c.456del (p.Val153SerfsTer21)</td>
        <td>5</td>
        <td>frameshift</td>
      </tr>
      
      <tr>
        <td>1</td>
        <td>12_114401873_114401874_TA_T</td>
        <td>c.194del (p.Leu65GlnfsTer10)</td>
        <td>3</td>
        <td>frameshift</td>
      </tr>
      
    </tbody>
  </table>
</div>

    </div>
</div></section>
</section>
<section id="partition-the-cohort-by-genotype-and-phenotype">
<h3><a class="toc-backref" href="#id14" role="doc-backlink">Partition the cohort by genotype and phenotype</a><a class="headerlink" href="#partition-the-cohort-by-genotype-and-phenotype" title="Link to this heading"></a></h3>
<p>Testing for a genotype-phenotype association uses genotype and phenotype as variables.
In GPSEA, the variable value for an individual is computed
either by a <a class="reference internal" href="apidocs/gpsea.analysis.clf.html#gpsea.analysis.clf.Classifier" title="gpsea.analysis.clf.Classifier"><code class="xref py py-class docutils literal notranslate"><span class="pre">Classifier</span></code></a>
or by a <a class="reference internal" href="apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScorer" title="gpsea.analysis.pscore.PhenotypeScorer"><code class="xref py py-class docutils literal notranslate"><span class="pre">PhenotypeScorer</span></code></a>.
A <cite>Classifier</cite> assigns the individual into a class,
whereas a <cite>PhenotypeScorer</cite> computes a continuous score.
The classifiers and scorers are applied on all individuals of the cohort
and the resulting variable distributions are then assessed by a statistical test.</p>
<p>In GPSEA, genotype is always treated as a class
and a genotype <cite>Classifier</cite> is a prerequisite for each analysis.
However, there is much more flexibility on the phenotype part,
where either a <cite>Classifier</cite> or a <cite>PhenotypeScorer</cite> can be used to compute the values,
depending on the analysis goals.</p>
<p>In this tutorial section, we first configure a <cite>Classifier</cite> for assigning
the individuals into a genotype class,
and we follow with generating classifiers for testing the presence or exclusion
of HPO terms in the individuals.</p>
<section id="partition-by-genotype">
<h4><a class="toc-backref" href="#id15" role="doc-backlink">Partition by genotype</a><a class="headerlink" href="#partition-by-genotype" title="Link to this heading"></a></h4>
<p>In context of the tutorial, we assign each cohort member into a class
depending on presence of a single allele of a missense or truncating variant
(e.g. frameshift, stop gain, or splice site region):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.model</span><span class="w"> </span><span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.predicate</span><span class="w"> </span><span class="kn">import</span> <span class="n">variant_effect</span><span class="p">,</span> <span class="n">anyof</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.clf</span><span class="w"> </span><span class="kn">import</span> <span class="n">monoallelic_classifier</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_missense</span> <span class="o">=</span> <span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">truncating_effects</span> <span class="o">=</span> <span class="p">(</span>
<span class="gp">... </span>   <span class="n">VariantEffect</span><span class="o">.</span><span class="n">TRANSCRIPT_ABLATION</span><span class="p">,</span>
<span class="gp">... </span>   <span class="n">VariantEffect</span><span class="o">.</span><span class="n">TRANSCRIPT_TRANSLOCATION</span><span class="p">,</span>
<span class="gp">... </span>   <span class="n">VariantEffect</span><span class="o">.</span><span class="n">FRAMESHIFT_VARIANT</span><span class="p">,</span>
<span class="gp">... </span>   <span class="n">VariantEffect</span><span class="o">.</span><span class="n">START_LOST</span><span class="p">,</span>
<span class="gp">... </span>   <span class="n">VariantEffect</span><span class="o">.</span><span class="n">STOP_GAINED</span><span class="p">,</span>
<span class="gp">... </span>   <span class="n">VariantEffect</span><span class="o">.</span><span class="n">SPLICE_DONOR_VARIANT</span><span class="p">,</span>
<span class="gp">... </span>   <span class="n">VariantEffect</span><span class="o">.</span><span class="n">SPLICE_ACCEPTOR_VARIANT</span><span class="p">,</span>
<span class="gp">... </span>   <span class="c1"># more effects could be listed here ...</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_truncating</span> <span class="o">=</span> <span class="n">anyof</span><span class="p">(</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">e</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">)</span> <span class="k">for</span> <span class="n">e</span> <span class="ow">in</span> <span class="n">truncating_effects</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span> <span class="o">=</span> <span class="n">monoallelic_classifier</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">a_predicate</span><span class="o">=</span><span class="n">is_missense</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">b_predicate</span><span class="o">=</span><span class="n">is_truncating</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">a_label</span><span class="o">=</span><span class="s2">&quot;Missense&quot;</span><span class="p">,</span> <span class="n">b_label</span><span class="o">=</span><span class="s2">&quot;Truncating&quot;</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span><span class="o">.</span><span class="n">class_labels</span>
<span class="go">(&#39;Missense&#39;, &#39;Truncating&#39;)</span>
</pre></div>
</div>
<p>This is a lot of code, and detailed explanations and examples are available in the <a class="reference internal" href="user-guide/analyses/partitioning/index.html#partitioning"><span class="std std-ref">Partitioning</span></a> section.
For now, it is enough to know that the <cite>gt_clf</cite> will assign the individuals
into <cite>Missense</cite> or <cite>Truncating</cite> class. The individuals with the number of missense (or truncating) variants
different than one will be omitted from the analysis.</p>
</section>
<section id="partition-by-phenotype">
<h4><a class="toc-backref" href="#id16" role="doc-backlink">Partition by phenotype</a><a class="headerlink" href="#partition-by-phenotype" title="Link to this heading"></a></h4>
<p>We use HPO terms to assign the individuals into phenotype classes,
according to the term’s presence or exclusion.
The testing leverages the <a class="reference internal" href="user-guide/glossary.html#true-path-rule"><span class="std std-ref">True path rule</span></a> of ontologies.</p>
<p>We now prepare the classifiers for assigning into phenotype classes:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.clf</span><span class="w"> </span><span class="kn">import</span> <span class="n">prepare_classifiers_for_terms_of_interest</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pheno_clfs</span> <span class="o">=</span> <span class="n">prepare_classifiers_for_terms_of_interest</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">hpo</span><span class="o">=</span><span class="n">hpo</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="multiple-testing-correction">
<h4><a class="toc-backref" href="#id17" role="doc-backlink">Multiple testing correction</a><a class="headerlink" href="#multiple-testing-correction" title="Link to this heading"></a></h4>
<p>By default, GPSEA performs a test for each HPO term used to annotate at least one individual in the cohort,
and there are 369 such terms in <em>TBX5</em> cohort:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">pheno_clfs</span><span class="p">)</span>
<span class="go">369</span>
</pre></div>
</div>
<p>However, testing multiple hypothesis on the same dataset increases the chance of receiving false positive result.
Luckily, GPSEA simplifies the application of an appropriate multiple testing correction.</p>
<p>For general use, we recommend using a combination
of a <em>phenotype MT filter</em> (<a class="reference internal" href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcFilter" title="gpsea.analysis.mtc_filter.PhenotypeMtcFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">PhenotypeMtcFilter</span></code></a>) with a <em>multiple testing correction</em>.
Phenotype MT filter chooses the HPO terms to test according to several heuristics, which
reduce the multiple testing burden and focus the analysis
on the most interesting terms (see <a class="reference internal" href="user-guide/analyses/mtc.html#hpo-if-filter"><span class="std std-ref">Independent filtering for HPO</span></a> for more info).
Then the multiple testing correction, such as Bonferroni or Benjamini-Hochberg,
is used to control the family-wise error rate or the false discovery rate.
See <a class="reference internal" href="user-guide/analyses/mtc.html#mtc"><span class="std std-ref">Multiple-testing correction</span></a> for more information.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.pcats</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_hpo_term_analysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">analysis</span> <span class="o">=</span> <span class="n">configure_hpo_term_analysis</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
</pre></div>
</div>
<p><a class="reference internal" href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.configure_hpo_term_analysis" title="gpsea.analysis.pcats.configure_hpo_term_analysis"><code class="xref py py-func docutils literal notranslate"><span class="pre">configure_hpo_term_analysis()</span></code></a> configures the analysis
that uses Independent filtering for HPO (<a class="reference internal" href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.IfHpoFilter" title="gpsea.analysis.mtc_filter.IfHpoFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">IfHpoFilter</span></code></a>) for selecting HPO terms of interest,
Fisher Exact test for computing nominal p values, and Benjamini-Hochberg for multiple testing correction.</p>
</section>
</section>
<section id="statistical-testing">
<h3><a class="toc-backref" href="#id18" role="doc-backlink">Statistical testing</a><a class="headerlink" href="#statistical-testing" title="Link to this heading"></a></h3>
<p>Now we can perform the testing and evaluate the results.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span> <span class="o">=</span> <span class="n">analysis</span><span class="o">.</span><span class="n">compare_genotype_vs_phenotypes</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">gt_clf</span><span class="o">=</span><span class="n">gt_clf</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">pheno_clfs</span><span class="o">=</span><span class="n">pheno_clfs</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">total_tests</span>
<span class="go">30</span>
</pre></div>
</div>
<p>We only tested 30 HPO terms. This is despite the individuals being collectively annotated with
369 direct and indirect HPO terms</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">result</span><span class="o">.</span><span class="n">phenotypes</span><span class="p">)</span>
<span class="go">369</span>
</pre></div>
</div>
<p>We can show the reasoning behind <em>not</em> testing 339 (<cite>369 - 30</cite>) HPO terms
by exploring the phenotype MTC filtering report:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">MtcStatsViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_viewer</span> <span class="o">=</span> <span class="n">MtcStatsViewer</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_report</span> <span class="o">=</span> <span class="n">mtc_viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_report</span>
</pre></div>
</div>
<div>
    <style>
        /** Light theme is the default */
        :root {
            --gpsea-color-text: black;
            --gpsea-color-link-text: rgb(0, 118, 187);
            --gpsea-color-link-hover: rgb(0, 99, 156);
            --gpsea-color-primary-a0: #2f5170;
            --gpsea-color-primary-a10: #4778a5;
            --gpsea-color-primary-a20: #5995ce;
            --gpsea-color-primary-a60: LightSkyBlue;
            --gpsea-color-surface-a0: #fcfcfc;
            --gpsea-color-surface-a10: #ffffff;
            --gpsea-color-surface-a20: #b3b3b3;
            --gpsea-color-surface-a30: #8d8d8d;
            --gpsea-color-surface-a40: #646464;
            --gpsea-color-surface-a50: #000000;
        }

        /** Dark theme */
        /*
        @media (prefers-color-scheme: dark) {
            :root {
                --gpsea-color-text: rgb(240, 240, 240);
                --gpsea-color-link-text: rgb(240, 240, 240);
                --gpsea-color-link-hover: lightblue;
                --gpsea-color-primary-a0: #64a3df;
                --gpsea-color-primary-a10: #5286b8;
                --gpsea-color-primary-a20: #375c7e;
                --gpsea-color-primary-a60: #233b52;
                --gpsea-color-surface-a0: #121212;
                --gpsea-color-surface-a10: #282828;
                --gpsea-color-surface-a20: #3f3f3f;
                --gpsea-color-surface-a30: #575757;
                --gpsea-color-surface-a40: #717171;
                --gpsea-color-surface-a50: #8b8b8b;
            }
        }
        */

        #gpsea-body {
            max-width: 1600px;
            color: var(--gpsea-color-text);
            background-color: var(--gpsea-color-surface-a0);
            padding: 10px 0;
        }

        #gpsea-body {

            h1,
            h2,
            h3,
            h4,
            h5 {
                color: var(--gpsea-color-primary-a0);
            }

            a {
                color: var(--gpsea-color-link-text);
                text-decoration: none;
            }

            a:hover {
                color: var(--gpsea-color-link-hover);
            }

            #cohort-summary {
                max-width: 800px;
            }

            td {
                padding: 5px 0;
            }

            .cards {
                display: flex;
                flex-flow: row wrap;
                align-items: top;
                gap: 1rem;
            }

            .card {
                border-radius: 5px;
                padding: 0px 10px;
                background: var(--gpsea-color-surface-a10);
                border: 1px solid var(--gpsea-color-surface-a30);
                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);
            }

            table {
                width: 100%;
                border-collapse: collapse;
                margin: 10px 0;
                font-size: 0.9em;
                font-family: sans-serif;
                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
                border: 1px solid var(--gpsea-color-primary-a60);
            }

            th {
                text-align: left;
                padding: 2px;
                font-weight: bold;
                font-size: 120%;
                color: var(--gpsea-color-text);
                background-color: var(--gpsea-color-primary-a60);
            }

            table td,
            tr {
                text-align: left;
                padding: 0px 5px;
            }

            tr:nth-child(even) {
                background-color: var(--gpsea-color-surface-a10);
            }

            tr:nth-child(odd) {
                background-color: var(--gpsea-color-surface-a20);
            }

            caption {
                caption-side: top;
                text-align: left;
                padding-bottom: 10px;
                font-weight: bold;
            }
        }
    </style>
    <div id="gpsea-body">
        
<h1>Phenotype testing report</h1>
<div class="cards">
  <div class="card">
    <ul>
      <li>Phenotype MTC filter: <em>Independent filtering HPO filter</em></li>
      <li>Multiple testing correction: <em>fdr_bh</em></li>
    </ul>
  </div>
  <div class="card">
    <table>
      <tbody>
        <tr>
          <th>Reason</th>
          <th>Count</th>
        </tr>
        
        <tr>
          
            <td><a href="https://monarch-initiative.github.io/gpsea/stable/user-guide/analyses/mtc.html/#skip-terms-if-all-counts-are-identical-to-counts-for-a-child-term">Skipping term because of a child term with the same individual counts</a></td>
          
          <td>1</td>
        </tr>
        
        <tr>
          
            <td><a href="https://monarch-initiative.github.io/gpsea/stable/user-guide/analyses/mtc.html/#skipping-general-level-terms">Skipping general term</a></td>
          
          <td>48</td>
        </tr>
        
        <tr>
          
            <td><a href="https://monarch-initiative.github.io/gpsea/stable/user-guide/analyses/mtc.html/#skipping-terms-that-are-rare-on-the-cohort-level">Skipping term that was annotated to less than 40% of the cohort members</a></td>
          
          <td>290</td>
        </tr>
        
      </tbody>
      <caption>
        Performed statistical tests for 30 out of the total of 369 HPO terms.
      </caption>
    </table>
  </div>
</div>

    </div>
</div><p>and these are the tested HPO terms ordered by the p value corrected with the Benjamini-Hochberg procedure:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">summarize_hpo_analysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span> <span class="o">=</span> <span class="n">summarize_hpo_analysis</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">result</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span>
</pre></div>
</div>
<table class="docutils align-default" id="id5">
<caption><span class="caption-text"><em>TBX5</em> truncating vs. missense</span><a class="headerlink" href="#id5" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"></th>
<th class="head"><p>Missense</p></th>
<th class="head"><p>Truncating</p></th>
<th class="head"><p>Corrected p values</p></th>
<th class="head"><p>p values</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Ventricular septal defect [HP:0001629]</p></td>
<td><p>31/60 (52%)</p></td>
<td><p>29/29 (100%)</p></td>
<td><p>1.685445780374555e-05</p></td>
<td><p>5.618152601248516e-07</p></td>
</tr>
<tr class="row-odd"><td><p>Patent foramen ovale [HP:0001655]</p></td>
<td><p>4/40 (10%)</p></td>
<td><p>0/62 (0%)</p></td>
<td><p>0.26028179887989955</p></td>
<td><p>0.021505679979762674</p></td>
</tr>
<tr class="row-even"><td><p>Triphalangeal thumb [HP:0001199]</p></td>
<td><p>13/72 (18%)</p></td>
<td><p>21/57 (37%)</p></td>
<td><p>0.26028179887989955</p></td>
<td><p>0.026028179887989958</p></td>
</tr>
<tr class="row-odd"><td><p>Absent thumb [HP:0009777]</p></td>
<td><p>12/71 (17%)</p></td>
<td><p>19/57 (33%)</p></td>
<td><p>0.2891465722255393</p></td>
<td><p>0.03855287629673857</p></td>
</tr>
<tr class="row-even"><td><p>Complete atrioventricular canal defect [HP:0001674]</p></td>
<td><p>5/37 (14%)</p></td>
<td><p>3/63 (5%)</p></td>
<td><p>0.8049542711586962</p></td>
<td><p>0.1419239358441169</p></td>
</tr>
<tr class="row-odd"><td><p>Perimembranous ventricular septal defect [HP:0011682]</p></td>
<td><p>3/59 (5%)</p></td>
<td><p>6/47 (13%)</p></td>
<td><p>0.8049542711586962</p></td>
<td><p>0.18069478744841608</p></td>
</tr>
<tr class="row-even"><td><p>Upper limb phocomelia [HP:0009813]</p></td>
<td><p>8/85 (9%)</p></td>
<td><p>2/65 (3%)</p></td>
<td><p>0.8049542711586962</p></td>
<td><p>0.18782266327036246</p></td>
</tr>
<tr class="row-odd"><td><p>Short thumb [HP:0009778]</p></td>
<td><p>11/41 (27%)</p></td>
<td><p>22/56 (39%)</p></td>
<td><p>1.0</p></td>
<td><p>0.2782218994250125</p></td>
</tr>
<tr class="row-even"><td><p>Hypoplasia of the radius [HP:0002984]</p></td>
<td><p>30/62 (48%)</p></td>
<td><p>10/27 (37%)</p></td>
<td><p>1.0</p></td>
<td><p>0.3614379675325876</p></td>
</tr>
<tr class="row-odd"><td><p>Muscular ventricular septal defect [HP:0011623]</p></td>
<td><p>6/59 (10%)</p></td>
<td><p>8/47 (17%)</p></td>
<td><p>1.0</p></td>
<td><p>0.3894578351610228</p></td>
</tr>
<tr class="row-even"><td><p>Amelia involving the upper limbs [HP:0009812]</p></td>
<td><p>0/83 (0%)</p></td>
<td><p>1/65 (2%)</p></td>
<td><p>1.0</p></td>
<td><p>0.4391891891891892</p></td>
</tr>
<tr class="row-odd"><td><p>Atrial septal defect [HP:0001631]</p></td>
<td><p>42/44 (95%)</p></td>
<td><p>38/38 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>0.49653718759409815</p></td>
</tr>
<tr class="row-even"><td><p>Secundum atrial septal defect [HP:0001684]</p></td>
<td><p>14/35 (40%)</p></td>
<td><p>13/40 (32%)</p></td>
<td><p>1.0</p></td>
<td><p>0.6304400561799244</p></td>
</tr>
<tr class="row-odd"><td><p>Short 1st metacarpal [HP:0010034]</p></td>
<td><p>0/30 (0%)</p></td>
<td><p>1/35 (3%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Common atrium [HP:0011565]</p></td>
<td><p>0/83 (0%)</p></td>
<td><p>0/66 (0%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Unroofed coronary sinus [HP:0031297]</p></td>
<td><p>0/85 (0%)</p></td>
<td><p>0/66 (0%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal thumb morphology [HP:0001172]</p></td>
<td><p>30/30 (100%)</p></td>
<td><p>56/56 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal finger morphology [HP:0001167]</p></td>
<td><p>36/36 (100%)</p></td>
<td><p>56/56 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal digit morphology [HP:0011297]</p></td>
<td><p>38/38 (100%)</p></td>
<td><p>59/59 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal atrial septum morphology [HP:0011994]</p></td>
<td><p>43/43 (100%)</p></td>
<td><p>38/38 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal cardiac atrium morphology [HP:0005120]</p></td>
<td><p>43/43 (100%)</p></td>
<td><p>38/38 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal long bone morphology [HP:0011314]</p></td>
<td><p>44/44 (100%)</p></td>
<td><p>19/19 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal hand morphology [HP:0005922]</p></td>
<td><p>53/53 (100%)</p></td>
<td><p>41/41 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia of the extremities [HP:0009815]</p></td>
<td><p>55/55 (100%)</p></td>
<td><p>44/44 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving bones of the upper limbs [HP:0006496]</p></td>
<td><p>55/55 (100%)</p></td>
<td><p>44/44 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia involving bones of the extremities [HP:0045060]</p></td>
<td><p>55/55 (100%)</p></td>
<td><p>44/44 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving the skeleton [HP:0009115]</p></td>
<td><p>56/56 (100%)</p></td>
<td><p>45/45 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiac septum morphology [HP:0001671]</p></td>
<td><p>62/62 (100%)</p></td>
<td><p>50/50 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal appendicular skeleton morphology [HP:0011844]</p></td>
<td><p>64/64 (100%)</p></td>
<td><p>60/60 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Absent radius [HP:0003974]</p></td>
<td><p>7/32 (22%)</p></td>
<td><p>7/35 (20%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
</tbody>
</table>
<p>We see that several HPO terms are significantly associated
with presence of a truncating variant in <em>TBX5</em>.
For example, <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001629">Ventricular septal defect</a>
was observed in 31/60 (52%) patients with a missense variant
but it was observed in 29/29 (100%) patients with a truncating variant.
Fisher exact test computed a p value of 5.61e<sup>-7</sup>
and the p value corrected by Benjamini-Hochberg procedure
is 9.55e<sup>-6</sup>.</p>
</section>
</section>
<section id="conclusion">
<h2>Conclusion<a class="headerlink" href="#conclusion" title="Link to this heading"></a></h2>
<p>We showed the high-level structure of genotype-phenotype association analysis using GPSEA
and we found an association between truncating <em>TBX5</em> variants
and <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001629">Ventricular septal defect</a>.</p>
<p>This is just one of many analysis types that are possible with GPSEA.
Please refer to the user guide (next section) to learn more.</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="setup.html" class="btn btn-neutral float-left" title="Setup" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="user-guide/index.html" class="btn btn-neutral float-right" title="User guide" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>