<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Tutorial &mdash; genophenocorr 0.1.1dev documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="_static/genophenocorr.css?v=a9fa5eb9" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=8427294d"></script>
        <script src="_static/doctools.js?v=9a2dae69"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="_static/copybutton.js?v=6dbb43f8"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="User guide" href="user-guide/index.html" />
    <link rel="prev" title="Installation" href="installation.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            genophenocorr
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Tutorial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#set-up-analysis">Set up analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#prepare-samples">Prepare samples</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">genophenocorr</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Tutorial</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/tutorial.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="tutorial">
<span id="id1"></span><h1>Tutorial<a class="headerlink" href="#tutorial" title="Link to this heading"></a></h1>
<p>The tutorial demonstrates how to load an example Phenopacket cohort and perform genotype-phenotype analysis.</p>
<section id="set-up-analysis">
<h2>Set up analysis<a class="headerlink" href="#set-up-analysis" title="Link to this heading"></a></h2>
<p><cite>genophenocorr</cite> needs HPO to do the analysis. Let’s load the ontology:</p>
<div class="highlight-pycon notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">os</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">load_minimal_ontology</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="s1">&#39;docs&#39;</span><span class="p">,</span> <span class="s1">&#39;data&#39;</span><span class="p">,</span> <span class="s1">&#39;hp.toy.json&#39;</span><span class="p">))</span>
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Use the latest HPO which you can get at <cite>http://purl.obolibrary.org/obo/hp.json</cite></p>
</div>
<p>TODO - move the code from <cite>workflow</cite> and the notebook here.</p>
</section>
<section id="prepare-samples">
<h2>Prepare samples<a class="headerlink" href="#prepare-samples" title="Link to this heading"></a></h2>
<p>Now we need some samples. To keep things simple in this tutorial, we will use a toy cohort that is shipped
with the package:</p>
<div class="highlight-pycon notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.data</span> <span class="kn">import</span> <span class="n">get_toy_cohort</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span> <span class="o">=</span> <span class="n">get_toy_cohort</span><span class="p">()</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>See <a class="reference internal" href="user-guide/input-data.html#input-data"><span class="std std-ref">Input data</span></a> section to learn about preparing your data for the analysis.</p>
</div>
<p>We can then view the data using the list commands.</p>
<div class="highlight-pycon notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">sorted</span><span class="p">(</span><span class="n">cohort</span><span class="o">.</span><span class="n">get_patient_ids</span><span class="p">())</span>
<span class="go">[&#39;A&#39;, &#39;B&#39;, &#39;C&#39;, &#39;D&#39;, &#39;E&#39;, &#39;F&#39;, &#39;G&#39;, &#39;H&#39;, &#39;I&#39;, &#39;J&#39;, &#39;K&#39;, &#39;L&#39;, &#39;M&#39;, &#39;N&#39;, &#39;O&#39;, &#39;P&#39;, &#39;Q&#39;, &#39;R&#39;, &#39;S&#39;, &#39;T&#39;, &#39;U&#39;, &#39;V&#39;, &#39;W&#39;, &#39;X&#39;, &#39;Y&#39;, &#39;Z&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">sorted</span><span class="p">(</span><span class="n">cohort</span><span class="o">.</span><span class="n">list_present_phenotypes</span><span class="p">())</span>
<span class="go">[(&#39;HP:0001166&#39;, 14), (&#39;HP:0001250&#39;, 20), (&#39;HP:0001257&#39;, 17)]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">sorted</span><span class="p">(</span><span class="n">cohort</span><span class="o">.</span><span class="n">list_all_variants</span><span class="p">())</span>
<span class="go">[(&#39;1_281_281_A_G&#39;, 16), (&#39;1_361_363_TTC_T&#39;, 13)]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">sorted</span><span class="p">(</span><span class="n">cohort</span><span class="o">.</span><span class="n">list_all_proteins</span><span class="p">())</span>
<span class="go">[(&#39;NP_09876.5&#39;, 29)]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_dict</span> <span class="o">=</span> <span class="n">cohort</span><span class="o">.</span><span class="n">variant_effect_count_by_tx</span><span class="p">(</span><span class="n">tx_id</span><span class="o">=</span><span class="s1">&#39;NM_1234.5&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">sorted</span><span class="p">(</span><span class="n">tx_dict</span><span class="p">[</span><span class="s1">&#39;NM_1234.5&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">items</span><span class="p">())</span>
<span class="go">[(&#39;FRAMESHIFT_VARIANT&#39;, 1), (&#39;MISSENSE_VARIANT&#39;, 1)]</span>
</pre></div>
</div>
<p>Using the counts, we can choose and run what analyses we want.</p>
<p>For instance, we can partition the patients into two groups based on presence/absence of a <em>missense</em> variant:</p>
<div class="highlight-pycon notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>

<span class="gp">&gt;&gt;&gt; </span><span class="n">missense_in_tx</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="s1">&#39;NM_1234.5&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>We created a predicate <cite>missense_in_tx</cite> that tests
if a variant leads to an aminoacid change in a transcript <cite>NM_1234.5</cite>.
We will use the predicate in the downstream analysis to assign the patients
into a group and test for genotype-phenotype correlation between the groups.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>There are many other ways to set up a predicate for testing
for genotype-phenotype correlation.
See the <a class="reference internal" href="user-guide/predicates.html#predicates"><span class="std std-ref">Predicates</span></a> section to learn more about building
a predicate for the analysis of interest.</p>
</div>
<p>Next, we will prepare and run the genotype-phenotype analysis
to test the correlation between phenotypes and presence of a missense variant.</p>
<div class="highlight-pycon notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.analysis</span> <span class="kn">import</span> <span class="n">configure_cohort_analysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">genophenocorr.analysis.predicate</span> <span class="kn">import</span> <span class="n">PatientCategories</span>  <span class="c1"># TODO - explain the predicate or update the API</span>

<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_analysis</span> <span class="o">=</span> <span class="n">configure_cohort_analysis</span><span class="p">(</span><span class="n">cohort</span><span class="p">,</span> <span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">missense</span> <span class="o">=</span> <span class="n">cohort_analysis</span><span class="o">.</span><span class="n">compare_hpo_vs_genotype</span><span class="p">(</span><span class="n">missense_in_tx</span><span class="p">)</span>
</pre></div>
</div>
<p>We prepared the <cite>cohort_analysis</cite> and we ran the analysis to test for correlation
between the clinical signs and symptoms of patients encoded into HPO terms
and presence of a missense variant in the patient.</p>
<p>We stored the results into <cite>missense</cite> variable. We can summarize the results
by showing a data frame with results for each tested HPO term.
We show the first two terms here:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pd</span><span class="o">.</span><span class="n">set_option</span><span class="p">(</span><span class="s1">&#39;expand_frame_repr&#39;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span> <span class="o">=</span> <span class="n">missense</span><span class="o">.</span><span class="n">summarize</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">PatientCategories</span><span class="o">.</span><span class="n">YES</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">2</span><span class="p">)</span>  
<span class="go">  MISSENSE_VARIANT on NM_1234.5    Yes             No</span>
<span class="go">                                  Count   Percent Count Percent   p value    Corrected p value</span>
<span class="go">  Arachnodactyly [HP:0001166]     13/16   81%     1/10  10%       0.000781   0.020299</span>
<span class="go">  Spasticity [HP:0001257]         11/16   69%     6/10  60%       0.692449   1.000000</span>
</pre></div>
</div>
<blockquote>
<div><p>We’re showing just 1 row above. This is due to 2-.. rows all having corrected p value of <cite>1.000</cite> resulting
in unstable sort order. We can show more rows with a better cohort, as soon as we have it!</p>
</div></blockquote>
<blockquote>
<div><p>We can show analysis for <cite>VariantEffect.FRAMESHIFT_VARIANT</cite> as well..</p>
</div></blockquote>
<p>The table presents the HPO terms that annotate the cohort members and report their counts and p values
for each genotype group. The rows are sorted by the corrected p value in ascending order.</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="installation.html" class="btn btn-neutral float-left" title="Installation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="user-guide/index.html" class="btn btn-neutral float-right" title="User guide" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>