<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Tutorial &mdash; GPSEA 0.1.1dev documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="_static/gpsea.css?v=547657ed" />

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=8427294d"></script>
        <script src="_static/doctools.js?v=9a2dae69"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="_static/copybutton.js?v=6dbb43f8"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="User guide" href="user-guide/index.html" />
    <link rel="prev" title="Installation" href="installation.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            GPSEA
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Tutorial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#the-analysis">The analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#load-hpo">Load HPO</a></li>
<li class="toctree-l3"><a class="reference internal" href="#prepare-cohort">Prepare cohort</a></li>
<li class="toctree-l3"><a class="reference internal" href="#explore-cohort">Explore cohort</a></li>
<li class="toctree-l3"><a class="reference internal" href="#prepare-genotype-and-phenotype-predicates">Prepare genotype and phenotype predicates</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Tutorial</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/tutorial.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="tutorial">
<span id="id1"></span><h1>Tutorial<a class="headerlink" href="#tutorial" title="Link to this heading"></a></h1>
<p>Here we demonstrate an end-to-end genotype-phenotype analysis with GPSEA.
The tutorial illustrates just one of the many ways GPSEA can be used to characterize genotype-phenotype correlations.
The <a class="reference external" href="https://ielis.github.io/hpo-toolkit/stable/user-guide/index.html#user-guide" title="(in hpo-toolkit)"><span>User guide</span></a> contains details about additional methods and functionalities.</p>
<p>The tutorial presents an analysis of a cohort of individuals with pathogenic variants in <em>TBX5</em> leading to
<a class="reference external" href="https://omim.org/entry/142900">Holt-Oram syndrome MIM:142900</a>.</p>
<p>Holt-Oram syndrome is an autosomal dominant disorder characterized by
upper limb defects, congenital heart defects, and arrhythmias (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/38336121/">PMID:38336121</a>).
It has been observed in the literature that congenital defects of the ventricular and atrial septum are more
common in the truncating than in the missense variants (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30552424/">PMID:30552424</a>).
Additionally, upper limb defects are more frequent in patients with protein-truncating variants (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/38336121/">PMID:38336121</a>).</p>
<p>To perform the analysis, we curated the literature and created on <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/35705716/">GA4GH phenopacket</a> for each
affected individual. The phenopackets are made available in <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a>.</p>
<section id="the-analysis">
<h2>The analysis<a class="headerlink" href="#the-analysis" title="Link to this heading"></a></h2>
<p>For the analysis, the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/refseq/MANE/">MANE</a> transcript
(i.e., the “main” biomedically relevant transcript of a gene) should be chosen unless
there is a specific reason not to (which should occur rarely if at all).</p>
<p>In the case of <em>TBX5</em> the MANE transcript is <cite>NM_181486.4</cite>. Note that the trascript identifier (<cite>NM_181486</cite>) and the version (<cite>4</cite>) are both required.
A good way to find the MANE transcript is to search on the gene symbol (e.g., <em>TBX5</em>) in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a> and to
choose a variant that is specifically located in the gene. The MANE transcript will be displayed here (e.g., <a class="reference external" href="https://www.ncbi.nlm.nih.gov/clinvar/variation/495227/">NM_181486.4(TBX5):c.1221C&gt;G (p.Tyr407Ter)</a>).</p>
<p>We additionally need the corresponding protein identifier.
A good way to find this is to search on the transcript id in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/">NCBI Nucleotide</a>.
In our case, search on <cite>NM_181486.4</cite> will bring us to <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/NM_181486.4">this page</a>.
If we search within this page for <cite>“NP_”</cite>, this will bring us to the
corresponding protein accession <cite>NP_852259.1</cite>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_name</span> <span class="o">=</span> <span class="s1">&#39;TBX5&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_181486.4&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">px_id</span> <span class="o">=</span> <span class="s1">&#39;NP_852259.1&#39;</span>
</pre></div>
</div>
<section id="load-hpo">
<h3>Load HPO<a class="headerlink" href="#load-hpo" title="Link to this heading"></a></h3>
<p>GPSEA needs HPO to do the analysis.
We use HPO toolkit to load HPO version <cite>v2024-07-01</cite>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Use the latest HPO release by omitting the <cite>release</cite> option.</p>
</div>
</section>
<section id="prepare-cohort">
<h3>Prepare cohort<a class="headerlink" href="#prepare-cohort" title="Link to this heading"></a></h3>
<p>Now we will load the samples to analyze. We will use the cohort of 156 individuals with mutations in <em>TBX5</em>
whose clinical signs and symptoms were encoded into HPO terms
and stored in <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">ppktstore.registry</span> <span class="kn">import</span> <span class="n">configure_phenopacket_registry</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">phenopacket_registry</span> <span class="o">=</span> <span class="n">configure_phenopacket_registry</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">phenopacket_registry</span><span class="o">.</span><span class="n">open_phenopacket_store</span><span class="p">(</span><span class="s1">&#39;0.1.18&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">ps</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">phenopackets</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">ps</span><span class="o">.</span><span class="n">iter_cohort_phenopackets</span><span class="p">(</span><span class="n">cohort_name</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">)</span>
<span class="go">156</span>
</pre></div>
</div>
<p>We loaded 156 phenopackets which need further preprocessing to prepare for the analysis.
We will compute functional annotations for the mutations and then include the individuals into
a <a class="reference internal" href="apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span><span class="p">,</span> <span class="n">load_phenopackets</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">validation</span> <span class="o">=</span> <span class="n">load_phenopackets</span><span class="p">(</span>  
<span class="gp">... </span>    <span class="n">phenopackets</span><span class="o">=</span><span class="n">phenopackets</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">cohort_creator</span><span class="o">=</span><span class="n">cohort_creator</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="go">Patients Created: ...</span>
</pre></div>
</div>
<p>and we will check that there are no Q/C issues:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">validation</span><span class="o">.</span><span class="n">summarize</span><span class="p">()</span>  
<span class="go">Validated under none policy</span>
<span class="go">No errors or warnings were found</span>
</pre></div>
</div>
<p>We loaded the patient data into a <cite>cohort</cite> which is ready for the next steps.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>Here we show how to create a <a class="reference internal" href="apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a> from phenopackets.
See <a class="reference internal" href="user-guide/input-data.html#input-data"><span class="std std-ref">Input data</span></a> section to learn how to create a cohort from another inputs.</p>
</div>
</section>
<section id="explore-cohort">
<h3>Explore cohort<a class="headerlink" href="#explore-cohort" title="Link to this heading"></a></h3>
<p>We can now explore the cohort to see how many patients are included.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">CohortViewable</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">viewer</span> <span class="o">=</span> <span class="n">CohortViewable</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span> <span class="o">=</span> <span class="n">viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span> <span class="n">transcript_id</span><span class="o">=</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;docs/report/tbx5_cohort_info.html&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">fh</span><span class="p">:</span>  
<span class="gp">... </span>    <span class="n">_</span> <span class="o">=</span> <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">report</span><span class="p">)</span>
</pre></div>
</div>
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Cohort</title>
    <style>
    table {
      border-collapse: collapse;
      margin: 25px 0;
      font-size: 0.9em;
      font-family: sans-serif;
      min-width: 400px;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
    }

    .table .column-1 {
      text-align: left;
    }
    th {
      background-color: LightSkyBlue;
      border: 1px solid #dddddd;
      text-align: left;
      padding: 2px;
      font-weight: bold;
      font-size: 120%;
    }

    tr {
      border: 1px solid #dddddd;
    }

    td {
      padding: 2px;
      font-weight: bold;
    }

    tr:nth-child(even) {
      background-color: #f2f2f2;
    }

    .table td, tr {
      text-align: right;
    }

    .lft {
      text-align: left;
    }

    div {
      display: block;
      width: 100%;
      padding-right: 1%;
      margin-bottom: 20px; /* Add margin to separate divs */
    }

    caption {
      caption-side: top;
      text-align: left;
      padding-bottom: 10px;
      font-weight: bold;
    }
  </style>
</head>

<body>
  <h1>GPSEA cohort analysis</h1>
  <p>Successfully loaded 156 individuals.</p>
  
  <p>No errors encountered.</p>
  

    <table>
    <caption style="color: black;">
        <h3>Top 10 HPO Terms</h3>
        A total of 116 HPO terms were used to annotated the cohort.
    </caption>
        <tbody>
            <tr class="strng">
                <th class="lft">HPO Term</th>
                <th>ID</th>
                <th>Seen in <em>n</em> individuals</th>
            </tr>
            
            <tr>
                <td class="lft">Atrial septal defect</td>
                <td>HP:0001631</td>
                <td>50</td>
                </tr>
            
            <tr>
                <td class="lft">Ventricular septal defect</td>
                <td>HP:0001629</td>
                <td>41</td>
                </tr>
            
            <tr>
                <td class="lft">Hypoplasia of the radius</td>
                <td>HP:0002984</td>
                <td>40</td>
                </tr>
            
            <tr>
                <td class="lft">Triphalangeal thumb</td>
                <td>HP:0001199</td>
                <td>36</td>
                </tr>
            
            <tr>
                <td class="lft">Absent thumb</td>
                <td>HP:0009777</td>
                <td>32</td>
                </tr>
            
            <tr>
                <td class="lft">Short thumb</td>
                <td>HP:0009778</td>
                <td>32</td>
                </tr>
            
            <tr>
                <td class="lft">Abnormal carpal morphology</td>
                <td>HP:0001191</td>
                <td>30</td>
                </tr>
            
            <tr>
                <td class="lft">Secundum atrial septal defect</td>
                <td>HP:0001684</td>
                <td>27</td>
                </tr>
            
            <tr>
                <td class="lft">Absent radius</td>
                <td>HP:0003974</td>
                <td>15</td>
                </tr>
            
            <tr>
                <td class="lft">Cardiac conduction abnormality</td>
                <td>HP:0031546</td>
                <td>14</td>
                </tr>
            
                </tbody>
    </table>

    <table>
    <caption style="color: black;">
        <h3>Top 10 Variants</h3>
        Variants are shown according to NM_181486.4. A total of 156 unique variants were identified in the cohort.
    </caption>
            <tbody>
                <tr class="strng">
                    <th>Count</th>
                    <th class="lft">Variant key</th>
                    <th>Variant Name</th>
                    <th>Protein Variant</th>
                    <th>Variant Class</th>
                </tr>
                
                <tr>
                    <td>22</td>
                    <td class="lft">12_114385521_114385521_C_T</td>
                    <td>c.710G>A</td>
                    <td>p.Arg237Gln</td>
                    <td>MISSENSE_VARIANT</td>
                </tr>
                
                <tr>
                    <td>20</td>
                    <td class="lft">12_114401830_114401830_C_T</td>
                    <td>c.238G>A</td>
                    <td>p.Gly80Arg</td>
                    <td>MISSENSE_VARIANT</td>
                </tr>
                
                <tr>
                    <td>8</td>
                    <td class="lft">12_114385563_114385563_G_A</td>
                    <td>c.668C>T</td>
                    <td>p.Thr223Met</td>
                    <td>MISSENSE_VARIANT</td>
                </tr>
                
                <tr>
                    <td>6</td>
                    <td class="lft">12_114398675_114398675_G_T</td>
                    <td>c.408C>A</td>
                    <td>p.Tyr136Ter</td>
                    <td>STOP_GAINED</td>
                </tr>
                
                <tr>
                    <td>5</td>
                    <td class="lft">12_114398682_114398682_C_CG</td>
                    <td>c.400dup</td>
                    <td>p.Arg134ProfsTer49</td>
                    <td>FRAMESHIFT_VARIANT</td>
                </tr>
                
                <tr>
                    <td>5</td>
                    <td class="lft">12_114403792_114403792_C_CG</td>
                    <td>c.106_107insC</td>
                    <td>p.Ser36ThrfsTer25</td>
                    <td>FRAMESHIFT_VARIANT</td>
                </tr>
                
                <tr>
                    <td>5</td>
                    <td class="lft">12_114399514_114399514_A_C</td>
                    <td>c.361T>G</td>
                    <td>p.Trp121Gly</td>
                    <td>MISSENSE_VARIANT, SPLICE_REGION_VARIANT</td>
                </tr>
                
                <tr>
                    <td>4</td>
                    <td class="lft">12_114385474_114385474_A_G</td>
                    <td>c.755+2T>C</td>
                    <td>None</td>
                    <td>SPLICE_DONOR_VARIANT</td>
                </tr>
                
                <tr>
                    <td>4</td>
                    <td class="lft">12_114398656_114398656_C_CG</td>
                    <td>c.426dup</td>
                    <td>p.Ala143ArgfsTer40</td>
                    <td>FRAMESHIFT_VARIANT</td>
                </tr>
                
                <tr>
                    <td>4</td>
                    <td class="lft">12_114366360_114366360_C_T</td>
                    <td>c.787G>A</td>
                    <td>p.Val263Met</td>
                    <td>MISSENSE_VARIANT</td>
                </tr>
                
            </tbody>
        </table>
    <table>
    <caption style="color: black;">
        <h3>Diseases</h3>
    </caption>
        <tbody>
            <tr class="strng">
                <th class="lft">Disease Name</th>
                <th >Disease ID</th>
                <th>Annotation Count</th>
              </tr>
              
              <tr>
                <td class="lft">Holt-Oram syndrome</td>
                <td>OMIM:142900</td>
                <td>156</td>
                </tr>
            
           </tbody>
    </table>
    
    <table>
    <caption style="color: black;">
        <h3>Variant categories for NM_181486.4</h3>
    </caption>
        <tbody>
            <tr class="strng">
                <th class="lft">Variant effect</th>
                <th>Annotation Count</th>
              </tr>
              
              <tr>
                <td class="lft">FRAMESHIFT_VARIANT</td>
                <td>38</td>
                </tr>
            
              <tr>
                <td class="lft">MISSENSE_VARIANT</td>
                <td>85</td>
                </tr>
            
              <tr>
                <td class="lft">STOP_GAINED</td>
                <td>19</td>
                </tr>
            
              <tr>
                <td class="lft">SPLICE_REGION_VARIANT</td>
                <td>10</td>
                </tr>
            
              <tr>
                <td class="lft">SPLICE_ACCEPTOR_VARIANT</td>
                <td>2</td>
                </tr>
            
              <tr>
                <td class="lft">SPLICE_DONOR_VARIANT</td>
                <td>7</td>
                </tr>
            
              <tr>
                <td class="lft">SPLICE_DONOR_5TH_BASE_VARIANT</td>
                <td>2</td>
                </tr>
            
              <tr>
                <td class="lft">INTRON_VARIANT</td>
                <td>2</td>
                </tr>
            
              <tr>
                <td class="lft">INFRAME_INSERTION</td>
                <td>2</td>
                </tr>
            
              <tr>
                <td class="lft">STOP_RETAINED_VARIANT</td>
                <td>2</td>
                </tr>
            
              <tr>
                <td class="lft">INFRAME_DELETION</td>
                <td>1</td>
                </tr>
            
           </tbody>
    </table>
  


</body>
</html>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The report can also be displayed directly in a Jupyter notebook by running:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">HTML</span><span class="p">,</span> <span class="n">display</span>
<span class="n">display</span><span class="p">(</span><span class="n">HTML</span><span class="p">(</span><span class="n">report</span><span class="p">))</span>
</pre></div>
</div>
</div>
<p>Now we can show the distribution of variants with respect to the encoded protein.
We first obtain <cite>tx_coordinates</cite> (<a class="reference internal" href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates" title="gpsea.model.TranscriptCoordinates"><code class="xref py py-class docutils literal notranslate"><span class="pre">TranscriptCoordinates</span></code></a>)
and <cite>protein_meta</cite> (<a class="reference internal" href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata" title="gpsea.model.ProteinMetadata"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinMetadata</span></code></a>)
with information about the transcript and protein “anatomy”:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model.genome</span> <span class="kn">import</span> <span class="n">GRCh38</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">configure_protein_metadata_service</span><span class="p">,</span> <span class="n">VVMultiCoordinateService</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">txc_service</span> <span class="o">=</span> <span class="n">VVMultiCoordinateService</span><span class="p">(</span><span class="n">genome_build</span><span class="o">=</span><span class="n">GRCh38</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pms</span> <span class="o">=</span> <span class="n">configure_protein_metadata_service</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span> <span class="o">=</span> <span class="n">txc_service</span><span class="o">.</span><span class="n">fetch</span><span class="p">(</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">protein_meta</span> <span class="o">=</span> <span class="n">pms</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="n">px_id</span><span class="p">)</span>
</pre></div>
</div>
<p>and we follow with plotting the diagram of the mutations on the protein:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">ProteinVisualizer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span> <span class="mi">8</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">visualizer</span> <span class="o">=</span> <span class="n">ProteinVisualizer</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">visualizer</span><span class="o">.</span><span class="n">draw_protein_diagram</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">tx_coordinates</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">protein_meta</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;docs/img/tutorial/tbx5_protein_diagram.png&#39;</span><span class="p">)</span>  
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/tbx5_protein_diagram.png"><img alt="TBX5 protein diagram" class="align-center" src="_images/tbx5_protein_diagram.png" style="width: 600px;" /></a>
</section>
<section id="prepare-genotype-and-phenotype-predicates">
<h3>Prepare genotype and phenotype predicates<a class="headerlink" href="#prepare-genotype-and-phenotype-predicates" title="Link to this heading"></a></h3>
<p>We will create a predicate to bin patients into group
depending on presence of a missense and frameshift variant to test
if there is a difference between frameshift and non-frameshift variants
in the individuals of the <em>TBX5</em> cohort.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span><span class="p">,</span> <span class="n">groups_predicate</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span> <span class="o">=</span> <span class="n">groups_predicate</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">predicates</span><span class="o">=</span><span class="p">(</span>
<span class="gp">... </span>        <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">),</span>
<span class="gp">... </span>        <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">FRAMESHIFT_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">)</span>
<span class="gp">... </span>    <span class="p">),</span>
<span class="gp">... </span>    <span class="n">group_names</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;Missense&#39;</span><span class="p">,</span> <span class="s1">&#39;Frameshift&#39;</span><span class="p">),</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;Genotype group: Missense, Frameshift&#39;</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>There are many other ways to set up a predicate for testing
for a GP correlation.
See the <a class="reference internal" href="user-guide/predicates.html#predicates"><span class="std std-ref">Predicates</span></a> section to learn more about building
a predicate of interest.</p>
</div>
<p>The phenotype grouping is based on presence or absence of an HPO term.
We take advantage of the ontology “true path rule” to infer presence
of the ancestor terms for all present HPO terms
(e.g. presence of <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0010438">Abnormal ventricular septum morphology</a>
in an individual annotated with <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001629">Ventricular septal defect</a>)
and exclusion of the descendant terms for all excluded terms (e.g. exclusion of
<a class="reference external" href="https://hpo.jax.org/browse/term/HP:0020219">Motor seizure</a>
in an individual where <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001250">Seizure</a>
was excluded):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.phenotype</span> <span class="kn">import</span> <span class="n">prepare_predicates_for_terms_of_interest</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pheno_predicates</span> <span class="o">=</span> <span class="n">prepare_predicates_for_terms_of_interest</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">hpo</span><span class="o">=</span><span class="n">hpo</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">min_n_of_patients_with_term</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>By default, GPSEA will perform one hypothesis test for each HPO term used to annotate two or more individuals in the cohort
(see <code class="docutils literal notranslate"><span class="pre">min_n_of_patients_with_term=2</span></code> above).
Testing multiple hypothesis on the same dataset increases the chance of receiving false positive result.
However, GPSEA simplifies the application of an appropriate multiple testing correction.</p>
<p>For general use, we recommend using a combination
of a <em>Phenotype MTC filter</em> (<code class="xref py py-class docutils literal notranslate"><span class="pre">PhenotypeMtcFilter</span></code>) with a <em>multiple testing correction</em>.
Phenotype MTC filter chooses the HPO terms to test according to several heuristics, which
reduce the multiple testing burden and focus the analysis
on the most interesting terms (see <a class="reference internal" href="user-guide/mtc.html#hpo-mtc-filter-strategy"><span class="std std-ref">HPO MTC filter</span></a> for more info).
Then the multiple testing correction, such as Bonferroni or Benjamini-Hochberg,
is used to control the family-wise error rate or the false discovery rate.
See <a class="reference internal" href="user-guide/mtc.html#mtc"><span class="std std-ref">Multiple-testing correction</span></a> for more information.</p>
<p>In this example, we will use a combination of the HPO MTC filter (<a class="reference internal" href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter" title="gpsea.analysis.mtc_filter.HpoMtcFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">HpoMtcFilter</span></code></a>)
with Benjamini-Hochberg procedure (<code class="docutils literal notranslate"><span class="pre">mtc_correction='fdr_bh'</span></code>)
with a false discovery control level at (<code class="docutils literal notranslate"><span class="pre">mtc_alpha=0.05</span></code>):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.mtc_filter</span> <span class="kn">import</span> <span class="n">HpoMtcFilter</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_filter</span> <span class="o">=</span> <span class="n">HpoMtcFilter</span><span class="o">.</span><span class="n">default_filter</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">term_frequency_threshold</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_correction</span> <span class="o">=</span> <span class="s1">&#39;fdr_bh&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_alpha</span> <span class="o">=</span> <span class="mf">0.05</span>
</pre></div>
</div>
<p>Choosing the statistical procedure for assessment of association between genotype and phenotype
groups is the last missing piece of the analysis. We will use Fisher Exact Test:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pcats.stats</span> <span class="kn">import</span> <span class="n">ScipyFisherExact</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">count_statistic</span> <span class="o">=</span> <span class="n">ScipyFisherExact</span><span class="p">()</span>
</pre></div>
</div>
<p>and we finalize the analysis setup by putting all components together
into <a class="reference internal" href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysis" title="gpsea.analysis.pcats.HpoTermAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">HpoTermAnalysis</span></code></a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pcats</span> <span class="kn">import</span> <span class="n">HpoTermAnalysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">analysis</span> <span class="o">=</span> <span class="n">HpoTermAnalysis</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">count_statistic</span><span class="o">=</span><span class="n">count_statistic</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_filter</span><span class="o">=</span><span class="n">mtc_filter</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_correction</span><span class="o">=</span><span class="n">mtc_correction</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_alpha</span><span class="o">=</span><span class="n">mtc_alpha</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>Now we can perform the analysis and investigate the results.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span> <span class="o">=</span> <span class="n">analysis</span><span class="o">.</span><span class="n">compare_genotype_vs_phenotypes</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">gt_predicate</span><span class="o">=</span><span class="n">gt_predicate</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">pheno_predicates</span><span class="o">=</span><span class="n">pheno_predicates</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">total_tests</span>
<span class="go">16</span>
</pre></div>
</div>
<p>We only tested 16 HPO terms. This is despite the individuals being collectively annotated with
260 direct and indirect HPO terms</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">result</span><span class="o">.</span><span class="n">phenotypes</span><span class="p">)</span>
<span class="go">260</span>
</pre></div>
</div>
<p>We can show the reasoning behind <em>not</em> testing 244 (<cite>260 - 16</cite>) HPO terms
by exploring the phenotype MTC filtering report.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">MtcStatsViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_viewer</span> <span class="o">=</span> <span class="n">MtcStatsViewer</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_report</span> <span class="o">=</span> <span class="n">mtc_viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;docs/report/tbx5_frameshift_vs_missense.mtc_report.html&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">fh</span><span class="p">:</span>  
<span class="gp">... </span>    <span class="n">_</span> <span class="o">=</span> <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">mtc_report</span><span class="p">)</span>
</pre></div>
</div>
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Cohort</title>
  <style>table {
    border-collapse: collapse;
    margin: 25px 0;
      font-size: 0.9em;
      font-family: sans-serif;
      min-width: 400px;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
  }


  th {
    background-color: #f2f2f2;
    border: 1px solid #dddddd;
    text-align: left;
    padding: 2px;
  }

  tr {
    border: 1px solid #dddddd;
  }

  td {
    padding: 2px;
    font-weight: bold;
  }

  tr:nth-child(even) {
    background-color: #f2f2f2;
  }

  .table td,tr {
    text-align: left;
}

.table td:first-child, tr:first-child {
    text-align: left;
}

  </style>
</head>

<body>
  <h1>Phenotype testing report</h1>
  <p>Phenotype MTC filter: <em>HPO MTC filter</em></p>
  <p>Multiple testing correction: <em>fdr_bh</em></p>
  <p>Performed statistical tests for 16 out of the total of 260 HPO terms.</p>
    <table>
      <caption>Using <em>HPO MTC filter</em>, 244 term(s) were omitted from statistical analysis.</caption>
        <tbody>
          <tr>
            <th>Code</th>
            <th>Reason</th>
            <th>Count</th>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping general term</td>
            <td>44</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term because all genotypes have same HPO observed proportions</td>
            <td>42</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 0 observations (not powered for 2x2)</td>
            <td>41</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 2 observations (not powered for 2x2)</td>
            <td>26</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 1 observations (not powered for 2x2)</td>
            <td>24</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 3 observations (not powered for 2x2)</td>
            <td>22</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 4 observations (not powered for 2x2)</td>
            <td>14</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 6 observations (not powered for 2x2)</td>
            <td>12</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with maximum frequency that was less than threshold 0.2</td>
            <td>10</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term with only 5 observations (not powered for 2x2)</td>
            <td>4</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term because no genotype has more than one observed HPO count</td>
            <td>3</td>
          </tr>
          
          <tr>
            <!-- TODO: plug the real reason code here -->
            <td>TODO</td>
            <td>Skipping term because one genotype had zero observations</td>
            <td>2</td>
          </tr>
          
        </tbody>
    </table>
</body>
</html>
<p>and these are the HPO terms ordered by the p value corrected with the Benjamini-Hochberg procedure:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate</span> <span class="kn">import</span> <span class="n">PatientCategories</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span> <span class="o">=</span> <span class="n">result</span><span class="o">.</span><span class="n">summarize</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">PatientCategories</span><span class="o">.</span><span class="n">YES</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;docs/report/tbx5_frameshift_vs_missense.csv&#39;</span><span class="p">)</span>  
</pre></div>
</div>
<table class="docutils align-default" id="id5">
<caption><span class="caption-text"><em>TBX5</em> frameshift vs missense</span><a class="headerlink" href="#id5" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"><p>Genotype group: Missense, Frameshift</p></th>
<th class="head"><p>Missense</p></th>
<th class="head"><p>Missense</p></th>
<th class="head"><p>Frameshift</p></th>
<th class="head"><p>Frameshift</p></th>
<th class="head"></th>
<th class="head"></th>
</tr>
<tr class="row-even"><th class="head"></th>
<th class="head"><p>Count</p></th>
<th class="head"><p>Percent</p></th>
<th class="head"><p>Count</p></th>
<th class="head"><p>Percent</p></th>
<th class="head"><p>Corrected p values</p></th>
<th class="head"><p>p values</p></th>
</tr>
</thead>
<tbody>
<tr class="row-odd"><td><p>Ventricular septal defect [HP:0001629]</p></td>
<td><p>31/60</p></td>
<td><p>52%</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td><p>0.0008990549794102921</p></td>
<td><p>5.6190936213143254e-05</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal atrioventricular conduction [HP:0005150]</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0.003478260869565217</p></td>
<td><p>0.00043478260869565214</p></td>
</tr>
<tr class="row-odd"><td><p>Atrioventricular block [HP:0001678]</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0.014492753623188406</p></td>
<td><p>0.0036231884057971015</p></td>
</tr>
<tr class="row-even"><td><p>Heart block [HP:0012722]</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0.014492753623188406</p></td>
<td><p>0.0036231884057971015</p></td>
</tr>
<tr class="row-odd"><td><p>Absent thumb [HP:0009777]</p></td>
<td><p>12/71</p></td>
<td><p>17%</p></td>
<td><p>14/31</p></td>
<td><p>45%</p></td>
<td><p>0.018011232501600187</p></td>
<td><p>0.005628510156750059</p></td>
</tr>
<tr class="row-even"><td><p>Patent ductus arteriosus [HP:0001643]</p></td>
<td><p>3/37</p></td>
<td><p>8%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0.03598740440845704</p></td>
<td><p>0.01349527665317139</p></td>
</tr>
<tr class="row-odd"><td><p>Triphalangeal thumb [HP:0001199]</p></td>
<td><p>13/72</p></td>
<td><p>18%</p></td>
<td><p>13/32</p></td>
<td><p>41%</p></td>
<td><p>0.058517997576307476</p></td>
<td><p>0.02560162393963452</p></td>
</tr>
<tr class="row-even"><td><p>Cardiac conduction abnormality [HP:0031546]</p></td>
<td><p>14/36</p></td>
<td><p>39%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0.14881278039172777</p></td>
<td><p>0.07440639019586388</p></td>
</tr>
<tr class="row-odd"><td><p>Secundum atrial septal defect [HP:0001684]</p></td>
<td><p>14/35</p></td>
<td><p>40%</p></td>
<td><p>4/22</p></td>
<td><p>18%</p></td>
<td><p>0.2532484592139712</p></td>
<td><p>0.1424522583078588</p></td>
</tr>
<tr class="row-even"><td><p>Muscular ventricular septal defect [HP:0011623]</p></td>
<td><p>6/59</p></td>
<td><p>10%</p></td>
<td><p>6/25</p></td>
<td><p>24%</p></td>
<td><p>0.2699930339748754</p></td>
<td><p>0.1687456462342971</p></td>
</tr>
<tr class="row-odd"><td><p>Pulmonary arterial hypertension [HP:0002092]</p></td>
<td><p>4/6</p></td>
<td><p>67%</p></td>
<td><p>0/2</p></td>
<td><p>0%</p></td>
<td><p>0.6233766233766234</p></td>
<td><p>0.42857142857142855</p></td>
</tr>
<tr class="row-even"><td><p>Hypoplasia of the ulna [HP:0003022]</p></td>
<td><p>1/12</p></td>
<td><p>8%</p></td>
<td><p>2/10</p></td>
<td><p>20%</p></td>
<td><p>0.7619047619047618</p></td>
<td><p>0.5714285714285713</p></td>
</tr>
<tr class="row-odd"><td><p>Hypoplasia of the radius [HP:0002984]</p></td>
<td><p>30/62</p></td>
<td><p>48%</p></td>
<td><p>6/14</p></td>
<td><p>43%</p></td>
<td><p>0.9520604334894502</p></td>
<td><p>0.7735491022101784</p></td>
</tr>
<tr class="row-even"><td><p>Atrial septal defect [HP:0001631]</p></td>
<td><p>42/44</p></td>
<td><p>95%</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Absent radius [HP:0003974]</p></td>
<td><p>7/32</p></td>
<td><p>22%</p></td>
<td><p>6/25</p></td>
<td><p>24%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Short humerus [HP:0005792]</p></td>
<td><p>7/17</p></td>
<td><p>41%</p></td>
<td><p>4/9</p></td>
<td><p>44%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal carpal morphology [HP:0001191]</p></td>
<td><p>30/32</p></td>
<td><p>94%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal hand morphology [HP:0005922]</p></td>
<td><p>53/53</p></td>
<td><p>100%</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the hand [HP:0001155]</p></td>
<td><p>60/60</p></td>
<td><p>100%</p></td>
<td><p>31/31</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the upper limb [HP:0002817]</p></td>
<td><p>73/73</p></td>
<td><p>100%</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of limbs [HP:0040064]</p></td>
<td><p>73/73</p></td>
<td><p>100%</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Phenotypic abnormality [HP:0000118]</p></td>
<td><p>82/82</p></td>
<td><p>100%</p></td>
<td><p>38/38</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>All [HP:0000001]</p></td>
<td><p>82/82</p></td>
<td><p>100%</p></td>
<td><p>38/38</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the wrist [HP:0003019]</p></td>
<td><p>30/30</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of upper limb joint [HP:0009810]</p></td>
<td><p>30/30</p></td>
<td><p>100%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal joint morphology [HP:0001367]</p></td>
<td><p>31/31</p></td>
<td><p>100%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal skeletal morphology [HP:0011842]</p></td>
<td><p>73/73</p></td>
<td><p>100%</p></td>
<td><p>35/35</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the skeletal system [HP:0000924]</p></td>
<td><p>73/73</p></td>
<td><p>100%</p></td>
<td><p>35/35</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the musculoskeletal system [HP:0033127]</p></td>
<td><p>74/74</p></td>
<td><p>100%</p></td>
<td><p>35/35</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short thumb [HP:0009778]</p></td>
<td><p>11/41</p></td>
<td><p>27%</p></td>
<td><p>8/30</p></td>
<td><p>27%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the thumb [HP:0009601]</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of fingers [HP:0006265]</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia involving bones of the hand [HP:0005927]</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving bones of the upper limbs [HP:0006496]</p></td>
<td><p>55/55</p></td>
<td><p>100%</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia involving bones of the extremities [HP:0045060]</p></td>
<td><p>55/55</p></td>
<td><p>100%</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia of the extremities [HP:0009815]</p></td>
<td><p>55/55</p></td>
<td><p>100%</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia involving the skeleton [HP:0009115]</p></td>
<td><p>56/56</p></td>
<td><p>100%</p></td>
<td><p>23/23</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal limb bone morphology [HP:0002813]</p></td>
<td><p>63/63</p></td>
<td><p>100%</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of limb bone [HP:0040068]</p></td>
<td><p>63/63</p></td>
<td><p>100%</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal appendicular skeleton morphology [HP:0011844]</p></td>
<td><p>64/64</p></td>
<td><p>100%</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal finger morphology [HP:0001167]</p></td>
<td><p>36/36</p></td>
<td><p>100%</p></td>
<td><p>31/31</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal digit morphology [HP:0011297]</p></td>
<td><p>38/38</p></td>
<td><p>100%</p></td>
<td><p>33/33</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal thumb morphology [HP:0001172]</p></td>
<td><p>30/30</p></td>
<td><p>100%</p></td>
<td><p>31/31</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short finger [HP:0009381]</p></td>
<td><p>11/11</p></td>
<td><p>100%</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Short digit [HP:0011927]</p></td>
<td><p>11/11</p></td>
<td><p>100%</p></td>
<td><p>10/10</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Complete atrioventricular canal defect [HP:0001674]</p></td>
<td><p>5/37</p></td>
<td><p>14%</p></td>
<td><p>3/36</p></td>
<td><p>8%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Atrioventricular canal defect [HP:0006695]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal cardiac septum morphology [HP:0001671]</p></td>
<td><p>62/62</p></td>
<td><p>100%</p></td>
<td><p>28/28</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal heart morphology [HP:0001627]</p></td>
<td><p>62/62</p></td>
<td><p>100%</p></td>
<td><p>30/30</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal cardiovascular system morphology [HP:0030680]</p></td>
<td><p>63/63</p></td>
<td><p>100%</p></td>
<td><p>30/30</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the cardiovascular system [HP:0001626]</p></td>
<td><p>65/65</p></td>
<td><p>100%</p></td>
<td><p>32/32</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>11 pairs of ribs [HP:0000878]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Missing ribs [HP:0000921]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the ribs [HP:0006712]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia involving bones of the thorax [HP:0006711]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia affecting bones of the axial skeleton [HP:0009122]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal axial skeleton morphology [HP:0009121]</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal thorax morphology [HP:0000765]</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal rib morphology [HP:0000772]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal rib cage morphology [HP:0001547]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiac ventricle morphology [HP:0001713]</p></td>
<td><p>31/31</p></td>
<td><p>100%</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal atrial septum morphology [HP:0011994]</p></td>
<td><p>43/43</p></td>
<td><p>100%</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiac atrium morphology [HP:0005120]</p></td>
<td><p>43/43</p></td>
<td><p>100%</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Finger aplasia [HP:0009380]</p></td>
<td><p>15/15</p></td>
<td><p>100%</p></td>
<td><p>14/14</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia involving forearm bones [HP:0009822]</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving forearm bones [HP:0006503]</p></td>
<td><p>37/37</p></td>
<td><p>100%</p></td>
<td><p>12/12</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal forearm bone morphology [HP:0040072]</p></td>
<td><p>37/37</p></td>
<td><p>100%</p></td>
<td><p>14/14</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal upper limb bone morphology [HP:0040070]</p></td>
<td><p>40/40</p></td>
<td><p>100%</p></td>
<td><p>14/14</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal forearm morphology [HP:0002973]</p></td>
<td><p>37/37</p></td>
<td><p>100%</p></td>
<td><p>14/14</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the radius [HP:0006501]</p></td>
<td><p>37/37</p></td>
<td><p>100%</p></td>
<td><p>11/11</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal morphology of the radius [HP:0002818]</p></td>
<td><p>37/37</p></td>
<td><p>100%</p></td>
<td><p>13/13</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Absent forearm bone [HP:0003953]</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Forearm undergrowth [HP:0009821]</p></td>
<td><p>30/30</p></td>
<td><p>100%</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Upper limb undergrowth [HP:0009824]</p></td>
<td><p>33/33</p></td>
<td><p>100%</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Limb undergrowth [HP:0009826]</p></td>
<td><p>33/33</p></td>
<td><p>100%</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short long bone [HP:0003026]</p></td>
<td><p>35/35</p></td>
<td><p>100%</p></td>
<td><p>9/9</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal long bone morphology [HP:0011314]</p></td>
<td><p>44/44</p></td>
<td><p>100%</p></td>
<td><p>13/13</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal ventricular septum morphology [HP:0010438]</p></td>
<td><p>31/31</p></td>
<td><p>100%</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of thumb phalanx [HP:0009602]</p></td>
<td><p>13/13</p></td>
<td><p>100%</p></td>
<td><p>13/13</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Limited pronation/supination of forearm [HP:0006394]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Limited elbow movement [HP:0002996]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the elbow [HP:0009811]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Limitation of joint mobility [HP:0001376]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of joint mobility [HP:0011729]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal joint physiology [HP:0034430]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal musculoskeletal physiology [HP:0011843]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of cardiovascular system electrophysiology [HP:0030956]</p></td>
<td><p>15/15</p></td>
<td><p>100%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal cardiovascular system physiology [HP:0011025]</p></td>
<td><p>23/23</p></td>
<td><p>100%</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Pre-capillary pulmonary hypertension [HP:0033578]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Elevated pulmonary artery pressure [HP:0004890]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of pulmonary circulation [HP:0030875]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal vascular physiology [HP:0030163]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the vasculature [HP:0002597]</p></td>
<td><p>10/10</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal respiratory system physiology [HP:0002795]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the respiratory system [HP:0002086]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Tricuspid regurgitation [HP:0005180]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Atrioventricular valve regurgitation [HP:0034376]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal atrioventricular valve physiology [HP:0031650]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal heart valve physiology [HP:0031653]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal tricuspid valve physiology [HP:0031651]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia of the 1st metacarpal [HP:0010035]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia of the proximal phalanges of the hand [HP:0010242]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the proximal phalanges of the hand [HP:0009851]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal proximal phalanx morphology of the hand [HP:0009834]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal finger phalanx morphology [HP:0005918]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the phalanges of the hand [HP:0009767]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia of the phalanges of the hand [HP:0009802]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia involving bones of the upper limbs [HP:0009823]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia involving bones of the extremities [HP:0009825]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia of metacarpal bones [HP:0010048]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia involving the metacarpal bones [HP:0005914]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal metacarpal morphology [HP:0005916]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the 1st metacarpal [HP:0010026]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal 1st metacarpal morphology [HP:0010009]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the phalanges of the thumb [HP:0009658]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Partial absence of thumb [HP:0009659]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>1-2 finger syndactyly [HP:0010704]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Finger syndactyly [HP:0006101]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Syndactyly [HP:0001159]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Upper limb phocomelia [HP:0009813]</p></td>
<td><p>8/85</p></td>
<td><p>9%</p></td>
<td><p>2/37</p></td>
<td><p>5%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Phocomelia [HP:0009829]</p></td>
<td><p>8/8</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Clinodactyly of the 5th finger [HP:0004209]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Finger clinodactyly [HP:0040019]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Clinodactyly [HP:0030084]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Deviation of finger [HP:0004097]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Deviation of the hand or of fingers of the hand [HP:0009484]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Deviation of the 5th finger [HP:0009179]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal 5th finger morphology [HP:0004207]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Patent foramen ovale [HP:0001655]</p></td>
<td><p>4/40</p></td>
<td><p>10%</p></td>
<td><p>0/36</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Small hypothenar eminence [HP:0010487]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the hypothenar eminence [HP:0010486]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the musculature of the hand [HP:0001421]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the musculature of the upper limbs [HP:0001446]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the musculature of the limbs [HP:0009127]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal skeletal muscle morphology [HP:0011805]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the musculature [HP:0003011]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Perimembranous ventricular septal defect [HP:0011682]</p></td>
<td><p>3/59</p></td>
<td><p>5%</p></td>
<td><p>3/25</p></td>
<td><p>12%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Deviation of the thumb [HP:0009603]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia of the humerus [HP:0006507]</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the humerus [HP:0003063]</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the upper arm [HP:0001454]</p></td>
<td><p>7/7</p></td>
<td><p>100%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Right atrial enlargement [HP:0030718]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal right atrium morphology [HP:0025580]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Atrial septal dilatation [HP:0011995]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Pectus excavatum [HP:0000767]</p></td>
<td><p>3/4</p></td>
<td><p>75%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal sternum morphology [HP:0000766]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Postaxial hand polydactyly [HP:0001162]</p></td>
<td><p>3/4</p></td>
<td><p>75%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Postaxial polydactyly [HP:0100259]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Polydactyly [HP:0010442]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Hand polydactyly [HP:0001161]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Duplication of phalanx of hand [HP:0009997]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Duplication of hand bones [HP:0004275]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Duplication of bones involving the upper extremities [HP:0009142]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>High palate [HP:0000218]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal palate morphology [HP:0000174]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal oral cavity morphology [HP:0000163]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal oral morphology [HP:0031816]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the mouth [HP:0000153]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the face [HP:0000271]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the head [HP:0000234]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of head or neck [HP:0000152]</p></td>
<td><p>5/5</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short neck [HP:0000470]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal neck morphology [HP:0025668]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the neck [HP:0000464]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the cervical spine [HP:0003319]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the vertebral column [HP:0000925]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Shield chest [HP:0000914]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Enlarged thorax [HP:0100625]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Y-shaped metatarsals [HP:0010567]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal metatarsal morphology [HP:0001832]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal lower limb bone morphology [HP:0040069]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the lower limb [HP:0002814]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal foot morphology [HP:0001760]</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Mitral regurgitation [HP:0001653]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal mitral valve physiology [HP:0031481]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>First degree atrioventricular block [HP:0011705]</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Congenital malformation of the great arteries [HP:0011603]</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal morphology of the great vessels [HP:0030962]</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal blood vessel morphology [HP:0033353]</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal vascular morphology [HP:0025015]</p></td>
<td><p>6/6</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal skull morphology [HP:0000929]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Sinus bradycardia [HP:0001688]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal electrophysiology of sinoatrial node origin [HP:0011702]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Arrhythmia [HP:0011675]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Bradycardia [HP:0001662]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal skin morphology [HP:0011121]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the skin [HP:0000951]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the integument [HP:0001574]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Sinus venosus atrial septal defect [HP:0011567]</p></td>
<td><p>0/2</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal morphology of ulna [HP:0040071]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>4/4</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the ulna [HP:0006495]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Persistent left superior vena cava [HP:0005301]</p></td>
<td><p>2/37</p></td>
<td><p>5%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal superior vena cava morphology [HP:0025575]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal vena cava morphology [HP:0005345]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal venous morphology [HP:0002624]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal shoulder morphology [HP:0003043]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Hypoplasia of deltoid muscle [HP:0030241]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Shoulder muscle hypoplasia [HP:0008952]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Hypoplasia of the musculature [HP:0009004]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia involving the skeletal musculature [HP:0001460]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of muscle size [HP:0030236]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia involving the shoulder musculature [HP:0001464]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia involving the musculature of the upper limbs [HP:0001467]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia involving the musculature of the extremities [HP:0009128]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the shoulder girdle musculature [HP:0001435]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Common atrium [HP:0011565]</p></td>
<td><p>0/83</p></td>
<td><p>0%</p></td>
<td><p>0/38</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Unroofed coronary sinus [HP:0031297]</p></td>
<td><p>0/85</p></td>
<td><p>0%</p></td>
<td><p>0/38</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Atrioventricular dissociation [HP:0011709]</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Bowed forearm bones [HP:0003956]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Bowing of the arm [HP:0006488]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Bowing of the long bones [HP:0006487]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal diaphysis morphology [HP:0000940]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Proximal placement of thumb [HP:0009623]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short middle phalanx of the 5th finger [HP:0004220]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Type A brachydactyly [HP:0009370]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Brachydactyly [HP:0001156]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Short 5th finger [HP:0009237]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the 5th finger [HP:0006262]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the middle phalanx of the 5th finger [HP:0009161]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the phalanges of the 5th finger [HP:0009376]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal 5th finger phalanx morphology [HP:0004213]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormality of the middle phalanx of the 5th finger [HP:0004219]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Short middle phalanx of finger [HP:0005819]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the middle phalanges of the hand [HP:0009843]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal middle phalanx morphology of the hand [HP:0009833]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short phalanx of finger [HP:0009803]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Micrognathia [HP:0000347]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the mandible [HP:0009118]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia involving bones of the skull [HP:0009116]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal mandible morphology [HP:0000277]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal jaw morphology [HP:0030791]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal facial skeleton morphology [HP:0011821]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Cleft soft palate [HP:0000185]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal soft palate morphology [HP:0100736]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Cleft palate [HP:0000175]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Orofacial cleft [HP:0000202]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Craniofacial cleft [HP:5201015]</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Hypoplastic scapulae [HP:0000882]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/Hypoplasia of the scapulae [HP:0006713]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal scapula morphology [HP:0000782]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Amelia involving the upper limbs [HP:0009812]</p></td>
<td><p>0/83</p></td>
<td><p>0%</p></td>
<td><p>1/37</p></td>
<td><p>3%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Third degree atrioventricular block [HP:0001709]</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Upper extremity joint dislocation [HP:0030310]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Joint dislocation [HP:0001373]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal toe morphology [HP:0001780]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal toe phalanx morphology [HP:0010161]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormality of the distal phalanges of the toes [HP:0010182]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of the 2nd finger [HP:0006264]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal 2nd finger morphology [HP:0004100]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Synostosis of joints [HP:0100240]</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Left ventricular noncompaction cardiomyopathy [HP:0011664]</p></td>
<td><p>0/1</p></td>
<td><p>0%</p></td>
<td><p>1/5</p></td>
<td><p>20%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Noncompaction cardiomyopathy [HP:0012817]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Cardiomyopathy [HP:0001638]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal myocardium morphology [HP:0001637]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Mitral valve prolapse [HP:0001634]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal mitral valve morphology [HP:0001633]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Abnormal atrioventricular valve morphology [HP:0006705]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Abnormal heart valve morphology [HP:0001654]</p></td>
<td><p>0/0</p></td>
<td><p>0%</p></td>
<td><p>1/1</p></td>
<td><p>100%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Short 1st metacarpal [HP:0010034]</p></td>
<td><p>0/30</p></td>
<td><p>0%</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Short phalanx of the thumb [HP:0009660]</p></td>
<td><p>0/30</p></td>
<td><p>0%</p></td>
<td><p>0/22</p></td>
<td><p>0%</p></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>We see that several HPO terms are significantly associated
with presence of a frameshift variant in <em>TBX5</em>.
For example, <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001629">Ventricular septal defect</a>
was observed in 31/60 (52%) patients with a missense variant
but it was observed in 19/19 (100%) patients with a frameshift variant.
Fisher exact test computed a p value of <cite>~0.0000562</cite>
and the p value corrected by Benjamini-Hochberg procedure
is <cite>~0.00112</cite>.</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="installation.html" class="btn btn-neutral float-left" title="Installation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="user-guide/index.html" class="btn btn-neutral float-right" title="User guide" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>