Search.setIndex({"alltitles": {"API reference": [[17, null]], "Allele count predicates": [[32, null]], "Alternative phenopacket sources": [[30, "alternative-phenopacket-sources"]], "Analysis": [[22, "analysis"], [23, "analysis"], [24, "analysis"]], "Autosomal dominant predicate": [[40, "autosomal-dominant-predicate"]], "Autosomal dominant predicate categories": [[40, "id6"]], "Autosomal recessive predicate": [[40, "autosomal-recessive-predicate"]], "Autosomal recessive predicate categories": [[40, "id7"]], "Background": [[31, "background"]], "Biallelic predicate": [[32, "biallelic-predicate"]], "Biallelic predicate genotype categories": [[32, "id5"]], "Building blocks": [[42, "building-blocks"]], "Categories": [[32, "categories"]], "Change length of an allele": [[28, "change-length-of-an-allele"]], "Choose the transcript": [[30, "choose-the-transcript"]], "Choose the transcript and protein of interest": [[30, "choose-the-transcript-and-protein-of-interest"]], "Cohort exploratory analysis": [[26, null]], "Cohort summary": [[26, "cohort-summary"]], "Compare Monoallelic vs. Biallelic": [[40, "compare-monoallelic-vs-biallelic"]], "Compare genotype and phenotype groups": [[23, null]], "Compare measurement values": [[22, null]], "Compare missense vs. at least one frameshift allele": [[32, "compare-missense-vs-at-least-one-frameshift-allele"]], "Compare missense vs. frameshift": [[32, "compare-missense-vs-frameshift"]], "Compare phenotype scores in genotype groups": [[24, null]], "Complex conditions": [[42, "complex-conditions"]], "Configure a cohort creator": [[30, "configure-a-cohort-creator"]], "Configure analysis": [[22, "configure-analysis"], [23, "configure-analysis"], [24, "configure-analysis"]], "Contents:": [[21, null], [29, null], [35, null], [38, null], [41, null]], "Create a cohort from GA4GH phenopackets": [[30, "create-a-cohort-from-ga4gh-phenopackets"]], "De Vries Score": [[33, null]], "Developmental delay": [[33, "developmental-delay"]], "Distribution of variants across protein domains": [[26, "distribution-of-variants-across-protein-domains"]], "Enter features manually": [[30, "enter-features-manually"]], "Example": [[30, "example"], [30, "id4"], [32, "example"], [36, "example"], [37, "example"]], "Example analysis": [[22, "example-analysis"], [23, "example-analysis"], [24, "example-analysis"]], "Examples": [[32, "examples"], [40, "examples"], [40, "id3"]], "Exploration": [[26, "exploration"]], "Explore cohort": [[20, "explore-cohort"]], "Facial dysmorphic features": [[33, "facial-dysmorphic-features"]], "Feedback": [[18, "feedback"]], "Fetch data from UniProt REST API": [[30, "fetch-data-from-uniprot-rest-api"]], "Fetch protein data": [[30, "fetch-protein-data"]], "Fetch transcript coordinates from Variant Validator REST API": [[30, "fetch-transcript-coordinates-from-variant-validator-rest-api"]], "Final analysis": [[22, "final-analysis"], [23, "final-analysis"], [24, "final-analysis"]], "Final score": [[33, "final-score"]], "Fisher exact test (FET)": [[23, "fisher-exact-test-fet"]], "GPSEA": [[18, null]], "Gallery": [[38, "gallery"]], "General HPO terms": [[27, null]], "Genotype Predicates": [[35, null]], "Genotype phenotype associations": [[23, "genotype-phenotype-associations"]], "Genotype predicate": [[22, "genotype-predicate"], [23, "genotype-predicate"], [24, "genotype-predicate"]], "Genotype-Phenotype Correlations in Autosomal Recessive Diseases": [[25, null]], "Get the transcript data": [[30, "get-the-transcript-data"]], "Glossary": [[28, null]], "Groups Predicate": [[36, null]], "HMF01 - Skip terms that occur very rarely": [[31, "hmf01-skip-terms-that-occur-very-rarely"]], "HMF02 - Skip terms if no genotype group has more than one count": [[31, "hmf02-skip-terms-if-no-genotype-group-has-more-than-one-count"]], "HMF03 - Skip terms if all counts are identical to counts for a child term": [[31, "hmf03-skip-terms-if-all-counts-are-identical-to-counts-for-a-child-term"]], "HMF05 - Skip term if one of the genotype groups has neither observed nor excluded observations": [[31, "hmf05-skip-term-if-one-of-the-genotype-groups-has-neither-observed-nor-excluded-observations"]], "HMF06 - Skip term if underpowered for 2x2 or 2x3 analysis": [[31, "hmf06-skip-term-if-underpowered-for-2x2-or-2x3-analysis"]], "HMF07 - Skipping terms that are not descendents of Phenotypic abnormality": [[31, "hmf07-skipping-terms-that-are-not-descendents-of-phenotypic-abnormality"]], "HMF08 - Skipping \u201cgeneral\u201d level terms": [[31, "hmf08-skipping-general-level-terms"]], "HPO MTC filter strategy": [[31, "hpo-mtc-filter-strategy"]], "HPO predicate": [[37, null]], "Implementation in GPSEA": [[31, "implementation-in-gpsea"]], "Input data": [[30, null]], "Installation": [[19, null]], "Interactive exploration": [[26, "interactive-exploration"]], "Inverting conditions": [[42, "inverting-conditions"]], "Latest release": [[19, "latest-release"]], "Length of the reference allele": [[28, "length-of-the-reference-allele"]], "Literature": [[18, "literature"]], "Load HPO": [[20, "load-hpo"], [24, "load-hpo"]], "Load cohort": [[22, "load-cohort"], [23, "load-cohort"], [24, "load-cohort"]], "Load phenopackets": [[30, "load-phenopackets"]], "MTC filters: Choosing which terms to test": [[31, "mtc-filters-choosing-which-terms-to-test"]], "Mann-Whitney U Test": [[24, "mann-whitney-u-test"]], "Mode of Inheritance Predicates": [[40, null]], "Monoallelic predicate": [[32, "monoallelic-predicate"]], "Monoallelic predicate genotype categories": [[32, "id4"]], "Multiple testing correction": [[23, "multiple-testing-correction"]], "Multiple-testing correction": [[31, null]], "Multiple-testing correction procedures": [[31, "multiple-testing-correction-procedures"]], "Need more?": [[38, "need-more"]], "Non-facial dysmorphism and congenital abnormalities": [[33, "non-facial-dysmorphism-and-congenital-abnormalities"]], "Parse UniProt JSON dump": [[30, "parse-uniprot-json-dump"]], "Partition by a diagnosis": [[34, null]], "Partition by the sex of the individual": [[39, null]], "Partitions": [[32, "partitions"], [40, "partitions"]], "Persist the cohort for later": [[30, "persist-the-cohort-for-later"]], "Phenotype Predicates": [[41, null]], "Phenotype predicates": [[23, "phenotype-predicates"]], "Phenotype score": [[22, "phenotype-score"], [24, "phenotype-score"]], "Plot distribution of cohort variants on the protein": [[30, "plot-distribution-of-cohort-variants-on-the-protein"]], "Plot distribution of variants with respect to the protein sequence": [[20, "plot-distribution-of-variants-with-respect-to-the-protein-sequence"], [26, "plot-distribution-of-variants-with-respect-to-the-protein-sequence"]], "Postnatal growth abnormalities": [[33, "postnatal-growth-abnormalities"]], "Power": [[26, "power"]], "Predicates": [[38, null]], "Predicates for all cohort phenotypes": [[37, "predicates-for-all-cohort-phenotypes"]], "Prenatal-onset growth retardation": [[33, "prenatal-onset-growth-retardation"]], "Prepare cohort": [[20, "prepare-cohort"]], "Prepare genotype and phenotype predicates": [[20, "prepare-genotype-and-phenotype-predicates"]], "Provide the transcript coordinates manually": [[30, "provide-the-transcript-coordinates-manually"]], "Quality control": [[30, "quality-control"]], "Run tests": [[19, "run-tests"]], "Show cohort summary": [[20, "show-cohort-summary"]], "Showcase transcript data": [[30, "showcase-transcript-data"]], "Specify terms strategy": [[31, "specify-terms-strategy"]], "Stable release": [[19, "stable-release"]], "Standardize genotype and phenotype data": [[30, "standardize-genotype-and-phenotype-data"]], "Statistical analyses": [[21, null]], "Statistical test": [[22, "statistical-test"], [23, "statistical-test"], [24, "statistical-test"]], "Submodules": [[0, "submodules"]], "Subpackages": [[0, "subpackages"], [1, "subpackages"], [3, "subpackages"], [5, "subpackages"], [8, "subpackages"], [12, "subpackages"]], "Summarize all variant alleles": [[20, "summarize-all-variant-alleles"]], "Supports shape": [[4, "supports-shape"]], "TBX5 frameshift vs missense": [[20, "id5"]], "TBX5 frameshift vs rest": [[23, "id1"]], "Test all terms": [[31, "test-all-terms"]], "The analysis": [[20, "the-analysis"]], "The sections of the score": [[33, "the-sections-of-the-score"]], "True path rule": [[28, "true-path-rule"]], "Tutorial": [[20, null], [25, "tutorial"]], "Types of statistical test": [[26, "types-of-statistical-test"]], "Use a variant subset": [[40, "use-a-variant-subset"], [40, "id5"]], "Use all variants": [[40, "use-all-variants"], [40, "id4"]], "User guide": [[29, null]], "Using the De Vries scorer in code": [[33, "using-the-de-vries-scorer-in-code"]], "Variant Predicates": [[42, null]], "gpsea package": [[0, null]], "gpsea.analysis package": [[1, null]], "gpsea.analysis.mtc_filter package": [[2, null]], "gpsea.analysis.pcats package": [[3, null]], "gpsea.analysis.pcats.stats package": [[4, null]], "gpsea.analysis.predicate package": [[5, null]], "gpsea.analysis.predicate.genotype package": [[6, null]], "gpsea.analysis.predicate.phenotype package": [[7, null]], "gpsea.analysis.pscore package": [[8, null]], "gpsea.analysis.pscore.stats package": [[9, null]], "gpsea.config module": [[10, null]], "gpsea.io module": [[11, null]], "gpsea.model package": [[12, null]], "gpsea.model.genome package": [[13, null]], "gpsea.preprocessing package": [[14, null]], "gpsea.util module": [[15, null]], "gpsea.view package": [[16, null]], "missing_implies_phenotype_excluded": [[37, "missing-implies-phenotype-excluded"]]}, "docnames": ["apidocs/gpsea", "apidocs/gpsea.analysis", "apidocs/gpsea.analysis.mtc_filter", "apidocs/gpsea.analysis.pcats", "apidocs/gpsea.analysis.pcats.stats", "apidocs/gpsea.analysis.predicate", "apidocs/gpsea.analysis.predicate.genotype", "apidocs/gpsea.analysis.predicate.phenotype", "apidocs/gpsea.analysis.pscore", "apidocs/gpsea.analysis.pscore.stats", "apidocs/gpsea.config", "apidocs/gpsea.io", "apidocs/gpsea.model", "apidocs/gpsea.model.genome", "apidocs/gpsea.preprocessing", "apidocs/gpsea.util", "apidocs/gpsea.view", "apidocs/modules", "index", "installation", "tutorial", "user-guide/analyses/index", "user-guide/analyses/measurements", "user-guide/analyses/phenotype-groups", "user-guide/analyses/phenotype-scores", "user-guide/autosomal_recessive", "user-guide/exploratory", "user-guide/general_hpo_terms", "user-guide/glossary", "user-guide/index", "user-guide/input-data", "user-guide/mtc", "user-guide/predicates/allele_count_predicates", "user-guide/predicates/devries", "user-guide/predicates/diagnosis_predicate", "user-guide/predicates/genotype_predicates", "user-guide/predicates/groups_predicate", "user-guide/predicates/hpo_predicate", "user-guide/predicates/index", "user-guide/predicates/male_female_predicate", "user-guide/predicates/mode_of_inheritance_predicate", "user-guide/predicates/phenotype_predicates", "user-guide/predicates/variant_predicates"], "envversion": {"sphinx": 64, "sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinx.ext.viewcode": 1}, "filenames": ["apidocs/gpsea.rst", "apidocs/gpsea.analysis.rst", "apidocs/gpsea.analysis.mtc_filter.rst", "apidocs/gpsea.analysis.pcats.rst", "apidocs/gpsea.analysis.pcats.stats.rst", "apidocs/gpsea.analysis.predicate.rst", "apidocs/gpsea.analysis.predicate.genotype.rst", "apidocs/gpsea.analysis.predicate.phenotype.rst", "apidocs/gpsea.analysis.pscore.rst", "apidocs/gpsea.analysis.pscore.stats.rst", "apidocs/gpsea.config.rst", "apidocs/gpsea.io.rst", "apidocs/gpsea.model.rst", "apidocs/gpsea.model.genome.rst", "apidocs/gpsea.preprocessing.rst", "apidocs/gpsea.util.rst", "apidocs/gpsea.view.rst", "apidocs/modules.rst", "index.rst", "installation.rst", "tutorial.rst", "user-guide/analyses/index.rst", "user-guide/analyses/measurements.rst", "user-guide/analyses/phenotype-groups.rst", "user-guide/analyses/phenotype-scores.rst", "user-guide/autosomal_recessive.rst", "user-guide/exploratory.rst", "user-guide/general_hpo_terms.rst", "user-guide/glossary.rst", "user-guide/index.rst", "user-guide/input-data.rst", "user-guide/mtc.rst", "user-guide/predicates/allele_count_predicates.rst", "user-guide/predicates/devries.rst", "user-guide/predicates/diagnosis_predicate.rst", "user-guide/predicates/genotype_predicates.rst", "user-guide/predicates/groups_predicate.rst", "user-guide/predicates/hpo_predicate.rst", "user-guide/predicates/index.rst", "user-guide/predicates/male_female_predicate.rst", "user-guide/predicates/mode_of_inheritance_predicate.rst", "user-guide/predicates/phenotype_predicates.rst", "user-guide/predicates/variant_predicates.rst"], "indexentries": {"add_error() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.add_error", false]], "add_issue() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.add_issue", false]], "add_subsection() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.add_subsection", false]], "add_subsection() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.add_subsection", false]], "add_warning() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.add_warning", false]], "all() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.all", false]], "all_counts (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.all_counts", false]], "all_diseases() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.all_diseases", false]], "all_patients (gpsea.model.cohort property)": [[12, "gpsea.model.Cohort.all_patients", false]], "all_phenotypes() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.all_phenotypes", false]], "all_transcript_ids (gpsea.model.cohort property)": [[12, "gpsea.model.Cohort.all_transcript_ids", false]], "all_variants() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.all_variants", false]], "allelecounter (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.AlleleCounter", false]], "alt (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.alt", false]], "annotate() (gpsea.preprocessing.defaultimprecisesvfunctionalannotator method)": [[14, "gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.functionalannotator method)": [[14, "gpsea.preprocessing.FunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.imprecisesvfunctionalannotator method)": [[14, "gpsea.preprocessing.ImpreciseSvFunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.protcachingmetadataservice method)": [[14, "gpsea.preprocessing.ProtCachingMetadataService.annotate", false]], "annotate() (gpsea.preprocessing.proteinmetadataservice method)": [[14, "gpsea.preprocessing.ProteinMetadataService.annotate", false]], "annotate() (gpsea.preprocessing.uniprotproteinmetadataservice method)": [[14, "gpsea.preprocessing.UniprotProteinMetadataService.annotate", false]], "annotate() (gpsea.preprocessing.varcachingfunctionalannotator method)": [[14, "gpsea.preprocessing.VarCachingFunctionalAnnotator.annotate", false]], "annotate() (gpsea.preprocessing.vepfunctionalannotator method)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator.annotate", false]], "any() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.any", false]], "apply_predicates_on_patients() (in module gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.apply_predicates_on_patients", false]], "auditor (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.Auditor", false]], "autosomal_dominant() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.autosomal_dominant", false]], "autosomal_recessive() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.autosomal_recessive", false]], "biallelic_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.biallelic_predicate", false]], "cache_env (in module gpsea.config)": [[10, "gpsea.config.CACHE_ENV", false]], "cat_id (gpsea.analysis.predicate.patientcategory property)": [[5, "gpsea.analysis.predicate.PatientCategory.cat_id", false]], "categorization (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.Categorization", false]], "category (gpsea.analysis.predicate.categorization property)": [[5, "gpsea.analysis.predicate.Categorization.category", false]], "cds_end (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.cds_end", false]], "cds_start (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.cds_start", false]], "change_length (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.change_length", false]], "change_length() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.change_length", false]], "children (gpsea.preprocessing.notepadtree property)": [[14, "gpsea.preprocessing.NotepadTree.children", false]], "chrom (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.chrom", false]], "code (gpsea.analysis.mtc_filter.phenotypemtcissue attribute)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcIssue.code", false]], "code (gpsea.model.genotype property)": [[12, "gpsea.model.Genotype.code", false]], "coding_sequence_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.CODING_SEQUENCE_VARIANT", false]], "cohort (class in gpsea.model)": [[12, "gpsea.model.Cohort", false]], "cohortcreator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.CohortCreator", false]], "cohortvariantviewer (class in gpsea.view)": [[16, "gpsea.view.CohortVariantViewer", false]], "cohortviewable (class in gpsea.view)": [[16, "gpsea.view.CohortViewable", false]], "coiled_coil (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.COILED_COIL", false]], "compare_genotype_vs_phenotype_score() (gpsea.analysis.pscore.phenotypescoreanalysis method)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysis.compare_genotype_vs_phenotype_score", false]], "compare_genotype_vs_phenotypes() (gpsea.analysis.pcats.multiphenotypeanalysis method)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysis.compare_genotype_vs_phenotypes", false]], "compositional_bias (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.COMPOSITIONAL_BIAS", false]], "compute_pval() (gpsea.analysis.pcats.stats.countstatistic method)": [[4, "gpsea.analysis.pcats.stats.CountStatistic.compute_pval", false]], "compute_pval() (gpsea.analysis.pcats.stats.fisherexacttest method)": [[4, "gpsea.analysis.pcats.stats.FisherExactTest.compute_pval", false]], "compute_pval() (gpsea.analysis.pscore.stats.mannwhitneystatistic method)": [[9, "gpsea.analysis.pscore.stats.MannWhitneyStatistic.compute_pval", false]], "compute_pval() (gpsea.analysis.pscore.stats.phenotypescorestatistic method)": [[9, "gpsea.analysis.pscore.stats.PhenotypeScoreStatistic.compute_pval", false]], "compute_pval() (gpsea.analysis.pscore.stats.tteststatistic method)": [[9, "gpsea.analysis.pscore.stats.TTestStatistic.compute_pval", false]], "configure_caching_cohort_creator() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_caching_cohort_creator", false]], "configure_caching_patient_creator() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_caching_patient_creator", false]], "configure_cohort_creator() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_cohort_creator", false]], "configure_default_protein_metadata_service() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_default_protein_metadata_service", false]], "configure_patient_creator() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_patient_creator", false]], "configure_protein_metadata_service() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.configure_protein_metadata_service", false]], "contains() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.contains", false]], "contains() (gpsea.model.genome.region method)": [[13, "gpsea.model.genome.Region.contains", false]], "contains_pos() (gpsea.model.genome.region method)": [[13, "gpsea.model.genome.Region.contains_pos", false]], "contig (class in gpsea.model.genome)": [[13, "gpsea.model.genome.Contig", false]], "contig (gpsea.model.genome.genomicregion property)": [[13, "gpsea.model.genome.GenomicRegion.contig", false]], "contig_by_name() (gpsea.model.genome.genomebuild method)": [[13, "gpsea.model.genome.GenomeBuild.contig_by_name", false]], "contigs (gpsea.model.genome.genomebuild property)": [[13, "gpsea.model.genome.GenomeBuild.contigs", false]], "corrected_pvals (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.corrected_pvals", false]], "count() (gpsea.analysis.predicate.genotype.allelecounter method)": [[6, "gpsea.analysis.predicate.genotype.AlleleCounter.count", false]], "countingphenotypescorer (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer", false]], "countstatistic (class in gpsea.analysis.pcats.stats)": [[4, "gpsea.analysis.pcats.stats.CountStatistic", false]], "create() (gpsea.model.proteinfeature static method)": [[12, "gpsea.model.ProteinFeature.create", false]], "create_variant_from_scratch() (gpsea.model.variant static method)": [[12, "gpsea.model.Variant.create_variant_from_scratch", false]], "curie (gpsea.model.varianteffect property)": [[12, "gpsea.model.VariantEffect.curie", false]], "datasanityissue (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.DataSanityIssue", false]], "default() (gpsea.io.gpseajsonencoder method)": [[11, "gpsea.io.GpseaJSONEncoder.default", false]], "default_cache_path (in module gpsea.config)": [[10, "gpsea.config.DEFAULT_CACHE_PATH", false]], "default_filter() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.default_filter", false]], "defaultimprecisesvfunctionalannotator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator", false]], "del (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.DEL", false]], "description (gpsea.analysis.predicate.patientcategory property)": [[5, "gpsea.analysis.predicate.PatientCategory.description", false]], "devriesphenotypescorer (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.DeVriesPhenotypeScorer", false]], "diagnosis_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.diagnosis_predicate", false]], "disease (class in gpsea.model)": [[12, "gpsea.model.Disease", false]], "diseaseanalysis (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.DiseaseAnalysis", false]], "diseasepresencepredicate (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate", false]], "diseases (gpsea.model.patient property)": [[12, "gpsea.model.Patient.diseases", false]], "diseaseviewable (class in gpsea.view)": [[16, "gpsea.view.DiseaseViewable", false]], "display_question() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.display_question", false]], "distance_to() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.distance_to", false]], "distance_to() (gpsea.model.genome.region method)": [[13, "gpsea.model.genome.Region.distance_to", false]], "domain (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.DOMAIN", false]], "domains() (gpsea.model.proteinmetadata method)": [[12, "gpsea.model.ProteinMetadata.domains", false]], "downstream_gene_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.DOWNSTREAM_GENE_VARIANT", false]], "draw_fig() (gpsea.view.proteinvisualizer method)": [[16, "gpsea.view.ProteinVisualizer.draw_fig", false]], "draw_protein_diagram() (gpsea.view.proteinvisualizer method)": [[16, "gpsea.view.ProteinVisualizer.draw_protein_diagram", false]], "draw_variants() (gpsea.view.varianttranscriptvisualizer method)": [[16, "gpsea.view.VariantTranscriptVisualizer.draw_variants", false]], "dup (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.DUP", false]], "empty() (gpsea.model.genotypes static method)": [[12, "gpsea.model.Genotypes.empty", false]], "end (gpsea.model.featureinfo property)": [[12, "gpsea.model.FeatureInfo.end", false]], "end (gpsea.model.genome.region property)": [[13, "gpsea.model.genome.Region.end", false]], "end (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.end", false]], "end_on_strand() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.end_on_strand", false]], "error (gpsea.preprocessing.level attribute)": [[14, "gpsea.preprocessing.Level.ERROR", false]], "error_count() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.error_count", false]], "errors() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.errors", false]], "excluded_diseases() (gpsea.model.patient method)": [[12, "gpsea.model.Patient.excluded_diseases", false]], "excluded_phenotypes() (gpsea.model.patient method)": [[12, "gpsea.model.Patient.excluded_phenotypes", false]], "exon() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.exon", false]], "exons (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.exons", false]], "fail() (gpsea.analysis.mtc_filter.phenotypemtcresult static method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.fail", false]], "feature_elongation (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.FEATURE_ELONGATION", false]], "feature_truncation (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.FEATURE_TRUNCATION", false]], "feature_type (gpsea.model.proteinfeature property)": [[12, "gpsea.model.ProteinFeature.feature_type", false]], "featureinfo (class in gpsea.model)": [[12, "gpsea.model.FeatureInfo", false]], "featuretype (class in gpsea.model)": [[12, "gpsea.model.FeatureType", false]], "female (gpsea.model.sex attribute)": [[12, "gpsea.model.Sex.FEMALE", false]], "fetch() (gpsea.preprocessing.transcriptcoordinateservice method)": [[14, "gpsea.preprocessing.TranscriptCoordinateService.fetch", false]], "fetch() (gpsea.preprocessing.vvmulticoordinateservice method)": [[14, "gpsea.preprocessing.VVMultiCoordinateService.fetch", false]], "fetch_for_gene() (gpsea.preprocessing.genecoordinateservice method)": [[14, "gpsea.preprocessing.GeneCoordinateService.fetch_for_gene", false]], "fetch_for_gene() (gpsea.preprocessing.vvmulticoordinateservice method)": [[14, "gpsea.preprocessing.VVMultiCoordinateService.fetch_for_gene", false]], "fetch_response() (gpsea.preprocessing.vepfunctionalannotator method)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator.fetch_response", false]], "filter() (gpsea.analysis.mtc_filter.hpomtcfilter method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.filter", false]], "filter() (gpsea.analysis.mtc_filter.phenotypemtcfilter method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter.filter", false]], "filter() (gpsea.analysis.mtc_filter.specifiedtermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.filter", false]], "filter() (gpsea.analysis.mtc_filter.usealltermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.filter", false]], "filter_method_name() (gpsea.analysis.mtc_filter.hpomtcfilter method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.filter_method_name", false]], "filter_method_name() (gpsea.analysis.mtc_filter.phenotypemtcfilter method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter.filter_method_name", false]], "filter_method_name() (gpsea.analysis.mtc_filter.specifiedtermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.filter_method_name", false]], "filter_method_name() (gpsea.analysis.mtc_filter.usealltermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.filter_method_name", false]], "find_coordinates() (gpsea.preprocessing.phenopacketvariantcoordinatefinder method)": [[14, "gpsea.preprocessing.PhenopacketVariantCoordinateFinder.find_coordinates", false]], "find_coordinates() (gpsea.preprocessing.variantcoordinatefinder method)": [[14, "gpsea.preprocessing.VariantCoordinateFinder.find_coordinates", false]], "find_coordinates() (gpsea.preprocessing.vvhgvsvariantcoordinatefinder method)": [[14, "gpsea.preprocessing.VVHgvsVariantCoordinateFinder.find_coordinates", false]], "fisherexacttest (class in gpsea.analysis.pcats.stats)": [[4, "gpsea.analysis.pcats.stats.FisherExactTest", false]], "five_prime_utr_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.FIVE_PRIME_UTR_VARIANT", false]], "for_sample() (gpsea.model.genotypes method)": [[12, "gpsea.model.Genotypes.for_sample", false]], "format_as_string() (gpsea.view.formatter method)": [[16, "gpsea.view.Formatter.format_as_string", false]], "format_as_string() (gpsea.view.variantformatter method)": [[16, "gpsea.view.VariantFormatter.format_as_string", false]], "format_coordinates_for_vep_query() (gpsea.preprocessing.vepfunctionalannotator static method)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator.format_coordinates_for_vep_query", false]], "formatter (class in gpsea.view)": [[16, "gpsea.view.Formatter", false]], "frameshift_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.FRAMESHIFT_VARIANT", false]], "from_feature_frame() (gpsea.model.proteinmetadata static method)": [[12, "gpsea.model.ProteinMetadata.from_feature_frame", false]], "from_mapping() (gpsea.model.genotypes static method)": [[12, "gpsea.model.Genotypes.from_mapping", false]], "from_measurement_id() (gpsea.analysis.pscore.measurementphenotypescorer static method)": [[8, "gpsea.analysis.pscore.MeasurementPhenotypeScorer.from_measurement_id", false]], "from_patients() (gpsea.model.cohort static method)": [[12, "gpsea.model.Cohort.from_patients", false]], "from_query_curies() (gpsea.analysis.pscore.countingphenotypescorer static method)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer.from_query_curies", false]], "from_raw_parts() (gpsea.analysis.predicate.categorization static method)": [[5, "gpsea.analysis.predicate.Categorization.from_raw_parts", false]], "from_raw_parts() (gpsea.model.patient static method)": [[12, "gpsea.model.Patient.from_raw_parts", false]], "from_raw_parts() (gpsea.model.phenotype static method)": [[12, "gpsea.model.Phenotype.from_raw_parts", false]], "from_string() (gpsea.model.featuretype static method)": [[12, "gpsea.model.FeatureType.from_string", false]], "from_term() (gpsea.model.phenotype static method)": [[12, "gpsea.model.Phenotype.from_term", false]], "from_uniprot_json() (gpsea.model.proteinmetadata static method)": [[12, "gpsea.model.ProteinMetadata.from_uniprot_json", false]], "from_vcf_literal() (gpsea.model.variantcoordinates static method)": [[12, "gpsea.model.VariantCoordinates.from_vcf_literal", false]], "from_vcf_symbolic() (gpsea.model.variantcoordinates static method)": [[12, "gpsea.model.VariantCoordinates.from_vcf_symbolic", false]], "functionalannotationaware (class in gpsea.model)": [[12, "gpsea.model.FunctionalAnnotationAware", false]], "functionalannotator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.FunctionalAnnotator", false]], "genbank_acc (gpsea.model.genome.contig property)": [[13, "gpsea.model.genome.Contig.genbank_acc", false]], "gene() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.gene", false]], "gene_id (gpsea.model.imprecisesvinfo property)": [[12, "gpsea.model.ImpreciseSvInfo.gene_id", false]], "gene_id (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.gene_id", false]], "gene_id (gpsea.model.transcriptinfoaware property)": [[12, "gpsea.model.TranscriptInfoAware.gene_id", false]], "gene_symbol (gpsea.model.imprecisesvinfo property)": [[12, "gpsea.model.ImpreciseSvInfo.gene_symbol", false]], "genecoordinateservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.GeneCoordinateService", false]], "genome_build_id (gpsea.model.genome.genomebuild property)": [[13, "gpsea.model.genome.GenomeBuild.genome_build_id", false]], "genomebuild (class in gpsea.model.genome)": [[13, "gpsea.model.genome.GenomeBuild", false]], "genomebuildidentifier (class in gpsea.model.genome)": [[13, "gpsea.model.genome.GenomeBuildIdentifier", false]], "genomicregion (class in gpsea.model.genome)": [[13, "gpsea.model.genome.GenomicRegion", false]], "genotype (class in gpsea.model)": [[12, "gpsea.model.Genotype", false]], "genotype_for_sample() (gpsea.model.genotyped method)": [[12, "gpsea.model.Genotyped.genotype_for_sample", false]], "genotype_phenotype_scores (gpsea.analysis.pscore.phenotypescoreanalysisresult property)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.genotype_phenotype_scores", false]], "genotyped (class in gpsea.model)": [[12, "gpsea.model.Genotyped", false]], "genotypepolypredicate (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.GenotypePolyPredicate", false]], "genotypes (class in gpsea.model)": [[12, "gpsea.model.Genotypes", false]], "genotypes (gpsea.model.genotyped property)": [[12, "gpsea.model.Genotyped.genotypes", false]], "genotypes (gpsea.model.variant property)": [[12, "gpsea.model.Variant.genotypes", false]], "get_annotations() (gpsea.preprocessing.proteinannotationcache method)": [[14, "gpsea.preprocessing.ProteinAnnotationCache.get_annotations", false]], "get_annotations() (gpsea.preprocessing.variantannotationcache method)": [[14, "gpsea.preprocessing.VariantAnnotationCache.get_annotations", false]], "get_cache_dir_path() (in module gpsea.config)": [[10, "gpsea.config.get_cache_dir_path", false]], "get_categories() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_categories", false]], "get_categorizations() (gpsea.analysis.predicate.phenotype.diseasepresencepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.get_categorizations", false]], "get_categorizations() (gpsea.analysis.predicate.phenotype.hpopredicate method)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate.get_categorizations", false]], "get_categorizations() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_categorizations", false]], "get_category() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_category", false]], "get_category_name() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_category_name", false]], "get_category_names() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_category_names", false]], "get_cds_regions() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.get_cds_regions", false]], "get_coding_base_count() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.get_coding_base_count", false]], "get_codon_count() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.get_codon_count", false]], "get_excluded_count() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.get_excluded_count", false]], "get_features_variant_overlaps() (gpsea.model.proteinmetadata method)": [[12, "gpsea.model.ProteinMetadata.get_features_variant_overlaps", false]], "get_five_prime_utrs() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.get_five_prime_utrs", false]], "get_hgvs_cdna_by_tx_id() (gpsea.model.functionalannotationaware method)": [[12, "gpsea.model.FunctionalAnnotationAware.get_hgvs_cdna_by_tx_id", false]], "get_maximum_group_observed_hpo_frequency() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.get_maximum_group_observed_HPO_frequency", false]], "get_number_of_observed_hpo_observations() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.get_number_of_observed_hpo_observations", false]], "get_patient_ids() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.get_patient_ids", false]], "get_preferred_tx_annotation() (gpsea.model.functionalannotationaware method)": [[12, "gpsea.model.FunctionalAnnotationAware.get_preferred_tx_annotation", false]], "get_question() (gpsea.analysis.predicate.genotype.allelecounter method)": [[6, "gpsea.analysis.predicate.genotype.AlleleCounter.get_question", false]], "get_question() (gpsea.analysis.predicate.genotype.variantpredicate method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicate.get_question", false]], "get_question() (gpsea.analysis.pscore.countingphenotypescorer method)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer.get_question", false]], "get_question_base() (gpsea.analysis.predicate.phenotype.diseasepresencepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.get_question_base", false]], "get_question_base() (gpsea.analysis.predicate.phenotype.hpopredicate method)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate.get_question_base", false]], "get_question_base() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.get_question_base", false]], "get_response() (gpsea.preprocessing.vvmulticoordinateservice method)": [[14, "gpsea.preprocessing.VVMultiCoordinateService.get_response", false]], "get_three_prime_utrs() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.get_three_prime_utrs", false]], "get_tx_anno_by_tx_id() (gpsea.model.functionalannotationaware method)": [[12, "gpsea.model.FunctionalAnnotationAware.get_tx_anno_by_tx_id", false]], "get_variant_by_key() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.get_variant_by_key", false]], "gpsea": [[0, "module-gpsea", false]], "gpsea.analysis": [[1, "module-gpsea.analysis", false]], "gpsea.analysis.mtc_filter": [[2, "module-gpsea.analysis.mtc_filter", false]], "gpsea.analysis.pcats": [[3, "module-gpsea.analysis.pcats", false]], "gpsea.analysis.pcats.stats": [[4, "module-gpsea.analysis.pcats.stats", false]], "gpsea.analysis.predicate": [[5, "module-gpsea.analysis.predicate", false]], "gpsea.analysis.predicate.genotype": [[6, "module-gpsea.analysis.predicate.genotype", false]], "gpsea.analysis.predicate.phenotype": [[7, "module-gpsea.analysis.predicate.phenotype", false]], "gpsea.analysis.pscore": [[8, "module-gpsea.analysis.pscore", false]], "gpsea.analysis.pscore.stats": [[9, "module-gpsea.analysis.pscore.stats", false]], "gpsea.config": [[10, "module-gpsea.config", false]], "gpsea.io": [[11, "module-gpsea.io", false]], "gpsea.model": [[12, "module-gpsea.model", false]], "gpsea.model.genome": [[13, "module-gpsea.model.genome", false]], "gpsea.preprocessing": [[14, "module-gpsea.preprocessing", false]], "gpsea.util": [[15, "module-gpsea.util", false]], "gpsea.view": [[16, "module-gpsea.view", false]], "gpseajsondecoder (class in gpsea.io)": [[11, "gpsea.io.GpseaJSONDecoder", false]], "gpseajsonencoder (class in gpsea.io)": [[11, "gpsea.io.GpseaJSONEncoder", false]], "gpseareport (class in gpsea.view)": [[16, "gpsea.view.GpseaReport", false]], "groups_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.groups_predicate", false]], "has_errors() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.has_errors", false]], "has_errors_or_warnings() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.has_errors_or_warnings", false]], "has_sv_info() (gpsea.model.variantinfo method)": [[12, "gpsea.model.VariantInfo.has_sv_info", false]], "has_variant_coordinates() (gpsea.model.variantinfo method)": [[12, "gpsea.model.VariantInfo.has_variant_coordinates", false]], "has_warnings() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.has_warnings", false]], "hemizygous (gpsea.model.genotype attribute)": [[12, "gpsea.model.Genotype.HEMIZYGOUS", false]], "heterozygous (gpsea.model.genotype attribute)": [[12, "gpsea.model.Genotype.HETEROZYGOUS", false]], "hgvs_cdna (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.hgvs_cdna", false]], "hgvsp (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.hgvsp", false]], "homozygous_alternate (gpsea.model.genotype attribute)": [[12, "gpsea.model.Genotype.HOMOZYGOUS_ALTERNATE", false]], "homozygous_reference (gpsea.model.genotype attribute)": [[12, "gpsea.model.Genotype.HOMOZYGOUS_REFERENCE", false]], "hpomtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter", false]], "hpopredicate (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate", false]], "hpotermanalysis (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.HpoTermAnalysis", false]], "hpotermanalysisresult (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult", false]], "identifier (gpsea.model.disease property)": [[12, "gpsea.model.Disease.identifier", false]], "identifier (gpsea.model.genome.genomebuild property)": [[13, "gpsea.model.genome.GenomeBuild.identifier", false]], "identifier (gpsea.model.genome.genomebuildidentifier property)": [[13, "gpsea.model.genome.GenomeBuildIdentifier.identifier", false]], "identifier (gpsea.model.measurement property)": [[12, "gpsea.model.Measurement.identifier", false]], "identifier (gpsea.model.phenotype property)": [[12, "gpsea.model.Phenotype.identifier", false]], "identifier (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.identifier", false]], "imprecisesvfunctionalannotator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.ImpreciseSvFunctionalAnnotator", false]], "imprecisesvinfo (class in gpsea.model)": [[12, "gpsea.model.ImpreciseSvInfo", false]], "incomplete_terminal_codon_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.INCOMPLETE_TERMINAL_CODON_VARIANT", false]], "info (gpsea.model.proteinfeature property)": [[12, "gpsea.model.ProteinFeature.info", false]], "inframe_deletion (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.INFRAME_DELETION", false]], "inframe_insertion (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.INFRAME_INSERTION", false]], "ins (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.INS", false]], "intergenic_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.INTERGENIC_VARIANT", false]], "intron_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.INTRON_VARIANT", false]], "inv (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.INV", false]], "is_coding() (gpsea.model.transcriptcoordinates method)": [[12, "gpsea.model.TranscriptCoordinates.is_coding", false]], "is_empty() (gpsea.model.genome.region method)": [[13, "gpsea.model.genome.Region.is_empty", false]], "is_female() (gpsea.model.sex method)": [[12, "gpsea.model.Sex.is_female", false]], "is_filtered_out() (gpsea.analysis.mtc_filter.phenotypemtcresult method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.is_filtered_out", false]], "is_large_imprecise_sv() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.is_large_imprecise_sv", false]], "is_male() (gpsea.model.sex method)": [[12, "gpsea.model.Sex.is_male", false]], "is_negative() (gpsea.model.genome.strand method)": [[13, "gpsea.model.genome.Strand.is_negative", false]], "is_observed (gpsea.model.phenotype property)": [[12, "gpsea.model.Phenotype.is_observed", false]], "is_ok() (gpsea.preprocessing.preprocessingvalidationresult method)": [[14, "gpsea.preprocessing.PreprocessingValidationResult.is_ok", false]], "is_passed() (gpsea.analysis.mtc_filter.phenotypemtcresult method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.is_passed", false]], "is_positive() (gpsea.model.genome.strand method)": [[13, "gpsea.model.genome.Strand.is_positive", false]], "is_preferred (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.is_preferred", false]], "is_preferred (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.is_preferred", false]], "is_present (gpsea.model.disease property)": [[12, "gpsea.model.Disease.is_present", false]], "is_present (gpsea.model.phenotype property)": [[12, "gpsea.model.Phenotype.is_present", false]], "is_provided() (gpsea.model.sex method)": [[12, "gpsea.model.Sex.is_provided", false]], "is_structural() (gpsea.model.variantcoordinates method)": [[12, "gpsea.model.VariantCoordinates.is_structural", false]], "is_structural() (gpsea.model.variantinfo method)": [[12, "gpsea.model.VariantInfo.is_structural", false]], "is_structural_deletion() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.is_structural_deletion", false]], "is_structural_variant() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.is_structural_variant", false]], "is_unknown() (gpsea.model.sex method)": [[12, "gpsea.model.Sex.is_unknown", false]], "issues (gpsea.preprocessing.notepad property)": [[14, "gpsea.preprocessing.Notepad.issues", false]], "iterate_nodes() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.iterate_nodes", false]], "label (gpsea.model.proteinmetadata property)": [[12, "gpsea.model.ProteinMetadata.label", false]], "label (gpsea.model.samplelabels property)": [[12, "gpsea.model.SampleLabels.label", false]], "label (gpsea.preprocessing.notepad property)": [[14, "gpsea.preprocessing.Notepad.label", false]], "label_summary() (gpsea.model.samplelabels method)": [[12, "gpsea.model.SampleLabels.label_summary", false]], "labels (gpsea.model.patient property)": [[12, "gpsea.model.Patient.labels", false]], "level (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.Level", false]], "level (gpsea.preprocessing.datasanityissue property)": [[14, "gpsea.preprocessing.DataSanityIssue.level", false]], "level (gpsea.preprocessing.notepadtree property)": [[14, "gpsea.preprocessing.NotepadTree.level", false]], "list_all_diseases() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.list_all_diseases", false]], "list_all_proteins() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.list_all_proteins", false]], "list_all_variants() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.list_all_variants", false]], "list_present_phenotypes() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.list_present_phenotypes", false]], "load_phenopacket_files() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.load_phenopacket_files", false]], "load_phenopacket_folder() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.load_phenopacket_folder", false]], "load_phenopackets() (in module gpsea.preprocessing)": [[14, "gpsea.preprocessing.load_phenopackets", false]], "major_assembly (gpsea.model.genome.genomebuildidentifier property)": [[13, "gpsea.model.genome.GenomeBuildIdentifier.major_assembly", false]], "male (gpsea.model.sex attribute)": [[12, "gpsea.model.Sex.MALE", false]], "mannwhitneystatistic (class in gpsea.analysis.pscore.stats)": [[9, "gpsea.analysis.pscore.stats.MannWhitneyStatistic", false]], "marker_counts (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.marker_counts", false]], "mature_mirna_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.MATURE_MIRNA_VARIANT", false]], "measurement (class in gpsea.model)": [[12, "gpsea.model.Measurement", false]], "measurement_by_id() (gpsea.model.patient method)": [[12, "gpsea.model.Patient.measurement_by_id", false]], "measurementphenotypescorer (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.MeasurementPhenotypeScorer", false]], "measurements (gpsea.model.patient property)": [[12, "gpsea.model.Patient.measurements", false]], "message (gpsea.preprocessing.datasanityissue property)": [[14, "gpsea.preprocessing.DataSanityIssue.message", false]], "meta_label (gpsea.model.samplelabels property)": [[12, "gpsea.model.SampleLabels.meta_label", false]], "missense_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.MISSENSE_VARIANT", false]], "mnv (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.MNV", false]], "module": [[0, "module-gpsea", false], [1, "module-gpsea.analysis", false], [2, "module-gpsea.analysis.mtc_filter", false], [3, "module-gpsea.analysis.pcats", false], [4, "module-gpsea.analysis.pcats.stats", false], [5, "module-gpsea.analysis.predicate", false], [6, "module-gpsea.analysis.predicate.genotype", false], [7, "module-gpsea.analysis.predicate.phenotype", false], [8, "module-gpsea.analysis.pscore", false], [9, "module-gpsea.analysis.pscore.stats", false], [10, "module-gpsea.config", false], [11, "module-gpsea.io", false], [12, "module-gpsea.model", false], [13, "module-gpsea.model.genome", false], [14, "module-gpsea.preprocessing", false], [15, "module-gpsea.util", false], [16, "module-gpsea.view", false]], "monoallelic_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.monoallelic_predicate", false]], "motif (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.MOTIF", false]], "motifs() (gpsea.model.proteinmetadata method)": [[12, "gpsea.model.ProteinMetadata.motifs", false]], "mtc_filter_name (gpsea.analysis.pcats.hpotermanalysisresult property)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_filter_name", false]], "mtc_filter_results (gpsea.analysis.pcats.hpotermanalysisresult property)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_filter_results", false]], "mtc_issue (gpsea.analysis.mtc_filter.phenotypemtcresult property)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.mtc_issue", false]], "mtc_name (gpsea.analysis.pcats.hpotermanalysisresult property)": [[3, "gpsea.analysis.pcats.HpoTermAnalysisResult.mtc_name", false]], "mtcstatsviewer (class in gpsea.view)": [[16, "gpsea.view.MtcStatsViewer", false]], "multiphenotypeanalysis (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysis", false]], "multiphenotypeanalysisresult (class in gpsea.analysis.pcats)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult", false]], "n_categorizations() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.n_categorizations", false]], "n_usable (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.n_usable", false]], "name (gpsea.analysis.predicate.patientcategory property)": [[5, "gpsea.analysis.predicate.PatientCategory.name", false]], "name (gpsea.model.disease property)": [[12, "gpsea.model.Disease.name", false]], "name (gpsea.model.featureinfo property)": [[12, "gpsea.model.FeatureInfo.name", false]], "name (gpsea.model.genome.contig property)": [[13, "gpsea.model.genome.Contig.name", false]], "name (gpsea.model.measurement property)": [[12, "gpsea.model.Measurement.name", false]], "negative (gpsea.model.genome.strand attribute)": [[13, "gpsea.model.genome.Strand.NEGATIVE", false]], "nmd_transcript_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.NMD_TRANSCRIPT_VARIANT", false]], "no (gpsea.analysis.predicate.patientcategories attribute)": [[5, "gpsea.analysis.predicate.PatientCategories.NO", false]], "no_call (gpsea.model.genotype attribute)": [[12, "gpsea.model.Genotype.NO_CALL", false]], "no_genotype_has_more_than_one_hpo (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.NO_GENOTYPE_HAS_MORE_THAN_ONE_HPO", false]], "non_coding_transcript_exon_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.NON_CODING_TRANSCRIPT_EXON_VARIANT", false]], "non_coding_transcript_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.NON_CODING_TRANSCRIPT_VARIANT", false]], "non_specified_term (gpsea.analysis.mtc_filter.specifiedtermsmtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.NON_SPECIFIED_TERM", false]], "noncoding_effects (gpsea.preprocessing.vepfunctionalannotator attribute)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator.NONCODING_EFFECTS", false]], "notepad (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.Notepad", false]], "notepadtree (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.NotepadTree", false]], "object_hook() (gpsea.io.gpseajsondecoder static method)": [[11, "gpsea.io.GpseaJSONDecoder.object_hook", false]], "observed (gpsea.model.phenotype property)": [[12, "gpsea.model.Phenotype.observed", false]], "ok (gpsea.analysis.mtc_filter.phenotypemtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter.OK", false]], "ok() (gpsea.analysis.mtc_filter.phenotypemtcresult static method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.ok", false]], "one_genotype_has_zero_hpo_observations() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.one_genotype_has_zero_hpo_observations", false]], "open_text_io_handle_for_reading() (in module gpsea.util)": [[15, "gpsea.util.open_text_io_handle_for_reading", false]], "open_text_io_handle_for_writing() (in module gpsea.util)": [[15, "gpsea.util.open_text_io_handle_for_writing", false]], "opposite() (gpsea.model.genome.strand method)": [[13, "gpsea.model.genome.Strand.opposite", false]], "overlapping_exons (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.overlapping_exons", false]], "overlaps_with() (gpsea.model.featureinfo method)": [[12, "gpsea.model.FeatureInfo.overlaps_with", false]], "overlaps_with() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.overlaps_with", false]], "overlaps_with() (gpsea.model.genome.region method)": [[13, "gpsea.model.genome.Region.overlaps_with", false]], "parse_multiple() (gpsea.preprocessing.vvmulticoordinateservice method)": [[14, "gpsea.preprocessing.VVMultiCoordinateService.parse_multiple", false]], "parse_response() (gpsea.preprocessing.vvmulticoordinateservice method)": [[14, "gpsea.preprocessing.VVMultiCoordinateService.parse_response", false]], "parse_uniprot_json() (gpsea.preprocessing.uniprotproteinmetadataservice static method)": [[14, "gpsea.preprocessing.UniprotProteinMetadataService.parse_uniprot_json", false]], "patch (gpsea.model.genome.genomebuildidentifier property)": [[13, "gpsea.model.genome.GenomeBuildIdentifier.patch", false]], "patient (class in gpsea.model)": [[12, "gpsea.model.Patient", false]], "patient_id (gpsea.model.patient property)": [[12, "gpsea.model.Patient.patient_id", false]], "patientcategories (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.PatientCategories", false]], "patientcategory (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.PatientCategory", false]], "patientcreator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.PatientCreator", false]], "phenopacketpatientcreator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.PhenopacketPatientCreator", false]], "phenopacketvariantcoordinatefinder (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.PhenopacketVariantCoordinateFinder", false]], "phenotype (class in gpsea.model)": [[12, "gpsea.model.Phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.diseasepresencepredicate property)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.hpopredicate property)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate.phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.phenotypecategorization property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypeCategorization.phenotype", false]], "phenotype (gpsea.analysis.predicate.phenotype.phenotypepolypredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.phenotype", false]], "phenotypecategorization (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypeCategorization", false]], "phenotypecreator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.PhenotypeCreator", false]], "phenotypemtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter", false]], "phenotypemtcissue (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcIssue", false]], "phenotypemtcresult (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult", false]], "phenotypepolypredicate (class in gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate", false]], "phenotypes (gpsea.model.patient property)": [[12, "gpsea.model.Patient.phenotypes", false]], "phenotypescoreanalysis (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysis", false]], "phenotypescoreanalysisresult (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult", false]], "phenotypescorer (class in gpsea.analysis.pscore)": [[8, "gpsea.analysis.pscore.PhenotypeScorer", false]], "phenotypescorestatistic (class in gpsea.analysis.pscore.stats)": [[9, "gpsea.analysis.pscore.stats.PhenotypeScoreStatistic", false]], "plot_boxplots() (gpsea.analysis.pscore.phenotypescoreanalysisresult method)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.plot_boxplots", false]], "policy (gpsea.preprocessing.preprocessingvalidationresult property)": [[14, "gpsea.preprocessing.PreprocessingValidationResult.policy", false]], "polypredicate (class in gpsea.analysis.predicate)": [[5, "gpsea.analysis.predicate.PolyPredicate", false]], "positive (gpsea.model.genome.strand attribute)": [[13, "gpsea.model.genome.Strand.POSITIVE", false]], "possible_results() (gpsea.analysis.mtc_filter.hpomtcfilter method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.possible_results", false]], "possible_results() (gpsea.analysis.mtc_filter.phenotypemtcfilter method)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcFilter.possible_results", false]], "possible_results() (gpsea.analysis.mtc_filter.specifiedtermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter.possible_results", false]], "possible_results() (gpsea.analysis.mtc_filter.usealltermsmtcfilter method)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter.possible_results", false]], "prepare_hpo_terms_of_interest() (in module gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.prepare_hpo_terms_of_interest", false]], "prepare_notepad() (gpsea.preprocessing.auditor static method)": [[14, "gpsea.preprocessing.Auditor.prepare_notepad", false]], "prepare_predicates_for_terms_of_interest() (in module gpsea.analysis.predicate.phenotype)": [[7, "gpsea.analysis.predicate.phenotype.prepare_predicates_for_terms_of_interest", false]], "preprocessingvalidationresult (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.PreprocessingValidationResult", false]], "present_diseases() (gpsea.model.patient method)": [[12, "gpsea.model.Patient.present_diseases", false]], "present_phenotype_categorization (gpsea.analysis.predicate.phenotype.diseasepresencepredicate property)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.present_phenotype_categorization", false]], "present_phenotype_categorization (gpsea.analysis.predicate.phenotype.hpopredicate property)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate.present_phenotype_categorization", false]], "present_phenotype_categorization (gpsea.analysis.predicate.phenotype.phenotypepolypredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.present_phenotype_categorization", false]], "present_phenotype_category (gpsea.analysis.predicate.phenotype.phenotypepolypredicate property)": [[7, "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate.present_phenotype_category", false]], "present_phenotypes() (gpsea.model.patient method)": [[12, "gpsea.model.Patient.present_phenotypes", false]], "process() (gpsea.preprocessing.auditor method)": [[14, "gpsea.preprocessing.Auditor.process", false]], "process() (gpsea.preprocessing.cohortcreator method)": [[14, "gpsea.preprocessing.CohortCreator.process", false]], "process() (gpsea.preprocessing.phenopacketpatientcreator method)": [[14, "gpsea.preprocessing.PhenopacketPatientCreator.process", false]], "process() (gpsea.preprocessing.phenotypecreator method)": [[14, "gpsea.preprocessing.PhenotypeCreator.process", false]], "process() (gpsea.view.cohortvariantviewer method)": [[16, "gpsea.view.CohortVariantViewer.process", false]], "process() (gpsea.view.cohortviewable method)": [[16, "gpsea.view.CohortViewable.process", false]], "process() (gpsea.view.diseaseviewable method)": [[16, "gpsea.view.DiseaseViewable.process", false]], "process() (gpsea.view.mtcstatsviewer method)": [[16, "gpsea.view.MtcStatsViewer.process", false]], "process() (gpsea.view.proteinvariantviewer method)": [[16, "gpsea.view.ProteinVariantViewer.process", false]], "process_response() (gpsea.preprocessing.vepfunctionalannotator method)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator.process_response", false]], "protcachingmetadataservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.ProtCachingMetadataService", false]], "protein_altering_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.PROTEIN_ALTERING_VARIANT", false]], "protein_effect_location (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.protein_effect_location", false]], "protein_feature() (gpsea.analysis.predicate.genotype.proteinpredicates method)": [[6, "gpsea.analysis.predicate.genotype.ProteinPredicates.protein_feature", false]], "protein_feature_ends (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_ends", false]], "protein_feature_names (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_names", false]], "protein_feature_starts (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_starts", false]], "protein_feature_type() (gpsea.analysis.predicate.genotype.proteinpredicates method)": [[6, "gpsea.analysis.predicate.genotype.ProteinPredicates.protein_feature_type", false]], "protein_feature_types (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_feature_types", false]], "protein_features (gpsea.model.proteinmetadata property)": [[12, "gpsea.model.ProteinMetadata.protein_features", false]], "protein_id (gpsea.model.proteinmetadata property)": [[12, "gpsea.model.ProteinMetadata.protein_id", false]], "protein_id (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.protein_id", false]], "protein_id (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_id", false]], "protein_length (gpsea.model.proteinmetadata property)": [[12, "gpsea.model.ProteinMetadata.protein_length", false]], "protein_length (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_length", false]], "protein_metadata (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.protein_metadata", false]], "proteinannotationcache (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.ProteinAnnotationCache", false]], "proteinfeature (class in gpsea.model)": [[12, "gpsea.model.ProteinFeature", false]], "proteinmetadata (class in gpsea.model)": [[12, "gpsea.model.ProteinMetadata", false]], "proteinmetadataservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.ProteinMetadataService", false]], "proteinpredicates (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.ProteinPredicates", false]], "proteinvariantviewer (class in gpsea.view)": [[16, "gpsea.view.ProteinVariantViewer", false]], "proteinvisualizable (class in gpsea.view)": [[16, "gpsea.view.ProteinVisualizable", false]], "proteinvisualizer (class in gpsea.view)": [[16, "gpsea.view.ProteinVisualizer", false]], "pval (gpsea.analysis.pscore.phenotypescoreanalysisresult property)": [[8, "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.pval", false]], "pvals (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.pvals", false]], "reason (gpsea.analysis.mtc_filter.phenotypemtcissue attribute)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcIssue.reason", false]], "reason (gpsea.analysis.mtc_filter.phenotypemtcresult property)": [[2, "gpsea.analysis.mtc_filter.PhenotypeMtcResult.reason", false]], "ref (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.ref", false]], "ref_length() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.ref_length", false]], "refseq_name (gpsea.model.genome.contig property)": [[13, "gpsea.model.genome.Contig.refseq_name", false]], "region (class in gpsea.model.genome)": [[13, "gpsea.model.genome.Region", false]], "region (gpsea.model.featureinfo property)": [[12, "gpsea.model.FeatureInfo.region", false]], "region (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.REGION", false]], "region (gpsea.model.transcriptcoordinates property)": [[12, "gpsea.model.TranscriptCoordinates.region", false]], "region (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.region", false]], "region() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.region", false]], "regions() (gpsea.model.proteinmetadata method)": [[12, "gpsea.model.ProteinMetadata.regions", false]], "regulatory_region_ablation (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.REGULATORY_REGION_ABLATION", false]], "regulatory_region_amplification (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.REGULATORY_REGION_AMPLIFICATION", false]], "regulatory_region_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.REGULATORY_REGION_VARIANT", false]], "repeat (gpsea.model.featuretype attribute)": [[12, "gpsea.model.FeatureType.REPEAT", false]], "repeats() (gpsea.model.proteinmetadata method)": [[12, "gpsea.model.ProteinMetadata.repeats", false]], "same_count_as_the_only_child (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.SAME_COUNT_AS_THE_ONLY_CHILD", false]], "samplelabels (class in gpsea.model)": [[12, "gpsea.model.SampleLabels", false]], "score() (gpsea.analysis.pscore.countingphenotypescorer method)": [[8, "gpsea.analysis.pscore.CountingPhenotypeScorer.score", false]], "score() (gpsea.analysis.pscore.devriesphenotypescorer method)": [[8, "gpsea.analysis.pscore.DeVriesPhenotypeScorer.score", false]], "score() (gpsea.analysis.pscore.measurementphenotypescorer method)": [[8, "gpsea.analysis.pscore.MeasurementPhenotypeScorer.score", false]], "score() (gpsea.analysis.pscore.phenotypescorer method)": [[8, "gpsea.analysis.pscore.PhenotypeScorer.score", false]], "sequence_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SEQUENCE_VARIANT", false]], "sex (class in gpsea.model)": [[12, "gpsea.model.Sex", false]], "sex (gpsea.model.patient property)": [[12, "gpsea.model.Patient.sex", false]], "sex_predicate() (in module gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.sex_predicate", false]], "single() (gpsea.model.genotypes static method)": [[12, "gpsea.model.Genotypes.single", false]], "skipping_general_term (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.SKIPPING_GENERAL_TERM", false]], "skipping_non_phenotype_term (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.SKIPPING_NON_PHENOTYPE_TERM", false]], "skipping_since_one_genotype_had_zero_observations (gpsea.analysis.mtc_filter.hpomtcfilter attribute)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.SKIPPING_SINCE_ONE_GENOTYPE_HAD_ZERO_OBSERVATIONS", false]], "snv (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.SNV", false]], "solution (gpsea.preprocessing.datasanityissue property)": [[14, "gpsea.preprocessing.DataSanityIssue.solution", false]], "some_cell_has_greater_than_one_count() (gpsea.analysis.mtc_filter.hpomtcfilter static method)": [[2, "gpsea.analysis.mtc_filter.HpoMtcFilter.some_cell_has_greater_than_one_count", false]], "specifiedtermsmtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter", false]], "splice_acceptor_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_ACCEPTOR_VARIANT", false]], "splice_donor_5th_base_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_DONOR_5TH_BASE_VARIANT", false]], "splice_donor_region_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_DONOR_REGION_VARIANT", false]], "splice_donor_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_DONOR_VARIANT", false]], "splice_polypyrimidine_tract_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_POLYPYRIMIDINE_TRACT_VARIANT", false]], "splice_region_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SPLICE_REGION_VARIANT", false]], "start (gpsea.model.featureinfo property)": [[12, "gpsea.model.FeatureInfo.start", false]], "start (gpsea.model.genome.region property)": [[13, "gpsea.model.genome.Region.start", false]], "start (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.start", false]], "start_lost (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.START_LOST", false]], "start_on_strand() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.start_on_strand", false]], "start_retained_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.START_RETAINED_VARIANT", false]], "stop_gained (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.STOP_GAINED", false]], "stop_lost (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.STOP_LOST", false]], "stop_retained_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.STOP_RETAINED_VARIANT", false]], "store_annotations() (gpsea.preprocessing.proteinannotationcache method)": [[14, "gpsea.preprocessing.ProteinAnnotationCache.store_annotations", false]], "store_annotations() (gpsea.preprocessing.variantannotationcache method)": [[14, "gpsea.preprocessing.VariantAnnotationCache.store_annotations", false]], "strand (class in gpsea.model.genome)": [[13, "gpsea.model.genome.Strand", false]], "strand (gpsea.model.genome.genomicregion property)": [[13, "gpsea.model.genome.GenomicRegion.strand", false]], "strand (gpsea.model.genome.stranded property)": [[13, "gpsea.model.genome.Stranded.strand", false]], "stranded (class in gpsea.model.genome)": [[13, "gpsea.model.genome.Stranded", false]], "structural_so_id_to_display() (gpsea.model.varianteffect static method)": [[12, "gpsea.model.VariantEffect.structural_so_id_to_display", false]], "structural_type (gpsea.model.imprecisesvinfo property)": [[12, "gpsea.model.ImpreciseSvInfo.structural_type", false]], "structural_type() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.structural_type", false]], "summarize() (gpsea.preprocessing.preprocessingvalidationresult method)": [[14, "gpsea.preprocessing.PreprocessingValidationResult.summarize", false]], "summarize_hpo_analysis() (in module gpsea.view)": [[16, "gpsea.view.summarize_hpo_analysis", false]], "supports_shape (gpsea.analysis.pcats.stats.countstatistic property)": [[4, "gpsea.analysis.pcats.stats.CountStatistic.supports_shape", false]], "supports_shape (gpsea.analysis.pcats.stats.fisherexacttest property)": [[4, "gpsea.analysis.pcats.stats.FisherExactTest.supports_shape", false]], "sv_info (gpsea.model.variantinfo property)": [[12, "gpsea.model.VariantInfo.sv_info", false]], "symbol (gpsea.model.genome.strand property)": [[13, "gpsea.model.genome.Strand.symbol", false]], "synonymous_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.SYNONYMOUS_VARIANT", false]], "test() (gpsea.analysis.predicate.genotype.variantpredicate method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicate.test", false]], "test() (gpsea.analysis.predicate.phenotype.diseasepresencepredicate method)": [[7, "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate.test", false]], "test() (gpsea.analysis.predicate.phenotype.hpopredicate method)": [[7, "gpsea.analysis.predicate.phenotype.HpoPredicate.test", false]], "test() (gpsea.analysis.predicate.polypredicate method)": [[5, "gpsea.analysis.predicate.PolyPredicate.test", false]], "test_result (gpsea.model.measurement property)": [[12, "gpsea.model.Measurement.test_result", false]], "tf_binding_site_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.TF_BINDING_SITE_VARIANT", false]], "tfbs_ablation (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.TFBS_ABLATION", false]], "tfbs_amplification (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.TFBS_AMPLIFICATION", false]], "three_prime_utr_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.THREE_PRIME_UTR_VARIANT", false]], "to_display() (gpsea.model.varianteffect method)": [[12, "gpsea.model.VariantEffect.to_display", false]], "to_negative_strand() (gpsea.model.genome.transposable method)": [[13, "gpsea.model.genome.Transposable.to_negative_strand", false]], "to_opposite_strand() (gpsea.model.genome.transposable method)": [[13, "gpsea.model.genome.Transposable.to_opposite_strand", false]], "to_positive_strand() (gpsea.model.genome.transposable method)": [[13, "gpsea.model.genome.Transposable.to_positive_strand", false]], "to_string() (gpsea.model.proteinfeature method)": [[12, "gpsea.model.ProteinFeature.to_string", false]], "total_patient_count (gpsea.model.cohort property)": [[12, "gpsea.model.Cohort.total_patient_count", false]], "total_tests (gpsea.analysis.pcats.multiphenotypeanalysisresult property)": [[3, "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult.total_tests", false]], "transcript() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.transcript", false]], "transcript_ablation (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.TRANSCRIPT_ABLATION", false]], "transcript_amplification (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.TRANSCRIPT_AMPLIFICATION", false]], "transcript_coordinates (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.transcript_coordinates", false]], "transcript_id (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.transcript_id", false]], "transcript_id (gpsea.model.transcriptinfoaware property)": [[12, "gpsea.model.TranscriptInfoAware.transcript_id", false]], "transcript_id (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.transcript_id", false]], "transcriptannotation (class in gpsea.model)": [[12, "gpsea.model.TranscriptAnnotation", false]], "transcriptcoordinates (class in gpsea.model)": [[12, "gpsea.model.TranscriptCoordinates", false]], "transcriptcoordinateservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.TranscriptCoordinateService", false]], "transcriptinfoaware (class in gpsea.model)": [[12, "gpsea.model.TranscriptInfoAware", false]], "translocation (gpsea.model.variantclass attribute)": [[12, "gpsea.model.VariantClass.TRANSLOCATION", false]], "transposable (class in gpsea.model.genome)": [[13, "gpsea.model.genome.Transposable", false]], "transpose_coordinate() (in module gpsea.model.genome)": [[13, "gpsea.model.genome.transpose_coordinate", false]], "true() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.true", false]], "tteststatistic (class in gpsea.analysis.pscore.stats)": [[9, "gpsea.analysis.pscore.stats.TTestStatistic", false]], "tx_annotations (gpsea.model.functionalannotationaware property)": [[12, "gpsea.model.FunctionalAnnotationAware.tx_annotations", false]], "tx_annotations (gpsea.model.variant property)": [[12, "gpsea.model.Variant.tx_annotations", false]], "ucsc_name (gpsea.model.genome.contig property)": [[13, "gpsea.model.genome.Contig.ucsc_name", false]], "uniprotproteinmetadataservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.UniprotProteinMetadataService", false]], "unit (gpsea.model.measurement property)": [[12, "gpsea.model.Measurement.unit", false]], "unknown_sex (gpsea.model.sex attribute)": [[12, "gpsea.model.Sex.UNKNOWN_SEX", false]], "upstream_gene_variant (gpsea.model.varianteffect attribute)": [[12, "gpsea.model.VariantEffect.UPSTREAM_GENE_VARIANT", false]], "usealltermsmtcfilter (class in gpsea.analysis.mtc_filter)": [[2, "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter", false]], "varcachingfunctionalannotator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VarCachingFunctionalAnnotator", false]], "variant (class in gpsea.model)": [[12, "gpsea.model.Variant", false]], "variant_class (gpsea.model.imprecisesvinfo property)": [[12, "gpsea.model.ImpreciseSvInfo.variant_class", false]], "variant_class (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.variant_class", false]], "variant_class (gpsea.model.variantinfo property)": [[12, "gpsea.model.VariantInfo.variant_class", false]], "variant_class() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.variant_class", false]], "variant_coordinates (gpsea.model.variantinfo property)": [[12, "gpsea.model.VariantInfo.variant_coordinates", false]], "variant_effect() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.variant_effect", false]], "variant_effect_count_by_tx() (gpsea.model.cohort method)": [[12, "gpsea.model.Cohort.variant_effect_count_by_tx", false]], "variant_effects (gpsea.model.transcriptannotation property)": [[12, "gpsea.model.TranscriptAnnotation.variant_effects", false]], "variant_effects (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.variant_effects", false]], "variant_info (gpsea.model.variant property)": [[12, "gpsea.model.Variant.variant_info", false]], "variant_info (gpsea.model.variantinfoaware property)": [[12, "gpsea.model.VariantInfoAware.variant_info", false]], "variant_key (gpsea.model.imprecisesvinfo property)": [[12, "gpsea.model.ImpreciseSvInfo.variant_key", false]], "variant_key (gpsea.model.variantcoordinates property)": [[12, "gpsea.model.VariantCoordinates.variant_key", false]], "variant_key (gpsea.model.variantinfo property)": [[12, "gpsea.model.VariantInfo.variant_key", false]], "variant_key() (gpsea.analysis.predicate.genotype.variantpredicates static method)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates.variant_key", false]], "variant_locations (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.variant_locations", false]], "variant_locations_counted_absolute (gpsea.view.proteinvisualizable property)": [[16, "gpsea.view.ProteinVisualizable.variant_locations_counted_absolute", false]], "variantannotationcache (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VariantAnnotationCache", false]], "variantclass (class in gpsea.model)": [[12, "gpsea.model.VariantClass", false]], "variantcoordinatefinder (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VariantCoordinateFinder", false]], "variantcoordinates (class in gpsea.model)": [[12, "gpsea.model.VariantCoordinates", false]], "varianteffect (class in gpsea.model)": [[12, "gpsea.model.VariantEffect", false]], "variantformatter (class in gpsea.view)": [[16, "gpsea.view.VariantFormatter", false]], "variantinfo (class in gpsea.model)": [[12, "gpsea.model.VariantInfo", false]], "variantinfoaware (class in gpsea.model)": [[12, "gpsea.model.VariantInfoAware", false]], "variantpredicate (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicate", false]], "variantpredicates (class in gpsea.analysis.predicate.genotype)": [[6, "gpsea.analysis.predicate.genotype.VariantPredicates", false]], "variants (gpsea.model.patient property)": [[12, "gpsea.model.Patient.variants", false]], "varianttranscriptvisualizer (class in gpsea.view)": [[16, "gpsea.view.VariantTranscriptVisualizer", false]], "vepfunctionalannotator (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VepFunctionalAnnotator", false]], "visit() (gpsea.preprocessing.notepadtree method)": [[14, "gpsea.preprocessing.NotepadTree.visit", false]], "vvhgvsvariantcoordinatefinder (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VVHgvsVariantCoordinateFinder", false]], "vvmulticoordinateservice (class in gpsea.preprocessing)": [[14, "gpsea.preprocessing.VVMultiCoordinateService", false]], "warn (gpsea.preprocessing.level attribute)": [[14, "gpsea.preprocessing.Level.WARN", false]], "warning_count() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.warning_count", false]], "warnings() (gpsea.preprocessing.notepad method)": [[14, "gpsea.preprocessing.Notepad.warnings", false]], "with_cache_folder() (gpsea.preprocessing.varcachingfunctionalannotator static method)": [[14, "gpsea.preprocessing.VarCachingFunctionalAnnotator.with_cache_folder", false]], "with_strand() (gpsea.model.genome.genomicregion method)": [[13, "gpsea.model.genome.GenomicRegion.with_strand", false]], "with_strand() (gpsea.model.genome.transposable method)": [[13, "gpsea.model.genome.Transposable.with_strand", false]], "wrap_scoring_function() (gpsea.analysis.pscore.phenotypescorer static method)": [[8, "gpsea.analysis.pscore.PhenotypeScorer.wrap_scoring_function", false]], "write() (gpsea.view.gpseareport method)": [[16, "gpsea.view.GpseaReport.write", false]], "yes (gpsea.analysis.predicate.patientcategories attribute)": [[5, "gpsea.analysis.predicate.PatientCategories.YES", false]]}, "objects": {"": [[0, 0, 0, "-", "gpsea"]], "gpsea": [[1, 0, 0, "-", "analysis"], [10, 0, 0, "-", "config"], [11, 0, 0, "-", "io"], [12, 0, 0, "-", "model"], [14, 0, 0, "-", "preprocessing"], [15, 0, 0, "-", "util"], [16, 0, 0, "-", "view"]], "gpsea.analysis": [[2, 0, 0, "-", "mtc_filter"], [3, 0, 0, "-", "pcats"], [5, 0, 0, "-", "predicate"], [8, 0, 0, "-", "pscore"]], "gpsea.analysis.mtc_filter": [[2, 1, 1, "", "HpoMtcFilter"], [2, 1, 1, "", "PhenotypeMtcFilter"], [2, 1, 1, "", "PhenotypeMtcIssue"], [2, 1, 1, "", "PhenotypeMtcResult"], [2, 1, 1, "", "SpecifiedTermsMtcFilter"], [2, 1, 1, "", "UseAllTermsMtcFilter"]], "gpsea.analysis.mtc_filter.HpoMtcFilter": [[2, 2, 1, "", "NO_GENOTYPE_HAS_MORE_THAN_ONE_HPO"], [2, 2, 1, "", "SAME_COUNT_AS_THE_ONLY_CHILD"], [2, 2, 1, "", "SKIPPING_GENERAL_TERM"], [2, 2, 1, "", "SKIPPING_NON_PHENOTYPE_TERM"], [2, 2, 1, "", "SKIPPING_SINCE_ONE_GENOTYPE_HAD_ZERO_OBSERVATIONS"], [2, 3, 1, "", "default_filter"], [2, 3, 1, "", "filter"], [2, 3, 1, "", "filter_method_name"], [2, 3, 1, "", "get_maximum_group_observed_HPO_frequency"], [2, 3, 1, "", "get_number_of_observed_hpo_observations"], [2, 3, 1, "", "one_genotype_has_zero_hpo_observations"], [2, 3, 1, "", "possible_results"], [2, 3, 1, "", "some_cell_has_greater_than_one_count"]], "gpsea.analysis.mtc_filter.PhenotypeMtcFilter": [[2, 2, 1, "", "OK"], [2, 3, 1, "", "filter"], [2, 3, 1, "", "filter_method_name"], [2, 3, 1, "", "possible_results"]], "gpsea.analysis.mtc_filter.PhenotypeMtcIssue": [[2, 2, 1, "", "code"], [2, 2, 1, "", "reason"]], "gpsea.analysis.mtc_filter.PhenotypeMtcResult": [[2, 3, 1, "", "fail"], [2, 3, 1, "", "is_filtered_out"], [2, 3, 1, "", "is_passed"], [2, 4, 1, "", "mtc_issue"], [2, 3, 1, "", "ok"], [2, 4, 1, "", "reason"]], "gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter": [[2, 2, 1, "", "NON_SPECIFIED_TERM"], [2, 3, 1, "", "filter"], [2, 3, 1, "", "filter_method_name"], [2, 3, 1, "", "possible_results"]], "gpsea.analysis.mtc_filter.UseAllTermsMtcFilter": [[2, 3, 1, "", "filter"], [2, 3, 1, "", "filter_method_name"], [2, 3, 1, "", "possible_results"]], "gpsea.analysis.pcats": [[3, 1, 1, "", "DiseaseAnalysis"], [3, 1, 1, "", "HpoTermAnalysis"], [3, 1, 1, "", "HpoTermAnalysisResult"], [3, 1, 1, "", "MultiPhenotypeAnalysis"], [3, 1, 1, "", "MultiPhenotypeAnalysisResult"], [3, 5, 1, "", "apply_predicates_on_patients"], [4, 0, 0, "-", "stats"]], "gpsea.analysis.pcats.HpoTermAnalysisResult": [[3, 4, 1, "", "mtc_filter_name"], [3, 4, 1, "", "mtc_filter_results"], [3, 4, 1, "", "mtc_name"]], "gpsea.analysis.pcats.MultiPhenotypeAnalysis": [[3, 3, 1, "", "compare_genotype_vs_phenotypes"]], "gpsea.analysis.pcats.MultiPhenotypeAnalysisResult": [[3, 4, 1, "", "all_counts"], [3, 4, 1, "", "corrected_pvals"], [3, 4, 1, "", "n_usable"], [3, 4, 1, "", "pvals"], [3, 4, 1, "", "total_tests"]], "gpsea.analysis.pcats.stats": [[4, 1, 1, "", "CountStatistic"], [4, 1, 1, "", "FisherExactTest"]], "gpsea.analysis.pcats.stats.CountStatistic": [[4, 3, 1, "", "compute_pval"], [4, 4, 1, "", "supports_shape"]], "gpsea.analysis.pcats.stats.FisherExactTest": [[4, 3, 1, "", "compute_pval"], [4, 4, 1, "", "supports_shape"]], "gpsea.analysis.predicate": [[5, 1, 1, "", "Categorization"], [5, 1, 1, "", "PatientCategories"], [5, 1, 1, "", "PatientCategory"], [5, 1, 1, "", "PolyPredicate"], [6, 0, 0, "-", "genotype"], [7, 0, 0, "-", "phenotype"]], "gpsea.analysis.predicate.Categorization": [[5, 4, 1, "", "category"], [5, 3, 1, "", "from_raw_parts"]], "gpsea.analysis.predicate.PatientCategories": [[5, 2, 1, "", "NO"], [5, 2, 1, "", "YES"]], "gpsea.analysis.predicate.PatientCategory": [[5, 4, 1, "", "cat_id"], [5, 4, 1, "", "description"], [5, 4, 1, "", "name"]], "gpsea.analysis.predicate.PolyPredicate": [[5, 3, 1, "", "display_question"], [5, 3, 1, "", "get_categories"], [5, 3, 1, "", "get_categorizations"], [5, 3, 1, "", "get_category"], [5, 3, 1, "", "get_category_name"], [5, 3, 1, "", "get_category_names"], [5, 3, 1, "", "get_question_base"], [5, 3, 1, "", "n_categorizations"], [5, 3, 1, "", "test"]], "gpsea.analysis.predicate.genotype": [[6, 1, 1, "", "AlleleCounter"], [6, 1, 1, "", "GenotypePolyPredicate"], [6, 1, 1, "", "ProteinPredicates"], [6, 1, 1, "", "VariantPredicate"], [6, 1, 1, "", "VariantPredicates"], [6, 5, 1, "", "autosomal_dominant"], [6, 5, 1, "", "autosomal_recessive"], [6, 5, 1, "", "biallelic_predicate"], [6, 5, 1, "", "diagnosis_predicate"], [6, 5, 1, "", "groups_predicate"], [6, 5, 1, "", "monoallelic_predicate"], [6, 5, 1, "", "sex_predicate"]], "gpsea.analysis.predicate.genotype.AlleleCounter": [[6, 3, 1, "", "count"], [6, 3, 1, "", "get_question"]], "gpsea.analysis.predicate.genotype.ProteinPredicates": [[6, 3, 1, "", "protein_feature"], [6, 3, 1, "", "protein_feature_type"]], "gpsea.analysis.predicate.genotype.VariantPredicate": [[6, 3, 1, "", "get_question"], [6, 3, 1, "", "test"]], "gpsea.analysis.predicate.genotype.VariantPredicates": [[6, 3, 1, "", "all"], [6, 3, 1, "", "any"], [6, 3, 1, "", "change_length"], [6, 3, 1, "", "exon"], [6, 3, 1, "", "gene"], [6, 3, 1, "", "is_large_imprecise_sv"], [6, 3, 1, "", "is_structural_deletion"], [6, 3, 1, "", "is_structural_variant"], [6, 3, 1, "", "ref_length"], [6, 3, 1, "", "region"], [6, 3, 1, "", "structural_type"], [6, 3, 1, "", "transcript"], [6, 3, 1, "", "true"], [6, 3, 1, "", "variant_class"], [6, 3, 1, "", "variant_effect"], [6, 3, 1, "", "variant_key"]], "gpsea.analysis.predicate.phenotype": [[7, 1, 1, "", "DiseasePresencePredicate"], [7, 1, 1, "", "HpoPredicate"], [7, 1, 1, "", "PhenotypeCategorization"], [7, 1, 1, "", "PhenotypePolyPredicate"], [7, 5, 1, "", "prepare_hpo_terms_of_interest"], [7, 5, 1, "", "prepare_predicates_for_terms_of_interest"]], "gpsea.analysis.predicate.phenotype.DiseasePresencePredicate": [[7, 3, 1, "", "get_categorizations"], [7, 3, 1, "", "get_question_base"], [7, 4, 1, "", "phenotype"], [7, 4, 1, "", "present_phenotype_categorization"], [7, 3, 1, "", "test"]], "gpsea.analysis.predicate.phenotype.HpoPredicate": [[7, 3, 1, "", "get_categorizations"], [7, 3, 1, "", "get_question_base"], [7, 4, 1, "", "phenotype"], [7, 4, 1, "", "present_phenotype_categorization"], [7, 3, 1, "", "test"]], "gpsea.analysis.predicate.phenotype.PhenotypeCategorization": [[7, 4, 1, "", "phenotype"]], "gpsea.analysis.predicate.phenotype.PhenotypePolyPredicate": [[7, 4, 1, "", "phenotype"], [7, 4, 1, "", "present_phenotype_categorization"], [7, 4, 1, "", "present_phenotype_category"]], "gpsea.analysis.pscore": [[8, 1, 1, "", "CountingPhenotypeScorer"], [8, 1, 1, "", "DeVriesPhenotypeScorer"], [8, 1, 1, "", "MeasurementPhenotypeScorer"], [8, 1, 1, "", "PhenotypeScoreAnalysis"], [8, 1, 1, "", "PhenotypeScoreAnalysisResult"], [8, 1, 1, "", "PhenotypeScorer"], [9, 0, 0, "-", "stats"]], "gpsea.analysis.pscore.CountingPhenotypeScorer": [[8, 3, 1, "", "from_query_curies"], [8, 3, 1, "", "get_question"], [8, 3, 1, "", "score"]], "gpsea.analysis.pscore.DeVriesPhenotypeScorer": [[8, 3, 1, "", "score"]], "gpsea.analysis.pscore.MeasurementPhenotypeScorer": [[8, 3, 1, "", "from_measurement_id"], [8, 3, 1, "", "score"]], "gpsea.analysis.pscore.PhenotypeScoreAnalysis": [[8, 3, 1, "", "compare_genotype_vs_phenotype_score"]], "gpsea.analysis.pscore.PhenotypeScoreAnalysisResult": [[8, 4, 1, "", "genotype_phenotype_scores"], [8, 3, 1, "", "plot_boxplots"], [8, 4, 1, "", "pval"]], "gpsea.analysis.pscore.PhenotypeScorer": [[8, 3, 1, "", "score"], [8, 3, 1, "", "wrap_scoring_function"]], "gpsea.analysis.pscore.stats": [[9, 1, 1, "", "MannWhitneyStatistic"], [9, 1, 1, "", "PhenotypeScoreStatistic"], [9, 1, 1, "", "TTestStatistic"]], "gpsea.analysis.pscore.stats.MannWhitneyStatistic": [[9, 3, 1, "", "compute_pval"]], "gpsea.analysis.pscore.stats.PhenotypeScoreStatistic": [[9, 3, 1, "", "compute_pval"]], "gpsea.analysis.pscore.stats.TTestStatistic": [[9, 3, 1, "", "compute_pval"]], "gpsea.config": [[10, 6, 1, "", "CACHE_ENV"], [10, 6, 1, "", "DEFAULT_CACHE_PATH"], [10, 5, 1, "", "get_cache_dir_path"]], "gpsea.io": [[11, 1, 1, "", "GpseaJSONDecoder"], [11, 1, 1, "", "GpseaJSONEncoder"]], "gpsea.io.GpseaJSONDecoder": [[11, 3, 1, "", "object_hook"]], "gpsea.io.GpseaJSONEncoder": [[11, 3, 1, "", "default"]], "gpsea.model": [[12, 1, 1, "", "Cohort"], [12, 1, 1, "", "Disease"], [12, 1, 1, "", "FeatureInfo"], [12, 1, 1, "", "FeatureType"], [12, 1, 1, "", "FunctionalAnnotationAware"], [12, 1, 1, "", "Genotype"], [12, 1, 1, "", "Genotyped"], [12, 1, 1, "", "Genotypes"], [12, 1, 1, "", "ImpreciseSvInfo"], [12, 1, 1, "", "Measurement"], [12, 1, 1, "", "Patient"], [12, 1, 1, "", "Phenotype"], [12, 1, 1, "", "ProteinFeature"], [12, 1, 1, "", "ProteinMetadata"], [12, 1, 1, "", "SampleLabels"], [12, 1, 1, "", "Sex"], [12, 1, 1, "", "TranscriptAnnotation"], [12, 1, 1, "", "TranscriptCoordinates"], [12, 1, 1, "", "TranscriptInfoAware"], [12, 1, 1, "", "Variant"], [12, 1, 1, "", "VariantClass"], [12, 1, 1, "", "VariantCoordinates"], [12, 1, 1, "", "VariantEffect"], [12, 1, 1, "", "VariantInfo"], [12, 1, 1, "", "VariantInfoAware"], [13, 0, 0, "-", "genome"]], "gpsea.model.Cohort": [[12, 3, 1, "", "all_diseases"], [12, 4, 1, "", "all_patients"], [12, 3, 1, "", "all_phenotypes"], [12, 4, 1, "", "all_transcript_ids"], [12, 3, 1, "", "all_variants"], [12, 3, 1, "", "from_patients"], [12, 3, 1, "", "get_excluded_count"], [12, 3, 1, "", "get_patient_ids"], [12, 3, 1, "", "get_variant_by_key"], [12, 3, 1, "", "list_all_diseases"], [12, 3, 1, "", "list_all_proteins"], [12, 3, 1, "", "list_all_variants"], [12, 3, 1, "", "list_present_phenotypes"], [12, 4, 1, "", "total_patient_count"], [12, 3, 1, "", "variant_effect_count_by_tx"]], "gpsea.model.Disease": [[12, 4, 1, "", "identifier"], [12, 4, 1, "", "is_present"], [12, 4, 1, "", "name"]], "gpsea.model.FeatureInfo": [[12, 4, 1, "", "end"], [12, 4, 1, "", "name"], [12, 3, 1, "", "overlaps_with"], [12, 4, 1, "", "region"], [12, 4, 1, "", "start"]], "gpsea.model.FeatureType": [[12, 2, 1, "", "COILED_COIL"], [12, 2, 1, "", "COMPOSITIONAL_BIAS"], [12, 2, 1, "", "DOMAIN"], [12, 2, 1, "", "MOTIF"], [12, 2, 1, "", "REGION"], [12, 2, 1, "", "REPEAT"], [12, 3, 1, "", "from_string"]], "gpsea.model.FunctionalAnnotationAware": [[12, 3, 1, "", "get_hgvs_cdna_by_tx_id"], [12, 3, 1, "", "get_preferred_tx_annotation"], [12, 3, 1, "", "get_tx_anno_by_tx_id"], [12, 4, 1, "", "tx_annotations"]], "gpsea.model.Genotype": [[12, 2, 1, "", "HEMIZYGOUS"], [12, 2, 1, "", "HETEROZYGOUS"], [12, 2, 1, "", "HOMOZYGOUS_ALTERNATE"], [12, 2, 1, "", "HOMOZYGOUS_REFERENCE"], [12, 2, 1, "", "NO_CALL"], [12, 4, 1, "", "code"]], "gpsea.model.Genotyped": [[12, 3, 1, "", "genotype_for_sample"], [12, 4, 1, "", "genotypes"]], "gpsea.model.Genotypes": [[12, 3, 1, "", "empty"], [12, 3, 1, "", "for_sample"], [12, 3, 1, "", "from_mapping"], [12, 3, 1, "", "single"]], "gpsea.model.ImpreciseSvInfo": [[12, 4, 1, "", "gene_id"], [12, 4, 1, "", "gene_symbol"], [12, 4, 1, "", "structural_type"], [12, 4, 1, "", "variant_class"], [12, 4, 1, "", "variant_key"]], "gpsea.model.Measurement": [[12, 4, 1, "", "identifier"], [12, 4, 1, "", "name"], [12, 4, 1, "", "test_result"], [12, 4, 1, "", "unit"]], "gpsea.model.Patient": [[12, 4, 1, "", "diseases"], [12, 3, 1, "", "excluded_diseases"], [12, 3, 1, "", "excluded_phenotypes"], [12, 3, 1, "", "from_raw_parts"], [12, 4, 1, "", "labels"], [12, 3, 1, "", "measurement_by_id"], [12, 4, 1, "", "measurements"], [12, 4, 1, "", "patient_id"], [12, 4, 1, "", "phenotypes"], [12, 3, 1, "", "present_diseases"], [12, 3, 1, "", "present_phenotypes"], [12, 4, 1, "", "sex"], [12, 4, 1, "", "variants"]], "gpsea.model.Phenotype": [[12, 3, 1, "", "from_raw_parts"], [12, 3, 1, "", "from_term"], [12, 4, 1, "", "identifier"], [12, 4, 1, "", "is_observed"], [12, 4, 1, "", "is_present"], [12, 4, 1, "", "observed"]], "gpsea.model.ProteinFeature": [[12, 3, 1, "", "create"], [12, 4, 1, "", "feature_type"], [12, 4, 1, "", "info"], [12, 3, 1, "", "to_string"]], "gpsea.model.ProteinMetadata": [[12, 3, 1, "", "domains"], [12, 3, 1, "", "from_feature_frame"], [12, 3, 1, "", "from_uniprot_json"], [12, 3, 1, "", "get_features_variant_overlaps"], [12, 4, 1, "", "label"], [12, 3, 1, "", "motifs"], [12, 4, 1, "", "protein_features"], [12, 4, 1, "", "protein_id"], [12, 4, 1, "", "protein_length"], [12, 3, 1, "", "regions"], [12, 3, 1, "", "repeats"]], "gpsea.model.SampleLabels": [[12, 4, 1, "", "label"], [12, 3, 1, "", "label_summary"], [12, 4, 1, "", "meta_label"]], "gpsea.model.Sex": [[12, 2, 1, "", "FEMALE"], [12, 2, 1, "", "MALE"], [12, 2, 1, "", "UNKNOWN_SEX"], [12, 3, 1, "", "is_female"], [12, 3, 1, "", "is_male"], [12, 3, 1, "", "is_provided"], [12, 3, 1, "", "is_unknown"]], "gpsea.model.TranscriptAnnotation": [[12, 4, 1, "", "gene_id"], [12, 4, 1, "", "hgvs_cdna"], [12, 4, 1, "", "hgvsp"], [12, 4, 1, "", "is_preferred"], [12, 4, 1, "", "overlapping_exons"], [12, 4, 1, "", "protein_effect_location"], [12, 4, 1, "", "protein_id"], [12, 4, 1, "", "transcript_id"], [12, 4, 1, "", "variant_effects"]], "gpsea.model.TranscriptCoordinates": [[12, 4, 1, "", "cds_end"], [12, 4, 1, "", "cds_start"], [12, 4, 1, "", "exons"], [12, 3, 1, "", "get_cds_regions"], [12, 3, 1, "", "get_coding_base_count"], [12, 3, 1, "", "get_codon_count"], [12, 3, 1, "", "get_five_prime_utrs"], [12, 3, 1, "", "get_three_prime_utrs"], [12, 4, 1, "", "identifier"], [12, 3, 1, "", "is_coding"], [12, 4, 1, "", "is_preferred"], [12, 4, 1, "", "region"]], "gpsea.model.TranscriptInfoAware": [[12, 4, 1, "", "gene_id"], [12, 4, 1, "", "transcript_id"]], "gpsea.model.Variant": [[12, 3, 1, "", "create_variant_from_scratch"], [12, 4, 1, "", "genotypes"], [12, 4, 1, "", "tx_annotations"], [12, 4, 1, "", "variant_info"]], "gpsea.model.VariantClass": [[12, 2, 1, "", "DEL"], [12, 2, 1, "", "DUP"], [12, 2, 1, "", "INS"], [12, 2, 1, "", "INV"], [12, 2, 1, "", "MNV"], [12, 2, 1, "", "SNV"], [12, 2, 1, "", "TRANSLOCATION"]], "gpsea.model.VariantCoordinates": [[12, 4, 1, "", "alt"], [12, 4, 1, "", "change_length"], [12, 4, 1, "", "chrom"], [12, 4, 1, "", "end"], [12, 3, 1, "", "from_vcf_literal"], [12, 3, 1, "", "from_vcf_symbolic"], [12, 3, 1, "", "is_structural"], [12, 4, 1, "", "ref"], [12, 4, 1, "", "region"], [12, 4, 1, "", "start"], [12, 4, 1, "", "variant_class"], [12, 4, 1, "", "variant_key"]], "gpsea.model.VariantEffect": [[12, 2, 1, "", "CODING_SEQUENCE_VARIANT"], [12, 2, 1, "", "DOWNSTREAM_GENE_VARIANT"], [12, 2, 1, "", "FEATURE_ELONGATION"], [12, 2, 1, "", "FEATURE_TRUNCATION"], [12, 2, 1, "", "FIVE_PRIME_UTR_VARIANT"], [12, 2, 1, "", "FRAMESHIFT_VARIANT"], [12, 2, 1, "", "INCOMPLETE_TERMINAL_CODON_VARIANT"], [12, 2, 1, "", "INFRAME_DELETION"], [12, 2, 1, "", "INFRAME_INSERTION"], [12, 2, 1, "", "INTERGENIC_VARIANT"], [12, 2, 1, "", "INTRON_VARIANT"], [12, 2, 1, "", "MATURE_MIRNA_VARIANT"], [12, 2, 1, "", "MISSENSE_VARIANT"], [12, 2, 1, "", "NMD_TRANSCRIPT_VARIANT"], [12, 2, 1, "", "NON_CODING_TRANSCRIPT_EXON_VARIANT"], [12, 2, 1, "", "NON_CODING_TRANSCRIPT_VARIANT"], [12, 2, 1, "", "PROTEIN_ALTERING_VARIANT"], [12, 2, 1, "", "REGULATORY_REGION_ABLATION"], [12, 2, 1, "", "REGULATORY_REGION_AMPLIFICATION"], [12, 2, 1, "", "REGULATORY_REGION_VARIANT"], [12, 2, 1, "", "SEQUENCE_VARIANT"], [12, 2, 1, "", "SPLICE_ACCEPTOR_VARIANT"], [12, 2, 1, "", "SPLICE_DONOR_5TH_BASE_VARIANT"], [12, 2, 1, "", "SPLICE_DONOR_REGION_VARIANT"], [12, 2, 1, "", "SPLICE_DONOR_VARIANT"], [12, 2, 1, "", "SPLICE_POLYPYRIMIDINE_TRACT_VARIANT"], [12, 2, 1, "", "SPLICE_REGION_VARIANT"], [12, 2, 1, "", "START_LOST"], [12, 2, 1, "", "START_RETAINED_VARIANT"], [12, 2, 1, "", "STOP_GAINED"], [12, 2, 1, "", "STOP_LOST"], [12, 2, 1, "", "STOP_RETAINED_VARIANT"], [12, 2, 1, "", "SYNONYMOUS_VARIANT"], [12, 2, 1, "", "TFBS_ABLATION"], [12, 2, 1, "", "TFBS_AMPLIFICATION"], [12, 2, 1, "", "TF_BINDING_SITE_VARIANT"], [12, 2, 1, "", "THREE_PRIME_UTR_VARIANT"], [12, 2, 1, "", "TRANSCRIPT_ABLATION"], [12, 2, 1, "", "TRANSCRIPT_AMPLIFICATION"], [12, 2, 1, "", "UPSTREAM_GENE_VARIANT"], [12, 4, 1, "", "curie"], [12, 3, 1, "", "structural_so_id_to_display"], [12, 3, 1, "", "to_display"]], "gpsea.model.VariantInfo": [[12, 3, 1, "", "has_sv_info"], [12, 3, 1, "", "has_variant_coordinates"], [12, 3, 1, "", "is_structural"], [12, 4, 1, "", "sv_info"], [12, 4, 1, "", "variant_class"], [12, 4, 1, "", "variant_coordinates"], [12, 4, 1, "", "variant_key"]], "gpsea.model.VariantInfoAware": [[12, 4, 1, "", "variant_info"]], "gpsea.model.genome": [[13, 1, 1, "", "Contig"], [13, 1, 1, "", "GenomeBuild"], [13, 1, 1, "", "GenomeBuildIdentifier"], [13, 1, 1, "", "GenomicRegion"], [13, 1, 1, "", "Region"], [13, 1, 1, "", "Strand"], [13, 1, 1, "", "Stranded"], [13, 1, 1, "", "Transposable"], [13, 5, 1, "", "transpose_coordinate"]], "gpsea.model.genome.Contig": [[13, 4, 1, "", "genbank_acc"], [13, 4, 1, "", "name"], [13, 4, 1, "", "refseq_name"], [13, 4, 1, "", "ucsc_name"]], "gpsea.model.genome.GenomeBuild": [[13, 3, 1, "", "contig_by_name"], [13, 4, 1, "", "contigs"], [13, 4, 1, "", "genome_build_id"], [13, 4, 1, "", "identifier"]], "gpsea.model.genome.GenomeBuildIdentifier": [[13, 4, 1, "", "identifier"], [13, 4, 1, "", "major_assembly"], [13, 4, 1, "", "patch"]], "gpsea.model.genome.GenomicRegion": [[13, 3, 1, "", "contains"], [13, 4, 1, "", "contig"], [13, 3, 1, "", "distance_to"], [13, 3, 1, "", "end_on_strand"], [13, 3, 1, "", "overlaps_with"], [13, 3, 1, "", "start_on_strand"], [13, 4, 1, "", "strand"], [13, 3, 1, "", "with_strand"]], "gpsea.model.genome.Region": [[13, 3, 1, "", "contains"], [13, 3, 1, "", "contains_pos"], [13, 3, 1, "", "distance_to"], [13, 4, 1, "", "end"], [13, 3, 1, "", "is_empty"], [13, 3, 1, "", "overlaps_with"], [13, 4, 1, "", "start"]], "gpsea.model.genome.Strand": [[13, 2, 1, "", "NEGATIVE"], [13, 2, 1, "", "POSITIVE"], [13, 3, 1, "", "is_negative"], [13, 3, 1, "", "is_positive"], [13, 3, 1, "", "opposite"], [13, 4, 1, "", "symbol"]], "gpsea.model.genome.Stranded": [[13, 4, 1, "", "strand"]], "gpsea.model.genome.Transposable": [[13, 3, 1, "", "to_negative_strand"], [13, 3, 1, "", "to_opposite_strand"], [13, 3, 1, "", "to_positive_strand"], [13, 3, 1, "", "with_strand"]], "gpsea.preprocessing": [[14, 1, 1, "", "Auditor"], [14, 1, 1, "", "CohortCreator"], [14, 1, 1, "", "DataSanityIssue"], [14, 1, 1, "", "DefaultImpreciseSvFunctionalAnnotator"], [14, 1, 1, "", "FunctionalAnnotator"], [14, 1, 1, "", "GeneCoordinateService"], [14, 1, 1, "", "ImpreciseSvFunctionalAnnotator"], [14, 1, 1, "", "Level"], [14, 1, 1, "", "Notepad"], [14, 1, 1, "", "NotepadTree"], [14, 1, 1, "", "PatientCreator"], [14, 1, 1, "", "PhenopacketPatientCreator"], [14, 1, 1, "", "PhenopacketVariantCoordinateFinder"], [14, 1, 1, "", "PhenotypeCreator"], [14, 1, 1, "", "PreprocessingValidationResult"], [14, 1, 1, "", "ProtCachingMetadataService"], [14, 1, 1, "", "ProteinAnnotationCache"], [14, 1, 1, "", "ProteinMetadataService"], [14, 1, 1, "", "TranscriptCoordinateService"], [14, 1, 1, "", "UniprotProteinMetadataService"], [14, 1, 1, "", "VVHgvsVariantCoordinateFinder"], [14, 1, 1, "", "VVMultiCoordinateService"], [14, 1, 1, "", "VarCachingFunctionalAnnotator"], [14, 1, 1, "", "VariantAnnotationCache"], [14, 1, 1, "", "VariantCoordinateFinder"], [14, 1, 1, "", "VepFunctionalAnnotator"], [14, 5, 1, "", "configure_caching_cohort_creator"], [14, 5, 1, "", "configure_caching_patient_creator"], [14, 5, 1, "", "configure_cohort_creator"], [14, 5, 1, "", "configure_default_protein_metadata_service"], [14, 5, 1, "", "configure_patient_creator"], [14, 5, 1, "", "configure_protein_metadata_service"], [14, 5, 1, "", "load_phenopacket_files"], [14, 5, 1, "", "load_phenopacket_folder"], [14, 5, 1, "", "load_phenopackets"]], "gpsea.preprocessing.Auditor": [[14, 3, 1, "", "prepare_notepad"], [14, 3, 1, "", "process"]], "gpsea.preprocessing.CohortCreator": [[14, 3, 1, "", "process"]], "gpsea.preprocessing.DataSanityIssue": [[14, 4, 1, "", "level"], [14, 4, 1, "", "message"], [14, 4, 1, "", "solution"]], "gpsea.preprocessing.DefaultImpreciseSvFunctionalAnnotator": [[14, 3, 1, "", "annotate"]], "gpsea.preprocessing.FunctionalAnnotator": [[14, 3, 1, "", "annotate"]], "gpsea.preprocessing.GeneCoordinateService": [[14, 3, 1, "", "fetch_for_gene"]], "gpsea.preprocessing.ImpreciseSvFunctionalAnnotator": [[14, 3, 1, "", "annotate"]], "gpsea.preprocessing.Level": [[14, 2, 1, "", "ERROR"], [14, 2, 1, "", "WARN"]], "gpsea.preprocessing.Notepad": [[14, 3, 1, "", "add_error"], [14, 3, 1, "", "add_issue"], [14, 3, 1, "", "add_subsection"], [14, 3, 1, "", "add_warning"], [14, 3, 1, "", "error_count"], [14, 3, 1, "", "errors"], [14, 4, 1, "", "issues"], [14, 4, 1, "", "label"], [14, 3, 1, "", "warning_count"], [14, 3, 1, "", "warnings"]], "gpsea.preprocessing.NotepadTree": [[14, 3, 1, "", "add_subsection"], [14, 4, 1, "", "children"], [14, 3, 1, "", "has_errors"], [14, 3, 1, "", "has_errors_or_warnings"], [14, 3, 1, "", "has_warnings"], [14, 3, 1, "", "iterate_nodes"], [14, 4, 1, "", "level"], [14, 3, 1, "", "visit"]], "gpsea.preprocessing.PhenopacketPatientCreator": [[14, 3, 1, "", "process"]], "gpsea.preprocessing.PhenopacketVariantCoordinateFinder": [[14, 3, 1, "", "find_coordinates"]], "gpsea.preprocessing.PhenotypeCreator": [[14, 3, 1, "", "process"]], "gpsea.preprocessing.PreprocessingValidationResult": [[14, 3, 1, "", "is_ok"], [14, 4, 1, "", "policy"], [14, 3, 1, "", "summarize"]], "gpsea.preprocessing.ProtCachingMetadataService": [[14, 3, 1, "", "annotate"]], "gpsea.preprocessing.ProteinAnnotationCache": [[14, 3, 1, "", "get_annotations"], [14, 3, 1, "", "store_annotations"]], "gpsea.preprocessing.ProteinMetadataService": [[14, 3, 1, "", "annotate"]], "gpsea.preprocessing.TranscriptCoordinateService": [[14, 3, 1, "", "fetch"]], "gpsea.preprocessing.UniprotProteinMetadataService": [[14, 3, 1, "", "annotate"], [14, 3, 1, "", "parse_uniprot_json"]], "gpsea.preprocessing.VVHgvsVariantCoordinateFinder": [[14, 3, 1, "", "find_coordinates"]], "gpsea.preprocessing.VVMultiCoordinateService": [[14, 3, 1, "", "fetch"], [14, 3, 1, "", "fetch_for_gene"], [14, 3, 1, "", "get_response"], [14, 3, 1, "", "parse_multiple"], [14, 3, 1, "", "parse_response"]], "gpsea.preprocessing.VarCachingFunctionalAnnotator": [[14, 3, 1, "", "annotate"], [14, 3, 1, "", "with_cache_folder"]], "gpsea.preprocessing.VariantAnnotationCache": [[14, 3, 1, "", "get_annotations"], [14, 3, 1, "", "store_annotations"]], "gpsea.preprocessing.VariantCoordinateFinder": [[14, 3, 1, "", "find_coordinates"]], "gpsea.preprocessing.VepFunctionalAnnotator": [[14, 2, 1, "", "NONCODING_EFFECTS"], [14, 3, 1, "", "annotate"], [14, 3, 1, "", "fetch_response"], [14, 3, 1, "", "format_coordinates_for_vep_query"], [14, 3, 1, "", "process_response"]], "gpsea.util": [[15, 5, 1, "", "open_text_io_handle_for_reading"], [15, 5, 1, "", "open_text_io_handle_for_writing"]], "gpsea.view": [[16, 1, 1, "", "CohortVariantViewer"], [16, 1, 1, "", "CohortViewable"], [16, 1, 1, "", "DiseaseViewable"], [16, 1, 1, "", "Formatter"], [16, 1, 1, "", "GpseaReport"], [16, 1, 1, "", "MtcStatsViewer"], [16, 1, 1, "", "ProteinVariantViewer"], [16, 1, 1, "", "ProteinVisualizable"], [16, 1, 1, "", "ProteinVisualizer"], [16, 1, 1, "", "VariantFormatter"], [16, 1, 1, "", "VariantTranscriptVisualizer"], [16, 5, 1, "", "summarize_hpo_analysis"]], "gpsea.view.CohortVariantViewer": [[16, 3, 1, "", "process"]], "gpsea.view.CohortViewable": [[16, 3, 1, "", "process"]], "gpsea.view.DiseaseViewable": [[16, 3, 1, "", "process"]], "gpsea.view.Formatter": [[16, 3, 1, "", "format_as_string"]], "gpsea.view.GpseaReport": [[16, 3, 1, "", "write"]], "gpsea.view.MtcStatsViewer": [[16, 3, 1, "", "process"]], "gpsea.view.ProteinVariantViewer": [[16, 3, 1, "", "process"]], "gpsea.view.ProteinVisualizable": [[16, 4, 1, "", "marker_counts"], [16, 4, 1, "", "protein_feature_ends"], [16, 4, 1, "", "protein_feature_names"], [16, 4, 1, "", "protein_feature_starts"], [16, 4, 1, "", "protein_feature_types"], [16, 4, 1, "", "protein_id"], [16, 4, 1, "", "protein_length"], [16, 4, 1, "", "protein_metadata"], [16, 4, 1, "", "transcript_coordinates"], [16, 4, 1, "", "transcript_id"], [16, 4, 1, "", "variant_effects"], [16, 4, 1, "", "variant_locations"], [16, 4, 1, "", "variant_locations_counted_absolute"]], "gpsea.view.ProteinVisualizer": [[16, 3, 1, "", "draw_fig"], [16, 3, 1, "", "draw_protein_diagram"]], "gpsea.view.VariantFormatter": [[16, 3, 1, "", "format_as_string"]], "gpsea.view.VariantTranscriptVisualizer": [[16, 3, 1, "", "draw_variants"]]}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "attribute", "Python attribute"], "3": ["py", "method", "Python method"], "4": ["py", "property", "Python property"], "5": ["py", "function", "Python function"], "6": ["py", "data", "Python data"]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:attribute", "3": "py:method", "4": "py:property", "5": "py:function", "6": "py:data"}, "terms": {"": [4, 6, 7, 9, 11, 12, 13, 14, 19, 20, 23, 24, 26, 28, 30, 31, 32, 42], "0": [2, 3, 5, 6, 8, 12, 13, 14, 16, 20, 22, 23, 24, 26, 28, 30, 31, 32, 33, 37, 40, 42], "000": 6, "0000002": 27, "0000047": 33, "0000078": 27, "0000079": 27, "0000080": 27, "0000098": 33, "0000118": [2, 27, 31], "0000119": 27, "0000152": 27, "0000223": 27, "0000234": 27, "0000252": 33, "0000256": 33, "0000271": 27, "0000316": 33, "0000356": 27, "0000359": 27, "0000370": 27, "0000377": 33, "0000407": 24, "0000464": 27, "0000478": 27, "0000486": 23, "0000496": 27, "0000504": 27, "0000517": 37, "0000539": 27, "0000553": 27, "0000562": 20, "0000591": 27, "0000598": 27, "0000707": [27, 31], "0000759": 27, "0000769": 27, "0000777": 27, "0000818": 27, "0000820": 27, "0000828": 27, "0000834": 27, "0000847": 27, "0000864": 27, "0000924": 27, "0000927": 27, "0000929": 27, "0000951": 27, "0000971": 27, "0001060": 12, "0001115": 31, "0001155": 27, "0001166": [3, 7], "0001167": [20, 23], "0001172": [20, 23], "0001191": [20, 26], "0001194": 27, "0001197": 27, "0001199": [20, 23, 26], "0001249": 33, "0001250": 12, "0001256": 33, "0001263": 33, "0001311": 27, "0001367": 23, "0001392": 27, "0001437": 27, "0001438": 27, "0001446": 27, "0001454": 27, "0001507": 27, "0001511": 33, "0001518": 33, "0001537": 6, "0001551": 27, "0001560": 27, "0001566": 12, "0001567": 12, "0001574": [12, 27], "0001575": 12, "0001578": 12, "0001580": 12, "0001583": 12, "0001587": 12, "0001589": 12, "0001595": 27, "0001597": 27, "0001608": 27, "0001619": 12, "0001620": 12, "0001621": 12, "0001623": 12, "0001624": 12, "0001626": [12, 27], "0001627": [12, 24, 27, 33], "0001628": 12, "0001629": [20, 23, 26], "0001630": 12, "0001631": [12, 20, 23, 26], "0001632": 12, "0001643": [20, 23], "0001671": [20, 23], "0001684": [20, 23, 26], "0001713": 23, "0001732": 27, "0001743": 27, "0001760": 27, "0001782": 12, "0001787": [12, 27], "0001792": 12, "0001818": 12, "0001819": 12, "0001821": 12, "0001822": 12, "0001871": 27, "0001872": 27, "0001877": 27, "0001881": 27, "0001889": 12, "0001891": 12, "0001892": [12, 27], "0001893": 12, "0001894": 12, "0001895": 12, "0001906": 12, "0001907": 12, "0001928": 27, "0001939": 27, "0001965": 27, "0001977": 27, "0002012": [12, 27], "0002019": 12, "0002086": 27, "0002169": 12, "0002170": 12, "0002187": 33, "0002270": 27, "0002339": 31, "0002342": 33, "00024192670136836825": 23, "000242": 23, "0002585": 27, "0002597": 27, "0002664": 27, "0002715": 27, "0002733": 27, "0002793": 27, "0002795": 27, "0002813": 27, "0002814": 27, "0002817": 27, "0002818": [20, 23], "0002916": 27, "0002926": 27, "0002973": 27, "0002981": 27, "0002984": [20, 23, 26], "0003011": 27, "0003026": [20, 23], "0003110": 27, "0003111": 27, "0003117": 27, "0003220": 27, "0003254": 27, "0003256": 27, "0003498": 33, "0003549": 27, "0003839": 27, "0003974": [20, 23, 26], "0004298": 27, "0004312": 27, "0004322": 33, "0004323": 27, "0004328": 27, "0004329": 27, "0004332": 27, "0004337": 27, "0004352": 27, "0004353": 27, "0004356": 27, "0004360": 27, "0004361": 27, "0004370": 27, "0004371": 27, "0004378": 27, "0004404": 27, "0004408": 27, "0004930": 27, "0005120": [20, 23], "0005368": 27, "0005561": 27, "0005922": [20, 23, 33], "0005927": [20, 23], "0006265": [20, 23], "0006476": 27, "0006490": 27, "0006496": [20, 23], "0006500": 27, "0006501": [20, 23], "0006503": [20, 23], "0006529": 27, "0006707": 27, "0006889": 33, "0007686": 27, "0008047": 27, "0009115": [20, 23], "0009124": 27, "0009127": 27, "0009601": [20, 23], "0009602": 23, "0009777": [20, 23, 26], "0009778": [20, 23, 26], "0009809": 27, "0009810": 27, "0009815": [20, 23], "0009821": 23, "0009824": [20, 23], "0010438": [20, 23], "0010696": 31, "0010864": 33, "0010876": 27, "0010881": 27, "0010932": 27, "0010935": 27, "0010936": 27, "0010938": 33, "0010948": 27, "0010968": 27, "0010969": 27, "0010974": 27, "0010977": 27, "0010978": 27, "0010987": 27, "0010991": 27, "0011014": 27, "0011017": 27, "0011018": 27, "0011019": 27, "0011024": 27, "0011025": 27, "0011028": 27, "0011032": 27, "0011043": 27, "0011111": 27, "0011121": 27, "0011122": 27, "0011138": 27, "0011277": 27, "0011297": [20, 23], "0011314": [20, 23], "0011342": 33, "0011343": 33, "0011344": 33, "0011409": 27, "0011442": 27, "0011446": 27, "0011620": 27, "0011623": [20, 23], "0011730": 27, "0011732": 27, "0011733": 27, "0011766": 27, "0011767": 27, "0011772": 27, "0011804": 27, "0011805": 27, "0011842": 27, "0011843": 27, "0011844": [20, 23], "0011869": 27, "0011875": 27, "0011884": 27, "0011893": 27, "0011902": 27, "0011994": [20, 23], "0012029": 27, "0012093": 27, "0012099": 27, "0012103": 27, "0012111": 27, "0012129": 27, "0012130": 27, "0012131": 27, "0012135": 27, "0012143": 27, "0012145": 27, "0012210": 24, "0012243": 27, "0012252": 27, "0012261": 27, "0012285": 27, "0012337": 27, "0012338": 27, "0012345": 27, "0012372": [24, 27], "0012373": 27, "0012379": 27, "0012415": 27, "0012443": [8, 24], "0012447": 27, "0012503": 27, "0012535": 27, "0012591": 27, "0012614": 27, "0012632": 27, "0012638": 27, "0012639": 27, "0012640": 27, "0012647": 27, "0012654": 27, "0012680": 27, "0012681": 27, "0012688": 27, "0012700": 27, "0012718": 27, "0012757": 27, "0012769": 27, "0012772": 27, "001910": 20, "0019104918312468707": 20, "0020047": 27, "0020054": 27, "0020058": 27, "0020061": 27, "0020155": 27, "0020169": 27, "0025015": 27, "0025021": 27, "0025031": 27, "0025032": 27, "0025033": 27, "0025065": 27, "0025142": 27, "0025155": 27, "0025276": 27, "0025354": 27, "0025427": 27, "0025429": 27, "0025443": 27, "0025454": 27, "0025456": 27, "0025461": 27, "0025463": 27, "0025546": 27, "0025590": 27, "0025640": 27, "0025668": 27, "0025669": 27, "0025688": 27, "0030085": 27, "0030163": 27, "0030272": 27, "0030338": 27, "0030352": 27, "0030453": 27, "0030680": 27, "0030684": 27, "0030687": 27, "0030800": 27, "0030829": 27, "0030860": 27, "0030872": 27, "0030875": 27, "0030878": 27, "0030956": 27, "0030972": 27, "0031071": 27, "0031072": 27, "0031073": 27, "0031093": 27, "0031094": 27, "0031097": 27, "0031099": 27, "0031101": 27, "0031285": 27, "0031331": 27, "0031340": 27, "0031377": 27, "0031383": 27, "0031389": 27, "0031409": 27, "0031411": 27, "0031416": 27, "0031427": 27, "0031476": 27, "0031508": 27, "0031546": [20, 23, 26], "0031550": 27, "0031602": 27, "0031653": 27, "0031657": 27, "0031685": 27, "0031703": 27, "0031704": 27, "0031818": 27, "0031850": 27, "0031871": 27, "0031884": 27, "0031910": 27, "0031982": 31, "0032120": 27, "0032180": 27, "0032226": 27, "0032243": 27, "0032245": 27, "0032251": 27, "0032314": 27, "0032367": 27, "0032481": 27, "0032485": 27, "0032488": 27, "0032943": 27, "0033012": 27, "0033013": 27, "0033072": 27, "0033127": 27, "0033170": 27, "0033334": 27, "0033335": 27, "0033354": 27, "0033358": 27, "0033796": 27, "0033799": 27, "0034190": 27, "0034200": 27, "0034251": 27, "0034263": 27, "0034370": 27, "0034430": 27, "0034442": 27, "0034482": 27, "0034552": 27, "0034644": 27, "0034684": 27, "0034698": 27, "0034737": 27, "0034858": 27, "0034899": 27, "0034915": 27, "0034916": 27, "0034977": 27, "003713671516019927": 23, "0040064": 27, "0040068": 27, "0040069": 27, "0040070": 27, "0040072": [20, 23], "0040084": 27, "0040085": 27, "0040127": 27, "0040172": 27, "0040203": 27, "0040207": 27, "0040214": 27, "0040224": 27, "0040231": 27, "0045026": 27, "0045027": 27, "0045060": [20, 23], "0045081": 27, "005628510156750059": 20, "00919": 23, "009193214651997994": 23, "01": [20, 23, 24, 26, 30, 31, 33, 37, 42], "0100016": 27, "0100022": 27, "0100491": 27, "0100508": 27, "0100530": 27, "0100536": 27, "0100685": 27, "0100705": 27, "0100763": 27, "0100765": 27, "0100766": 27, "0100767": 27, "0100886": 27, "0100887": 27, "012074957610483744": 24, "01349527665317139": 20, "02560162393963452": 20, "03252032520325203": 23, "0410008": 27, "0410009": 27, "0410014": 27, "0430071": 27, "05": [3, 20, 23, 31], "0500012": 27, "0500015": 27, "0500016": 27, "0500017": 27, "0500018": 27, "0500019": 27, "0500020": 27, "0500114": 27, "0500117": 27, "0500166": 27, "0500183": 27, "0500238": 27, "06": [23, 24], "06544319142266644": 23, "06932119159387057": 23, "07": [20, 23, 24, 26, 30, 31, 33, 37, 42], "07055975880437862": 23, "07440639019586388": 20, "08259109311740892": 23, "08470708701170182": 23, "095684672664751": 20, "1": [2, 3, 5, 6, 8, 12, 13, 14, 20, 22, 23, 24, 26, 27, 28, 30, 31, 32, 33, 37, 40, 42], "10": [2, 6, 8, 12, 14, 16, 20, 22, 23, 24, 26, 31, 33], "100": [2, 20, 23, 31], "1000": 12, "1000029": [6, 12, 42], "1001": [12, 28], "1003": 28, "100del": 20, "100dup": [20, 26], "101": 12, "1024del": 20, "106_107insc": [20, 26], "11": [12, 13, 20, 22, 23, 31], "113": 26, "116": [20, 26], "12": [20, 22, 23, 30], "120": [12, 24, 30], "1221c": [20, 30], "123": 8, "129": 20, "129600": 34, "12_114355723_114355723_g_a": 20, "12_114355755_114355756_tg_t": 20, "12_114355784_114355785_ca_c": 20, "12_114356064_114356065_ta_t": 20, "12_114366207_114366208_gc_g": 20, "12_114366241_114366242_ct_c": 20, "12_114366267_114366267_c_a": 20, "12_114366274_114366274_g_t": 20, "12_114366312_114366312_g_a": 20, "12_114366348_114366349_ct_c": 20, "12_114366360_114366360_c_t": 20, "12_114366366_114366366_t_a": 20, "12_114385474_114385474_a_g": [20, 26], "12_114385475_114385475_c_t": 20, "12_114385521_114385521_c_g": 20, "12_114385521_114385521_c_t": [20, 26], "12_114385522_114385522_g_a": 20, "12_114385550_114385550_a_aattattctcag": 20, "12_114385553_114385553_c_a": 20, "12_114385563_114385563_g_a": [20, 26], "12_114394743_114394746_tgtg_t": 20, "12_114394762_114394763_ca_c": 20, "12_114394817_114394817_g_c": 20, "12_114394820_114394820_c_g": 20, "12_114398568_114398568_c_a": 20, "12_114398578_114398579_ca_c": 20, "12_114398602_114398602_t_g": 20, "12_114398626_114398627_cg_c": 20, "12_114398632_114398632_g_a": 20, "12_114398656_114398656_c_cg": [20, 26], "12_114398666_114398667_tg_t": 20, "12_114398675_114398675_g_t": [20, 26], "12_114398682_114398682_c_cg": [20, 26], "12_114398708_114398709_gc_g": 20, "12_114399514_114399514_a_c": [20, 26], "12_114399559_114399559_t_c": 20, "12_114399594_114399594_a_c": 20, "12_114399613_114399613_t_a": 20, "12_114399622_114399622_g_t": 20, "12_114399625_114399629_acatc_a": 20, "12_114399633_114399633_c_g": 20, "12_114401827_114401827_t_a": 20, "12_114401830_114401830_c_t": [20, 26], "12_114401846_114401846_c_g": 20, "12_114401853_114401853_g_t": 20, "12_114401873_114401874_ta_t": 20, "12_114401907_114401907_a_g": 20, "12_114401921_114401921_c_g": 20, "12_114403754_114403754_g_t": 20, "12_114403792_114403792_c_cg": [20, 26], "12_114403798_114403798_g_gc": [20, 26], "12_114403798_114403799_gc_g": 20, "12_114403859_114403859_g_t": 20, "13": [2, 3, 20, 22, 23, 26], "1304del": 20, "1333del": 20, "1366c": 20, "14": [20, 22, 23, 26], "1424522583078588": 20, "142900": [20, 26, 42], "1435th": 42, "145c": 20, "148": 20, "15": [13, 20, 23, 26], "15294646873594242": 20, "154700": 34, "1554": 30, "156": [20, 23, 26, 30, 37], "161t": 20, "167": 23, "1687456462342971": 20, "17": [20, 23], "18": [20, 23, 30, 37, 42], "1824c": 12, "18869165": 30, "18869682": 30, "18921399": 30, "19": [20, 23, 24, 26, 30, 42], "194del": 20, "1_8358231_8358231_t_c": 42, "1d": 16, "1g": 20, "1st": [6, 12, 30], "2": [2, 3, 4, 6, 7, 8, 9, 12, 13, 14, 20, 22, 23, 24, 26, 27, 28, 30, 31, 32, 33, 40, 42], "20": [2, 6, 12, 20, 22, 23, 24, 26, 31, 42], "2011": 33, "2015": 18, "2018": [24, 30], "2019": 22, "2022": [25, 33], "2024": 24, "205": 12, "206": 12, "21": 23, "215c": 20, "2176138034868934": 20, "22": [2, 20, 23, 26], "222": 23, "222g": 20, "223": 12, "224": 12, "226": 20, "22_10001_20000_inv": [6, 12], "23": [20, 23, 27], "238g": [20, 26], "24": [20, 23], "241a": 20, "243": 20, "246_249del": 20, "25": [2, 20, 23], "250": [26, 30], "253a": 14, "253c": 20, "26": 23, "260": [20, 23, 37], "262a": 20, "263": 30, "27": [20, 23, 26], "27bp": 12, "28": [20, 23], "281t": 20, "2986": [8, 12, 22], "299": 30, "2d": 2, "2nd": 6, "2t": [20, 26], "2x2": 4, "2x3": 4, "3": [2, 4, 8, 12, 14, 20, 23, 24, 27, 28, 31, 40], "30": [14, 20, 23, 26], "3000013": 27, "3000036": 27, "3000050": 27, "303": 22, "30552424": 20, "31": [20, 23], "316a": 20, "32": [20, 23, 26], "320": 30, "321681": 12, "321682": 12, "321887": 12, "33": [20, 23], "34": [20, 23], "346": 30, "348081479150901e": 24, "348081479150902e": 24, "35": [20, 23], "356": [26, 30], "36": [20, 23, 26, 31], "3603": [12, 14], "361t": [20, 26], "36303223": 25, "37": [20, 23], "374del": 20, "38": [20, 23, 26], "38336121": 20, "39": 20, "4": [6, 8, 12, 14, 20, 23, 24, 26, 30], "40": [20, 23, 26], "4000056": 27, "4000183": 27, "400dup": [20, 26], "408c": [20, 26], "40c": 20, "41": [20, 23, 26], "4119241192411924": 23, "416del": 20, "42": [16, 20, 23], "426dup": [20, 26], "43": [20, 23], "4304a": 42, "432292015291845e": 23, "44": [20, 23], "45": [20, 23], "450": 31, "451c": 20, "456del": 20, "45983847234923847": 23, "46": [26, 30], "48": 20, "481a": 20, "48700997145537483": 23, "5": [3, 6, 12, 14, 20, 23, 24, 26, 30, 32, 33, 36, 40, 42], "50": [6, 12, 20, 23, 26, 42], "504del": 20, "5059634533318744": 20, "50bp": 6, "51": 20, "510": 20, "518": [26, 30], "52": 20, "53": [20, 23], "55": [20, 23], "56": 20, "578": 12, "58": 23, "584g": 20, "587c": 20, "59": [20, 23], "5g": 20, "5th": [12, 20], "5x": 24, "6": [12, 20, 23, 24, 25, 26, 28, 31, 42], "60": 20, "6000062": 27, "6000231": 27, "6000489": 27, "6000673": 27, "614": 22, "6190936213143254e": 20, "62": 20, "63": 23, "630": 22, "64": [20, 23, 31], "641del": 20, "65": 23, "658_660del": 20, "664": 22, "668c": [20, 26], "67": 23, "678g": 20, "680_681insctgagaataat": 20, "69": [22, 23], "7": [3, 6, 12, 20, 23, 26, 31], "709c": 20, "71": [20, 23], "710g": [20, 26], "72": 20, "741216622359659": 22, "75": 23, "755": [20, 26], "7703831604944444": 23, "7735491022101784": 20, "78": 23, "781a": 20, "787g": 20, "798del": 20, "8": [8, 12, 14, 15, 20, 22, 23, 26, 30, 31], "80": [23, 30], "8072294637445331": 20, "8196217102808717": 20, "833333333333336": 23, "835c": 20, "84": 23, "85": [20, 26], "873c": 20, "880g": 20, "89": 23, "9": [20, 22, 23, 30], "90": 31, "905del": 20, "93": 23, "939del": 20, "95": [20, 31], "97": 23, "9876g": 12, "99": [23, 31], "A": [2, 3, 5, 6, 7, 8, 12, 14, 16, 18, 20, 26, 28, 30, 31, 32, 33, 35, 38], "AND": [6, 24, 42], "AT": 28, "As": [5, 7, 24, 26, 38, 42], "But": 26, "By": [2, 16, 20, 31, 37, 40], "For": [2, 4, 8, 11, 12, 13, 14, 20, 22, 23, 24, 25, 26, 28, 30, 31, 32, 37, 40, 42], "IN": 14, "INS": 12, "If": [2, 3, 6, 10, 12, 14, 15, 20, 31, 33], "In": [2, 4, 6, 8, 14, 18, 20, 23, 24, 25, 26, 29, 30, 31, 33, 37, 38], "It": [3, 4, 20, 26, 31, 34, 39], "NO": [5, 7], "NOT": [2, 12, 13, 22, 40, 42], "No": [3, 5, 13, 20, 23, 26, 30, 37, 40], "Not": 12, "OR": [6, 22, 24, 32, 42], "On": [3, 7, 30], "One": [33, 42], "That": 19, "The": [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 16, 18, 19, 22, 23, 24, 26, 27, 28, 29, 30, 31, 32, 34, 35, 36, 37, 38, 39, 40, 42], "Then": [12, 19, 20, 30], "There": [12, 20, 21, 40], "These": [12, 40], "To": [14, 19, 20, 23, 24, 30, 31, 40, 42], "With": 32, "_": [24, 30, 37, 42], "__eq__": 6, "__hash__": 6, "__init__": [12, 31], "__repr__": 6, "__str__": 6, "_api": 2, "_io": 14, "_missing_implies_phenotype_exclud": 7, "_t": 12, "_term_id": 2, "a_": 32, "a_pred": [6, 20, 22, 24, 32], "aa": 6, "aat": 28, "ab": 6, "abbrevi": [16, 31], "abdomen": 27, "abdomin": 27, "abil": 26, "abnorm": [8, 18, 20, 23, 24, 26, 27, 28, 37], "about": [2, 12, 14, 16, 19, 20, 26, 29, 30, 31, 33, 35, 37], "abov": [6, 19, 20, 23, 27, 30, 31, 32, 33, 40], "absenc": [3, 7, 20, 26, 37], "absens": 26, "absent": [20, 23, 26], "absolut": 6, "abstract": [2, 3, 4, 5, 6, 7, 8, 9, 12, 13, 14, 16], "ac": 32, "accept": [12, 14], "acceptor": 20, "access": [6, 12, 13, 14, 19, 20, 24, 30, 31, 37], "accommod": 27, "accord": [2, 3, 20, 23, 26, 27], "account": 31, "acetylaspart": 27, "achiev": 32, "acid": [12, 26, 27, 30], "across": [2, 16, 29, 39], "actg": 28, "activ": [26, 27], "actual": [26, 30], "ad": [14, 33], "adapt": [8, 27], "add": [14, 19, 35], "add_error": 14, "add_issu": 14, "add_subsect": 14, "add_warn": 14, "addit": [4, 19, 20], "addition": 20, "address": 3, "adenosin": 27, "adher": 40, "adiponectin": 27, "adipos": 27, "adjac": 13, "adjust": 33, "adnexa": 27, "adren": 27, "adrenocorticotropin": 27, "advanc": 2, "advantag": [20, 23, 31, 37], "affect": [6, 12, 20, 25, 28, 30, 42], "affected_exon": 12, "after": [3, 14, 19], "ag": [12, 26, 33], "against": 25, "al": [8, 18, 22, 24, 25, 30, 33], "ala143argfster40": [20, 26], "ala34glyfster27": [20, 26], "ala34profster32": [20, 26], "albeit": 14, "albumin": 27, "aldosteron": 27, "align": 12, "alik": 40, "all": [2, 3, 5, 6, 7, 8, 12, 14, 19, 23, 27, 28, 29, 30, 32, 35, 36, 38, 41, 42], "all_count": 3, "all_diseas": 12, "all_pati": 12, "all_phenotyp": 12, "all_transcript_id": 12, "all_vari": 12, "allel": [6, 12, 18, 22, 23, 24, 25, 29, 35, 36, 40, 42], "allelecount": [5, 6], "allianc": 18, "allow": [3, 26, 29, 30, 31, 36, 38], "allow_nan": 11, "alon": 31, "along": [11, 30, 35, 42], "alpha": [23, 27, 31], "alreadi": 30, "also": [12, 27, 30, 34, 35, 37], "alt": [12, 28, 40], "altern": [6, 12, 24, 26, 28, 31, 40], "although": 26, "alwai": [6, 13, 30], "amaz": 30, "amino": [12, 26, 27], "aminoacid": [12, 13, 16, 30], "amniot": 27, "among": [2, 12], "amount": [8, 30, 42], "amyloid": 27, "an": [2, 3, 4, 5, 6, 7, 8, 9, 12, 13, 14, 15, 16, 18, 19, 20, 21, 22, 23, 24, 25, 26, 29, 30, 31, 32, 34, 35, 37, 38, 40, 42], "analys": [29, 30, 38], "analysi": [0, 14, 16, 18, 21, 25, 29, 30, 32, 33, 34, 36, 37, 38, 39, 40, 42], "analyz": [0, 8, 16, 20, 22, 24, 26, 32, 38, 40], "anatomi": [20, 30], "ancestor": [7, 20, 28, 37], "andolfo": 30, "aneurysm": 28, "angiotensin": 27, "ani": [2, 5, 6, 8, 11, 12, 14, 23, 24, 26, 30, 31, 32, 33, 34, 40, 42], "ank": [6, 12], "ankyrin": 12, "annot": [2, 6, 7, 12, 14, 16, 20, 23, 26, 28, 30, 31, 37, 42], "annotation_frequency_threshold": 2, "anomali": [24, 33], "anoth": [7, 13, 20, 30], "answer": [5, 6, 7], "anterior": [27, 37], "antimullerian": 27, "anu": 27, "api": [12, 14, 18, 23, 26, 38, 42], "aplasia": [20, 23], "apoptosi": 27, "appear": 27, "appendicular": [20, 23], "appendix": 40, "appli": [3, 9, 28, 31], "applic": 20, "apply_predicates_on_pati": [1, 3], "approach": [29, 31], "appropri": [20, 21, 26], "ar": [2, 3, 6, 8, 9, 12, 13, 14, 16, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 30, 32, 33, 36, 37, 38, 40, 42], "arachnodactyli": 7, "arbitrari": [11, 13], "areolar": 27, "arg": 11, "arg134profster49": [20, 26], "arg237gln": [20, 26], "arg237pro": 20, "arg237trp": 20, "arg279ter": [20, 26], "arg81trp": 20, "arginin": 42, "argument": [8, 10, 14, 16, 31, 32, 37], "arithmet": 13, "arm": 27, "aromatas": 27, "around": [4, 9, 12], "arrai": [3, 4, 16], "arrang": 27, "arrhythmia": 20, "arriv": 31, "arteriosu": [20, 23], "ask": [2, 18], "aspect": [16, 26], "assembl": 14, "assembli": 13, "assert": [12, 13], "assess": [3, 12, 20, 24, 33], "assign": [3, 5, 6, 7, 8, 12, 23, 24, 32, 33, 34, 35, 36, 38, 40, 42], "assist": 29, "associ": [3, 8, 12, 14, 18, 20, 22, 25, 26, 30, 38], "assum": [12, 13, 14, 22, 31], "assume_karyotypic_sex": 14, "assumpt": [31, 37], "atrial": [20, 23, 26, 27], "atrium": [20, 23], "atrophi": 24, "atrophin": 24, "attach": 12, "attribut": [3, 6, 7, 12, 18], "audit": 14, "auditor": [0, 14], "automat": [24, 34], "autonom": 27, "autosom": [6, 20, 23, 32], "autosomal_domin": [5, 6, 23, 40], "autosomal_recess": [5, 6, 40], "avail": [2, 12, 14, 18, 20, 24, 26, 30, 38, 42], "avoid": [16, 27], "ax": [8, 16, 20, 24, 26, 30], "axi": [6, 24, 27, 35], "b": [6, 12, 13, 31, 32, 40], "b_": 32, "b_predic": [6, 20, 22, 24, 32], "back": [10, 14, 30], "background": [18, 29], "backup": 12, "bandwidth": 30, "base": [2, 3, 4, 5, 6, 7, 8, 9, 11, 12, 13, 14, 15, 16, 20, 23, 24, 27, 28, 30, 32, 33, 34, 38, 39, 40], "basemultiphenotypeanalysisresult": 3, "basi": 26, "basic": [12, 13, 30, 40, 42], "batteri": 7, "bb": 6, "becaus": [2, 12, 20, 23, 25, 26, 27, 28, 31, 33, 39], "been": [12, 20, 23, 30, 31], "befor": [31, 32, 42], "behavior": 18, "behind": [20, 37], "being": [2, 13, 14, 16, 20, 26, 31, 33, 38, 42], "belong": [2, 5, 12], "below": [6, 28], "benjamini": [20, 23, 31], "besid": 30, "best": 18, "beta": [27, 30], "better": 42, "between": [3, 8, 12, 13, 16, 18, 20, 22, 23, 24, 26, 28, 31, 33, 36, 39], "beyond": 6, "bia": [12, 30], "biallel": 6, "biallelic_pred": [5, 6, 22, 32], "big": 22, "bin": [3, 5, 6, 20, 24, 31, 34], "bind": 12, "biolog": [12, 13, 24, 30], "biologi": 18, "biomark": 27, "biomed": [20, 30], "biopterin": 27, "birth": [12, 27], "bit": 37, "bleed": 27, "blood": 27, "bodi": 27, "boiler": 30, "boilerpl": 30, "bone": [20, 23, 27], "bonferroni": [3, 20, 31], "bool": [2, 6, 7, 12, 13, 14, 16], "boolean": 12, "borderlin": 33, "both": [4, 12, 14, 20, 32, 33, 34, 38, 40], "bound": 13, "boundari": [12, 13, 14], "box": [8, 24, 26, 30], "bp": [6, 12, 28, 42], "brain": [8, 24], "branch": [19, 31], "break": 12, "breakend": 12, "breakpoint": [6, 12, 14], "breast": 27, "breath": 27, "brief": 18, "briefli": 42, "bring": [20, 27], "bronchu": 27, "buccal": 27, "bug": 18, "build": [6, 12, 13, 14, 20, 23, 36], "built": 30, "bunch": 42, "burden": [2, 3, 20, 31], "c": [5, 12, 14, 18, 20, 22, 24, 26, 28, 30, 42], "c17998": 12, "c46112": 12, "c46113": 12, "cach": [10, 14, 30], "cache_dir": 14, "cache_env": [0, 10], "calcium": 27, "calcul": [4, 8, 9, 12, 13, 23, 26], "calf": 27, "call": [11, 12, 14, 27, 30, 31], "callabl": [8, 14], "caller": 14, "can": [2, 3, 4, 5, 6, 7, 8, 12, 13, 14, 16, 19, 20, 23, 24, 26, 27, 30, 31, 32, 33, 35, 36, 37, 40, 42], "cannot": [5, 8, 12, 14, 24, 30, 33, 38], "canon": 12, "cap": 31, "capabl": 38, "carboxyl": 27, "cardiac": [20, 23, 26, 27], "cardiomyocyt": 27, "cardiovascular": 27, "carpal": [20, 26], "carri": [24, 25, 30, 40], "cart": 27, "cascad": 27, "case": [2, 6, 8, 12, 14, 20, 23, 24, 26, 30, 32, 37], "cat_id": [5, 8, 22, 24], "catalyt": 26, "cataract": 31, "catecholamin": 27, "categor": [1, 2, 5, 6, 7, 26, 32, 35, 40], "categori": [3, 5, 7, 8, 12, 20, 22, 23, 24, 26, 30, 33], "catheter": 27, "caudat": 31, "caus": [18, 37], "caviti": 27, "ccc": 28, "cct": 12, "cd": [12, 19], "cdna": [12, 20], "cds_end": 12, "cds_start": 12, "cell": [26, 27], "cellular": 27, "central": 27, "cerebellar": 24, "certain": [6, 12, 14, 26, 35, 42], "certainli": 30, "cg": 28, "chanc": [20, 23, 31], "chang": [6, 12, 24, 29, 30, 42], "change_length": [6, 12, 24, 42], "charact": 12, "character": [12, 20], "characterist": 12, "check": [3, 4, 6, 12, 13, 14, 20, 31, 42], "check_circular": 11, "checker": 30, "checkout": 19, "child": [2, 20, 23], "children": [14, 27, 31], "choic": [12, 29], "choos": [6, 14, 20, 23, 26, 29, 32, 40], "chosen": [12, 20, 23, 30, 31], "chr1": [12, 13, 28], "chr5": 12, "chrom": [12, 28], "chromosom": [6, 12, 25, 27, 42], "chromosomal_delet": 42, "chrx": 13, "cilium": 27, "circul": 27, "cl": [11, 22, 23, 24, 26, 30], "class": [2, 3, 4, 5, 6, 7, 8, 9, 11, 12, 13, 14, 16, 20, 26, 30, 31, 42], "clearanc": 27, "clearli": 42, "click": [12, 30], "clinic": [12, 20, 22, 30, 31, 33, 37, 40, 42], "clinvar": [20, 30], "clone": 19, "cluster": 35, "cm000663": 13, "coagul": 27, "code": [2, 8, 12, 14, 19, 20, 23, 24, 26, 30], "coding_sequence_vari": 12, "codon": [12, 18, 30, 42], "cohort": [0, 2, 3, 7, 8, 12, 14, 16, 18, 29, 31, 38, 40, 41, 42], "cohort_cr": [14, 20, 30, 37, 42], "cohort_nam": 20, "cohort_s": 2, "cohortcr": [0, 14, 30], "cohortvariantview": [0, 16, 20], "cohortview": [0, 16, 20, 26], "coiled_coil": 12, "collect": [2, 6, 7, 9, 12, 14, 18, 20, 30, 37], "color": 8, "column": [2, 3, 8, 12], "com": 19, "combin": [12, 20, 23, 32, 42], "come": [13, 24, 26], "command": 19, "common": [5, 20, 25], "commonli": [13, 31], "compact": [6, 12], "compar": [4, 9, 12, 21, 25, 29, 36, 39], "compare_genotype_vs_phenotyp": [3, 20, 23], "compare_genotype_vs_phenotype_scor": [8, 22, 24], "comparison": [25, 32], "complain": 14, "complet": [14, 30, 31], "complex": 31, "compon": [20, 23, 24, 42], "composit": [12, 27, 30], "compositional_bia": 12, "compound": [6, 25, 27, 40, 42], "comprehens": [18, 37], "comput": [3, 6, 8, 9, 14, 20, 23, 24, 32], "compute_pv": [4, 9], "concentr": 27, "concept": [18, 28, 31], "conceptu": 12, "concern": [18, 24], "concis": 12, "conclus": 31, "condens": 27, "condit": [24, 31], "conduct": [20, 23, 26], "config": 0, "configur": [6, 14, 42], "configure_caching_cohort_cr": [0, 14, 20, 30, 37, 42], "configure_caching_patient_cr": [0, 14], "configure_cohort_cr": [0, 14], "configure_default_protein_metadata_servic": [0, 14, 26, 30], "configure_ontology_stor": [20, 23, 24, 26, 30, 31, 33, 37, 42], "configure_patient_cr": [0, 14], "configure_phenopacket_registri": [20, 30, 37, 42], "configure_protein_metadata_servic": [0, 14, 20], "congenit": [20, 23, 24], "connect": 27, "consequ": 12, "conserv": 12, "consid": [4, 6, 12, 13, 14, 23, 24, 27, 33, 40], "consider": [27, 42], "consist": [2, 3, 12, 13, 30], "consortium": 24, "constitut": 27, "construct": [14, 24, 37], "constructor": [2, 12, 31], "consult": [6, 10, 26, 40], "contain": [2, 8, 12, 13, 14, 16, 20, 27, 30], "contains_po": 13, "contemporari": 26, "content": 30, "context": [2, 7, 18, 24, 31, 32], "contig": [12, 13, 30], "contig_by_nam": [12, 13, 28], "contigu": 13, "conting": [2, 3, 4, 23], "contingency_t": 23, "contribut": 26, "control": [20, 23, 31], "conveni": [7, 14, 30], "convers": 28, "convert": [14, 42], "coordin": [6, 12, 13, 14, 16, 26, 28], "copi": 25, "cord": 27, "corneal": 27, "corpuscular": 27, "correct": [3, 20, 27, 29], "corrected_pv": 3, "correctli": 14, "correl": [0, 2, 3, 18, 20, 21, 26, 27, 29, 30, 31, 42], "correspond": [2, 3, 7, 8, 12, 13, 14, 20, 23, 24, 26, 28, 29, 31, 32, 33, 42], "cortisol": 27, "could": [3, 11, 23], "count": [2, 3, 4, 6, 7, 8, 12, 14, 16, 20, 23, 24, 26, 27, 29, 30, 33, 35, 36, 40], "count_statist": [3, 20, 23, 31], "counter": 12, "countingphenotypescor": [1, 8, 24], "counts_fram": 2, "countstatist": [3, 4], "coveni": 12, "cover": [5, 38], "cpd_viewer": 26, "cranial": 27, "creat": [2, 5, 6, 7, 8, 10, 12, 13, 14, 16, 18, 20, 22, 23, 24, 26, 29, 31, 32, 33, 37, 40, 42], "create_variant_from_scratch": 12, "creator": [14, 42], "criteria": [6, 35, 42], "criterion": 6, "cry": 27, "csf": 27, "ct": 12, "culprit": 2, "curat": 20, "curi": [6, 8, 12, 14], "current": [10, 12, 13, 14, 16, 19, 30, 39], "custom": [30, 32], "cut": 37, "cycl": 27, "cyp21a2": 22, "cytologi": 27, "cytometri": 27, "danger": 26, "darksalmon": 8, "data": [2, 3, 4, 12, 13, 14, 16, 18, 20, 22, 23, 24, 26, 29, 31, 35, 37, 42], "databas": 14, "datadir": 14, "datafram": [2, 3, 4, 8, 12, 16, 30], "datasanityissu": [0, 14], "dataset": [20, 23, 26], "de": [8, 29, 41], "decid": [2, 12, 26, 27], "decilit": 12, "decis": 2, "decod": [11, 15, 30], "deem": 27, "deep": 18, "def": [8, 11], "default": [2, 6, 10, 11, 14, 15, 16, 20, 23, 30, 31, 32, 37, 40], "default_cache_path": [0, 10], "default_filt": [2, 20, 23, 31], "defaultimprecisesvfunctionalannot": [0, 14], "defaulttermid": 2, "defect": [20, 23, 24, 26, 28], "defin": [3, 5, 6, 12, 28], "definit": 12, "degre": 33, "dehydr": 30, "del": [6, 12, 28, 42], "delet": [6, 12, 20, 28, 42], "deliveri": 27, "demand": 42, "demonstr": [20, 42], "denot": [12, 18], "depend": [20, 21], "deposit": [23, 24], "depth": 14, "deriv": 31, "descend": [6, 7, 8, 20, 24, 28, 33, 37], "describ": [8, 12, 14, 19, 22, 23, 24, 26, 31, 37, 42], "descript": [5, 12, 14, 33], "design": [18, 26, 30, 31], "desir": [6, 14, 25, 27, 38], "despit": 20, "detail": [5, 12, 20, 31, 42], "detect": 26, "determin": [3, 6, 12, 14, 24, 31, 32, 33], "develop": [19, 27, 31, 33], "deviat": [18, 23], "devri": 26, "devriesphenotypescor": [1, 8, 24, 33], "df": [12, 30], "diagnos": [6, 7, 12, 34, 35], "diagnosi": [6, 7, 29, 35], "diagnosis_pred": [5, 6, 34], "diagram": [16, 20, 26, 30], "dialog": 30, "dichotom": 23, "dict": [11, 14], "dictat": 32, "differ": [8, 12, 13, 20, 21, 22, 23, 24, 28, 29, 30, 31, 32, 33, 36, 38, 39], "differenti": 18, "digest": 27, "digit": [20, 23], "dimension": 12, "dingeman": 33, "diploid": 12, "direct": [7, 12, 20, 31], "directli": 7, "directori": [10, 14, 30], "disabl": [24, 31, 33], "discharg": 27, "discoveri": [20, 23, 31], "discret": [3, 4, 5, 14, 38], "diseas": [0, 6, 7, 12, 18, 20, 26, 32, 34, 35, 37, 40], "disease_id_queri": 7, "diseaseanalysi": [1, 3], "diseasepresencepred": [5, 7], "diseaseview": [0, 16], "disord": [20, 26, 30], "displai": [2, 16, 20, 26, 30], "display_quest": [5, 20, 22, 23, 24, 32, 34, 36, 37, 39, 40], "disproportion": 33, "dissect": 23, "distanc": 13, "distance_to": 13, "distinct": 26, "distribut": [8, 12, 16, 24, 29, 39], "divid": [23, 35], "dl": 12, "dna": [13, 27], "do": [2, 6, 8, 13, 14, 16, 19, 20, 30, 31, 36, 37, 40, 42], "doc": [22, 23, 24, 26, 30], "document": [18, 19, 21, 26, 30, 31], "doe": [2, 5, 10, 12, 14, 28], "domain": [2, 6, 12, 24, 27, 29, 30, 31, 32, 40, 42], "domin": [6, 20, 23, 25, 32], "done": [13, 14, 19, 35, 42], "donor": [12, 20], "dopamin": 27, "doubl": [8, 13], "down": [14, 37], "download": [12, 30], "downloaded_uniprot_json": 30, "downstream": [2, 13, 14, 30, 38], "downstream_gene_vari": [12, 14], "dpi": [24, 30], "draw": [8, 16, 30], "draw_fig": [16, 30], "draw_protein_diagram": [16, 20, 26], "draw_vari": 16, "drawn": 16, "driven": 27, "drug": 27, "ductu": [20, 23], "due": [12, 14], "dump": 11, "dup": 12, "duplic": [12, 14, 28], "dure": [12, 14], "e": [2, 3, 6, 7, 8, 12, 13, 14, 18, 20, 24, 26, 27, 28, 30, 31, 32, 34, 37], "each": [2, 3, 7, 8, 12, 14, 16, 18, 20, 23, 24, 25, 26, 29, 31, 33, 38], "ear": [27, 33], "easi": [19, 39], "easiest": 30, "easili": 16, "ectopia": [34, 37], "eff": [24, 42], "effect": [6, 12, 14, 20, 24, 26, 30, 32, 40, 42], "either": [2, 6, 7, 8, 12, 13, 14, 16, 30, 32, 33], "elabor": 42, "electrolyt": 27, "electrophysiologi": 27, "element": [12, 13, 14, 16], "els": 11, "elsewher": 26, "embryon": 27, "empti": [12, 13, 14, 32, 40], "enabl": 19, "encod": [7, 11, 12, 14, 15, 20, 22, 23, 24, 30, 42], "encount": [14, 20, 26], "end": [6, 12, 13, 20, 28, 30], "end_on_strand": 13, "endocrin": 27, "energi": 27, "enrich": 31, "ensembl": 12, "enst00000123456": 12, "ensur": 15, "ensure_ascii": 11, "enter": 26, "entir": [14, 30], "entiti": [7, 12, 13, 14], "entri": [14, 30, 37], "enum": [12, 13, 14], "enumer": 16, "environ": [10, 16, 19, 26], "enzym": 27, "epiphysi": 27, "equal": [6, 12, 24, 31], "equival": 8, "error": [14, 20, 26, 30, 31], "error_count": 14, "erythrocyt": 27, "erythroid": 27, "erythropoietin": 27, "et": [8, 18, 22, 24, 25, 30, 33], "etc": [12, 14], "evalu": [6, 18, 23, 24], "even": [3, 27, 30, 31], "ever": 2, "exact": [4, 6, 8, 12, 14, 18, 20, 21, 24, 25, 26, 29, 31], "exactli": [6, 8, 31], "exampl": [3, 4, 6, 7, 8, 9, 11, 12, 14, 16, 20, 21, 25, 28, 29, 31, 38, 42], "except": [11, 14, 40, 42], "exclud": [2, 5, 7, 12, 13, 20, 23, 36, 37], "excluded_diseas": 12, "excluded_member_count": 12, "excluded_phenotyp": 12, "exclus": [5, 7, 20, 28, 38], "execut": [22, 23, 24], "exemplifi": 26, "exercis": [27, 30], "exhaust": 5, "exibit": 7, "exist": [10, 14, 26, 31], "exon": [6, 12, 30, 42], "exon20": 42, "exon_numb": 6, "exons_effect": 12, "expect": [2, 12, 14, 23, 24, 27, 30, 31, 35], "expenditur": 27, "experiment": 26, "explan": [2, 21, 41], "explicit": [2, 7], "explicitli": [12, 37], "explor": [23, 24, 29, 30, 37], "exploratori": 29, "expos": 23, "express": [14, 23, 27], "extend": 24, "extern": 33, "extra": 32, "extract": [12, 14], "extraocular": 27, "extrem": [20, 23, 31], "ey": [24, 27, 37], "f": [8, 18], "face": 27, "fact": [31, 42], "factor": [26, 27, 30, 31, 35], "factori": 2, "fail": [2, 12], "fall": [8, 10], "fallback": [10, 14], "fals": [2, 3, 6, 7, 11, 12, 13, 14, 20, 23, 26, 31], "famili": [12, 20, 31, 34], "far": 24, "farthest": 24, "fascia": 27, "fashion": 14, "fbn1": [6, 12], "fdr": [23, 31], "fdr_bh": [3, 20, 23, 31], "fdr_by": 31, "fdr_gb": 31, "fdr_tsbh": 31, "fdr_tsbky": 31, "featur": [6, 7, 12, 14, 18, 19, 23, 24, 26, 37, 42], "feature_elong": 12, "feature_id": 6, "feature_trunc": 12, "feature_typ": [6, 12], "featureinfo": [0, 12], "featuretyp": [0, 6, 12], "fecal": 27, "feel": 38, "feenstra": [8, 33], "femal": [12, 39], "fet": [21, 26, 29], "fetal": 27, "fetch": [14, 20, 26, 29], "fetch_for_gen": 14, "fetch_respons": 14, "few": [28, 30, 42], "fh": [15, 16, 22, 23, 24, 26], "fiber": 27, "fibrillin": 12, "fibrinolysi": 27, "fibroblast": 27, "fiction": [6, 32, 40], "field": [12, 25], "fig": [20, 24, 26, 30], "figsiz": [20, 24, 26, 30], "figur": [14, 16, 30], "file": [12, 14, 15, 16, 22, 23, 24, 26, 30], "filter": [2, 3, 12, 20, 23, 27, 28, 29], "filter_method_nam": 2, "final": [20, 37], "find": [3, 13, 14, 20, 23, 30, 31, 35], "find_coordin": 14, "finder": 14, "finger": [20, 23], "fingertip": 30, "finish": 22, "first": [6, 10, 12, 14, 20, 23, 24, 26, 30, 31, 32, 35, 42], "fisher": [4, 18, 20, 21, 25, 26, 29, 31, 41], "fisher_exact": [4, 23], "fisherexacttest": [3, 4, 20, 23, 31], "fit": 12, "five": [24, 40], "five_prime_utr_vari": [12, 14], "fix": 30, "flexibl": [23, 42], "flip": 13, "float": [2, 3, 4, 8, 9, 12, 14, 23, 24], "flow": 27, "fluid": 27, "focal": 28, "focu": [18, 20], "fold": 12, "folder": [10, 14, 30], "follow": [3, 8, 12, 19, 20, 22, 23, 24, 26, 27, 28, 30, 31, 32, 33, 38, 40], "foot": 27, "for_sampl": 12, "forearm": [20, 23, 27], "form": [4, 27, 30, 32], "formal": 26, "format": [7, 12, 13, 14, 16, 26, 30], "format_as_str": 16, "format_coordinates_for_vep_queri": 14, "formatt": [0, 16], "formul": [26, 30], "forward": 13, "found": [12, 14, 20, 24, 26, 30, 31, 33, 36], "four": [26, 30], "fpath_cache_dir": 14, "fpath_cohort": 26, "fpath_cohort_json": [22, 23, 24], "fragment": 12, "frame": [2, 3, 4, 12, 16, 22, 24, 30, 42], "frameshift": [36, 40, 42], "frameshift_vari": [12, 20, 23, 24, 26, 32, 36, 42], "framework": 24, "free": 38, "frequenc": [2, 16, 20, 23], "frequent": [20, 26, 28], "friendli": [2, 14], "from": [2, 5, 6, 8, 11, 12, 13, 14, 15, 18, 19, 20, 22, 23, 24, 26, 27, 28, 29, 31, 32, 33, 34, 36, 37, 38, 39, 40, 42], "from_curi": 37, "from_feature_fram": [12, 30], "from_map": 12, "from_measurement_id": [8, 22], "from_pati": 12, "from_query_curi": [8, 24], "from_raw_part": [5, 12], "from_str": 12, "from_term": 12, "from_uniprot_json": [12, 30], "from_vcf_liter": [12, 28], "from_vcf_symbol": [12, 28], "fuction": 14, "func": 8, "function": [4, 6, 7, 8, 9, 12, 14, 15, 16, 20, 23, 24, 26, 27, 30, 32, 37, 40, 42], "functional_annot": 14, "functionalannot": [0, 14], "functionalannotationawar": [0, 12], "further": [20, 24, 37], "furthermor": 30, "fwer": 31, "g": [2, 3, 6, 7, 8, 12, 13, 14, 18, 20, 24, 26, 27, 28, 30, 31, 32, 34, 37, 42], "ga": 27, "ga4gh": [12, 18, 20, 22, 23, 24, 29], "gain": 20, "galleri": 29, "ganglion": 27, "gastrin": 27, "gastrointestin": 27, "gather": 26, "gb_acc": 13, "ge": 40, "genbank": 13, "genbank_acc": 13, "gene": [6, 12, 14, 20, 23, 24, 25, 26, 30, 37, 40, 42], "gene_coordinate_servic": 14, "gene_id": 12, "gene_nam": 12, "gene_symbol": 12, "genecoordinateservic": [0, 14], "gener": [2, 3, 5, 7, 12, 14, 16, 18, 20, 23, 26, 29, 38], "general_hpo_term": 2, "genet": [18, 40], "genit": 27, "genitourinari": 27, "genom": [6, 12, 14, 18, 20, 26, 28, 30, 42], "genome_build": [14, 20, 26, 30], "genome_build_id": 13, "genomebuild": [12, 13, 14], "genomebuildidentifi": [12, 13], "genomicinterpret": 14, "genomicregion": [12, 13, 30], "genotyp": [0, 2, 3, 4, 5, 8, 9, 12, 14, 18, 21, 26, 27, 29, 34, 36, 38, 39, 40, 42], "genotype_for_sampl": 12, "genotype_phenotype_scor": [8, 22, 24], "genotypepolypred": [2, 3, 5, 6, 8, 36, 38], "gestat": 33, "get": [2, 3, 4, 5, 6, 7, 8, 10, 12, 13, 14, 15, 16, 29, 31], "get_annot": 14, "get_cache_dir_path": [0, 10, 14], "get_categor": [5, 7, 22, 24], "get_categori": 5, "get_category_nam": 5, "get_cds_region": [12, 30], "get_coding_base_count": [12, 30], "get_codon_count": [12, 30], "get_excluded_count": 12, "get_features_variant_overlap": 12, "get_five_prime_utr": [12, 30], "get_hgvs_cdna_by_tx_id": 12, "get_maximum_group_observed_hpo_frequ": 2, "get_number_of_observed_hpo_observ": 2, "get_patient_id": 12, "get_preferred_tx_annot": 12, "get_quest": [6, 8, 22, 23, 24, 40, 42], "get_question_bas": [5, 7], "get_respons": 14, "get_three_prime_utr": 12, "get_tx_anno_by_tx_id": 12, "get_variant_by_kei": [12, 42], "gg": [12, 14], "git": 19, "github": [19, 38], "give": [6, 30], "given": [2, 6, 7, 12, 14, 15, 33, 37], "gland": 27, "glial": 27, "gln151ter": 20, "gln302argfster92": [20, 26], "gln315argfster79": [20, 26], "gln421ter": 12, "gln456ter": 20, "gln49ly": 20, "global": [18, 33], "globe": 27, "glossari": 29, "glu294ter": [20, 26], "glucagon": 27, "glucocorticoid": 27, "glucos": 27, "gly125alafster25": 20, "gly195ala": 20, "gly80arg": [20, 26], "glycolipid": 27, "glycoprotein": 27, "glycosaminoglycan": 27, "glycosyl": 27, "go": 12, "gonadotropin": 27, "good": [20, 30], "gov": 25, "gp": 20, "gpc": 29, "gpsea": [17, 19, 20, 21, 22, 23, 24, 25, 26, 28, 29, 30, 32, 33, 34, 36, 37, 38, 39, 40, 41, 42], "gpsea_cach": [10, 14], "gpsea_cachedir": 10, "gpseajsondecod": [0, 11, 22, 23, 24, 26, 30], "gpseajsonencod": [0, 11, 30], "gpseareport": [0, 16, 26], "granulocytopoiet": 27, "graph": [7, 23], "graphawar": 7, "graphic": [16, 23, 26], "grch37": [13, 14, 30], "grch38": [12, 13, 14, 20, 26, 28, 30], "great": 31, "greater": [31, 42], "grid": 24, "group": [2, 3, 4, 5, 6, 7, 8, 9, 20, 21, 22, 25, 29, 30, 32, 33, 34, 35, 38, 40, 42], "group1": 24, "group2": 24, "group_nam": [6, 36], "groups_pred": [5, 6, 36], "growth": [27, 30], "gt": 12, "gt_id_to_nam": [22, 24], "gt_predic": [2, 3, 8, 20, 22, 23, 24, 32, 34, 36, 39, 40], "guid": [18, 20, 30], "h": 31, "h_1": 4, "ha": [2, 3, 5, 6, 7, 8, 12, 13, 14, 20, 23, 24, 26, 27, 30, 33, 36], "had": 2, "hair": 27, "hallmark": 7, "hand": [20, 23, 27, 30, 33], "handl": [15, 16], "happen": 3, "harbor": [24, 32], "has_error": 14, "has_errors_or_warn": 14, "has_sv_info": 12, "has_variant_coordin": 12, "has_warn": 14, "hashabl": 12, "have": [2, 8, 12, 14, 23, 30, 31, 32, 33, 39], "head": [22, 24, 27], "health": 18, "hear": 24, "heart": [20, 23, 24, 27, 33], "height": 27, "helic": 12, "help": 20, "hematocrit": 27, "hemizyg": 12, "hemoglobin": 27, "henc": [14, 19, 23], "hepat": 27, "hepatobiliari": 27, "here": [6, 13, 20, 23, 24, 28, 30, 32, 33, 36, 37, 38], "hereditari": 30, "heterozyg": [6, 12, 23, 35, 40], "heterozygot": [25, 40], "heurist": [2, 20, 27, 31], "hgnc": [12, 14], "hgv": [12, 14, 16], "hgvs_cdna": 12, "hgvs_coordinate_find": 14, "hgvsp": 12, "hierarch": [14, 27], "hierarchi": [2, 33, 37], "high": 12, "higher": [23, 24, 33], "highest": 12, "hip": 27, "his1435arg": 42, "his220del": 20, "his445metfster137": 20, "histidin": 42, "histori": 31, "hkd": 26, "hmf01": [20, 23], "hmf02": 2, "hmf03": [2, 20, 23], "hmf05": 2, "hmf07": 2, "hmf08": [2, 20, 23], "hmf09": [20, 23], "ho": 31, "hochberg": [20, 23, 31], "holm": 31, "holt": [20, 26, 42], "homeostasi": 27, "hommel": 31, "homo": 6, "homovanil": 27, "homozyg": [6, 40], "homozygot": 25, "homozygous_altern": 12, "homozygous_refer": 12, "honeydew": 8, "hormon": 27, "how": [13, 19, 20, 21, 30, 36, 37, 42], "howev": [5, 13, 14, 19, 20, 23, 28, 35, 36, 38, 40, 42], "hp": [2, 3, 7, 8, 12, 20, 23, 24, 26, 27, 31, 33, 37], "hpo": [2, 3, 7, 8, 12, 14, 16, 18, 22, 23, 26, 29, 30, 33, 38, 41, 42], "hpo_mtc": 31, "hpomtcfilt": [1, 2, 20, 23, 31], "hpopred": [5, 7, 37], "hpotermanalysi": [1, 3, 20, 23, 31, 37], "hpotermanalysisresult": [1, 3, 16], "hpotk": [2, 20, 23, 24, 26, 30, 31, 33, 37, 42], "html": 16, "http": [19, 25], "human": [2, 5, 12, 13, 14, 16, 18, 26, 30, 33], "humor": 27, "hundr": 31, "hypertelor": 33, "hypoplasia": [20, 23, 26], "hypospadia": 33, "hypothalamu": 27, "hypothes": [23, 26, 29, 30, 31], "hypothesi": [2, 20, 23, 26, 31], "i": [0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 18, 19, 20, 23, 24, 25, 26, 27, 28, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 42], "id": [5, 6, 7, 8, 12, 14, 16, 20, 26, 27, 28, 30, 35], "ident": 25, "identifi": [5, 6, 8, 12, 13, 14, 20, 26, 27, 30, 33], "idx": 2, "ignor": [9, 14, 30], "ii": 27, "ile106v": 20, "ile227_glu228inst": 20, "ile54thr": 20, "illustr": [20, 23], "img": [24, 30], "immun": 27, "impact": 6, "impair": 24, "implement": [4, 6, 11, 14, 21, 23, 25, 29, 30, 33, 39, 42], "implementor": 12, "impli": [7, 23, 28, 31], "implicitli": [28, 37], "import": [4, 6, 8, 12, 13, 20, 22, 23, 24, 26, 28, 30, 31, 32, 33, 34, 36, 37, 39, 40, 42], "impos": 4, "imprecis": 12, "imprecise_sv_functional_annot": 14, "imprecisesvfunctionalannot": [0, 14], "imprecisesvinfo": [0, 12, 14], "includ": [2, 3, 4, 5, 6, 7, 8, 12, 13, 14, 16, 18, 20, 22, 23, 24, 29, 30, 32, 35, 36, 37, 40], "include_computational_tx": 14, "include_patients_with_no_hpo": 12, "include_patients_with_no_vari": 12, "include_subsect": 14, "inclus": [6, 35, 42], "incomplete_terminal_codon_vari": 12, "increas": [20, 23, 26], "indel": 12, "indent": [11, 14], "independ": 31, "index": [6, 8, 12, 19, 27, 32, 40], "indic": [12, 32, 40], "indirect": [20, 23, 31], "indirectli": 7, "individu": [2, 3, 6, 7, 8, 12, 18, 20, 22, 23, 24, 25, 26, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 40, 42], "infer": 20, "inflammatori": 27, "info": [2, 3, 6, 12, 14, 20, 23, 26, 28, 42], "inform": [6, 12, 16, 18, 20, 26, 29, 30, 31, 33, 38], "infram": 20, "inframe_delet": [12, 20, 26], "inframe_insert": [12, 20, 26], "ingest": 18, "inherit": [6, 22, 23, 29, 31, 32, 35], "inhibin": 27, "initi": [7, 12, 19], "inner": 27, "input": [2, 3, 14, 15, 16, 18, 20, 26, 29, 42], "insert": [12, 14, 20, 28], "insight": [20, 27, 30], "instal": [18, 30], "instanc": [2, 4, 5, 6, 7, 8, 12, 14, 23, 24, 25, 26, 28, 30, 31, 32, 33, 37, 40, 42], "instanti": 30, "instead": [6, 12, 31, 32], "instruct": [18, 22, 23, 24], "insulin": 27, "int": [2, 3, 4, 5, 6, 7, 12, 13, 14, 16], "integ": 12, "integr": [19, 26, 30], "integu": 27, "intellectu": [24, 33], "intend": [8, 12], "inter": [24, 38], "interact": [12, 16, 29], "interest": [6, 7, 12, 14, 20, 23, 26, 27, 29, 31, 32, 40], "intergenic_vari": [12, 14], "internet": 19, "interpret": [14, 15, 24, 35], "intersect": [13, 32, 40], "interspers": 12, "interv": 13, "intestin": 27, "intracrani": 27, "intraocular": 27, "intrauterin": 33, "introduct": 29, "intron": [12, 20], "intron_vari": [12, 14, 20, 26, 42], "inv": 12, "invers": [12, 42], "investig": [7, 12, 20, 22, 32, 39], "invoc": 19, "involv": [12, 20, 23, 30, 31], "io": [0, 12, 14, 15, 16, 22, 23, 24, 26, 30], "iobas": [12, 15, 16], "ion": 27, "ip3": 12, "ipython": [16, 26], "iqr": 24, "is_cod": [12, 30], "is_empti": 13, "is_femal": 12, "is_filtered_out": 2, "is_frameshift": [23, 32], "is_large_imprecise_sv": 6, "is_mal": 12, "is_missens": [22, 32, 40], "is_neg": 13, "is_observ": 12, "is_ok": 14, "is_pass": 2, "is_posit": 13, "is_pref": 12, "is_pres": 12, "is_provid": 12, "is_structur": 12, "is_structural_delet": 6, "is_structural_vari": 6, "is_unknown": 12, "islet": 27, "isoform": [12, 30], "issu": [2, 12, 14, 18, 20, 30, 38], "item": [2, 3, 5, 14, 16, 28], "iter": [2, 3, 5, 6, 7, 8, 11, 12, 13, 14, 16, 31], "iter_cohort_phenopacket": [20, 30, 37, 42], "iterate_nod": 14, "its": [7, 12, 14, 16, 24, 28, 37, 38, 42], "j": 3, "jinja2": 16, "joint": [23, 27], "jordan": 24, "json": [11, 12, 14, 22, 23, 24, 26], "jsondecod": 11, "jsonencod": 11, "jt": 25, "judgment": [27, 31, 40], "jupyt": [16, 18, 26], "just": [12, 14, 16, 20, 22, 30], "keep": [7, 14], "kei": [6, 12, 18, 20, 26], "kidnei": [8, 24], "kind": [14, 25, 29], "know": [12, 13, 14, 30], "known": [12, 26, 37], "kwarg": 11, "lab": 22, "label": [6, 12, 14, 16, 27, 30, 34, 36], "label_summari": 12, "labeling_method": 16, "laboratori": 24, "lack": [7, 14, 31], "lactat": 27, "langerhan": 27, "larg": [6, 12, 13, 14, 27], "larger": 12, "last": [10, 12, 20, 23, 24, 31], "later": [22, 23, 24, 29], "latest": 20, "lead": [6, 12, 20, 23, 24, 27, 28, 31, 32, 42], "learn": [20, 23], "least": [2, 6, 14, 23, 24, 31, 36, 38, 40, 42], "leav": [18, 30], "left": 13, "legend": [16, 22], "leigh": 12, "len": [12, 20, 22, 23, 24, 26, 28, 30, 37, 42], "length": [6, 12, 13, 24, 26, 29, 42], "lenient": 14, "lenti": [34, 37], "leptin": 27, "less": [6, 20, 23, 31], "let": [11, 12, 23, 24, 28, 30, 31, 32, 40, 42], "letter": 30, "leu435argfster147": 20, "leu65glnfster10": 20, "leu94arg": 20, "leukocyt": 27, "level": [0, 8, 12, 14, 20, 23, 24, 26, 27, 30, 42], "leverag": [2, 26, 30, 33, 37, 42], "li": [13, 25], "librari": [0, 13, 19, 21, 30], "life": 23, "like": [11, 16, 18, 27, 31], "limb": [20, 23, 27], "limit": [2, 31, 33], "line": [6, 12, 19, 24], "lineag": 27, "link": [35, 40], "liposaccharid": 27, "list": [6, 7, 8, 11, 12, 20, 27, 37, 38, 40], "list_all_diseas": 12, "list_all_protein": 12, "list_all_vari": 12, "list_present_phenotyp": 12, "liter": [6, 12, 14, 16, 28], "literatur": [6, 20], "liver": [8, 27], "ll": 30, "load": [11, 14, 26, 33, 37, 42], "load_minimal_hpo": [20, 23, 24, 26, 30, 31, 33, 37, 42], "load_phenopacket": [0, 14, 20, 30, 37, 42], "load_phenopacket_fil": [0, 14, 30], "load_phenopacket_fold": [0, 14, 30], "loader": 30, "local": [14, 15, 26], "locat": [12, 13, 14, 18, 20, 26, 27, 30, 42], "locu": [12, 32], "lof": 24, "lof_effect": 24, "lof_mut": 24, "log": [26, 41], "logic": [31, 33, 42], "loinc": [8, 12, 22], "long": [20, 23], "loss": [12, 24], "lower": [23, 27, 42], "lump": [32, 40], "ly": 24, "lymph": 27, "lymphat": 27, "lymphocyt": 27, "lys226asn": 20, "lys88ter": 20, "lysosom": 27, "m": [4, 19], "macrocephali": 33, "macroscop": 27, "made": [20, 31], "magic": 26, "mai": [3, 8, 12, 14, 23, 24, 26, 30, 31, 32, 40, 42], "main": [20, 24, 30], "major": [13, 38], "major_assembli": 13, "make": [6, 26, 31], "malarkei": 14, "male": [12, 39], "malform": 14, "mandatori": 12, "mane": [12, 20, 22, 23, 24, 30, 42], "mane_pt_id": 30, "mane_tx_id": 30, "mani": [20, 21, 26, 27, 29, 30, 31, 37, 38, 42], "manifest": [18, 30], "mann": [8, 9, 21, 26, 29, 33, 41], "mannwhitneystatist": [8, 9, 24], "mannwhitneyu": [9, 24], "manual": [12, 26], "map": [2, 12, 14, 16], "marfan": 34, "marker": [16, 27], "marker_count": 16, "marrow": 27, "mass": [8, 12, 22, 24, 27], "match": [6, 8, 24, 34, 35, 36], "matern": 25, "math": [2, 8], "matplotlib": [8, 16, 20, 24, 26, 30], "matrix": [2, 3], "matter": 14, "matur": 27, "mature_mirna_vari": 12, "maxim": 38, "maximum": [2, 20, 23, 33], "me": 12, "mean": [12, 23, 24, 27, 31], "meaning": 5, "measur": [0, 8, 12, 21, 24, 26, 27, 29], "measurement_by_id": 12, "measurementphenotypescor": [1, 8, 22, 24], "median": 24, "mediastinum": 27, "medic": [18, 27, 31], "medicin": 18, "meet": [6, 12, 24, 35, 42], "megakaryocyt": 27, "member": [12, 14, 23, 30], "membran": 27, "mendelian": 18, "mental": 27, "mesenteri": 27, "messag": [11, 14], "met": 42, "met74il": 20, "met83phefster6": 20, "meta_label": [12, 30], "metabol": 27, "metabolit": 27, "metadata": [14, 16], "metaphysi": 27, "method": [2, 3, 6, 7, 11, 12, 13, 14, 20, 24, 30, 31], "mhc": 27, "micro": 33, "microcephali": 33, "middl": 27, "midwif": 12, "might": [25, 26], "mild": 33, "mim": [20, 42], "min_n_of_patients_with_term": [7, 20, 23], "miner": 27, "minimalontologi": [2, 7, 8, 14, 16, 37], "minimalterm": 12, "minimum": [2, 7], "mininum": 31, "miss": [8, 14, 20, 31, 38], "missens": [2, 3, 6, 12, 18, 22, 23, 24, 25, 28, 31, 35, 36, 40, 42], "missense_and_exon20": 42, "missense_or_nonsens": 42, "missense_vari": [3, 6, 12, 20, 22, 24, 26, 32, 36, 40, 42], "missing_implies_exclud": 7, "missing_implies_phenotype_exclud": 7, "mitig": [23, 31], "mitochondri": 40, "mitochondrion": 27, "mixin": 13, "mnv": 12, "mode": [6, 14, 23, 29, 31, 32, 35], "model": [0, 2, 6, 20, 22, 23, 24, 26, 28, 30, 32, 36, 40, 42], "moder": 33, "modifi": 31, "modul": [0, 12, 13, 14, 24, 26, 30], "moi": 40, "moment": 14, "monarch": 19, "mondo": 7, "monoallel": [6, 23], "monoallelic_pred": [5, 6, 20, 24, 32], "more": [2, 3, 6, 7, 8, 9, 12, 14, 16, 19, 20, 23, 24, 26, 27, 29, 30, 33, 34, 36, 42], "morpholog": 27, "morphologi": [8, 18, 20, 23, 24, 26, 27, 28, 31, 33, 37], "mortal": 26, "most": [12, 13, 20, 26, 27, 30, 31, 32, 38], "mostli": [12, 38], "motif": [12, 18], "motil": 27, "motor": [20, 27], "movement": 27, "mri": 27, "mtc": [2, 3, 16, 20, 23, 27, 29], "mtc_alpha": [3, 20, 23], "mtc_correct": [3, 20, 23, 31], "mtc_filter": [1, 3, 20, 23, 31], "mtc_filter_nam": 3, "mtc_filter_result": 3, "mtc_issu": 2, "mtc_name": 3, "mtc_report": [20, 23], "mtc_viewer": [20, 23], "mtcstatsview": [0, 16, 20, 23], "much": [2, 26, 27, 31, 40, 42], "mucociliari": 27, "mucosa": 27, "mucu": 27, "multi": [12, 42], "multiphenotypeanalysi": [1, 3, 31], "multiphenotypeanalysisresult": [1, 3], "multipl": [2, 3, 20, 27, 29, 42], "multipli": 31, "muscl": 27, "muscular": [20, 23], "musculatur": 27, "musculoskelet": 27, "must": [2, 3, 5, 6, 7, 8, 12, 14, 19, 24, 26, 30, 31, 34, 38, 42], "mutat": [20, 22, 23, 24, 26, 30, 37, 40, 42], "mutlipl": 31, "myelin": 27, "myeloid": 27, "n": [2, 4, 6, 12, 14, 20, 23, 26, 27, 36, 37], "n_categor": 5, "n_usabl": [2, 3], "nacgtacgtac": 12, "nail": 27, "name": [2, 3, 5, 6, 10, 12, 13, 14, 20, 22, 24, 26, 30, 32], "nan": [3, 8, 9, 22, 24], "nanogram": 12, "nasal": [27, 33], "natur": 12, "nc_000001": 13, "ncbi": [20, 25], "ncit": 12, "ndarrai": 16, "neck": 27, "need": [6, 14, 20, 23, 29, 30, 31, 32, 37, 40], "neg": [6, 13, 14, 28], "neoplasm": 27, "neopterin": 27, "nerv": 27, "nervou": [27, 28, 31], "net": 12, "network": 30, "neuron": 27, "never": [13, 31], "new": 16, "next": [20, 30, 42], "ng": 12, "nih": 25, "nippl": 27, "nlm": 25, "nm_000500": 22, "nm_001042681": [24, 42], "nm_002834": 14, "nm_005912": 14, "nm_013275": 42, "nm_123": 6, "nm_1234": [32, 36, 40], "nm_123456": [3, 6, 12], "nm_170707": 12, "nm_181486": [20, 23, 26, 30], "nmd_transcript_vari": 12, "no_cal": 12, "no_genotype_has_more_than_one_hpo": 2, "node": [14, 27], "nomin": [3, 31], "non": [2, 3, 6, 12, 14, 20, 24, 31, 36, 42], "non_coding_transcript_exon_vari": [12, 14], "non_coding_transcript_vari": [12, 14], "non_frameshift_del": 42, "non_frameshift_effect": 42, "non_frameshift_pred": 42, "non_specified_term": 2, "noncoding_effect": 14, "none": [2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 16, 20, 24, 26, 30, 31, 32, 33, 36, 38, 40], "nonfaci": 33, "nons": 25, "nonsens": [2, 3, 6, 22, 23, 25, 31, 42], "normal": 18, "nose": 33, "notat": [12, 28], "note": [2, 3, 7, 10, 12, 13, 20, 23, 30, 31, 33, 34, 39, 42], "notebook": [16, 18, 26], "notepad": [0, 14], "notepadtre": [0, 14], "noth": 42, "novel": 12, "now": [12, 20, 22, 23, 24, 30, 31, 32, 37], "np_": 20, "np_000129": 12, "np_001027558": 14, "np_001027559": 12, "np_037407": 14, "np_852259": [20, 26, 30], "nuclear": 12, "nucleobas": 27, "nucleotid": [12, 20, 28, 42], "nucleu": 31, "nul": 12, "null": 31, "number": [2, 3, 4, 5, 6, 7, 8, 12, 13, 20, 24, 27, 31, 33, 40], "numer": [5, 12, 26], "o": 11, "obj": 11, "object": [2, 4, 5, 6, 8, 9, 11, 12, 13, 14, 16, 30], "object_hook": 11, "observ": [2, 8, 12, 14, 18, 20, 23, 26, 37], "observablefeatur": 12, "obtain": [12, 14, 20, 30, 31, 33], "obvious": 26, "occipitofrontali": 27, "occur": [12, 20, 24, 26, 30], "occurr": [20, 26], "oddsratio": 23, "odontoid": 27, "off": [38, 42], "offer": [21, 26, 31, 33, 41, 42], "offlin": 14, "often": 12, "ok": [2, 14], "oldest": 31, "omim": [7, 20, 26, 34], "omit": [2, 20, 23, 32, 34, 38, 39], "onc": [23, 30], "one": [2, 5, 6, 7, 8, 12, 14, 19, 20, 23, 24, 25, 26, 27, 30, 33, 34, 36, 38, 40], "one_genotype_has_zero_hpo_observ": 2, "ones": 14, "onli": [2, 5, 7, 12, 14, 20, 23, 25, 26, 31, 32, 34, 36, 37, 40], "onlin": 19, "only_hgv": 16, "onset": [26, 27, 28], "onto": 16, "ontologi": [2, 7, 12, 18, 20, 28, 30, 33], "oocyt": 27, "open": [15, 22, 23, 24, 26, 30, 38], "open_phenopacket_stor": [20, 30, 37, 42], "open_text_io_handle_for_read": [0, 15], "open_text_io_handle_for_writ": [0, 15], "oper": [6, 42], "oppos": 12, "opposit": 13, "optim": 14, "option": [2, 5, 7, 12, 14, 19, 20, 30, 31, 32, 37, 40], "oram": [20, 26, 42], "order": [2, 3, 12, 20, 23, 26, 30], "organ": [12, 14, 24, 27], "orgogozo": 18, "origin": 33, "orthogon": 35, "osmol": 27, "osteocalcin": 27, "other": [2, 12, 13, 14, 16, 18, 20, 22, 25, 30, 31, 32, 33, 35, 39], "otherwis": [6, 7, 12, 33, 36], "our": [20, 21, 23, 27, 31, 33, 38], "out": [2, 13, 14, 20, 23, 24, 28, 30, 31], "outcom": 3, "outer": 27, "output": [14, 15, 16], "over": [12, 14], "overal": [2, 31], "overlap": [3, 6, 12, 13, 16, 32, 40, 42], "overlapping_exon": 12, "overlaps_with": [12, 13], "overload": 42, "overrepres": 12, "overview": [20, 21, 26], "own": 13, "p": [2, 3, 4, 7, 8, 9, 12, 16, 18, 20, 23, 26, 30, 31, 37, 42], "p13": [13, 14], "p_valu": [23, 24], "packag": [17, 18, 19, 31], "page": [12, 20, 26, 29, 38], "pair": [3, 4, 6, 12, 28], "pancrea": 27, "pancreat": 27, "panda": [2, 12, 30], "param": [6, 12, 13, 14], "paramet": [2, 3, 5, 6, 7, 8, 11, 12, 13, 14, 15, 16], "parametr": 24, "parathyroid": 27, "parent": [3, 14, 31, 33], "pars": [12, 14], "parse_multipl": 14, "parse_respons": 14, "parse_uniprot_json": 14, "part": [12, 16, 30, 33], "particular": 26, "partit": [6, 22, 29, 35, 38], "pass": [2, 6, 11, 12, 28, 31], "past": 31, "patch": [8, 13], "patent": [20, 23], "patern": 25, "path": [10, 12, 14, 15, 16, 20, 23, 29, 30, 37], "pathogen": [20, 30, 42], "patient": [0, 2, 3, 5, 6, 7, 8, 12, 14, 16, 18, 20, 23, 24, 30, 31, 36, 37, 38], "patient_1": 8, "patient_2": 8, "patient_3": 8, "patient_4": 8, "patient_cr": 14, "patient_id": [8, 12, 22, 24], "patientcategori": [1, 5, 7], "patientcr": [0, 14], "pattern": [14, 27], "payload": 14, "pcat": [1, 20, 23, 31], "pd": [12, 30], "pehnotyp": 2, "peptid": 27, "per": 12, "percent": [20, 23], "perform": [3, 4, 12, 14, 18, 20, 23, 24, 25, 26, 29, 30, 31], "perifollicular": 27, "peripher": 27, "peritoneum": 27, "permiss": [14, 30], "peroxisom": 27, "persist": [22, 23, 24, 29], "ph": [26, 27], "ph_predic": 2, "phagocytosi": 27, "phalanx": 23, "phe168leufster6": 20, "pheno_pred": [3, 20, 23, 37], "pheno_scor": [8, 22, 24, 33], "phenopacket": [12, 14, 18, 20, 22, 23, 24, 26, 29, 37, 42], "phenopacket1": 30, "phenopacket2": 30, "phenopacket_registri": 20, "phenopacketpatientcr": [0, 14], "phenopacketvariantcoordinatefind": [0, 14], "phenotyp": [0, 2, 3, 4, 5, 8, 9, 12, 14, 18, 21, 26, 27, 29, 32, 33, 35, 38, 39, 40, 42], "phenotype_cr": 14, "phenotype_scor": 8, "phenotypecategor": [5, 7], "phenotypecr": [0, 14], "phenotypemtcfilt": [1, 2, 3, 20, 31], "phenotypemtcissu": [1, 2], "phenotypemtcresult": [1, 2, 3], "phenotypepolypred": [2, 3, 5, 7, 38], "phenotypescor": [1, 8, 9], "phenotypescoreanalysi": [1, 8, 22, 24], "phenotypescoreanalysisresult": [1, 8], "phenotypescorestatist": [8, 9], "phenotypic_abnorm": 2, "phosphat": 27, "physician": 12, "physiolog": 27, "physiologi": [18, 27, 28], "pickl": 14, "piec": [12, 20], "piezo1": 30, "pineal": 27, "pinna": 33, "pip": 19, "pituitari": 27, "place": [18, 30, 39], "placent": 27, "placenta": 27, "plan": 26, "plasma": [8, 12, 22, 24], "plate": 30, "platelet": 27, "platysma": 27, "pld1": 26, "pleas": [26, 38], "plot": [8, 16, 29], "plot_boxplot": [8, 24], "plt": [16, 20, 24, 26, 30], "plu": 6, "pm": [20, 30], "pm_servic": 26, "pmid": 20, "pmid_27087320_subject_1": 24, "pmid_27087320_subject_10": 24, "pmid_27087320_subject_2": 24, "pmid_29330883_subj": 30, "pmid_29330883_subject_1": 24, "pmid_29330883_subject_2": 24, "pmid_30968594_individual_10": 22, "pmid_30968594_individual_11": 22, "pmid_30968594_individual_12": 22, "pmid_30968594_individual_13": 22, "pmid_30968594_individual_14": 22, "png": [24, 30], "po": [12, 13, 28], "point": [12, 14, 24, 30, 33], "point_mut": 24, "point_mutation_effect": 24, "polar": [30, 31], "polici": [14, 20, 30], "polypred": [1, 5, 6, 7], "pore": 30, "port": 13, "posit": [3, 6, 12, 13, 14, 20, 26, 28, 31], "posixpath": 10, "possibl": [2, 3, 5, 8, 25, 26, 30, 34, 39], "possible_result": 2, "possibli": 35, "posterior": [27, 31], "potassium": 26, "potenti": [24, 29], "power": [29, 31, 38], "pp": 14, "pp_dir": 30, "pp_directori": 14, "pp_file": 14, "pp_file_path": 30, "ppktstore": [20, 30, 37, 42], "practic": 25, "pre": 2, "precis": 18, "precursor": 27, "predic": [1, 2, 3, 4, 8, 29, 34, 39], "predict": [6, 12, 20, 24, 30, 42], "predictor": [14, 30], "prefer": [12, 26], "preimplant": 27, "prematur": 18, "prenat": 27, "prepar": [3, 5, 6, 7, 12, 14, 22, 23, 24, 30, 34], "prepare_hpo_terms_of_interest": [5, 7], "prepare_notepad": 14, "prepare_predicates_for_terms_of_interest": [5, 7, 20, 23, 37], "preprocess": [0, 20, 26, 30, 37, 42], "preprocessingvalidationresult": [0, 14, 30], "preproprotein": 12, "prerequisit": 33, "presenc": [3, 6, 7, 12, 20, 23, 24, 26, 30, 31, 35, 37, 42], "present": [3, 7, 8, 12, 13, 16, 20, 23, 26, 32, 33, 37], "present_diseas": 12, "present_phenotyp": 12, "present_phenotype_categor": 7, "present_phenotype_categori": 7, "pressur": 27, "pretti": 16, "prevent": [2, 3, 42], "previou": [31, 32], "previous": [23, 42], "primari": [12, 27, 42], "principl": 25, "print": [12, 14, 30], "pro139glnfster11": 20, "pro14thr": [20, 26], "pro85thr": 20, "probabl": [23, 31], "problem": 31, "proce": 14, "procedur": [3, 20, 23, 27, 29], "process": [14, 16, 20, 23, 26, 30, 37, 42], "process_respons": 14, "produc": [2, 4, 5, 7, 8, 26], "product": 16, "profound": 33, "programmat": [6, 14], "programmaticali": 30, "project": 12, "prolifer": 27, "prong": [23, 38], "pronounc": 18, "propagatingphenotypepred": 7, "properti": [2, 3, 4, 5, 7, 8, 12, 13, 14, 16, 32, 40], "proport": 31, "protcachingmetadataservic": [0, 14], "protein": [6, 12, 13, 14, 16, 18, 27, 29, 32, 40, 42], "protein_altering_vari": 12, "protein_cach": 14, "protein_effect_coordin": 12, "protein_effect_end": 12, "protein_effect_loc": 12, "protein_effect_start": 12, "protein_featur": [6, 12, 30], "protein_feature_end": 16, "protein_feature_nam": 16, "protein_feature_start": 16, "protein_feature_typ": [6, 16], "protein_id": [12, 14, 16, 30], "protein_length": [12, 16, 30], "protein_meta": [12, 16, 20, 26, 30], "protein_metadata": [16, 26], "protein_metadata_servic": 6, "protein_sourc": 14, "proteinannotationcach": [0, 14], "proteinfeatur": [0, 12], "proteinmetadata": [0, 12, 14, 16, 20, 30], "proteinmetadataservic": [0, 6, 14, 30], "proteinpred": [5, 6, 38], "proteinvariantview": [0, 16, 26], "proteinvisu": [0, 16, 20, 26, 30], "proteinvisualiz": [0, 16, 30], "provid": [2, 3, 6, 7, 8, 12, 13, 14, 16, 18, 20, 21, 24, 26, 29, 31, 32, 33, 37, 38, 39, 40], "proxi": 33, "prune": 14, "pscore": [1, 22, 24, 33], "pt_id": 26, "publish": [19, 37], "pubm": 25, "pulmonari": 27, "pupillari": 27, "purin": 27, "purpos": [2, 22, 23, 26, 30], "put": [12, 20, 22, 23, 24], "putamen": 31, "pval": [3, 8, 22, 24], "pvalu": 24, "pvi": [16, 30], "px": 26, "px_id": 20, "py": 12, "pydoc": 30, "pypi": 19, "pyplot": [20, 24, 26, 30], "pyrimidin": 27, "pytest": 19, "python": [4, 11, 18, 19, 23, 26, 30, 31, 42], "python3": 19, "q": [20, 30], "q1": 24, "q3": 24, "q99593": 30, "qc_result": 30, "qual": [12, 28], "qualnam": [12, 13, 14], "quartil": 24, "queri": [7, 8, 12, 14, 24, 30, 37], "question": [5, 6, 7, 18, 21, 42], "r": 24, "radiu": [20, 23, 26], "rais": [5, 11, 12, 13, 14], "random": [4, 12], "random_se": 16, "rang": [2, 24, 33], "rank": [9, 24, 26, 41], "rare": [0, 2, 20, 30, 37], "rate": [20, 27, 31], "rather": 27, "ratio": 27, "re": 12, "reach": 14, "read": [15, 16, 18, 42], "readabl": [5, 12, 16], "reader": 30, "readi": 20, "readili": 30, "real": [22, 23], "realiz": 31, "realli": 30, "reason": [2, 20, 23, 27, 30], "receiv": 20, "recent": 31, "receptor": 30, "recess": [6, 32], "recommend": [2, 6, 12, 18, 20, 23, 26, 30, 31, 37], "record": [14, 37, 40], "red": [24, 27], "redox": 27, "reduc": [2, 20, 30, 31], "redund": 31, "ref": [6, 12, 16, 23, 24, 28], "ref_length": [6, 24], "refer": [6, 12, 18, 24, 27, 29, 30, 32, 33], "reflect": 38, "reflex": 27, "refract": 27, "refseq": [12, 13, 14, 30], "refseq_nam": 13, "regard": [2, 12, 16], "regardless": 12, "region": [6, 12, 13, 16, 20, 26, 28, 30, 42], "registri": [20, 30, 37, 42], "regul": [26, 27], "regulatory_region_abl": 12, "regulatory_region_amplif": 12, "regulatory_region_vari": 12, "rel": [6, 33], "relat": [14, 31], "relationship": [18, 23], "releas": [20, 23, 24, 26, 27, 30, 31, 33, 37, 42], "relev": [2, 12, 20, 24, 30], "remot": [14, 30], "remov": [6, 14, 27, 28, 42], "renal": 24, "renin": 27, "repair": 27, "repeat": 12, "replac": 42, "report": [3, 8, 14, 16, 18, 20, 22, 23, 24, 26, 37], "repositori": 19, "repres": [2, 3, 5, 7, 8, 12, 13, 14, 16, 18, 24, 30, 31, 32], "represent": [12, 14], "reproduct": 27, "request": 14, "requir": [4, 12, 14, 19, 20, 31, 32, 38, 40], "requisit": 30, "rere": [24, 30, 42], "rere_mane_tx_id": 42, "rere_phenotype_score_boxplot": 24, "residu": [26, 30], "resourc": [6, 12, 14, 30], "respect": [3, 12, 28, 29, 30, 32, 36, 42], "respir": 27, "respiratori": 27, "respons": [14, 24, 27, 30], "rest": [12, 14, 22, 27], "restrict": 25, "result": [2, 3, 8, 12, 14, 16, 18, 20, 22, 23, 24, 26, 27, 30, 31, 37, 38, 42], "retain": 31, "retainin": 20, "reticulocyt": 27, "retriev": [12, 13, 14], "return": [2, 3, 4, 5, 6, 7, 8, 11, 12, 13, 14, 15, 16, 31, 36, 38], "reus": 32, "revers": 13, "right": [13, 14, 16], "risk": 23, "rna": 13, "root": [2, 14, 27, 31], "row": [2, 3, 20], "rule": [5, 20, 23, 26, 29, 37, 40, 42], "run": [3, 14, 22, 24, 30, 31], "runner": 14, "runonlin": 19, "rust": 12, "sai": 31, "salient": 26, "salivari": 27, "same": [2, 3, 12, 13, 20, 23, 24, 27, 28, 30, 31, 32, 33, 40], "same_count_as_the_only_child": 2, "sampl": [12, 20], "sample_id": 12, "samplelabel": [0, 12], "sanit": 14, "sanitationresult": 14, "saniti": 14, "sapien": 6, "save": 30, "savefig": [24, 30], "scalp": 27, "scenario": 5, "schema": [12, 14, 16, 23, 30], "scienc": 6, "scientif": 26, "scipi": [4, 9, 23, 24], "scipyfisherexact": 3, "sclera": 27, "score": [8, 9, 21, 26, 29, 41], "score_analysi": [22, 24], "score_statist": [8, 22, 24], "scorer": [8, 24], "screenshot": 30, "search": [14, 20, 27, 30, 38], "sebac": 27, "second": [6, 14, 26, 32], "secondari": 12, "secret": 27, "section": [2, 3, 6, 7, 8, 12, 14, 20, 22, 23, 24, 26, 31, 32, 42], "secundum": [20, 23, 26], "sediment": 27, "see": [2, 3, 6, 7, 8, 9, 12, 16, 20, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 38, 40, 42], "seek": 35, "seem": 38, "seen": [20, 24, 26, 31], "segment": [12, 27, 37], "segreg": [27, 32], "seizur": [20, 28], "select": [2, 6, 24, 32, 40], "self": [11, 13], "sens": [26, 27, 31], "sensat": 27, "sensibl": 26, "sensori": 27, "sensorineur": 24, "separ": [11, 23, 41], "septal": [20, 23, 26, 28], "septum": [20, 23, 28], "sequenc": [2, 3, 4, 5, 6, 7, 9, 12, 13, 14, 16, 29, 30], "sequence_vari": 12, "ser196ter": 20, "ser261ci": [20, 26], "ser36thrfster25": [20, 26], "serial": [22, 23, 24], "serializ": 11, "seriou": 14, "serum": [8, 12, 22, 24, 27], "serv": [5, 38], "servic": 14, "set": [2, 8, 10, 12, 20, 23, 24, 31, 32, 37, 40], "setup": [20, 23, 26, 31], "sever": [3, 5, 12, 14, 20, 22, 23, 24, 26, 28, 30, 31, 32, 33, 40, 41, 42], "sex": [0, 6, 12, 27, 29, 35], "sex_pred": [5, 6, 39], "sh": 31, "sharpei": 27, "shelf": [24, 38, 42], "shift": 42, "ship": 30, "short": [12, 20, 23, 26, 33], "shortcut": 12, "shorter": [6, 23], "should": [2, 3, 6, 7, 10, 12, 13, 14, 16, 20, 22, 24, 30, 40], "show": [12, 16, 22, 23, 24, 26, 27, 28, 30, 31, 36, 37, 38, 42], "shown": [16, 20, 26, 29, 30], "sidak": 31, "side": [4, 24], "sign": [12, 13, 20, 22, 23, 24, 42], "signal": [27, 31], "signific": [12, 23, 24, 27, 31, 33], "significantli": [20, 23, 24], "sime": 31, "similar": [12, 24, 32], "similarli": 37, "simpl": [6, 14, 19, 25, 28, 30, 33, 42], "simplest": 31, "simpli": 2, "simplifi": [20, 23], "sinc": [3, 6, 12, 24], "sine": 12, "singl": [2, 12, 14, 20, 40, 42], "situ": 27, "size": [2, 12, 13, 26, 27], "skelet": 27, "skeleton": [20, 23], "skin": [8, 27], "skinfold": 27, "skip": [2, 20, 23, 27], "skipkei": 11, "skipping_general_term": 2, "skipping_non_phenotype_term": 2, "skipping_since_one_genotype_had_zero_observ": 2, "skull": 27, "slice": 13, "small": [28, 31, 33], "smell": 27, "snv": [12, 28, 42], "so": [6, 12, 24, 30, 31, 42], "so_term": 12, "softwar": 21, "solut": [14, 30], "somat": 27, "some": [4, 14, 18, 19, 26, 30, 31, 38], "some_cell_has_greater_than_one_count": 2, "someth": 14, "sometim": [32, 36, 40, 42], "sort": 14, "sort_index": [22, 24], "sort_kei": 11, "sound": [27, 37], "sourc": [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 19, 42], "span": [6, 12, 13, 27, 28, 30], "specif": [3, 6, 12, 14, 18, 20, 24, 26, 27, 30, 31, 32, 37, 40], "specifi": 2, "specified_term": 31, "specifiedtermsmtcfilt": [1, 2, 31], "spell": 28, "spinal": 27, "spleen": 27, "splice": [12, 20], "splice_acceptor_vari": [12, 14, 20, 26], "splice_donor_5th_base_vari": [12, 14, 20, 26], "splice_donor_region_vari": 12, "splice_donor_vari": [12, 14, 20, 26], "splice_polypyrimidine_tract_vari": [12, 14], "splice_region_vari": [12, 20, 26], "spuriou": 31, "st1": 2, "stabil": 27, "stalk": 27, "standalon": 16, "standard": [12, 29], "start": [6, 12, 13, 14, 16, 24, 26, 27, 30, 32], "start_lost": [12, 24], "start_on_strand": 13, "start_retained_vari": 12, "stat": [3, 8, 20, 22, 23, 24, 31], "state": [12, 14, 28], "static": [2, 5, 6, 8, 11, 12, 14, 31], "statist": [4, 18, 20, 27, 29, 31, 33, 38], "statsmodel": 31, "statu": [2, 14], "statur": 33, "stdout": 14, "step": [19, 20, 23, 29, 30, 35], "stimul": 27, "stomatocytosi": 30, "stool": 27, "stop": [20, 27], "stop_gain": [6, 12, 20, 24, 26, 42], "stop_lost": 12, "stop_retained_vari": [12, 20, 26], "storag": 14, "store": [14, 16, 20, 23, 24, 26, 30, 31, 33, 37, 42], "store_annot": 14, "str": [2, 3, 5, 6, 7, 8, 10, 12, 13, 14, 15, 16, 28, 30], "strabismu": 23, "strand": [6, 12, 13, 30], "strategi": [2, 16, 23, 27], "stress": 27, "strict": 14, "string": [8, 12, 14, 16, 27], "stringio": 14, "stromal": 27, "strongli": [6, 12], "structur": [6, 8, 12, 14, 23, 24, 26, 27, 28, 31, 33, 42], "structural_defect": 24, "structural_so_id_to_displai": 12, "structural_typ": [6, 12, 42], "structural_vari": 6, "student": 12, "studi": 18, "stump": 27, "sub": [12, 14, 30], "subclass": [6, 11], "subgroup": 12, "subject": [12, 24, 30, 37, 42], "submodul": 17, "subpackag": 17, "subplot": [20, 24, 26, 30], "subsect": [12, 14], "subsequ": 30, "subset": 32, "subunit": 27, "successfulli": [20, 26], "suffix": [14, 30], "suggest": 26, "suitabl": 12, "sum": [24, 33], "summar": [12, 14, 16, 28, 30], "summari": [14, 29, 30], "summarize_hpo_analysi": [0, 16, 20, 23], "summary_df": [20, 23], "suox": 25, "super": 11, "supercoil": 12, "suppli": 12, "support": [6, 11, 12, 13, 18, 25, 42], "supports_shap": 4, "suppos": 11, "suppresor": 12, "surf1": [6, 12], "surf2": 6, "surfac": 27, "surviv": [31, 41], "sv": [6, 12, 14, 22], "sv_delet": 28, "sv_info": 12, "svart": 13, "svlen": [12, 28], "svtype": [12, 28], "sweat": 27, "switch": 19, "symbol": [6, 12, 13, 20, 28, 30], "symptom": [12, 20, 22, 23, 24, 27, 42], "synapt": 27, "syndrom": [12, 20, 26, 34, 42], "synonym": [36, 42], "synonymous_vari": [12, 36, 42], "system": [12, 13, 27, 28, 31, 33], "t": [9, 12, 14, 16, 20, 22, 26, 28], "tabl": [4, 16, 20, 23, 26, 27, 28, 29, 31, 32, 40], "tail": 23, "take": [4, 8, 14, 20, 23, 30, 31, 32, 36, 37], "tall": 33, "tandem": 12, "target": [6, 7, 26, 30, 32, 42], "tast": 27, "taxon": 6, "tbx5": [26, 30, 37], "tbx5_gpsea_with_uniprot_domain": 30, "technic": 18, "tediou": [37, 42], "tedium": 37, "temperatur": 27, "templat": 16, "ten": [12, 31], "tend": 12, "term": [2, 3, 7, 8, 12, 14, 16, 18, 20, 22, 23, 24, 26, 28, 29, 33, 37, 38, 42], "term_frequency_threshold": [2, 20, 23, 31], "term_id": [8, 12, 22], "termid": [2, 3, 6, 7, 8, 12, 37], "termin": [12, 18], "terms_to_test": [2, 31], "test": [2, 3, 4, 5, 6, 7, 8, 9, 12, 13, 14, 16, 18, 20, 21, 25, 27, 29, 33, 35, 36, 37, 38, 41, 42], "test_nam": 12, "test_result": 12, "test_term_id": 12, "test_uniprot_json": 12, "testabl": 26, "testosteron": [8, 12, 22, 24], "text": 15, "textio": 15, "textiobas": 14, "textiowrapp": [14, 15], "tf_binding_site_vari": 12, "tfbs_ablat": 12, "tfbs_amplif": 12, "th": 3, "than": [2, 6, 12, 20, 23, 24, 27], "thank": 23, "thei": [13, 18, 21, 23, 24, 27, 31, 42], "them": [14, 26], "themselv": 13, "therebi": 2, "therefor": [2, 3, 6, 27, 31, 42], "thi": [2, 3, 6, 7, 8, 11, 12, 13, 14, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 30, 31, 32, 33, 35, 36, 37, 39, 40, 42], "thick": 27, "thin": 4, "thing": 26, "think": 31, "third": 24, "thorac": 27, "those": [32, 40], "thr161pro": 20, "thr223met": [20, 26], "thr72ly": 20, "three": [6, 12, 25, 31, 32], "three_prime_utr_vari": [12, 14], "threshold": [2, 6, 20, 23, 31], "thrombocyt": 27, "thrombosi": 27, "through": [14, 30], "thu": [3, 12, 26, 31, 32, 33], "thumb": [5, 20, 23, 26, 42], "thymu": 27, "thyroid": 27, "tight_layout": 30, "tild": 42, "time": [3, 12, 14, 19, 30, 31, 32], "timeout": 14, "tissu": 27, "to_displai": 12, "to_negative_strand": 13, "to_opposite_strand": 13, "to_positive_strand": 13, "to_str": 12, "todo": [16, 22, 25, 30, 35, 38, 41], "togeth": [12, 20, 22, 23, 24, 32], "tonsil": 27, "too": 27, "toolkit": [8, 14, 20, 30, 33], "top": [3, 7, 8, 12, 14, 20, 24, 26, 27, 30], "top_phenotype_count": 16, "top_variant_count": 16, "total": [3, 12, 20, 23, 24, 26, 31, 33], "total_patient_count": 12, "total_test": [3, 20, 23], "traceback": 31, "track": [7, 14], "tracker": [18, 38], "tract": 27, "trans_id": 12, "transcript": [3, 6, 12, 14, 16, 20, 22, 23, 24, 26, 29, 32, 40, 42], "transcript_abl": [12, 24], "transcript_amplif": 12, "transcript_coordin": 16, "transcript_id": [6, 12, 16, 20, 26], "transcriptannot": [0, 12, 14], "transcriptcoordin": [0, 12, 14, 16, 20, 30], "transcriptcoordinateservic": [0, 14], "transcriptinfoawar": [0, 12, 14], "transform": [14, 30], "transit": 13, "translat": 12, "transloc": [6, 12], "transmiss": 27, "transpos": [12, 13], "transpose_coordin": [12, 13], "trascript": 20, "travers": 14, "tree": 14, "tri": 14, "triphalang": [20, 23, 26], "triphosph": 27, "trp121gly": [20, 26], "true": [2, 6, 7, 11, 12, 13, 14, 16, 20, 23, 26, 29, 30, 31, 37, 42], "truli": 26, "truncat": 20, "try": [10, 11, 13, 14], "ttest_ind": 9, "tteststatist": [8, 9, 22], "tupl": [3, 6, 8, 12, 14, 20, 30, 31, 37, 42], "turn": [14, 30], "tutori": [18, 23, 29], "twice": 31, "two": [3, 4, 6, 8, 12, 13, 20, 23, 24, 25, 26, 30, 31, 32, 33, 34, 35, 38], "tx": [14, 16, 42], "tx_annot": 12, "tx_coordin": [16, 20, 26, 30], "tx_id": [6, 12, 14, 16, 20, 22, 23, 24, 26, 32, 36, 40, 42], "tx_servic": 26, "txc_servic": [20, 30], "type": [2, 3, 6, 7, 8, 11, 12, 13, 14, 16, 18, 29, 30, 31, 42], "typeerror": [11, 31], "typic": [3, 12, 23, 35, 38], "tyr136ter": [20, 26], "tyr291ter": [20, 26], "tyr342thrfster52": [20, 26], "tyr407ter": [20, 30], "u": [8, 9, 20, 21, 23, 26, 29, 30, 32, 33, 40, 42], "ucsc": 13, "ucsc_nam": 13, "ucum": 12, "umbil": 27, "umbilicu": 27, "unavail": 12, "under": [14, 20, 30, 31], "undergrowth": [20, 23], "underli": 42, "understand": 40, "undoubtedli": 30, "uniform": 6, "uninterest": 27, "union": [32, 40], "uniprot": [12, 14, 26], "uniprot_json": [12, 30], "uniprotproteinmetadataservic": [0, 12, 14], "uniqu": [2, 3, 5, 12, 20, 26], "unit": [12, 13, 19, 26], "unknown": [6, 12], "unknown_sex": [12, 39], "unless": [7, 20, 30, 31, 37], "unlik": [14, 27, 31], "unnecessari": 27, "unsur": 26, "untransl": [12, 30], "up": [20, 24, 31, 33, 37], "upper": [20, 23, 27], "upstream": 13, "upstream_gene_vari": [12, 14], "urat": 27, "uri": 12, "urin": 27, "urinari": 27, "urobilinogen": 27, "us": [2, 3, 6, 7, 8, 10, 11, 12, 13, 14, 15, 16, 18, 20, 22, 23, 24, 26, 28, 30, 31, 32, 35, 37, 38, 39, 42], "usabl": 3, "usag": [6, 7, 9, 16], "use_al": 31, "usealltermsmtcfilt": [1, 2, 31], "user": [12, 16, 18, 20, 26, 27, 30], "usual": [6, 12, 23, 27, 38], "utf": [14, 15], "util": [0, 5, 6, 23], "utr": [12, 30], "uvea": 27, "v": [18, 22, 24, 31, 36], "v2024": [20, 23, 24, 26, 30, 31, 33, 37, 42], "vagin": 27, "val153serfster21": 20, "val214glyfster12": 20, "val263met": [20, 26], "val267trpfster127": [20, 26], "valid": [14, 16, 20, 26, 31], "validation_polici": 14, "validation_runn": 14, "validationrunn": 14, "valu": [2, 3, 4, 8, 9, 12, 13, 14, 16, 20, 21, 23, 24, 29, 31], "valueerror": [5, 12, 13, 14], "valv": 27, "varcachingfunctionalannot": [0, 14], "vari": [6, 23], "variabl": [4, 10, 12, 23, 26, 35], "variant": [0, 3, 4, 6, 12, 14, 16, 18, 22, 23, 24, 28, 29, 31, 32, 35, 36, 38], "variant_class": [6, 12, 42], "variant_coordin": [12, 14], "variant_effect": [6, 12, 16, 20, 22, 23, 24, 32, 36, 40, 42], "variant_effect_count_by_tx": 12, "variant_fallback": 14, "variant_info": 12, "variant_kei": [6, 12, 14], "variant_key_of_interest": 42, "variant_loc": 16, "variant_locations_counted_absolut": 16, "variant_pred": 6, "variantannotationcach": [0, 14], "variantclass": [0, 6, 12, 42], "variantcoordin": [0, 12, 14, 28], "variantcoordinatefind": [0, 14], "varianteffect": [0, 6, 12, 14, 16, 20, 22, 23, 24, 32, 36, 40, 42], "variantformatt": [0, 16], "variantinfo": [0, 12], "variantinfoawar": [0, 12], "variantpred": [5, 6, 20, 22, 23, 24, 32, 35, 36, 38, 40, 42], "varianttranscriptproteinartist": 16, "varianttranscriptvisu": [0, 16], "variou": 13, "vascular": 27, "vasculatur": 27, "vc": [12, 14], "vcf": 12, "vcv000522858": 42, "ventricl": 23, "ventricular": [20, 23, 26, 27, 28], "vep": 14, "vepfunctionalannot": [0, 14], "verbos": 5, "veri": [2, 32, 40], "version": [4, 20, 23, 24, 33, 37, 40], "vessel": 27, "via": [7, 11, 14, 16, 30], "view": [0, 18, 20, 23, 26, 30], "viewabl": 16, "viewer": [20, 26], "virtual": 19, "vision": 27, "visit": 14, "visitor": 14, "visual": [16, 20, 24, 26, 27, 30], "vitamin": 27, "voic": 27, "vol": 12, "volum": [8, 12, 22, 24, 27], "vri": [8, 29, 41], "vvhgvsvariantcoordinatefind": [0, 14], "vvmulticoordinateservic": [0, 14, 20, 26, 30], "w": 14, "wa": [2, 3, 5, 7, 12, 14, 16, 20, 22, 23, 24, 26, 31, 33, 34, 37], "wai": [20, 21, 23, 26, 30], "waist": 27, "wall": 27, "want": [2, 8, 23, 24, 30, 31, 32, 36, 40, 42], "warn": [14, 20, 30], "warning_count": 14, "we": [2, 3, 6, 8, 12, 14, 15, 16, 18, 19, 20, 22, 23, 24, 26, 27, 30, 31, 32, 33, 35, 36, 37, 38, 39, 40, 42], "websit": 30, "weight": 27, "well": [6, 7, 12, 26, 30], "were": [3, 14, 16, 20, 23, 24, 26, 30, 37, 42], "what": [23, 34, 35, 40], "whatsoev": 6, "when": [2, 7, 12, 14, 30, 31, 37], "where": [2, 3, 12, 14, 20, 33, 36], "wherea": 26, "whether": [7, 8, 12, 18, 23, 26], "which": [2, 4, 5, 7, 12, 20, 26, 29, 30, 33, 40, 42], "while": [4, 23, 25, 38], "whisker": 24, "whitnei": [8, 9, 21, 26, 29, 33, 41], "who": [5, 7, 20, 24, 25, 32, 36], "whose": [12, 20, 23, 24, 37], "why": 2, "widespread": 26, "wikipedia": 26, "wilcoxon": 24, "wise": [20, 31], "with_cache_fold": 14, "with_strand": 13, "within": [12, 14, 20, 23, 24, 26], "without": [6, 23, 26], "word": [18, 27], "wordsmith": 22, "work": [2, 10, 14, 30], "workflow": 26, "worth": 2, "would": [2, 8, 12, 18, 23, 24, 25, 26, 31, 33, 42], "wrap": [8, 12, 30, 42], "wrap_scoring_funct": 8, "wrapper": [4, 9, 15], "write": [14, 15, 16], "written": 16, "wwox": 37, "x": [24, 36, 40], "x_1000001_1000027_": 12, "x_1000001_1000027_taaaaaaaaaaaaaaaaaaaaaaaaaa_t": 12, "x_12345_12345_c_g": [6, 12], "xm_": 14, "xu": 22, "y": [24, 36], "ye": [3, 5, 7, 37], "yet": 12, "yield": 31, "ylabel": 24, "ylim": 24, "you": [11, 12, 13, 26, 30, 31], "your": [13, 19, 30], "z": 36, "zero": [2, 6, 13, 28, 33, 36], "\u03b1": 12, "\u03c72": 4}, "titles": ["gpsea package", "gpsea.analysis package", "gpsea.analysis.mtc_filter package", "gpsea.analysis.pcats package", "gpsea.analysis.pcats.stats package", "gpsea.analysis.predicate package", "gpsea.analysis.predicate.genotype package", "gpsea.analysis.predicate.phenotype package", "gpsea.analysis.pscore package", "gpsea.analysis.pscore.stats package", "gpsea.config module", "gpsea.io module", "gpsea.model package", "gpsea.model.genome package", "gpsea.preprocessing package", "gpsea.util module", "gpsea.view package", "API reference", "GPSEA", "Installation", "Tutorial", "Statistical analyses", "Compare measurement values", "Compare genotype and phenotype groups", "Compare phenotype scores in genotype groups", "Genotype-Phenotype Correlations in Autosomal Recessive Diseases", "Cohort exploratory analysis", "General HPO terms", "Glossary", "User guide", "Input data", "Multiple-testing correction", "Allele count predicates", "De Vries Score", "Partition by a diagnosis", "Genotype Predicates", "Groups Predicate", "HPO predicate", "Predicates", "Partition by the sex of the individual", "Mode of Inheritance Predicates", "Phenotype Predicates", "Variant Predicates"], "titleterms": {"2x2": 31, "2x3": 31, "The": [20, 33], "abnorm": [31, 33], "across": 26, "all": [20, 31, 37, 40], "allel": [20, 28, 32], "altern": 30, "an": 28, "analys": 21, "analysi": [1, 2, 3, 4, 5, 6, 7, 8, 9, 20, 22, 23, 24, 26, 31], "api": [17, 30], "ar": 31, "associ": 23, "autosom": [25, 40], "background": 31, "biallel": [32, 40], "block": 42, "build": 42, "categori": [32, 40], "chang": 28, "child": 31, "choos": [30, 31], "code": 33, "cohort": [20, 22, 23, 24, 26, 30, 37], "compar": [22, 23, 24, 32, 40], "complex": 42, "condit": 42, "config": 10, "configur": [22, 23, 24, 30], "congenit": 33, "content": [21, 29, 35, 38, 41], "control": 30, "coordin": 30, "correct": [23, 31], "correl": 25, "count": [31, 32], "creat": 30, "creator": 30, "data": 30, "de": 33, "delai": 33, "descend": 31, "development": 33, "diagnosi": 34, "diseas": 25, "distribut": [20, 26, 30], "domain": 26, "domin": 40, "dump": 30, "dysmorph": 33, "enter": 30, "exact": 23, "exampl": [22, 23, 24, 30, 32, 36, 37, 40], "exclud": 31, "explor": [20, 26], "exploratori": 26, "facial": 33, "featur": [30, 33], "feedback": 18, "fet": 23, "fetch": 30, "filter": 31, "final": [22, 23, 24, 33], "fisher": 23, "frameshift": [20, 23, 32], "from": 30, "ga4gh": 30, "galleri": 38, "gener": [27, 31], "genom": 13, "genotyp": [6, 20, 22, 23, 24, 25, 30, 31, 32, 35], "get": 30, "glossari": 28, "gpsea": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 18, 31], "group": [23, 24, 31, 36], "growth": 33, "guid": 29, "ha": 31, "hmf01": 31, "hmf02": 31, "hmf03": 31, "hmf05": 31, "hmf06": 31, "hmf07": 31, "hmf08": 31, "hpo": [20, 24, 27, 31, 37], "ident": 31, "implement": 31, "individu": 39, "inherit": 40, "input": 30, "instal": 19, "interact": 26, "interest": 30, "invert": 42, "io": 11, "json": 30, "later": 30, "latest": 19, "least": 32, "length": 28, "level": 31, "literatur": 18, "load": [20, 22, 23, 24, 30], "mann": 24, "manual": 30, "measur": 22, "missens": [20, 32], "missing_implies_phenotype_exclud": 37, "mode": 40, "model": [12, 13], "modul": [10, 11, 15], "monoallel": [32, 40], "more": [31, 38], "mtc": 31, "mtc_filter": 2, "multipl": [23, 31], "need": 38, "neither": 31, "non": 33, "nor": 31, "observ": 31, "occur": 31, "one": [31, 32], "onset": 33, "packag": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 12, 13, 14, 16], "pars": 30, "partit": [32, 34, 39, 40], "path": 28, "pcat": [3, 4], "persist": 30, "phenopacket": 30, "phenotyp": [7, 20, 22, 23, 24, 25, 30, 31, 37, 41], "plot": [20, 26, 30], "postnat": 33, "power": 26, "predic": [5, 6, 7, 20, 22, 23, 24, 32, 35, 36, 37, 38, 40, 41, 42], "prenat": 33, "prepar": 20, "preprocess": 14, "procedur": 31, "protein": [20, 26, 30], "provid": 30, "pscore": [8, 9], "qualiti": 30, "rare": 31, "recess": [25, 40], "refer": [17, 28], "releas": 19, "respect": [20, 26], "rest": [23, 30], "retard": 33, "rule": 28, "run": 19, "score": [22, 24, 33], "scorer": 33, "section": 33, "sequenc": [20, 26], "sex": 39, "shape": 4, "show": 20, "showcas": 30, "skip": 31, "sourc": 30, "specifi": 31, "stabl": 19, "standard": 30, "stat": [4, 9], "statist": [21, 22, 23, 24, 26], "strategi": 31, "submodul": 0, "subpackag": [0, 1, 3, 5, 8, 12], "subset": 40, "summar": 20, "summari": [20, 26], "support": 4, "tbx5": [20, 23], "term": [27, 31], "test": [19, 22, 23, 24, 26, 31], "than": 31, "transcript": 30, "true": 28, "tutori": [20, 25], "type": 26, "u": 24, "underpow": 31, "uniprot": 30, "us": [33, 40], "user": 29, "util": 15, "v": [20, 23, 32, 40], "valid": 30, "valu": 22, "variant": [20, 26, 30, 40, 42], "veri": 31, "view": 16, "vri": 33, "which": 31, "whitnei": 24}})