

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Tutorial &mdash; GPSEA 0.7.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="_static/gpsea.css?v=547657ed" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=fe7df9b0"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="_static/copybutton.js?v=6dbb43f8"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="User guide" href="user-guide/index.html" />
    <link rel="prev" title="Installation" href="installation.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Tutorial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#the-analysis">The analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#load-hpo">Load HPO</a></li>
<li class="toctree-l3"><a class="reference internal" href="#prepare-cohort">Prepare cohort</a></li>
<li class="toctree-l3"><a class="reference internal" href="#explore-cohort">Explore cohort</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#show-cohort-summary">Show cohort summary</a></li>
<li class="toctree-l4"><a class="reference internal" href="#plot-distribution-of-variants-with-respect-to-the-protein-sequence">Plot distribution of variants with respect to the protein sequence</a></li>
<li class="toctree-l4"><a class="reference internal" href="#summarize-all-variant-alleles">Summarize all variant alleles</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#prepare-genotype-and-phenotype-predicates">Prepare genotype and phenotype predicates</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Tutorial</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/tutorial.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="tutorial">
<span id="id1"></span><h1>Tutorial<a class="headerlink" href="#tutorial" title="Link to this heading"></a></h1>
<p>Here we demonstrate an end-to-end genotype-phenotype analysis with GPSEA.
The tutorial illustrates just one of the many ways GPSEA can be used to characterize genotype-phenotype correlations.
The <a class="reference internal" href="user-guide/index.html#user-guide"><span class="std std-ref">User guide</span></a> contains details about additional methods and functionalities.</p>
<p>The tutorial presents an analysis of a cohort of individuals with pathogenic variants in <em>TBX5</em> leading to
<a class="reference external" href="https://omim.org/entry/142900">Holt-Oram syndrome MIM:142900</a>.</p>
<p>Holt-Oram syndrome is an autosomal dominant disorder characterized by
upper limb defects, congenital heart defects, and arrhythmias (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/38336121/">PMID:38336121</a>).
It has been observed in the literature that congenital defects of the ventricular and atrial septum are more
common in the truncating than in the missense variants (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30552424/">PMID:30552424</a>).
Additionally, upper limb defects are more frequent in patients with protein-truncating variants (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/38336121/">PMID:38336121</a>).</p>
<p>To perform the analysis, we curated the literature and created on <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/35705716/">GA4GH phenopacket</a> for each
affected individual. The phenopackets are made available in <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a>.</p>
<section id="the-analysis">
<h2>The analysis<a class="headerlink" href="#the-analysis" title="Link to this heading"></a></h2>
<p>For the analysis, the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/refseq/MANE/">MANE</a> transcript
(i.e., the “main” biomedically relevant transcript of a gene) should be chosen unless
there is a specific reason not to (which should occur rarely if at all).</p>
<p>In the case of <em>TBX5</em> the MANE transcript is <cite>NM_181486.4</cite>. Note that the trascript identifier (<cite>NM_181486</cite>) and the version (<cite>4</cite>) are both required.
A good way to find the MANE transcript is to search on the gene symbol (e.g., <em>TBX5</em>) in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a> and to
choose a variant that is specifically located in the gene. The MANE transcript will be displayed here (e.g., <a class="reference external" href="https://www.ncbi.nlm.nih.gov/clinvar/variation/495227/">NM_181486.4(TBX5):c.1221C&gt;G (p.Tyr407Ter)</a>).</p>
<p>We additionally need the corresponding protein identifier.
A good way to find this is to search on the transcript id in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/">NCBI Nucleotide</a>.
In our case, search on <cite>NM_181486.4</cite> will bring us to <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/NM_181486.4">this page</a>.
If we search within this page for <cite>“NP_”</cite>, this will bring us to the
corresponding protein accession <cite>NP_852259.1</cite>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_name</span> <span class="o">=</span> <span class="s1">&#39;TBX5&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_181486.4&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">px_id</span> <span class="o">=</span> <span class="s1">&#39;NP_852259.1&#39;</span>
</pre></div>
</div>
<section id="load-hpo">
<h3>Load HPO<a class="headerlink" href="#load-hpo" title="Link to this heading"></a></h3>
<p>GPSEA needs HPO to do the analysis.
We use HPO toolkit to load HPO version <cite>v2024-07-01</cite>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Use the latest HPO release by omitting the <cite>release</cite> option.</p>
</div>
</section>
<section id="prepare-cohort">
<h3>Prepare cohort<a class="headerlink" href="#prepare-cohort" title="Link to this heading"></a></h3>
<p>Now we will load the samples to analyze. We will use the cohort of 156 individuals with mutations in <em>TBX5</em>
whose clinical signs and symptoms were encoded into HPO terms
and stored in <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">ppktstore.registry</span> <span class="kn">import</span> <span class="n">configure_phenopacket_registry</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">phenopacket_registry</span> <span class="o">=</span> <span class="n">configure_phenopacket_registry</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">phenopacket_registry</span><span class="o">.</span><span class="n">open_phenopacket_store</span><span class="p">(</span><span class="s2">&quot;0.1.20&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">ps</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">phenopackets</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">ps</span><span class="o">.</span><span class="n">iter_cohort_phenopackets</span><span class="p">(</span><span class="n">cohort_name</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">)</span>
<span class="go">156</span>
</pre></div>
</div>
<p>We loaded 156 phenopackets which need further preprocessing to prepare for the analysis.
We will compute functional annotations for the mutations and then include the individuals into
a <a class="reference internal" href="apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span><span class="p">,</span> <span class="n">load_phenopackets</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">validation</span> <span class="o">=</span> <span class="n">load_phenopackets</span><span class="p">(</span>  
<span class="gp">... </span>    <span class="n">phenopackets</span><span class="o">=</span><span class="n">phenopackets</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">cohort_creator</span><span class="o">=</span><span class="n">cohort_creator</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="go">Individuals Processed: ...</span>
</pre></div>
</div>
<p>and we will check that there are no Q/C issues:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">validation</span><span class="o">.</span><span class="n">summarize</span><span class="p">()</span>  
<span class="go">Validated under none policy</span>
<span class="go">No errors or warnings were found</span>
</pre></div>
</div>
<p>We loaded the patient data into a <cite>cohort</cite> which is ready for the next steps.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>Here we show how to create a <a class="reference internal" href="apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a> from phenopackets.
See <a class="reference internal" href="user-guide/input-data.html#input-data"><span class="std std-ref">Input data</span></a> section to learn how to create a cohort from another inputs.</p>
</div>
</section>
<section id="explore-cohort">
<h3>Explore cohort<a class="headerlink" href="#explore-cohort" title="Link to this heading"></a></h3>
<p>GPSEA helps with gaining insight into the cohort by providing</p>
<section id="show-cohort-summary">
<h4>Show cohort summary<a class="headerlink" href="#show-cohort-summary" title="Link to this heading"></a></h4>
<p>The summary report provides an overview about the HPO terms, variants, diseases, and variant effects that occur most frequently:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">CohortViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">viewer</span> <span class="o">=</span> <span class="n">CohortViewer</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span> <span class="o">=</span> <span class="n">viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span> <span class="n">transcript_id</span><span class="o">=</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span>  
</pre></div>
</div>
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Cohort</title>
    <style>
    table {
      border-collapse: collapse;
      margin: 25px 0;
      font-size: 0.9em;
      font-family: sans-serif;
      min-width: 400px;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
    }

    .table .column-1 {
      text-align: left;
    }
    th {
      background-color: LightSkyBlue;
      border: 1px solid #dddddd;
      text-align: left;
      padding: 2px;
      font-weight: bold;
      font-size: 120%;
    }

    tr {
      border: 1px solid #dddddd;
    }

    td {
      padding: 2px;
      font-weight: bold;
    }

    tr:nth-child(even) {
      background-color: #f2f2f2;
    }

    .table td, tr {
      text-align: right;
    }

    .lft {
      text-align: left;
    }

    div {
      display: block;
      width: 100%;
      padding-right: 1%;
      margin-bottom: 20px; /* Add margin to separate divs */
    }

    caption {
      caption-side: top;
      text-align: left;
      padding-bottom: 10px;
      font-weight: bold;
    }
  </style>
</head>

<body>
  <h1>GPSEA cohort analysis</h1>
  <p>Successfully loaded 156 individuals. 46 were recorded as male, 56 as female,
      and 54 as unknown sex. 156 were reported to be alive at the time of last
      encounter, and 0 were deceased. 1 had disease onset information
      and 51 had information about the age at last encounter. In the cohort,
      there were a total of 117 HPO terms used for annotation.
    
  </p>
  
  <p>No errors encountered.</p>
  

    <table>
    <caption style="color: black;">
        <h3>Top 10 HPO Terms</h3>
        A total of 117 HPO terms were used to annotated the cohort.
    </caption>
        <tbody>
            <tr class="strng">
                <th class="lft">HPO Term</th>
                <th>ID</th>
                <th>Seen in <em>n</em> individuals</th>
            </tr>
            
            <tr>
                <td class="lft">Atrial septal defect</td>
                <td>HP:0001631</td>
                <td>50</td>
                </tr>
            
            <tr>
                <td class="lft">Ventricular septal defect</td>
                <td>HP:0001629</td>
                <td>41</td>
                </tr>
            
            <tr>
                <td class="lft">Hypoplasia of the radius</td>
                <td>HP:0002984</td>
                <td>40</td>
                </tr>
            
            <tr>
                <td class="lft">Triphalangeal thumb</td>
                <td>HP:0001199</td>
                <td>36</td>
                </tr>
            
            <tr>
                <td class="lft">Short thumb</td>
                <td>HP:0009778</td>
                <td>32</td>
                </tr>
            
            <tr>
                <td class="lft">Absent thumb</td>
                <td>HP:0009777</td>
                <td>32</td>
                </tr>
            
            <tr>
                <td class="lft">Abnormal carpal morphology</td>
                <td>HP:0001191</td>
                <td>30</td>
                </tr>
            
            <tr>
                <td class="lft">Secundum atrial septal defect</td>
                <td>HP:0001684</td>
                <td>27</td>
                </tr>
            
            <tr>
                <td class="lft">Absent radius</td>
                <td>HP:0003974</td>
                <td>15</td>
                </tr>
            
            <tr>
                <td class="lft">Cardiac conduction abnormality</td>
                <td>HP:0031546</td>
                <td>14</td>
                </tr>
            
                </tbody>
    </table>

  

    <table>
    <caption style="color: black;">
        <h3>Top 10 Variants</h3>
        Variants are shown according to NM_181486.4. A total of 53 unique variants were identified in the cohort.
    </caption>
            <tbody>
                <tr class="strng">
                    <th>Seen in <em>n</em> individuals</th>
                    <th class="lft">Variant key</th>
                    <th>Variant Name</th>
                    <th>Protein Variant</th>
                    <th>Variant Class</th>
                </tr>
                
                <tr>
                    <td>22</td>
                    <td class="lft">12_114385521_114385521_C_T</td>
                    <td>c.710G>A</td>
                    <td>p.Arg237Gln</td>
                    <td>MISSENSE_VARIANT</td>
                </tr>
                
                <tr>
                    <td>20</td>
                    <td class="lft">12_114401830_114401830_C_T</td>
                    <td>c.238G>A</td>
                    <td>p.Gly80Arg</td>
                    <td>MISSENSE_VARIANT</td>
                </tr>
                
                <tr>
                    <td>8</td>
                    <td class="lft">12_114385563_114385563_G_A</td>
                    <td>c.668C>T</td>
                    <td>p.Thr223Met</td>
                    <td>MISSENSE_VARIANT</td>
                </tr>
                
                <tr>
                    <td>6</td>
                    <td class="lft">12_114398675_114398675_G_T</td>
                    <td>c.408C>A</td>
                    <td>p.Tyr136Ter</td>
                    <td>STOP_GAINED</td>
                </tr>
                
                <tr>
                    <td>5</td>
                    <td class="lft">12_114399514_114399514_A_C</td>
                    <td>c.361T>G</td>
                    <td>p.Trp121Gly</td>
                    <td>MISSENSE_VARIANT, SPLICE_REGION_VARIANT</td>
                </tr>
                
                <tr>
                    <td>5</td>
                    <td class="lft">12_114398682_114398682_C_CG</td>
                    <td>c.400dup</td>
                    <td>p.Arg134ProfsTer49</td>
                    <td>FRAMESHIFT_VARIANT</td>
                </tr>
                
                <tr>
                    <td>5</td>
                    <td class="lft">12_114403792_114403792_C_CG</td>
                    <td>c.106_107insC</td>
                    <td>p.Ser36ThrfsTer25</td>
                    <td>FRAMESHIFT_VARIANT</td>
                </tr>
                
                <tr>
                    <td>4</td>
                    <td class="lft">12_114385474_114385474_A_G</td>
                    <td>c.755+2T>C</td>
                    <td>None</td>
                    <td>SPLICE_DONOR_VARIANT</td>
                </tr>
                
                <tr>
                    <td>4</td>
                    <td class="lft">12_114398656_114398656_C_CG</td>
                    <td>c.426dup</td>
                    <td>p.Ala143ArgfsTer40</td>
                    <td>FRAMESHIFT_VARIANT</td>
                </tr>
                
                <tr>
                    <td>4</td>
                    <td class="lft">12_114403798_114403798_G_GC</td>
                    <td>c.100dup</td>
                    <td>p.Ala34GlyfsTer27</td>
                    <td>FRAMESHIFT_VARIANT</td>
                </tr>
                
            </tbody>
        </table>
    <table>
    <caption style="color: black;">
        <h3>Diseases</h3>
    </caption>
        <tbody>
            <tr class="strng">
                <th class="lft">Disease Name</th>
                <th >Disease ID</th>
                <th>Annotation Count</th>
              </tr>
              
              <tr>
                <td class="lft">Holt-Oram syndrome</td>
                <td>OMIM:142900</td>
                <td>156</td>
                </tr>
            
           </tbody>
    </table>
    
    <table>
    <caption style="color: black;">
        <h3>Variant categories for NM_181486.4</h3>
    </caption>
        <tbody>
            <tr class="strng">
                <th class="lft">Variant effect</th>
                <th>Annotation Count</th>
                <th>Percent</th>
              </tr>
              
              <tr>
                <td class="lft">MISSENSE_VARIANT</td>
                <td>85</td>
               <td>50%</td>
                </tr>
            
              <tr>
                <td class="lft">FRAMESHIFT_VARIANT</td>
                <td>38</td>
               <td>22%</td>
                </tr>
            
              <tr>
                <td class="lft">STOP_GAINED</td>
                <td>19</td>
               <td>11%</td>
                </tr>
            
              <tr>
                <td class="lft">SPLICE_REGION_VARIANT</td>
                <td>10</td>
               <td>6%</td>
                </tr>
            
              <tr>
                <td class="lft">SPLICE_DONOR_VARIANT</td>
                <td>7</td>
               <td>4%</td>
                </tr>
            
              <tr>
                <td class="lft">INFRAME_INSERTION</td>
                <td>2</td>
               <td>1%</td>
                </tr>
            
              <tr>
                <td class="lft">STOP_RETAINED_VARIANT</td>
                <td>2</td>
               <td>1%</td>
                </tr>
            
              <tr>
                <td class="lft">SPLICE_ACCEPTOR_VARIANT</td>
                <td>2</td>
               <td>1%</td>
                </tr>
            
              <tr>
                <td class="lft">SPLICE_DONOR_5TH_BASE_VARIANT</td>
                <td>2</td>
               <td>1%</td>
                </tr>
            
              <tr>
                <td class="lft">INTRON_VARIANT</td>
                <td>2</td>
               <td>1%</td>
                </tr>
            
              <tr>
                <td class="lft">INFRAME_DELETION</td>
                <td>1</td>
               <td>1%</td>
                </tr>
            
           </tbody>
    </table>
  


</body>
</html></section>
<section id="plot-distribution-of-variants-with-respect-to-the-protein-sequence">
<h4>Plot distribution of variants with respect to the protein sequence<a class="headerlink" href="#plot-distribution-of-variants-with-respect-to-the-protein-sequence" title="Link to this heading"></a></h4>
<p>Now we can show the distribution of variants with respect to the encoded protein.
We first obtain <cite>tx_coordinates</cite> (<a class="reference internal" href="apidocs/gpsea.model.html#gpsea.model.TranscriptCoordinates" title="gpsea.model.TranscriptCoordinates"><code class="xref py py-class docutils literal notranslate"><span class="pre">TranscriptCoordinates</span></code></a>)
and <cite>protein_meta</cite> (<a class="reference internal" href="apidocs/gpsea.model.html#gpsea.model.ProteinMetadata" title="gpsea.model.ProteinMetadata"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinMetadata</span></code></a>)
with information about the transcript and protein “anatomy”:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model.genome</span> <span class="kn">import</span> <span class="n">GRCh38</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">configure_protein_metadata_service</span><span class="p">,</span> <span class="n">VVMultiCoordinateService</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">txc_service</span> <span class="o">=</span> <span class="n">VVMultiCoordinateService</span><span class="p">(</span><span class="n">genome_build</span><span class="o">=</span><span class="n">GRCh38</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pms</span> <span class="o">=</span> <span class="n">configure_protein_metadata_service</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_coordinates</span> <span class="o">=</span> <span class="n">txc_service</span><span class="o">.</span><span class="n">fetch</span><span class="p">(</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">protein_meta</span> <span class="o">=</span> <span class="n">pms</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="n">px_id</span><span class="p">)</span>
</pre></div>
</div>
<p>and we follow with plotting the diagram of the mutations on the protein:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">ProteinVisualizer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span> <span class="mi">8</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">visualizer</span> <span class="o">=</span> <span class="n">ProteinVisualizer</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">visualizer</span><span class="o">.</span><span class="n">draw_protein_diagram</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">tx_coordinates</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">protein_meta</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/tbx5_protein_diagram.png"><img alt="TBX5 protein diagram" class="align-center" src="_images/tbx5_protein_diagram.png" style="width: 600px;" />
</a>
</section>
<section id="summarize-all-variant-alleles">
<span id="show-cohort-variants"></span><h4>Summarize all variant alleles<a class="headerlink" href="#summarize-all-variant-alleles" title="Link to this heading"></a></h4>
<p>We can prepare a table of all variant alleles that occurr in the cohort.
Each table row corresponds to a single allele and lists the variant key,
the predicted effect on the transcript (<em>cDNA</em>) and protein of interest,
the variant effects, and the number of patients who present
with one or more variant alleles (<em>Count</em>):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">CohortVariantViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">viewer</span> <span class="o">=</span> <span class="n">CohortVariantViewer</span><span class="p">(</span><span class="n">tx_id</span><span class="o">=</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span> <span class="o">=</span> <span class="n">viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">report</span>  
</pre></div>
</div>
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Cohort</title>
    <style>
    table {
      border-collapse: collapse;
      margin: 25px 0;
      font-size: 0.9em;
      font-family: sans-serif;
      min-width: 400px;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
    }

    .table .column-1 {
      text-align: left;
    }
    th {
      background-color: LightSkyBlue;
      border: 1px solid #dddddd;
      text-align: left;
      padding: 2px;
      font-weight: bold;
      font-size: 120%;
    }

    tr {
      border: 1px solid #dddddd;
    }

    td {
      padding: 2px;
      font-weight: bold;
    }

    tr:nth-child(even) {
      background-color: #f2f2f2;
    }

    .table td, tr {
      text-align: right;
    }

    .lft {
      text-align: left;
    }

    div {
      display: block;
      width: 100%;
      padding-right: 1%;
      margin-bottom: 20px; /* Add margin to separate divs */
    }

    caption {
      caption-side: top;
      text-align: left;
      padding-bottom: 10px;
      font-weight: bold;
    }
  </style>
</head>

<body>
  <h1>GPSEA cohort analysis: All variant alleles</h1>


    <table>
    <caption style="color: black;">
        <h3>Variant alleles</h3>
        A total of 53 unique alleles were identified in the cohort.
    </caption>


        <tbody>
            <tr class="strng">
                <th class="lft">Variant key</th>
                <th>Variant (cDNA)</th>
                <th>Variant (protein)</th>
                <th>Effects</th>
                <th>Count</th>
            </tr>
            
            <tr>
                <td class="lft">12_114385521_114385521_C_T</td>
                <td>c.710G>A</td>
                <td>p.Arg237Gln</td>
                <td>missense</td>
                <td>22</td>
                </tr>
            
            <tr>
                <td class="lft">12_114401830_114401830_C_T</td>
                <td>c.238G>A</td>
                <td>p.Gly80Arg</td>
                <td>missense</td>
                <td>20</td>
                </tr>
            
            <tr>
                <td class="lft">12_114385563_114385563_G_A</td>
                <td>c.668C>T</td>
                <td>p.Thr223Met</td>
                <td>missense</td>
                <td>8</td>
                </tr>
            
            <tr>
                <td class="lft">12_114398675_114398675_G_T</td>
                <td>c.408C>A</td>
                <td>p.Tyr136Ter</td>
                <td>stop gained</td>
                <td>6</td>
                </tr>
            
            <tr>
                <td class="lft">12_114399514_114399514_A_C</td>
                <td>c.361T>G</td>
                <td>p.Trp121Gly</td>
                <td>missense, splice region</td>
                <td>5</td>
                </tr>
            
            <tr>
                <td class="lft">12_114403792_114403792_C_CG</td>
                <td>c.106_107insC</td>
                <td>p.Ser36ThrfsTer25</td>
                <td>frameshift</td>
                <td>5</td>
                </tr>
            
            <tr>
                <td class="lft">12_114398682_114398682_C_CG</td>
                <td>c.400dup</td>
                <td>p.Arg134ProfsTer49</td>
                <td>frameshift</td>
                <td>5</td>
                </tr>
            
            <tr>
                <td class="lft">12_114385474_114385474_A_G</td>
                <td>c.755+2T>C</td>
                <td>None</td>
                <td>splice donor</td>
                <td>4</td>
                </tr>
            
            <tr>
                <td class="lft">12_114398656_114398656_C_CG</td>
                <td>c.426dup</td>
                <td>p.Ala143ArgfsTer40</td>
                <td>frameshift</td>
                <td>4</td>
                </tr>
            
            <tr>
                <td class="lft">12_114366360_114366360_C_T</td>
                <td>c.787G>A</td>
                <td>p.Val263Met</td>
                <td>missense</td>
                <td>4</td>
                </tr>
            
            <tr>
                <td class="lft">12_114385522_114385522_G_A</td>
                <td>c.709C>T</td>
                <td>p.Arg237Trp</td>
                <td>missense</td>
                <td>4</td>
                </tr>
            
            <tr>
                <td class="lft">12_114403798_114403798_G_GC</td>
                <td>c.100dup</td>
                <td>p.Ala34GlyfsTer27</td>
                <td>frameshift</td>
                <td>4</td>
                </tr>
            
            <tr>
                <td class="lft">12_114399613_114399613_T_A</td>
                <td>c.262A>T</td>
                <td>p.Lys88Ter</td>
                <td>stop gained</td>
                <td>3</td>
                </tr>
            
            <tr>
                <td class="lft">12_114401827_114401827_T_A</td>
                <td>c.241A>T</td>
                <td>p.Arg81Trp</td>
                <td>missense, splice region</td>
                <td>3</td>
                </tr>
            
            <tr>
                <td class="lft">12_114366312_114366312_G_A</td>
                <td>c.835C>T</td>
                <td>p.Arg279Ter</td>
                <td>stop gained</td>
                <td>3</td>
                </tr>
            
            <tr>
                <td class="lft">12_114366366_114366366_T_A</td>
                <td>c.781A>T</td>
                <td>p.Ser261Cys</td>
                <td>missense</td>
                <td>3</td>
                </tr>
            
            <tr>
                <td class="lft">12_114403798_114403799_GC_G</td>
                <td>c.100del</td>
                <td>p.Ala34ProfsTer32</td>
                <td>frameshift</td>
                <td>3</td>
                </tr>
            
            <tr>
                <td class="lft">12_114385475_114385475_C_T</td>
                <td>c.755+1G>A</td>
                <td>None</td>
                <td>splice donor</td>
                <td>3</td>
                </tr>
            
            <tr>
                <td class="lft">12_114401853_114401853_G_T</td>
                <td>c.215C>A</td>
                <td>p.Thr72Lys</td>
                <td>missense</td>
                <td>3</td>
                </tr>
            
            <tr>
                <td class="lft">12_114385521_114385521_C_G</td>
                <td>c.710G>C</td>
                <td>p.Arg237Pro</td>
                <td>missense</td>
                <td>2</td>
                </tr>
            
            <tr>
                <td class="lft">12_114398568_114398568_C_A</td>
                <td>c.510+5G>T</td>
                <td>None</td>
                <td>splice donor 5th base, intronic</td>
                <td>2</td>
                </tr>
            
            <tr>
                <td class="lft">12_114366207_114366208_GC_G</td>
                <td>c.939del</td>
                <td>p.Gln315ArgfsTer79</td>
                <td>frameshift</td>
                <td>2</td>
                </tr>
            
            <tr>
                <td class="lft">12_114366274_114366274_G_T</td>
                <td>c.873C>A</td>
                <td>p.Tyr291Ter</td>
                <td>stop gained</td>
                <td>2</td>
                </tr>
            
            <tr>
                <td class="lft">12_114366267_114366267_C_A</td>
                <td>c.880G>T</td>
                <td>p.Glu294Ter</td>
                <td>stop gained</td>
                <td>2</td>
                </tr>
            
            <tr>
                <td class="lft">12_114398578_114398579_CA_C</td>
                <td>c.504del</td>
                <td>p.Phe168LeufsTer6</td>
                <td>frameshift</td>
                <td>2</td>
                </tr>
            
            <tr>
                <td class="lft">12_114394762_114394763_CA_C</td>
                <td>c.641del</td>
                <td>p.Val214GlyfsTer12</td>
                <td>frameshift</td>
                <td>2</td>
                </tr>
            
            <tr>
                <td class="lft">12_114398666_114398667_TG_T</td>
                <td>c.416del</td>
                <td>p.Pro139GlnfsTer11</td>
                <td>frameshift</td>
                <td>2</td>
                </tr>
            
            <tr>
                <td class="lft">12_114403754_114403754_G_T</td>
                <td>c.145C>A</td>
                <td>p.Gln49Lys</td>
                <td>missense, splice region</td>
                <td>2</td>
                </tr>
            
            <tr>
                <td class="lft">12_114385550_114385550_A_AATTATTCTCAG</td>
                <td>c.680_681insCTGAGAATAAT</td>
                <td>p.Ile227_Glu228insTer</td>
                <td>inframe insertion, stop retainined</td>
                <td>2</td>
                </tr>
            
            <tr>
                <td class="lft">12_114366348_114366349_CT_C</td>
                <td>c.798del</td>
                <td>p.Val267TrpfsTer127</td>
                <td>frameshift</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114399559_114399559_T_C</td>
                <td>c.316A>G</td>
                <td>p.Ile106Val</td>
                <td>missense</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114356064_114356065_TA_T</td>
                <td>c.1024del</td>
                <td>p.Tyr342ThrfsTer52</td>
                <td>frameshift</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114355755_114355756_TG_T</td>
                <td>c.1333del</td>
                <td>p.His445MetfsTer137</td>
                <td>frameshift</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114398632_114398632_G_A</td>
                <td>c.451C>T</td>
                <td>p.Gln151Ter</td>
                <td>stop gained</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114401907_114401907_A_G</td>
                <td>c.161T>C</td>
                <td>p.Ile54Thr</td>
                <td>missense</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114355723_114355723_G_A</td>
                <td>c.1366C>T</td>
                <td>p.Gln456Ter</td>
                <td>stop gained</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114398602_114398602_T_G</td>
                <td>c.481A>C</td>
                <td>p.Thr161Pro</td>
                <td>missense</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114398708_114398709_GC_G</td>
                <td>c.374del</td>
                <td>p.Gly125AlafsTer25</td>
                <td>frameshift</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114385553_114385553_C_A</td>
                <td>c.678G>T</td>
                <td>p.Lys226Asn</td>
                <td>missense</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114399633_114399633_C_G</td>
                <td>c.243-1G>C</td>
                <td>None</td>
                <td>splice acceptor</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114401873_114401874_TA_T</td>
                <td>c.194del</td>
                <td>p.Leu65GlnfsTer10</td>
                <td>frameshift</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114399594_114399594_A_C</td>
                <td>c.281T>G</td>
                <td>p.Leu94Arg</td>
                <td>missense</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114399622_114399622_G_T</td>
                <td>c.253C>A</td>
                <td>p.Pro85Thr</td>
                <td>missense</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114394743_114394746_TGTG_T</td>
                <td>c.658_660del</td>
                <td>p.His220del</td>
                <td>inframe deletion</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114394820_114394820_C_G</td>
                <td>c.584G>C</td>
                <td>p.Gly195Ala</td>
                <td>missense</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114401846_114401846_C_G</td>
                <td>c.222G>C</td>
                <td>p.Met74Ile</td>
                <td>missense</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114366241_114366242_CT_C</td>
                <td>c.905del</td>
                <td>p.Gln302ArgfsTer92</td>
                <td>frameshift</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114355784_114355785_CA_C</td>
                <td>c.1304del</td>
                <td>p.Leu435ArgfsTer147</td>
                <td>frameshift</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114398626_114398627_CG_C</td>
                <td>c.456del</td>
                <td>p.Val153SerfsTer21</td>
                <td>frameshift</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114403859_114403859_G_T</td>
                <td>c.40C>A</td>
                <td>p.Pro14Thr</td>
                <td>missense</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114399625_114399629_ACATC_A</td>
                <td>c.246_249del</td>
                <td>p.Met83PhefsTer6</td>
                <td>frameshift</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114394817_114394817_G_C</td>
                <td>c.587C>G</td>
                <td>p.Ser196Ter</td>
                <td>stop gained</td>
                <td>1</td>
                </tr>
            
            <tr>
                <td class="lft">12_114401921_114401921_C_G</td>
                <td>c.148-1G>C</td>
                <td>None</td>
                <td>splice acceptor</td>
                <td>1</td>
                </tr>
            
                </tbody>
    </table>
<!--
   <table>
    <caption style="color: black;">
        <h3>Variant category counts for </h3>
    </caption>
        <tbody>
            <tr class="strng">
                <th class="lft">Variant effect</th>
                <th>Annotation Count</th>
              </tr>
              
              <tr>
                <td class="lft">missense</td>
                <td>85</td>
                </tr>
            
              <tr>
                <td class="lft">frameshift</td>
                <td>38</td>
                </tr>
            
              <tr>
                <td class="lft">stop gained</td>
                <td>19</td>
                </tr>
            
              <tr>
                <td class="lft">splice region</td>
                <td>10</td>
                </tr>
            
              <tr>
                <td class="lft">splice donor</td>
                <td>7</td>
                </tr>
            
              <tr>
                <td class="lft">splice donor 5th base</td>
                <td>2</td>
                </tr>
            
              <tr>
                <td class="lft">intronic</td>
                <td>2</td>
                </tr>
            
              <tr>
                <td class="lft">splice acceptor</td>
                <td>2</td>
                </tr>
            
              <tr>
                <td class="lft">inframe insertion</td>
                <td>2</td>
                </tr>
            
              <tr>
                <td class="lft">stop retainined</td>
                <td>2</td>
                </tr>
            
              <tr>
                <td class="lft">inframe deletion</td>
                <td>1</td>
                </tr>
            
           </tbody>
    </table>
 -->


</body>
</html></section>
</section>
<section id="prepare-genotype-and-phenotype-predicates">
<h3>Prepare genotype and phenotype predicates<a class="headerlink" href="#prepare-genotype-and-phenotype-predicates" title="Link to this heading"></a></h3>
<p>We will create a predicate to bin patients into group
depending on presence of a single allele of a missense or frameshift variant to test
if there is a difference between frameshift and non-frameshift variants
in the individuals of the <em>TBX5</em> cohort.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span><span class="p">,</span> <span class="n">monoallelic_predicate</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span> <span class="o">=</span> <span class="n">monoallelic_predicate</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">a_predicate</span><span class="o">=</span><span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">),</span>
<span class="gp">... </span>    <span class="n">b_predicate</span><span class="o">=</span><span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">FRAMESHIFT_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">),</span>
<span class="gp">... </span>    <span class="n">a_label</span><span class="o">=</span><span class="s2">&quot;Missense&quot;</span><span class="p">,</span> <span class="n">b_label</span><span class="o">=</span><span class="s2">&quot;Frameshift&quot;</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span><span class="o">.</span><span class="n">group_labels</span>
<span class="go">(&#39;Missense&#39;, &#39;Frameshift&#39;)</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>There are many other ways to set up a predicate for testing
for a GP correlation.
See the <a class="reference internal" href="user-guide/predicates/index.html#predicates"><span class="std std-ref">Predicates</span></a> section to learn more about building
a predicate of interest.</p>
</div>
<p>The phenotype grouping is based on presence or absence of an HPO term.
We take advantage of the ontology “true path rule” to infer presence
of the ancestor terms for all present HPO terms
(e.g. presence of <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0010438">Abnormal ventricular septum morphology</a>
in an individual annotated with <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001629">Ventricular septal defect</a>)
and exclusion of the descendant terms for all excluded terms (e.g. exclusion of
<a class="reference external" href="https://hpo.jax.org/browse/term/HP:0020219">Motor seizure</a>
in an individual where <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001250">Seizure</a>
was excluded):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.phenotype</span> <span class="kn">import</span> <span class="n">prepare_predicates_for_terms_of_interest</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pheno_predicates</span> <span class="o">=</span> <span class="n">prepare_predicates_for_terms_of_interest</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">hpo</span><span class="o">=</span><span class="n">hpo</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>By default, GPSEA will perform one hypothesis test for each HPO term used to annotate at least one individual in the cohort.
Testing multiple hypothesis on the same dataset increases the chance of receiving false positive result.
However, GPSEA simplifies the application of an appropriate multiple testing correction.</p>
<p>For general use, we recommend using a combination
of a <em>Phenotype MTC filter</em> (<a class="reference internal" href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.PhenotypeMtcFilter" title="gpsea.analysis.mtc_filter.PhenotypeMtcFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">PhenotypeMtcFilter</span></code></a>) with a <em>multiple testing correction</em>.
Phenotype MTC filter chooses the HPO terms to test according to several heuristics, which
reduce the multiple testing burden and focus the analysis
on the most interesting terms (see <a class="reference internal" href="user-guide/mtc.html#hpo-mtc-filter-strategy"><span class="std std-ref">HPO MTC filter</span></a> for more info).
Then the multiple testing correction, such as Bonferroni or Benjamini-Hochberg,
is used to control the family-wise error rate or the false discovery rate.
See <a class="reference internal" href="user-guide/mtc.html#mtc"><span class="std std-ref">Multiple-testing correction</span></a> for more information.</p>
<p>In this example, we will use a combination of the HPO MTC filter (<a class="reference internal" href="apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter" title="gpsea.analysis.mtc_filter.HpoMtcFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">HpoMtcFilter</span></code></a>)
with Benjamini-Hochberg procedure (<code class="docutils literal notranslate"><span class="pre">mtc_correction='fdr_bh'</span></code>)
with a false discovery control level at (<code class="docutils literal notranslate"><span class="pre">mtc_alpha=0.05</span></code>):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.mtc_filter</span> <span class="kn">import</span> <span class="n">HpoMtcFilter</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_filter</span> <span class="o">=</span> <span class="n">HpoMtcFilter</span><span class="o">.</span><span class="n">default_filter</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_correction</span> <span class="o">=</span> <span class="s1">&#39;fdr_bh&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_alpha</span> <span class="o">=</span> <span class="mf">0.05</span>
</pre></div>
</div>
<p>Choosing the statistical procedure for assessment of association between genotype and phenotype
groups is the last missing piece of the analysis. We will use Fisher Exact Test:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pcats.stats</span> <span class="kn">import</span> <span class="n">FisherExactTest</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">count_statistic</span> <span class="o">=</span> <span class="n">FisherExactTest</span><span class="p">()</span>
</pre></div>
</div>
<p>and we finalize the analysis setup by putting all components together
into <a class="reference internal" href="apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysis" title="gpsea.analysis.pcats.HpoTermAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">HpoTermAnalysis</span></code></a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pcats</span> <span class="kn">import</span> <span class="n">HpoTermAnalysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">analysis</span> <span class="o">=</span> <span class="n">HpoTermAnalysis</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">count_statistic</span><span class="o">=</span><span class="n">count_statistic</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_filter</span><span class="o">=</span><span class="n">mtc_filter</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_correction</span><span class="o">=</span><span class="n">mtc_correction</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_alpha</span><span class="o">=</span><span class="n">mtc_alpha</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>Now we can perform the analysis and investigate the results.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span> <span class="o">=</span> <span class="n">analysis</span><span class="o">.</span><span class="n">compare_genotype_vs_phenotypes</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">gt_predicate</span><span class="o">=</span><span class="n">gt_predicate</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">pheno_predicates</span><span class="o">=</span><span class="n">pheno_predicates</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">total_tests</span>
<span class="go">16</span>
</pre></div>
</div>
<p>We only tested 16 HPO terms. This is despite the individuals being collectively annotated with
369 direct and indirect HPO terms</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">result</span><span class="o">.</span><span class="n">phenotypes</span><span class="p">)</span>
<span class="go">369</span>
</pre></div>
</div>
<p>We can show the reasoning behind <em>not</em> testing 353 (<cite>369 - 16</cite>) HPO terms
by exploring the phenotype MTC filtering report.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">MtcStatsViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_viewer</span> <span class="o">=</span> <span class="n">MtcStatsViewer</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_report</span> <span class="o">=</span> <span class="n">mtc_viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_report</span>  
</pre></div>
</div>
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Cohort</title>
  <style>table {
    border-collapse: collapse;
    margin: 25px 0;
      font-size: 0.9em;
      font-family: sans-serif;
      min-width: 400px;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
  }


  th {
    background-color: #f2f2f2;
    border: 1px solid #dddddd;
    text-align: left;
    padding: 2px;
  }

  tr {
    border: 1px solid #dddddd;
  }

  td {
    padding: 2px;
    font-weight: bold;
  }

  tr:nth-child(even) {
    background-color: #f2f2f2;
  }

  .table td,tr {
    text-align: left;
}

.table td:first-child, tr:first-child {
    text-align: left;
}

  </style>
</head>

<body>
  <h1>Phenotype testing report</h1>
  <p>Phenotype MTC filter: <em>HPO MTC filter</em></p>
  <p>Multiple testing correction: <em>fdr_bh</em></p>
  <p>Performed statistical tests for 16 out of the total of 369 HPO terms.</p>
    <table>
      <caption>Using <em>HPO MTC filter</em>, 353 term(s) were omitted from statistical analysis.</caption>
        <tbody>
          <tr>
            <th>Code</th>
            <th>Reason</th>
            <th>Count</th>
          </tr>
          
          <tr>
            <td>HMF01</td>
            <td>Skipping term with maximum frequency that was less than threshold 0.4</td>
            <td>94</td>
          </tr>
          
          <tr>
            <td>HMF08</td>
            <td>Skipping general term</td>
            <td>48</td>
          </tr>
          
          <tr>
            <td>HMF09</td>
            <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>
            <td>211</td>
          </tr>
          
        </tbody>
    </table>
</body>
</html><p>and these are the tested HPO terms ordered by the p value corrected with the Benjamini-Hochberg procedure:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">summarize_hpo_analysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span> <span class="o">=</span> <span class="n">summarize_hpo_analysis</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">result</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span>  
</pre></div>
</div>
<table class="docutils align-default" id="id5">
<caption><span class="caption-text"><em>TBX5</em> frameshift vs missense</span><a class="headerlink" href="#id5" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"><p>Allele group</p></th>
<th class="head"><p>Missense</p></th>
<th class="head"><p>Missense</p></th>
<th class="head"><p>Frameshift</p></th>
<th class="head"><p>Frameshift</p></th>
<th class="head"></th>
<th class="head"></th>
</tr>
<tr class="row-even"><th class="head"></th>
<th class="head"><p>Count</p></th>
<th class="head"><p>Percent</p></th>
<th class="head"><p>Count</p></th>
<th class="head"><p>Percent</p></th>
<th class="head"><p>Corrected p values</p></th>
<th class="head"><p>p values</p></th>
</tr>
</thead>
<tbody>
<tr class="row-odd"><td><p>Ventricular septal defect [HP:0001629]</p></td>
<td><p>31/60</p></td>
<td><p>52%</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td><p>0.0008990549794102921</p></td>
<td><p>5.6190936213143254e-05</p></td>
</tr>
<tr class="row-even"><td><p>Absent thumb [HP:0009777]</p></td>
<td><p>12/71</p></td>
<td><p>17%</p></td>
<td><p>14/31</p></td>
<td><p>45%</p></td>
<td><p>0.04502808125400047</p></td>
<td><p>0.005628510156750059</p></td>
</tr>
<tr class="row-odd"><td><p>Triphalangeal thumb [HP:0001199]</p></td>
<td><p>13/72</p></td>
<td><p>18%</p></td>
<td><p>13/32</p></td>
<td><p>41%</p></td>
<td><p>0.13654199434471745</p></td>
<td><p>0.02560162393963452</p></td>
</tr>
<tr class="row-even"><td><p>Hypoplasia of the radius [HP:0002984]</p></td>
<td><p>30/62</p></td>
<td><p>48%</p></td>
<td><p>6/14</p></td>
<td><p>43%</p></td>
<td><p>1.0</p></td>
<td><p>0.7735491022101784</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiac septum morphology [HP:0001671]</p></td>
<td><p>62/62</p></td>
<td><p>100%</p></td>
<td><p>28/28</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal hand morphology [HP:0005922]</p></td>
<td><p>53/53</p></td>
<td><p>100%</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Atrial septal defect [HP:0001631]</p></td>
<td><p>42/44</p></td>
<td><p>95%</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal atrial septum morphology [HP:0011994]</p></td>
<td><p>43/43</p></td>
<td><p>100%</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiac atrium morphology [HP:0005120]</p></td>
<td><p>43/43</p></td>
<td><p>100%</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal appendicular skeleton morphology [HP:0011844]</p></td>
<td><p>64/64</p></td>
<td><p>100%</p></td>
<td><p>34/34</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia of the extremities [HP:0009815]</p></td>
<td><p>55/55</p></td>
<td><p>100%</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving the skeleton [HP:0009115]</p></td>
<td><p>56/56</p></td>
<td><p>100%</p></td>
<td><p>23/23</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia involving bones of the upper limbs [HP:0006496]</p></td>
<td><p>55/55</p></td>
<td><p>100%</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving bones of the extremities [HP:0045060]</p></td>
<td><p>55/55</p></td>
<td><p>100%</p></td>
<td><p>22/22</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal finger morphology [HP:0001167]</p></td>
<td><p>36/36</p></td>
<td><p>100%</p></td>
<td><p>31/31</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal digit morphology [HP:0011297]</p></td>
<td><p>38/38</p></td>
<td><p>100%</p></td>
<td><p>33/33</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
</tbody>
</table>
<p>We see that several HPO terms are significantly associated
with presence of a frameshift variant in <em>TBX5</em>.
For example, <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001629">Ventricular septal defect</a>
was observed in 31/60 (52%) patients with a missense variant
but it was observed in 19/19 (100%) patients with a frameshift variant.
Fisher exact test computed a p value of <cite>~0.0000562</cite>
and the p value corrected by Benjamini-Hochberg procedure
is <cite>~0.000899</cite>.</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="installation.html" class="btn btn-neutral float-left" title="Installation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="user-guide/index.html" class="btn btn-neutral float-right" title="User guide" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>