<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../../../">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>genophenocorr.analysis.predicate._all_predicates &mdash; genophenocorr 0.1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../../../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/genophenocorr.css?v=a9fa5eb9" />

  
  <!--[if lt IE 9]>
    <script src="../../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../../../_static/documentation_options.js?v=01f34227"></script>
        <script src="../../../../_static/doctools.js?v=888ff710"></script>
        <script src="../../../../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../../../../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../../../../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../../index.html" class="icon icon-home">
            genophenocorr
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">genophenocorr</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../../index.html">Module code</a></li>
      <li class="breadcrumb-item active">genophenocorr.analysis.predicate._all_predicates</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <h1>Source code for genophenocorr.analysis.predicate._all_predicates</h1><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">typing</span>

<span class="kn">import</span> <span class="nn">hpotk</span>

<span class="kn">from</span> <span class="nn">genophenocorr.constants</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="kn">from</span> <span class="nn">genophenocorr.model</span> <span class="kn">import</span> <span class="n">Patient</span><span class="p">,</span> <span class="n">FeatureType</span>
<span class="kn">from</span> <span class="nn">._api</span> <span class="kn">import</span> <span class="n">PolyPredicate</span><span class="p">,</span> <span class="n">PatientCategory</span>


<div class="viewcode-block" id="HPOPresentPredicate">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.HPOPresentPredicate">[docs]</a>
<span class="k">class</span> <span class="nc">HPOPresentPredicate</span><span class="p">(</span><span class="n">PolyPredicate</span><span class="p">[</span><span class="n">hpotk</span><span class="o">.</span><span class="n">TermId</span><span class="p">]):</span>

    <span class="n">OBSERVED</span> <span class="o">=</span> <span class="n">PatientCategory</span><span class="p">(</span><span class="n">cat_id</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                               <span class="n">name</span><span class="o">=</span><span class="s1">&#39;Observed&#39;</span><span class="p">,</span>
                               <span class="n">description</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;</span>
<span class="s2">                               The sample *is* annotated with the tested phenotype feature `q`.</span>
<span class="s2">                               </span>
<span class="s2">                               This is either because the sample is annotated with `q` (exact match),</span>
<span class="s2">                               or because one of sample&#39;s annotations is a descendant `q` (annotation propagation).</span>
<span class="s2">                               For instance, we tested for a Seizure and the sample *had* a Clonic seizure </span>
<span class="s2">                               (a descendant of Seizure).</span>
<span class="s2">                               &quot;&quot;&quot;</span><span class="p">)</span>

    <span class="n">NOT_OBSERVED</span> <span class="o">=</span> <span class="n">PatientCategory</span><span class="p">(</span><span class="n">cat_id</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
                                   <span class="n">name</span><span class="o">=</span><span class="s1">&#39;Not observed&#39;</span><span class="p">,</span>
                                   <span class="n">description</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;</span>
<span class="s2">                                   We are particular about the sample *not* having the tested feature `q`.</span>
<span class="s2">                                   </span>
<span class="s2">                                   In other words, `q` was *excluded* in the sample or the sample is annotated with an excluded ancestor of `q`.</span>
<span class="s2">                                   </span>
<span class="s2">                                   For instance, we tested for a Clonic seizure and the sample did *not* have any Seizure, which implies </span>
<span class="s2">                                   *not* Clonic seizure.  </span>
<span class="s2">                                   &quot;&quot;&quot;</span><span class="p">)</span>

    <span class="n">NOT_MEASURED</span> <span class="o">=</span> <span class="n">PatientCategory</span><span class="p">(</span><span class="n">cat_id</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                                   <span class="n">name</span><span class="o">=</span><span class="s1">&#39;Not measured&#39;</span><span class="p">,</span>
                                   <span class="n">description</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;</span>
<span class="s2">                                   We do not know if the sample has or has not the tested feature.</span>
<span class="s2">                                   &quot;&quot;&quot;</span><span class="p">)</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> 
                 <span class="n">hpo</span><span class="p">:</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">MinimalOntology</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_hpo</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">MinimalOntology</span><span class="p">,</span> <span class="s1">&#39;hpo&#39;</span><span class="p">)</span>

<div class="viewcode-block" id="HPOPresentPredicate.categories">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.HPOPresentPredicate.categories">[docs]</a>
    <span class="k">def</span> <span class="nf">categories</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">OBSERVED</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">NOT_OBSERVED</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">NOT_MEASURED</span></div>


<div class="viewcode-block" id="HPOPresentPredicate.test">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.HPOPresentPredicate.test">[docs]</a>
    <span class="k">def</span> <span class="nf">test</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">patient</span><span class="p">:</span> <span class="n">Patient</span><span class="p">,</span> <span class="n">query</span><span class="p">:</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">TermId</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">patient</span><span class="p">,</span> <span class="n">Patient</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;patient must be type Patient but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">patient</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="n">query</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">query</span><span class="p">,</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">TermId</span><span class="p">,</span> <span class="s1">&#39;query&#39;</span><span class="p">)</span>

        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">patient</span><span class="o">.</span><span class="n">phenotypes</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="k">return</span> <span class="kc">None</span>

        <span class="k">for</span> <span class="n">pheno</span> <span class="ow">in</span> <span class="n">patient</span><span class="o">.</span><span class="n">phenotypes</span><span class="p">:</span>
            <span class="k">if</span> <span class="n">pheno</span><span class="o">.</span><span class="n">observed</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="k">continue</span>
            <span class="k">if</span> <span class="n">pheno</span><span class="o">.</span><span class="n">observed</span><span class="p">:</span>
                <span class="k">if</span> <span class="nb">any</span><span class="p">((</span><span class="n">query</span> <span class="o">==</span> <span class="n">ancestor</span>
                        <span class="k">for</span> <span class="n">ancestor</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hpo</span><span class="o">.</span><span class="n">graph</span><span class="o">.</span><span class="n">get_ancestors</span><span class="p">(</span><span class="n">pheno</span><span class="p">,</span> <span class="n">include_source</span><span class="o">=</span><span class="kc">True</span><span class="p">))):</span>
                    <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">OBSERVED</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">if</span> <span class="nb">any</span><span class="p">((</span><span class="n">query</span> <span class="o">==</span> <span class="n">descendant</span>
                        <span class="k">for</span> <span class="n">descendant</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hpo</span><span class="o">.</span><span class="n">graph</span><span class="o">.</span><span class="n">get_descendants</span><span class="p">(</span><span class="n">pheno</span><span class="p">,</span> <span class="n">include_source</span><span class="o">=</span><span class="kc">True</span><span class="p">))):</span>
                    <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">NOT_OBSERVED</span>

        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">NOT_MEASURED</span></div>
</div>



<span class="n">HETEROZYGOUS</span> <span class="o">=</span> <span class="n">PatientCategory</span><span class="p">(</span><span class="n">cat_id</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                            <span class="n">name</span><span class="o">=</span><span class="s1">&#39;Heterozygous&#39;</span><span class="p">,</span>
                            <span class="n">description</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;</span>
<span class="s2">                            This sample has the tested attribute on one allele. </span>
<span class="s2">                            &quot;&quot;&quot;</span><span class="p">)</span>

<span class="n">HOMOZYGOUS</span> <span class="o">=</span> <span class="n">PatientCategory</span><span class="p">(</span><span class="n">cat_id</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
                                <span class="n">name</span><span class="o">=</span><span class="s1">&#39;Homozygous&#39;</span><span class="p">,</span>
                                <span class="n">description</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;</span>
<span class="s2">                                This sample has the tested attribute on both alleles. </span>
<span class="s2">                                &quot;&quot;&quot;</span><span class="p">)</span>

<span class="n">NO_VARIANT</span> <span class="o">=</span> <span class="n">PatientCategory</span><span class="p">(</span><span class="n">cat_id</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                                <span class="n">name</span><span class="o">=</span><span class="s1">&#39;No Variant&#39;</span><span class="p">,</span>
                                <span class="n">description</span><span class="o">=</span><span class="s2">&quot;&quot;&quot;</span>
<span class="s2">                                The sample does not have the tested attribute.</span>
<span class="s2">                                &quot;&quot;&quot;</span><span class="p">)</span>



<div class="viewcode-block" id="VariantEffectPredicate">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.VariantEffectPredicate">[docs]</a>
<span class="k">class</span> <span class="nc">VariantEffectPredicate</span><span class="p">(</span><span class="n">PolyPredicate</span><span class="p">[</span><span class="n">VariantEffect</span><span class="p">]):</span>
    
    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span><span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_transcript</span> <span class="o">=</span> <span class="n">transcript</span>

<div class="viewcode-block" id="VariantEffectPredicate.categories">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.VariantEffectPredicate.categories">[docs]</a>
    <span class="k">def</span> <span class="nf">categories</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">return</span> <span class="n">HETEROZYGOUS</span><span class="p">,</span> <span class="n">HOMOZYGOUS</span><span class="p">,</span> <span class="n">NO_VARIANT</span></div>


<div class="viewcode-block" id="VariantEffectPredicate.test">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.VariantEffectPredicate.test">[docs]</a>
    <span class="k">def</span> <span class="nf">test</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">patient</span><span class="p">:</span> <span class="n">Patient</span><span class="p">,</span> <span class="n">query</span><span class="p">:</span><span class="n">VariantEffect</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">patient</span><span class="p">,</span> <span class="n">Patient</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;patient must be type Patient but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">patient</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">query</span><span class="p">,</span> <span class="n">VariantEffect</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;query must be type VariantEffect but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="nb">vars</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">var</span> <span class="ow">in</span> <span class="n">patient</span><span class="o">.</span><span class="n">variants</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">trans</span> <span class="ow">in</span> <span class="n">var</span><span class="o">.</span><span class="n">tx_annotations</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">trans</span><span class="o">.</span><span class="n">transcript_id</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">_transcript</span><span class="p">:</span>
                    <span class="k">for</span> <span class="n">var_eff</span> <span class="ow">in</span> <span class="n">trans</span><span class="o">.</span><span class="n">variant_effects</span><span class="p">:</span>
                        <span class="k">if</span> <span class="n">var_eff</span> <span class="o">==</span> <span class="nb">str</span><span class="p">(</span><span class="n">query</span><span class="p">):</span>
                            <span class="nb">vars</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">var</span><span class="p">)</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="nb">vars</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">v</span> <span class="ow">in</span> <span class="nb">vars</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">v</span><span class="o">.</span><span class="n">genotype</span> <span class="o">==</span> <span class="s2">&quot;heterozygous&quot;</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HETEROZYGOUS</span>
                <span class="k">elif</span> <span class="n">v</span><span class="o">.</span><span class="n">genotype</span> <span class="o">==</span> <span class="s2">&quot;homozygous&quot;</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HOMOZYGOUS</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HETEROZYGOUS</span>
        <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="nb">vars</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">HOMOZYGOUS</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">NO_VARIANT</span></div>
</div>


<div class="viewcode-block" id="VariantPredicate">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.VariantPredicate">[docs]</a>
<span class="k">class</span> <span class="nc">VariantPredicate</span><span class="p">(</span><span class="n">PolyPredicate</span><span class="p">[</span><span class="nb">str</span><span class="p">]):</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span><span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_transcript</span> <span class="o">=</span> <span class="n">transcript</span>

<div class="viewcode-block" id="VariantPredicate.categories">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.VariantPredicate.categories">[docs]</a>
    <span class="k">def</span> <span class="nf">categories</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">return</span> <span class="n">HETEROZYGOUS</span><span class="p">,</span> <span class="n">HOMOZYGOUS</span><span class="p">,</span> <span class="n">NO_VARIANT</span></div>


<div class="viewcode-block" id="VariantPredicate.test">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.VariantPredicate.test">[docs]</a>
    <span class="k">def</span> <span class="nf">test</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">patient</span><span class="p">:</span> <span class="n">Patient</span><span class="p">,</span> <span class="n">query</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">patient</span><span class="p">,</span> <span class="n">Patient</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;patient must be type Patient but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">patient</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">query</span><span class="p">,</span> <span class="nb">str</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;query must be type string but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="nb">vars</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">var</span> <span class="ow">in</span> <span class="n">patient</span><span class="o">.</span><span class="n">variants</span><span class="p">:</span>
            <span class="c1">#print(f&quot;{var.variant_string} == {query}&quot;)</span>
            <span class="k">if</span> <span class="n">var</span><span class="o">.</span><span class="n">variant_string</span> <span class="o">==</span> <span class="n">query</span><span class="p">:</span>
                <span class="nb">vars</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">var</span><span class="p">)</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="nb">vars</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">v</span> <span class="ow">in</span> <span class="nb">vars</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">v</span><span class="o">.</span><span class="n">genotype</span> <span class="o">==</span> <span class="s2">&quot;heterozygous&quot;</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HETEROZYGOUS</span>
                <span class="k">elif</span> <span class="n">v</span><span class="o">.</span><span class="n">genotype</span> <span class="o">==</span> <span class="s2">&quot;homozygous&quot;</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HOMOZYGOUS</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HETEROZYGOUS</span>
        <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="nb">vars</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">HOMOZYGOUS</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">NO_VARIANT</span></div>
</div>


<div class="viewcode-block" id="ExonPredicate">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ExonPredicate">[docs]</a>
<span class="k">class</span> <span class="nc">ExonPredicate</span><span class="p">(</span><span class="n">PolyPredicate</span><span class="p">[</span><span class="nb">int</span><span class="p">]):</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span><span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_transcript</span> <span class="o">=</span> <span class="n">transcript</span>

<div class="viewcode-block" id="ExonPredicate.categories">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ExonPredicate.categories">[docs]</a>
    <span class="k">def</span> <span class="nf">categories</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">return</span> <span class="n">HETEROZYGOUS</span><span class="p">,</span> <span class="n">HOMOZYGOUS</span><span class="p">,</span> <span class="n">NO_VARIANT</span></div>


<div class="viewcode-block" id="ExonPredicate.test">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ExonPredicate.test">[docs]</a>
    <span class="k">def</span> <span class="nf">test</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">patient</span><span class="p">:</span> <span class="n">Patient</span><span class="p">,</span> <span class="n">query</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">patient</span><span class="p">,</span> <span class="n">Patient</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;patient must be type Patient but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">patient</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">query</span><span class="p">,</span> <span class="nb">int</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;query must be type integer but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="nb">vars</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">var</span> <span class="ow">in</span> <span class="n">patient</span><span class="o">.</span><span class="n">variants</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">trans</span> <span class="ow">in</span> <span class="n">var</span><span class="o">.</span><span class="n">tx_annotations</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">trans</span><span class="o">.</span><span class="n">transcript_id</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">_transcript</span><span class="p">:</span>
                    <span class="k">if</span> <span class="n">trans</span><span class="o">.</span><span class="n">overlapping_exons</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
                        <span class="k">if</span> <span class="n">query</span> <span class="ow">in</span> <span class="n">trans</span><span class="o">.</span><span class="n">overlapping_exons</span><span class="p">:</span>
                            <span class="nb">vars</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">var</span><span class="p">)</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="nb">vars</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">v</span> <span class="ow">in</span> <span class="nb">vars</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">v</span><span class="o">.</span><span class="n">genotype</span> <span class="o">==</span> <span class="s2">&quot;heterozygous&quot;</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HETEROZYGOUS</span>
                <span class="k">elif</span> <span class="n">v</span><span class="o">.</span><span class="n">genotype</span> <span class="o">==</span> <span class="s2">&quot;homozygous&quot;</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HOMOZYGOUS</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HETEROZYGOUS</span>
        <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="nb">vars</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">HOMOZYGOUS</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">NO_VARIANT</span></div>
</div>


<div class="viewcode-block" id="ProtFeatureTypePredicate">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ProtFeatureTypePredicate">[docs]</a>
<span class="k">class</span> <span class="nc">ProtFeatureTypePredicate</span><span class="p">(</span><span class="n">PolyPredicate</span><span class="p">[</span><span class="n">FeatureType</span><span class="p">]):</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span><span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
         <span class="bp">self</span><span class="o">.</span><span class="n">_transcript</span> <span class="o">=</span> <span class="n">transcript</span>

<div class="viewcode-block" id="ProtFeatureTypePredicate.categories">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ProtFeatureTypePredicate.categories">[docs]</a>
    <span class="k">def</span> <span class="nf">categories</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">return</span> <span class="n">HETEROZYGOUS</span><span class="p">,</span> <span class="n">HOMOZYGOUS</span><span class="p">,</span> <span class="n">NO_VARIANT</span></div>


<div class="viewcode-block" id="ProtFeatureTypePredicate.test">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ProtFeatureTypePredicate.test">[docs]</a>
    <span class="k">def</span> <span class="nf">test</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">patient</span><span class="p">:</span> <span class="n">Patient</span><span class="p">,</span> <span class="n">query</span><span class="p">:</span><span class="n">FeatureType</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">patient</span><span class="p">,</span> <span class="n">Patient</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;patient must be type Patient but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">patient</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">query</span><span class="p">,</span> <span class="n">FeatureType</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;query must be type FeatureType but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="nb">vars</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">var</span> <span class="ow">in</span> <span class="n">patient</span><span class="o">.</span><span class="n">variants</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">trans</span> <span class="ow">in</span> <span class="n">var</span><span class="o">.</span><span class="n">tx_annotations</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">trans</span><span class="o">.</span><span class="n">transcript_id</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">_transcript</span><span class="p">:</span>
                    <span class="k">if</span> <span class="n">trans</span><span class="o">.</span><span class="n">protein_effect_location</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="ow">and</span> <span class="n">trans</span><span class="o">.</span><span class="n">protein_effect_location</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="ow">and</span> <span class="n">trans</span><span class="o">.</span><span class="n">protein_effect_location</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
                        <span class="k">for</span> <span class="n">prot</span> <span class="ow">in</span> <span class="n">trans</span><span class="o">.</span><span class="n">protein_affected</span><span class="p">:</span>
                            <span class="k">for</span> <span class="n">feat</span> <span class="ow">in</span> <span class="n">prot</span><span class="o">.</span><span class="n">protein_features</span><span class="p">:</span>
                                <span class="k">if</span> <span class="n">feat</span><span class="o">.</span><span class="n">feature_type</span> <span class="o">==</span> <span class="n">query</span><span class="p">:</span>
                                    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="nb">max</span><span class="p">(</span><span class="n">trans</span><span class="o">.</span><span class="n">protein_effect_location</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">feat</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">start</span><span class="p">),</span> <span class="nb">min</span><span class="p">(</span><span class="n">trans</span><span class="o">.</span><span class="n">protein_effect_location</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">feat</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">end</span><span class="p">)</span> <span class="o">+</span> <span class="mi">1</span><span class="p">)))</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
                                        <span class="nb">vars</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">var</span><span class="p">)</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="nb">vars</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">v</span> <span class="ow">in</span> <span class="nb">vars</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">v</span><span class="o">.</span><span class="n">genotype</span> <span class="o">==</span> <span class="s2">&quot;heterozygous&quot;</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HETEROZYGOUS</span>
                <span class="k">elif</span> <span class="n">v</span><span class="o">.</span><span class="n">genotype</span> <span class="o">==</span> <span class="s2">&quot;homozygous&quot;</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HOMOZYGOUS</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HETEROZYGOUS</span>
        <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="nb">vars</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">HOMOZYGOUS</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">NO_VARIANT</span></div>
</div>


<div class="viewcode-block" id="ProtFeaturePredicate">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ProtFeaturePredicate">[docs]</a>
<span class="k">class</span> <span class="nc">ProtFeaturePredicate</span><span class="p">(</span><span class="n">PolyPredicate</span><span class="p">[</span><span class="nb">str</span><span class="p">]):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Test if the patient has a variant that overlaps with a protein feature.</span>

<span class="sd">    The predicate is generic over a string that represents the name of the protein feature.</span>
<span class="sd">    For instance, `EGF-like 2` for `FBN1 &lt;https://www.uniprot.org/uniprotkb/P35555/entry#family_and_domains&gt;`_</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span><span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
         <span class="bp">self</span><span class="o">.</span><span class="n">_transcript</span> <span class="o">=</span> <span class="n">transcript</span>

<div class="viewcode-block" id="ProtFeaturePredicate.categories">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ProtFeaturePredicate.categories">[docs]</a>
    <span class="k">def</span> <span class="nf">categories</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">return</span> <span class="n">HETEROZYGOUS</span><span class="p">,</span> <span class="n">HOMOZYGOUS</span><span class="p">,</span> <span class="n">NO_VARIANT</span></div>


<div class="viewcode-block" id="ProtFeaturePredicate.test">
<a class="viewcode-back" href="../../../../apidocs/genophenocorr.analysis.predicate.html#genophenocorr.analysis.predicate.ProtFeaturePredicate.test">[docs]</a>
    <span class="k">def</span> <span class="nf">test</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">patient</span><span class="p">:</span> <span class="n">Patient</span><span class="p">,</span> <span class="n">query</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="n">PatientCategory</span><span class="p">]:</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">patient</span><span class="p">,</span> <span class="n">Patient</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;patient must be type Patient but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">patient</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">query</span><span class="p">,</span> <span class="nb">str</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;query must be type string but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="nb">vars</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">var</span> <span class="ow">in</span> <span class="n">patient</span><span class="o">.</span><span class="n">variants</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">trans</span> <span class="ow">in</span> <span class="n">var</span><span class="o">.</span><span class="n">tx_annotations</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">trans</span><span class="o">.</span><span class="n">transcript_id</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">_transcript</span><span class="p">:</span>
                    <span class="k">if</span> <span class="n">trans</span><span class="o">.</span><span class="n">protein_effect_location</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="ow">and</span> <span class="n">trans</span><span class="o">.</span><span class="n">protein_effect_location</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="ow">and</span> <span class="n">trans</span><span class="o">.</span><span class="n">protein_effect_location</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
                        <span class="k">for</span> <span class="n">prot</span> <span class="ow">in</span> <span class="n">trans</span><span class="o">.</span><span class="n">protein_affected</span><span class="p">:</span>
                            <span class="k">for</span> <span class="n">feat</span> <span class="ow">in</span> <span class="n">prot</span><span class="o">.</span><span class="n">protein_features</span><span class="p">:</span>
                                <span class="k">if</span> <span class="n">feat</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">name</span> <span class="o">==</span> <span class="n">query</span><span class="p">:</span>
                                    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="nb">max</span><span class="p">(</span><span class="n">trans</span><span class="o">.</span><span class="n">protein_effect_location</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">feat</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">start</span><span class="p">),</span> <span class="nb">min</span><span class="p">(</span><span class="n">trans</span><span class="o">.</span><span class="n">protein_effect_location</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">feat</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">end</span><span class="p">)</span> <span class="o">+</span> <span class="mi">1</span><span class="p">)))</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
                                        <span class="nb">vars</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">var</span><span class="p">)</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="nb">vars</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">v</span> <span class="ow">in</span> <span class="nb">vars</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">v</span><span class="o">.</span><span class="n">genotype</span> <span class="o">==</span> <span class="s2">&quot;heterozygous&quot;</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HETEROZYGOUS</span>
                <span class="k">elif</span> <span class="n">v</span><span class="o">.</span><span class="n">genotype</span> <span class="o">==</span> <span class="s2">&quot;homozygous&quot;</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HOMOZYGOUS</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="k">return</span> <span class="n">HETEROZYGOUS</span>
        <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="nb">vars</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">HOMOZYGOUS</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">NO_VARIANT</span></div>
</div>

</pre></div>

           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>