

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../../">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>gpsea.view._cohort &mdash; GPSEA 0.5.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../../../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../../_static/gpsea.css?v=547657ed" />

  
      <script src="../../../_static/jquery.js?v=5d32c60e"></script>
      <script src="../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../../../_static/documentation_options.js?v=1dd76d02"></script>
      <script src="../../../_static/doctools.js?v=9a2dae69"></script>
      <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../../../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Module code</a></li>
      <li class="breadcrumb-item active">gpsea.view._cohort</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <h1>Source code for gpsea.view._cohort</h1><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">typing</span>

<span class="kn">from</span> <span class="nn">hpotk</span> <span class="kn">import</span> <span class="n">MinimalOntology</span>
<span class="kn">from</span> <span class="nn">jinja2</span> <span class="kn">import</span> <span class="n">Environment</span><span class="p">,</span> <span class="n">PackageLoader</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">namedtuple</span>

<span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">Cohort</span>
<span class="kn">from</span> <span class="nn">._report</span> <span class="kn">import</span> <span class="n">GpseaReport</span><span class="p">,</span> <span class="n">HtmlGpseaReport</span>
<span class="kn">from</span> <span class="nn">._formatter</span> <span class="kn">import</span> <span class="n">VariantFormatter</span>


<span class="n">ToDisplay</span> <span class="o">=</span> <span class="n">namedtuple</span><span class="p">(</span><span class="s1">&#39;ToDisplay&#39;</span><span class="p">,</span> <span class="p">[</span><span class="s1">&#39;hgvs_cdna&#39;</span><span class="p">,</span> <span class="s1">&#39;hgvsp&#39;</span><span class="p">,</span> <span class="s1">&#39;variant_effects&#39;</span><span class="p">])</span>

<div class="viewcode-block" id="CohortViewable">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.CohortViewable">[docs]</a>
<span class="k">class</span> <span class="nc">CohortViewable</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Class to create a viewable object that is uses a Jinja2 template to create an HTML element</span>
<span class="sd">    for display in the Jupyter notebook.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
            <span class="bp">self</span><span class="p">,</span>
            <span class="n">hpo</span><span class="p">:</span> <span class="n">MinimalOntology</span><span class="p">,</span>
            <span class="n">top_phenotype_count</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">10</span><span class="p">,</span>
            <span class="n">top_variant_count</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">10</span><span class="p">,</span>
    <span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Args:</span>
<span class="sd">            hpo(MinimalOntology): An HPO ontology object from hpo-toolkit</span>
<span class="sd">            top_phenotype_count(int): Maximum number of HPO terms to display in the HTML table (default: 10)</span>
<span class="sd">            top_variant_count(int): Maximum number of variants to display in the HTML table (default: 10)</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_hpo</span> <span class="o">=</span> <span class="n">hpo</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_top_phenotype_count</span> <span class="o">=</span> <span class="n">top_phenotype_count</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_top_variant_count</span> <span class="o">=</span> <span class="n">top_variant_count</span>
        <span class="n">environment</span> <span class="o">=</span> <span class="n">Environment</span><span class="p">(</span><span class="n">loader</span><span class="o">=</span><span class="n">PackageLoader</span><span class="p">(</span><span class="s1">&#39;gpsea.view&#39;</span><span class="p">,</span> <span class="s1">&#39;templates&#39;</span><span class="p">))</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span> <span class="o">=</span> <span class="n">environment</span><span class="o">.</span><span class="n">get_template</span><span class="p">(</span><span class="s2">&quot;cohort.html&quot;</span><span class="p">)</span>

<div class="viewcode-block" id="CohortViewable.process">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.CohortViewable.process">[docs]</a>
    <span class="k">def</span> <span class="nf">process</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
        <span class="n">transcript_id</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">GpseaReport</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Create an HTML that should be shown with display(HTML(..)) of the ipython package.</span>

<span class="sd">        Args:</span>
<span class="sd">            cohort (Cohort): The cohort being analyzed in the current Notebook</span>
<span class="sd">            transcript_id (str): the transcript that we map variants onto</span>

<span class="sd">        Returns:</span>
<span class="sd">            GpseaReport: a report that can be stored to a path or displayed in</span>
<span class="sd">                interactive environment such as Jupyter notebook.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">context</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_prepare_context</span><span class="p">(</span><span class="n">cohort</span><span class="p">,</span> <span class="n">transcript_id</span><span class="o">=</span><span class="n">transcript_id</span><span class="p">)</span>
        <span class="n">report</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span><span class="o">.</span><span class="n">render</span><span class="p">(</span><span class="n">context</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">HtmlGpseaReport</span><span class="p">(</span><span class="n">html</span><span class="o">=</span><span class="n">report</span><span class="p">)</span></div>


    <span class="k">def</span> <span class="nf">_prepare_context</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
        <span class="n">transcript_id</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Mapping</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">typing</span><span class="o">.</span><span class="n">Any</span><span class="p">]:</span>

        <span class="n">hpo_counts</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">hpo</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">list_present_phenotypes</span><span class="p">(</span><span class="n">top</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">_top_phenotype_count</span><span class="p">):</span>
            <span class="n">hpo_id</span> <span class="o">=</span> <span class="n">hpo</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
            <span class="n">individual_count</span> <span class="o">=</span> <span class="n">hpo</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
            <span class="n">hpo_label</span> <span class="o">=</span> <span class="s2">&quot;n/a&quot;</span>
            <span class="k">if</span> <span class="n">hpo_id</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hpo</span><span class="p">:</span>
                <span class="n">hpo_label</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hpo</span><span class="o">.</span><span class="n">get_term_name</span><span class="p">(</span><span class="n">hpo_id</span><span class="p">)</span>
            <span class="n">hpo_counts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                <span class="p">{</span>
                    <span class="s2">&quot;HPO&quot;</span><span class="p">:</span> <span class="n">hpo_label</span><span class="p">,</span>
                    <span class="s2">&quot;ID&quot;</span><span class="p">:</span> <span class="n">hpo_id</span><span class="p">,</span>
                    <span class="s2">&quot;Count&quot;</span><span class="p">:</span> <span class="n">individual_count</span><span class="p">,</span>
                <span class="p">}</span>
            <span class="p">)</span>

        <span class="n">variant_counts</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="n">variant_to_display_d</span> <span class="o">=</span> <span class="n">CohortViewable</span><span class="o">.</span><span class="n">_get_variant_description</span><span class="p">(</span><span class="n">cohort</span><span class="p">,</span> <span class="n">transcript_id</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">variant_key</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">list_all_variants</span><span class="p">(</span><span class="n">top</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">_top_variant_count</span><span class="p">):</span>
            <span class="c1"># get HGVS or human readable variant</span>
            <span class="k">if</span> <span class="n">variant_key</span> <span class="ow">in</span> <span class="n">variant_to_display_d</span><span class="p">:</span>
                <span class="n">display</span> <span class="o">=</span> <span class="n">variant_to_display_d</span><span class="p">[</span><span class="n">variant_key</span><span class="p">]</span>
                <span class="n">hgvs_cdna</span> <span class="o">=</span> <span class="n">display</span><span class="o">.</span><span class="n">hgvs_cdna</span>
                <span class="n">protein_name</span> <span class="o">=</span> <span class="n">display</span><span class="o">.</span><span class="n">hgvsp</span>
                <span class="n">effects</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span> <span class="k">if</span> <span class="n">display</span><span class="o">.</span><span class="n">variant_effects</span> <span class="ow">is</span> <span class="kc">None</span> <span class="k">else</span> <span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">display</span><span class="o">.</span><span class="n">variant_effects</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">display</span> <span class="o">=</span> <span class="n">variant_key</span>
                <span class="n">hgvs_cdna</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
                <span class="n">protein_name</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
                <span class="n">effects</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
            
            <span class="n">variant_counts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                <span class="p">{</span>
                    <span class="s2">&quot;variant&quot;</span><span class="p">:</span> <span class="n">variant_key</span><span class="p">,</span>
                    <span class="s2">&quot;variant_name&quot;</span><span class="p">:</span> <span class="n">hgvs_cdna</span><span class="p">,</span>
                    <span class="s2">&quot;protein_name&quot;</span><span class="p">:</span> <span class="n">protein_name</span><span class="p">,</span>
                    <span class="s2">&quot;variant_effects&quot;</span><span class="p">:</span> <span class="n">effects</span><span class="p">,</span>
                    <span class="s2">&quot;Count&quot;</span><span class="p">:</span> <span class="n">count</span><span class="p">,</span>
                <span class="p">}</span>
            <span class="p">)</span>

        <span class="n">disease_counts</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">disease_id</span><span class="p">,</span> <span class="n">disease_count</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">list_all_diseases</span><span class="p">():</span>
            <span class="n">disease_name</span> <span class="o">=</span> <span class="s2">&quot;Unknown&quot;</span>
            <span class="k">for</span> <span class="n">disease</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_diseases</span><span class="p">():</span>
                <span class="k">if</span> <span class="n">disease</span><span class="o">.</span><span class="n">identifier</span> <span class="o">==</span> <span class="n">disease_id</span><span class="p">:</span>
                    <span class="n">disease_name</span> <span class="o">=</span> <span class="n">disease</span><span class="o">.</span><span class="n">name</span>
            <span class="n">disease_counts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                <span class="p">{</span>
                    <span class="s2">&quot;disease_id&quot;</span><span class="p">:</span> <span class="n">disease_id</span><span class="p">,</span>
                    <span class="s2">&quot;disease_name&quot;</span><span class="p">:</span> <span class="n">disease_name</span><span class="p">,</span>
                    <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="n">disease_count</span><span class="p">,</span>
                <span class="p">}</span>
            <span class="p">)</span>
        
        <span class="n">n_diseases</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">disease_counts</span><span class="p">)</span>

        <span class="n">var_effects_list</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="k">if</span> <span class="n">transcript_id</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">has_transcript</span> <span class="o">=</span> <span class="kc">True</span>
            <span class="n">data_by_tx</span> <span class="o">=</span> <span class="n">cohort</span><span class="o">.</span><span class="n">variant_effect_count_by_tx</span><span class="p">(</span><span class="n">tx_id</span><span class="o">=</span><span class="n">transcript_id</span><span class="p">)</span>
            <span class="c1"># e.g., data structure -- {&#39;effect}&#39;: &#39;FRAMESHIFT_VARIANT&#39;, &#39;count&#39;: 175}, {&#39;effect}&#39;: &#39;STOP_GAINED&#39;, &#39;count&#39;: 67},</span>
            <span class="k">for</span> <span class="n">tx_id</span><span class="p">,</span> <span class="n">counter</span> <span class="ow">in</span> <span class="n">data_by_tx</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
                <span class="k">if</span> <span class="n">tx_id</span> <span class="o">==</span> <span class="n">transcript_id</span><span class="p">:</span>
                    <span class="k">for</span> <span class="n">effect</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">counter</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
                        <span class="n">var_effects_list</span><span class="o">.</span><span class="n">append</span><span class="p">({</span><span class="s2">&quot;effect&quot;</span><span class="p">:</span> <span class="n">effect</span><span class="p">,</span> <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="n">count</span><span class="p">})</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">has_transcript</span> <span class="o">=</span> <span class="kc">False</span>
        <span class="k">if</span> <span class="n">transcript_id</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">transcript_id</span> <span class="o">=</span> <span class="s2">&quot;MANE transcript ID&quot;</span>
        <span class="c1"># The following dictionary is used by the Jinja2 HTML template</span>
        <span class="k">return</span> <span class="p">{</span>
            <span class="s2">&quot;n_individuals&quot;</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="o">.</span><span class="n">all_patients</span><span class="p">),</span>
            <span class="s2">&quot;n_excluded&quot;</span><span class="p">:</span> <span class="n">cohort</span><span class="o">.</span><span class="n">get_excluded_count</span><span class="p">(),</span>
            <span class="s2">&quot;total_hpo_count&quot;</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="o">.</span><span class="n">all_phenotypes</span><span class="p">()),</span>
            <span class="s2">&quot;top_hpo_count&quot;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">_top_phenotype_count</span><span class="p">,</span>
            <span class="s2">&quot;hpo_counts&quot;</span><span class="p">:</span> <span class="n">hpo_counts</span><span class="p">,</span>
            <span class="s2">&quot;unique_variant_count&quot;</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="o">.</span><span class="n">all_variants</span><span class="p">()),</span>
            <span class="s2">&quot;top_var_count&quot;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">_top_variant_count</span><span class="p">,</span>
            <span class="s2">&quot;var_counts&quot;</span><span class="p">:</span> <span class="n">variant_counts</span><span class="p">,</span>
            <span class="s2">&quot;n_diseases&quot;</span><span class="p">:</span> <span class="n">n_diseases</span><span class="p">,</span>
            <span class="s2">&quot;disease_counts&quot;</span><span class="p">:</span> <span class="n">disease_counts</span><span class="p">,</span>
            <span class="s2">&quot;has_transcript&quot;</span><span class="p">:</span> <span class="n">has_transcript</span><span class="p">,</span>
            <span class="s2">&quot;var_effects_list&quot;</span><span class="p">:</span> <span class="n">var_effects_list</span><span class="p">,</span>
            <span class="s2">&quot;transcript_id&quot;</span><span class="p">:</span> <span class="n">transcript_id</span><span class="p">,</span>
        <span class="p">}</span>

    <span class="nd">@staticmethod</span>
    <span class="k">def</span> <span class="nf">_get_variant_description</span><span class="p">(</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
        <span class="n">transcript_id</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
        <span class="n">only_hgvs</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Mapping</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">ToDisplay</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get user-friendly strings (e.g., HGVS for our target transcript) to match to the chromosomal strings</span>
<span class="sd">        Args:</span>
<span class="sd">            cohort (Cohort): The cohort being analyzed in the current Notebook</span>
<span class="sd">            transcript_id (str): the transcript that we map variants onto</span>
<span class="sd">            only_hgvs (bool): do not show the transcript ID part of the HGVS annotation, just the annotation.</span>

<span class="sd">        Returns:</span>
<span class="sd">            typing.Mapping[str, ToDisplay]: key: variant key, value: namedtuple(display (e.g. HGVS) string of variant, hgvsp protein string of variant)</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">chrom_to_display</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
        <span class="n">var_formatter</span> <span class="o">=</span> <span class="n">VariantFormatter</span><span class="p">(</span><span class="n">transcript_id</span><span class="p">)</span>
        
        <span class="k">for</span> <span class="n">var</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_variants</span><span class="p">():</span>
            <span class="n">variant_key</span> <span class="o">=</span> <span class="n">var</span><span class="o">.</span><span class="n">variant_info</span><span class="o">.</span><span class="n">variant_key</span>
            <span class="n">display</span> <span class="o">=</span> <span class="n">var_formatter</span><span class="o">.</span><span class="n">format_as_string</span><span class="p">(</span><span class="n">var</span><span class="p">)</span>
            <span class="n">tx_annotation</span> <span class="o">=</span> <span class="n">var</span><span class="o">.</span><span class="n">get_tx_anno_by_tx_id</span><span class="p">(</span><span class="n">transcript_id</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">tx_annotation</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
                <span class="n">hgvsp</span> <span class="o">=</span> <span class="n">tx_annotation</span><span class="o">.</span><span class="n">hgvsp</span>
                <span class="n">var_effects</span> <span class="o">=</span> <span class="p">[</span><span class="n">var_eff</span><span class="o">.</span><span class="n">name</span> <span class="k">for</span> <span class="n">var_eff</span> <span class="ow">in</span> <span class="n">tx_annotation</span><span class="o">.</span><span class="n">variant_effects</span><span class="p">]</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">hgvsp</span> <span class="o">=</span> <span class="kc">None</span>
                <span class="n">var_effects</span> <span class="o">=</span> <span class="kc">None</span>
            <span class="k">if</span> <span class="n">only_hgvs</span><span class="p">:</span>
                <span class="c1"># do not show the transcript id</span>
                <span class="n">fields_dna</span> <span class="o">=</span> <span class="n">display</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)</span>
                <span class="n">fields_ps</span> <span class="o">=</span> <span class="n">hgvsp</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)</span> <span class="k">if</span> <span class="n">hgvsp</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="k">else</span> <span class="p">[</span><span class="kc">None</span><span class="p">]</span>
                <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">fields_dna</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                    <span class="n">display</span> <span class="o">=</span> <span class="n">fields_dna</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">display</span> <span class="o">=</span> <span class="n">fields_dna</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
                <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">fields_ps</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                    <span class="n">hgvsp</span> <span class="o">=</span> <span class="n">fields_ps</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">hgvsp</span> <span class="o">=</span> <span class="n">fields_ps</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
            <span class="n">chrom_to_display</span><span class="p">[</span><span class="n">variant_key</span><span class="p">]</span> <span class="o">=</span> <span class="n">ToDisplay</span><span class="p">(</span><span class="n">display</span><span class="p">,</span> <span class="n">hgvsp</span><span class="p">,</span> <span class="n">var_effects</span><span class="p">)</span>

        <span class="k">return</span> <span class="n">chrom_to_display</span></div>

</pre></div>

           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>