<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Input data &mdash; GPSEA 0.4.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/gpsea.css?v=547657ed" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=6c02275b"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Cohort exploratory analysis" href="exploratory.html" />
    <link rel="prev" title="User guide" href="index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            GPSEA
          </a>
              <div class="version">
                0.4
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="index.html">User guide</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">Input data</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#create-a-cohort-from-ga4gh-phenopackets">Create a cohort from GA4GH phenopackets</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#create-cohort-creator">Create cohort creator</a></li>
<li class="toctree-l4"><a class="reference internal" href="#load-phenopackets">Load phenopackets</a></li>
<li class="toctree-l4"><a class="reference internal" href="#alternative-phenopacket-sources">Alternative phenopacket sources</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="exploratory.html">Cohort exploratory analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="predicates.html">Predicates</a></li>
<li class="toctree-l2"><a class="reference internal" href="stats.html">Statistical tests</a></li>
<li class="toctree-l2"><a class="reference internal" href="mtc.html">Multiple-testing correction</a></li>
<li class="toctree-l2"><a class="reference internal" href="glossary.html">Glossary</a></li>
<li class="toctree-l2"><a class="reference internal" href="autosomal_recessive.html">Genotype-Phenotype Correlations in Autosomal Recessive Diseases</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="index.html">User guide</a></li>
      <li class="breadcrumb-item active">Input data</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/user-guide/input-data.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="input-data">
<span id="id1"></span><h1>Input data<a class="headerlink" href="#input-data" title="Link to this heading"></a></h1>
<p>The <cite>gpsea</cite> analysis needs to be provided with a standardized form of genotype and phenotype data.
The analyses require an instance of <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a> that consists
of individuals in form of a <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Patient" title="gpsea.model.Patient"><code class="xref py py-class docutils literal notranslate"><span class="pre">Patient</span></code></a> class.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>See the <span class="xref std std-ref">cohort-exploratory</span> section for more info.</p>
</div>
<p>The first step of the <cite>gpsea</cite> analysis involves standardization of the genotype and phenotype data
and performing functional annotation of the variants. Here we describe how to prepare a <cite>Cohort</cite>
for the exploratory and downstream analysis.</p>
<section id="create-a-cohort-from-ga4gh-phenopackets">
<h2>Create a cohort from GA4GH phenopackets<a class="headerlink" href="#create-a-cohort-from-ga4gh-phenopackets" title="Link to this heading"></a></h2>
<p>The easiest way to input data into <cite>gpsea</cite> is to use the
<a class="reference external" href="https://phenopacket-schema.readthedocs.io/en/latest">GA4GH Phenopacket Schema</a> phenopackets.
<cite>gpsea</cite> provides an out-of-the-box solution for loading a cohort from a folder of phenopacket JSON files.</p>
<section id="create-cohort-creator">
<h3>Create cohort creator<a class="headerlink" href="#create-cohort-creator" title="Link to this heading"></a></h3>
<p>Next, let’s prepare a <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.CohortCreator" title="gpsea.preprocessing.CohortCreator"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortCreator</span></code></a> that will turn a phenopacket collection
into a <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>. The cohort creator also performs an input validation.
The validation needs Human Phenotype Ontology data.
Let’s start with loading Human Phenotype Ontology, a requisite for the input Q/C steps. We’ll use the amazing
<a class="reference external" href="https://github.com/TheJacksonLaboratory/hpo-toolkit">hpo-toolkit</a> library which is installed along with
the standard <cite>gpsea</cite> installation:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>The easiest way to get the <cite>CohortCreator</cite> is to use the
<a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_caching_cohort_creator" title="gpsea.preprocessing.configure_caching_cohort_creator"><code class="xref py py-func docutils literal notranslate"><span class="pre">configure_caching_cohort_creator()</span></code></a> convenience method:</p>
<div class="highlight-pycon notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span>

<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The default <cite>CohortCreator</cite> will call Variant Effect Predictor and Uniprot APIs
to perform the functional annotation and protein annotation,
and the responses will be cached in the current working directory to reduce the network bandwidth.
See the <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.configure_caching_cohort_creator" title="gpsea.preprocessing.configure_caching_cohort_creator"><code class="xref py py-func docutils literal notranslate"><span class="pre">configure_caching_cohort_creator()</span></code></a> pydoc for more options.</p>
</div>
</section>
<section id="load-phenopackets">
<h3>Load phenopackets<a class="headerlink" href="#load-phenopackets" title="Link to this heading"></a></h3>
<p>We can create a cohort starting from a collection of <cite>Phenopacket</cite> objects
provided by Python  <a class="reference external" href="https://pypi.org/project/phenopackets">Phenopackets</a> library.
For the purpose of this example, we will load a cohort of patients with pathogenic mutations in <em>RERE</em> gene
included in the release <cite>0.1.18</cite> of <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a>.
We use <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store-toolkit">Phenopacket Store Toolkit</a>
(<code class="docutils literal notranslate"><span class="pre">ppktstore</span></code> in the code) to reduce the boilerplate code needed to load the phenopackets:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">ppktstore.registry</span> <span class="kn">import</span> <span class="n">configure_phenopacket_registry</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">registry</span> <span class="o">=</span> <span class="n">configure_phenopacket_registry</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">registry</span><span class="o">.</span><span class="n">open_phenopacket_store</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;0.1.18&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">ps</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">phenopackets</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">ps</span><span class="o">.</span><span class="n">iter_cohort_phenopackets</span><span class="p">(</span><span class="s1">&#39;RERE&#39;</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">)</span>
<span class="go">19</span>
</pre></div>
</div>
<p>We loaded 19 phenopackets. Now we can turn the phenopackets into a <cite>Cohort</cite>
using the <cite>cohort_creator</cite> and the <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.load_phenopackets" title="gpsea.preprocessing.load_phenopackets"><code class="xref py py-func docutils literal notranslate"><span class="pre">load_phenopackets()</span></code></a>
loader function:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">load_phenopackets</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">qc_results</span> <span class="o">=</span> <span class="n">load_phenopackets</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>  
<span class="go">Individuals Processed: ...</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
<span class="go">19</span>
</pre></div>
</div>
<p>The cohort includes all 19 individuals.
On top of the <code class="docutils literal notranslate"><span class="pre">cohort</span></code>, the loader function also provides Q/C results  <code class="docutils literal notranslate"><span class="pre">qc_results</span></code>.
We call <code class="xref py py-meth docutils literal notranslate"><span class="pre">summarize()</span></code>
to display the Q/C summary:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">qc_results</span><span class="o">.</span><span class="n">summarize</span><span class="p">()</span>  
<span class="go">Validated under none policy</span>
<span class="go">No errors or warnings were found</span>
</pre></div>
</div>
</section>
<section id="alternative-phenopacket-sources">
<h3>Alternative phenopacket sources<a class="headerlink" href="#alternative-phenopacket-sources" title="Link to this heading"></a></h3>
<p>In case you do not already have a <cite>Phenopacket</cite> collection at your fingertips,
GPSEA provides a few other convenience functions for loading phenopackets from JSON files.</p>
<p>The <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.load_phenopacket_files" title="gpsea.preprocessing.load_phenopacket_files"><code class="xref py py-func docutils literal notranslate"><span class="pre">load_phenopacket_files()</span></code></a> function can be used to load
a bunch of phenopacket JSON files:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">load_phenopacket_files</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pp_files</span> <span class="o">=</span> <span class="p">(</span><span class="s1">&#39;path/to/phenopacket1.json&#39;</span><span class="p">,</span> <span class="s1">&#39;path/to/phenopacket2.json&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">qc_results</span> <span class="o">=</span> <span class="n">load_phenopacket_files</span><span class="p">(</span><span class="n">pp_files</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>  
</pre></div>
</div>
<p>or you can load an entire directory of JSON files with <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.load_phenopacket_folder" title="gpsea.preprocessing.load_phenopacket_folder"><code class="xref py py-func docutils literal notranslate"><span class="pre">load_phenopacket_folder()</span></code></a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">load_phenopacket_folder</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pp_dir</span> <span class="o">=</span> <span class="s1">&#39;path/to/folder/with/many/phenopacket/json/files&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">qc_results</span> <span class="o">=</span> <span class="n">load_phenopacket_folder</span><span class="p">(</span><span class="n">pp_dir</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>  
</pre></div>
</div>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="index.html" class="btn btn-neutral float-left" title="User guide" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="exploratory.html" class="btn btn-neutral float-right" title="Cohort exploratory analysis" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>