<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Statistical tests &mdash; GPSEA 0.4.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../_static/gpsea.css?v=547657ed" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=c87aa342"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../_static/copybutton.js?v=6dbb43f8"></script>
        <script>window.MathJax = {"TeX": {"Macros": {"RR": "{\\mathbb{R}}"}}}</script>
        <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Multiple-testing correction" href="mtc.html" />
    <link rel="prev" title="Groups Predicate" href="predicates/groups_predicate.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            GPSEA
          </a>
              <div class="version">
                0.4
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="index.html">User guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="input-data.html">Input data</a></li>
<li class="toctree-l2"><a class="reference internal" href="exploratory.html">Cohort exploratory analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="predicates.html">Predicates</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Statistical tests</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#compare-genotype-and-phenotype-groups">Compare genotype and phenotype groups</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#fisher-exact-test-fet">Fisher exact test (FET)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#phenotype-score">Phenotype score</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#mann-whitney-u-test">Mann-Whitney U Test</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="mtc.html">Multiple-testing correction</a></li>
<li class="toctree-l2"><a class="reference internal" href="glossary.html">Glossary</a></li>
<li class="toctree-l2"><a class="reference internal" href="autosomal_recessive.html">Genotype-Phenotype Correlations in Autosomal Recessive Diseases</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="index.html">User guide</a></li>
      <li class="breadcrumb-item active">Statistical tests</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/user-guide/stats.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="statistical-tests">
<span id="stats"></span><h1>Statistical tests<a class="headerlink" href="#statistical-tests" title="Link to this heading"></a></h1>
<p>There are many different ways of statistically testing for genotype-phenotype correlations,
and the appropriate statistical test depends on the question.
This document provides an overview of the tests offered by the GPSEA library
and explanations of how they are implemented by our software.</p>
<section id="compare-genotype-and-phenotype-groups">
<h2>Compare genotype and phenotype groups<a class="headerlink" href="#compare-genotype-and-phenotype-groups" title="Link to this heading"></a></h2>
<section id="fisher-exact-test-fet">
<span id="fisher-exact-test"></span><h3>Fisher exact test (FET)<a class="headerlink" href="#fisher-exact-test-fet" title="Link to this heading"></a></h3>
<p>The Fisher exact test (FET) calculates the exact probability value
for the relationship between two dichotomous variables.
In our implementation, the two dichotomous variables are the genotype and the phenotype.
For instance, the individuals of the cohort may be divided
according to whether or not they have a nonsense variant
and according to whether or not they have Strabismus (<a class="reference external" href="https://hpo.jax.org/browse/term/HP:0000486">HP:0000486</a>).</p>
<p>The results of FET are expressed in terms of an exact probability (P-value), varying within 0 and 1.
Two groups are considered statistically significant if the P-value is less
than the chosen significance level (usually <span class="math notranslate nohighlight">\(\alpha = 0.05\)</span>).</p>
<p>The following graphic shows an example contingency table that is used to conduct a Fisher exact test.
We are comparing the frequency of <em>strabismus</em> in individuals with missense and nonsense variants:</p>
<a class="reference internal image-reference" href="../_images/fisher.png"><img alt="Fisher exact text example" class="align-center" src="../_images/fisher.png" style="width: 600px;" /></a>
<p>To perform the corresponding test in Python, we would use the following code.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">scipy.stats</span> <span class="k">as</span> <span class="nn">stats</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">contingency_table</span> <span class="o">=</span> <span class="p">[</span>
<span class="gp">... </span><span class="c1">#   Missense,    Nonsense</span>
<span class="gp">... </span>   <span class="p">[</span><span class="mi">17</span><span class="p">,</span>          <span class="mi">6</span>       <span class="p">],</span>  <span class="c1"># Strabismus observed</span>
<span class="gp">... </span>   <span class="p">[</span><span class="mi">1</span><span class="p">,</span>          <span class="mi">19</span>       <span class="p">],</span>  <span class="c1"># Strabismus excluded</span>
<span class="gp">... </span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">oddsratio</span><span class="p">,</span> <span class="n">p_value</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">fisher_exact</span><span class="p">(</span><span class="n">contingency_table</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">float</span><span class="p">(</span><span class="n">oddsratio</span><span class="p">)</span>
<span class="go">53.833333333333336</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">float</span><span class="p">(</span><span class="n">p_value</span><span class="p">)</span>
<span class="go">5.432292015291845e-06</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">p_value</span></code> evaluates to <cite>5.432292015291845e-06</cite>, meaning there is a significant difference between the groups.</p>
<p>The Fisher exact test evaluates whether the observed frequencies in a contingency table significantly
deviate from the frequencies we would expect if there were no association between the variables.
We want to test whether the frequency of <cite>HP:0000486`</cite> is significantly higher or lower in
one genotype group compared to what would be expected if there were no association.
Note that by default, the <em>two-tailed</em> Fisher exact test is performed, meaning we have no
hypothesis as to whether there is a higher or lower frequency in one of the genotype groups.</p>
<p>However, we are typically interested in testing the associations between the genotype and multiple phenotypic features at once.
GPSEA takes advantage of the HPO structure and simplifies the testing for all HPO terms encoded in the cohort.</p>
<section id="example">
<h4>Example<a class="headerlink" href="#example" title="Link to this heading"></a></h4>
<p>Let’s illustrate this in a real-life example of the analysis of the association between frameshift variants in <em>TBX5</em> gene
and congenital heart defects in the dataset of 156 individuals with mutations in <em>TBX5</em> whose signs and symptoms were
encoded into HPO terms, stored as phenopackets of the <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/35705716">GA4GH Phenopacket Schema</a>,
and deposited in <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a> (version <cite>0.1.18</cite>).</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The shorter version of the same analysis has been presented in the <a class="reference internal" href="../tutorial.html#tutorial"><span class="std std-ref">Tutorial</span></a>.</p>
</div>
<section id="create-cohort">
<h5>Create cohort<a class="headerlink" href="#create-cohort" title="Link to this heading"></a></h5>
<p>We will load and transform the phenopackets into a <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>,
as described in <a class="reference internal" href="input-data.html#input-data"><span class="std std-ref">Input data</span></a> section. Briefly, we will load the phenopackets:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">ppktstore.registry</span> <span class="kn">import</span> <span class="n">configure_phenopacket_registry</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">registry</span> <span class="o">=</span> <span class="n">configure_phenopacket_registry</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">registry</span><span class="o">.</span><span class="n">open_phenopacket_store</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;0.1.18&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">ps</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">phenopackets</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">ps</span><span class="o">.</span><span class="n">iter_cohort_phenopackets</span><span class="p">(</span><span class="s1">&#39;TBX5&#39;</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">)</span>
<span class="go">156</span>
</pre></div>
</div>
<p>followed by loading HPO release <cite>v2024-07-01</cite>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>and we will perform Q/C and functional annotations for the mutations
with the default cohort creator:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span><span class="p">,</span> <span class="n">load_phenopackets</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">qc_results</span> <span class="o">=</span> <span class="n">load_phenopackets</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>  
<span class="go">Individuals Processed: ...</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">qc_results</span><span class="o">.</span><span class="n">summarize</span><span class="p">()</span>  
<span class="go">Validated under none policy</span>
<span class="go">No errors or warnings were found</span>
</pre></div>
</div>
</section>
<section id="configure-analysis">
<h5>Configure analysis<a class="headerlink" href="#configure-analysis" title="Link to this heading"></a></h5>
<p>We want to test the association between frameshift <em>TBX5</em> variants and phenotypic abnormalities.
GPSEA exposes a flexible predicate API that lets us create genotype and phenotype predicates
to assign the cohort members into genotype and phenotype categories based on the variants
and the HPO terms. We need to create one genotype predicate and one or more phenotype predicates.</p>
<p><strong>Genotype predicate</strong></p>
<p>We want to separate the patients into two groups: a group <em>with</em> a frameshift variant
and a group <em>without</em> a frameshift variant, based on the functional annotation.
We will use the <em>MANE</em> transcript for the analysis:</p>
<p>Building a genotype predicate is a two step process.
First, we create a <a class="reference internal" href="../apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate" title="gpsea.analysis.predicate.genotype.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a>
to test if the variant leads to a frameshift (in this case):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_181486.4&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_frameshift</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">FRAMESHIFT_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_frameshift</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;FRAMESHIFT_VARIANT on NM_181486.4&#39;</span>
</pre></div>
</div>
<p>and then we choose the expected mode of inheritance to test. In case of <em>TBX5</em>,
we expect the autosomal dominant mode of inheritance:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">autosomal_dominant</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span> <span class="o">=</span> <span class="n">autosomal_dominant</span><span class="p">(</span><span class="n">is_frameshift</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span><span class="o">.</span><span class="n">display_question</span><span class="p">()</span>
<span class="go">&#39;What is the genotype group: HOM_REF, HET&#39;</span>
</pre></div>
</div>
<p><cite>gt_predicate</cite> will assign the patients with no frameshift variant allele into <cite>HOM_REF</cite> group
and the patients with one frameshift allele will be assigned into <cite>HET</cite> group.
Note, any patient with 2 or more alleles will be <em>omitted</em> from the analysis.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Mode of inheritance testing is not the only way to dissect by a genotype.
See the <a class="reference internal" href="predicates/genotype_predicates.html#genotype-predicates"><span class="std std-ref">Genotype Predicates</span></a> section for more info.</p>
</div>
<p><strong>Phenotype predicates</strong></p>
<p>We recommend testing the genotype phenotype association for all HPO terms that are present in 2 or more cohort members,
while taking advantage of the HPO graph structure and of the <a class="reference internal" href="glossary.html#true-path-rule"><span class="std std-ref">True path rule</span></a>.
We will use the <a class="reference internal" href="../apidocs/gpsea.analysis.predicate.phenotype.html#gpsea.analysis.predicate.phenotype.prepare_predicates_for_terms_of_interest" title="gpsea.analysis.predicate.phenotype.prepare_predicates_for_terms_of_interest"><code class="xref py py-func docutils literal notranslate"><span class="pre">prepare_predicates_for_terms_of_interest()</span></code></a>
utility function to generate phenotype predicates for all HPO terms:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.phenotype</span> <span class="kn">import</span> <span class="n">prepare_predicates_for_terms_of_interest</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pheno_predicates</span> <span class="o">=</span> <span class="n">prepare_predicates_for_terms_of_interest</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">hpo</span><span class="o">=</span><span class="n">hpo</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">min_n_of_patients_with_term</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">pheno_predicates</span><span class="p">)</span>
<span class="go">260</span>
</pre></div>
</div>
<p>The function finds all HPO terms that annotate at least <em>n</em> (<code class="docutils literal notranslate"><span class="pre">min_n_of_patients_with_term=2</span></code> above) individuals,
including the <em>indirect</em> annotations whose presence is implied by the true path rule.</p>
<p><strong>Statistical test</strong></p>
<p>We will use :ref:&lt;fisher-exact-test&gt; to test the association
between genotype and phenotype groups, as described previously.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pcats.stats</span> <span class="kn">import</span> <span class="n">FisherExactTest</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">count_statistic</span> <span class="o">=</span> <span class="n">FisherExactTest</span><span class="p">()</span>
</pre></div>
</div>
<p>FET will compute a p value for each genotype phenotype group.</p>
<p><strong>Multiple testing correction</strong></p>
<p>In the case of this cohort, we could test association between having a frameshift variant and one of 260 HPO terms.
However, testing multiple hypotheses on the same dataset increases the risk of finding a significant association
by chance.
GPSEA uses a two-pronged strategy to mitigate this risk - use Phenotype MTC filter and multiple testing correction.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>See the <a class="reference internal" href="mtc.html#mtc"><span class="std std-ref">Multiple-testing correction</span></a> section for more info on multiple testing procedures.</p>
</div>
<p>Here we will use a combination of the HPO MTC filter (<a class="reference internal" href="../apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.HpoMtcFilter" title="gpsea.analysis.mtc_filter.HpoMtcFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">HpoMtcFilter</span></code></a>)
with Benjamini-Hochberg procedure (<code class="docutils literal notranslate"><span class="pre">mtc_correction='fdr_bh'</span></code>)
with a false discovery control level set to <cite>0.05</cite> (<code class="docutils literal notranslate"><span class="pre">mtc_alpha=0.05</span></code>):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.mtc_filter</span> <span class="kn">import</span> <span class="n">HpoMtcFilter</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_filter</span> <span class="o">=</span> <span class="n">HpoMtcFilter</span><span class="o">.</span><span class="n">default_filter</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">term_frequency_threshold</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_correction</span> <span class="o">=</span> <span class="s1">&#39;fdr_bh&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_alpha</span> <span class="o">=</span> <span class="mf">0.05</span>
</pre></div>
</div>
<p><strong>Final analysis</strong></p>
<p>We finalize the analysis setup by putting all components together
into <a class="reference internal" href="../apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysis" title="gpsea.analysis.pcats.HpoTermAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">HpoTermAnalysis</span></code></a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pcats</span> <span class="kn">import</span> <span class="n">HpoTermAnalysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">analysis</span> <span class="o">=</span> <span class="n">HpoTermAnalysis</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">count_statistic</span><span class="o">=</span><span class="n">count_statistic</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_filter</span><span class="o">=</span><span class="n">mtc_filter</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_correction</span><span class="o">=</span><span class="n">mtc_correction</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_alpha</span><span class="o">=</span><span class="n">mtc_alpha</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="analysis">
<h5>Analysis<a class="headerlink" href="#analysis" title="Link to this heading"></a></h5>
<p>We can now execute the analysis:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span> <span class="o">=</span> <span class="n">analysis</span><span class="o">.</span><span class="n">compare_genotype_vs_phenotypes</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">gt_predicate</span><span class="o">=</span><span class="n">gt_predicate</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">pheno_predicates</span><span class="o">=</span><span class="n">pheno_predicates</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">result</span><span class="o">.</span><span class="n">phenotypes</span><span class="p">)</span>
<span class="go">260</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">total_tests</span>
<span class="go">16</span>
</pre></div>
</div>
<p>Thanks to Phenotype MTC filter, we only tested 16 out of 260 terms.
We can learn more by showing the MTC filter report:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">MtcStatsViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_viewer</span> <span class="o">=</span> <span class="n">MtcStatsViewer</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_report</span> <span class="o">=</span> <span class="n">mtc_viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;docs/user-guide/report/tbx5_frameshift.mtc_report.html&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">fh</span><span class="p">:</span>  
<span class="gp">... </span>    <span class="n">_</span> <span class="o">=</span> <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">mtc_report</span><span class="p">)</span>
</pre></div>
</div>
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Cohort</title>
  <style>table {
    border-collapse: collapse;
    margin: 25px 0;
      font-size: 0.9em;
      font-family: sans-serif;
      min-width: 400px;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
  }


  th {
    background-color: #f2f2f2;
    border: 1px solid #dddddd;
    text-align: left;
    padding: 2px;
  }

  tr {
    border: 1px solid #dddddd;
  }

  td {
    padding: 2px;
    font-weight: bold;
  }

  tr:nth-child(even) {
    background-color: #f2f2f2;
  }

  .table td,tr {
    text-align: left;
}

.table td:first-child, tr:first-child {
    text-align: left;
}

  </style>
</head>

<body>
  <h1>Phenotype testing report</h1>
  <p>Phenotype MTC filter: <em>HPO MTC filter</em></p>
  <p>Multiple testing correction: <em>fdr_bh</em></p>
  <p>Performed statistical tests for 16 out of the total of 260 HPO terms.</p>
    <table>
      <caption>Using <em>HPO MTC filter</em>, 244 term(s) were omitted from statistical analysis.</caption>
        <tbody>
          <tr>
            <th>Code</th>
            <th>Reason</th>
            <th>Count</th>
          </tr>
          
          <tr>
            <td>HMF01</td>
            <td>Skipping term with maximum frequency that was less than threshold 0.2</td>
            <td>10</td>
          </tr>
          
          <tr>
            <td>HMF02</td>
            <td>Skipping term because no genotype has more than one observed HPO count</td>
            <td>4</td>
          </tr>
          
          <tr>
            <td>HMF03</td>
            <td>Skipping term because of a child term with the same individual counts</td>
            <td>1</td>
          </tr>
          
          <tr>
            <td>HMF04</td>
            <td>Skipping term because all genotypes have same HPO observed proportions</td>
            <td>50</td>
          </tr>
          
          <tr>
            <td>HMF05</td>
            <td>Skipping term because one genotype had zero observations</td>
            <td>5</td>
          </tr>
          
          <tr>
            <td>HMF06</td>
            <td>Skipping term with less than 7 observations (not powered for 2x2)</td>
            <td>130</td>
          </tr>
          
          <tr>
            <td>HMF08</td>
            <td>Skipping general term</td>
            <td>44</td>
          </tr>
          
        </tbody>
    </table>
</body>
</html>
</section>
<section id="genotype-phenotype-associations">
<h5>Genotype phenotype associations<a class="headerlink" href="#genotype-phenotype-associations" title="Link to this heading"></a></h5>
<p>Last, let’s explore the associations. The results include a table with all tested HPO terms
ordered by the corrected p value (Benjamini-Hochberg FDR):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.view</span> <span class="kn">import</span> <span class="n">summarize_hpo_analysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span> <span class="o">=</span> <span class="n">summarize_hpo_analysis</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">result</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;docs/user-guide/report/tbx5_frameshift.csv&#39;</span><span class="p">)</span>  
</pre></div>
</div>
<table class="docutils align-default" id="id5">
<caption><span class="caption-text"><em>TBX5</em> frameshift vs rest</span><a class="headerlink" href="#id5" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"><p>What is the genotype group</p></th>
<th class="head"><p>HOM_REF</p></th>
<th class="head"><p>HOM_REF</p></th>
<th class="head"><p>HET</p></th>
<th class="head"><p>HET</p></th>
<th class="head"></th>
<th class="head"></th>
</tr>
<tr class="row-even"><th class="head"></th>
<th class="head"><p>Count</p></th>
<th class="head"><p>Percent</p></th>
<th class="head"><p>Count</p></th>
<th class="head"><p>Percent</p></th>
<th class="head"><p>Corrected p values</p></th>
<th class="head"><p>p values</p></th>
</tr>
</thead>
<tbody>
<tr class="row-odd"><td><p>Ventricular septal defect [HP:0001629]</p></td>
<td><p>42/71</p></td>
<td><p>59%</p></td>
<td><p>19/19</p></td>
<td><p>100%</p></td>
<td><p>0.003870827221893892</p></td>
<td><p>0.00024192670136836825</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal atrioventricular conduction [HP:0005150]</p></td>
<td><p>1/23</p></td>
<td><p>4%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0.01230769230769231</p></td>
<td><p>0.0015384615384615387</p></td>
</tr>
<tr class="row-odd"><td><p>Absent thumb [HP:0009777]</p></td>
<td><p>18/100</p></td>
<td><p>18%</p></td>
<td><p>14/31</p></td>
<td><p>45%</p></td>
<td><p>0.01980624808543961</p></td>
<td><p>0.003713671516019927</p></td>
</tr>
<tr class="row-even"><td><p>Atrioventricular block [HP:0001678]</p></td>
<td><p>1/23</p></td>
<td><p>4%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0.04</p></td>
<td><p>0.01</p></td>
</tr>
<tr class="row-odd"><td><p>Patent ductus arteriosus [HP:0001643]</p></td>
<td><p>6/40</p></td>
<td><p>15%</p></td>
<td><p>2/2</p></td>
<td><p>100%</p></td>
<td><p>0.10406504065040649</p></td>
<td><p>0.03252032520325203</p></td>
</tr>
<tr class="row-even"><td><p>Secundum atrial septal defect [HP:0001684]</p></td>
<td><p>23/55</p></td>
<td><p>42%</p></td>
<td><p>4/22</p></td>
<td><p>18%</p></td>
<td><p>0.15059037690969213</p></td>
<td><p>0.06544319142266644</p></td>
</tr>
<tr class="row-odd"><td><p>Triphalangeal thumb [HP:0001199]</p></td>
<td><p>23/99</p></td>
<td><p>23%</p></td>
<td><p>13/32</p></td>
<td><p>41%</p></td>
<td><p>0.15059037690969213</p></td>
<td><p>0.06932119159387057</p></td>
</tr>
<tr class="row-even"><td><p>Cardiac conduction abnormality [HP:0031546]</p></td>
<td><p>15/37</p></td>
<td><p>41%</p></td>
<td><p>3/3</p></td>
<td><p>100%</p></td>
<td><p>0.15059037690969213</p></td>
<td><p>0.08259109311740892</p></td>
</tr>
<tr class="row-odd"><td><p>Muscular ventricular septal defect [HP:0011623]</p></td>
<td><p>8/84</p></td>
<td><p>10%</p></td>
<td><p>6/25</p></td>
<td><p>24%</p></td>
<td><p>0.15059037690969213</p></td>
<td><p>0.08470708701170182</p></td>
</tr>
<tr class="row-even"><td><p>Pulmonary arterial hypertension [HP:0002092]</p></td>
<td><p>8/14</p></td>
<td><p>57%</p></td>
<td><p>0/2</p></td>
<td><p>0%</p></td>
<td><p>0.708378140298727</p></td>
<td><p>0.4666666666666667</p></td>
</tr>
<tr class="row-odd"><td><p>Short thumb [HP:0009778]</p></td>
<td><p>25/69</p></td>
<td><p>36%</p></td>
<td><p>8/30</p></td>
<td><p>27%</p></td>
<td><p>0.708378140298727</p></td>
<td><p>0.48700997145537483</p></td>
</tr>
<tr class="row-even"><td><p>Absent radius [HP:0003974]</p></td>
<td><p>9/43</p></td>
<td><p>21%</p></td>
<td><p>6/25</p></td>
<td><p>24%</p></td>
<td><p>1.0</p></td>
<td><p>0.7703831604944444</p></td>
</tr>
<tr class="row-odd"><td><p>Atrial septal defect [HP:0001631]</p></td>
<td><p>63/65</p></td>
<td><p>97%</p></td>
<td><p>20/20</p></td>
<td><p>100%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Hypoplasia of the radius [HP:0002984]</p></td>
<td><p>34/75</p></td>
<td><p>45%</p></td>
<td><p>6/14</p></td>
<td><p>43%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Hypoplasia of the ulna [HP:0003022]</p></td>
<td><p>3/17</p></td>
<td><p>18%</p></td>
<td><p>2/10</p></td>
<td><p>20%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Short humerus [HP:0005792]</p></td>
<td><p>8/21</p></td>
<td><p>38%</p></td>
<td><p>4/9</p></td>
<td><p>44%</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
</tbody>
</table>
<p>The table shows that several HPO terms are significantly associated
with presence of a heterozygous (<cite>HET</cite>) frameshift variant in <em>TBX5</em>.
For example, <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001629">Ventricular septal defect</a>
was observed in 31/60 (52%) patients with a missense variant
but it was observed in 19/19 (100%) patients with a frameshift variant.
Fisher exact test computed a p value of <cite>~0.000242</cite>
and the p value corrected by Benjamini-Hochberg procedure
is <cite>~0.00387</cite>.</p>
<p>The table includes all HPO terms of the cohort, including the terms that were not selected for testing
and thus have no associated p value.</p>
</section>
</section>
</section>
</section>
<section id="phenotype-score">
<span id="phenotype-score-stats"></span><h2>Phenotype score<a class="headerlink" href="#phenotype-score" title="Link to this heading"></a></h2>
<section id="mann-whitney-u-test">
<span id="id1"></span><h3>Mann-Whitney U Test<a class="headerlink" href="#mann-whitney-u-test" title="Link to this heading"></a></h3>
<p>We may want to compare the total number of occurences of a specific set of phenotypic features between two different genotype groups.
For instance, <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/29330883/">Jordan et al (2018)</a> found that the total number of structural defects
of the brain, eye, heart, and kidney and sensorineural hearing loss seen in individuals with point mutations in the Atrophin-1 domain of the RERE gene
is significantly higher than expected based on the number of similar defects seen in individuals with putative loss-of-function variants.
Since there are five potential defects, each individual has a count ranging between 0 and 5.</p>
<p>We perform a Mann-Whitney U Test (or Wilcoxon Rank-Sum Test) to compare the distribution of such counts between genotype groups.
This is a non-parametric test that compares the medians of the two groups to determine if they come from the same distribution.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">scipy.stats</span> <span class="k">as</span> <span class="nn">stats</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">group1</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">0</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">group2</span> <span class="o">=</span> <span class="p">[</span><span class="mi">4</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">3</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">r</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">mannwhitneyu</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">group1</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">group2</span><span class="p">,</span> <span class="n">alternative</span> <span class="o">=</span> <span class="s1">&#39;two-sided&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">p_value</span> <span class="o">=</span> <span class="n">r</span><span class="o">.</span><span class="n">pvalue</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">float</span><span class="p">(</span><span class="n">p_value</span><span class="p">)</span>
<span class="go">6.348081479150902e-06</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">p_value</span></code> evaluates to <cite>6.348081479150901e-06</cite>, meaning there is a significant difference between the groups.</p>
<section id="id2">
<h4>Example<a class="headerlink" href="#id2" title="Link to this heading"></a></h4>
<p>Let’s now analyze the subjects reported in <em>Jordan et al</em>.
We will load 19 phenopackets that represent individuals with mutations in <em>RERE</em>
whose signs and symptoms were encoded into HPO terms and deposited into Phenopacket Store.
The phenopackets will be processed into <a class="reference internal" href="../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>
as described in the <a class="reference internal" href="input-data.html#input-data"><span class="std std-ref">Input data</span></a> section.</p>
<p>Briefly, we will first load 19 phenopackets</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">ppktstore.registry</span> <span class="kn">import</span> <span class="n">configure_phenopacket_registry</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">registry</span> <span class="o">=</span> <span class="n">configure_phenopacket_registry</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">registry</span><span class="o">.</span><span class="n">open_phenopacket_store</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;0.1.18&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">ps</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">phenopackets</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">ps</span><span class="o">.</span><span class="n">iter_cohort_phenopackets</span><span class="p">(</span><span class="s1">&#39;RERE&#39;</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">)</span>
<span class="go">19</span>
</pre></div>
</div>
<p>and load HPO (version <cite>v2024-07-01</cite>)</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>to create a <a class="reference internal" href="../apidocs/gpsea.preprocessing.html#gpsea.preprocessing.CohortCreator" title="gpsea.preprocessing.CohortCreator"><code class="xref py py-class docutils literal notranslate"><span class="pre">CohortCreator</span></code></a></p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
</pre></div>
</div>
<p>which we will use to preprocess the cohort</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">load_phenopackets</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">load_phenopackets</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>  
<span class="go">Individuals Processed: ...</span>
</pre></div>
</div>
<p>resulting in a cohort consisting of 19 individuals</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
<span class="go">19</span>
</pre></div>
</div>
<section id="id3">
<h5>Configure analysis<a class="headerlink" href="#id3" title="Link to this heading"></a></h5>
<p>Now we can set up the analysis of genotype and phenotype.
We will perform the analysis using the <em>RERE</em> transcript selected
as the “main” biologically relevant by the <cite>MANE</cite> consortium.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_001042681.2&#39;</span>
</pre></div>
</div>
<p><strong>Genotype predicate</strong></p>
<p><em>Jordan et al.</em> compare phenotype of individuals harboring point mutations
with the individuals carrying loss of function mutations.
Let’s create a predicate for testing if the variant
is a point mutation or a loss of function mutation.</p>
<p>In this example, the point mutation is a mutation that meets the following conditions:</p>
<ul class="simple">
<li><p>predicted to lead to a missense variant on the <cite>MANE</cite> transcript</p></li>
<li><p>the <a class="reference internal" href="glossary.html#length-of-the-reference-allele"><span class="std std-ref">Length of the reference allele</span></a> is equal to <cite>1</cite></p></li>
<li><p>the <a class="reference internal" href="glossary.html#change-length-of-an-allele"><span class="std std-ref">Change length of an allele</span></a> is equal to <cite>0</cite></p></li>
</ul>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">point_mutation_effects</span> <span class="o">=</span> <span class="p">(</span>
<span class="gp">... </span>    <span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">point_mutation</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">change_length</span><span class="p">(</span><span class="s1">&#39;==&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">)</span> \
<span class="gp">... </span>    <span class="o">&amp;</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">ref_length</span><span class="p">(</span><span class="s1">&#39;==&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span> \
<span class="gp">... </span>    <span class="o">&amp;</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">any</span><span class="p">(</span><span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">effect</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">)</span> <span class="k">for</span> <span class="n">effect</span> <span class="ow">in</span> <span class="n">point_mutation_effects</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">point_mutation</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;((change length == 0 AND ref allele length == 1) AND MISSENSE_VARIANT on NM_001042681.2)&#39;</span>
</pre></div>
</div>
<p>For the loss of function predicate, the following variant effects are considered loss of function:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">lof_effects</span> <span class="o">=</span> <span class="p">(</span>
<span class="gp">... </span>    <span class="n">VariantEffect</span><span class="o">.</span><span class="n">TRANSCRIPT_ABLATION</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">VariantEffect</span><span class="o">.</span><span class="n">FRAMESHIFT_VARIANT</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">VariantEffect</span><span class="o">.</span><span class="n">START_LOST</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">VariantEffect</span><span class="o">.</span><span class="n">STOP_GAINED</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lof_mutation</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">any</span><span class="p">(</span><span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">eff</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">)</span> <span class="k">for</span> <span class="n">eff</span> <span class="ow">in</span> <span class="n">lof_effects</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lof_mutation</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;(TRANSCRIPT_ABLATION on NM_001042681.2 OR FRAMESHIFT_VARIANT on NM_001042681.2 OR START_LOST on NM_001042681.2 OR STOP_GAINED on NM_001042681.2)&#39;</span>
</pre></div>
</div>
<p>The genotype predicate will bin the patient into two groups: a point mutation group or the loss of function group:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">groups_predicate</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span> <span class="o">=</span> <span class="n">groups_predicate</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">predicates</span><span class="o">=</span><span class="p">(</span><span class="n">point_mutation</span><span class="p">,</span> <span class="n">lof_mutation</span><span class="p">),</span>
<span class="gp">... </span>    <span class="n">group_names</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;Point&#39;</span><span class="p">,</span> <span class="s1">&#39;LoF&#39;</span><span class="p">),</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span><span class="o">.</span><span class="n">display_question</span><span class="p">()</span>
<span class="go">&#39;Genotype group: Point, LoF&#39;</span>
</pre></div>
</div>
<p><strong>Phenotype score</strong></p>
<p>The authors score the individuals based on the number of structural defects
from the following 5 categories:</p>
<ul class="simple">
<li><p>Brain anomalies</p></li>
<li><p>Eye anomalies</p></li>
<li><p>Congenital heart defects</p></li>
<li><p>Renal anomalies</p></li>
<li><p>Sensorineural hearing loss</p></li>
</ul>
<p>and they assign each individual a score based on the number structural defects.
For example, an individual with a congenital heart defect would be assigned a score of <cite>1</cite>,
an individual with congenital heart defect and a renal anomaly would be assigned a score of <cite>2</cite>,
and so on.</p>
<p>We automatize this scoring method by encoding the categories into HPO terms</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">structural_defects</span> <span class="o">=</span> <span class="p">(</span>
<span class="gp">... </span>    <span class="s1">&#39;HP:0012443&#39;</span><span class="p">,</span>  <span class="c1"># Abnormal brain morphology (Brain anomalies)</span>
<span class="gp">... </span>    <span class="s1">&#39;HP:0012372&#39;</span><span class="p">,</span>  <span class="c1"># Abnormal eye morphology (Eye anomalies)</span>
<span class="gp">... </span>    <span class="s1">&#39;HP:0001627&#39;</span><span class="p">,</span>  <span class="c1"># Abnormal heart morphology (Congenital heart defects)</span>
<span class="gp">... </span>    <span class="s1">&#39;HP:0012210&#39;</span><span class="p">,</span>  <span class="c1"># Abnormal renal morphology (Renal anomalies)</span>
<span class="gp">... </span>    <span class="s1">&#39;HP:0000407&#39;</span><span class="p">,</span>  <span class="c1"># Sensorineural hearing impairment (Sensorineural hearing loss)</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>and then test the individuals for presence of at least one HPO term
that corresponds to the structural defect category
(e.g. <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0012443">Abnormal brain morphology</a>)
or that is its descendant
(e.g. <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001272">Cerebellar atrophy</a>).</p>
<p>GPSEA implements this scoring method in <a class="reference internal" href="../apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.CountingPhenotypeScorer" title="gpsea.analysis.pscore.CountingPhenotypeScorer"><code class="xref py py-class docutils literal notranslate"><span class="pre">CountingPhenotypeScorer</span></code></a>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pscore</span> <span class="kn">import</span> <span class="n">CountingPhenotypeScorer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pheno_scorer</span> <span class="o">=</span> <span class="n">CountingPhenotypeScorer</span><span class="o">.</span><span class="n">from_query_curies</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">hpo</span><span class="o">=</span><span class="n">hpo</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">query</span><span class="o">=</span><span class="n">structural_defects</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Statistical test</strong></p>
<p>We will use <a class="reference internal" href="#mann-whitney-u-test"><span class="std std-ref">Mann-Whitney U Test</span></a> as described above.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pscore.stats</span> <span class="kn">import</span> <span class="n">MannWhitneyStatistic</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">score_statistic</span> <span class="o">=</span> <span class="n">MannWhitneyStatistic</span><span class="p">()</span>
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>See <a class="reference internal" href="../apidocs/gpsea.analysis.pscore.stats.html#module-gpsea.analysis.pscore.stats" title="gpsea.analysis.pscore.stats"><code class="xref py py-mod docutils literal notranslate"><span class="pre">gpsea.analysis.pscore.stats</span></code></a> module for more statistical tests available
for using with phenotype scores.</p>
</div>
<p><strong>Final analysis</strong></p>
<p>We will put the final analysis together into <a class="reference internal" href="../apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysis" title="gpsea.analysis.pscore.PhenotypeScoreAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">PhenotypeScoreAnalysis</span></code></a>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.pscore</span> <span class="kn">import</span> <span class="n">PhenotypeScoreAnalysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">score_analysis</span> <span class="o">=</span> <span class="n">PhenotypeScoreAnalysis</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">score_statistic</span><span class="o">=</span><span class="n">score_statistic</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="id4">
<h5>Analysis<a class="headerlink" href="#id4" title="Link to this heading"></a></h5>
<p>We execute the analysis by running</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span> <span class="o">=</span> <span class="n">score_analysis</span><span class="o">.</span><span class="n">compare_genotype_vs_phenotype_score</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">gt_predicate</span><span class="o">=</span><span class="n">gt_predicate</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">pheno_scorer</span><span class="o">=</span><span class="n">pheno_scorer</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>The analysis shows a significant difference between the number of structural defects
in individuals with point vs. loss-of-function mutations.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">pval</span>
<span class="go">0.012074957610483744</span>
</pre></div>
</div>
<p>To explore further, we can access a data frame with genotype categories and phenotype counts:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">scores</span> <span class="o">=</span> <span class="n">result</span><span class="o">.</span><span class="n">genotype_phenotype_scores</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">scores</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>  
<span class="go">                                     genotype phenotype</span>
<span class="go">patient_id</span>
<span class="go">Subject 10[PMID_27087320_Subject_10]        1         0</span>
<span class="go">Subject 1[PMID_27087320_Subject_1]          0         4</span>
<span class="go">Subject 1[PMID_29330883_Subject_1]          1         0</span>
<span class="go">Subject 2[PMID_27087320_Subject_2]       None         4</span>
<span class="go">Subject 2[PMID_29330883_Subject_2]          1         1</span>
</pre></div>
</div>
<p>The data frame provides a <cite>genotype</cite> category and a <cite>phenotype</cite> score for each patient.
The genotype category should be interpreted in the context of the genotype predicate:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">gt_id_to_name</span> <span class="o">=</span> <span class="p">{</span><span class="n">c</span><span class="o">.</span><span class="n">category</span><span class="o">.</span><span class="n">cat_id</span><span class="p">:</span> <span class="n">c</span><span class="o">.</span><span class="n">category</span><span class="o">.</span><span class="n">name</span> <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">gt_predicate</span><span class="o">.</span><span class="n">get_categorizations</span><span class="p">()}</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_id_to_name</span>
<span class="go">{0: &#39;Point&#39;, 1: &#39;LoF&#39;}</span>
</pre></div>
</div>
<p>The genotype code <cite>0</cite> is assigned to patients with a point mutation, <cite>1</cite> corresponds to the loss-of-function mutations,
and <cite>None</cite> is assigned to patients who cannot be assigned into any of the groups.</p>
<p>Last, let’s use <a class="reference internal" href="../apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.plot_boxplots" title="gpsea.analysis.pscore.PhenotypeScoreAnalysisResult.plot_boxplots"><code class="xref py py-meth docutils literal notranslate"><span class="pre">plot_boxplots()</span></code></a> method
to show a box plot of the phenotype score distributions:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">4</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">120</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">plot_boxplots</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">gt_predicate</span><span class="o">=</span><span class="n">gt_predicate</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">_</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s2">&quot;y&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">_</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">set</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">ylabel</span><span class="o">=</span><span class="s2">&quot;Phenotype score&quot;</span><span class="p">,</span> <span class="n">ylim</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">structural_defects</span><span class="p">)</span> <span class="o">+</span> <span class="mf">0.5</span><span class="p">)</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;docs/img/rere_phenotype_score_boxplot.png&#39;</span><span class="p">)</span>  
</pre></div>
</div>
<a class="reference internal image-reference" href="../_images/rere_phenotype_score_boxplot.png"><img alt="Phenotype score distribution" class="align-center" src="../_images/rere_phenotype_score_boxplot.png" style="width: 600px;" /></a>
<p>We see that the individuals with the point mutations feature structural defects
than the individuals with the loss-of-function mutations.</p>
<p>The box extends from the first quartile (Q1) to the third quartile (Q3) of the data,
with a red line at the median.
The whiskers extend from the box to the farthest data point
lying within 1.5x the inter-quartile range (IQR) from the box.</p>
</section>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="predicates/groups_predicate.html" class="btn btn-neutral float-left" title="Groups Predicate" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="mtc.html" class="btn btn-neutral float-right" title="Multiple-testing correction" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>