

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../../">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>gpsea.view._protein_viewer &mdash; GPSEA 0.5.1.dev0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../../../_static/css/theme.css?v=7ab3649f" />
      <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../../_static/gpsea.css?v=547657ed" />

  
      <script src="../../../_static/jquery.js?v=5d32c60e"></script>
      <script src="../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../../../_static/documentation_options.js?v=215e94d7"></script>
      <script src="../../../_static/doctools.js?v=9a2dae69"></script>
      <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../../../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Module code</a></li>
      <li class="breadcrumb-item active">gpsea.view._protein_viewer</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <h1>Source code for gpsea.view._protein_viewer</h1><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">typing</span>

<span class="kn">from</span> <span class="nn">dataclasses</span> <span class="kn">import</span> <span class="n">dataclass</span>

<span class="kn">from</span> <span class="nn">jinja2</span> <span class="kn">import</span> <span class="n">Environment</span><span class="p">,</span> <span class="n">PackageLoader</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">defaultdict</span>

<span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">Cohort</span><span class="p">,</span> <span class="n">ProteinMetadata</span>
<span class="kn">from</span> <span class="nn">gpsea.model.genome</span> <span class="kn">import</span> <span class="n">Region</span>

<span class="kn">from</span> <span class="nn">._report</span> <span class="kn">import</span> <span class="n">GpseaReport</span><span class="p">,</span> <span class="n">HtmlGpseaReport</span>


<span class="nd">@dataclass</span><span class="p">(</span><span class="n">frozen</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="k">class</span> <span class="nc">Feature</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    A private dataclass for representing a table row.</span>

<span class="sd">    Any edits to the dataclass must also be followed by an update of the Jinja template.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">name</span><span class="p">:</span> <span class="nb">str</span>
    <span class="nb">type</span><span class="p">:</span> <span class="nb">str</span>
    <span class="n">region</span><span class="p">:</span> <span class="n">Region</span>
    <span class="n">variant_count</span><span class="p">:</span> <span class="nb">int</span>


<div class="viewcode-block" id="ProteinVariantViewer">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.ProteinVariantViewer">[docs]</a>
<span class="k">class</span> <span class="nc">ProteinVariantViewer</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Class to create a pretty HTML table to display the protein information in the Jupyter notebook.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span>
                 <span class="n">protein_metadata</span><span class="p">:</span> <span class="n">ProteinMetadata</span><span class="p">,</span>
                 <span class="n">tx_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">environment</span> <span class="o">=</span> <span class="n">Environment</span><span class="p">(</span><span class="n">loader</span><span class="o">=</span><span class="p">(</span><span class="n">PackageLoader</span><span class="p">(</span><span class="s1">&#39;gpsea.view&#39;</span><span class="p">,</span> <span class="s1">&#39;templates&#39;</span><span class="p">)))</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span> <span class="o">=</span> <span class="n">environment</span><span class="o">.</span><span class="n">get_template</span><span class="p">(</span><span class="s2">&quot;protein.html&quot;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span> <span class="o">=</span> <span class="n">protein_metadata</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_tx_id</span> <span class="o">=</span> <span class="n">tx_id</span>

<div class="viewcode-block" id="ProteinVariantViewer.process">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.ProteinVariantViewer.process">[docs]</a>
    <span class="k">def</span> <span class="nf">process</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">GpseaReport</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Summarize the data regarding the protein into a HTML table.</span>

<span class="sd">        Args:</span>
<span class="sd">            cohort (Cohort): the cohort of patients being analyzed</span>

<span class="sd">        Returns:</span>
<span class="sd">            GpseaReport: a report that can be stored to a path or displayed in</span>
<span class="sd">                interactive environment such as Jupyter notebook.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">context</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_prepare_context</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
        <span class="n">html</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span><span class="o">.</span><span class="n">render</span><span class="p">(</span><span class="n">context</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">HtmlGpseaReport</span><span class="p">(</span><span class="n">html</span><span class="o">=</span><span class="n">html</span><span class="p">)</span></div>


    <span class="k">def</span> <span class="nf">_prepare_context</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Mapping</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">typing</span><span class="o">.</span><span class="n">Any</span><span class="p">]:</span>
        <span class="n">protein_id</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span><span class="o">.</span><span class="n">protein_id</span>
        <span class="n">protein_label</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span><span class="o">.</span><span class="n">label</span>
        <span class="n">protein_features</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span><span class="o">.</span><span class="n">protein_features</span>
        <span class="n">protein_features</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">protein_features</span><span class="p">,</span> <span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">f</span><span class="p">:</span> <span class="n">f</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">start</span><span class="p">)</span>
        <span class="n">protein_length</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span><span class="o">.</span><span class="n">protein_length</span>
        <span class="c1"># collect variants that are located in the protein features as well as other variants that are located</span>
        <span class="c1"># &quot;in-between&quot; the features</span>
        <span class="n">feature_to_variant_list_d</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">list</span><span class="p">)</span>
        <span class="n">non_feature_to_variant_list</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="n">n_variants_in_features</span> <span class="o">=</span> <span class="mi">0</span>

        <span class="k">for</span> <span class="n">var</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_variants</span><span class="p">():</span>
            <span class="n">target_annot</span> <span class="o">=</span> <span class="nb">next</span><span class="p">((</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">var</span><span class="o">.</span><span class="n">tx_annotations</span> <span class="k">if</span> <span class="n">x</span><span class="o">.</span><span class="n">transcript_id</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">_tx_id</span><span class="p">),</span> <span class="kc">None</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">target_annot</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="c1"># structural variants do not have a transcript id, and we skip them</span>
                <span class="c1"># It should never happen that HGVS variants do not have the proper transcript id but we do not check</span>
                <span class="c1"># that here in this visualization function</span>
                <span class="k">continue</span>
            <span class="n">hgvs_p</span> <span class="o">=</span> <span class="n">target_annot</span><span class="o">.</span><span class="n">hgvsp</span>
            <span class="k">if</span> <span class="n">hgvs_p</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
                <span class="c1"># hgvs_p is now a string such as NP_001305781.1:p.Tyr15Ter. We remove the accession id, which</span>
                <span class="c1"># is shown in the title and caption and is redundant here</span>
                <span class="n">fields</span> <span class="o">=</span> <span class="n">hgvs_p</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)</span>
                <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">fields</span><span class="p">)</span> <span class="o">==</span> <span class="mi">2</span><span class="p">:</span>
                    <span class="n">hgvs_p</span> <span class="o">=</span> <span class="n">fields</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">continue</span>
            <span class="n">target_region</span> <span class="o">=</span> <span class="n">target_annot</span><span class="o">.</span><span class="n">protein_effect_location</span>
            <span class="k">if</span> <span class="n">target_region</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="c1"># can happen for certain variant classes such as splice variant. Not an error</span>
                <span class="k">continue</span>
            <span class="n">found_overlap</span> <span class="o">=</span> <span class="kc">False</span>
            <span class="k">for</span> <span class="n">feature</span> <span class="ow">in</span> <span class="n">protein_features</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">feature</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">region</span><span class="o">.</span><span class="n">overlaps_with</span><span class="p">(</span><span class="n">target_region</span><span class="p">):</span>
                    <span class="n">found_overlap</span> <span class="o">=</span> <span class="kc">True</span>
                    <span class="n">feature_to_variant_list_d</span><span class="p">[</span><span class="n">feature</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">hgvs_p</span><span class="p">)</span>
            
            <span class="k">if</span> <span class="n">found_overlap</span><span class="p">:</span>
                <span class="n">n_variants_in_features</span> <span class="o">+=</span> <span class="mi">1</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">non_feature_to_variant_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">hgvs_p</span><span class="p">)</span>

        <span class="n">feature_list</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">feature</span> <span class="ow">in</span> <span class="n">protein_features</span><span class="p">:</span>
            <span class="n">variant_list</span> <span class="o">=</span> <span class="s2">&quot;; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">feature_to_variant_list_d</span><span class="p">[</span><span class="n">feature</span><span class="p">]))</span>
            <span class="n">feature_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                <span class="p">{</span>
                    <span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="n">feature</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">name</span><span class="p">,</span>
                    <span class="s1">&#39;type&#39;</span><span class="p">:</span> <span class="n">feature</span><span class="o">.</span><span class="n">feature_type</span><span class="o">.</span><span class="n">name</span><span class="p">,</span>
                    <span class="s1">&#39;start&#39;</span><span class="p">:</span> <span class="n">feature</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">start</span><span class="p">,</span>
                    <span class="s1">&#39;end&#39;</span><span class="p">:</span> <span class="n">feature</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">end</span><span class="p">,</span>
                    <span class="s1">&#39;count&#39;</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="n">feature_to_variant_list_d</span><span class="p">[</span><span class="n">feature</span><span class="p">]),</span>
                    <span class="s1">&#39;variants&#39;</span><span class="p">:</span> <span class="n">variant_list</span><span class="p">,</span>
                <span class="p">}</span>
            <span class="p">)</span>
            
        <span class="n">non_feature_count</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">non_feature_to_variant_list</span><span class="p">)</span>
        <span class="n">non_feature_variants</span> <span class="o">=</span> <span class="s2">&quot;; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">non_feature_to_variant_list</span><span class="p">))</span>

        <span class="k">return</span> <span class="p">{</span>
            <span class="s1">&#39;protein_id&#39;</span><span class="p">:</span> <span class="n">protein_id</span><span class="p">,</span>
            <span class="s1">&#39;protein_length&#39;</span><span class="p">:</span> <span class="n">protein_length</span><span class="p">,</span>
            <span class="s1">&#39;protein_label&#39;</span><span class="p">:</span> <span class="n">protein_label</span><span class="p">,</span>
            <span class="s1">&#39;feature_list&#39;</span><span class="p">:</span> <span class="n">feature_list</span><span class="p">,</span>
            <span class="s1">&#39;n_variants_in_features&#39;</span><span class="p">:</span> <span class="n">n_variants_in_features</span><span class="p">,</span>
            <span class="s1">&#39;n_features&#39;</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="n">feature_list</span><span class="p">),</span>
            <span class="s1">&#39;non_feature_count&#39;</span><span class="p">:</span> <span class="n">non_feature_count</span><span class="p">,</span>
            <span class="s1">&#39;non_feature_variants&#39;</span><span class="p">:</span> <span class="n">non_feature_variants</span><span class="p">,</span>
        <span class="p">}</span></div>


</pre></div>

           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>