

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../../">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>gpsea.view._txp &mdash; GPSEA 0.8.1.dev0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../../../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../../_static/gpsea.css?v=547657ed" />

  
      <script src="../../../_static/jquery.js?v=5d32c60e"></script>
      <script src="../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../../../_static/documentation_options.js?v=2556cadf"></script>
      <script src="../../../_static/doctools.js?v=9bcbadda"></script>
      <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../../../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Module code</a></li>
      <li class="breadcrumb-item active">gpsea.view._txp</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <h1>Source code for gpsea.view._txp</h1><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">typing</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">defaultdict</span>
<span class="kn">from</span> <span class="nn">matplotlib</span> <span class="kn">import</span> <span class="n">pyplot</span> <span class="k">as</span> <span class="n">plt</span>
<span class="kn">from</span> <span class="nn">matplotlib.patches</span> <span class="kn">import</span> <span class="n">Rectangle</span>
<span class="kn">from</span> <span class="nn">matplotlib.collections</span> <span class="kn">import</span> <span class="n">PatchCollection</span>
<span class="kn">from</span> <span class="nn">matplotlib.lines</span> <span class="kn">import</span> <span class="n">Line2D</span>
<span class="kn">import</span> <span class="nn">typing</span>
<span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">Variant</span><span class="p">,</span> <span class="n">TranscriptCoordinates</span><span class="p">,</span> <span class="n">ProteinMetadata</span>


<span class="k">def</span> <span class="nf">get_interpolated_location_in_protein</span><span class="p">(</span><span class="n">aa_pos_in_prot</span><span class="p">:</span><span class="nb">int</span><span class="p">,</span>
                                        <span class="n">prot_len</span><span class="p">:</span><span class="nb">int</span><span class="p">,</span>
                                        <span class="n">plot_width</span><span class="p">:</span><span class="nb">int</span><span class="o">=</span><span class="mi">1200</span><span class="p">,</span>
                                        <span class="n">margin_size</span><span class="p">:</span><span class="nb">int</span><span class="o">=</span><span class="mi">20</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Tuple</span><span class="p">[</span><span class="nb">int</span><span class="p">,</span><span class="nb">int</span><span class="p">]:</span>
    <span class="n">used_plot_width</span> <span class="o">=</span> <span class="n">plot_width</span> <span class="o">-</span> <span class="mi">2</span><span class="o">*</span><span class="n">margin_size</span>
    <span class="n">rel_pos_in_prot</span> <span class="o">=</span> <span class="n">aa_pos_in_prot</span> <span class="o">/</span> <span class="n">prot_len</span>
    <span class="n">interpolated_x</span> <span class="o">=</span> <span class="n">margin_size</span> <span class="o">+</span> <span class="n">used_plot_width</span> <span class="o">*</span> <span class="n">rel_pos_in_prot</span>
    <span class="k">return</span> <span class="n">interpolated_x</span>



<div class="viewcode-block" id="VariantTranscriptVisualizer">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.VariantTranscriptVisualizer">[docs]</a>
<span class="k">class</span> <span class="nc">VariantTranscriptVisualizer</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    `VariantTranscriptProteinArtist` creates a graphic to show distributions of variants across a provided transcript</span>
<span class="sd">    and its protein product.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="c1"># noinspection PyMethodMayBeStatic</span>
<div class="viewcode-block" id="VariantTranscriptVisualizer.draw_variants">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.VariantTranscriptVisualizer.draw_variants">[docs]</a>
    <span class="k">def</span> <span class="nf">draw_variants</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">variants</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Iterable</span><span class="p">[</span><span class="n">Variant</span><span class="p">],</span>
                    <span class="n">tx</span><span class="p">:</span> <span class="n">TranscriptCoordinates</span><span class="p">,</span>
                    <span class="n">protein</span><span class="p">:</span> <span class="n">ProteinMetadata</span><span class="p">):</span>
        <span class="n">title</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">protein</span><span class="o">.</span><span class="n">protein_id</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="n">protein</span><span class="o">.</span><span class="n">label</span><span class="si">}</span><span class="s2">)&quot;</span>
        <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">10</span><span class="p">))</span>
        <span class="n">protein_domains</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
        <span class="n">THRESHOLD</span> <span class="o">=</span> <span class="mi">2</span>
        <span class="n">BOTTOM_MARGIN</span> <span class="o">=</span> <span class="mi">20</span>
        <span class="n">amino_acid_len</span> <span class="o">=</span> <span class="n">tx</span><span class="o">.</span><span class="n">get_codon_count</span><span class="p">()</span>
        <span class="c1"># draw a box that is ten aax tall, where aax is the dimension of one amino acid</span>
        <span class="n">prot_start</span> <span class="o">=</span> <span class="n">get_interpolated_location_in_protein</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">amino_acid_len</span><span class="p">)</span>
        <span class="n">prot_end</span> <span class="o">=</span> <span class="n">get_interpolated_location_in_protein</span><span class="p">(</span><span class="n">amino_acid_len</span><span class="p">,</span> <span class="n">amino_acid_len</span><span class="p">)</span>
        <span class="n">box_height</span> <span class="o">=</span> <span class="mi">10</span><span class="o">/</span><span class="n">amino_acid_len</span>
        <span class="n">prot_width</span> <span class="o">=</span> <span class="n">prot_end</span> <span class="o">-</span> <span class="n">prot_start</span> <span class="o">+</span> <span class="mi">1</span>
        <span class="n">protein_height</span> <span class="o">=</span> <span class="n">prot_width</span><span class="o">/</span><span class="mi">20</span>
        <span class="c1">#rect = Rectangle((prot_start, BOTTOM_MARGIN), prot_width, protein_height)</span>
        <span class="n">rect</span> <span class="o">=</span> <span class="n">Rectangle</span><span class="p">((</span><span class="mi">50</span><span class="p">,</span> <span class="mi">30</span><span class="p">),</span> <span class="n">prot_width</span><span class="p">,</span> <span class="mi">40</span><span class="p">,</span> <span class="n">edgecolor</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">facecolor</span><span class="o">=</span><span class="s1">&#39;white&#39;</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.5</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">add_patch</span><span class="p">(</span><span class="n">rect</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">variant</span> <span class="ow">in</span> <span class="n">variants</span><span class="p">:</span>
            <span class="n">tx_annot_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">txa</span> <span class="k">for</span> <span class="n">txa</span> <span class="ow">in</span> <span class="n">variant</span><span class="o">.</span><span class="n">tx_annotations</span> <span class="k">if</span> <span class="n">txa</span><span class="o">.</span><span class="n">transcript_id</span> <span class="o">==</span> <span class="n">tx</span><span class="o">.</span><span class="n">identifier</span><span class="p">]</span>
            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">tx_annot_list</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Could not find variant transcript annotations for </span><span class="si">{</span><span class="n">tx</span><span class="o">.</span><span class="n">identifier</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">tx_annot_list</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                <span class="c1"># Should never happen</span>
                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Found mutiple variant transcript annotations for </span><span class="si">{</span><span class="n">tx</span><span class="o">.</span><span class="n">identifier</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
            <span class="n">tx_annot</span> <span class="o">=</span> <span class="n">tx_annot_list</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
            <span class="n">hgvs</span> <span class="o">=</span> <span class="n">tx_annot</span><span class="o">.</span><span class="n">hgvsc_id</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)</span>
            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">hgvs</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                <span class="n">hgvs_cdna</span> <span class="o">=</span> <span class="n">hgvs</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">hgvs_cdna</span> <span class="o">=</span> <span class="n">hgvs</span>
            <span class="n">variant_effects</span> <span class="o">=</span> <span class="n">tx_annot</span><span class="o">.</span><span class="n">variant_effects</span>
            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">variant_effects</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                <span class="n">var_effect</span> <span class="o">=</span> <span class="s2">&quot;MULTIPLE&quot;</span>
            <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="n">variant_effects</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
                <span class="n">var_effect</span> <span class="o">=</span> <span class="s2">&quot;UNKNOWN&quot;</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">var_effect</span> <span class="o">=</span> <span class="n">variant_effects</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">name</span>
            <span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">tx_annot</span><span class="o">.</span><span class="n">protein_affected</span><span class="p">:</span>
                <span class="k">for</span> <span class="n">f</span> <span class="ow">in</span> <span class="n">p</span><span class="o">.</span><span class="n">domains</span><span class="p">():</span>
                    <span class="n">protein_domains</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">f</span><span class="o">.</span><span class="n">info</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">tx_annot</span><span class="o">.</span><span class="n">protein_effect_location</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;skipping </span><span class="si">{</span><span class="n">hgvs_cdna</span><span class="si">}</span><span class="s2"> because no protein effect location was found&quot;</span><span class="p">)</span>
                <span class="k">continue</span>
            <span class="c1"># for plotting, we will just show the start position; note there is also end</span>
            <span class="n">pe_loc</span> <span class="o">=</span> <span class="n">tx_annot</span><span class="o">.</span><span class="n">protein_effect_location</span><span class="o">.</span><span class="n">start</span>
            <span class="n">prot_variant_location</span> <span class="o">=</span> <span class="n">get_interpolated_location_in_protein</span><span class="p">(</span><span class="n">pe_loc</span><span class="p">,</span> <span class="n">amino_acid_len</span><span class="p">)</span>
            <span class="n">x</span> <span class="o">=</span> <span class="p">[</span><span class="n">prot_variant_location</span><span class="p">,</span> <span class="n">prot_variant_location</span><span class="p">]</span>
            <span class="n">ybase</span> <span class="o">=</span> <span class="n">BOTTOM_MARGIN</span> <span class="o">+</span> <span class="n">protein_height</span>
            <span class="n">ytop</span> <span class="o">=</span> <span class="n">ybase</span> <span class="o">+</span> <span class="mi">10</span>
            <span class="n">y</span> <span class="o">=</span> <span class="p">[</span><span class="n">ybase</span><span class="p">,</span> <span class="n">ytop</span><span class="p">]</span>
            <span class="n">line</span> <span class="o">=</span> <span class="n">Line2D</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">)</span>
            <span class="n">ax</span><span class="o">.</span><span class="n">add_line</span><span class="p">(</span><span class="n">line</span><span class="p">)</span>
            <span class="c1"># TODO show one box per mutation or otherwise represent frequency</span>
            <span class="n">color_list</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;red&#39;</span><span class="p">,</span> <span class="s1">&#39;blue&#39;</span><span class="p">,</span> <span class="s1">&#39;green&#39;</span><span class="p">,</span> <span class="s1">&#39;orange&#39;</span><span class="p">,</span> <span class="s1">&#39;brown&#39;</span><span class="p">,</span> <span class="s1">&#39;yellow&#39;</span><span class="p">,</span> <span class="s1">&#39;purple&#39;</span><span class="p">]</span>
            <span class="n">feature_to_color_index_d</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">int</span><span class="p">)</span> <span class="c1"># default is red</span>
            <span class="k">for</span> <span class="n">feature</span> <span class="ow">in</span> <span class="n">protein_domains</span><span class="p">:</span>
                <span class="n">name</span> <span class="o">=</span> <span class="n">feature</span><span class="o">.</span><span class="n">name</span>
                <span class="k">if</span> <span class="n">name</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">feature_to_color_index_d</span><span class="p">:</span>
                    <span class="n">new_idx</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">feature_to_color_index_d</span><span class="p">)</span> <span class="o">%</span> <span class="nb">len</span><span class="p">(</span><span class="n">color_list</span><span class="p">)</span>
                    <span class="n">feature_to_color_index_d</span><span class="p">[</span><span class="n">name</span><span class="p">]</span> <span class="o">=</span> <span class="n">new_idx</span>
                <span class="n">color_idx</span> <span class="o">=</span> <span class="n">feature_to_color_index_d</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">name</span><span class="p">)</span>
                <span class="n">box_color</span> <span class="o">=</span> <span class="n">color_list</span><span class="p">[</span><span class="n">color_idx</span><span class="p">]</span>
                <span class="n">start</span> <span class="o">=</span> <span class="n">feature</span><span class="o">.</span><span class="n">start</span>
                <span class="n">end</span> <span class="o">=</span> <span class="n">feature</span><span class="o">.</span><span class="n">end</span>
                <span class="c1">#print(name, start, end, box_color)</span>
                <span class="n">box_height</span> <span class="o">=</span> <span class="mi">10</span><span class="o">/</span><span class="n">amino_acid_len</span>
                <span class="n">prot_width</span> <span class="o">=</span> <span class="n">prot_end</span> <span class="o">-</span> <span class="n">prot_start</span> <span class="o">+</span> <span class="mi">1</span>
                <span class="n">protein_height</span> <span class="o">=</span> <span class="n">prot_width</span><span class="o">/</span><span class="mi">20</span>
                <span class="c1">#rect = Rectangle((prot_start, BOTTOM_MARGIN), prot_width, protein_height)</span>
                <span class="n">box_start</span> <span class="o">=</span> <span class="n">get_interpolated_location_in_protein</span><span class="p">(</span><span class="n">start</span><span class="p">,</span>  <span class="n">amino_acid_len</span><span class="p">)</span>
                <span class="n">box_end</span> <span class="o">=</span> <span class="n">get_interpolated_location_in_protein</span><span class="p">(</span><span class="n">end</span><span class="p">,</span> <span class="n">amino_acid_len</span><span class="p">)</span>
                <span class="n">box_width</span> <span class="o">=</span> <span class="n">box_end</span> <span class="o">-</span> <span class="n">box_start</span>
                <span class="n">rect</span> <span class="o">=</span> <span class="n">Rectangle</span><span class="p">((</span><span class="n">box_start</span><span class="p">,</span> <span class="mi">30</span><span class="p">),</span> <span class="n">box_width</span><span class="p">,</span> <span class="mi">40</span><span class="p">,</span> <span class="n">edgecolor</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">facecolor</span><span class="o">=</span><span class="n">box_color</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.5</span><span class="p">)</span>
                <span class="n">ax</span><span class="o">.</span><span class="n">add_patch</span><span class="p">(</span><span class="n">rect</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">title</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="s1">&#39;x-large&#39;</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;X Label&#39;</span><span class="p">,</span> <span class="n">labelpad</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mf">14.5</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Y label&#39;</span><span class="p">,</span> <span class="n">labelpad</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mf">14.5</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mi">10</span><span class="p">,</span> <span class="mi">1200</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">120</span><span class="p">)</span></div>
</div>

</pre></div>

           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>