

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../../">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>gpsea.view._viewers &mdash; GPSEA 0.9.4.dev0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="../../../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../../_static/gpsea.css?v=547657ed" />

  
      <script src="../../../_static/jquery.js?v=5d32c60e"></script>
      <script src="../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../../../_static/documentation_options.js?v=d641ac61"></script>
      <script src="../../../_static/doctools.js?v=9bcbadda"></script>
      <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../../../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Module code</a></li>
      <li class="breadcrumb-item active">gpsea.view._viewers</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <h1>Source code for gpsea.view._viewers</h1><div class="highlight"><pre>
<span></span><span class="kn">import</span><span class="w"> </span><span class="nn">typing</span>

<span class="kn">from</span><span class="w"> </span><span class="nn">collections</span><span class="w"> </span><span class="kn">import</span> <span class="n">namedtuple</span><span class="p">,</span> <span class="n">defaultdict</span><span class="p">,</span> <span class="n">Counter</span>

<span class="kn">import</span><span class="w"> </span><span class="nn">hpotk</span>

<span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.model</span><span class="w"> </span><span class="kn">import</span> <span class="n">Cohort</span><span class="p">,</span> <span class="n">Variant</span><span class="p">,</span> <span class="n">VariantEffect</span><span class="p">,</span> <span class="n">ProteinMetadata</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.pcats</span><span class="w"> </span><span class="kn">import</span> <span class="n">HpoTermAnalysisResult</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">._base</span><span class="w"> </span><span class="kn">import</span> <span class="n">BaseViewer</span><span class="p">,</span> <span class="n">GpseaReport</span><span class="p">,</span> <span class="n">HtmlGpseaReport</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">._formatter</span><span class="w"> </span><span class="kn">import</span> <span class="n">VariantFormatter</span>


<span class="n">ToDisplay</span> <span class="o">=</span> <span class="n">namedtuple</span><span class="p">(</span><span class="s2">&quot;ToDisplay&quot;</span><span class="p">,</span> <span class="p">[</span><span class="s2">&quot;hgvs_cdna&quot;</span><span class="p">,</span> <span class="s2">&quot;hgvsp&quot;</span><span class="p">,</span> <span class="s2">&quot;variant_effects&quot;</span><span class="p">])</span>
<span class="n">IdentifiedCount</span> <span class="o">=</span> <span class="n">namedtuple</span><span class="p">(</span><span class="s2">&quot;IdentifiedCount&quot;</span><span class="p">,</span> <span class="p">(</span><span class="s2">&quot;term_id&quot;</span><span class="p">,</span> <span class="s2">&quot;label&quot;</span><span class="p">,</span> <span class="s2">&quot;count&quot;</span><span class="p">))</span>
<span class="n">VariantData</span> <span class="o">=</span> <span class="n">namedtuple</span><span class="p">(</span>
    <span class="s2">&quot;VariantData&quot;</span><span class="p">,</span> <span class="p">[</span><span class="s2">&quot;variant_key&quot;</span><span class="p">,</span> <span class="s2">&quot;hgvsc&quot;</span><span class="p">,</span> <span class="s2">&quot;hgvsp&quot;</span><span class="p">,</span> <span class="s2">&quot;variant_effects&quot;</span><span class="p">,</span> <span class="s2">&quot;exons&quot;</span><span class="p">]</span>
<span class="p">)</span>


<div class="viewcode-block" id="CohortViewer">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.CohortViewer">[docs]</a>
<span class="k">class</span><span class="w"> </span><span class="nc">CohortViewer</span><span class="p">(</span><span class="n">BaseViewer</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    `CohortViewer` summarizes the most salient :class:`~gpsea.model.Cohort` aspects into an HTML report.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span><span class="w"> </span><span class="fm">__init__</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">hpo</span><span class="p">:</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">MinimalOntology</span><span class="p">,</span>
        <span class="n">top_phenotype_count</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">10</span><span class="p">,</span>
        <span class="n">top_variant_count</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">10</span><span class="p">,</span>
    <span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Args:</span>
<span class="sd">            hpo(MinimalOntology): An HPO ontology object from hpo-toolkit</span>
<span class="sd">            tx_id(str): the transcript accession `str` (e.g. `NM_123456.7`)</span>
<span class="sd">            top_phenotype_count(int): Maximum number of phenotype items (HPO terms, measurements, ...) to display in the HTML table (default: 10)</span>
<span class="sd">            top_variant_count(int): Maximum number of variants to display in the HTML table (default: 10)</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="nb">super</span><span class="p">()</span><span class="o">.</span><span class="fm">__init__</span><span class="p">()</span>
        <span class="k">assert</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">MinimalOntology</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_hpo</span> <span class="o">=</span> <span class="n">hpo</span>
        <span class="k">assert</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">top_phenotype_count</span><span class="p">,</span> <span class="nb">int</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_top_phenotype_count</span> <span class="o">=</span> <span class="n">top_phenotype_count</span>
        <span class="k">assert</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">top_variant_count</span><span class="p">,</span> <span class="nb">int</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_top_variant_count</span> <span class="o">=</span> <span class="n">top_variant_count</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_environment</span><span class="o">.</span><span class="n">get_template</span><span class="p">(</span><span class="s2">&quot;cohort.html&quot;</span><span class="p">)</span>

<div class="viewcode-block" id="CohortViewer.process">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.CohortViewer.process">[docs]</a>
    <span class="k">def</span><span class="w"> </span><span class="nf">process</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
        <span class="n">transcript_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">GpseaReport</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Generate the report for a given `cohort`.</span>

<span class="sd">        The variant effects will be reported with respect to the provided transcript.</span>

<span class="sd">        Args:</span>
<span class="sd">            cohort (Cohort): the cohort to visualize</span>
<span class="sd">            transcript_id (str): the accession of the target transcript (e.g. `NM_123456.7`)</span>

<span class="sd">        Returns:</span>
<span class="sd">            GpseaReport: a report that can be stored to a path or displayed in</span>
<span class="sd">                interactive environment such as Jupyter notebook.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">context</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_prepare_context</span><span class="p">(</span><span class="n">cohort</span><span class="p">,</span> <span class="n">transcript_id</span><span class="o">=</span><span class="n">transcript_id</span><span class="p">)</span>
        <span class="n">report</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span><span class="o">.</span><span class="n">render</span><span class="p">(</span><span class="n">context</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">HtmlGpseaReport</span><span class="p">(</span><span class="n">html</span><span class="o">=</span><span class="n">report</span><span class="p">)</span></div>


    <span class="k">def</span><span class="w"> </span><span class="nf">_prepare_context</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
        <span class="n">transcript_id</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Mapping</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">typing</span><span class="o">.</span><span class="n">Any</span><span class="p">]:</span>
        <span class="n">hpo_counts</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_summarize_hpo_counts</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>

        <span class="n">measurement_counts</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_summarize_measurement_counts</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>

        <span class="n">disease_counts</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_summarize_disease_counts</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>

        <span class="n">variant_counts</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="n">variant_to_display_d</span> <span class="o">=</span> <span class="n">CohortViewer</span><span class="o">.</span><span class="n">_get_variant_description</span><span class="p">(</span>
            <span class="n">cohort</span><span class="p">,</span> <span class="n">transcript_id</span>
        <span class="p">)</span>
        <span class="k">for</span> <span class="n">variant_key</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">list_all_variants</span><span class="p">(</span><span class="n">top</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">_top_variant_count</span><span class="p">):</span>
            <span class="c1"># get HGVS or human readable variant</span>
            <span class="k">if</span> <span class="n">variant_key</span> <span class="ow">in</span> <span class="n">variant_to_display_d</span><span class="p">:</span>
                <span class="n">display</span> <span class="o">=</span> <span class="n">variant_to_display_d</span><span class="p">[</span><span class="n">variant_key</span><span class="p">]</span>
                <span class="n">hgvsc</span> <span class="o">=</span> <span class="n">display</span><span class="o">.</span><span class="n">hgvs_cdna</span>
                <span class="n">hgvsp</span> <span class="o">=</span> <span class="n">display</span><span class="o">.</span><span class="n">hgvsp</span>
                <span class="n">effects</span> <span class="o">=</span> <span class="p">(</span>
                    <span class="s2">&quot;&quot;</span>
                    <span class="k">if</span> <span class="n">display</span><span class="o">.</span><span class="n">variant_effects</span> <span class="ow">is</span> <span class="kc">None</span>
                    <span class="k">else</span> <span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">display</span><span class="o">.</span><span class="n">variant_effects</span><span class="p">)</span>
                <span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">hgvsc</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
                <span class="n">hgvsp</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
                <span class="n">effects</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>

            <span class="n">variant_counts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                <span class="p">{</span>
                    <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="n">count</span><span class="p">,</span>
                    <span class="s2">&quot;key&quot;</span><span class="p">:</span> <span class="n">variant_key</span><span class="p">,</span>
                    <span class="s2">&quot;hgvsc&quot;</span><span class="p">:</span> <span class="n">hgvsc</span><span class="p">,</span>
                    <span class="s2">&quot;hgvsp&quot;</span><span class="p">:</span> <span class="n">hgvsp</span><span class="p">,</span>
                    <span class="s2">&quot;effects&quot;</span><span class="p">:</span> <span class="n">effects</span><span class="p">,</span>
                <span class="p">}</span>
            <span class="p">)</span>

        <span class="n">variant_effects</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="n">has_transcript</span> <span class="o">=</span> <span class="n">transcript_id</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span>
        <span class="k">if</span> <span class="n">has_transcript</span><span class="p">:</span>
            <span class="n">var_effects_d</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
            <span class="k">for</span> <span class="n">tx_id</span><span class="p">,</span> <span class="n">counter</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">variant_effect_count_by_tx</span><span class="p">(</span>
                <span class="n">tx_id</span><span class="o">=</span><span class="n">transcript_id</span>
            <span class="p">)</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
                <span class="c1"># e.g., data structure</span>
                <span class="c1">#   -- {&#39;effect}&#39;: &#39;FRAMESHIFT_VARIANT&#39;, &#39;count&#39;: 175},</span>
                <span class="c1">#   -- {&#39;effect}&#39;: &#39;STOP_GAINED&#39;, &#39;count&#39;: 67},</span>
                <span class="k">if</span> <span class="n">tx_id</span> <span class="o">==</span> <span class="n">transcript_id</span><span class="p">:</span>
                    <span class="k">for</span> <span class="n">effect</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">counter</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
                        <span class="n">var_effects_d</span><span class="p">[</span><span class="n">effect</span><span class="p">]</span> <span class="o">=</span> <span class="n">count</span>
                    <span class="k">break</span>
            <span class="n">total</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">var_effects_d</span><span class="o">.</span><span class="n">values</span><span class="p">())</span>
            <span class="c1"># Sort in descending order based on counts</span>
            <span class="k">for</span> <span class="n">effect</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="nb">sorted</span><span class="p">(</span>
                <span class="n">var_effects_d</span><span class="o">.</span><span class="n">items</span><span class="p">(),</span> <span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">item</span><span class="p">:</span> <span class="n">item</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">reverse</span><span class="o">=</span><span class="kc">True</span>
            <span class="p">):</span>
                <span class="n">variant_effects</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                    <span class="p">{</span>
                        <span class="s2">&quot;effect&quot;</span><span class="p">:</span> <span class="n">effect</span><span class="p">,</span>
                        <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="n">count</span><span class="p">,</span>
                        <span class="s2">&quot;percent&quot;</span><span class="p">:</span> <span class="nb">round</span><span class="p">(</span><span class="n">count</span> <span class="o">/</span> <span class="n">total</span> <span class="o">*</span> <span class="mi">100</span><span class="p">),</span>
                    <span class="p">}</span>
                <span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">transcript_id</span> <span class="o">=</span> <span class="s2">&quot;MANE transcript ID&quot;</span>

        <span class="n">n_has_disease_onset</span> <span class="o">=</span> <span class="mi">0</span>
        <span class="n">hpo_id_to_has_cnset_count_d</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">int</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">pat</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_patients</span><span class="p">:</span>
            <span class="n">diseases</span> <span class="o">=</span> <span class="n">pat</span><span class="o">.</span><span class="n">diseases</span>
            <span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">diseases</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">d</span><span class="o">.</span><span class="n">onset</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
                    <span class="n">n_has_disease_onset</span> <span class="o">+=</span> <span class="mi">1</span>
                    <span class="k">break</span>  <span class="c1"># for now, do this for any diseases. All of our current phenopackets habve but one disease</span>
            <span class="n">terms_and_ancestors_with_onset_information</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
            <span class="k">for</span> <span class="n">pf</span> <span class="ow">in</span> <span class="n">pat</span><span class="o">.</span><span class="n">present_phenotypes</span><span class="p">():</span>
                <span class="k">if</span> <span class="n">pf</span><span class="o">.</span><span class="n">onset</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
                    <span class="n">ancs</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hpo</span><span class="o">.</span><span class="n">graph</span><span class="o">.</span><span class="n">get_ancestors</span><span class="p">(</span>
                        <span class="n">pf</span><span class="o">.</span><span class="n">identifier</span><span class="p">,</span> <span class="n">include_source</span><span class="o">=</span><span class="kc">True</span>
                    <span class="p">)</span>
                    <span class="n">terms_and_ancestors_with_onset_information</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">ancs</span><span class="p">)</span>
            <span class="k">for</span> <span class="n">hpo_id</span> <span class="ow">in</span> <span class="n">terms_and_ancestors_with_onset_information</span><span class="p">:</span>
                <span class="n">hpo_id_to_has_cnset_count_d</span><span class="p">[</span><span class="n">hpo_id</span><span class="p">]</span> <span class="o">+=</span> <span class="mi">1</span>

        <span class="c1"># When we get here, we want to present the counts of HPO terms that have onset information</span>
        <span class="c1"># onset_threshold = int(0.2 * len(cohort))  # only show terms with a decent amount of information</span>
        <span class="c1"># has_onset_information = list()</span>
        <span class="c1"># for hpo_id, count in hpo_id_to_has_cnset_count_d.items():</span>
        <span class="c1">#     if count &lt; onset_threshold:</span>
        <span class="c1">#         continue</span>
        <span class="c1">#     label = self._hpo.get_term_name(hpo_id)</span>
        <span class="c1">#     if label is not None:</span>
        <span class="c1">#         display_label = f&quot;{label} ({hpo_id})&quot;</span>
        <span class="c1">#     else:</span>
        <span class="c1">#         display_label = hpo_id</span>
        <span class="c1">#     has_onset_information.append(</span>
        <span class="c1">#         {&quot;HPO&quot;: display_label, &quot;count&quot;: count})</span>
        <span class="c1"># n_has_onset_info = len(has_onset_information)</span>

        <span class="c1"># The following dictionary is used by the Jinja2 HTML template</span>
        <span class="k">return</span> <span class="p">{</span>
            <span class="s2">&quot;cohort&quot;</span><span class="p">:</span> <span class="n">cohort</span><span class="p">,</span>
            <span class="s2">&quot;transcript_id&quot;</span><span class="p">:</span> <span class="n">transcript_id</span><span class="p">,</span>
            <span class="s2">&quot;has_transcript&quot;</span><span class="p">:</span> <span class="n">has_transcript</span><span class="p">,</span>
            <span class="s2">&quot;top_phenotype_count&quot;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">_top_phenotype_count</span><span class="p">,</span>
            <span class="s2">&quot;hpo_counts&quot;</span><span class="p">:</span> <span class="n">hpo_counts</span><span class="p">,</span>
            <span class="s2">&quot;measurement_counts&quot;</span><span class="p">:</span> <span class="n">measurement_counts</span><span class="p">,</span>
            <span class="s2">&quot;disease_counts&quot;</span><span class="p">:</span> <span class="n">disease_counts</span><span class="p">,</span>
            <span class="s2">&quot;top_var_count&quot;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">_top_variant_count</span><span class="p">,</span>
            <span class="s2">&quot;var_counts&quot;</span><span class="p">:</span> <span class="n">variant_counts</span><span class="p">,</span>
            <span class="s2">&quot;variant_effects&quot;</span><span class="p">:</span> <span class="n">variant_effects</span><span class="p">,</span>
        <span class="p">}</span>

    <span class="k">def</span><span class="w"> </span><span class="nf">_summarize_hpo_counts</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">IdentifiedCount</span><span class="p">]:</span>
        <span class="n">counts</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">term_id</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">list_present_phenotypes</span><span class="p">(</span>
            <span class="n">top</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">_top_phenotype_count</span><span class="p">,</span>
        <span class="p">):</span>
            <span class="n">label</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hpo</span><span class="o">.</span><span class="n">get_term_name</span><span class="p">(</span><span class="n">term_id</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">label</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;N/A&quot;</span>

            <span class="n">counts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                <span class="n">IdentifiedCount</span><span class="p">(</span>
                    <span class="n">term_id</span><span class="o">=</span><span class="n">term_id</span><span class="p">,</span>
                    <span class="n">label</span><span class="o">=</span><span class="n">label</span><span class="p">,</span>
                    <span class="n">count</span><span class="o">=</span><span class="n">count</span><span class="p">,</span>
                <span class="p">)</span>
            <span class="p">)</span>

        <span class="k">return</span> <span class="n">counts</span>

    <span class="k">def</span><span class="w"> </span><span class="nf">_summarize_measurement_counts</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">IdentifiedCount</span><span class="p">]:</span>
        <span class="n">measurement2label</span> <span class="o">=</span> <span class="p">{</span>
            <span class="n">m</span><span class="o">.</span><span class="n">identifier</span><span class="o">.</span><span class="n">value</span><span class="p">:</span> <span class="n">m</span><span class="o">.</span><span class="n">name</span> <span class="k">for</span> <span class="n">m</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_measurements</span><span class="p">()</span>
        <span class="p">}</span>

        <span class="k">return</span> <span class="p">[</span>
            <span class="n">IdentifiedCount</span><span class="p">(</span>
                <span class="n">term_id</span><span class="o">=</span><span class="n">measurement_id</span><span class="p">,</span>
                <span class="n">label</span><span class="o">=</span><span class="n">measurement2label</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">measurement_id</span><span class="p">,</span> <span class="s2">&quot;N/A&quot;</span><span class="p">),</span>
                <span class="n">count</span><span class="o">=</span><span class="n">count</span><span class="p">,</span>
            <span class="p">)</span>
            <span class="k">for</span> <span class="n">measurement_id</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">list_measurements</span><span class="p">(</span>
                <span class="n">top</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">_top_phenotype_count</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="p">]</span>

    <span class="k">def</span><span class="w"> </span><span class="nf">_summarize_disease_counts</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">IdentifiedCount</span><span class="p">]:</span>
        <span class="n">disease2label</span> <span class="o">=</span> <span class="p">{</span><span class="n">m</span><span class="o">.</span><span class="n">identifier</span><span class="o">.</span><span class="n">value</span><span class="p">:</span> <span class="n">m</span><span class="o">.</span><span class="n">name</span> <span class="k">for</span> <span class="n">m</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_diseases</span><span class="p">()}</span>

        <span class="k">return</span> <span class="p">[</span>
            <span class="n">IdentifiedCount</span><span class="p">(</span>
                <span class="n">term_id</span><span class="o">=</span><span class="n">disease_id</span><span class="p">,</span>
                <span class="n">label</span><span class="o">=</span><span class="n">disease2label</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">disease_id</span><span class="p">,</span> <span class="s2">&quot;N/A&quot;</span><span class="p">),</span>
                <span class="n">count</span><span class="o">=</span><span class="n">count</span><span class="p">,</span>
            <span class="p">)</span>
            <span class="k">for</span> <span class="n">disease_id</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">list_all_diseases</span><span class="p">(</span>
                <span class="n">top</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">_top_phenotype_count</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="p">]</span>

    <span class="nd">@staticmethod</span>
    <span class="k">def</span><span class="w"> </span><span class="nf">_get_variant_description</span><span class="p">(</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
        <span class="n">transcript_id</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">],</span>
        <span class="n">only_hgvs</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Mapping</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">ToDisplay</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get user-friendly strings (e.g., HGVS for our target transcript) to match to the chromosomal strings</span>
<span class="sd">        Args:</span>
<span class="sd">            cohort (Cohort): The cohort being analyzed in the current Notebook</span>
<span class="sd">            transcript_id (str): the transcript that we map variants onto</span>
<span class="sd">            only_hgvs (bool): do not show the transcript ID part of the HGVS annotation, just the annotation.</span>

<span class="sd">        Returns:</span>
<span class="sd">            typing.Mapping[str, ToDisplay]:</span>
<span class="sd">              key: variant key,</span>
<span class="sd">              value: namedtuple(display (e.g. HGVS) string of variant, hgvsp protein string of variant)</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">chrom_to_display</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>
        <span class="n">var_formatter</span> <span class="o">=</span> <span class="n">VariantFormatter</span><span class="p">(</span><span class="n">transcript_id</span><span class="p">)</span>

        <span class="k">for</span> <span class="n">var</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_variants</span><span class="p">():</span>
            <span class="n">variant_key</span> <span class="o">=</span> <span class="n">var</span><span class="o">.</span><span class="n">variant_info</span><span class="o">.</span><span class="n">variant_key</span>
            <span class="n">display</span> <span class="o">=</span> <span class="n">var_formatter</span><span class="o">.</span><span class="n">format_as_string</span><span class="p">(</span><span class="n">var</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">transcript_id</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="n">tx_annotation</span> <span class="o">=</span> <span class="kc">None</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">tx_annotation</span> <span class="o">=</span> <span class="n">var</span><span class="o">.</span><span class="n">get_tx_anno_by_tx_id</span><span class="p">(</span><span class="n">transcript_id</span><span class="p">)</span>

            <span class="k">if</span> <span class="n">tx_annotation</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="n">hgvsp</span> <span class="o">=</span> <span class="kc">None</span>
                <span class="n">var_effects</span> <span class="o">=</span> <span class="kc">None</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">hgvsp</span> <span class="o">=</span> <span class="n">tx_annotation</span><span class="o">.</span><span class="n">hgvsp</span>
                <span class="n">var_effects</span> <span class="o">=</span> <span class="p">[</span>
                    <span class="n">var_eff</span><span class="o">.</span><span class="n">name</span> <span class="k">for</span> <span class="n">var_eff</span> <span class="ow">in</span> <span class="n">tx_annotation</span><span class="o">.</span><span class="n">variant_effects</span>
                <span class="p">]</span>

            <span class="k">if</span> <span class="n">only_hgvs</span><span class="p">:</span>
                <span class="c1"># do not show the transcript id</span>
                <span class="n">fields_dna</span> <span class="o">=</span> <span class="n">display</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)</span>
                <span class="n">fields_ps</span> <span class="o">=</span> <span class="n">hgvsp</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)</span> <span class="k">if</span> <span class="n">hgvsp</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span> <span class="k">else</span> <span class="p">[</span><span class="kc">None</span><span class="p">]</span>
                <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">fields_dna</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                    <span class="n">display</span> <span class="o">=</span> <span class="n">fields_dna</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">display</span> <span class="o">=</span> <span class="n">fields_dna</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
                <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">fields_ps</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                    <span class="n">hgvsp</span> <span class="o">=</span> <span class="n">fields_ps</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">hgvsp</span> <span class="o">=</span> <span class="n">fields_ps</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
            <span class="n">chrom_to_display</span><span class="p">[</span><span class="n">variant_key</span><span class="p">]</span> <span class="o">=</span> <span class="n">ToDisplay</span><span class="p">(</span><span class="n">display</span><span class="p">,</span> <span class="n">hgvsp</span><span class="p">,</span> <span class="n">var_effects</span><span class="p">)</span>

        <span class="k">return</span> <span class="n">chrom_to_display</span></div>



<div class="viewcode-block" id="DiseaseViewer">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.DiseaseViewer">[docs]</a>
<span class="k">class</span><span class="w"> </span><span class="nc">DiseaseViewer</span><span class="p">(</span><span class="n">BaseViewer</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    TODO</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span><span class="w"> </span><span class="fm">__init__</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span> <span class="n">hpo</span><span class="p">:</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">MinimalOntology</span><span class="p">,</span> <span class="n">transcript_id</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span>
    <span class="p">):</span>
        <span class="nb">super</span><span class="p">()</span><span class="o">.</span><span class="fm">__init__</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_hpo</span> <span class="o">=</span> <span class="n">hpo</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_tx_id</span> <span class="o">=</span> <span class="n">transcript_id</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_environment</span><span class="o">.</span><span class="n">get_template</span><span class="p">(</span><span class="s2">&quot;disease.html&quot;</span><span class="p">)</span>

<div class="viewcode-block" id="DiseaseViewer.process">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.DiseaseViewer.process">[docs]</a>
    <span class="k">def</span><span class="w"> </span><span class="nf">process</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">GpseaReport</span><span class="p">:</span>
        <span class="n">context</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_prepare_context</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
        <span class="n">html</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span><span class="o">.</span><span class="n">render</span><span class="p">(</span><span class="n">context</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">HtmlGpseaReport</span><span class="p">(</span><span class="n">html</span><span class="o">=</span><span class="n">html</span><span class="p">)</span></div>


    <span class="k">def</span><span class="w"> </span><span class="nf">_prepare_context</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Mapping</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">typing</span><span class="o">.</span><span class="n">Any</span><span class="p">]:</span>
        <span class="n">diseases</span> <span class="o">=</span> <span class="n">cohort</span><span class="o">.</span><span class="n">list_all_diseases</span><span class="p">()</span>
        <span class="n">n_diseases</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">diseases</span><span class="p">)</span>
        <span class="n">disease_counts</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="n">disease_variants_counts</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="n">Counter</span><span class="p">)</span>
        <span class="n">disease_effects_counts</span> <span class="o">=</span> <span class="p">{}</span>
        <span class="k">for</span> <span class="n">curie</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">diseases</span><span class="p">:</span>
            <span class="n">disease_name</span> <span class="o">=</span> <span class="s2">&quot;Unknown&quot;</span>
            <span class="n">disease_variants_counts</span><span class="p">[</span><span class="n">curie</span><span class="p">]</span> <span class="o">=</span> <span class="n">Counter</span><span class="p">()</span>
            <span class="n">disease_effects_counts</span><span class="p">[</span><span class="n">curie</span><span class="p">]</span> <span class="o">=</span> <span class="n">Counter</span><span class="p">()</span>

            <span class="k">for</span> <span class="n">dis</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_diseases</span><span class="p">():</span>
                <span class="k">if</span> <span class="n">dis</span><span class="o">.</span><span class="n">identifier</span><span class="o">.</span><span class="n">value</span> <span class="o">==</span> <span class="n">curie</span><span class="p">:</span>
                    <span class="n">disease_name</span> <span class="o">=</span> <span class="n">dis</span><span class="o">.</span><span class="n">name</span>
                    <span class="k">break</span>

            <span class="n">disease_counts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                <span class="p">{</span>
                    <span class="s2">&quot;disease_id&quot;</span><span class="p">:</span> <span class="n">curie</span><span class="p">,</span>
                    <span class="s2">&quot;disease_name&quot;</span><span class="p">:</span> <span class="n">disease_name</span><span class="p">,</span>
                    <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="n">count</span><span class="p">,</span>
                <span class="p">}</span>
            <span class="p">)</span>

            <span class="k">for</span> <span class="n">pat</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_patients</span><span class="p">:</span>
                <span class="k">for</span> <span class="n">disease</span> <span class="ow">in</span> <span class="n">pat</span><span class="o">.</span><span class="n">diseases</span><span class="p">:</span>
                    <span class="n">variant_key_count</span> <span class="o">=</span> <span class="n">disease_variants_counts</span><span class="p">[</span>
                        <span class="n">disease</span><span class="o">.</span><span class="n">identifier</span><span class="o">.</span><span class="n">value</span>
                    <span class="p">]</span>
                    <span class="n">variant_key_count</span><span class="o">.</span><span class="n">update</span><span class="p">(</span>
                        <span class="n">variant</span><span class="o">.</span><span class="n">variant_info</span><span class="o">.</span><span class="n">variant_key</span> <span class="k">for</span> <span class="n">variant</span> <span class="ow">in</span> <span class="n">pat</span><span class="o">.</span><span class="n">variants</span>
                    <span class="p">)</span>
                    <span class="k">for</span> <span class="n">variant</span> <span class="ow">in</span> <span class="n">pat</span><span class="o">.</span><span class="n">variants</span><span class="p">:</span>
                        <span class="k">for</span> <span class="n">txa</span> <span class="ow">in</span> <span class="n">variant</span><span class="o">.</span><span class="n">tx_annotations</span><span class="p">:</span>
                            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">_tx_id</span> <span class="o">==</span> <span class="n">txa</span><span class="o">.</span><span class="n">transcript_id</span><span class="p">:</span>
                                <span class="n">disease_effects_counts</span><span class="p">[</span><span class="n">curie</span><span class="p">]</span><span class="o">.</span><span class="n">update</span><span class="p">(</span>
                                    <span class="n">eff</span><span class="o">.</span><span class="n">name</span> <span class="k">for</span> <span class="n">eff</span> <span class="ow">in</span> <span class="n">txa</span><span class="o">.</span><span class="n">variant_effects</span>
                                <span class="p">)</span>

        <span class="k">return</span> <span class="p">{</span>
            <span class="s2">&quot;n_diseases&quot;</span><span class="p">:</span> <span class="n">n_diseases</span><span class="p">,</span>
            <span class="s2">&quot;disease_counts&quot;</span><span class="p">:</span> <span class="n">disease_counts</span><span class="p">,</span>
            <span class="s2">&quot;disease_variant_counts&quot;</span><span class="p">:</span> <span class="n">disease_variants_counts</span><span class="p">,</span>
            <span class="s2">&quot;disease_effects_counts&quot;</span><span class="p">:</span> <span class="n">disease_effects_counts</span><span class="p">,</span>
        <span class="p">}</span></div>



<div class="viewcode-block" id="CohortVariantViewer">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.CohortVariantViewer">[docs]</a>
<span class="k">class</span><span class="w"> </span><span class="nc">CohortVariantViewer</span><span class="p">(</span><span class="n">BaseViewer</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    `CohortVariantViewer` creates an HTML report with the cohort variants.</span>

<span class="sd">    The report can be either written into an HTML file or displayed in a Jupyter notebook.</span>

<span class="sd">    See :ref:show-cohort-variants: for an example usage.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span><span class="w"> </span><span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">):</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Args:</span>
<span class="sd">            tx_id (str): The transcript identifier (Usually, the MANE RefSeq transcript, that should start with &quot;NM_&quot;)</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="nb">super</span><span class="p">()</span><span class="o">.</span><span class="fm">__init__</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_environment</span><span class="o">.</span><span class="n">get_template</span><span class="p">(</span><span class="s2">&quot;all_variants.html&quot;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_var_formatter</span> <span class="o">=</span> <span class="n">VariantFormatter</span><span class="p">(</span><span class="n">tx_id</span><span class="p">)</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">tx_id</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;NM&quot;</span><span class="p">):</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;[WARNING] Non-RefSeq transcript id: </span><span class="si">{</span><span class="n">tx_id</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_transcript_id</span> <span class="o">=</span> <span class="n">tx_id</span>

<div class="viewcode-block" id="CohortVariantViewer.process">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.CohortVariantViewer.process">[docs]</a>
    <span class="k">def</span><span class="w"> </span><span class="nf">process</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span> <span class="n">only_hgvs</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">GpseaReport</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Generate the variant report.</span>

<span class="sd">        Args:</span>
<span class="sd">            cohort (Cohort): The cohort being analyzed in the current notebook.</span>
<span class="sd">            only_hgvs (bool): Do not show the transcript ID part of the HGVS annotation, just the annotation.</span>

<span class="sd">        Returns:</span>
<span class="sd">            GpseaReport: a report that can be stored to a path or displayed in</span>
<span class="sd">                interactive environment such as Jupyter notebook.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">context</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_prepare_context</span><span class="p">(</span><span class="n">cohort</span><span class="p">,</span> <span class="n">only_hgvs</span><span class="o">=</span><span class="n">only_hgvs</span><span class="p">)</span>
        <span class="n">html</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span><span class="o">.</span><span class="n">render</span><span class="p">(</span><span class="n">context</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">HtmlGpseaReport</span><span class="p">(</span><span class="n">html</span><span class="o">=</span><span class="n">html</span><span class="p">)</span></div>


    <span class="k">def</span><span class="w"> </span><span class="nf">_prepare_context</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">,</span>
        <span class="n">only_hgvs</span><span class="p">:</span> <span class="nb">bool</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Mapping</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">typing</span><span class="o">.</span><span class="n">Any</span><span class="p">]:</span>
        <span class="n">variant_count_dictionary</span> <span class="o">=</span> <span class="n">Counter</span><span class="p">()</span>
        <span class="n">variant_key_to_variant</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">()</span>

        <span class="k">for</span> <span class="n">var</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_variants</span><span class="p">():</span>
            <span class="n">var_key</span> <span class="o">=</span> <span class="n">var</span><span class="o">.</span><span class="n">variant_info</span><span class="o">.</span><span class="n">variant_key</span>
            <span class="n">vdata</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_get_variant_data</span><span class="p">(</span><span class="n">var</span><span class="p">,</span> <span class="n">only_hgvs</span><span class="p">)</span>
            <span class="n">variant_key_to_variant</span><span class="p">[</span><span class="n">var_key</span><span class="p">]</span> <span class="o">=</span> <span class="n">vdata</span>
            <span class="n">variant_count_dictionary</span><span class="p">[</span><span class="n">var_key</span><span class="p">]</span> <span class="o">+=</span> <span class="mi">1</span>

        <span class="n">variant_counts</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">var_key</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">variant_count_dictionary</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
            <span class="n">var_data</span> <span class="o">=</span> <span class="n">variant_key_to_variant</span><span class="p">[</span><span class="n">var_key</span><span class="p">]</span>
            <span class="n">variant_counts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                <span class="p">{</span>
                    <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="n">count</span><span class="p">,</span>
                    <span class="s2">&quot;variant_key&quot;</span><span class="p">:</span> <span class="n">var_data</span><span class="o">.</span><span class="n">variant_key</span><span class="p">,</span>
                    <span class="s2">&quot;hgvs&quot;</span><span class="p">:</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">var_data</span><span class="o">.</span><span class="n">hgvsc</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="n">var_data</span><span class="o">.</span><span class="n">hgvsp</span><span class="si">}</span><span class="s2">)&quot;</span><span class="p">,</span>
                    <span class="s2">&quot;variant_effects&quot;</span><span class="p">:</span> <span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">var_data</span><span class="o">.</span><span class="n">variant_effects</span><span class="p">),</span>
                    <span class="s2">&quot;exons&quot;</span><span class="p">:</span> <span class="p">(</span>
                        <span class="s2">&quot;-&quot;</span>
                        <span class="k">if</span> <span class="n">var_data</span><span class="o">.</span><span class="n">exons</span> <span class="ow">is</span> <span class="kc">None</span>
                        <span class="k">else</span> <span class="s2">&quot;, &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="n">var_data</span><span class="o">.</span><span class="n">exons</span><span class="p">))</span>
                    <span class="p">),</span>
                <span class="p">}</span>
            <span class="p">)</span>

        <span class="n">variant_counts</span><span class="o">.</span><span class="n">sort</span><span class="p">(</span><span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;count&quot;</span><span class="p">],</span> <span class="n">reverse</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

        <span class="k">return</span> <span class="p">{</span>
            <span class="s2">&quot;variant_counts&quot;</span><span class="p">:</span> <span class="n">variant_counts</span><span class="p">,</span>
        <span class="p">}</span>

    <span class="k">def</span><span class="w"> </span><span class="nf">_get_variant_data</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">variant</span><span class="p">:</span> <span class="n">Variant</span><span class="p">,</span> <span class="n">only_hgvs</span><span class="p">:</span> <span class="nb">bool</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">VariantData</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Get user-friendly strings (e.g., HGVS for our target transcript) to match to the chromosomal strings</span>
<span class="sd">        Args:</span>
<span class="sd">            variant (Variant): The variant to be formatted.</span>
<span class="sd">            only_hgvs (bool): do not show the transcript ID part of the HGVS annotation, just the annotation.</span>

<span class="sd">        Returns:</span>
<span class="sd">            VariantData: a named tuple with variant data formatted for human consumption.</span>
<span class="sd">        &quot;&quot;&quot;</span>

        <span class="k">if</span> <span class="n">variant</span><span class="o">.</span><span class="n">variant_info</span><span class="o">.</span><span class="n">has_sv_info</span><span class="p">():</span>
            <span class="n">sv_info</span> <span class="o">=</span> <span class="n">variant</span><span class="o">.</span><span class="n">variant_info</span><span class="o">.</span><span class="n">sv_info</span>
            <span class="k">assert</span> <span class="n">sv_info</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span>
            <span class="n">gene_symbol</span> <span class="o">=</span> <span class="n">sv_info</span><span class="o">.</span><span class="n">gene_symbol</span>
            <span class="n">display</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;SV involving </span><span class="si">{</span><span class="n">gene_symbol</span><span class="si">}</span><span class="s2">&quot;</span>
            <span class="n">effect</span> <span class="o">=</span> <span class="n">VariantEffect</span><span class="o">.</span><span class="n">structural_so_id_to_display</span><span class="p">(</span>
                <span class="n">so_term</span><span class="o">=</span><span class="n">sv_info</span><span class="o">.</span><span class="n">structural_type</span>
            <span class="p">)</span>
            <span class="k">return</span> <span class="n">VariantData</span><span class="p">(</span>
                <span class="n">variant_key</span><span class="o">=</span><span class="n">variant</span><span class="o">.</span><span class="n">variant_info</span><span class="o">.</span><span class="n">variant_key</span><span class="p">,</span>
                <span class="n">hgvsc</span><span class="o">=</span><span class="n">display</span><span class="p">,</span>
                <span class="n">hgvsp</span><span class="o">=</span><span class="s2">&quot;p.?&quot;</span><span class="p">,</span>
                <span class="n">variant_effects</span><span class="o">=</span><span class="p">(</span><span class="n">effect</span><span class="p">,),</span>
                <span class="n">exons</span><span class="o">=</span><span class="p">(),</span>
            <span class="p">)</span>
        <span class="k">elif</span> <span class="n">variant</span><span class="o">.</span><span class="n">variant_info</span><span class="o">.</span><span class="n">has_variant_coordinates</span><span class="p">():</span>
            <span class="n">tx_annotation</span> <span class="o">=</span> <span class="n">variant</span><span class="o">.</span><span class="n">get_tx_anno_by_tx_id</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">_transcript_id</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">tx_annotation</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="n">hgvsc</span> <span class="o">=</span> <span class="s2">&quot;-&quot;</span>
                <span class="n">hgvsp</span> <span class="o">=</span> <span class="s2">&quot;-&quot;</span>
                <span class="n">var_effects</span> <span class="o">=</span> <span class="p">()</span>
                <span class="n">exons</span> <span class="o">=</span> <span class="p">()</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">only_hgvs</span><span class="p">:</span>
                    <span class="k">if</span> <span class="n">tx_annotation</span><span class="o">.</span><span class="n">hgvs_cdna</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                        <span class="n">hgvsc</span> <span class="o">=</span> <span class="s2">&quot;-&quot;</span>
                        <span class="n">hgvsp</span> <span class="o">=</span> <span class="s2">&quot;-&quot;</span>
                    <span class="k">else</span><span class="p">:</span>
                        <span class="n">fields_dna</span> <span class="o">=</span> <span class="n">tx_annotation</span><span class="o">.</span><span class="n">hgvs_cdna</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)</span>
                        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">fields_dna</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                            <span class="n">hgvsc</span> <span class="o">=</span> <span class="n">fields_dna</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
                        <span class="k">else</span><span class="p">:</span>
                            <span class="n">hgvsc</span> <span class="o">=</span> <span class="n">fields_dna</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>

                        <span class="n">fields_ps</span> <span class="o">=</span> <span class="p">(</span>
                            <span class="n">tx_annotation</span><span class="o">.</span><span class="n">hgvsp</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)</span>
                            <span class="k">if</span> <span class="n">tx_annotation</span><span class="o">.</span><span class="n">hgvsp</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span>
                            <span class="k">else</span> <span class="p">(</span><span class="s2">&quot;-&quot;</span><span class="p">,)</span>
                        <span class="p">)</span>
                        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">fields_ps</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
                            <span class="n">hgvsp</span> <span class="o">=</span> <span class="n">fields_ps</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
                        <span class="k">else</span><span class="p">:</span>
                            <span class="n">hgvsp</span> <span class="o">=</span> <span class="n">fields_ps</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">hgvsc</span> <span class="o">=</span> <span class="n">tx_annotation</span><span class="o">.</span><span class="n">hgvs_cdna</span>

                <span class="n">var_effects</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span>
                    <span class="n">var_eff</span><span class="o">.</span><span class="n">to_display</span><span class="p">()</span> <span class="k">for</span> <span class="n">var_eff</span> <span class="ow">in</span> <span class="n">tx_annotation</span><span class="o">.</span><span class="n">variant_effects</span>
                <span class="p">)</span>
                <span class="n">exons</span> <span class="o">=</span> <span class="n">tx_annotation</span><span class="o">.</span><span class="n">overlapping_exons</span>

            <span class="k">return</span> <span class="n">VariantData</span><span class="p">(</span>
                <span class="n">variant_key</span><span class="o">=</span><span class="n">variant</span><span class="o">.</span><span class="n">variant_info</span><span class="o">.</span><span class="n">variant_key</span><span class="p">,</span>
                <span class="n">hgvsc</span><span class="o">=</span><span class="n">hgvsc</span><span class="p">,</span>
                <span class="n">hgvsp</span><span class="o">=</span><span class="n">hgvsp</span><span class="p">,</span>
                <span class="n">variant_effects</span><span class="o">=</span><span class="n">var_effects</span><span class="p">,</span>
                <span class="n">exons</span><span class="o">=</span><span class="n">exons</span><span class="p">,</span>
            <span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Neither variant coordinates nor SV info are available&quot;</span><span class="p">)</span></div>



<div class="viewcode-block" id="ProteinVariantViewer">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.ProteinVariantViewer">[docs]</a>
<span class="k">class</span><span class="w"> </span><span class="nc">ProteinVariantViewer</span><span class="p">(</span><span class="n">BaseViewer</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Class to create a pretty HTML table to display the protein information in the Jupyter notebook.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span><span class="w"> </span><span class="fm">__init__</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">protein_metadata</span><span class="p">:</span> <span class="n">ProteinMetadata</span><span class="p">,</span>
        <span class="n">tx_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="p">):</span>
        <span class="c1"># TODO[v1.0.0] - remove `tx_id`.</span>
        <span class="nb">super</span><span class="p">()</span><span class="o">.</span><span class="fm">__init__</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_environment</span><span class="o">.</span><span class="n">get_template</span><span class="p">(</span><span class="s2">&quot;protein.html&quot;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span> <span class="o">=</span> <span class="n">protein_metadata</span>

<div class="viewcode-block" id="ProteinVariantViewer.process">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.ProteinVariantViewer.process">[docs]</a>
    <span class="k">def</span><span class="w"> </span><span class="nf">process</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">GpseaReport</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Summarize the data regarding the protein into a HTML table.</span>

<span class="sd">        Args:</span>
<span class="sd">            cohort (Cohort): the cohort of patients being analyzed</span>

<span class="sd">        Returns:</span>
<span class="sd">            GpseaReport: a report that can be stored to a path or displayed in</span>
<span class="sd">                interactive environment such as Jupyter notebook.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">context</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_prepare_context</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
        <span class="n">html</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span><span class="o">.</span><span class="n">render</span><span class="p">(</span><span class="n">context</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">HtmlGpseaReport</span><span class="p">(</span><span class="n">html</span><span class="o">=</span><span class="n">html</span><span class="p">)</span></div>


    <span class="k">def</span><span class="w"> </span><span class="nf">_prepare_context</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">cohort</span><span class="p">:</span> <span class="n">Cohort</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Mapping</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">typing</span><span class="o">.</span><span class="n">Any</span><span class="p">]:</span>
        <span class="n">protein_id</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span><span class="o">.</span><span class="n">protein_id</span>
        <span class="n">protein_label</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span><span class="o">.</span><span class="n">label</span>
        <span class="n">protein_features</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span><span class="o">.</span><span class="n">protein_features</span><span class="p">,</span> <span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">f</span><span class="p">:</span> <span class="n">f</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">start</span>
        <span class="p">)</span>

        <span class="c1"># collect variants that are located in the protein features as well as other variants that are located</span>
        <span class="c1"># &quot;in-between&quot; the features</span>
        <span class="n">feature_to_variants</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">list</span><span class="p">)</span>
        <span class="n">non_feature_count</span> <span class="o">=</span> <span class="mi">0</span>
        <span class="n">n_variants_in_features</span> <span class="o">=</span> <span class="mi">0</span>

        <span class="c1"># This iterates over variants as recorded in individuals,</span>
        <span class="c1"># not over *unique* `VariantInfo`s</span>
        <span class="k">for</span> <span class="n">var</span> <span class="ow">in</span> <span class="n">cohort</span><span class="o">.</span><span class="n">all_variants</span><span class="p">():</span>
            <span class="n">target_annot</span> <span class="o">=</span> <span class="nb">next</span><span class="p">(</span>
                <span class="p">(</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">var</span><span class="o">.</span><span class="n">tx_annotations</span> <span class="k">if</span> <span class="n">x</span><span class="o">.</span><span class="n">protein_id</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span><span class="o">.</span><span class="n">protein_id</span><span class="p">),</span> <span class="kc">None</span>
            <span class="p">)</span>
            <span class="k">if</span> <span class="n">target_annot</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="c1"># structural variants do not have a transcript id, and we skip them</span>
                <span class="c1"># It should never happen that HGVS variants do not have the proper transcript id but we do not check</span>
                <span class="c1"># that here in this visualization function</span>
                <span class="k">continue</span>
            <span class="n">hgvs_p</span> <span class="o">=</span> <span class="n">target_annot</span><span class="o">.</span><span class="n">hgvsp</span>
            <span class="k">if</span> <span class="n">hgvs_p</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
                <span class="c1"># hgvs_p is now a string such as NP_001305781.1:p.Tyr15Ter. We remove the accession id, which</span>
                <span class="c1"># is shown in the title and caption and is redundant here</span>
                <span class="n">fields</span> <span class="o">=</span> <span class="n">hgvs_p</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)</span>
                <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">fields</span><span class="p">)</span> <span class="o">==</span> <span class="mi">2</span><span class="p">:</span>
                    <span class="n">hgvs_p</span> <span class="o">=</span> <span class="n">fields</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">continue</span>
            <span class="n">target_region</span> <span class="o">=</span> <span class="n">target_annot</span><span class="o">.</span><span class="n">protein_effect_location</span>
            <span class="k">if</span> <span class="n">target_region</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
                <span class="c1"># can happen for certain variant classes such as splice variant. Not an error</span>
                <span class="k">continue</span>
            <span class="n">found_overlap</span> <span class="o">=</span> <span class="kc">False</span>
            <span class="k">for</span> <span class="n">feature</span> <span class="ow">in</span> <span class="n">protein_features</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">feature</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">region</span><span class="o">.</span><span class="n">overlaps_with</span><span class="p">(</span><span class="n">target_region</span><span class="p">):</span>
                    <span class="n">found_overlap</span> <span class="o">=</span> <span class="kc">True</span>
                    <span class="n">feature_to_variants</span><span class="p">[</span><span class="n">feature</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">hgvs_p</span><span class="p">)</span>

            <span class="k">if</span> <span class="n">found_overlap</span><span class="p">:</span>
                <span class="n">n_variants_in_features</span> <span class="o">+=</span> <span class="mi">1</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">non_feature_count</span> <span class="o">+=</span> <span class="mi">1</span>

        <span class="n">feature_list</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">feature</span> <span class="ow">in</span> <span class="n">protein_features</span><span class="p">:</span>
            <span class="n">variant_list</span> <span class="o">=</span> <span class="s2">&quot;; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">feature_to_variants</span><span class="p">[</span><span class="n">feature</span><span class="p">]))</span>
            <span class="n">feature_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                <span class="p">{</span>
                    <span class="s2">&quot;name&quot;</span><span class="p">:</span> <span class="n">feature</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">name</span><span class="p">,</span>
                    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="n">feature</span><span class="o">.</span><span class="n">feature_type</span><span class="p">,</span>
                    <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="n">feature</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">start</span><span class="p">,</span>
                    <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="n">feature</span><span class="o">.</span><span class="n">info</span><span class="o">.</span><span class="n">end</span><span class="p">,</span>
                    <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="n">feature_to_variants</span><span class="p">[</span><span class="n">feature</span><span class="p">]),</span>
                    <span class="s2">&quot;variants&quot;</span><span class="p">:</span> <span class="n">variant_list</span><span class="p">,</span>
                <span class="p">}</span>
            <span class="p">)</span>

        <span class="k">return</span> <span class="p">{</span>
            <span class="s2">&quot;protein_label&quot;</span><span class="p">:</span> <span class="n">protein_label</span><span class="p">,</span>
            <span class="s2">&quot;protein_id&quot;</span><span class="p">:</span> <span class="n">protein_id</span><span class="p">,</span>
            <span class="s2">&quot;protein_length&quot;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">_protein_meta</span><span class="o">.</span><span class="n">protein_length</span><span class="p">,</span>
            <span class="s2">&quot;feature_list&quot;</span><span class="p">:</span> <span class="n">feature_list</span><span class="p">,</span>
            <span class="s2">&quot;n_variants_in_features&quot;</span><span class="p">:</span> <span class="n">n_variants_in_features</span><span class="p">,</span>
            <span class="s2">&quot;non_feature_count&quot;</span><span class="p">:</span> <span class="n">non_feature_count</span><span class="p">,</span>
        <span class="p">}</span></div>



<div class="viewcode-block" id="MtcStatsViewer">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.MtcStatsViewer">[docs]</a>
<span class="k">class</span><span class="w"> </span><span class="nc">MtcStatsViewer</span><span class="p">(</span><span class="n">BaseViewer</span><span class="p">):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    `MtcStatsViewer` uses a Jinja2 template to create an HTML element for showing in the Jupyter notebook</span>
<span class="sd">    or for writing into a standalone HTML file.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span><span class="w"> </span><span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="nb">super</span><span class="p">()</span><span class="o">.</span><span class="fm">__init__</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_environment</span><span class="o">.</span><span class="n">get_template</span><span class="p">(</span><span class="s2">&quot;stats.html&quot;</span><span class="p">)</span>

<div class="viewcode-block" id="MtcStatsViewer.process">
<a class="viewcode-back" href="../../../apidocs/gpsea.view.html#gpsea.view.MtcStatsViewer.process">[docs]</a>
    <span class="k">def</span><span class="w"> </span><span class="nf">process</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span>
        <span class="n">result</span><span class="p">:</span> <span class="n">HpoTermAnalysisResult</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">GpseaReport</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Create an HTML to present MTC part of the :class:`~gpsea.analysis.pcats.HpoTermAnalysisResult`.</span>

<span class="sd">        Args:</span>
<span class="sd">            result (HpoTermAnalysisResult): the result to show</span>

<span class="sd">        Returns:</span>
<span class="sd">            GpseaReport: a report that can be stored to a path or displayed in</span>
<span class="sd">                interactive environment such as Jupyter notebook.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">assert</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">result</span><span class="p">,</span> <span class="n">HpoTermAnalysisResult</span><span class="p">)</span>
        <span class="n">context</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_prepare_context</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
        <span class="n">html</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_cohort_template</span><span class="o">.</span><span class="n">render</span><span class="p">(</span><span class="n">context</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">HtmlGpseaReport</span><span class="p">(</span><span class="n">html</span><span class="o">=</span><span class="n">html</span><span class="p">)</span></div>


    <span class="nd">@staticmethod</span>
    <span class="k">def</span><span class="w"> </span><span class="nf">_prepare_context</span><span class="p">(</span>
        <span class="n">report</span><span class="p">:</span> <span class="n">HpoTermAnalysisResult</span><span class="p">,</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Mapping</span><span class="p">[</span><span class="nb">str</span><span class="p">,</span> <span class="n">typing</span><span class="o">.</span><span class="n">Any</span><span class="p">]:</span>
        <span class="n">counts</span> <span class="o">=</span> <span class="n">Counter</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">result</span> <span class="ow">in</span> <span class="n">report</span><span class="o">.</span><span class="n">mtc_filter_results</span><span class="p">:</span>
            <span class="k">if</span> <span class="n">result</span><span class="o">.</span><span class="n">is_filtered_out</span><span class="p">():</span>
                <span class="n">counts</span><span class="p">[</span><span class="n">result</span><span class="o">.</span><span class="n">mtc_issue</span><span class="p">]</span> <span class="o">+=</span> <span class="mi">1</span>

        <span class="n">n_skipped</span> <span class="o">=</span> <span class="mi">0</span>
        <span class="n">issue_to_count</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
        <span class="k">for</span> <span class="n">mtc_issue</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="nb">sorted</span><span class="p">(</span>
            <span class="n">counts</span><span class="o">.</span><span class="n">items</span><span class="p">(),</span>
            <span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">issue2count</span><span class="p">:</span> <span class="p">(</span><span class="n">issue2count</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">code</span><span class="p">,</span> <span class="n">issue2count</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">reason</span><span class="p">),</span>
        <span class="p">):</span>
            <span class="n">issue_to_count</span><span class="o">.</span><span class="n">append</span><span class="p">(</span>
                <span class="p">{</span>
                    <span class="s2">&quot;code&quot;</span><span class="p">:</span> <span class="n">mtc_issue</span><span class="o">.</span><span class="n">code</span><span class="p">,</span>
                    <span class="s2">&quot;reason&quot;</span><span class="p">:</span> <span class="n">mtc_issue</span><span class="o">.</span><span class="n">reason</span><span class="p">,</span>
                    <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="n">count</span><span class="p">,</span>
                <span class="p">}</span>
            <span class="p">)</span>
            <span class="n">n_skipped</span> <span class="o">+=</span> <span class="n">count</span>

        <span class="n">n_all</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">report</span><span class="o">.</span><span class="n">phenotypes</span><span class="p">)</span>
        <span class="n">n_tested</span> <span class="o">=</span> <span class="n">n_all</span> <span class="o">-</span> <span class="n">n_skipped</span>

        <span class="c1"># The following dictionary is used by the Jinja2 HTML template</span>
        <span class="k">return</span> <span class="p">{</span>
            <span class="s2">&quot;mtc_name&quot;</span><span class="p">:</span> <span class="n">report</span><span class="o">.</span><span class="n">mtc_correction</span><span class="p">,</span>
            <span class="s2">&quot;hpo_mtc_filter_name&quot;</span><span class="p">:</span> <span class="n">report</span><span class="o">.</span><span class="n">mtc_filter_name</span><span class="p">,</span>
            <span class="s2">&quot;skipped_hpo_count&quot;</span><span class="p">:</span> <span class="n">n_skipped</span><span class="p">,</span>
            <span class="s2">&quot;tested_hpo_count&quot;</span><span class="p">:</span> <span class="n">n_tested</span><span class="p">,</span>
            <span class="s2">&quot;total_hpo_count&quot;</span><span class="p">:</span> <span class="n">n_all</span><span class="p">,</span>
            <span class="s2">&quot;issue_to_count&quot;</span><span class="p">:</span> <span class="n">issue_to_count</span><span class="p">,</span>
        <span class="p">}</span></div>

</pre></div>

           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>