<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Variant Predicates &mdash; GPSEA 0.3.1.dev0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../_static/gpsea.css?v=547657ed" />

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../_static/documentation_options.js?v=f0846db4"></script>
        <script src="../../_static/doctools.js?v=9a2dae69"></script>
        <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Mode of Inheritance Predicates" href="mode_of_inheritance_predicate.html" />
    <link rel="prev" title="Genotype Predicates" href="genotype_predicates.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../index.html" class="icon icon-home">
            GPSEA
          </a>
              <div class="version">
                0.3
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../index.html">User guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../input-data.html">Input data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exploratory.html">Cohort exploratory analysis</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../predicates.html">Predicates</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="phenotype_predicates.html">Phenotype Predicates</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="genotype_predicates.html">Genotype Predicates</a><ul class="current">
<li class="toctree-l4 current"><a class="current reference internal" href="#">Variant Predicates</a></li>
<li class="toctree-l4"><a class="reference internal" href="mode_of_inheritance_predicate.html">Mode of Inheritance Predicates</a></li>
<li class="toctree-l4"><a class="reference internal" href="male_female_predicate.html">Partition by the sex of the individual</a></li>
<li class="toctree-l4"><a class="reference internal" href="diagnosis_predicate.html">Partition by a diagnosis</a></li>
<li class="toctree-l4"><a class="reference internal" href="groups_predicate.html">Groups Predicate</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../predicates.html#gallery">Gallery</a></li>
<li class="toctree-l3"><a class="reference internal" href="../predicates.html#need-more">Need more?</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../stats.html">Statistical tests</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mtc.html">Multiple-testing correction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../glossary.html">Glossary</a></li>
<li class="toctree-l2"><a class="reference internal" href="../autosomal_recessive.html">Genotype-Phenotype Correlations in Autosomal Recessive Diseases</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../index.html">User guide</a></li>
          <li class="breadcrumb-item"><a href="../predicates.html">Predicates</a></li>
          <li class="breadcrumb-item"><a href="genotype_predicates.html">Genotype Predicates</a></li>
      <li class="breadcrumb-item active">Variant Predicates</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/user-guide/predicates/variant_predicates.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="variant-predicates">
<span id="id1"></span><h1>Variant Predicates<a class="headerlink" href="#variant-predicates" title="Link to this heading"></a></h1>
<p>GPSEA uses the <a class="reference internal" href="../../apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate" title="gpsea.analysis.predicate.genotype.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a> class
to test if a <a class="reference internal" href="../../apidocs/gpsea.model.html#gpsea.model.Variant" title="gpsea.model.Variant"><code class="xref py py-class docutils literal notranslate"><span class="pre">Variant</span></code></a> meets the inclusion criteria.
The variant predicate can leverage multiple primary data:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Primary data source</p></th>
<th class="head"><p>Example</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Allele</p></td>
<td><p>the variant being a deletion or a single nucleotide variant (SNV)</p></td>
</tr>
<tr class="row-odd"><td><p>Genome</p></td>
<td><p>overlaps of a target genomic region</p></td>
</tr>
<tr class="row-even"><td><p>Functional annotation</p></td>
<td><p>variant is predicted to lead to a missense change or affect an exon of certain transcript</p></td>
</tr>
<tr class="row-odd"><td><p>Protein data</p></td>
<td><p>variant is located in a region encoding a protein domain, protein feature type</p></td>
</tr>
</tbody>
</table>
<p>which demands a considerable amount of flexibility for creating the predicate.</p>
<p>As a rule of thumb, the predicates for testing basic conditions are available off the shelf,
and they can be used as building block for testing for more complex conditions,
such as testing if the variant is “a missense or synonymous variant located in exon 6 of transcript <cite>NM_013275.6</cite>”.</p>
<p>Let’s demonstrate this on few examples.
We will load a cohort of 19 subjects with variants in <em>RERE</em>,
leading to <a class="reference external" href="https://omim.org/entry/142900">Holt-Oram syndrome MIM:142900</a>.
The the clinical signs and symptoms of the subjects were encoded into HPO terms
along with the pathogenic <em>RERE</em> variant.</p>
<p>Let’s load the phenopackets, as previously described in greater detail the <a class="reference internal" href="../input-data.html#input-data"><span class="std std-ref">Input data</span></a> section.
Briefly, we first load HPO:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>then, we configure the cohort creator:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">configure_caching_cohort_creator</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort_creator</span> <span class="o">=</span> <span class="n">configure_caching_cohort_creator</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
</pre></div>
</div>
<p>which we use to create a <a class="reference internal" href="../../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a> from a bunch of phenopackets
from the release <cite>0.1.18</cite> of <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a>.
We will load 19 individuals with mutations in <em>RERE</em> gene:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">ppktstore.registry</span> <span class="kn">import</span> <span class="n">configure_phenopacket_registry</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">registry</span> <span class="o">=</span> <span class="n">configure_phenopacket_registry</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="n">registry</span><span class="o">.</span><span class="n">open_phenopacket_store</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;0.1.18&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">ps</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">phenopackets</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">ps</span><span class="o">.</span><span class="n">iter_cohort_phenopackets</span><span class="p">(</span><span class="s1">&#39;RERE&#39;</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">)</span>
<span class="go">19</span>
</pre></div>
</div>
<p>and we will convert the phenopacket into a <a class="reference internal" href="../../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.preprocessing</span> <span class="kn">import</span> <span class="n">load_phenopackets</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cohort</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">load_phenopackets</span><span class="p">(</span><span class="n">phenopackets</span><span class="p">,</span> <span class="n">cohort_creator</span><span class="p">)</span>  
<span class="go">Individuals Processed: ...</span>
</pre></div>
</div>
<p>To demonstrate the predicate API, we will use the variant <code class="docutils literal notranslate"><span class="pre">1_8358231_8358231_T_C</span></code> that corresponds
to a pathogenic variant <a class="reference external" href="https://www.ncbi.nlm.nih.gov/clinvar/variation/522858/">VCV000522858.5</a>
that replaces the histidine encoded by the 1435th codon of <cite>NM_001042681.2</cite> with arginine: <code class="docutils literal notranslate"><span class="pre">NM_001042681.2(RERE):c.4304A&gt;G</span> <span class="pre">(p.His1435Arg)</span></code>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">variant_key_of_interest</span> <span class="o">=</span> <span class="s1">&#39;1_8358231_8358231_T_C&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant</span> <span class="o">=</span> <span class="n">cohort</span><span class="o">.</span><span class="n">get_variant_by_key</span><span class="p">(</span><span class="n">variant_key_of_interest</span><span class="p">)</span>
</pre></div>
</div>
<section id="building-blocks">
<h2>Building blocks<a class="headerlink" href="#building-blocks" title="Link to this heading"></a></h2>
<p>We can check that the variant overlaps with <em>RERE</em>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gene</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">gene</span><span class="p">(</span><span class="s1">&#39;RERE&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gene</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>it overlaps with the <em>MANE</em> transcript:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">rere_mane_tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_001042681.2&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">transcript</span><span class="p">(</span><span class="n">rere_mane_tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tx</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>it in fact overlaps with the exon 20:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">exon20</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">exon</span><span class="p">(</span><span class="n">exon</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">rere_mane_tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">exon20</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>and leads to a missense mutation with respect to the MANE transcript:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">missense</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">rere_mane_tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">missense</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>See <a class="reference internal" href="../../apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicates" title="gpsea.analysis.predicate.genotype.VariantPredicates"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicates</span></code></a>
for more info on the predicates available off the shelf.</p>
</section>
<section id="complex-conditions">
<h2>Complex conditions<a class="headerlink" href="#complex-conditions" title="Link to this heading"></a></h2>
<p>We can combine the building blocks to test for more elaborate conditions.
For instance, we can test if the variant meets <em>any</em> or several conditions:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">nonsense</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">STOP_GAINED</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">rere_mane_tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">missense_or_nonsense</span> <span class="o">=</span> <span class="n">missense</span> <span class="o">|</span> <span class="n">nonsense</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">missense_or_nonsense</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>or <em>all</em> conditions:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">missense_and_exon20</span> <span class="o">=</span> <span class="n">missense</span> <span class="o">&amp;</span> <span class="n">exon20</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">missense_and_exon20</span><span class="o">.</span><span class="n">test</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>The <cite>VariantPredicate</cite> overloads Python <code class="docutils literal notranslate"><span class="pre">&amp;</span></code> (AND) and <code class="docutils literal notranslate"><span class="pre">|</span></code> (OR) operators to build a compound predicate from lower level building blocks.</p>
<p>Therefore, there is nothing that prevents us to combine the predicates into multi-level tests,
such as testing if the variant is a <em>“chromosomal deletion” or a deletion which removes at least 50 bp</em>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">VariantClass</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">chromosomal_deletion</span> <span class="o">=</span> <span class="s2">&quot;SO:1000029&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">predicate</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">structural_type</span><span class="p">(</span><span class="n">chromosomal_deletion</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_class</span><span class="p">(</span><span class="n">VariantClass</span><span class="o">.</span><span class="n">DEL</span><span class="p">)</span> <span class="o">&amp;</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">change_length</span><span class="p">(</span><span class="s2">&quot;&lt;=&quot;</span><span class="p">,</span> <span class="o">-</span><span class="mi">50</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">predicate</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;(structural type is SO:1000029 OR (variant class is DEL AND change length &lt;= -50))&#39;</span>
</pre></div>
</div>
</section>
<section id="inverting-conditions">
<h2>Inverting conditions<a class="headerlink" href="#inverting-conditions" title="Link to this heading"></a></h2>
<p>Sometimes we may want to test the variant for a condition that must <em>not</em> be met.
For instance, we may want to test if the variant is a deletion
that is <em>not</em> predicted to shift the transcript reading frame.
One of doing this would be to build a compound predicates
for all variant effects except of <a class="reference internal" href="../../apidocs/gpsea.model.html#gpsea.model.VariantEffect.FRAMESHIFT_VARIANT" title="gpsea.model.VariantEffect.FRAMESHIFT_VARIANT"><code class="xref py py-class docutils literal notranslate"><span class="pre">FRAMESHIFT_VARIANT</span></code></a>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">non_frameshift_effects</span> <span class="o">=</span> <span class="p">(</span>
<span class="gp">... </span>  <span class="n">VariantEffect</span><span class="o">.</span><span class="n">SYNONYMOUS_VARIANT</span><span class="p">,</span> <span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">VariantEffect</span><span class="o">.</span><span class="n">INTRON_VARIANT</span><span class="p">,</span>
<span class="gp">... </span>  <span class="c1"># and many more effects..</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">non_frameshift_predicate</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">all</span><span class="p">(</span><span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">eff</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">rere_mane_tx_id</span><span class="p">)</span> <span class="k">for</span> <span class="n">eff</span> <span class="ow">in</span> <span class="n">non_frameshift_effects</span><span class="p">)</span>
</pre></div>
</div>
<p>However, this is clearly tedious and it would be much better implemented
by a simple logical not of a predicate for a frameshift variant effect.</p>
<p>To support this, <cite>VariantPredicate</cite> implements <em>logical inversion</em>
which corresponds to Python’s <code class="docutils literal notranslate"><span class="pre">~</span></code> operator (tilde), to wrap
the underlying predicate and to invert its test result.</p>
<p>This is how we can use the predicate inversion to build the predicate for non-frameshift deletions:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">non_frameshift_del</span> <span class="o">=</span> <span class="o">~</span><span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">FRAMESHIFT_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">rere_mane_tx_id</span><span class="p">)</span> <span class="o">&amp;</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_class</span><span class="p">(</span><span class="n">VariantClass</span><span class="o">.</span><span class="n">DEL</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">non_frameshift_del</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;(NOT FRAMESHIFT_VARIANT on NM_001042681.2 AND variant class is DEL)&#39;</span>
</pre></div>
</div>
<p>Note the presence of a tilde <code class="docutils literal notranslate"><span class="pre">~</span></code> before the variant effect predicate and resulting <code class="docutils literal notranslate"><span class="pre">NOT</span></code> in the predicate question.</p>
<p>The variant predicate offers a flexible API for testing if variants meet a condition.
However, the genotype phenotype correlations are done on the individual level
and the variant predicates are used as a component of the genotype predicate.
The next sections show how to use variant predicates to assign individuals into groups.</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="genotype_predicates.html" class="btn btn-neutral float-left" title="Genotype Predicates" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="mode_of_inheritance_predicate.html" class="btn btn-neutral float-right" title="Mode of Inheritance Predicates" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>