

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../../../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Group by allele count &mdash; GPSEA 0.9.5.dev0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../../../_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../../../_static/gpsea.css?v=547657ed" />

  
      <script src="../../../../_static/jquery.js?v=5d32c60e"></script>
      <script src="../../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../../../../_static/documentation_options.js?v=490338fb"></script>
      <script src="../../../../_static/doctools.js?v=9bcbadda"></script>
      <script src="../../../../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../../../../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../../../../_static/copybutton.js?v=6dbb43f8"></script>
      <script>window.MathJax = {"TeX": {"Macros": {"RR": "{\\mathbb{R}}"}}}</script>
      <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" />
    <link rel="next" title="Phenotype classifiers and scorers" href="../phenotype/index.html" />
    <link rel="prev" title="Group by variant category" href="variant_category.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../index.html">User guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../../input-data.html">Input data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../exploratory.html">Cohort exploratory analysis</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../index.html">Statistical analyses</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="../index.html">Partitioning</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="index.html">Genotype classifiers</a></li>
<li class="toctree-l4"><a class="reference internal" href="../phenotype/index.html">Phenotype classifiers and scorers</a></li>
<li class="toctree-l4"><a class="reference internal" href="../gallery.html">Gallery</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../phenotype-classes.html">Compare genotype and phenotype classes</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../mtc.html">Multiple-testing correction</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../phenotype-scores.html">Compare phenotype scores</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../measurements.html">Compare measurements</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../survival.html">Survival analysis</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../custom-components.html">Custom components</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../glossary.html">Glossary</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../../index.html">User guide</a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Statistical analyses</a></li>
          <li class="breadcrumb-item"><a href="../index.html">Partitioning</a></li>
          <li class="breadcrumb-item"><a href="index.html">Genotype classifiers</a></li>
      <li class="breadcrumb-item active">Group by allele count</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../../../_sources/user-guide/analyses/partitioning/genotype/allele_count.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="group-by-allele-count">
<span id="allele-count"></span><h1>Group by allele count<a class="headerlink" href="#group-by-allele-count" title="Link to this heading"></a></h1>
<p>Sometimes, we may want to compare individuals with different allele counts of a single variant category.</p>
<p>For instance, we may want to compare the survival of individuals harboring a mutation in <em>EGFR</em> (<span class="math notranslate nohighlight">\(AC = 1\)</span>)
with those with no such mutation (<span class="math notranslate nohighlight">\(AC = 0\)</span>).
Alternatively, in some genes, heterozygous mutations (<span class="math notranslate nohighlight">\(AC = 1\)</span>) and biallelic mutations (<span class="math notranslate nohighlight">\(AC = 2\)</span>)
can lead to different diseases.</p>
<p>It is worth pointing out the difference of the allele count analysis
with the variant category analysis described in the <a class="reference internal" href="variant_category.html#variant-category"><span class="std std-ref">Group by variant category</span></a> section.
The allele count analysis compares the allele counts while keeping the variant type constant,
whereas the variant category analysis compares variant types while keeping the total allele count constant.</p>
<p>A classifier for the allele count analysis is created with
the <a class="reference internal" href="../../../../apidocs/gpsea.analysis.clf.html#gpsea.analysis.clf.allele_count" title="gpsea.analysis.clf.allele_count"><code class="xref py py-func docutils literal notranslate"><span class="pre">allele_count()</span></code></a> function.
The function needs two arguments:
<code class="docutils literal notranslate"><span class="pre">target</span></code> with a <a class="reference internal" href="../../../../apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.VariantPredicate" title="gpsea.analysis.predicate.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a> for selecting the target variant type,
and <code class="docutils literal notranslate"><span class="pre">counts</span></code> with the target allele counts.
The <code class="docutils literal notranslate"><span class="pre">target</span></code> variant predicate typically selects a broad variant types - for instance, all variants that affect a gene of interest.
The <code class="docutils literal notranslate"><span class="pre">counts</span></code> take the allele counts to compare.</p>
<p>Let’s show a few examples.</p>
<section id="examples">
<h2>Examples<a class="headerlink" href="#examples" title="Link to this heading"></a></h2>
<section id="compare-the-individuals-with-egfr-mutation">
<h3>Compare the individuals with <em>EGFR</em> mutation<a class="headerlink" href="#compare-the-individuals-with-egfr-mutation" title="Link to this heading"></a></h3>
<p>We can use the allele count analysis to test for G/P associations
between the individuals with no mutation in <em>EGFR</em>
and the individuals harboring 1 variant allele.</p>
<p>First, let’s create a <a class="reference internal" href="../../../../apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.VariantPredicate" title="gpsea.analysis.predicate.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a>
to select variants that affect <em>EGFR</em>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.predicate</span><span class="w"> </span><span class="kn">import</span> <span class="n">gene</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">affects_egfr</span> <span class="o">=</span> <span class="n">gene</span><span class="p">(</span><span class="n">symbol</span><span class="o">=</span><span class="s2">&quot;EGFR&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">affects_egfr</span><span class="o">.</span><span class="n">description</span>
<span class="go">&#39;affects EGFR&#39;</span>
</pre></div>
</div>
<p>Next, we create allele count classifier to partition the individuals
based on presence of zero or one <em>EGFR</em> mutation allele:</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../../../../_images/allele-count-zero-one.png"><img alt="No mutation vs. monoallelic mutation" src="../../../../_images/allele-count-zero-one.png" style="width: 600px;" />
</a>
</figure>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.clf</span><span class="w"> </span><span class="kn">import</span> <span class="n">allele_count</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span> <span class="o">=</span> <span class="n">allele_count</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">counts</span><span class="o">=</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">),</span>
<span class="gp">... </span>    <span class="n">target</span><span class="o">=</span><span class="n">affects_egfr</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span><span class="o">.</span><span class="n">class_labels</span>
<span class="go">(&#39;0 alleles&#39;, &#39;1 allele&#39;)</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">allele_count</span></code> needs two inputs.
The <code class="docutils literal notranslate"><span class="pre">counts</span></code> takes a tuple of the target allele counts,
in order to partition the individuals based on zero or one allele.
The <code class="docutils literal notranslate"><span class="pre">target</span></code> takes a <a class="reference internal" href="../../../../apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.VariantPredicate" title="gpsea.analysis.predicate.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a>
to select the target variants.
The <code class="docutils literal notranslate"><span class="pre">target</span></code> is particularly important
if the cohort members include variants in other genes than <em>EGFR</em>.</p>
<p>The resulting <code class="docutils literal notranslate"><span class="pre">gt_clf</span></code> can partition a cohort along the genotype axis,
e.g. to compare the patient survivals in a <cite>survival analysis &lt;survival&gt;</cite>.</p>
</section>
<section id="compare-the-individuals-with-monoallelic-and-biallelic-mutations">
<h3>Compare the individuals with monoallelic and biallelic mutations<a class="headerlink" href="#compare-the-individuals-with-monoallelic-and-biallelic-mutations" title="Link to this heading"></a></h3>
<p>As another example, let’s partition individuals based on one or two alleles of a target mutation.</p>
<p>For this example, the target mutation is any mutation that affects <em>LMNA</em>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.predicate</span><span class="w"> </span><span class="kn">import</span> <span class="n">gene</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">affects_lmna</span> <span class="o">=</span> <span class="n">gene</span><span class="p">(</span><span class="n">symbol</span><span class="o">=</span><span class="s2">&quot;LMNA&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">affects_lmna</span><span class="o">.</span><span class="n">description</span>
<span class="go">&#39;affects LMNA&#39;</span>
</pre></div>
</div>
<p>and we will compare the individuals with one allele with those with two alleles:</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../../../../_images/allele-count-one-two.png"><img alt="Monoallelic vs biallelic mutations" src="../../../../_images/allele-count-one-two.png" style="width: 600px;" />
</a>
</figure>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span> <span class="o">=</span> <span class="n">allele_count</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">counts</span><span class="o">=</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">),</span>
<span class="gp">... </span>    <span class="n">target</span><span class="o">=</span><span class="n">affects_lmna</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span><span class="o">.</span><span class="n">class_labels</span>
<span class="go">(&#39;1 allele&#39;, &#39;2 alleles&#39;)</span>
</pre></div>
</div>
<p>The classifier assigns the individuals into one of two classes:
those with one <em>LMNA</em> variant allele and those with two <em>LMNA</em> variant alleles.
Any cohort member with other allele counts (e.g. <cite>0 allele</cite> or <cite>3 alleles</cite>) is ignored.</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="variant_category.html" class="btn btn-neutral float-left" title="Group by variant category" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../phenotype/index.html" class="btn btn-neutral float-right" title="Phenotype classifiers and scorers" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>