<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../../">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>genophenocorr.preprocessing._phenopacket &mdash; genophenocorr 0.1.1dev documentation</title>
      <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../../../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../../_static/genophenocorr.css?v=a9fa5eb9" />

  
  <!--[if lt IE 9]>
    <script src="../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../../_static/documentation_options.js?v=8427294d"></script>
        <script src="../../../_static/doctools.js?v=9a2dae69"></script>
        <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../../../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../index.html" class="icon icon-home">
            genophenocorr
          </a>
              <div class="version">
                0.1
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../user-guide/index.html">User guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">genophenocorr</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Module code</a></li>
      <li class="breadcrumb-item active">genophenocorr.preprocessing._phenopacket</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <h1>Source code for genophenocorr.preprocessing._phenopacket</h1><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">logging</span>
<span class="kn">import</span> <span class="nn">typing</span>

<span class="kn">import</span> <span class="nn">hpotk</span>

<span class="kn">from</span> <span class="nn">phenopackets</span> <span class="kn">import</span> <span class="n">GenomicInterpretation</span><span class="p">,</span> <span class="n">Phenopacket</span><span class="p">,</span> <span class="n">Disease</span> <span class="k">as</span> <span class="n">PPDisease</span>

<span class="kn">from</span> <span class="nn">genophenocorr.model</span> <span class="kn">import</span> <span class="n">Patient</span><span class="p">,</span> <span class="n">SampleLabels</span><span class="p">,</span> <span class="n">Disease</span>
<span class="kn">from</span> <span class="nn">genophenocorr.model</span> <span class="kn">import</span> <span class="n">VariantCoordinates</span><span class="p">,</span> <span class="n">Variant</span><span class="p">,</span> <span class="n">Genotype</span><span class="p">,</span> <span class="n">Genotypes</span>
<span class="kn">from</span> <span class="nn">genophenocorr.model.genome</span> <span class="kn">import</span> <span class="n">GenomeBuild</span><span class="p">,</span> <span class="n">GenomicRegion</span><span class="p">,</span> <span class="n">Strand</span>
<span class="kn">from</span> <span class="nn">._api</span> <span class="kn">import</span> <span class="n">VariantCoordinateFinder</span><span class="p">,</span> <span class="n">FunctionalAnnotator</span>
<span class="kn">from</span> <span class="nn">._audit</span> <span class="kn">import</span> <span class="n">Notepad</span>
<span class="kn">from</span> <span class="nn">._patient</span> <span class="kn">import</span> <span class="n">PatientCreator</span>
<span class="kn">from</span> <span class="nn">._phenotype</span> <span class="kn">import</span> <span class="n">PhenotypeCreator</span>


<div class="viewcode-block" id="PhenopacketVariantCoordinateFinder">
<a class="viewcode-back" href="../../../apidocs/genophenocorr.preprocessing.html#genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder">[docs]</a>
<span class="k">class</span> <span class="nc">PhenopacketVariantCoordinateFinder</span><span class="p">(</span><span class="n">VariantCoordinateFinder</span><span class="p">[</span><span class="n">GenomicInterpretation</span><span class="p">]):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    `PhenopacketVariantCoordinateFinder` figures out :class:`genophenocorr.model.VariantCoordinates`</span>
<span class="sd">    and :class:`genophenocorr.model.Genotype` from `GenomicInterpretation` element of Phenopacket Schema.</span>

<span class="sd">    :param build: genome build to use in `VariantCoordinates</span>
<span class="sd">    :param hgvs_coordinate_finder: the coordinate finder to use for parsing HGVS expressions</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">build</span><span class="p">:</span> <span class="n">GenomeBuild</span><span class="p">,</span>
                 <span class="n">hgvs_coordinate_finder</span><span class="p">:</span> <span class="n">VariantCoordinateFinder</span><span class="p">[</span><span class="nb">str</span><span class="p">]):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_logger</span> <span class="o">=</span> <span class="n">logging</span><span class="o">.</span><span class="n">getLogger</span><span class="p">(</span><span class="vm">__name__</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_build</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">build</span><span class="p">,</span> <span class="n">GenomeBuild</span><span class="p">,</span> <span class="s1">&#39;build&#39;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_hgvs_finder</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">hgvs_coordinate_finder</span><span class="p">,</span> <span class="n">VariantCoordinateFinder</span><span class="p">,</span>
                                                         <span class="s1">&#39;hgvs_coordinate_finder&#39;</span><span class="p">)</span>

<div class="viewcode-block" id="PhenopacketVariantCoordinateFinder.find_coordinates">
<a class="viewcode-back" href="../../../apidocs/genophenocorr.preprocessing.html#genophenocorr.preprocessing.PhenopacketVariantCoordinateFinder.find_coordinates">[docs]</a>
    <span class="k">def</span> <span class="nf">find_coordinates</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">item</span><span class="p">:</span> <span class="n">GenomicInterpretation</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Tuple</span><span class="p">[</span><span class="n">VariantCoordinates</span><span class="p">,</span> <span class="n">Genotype</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Creates a VariantCoordinates object from the data in a given Phenopacket</span>

<span class="sd">        Args:</span>
<span class="sd">            item (GenomicInterpretation): a genomic interpretation element from Phenopacket Schema</span>
<span class="sd">        Returns:</span>
<span class="sd">            tuple[VariantCall, Genotype]: A tuple of :class:`VariantCoordinates` and :class:`Genotype`</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">item</span><span class="p">,</span> <span class="n">GenomicInterpretation</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;item must be a Phenopacket GenomicInterpretation but was type </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">item</span><span class="p">)</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

        <span class="n">variant_descriptor</span> <span class="o">=</span> <span class="n">item</span><span class="o">.</span><span class="n">variant_interpretation</span><span class="o">.</span><span class="n">variation_descriptor</span>

        <span class="n">vc</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">_vcf_is_available</span><span class="p">(</span><span class="n">variant_descriptor</span><span class="o">.</span><span class="n">vcf_record</span><span class="p">):</span>
            <span class="c1"># We have a VCF record.</span>
            <span class="k">if</span> <span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">_check_assembly</span><span class="p">(</span><span class="n">variant_descriptor</span><span class="o">.</span><span class="n">vcf_record</span><span class="o">.</span><span class="n">genome_assembly</span><span class="p">):</span>
                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Variant id </span><span class="si">{</span><span class="n">variant_descriptor</span><span class="o">.</span><span class="n">id</span><span class="si">}</span><span class="s2"> for patient </span><span class="si">{</span><span class="n">item</span><span class="o">.</span><span class="n">subject_or_biosample_id</span><span class="si">}</span><span class="s2"> has a different Genome Assembly than what was given. &quot;</span> \
                                 <span class="o">+</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">variant_descriptor</span><span class="o">.</span><span class="n">vcf_record</span><span class="o">.</span><span class="n">genome_assembly</span><span class="si">}</span><span class="s2"> is not </span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">_build</span><span class="o">.</span><span class="n">identifier</span><span class="si">}</span><span class="s2">.&quot;</span><span class="p">)</span>
            <span class="n">contig</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_build</span><span class="o">.</span><span class="n">contig_by_name</span><span class="p">(</span><span class="n">variant_descriptor</span><span class="o">.</span><span class="n">vcf_record</span><span class="o">.</span><span class="n">chrom</span><span class="p">)</span>
            <span class="n">start</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">variant_descriptor</span><span class="o">.</span><span class="n">vcf_record</span><span class="o">.</span><span class="n">pos</span><span class="p">)</span> <span class="o">-</span> <span class="mi">1</span>
            <span class="n">ref</span> <span class="o">=</span> <span class="n">variant_descriptor</span><span class="o">.</span><span class="n">vcf_record</span><span class="o">.</span><span class="n">ref</span>
            <span class="n">alt</span> <span class="o">=</span> <span class="n">variant_descriptor</span><span class="o">.</span><span class="n">vcf_record</span><span class="o">.</span><span class="n">alt</span>
            <span class="n">end</span> <span class="o">=</span> <span class="n">start</span> <span class="o">+</span> <span class="nb">len</span><span class="p">(</span><span class="n">ref</span><span class="p">)</span>
            <span class="n">change_length</span> <span class="o">=</span> <span class="n">end</span> <span class="o">-</span> <span class="n">start</span>

            <span class="n">region</span> <span class="o">=</span> <span class="n">GenomicRegion</span><span class="p">(</span><span class="n">contig</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">Strand</span><span class="o">.</span><span class="n">POSITIVE</span><span class="p">)</span>
            <span class="n">vc</span> <span class="o">=</span> <span class="n">VariantCoordinates</span><span class="p">(</span><span class="n">region</span><span class="p">,</span> <span class="n">ref</span><span class="p">,</span> <span class="n">alt</span><span class="p">,</span> <span class="n">change_length</span><span class="p">)</span>
        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">_cnv_is_available</span><span class="p">(</span><span class="n">variant_descriptor</span><span class="o">.</span><span class="n">variation</span><span class="p">):</span>
            <span class="c1"># We have a CNV.</span>
            <span class="n">variation</span> <span class="o">=</span> <span class="n">variant_descriptor</span><span class="o">.</span><span class="n">variation</span>
            <span class="n">seq_location</span> <span class="o">=</span> <span class="n">variation</span><span class="o">.</span><span class="n">copy_number</span><span class="o">.</span><span class="n">allele</span><span class="o">.</span><span class="n">sequence_location</span>
            <span class="n">refseq_contig_name</span> <span class="o">=</span> <span class="n">seq_location</span><span class="o">.</span><span class="n">sequence_id</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span>
            <span class="n">contig</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_build</span><span class="o">.</span><span class="n">contig_by_name</span><span class="p">(</span><span class="n">refseq_contig_name</span><span class="p">)</span>

            <span class="c1"># Assuming SV coordinates are 1-based (VCF style),</span>
            <span class="c1"># so we subtract 1 to transform to 0-based coordinate system</span>
            <span class="n">start</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">seq_location</span><span class="o">.</span><span class="n">sequence_interval</span><span class="o">.</span><span class="n">start_number</span><span class="o">.</span><span class="n">value</span><span class="p">)</span> <span class="o">-</span> <span class="mi">1</span>
            <span class="n">end</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">seq_location</span><span class="o">.</span><span class="n">sequence_interval</span><span class="o">.</span><span class="n">end_number</span><span class="o">.</span><span class="n">value</span><span class="p">)</span>
            <span class="n">ref</span> <span class="o">=</span> <span class="s1">&#39;N&#39;</span>
            <span class="n">number</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">variation</span><span class="o">.</span><span class="n">copy_number</span><span class="o">.</span><span class="n">number</span><span class="o">.</span><span class="n">value</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">number</span> <span class="o">&gt;</span> <span class="mi">2</span><span class="p">:</span>
                <span class="n">alt</span> <span class="o">=</span> <span class="s1">&#39;&lt;DUP&gt;&#39;</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">alt</span> <span class="o">=</span> <span class="s1">&#39;&lt;DEL&gt;&#39;</span>
            <span class="n">change_length</span> <span class="o">=</span> <span class="n">end</span> <span class="o">-</span> <span class="n">start</span>

            <span class="n">region</span> <span class="o">=</span> <span class="n">GenomicRegion</span><span class="p">(</span><span class="n">contig</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">Strand</span><span class="o">.</span><span class="n">POSITIVE</span><span class="p">)</span>
            <span class="n">vc</span> <span class="o">=</span> <span class="n">VariantCoordinates</span><span class="p">(</span><span class="n">region</span><span class="p">,</span> <span class="n">ref</span><span class="p">,</span> <span class="n">alt</span><span class="p">,</span> <span class="n">change_length</span><span class="p">)</span>
        <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="n">variant_descriptor</span><span class="o">.</span><span class="n">expressions</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="c1"># We have some expressions. Let&#39;s try to find the 1st expression with `hgvs.c` syntax.</span>
            <span class="k">for</span> <span class="n">expression</span> <span class="ow">in</span> <span class="n">variant_descriptor</span><span class="o">.</span><span class="n">expressions</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">expression</span><span class="o">.</span><span class="n">syntax</span> <span class="o">==</span> <span class="s1">&#39;hgvs.c&#39;</span><span class="p">:</span>
                    <span class="n">vc</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_hgvs_finder</span><span class="o">.</span><span class="n">find_coordinates</span><span class="p">(</span><span class="n">expression</span><span class="o">.</span><span class="n">value</span><span class="p">)</span>
                    <span class="k">break</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Unable to find variant coordinates.&quot;</span><span class="p">)</span>

        <span class="c1"># Last, parse genotype.</span>
        <span class="n">genotype</span> <span class="o">=</span> <span class="n">variant_descriptor</span><span class="o">.</span><span class="n">allelic_state</span><span class="o">.</span><span class="n">label</span>
        <span class="n">gt</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_map_geno_genotype_label</span><span class="p">(</span><span class="n">genotype</span><span class="p">)</span>

        <span class="k">return</span> <span class="n">vc</span><span class="p">,</span> <span class="n">gt</span></div>


    <span class="k">def</span> <span class="nf">_check_assembly</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">genome_assembly</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
        <span class="k">if</span> <span class="s1">&#39;38&#39;</span> <span class="ow">in</span> <span class="n">genome_assembly</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_build</span><span class="o">.</span><span class="n">identifier</span> <span class="o">==</span> <span class="s1">&#39;GRCh38.p13&#39;</span><span class="p">:</span>
            <span class="k">return</span> <span class="kc">True</span>
        <span class="k">elif</span> <span class="p">(</span><span class="s1">&#39;37&#39;</span> <span class="ow">in</span> <span class="n">genome_assembly</span> <span class="ow">or</span> <span class="s1">&#39;19&#39;</span> <span class="ow">in</span> <span class="n">genome_assembly</span><span class="p">)</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">_build</span><span class="o">.</span><span class="n">identifier</span> <span class="o">==</span> <span class="s1">&#39;GRCh37.p13&#39;</span><span class="p">:</span>
            <span class="k">return</span> <span class="kc">True</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="kc">False</span>


    <span class="nd">@staticmethod</span>
    <span class="k">def</span> <span class="nf">_vcf_is_available</span><span class="p">(</span><span class="n">vcf_record</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Check if we can parse data out of VCF record.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">return</span> <span class="p">(</span><span class="n">vcf_record</span><span class="o">.</span><span class="n">genome_assembly</span> <span class="o">!=</span> <span class="s1">&#39;&#39;</span>
                <span class="ow">and</span> <span class="n">vcf_record</span><span class="o">.</span><span class="n">chrom</span> <span class="o">!=</span> <span class="s1">&#39;&#39;</span>
                <span class="ow">and</span> <span class="n">vcf_record</span><span class="o">.</span><span class="n">pos</span> <span class="o">&gt;=</span> <span class="mi">0</span>
                <span class="ow">and</span> <span class="n">vcf_record</span><span class="o">.</span><span class="n">ref</span> <span class="o">!=</span> <span class="s1">&#39;&#39;</span>
                <span class="ow">and</span> <span class="n">vcf_record</span><span class="o">.</span><span class="n">alt</span> <span class="o">!=</span> <span class="s1">&#39;&#39;</span><span class="p">)</span>

    <span class="nd">@staticmethod</span>
    <span class="k">def</span> <span class="nf">_cnv_is_available</span><span class="p">(</span><span class="n">variation</span><span class="p">):</span>
        <span class="n">seq_location</span> <span class="o">=</span> <span class="n">variation</span><span class="o">.</span><span class="n">copy_number</span><span class="o">.</span><span class="n">allele</span><span class="o">.</span><span class="n">sequence_location</span>
        <span class="k">return</span> <span class="p">(</span><span class="n">seq_location</span><span class="o">.</span><span class="n">sequence_id</span> <span class="o">!=</span> <span class="s1">&#39;&#39;</span>
                <span class="ow">and</span> <span class="n">seq_location</span><span class="o">.</span><span class="n">sequence_interval</span><span class="o">.</span><span class="n">start_number</span><span class="o">.</span><span class="n">value</span> <span class="o">&gt;=</span> <span class="mi">0</span>
                <span class="ow">and</span> <span class="n">seq_location</span><span class="o">.</span><span class="n">sequence_interval</span><span class="o">.</span><span class="n">end_number</span><span class="o">.</span><span class="n">value</span> <span class="o">&gt;=</span> <span class="mi">0</span>
                <span class="ow">and</span> <span class="n">variation</span><span class="o">.</span><span class="n">copy_number</span><span class="o">.</span><span class="n">number</span><span class="o">.</span><span class="n">value</span> <span class="o">!=</span> <span class="s1">&#39;&#39;</span><span class="p">)</span>

    <span class="nd">@staticmethod</span>
    <span class="k">def</span> <span class="nf">_map_geno_genotype_label</span><span class="p">(</span><span class="n">genotype</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Genotype</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">        Mapping from labels of the relevant GENO terms that is valid as of Oct 2nd, 2023.</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="n">genotype</span> <span class="ow">in</span> <span class="p">(</span><span class="s1">&#39;heterozygous&#39;</span><span class="p">,</span> <span class="s1">&#39;compound heterozygous&#39;</span><span class="p">,</span> <span class="s1">&#39;simple heterozygous&#39;</span><span class="p">):</span>
            <span class="k">return</span> <span class="n">Genotype</span><span class="o">.</span><span class="n">HETEROZYGOUS</span>
        <span class="k">elif</span> <span class="n">genotype</span> <span class="o">==</span> <span class="s1">&#39;homozygous&#39;</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">Genotype</span><span class="o">.</span><span class="n">HOMOZYGOUS_ALTERNATE</span>
        <span class="k">elif</span> <span class="n">genotype</span> <span class="ow">in</span> <span class="p">(</span><span class="s1">&#39;hemizygous&#39;</span><span class="p">,</span> <span class="s1">&#39;hemizygous X-linked&#39;</span><span class="p">,</span> <span class="s1">&#39;hemizygous Y-linked&#39;</span><span class="p">):</span>
            <span class="k">return</span> <span class="n">Genotype</span><span class="o">.</span><span class="n">HEMIZYGOUS</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Unknown genotype </span><span class="si">{</span><span class="n">genotype</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span></div>



<div class="viewcode-block" id="PhenopacketPatientCreator">
<a class="viewcode-back" href="../../../apidocs/genophenocorr.preprocessing.html#genophenocorr.preprocessing.PhenopacketPatientCreator">[docs]</a>
<span class="k">class</span> <span class="nc">PhenopacketPatientCreator</span><span class="p">(</span><span class="n">PatientCreator</span><span class="p">[</span><span class="n">Phenopacket</span><span class="p">]):</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    `PhenopacketPatientCreator` transforms `Phenopacket` into :class:`genophenocorr.model.Patient`.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">build</span><span class="p">:</span> <span class="n">GenomeBuild</span><span class="p">,</span>
                 <span class="n">phenotype_creator</span><span class="p">:</span> <span class="n">PhenotypeCreator</span><span class="p">,</span>
                 <span class="n">var_func_ann</span><span class="p">:</span> <span class="n">FunctionalAnnotator</span><span class="p">,</span>
                 <span class="n">hgvs_coordinate_finder</span><span class="p">:</span> <span class="n">VariantCoordinateFinder</span><span class="p">[</span><span class="nb">str</span><span class="p">]):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_logger</span> <span class="o">=</span> <span class="n">logging</span><span class="o">.</span><span class="n">getLogger</span><span class="p">(</span><span class="vm">__name__</span><span class="p">)</span>
        <span class="c1"># Violates DI, but it is specific to this class, so I&#39;ll leave it &quot;as is&quot;.</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_coord_finder</span> <span class="o">=</span> <span class="n">PhenopacketVariantCoordinateFinder</span><span class="p">(</span><span class="n">build</span><span class="p">,</span> <span class="n">hgvs_coordinate_finder</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_phenotype_creator</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">phenotype_creator</span><span class="p">,</span> <span class="n">PhenotypeCreator</span><span class="p">,</span> <span class="s1">&#39;phenotype_creator&#39;</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_func_ann</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">util</span><span class="o">.</span><span class="n">validate_instance</span><span class="p">(</span><span class="n">var_func_ann</span><span class="p">,</span> <span class="n">FunctionalAnnotator</span><span class="p">,</span> <span class="s1">&#39;var_func_ann&#39;</span><span class="p">)</span>

<div class="viewcode-block" id="PhenopacketPatientCreator.process">
<a class="viewcode-back" href="../../../apidocs/genophenocorr.preprocessing.html#genophenocorr.preprocessing.PhenopacketPatientCreator.process">[docs]</a>
    <span class="k">def</span> <span class="nf">process</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">inputs</span><span class="p">:</span> <span class="n">Phenopacket</span><span class="p">,</span> <span class="n">notepad</span><span class="p">:</span> <span class="n">Notepad</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Patient</span><span class="p">:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Creates a Patient from the data in a given Phenopacket</span>

<span class="sd">        Args:</span>
<span class="sd">            inputs (Phenopacket): A Phenopacket object</span>
<span class="sd">            notepad (Notepad): notepad to write down the issues</span>
<span class="sd">        Returns:</span>
<span class="sd">            Patient: A Patient object</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">sample_id</span> <span class="o">=</span> <span class="n">SampleLabels</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="n">inputs</span><span class="o">.</span><span class="n">subject</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="n">meta_label</span><span class="o">=</span><span class="n">inputs</span><span class="o">.</span><span class="n">id</span> <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">inputs</span><span class="o">.</span><span class="n">id</span><span class="p">)</span> <span class="o">&gt;</span> <span class="mi">0</span> <span class="k">else</span> <span class="kc">None</span><span class="p">)</span>

        <span class="c1"># Check phenotypes </span>
        <span class="n">pfs</span> <span class="o">=</span> <span class="n">notepad</span><span class="o">.</span><span class="n">add_subsection</span><span class="p">(</span><span class="s1">&#39;phenotype-features&#39;</span><span class="p">)</span>
        <span class="n">phenotypes</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_phenotype_creator</span><span class="o">.</span><span class="n">process</span><span class="p">(</span>
            <span class="p">((</span><span class="n">pf</span><span class="o">.</span><span class="n">type</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="ow">not</span> <span class="n">pf</span><span class="o">.</span><span class="n">excluded</span><span class="p">)</span> <span class="k">for</span> <span class="n">pf</span> <span class="ow">in</span> <span class="n">inputs</span><span class="o">.</span><span class="n">phenotypic_features</span><span class="p">),</span>
            <span class="n">pfs</span>
        <span class="p">)</span>
        
        <span class="c1"># Check diseases</span>
        <span class="n">diseases</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_add_diseases</span><span class="p">([</span><span class="n">dis</span> <span class="k">for</span> <span class="n">dis</span> <span class="ow">in</span> <span class="n">inputs</span><span class="o">.</span><span class="n">diseases</span><span class="p">],</span> <span class="n">pfs</span><span class="p">)</span>

        <span class="c1"># Check variants</span>
        <span class="n">vs</span> <span class="o">=</span> <span class="n">notepad</span><span class="o">.</span><span class="n">add_subsection</span><span class="p">(</span><span class="s1">&#39;variants&#39;</span><span class="p">)</span>
        <span class="n">variants</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_add_variants</span><span class="p">(</span><span class="n">sample_id</span><span class="p">,</span> <span class="n">inputs</span><span class="p">,</span> <span class="n">vs</span><span class="p">)</span>
    
        <span class="k">return</span> <span class="n">Patient</span><span class="p">(</span><span class="n">sample_id</span><span class="p">,</span> <span class="n">phenotypes</span><span class="o">=</span><span class="n">phenotypes</span><span class="p">,</span> <span class="n">variants</span><span class="o">=</span><span class="n">variants</span><span class="p">,</span> <span class="n">diseases</span><span class="o">=</span><span class="n">diseases</span><span class="p">)</span></div>


    <span class="k">def</span> <span class="nf">_add_diseases</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">diseases</span><span class="p">:</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">PPDisease</span><span class="p">],</span> <span class="n">notepad</span><span class="p">:</span> <span class="n">Notepad</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">Disease</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Creates a list of Disease objects from the data in a given Phenopacket</span>

<span class="sd">        Args:</span>
<span class="sd">            diseases (Sequence[Disease]): A sequence of Phenopacket Disease objects</span>
<span class="sd">        Returns:</span>
<span class="sd">            Sequence[Dis]: A list of Disease objects</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">diseases</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">notepad</span><span class="o">.</span><span class="n">add_warning</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;No diseases found.&quot;</span><span class="p">)</span>
            <span class="k">return</span> <span class="p">[]</span>
        <span class="n">final_diseases</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">dis</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">diseases</span><span class="p">):</span>
            <span class="k">if</span> <span class="ow">not</span> <span class="n">dis</span><span class="o">.</span><span class="n">HasField</span><span class="p">(</span><span class="s2">&quot;term&quot;</span><span class="p">):</span>
                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s1">&#39;Could not find term in Disease.&#39;</span><span class="p">)</span>
            <span class="n">term_id</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">TermId</span><span class="o">.</span><span class="n">from_curie</span><span class="p">(</span><span class="n">dis</span><span class="o">.</span><span class="n">term</span><span class="o">.</span><span class="n">id</span><span class="p">)</span>
            <span class="c1"># Do not include excluded diseases if we decide to assume excluded if not included </span>
            <span class="n">final_diseases</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Disease</span><span class="p">(</span><span class="n">term_id</span><span class="p">,</span> <span class="n">dis</span><span class="o">.</span><span class="n">term</span><span class="o">.</span><span class="n">label</span><span class="p">,</span> <span class="ow">not</span> <span class="n">dis</span><span class="o">.</span><span class="n">excluded</span><span class="p">))</span>
        <span class="k">return</span> <span class="n">final_diseases</span>
        

    <span class="k">def</span> <span class="nf">_add_variants</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">sample_id</span><span class="p">:</span> <span class="n">SampleLabels</span><span class="p">,</span> <span class="n">pp</span><span class="p">:</span> <span class="n">Phenopacket</span><span class="p">,</span> <span class="n">notepad</span><span class="p">:</span> <span class="n">Notepad</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">typing</span><span class="o">.</span><span class="n">Sequence</span><span class="p">[</span><span class="n">Variant</span><span class="p">]:</span>
<span class="w">        </span><span class="sd">&quot;&quot;&quot;Creates a list of Variant objects from the data in a given Phenopacket</span>

<span class="sd">        Args:</span>
<span class="sd">            pp (Phenopacket): A Phenopacket object</span>
<span class="sd">        Returns:</span>
<span class="sd">            Sequence[Variant]: A list of Variant objects</span>
<span class="sd">        &quot;&quot;&quot;</span>
        <span class="n">variants</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">interp</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">pp</span><span class="o">.</span><span class="n">interpretations</span><span class="p">):</span>
            <span class="k">for</span> <span class="n">genomic_interp</span> <span class="ow">in</span> <span class="n">interp</span><span class="o">.</span><span class="n">diagnosis</span><span class="o">.</span><span class="n">genomic_interpretations</span><span class="p">:</span>
                <span class="k">try</span><span class="p">:</span>
                    <span class="n">vc</span><span class="p">,</span> <span class="n">gt</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_coord_finder</span><span class="o">.</span><span class="n">find_coordinates</span><span class="p">(</span><span class="n">genomic_interp</span><span class="p">)</span>
                <span class="k">except</span> <span class="ne">ValueError</span><span class="p">:</span>
                    <span class="n">notepad</span><span class="o">.</span><span class="n">add_warning</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Expected a VCF record, a VRS CNV, or an expression with `hgvs.c` but had an error retrieving any from patient </span><span class="si">{</span><span class="n">sample_id</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">,</span>
                                        <span class="s2">&quot;Remove variant from testing&quot;</span><span class="p">)</span>
                    <span class="k">continue</span>

                <span class="k">try</span><span class="p">:</span>
                    <span class="n">tx_annotations</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_func_ann</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="n">vc</span><span class="p">)</span>
                <span class="k">except</span> <span class="ne">ValueError</span><span class="p">:</span>
                    <span class="n">notepad</span><span class="o">.</span><span class="n">add_warning</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Patient </span><span class="si">{</span><span class="n">pp</span><span class="o">.</span><span class="n">id</span><span class="si">}</span><span class="s2"> has an error with variant </span><span class="si">{</span><span class="n">vc</span><span class="o">.</span><span class="n">variant_key</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">,</span>
                                        <span class="s2">&quot;Try again or remove variant form testing&quot;</span><span class="p">)</span>
                    <span class="k">continue</span>

                <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">tx_annotations</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
                    <span class="n">notepad</span><span class="o">.</span><span class="n">add_warning</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Patient </span><span class="si">{</span><span class="n">pp</span><span class="o">.</span><span class="n">id</span><span class="si">}</span><span class="s2"> has an error with variant </span><span class="si">{</span><span class="n">vc</span><span class="o">.</span><span class="n">variant_key</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">,</span>
                                        <span class="s2">&quot;Remove variant from testing&quot;</span><span class="p">)</span>
                    <span class="k">continue</span>

                <span class="n">genotype</span> <span class="o">=</span> <span class="n">Genotypes</span><span class="o">.</span><span class="n">single</span><span class="p">(</span><span class="n">sample_id</span><span class="p">,</span> <span class="n">gt</span><span class="p">)</span>
                <span class="n">variants</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Variant</span><span class="p">(</span><span class="n">vc</span><span class="p">,</span> <span class="n">tx_annotations</span><span class="p">,</span> <span class="n">genotype</span><span class="p">))</span>

        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">variants</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">notepad</span><span class="o">.</span><span class="n">add_warning</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Patient </span><span class="si">{</span><span class="n">pp</span><span class="o">.</span><span class="n">id</span><span class="si">}</span><span class="s2"> has no variants to work with&quot;</span><span class="p">)</span>

        <span class="k">return</span> <span class="n">variants</span></div>


</pre></div>

           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>