

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Mode of Inheritance Predicates &mdash; GPSEA 0.6.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=fa44fd50" />
      <link rel="stylesheet" type="text/css" href="../../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../_static/gpsea.css?v=547657ed" />

  
      <script src="../../_static/jquery.js?v=5d32c60e"></script>
      <script src="../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../../_static/documentation_options.js?v=8fa8b3e9"></script>
      <script src="../../_static/doctools.js?v=9bcbadda"></script>
      <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../../_static/copybutton.js?v=6dbb43f8"></script>
      <script>window.MathJax = {"TeX": {"Macros": {"RR": "{\\mathbb{R}}"}}}</script>
      <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Partition by the sex of the individual" href="male_female_predicate.html" />
    <link rel="prev" title="Allele count predicates" href="allele_count_predicates.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Installation &amp; Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../index.html">User guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../input-data.html">Input data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exploratory.html">Cohort exploratory analysis</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="index.html">Predicates</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="genotype_predicates.html">Genotype Predicates</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="variant_predicates.html">Variant Predicates</a></li>
<li class="toctree-l4"><a class="reference internal" href="allele_count_predicates.html">Allele count predicates</a></li>
<li class="toctree-l4 current"><a class="current reference internal" href="#">Mode of Inheritance Predicates</a></li>
<li class="toctree-l4"><a class="reference internal" href="male_female_predicate.html">Partition by the sex of the individual</a></li>
<li class="toctree-l4"><a class="reference internal" href="diagnosis_predicate.html">Partition by a diagnosis</a></li>
<li class="toctree-l4"><a class="reference internal" href="groups_predicate.html">Groups Predicate</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="phenotype_predicates.html">Phenotype Predicates</a></li>
<li class="toctree-l3"><a class="reference internal" href="index.html#gallery">Gallery</a></li>
<li class="toctree-l3"><a class="reference internal" href="index.html#need-more">Need more?</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../analyses/index.html">Statistical analyses</a></li>
<li class="toctree-l2"><a class="reference internal" href="../mtc.html">Multiple-testing correction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../general_hpo_terms.html">General HPO terms</a></li>
<li class="toctree-l2"><a class="reference internal" href="../glossary.html">Glossary</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../index.html">User guide</a></li>
          <li class="breadcrumb-item"><a href="index.html">Predicates</a></li>
          <li class="breadcrumb-item"><a href="genotype_predicates.html">Genotype Predicates</a></li>
      <li class="breadcrumb-item active">Mode of Inheritance Predicates</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/user-guide/predicates/mode_of_inheritance_predicate.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="mode-of-inheritance-predicates">
<span id="mode-of-inheritance-predicate"></span><h1>Mode of Inheritance Predicates<a class="headerlink" href="#mode-of-inheritance-predicates" title="Link to this heading"></a></h1>
<p>There are five basic modes of inheritance for single-gene diseases: autosomal dominant,
autosomal recessive, X-linked dominant, X-linked recessive, and mitochondrial
(See <a class="reference external" href="https://www.ncbi.nlm.nih.gov/books/NBK132145/">Understanding Genetics, Appendix B</a>).</p>
<p>GPSEA provides Mode of Inheritance predicates that categorize individuals
based on the number of variant alleles. Clinical judgment should be used to choose the MoI
for the cohort analysis.</p>
<p>By default, MoI predicates use <em>all</em> variants recorded in the individual.
However, a <a class="reference internal" href="../../apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate" title="gpsea.analysis.predicate.genotype.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a>
can be provided to select the variants of interest. For instance, missense variants
or the variants that overlap with a specific protein domain.</p>
<section id="autosomal-dominant-predicate">
<span id="id1"></span><h2>Autosomal dominant predicate<a class="headerlink" href="#autosomal-dominant-predicate" title="Link to this heading"></a></h2>
<p>Autosomal dominant predicate counts the alleles of variants of interest
to assign an individual into one of the following categories:</p>
<table class="docutils align-default" id="id6">
<caption><span class="caption-text">Autosomal dominant predicate categories</span><a class="headerlink" href="#id6" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"><p>Allele count</p></th>
<th class="head"><p>Category</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>0</p></td>
<td><p><cite>No allele</cite></p></td>
</tr>
<tr class="row-odd"><td><p>1</p></td>
<td><p><cite>Monoallelic</cite></p></td>
</tr>
<tr class="row-even"><td><p><span class="math notranslate nohighlight">\(\ge 2\)</span></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">None</span></code></p></td>
</tr>
</tbody>
</table>
<section id="examples">
<h3>Examples<a class="headerlink" href="#examples" title="Link to this heading"></a></h3>
<section id="use-all-variants">
<h4>Use all variants<a class="headerlink" href="#use-all-variants" title="Link to this heading"></a></h4>
<p>We can create the predicate with the <a class="reference internal" href="../../apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.autosomal_dominant" title="gpsea.analysis.predicate.genotype.autosomal_dominant"><code class="xref py py-func docutils literal notranslate"><span class="pre">autosomal_dominant()</span></code></a> function:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">autosomal_dominant</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span> <span class="o">=</span> <span class="n">autosomal_dominant</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span><span class="o">.</span><span class="n">display_question</span><span class="p">()</span>
<span class="go">&#39;What is the genotype group: No allele, Monoallelic&#39;</span>
</pre></div>
</div>
</section>
<section id="use-a-variant-subset">
<h4>Use a variant subset<a class="headerlink" href="#use-a-variant-subset" title="Link to this heading"></a></h4>
<p>To only consider a subset of variants, a <a class="reference internal" href="../../apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate" title="gpsea.analysis.predicate.genotype.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a> can be provided,
to, for example, only include the missense variants on transcript <code class="docutils literal notranslate"><span class="pre">NM_1234.5</span></code>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_missense</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">effect</span><span class="o">=</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="s2">&quot;NM_1234.5&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_missense</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;MISSENSE_VARIANT on NM_1234.5&#39;</span>
</pre></div>
</div>
</section>
</section>
</section>
<section id="autosomal-recessive-predicate">
<span id="id2"></span><h2>Autosomal recessive predicate<a class="headerlink" href="#autosomal-recessive-predicate" title="Link to this heading"></a></h2>
<p>The autosomal recessive predicate uses the allele count to assign
an individual into one of the genotype categories:</p>
<table class="docutils align-default" id="id7">
<caption><span class="caption-text">Autosomal recessive predicate categories</span><a class="headerlink" href="#id7" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"><p>Allele count</p></th>
<th class="head"><p>Category</p></th>
<th class="head"><p>Category index</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>0</p></td>
<td><p><cite>No allele</cite></p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><p>1</p></td>
<td><p><cite>Monoallelic</cite></p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><p>2</p></td>
<td><p><cite>Biallelic</cite></p></td>
<td><p>2</p></td>
</tr>
<tr class="row-odd"><td><p><span class="math notranslate nohighlight">\(\ge 3\)</span></p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">None</span></code></p></td>
<td></td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><cite>Biallelic</cite> includes both homozygous and compound heterozygous genotypes.</p>
</div>
<section id="partitions">
<h3>Partitions<a class="headerlink" href="#partitions" title="Link to this heading"></a></h3>
<p>Sometimes we are interested in lumping several genotype categories into a group or and then comparing the groups.
For instance, to compare phenotype of the individuals with <em>at least one</em> frameshift allele
with those with <em>no</em> frameshift allele. Alternatively, we may only want to analyze a subset of the genotype categories,
such as <cite>Monoallelic</cite> vs. <cite>Biallelic</cite>.</p>
<p>The <cite>partitions</cite> option of the <a class="reference internal" href="../../apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.autosomal_recessive" title="gpsea.analysis.predicate.genotype.autosomal_recessive"><code class="xref py py-func docutils literal notranslate"><span class="pre">autosomal_recessive()</span></code></a> function
lets us do this.
The option needs a set of sets of category indices (see table above).
The set is a partition of a set with the following properties:</p>
<ul class="simple">
<li><p>no subset is empty</p></li>
<li><p>the intersection of any subsets is empty</p></li>
</ul>
<p>These rules are very much alike the properties of the <a class="reference external" href="https://en.wikipedia.org/wiki/Partition_of_a_set">partitions of a set</a>,
with the exception that adherence to <em>the union of the subsets include all group indices</em> rule is <em>NOT</em> required.</p>
</section>
<section id="id3">
<h3>Examples<a class="headerlink" href="#id3" title="Link to this heading"></a></h3>
<section id="id4">
<h4>Use all variants<a class="headerlink" href="#id4" title="Link to this heading"></a></h4>
<p>We create the predicate
with the <a class="reference internal" href="../../apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.autosomal_recessive" title="gpsea.analysis.predicate.genotype.autosomal_recessive"><code class="xref py py-func docutils literal notranslate"><span class="pre">autosomal_recessive()</span></code></a> function:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">autosomal_recessive</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span> <span class="o">=</span> <span class="n">autosomal_recessive</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span><span class="o">.</span><span class="n">display_question</span><span class="p">()</span>
<span class="go">&#39;What is the genotype group: No allele, Monoallelic, Biallelic&#39;</span>
</pre></div>
</div>
</section>
<section id="id5">
<h4>Use a variant subset<a class="headerlink" href="#id5" title="Link to this heading"></a></h4>
<p>Same as in the autosomal dominant version,
we can use a <a class="reference internal" href="../../apidocs/gpsea.analysis.predicate.genotype.html#gpsea.analysis.predicate.genotype.VariantPredicate" title="gpsea.analysis.predicate.genotype.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a>
to only count the alleles of the variants of interest, such as the missense variants
of a fictional transcript <code class="docutils literal notranslate"><span class="pre">NM_1234.5</span></code>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.model</span> <span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">gpsea.analysis.predicate.genotype</span> <span class="kn">import</span> <span class="n">VariantPredicates</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_missense</span> <span class="o">=</span> <span class="n">VariantPredicates</span><span class="o">.</span><span class="n">variant_effect</span><span class="p">(</span><span class="n">effect</span><span class="o">=</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="s2">&quot;NM_1234.5&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_missense</span><span class="o">.</span><span class="n">get_question</span><span class="p">()</span>
<span class="go">&#39;MISSENSE_VARIANT on NM_1234.5&#39;</span>
</pre></div>
</div>
<p>and then use it to create the autosomal recessive predicate:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span> <span class="o">=</span> <span class="n">autosomal_recessive</span><span class="p">(</span><span class="n">is_missense</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span><span class="o">.</span><span class="n">display_question</span><span class="p">()</span>
<span class="go">&#39;What is the genotype group: No allele, Monoallelic, Biallelic&#39;</span>
</pre></div>
</div>
<p>This predicate will assign the individuals into one of the listed genotype categories
based on the allele counts of the missense variants.</p>
</section>
<section id="compare-monoallelic-vs-biallelic">
<h4>Compare <cite>Monoallelic</cite> vs. <cite>Biallelic</cite><a class="headerlink" href="#compare-monoallelic-vs-biallelic" title="Link to this heading"></a></h4>
<p>We can provide <code class="docutils literal notranslate"><span class="pre">partitions</span></code> to only compare the heterozygotes with those carrying
biallelic alt mutations (homozygous alternate or compound heterozygous):</p>
<p>We consult the <em>Autosomal recessive predicate categories</em> table for the category indices
and we create the genotype group partitions:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="c1"># `1` for `Monoallelic` and `2` for `Biallelic`</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">partitions</span> <span class="o">=</span> <span class="p">({</span><span class="mi">1</span><span class="p">,},</span> <span class="p">{</span><span class="mi">2</span><span class="p">,})</span>
</pre></div>
</div>
<p>which we use to create the autosomal recessive predicate:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span> <span class="o">=</span> <span class="n">autosomal_recessive</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">partitions</span><span class="o">=</span><span class="n">partitions</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_predicate</span><span class="o">.</span><span class="n">display_question</span><span class="p">()</span>
<span class="go">&#39;What is the genotype group: Monoallelic, Biallelic&#39;</span>
</pre></div>
</div>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="allele_count_predicates.html" class="btn btn-neutral float-left" title="Allele count predicates" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="male_female_predicate.html" class="btn btn-neutral float-right" title="Partition by the sex of the individual" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>