

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Compare measurements &mdash; GPSEA 0.9.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="../../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../_static/gpsea.css?v=547657ed" />

  
      <script src="../../_static/jquery.js?v=5d32c60e"></script>
      <script src="../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../../_static/documentation_options.js?v=39bb1c6d"></script>
      <script src="../../_static/doctools.js?v=9bcbadda"></script>
      <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../../_static/copybutton.js?v=6dbb43f8"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Survival analysis" href="survival.html" />
    <link rel="prev" title="Compare phenotype scores" href="phenotype-scores.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../index.html">User guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../input-data.html">Input data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exploratory.html">Cohort exploratory analysis</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="index.html">Statistical analyses</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="partitioning/index.html">Partitioning</a></li>
<li class="toctree-l3"><a class="reference internal" href="phenotype-classes.html">Compare genotype and phenotype classes</a></li>
<li class="toctree-l3"><a class="reference internal" href="mtc.html">Multiple-testing correction</a></li>
<li class="toctree-l3"><a class="reference internal" href="phenotype-scores.html">Compare phenotype scores</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Compare measurements</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#example-analysis">Example analysis</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="survival.html">Survival analysis</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../glossary.html">Glossary</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../index.html">User guide</a></li>
          <li class="breadcrumb-item"><a href="index.html">Statistical analyses</a></li>
      <li class="breadcrumb-item active">Compare measurements</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/user-guide/analyses/measurements.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="compare-measurements">
<span id="measurement-stat"></span><h1>Compare measurements<a class="headerlink" href="#compare-measurements" title="Link to this heading"></a></h1>
<section id="example-analysis">
<h2>Example analysis<a class="headerlink" href="#example-analysis" title="Link to this heading"></a></h2>
<p>We will analyze 69 individuals reported by
<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30968594/">Xu et al. (2019)</a>.
The reported data includes the <em>CYP21A2</em> mutations, the lab measurement,
and other clinical signs and symptoms encoded as HPO terms.</p>
<p>TODO: wordsmith</p>
<section id="load-cohort">
<h3>Load cohort<a class="headerlink" href="#load-cohort" title="Link to this heading"></a></h3>
<p>For the purpose of this analysis, we will load the <a class="reference internal" href="../../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>
from a <a class="reference external" href="https://github.com/monarch-initiative/gpsea/tree/main/docs/cohort-data/CYP21A2.0.1.20.json">JSON file</a>.
The cohort was prepared from phenopackets as described in <a class="reference internal" href="../input-data.html#create-a-cohort"><span class="std std-ref">Create a cohort</span></a> section,
and then serialized as a JSON file following the instructions in <a class="reference internal" href="../input-data.html#cohort-persistence"><span class="std std-ref">Persist the cohort for later</span></a> section.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.io</span><span class="w"> </span><span class="kn">import</span> <span class="n">GpseaJSONDecoder</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fpath_cohort_json</span> <span class="o">=</span> <span class="s1">&#39;docs/cohort-data/CYP21A2.0.1.20.json&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">fpath_cohort_json</span><span class="p">)</span> <span class="k">as</span> <span class="n">fh</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">cohort</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">fh</span><span class="p">,</span> <span class="bp">cls</span><span class="o">=</span><span class="n">GpseaJSONDecoder</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
<span class="go">69</span>
</pre></div>
</div>
</section>
<section id="configure-analysis">
<h3>Configure analysis<a class="headerlink" href="#configure-analysis" title="Link to this heading"></a></h3>
<p><em>MANE</em> transcript of <em>CYP21A2</em>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_000500.9&#39;</span>
</pre></div>
</div>
<section id="genotype-predicate">
<h4>Genotype predicate<a class="headerlink" href="#genotype-predicate" title="Link to this heading"></a></h4>
<p>For now, just Missense vs rest.
Nonsense variants + big SVs,
Missense should <em>NOT</em> be severe.
TODO - create real predicate.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.model</span><span class="w"> </span><span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.predicate</span><span class="w"> </span><span class="kn">import</span> <span class="n">variant_effect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_missense</span> <span class="o">=</span> <span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">MISSENSE_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="o">=</span><span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_missense</span><span class="o">.</span><span class="n">description</span>
<span class="go">&#39;MISSENSE_VARIANT on NM_000500.9&#39;</span>
</pre></div>
</div>
<p>Assuming AR inheritance, we compare missense vs. rest:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.clf</span><span class="w"> </span><span class="kn">import</span> <span class="n">biallelic_classifier</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span> <span class="o">=</span> <span class="n">biallelic_classifier</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">a_predicate</span><span class="o">=</span><span class="n">is_missense</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">b_predicate</span><span class="o">=~</span><span class="n">is_missense</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">a_label</span><span class="o">=</span><span class="s2">&quot;Missense&quot;</span><span class="p">,</span> <span class="n">b_label</span><span class="o">=</span><span class="s2">&quot;Other&quot;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">partitions</span><span class="o">=</span><span class="p">({</span><span class="mi">0</span><span class="p">,},</span> <span class="p">{</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">}),</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span><span class="o">.</span><span class="n">class_labels</span>
<span class="go">(&#39;Missense/Missense&#39;, &#39;Missense/Other OR Other/Other&#39;)</span>
</pre></div>
</div>
</section>
<section id="phenotype-score">
<h4>Phenotype score<a class="headerlink" href="#phenotype-score" title="Link to this heading"></a></h4>
<p>We investigate an association between a measurement value and a genotype group.</p>
<p>We use the measurement of <a class="reference external" href="https://loinc.org/2986-8/">Testosterone [Mass/volume] in Serum or Plasma</a>
(<cite>LOINC:2986-8</cite>).</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.pscore</span><span class="w"> </span><span class="kn">import</span> <span class="n">MeasurementPhenotypeScorer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">testosterone</span> <span class="o">=</span> <span class="s1">&#39;LOINC:2986-8&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pheno_scorer</span> <span class="o">=</span> <span class="n">MeasurementPhenotypeScorer</span><span class="o">.</span><span class="n">from_measurement_id</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">term_id</span><span class="o">=</span><span class="n">testosterone</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">label</span><span class="o">=</span><span class="s2">&quot;Testosterone [Mass/volume] in Serum or Plasma&quot;</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pheno_scorer</span><span class="o">.</span><span class="n">description</span>
<span class="go">&#39;Value of Testosterone [Mass/volume] in Serum or Plasma [LOINC:2986-8]&#39;</span>
</pre></div>
</div>
</section>
<section id="statistical-test">
<h4>Statistical test<a class="headerlink" href="#statistical-test" title="Link to this heading"></a></h4>
<p>We will use T test to test for differences between scores
of the different genotype groups</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.pscore.stats</span><span class="w"> </span><span class="kn">import</span> <span class="n">TTestStatistic</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">score_statistic</span> <span class="o">=</span> <span class="n">TTestStatistic</span><span class="p">()</span>
</pre></div>
</div>
</section>
<section id="final-analysis">
<h4>Final analysis<a class="headerlink" href="#final-analysis" title="Link to this heading"></a></h4>
<p>We will put the final analysis together into <a class="reference internal" href="../../apidocs/gpsea.analysis.pscore.html#gpsea.analysis.pscore.PhenotypeScoreAnalysis" title="gpsea.analysis.pscore.PhenotypeScoreAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">PhenotypeScoreAnalysis</span></code></a>.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.pscore</span><span class="w"> </span><span class="kn">import</span> <span class="n">PhenotypeScoreAnalysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">score_analysis</span> <span class="o">=</span> <span class="n">PhenotypeScoreAnalysis</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">score_statistic</span><span class="o">=</span><span class="n">score_statistic</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>
<section id="analysis">
<h3>Analysis<a class="headerlink" href="#analysis" title="Link to this heading"></a></h3>
<p>We execute the analysis by running</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span> <span class="o">=</span> <span class="n">score_analysis</span><span class="o">.</span><span class="n">compare_genotype_vs_phenotype_score</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">gt_clf</span><span class="o">=</span><span class="n">gt_clf</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">pheno_scorer</span><span class="o">=</span><span class="n">pheno_scorer</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">pval</span>
<span class="go">0.741216622359659</span>
</pre></div>
</div>
<p>Show data frame with scores</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">scores</span> <span class="o">=</span> <span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">sort_index</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">scores</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>  
<span class="go">                                            genotype  phenotype</span>
<span class="go">patient_id</span>
<span class="go">individual 10[PMID_30968594_individual_10]         1      614.0</span>
<span class="go">individual 11[PMID_30968594_individual_11]         1      630.0</span>
<span class="go">individual 12[PMID_30968594_individual_12]         1        NaN</span>
<span class="go">individual 13[PMID_30968594_individual_13]         1      303.0</span>
<span class="go">individual 14[PMID_30968594_individual_14]         1      664.0</span>
</pre></div>
</div>
<p>Prepare genotype category legend:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">gt_id_to_name</span> <span class="o">=</span> <span class="p">{</span><span class="n">c</span><span class="o">.</span><span class="n">category</span><span class="o">.</span><span class="n">cat_id</span><span class="p">:</span> <span class="n">c</span><span class="o">.</span><span class="n">category</span><span class="o">.</span><span class="n">name</span> <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">gt_clf</span><span class="o">.</span><span class="n">get_categorizations</span><span class="p">()}</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_id_to_name</span>
<span class="go">{0: &#39;Missense/Missense&#39;, 1: &#39;Missense/Other OR Other/Other&#39;}</span>
</pre></div>
</div>
<p>TODO: wordsmith &amp; finish!</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="phenotype-scores.html" class="btn btn-neutral float-left" title="Compare phenotype scores" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="survival.html" class="btn btn-neutral float-right" title="Survival analysis" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>