

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Compare genotype and phenotype classes &mdash; GPSEA 0.9.8 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="../../_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../_static/gpsea.css?v=547657ed" />

  
      <script src="../../_static/jquery.js?v=5d32c60e"></script>
      <script src="../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../../_static/documentation_options.js?v=1a282f62"></script>
      <script src="../../_static/doctools.js?v=9bcbadda"></script>
      <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="../../_static/copybutton.js?v=6dbb43f8"></script>
      <script>window.MathJax = {"TeX": {"Macros": {"RR": "{\\mathbb{R}}"}}}</script>
      <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Multiple-testing correction" href="mtc.html" />
    <link rel="prev" title="Gallery" href="partitioning/gallery.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../index.html" class="icon icon-home">
            GPSEA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../tutorial.html">Tutorial</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../index.html">User guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../input-data.html">Input data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exploratory.html">Cohort exploratory analysis</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="index.html">Statistical analyses</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="partitioning/index.html">Partitioning</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Compare genotype and phenotype classes</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#fisher-exact-test">Fisher exact test</a></li>
<li class="toctree-l4"><a class="reference internal" href="#example-analysis">Example analysis</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="mtc.html">Multiple-testing correction</a></li>
<li class="toctree-l3"><a class="reference internal" href="phenotype-scores.html">Compare phenotype scores</a></li>
<li class="toctree-l3"><a class="reference internal" href="measurements.html">Compare measurements</a></li>
<li class="toctree-l3"><a class="reference internal" href="survival.html">Survival analysis</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../custom-components.html">Custom components</a></li>
<li class="toctree-l2"><a class="reference internal" href="../glossary.html">Glossary</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../apidocs/modules.html">API reference</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">GPSEA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../index.html">User guide</a></li>
          <li class="breadcrumb-item"><a href="index.html">Statistical analyses</a></li>
      <li class="breadcrumb-item active">Compare genotype and phenotype classes</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/user-guide/analyses/phenotype-classes.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="compare-genotype-and-phenotype-classes">
<span id="genotype-phenotype-classes"></span><h1>Compare genotype and phenotype classes<a class="headerlink" href="#compare-genotype-and-phenotype-classes" title="Link to this heading"></a></h1>
<p>In this section, we show how to test the association between genotype and phenotype classes.
We assume a cohort was preprocessed following the <a class="reference internal" href="../input-data.html#input-data"><span class="std std-ref">Input data</span></a> section,
and we use classifiers described in the <a class="reference internal" href="partitioning/index.html#partitioning"><span class="std std-ref">Partitioning</span></a> to assign each cohort member
into a class along the genotype and phenotype axes.
We use Fisher exact test (FET) to test for differences between the classes
and we apply multiple testing correction to mitigate finding significant associations by chance.</p>
<section id="fisher-exact-test">
<span id="id1"></span><h2>Fisher exact test<a class="headerlink" href="#fisher-exact-test" title="Link to this heading"></a></h2>
<p>The Fisher exact test calculates the exact probability value
for the relationship between two dichotomous variables.
In our implementation, the two dichotomous variables are the genotype and the phenotype.
For instance, the individuals of the cohort may be divided
according to whether or not they have a nonsense variant
and according to whether or not they have Strabismus (<a class="reference external" href="https://hpo.jax.org/browse/term/HP:0000486">HP:0000486</a>).</p>
<p>The results of FET are expressed in terms of an exact probability (P-value), varying within 0 and 1.
Two groups are considered statistically significant if the P-value is less
than the chosen significance level (usually <span class="math notranslate nohighlight">\(\alpha = 0.05\)</span>).</p>
<p>The following graphic shows an example contingency table that is used to conduct a Fisher exact test.
We are comparing the frequency of <em>strabismus</em> in individuals with missense and nonsense variants:</p>
<a class="reference internal image-reference" href="../../_images/fisher.png"><img alt="Fisher exact text example" class="align-center" src="../../_images/fisher.png" style="width: 600px;" />
</a>
<p>To perform the corresponding test in Python, we would use the following code.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">scipy.stats</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">stats</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">contingency_table</span> <span class="o">=</span> <span class="p">[</span>
<span class="gp">... </span><span class="c1">#   Missense,    Nonsense</span>
<span class="gp">... </span>   <span class="p">[</span><span class="mi">17</span><span class="p">,</span>          <span class="mi">6</span>       <span class="p">],</span>  <span class="c1"># Strabismus observed</span>
<span class="gp">... </span>   <span class="p">[</span><span class="mi">1</span><span class="p">,</span>          <span class="mi">19</span>       <span class="p">],</span>  <span class="c1"># Strabismus excluded</span>
<span class="gp">... </span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">oddsratio</span><span class="p">,</span> <span class="n">p_value</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">fisher_exact</span><span class="p">(</span><span class="n">contingency_table</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">float</span><span class="p">(</span><span class="n">oddsratio</span><span class="p">)</span>
<span class="go">53.833333333333336</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">float</span><span class="p">(</span><span class="n">p_value</span><span class="p">)</span>
<span class="go">5.432292015291845e-06</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">p_value</span></code> evaluates to <cite>5.432292015291845e-06</cite>, meaning there is a significant difference between the groups.</p>
<p>The Fisher exact test evaluates whether the observed frequencies in a contingency table significantly
deviate from the frequencies we would expect if there were no association between the variables.
We want to test whether the frequency of <cite>HP:0000486`</cite> is significantly higher or lower in
one genotype class compared to what would be expected if there were no association.
Note that by default, the <em>two-tailed</em> Fisher exact test is performed, meaning we have no
hypothesis as to whether there is a higher or lower frequency in one of the genotype classes.</p>
<p>However, we are typically interested in testing the associations between the genotype and multiple phenotypic features at once.
GPSEA takes advantage of the HPO structure and simplifies the testing for all HPO terms encoded in the cohort.</p>
</section>
<section id="example-analysis">
<h2>Example analysis<a class="headerlink" href="#example-analysis" title="Link to this heading"></a></h2>
<p>Let’s illustrate this in a real-life example of the analysis of the association between frameshift variants in <em>TBX5</em> gene
and congenital heart defects in the dataset of 156 individuals with mutations in <em>TBX5</em> whose signs and symptoms were
encoded into HPO terms, stored as phenopackets of the <a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/35705716">GA4GH Phenopacket Schema</a>,
and deposited in <a class="reference external" href="https://github.com/monarch-initiative/phenopacket-store">Phenopacket Store</a>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The shorter version of the same analysis has been presented in the <a class="reference internal" href="../../tutorial.html#tutorial"><span class="std std-ref">Tutorial</span></a>.</p>
</div>
<section id="load-cohort">
<h3>Load cohort<a class="headerlink" href="#load-cohort" title="Link to this heading"></a></h3>
<p>For the purpose of this analysis, we will load the <a class="reference internal" href="../../apidocs/gpsea.model.html#gpsea.model.Cohort" title="gpsea.model.Cohort"><code class="xref py py-class docutils literal notranslate"><span class="pre">Cohort</span></code></a>
from a <a class="reference external" href="https://github.com/P2GX/gpsea/tree/main/docs/cohort-data/TBX5.0.1.20.json">JSON file</a>.
The cohort was prepared from phenopackets as described in <a class="reference internal" href="../input-data.html#create-a-cohort"><span class="std std-ref">Create a cohort</span></a> section,
and then serialized as a JSON file following the instructions in <a class="reference internal" href="../input-data.html#cohort-persistence"><span class="std std-ref">Persist the cohort for later</span></a> section.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.io</span><span class="w"> </span><span class="kn">import</span> <span class="n">GpseaJSONDecoder</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fpath_cohort_json</span> <span class="o">=</span> <span class="s1">&#39;docs/cohort-data/TBX5.0.1.20.json&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">fpath_cohort_json</span><span class="p">)</span> <span class="k">as</span> <span class="n">fh</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">cohort</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">fh</span><span class="p">,</span> <span class="bp">cls</span><span class="o">=</span><span class="n">GpseaJSONDecoder</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">cohort</span><span class="p">)</span>
<span class="go">156</span>
</pre></div>
</div>
</section>
<section id="configure-analysis">
<h3>Configure analysis<a class="headerlink" href="#configure-analysis" title="Link to this heading"></a></h3>
<p>We want to test the association between frameshift <em>TBX5</em> variants and phenotypic abnormalities.
GPSEA exposes a flexible classifier API that lets us create genotype and phenotype classifiers
to assign the cohort members into genotype and phenotype categories based on the variants
and the HPO terms.
We need to create one genotype classifier and one or more phenotype classifiers.</p>
<section id="genotype-classifier">
<h4>Genotype classifier<a class="headerlink" href="#genotype-classifier" title="Link to this heading"></a></h4>
<p>We want to separate the patients into two classes: a class <em>with</em> a frameshift variant
and a class <em>without</em> a frameshift variant (i.e. any other heterozygous variant).
We will use the <em>MANE</em> transcript for the analysis:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">tx_id</span> <span class="o">=</span> <span class="s1">&#39;NM_181486.4&#39;</span>
</pre></div>
</div>
<p>Building a genotype classifier is a two step process.
First, we create a <a class="reference internal" href="../../apidocs/gpsea.analysis.predicate.html#gpsea.analysis.predicate.VariantPredicate" title="gpsea.analysis.predicate.VariantPredicate"><code class="xref py py-class docutils literal notranslate"><span class="pre">VariantPredicate</span></code></a>
to test if the variant is predicted to lead to a frameshift in <cite>NM_181486.4</cite>:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.model</span><span class="w"> </span><span class="kn">import</span> <span class="n">VariantEffect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.predicate</span><span class="w"> </span><span class="kn">import</span> <span class="n">variant_effect</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_frameshift</span> <span class="o">=</span> <span class="n">variant_effect</span><span class="p">(</span><span class="n">VariantEffect</span><span class="o">.</span><span class="n">FRAMESHIFT_VARIANT</span><span class="p">,</span> <span class="n">tx_id</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">is_frameshift</span><span class="o">.</span><span class="n">description</span>
<span class="go">&#39;FRAMESHIFT_VARIANT on NM_181486.4&#39;</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The <a class="reference internal" href="../../apidocs/gpsea.analysis.predicate.html#module-gpsea.analysis.predicate" title="gpsea.analysis.predicate"><code class="xref py py-mod docutils literal notranslate"><span class="pre">gpsea.analysis.predicate</span></code></a> documentation lists all available variant predicates
and <a class="reference internal" href="partitioning/genotype/variant_predicates.html#variant-predicates"><span class="std std-ref">Variant Predicates</span></a> exemplifies their usage.</p>
</div>
<p>To build a genotype classifier, we wrap <cite>is_frameshift</cite>
in a Monoallelic classifier (<a class="reference internal" href="../../apidocs/gpsea.analysis.clf.html#gpsea.analysis.clf.monoallelic_classifier" title="gpsea.analysis.clf.monoallelic_classifier"><code class="xref py py-class docutils literal notranslate"><span class="pre">monoallelic_classifier</span></code></a>),
to classify each <em>TBX5</em> cohort member either as an individual with one <em>frameshift</em> allele (<cite>Frameshift</cite>)
or as an individual with one <em>non-frameshift</em> allele (<cite>Other</cite>):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.clf</span><span class="w"> </span><span class="kn">import</span> <span class="n">monoallelic_classifier</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span> <span class="o">=</span> <span class="n">monoallelic_classifier</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">a_predicate</span><span class="o">=</span><span class="n">is_frameshift</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">a_label</span><span class="o">=</span><span class="s2">&quot;Frameshift&quot;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">b_label</span><span class="o">=</span><span class="s2">&quot;Other&quot;</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gt_clf</span><span class="o">.</span><span class="n">class_labels</span>
<span class="go">(&#39;Frameshift&#39;, &#39;Other&#39;)</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>See the <a class="reference internal" href="partitioning/genotype/index.html#genotype-classifiers"><span class="std std-ref">Genotype classifiers</span></a> for other genotype classifier examples.</p>
</div>
<p>In the subsequent analysis, <cite>gt_clf</cite> assigns an individual into a genotype class.
Note, any individual with <span class="math notranslate nohighlight">\(0\)</span> or <span class="math notranslate nohighlight">\(\ge 2\)</span> alleles will be <em>omitted</em> from the analysis.</p>
</section>
<section id="phenotype-classifiers">
<h4>Phenotype classifiers<a class="headerlink" href="#phenotype-classifiers" title="Link to this heading"></a></h4>
<p>We recommend testing the genotype phenotype association for all HPO terms that annotate the cohort members,
while taking advantage of the HPO graph structure and of the <a class="reference internal" href="../glossary.html#true-path-rule"><span class="std std-ref">True path rule</span></a>.
We will use the <a class="reference internal" href="../../apidocs/gpsea.analysis.clf.html#gpsea.analysis.clf.prepare_classifiers_for_terms_of_interest" title="gpsea.analysis.clf.prepare_classifiers_for_terms_of_interest"><code class="xref py py-func docutils literal notranslate"><span class="pre">prepare_classifiers_for_terms_of_interest()</span></code></a>
utility function to generate phenotype classifiers for all HPO terms.</p>
<p>The function needs HPO to prepare classifiers, hence we need to load HPO:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span><span class="w"> </span><span class="nn">hpotk</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">store</span> <span class="o">=</span> <span class="n">hpotk</span><span class="o">.</span><span class="n">configure_ontology_store</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">hpo</span> <span class="o">=</span> <span class="n">store</span><span class="o">.</span><span class="n">load_minimal_hpo</span><span class="p">(</span><span class="n">release</span><span class="o">=</span><span class="s1">&#39;v2024-07-01&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>and then we can create the classifiers</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.clf</span><span class="w"> </span><span class="kn">import</span> <span class="n">prepare_classifiers_for_terms_of_interest</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">pheno_clfs</span> <span class="o">=</span> <span class="n">prepare_classifiers_for_terms_of_interest</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">hpo</span><span class="o">=</span><span class="n">hpo</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">pheno_clfs</span><span class="p">)</span>
<span class="go">369</span>
</pre></div>
</div>
<p>The function finds 369 HPO terms that annotate at least one individual,
including the <em>indirect</em> annotations whose presence is implied by the <a class="reference internal" href="../glossary.html#true-path-rule"><span class="std std-ref">True path rule</span></a>.</p>
</section>
<section id="statistical-analysis">
<span id="phenotype-classes-statistical-analysis"></span><h4>Statistical analysis<a class="headerlink" href="#statistical-analysis" title="Link to this heading"></a></h4>
<p>We will use <a class="reference internal" href="#fisher-exact-test"><span class="std std-ref">Fisher exact test</span></a> to test the association
between genotype and phenotype classes, as described previously.</p>
<p>In the case of this cohort, we can test association between having a frameshift variant and one of 369 HPO terms.
However, testing multiple hypotheses on the same dataset increases the risk of finding
a significant association solely by chance.
GPSEA uses a two-pronged strategy to reduce the number of tests and, therefore, mitigate this risk:
a phenotype multiple testing (MT) filter and multiple testing correction (MTC).</p>
<p>Phenotype MT filter selects a (sub)set of HPO terms for testing,
for instance only the user-selected terms (see <a class="reference internal" href="../../apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter" title="gpsea.analysis.mtc_filter.SpecifiedTermsMtcFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">SpecifiedTermsMtcFilter</span></code></a>)
or the terms selected by <a class="reference internal" href="../../apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.IfHpoFilter" title="gpsea.analysis.mtc_filter.IfHpoFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">IfHpoFilter</span></code></a>.</p>
<p>MTC then adjusts the nominal p values for the increased risk
of false positive G/P associations.
The available MTC procedures are listed in the <a class="reference internal" href="mtc.html#mtc-correction-procedures"><span class="std std-ref">Multiple-testing correction procedures</span></a> section.</p>
<p>We must choose a phenotype MT filter as well as a MTC procedure to perform genotype-phenotype analysis.</p>
<section id="default-analysis">
<span id="default-hpo-analysis"></span><h5>Default analysis<a class="headerlink" href="#default-analysis" title="Link to this heading"></a></h5>
<p>We recommend using Independent filtering for HPO (<a class="reference internal" href="../../apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.IfHpoFilter" title="gpsea.analysis.mtc_filter.IfHpoFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">IfHpoFilter</span></code></a>)
and Benjamini-Hochberg MT correction.
The default analysis can be configured with <a class="reference internal" href="../../apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.configure_hpo_term_analysis" title="gpsea.analysis.pcats.configure_hpo_term_analysis"><code class="xref py py-func docutils literal notranslate"><span class="pre">configure_hpo_term_analysis()</span></code></a> convenience method.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.pcats</span><span class="w"> </span><span class="kn">import</span> <span class="n">configure_hpo_term_analysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">analysis</span> <span class="o">=</span> <span class="n">configure_hpo_term_analysis</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
</pre></div>
</div>
<p>At this point, the <code class="docutils literal notranslate"><span class="pre">analysis</span></code> configured to test
a cohort for G/P associations.</p>
</section>
<section id="custom-analysis">
<span id="custom-hpo-analysis"></span><h5>Custom analysis<a class="headerlink" href="#custom-analysis" title="Link to this heading"></a></h5>
<p>If the default selection of phenotype MT filter and multiple testing correction is not an option,
we can configure the analysis manually.</p>
<p>First, we choose a phenotype MT filter (e.g. <a class="reference internal" href="../../apidocs/gpsea.analysis.mtc_filter.html#gpsea.analysis.mtc_filter.IfHpoFilter" title="gpsea.analysis.mtc_filter.IfHpoFilter"><code class="xref py py-class docutils literal notranslate"><span class="pre">IfHpoFilter</span></code></a>):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.mtc_filter</span><span class="w"> </span><span class="kn">import</span> <span class="n">IfHpoFilter</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_filter</span> <span class="o">=</span> <span class="n">IfHpoFilter</span><span class="o">.</span><span class="n">default_filter</span><span class="p">(</span><span class="n">hpo</span><span class="p">)</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>See the <a class="reference internal" href="mtc.html#mtc-filters"><span class="std std-ref">MT filters: Choosing which terms to test</span></a> section for info regarding other phenotype MT filters.</p>
</div>
<p>then a statistical test (e.g. Fisher Exact test):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.pcats.stats</span><span class="w"> </span><span class="kn">import</span> <span class="n">FisherExactTest</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">count_statistic</span> <span class="o">=</span> <span class="n">FisherExactTest</span><span class="p">()</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>See the <a class="reference internal" href="../../apidocs/gpsea.analysis.pcats.stats.html#module-gpsea.analysis.pcats.stats" title="gpsea.analysis.pcats.stats"><code class="xref py py-mod docutils literal notranslate"><span class="pre">gpsea.analysis.pcats.stats</span></code></a> module for the available multiple testing procedures
(TL;DR, just Fisher Exact test at this time).</p>
</div>
<p>and we finalize the setup by choosing a MTC procedure
(e.g. <cite>fdr_bh</cite> for Benjamini-Hochberg) along with the MTC alpha:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_correction</span> <span class="o">=</span> <span class="s1">&#39;fdr_bh&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_alpha</span> <span class="o">=</span> <span class="mf">0.05</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>See the <a class="reference internal" href="mtc.html#mtc-correction-procedures"><span class="std std-ref">Multiple-testing correction procedures</span></a> section for a list of available MTC procedure codes.</p>
</div>
<p>The final <a class="reference internal" href="../../apidocs/gpsea.analysis.pcats.html#gpsea.analysis.pcats.HpoTermAnalysis" title="gpsea.analysis.pcats.HpoTermAnalysis"><code class="xref py py-class docutils literal notranslate"><span class="pre">HpoTermAnalysis</span></code></a> is created as:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.analysis.pcats</span><span class="w"> </span><span class="kn">import</span> <span class="n">HpoTermAnalysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">analysis</span> <span class="o">=</span> <span class="n">HpoTermAnalysis</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">count_statistic</span><span class="o">=</span><span class="n">count_statistic</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_filter</span><span class="o">=</span><span class="n">mtc_filter</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_correction</span><span class="o">=</span><span class="s1">&#39;fdr_bh&#39;</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">mtc_alpha</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">analysis</span></code> is identical to the one configured in the <a class="reference internal" href="#default-hpo-analysis"><span class="std std-ref">Default analysis</span></a> section.</p>
</section>
</section>
</section>
<section id="analysis">
<h3>Analysis<a class="headerlink" href="#analysis" title="Link to this heading"></a></h3>
<p>We can now test associations between the genotype classes and the HPO terms:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">result</span> <span class="o">=</span> <span class="n">analysis</span><span class="o">.</span><span class="n">compare_genotype_vs_phenotypes</span><span class="p">(</span>
<span class="gp">... </span>    <span class="n">cohort</span><span class="o">=</span><span class="n">cohort</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">gt_clf</span><span class="o">=</span><span class="n">gt_clf</span><span class="p">,</span>
<span class="gp">... </span>    <span class="n">pheno_clfs</span><span class="o">=</span><span class="n">pheno_clfs</span><span class="p">,</span>
<span class="gp">... </span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">result</span><span class="o">.</span><span class="n">phenotypes</span><span class="p">)</span>
<span class="go">369</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">result</span><span class="o">.</span><span class="n">total_tests</span>
<span class="go">32</span>
</pre></div>
</div>
<p>We tested the <code class="docutils literal notranslate"><span class="pre">cohort</span></code> for association between the genotype classes (<code class="docutils literal notranslate"><span class="pre">gt_clf</span></code>)
and HPO terms (<code class="docutils literal notranslate"><span class="pre">pheno_clfs</span></code>).
Thanks to phenotype MT filter, we only tested 32 out of 369 terms.
The MT filter report shows the filtering details:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">MtcStatsViewer</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_viewer</span> <span class="o">=</span> <span class="n">MtcStatsViewer</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_report</span> <span class="o">=</span> <span class="n">mtc_viewer</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mtc_report</span>
</pre></div>
</div>
<div>
    <style>
        /** Light theme is the default */
        :root {
            --gpsea-color-text: black;
            --gpsea-color-link-text: rgb(0, 118, 187);
            --gpsea-color-link-hover: rgb(0, 99, 156);
            --gpsea-color-primary-a0: #2f5170;
            --gpsea-color-primary-a10: #4778a5;
            --gpsea-color-primary-a20: #5995ce;
            --gpsea-color-primary-a60: LightSkyBlue;
            --gpsea-color-surface-a0: #fcfcfc;
            --gpsea-color-surface-a10: #ffffff;
            --gpsea-color-surface-a20: #b3b3b3;
            --gpsea-color-surface-a30: #8d8d8d;
            --gpsea-color-surface-a40: #646464;
            --gpsea-color-surface-a50: #000000;
        }

        /** Dark theme */
        /*
        @media (prefers-color-scheme: dark) {
            :root {
                --gpsea-color-text: rgb(240, 240, 240);
                --gpsea-color-link-text: rgb(240, 240, 240);
                --gpsea-color-link-hover: lightblue;
                --gpsea-color-primary-a0: #64a3df;
                --gpsea-color-primary-a10: #5286b8;
                --gpsea-color-primary-a20: #375c7e;
                --gpsea-color-primary-a60: #233b52;
                --gpsea-color-surface-a0: #121212;
                --gpsea-color-surface-a10: #282828;
                --gpsea-color-surface-a20: #3f3f3f;
                --gpsea-color-surface-a30: #575757;
                --gpsea-color-surface-a40: #717171;
                --gpsea-color-surface-a50: #8b8b8b;
            }
        }
        */

        #gpsea-body {
            max-width: 1600px;
            color: var(--gpsea-color-text);
            background-color: var(--gpsea-color-surface-a0);
            padding: 10px 0;
        }

        #gpsea-body {

            h1,
            h2,
            h3,
            h4,
            h5 {
                color: var(--gpsea-color-primary-a0);
            }

            a {
                color: var(--gpsea-color-link-text);
                text-decoration: none;
            }

            a:hover {
                color: var(--gpsea-color-link-hover);
            }

            #cohort-summary {
                max-width: 800px;
            }

            td {
                padding: 5px 0;
            }

            .cards {
                display: flex;
                flex-flow: row wrap;
                align-items: top;
                gap: 1rem;
            }

            .card {
                border-radius: 5px;
                padding: 0px 10px;
                background: var(--gpsea-color-surface-a10);
                border: 1px solid var(--gpsea-color-surface-a30);
                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);
            }

            table {
                width: 100%;
                border-collapse: collapse;
                margin: 10px 0;
                font-size: 0.9em;
                font-family: sans-serif;
                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
                border: 1px solid var(--gpsea-color-primary-a60);
            }

            th {
                text-align: left;
                padding: 2px;
                font-weight: bold;
                font-size: 120%;
                color: var(--gpsea-color-text);
                background-color: var(--gpsea-color-primary-a60);
            }

            table td,
            tr {
                text-align: left;
                padding: 0px 5px;
            }

            tr:nth-child(even) {
                background-color: var(--gpsea-color-surface-a10);
            }

            tr:nth-child(odd) {
                background-color: var(--gpsea-color-surface-a20);
            }

            caption {
                caption-side: top;
                text-align: left;
                padding-bottom: 10px;
                font-weight: bold;
            }
        }
    </style>
    <div id="gpsea-body">
        
<h1>Phenotype testing report</h1>
<div class="cards">
  <div class="card">
    <ul>
      <li>Phenotype MTC filter: <em>Independent filtering HPO filter</em></li>
      <li>Multiple testing correction: <em>fdr_bh</em></li>
    </ul>
  </div>
  <div class="card">
    <table>
      <tbody>
        <tr>
          <th>Reason</th>
          <th>Count</th>
        </tr>
        
        <tr>
          
            <td><a href="https://monarch-initiative.github.io/gpsea/stable/user-guide/analyses/mtc.html#skip-terms-if-all-counts-are-identical-to-counts-for-a-child-term">Skip terms if all counts are identical to counts for a child term</a></td>
          
          <td>1</td>
        </tr>
        
        <tr>
          
            <td><a href="https://monarch-initiative.github.io/gpsea/stable/user-guide/analyses/mtc.html#skipping-general-level-terms">Skipping "general" level terms</a></td>
          
          <td>48</td>
        </tr>
        
        <tr>
          
            <td><a href="https://monarch-initiative.github.io/gpsea/stable/user-guide/analyses/mtc.html#skipping-terms-that-are-rare-on-the-cohort-level">Skipping terms that are rare on the cohort level (in less than 40% of the cohort members)</a></td>
          
          <td>288</td>
        </tr>
        
      </tbody>
      <caption>
        Performed statistical tests for 32 out of the total of 369 HPO terms.
      </caption>
    </table>
  </div>
</div>

    </div>
</div></section>
<section id="genotype-phenotype-associations">
<h3>Genotype phenotype associations<a class="headerlink" href="#genotype-phenotype-associations" title="Link to this heading"></a></h3>
<p>Last, let’s explore the associations.</p>
<p>GPSEA displays the associations between genotypes and HPO terms in a table,
one HPO term per row. The rows are ordered by the corrected p value and nominal p value in descending order.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span><span class="w"> </span><span class="nn">gpsea.view</span><span class="w"> </span><span class="kn">import</span> <span class="n">summarize_hpo_analysis</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span> <span class="o">=</span> <span class="n">summarize_hpo_analysis</span><span class="p">(</span><span class="n">hpo</span><span class="p">,</span> <span class="n">result</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">summary_df</span>
</pre></div>
</div>
<table class="docutils align-default" id="id2">
<caption><span class="caption-text"><em>TBX5</em> frameshift vs rest</span><a class="headerlink" href="#id2" title="Link to this table"></a></caption>
<thead>
<tr class="row-odd"><th class="head"></th>
<th class="head"><p>Frameshift</p></th>
<th class="head"><p>Other</p></th>
<th class="head"><p>Corrected p values</p></th>
<th class="head"><p>p values</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Ventricular septal defect [HP:0001629]</p></td>
<td><p>19/19 (100%)</p></td>
<td><p>42/71 (59%)</p></td>
<td><p>0.007741654443787785</p></td>
<td><p>0.00024192670136836828</p></td>
</tr>
<tr class="row-odd"><td><p>Absent thumb [HP:0009777]</p></td>
<td><p>14/31 (45%)</p></td>
<td><p>18/100 (18%)</p></td>
<td><p>0.05941874425631884</p></td>
<td><p>0.0037136715160199273</p></td>
</tr>
<tr class="row-even"><td><p>Secundum atrial septal defect [HP:0001684]</p></td>
<td><p>4/22 (18%)</p></td>
<td><p>23/55 (42%)</p></td>
<td><p>0.5421253568748916</p></td>
<td><p>0.06544319142266644</p></td>
</tr>
<tr class="row-odd"><td><p>Triphalangeal thumb [HP:0001199]</p></td>
<td><p>13/32 (41%)</p></td>
<td><p>23/99 (23%)</p></td>
<td><p>0.5421253568748916</p></td>
<td><p>0.06932119159387057</p></td>
</tr>
<tr class="row-even"><td><p>Muscular ventricular septal defect [HP:0011623]</p></td>
<td><p>6/25 (24%)</p></td>
<td><p>8/84 (10%)</p></td>
<td><p>0.5421253568748916</p></td>
<td><p>0.08470708701170182</p></td>
</tr>
<tr class="row-odd"><td><p>Amelia involving the upper limbs [HP:0009812]</p></td>
<td><p>1/37 (3%)</p></td>
<td><p>0/114 (0%)</p></td>
<td><p>1.0</p></td>
<td><p>0.24503311258278143</p></td>
</tr>
<tr class="row-even"><td><p>Patent foramen ovale [HP:0001655]</p></td>
<td><p>0/36 (0%)</p></td>
<td><p>4/69 (6%)</p></td>
<td><p>1.0</p></td>
<td><p>0.29623386322415446</p></td>
</tr>
<tr class="row-odd"><td><p>Perimembranous ventricular septal defect [HP:0011682]</p></td>
<td><p>3/25 (12%)</p></td>
<td><p>6/84 (7%)</p></td>
<td><p>1.0</p></td>
<td><p>0.4256518230307461</p></td>
</tr>
<tr class="row-even"><td><p>Short thumb [HP:0009778]</p></td>
<td><p>8/30 (27%)</p></td>
<td><p>25/69 (36%)</p></td>
<td><p>1.0</p></td>
<td><p>0.4870099714553749</p></td>
</tr>
<tr class="row-odd"><td><p>Absent radius [HP:0003974]</p></td>
<td><p>6/25 (24%)</p></td>
<td><p>9/43 (21%)</p></td>
<td><p>1.0</p></td>
<td><p>0.7703831604944444</p></td>
</tr>
<tr class="row-even"><td><p>Short 1st metacarpal [HP:0010034]</p></td>
<td><p>0/22 (0%)</p></td>
<td><p>1/45 (2%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Common atrium [HP:0011565]</p></td>
<td><p>0/38 (0%)</p></td>
<td><p>1/115 (1%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Unroofed coronary sinus [HP:0031297]</p></td>
<td><p>0/38 (0%)</p></td>
<td><p>1/117 (1%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal long bone morphology [HP:0011314]</p></td>
<td><p>13/13 (100%)</p></td>
<td><p>50/50 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Aplasia/Hypoplasia of fingers [HP:0006265]</p></td>
<td><p>19/19 (100%)</p></td>
<td><p>44/44 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia involving bones of the hand [HP:0005927]</p></td>
<td><p>19/19 (100%)</p></td>
<td><p>44/44 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Upper limb phocomelia [HP:0009813]</p></td>
<td><p>2/37 (5%)</p></td>
<td><p>8/116 (7%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Atrial septal defect [HP:0001631]</p></td>
<td><p>20/20 (100%)</p></td>
<td><p>63/65 (97%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal atrial septum morphology [HP:0011994]</p></td>
<td><p>20/20 (100%)</p></td>
<td><p>64/64 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiac atrium morphology [HP:0005120]</p></td>
<td><p>20/20 (100%)</p></td>
<td><p>64/64 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal hand morphology [HP:0005922]</p></td>
<td><p>20/20 (100%)</p></td>
<td><p>75/75 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia of the extremities [HP:0009815]</p></td>
<td><p>22/22 (100%)</p></td>
<td><p>78/78 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving bones of the upper limbs [HP:0006496]</p></td>
<td><p>22/22 (100%)</p></td>
<td><p>78/78 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Aplasia/hypoplasia involving bones of the extremities [HP:0045060]</p></td>
<td><p>22/22 (100%)</p></td>
<td><p>78/78 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Aplasia/hypoplasia involving the skeleton [HP:0009115]</p></td>
<td><p>23/23 (100%)</p></td>
<td><p>80/80 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal cardiac septum morphology [HP:0001671]</p></td>
<td><p>28/28 (100%)</p></td>
<td><p>89/89 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Complete atrioventricular canal defect [HP:0001674]</p></td>
<td><p>3/36 (8%)</p></td>
<td><p>6/67 (9%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal thumb morphology [HP:0001172]</p></td>
<td><p>31/31 (100%)</p></td>
<td><p>58/58 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal finger morphology [HP:0001167]</p></td>
<td><p>31/31 (100%)</p></td>
<td><p>64/64 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Abnormal digit morphology [HP:0011297]</p></td>
<td><p>33/33 (100%)</p></td>
<td><p>67/67 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-even"><td><p>Abnormal appendicular skeleton morphology [HP:0011844]</p></td>
<td><p>34/34 (100%)</p></td>
<td><p>93/93 (100%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
<tr class="row-odd"><td><p>Hypoplasia of the radius [HP:0002984]</p></td>
<td><p>6/14 (43%)</p></td>
<td><p>34/75 (45%)</p></td>
<td><p>1.0</p></td>
<td><p>1.0</p></td>
</tr>
</tbody>
</table>
<p>The table shows that several HPO terms are significantly associated
with presence of a heterozygous (<cite>Frameshift</cite>) frameshift variant in <em>TBX5</em>.
For example, <a class="reference external" href="https://hpo.jax.org/browse/term/HP:0001629">Ventricular septal defect</a>
was observed in 42/71 (59%) patients with no frameshift allele (<cite>Other</cite>)
but it was observed in 19/19 (100%) patients with a frameshift allele (<cite>Frameshift</cite>).
Fisher exact test computed a p value of <cite>~0.000242</cite>
and the p value corrected by Benjamini-Hochberg procedure
is <cite>~0.00774</cite>.</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="partitioning/gallery.html" class="btn btn-neutral float-left" title="Gallery" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="mtc.html" class="btn btn-neutral float-right" title="Multiple-testing correction" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>